<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Genes in AC toxicity-associated loci response to CX5461 and DOX</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CX5461_Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Genes in AC toxicity-associated loci
response to CX5461 and DOX</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-05-04
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>CX5461_Project/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree03e5ebaf0e91670502b88762870a9fdf3611b875targetblank03e5ebaa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/03e5ebaf0e91670502b88762870a9fdf3611b875" target="_blank">03e5eba</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree03e5ebaf0e91670502b88762870a9fdf3611b875targetblank03e5ebaa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/03e5ebaf0e91670502b88762870a9fdf3611b875" target="_blank">03e5eba</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    0.1 box.svg
    Ignored:    Rplot04.svg

Untracked files:
    Untracked:  0.1 density.svg
    Untracked:  0.1.emf
    Untracked:  0.1.svg
    Untracked:  0.5 box.svg
    Untracked:  0.5 density.svg
    Untracked:  0.5.svg
    Untracked:  Additional/
    Untracked:  CX_5461_Pattern_Genes_24hr.csv
    Untracked:  CX_5461_Pattern_Genes_3hr.csv
    Untracked:  Cell viability box plot.svg
    Untracked:  DRC1.svg
    Untracked:  Figure 1.jpeg
    Untracked:  Figure 1.pdf
    Untracked:  Figure_CM_Purity.pdf
    Untracked:  Rplot.svg
    Untracked:  Rplot01.svg
    Untracked:  Rplot02.svg
    Untracked:  Rplot03.svg
    Untracked:  Rplot05.svg
    Untracked:  Rplot06.svg
    Untracked:  Rplot07.svg
    Untracked:  Rplot08.jpeg
    Untracked:  Rplot08.svg
    Untracked:  Rplot09.svg
    Untracked:  Rplot10.svg
    Untracked:  Rplot11.svg
    Untracked:  TS HPA (Violin).svg
    Untracked:  TS HPA.svg
    Untracked:  TS_HA.svg
    Untracked:  TS_HV.svg
    Untracked:  Violin HA.svg
    Untracked:  Violin HV (CX vs DOX).svg
    Untracked:  Violin HV.svg
    Untracked:  data/AF.csv
    Untracked:  data/AF_Mapped.csv
    Untracked:  data/AF_genes.csv
    Untracked:  data/Annotated_DOX_Gene_Table.csv
    Untracked:  data/CAD_genes.csv
    Untracked:  data/Cardiotox.csv
    Untracked:  data/DDR.csv
    Untracked:  data/DDR_Mapped.csv
    Untracked:  data/DOX_Vald.csv
    Untracked:  data/DOX_Vald_Mapped.csv
    Untracked:  data/Entrez_Cardiotox.csv
    Untracked:  data/Entrez_Cardiotox_Mapped.csv
    Untracked:  data/GWAS.xlsx
    Untracked:  data/GWAS_SNPs.bed
    Untracked:  data/HF.csv
    Untracked:  data/HF_Mapped.csv
    Untracked:  data/HF_genes.csv
    Untracked:  data/Hypertension_genes.csv
    Untracked:  data/MI_genes.csv
    Untracked:  data/TS.csv
    Untracked:  data/TS_HPA.csv
    Untracked:  data/TS_HPA_mapped.csv
    Untracked:  data/Toptable_CX_0.1_24.csv
    Untracked:  data/Toptable_CX_0.1_3.csv
    Untracked:  data/Toptable_CX_0.1_48.csv
    Untracked:  data/Toptable_CX_0.5_24.csv
    Untracked:  data/Toptable_CX_0.5_3.csv
    Untracked:  data/Toptable_CX_0.5_48.csv
    Untracked:  data/Toptable_DOX_0.1_24.csv
    Untracked:  data/Toptable_DOX_0.1_3.csv
    Untracked:  data/Toptable_DOX_0.1_48.csv
    Untracked:  data/Toptable_DOX_0.5_24.csv
    Untracked:  data/Toptable_DOX_0.5_3.csv
    Untracked:  data/Toptable_DOX_0.5_48.csv
    Untracked:  data/ts_data_mapped
    Untracked:  results/
    Untracked:  run_bedtools.bat

Unstaged changes:
    Deleted:    analysis/Actox.Rmd
    Modified:   analysis/Cardiotox.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Cardiotox.Rmd</code>) and HTML
(<code>docs/Cardiotox.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2f741230924f46f6de464fd0118e4c8b28c0e751/analysis/Cardiotox.Rmd" target="_blank">2f74123</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-24
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/2f741230924f46f6de464fd0118e4c8b28c0e751/docs/Cardiotox.html" target="_blank">2f74123</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-24
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/693aad487a768e0acf96281271ab6f7952f71e89/docs/Cardiotox.html" target="_blank">693aad4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/0d15a959ebf294d5e1489f6ee2c258a23d0c860e/docs/Cardiotox.html" target="_blank">0d15a95</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-20
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/b5ace31718a78c1fa2427d1262bad0df7ce9de7c/analysis/Cardiotox.Rmd" target="_blank">b5ace31</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-20
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/910b6fbf5b9b6eab1020380fd13e152c538c5c6b/docs/Cardiotox.html" target="_blank">910b6fb</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-20
</td>
<td>
Commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="ac-cardiotoxicity" class="section level2">
<h2><strong>📌 AC Cardiotoxicity</strong></h2>
</div>
<div id="load-required-libraries" class="section level2">
<h2><strong>📌 Load Required Libraries</strong></h2>
<pre class="r"><code>library(tidyverse) 
library(ggfortify)
library(cluster)
library(edgeR)
library(limma)
library(Homo.sapiens)
library(BiocParallel)
library(qvalue)
library(pheatmap)
library(clusterProfiler)
library(AnnotationDbi)
library(org.Hs.eg.db)
library(RColorBrewer)
library(readr)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(ComplexHeatmap)
library(circlize)
library(grid)
library(reshape2)
library(dplyr)

# Load UCSC transcript database
txdb &lt;- TxDb.Hsapiens.UCSC.hg38.knownGene</code></pre>
</div>
<div id="read-and-process-deg-data" class="section level2">
<h2><strong>📌 Read and Process DEG Data</strong></h2>
<pre class="r"><code># Load DEGs Data
CX_0.1_3 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;)
CX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;)
CX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;)
CX_0.5_3 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)

DOX_0.1_3 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;)
DOX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;)
DOX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;)
DOX_0.5_3 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

Entrez_IDs &lt;- c(
  6272, 8029, 11128, 79899, 54477, 121665, 5095, 22863, 57161, 4692,
  8214, 23151, 56606, 108, 22999, 56895, 9603, 3181, 4023, 10499,
  92949, 4363, 10057, 5243, 5244, 5880, 1535, 2950, 847, 5447,
  3038, 3077, 4846, 3958, 23327, 29899, 23155, 80856, 55020, 78996,
  150383, 79730, 344595, 6251, 3482, 23262, 9588, 87769, 23409, 339416, 7292, 55157, 720, 5066, 3107, 54535, 1590, 80059, 7991, 57110, 8803, 9620, 323, 54826, 5916, 23371, 283337, 64078, 80010, 1933, 10818, 51020, 873, 874, 2064, 2878, 2944, 51196, 6687, 7799, 4292, 51310, 9154, 10060, 89845, 56853, 4625, 1573, 79890
)

# Subset the toptable based on the entrez IDs and select specific columns

subset_toptable1 &lt;- CX_0.1_3[CX_0.1_3$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]


subset_toptable2 &lt;- CX_0.1_24[CX_0.1_24$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]


subset_toptable3 &lt;- CX_0.1_48[CX_0.1_48$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]


subset_toptable3 &lt;- CX_0.1_48[CX_0.1_48$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]

subset_toptable4 &lt;- CX_0.5_3[CX_0.5_3$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]

subset_toptable5 &lt;- CX_0.5_24[CX_0.5_24$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]

subset_toptable6 &lt;- CX_0.5_48[CX_0.5_48$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]

subset_toptable7 &lt;- DOX_0.1_3[DOX_0.1_3$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]

subset_toptable8 &lt;- DOX_0.1_24[DOX_0.1_24$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]

subset_toptable9 &lt;- DOX_0.1_48[DOX_0.1_48$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]

subset_toptable10 &lt;- DOX_0.5_3[DOX_0.5_3$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]

subset_toptable11 &lt;- DOX_0.5_24[DOX_0.5_24$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]

subset_toptable12 &lt;- DOX_0.5_48[DOX_0.5_48$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]</code></pre>
</div>
<div id="add-metadata-and-combine-all-subsets" class="section level2">
<h2><strong>📌 Add metadata and combine all subsets</strong></h2>
</div>
<div id="create-a-matrix-and-heatmap" class="section level2">
<h2><strong>📌Create a matrix and heatmap</strong></h2>
<pre class="r"><code># Assuming your dataframe is named data
# Add a column for significance stars
final_data &lt;- final_data %&gt;%
  mutate(Significance = ifelse(adj.P.Val &lt; 0.05, &quot;*&quot;, &quot;&quot;))

# Create a matrix for the heatmap (logFC values)
logFC_matrix &lt;- acast(final_data, Gene ~ paste(Drug, Conc, Time, sep = &quot;_&quot;), value.var = &quot;logFC&quot;)

# Create a matrix for the significance annotations
signif_matrix &lt;- acast(final_data, Gene ~ paste(Drug, Conc, Time, sep = &quot;_&quot;), value.var = &quot;Significance&quot;)

# Split column names into Drug, Conc, and Time
colnames_split &lt;- strsplit(colnames(logFC_matrix), &quot;_&quot;)
drug &lt;- sapply(colnames_split, function(x) x[1])
conc &lt;- sapply(colnames_split, function(x) x[2])
time &lt;- sapply(colnames_split, function(x) x[3])

# Create the desired column order: CX 0.1 3hr, CX 0.5 3hr, CX 0.1 24hr, CX 0.5 24hr, CX 0.1 48h, CX 0.5 48h,
# DOX 0.1 3hr, DOX 0.5 3hr, DOX 0.1 24hr, DOX 0.5 24hr, DOX 0.1 48h, DOX 0.5 48h
desired_order &lt;- c(&quot;CX_0.1_3&quot;, &quot;CX_0.5_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.1_48&quot;, &quot;CX_0.5_48&quot;,
                   &quot;DOX_0.1_3&quot;, &quot;DOX_0.5_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.1_48&quot;, &quot;DOX_0.5_48&quot;)

# Reorder columns in the matrix based on the desired order
column_names &lt;- paste(drug, conc, time, sep = &quot;_&quot;)
column_order &lt;- match(desired_order, column_names)
logFC_matrix &lt;- logFC_matrix[, column_order]
signif_matrix &lt;- signif_matrix[, column_order]
drug &lt;- drug[column_order]
conc &lt;- conc[column_order]
time &lt;- time[column_order]

# Prepare annotations matching the column structure
ha_top &lt;- HeatmapAnnotation(
  Drug = drug,
  Conc = conc,
  Time = time,
  col = list(Drug = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;),
             Conc = c(&quot;0.1&quot; = &quot;lightgreen&quot;, &quot;0.5&quot; = &quot;darkgreen&quot;),
             Time = c(&quot;3&quot; = &quot;yellow&quot;, &quot;24&quot; = &quot;orange&quot;, &quot;48&quot; = &quot;purple&quot;)),
  annotation_height = unit(c(2, 2, 2), &quot;cm&quot;)
)

# Create the heatmap
heatmap &lt;- Heatmap(logFC_matrix, name = &quot;logFC&quot;, top_annotation = ha_top,
                   cell_fun = function(j, i, x, y, width, height, fill) {
                     grid.text(signif_matrix[i, j], x, y, gp = gpar(fontsize = 10))
                   },
                   show_row_names = TRUE, show_column_names = FALSE,
                   column_title = &quot;Genes in AC toxicity-associated loci\nresponse to CX5461 and DOX&quot;,
                   column_title_gp = gpar(fontsize = 16, fontface = &quot;bold&quot;),
                   cluster_columns = FALSE)  # Disable column clustering

# Draw the heatmap
draw(heatmap, heatmap_legend_side = &quot;right&quot;, annotation_legend_side = &quot;right&quot;)</code></pre>
<p><img src="figure/Cardiotox.Rmd/Data_prep-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Data_prep-1">
Past versions of Data_prep-1.png
</button>
</p>
<div id="fig-Data_prep-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2f741230924f46f6de464fd0118e4c8b28c0e751/docs/figure/Cardiotox.Rmd/Data_prep-1.png" target="_blank">2f74123</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/910b6fbf5b9b6eab1020380fd13e152c538c5c6b/docs/figure/Cardiotox.Rmd/Data_prep-1.png" target="_blank">910b6fb</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dox-cardiotoxicity" class="section level2">
<h2><strong>📌 DOX Cardiotoxicity</strong></h2>
<pre class="r"><code># Load necessary libraries
library(tidyverse)
library(ComplexHeatmap)
library(circlize)
library(org.Hs.eg.db)
library(reshape2)
library(grid)

# Set Entrez ID order
Entrez_IDs &lt;- c(847, 873, 2064, 2878, 2944, 3038, 4846, 51196, 5880, 6687,
                7799, 4292, 5916, 3077, 51310, 9154, 64078, 5244, 10057, 10060,
                89845, 56853, 4625, 1573, 79890)

# Load annotated gene table
annotated_genes &lt;- read.csv(&quot;data/Annotated_DOX_Gene_Table.csv&quot;)

# Load DEG data
CX_0.1_3 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;)
CX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;)
CX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;)
CX_0.5_3 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)
DOX_0.1_3 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;)
DOX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;)
DOX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;)
DOX_0.5_3 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

# Subsetting helper
get_subset &lt;- function(df) {
  df[df$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]
}

# Subset and annotate
add_metadata &lt;- function(data, drug, conc, time) {
  data %&gt;% mutate(Drug = drug, Conc = conc, Time = time)
}

combined_data &lt;- bind_rows(
  add_metadata(get_subset(CX_0.1_3), &quot;CX&quot;, 0.1, 3),
  add_metadata(get_subset(CX_0.1_24), &quot;CX&quot;, 0.1, 24),
  add_metadata(get_subset(CX_0.1_48), &quot;CX&quot;, 0.1, 48),
  add_metadata(get_subset(CX_0.5_3), &quot;CX&quot;, 0.5, 3),
  add_metadata(get_subset(CX_0.5_24), &quot;CX&quot;, 0.5, 24),
  add_metadata(get_subset(CX_0.5_48), &quot;CX&quot;, 0.5, 48),
  add_metadata(get_subset(DOX_0.1_3), &quot;DOX&quot;, 0.1, 3),
  add_metadata(get_subset(DOX_0.1_24), &quot;DOX&quot;, 0.1, 24),
  add_metadata(get_subset(DOX_0.1_48), &quot;DOX&quot;, 0.1, 48),
  add_metadata(get_subset(DOX_0.5_3), &quot;DOX&quot;, 0.5, 3),
  add_metadata(get_subset(DOX_0.5_24), &quot;DOX&quot;, 0.5, 24),
  add_metadata(get_subset(DOX_0.5_48), &quot;DOX&quot;, 0.5, 48)
)

# Add gene symbol and significance
combined_data &lt;- combined_data %&gt;%
  mutate(Gene = mapIds(org.Hs.eg.db, keys = as.character(Entrez_ID),
                       column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
         Significance = ifelse(adj.P.Val &lt; 0.05, &quot;*&quot;, &quot;&quot;))

# Merge with mechanistic category
combined_data &lt;- left_join(combined_data, annotated_genes, by = c(&quot;Entrez_ID&quot; = &quot;ENTREZID&quot;))

# Reorder by Entrez_ID
combined_data$Entrez_ID &lt;- factor(combined_data$Entrez_ID, levels = Entrez_IDs)

# Preserve correct gene order
ordered_genes &lt;- combined_data %&gt;%
  distinct(Entrez_ID, Gene) %&gt;%
  arrange(factor(Entrez_ID, levels = Entrez_IDs)) %&gt;%
  pull(Gene)

# Create logFC and significance matrices
logFC_matrix &lt;- acast(combined_data, Gene ~ paste(Drug, Conc, Time, sep = &quot;_&quot;), value.var = &quot;logFC&quot;)
logFC_matrix &lt;- logFC_matrix[ordered_genes, ]

signif_matrix &lt;- acast(combined_data, Gene ~ paste(Drug, Conc, Time, sep = &quot;_&quot;), value.var = &quot;Significance&quot;)
signif_matrix &lt;- signif_matrix[ordered_genes, ]

# Desired column order: group by drug → conc → time
desired_order &lt;- c(
  &quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;,
  &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
  &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;,
  &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;
)
logFC_matrix &lt;- logFC_matrix[, desired_order]
signif_matrix &lt;- signif_matrix[, desired_order]

# Column metadata
split_cols &lt;- strsplit(colnames(logFC_matrix), &quot;_&quot;)
drug &lt;- sapply(split_cols, function(x) x[1])
conc &lt;- sapply(split_cols, function(x) x[2])
time &lt;- sapply(split_cols, function(x) x[3])

# Top annotation
ha_top &lt;- HeatmapAnnotation(
  Drug = drug,
  Conc = conc,
  Time = time,
  col = list(
    Drug = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;),
    Conc = c(&quot;0.1&quot; = &quot;lightgreen&quot;, &quot;0.5&quot; = &quot;darkgreen&quot;),
    Time = c(&quot;3&quot; = &quot;yellow&quot;, &quot;24&quot; = &quot;orange&quot;, &quot;48&quot; = &quot;purple&quot;)
  ),
  annotation_height = unit(c(2, 2, 2), &quot;cm&quot;)
)

# Mechanistic category row annotation
category_mapping &lt;- combined_data %&gt;%
  distinct(Gene, Mechanistic_Implication) %&gt;%
  filter(Gene %in% ordered_genes) %&gt;%
  arrange(match(Gene, ordered_genes))

category_colors &lt;- structure(
  c(&quot;darkorange&quot;, &quot;steelblue&quot;, &quot;darkgreen&quot;, &quot;firebrick&quot;, &quot;gold&quot;, &quot;mediumpurple&quot;, &quot;cyan&quot;),
  names = unique(category_mapping$Mechanistic_Implication)
)

ha_left &lt;- rowAnnotation(
  Category = category_mapping$Mechanistic_Implication,
  col = list(Category = category_colors),
  show_annotation_name = TRUE,
  annotation_name_side = &quot;top&quot;
)

# Draw heatmap
Heatmap(logFC_matrix,
        name = &quot;logFC&quot;,
        top_annotation = ha_top,
        left_annotation = ha_left,
        show_row_names = TRUE,
        row_names_gp = gpar(fontsize = 10),
        show_column_names = FALSE,
        cluster_rows = FALSE,
        cluster_columns = FALSE,
        column_title = &quot;Genes in DOX cardiotoxicity-associated loci\nresponse to CX5461 and DOX&quot;,
        column_title_gp = gpar(fontsize = 16, fontface = &quot;bold&quot;),
        cell_fun = function(j, i, x, y, width, height, fill) {
          grid.text(signif_matrix[i, j], x, y, gp = gpar(fontsize = 10))
        }
)</code></pre>
<p><img src="figure/Cardiotox.Rmd/load_DEGs2-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-load_DEGs2-1">
Past versions of load_DEGs2-1.png
</button>
</p>
<div id="fig-load_DEGs2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2f741230924f46f6de464fd0118e4c8b28c0e751/docs/figure/Cardiotox.Rmd/load_DEGs2-1.png" target="_blank">2f74123</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 26100)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] reshape2_1.4.4                          
 [2] circlize_0.4.16                         
 [3] ComplexHeatmap_2.18.0                   
 [4] TxDb.Hsapiens.UCSC.hg38.knownGene_3.18.0
 [5] RColorBrewer_1.1-3                      
 [6] clusterProfiler_4.10.1                  
 [7] pheatmap_1.0.12                         
 [8] qvalue_2.34.0                           
 [9] BiocParallel_1.36.0                     
[10] Homo.sapiens_1.3.1                      
[11] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2 
[12] org.Hs.eg.db_3.18.0                     
[13] GO.db_3.18.0                            
[14] OrganismDbi_1.44.0                      
[15] GenomicFeatures_1.54.4                  
[16] GenomicRanges_1.54.1                    
[17] GenomeInfoDb_1.38.8                     
[18] AnnotationDbi_1.64.1                    
[19] IRanges_2.36.0                          
[20] S4Vectors_0.40.2                        
[21] Biobase_2.62.0                          
[22] BiocGenerics_0.48.1                     
[23] edgeR_4.0.16                            
[24] limma_3.58.1                            
[25] cluster_2.1.8.1                         
[26] ggfortify_0.4.17                        
[27] lubridate_1.9.4                         
[28] forcats_1.0.0                           
[29] stringr_1.5.1                           
[30] dplyr_1.1.4                             
[31] purrr_1.0.4                             
[32] readr_2.1.5                             
[33] tidyr_1.3.1                             
[34] tibble_3.2.1                            
[35] ggplot2_3.5.2                           
[36] tidyverse_2.0.0                         

loaded via a namespace (and not attached):
  [1] splines_4.3.0               later_1.3.2                
  [3] BiocIO_1.12.0               bitops_1.0-9               
  [5] ggplotify_0.1.2             filelock_1.0.3             
  [7] polyclip_1.10-7             graph_1.80.0               
  [9] XML_3.99-0.18               lifecycle_1.0.4            
 [11] doParallel_1.0.17           rprojroot_2.0.4            
 [13] lattice_0.22-7              MASS_7.3-60                
 [15] magrittr_2.0.3              sass_0.4.10                
 [17] rmarkdown_2.29              jquerylib_0.1.4            
 [19] yaml_2.3.10                 httpuv_1.6.15              
 [21] cowplot_1.1.3               DBI_1.2.3                  
 [23] abind_1.4-8                 zlibbioc_1.48.2            
 [25] ggraph_2.2.1                RCurl_1.98-1.17            
 [27] yulab.utils_0.2.0           tweenr_2.0.3               
 [29] rappdirs_0.3.3              git2r_0.36.2               
 [31] GenomeInfoDbData_1.2.11     enrichplot_1.22.0          
 [33] ggrepel_0.9.6               tidytree_0.4.6             
 [35] codetools_0.2-20            DelayedArray_0.28.0        
 [37] DOSE_3.28.2                 xml2_1.3.8                 
 [39] ggforce_0.4.2               shape_1.4.6.1              
 [41] tidyselect_1.2.1            aplot_0.2.5                
 [43] farver_2.1.2                viridis_0.6.5              
 [45] matrixStats_1.5.0           BiocFileCache_2.10.2       
 [47] GenomicAlignments_1.38.2    jsonlite_2.0.0             
 [49] GetoptLong_1.0.5            tidygraph_1.3.1            
 [51] iterators_1.0.14            foreach_1.5.2              
 [53] tools_4.3.0                 progress_1.2.3             
 [55] treeio_1.26.0               Rcpp_1.0.12                
 [57] glue_1.7.0                  gridExtra_2.3              
 [59] SparseArray_1.2.4           xfun_0.52                  
 [61] MatrixGenerics_1.14.0       withr_3.0.2                
 [63] BiocManager_1.30.25         fastmap_1.2.0              
 [65] digest_0.6.34               timechange_0.3.0           
 [67] R6_2.6.1                    gridGraphics_0.5-1         
 [69] colorspace_2.1-0            Cairo_1.6-2                
 [71] biomaRt_2.58.2              RSQLite_2.3.9              
 [73] generics_0.1.3              data.table_1.17.0          
 [75] rtracklayer_1.62.0          prettyunits_1.2.0          
 [77] graphlayouts_1.2.2          httr_1.4.7                 
 [79] S4Arrays_1.2.1              scatterpie_0.2.4           
 [81] whisker_0.4.1               pkgconfig_2.0.3            
 [83] gtable_0.3.6                blob_1.2.4                 
 [85] workflowr_1.7.1             XVector_0.42.0             
 [87] shadowtext_0.1.4            htmltools_0.5.8.1          
 [89] fgsea_1.28.0                RBGL_1.78.0                
 [91] clue_0.3-66                 scales_1.3.0               
 [93] png_0.1-8                   ggfun_0.1.8                
 [95] knitr_1.50                  rstudioapi_0.17.1          
 [97] tzdb_0.5.0                  rjson_0.2.23               
 [99] nlme_3.1-168                curl_6.2.2                 
[101] GlobalOptions_0.1.2         cachem_1.1.0               
[103] parallel_4.3.0              HDO.db_0.99.1              
[105] restfulr_0.0.15             pillar_1.10.2              
[107] vctrs_0.6.5                 promises_1.3.2             
[109] dbplyr_2.5.0                evaluate_1.0.3             
[111] magick_2.8.6                cli_3.6.1                  
[113] locfit_1.5-9.12             compiler_4.3.0             
[115] Rsamtools_2.18.0            rlang_1.1.3                
[117] crayon_1.5.3                plyr_1.8.9                 
[119] fs_1.6.3                    stringi_1.8.3              
[121] viridisLite_0.4.2           munsell_0.5.1              
[123] Biostrings_2.70.3           lazyeval_0.2.2             
[125] GOSemSim_2.28.1             Matrix_1.6-1.1             
[127] patchwork_1.3.0             hms_1.1.3                  
[129] bit64_4.6.0-1               KEGGREST_1.42.0            
[131] statmod_1.5.0               SummarizedExperiment_1.32.0
[133] igraph_2.1.4                memoise_2.0.1              
[135] bslib_0.9.0                 ggtree_3.10.1              
[137] fastmatch_1.1-6             bit_4.6.0                  
[139] gson_0.1.0                  ape_5.8-1                  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
