<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Cardiac, TOP2 and DNA damage Genes</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CX5461_Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Cardiac, TOP2 and DNA damage Genes</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-03-02
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>CX5461_Project/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree6787f4ebe083b7dfd7f9aa934222255360b60d71targetblank6787f4ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/6787f4ebe083b7dfd7f9aa934222255360b60d71" target="_blank">6787f4e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree6787f4ebe083b7dfd7f9aa934222255360b60d71targetblank6787f4ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/6787f4ebe083b7dfd7f9aa934222255360b60d71" target="_blank">6787f4e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  data/P53_Target.csv

Unstaged changes:
    Modified:   analysis/Cardiac_and_TOP2_Genes.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/Cardiac_and_TOP2_Genes.Rmd</code>) and HTML
(<code>docs/Cardiac_and_TOP2_Genes.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/c14f8de07187bea3fb9b865466ef1fd4dd40b3d4/docs/Cardiac_and_TOP2_Genes.html" target="_blank">c14f8de</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/6d5a0a93617d94740c63bf16a444354704c8aa99/docs/Cardiac_and_TOP2_Genes.html" target="_blank">6d5a0a9</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/c2a6b920376e71b5c6fc084f6763298921b6264a/analysis/Cardiac_and_TOP2_Genes.Rmd" target="_blank">c2a6b92</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
<td>
Updated analysis of Cardiac, TOP2, and DNA Damage Genes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/1337c6888968145907547d61ab32b7a2910b6fea/docs/Cardiac_and_TOP2_Genes.html" target="_blank">1337c68</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/55dbfa037afc81e5390a8b110f2421a36636623b/docs/Cardiac_and_TOP2_Genes.html" target="_blank">55dbfa0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/1b220adf2523368d0132d73371074f673839cf15/analysis/Cardiac_and_TOP2_Genes.Rmd" target="_blank">1b220ad</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
<td>
Updated analysis of Cardiac, TOP2, and DNA Damage Genes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/0a7f53eb3fdbf5160350e69f74f931b54aa0e73a/docs/Cardiac_and_TOP2_Genes.html" target="_blank">0a7f53e</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/565ede2e548ade39425a2be455d3cd0d62edf158/analysis/Cardiac_and_TOP2_Genes.Rmd" target="_blank">565ede2</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
<td>
Updated analysis of Cardiac, TOP2, and DNA Damage Genes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/3313173f542c38cf7a8bdab7d6e8ec20e3b456f2/docs/Cardiac_and_TOP2_Genes.html" target="_blank">3313173</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/456790a1f560bd70267c93b27598f54a72c51e61/docs/Cardiac_and_TOP2_Genes.html" target="_blank">456790a</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/e943e2d7775dac9d6c268443840ca24303996b84/analysis/Cardiac_and_TOP2_Genes.Rmd" target="_blank">e943e2d</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
<td>
Added TP53 as a DNA damage marker in log2CPM boxplots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/6ed78e299a335e7a75dd8448fc6eb6a810537b3d/docs/Cardiac_and_TOP2_Genes.html" target="_blank">6ed78e2</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/72599accbf90ae9bf96e52d2b799ed6eee6baf48/docs/Cardiac_and_TOP2_Genes.html" target="_blank">72599ac</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e79f5b46c58a97a56b44cd383e25e397f65a5dd/analysis/Cardiac_and_TOP2_Genes.Rmd" target="_blank">0e79f5b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
<td>
Fix Timepoint Ordering in Boxplots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/Cardiac_and_TOP2_Genes.html" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91f5f8f51cd57fdd0d6ab9be529ef747ba23f697/analysis/Cardiac_and_TOP2_Genes.Rmd" target="_blank">91f5f8f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
<td>
Fix Timepoint Ordering in Boxplots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/bcc58f6f39f4dd5e241b7d0026e9d7b927d68f83/docs/Cardiac_and_TOP2_Genes.html" target="_blank">bcc58f6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/Cardiac_and_TOP2_Genes.html" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/c6b5ccd218c3fd41470b2b766ba41ed8c2aed105/analysis/Cardiac_and_TOP2_Genes.Rmd" target="_blank">c6b5ccd</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
<td>
Fixed boxplot significance issue for Cardiac and TOP2 genes
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="log2cpm-boxplots-for-cardiac-and-top2-genes"
class="section level2">
<h2><strong>📌 Log2CPM Boxplots for Cardiac and TOP2 Genes</strong></h2>
<p>This analysis generates <strong>boxplots for cardiac genes and TOP2
genes</strong> across different treatments and timepoints.</p>
<hr />
</div>
<div id="load-required-libraries" class="section level2">
<h2><strong>📌 Load Required Libraries</strong></h2>
<pre class="r"><code>library(ggplot2)</code></pre>
<pre><code>Warning: package &#39;ggplot2&#39; was built under R version 4.3.3</code></pre>
<pre class="r"><code>library(dplyr)</code></pre>
<pre><code>Warning: package &#39;dplyr&#39; was built under R version 4.3.2</code></pre>
<pre class="r"><code>library(tidyr)</code></pre>
<pre><code>Warning: package &#39;tidyr&#39; was built under R version 4.3.3</code></pre>
<pre class="r"><code>library(org.Hs.eg.db)</code></pre>
<pre><code>Warning: package &#39;AnnotationDbi&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;BiocGenerics&#39; was built under R version 4.3.1</code></pre>
<pre><code>Warning: package &#39;Biobase&#39; was built under R version 4.3.1</code></pre>
<pre><code>Warning: package &#39;IRanges&#39; was built under R version 4.3.1</code></pre>
<pre><code>Warning: package &#39;S4Vectors&#39; was built under R version 4.3.1</code></pre>
<pre class="r"><code>library(clusterProfiler)</code></pre>
<pre><code>Warning: package &#39;clusterProfiler&#39; was built under R version 4.3.3</code></pre>
</div>
<div id="read-log2cpm-data" class="section level2">
<h2><strong>📌 Read Log2CPM Data</strong></h2>
<pre class="r"><code># Load feature count matrix
boxplot1 &lt;- read.csv(&quot;data/Feature_count_Matrix_Log2CPM_filtered.csv&quot;) %&gt;% as.data.frame()

# Ensure column names are cleaned
colnames(boxplot1) &lt;- trimws(gsub(&quot;^X&quot;, &quot;&quot;, colnames(boxplot1)))  </code></pre>
</div>
<div id="define-genes-of-interest" class="section level2">
<h2><strong>📌 Define Genes of Interest</strong></h2>
<pre class="r"><code># Define the genes of interest
top2_genes &lt;- c(&quot;TOP2A&quot;, &quot;TOP2B&quot;)
cardiac_genes &lt;- c(&quot;ACTN2&quot;, &quot;CALR&quot;, &quot;MYBPC3&quot;, &quot;MYH6&quot;, &quot;MYH7&quot;, 
                   &quot;MYL2&quot;, &quot;RYR2&quot;, &quot;SCN5A&quot;, &quot;TNNI3&quot;, &quot;TNNT2&quot;, &quot;TTN&quot;)
dna_damage_genes &lt;- c(&quot;TP53&quot;)  # Using correct gene symbol TP53</code></pre>
</div>
<div id="read-and-process-degs-data" class="section level2">
<h2><strong>📌 Read and Process DEGs Data</strong></h2>
<pre class="r"><code># Load Toptables
deg_files &lt;- list.files(&quot;data/DEGs&quot;, pattern = &quot;Toptable_.*\\.csv&quot;, full.names = TRUE)
deg_list &lt;- lapply(deg_files, read.csv)
names(deg_list) &lt;- gsub(&quot;data/DEGs/Toptable_|\\.csv&quot;, &quot;&quot;, deg_files)  

# Function to check significance based on **Entrez_ID in the correct sample**
is_significant &lt;- function(gene, drug, conc, timepoint) {
  condition &lt;- paste(drug, conc, timepoint, sep = &quot;_&quot;)
  if (!condition %in% names(deg_list)) return(FALSE)
  
  toptable &lt;- deg_list[[condition]]
  gene_entrez &lt;- boxplot1$ENTREZID[boxplot1$SYMBOL == gene]
  
  if (length(gene_entrez) == 0) return(FALSE)
  
  return(any(gene_entrez %in% toptable$Entrez_ID[toptable$adj.P.Val &lt; 0.05]))
}</code></pre>
</div>
<div id="process-data-for-plotting" class="section level2">
<h2><strong>📌 Process Data for Plotting</strong></h2>
<pre class="r"><code>process_gene_data &lt;- function(gene) {
  # Filter log2CPM data for the gene
  gene_data &lt;- boxplot1 %&gt;% filter(SYMBOL == gene)
  
  # Reshape data
  long_data &lt;- gene_data %&gt;%
    pivot_longer(cols = -c(ENTREZID, SYMBOL, GENENAME), names_to = &quot;Sample&quot;, values_to = &quot;log2CPM&quot;) %&gt;%
    mutate(
      Indv = case_when(
        grepl(&quot;75.1&quot;, Sample) ~ &quot;1&quot;,
        grepl(&quot;78.1&quot;, Sample) ~ &quot;2&quot;,
        grepl(&quot;87.1&quot;, Sample) ~ &quot;3&quot;,
        grepl(&quot;17.3&quot;, Sample) ~ &quot;4&quot;,
        grepl(&quot;84.1&quot;, Sample) ~ &quot;5&quot;,
        grepl(&quot;90.1&quot;, Sample) ~ &quot;6&quot;,
        TRUE ~ NA_character_
      ),
      Drug = case_when(
        grepl(&quot;CX.5461&quot;, Sample) ~ &quot;CX&quot;,
        grepl(&quot;DOX&quot;, Sample) ~ &quot;DOX&quot;,
        grepl(&quot;VEH&quot;, Sample) ~ &quot;VEH&quot;,
        TRUE ~ NA_character_
      ),
      Conc. = case_when(
        grepl(&quot;_0.1_&quot;, Sample) ~ &quot;0.1&quot;,
        grepl(&quot;_0.5_&quot;, Sample) ~ &quot;0.5&quot;,
        TRUE ~ NA_character_
      ),
      Timepoint = case_when(
        grepl(&quot;_3$&quot;, Sample) ~ &quot;3&quot;,
        grepl(&quot;_24$&quot;, Sample) ~ &quot;24&quot;,
        grepl(&quot;_48$&quot;, Sample) ~ &quot;48&quot;,
        TRUE ~ NA_character_
      ),
      Condition = paste(Drug, Conc., Timepoint, sep = &quot;_&quot;)
    )

  # **Ensure Condition is Ordered Correctly**
  long_data$Condition &lt;- factor(
    long_data$Condition, 
    levels = c(
      &quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;, &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
      &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;, &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;,
      &quot;VEH_0.1_3&quot;, &quot;VEH_0.1_24&quot;, &quot;VEH_0.1_48&quot;, &quot;VEH_0.5_3&quot;, &quot;VEH_0.5_24&quot;, &quot;VEH_0.5_48&quot;
    )
  )
  
  # Identify significant conditions **per Drug, Conc, and Timepoint**
  significance_labels &lt;- long_data %&gt;%
    distinct(Drug, Conc., Timepoint, Condition) %&gt;%
    rowwise() %&gt;%
    mutate(
      max_log2CPM = max(long_data$log2CPM[long_data$Condition == Condition], na.rm = TRUE),
      Significance = ifelse(is_significant(gene, Drug, Conc., Timepoint), &quot;*&quot;, &quot;&quot;)
    ) %&gt;%
    filter(Significance != &quot;&quot;) %&gt;% ungroup()
  
  list(long_data = long_data, significance_labels = significance_labels)
}</code></pre>
</div>
<div id="generate-boxplots-for-cardiac-genes" class="section level2">
<h2><strong>📌Generate Boxplots for Cardiac Genes</strong></h2>
<pre class="r"><code>for (gene in cardiac_genes) {
  data_info &lt;- process_gene_data(gene)
  p &lt;- ggplot(data_info$long_data, aes(x = Condition, y = log2CPM, fill = Drug)) +
    geom_boxplot(outlier.shape = NA) +
    scale_fill_manual(values = c(&quot;CX&quot; = &quot;#0000FF&quot;, &quot;DOX&quot; = &quot;#e6d800&quot;, &quot;VEH&quot; = &quot;#FF00FF&quot;)) +
    geom_point(aes(color = Indv), size = 2, alpha = 0.5, position = position_jitter(width = -1, height = 0)) +
    geom_text(data = data_info$significance_labels, aes(x = Condition, y = max_log2CPM + 0.5, label = Significance),
              inherit.aes = FALSE, size = 6, color = &quot;black&quot;) +
    ggtitle(paste(&quot;Log2CPM Expression of&quot;, gene)) +
    labs(x = &quot;Treatment&quot;, y = &quot;log2CPM&quot;) +
    theme_bw() +
    theme(plot.title = element_text(size = rel(2), hjust = 0.5),
          axis.title = element_text(size = 15, color = &quot;black&quot;),
          axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 90, hjust = 1))
  
  print(p)
}</code></pre>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cardiac-1">
Past versions of Cardiac-1.png
</button>
</p>
<div id="fig-Cardiac-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-1.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-1.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cardiac-2">
Past versions of Cardiac-2.png
</button>
</p>
<div id="fig-Cardiac-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-2.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-2.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cardiac-3">
Past versions of Cardiac-3.png
</button>
</p>
<div id="fig-Cardiac-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-3.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-3.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cardiac-4">
Past versions of Cardiac-4.png
</button>
</p>
<div id="fig-Cardiac-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-4.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-4.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cardiac-5">
Past versions of Cardiac-5.png
</button>
</p>
<div id="fig-Cardiac-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-5.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-5.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cardiac-6">
Past versions of Cardiac-6.png
</button>
</p>
<div id="fig-Cardiac-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-6.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-6.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cardiac-7">
Past versions of Cardiac-7.png
</button>
</p>
<div id="fig-Cardiac-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-7.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-7.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cardiac-8">
Past versions of Cardiac-8.png
</button>
</p>
<div id="fig-Cardiac-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-8.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-8.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cardiac-9">
Past versions of Cardiac-9.png
</button>
</p>
<div id="fig-Cardiac-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-9.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-9.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cardiac-10">
Past versions of Cardiac-10.png
</button>
</p>
<div id="fig-Cardiac-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-10.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-10.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cardiac-11">
Past versions of Cardiac-11.png
</button>
</p>
<div id="fig-Cardiac-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-11.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-11.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="generate-boxplots-for-top2-genes" class="section level2">
<h2><strong>📌Generate Boxplots for TOP2 Genes</strong></h2>
<pre class="r"><code>for (gene in top2_genes) {
  data_info &lt;- process_gene_data(gene)
  p &lt;- ggplot(data_info$long_data, aes(x = Condition, y = log2CPM, fill = Drug)) +
    geom_boxplot(outlier.shape = NA) +
    scale_fill_manual(values = c(&quot;CX&quot; = &quot;#0000FF&quot;, &quot;DOX&quot; = &quot;#e6d800&quot;, &quot;VEH&quot; = &quot;#FF00FF&quot;)) +
    geom_point(aes(color = Indv), size = 2, alpha = 0.5, position = position_jitter(width = -1, height = 0)) +
    geom_text(data = data_info$significance_labels, aes(x = Condition, y = max_log2CPM + 0.5, label = Significance),
              inherit.aes = FALSE, size = 6, color = &quot;black&quot;) +
    ggtitle(paste(&quot;Log2CPM Expression of&quot;, gene)) +
    labs(x = &quot;Treatment&quot;, y = &quot;log2CPM&quot;) +
    theme_bw() +
    theme(plot.title = element_text(size = rel(2), hjust = 0.5),
          axis.title = element_text(size = 15, color = &quot;black&quot;),
          axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 90, hjust = 1))
  
  print(p)
}</code></pre>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/TOP2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-TOP2-1">
Past versions of TOP2-1.png
</button>
</p>
<div id="fig-TOP2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/72599accbf90ae9bf96e52d2b799ed6eee6baf48/docs/figure/Cardiac_and_TOP2_Genes.Rmd/TOP2-1.png" target="_blank">72599ac</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/TOP2-1.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/TOP2-1.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/TOP2-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-TOP2-2">
Past versions of TOP2-2.png
</button>
</p>
<div id="fig-TOP2-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/72599accbf90ae9bf96e52d2b799ed6eee6baf48/docs/figure/Cardiac_and_TOP2_Genes.Rmd/TOP2-2.png" target="_blank">72599ac</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/TOP2-2.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/TOP2-2.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="generate-boxplots-for-tp53-genes" class="section level2">
<h2><strong>📌 Generate Boxplots for TP53 Genes</strong></h2>
<pre class="r"><code>for (gene in dna_damage_genes) {
  data_info &lt;- process_gene_data(gene)
  p &lt;- ggplot(data_info$long_data, aes(x = Condition, y = log2CPM, fill = Drug)) +
    geom_boxplot(outlier.shape = NA) +
    scale_fill_manual(values = c(&quot;CX&quot; = &quot;#0000FF&quot;, &quot;DOX&quot; = &quot;#e6d800&quot;, &quot;VEH&quot; = &quot;#FF00FF&quot;)) +
    geom_point(aes(color = Indv), size = 2, alpha = 0.5, position = position_jitter(width = -1, height = 0)) +
    geom_text(data = data_info$significance_labels, aes(x = Condition, y = max_log2CPM + 0.5, label = Significance),
              inherit.aes = FALSE, size = 6, color = &quot;black&quot;) +
    ggtitle(paste(&quot;Log2CPM Expression of&quot;, gene)) +
    labs(x = &quot;Treatment&quot;, y = &quot;log2CPM&quot;) +
    theme_bw() +
    theme(plot.title = element_text(size = rel(2), hjust = 0.5),
          axis.title = element_text(size = 15, color = &quot;black&quot;),
          axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 90, hjust = 1))
  
  print(p)
}</code></pre>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/DNA%20Damage-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-DNA-Damage-1">
Past versions of "DNA Damage-1.png"
</button>
</p>
<div id="fig-no-spaces-DNA-Damage-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/456790a1f560bd70267c93b27598f54a72c51e61/docs/figure/Cardiac_and_TOP2_Genes.Rmd/DNA Damage-1.png" target="_blank">456790a</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>##<strong>📌 DNA Damage Proportion</strong></p>
</div>
<div id="read-and-process-deg-data" class="section level2">
<h2><strong>📌 Read and Process DEG Data</strong></h2>
<pre class="r"><code># Load DEGs Data
CX_0.1_3 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;)
CX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;)
CX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;)
CX_0.5_3 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)

DOX_0.1_3 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;)
DOX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;)
DOX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;)
DOX_0.5_3 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

# Extract Significant DEGs
DEGs &lt;- list(
  &quot;CX_0.1_3&quot; = CX_0.1_3$Entrez_ID[CX_0.1_3$adj.P.Val &lt; 0.05],
  &quot;CX_0.1_24&quot; = CX_0.1_24$Entrez_ID[CX_0.1_24$adj.P.Val &lt; 0.05],
  &quot;CX_0.1_48&quot; = CX_0.1_48$Entrez_ID[CX_0.1_48$adj.P.Val &lt; 0.05],
  &quot;CX_0.5_3&quot; = CX_0.5_3$Entrez_ID[CX_0.5_3$adj.P.Val &lt; 0.05],
  &quot;CX_0.5_24&quot; = CX_0.5_24$Entrez_ID[CX_0.5_24$adj.P.Val &lt; 0.05],
  &quot;CX_0.5_48&quot; = CX_0.5_48$Entrez_ID[CX_0.5_48$adj.P.Val &lt; 0.05],
  &quot;DOX_0.1_3&quot; = DOX_0.1_3$Entrez_ID[DOX_0.1_3$adj.P.Val &lt; 0.05],
  &quot;DOX_0.1_24&quot; = DOX_0.1_24$Entrez_ID[DOX_0.1_24$adj.P.Val &lt; 0.05],
  &quot;DOX_0.1_48&quot; = DOX_0.1_48$Entrez_ID[DOX_0.1_48$adj.P.Val &lt; 0.05],
  &quot;DOX_0.5_3&quot; = DOX_0.5_3$Entrez_ID[DOX_0.5_3$adj.P.Val &lt; 0.05],
  &quot;DOX_0.5_24&quot; = DOX_0.5_24$Entrez_ID[DOX_0.5_24$adj.P.Val &lt; 0.05],
  &quot;DOX_0.5_48&quot; = DOX_0.5_48$Entrez_ID[DOX_0.5_48$adj.P.Val &lt; 0.05]
)

# Extract Significant DEGs
DEG1 &lt;- CX_0.1_3$Entrez_ID[CX_0.1_3$adj.P.Val &lt; 0.05]
DEG2 &lt;- CX_0.1_24$Entrez_ID[CX_0.1_24$adj.P.Val &lt; 0.05]
DEG3 &lt;- CX_0.1_48$Entrez_ID[CX_0.1_48$adj.P.Val &lt; 0.05]
DEG4 &lt;- CX_0.5_3$Entrez_ID[CX_0.5_3$adj.P.Val &lt; 0.05]
DEG5 &lt;- CX_0.5_24$Entrez_ID[CX_0.5_24$adj.P.Val &lt; 0.05]
DEG6 &lt;- CX_0.5_48$Entrez_ID[CX_0.5_48$adj.P.Val &lt; 0.05]

DEG7 &lt;- DOX_0.1_3$Entrez_ID[DOX_0.1_3$adj.P.Val &lt; 0.05]
DEG8 &lt;- DOX_0.1_24$Entrez_ID[DOX_0.1_24$adj.P.Val &lt; 0.05]
DEG9 &lt;- DOX_0.1_48$Entrez_ID[DOX_0.1_48$adj.P.Val &lt; 0.05]
DEG10 &lt;- DOX_0.5_3$Entrez_ID[DOX_0.5_3$adj.P.Val &lt; 0.05]
DEG11 &lt;- DOX_0.5_24$Entrez_ID[DOX_0.5_24$adj.P.Val &lt; 0.05]
DEG12 &lt;- DOX_0.5_48$Entrez_ID[DOX_0.5_48$adj.P.Val &lt; 0.05]</code></pre>
</div>
<div id="dna-damage-repair-proportion-with-chi-square-test"
class="section level2">
<h2><strong>📌 DNA Damage Repair Proportion with Chi-Square
Test</strong></h2>
<pre class="r"><code># Load DNA Damage Genes List
DNA_damage &lt;- read.csv(&quot;data/DNA_Damage.csv&quot;, stringsAsFactors = FALSE)
DNA_damage$Entrez_ID &lt;- mapIds(org.Hs.eg.db,
                               keys = DNA_damage$Symbol,
                               column = &quot;ENTREZID&quot;,
                               keytype = &quot;SYMBOL&quot;,
                               multiVals = &quot;first&quot;)

# Define CX-5461 DEG lists
CX_DEGs &lt;- list(
  &quot;CX_0.1_3&quot; = DEG1, &quot;CX_0.1_24&quot; = DEG2, &quot;CX_0.1_48&quot; = DEG3,
  &quot;CX_0.5_3&quot; = DEG4, &quot;CX_0.5_24&quot; = DEG5, &quot;CX_0.5_48&quot; = DEG6
)

# Define DOX DEG lists
DOX_DEGs &lt;- list(
  &quot;DOX_0.1_3&quot; = DEG7, &quot;DOX_0.1_24&quot; = DEG8, &quot;DOX_0.1_48&quot; = DEG9,
  &quot;DOX_0.5_3&quot; = DEG10, &quot;DOX_0.5_24&quot; = DEG11, &quot;DOX_0.5_48&quot; = DEG12
)

# Extract Entrez_IDs from DNA Damage gene dataset
DNA_damage_genes &lt;- na.omit(DNA_damage$Entrez_ID)

# Combine CX-5461 DEGs into a dataframe with a &quot;Drug&quot; column
CX_DEGs_df &lt;- bind_rows(
  lapply(CX_DEGs, function(ids) data.frame(Entrez_ID = ids, Drug = &quot;CX-5461&quot;)),
  .id = &quot;Sample&quot;
)

# Combine DOX DEGs into a dataframe with a &quot;Drug&quot; column
DOX_DEGs_df &lt;- bind_rows(
  lapply(DOX_DEGs, function(ids) data.frame(Entrez_ID = ids, Drug = &quot;DOX&quot;)),
  .id = &quot;Sample&quot;
)

# Merge CX-5461 and DOX datasets
DEGs_df &lt;- bind_rows(CX_DEGs_df, DOX_DEGs_df)

# Check if genes are in DNA Damage list
DEGs_df &lt;- DEGs_df %&gt;%
  mutate(Category = ifelse(Entrez_ID %in% DNA_damage_genes, &quot;Yes&quot;, &quot;No&quot;))

# Count DNA damage genes in each sample
proportion_data &lt;- DEGs_df %&gt;%
  group_by(Sample, Drug, Category) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Sample, Drug) %&gt;%
  mutate(Percentage = (Count / sum(Count)) * 100)

# Normalize Percentages to Sum Exactly 100%
proportion_data &lt;- proportion_data %&gt;%
  group_by(Sample) %&gt;%
  mutate(Percentage = round(Percentage, 2)) %&gt;%
  mutate(Adjustment = 100 - sum(Percentage, na.rm = TRUE)) %&gt;%
  mutate(Percentage = ifelse(Category == &quot;No&quot;, Percentage + Adjustment, Percentage)) %&gt;%
  mutate(Percentage = ifelse(Percentage &lt; 0, 0, Percentage)) %&gt;%
  mutate(Percentage = ifelse(Percentage &gt; 100, 100, Percentage)) %&gt;%
  ungroup() %&gt;%
  replace_na(list(Percentage = 0))

# Ensure &quot;Yes&quot; is at the Bottom and &quot;No&quot; is at the Top
proportion_data$Category &lt;- factor(proportion_data$Category, levels = c(&quot;Yes&quot;, &quot;No&quot;))

# **🔹 Maintain Correct X-Axis Order (3 → 24 → 48)**
sample_order &lt;- c(
  &quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;, 
  &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
  &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;, 
  &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;
)
proportion_data$Sample &lt;- factor(proportion_data$Sample, levels = sample_order, ordered = TRUE)

# **Perform Chi-Square Test for CX vs DOX Pairs**
chi_square_results &lt;- data.frame(Sample = character(), P_Value = numeric())

for (i in seq(1, 6)) {  # Pairwise comparison (CX vs DOX)
  cx_sample &lt;- sample_order[i]
  dox_sample &lt;- sample_order[i + 6]  # Correctly pairs CX_0.1_3 with DOX_0.1_3, etc.
  
  cx_data &lt;- filter(proportion_data, Sample == cx_sample)
  dox_data &lt;- filter(proportion_data, Sample == dox_sample)
  
  # Construct contingency table for Chi-Square test
  contingency_table &lt;- matrix(
    c(sum(cx_data$Count[cx_data$Category == &quot;Yes&quot;]), sum(cx_data$Count[cx_data$Category == &quot;No&quot;]),
      sum(dox_data$Count[dox_data$Category == &quot;Yes&quot;]), sum(dox_data$Count[dox_data$Category == &quot;No&quot;])),
    nrow = 2, byrow = TRUE
  )
  
  # Run Chi-Square Test
  test_result &lt;- chisq.test(contingency_table)
  p_value &lt;- test_result$p.value
  
  # Store results
  chi_square_results &lt;- rbind(chi_square_results, data.frame(Sample = cx_sample, P_Value = p_value))
}

# Identify significant CX samples (p &lt; 0.05)
chi_square_results$Significant &lt;- ifelse(chi_square_results$P_Value &lt; 0.05, &quot;*&quot;, &quot;&quot;)

# **🔹 Merge Chi-Square Results WITHOUT Modifying Order**
proportion_data &lt;- left_join(proportion_data, chi_square_results, by = &quot;Sample&quot;)

# **Reapply Factor Order to Prevent Changes**
proportion_data$Sample &lt;- factor(proportion_data$Sample, levels = sample_order, ordered = TRUE)</code></pre>
</div>
<div id="dna-damage-repair-proportion-plot" class="section level2">
<h2><strong>📌 DNA Damage Repair Proportion Plot</strong></h2>
<pre class="r"><code># **Generate Proportion Plot for CX-5461 and DOX**
ggplot(proportion_data, aes(x = Sample, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(data = subset(proportion_data, Significant == &quot;*&quot;),
            aes(x = Sample, y = 102, label = &quot;*&quot;),
            size = 6, color = &quot;black&quot;, fontface = &quot;bold&quot;) +
  scale_y_continuous(labels = scales::percent_format(scale = 1), limits = c(0, 105)) +
  scale_fill_manual(values = c(&quot;Yes&quot; = &quot;#e41a1c&quot;, &quot;No&quot; = &quot;#377eb8&quot;)) +
  labs(
    title = &quot;Proportion of DNA Damage Repair Genes in\nCX-5461 and DOX DEGs with Significance&quot;,
    x = &quot;Samples (CX-5461 and DOX)&quot;,
    y = &quot;Percentage&quot;,
    fill = &quot;Category&quot;
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
    legend.title = element_blank(),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    strip.background = element_blank(),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/ggplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-ggplot-1">
Past versions of ggplot-1.png
</button>
</p>
<div id="fig-ggplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/6d5a0a93617d94740c63bf16a444354704c8aa99/docs/figure/Cardiac_and_TOP2_Genes.Rmd/ggplot-1.png" target="_blank">6d5a0a9</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/55dbfa037afc81e5390a8b110f2421a36636623b/docs/figure/Cardiac_and_TOP2_Genes.Rmd/ggplot-1.png" target="_blank">55dbfa0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0a7f53eb3fdbf5160350e69f74f931b54aa0e73a/docs/figure/Cardiac_and_TOP2_Genes.Rmd/ggplot-1.png" target="_blank">0a7f53e</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="p53-target-genes-proportion" class="section level2">
<h2><strong>📌 P53 target genes Proportion</strong></h2>
</div>
<div id="p53-target-genes-proportion-with-chi-square-test"
class="section level2">
<h2><strong>📌 P53 target genes Proportion with Chi-Square
Test</strong></h2>
<pre class="r"><code># Read DNA Damage Genes List
P53_Target &lt;- read.csv(&quot;data/P53_Target.csv&quot;, stringsAsFactors = FALSE)

# Map gene symbols to Entrez IDs using org.Hs.eg.db
P53_Target &lt;- P53_Target %&gt;%
  mutate(Entrez_ID = mapIds(org.Hs.eg.db,
                            keys = Symbol,
                            column = &quot;ENTREZID&quot;,
                            keytype = &quot;SYMBOL&quot;,
                            multiVals = &quot;first&quot;))

# Extract valid Entrez_IDs
P53_Target_genes &lt;- na.omit(P53_Target$Entrez_ID)

# Define CX-5461 DEG lists
CX_DEGs &lt;- list(
  &quot;CX_0.1_3&quot; = DEG1, &quot;CX_0.1_24&quot; = DEG2, &quot;CX_0.1_48&quot; = DEG3,
  &quot;CX_0.5_3&quot; = DEG4, &quot;CX_0.5_24&quot; = DEG5, &quot;CX_0.5_48&quot; = DEG6
)

# Define DOX DEG lists
DOX_DEGs &lt;- list(
  &quot;DOX_0.1_3&quot; = DEG7, &quot;DOX_0.1_24&quot; = DEG8, &quot;DOX_0.1_48&quot; = DEG9,
  &quot;DOX_0.5_3&quot; = DEG10, &quot;DOX_0.5_24&quot; = DEG11, &quot;DOX_0.5_48&quot; = DEG12
)

# Combine CX-5461 DEGs into a dataframe with a &quot;Drug&quot; column
CX_DEGs_df &lt;- bind_rows(
  lapply(CX_DEGs, function(ids) data.frame(Entrez_ID = ids, Drug = &quot;CX-5461&quot;)),
  .id = &quot;Sample&quot;
)

# Combine DOX DEGs into a dataframe with a &quot;Drug&quot; column
DOX_DEGs_df &lt;- bind_rows(
  lapply(DOX_DEGs, function(ids) data.frame(Entrez_ID = ids, Drug = &quot;DOX&quot;)),
  .id = &quot;Sample&quot;
)

# Merge CX-5461 and DOX datasets
DEGs_df &lt;- bind_rows(CX_DEGs_df, DOX_DEGs_df)

# Check if genes are in DNA Damage list
DEGs_df &lt;- DEGs_df %&gt;%
  mutate(Category = ifelse(Entrez_ID %in% P53_Target_genes, &quot;Yes&quot;, &quot;No&quot;))

# Count DNA damage genes in each sample
proportion_data &lt;- DEGs_df %&gt;%
  group_by(Sample, Drug, Category) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Sample, Drug) %&gt;%
  mutate(Percentage = (Count / sum(Count)) * 100)

# Normalize Percentages to Sum Exactly 100%
proportion_data &lt;- proportion_data %&gt;%
  group_by(Sample) %&gt;%
  mutate(Percentage = round(Percentage, 2)) %&gt;%
  mutate(Adjustment = 100 - sum(Percentage, na.rm = TRUE)) %&gt;%
  mutate(Percentage = ifelse(Category == &quot;No&quot;, Percentage + Adjustment, Percentage)) %&gt;%
  mutate(Percentage = pmax(pmin(Percentage, 100), 0)) %&gt;%
  ungroup() %&gt;%
  replace_na(list(Percentage = 0))

# ✅ **Ensure &quot;Yes&quot; is at the Top and &quot;No&quot; at the Bottom**
proportion_data$Category &lt;- factor(proportion_data$Category, levels = c(&quot;Yes&quot;, &quot;No&quot;))

# **Ensure Correct X-Axis Order (3 → 24 → 48)**
sample_order &lt;- c(
  &quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;,
  &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
  &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;,
  &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;
)

# **Perform Chi-Square Test for CX vs DOX Pairs**
chi_square_results &lt;- data.frame(Sample = character(), P_Value = numeric(), Significant = character(), stringsAsFactors = FALSE)

for (i in seq(1, 6)) {  # Pairwise comparison (CX vs DOX)
  cx_sample &lt;- sample_order[i]
  dox_sample &lt;- sample_order[i + 6]  
  
  cx_data &lt;- filter(proportion_data, Sample == cx_sample)
  dox_data &lt;- filter(proportion_data, Sample == dox_sample)
  
  # Construct contingency table
  contingency_table &lt;- matrix(
    c(sum(cx_data$Count[cx_data$Category == &quot;Yes&quot;], na.rm = TRUE),
      sum(cx_data$Count[cx_data$Category == &quot;No&quot;], na.rm = TRUE),
      sum(dox_data$Count[dox_data$Category == &quot;Yes&quot;], na.rm = TRUE),
      sum(dox_data$Count[dox_data$Category == &quot;No&quot;], na.rm = TRUE)),
    nrow = 2, byrow = TRUE
  )
  
  # Run Chi-Square Test (only if there are nonzero values)
  if (all(contingency_table &gt; 0)) {
    test_result &lt;- chisq.test(contingency_table)
    p_value &lt;- test_result$p.value
    
    # ✅ Assign * only if CX is enriched compared to DOX
    if (p_value &lt; 0.05 &amp;&amp; 
        cx_data$Percentage[cx_data$Category == &quot;Yes&quot;] &gt; dox_data$Percentage[dox_data$Category == &quot;Yes&quot;]) {
      chi_square_results &lt;- rbind(chi_square_results, data.frame(Sample = cx_sample, P_Value = p_value, Significant = &quot;*&quot;))
    }
  }
}

# **Merge Chi-Square Results WITHOUT Modifying Order**
proportion_data &lt;- left_join(proportion_data, chi_square_results, by = &quot;Sample&quot;)

proportion_data$Sample &lt;- factor(proportion_data$Sample, levels = sample_order, ordered = TRUE)

# **Generate Proportion Plot for CX-5461 and DOX**
ggplot(proportion_data, aes(x = Sample, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +  # Stacked bars
  geom_text(data = subset(proportion_data, Significant == &quot;*&quot;),
            aes(x = Sample, y = max(Percentage, na.rm = TRUE) + 2, label = &quot;*&quot;),  # ✅ Fix applied here
            size = 6, color = &quot;black&quot;, fontface = &quot;bold&quot;) +
  scale_y_continuous(labels = scales::percent_format(scale = 1), limits = c(0, 105)) +  # Increase Y-axis slightly for better spacing
  scale_fill_manual(values = c(&quot;Yes&quot; = &quot;#e41a1c&quot;, &quot;No&quot; = &quot;#377eb8&quot;)) +  # Yes (Red), No (Blue)
  labs(
    title = &quot;Proportion of P53 target Genes in\nCX-5461 and DOX DEGs with Significance&quot;,
    x = &quot;Samples (CX-5461 and DOX)&quot;,
    y = &quot;Percentage&quot;,
    fill = &quot;Category&quot;
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
    legend.title = element_blank(),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    strip.background = element_blank(),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/P53_target_Proportion1-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 22631)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] clusterProfiler_4.10.1 org.Hs.eg.db_3.18.0    AnnotationDbi_1.64.1  
 [4] IRanges_2.36.0         S4Vectors_0.40.1       Biobase_2.62.0        
 [7] BiocGenerics_0.48.1    tidyr_1.3.1            dplyr_1.1.4           
[10] ggplot2_3.5.1         

loaded via a namespace (and not attached):
  [1] DBI_1.2.3               bitops_1.0-7            gson_0.1.0             
  [4] shadowtext_0.1.4        gridExtra_2.3           rlang_1.1.3            
  [7] magrittr_2.0.3          DOSE_3.28.2             git2r_0.35.0           
 [10] compiler_4.3.0          RSQLite_2.3.3           png_0.1-8              
 [13] vctrs_0.6.5             reshape2_1.4.4          stringr_1.5.1          
 [16] pkgconfig_2.0.3         crayon_1.5.3            fastmap_1.1.1          
 [19] XVector_0.42.0          labeling_0.4.3          ggraph_2.2.1           
 [22] HDO.db_0.99.1           promises_1.3.0          rmarkdown_2.29         
 [25] enrichplot_1.22.0       purrr_1.0.2             bit_4.0.5              
 [28] xfun_0.50               zlibbioc_1.48.0         cachem_1.0.8           
 [31] aplot_0.2.3             GenomeInfoDb_1.38.8     jsonlite_1.8.9         
 [34] blob_1.2.4              later_1.3.2             BiocParallel_1.36.0    
 [37] tweenr_2.0.3            parallel_4.3.0          R6_2.5.1               
 [40] bslib_0.8.0             stringi_1.8.3           RColorBrewer_1.1-3     
 [43] jquerylib_0.1.4         GOSemSim_2.28.1         Rcpp_1.0.12            
 [46] knitr_1.49              httpuv_1.6.15           Matrix_1.6-1.1         
 [49] splines_4.3.0           igraph_2.1.1            tidyselect_1.2.1       
 [52] qvalue_2.34.0           rstudioapi_0.17.1       yaml_2.3.10            
 [55] viridis_0.6.5           codetools_0.2-20        lattice_0.22-5         
 [58] tibble_3.2.1            plyr_1.8.9              treeio_1.26.0          
 [61] withr_3.0.2             KEGGREST_1.42.0         evaluate_1.0.3         
 [64] gridGraphics_0.5-1      scatterpie_0.2.4        polyclip_1.10-7        
 [67] Biostrings_2.70.1       ggtree_3.10.1           pillar_1.10.1          
 [70] whisker_0.4.1           ggfun_0.1.8             generics_0.1.3         
 [73] rprojroot_2.0.4         RCurl_1.98-1.13         tidytree_0.4.6         
 [76] munsell_0.5.1           scales_1.3.0            glue_1.7.0             
 [79] lazyeval_0.2.2          tools_4.3.0             data.table_1.14.10     
 [82] fgsea_1.28.0            fs_1.6.3                graphlayouts_1.2.0     
 [85] fastmatch_1.1-4         tidygraph_1.3.1         cowplot_1.1.3          
 [88] grid_4.3.0              ape_5.8                 colorspace_2.1-0       
 [91] nlme_3.1-166            patchwork_1.3.0         GenomeInfoDbData_1.2.11
 [94] ggforce_0.4.2           cli_3.6.1               workflowr_1.7.1        
 [97] viridisLite_0.4.2       gtable_0.3.6            yulab.utils_0.1.8      
[100] sass_0.4.9              digest_0.6.34           ggplotify_0.1.2        
[103] ggrepel_0.9.6           farver_2.1.2            memoise_2.0.1          
[106] htmltools_0.5.8.1       lifecycle_1.0.4         httr_1.4.7             
[109] GO.db_3.18.0            bit64_4.0.5             MASS_7.3-60            </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
