<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Cardiac, TOP2 and DNA damage Genes</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Paul_CX_2025</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Cardiac, TOP2 and DNA damage Genes</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-07-03
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Paul_CX_2025/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01PaulCX2025tree73dbcc0ec63fb338bacfcb8aa523ae400937601dtargetblank73dbcc0a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/tree/73dbcc0ec63fb338bacfcb8aa523ae400937601d" target="_blank">73dbcc0</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01PaulCX2025tree73dbcc0ec63fb338bacfcb8aa523ae400937601dtargetblank73dbcc0a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sayanpaul01/Paul_CX_2025/tree/73dbcc0ec63fb338bacfcb8aa523ae400937601d" target="_blank">73dbcc0</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    0.1 box.svg
    Ignored:    Rplot04.svg

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/Cardiac_and_TOP2_Genes.Rmd</code>) and HTML
(<code>docs/Cardiac_and_TOP2_Genes.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/4c85c1a279e68367101af8a07e1393ae1360bd9b/docs/Cardiac_and_TOP2_Genes.html" target="_blank">4c85c1a</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-07-03
</td>
<td>
Fix homepage links and update GitHub badge to lab site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/230ea5c5af12358c25015da4bbf86e06c1205626/analysis/Cardiac_and_TOP2_Genes.Rmd" target="_blank">230ea5c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-06-18
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/230ea5c5af12358c25015da4bbf86e06c1205626/docs/Cardiac_and_TOP2_Genes.html" target="_blank">230ea5c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-06-18
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/analysis/Cardiac_and_TOP2_Genes.Rmd" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/Cardiac_and_TOP2_Genes.html" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/a0b0fa3016aaba8e1f257c6afc54a4e696f92bdb/analysis/Cardiac_and_TOP2_Genes.Rmd" target="_blank">a0b0fa3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/a0b0fa3016aaba8e1f257c6afc54a4e696f92bdb/docs/Cardiac_and_TOP2_Genes.html" target="_blank">a0b0fa3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/22e2dc22a7064b6e988e03d2f617eb436c5e3bb8/docs/Cardiac_and_TOP2_Genes.html" target="_blank">22e2dc2</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/01405224251c4c3d17da1027adf4fa117a4ee735/docs/Cardiac_and_TOP2_Genes.html" target="_blank">0140522</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/d735a03cfeb5f79c1fc9bdc94f884fea2487e8f7/analysis/Cardiac_and_TOP2_Genes.Rmd" target="_blank">d735a03</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/d735a03cfeb5f79c1fc9bdc94f884fea2487e8f7/docs/Cardiac_and_TOP2_Genes.html" target="_blank">d735a03</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/b179eba8264934b9d3462c17e85fedf920f45e05/analysis/Cardiac_and_TOP2_Genes.Rmd" target="_blank">b179eba</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/b179eba8264934b9d3462c17e85fedf920f45e05/docs/Cardiac_and_TOP2_Genes.html" target="_blank">b179eba</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/04b721c5960eeb144f48be02a96a77afd8d17dce/analysis/Cardiac_and_TOP2_Genes.Rmd" target="_blank">04b721c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-02
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/04b721c5960eeb144f48be02a96a77afd8d17dce/docs/Cardiac_and_TOP2_Genes.html" target="_blank">04b721c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-02
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/b899d4c2ddc969bf68d432895a791e48654e83c5/analysis/Cardiac_and_TOP2_Genes.Rmd" target="_blank">b899d4c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-02
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/b899d4c2ddc969bf68d432895a791e48654e83c5/docs/Cardiac_and_TOP2_Genes.html" target="_blank">b899d4c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-02
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/c14f8de07187bea3fb9b865466ef1fd4dd40b3d4/docs/Cardiac_and_TOP2_Genes.html" target="_blank">c14f8de</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/6d5a0a93617d94740c63bf16a444354704c8aa99/docs/Cardiac_and_TOP2_Genes.html" target="_blank">6d5a0a9</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c2a6b920376e71b5c6fc084f6763298921b6264a/analysis/Cardiac_and_TOP2_Genes.Rmd" target="_blank">c2a6b92</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
<td>
Updated analysis of Cardiac, TOP2, and DNA Damage Genes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/1337c6888968145907547d61ab32b7a2910b6fea/docs/Cardiac_and_TOP2_Genes.html" target="_blank">1337c68</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/55dbfa037afc81e5390a8b110f2421a36636623b/docs/Cardiac_and_TOP2_Genes.html" target="_blank">55dbfa0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/1b220adf2523368d0132d73371074f673839cf15/analysis/Cardiac_and_TOP2_Genes.Rmd" target="_blank">1b220ad</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
<td>
Updated analysis of Cardiac, TOP2, and DNA Damage Genes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/0a7f53eb3fdbf5160350e69f74f931b54aa0e73a/docs/Cardiac_and_TOP2_Genes.html" target="_blank">0a7f53e</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/565ede2e548ade39425a2be455d3cd0d62edf158/analysis/Cardiac_and_TOP2_Genes.Rmd" target="_blank">565ede2</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
<td>
Updated analysis of Cardiac, TOP2, and DNA Damage Genes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/3313173f542c38cf7a8bdab7d6e8ec20e3b456f2/docs/Cardiac_and_TOP2_Genes.html" target="_blank">3313173</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/456790a1f560bd70267c93b27598f54a72c51e61/docs/Cardiac_and_TOP2_Genes.html" target="_blank">456790a</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/e943e2d7775dac9d6c268443840ca24303996b84/analysis/Cardiac_and_TOP2_Genes.Rmd" target="_blank">e943e2d</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
<td>
Added TP53 as a DNA damage marker in log2CPM boxplots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/6ed78e299a335e7a75dd8448fc6eb6a810537b3d/docs/Cardiac_and_TOP2_Genes.html" target="_blank">6ed78e2</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/72599accbf90ae9bf96e52d2b799ed6eee6baf48/docs/Cardiac_and_TOP2_Genes.html" target="_blank">72599ac</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/0e79f5b46c58a97a56b44cd383e25e397f65a5dd/analysis/Cardiac_and_TOP2_Genes.Rmd" target="_blank">0e79f5b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
<td>
Fix Timepoint Ordering in Boxplots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/Cardiac_and_TOP2_Genes.html" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/91f5f8f51cd57fdd0d6ab9be529ef747ba23f697/analysis/Cardiac_and_TOP2_Genes.Rmd" target="_blank">91f5f8f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
<td>
Fix Timepoint Ordering in Boxplots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/bcc58f6f39f4dd5e241b7d0026e9d7b927d68f83/docs/Cardiac_and_TOP2_Genes.html" target="_blank">bcc58f6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/Cardiac_and_TOP2_Genes.html" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c6b5ccd218c3fd41470b2b766ba41ed8c2aed105/analysis/Cardiac_and_TOP2_Genes.Rmd" target="_blank">c6b5ccd</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
<td>
Fixed boxplot significance issue for Cardiac and TOP2 genes
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="log2cpm-boxplots-for-cardiac-and-top2-genes"
class="section level2">
<h2><strong>📌 Log2CPM Boxplots for Cardiac and TOP2 Genes</strong></h2>
<p>This analysis generates <strong>boxplots for cardiac genes and TOP2
genes</strong> across different treatments and timepoints.</p>
<hr />
</div>
<div id="load-required-libraries" class="section level2">
<h2><strong>📌 Load Required Libraries</strong></h2>
<pre class="r"><code>library(ggplot2)
library(dplyr)</code></pre>
<pre><code>Warning: package &#39;dplyr&#39; was built under R version 4.3.2</code></pre>
<pre class="r"><code>library(tidyr)</code></pre>
<pre><code>Warning: package &#39;tidyr&#39; was built under R version 4.3.3</code></pre>
<pre class="r"><code>library(org.Hs.eg.db)</code></pre>
<pre><code>Warning: package &#39;AnnotationDbi&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;BiocGenerics&#39; was built under R version 4.3.1</code></pre>
<pre><code>Warning: package &#39;Biobase&#39; was built under R version 4.3.1</code></pre>
<pre><code>Warning: package &#39;IRanges&#39; was built under R version 4.3.1</code></pre>
<pre><code>Warning: package &#39;S4Vectors&#39; was built under R version 4.3.2</code></pre>
<pre class="r"><code>library(clusterProfiler)</code></pre>
<pre><code>Warning: package &#39;clusterProfiler&#39; was built under R version 4.3.3</code></pre>
</div>
<div id="read-log2cpm-data" class="section level2">
<h2><strong>📌 Read Log2CPM Data</strong></h2>
<pre class="r"><code># Load feature count matrix
boxplot1 &lt;- read.csv(&quot;data/Feature_count_Matrix_Log2CPM_filtered.csv&quot;) %&gt;% as.data.frame()

# Ensure column names are cleaned
colnames(boxplot1) &lt;- trimws(gsub(&quot;^X&quot;, &quot;&quot;, colnames(boxplot1)))  </code></pre>
</div>
<div id="define-genes-of-interest" class="section level2">
<h2><strong>📌 Define Genes of Interest</strong></h2>
<pre class="r"><code># Define the genes of interest
top2_genes &lt;- c(&quot;TOP2A&quot;, &quot;TOP2B&quot;)
cardiac_genes &lt;- c(&quot;ACTN2&quot;, &quot;CALR&quot;, &quot;MYBPC3&quot;, &quot;MYH6&quot;, &quot;MYH7&quot;, &quot;NKX2-5&quot;,&quot;MYL2&quot;, &quot;RYR2&quot;, &quot;SCN5A&quot;, &quot;TNNI3&quot;, &quot;TNNT2&quot;, &quot;TTN&quot;)
dna_damage_genes &lt;- c(&quot;TP53&quot;)  # Using correct gene symbol TP53
dna_repair_genes &lt;- c(&quot;H2AZ1&quot;, &quot;UBE2T&quot;, &quot;MMS22L&quot;,&quot;PIAS3&quot;)
p53_target_genes &lt;- c(&quot;IER5&quot;, &quot;HHAT&quot;, &quot;EPS8L2&quot;)</code></pre>
</div>
<div id="read-and-process-degs-data" class="section level2">
<h2><strong>📌 Read and Process DEGs Data</strong></h2>
<pre class="r"><code># Load Toptables
deg_files &lt;- list.files(&quot;data/DEGs&quot;, pattern = &quot;Toptable_.*\\.csv&quot;, full.names = TRUE)
deg_list &lt;- lapply(deg_files, read.csv)
names(deg_list) &lt;- gsub(&quot;data/DEGs/Toptable_|\\.csv&quot;, &quot;&quot;, deg_files)  

# Function to check significance based on **Entrez_ID in the correct sample**
is_significant &lt;- function(gene, drug, conc, timepoint) {
  condition &lt;- paste(drug, conc, timepoint, sep = &quot;_&quot;)
  if (!condition %in% names(deg_list)) return(FALSE)
  
  toptable &lt;- deg_list[[condition]]
  gene_entrez &lt;- boxplot1$ENTREZID[boxplot1$SYMBOL == gene]
  
  if (length(gene_entrez) == 0) return(FALSE)
  
  return(any(gene_entrez %in% toptable$Entrez_ID[toptable$adj.P.Val &lt; 0.05]))
}</code></pre>
</div>
<div id="process-data-for-plotting" class="section level2">
<h2><strong>📌 Process Data for Plotting</strong></h2>
<pre class="r"><code>process_gene_data &lt;- function(gene) {
  # Filter log2CPM data for the gene
  gene_data &lt;- boxplot1 %&gt;% filter(SYMBOL == gene)
  
  # Reshape data
  long_data &lt;- gene_data %&gt;%
    pivot_longer(cols = -c(ENTREZID, SYMBOL, GENENAME), names_to = &quot;Sample&quot;, values_to = &quot;log2CPM&quot;) %&gt;%
    mutate(
      Indv = case_when(
        grepl(&quot;75.1&quot;, Sample) ~ &quot;1&quot;,
        grepl(&quot;78.1&quot;, Sample) ~ &quot;2&quot;,
        grepl(&quot;87.1&quot;, Sample) ~ &quot;3&quot;,
        grepl(&quot;17.3&quot;, Sample) ~ &quot;4&quot;,
        grepl(&quot;84.1&quot;, Sample) ~ &quot;5&quot;,
        grepl(&quot;90.1&quot;, Sample) ~ &quot;6&quot;,
        TRUE ~ NA_character_
      ),
      Drug = case_when(
        grepl(&quot;CX.5461&quot;, Sample) ~ &quot;CX&quot;,
        grepl(&quot;DOX&quot;, Sample) ~ &quot;DOX&quot;,
        grepl(&quot;VEH&quot;, Sample) ~ &quot;VEH&quot;,
        TRUE ~ NA_character_
      ),
      Conc. = case_when(
        grepl(&quot;_0.1_&quot;, Sample) ~ &quot;0.1&quot;,
        grepl(&quot;_0.5_&quot;, Sample) ~ &quot;0.5&quot;,
        TRUE ~ NA_character_
      ),
      Timepoint = case_when(
        grepl(&quot;_3$&quot;, Sample) ~ &quot;3&quot;,
        grepl(&quot;_24$&quot;, Sample) ~ &quot;24&quot;,
        grepl(&quot;_48$&quot;, Sample) ~ &quot;48&quot;,
        TRUE ~ NA_character_
      ),
      Condition = paste(Drug, Conc., Timepoint, sep = &quot;_&quot;)
    )

  # **Ensure Condition is Ordered Correctly**
  long_data$Condition &lt;- factor(
    long_data$Condition, 
    levels = c(
      &quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;, &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
      &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;, &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;,
      &quot;VEH_0.1_3&quot;, &quot;VEH_0.1_24&quot;, &quot;VEH_0.1_48&quot;, &quot;VEH_0.5_3&quot;, &quot;VEH_0.5_24&quot;, &quot;VEH_0.5_48&quot;
    )
  )
  
  # Identify significant conditions **per Drug, Conc, and Timepoint**
  significance_labels &lt;- long_data %&gt;%
    distinct(Drug, Conc., Timepoint, Condition) %&gt;%
    rowwise() %&gt;%
    mutate(
      max_log2CPM = max(long_data$log2CPM[long_data$Condition == Condition], na.rm = TRUE),
      Significance = ifelse(is_significant(gene, Drug, Conc., Timepoint), &quot;*&quot;, &quot;&quot;)
    ) %&gt;%
    filter(Significance != &quot;&quot;) %&gt;% ungroup()
  
  list(long_data = long_data, significance_labels = significance_labels)
}</code></pre>
</div>
<div id="generate-boxplots-for-cardiac-genes" class="section level2">
<h2><strong>📌Generate Boxplots for Cardiac Genes</strong></h2>
<pre class="r"><code>for (gene in cardiac_genes) {
  data_info &lt;- process_gene_data(gene)
  p &lt;- ggplot(data_info$long_data, aes(x = Condition, y = log2CPM, fill = Drug)) +
    geom_boxplot(outlier.shape = NA) +
    scale_fill_manual(values = c(&quot;CX&quot; = &quot;#0000FF&quot;, &quot;DOX&quot; = &quot;#e6d800&quot;, &quot;VEH&quot; = &quot;#FF00FF&quot;)) +
    geom_point(aes(color = Indv), size = 2, alpha = 0.5, position = position_jitter(width = -1, height = 0)) +
    geom_text(data = data_info$significance_labels, aes(x = Condition, y = max_log2CPM + 0.5, label = Significance),
              inherit.aes = FALSE, size = 6, color = &quot;black&quot;) +
    ggtitle(paste(&quot;Log2CPM Expression of&quot;, gene)) +
    labs(x = &quot;Treatment&quot;, y = &quot;log2CPM&quot;) +
    theme_bw() +
    theme(plot.title = element_text(size = rel(2), hjust = 0.5),
          axis.title = element_text(size = 15, color = &quot;black&quot;),
          axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 90, hjust = 1))
  
  print(p)
}</code></pre>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cardiac-1">
Past versions of Cardiac-1.png
</button>
</p>
<div id="fig-Cardiac-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-1.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-1.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-1.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cardiac-2">
Past versions of Cardiac-2.png
</button>
</p>
<div id="fig-Cardiac-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-2.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-2.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-2.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cardiac-3">
Past versions of Cardiac-3.png
</button>
</p>
<div id="fig-Cardiac-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-3.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-3.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-3.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cardiac-4">
Past versions of Cardiac-4.png
</button>
</p>
<div id="fig-Cardiac-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-4.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-4.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-4.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cardiac-5">
Past versions of Cardiac-5.png
</button>
</p>
<div id="fig-Cardiac-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-5.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-5.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-5.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cardiac-6">
Past versions of Cardiac-6.png
</button>
</p>
<div id="fig-Cardiac-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-6.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-6.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-6.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cardiac-7">
Past versions of Cardiac-7.png
</button>
</p>
<div id="fig-Cardiac-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-7.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-7.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-7.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cardiac-8">
Past versions of Cardiac-8.png
</button>
</p>
<div id="fig-Cardiac-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-8.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-8.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-8.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cardiac-9">
Past versions of Cardiac-9.png
</button>
</p>
<div id="fig-Cardiac-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-9.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-9.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-9.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cardiac-10">
Past versions of Cardiac-10.png
</button>
</p>
<div id="fig-Cardiac-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-10.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-10.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-10.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cardiac-11">
Past versions of Cardiac-11.png
</button>
</p>
<div id="fig-Cardiac-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-11.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-11.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-11.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cardiac-12">
Past versions of Cardiac-12.png
</button>
</p>
<div id="fig-Cardiac-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Cardiac-12.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="generate-boxplots-for-veh_cardiac-genes"
class="section level2">
<h2><strong>📌Generate Boxplots for Veh_Cardiac Genes</strong></h2>
<pre class="r"><code># 📌 Prepare data
cardiac_data &lt;- boxplot1 %&gt;%
  filter(SYMBOL %in% cardiac_genes) %&gt;%
  pivot_longer(cols = -c(ENTREZID, SYMBOL, GENENAME), names_to = &quot;Sample&quot;, values_to = &quot;log2CPM&quot;) %&gt;%
  mutate(
    Indv = case_when(
      grepl(&quot;75.1&quot;, Sample) ~ &quot;1&quot;,
      grepl(&quot;78.1&quot;, Sample) ~ &quot;2&quot;,
      grepl(&quot;87.1&quot;, Sample) ~ &quot;3&quot;,
      grepl(&quot;17.3&quot;, Sample) ~ &quot;4&quot;,
      grepl(&quot;84.1&quot;, Sample) ~ &quot;5&quot;,
      grepl(&quot;90.1&quot;, Sample) ~ &quot;6&quot;,
      TRUE ~ NA_character_
    ),
    Drug = case_when(
      grepl(&quot;CX.5461&quot;, Sample) ~ &quot;CX&quot;,
      grepl(&quot;DOX&quot;, Sample) ~ &quot;DOX&quot;,
      grepl(&quot;VEH&quot;, Sample) ~ &quot;VEH&quot;,
      TRUE ~ NA_character_
    ),
    Conc. = case_when(
      grepl(&quot;_0.1_&quot;, Sample) ~ &quot;0.1&quot;,
      grepl(&quot;_0.5_&quot;, Sample) ~ &quot;0.5&quot;,
      TRUE ~ NA_character_
    ),
    Timepoint = case_when(
      grepl(&quot;_3$&quot;, Sample) ~ &quot;3&quot;,
      grepl(&quot;_24$&quot;, Sample) ~ &quot;24&quot;,
      grepl(&quot;_48$&quot;, Sample) ~ &quot;48&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(Drug == &quot;VEH&quot;)

# 📌 Set factors
cardiac_data$SYMBOL &lt;- factor(cardiac_data$SYMBOL, levels = cardiac_genes)
cardiac_data$Timepoint &lt;- factor(cardiac_data$Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
cardiac_data$Conc. &lt;- factor(cardiac_data$Conc., levels = c(&quot;0.1&quot;, &quot;0.5&quot;))

# Add a new combined X-axis label: Gene + Timepoint
cardiac_data &lt;- cardiac_data %&gt;%
  mutate(Gene_Time = interaction(SYMBOL, Timepoint, sep = &quot;_&quot;),
         SYMBOL = factor(SYMBOL, levels = cardiac_genes),
         Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;)),
         Conc. = factor(Conc., levels = c(&quot;0.1&quot;, &quot;0.5&quot;)))

# Plot using Gene-Time combination for clean x-axis dodge
ggplot(cardiac_data, aes(x = SYMBOL, y = log2CPM, fill = Timepoint)) +
  geom_boxplot(
    aes(group = interaction(SYMBOL, Timepoint)),
    position = position_dodge(width = 0.8),
    outlier.shape = NA,
    width = 0.6
  ) +
  geom_point(
    aes(color = Indv, group = interaction(SYMBOL, Timepoint)),
    position = position_dodge(width = 0.8),
    size = 2,
    alpha = 0.8
  ) +
  facet_grid(. ~ Conc., labeller = label_both) +
  labs(
    title = &quot;Cardiac Gene Expression in Vehicle-Treated iPSC-CMs&quot;,
    x = &quot;Cardiac Gene&quot;,
    y = &quot;log2CPM&quot;
  ) +
  theme_bw() +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.text.x = element_text(angle = 45, hjust = 1, size = 11),
    axis.title = element_text(size = 14),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    legend.title = element_text(face = &quot;bold&quot;),
    legend.position = &quot;right&quot;
  )</code></pre>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/Veh_cardiac-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Veh_cardiac-1">
Past versions of Veh_cardiac-1.png
</button>
</p>
<div id="fig-Veh_cardiac-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/230ea5c5af12358c25015da4bbf86e06c1205626/docs/figure/Cardiac_and_TOP2_Genes.Rmd/Veh_cardiac-1.png" target="_blank">230ea5c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-06-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="generate-boxplots-for-top2-genes" class="section level2">
<h2><strong>📌Generate Boxplots for TOP2 Genes</strong></h2>
<pre class="r"><code>for (gene in top2_genes) {
  data_info &lt;- process_gene_data(gene)
  p &lt;- ggplot(data_info$long_data, aes(x = Condition, y = log2CPM, fill = Drug)) +
    geom_boxplot(outlier.shape = NA) +
    scale_fill_manual(values = c(&quot;CX&quot; = &quot;#0000FF&quot;, &quot;DOX&quot; = &quot;#e6d800&quot;, &quot;VEH&quot; = &quot;#FF00FF&quot;)) +
    geom_point(aes(color = Indv), size = 2, alpha = 0.5, position = position_jitter(width = -1, height = 0)) +
    geom_text(data = data_info$significance_labels, aes(x = Condition, y = max_log2CPM + 0.5, label = Significance),
              inherit.aes = FALSE, size = 6, color = &quot;black&quot;) +
    ggtitle(paste(&quot;Log2CPM Expression of&quot;, gene)) +
    labs(x = &quot;Treatment&quot;, y = &quot;log2CPM&quot;) +
    theme_bw() +
    theme(plot.title = element_text(size = rel(2), hjust = 0.5),
          axis.title = element_text(size = 15, color = &quot;black&quot;),
          axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 90, hjust = 1))
  
  print(p)
}</code></pre>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/TOP2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-TOP2-1">
Past versions of TOP2-1.png
</button>
</p>
<div id="fig-TOP2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/TOP2-1.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/72599accbf90ae9bf96e52d2b799ed6eee6baf48/docs/figure/Cardiac_and_TOP2_Genes.Rmd/TOP2-1.png" target="_blank">72599ac</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/TOP2-1.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/TOP2-1.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/TOP2-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-TOP2-2">
Past versions of TOP2-2.png
</button>
</p>
<div id="fig-TOP2-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/TOP2-2.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/72599accbf90ae9bf96e52d2b799ed6eee6baf48/docs/figure/Cardiac_and_TOP2_Genes.Rmd/TOP2-2.png" target="_blank">72599ac</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/a41bd50f54ee5b4135624dfb8e15703031b2b998/docs/figure/Cardiac_and_TOP2_Genes.Rmd/TOP2-2.png" target="_blank">a41bd50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/8c1912ffcb4a6b781973fbafe15cbad8209cc4ea/docs/figure/Cardiac_and_TOP2_Genes.Rmd/TOP2-2.png" target="_blank">8c1912f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="generate-boxplots-for-tp53-genes" class="section level2">
<h2><strong>📌 Generate Boxplots for TP53 Genes</strong></h2>
<pre class="r"><code>for (gene in dna_damage_genes) {
  data_info &lt;- process_gene_data(gene)
  p &lt;- ggplot(data_info$long_data, aes(x = Condition, y = log2CPM, fill = Drug)) +
    geom_boxplot(outlier.shape = NA) +
    scale_fill_manual(values = c(&quot;CX&quot; = &quot;#0000FF&quot;, &quot;DOX&quot; = &quot;#e6d800&quot;, &quot;VEH&quot; = &quot;#FF00FF&quot;)) +
    geom_point(aes(color = Indv), size = 2, alpha = 0.5, position = position_jitter(width = -1, height = 0)) +
    geom_text(data = data_info$significance_labels, aes(x = Condition, y = max_log2CPM + 0.5, label = Significance),
              inherit.aes = FALSE, size = 6, color = &quot;black&quot;) +
    ggtitle(paste(&quot;Log2CPM Expression of&quot;, gene)) +
    labs(x = &quot;Treatment&quot;, y = &quot;log2CPM&quot;) +
    theme_bw() +
    theme(plot.title = element_text(size = rel(2), hjust = 0.5),
          axis.title = element_text(size = 15, color = &quot;black&quot;),
          axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 90, hjust = 1))
  
  print(p)
}</code></pre>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/DNA%20Damage-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-DNA-Damage-1">
Past versions of "DNA Damage-1.png"
</button>
</p>
<div id="fig-no-spaces-DNA-Damage-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/DNA Damage-1.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/456790a1f560bd70267c93b27598f54a72c51e61/docs/figure/Cardiac_and_TOP2_Genes.Rmd/DNA Damage-1.png" target="_blank">456790a</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>##<strong>📌 DNA Damage Repair Genes Proportion</strong></p>
</div>
<div id="read-and-process-deg-data" class="section level2">
<h2><strong>📌 Read and Process DEG Data</strong></h2>
<pre class="r"><code># Load DEGs Data
CX_0.1_3 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;)
CX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;)
CX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;)
CX_0.5_3 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)

DOX_0.1_3 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;)
DOX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;)
DOX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;)
DOX_0.5_3 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

# Extract Significant DEGs
DEGs &lt;- list(
  &quot;CX_0.1_3&quot; = CX_0.1_3$Entrez_ID[CX_0.1_3$adj.P.Val &lt; 0.05],
  &quot;CX_0.1_24&quot; = CX_0.1_24$Entrez_ID[CX_0.1_24$adj.P.Val &lt; 0.05],
  &quot;CX_0.1_48&quot; = CX_0.1_48$Entrez_ID[CX_0.1_48$adj.P.Val &lt; 0.05],
  &quot;CX_0.5_3&quot; = CX_0.5_3$Entrez_ID[CX_0.5_3$adj.P.Val &lt; 0.05],
  &quot;CX_0.5_24&quot; = CX_0.5_24$Entrez_ID[CX_0.5_24$adj.P.Val &lt; 0.05],
  &quot;CX_0.5_48&quot; = CX_0.5_48$Entrez_ID[CX_0.5_48$adj.P.Val &lt; 0.05],
  &quot;DOX_0.1_3&quot; = DOX_0.1_3$Entrez_ID[DOX_0.1_3$adj.P.Val &lt; 0.05],
  &quot;DOX_0.1_24&quot; = DOX_0.1_24$Entrez_ID[DOX_0.1_24$adj.P.Val &lt; 0.05],
  &quot;DOX_0.1_48&quot; = DOX_0.1_48$Entrez_ID[DOX_0.1_48$adj.P.Val &lt; 0.05],
  &quot;DOX_0.5_3&quot; = DOX_0.5_3$Entrez_ID[DOX_0.5_3$adj.P.Val &lt; 0.05],
  &quot;DOX_0.5_24&quot; = DOX_0.5_24$Entrez_ID[DOX_0.5_24$adj.P.Val &lt; 0.05],
  &quot;DOX_0.5_48&quot; = DOX_0.5_48$Entrez_ID[DOX_0.5_48$adj.P.Val &lt; 0.05]
)

# Extract Significant DEGs
DEG1 &lt;- CX_0.1_3$Entrez_ID[CX_0.1_3$adj.P.Val &lt; 0.05]
DEG2 &lt;- CX_0.1_24$Entrez_ID[CX_0.1_24$adj.P.Val &lt; 0.05]
DEG3 &lt;- CX_0.1_48$Entrez_ID[CX_0.1_48$adj.P.Val &lt; 0.05]
DEG4 &lt;- CX_0.5_3$Entrez_ID[CX_0.5_3$adj.P.Val &lt; 0.05]
DEG5 &lt;- CX_0.5_24$Entrez_ID[CX_0.5_24$adj.P.Val &lt; 0.05]
DEG6 &lt;- CX_0.5_48$Entrez_ID[CX_0.5_48$adj.P.Val &lt; 0.05]

DEG7 &lt;- DOX_0.1_3$Entrez_ID[DOX_0.1_3$adj.P.Val &lt; 0.05]
DEG8 &lt;- DOX_0.1_24$Entrez_ID[DOX_0.1_24$adj.P.Val &lt; 0.05]
DEG9 &lt;- DOX_0.1_48$Entrez_ID[DOX_0.1_48$adj.P.Val &lt; 0.05]
DEG10 &lt;- DOX_0.5_3$Entrez_ID[DOX_0.5_3$adj.P.Val &lt; 0.05]
DEG11 &lt;- DOX_0.5_24$Entrez_ID[DOX_0.5_24$adj.P.Val &lt; 0.05]
DEG12 &lt;- DOX_0.5_48$Entrez_ID[DOX_0.5_48$adj.P.Val &lt; 0.05]</code></pre>
</div>
<div id="dna-damage-repair-proportion-with-chi-square-test"
class="section level2">
<h2><strong>📌 DNA Damage Repair Proportion with Chi-Square
Test</strong></h2>
<pre class="r"><code># Read DNA Damage Genes List
DNA_damage &lt;- read.csv(&quot;data/DNA_Damage.csv&quot;, stringsAsFactors = FALSE)

# Convert gene symbols to Entrez IDs
DNA_damage &lt;- DNA_damage %&gt;%
  mutate(Entrez_ID = mapIds(org.Hs.eg.db,
                            keys = Symbol,
                            column = &quot;ENTREZID&quot;,
                            keytype = &quot;SYMBOL&quot;,
                            multiVals = &quot;first&quot;))

# Extract DNA damage gene Entrez IDs
DNA_damage_genes &lt;- na.omit(DNA_damage$Entrez_ID)
total_DNA_damage_genes &lt;- length(DNA_damage_genes)  # Total number of DNA damage genes

# Define CX-5461 DEG lists
CX_DEGs &lt;- list(
  &quot;CX_0.1_3&quot; = DEG1, &quot;CX_0.1_24&quot; = DEG2, &quot;CX_0.1_48&quot; = DEG3,
  &quot;CX_0.5_3&quot; = DEG4, &quot;CX_0.5_24&quot; = DEG5, &quot;CX_0.5_48&quot; = DEG6
)

# Define DOX DEG lists
DOX_DEGs &lt;- list(
  &quot;DOX_0.1_3&quot; = DEG7, &quot;DOX_0.1_24&quot; = DEG8, &quot;DOX_0.1_48&quot; = DEG9,
  &quot;DOX_0.5_3&quot; = DEG10, &quot;DOX_0.5_24&quot; = DEG11, &quot;DOX_0.5_48&quot; = DEG12
)

# Function to calculate the presence of DNA damage genes in DEGs
calculate_proportion &lt;- function(deg_list, drug_name) {
  data.frame(
    Sample = names(deg_list),
    Drug = drug_name,
    DNA_Damage_DEGs = sapply(deg_list, function(ids) sum(ids %in% DNA_damage_genes)),  # DEGs present in DNA damage set
    Non_DNA_Damage_DEGs = sapply(deg_list, function(ids) total_DNA_damage_genes - sum(ids %in% DNA_damage_genes))  # Remaining DNA damage genes
  ) %&gt;%
    mutate(
      Yes_Proportion = (DNA_Damage_DEGs / total_DNA_damage_genes) * 100,  # Percentage of DEGs in DNA damage genes
      No_Proportion = (Non_DNA_Damage_DEGs / total_DNA_damage_genes) * 100  # Remaining DNA damage genes as No
    )
}

# Calculate proportions for CX-5461 and DOX
CX_proportion &lt;- calculate_proportion(CX_DEGs, &quot;CX-5461&quot;)
DOX_proportion &lt;- calculate_proportion(DOX_DEGs, &quot;DOX&quot;)

# Combine data
proportion_data &lt;- bind_rows(CX_proportion, DOX_proportion)

# Convert to long format for stacked bar plot
proportion_long &lt;- proportion_data %&gt;%
  select(Sample, Drug, Yes_Proportion, No_Proportion) %&gt;%
  pivot_longer(cols = c(Yes_Proportion, No_Proportion), names_to = &quot;Category&quot;, values_to = &quot;Percentage&quot;) %&gt;%
  mutate(Category = ifelse(Category == &quot;Yes_Proportion&quot;, &quot;Yes&quot;, &quot;No&quot;))

# **Ensure correct order of samples on X-axis**
sample_order &lt;- c(
  &quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;, &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
  &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;, &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;
)
proportion_long$Sample &lt;- factor(proportion_long$Sample, levels = sample_order, ordered = TRUE)

# **Fix: Ensure &quot;Yes&quot; is on top and &quot;No&quot; is at the bottom in stacked bars**
proportion_long$Category &lt;- factor(proportion_long$Category, levels = c(&quot;Yes&quot;, &quot;No&quot;))  # Ensures &quot;Yes&quot; on top, &quot;No&quot; at bottom

# **Perform Chi-Square Test for CX vs DOX at each timepoint**
chi_square_results &lt;- data.frame(Sample = character(), P_Value = numeric())

for (i in seq(1, 6)) {  # Pairwise comparison (CX vs DOX)
  cx_sample &lt;- sample_order[i]
  dox_sample &lt;- sample_order[i + 6]  # Matches CX_0.1_3 with DOX_0.1_3, etc.
  
  cx_data &lt;- filter(proportion_data, Sample == cx_sample)
  dox_data &lt;- filter(proportion_data, Sample == dox_sample)
  
  # Construct contingency table for Chi-Square test
  contingency_table &lt;- matrix(
    c(cx_data$DNA_Damage_DEGs, cx_data$Non_DNA_Damage_DEGs,
      dox_data$DNA_Damage_DEGs, dox_data$Non_DNA_Damage_DEGs),
    nrow = 2, byrow = TRUE
  )
  
  # Run Chi-Square Test
  test_result &lt;- chisq.test(contingency_table)
  p_value &lt;- test_result$p.value
  
  # Store results
  chi_square_results &lt;- rbind(chi_square_results, data.frame(Sample = cx_sample, P_Value = p_value))
}

# Add significance stars
chi_square_results$Significant &lt;- ifelse(chi_square_results$P_Value &lt; 0.05, &quot;*&quot;, &quot;&quot;)

# Merge Chi-Square results
proportion_long &lt;- left_join(proportion_long, chi_square_results, by = &quot;Sample&quot;)

# **Save output**
write.csv(proportion_long, &quot;C:/Work/Postdoc_UTMB/CX-5461 Project/Transcriptome literatures/lit2/Proportion_Stacked_DNA_Damage_DEGs_with_ChiSquare.csv&quot;, row.names = FALSE)


# Define correct factor orders for samples
sample_order &lt;- c(
  &quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;, &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
  &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;, &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;
)

# Reapply factor levels for correct order in both proportion_data and proportion_long
proportion_data$Sample &lt;- factor(proportion_data$Sample, levels = sample_order, ordered = TRUE)
proportion_long$Sample &lt;- factor(proportion_long$Sample, levels = sample_order, ordered = TRUE)

# **Fix: Ensure &quot;Yes&quot; is on top and &quot;No&quot; is at the bottom in stacked bars**
proportion_long$Category &lt;- factor(proportion_long$Category, levels = c(&quot;Yes&quot;, &quot;No&quot;))</code></pre>
</div>
<div id="dna-damage-repair-proportion-plot" class="section level2">
<h2><strong>📌 DNA Damage Repair Proportion Plot</strong></h2>
<pre class="r"><code># **Generate Stacked Bar Plot with Correct X-Axis Order**
ggplot(proportion_long, aes(x = Sample, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +  # Stacked bars
  geom_text(data = subset(proportion_long, Significant == &quot;*&quot;),
            aes(x = Sample, y = 102, label = &quot;*&quot;),  # Position stars slightly above 100%
            size = 6, color = &quot;black&quot;, fontface = &quot;bold&quot;) +
  scale_y_continuous(labels = scales::percent_format(scale = 1), limits = c(0, 105)) +  # Increase Y-axis slightly
  scale_fill_manual(values = c(&quot;Yes&quot; = &quot;#e41a1c&quot;, &quot;No&quot; = &quot;#377eb8&quot;)) +  # Yes (Red), No (Blue)
  labs(
    title = &quot;Proportion of CX-5461 and DOX DEGs in\nDNA Damage Repair Genes with Significance&quot;,
    x = &quot;Samples (CX-5461 and DOX)&quot;,
    y = &quot;Percentage&quot;,
    fill = &quot;Category&quot;
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
    legend.title = element_blank(),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    strip.background = element_blank(),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/ggplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-ggplot-1">
Past versions of ggplot-1.png
</button>
</p>
<div id="fig-ggplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/ggplot-1.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/d735a03cfeb5f79c1fc9bdc94f884fea2487e8f7/docs/figure/Cardiac_and_TOP2_Genes.Rmd/ggplot-1.png" target="_blank">d735a03</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/6d5a0a93617d94740c63bf16a444354704c8aa99/docs/figure/Cardiac_and_TOP2_Genes.Rmd/ggplot-1.png" target="_blank">6d5a0a9</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/55dbfa037afc81e5390a8b110f2421a36636623b/docs/figure/Cardiac_and_TOP2_Genes.Rmd/ggplot-1.png" target="_blank">55dbfa0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/0a7f53eb3fdbf5160350e69f74f931b54aa0e73a/docs/figure/Cardiac_and_TOP2_Genes.Rmd/ggplot-1.png" target="_blank">0a7f53e</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dna-damage-repair-abs-logfc-distribution"
class="section level2">
<h2><strong>📌 DNA Damage Repair abs logFC distribution</strong></h2>
<pre class="r"><code># Load necessary libraries
library(dplyr)
library(ggplot2)
library(tidyr)
library(org.Hs.eg.db)
library(rstatix)</code></pre>
<pre><code>Warning: package &#39;rstatix&#39; was built under R version 4.3.1</code></pre>
<pre class="r"><code># Read DNA Damage Response Gene List
DNA_damage &lt;- read.csv(&quot;data/DNA_Damage.csv&quot;, stringsAsFactors = FALSE)

# Convert gene symbols to Entrez IDs
DNA_damage &lt;- DNA_damage %&gt;%
  mutate(Entrez_ID = mapIds(org.Hs.eg.db,
                            keys = Symbol,
                            column = &quot;ENTREZID&quot;,
                            keytype = &quot;SYMBOL&quot;,
                            multiVals = &quot;first&quot;))

DNA_damage_genes &lt;- na.omit(DNA_damage$Entrez_ID)

all_toptables &lt;- bind_rows(
  CX_0.1_3 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  CX_0.1_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  CX_0.1_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  CX_0.5_3 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  CX_0.5_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  CX_0.5_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;),
  DOX_0.1_3 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  DOX_0.1_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  DOX_0.1_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  DOX_0.5_3 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  DOX_0.5_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  DOX_0.5_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;)
)

filtered_toptables &lt;- all_toptables %&gt;%
  filter(Entrez_ID %in% DNA_damage_genes) %&gt;%
  mutate(abs_logFC = abs(logFC))

filtered_toptables &lt;- filtered_toptables %&gt;%
  mutate(
    Drug = factor(Drug, levels = c(&quot;CX.5461&quot;, &quot;DOX&quot;)),  # CX first
    Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;), 
                       labels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Concentration = factor(Concentration, levels = c(0.1, 0.5), 
                           labels = c(&quot;Concentration: 0.1&quot;, &quot;Concentration: 0.5&quot;))
  )

wilcox_results &lt;- filtered_toptables %&gt;%
  group_by(Timepoint, Concentration) %&gt;%
  wilcox_test(abs_logFC ~ Drug) %&gt;%
  adjust_pvalue(method = &quot;bonferroni&quot;) %&gt;%
  mutate(significance = ifelse(p &lt; 0.05, &quot;*&quot;, &quot;&quot;)) 

star_positions &lt;- filtered_toptables %&gt;%
  group_by(Timepoint, Concentration, Drug) %&gt;%
  summarise(y_pos = max(abs_logFC, na.rm = TRUE) + 0.2, .groups = &quot;drop&quot;) %&gt;%
  group_by(Timepoint, Concentration) %&gt;%
  summarise(y_pos = max(y_pos), .groups = &quot;drop&quot;)

wilcox_results_plot &lt;- wilcox_results %&gt;%
  left_join(star_positions, by = c(&quot;Timepoint&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(x_position = 1.5)  # Place stars in the middle between CX &amp; DOX

ggplot(filtered_toptables, aes(x = Drug, y = abs_logFC, fill = Drug)) +
  geom_boxplot() +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +  
  facet_grid(Concentration ~ Timepoint) +  
  geom_text(
    data = wilcox_results_plot, 
    aes(x = x_position, y = y_pos, label = significance), 
    size = 6, fontface = &quot;bold&quot;, color = &quot;black&quot;, inherit.aes = FALSE
  ) +
  theme_bw() +
  xlab(&quot;Drugs&quot;) +
  ylab(&quot;|Log Fold Change|&quot;) +  
  ggtitle(&quot;|Log Fold Change| for DNA Damage Repair Genes&quot;) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.line = element_line(linewidth = 1.5),
    strip.background = element_rect(fill = &quot;gray&quot;),  
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),  
    axis.text.x = element_text(size = 8, color = &quot;black&quot;, angle = 15)
  )</code></pre>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/abslogFC%20DNA%20damage-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-abslogFC-DNA-damage-1">
Past versions of "abslogFC DNA damage-1.png"
</button>
</p>
<div id="fig-no-spaces-abslogFC-DNA-damage-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/abslogFC DNA damage-1.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/a0b0fa3016aaba8e1f257c6afc54a4e696f92bdb/docs/figure/Cardiac_and_TOP2_Genes.Rmd/abslogFC DNA damage-1.png" target="_blank">a0b0fa3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dna-damage-repair-logfc-distribution" class="section level2">
<h2><strong>📌 DNA Damage Repair logFC distribution</strong></h2>
<pre class="r"><code># Load necessary libraries
library(dplyr)
library(ggplot2)
library(tidyr)
library(org.Hs.eg.db)
library(car)  # For Levene&#39;s Test</code></pre>
<pre><code>Warning: package &#39;car&#39; was built under R version 4.3.3</code></pre>
<pre><code>Warning: package &#39;carData&#39; was built under R version 4.3.1</code></pre>
<pre class="r"><code># Read DNA Damage Response Gene List
DNA_damage &lt;- read.csv(&quot;data/DNA_Damage.csv&quot;, stringsAsFactors = FALSE)

# Convert gene symbols to Entrez IDs
DNA_damage &lt;- DNA_damage %&gt;%
  mutate(Entrez_ID = mapIds(org.Hs.eg.db,
                            keys = Symbol,
                            column = &quot;ENTREZID&quot;,
                            keytype = &quot;SYMBOL&quot;,
                            multiVals = &quot;first&quot;))

DNA_damage_genes &lt;- na.omit(DNA_damage$Entrez_ID)

all_toptables &lt;- bind_rows(
  CX_0.1_3 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  CX_0.1_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  CX_0.1_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  CX_0.5_3 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  CX_0.5_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  CX_0.5_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;),
  DOX_0.1_3 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  DOX_0.1_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  DOX_0.1_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  DOX_0.5_3 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  DOX_0.5_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  DOX_0.5_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;)
)

filtered_toptables &lt;- all_toptables %&gt;%
  filter(Entrez_ID %in% DNA_damage_genes)

filtered_toptables &lt;- filtered_toptables %&gt;%
  mutate(
    Drug = factor(Drug, levels = c(&quot;CX.5461&quot;, &quot;DOX&quot;)),  # CX first
    Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;), 
                       labels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Concentration = factor(Concentration, levels = c(0.1, 0.5), 
                           labels = c(&quot;Concentration: 0.1&quot;, &quot;Concentration: 0.5&quot;))
  )

levene_results &lt;- filtered_toptables %&gt;%
  group_by(Timepoint, Concentration) %&gt;%
  summarise(p_value = leveneTest(logFC ~ Drug, data = .)$`Pr(&gt;F)`[1], .groups = &quot;drop&quot;) %&gt;%
  mutate(significance = ifelse(p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))  # Use p &lt; 0.05 threshold for stars

# **🔹 Determine Y-axis position for Stars**
star_positions &lt;- filtered_toptables %&gt;%
  group_by(Timepoint, Concentration, Drug) %&gt;%
  summarise(y_pos = max(logFC, na.rm = TRUE) + 0.5, .groups = &quot;drop&quot;) %&gt;%
  group_by(Timepoint, Concentration) %&gt;%
  summarise(y_pos = max(y_pos), .groups = &quot;drop&quot;)

# **🔹 Merge Levene test results with Y positions**
levene_results_plot &lt;- levene_results %&gt;%
  left_join(star_positions, by = c(&quot;Timepoint&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(x_position = 1.5)  # Centered between CX &amp; DOX

ggplot(filtered_toptables, aes(x = Drug, y = logFC, fill = Drug)) +
  geom_violin(trim = FALSE, alpha = 0.5) +  # Violin plot for logFC
  geom_boxplot(width = 0.1, outlier.shape = NA, color = &quot;black&quot;, alpha = 0.5) +  # Add boxplot inside violin
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +  
  facet_grid(Concentration ~ Timepoint) +  
  geom_text(
    data = levene_results_plot %&gt;% filter(significance == &quot;*&quot;),  # Only plot significant comparisons
    aes(x = x_position, y = y_pos, label = significance), 
    size = 6, fontface = &quot;bold&quot;, color = &quot;black&quot;, inherit.aes = FALSE
  ) +
  theme_bw() +
  xlab(&quot;Drugs&quot;) +
  ylab(&quot;Log Fold Change&quot;) +  
  ggtitle(&quot;Log Fold Change for DNA Damage Repair Genes&quot;) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.line = element_line(linewidth = 1.5),
    strip.background = element_rect(fill = &quot;gray&quot;),  
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),  
    axis.text.x = element_text(size = 8, color = &quot;black&quot;, angle = 15)
  )</code></pre>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/logFC%20DNA%20damage-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-logFC-DNA-damage-1">
Past versions of "logFC DNA damage-1.png"
</button>
</p>
<div id="fig-no-spaces-logFC-DNA-damage-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/logFC DNA damage-1.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/a0b0fa3016aaba8e1f257c6afc54a4e696f92bdb/docs/figure/Cardiac_and_TOP2_Genes.Rmd/logFC DNA damage-1.png" target="_blank">a0b0fa3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="generate-boxplots-for-dna-damage-repair-genes"
class="section level2">
<h2><strong>📌 Generate Boxplots for DNA Damage Repair
Genes</strong></h2>
<pre class="r"><code>for (gene in dna_repair_genes) {
  data_info &lt;- process_gene_data(gene)
  p &lt;- ggplot(data_info$long_data, aes(x = Condition, y = log2CPM, fill = Drug)) +
    geom_boxplot(outlier.shape = NA) +
    scale_fill_manual(values = c(&quot;CX&quot; = &quot;#0000FF&quot;, &quot;DOX&quot; = &quot;#e6d800&quot;, &quot;VEH&quot; = &quot;#FF00FF&quot;)) +
    geom_point(aes(color = Indv), size = 2, alpha = 0.5, position = position_jitter(width = 0.2, height = 0)) +
    geom_text(data = data_info$significance_labels, aes(x = Condition, y = max_log2CPM + 0.5, label = Significance),
              inherit.aes = FALSE, size = 6, color = &quot;black&quot;) +
    ggtitle(paste(&quot;Log2CPM Expression of&quot;, gene)) +
    labs(x = &quot;Treatment&quot;, y = &quot;log2CPM&quot;) +
    theme_bw() +
    theme(plot.title = element_text(size = rel(2), hjust = 0.5),
          axis.title = element_text(size = 15, color = &quot;black&quot;),
          axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 90, hjust = 1))
  
  print(p)
}</code></pre>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/DNA%20Damage%20Repair-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-DNA-Damage-Repair-1">
Past versions of "DNA Damage Repair-1.png"
</button>
</p>
<div id="fig-no-spaces-DNA-Damage-Repair-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/DNA Damage Repair-1.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/04b721c5960eeb144f48be02a96a77afd8d17dce/docs/figure/Cardiac_and_TOP2_Genes.Rmd/DNA Damage Repair-1.png" target="_blank">04b721c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/DNA%20Damage%20Repair-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-DNA-Damage-Repair-2">
Past versions of "DNA Damage Repair-2.png"
</button>
</p>
<div id="fig-no-spaces-DNA-Damage-Repair-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/DNA Damage Repair-2.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/04b721c5960eeb144f48be02a96a77afd8d17dce/docs/figure/Cardiac_and_TOP2_Genes.Rmd/DNA Damage Repair-2.png" target="_blank">04b721c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/DNA%20Damage%20Repair-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-DNA-Damage-Repair-3">
Past versions of "DNA Damage Repair-3.png"
</button>
</p>
<div id="fig-no-spaces-DNA-Damage-Repair-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/DNA Damage Repair-3.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/04b721c5960eeb144f48be02a96a77afd8d17dce/docs/figure/Cardiac_and_TOP2_Genes.Rmd/DNA Damage Repair-3.png" target="_blank">04b721c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/DNA%20Damage%20Repair-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-DNA-Damage-Repair-4">
Past versions of "DNA Damage Repair-4.png"
</button>
</p>
<div id="fig-no-spaces-DNA-Damage-Repair-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/DNA Damage Repair-4.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/b179eba8264934b9d3462c17e85fedf920f45e05/docs/figure/Cardiac_and_TOP2_Genes.Rmd/DNA Damage Repair-4.png" target="_blank">b179eba</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="p53-target-genes-proportion" class="section level2">
<h2><strong>📌 P53 target genes Proportion</strong></h2>
</div>
<div id="p53-target-genes-proportion-with-chi-square-test"
class="section level2">
<h2><strong>📌 P53 target genes Proportion with Chi-Square
Test</strong></h2>
<pre class="r"><code># Load necessary libraries
library(dplyr)
library(ggplot2)
library(tidyr)
library(org.Hs.eg.db)

# Read P53 Target Genes List
P53_Target &lt;- read.csv(&quot;data/P53_Target.csv&quot;, stringsAsFactors = FALSE)

# Convert gene symbols to Entrez IDs
P53_Target &lt;- P53_Target %&gt;%
  mutate(Entrez_ID = mapIds(org.Hs.eg.db,
                            keys = Symbol,
                            column = &quot;ENTREZID&quot;,
                            keytype = &quot;SYMBOL&quot;,
                            multiVals = &quot;first&quot;))

# Extract valid Entrez_IDs
P53_Target_genes &lt;- na.omit(P53_Target$Entrez_ID)
total_P53_Target_genes &lt;- length(P53_Target_genes)  # Total number of P53 target genes

# Function to calculate the presence of P53 target genes in DEGs
calculate_proportion &lt;- function(deg_list, drug_name) {
  data.frame(
    Sample = names(deg_list),
    Drug = drug_name,
    P53_Target_DEGs = sapply(deg_list, function(ids) sum(ids %in% P53_Target_genes)),  # DEGs present in P53 target set
    Non_P53_Target_DEGs = sapply(deg_list, function(ids) total_P53_Target_genes - sum(ids %in% P53_Target_genes))  # Remaining P53 target genes
  ) %&gt;%
    mutate(
      Yes_Proportion = (P53_Target_DEGs / total_P53_Target_genes) * 100,  # Percentage of DEGs in P53 target genes
      No_Proportion = (Non_P53_Target_DEGs / total_P53_Target_genes) * 100  # Remaining P53 target genes as No
    )
}

# Define CX-5461 and DOX DEG lists
CX_DEGs &lt;- list(
  &quot;CX_0.1_3&quot; = DEG1, &quot;CX_0.1_24&quot; = DEG2, &quot;CX_0.1_48&quot; = DEG3,
  &quot;CX_0.5_3&quot; = DEG4, &quot;CX_0.5_24&quot; = DEG5, &quot;CX_0.5_48&quot; = DEG6
)

DOX_DEGs &lt;- list(
  &quot;DOX_0.1_3&quot; = DEG7, &quot;DOX_0.1_24&quot; = DEG8, &quot;DOX_0.1_48&quot; = DEG9,
  &quot;DOX_0.5_3&quot; = DEG10, &quot;DOX_0.5_24&quot; = DEG11, &quot;DOX_0.5_48&quot; = DEG12
)

# Calculate proportions for CX-5461 and DOX
CX_proportion &lt;- calculate_proportion(CX_DEGs, &quot;CX-5461&quot;)
DOX_proportion &lt;- calculate_proportion(DOX_DEGs, &quot;DOX&quot;)

# Combine data
proportion_data &lt;- bind_rows(CX_proportion, DOX_proportion)

# Convert to long format for stacked bar plot
proportion_long &lt;- proportion_data %&gt;%
  select(Sample, Drug, Yes_Proportion, No_Proportion) %&gt;%
  pivot_longer(cols = c(Yes_Proportion, No_Proportion), names_to = &quot;Category&quot;, values_to = &quot;Percentage&quot;) %&gt;%
  mutate(Category = ifelse(Category == &quot;Yes_Proportion&quot;, &quot;Yes&quot;, &quot;No&quot;))

# **Ensure correct order of samples on X-axis**
sample_order &lt;- c(
  &quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;, &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
  &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;, &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;
)
proportion_long$Sample &lt;- factor(proportion_long$Sample, levels = sample_order, ordered = TRUE)

# **Ensure &quot;Yes&quot; is on top and &quot;No&quot; is at the bottom in stacked bars**
proportion_long$Category &lt;- factor(proportion_long$Category, levels = c(&quot;Yes&quot;, &quot;No&quot;))

# **Perform Chi-Square Test for CX vs DOX at each timepoint**
chi_square_results &lt;- data.frame(Sample = character(), P_Value = numeric())

for (i in seq(1, 6)) {  # Pairwise comparison (CX vs DOX)
  cx_sample &lt;- sample_order[i]
  dox_sample &lt;- sample_order[i + 6]  # Matches CX_0.1_3 with DOX_0.1_3, etc.
  
  cx_data &lt;- filter(proportion_data, Sample == cx_sample)
  dox_data &lt;- filter(proportion_data, Sample == dox_sample)
  
  # Construct contingency table for Chi-Square test
  contingency_table &lt;- matrix(
    c(cx_data$P53_Target_DEGs, cx_data$Non_P53_Target_DEGs,
      dox_data$P53_Target_DEGs, dox_data$Non_P53_Target_DEGs),
    nrow = 2, byrow = TRUE
  )
  
  # Run Chi-Square Test
  test_result &lt;- chisq.test(contingency_table)
  p_value &lt;- test_result$p.value
  
  # Store results
  chi_square_results &lt;- rbind(chi_square_results, data.frame(Sample = cx_sample, P_Value = p_value))
}

# Add significance stars
chi_square_results$Significant &lt;- ifelse(chi_square_results$P_Value &lt; 0.05, &quot;*&quot;, &quot;&quot;)

# Merge Chi-Square results
proportion_long &lt;- left_join(proportion_long, chi_square_results, by = &quot;Sample&quot;)

# **Save output**
write.csv(proportion_long, &quot;C:/Work/Postdoc_UTMB/CX-5461 Project/Transcriptome literatures/lit2/Proportion_Stacked_P53_Target_DEGs_with_ChiSquare.csv&quot;, row.names = FALSE)

# Define correct factor orders for samples
sample_order &lt;- c(
  &quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;, &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
  &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;, &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;
)

# Reapply factor levels for correct order in both proportion_data and proportion_long
proportion_data$Sample &lt;- factor(proportion_data$Sample, levels = sample_order, ordered = TRUE)
proportion_long$Sample &lt;- factor(proportion_long$Sample, levels = sample_order, ordered = TRUE)

# **Fix: Ensure &quot;Yes&quot; is on top and &quot;No&quot; is at the bottom in stacked bars**
proportion_long$Category &lt;- factor(proportion_long$Category, levels = c(&quot;Yes&quot;, &quot;No&quot;))


# **Generate Stacked Bar Plot with Correct X-Axis Order**
ggplot(proportion_long, aes(x = Sample, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +  # Stacked bars
  geom_text(data = subset(proportion_long, Significant == &quot;*&quot;),
            aes(x = Sample, y = 102, label = &quot;*&quot;),  # Position stars slightly above 100%
            size = 6, color = &quot;black&quot;, fontface = &quot;bold&quot;) +
  scale_y_continuous(labels = scales::percent_format(scale = 1), limits = c(0, 105)) +  # Increase Y-axis slightly
  scale_fill_manual(values = c(&quot;Yes&quot; = &quot;#e41a1c&quot;, &quot;No&quot; = &quot;#377eb8&quot;)) +  # Yes (Red), No (Blue)
  labs(
    title = &quot;Proportion of CX-5461 and DOX DEGs in\n P53 Target Genes with Significance&quot;,
    x = &quot;Samples (CX-5461 and DOX)&quot;,
    y = &quot;Percentage&quot;,
    fill = &quot;Category&quot;
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
    legend.title = element_blank(),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    strip.background = element_blank(),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/P53_target_Proportion1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-P53_target_Proportion1-1">
Past versions of P53_target_Proportion1-1.png
</button>
</p>
<div id="fig-P53_target_Proportion1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/230ea5c5af12358c25015da4bbf86e06c1205626/docs/figure/Cardiac_and_TOP2_Genes.Rmd/P53_target_Proportion1-1.png" target="_blank">230ea5c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-06-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/P53_target_Proportion1-1.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/d735a03cfeb5f79c1fc9bdc94f884fea2487e8f7/docs/figure/Cardiac_and_TOP2_Genes.Rmd/P53_target_Proportion1-1.png" target="_blank">d735a03</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/b899d4c2ddc969bf68d432895a791e48654e83c5/docs/figure/Cardiac_and_TOP2_Genes.Rmd/P53_target_Proportion1-1.png" target="_blank">b899d4c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="p53-target-genes-abs-logfc-distribution"
class="section level2">
<h2><strong>📌 P53 target genes abs logFC distribution</strong></h2>
<pre class="r"><code># Load necessary libraries
library(dplyr)
library(ggplot2)
library(tidyr)
library(org.Hs.eg.db)
library(rstatix)

# **🔹 Read P53 Target Genes List**
P53_Target &lt;- read.csv(&quot;data/P53_Target.csv&quot;, stringsAsFactors = FALSE)

# **🔹 Convert gene symbols to Entrez IDs**
P53_Target &lt;- P53_Target %&gt;%
  mutate(Entrez_ID = mapIds(org.Hs.eg.db,
                            keys = Symbol,
                            column = &quot;ENTREZID&quot;,
                            keytype = &quot;SYMBOL&quot;,
                            multiVals = &quot;first&quot;))

P53_Target_genes &lt;- na.omit(P53_Target$Entrez_ID)

# **🔹 Combine all subset_toptables into a single data frame with metadata**
all_toptables &lt;- bind_rows(
  CX_0.1_3 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  CX_0.1_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  CX_0.1_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  CX_0.5_3 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  CX_0.5_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  CX_0.5_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;),
  DOX_0.1_3 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  DOX_0.1_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  DOX_0.1_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  DOX_0.5_3 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  DOX_0.5_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  DOX_0.5_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;)
)

# **🔹 Filter data for P53 Target Genes**
filtered_toptables &lt;- all_toptables %&gt;%
  filter(Entrez_ID %in% P53_Target_genes) %&gt;%
  mutate(abs_logFC = abs(logFC))  # Compute absolute logFC

# **🔹 Ensure correct ordering of Timepoints &amp; Concentrations**
filtered_toptables &lt;- filtered_toptables %&gt;%
  mutate(
    Drug = factor(Drug, levels = c(&quot;CX.5461&quot;, &quot;DOX&quot;)),  # CX first
    Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;), 
                       labels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Concentration = factor(Concentration, levels = c(0.1, 0.5), 
                           labels = c(&quot;Concentration: 0.1&quot;, &quot;Concentration: 0.5&quot;))
  )

# **🔹 Wilcoxon Test for CX vs DOX**
wilcox_results &lt;- filtered_toptables %&gt;%
  group_by(Timepoint, Concentration) %&gt;%
  wilcox_test(abs_logFC ~ Drug) %&gt;%
  adjust_pvalue(method = &quot;bonferroni&quot;) %&gt;%
  mutate(significance = ifelse(p &lt; 0.05, &quot;*&quot;, &quot;&quot;))  # Assign stars if p &lt; 0.05

# **🔹 Determine Y-axis position for Stars**
star_positions &lt;- filtered_toptables %&gt;%
  group_by(Timepoint, Concentration, Drug) %&gt;%
  summarise(y_pos = max(abs_logFC, na.rm = TRUE) + 0.2, .groups = &quot;drop&quot;) %&gt;%
  group_by(Timepoint, Concentration) %&gt;%
  summarise(y_pos = max(y_pos), .groups = &quot;drop&quot;)

# **🔹 Merge Wilcoxon Test results with Star Positions**
wilcox_results_plot &lt;- wilcox_results %&gt;%
  left_join(star_positions, by = c(&quot;Timepoint&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(x_position = 1.5)  # Place stars in the middle between CX &amp; DOX

# **🔹 Generate Boxplot for LogFC of P53 Target Genes**
ggplot(filtered_toptables, aes(x = Drug, y = abs_logFC, fill = Drug)) +
  geom_boxplot() +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +  
  facet_grid(Concentration ~ Timepoint) +  
  geom_text(
    data = wilcox_results_plot, 
    aes(x = x_position, y = y_pos, label = significance), 
    size = 6, fontface = &quot;bold&quot;, color = &quot;black&quot;, inherit.aes = FALSE
  ) +
  theme_bw() +
  xlab(&quot;Drugs&quot;) +
  ylab(&quot;|Log Fold Change|&quot;) +  
  ggtitle(&quot;|Log Fold Change| for P53 Target Genes&quot;) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.line = element_line(linewidth = 1.5),
    strip.background = element_rect(fill = &quot;gray&quot;),  
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),  
    axis.text.x = element_text(size = 8, color = &quot;black&quot;, angle = 15)
  )</code></pre>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/abslogFC%20P53-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-abslogFC-P53-1">
Past versions of "abslogFC P53-1.png"
</button>
</p>
<div id="fig-no-spaces-abslogFC-P53-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/abslogFC P53-1.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/a0b0fa3016aaba8e1f257c6afc54a4e696f92bdb/docs/figure/Cardiac_and_TOP2_Genes.Rmd/abslogFC P53-1.png" target="_blank">a0b0fa3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="p53-target-genes-logfc-distribution" class="section level2">
<h2><strong>📌 P53 Target Genes logFC distribution</strong></h2>
<pre class="r"><code># Load necessary libraries
library(dplyr)
library(ggplot2)
library(tidyr)
library(org.Hs.eg.db)
library(car)  # For Levene&#39;s Test

# Read P53 Target Genes List
P53_Target &lt;- read.csv(&quot;data/P53_Target.csv&quot;, stringsAsFactors = FALSE)

# Convert gene symbols to Entrez IDs
P53_Target &lt;- P53_Target %&gt;%
  mutate(Entrez_ID = mapIds(org.Hs.eg.db,
                            keys = Symbol,
                            column = &quot;ENTREZID&quot;,
                            keytype = &quot;SYMBOL&quot;,
                            multiVals = &quot;first&quot;))

P53_Target_genes &lt;- na.omit(P53_Target$Entrez_ID)

# **🔹 Combine all subset_toptables into a single data frame with metadata**
all_toptables &lt;- bind_rows(
  CX_0.1_3 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  CX_0.1_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  CX_0.1_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  CX_0.5_3 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  CX_0.5_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  CX_0.5_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;),
  DOX_0.1_3 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  DOX_0.1_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  DOX_0.1_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  DOX_0.5_3 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  DOX_0.5_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  DOX_0.5_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;)
)

# **🔹 Filter data for P53 target genes**
filtered_toptables &lt;- all_toptables %&gt;%
  filter(Entrez_ID %in% P53_Target_genes)

# **🔹 Factorize variables for ordered plotting**
filtered_toptables &lt;- filtered_toptables %&gt;%
  mutate(
    Drug = factor(Drug, levels = c(&quot;CX.5461&quot;, &quot;DOX&quot;)),  # CX first
    Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;), 
                       labels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Concentration = factor(Concentration, levels = c(0.1, 0.5), 
                           labels = c(&quot;Concentration: 0.1&quot;, &quot;Concentration: 0.5&quot;))
  )

# **🔹 Perform Levene&#39;s test for variability differences**
levene_results &lt;- filtered_toptables %&gt;%
  group_by(Timepoint, Concentration) %&gt;%
  summarise(p_value = leveneTest(logFC ~ Drug, data = .)$`Pr(&gt;F)`[1], .groups = &quot;drop&quot;) %&gt;%
  mutate(significance = ifelse(p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))  # Use p &lt; 0.05 threshold for stars

# **🔹 Determine Y-axis position for Stars**
star_positions &lt;- filtered_toptables %&gt;%
  group_by(Timepoint, Concentration, Drug) %&gt;%
  summarise(y_pos = max(logFC, na.rm = TRUE) + 0.5, .groups = &quot;drop&quot;) %&gt;%
  group_by(Timepoint, Concentration) %&gt;%
  summarise(y_pos = max(y_pos), .groups = &quot;drop&quot;)

# **🔹 Merge Levene test results with Y positions**
levene_results_plot &lt;- levene_results %&gt;%
  left_join(star_positions, by = c(&quot;Timepoint&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(x_position = 1.5)  # Centered between CX &amp; DOX

# **🔹 Generate Violin Plot with Boxplot &amp; Significance Stars**
ggplot(filtered_toptables, aes(x = Drug, y = logFC, fill = Drug)) +
  geom_violin(trim = FALSE, alpha = 0.5) +  # Violin plot for logFC
  geom_boxplot(width = 0.1, outlier.shape = NA, color = &quot;black&quot;, alpha = 0.5) +  # Add boxplot inside violin
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +  
  facet_grid(Concentration ~ Timepoint) +  
  geom_text(
    data = levene_results_plot %&gt;% filter(significance == &quot;*&quot;),  # Only plot significant comparisons
    aes(x = x_position, y = y_pos, label = significance), 
    size = 6, fontface = &quot;bold&quot;, color = &quot;black&quot;, inherit.aes = FALSE
  ) +
  theme_bw() +
  xlab(&quot;Drugs&quot;) +
  ylab(&quot;Log Fold Change&quot;) +  
  ggtitle(&quot;Log Fold Change for P53 Target Genes&quot;) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.line = element_line(linewidth = 1.5),
    strip.background = element_rect(fill = &quot;gray&quot;),  
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),  
    axis.text.x = element_text(size = 8, color = &quot;black&quot;, angle = 15)
  )</code></pre>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/logFC%20P53-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-logFC-P53-1">
Past versions of "logFC P53-1.png"
</button>
</p>
<div id="fig-no-spaces-logFC-P53-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/logFC P53-1.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/a0b0fa3016aaba8e1f257c6afc54a4e696f92bdb/docs/figure/Cardiac_and_TOP2_Genes.Rmd/logFC P53-1.png" target="_blank">a0b0fa3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="generate-boxplots-for-p53-target-genes" class="section level2">
<h2><strong>📌 Generate Boxplots for P53 Target Genes</strong></h2>
<pre class="r"><code>for (gene in p53_target_genes) {
  data_info &lt;- process_gene_data(gene)
  p &lt;- ggplot(data_info$long_data, aes(x = Condition, y = log2CPM, fill = Drug)) +
    geom_boxplot(outlier.shape = NA) +
    scale_fill_manual(values = c(&quot;CX&quot; = &quot;#0000FF&quot;, &quot;DOX&quot; = &quot;#e6d800&quot;, &quot;VEH&quot; = &quot;#FF00FF&quot;)) +
    geom_point(aes(color = Indv), size = 2, alpha = 0.5, position = position_jitter(width = 0.2, height = 0)) +
    geom_text(data = data_info$significance_labels, aes(x = Condition, y = max_log2CPM + 0.5, label = Significance),
              inherit.aes = FALSE, size = 6, color = &quot;black&quot;) +
    ggtitle(paste(&quot;Log2CPM Expression of&quot;, gene)) +
    labs(x = &quot;Treatment&quot;, y = &quot;log2CPM&quot;) +
    theme_bw() +
    theme(plot.title = element_text(size = rel(2), hjust = 0.5),
          axis.title = element_text(size = 15, color = &quot;black&quot;),
          axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 90, hjust = 1))
  
  print(p)
}</code></pre>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/P53%20Target%20Genes-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-P53-Target-Genes-1">
Past versions of "P53 Target Genes-1.png"
</button>
</p>
<div id="fig-no-spaces-P53-Target-Genes-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/P53 Target Genes-1.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/b179eba8264934b9d3462c17e85fedf920f45e05/docs/figure/Cardiac_and_TOP2_Genes.Rmd/P53 Target Genes-1.png" target="_blank">b179eba</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/04b721c5960eeb144f48be02a96a77afd8d17dce/docs/figure/Cardiac_and_TOP2_Genes.Rmd/P53 Target Genes-1.png" target="_blank">04b721c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/P53%20Target%20Genes-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-P53-Target-Genes-2">
Past versions of "P53 Target Genes-2.png"
</button>
</p>
<div id="fig-no-spaces-P53-Target-Genes-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/P53 Target Genes-2.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/b179eba8264934b9d3462c17e85fedf920f45e05/docs/figure/Cardiac_and_TOP2_Genes.Rmd/P53 Target Genes-2.png" target="_blank">b179eba</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/04b721c5960eeb144f48be02a96a77afd8d17dce/docs/figure/Cardiac_and_TOP2_Genes.Rmd/P53 Target Genes-2.png" target="_blank">04b721c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Cardiac_and_TOP2_Genes.Rmd/P53%20Target%20Genes-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-P53-Target-Genes-3">
Past versions of "P53 Target Genes-3.png"
</button>
</p>
<div id="fig-no-spaces-P53-Target-Genes-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c8ef284a979f76910a228ec8df6e627c4a6a42dd/docs/figure/Cardiac_and_TOP2_Genes.Rmd/P53 Target Genes-3.png" target="_blank">c8ef284</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/b179eba8264934b9d3462c17e85fedf920f45e05/docs/figure/Cardiac_and_TOP2_Genes.Rmd/P53 Target Genes-3.png" target="_blank">b179eba</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/04b721c5960eeb144f48be02a96a77afd8d17dce/docs/figure/Cardiac_and_TOP2_Genes.Rmd/P53 Target Genes-3.png" target="_blank">04b721c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 26100)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] car_3.1-3              carData_3.0-5          rstatix_0.7.2         
 [4] clusterProfiler_4.10.1 org.Hs.eg.db_3.18.0    AnnotationDbi_1.64.1  
 [7] IRanges_2.36.0         S4Vectors_0.40.2       Biobase_2.62.0        
[10] BiocGenerics_0.48.1    tidyr_1.3.1            dplyr_1.1.4           
[13] ggplot2_3.5.2          workflowr_1.7.1       

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3      rstudioapi_0.17.1       jsonlite_2.0.0         
  [4] magrittr_2.0.3          farver_2.1.2            rmarkdown_2.29         
  [7] fs_1.6.3                zlibbioc_1.48.2         vctrs_0.6.5            
 [10] memoise_2.0.1           RCurl_1.98-1.17         ggtree_3.10.1          
 [13] htmltools_0.5.8.1       broom_1.0.8             Formula_1.2-5          
 [16] gridGraphics_0.5-1      sass_0.4.10             bslib_0.9.0            
 [19] plyr_1.8.9              cachem_1.1.0            whisker_0.4.1          
 [22] igraph_2.1.4            lifecycle_1.0.4         pkgconfig_2.0.3        
 [25] Matrix_1.6-1.1          R6_2.6.1                fastmap_1.2.0          
 [28] gson_0.1.0              GenomeInfoDbData_1.2.11 digest_0.6.34          
 [31] aplot_0.2.5             enrichplot_1.22.0       colorspace_2.1-0       
 [34] patchwork_1.3.0         ps_1.8.1                rprojroot_2.0.4        
 [37] RSQLite_2.3.9           labeling_0.4.3          abind_1.4-8            
 [40] httr_1.4.7              polyclip_1.10-7         compiler_4.3.0         
 [43] bit64_4.6.0-1           withr_3.0.2             backports_1.5.0        
 [46] BiocParallel_1.36.0     viridis_0.6.5           DBI_1.2.3              
 [49] ggforce_0.4.2           MASS_7.3-60             HDO.db_0.99.1          
 [52] tools_4.3.0             ape_5.8-1               scatterpie_0.2.4       
 [55] httpuv_1.6.15           glue_1.7.0              callr_3.7.6            
 [58] nlme_3.1-168            GOSemSim_2.28.1         promises_1.3.2         
 [61] grid_4.3.0              shadowtext_0.1.4        getPass_0.2-4          
 [64] reshape2_1.4.4          fgsea_1.28.0            generics_0.1.3         
 [67] gtable_0.3.6            data.table_1.17.0       tidygraph_1.3.1        
 [70] XVector_0.42.0          ggrepel_0.9.6           pillar_1.10.2          
 [73] stringr_1.5.1           yulab.utils_0.2.0       later_1.3.2            
 [76] splines_4.3.0           tweenr_2.0.3            treeio_1.26.0          
 [79] lattice_0.22-7          bit_4.6.0               tidyselect_1.2.1       
 [82] GO.db_3.18.0            Biostrings_2.70.3       knitr_1.50             
 [85] git2r_0.36.2            gridExtra_2.3           xfun_0.52              
 [88] graphlayouts_1.2.2      stringi_1.8.3           lazyeval_0.2.2         
 [91] ggfun_0.1.8             yaml_2.3.10             evaluate_1.0.3         
 [94] codetools_0.2-20        ggraph_2.2.1            tibble_3.2.1           
 [97] qvalue_2.34.0           ggplotify_0.1.2         cli_3.6.1              
[100] munsell_0.5.1           processx_3.8.6          jquerylib_0.1.4        
[103] Rcpp_1.0.12             GenomeInfoDb_1.38.8     png_0.1-8              
[106] parallel_4.3.0          blob_1.2.4              DOSE_3.28.2            
[109] bitops_1.0-9            viridisLite_0.4.2       tidytree_0.4.6         
[112] scales_1.3.0            purrr_1.0.4             crayon_1.5.3           
[115] rlang_1.1.3             cowplot_1.1.3           fastmatch_1.1-6        
[118] KEGGREST_1.42.0        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
