<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Corrmotif Conc Analysis</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CX5461_Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Corrmotif Conc Analysis</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-03-09
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>CX5461_Project/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttreebc36cac84264919e8a1a9242f63fff6d6be0bc54targetblankbc36caca">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/bc36cac84264919e8a1a9242f63fff6d6be0bc54" target="_blank">bc36cac</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttreebc36cac84264919e8a1a9242f63fff6d6be0bc54targetblankbc36caca"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/bc36cac84264919e8a1a9242f63fff6d6be0bc54" target="_blank">bc36cac</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Unstaged changes:
    Modified:   analysis/Corrmotif_Conc.Rmd

Staged changes:
    Modified:   analysis/Colorectal.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Corrmotif_Conc.Rmd</code>) and
HTML (<code>docs/Corrmotif_Conc.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/81247f64f08df43c2b40a386707fc8e9cc240143/analysis/Corrmotif_Conc.Rmd" target="_blank">81247f6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-06
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/81247f64f08df43c2b40a386707fc8e9cc240143/docs/Corrmotif_Conc.html" target="_blank">81247f6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-06
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/96d0db145cf68e6db0d39e70bad9af0f75793aae/analysis/Corrmotif_Conc.Rmd" target="_blank">96d0db1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/96d0db145cf68e6db0d39e70bad9af0f75793aae/docs/Corrmotif_Conc.html" target="_blank">96d0db1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/analysis/Corrmotif_Conc.Rmd" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/Corrmotif_Conc.html" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/analysis/Corrmotif_Conc.Rmd" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/Corrmotif_Conc.html" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2de67e6a82f82edbc78941b11ba72ed8aba9964e/analysis/Corrmotif_Conc.Rmd" target="_blank">2de67e6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/2de67e6a82f82edbc78941b11ba72ed8aba9964e/docs/Corrmotif_Conc.html" target="_blank">2de67e6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/41cd1be487c4309fb8ab2f2d36b18ad63cdc31e4/analysis/Corrmotif_Conc.Rmd" target="_blank">41cd1be</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/41cd1be487c4309fb8ab2f2d36b18ad63cdc31e4/docs/Corrmotif_Conc.html" target="_blank">41cd1be</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/f84821f0421d3805726f2ccfce5b5ca2275c6756/analysis/Corrmotif_Conc.Rmd" target="_blank">f84821f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-26
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/b6e38a14f7f28ac96f0b16857be1f89d71989e58/analysis/Corrmotif_Conc.Rmd" target="_blank">b6e38a1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-25
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/b6e38a14f7f28ac96f0b16857be1f89d71989e58/docs/Corrmotif_Conc.html" target="_blank">b6e38a1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-25
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ce4b32557fc3c7ee693369db9edf0a9d22968203/analysis/Corrmotif_Conc.Rmd" target="_blank">ce4b325</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-25
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/ce4b32557fc3c7ee693369db9edf0a9d22968203/docs/Corrmotif_Conc.html" target="_blank">ce4b325</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-25
</td>
<td>
Commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="micromolar" class="section level2">
<h2><strong>📌 0.1 Micromolar</strong></h2>
</div>
<div id="fit-limma-model-functions" class="section level2">
<h2><strong>📌 Fit Limma Model Functions</strong></h2>
<pre class="r"><code>## Fit limma model using code as it is found in the original cormotif code. It has
## only been modified to add names to the matrix of t values, as well as the
## limma fits

limmafit.default &lt;- function(exprs,groupid,compid) {
  limmafits  &lt;- list()
  compnum    &lt;- nrow(compid)
  genenum    &lt;- nrow(exprs)
  limmat     &lt;- matrix(0,genenum,compnum)
  limmas2    &lt;- rep(0,compnum)
  limmadf    &lt;- rep(0,compnum)
  limmav0    &lt;- rep(0,compnum)
  limmag1num &lt;- rep(0,compnum)
  limmag2num &lt;- rep(0,compnum)

  rownames(limmat)  &lt;- rownames(exprs)
  colnames(limmat)  &lt;- rownames(compid)
  names(limmas2)    &lt;- rownames(compid)
  names(limmadf)    &lt;- rownames(compid)
  names(limmav0)    &lt;- rownames(compid)
  names(limmag1num) &lt;- rownames(compid)
  names(limmag2num) &lt;- rownames(compid)

  for(i in 1:compnum) {
    selid1 &lt;- which(groupid == compid[i,1])
    selid2 &lt;- which(groupid == compid[i,2])
    eset   &lt;- new(&quot;ExpressionSet&quot;, exprs=cbind(exprs[,selid1],exprs[,selid2]))
    g1num  &lt;- length(selid1)
    g2num  &lt;- length(selid2)
    designmat &lt;- cbind(base=rep(1,(g1num+g2num)), delta=c(rep(0,g1num),rep(1,g2num)))
    fit &lt;- lmFit(eset,designmat)
    fit &lt;- eBayes(fit)
    limmat[,i] &lt;- fit$t[,2]
    limmas2[i] &lt;- fit$s2.prior
    limmadf[i] &lt;- fit$df.prior
    limmav0[i] &lt;- fit$var.prior[2]
    limmag1num[i] &lt;- g1num
    limmag2num[i] &lt;- g2num
    limmafits[[i]] &lt;- fit

    # log odds
    # w&lt;-sqrt(1+fit$var.prior[2]/(1/g1num+1/g2num))
    # log(0.99)+dt(fit$t[1,2],g1num+g2num-2+fit$df.prior,log=TRUE)-log(0.01)-dt(fit$t[1,2]/w, g1num+g2num-2+fit$df.prior, log=TRUE)+log(w)
  }
  names(limmafits) &lt;- rownames(compid)
  limmacompnum&lt;-nrow(compid)
  result&lt;-list(t       = limmat,
               v0      = limmav0,
               df0     = limmadf,
               s20     = limmas2,
               g1num   = limmag1num,
               g2num   = limmag2num,
               compnum = limmacompnum,
               fits    = limmafits)
}

limmafit.counts &lt;-
  function (exprs, groupid, compid, norm.factor.method = &quot;TMM&quot;, voom.normalize.method = &quot;none&quot;)
  {
    limmafits  &lt;- list()
    compnum    &lt;- nrow(compid)
    genenum    &lt;- nrow(exprs)
    limmat     &lt;- matrix(NA,genenum,compnum)
    limmas2    &lt;- rep(0,compnum)
    limmadf    &lt;- rep(0,compnum)
    limmav0    &lt;- rep(0,compnum)
    limmag1num &lt;- rep(0,compnum)
    limmag2num &lt;- rep(0,compnum)

    rownames(limmat)  &lt;- rownames(exprs)
    colnames(limmat)  &lt;- rownames(compid)
    names(limmas2)    &lt;- rownames(compid)
    names(limmadf)    &lt;- rownames(compid)
    names(limmav0)    &lt;- rownames(compid)
    names(limmag1num) &lt;- rownames(compid)
    names(limmag2num) &lt;- rownames(compid)

    for (i in 1:compnum) {
      message(paste(&quot;Running limma for comparision&quot;,i,&quot;/&quot;,compnum))
      selid1 &lt;- which(groupid == compid[i, 1])
      selid2 &lt;- which(groupid == compid[i, 2])
      # make a new count data frame
      counts &lt;- cbind(exprs[, selid1], exprs[, selid2])

      # remove NAs
      not.nas &lt;- which(apply(counts, 1, function(x) !any(is.na(x))) == TRUE)

      # runn voom/limma
      d &lt;- DGEList(counts[not.nas,])
      d &lt;- calcNormFactors(d, method = norm.factor.method)
      g1num &lt;- length(selid1)
      g2num &lt;- length(selid2)
      designmat &lt;- cbind(base = rep(1, (g1num + g2num)), delta = c(rep(0,
                                                                       g1num), rep(1, g2num)))

      y &lt;- voom(d, designmat, normalize.method = voom.normalize.method)
      fit &lt;- lmFit(y, designmat)
      fit &lt;- eBayes(fit)

      limmafits[[i]] &lt;- fit
      limmat[not.nas, i] &lt;- fit$t[, 2]
      limmas2[i] &lt;- fit$s2.prior
      limmadf[i] &lt;- fit$df.prior
      limmav0[i] &lt;- fit$var.prior[2]
      limmag1num[i] &lt;- g1num
      limmag2num[i] &lt;- g2num
    }
    limmacompnum &lt;- nrow(compid)
    names(limmafits) &lt;- rownames(compid)
    result &lt;- list(t       = limmat,
                   v0      = limmav0,
                   df0     = limmadf,
                   s20     = limmas2,
                   g1num   = limmag1num,
                   g2num   = limmag2num,
                   compnum = limmacompnum,
                   fits    = limmafits)
  }

limmafit.list &lt;-
  function (fitlist, cmp.idx=2)
  {
    compnum    &lt;- length(fitlist)

    genes &lt;- c()
    for (i in 1:compnum) genes &lt;- unique(c(genes, rownames(fitlist[[i]])))

    genenum    &lt;- length(genes)
    limmat     &lt;- matrix(NA,genenum,compnum)
    limmas2    &lt;- rep(0,compnum)
    limmadf    &lt;- rep(0,compnum)
    limmav0    &lt;- rep(0,compnum)
    limmag1num &lt;- rep(0,compnum)
    limmag2num &lt;- rep(0,compnum)

    rownames(limmat)  &lt;- genes
    colnames(limmat)  &lt;- names(fitlist)
    names(limmas2)    &lt;- names(fitlist)
    names(limmadf)    &lt;- names(fitlist)
    names(limmav0)    &lt;- names(fitlist)
    names(limmag1num) &lt;- names(fitlist)
    names(limmag2num) &lt;- names(fitlist)

    for (i in 1:compnum) {
      this.t &lt;- fitlist[[i]]$t[,cmp.idx]
      limmat[names(this.t),i] &lt;- this.t

      limmas2[i]    &lt;- fitlist[[i]]$s2.prior
      limmadf[i]    &lt;- fitlist[[i]]$df.prior
      limmav0[i]    &lt;- fitlist[[i]]$var.prior[cmp.idx]
      limmag1num[i] &lt;- sum(fitlist[[i]]$design[,cmp.idx]==0)
      limmag2num[i] &lt;- sum(fitlist[[i]]$design[,cmp.idx]==1)
    }

    limmacompnum &lt;- compnum
    result &lt;- list(t       = limmat,
                   v0      = limmav0,
                   df0     = limmadf,
                   s20     = limmas2,
                   g1num   = limmag1num,
                   g2num   = limmag2num,
                   compnum = limmacompnum,
                   fits    = limmafits)

  }

## Rank genes based on statistics
generank&lt;-function(x) {
  xcol&lt;-ncol(x)
  xrow&lt;-nrow(x)
  result&lt;-matrix(0,xrow,xcol)
  z&lt;-(1:1:xrow)
  for(i in 1:xcol) {
    y&lt;-sort(x[,i],decreasing=TRUE,na.last=TRUE)
    result[,i]&lt;-match(x[,i],y)
    result[,i]&lt;-order(result[,i])
  }
  result
}

## Log-likelihood for moderated t under H0
modt.f0.loglike&lt;-function(x,df) {
  a&lt;-dt(x, df, log=TRUE)
  result&lt;-as.vector(a)
  flag&lt;-which(is.na(result)==TRUE)
  result[flag]&lt;-0
  result
}

## Log-likelihood for moderated t under H1
## param=c(df,g1num,g2num,v0)
modt.f1.loglike&lt;-function(x,param) {
  df&lt;-param[1]
  g1num&lt;-param[2]
  g2num&lt;-param[3]
  v0&lt;-param[4]
  w&lt;-sqrt(1+v0/(1/g1num+1/g2num))
  dt(x/w, df, log=TRUE)-log(w)
  a&lt;-dt(x/w, df, log=TRUE)-log(w)
  result&lt;-as.vector(a)
  flag&lt;-which(is.na(result)==TRUE)
  result[flag]&lt;-0
  result
}

## Correlation Motif Fit
cmfit.X&lt;-function(x, type, K=1, tol=1e-3, max.iter=100) {
  ## initialize
  xrow &lt;- nrow(x)
  xcol &lt;- ncol(x)
  loglike0 &lt;- list()
  loglike1 &lt;- list()
  p &lt;- rep(1, K)/K
  q &lt;- matrix(runif(K * xcol), K, xcol)
  q[1, ] &lt;- rep(0.01, xcol)
  for (i in 1:xcol) {
    f0 &lt;- type[[i]][[1]]
    f0param &lt;- type[[i]][[2]]
    f1 &lt;- type[[i]][[3]]
    f1param &lt;- type[[i]][[4]]
    loglike0[[i]] &lt;- f0(x[, i], f0param)
    loglike1[[i]] &lt;- f1(x[, i], f1param)
  }
  condlike &lt;- list()
  for (i in 1:xcol) {
    condlike[[i]] &lt;- matrix(0, xrow, K)
  }
  loglike.old &lt;- -1e+10
  for (i.iter in 1:max.iter) {
    if ((i.iter%%50) == 0) {
      print(paste(&quot;We have run the first &quot;, i.iter, &quot; iterations for K=&quot;,
                  K, sep = &quot;&quot;))
    }
    err &lt;- tol + 1
    clustlike &lt;- matrix(0, xrow, K)
    #templike &lt;- matrix(0, xrow, 2)
    templike1 &lt;- rep(0, xrow)
    templike2 &lt;- rep(0, xrow)
    for (j in 1:K) {
      for (i in 1:xcol) {
        templike1 &lt;- log(q[j, i]) + loglike1[[i]]
        templike2 &lt;- log(1 - q[j, i]) + loglike0[[i]]
        tempmax &lt;- Rfast::Pmax(templike1, templike2)

        templike1 &lt;- exp(templike1 - tempmax)
        templike2 &lt;- exp(templike2 - tempmax)

        tempsum &lt;- templike1 + templike2
        clustlike[, j] &lt;- clustlike[, j] + tempmax +
          log(tempsum)
        condlike[[i]][, j] &lt;- templike1/tempsum
      }
      clustlike[, j] &lt;- clustlike[, j] + log(p[j])
    }
    #tempmax &lt;- apply(clustlike, 1, max)
    tempmax &lt;- Rfast::rowMaxs(clustlike, value=TRUE)
    for (j in 1:K) {
      clustlike[, j] &lt;- exp(clustlike[, j] - tempmax)
    }
    #tempsum &lt;- apply(clustlike, 1, sum)
    tempsum &lt;- Rfast::rowsums(clustlike)
    for (j in 1:K) {
      clustlike[, j] &lt;- clustlike[, j]/tempsum
    }
    #p.new &lt;- (apply(clustlike, 2, sum) + 1)/(xrow + K)
    p.new &lt;- (Rfast::colsums(clustlike) + 1)/(xrow + K)
    q.new &lt;- matrix(0, K, xcol)
    for (j in 1:K) {
      clustpsum &lt;- sum(clustlike[, j])
      for (i in 1:xcol) {
        q.new[j, i] &lt;- (sum(clustlike[, j] * condlike[[i]][,
                                                           j]) + 1)/(clustpsum + 2)
      }
    }
    err.p &lt;- max(abs(p.new - p)/p)
    err.q &lt;- max(abs(q.new - q)/q)
    err &lt;- max(err.p, err.q)
    loglike.new &lt;- (sum(tempmax + log(tempsum)) + sum(log(p.new)) +
                      sum(log(q.new) + log(1 - q.new)))/xrow
    p &lt;- p.new
    q &lt;- q.new
    loglike.old &lt;- loglike.new
    if (err &lt; tol) {
      break
    }
  }
  clustlike &lt;- matrix(0, xrow, K)
  for (j in 1:K) {
    for (i in 1:xcol) {
      templike1 &lt;- log(q[j, i]) + loglike1[[i]]
      templike2 &lt;- log(1 - q[j, i]) + loglike0[[i]]
      tempmax &lt;- Rfast::Pmax(templike1, templike2)

      templike1 &lt;- exp(templike1 - tempmax)
      templike2 &lt;- exp(templike2 - tempmax)

      tempsum &lt;- templike1 + templike2
      clustlike[, j] &lt;- clustlike[, j] + tempmax + log(tempsum)
      condlike[[i]][, j] &lt;- templike1/tempsum
    }
    clustlike[, j] &lt;- clustlike[, j] + log(p[j])
  }
  #tempmax &lt;- apply(clustlike, 1, max)
  tempmax &lt;- Rfast::rowMaxs(clustlike, value=TRUE)
  for (j in 1:K) {
    clustlike[, j] &lt;- exp(clustlike[, j] - tempmax)
  }
  #tempsum &lt;- apply(clustlike, 1, sum)
  tempsum &lt;- Rfast::rowsums(clustlike)
  for (j in 1:K) {
    clustlike[, j] &lt;- clustlike[, j]/tempsum
  }
  p.post &lt;- matrix(0, xrow, xcol)
  for (j in 1:K) {
    for (i in 1:xcol) {
      p.post[, i] &lt;- p.post[, i] + clustlike[, j] * condlike[[i]][,
                                                                  j]
    }
  }
  loglike.old &lt;- loglike.old - (sum(log(p)) + sum(log(q) +
                                                    log(1 - q)))/xrow
  loglike.old &lt;- loglike.old * xrow
  result &lt;- list(p.post = p.post, motif.prior = p, motif.q = q,
                 loglike = loglike.old, clustlike=clustlike, condlike=condlike)
}

## Fit using (0,0,...,0) and (1,1,...,1)
cmfitall&lt;-function(x, type, tol=1e-3, max.iter=100) {
  ## initialize
  xrow&lt;-nrow(x)
  xcol&lt;-ncol(x)
  loglike0&lt;-list()
  loglike1&lt;-list()
  p&lt;-0.01

  ## compute loglikelihood
  L0&lt;-matrix(0,xrow,1)
  L1&lt;-matrix(0,xrow,1)
  for(i in 1:xcol) {
    f0&lt;-type[[i]][[1]]
    f0param&lt;-type[[i]][[2]]
    f1&lt;-type[[i]][[3]]
    f1param&lt;-type[[i]][[4]]
    loglike0[[i]]&lt;-f0(x[,i],f0param)
    loglike1[[i]]&lt;-f1(x[,i],f1param)
    L0&lt;-L0+loglike0[[i]]
    L1&lt;-L1+loglike1[[i]]
  }


  ## EM algorithm to get MLE of p and q
  loglike.old &lt;- -1e10
  for(i.iter in 1:max.iter) {
    if((i.iter%%50) == 0) {
      print(paste(&quot;We have run the first &quot;, i.iter, &quot; iterations&quot;,sep=&quot;&quot;))
    }
    err&lt;-tol+1

    ## compute posterior cluster membership
    clustlike&lt;-matrix(0,xrow,2)
    clustlike[,1]&lt;-log(1-p)+L0
    clustlike[,2]&lt;-log(p)+L1

    tempmax&lt;-apply(clustlike,1,max)
    for(j in 1:2) {
      clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
    }
    tempsum&lt;-apply(clustlike,1,sum)

    ## update motif occurrence rate
    for(j in 1:2) {
      clustlike[,j]&lt;-clustlike[,j]/tempsum
    }

    p.new&lt;-(sum(clustlike[,2])+1)/(xrow+2)

    ## evaluate convergence
    err&lt;-abs(p.new-p)/p

    ## evaluate whether the log.likelihood increases
    loglike.new&lt;-(sum(tempmax+log(tempsum))+log(p.new)+log(1-p.new))/xrow

    loglike.old&lt;-loglike.new
    p&lt;-p.new

    if(err&lt;tol) {
      break;
    }
  }

  ## compute posterior p
  clustlike&lt;-matrix(0,xrow,2)
  clustlike[,1]&lt;-log(1-p)+L0
  clustlike[,2]&lt;-log(p)+L1

  tempmax&lt;-apply(clustlike,1,max)
  for(j in 1:2) {
    clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
  }
  tempsum&lt;-apply(clustlike,1,sum)

  for(j in 1:2) {
    clustlike[,j]&lt;-clustlike[,j]/tempsum
  }

  p.post&lt;-matrix(0,xrow,xcol)
  for(i in 1:xcol) {
    p.post[,i]&lt;-clustlike[,2]
  }

  ## return

  #calculate back loglikelihood
  loglike.old&lt;-loglike.old-(log(p)+log(1-p))/xrow
  loglike.old&lt;-loglike.old*xrow
  result&lt;-list(p.post=p.post, motif.prior=p, loglike=loglike.old)
}

## Fit each dataset separately
cmfitsep&lt;-function(x, type, tol=1e-3, max.iter=100) {
  ## initialize
  xrow&lt;-nrow(x)
  xcol&lt;-ncol(x)
  loglike0&lt;-list()
  loglike1&lt;-list()
  p&lt;-0.01*rep(1,xcol)
  loglike.final&lt;-rep(0,xcol)

  ## compute loglikelihood
  for(i in 1:xcol) {
    f0&lt;-type[[i]][[1]]
    f0param&lt;-type[[i]][[2]]
    f1&lt;-type[[i]][[3]]
    f1param&lt;-type[[i]][[4]]
    loglike0[[i]]&lt;-f0(x[,i],f0param)
    loglike1[[i]]&lt;-f1(x[,i],f1param)
  }

  p.post&lt;-matrix(0,xrow,xcol)

  ## EM algorithm to get MLE of p
  for(coli in 1:xcol) {
    loglike.old &lt;- -1e10
    for(i.iter in 1:max.iter) {
      if((i.iter%%50) == 0) {
        print(paste(&quot;We have run the first &quot;, i.iter, &quot; iterations&quot;,sep=&quot;&quot;))
      }
      err&lt;-tol+1

      ## compute posterior cluster membership
      clustlike&lt;-matrix(0,xrow,2)
      clustlike[,1]&lt;-log(1-p[coli])+loglike0[[coli]]
      clustlike[,2]&lt;-log(p[coli])+loglike1[[coli]]

      tempmax&lt;-apply(clustlike,1,max)
      for(j in 1:2) {
        clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
      }
      tempsum&lt;-apply(clustlike,1,sum)

      ## evaluate whether the log.likelihood increases
      loglike.new&lt;-sum(tempmax+log(tempsum))/xrow

      ## update motif occurrence rate
      for(j in 1:2) {
        clustlike[,j]&lt;-clustlike[,j]/tempsum
      }

      p.new&lt;-(sum(clustlike[,2]))/(xrow)

      ## evaluate convergence
      err&lt;-abs(p.new-p[coli])/p[coli]
      loglike.old&lt;-loglike.new
      p[coli]&lt;-p.new

      if(err&lt;tol) {
        break;
      }
    }

    ## compute posterior p
    clustlike&lt;-matrix(0,xrow,2)
    clustlike[,1]&lt;-log(1-p[coli])+loglike0[[coli]]
    clustlike[,2]&lt;-log(p[coli])+loglike1[[coli]]

    tempmax&lt;-apply(clustlike,1,max)
    for(j in 1:2) {
      clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
    }
    tempsum&lt;-apply(clustlike,1,sum)

    for(j in 1:2) {
      clustlike[,j]&lt;-clustlike[,j]/tempsum
    }

    p.post[,coli]&lt;-clustlike[,2]
    loglike.final[coli]&lt;-loglike.old
  }


  ## return
  loglike.final&lt;-loglike.final*xrow
  result&lt;-list(p.post=p.post, motif.prior=p, loglike=loglike.final)
}

## Fit the full model
cmfitfull&lt;-function(x, type, tol=1e-3, max.iter=100) {
  ## initialize
  xrow&lt;-nrow(x)
  xcol&lt;-ncol(x)
  loglike0&lt;-list()
  loglike1&lt;-list()
  K&lt;-2^xcol
  p&lt;-rep(1,K)/K
  pattern&lt;-rep(0,xcol)
  patid&lt;-matrix(0,K,xcol)

  ## compute loglikelihood
  for(i in 1:xcol) {
    f0&lt;-type[[i]][[1]]
    f0param&lt;-type[[i]][[2]]
    f1&lt;-type[[i]][[3]]
    f1param&lt;-type[[i]][[4]]
    loglike0[[i]]&lt;-f0(x[,i],f0param)
    loglike1[[i]]&lt;-f1(x[,i],f1param)
  }
  L&lt;-matrix(0,xrow,K)
  for(i in 1:K)
  {
    patid[i,]&lt;-pattern
    for(j in 1:xcol) {
      if(pattern[j] &lt; 0.5) {
        L[,i]&lt;-L[,i]+loglike0[[j]]
      } else {
        L[,i]&lt;-L[,i]+loglike1[[j]]
      }
    }

    if(i &lt; K) {
      pattern[xcol]&lt;-pattern[xcol]+1
      j&lt;-xcol
      while(pattern[j] &gt; 1) {
        pattern[j]&lt;-0
        j&lt;-j-1
        pattern[j]&lt;-pattern[j]+1
      }
    }
  }

  ## EM algorithm to get MLE of p and q
  loglike.old &lt;- -1e10
  for(i.iter in 1:max.iter) {
    if((i.iter%%50) == 0) {
      print(paste(&quot;We have run the first &quot;, i.iter, &quot; iterations&quot;,sep=&quot;&quot;))
    }
    err&lt;-tol+1

    ## compute posterior cluster membership
    clustlike&lt;-matrix(0,xrow,K)
    for(j in 1:K) {
      clustlike[,j]&lt;-log(p[j])+L[,j]
    }

    tempmax&lt;-apply(clustlike,1,max)
    for(j in 1:K) {
      clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
    }
    tempsum&lt;-apply(clustlike,1,sum)

    ## update motif occurrence rate
    for(j in 1:K) {
      clustlike[,j]&lt;-clustlike[,j]/tempsum
    }

    p.new&lt;-(apply(clustlike,2,sum)+1)/(xrow+K)

    ## evaluate convergence
    err&lt;-max(abs(p.new-p)/p)

    ## evaluate whether the log.likelihood increases
    loglike.new&lt;-(sum(tempmax+log(tempsum))+sum(log(p.new)))/xrow

    loglike.old&lt;-loglike.new
    p&lt;-p.new

    if(err&lt;tol) {
      break;
    }
  }

  ## compute posterior p
  clustlike&lt;-matrix(0,xrow,K)
  for(j in 1:K) {
    clustlike[,j]&lt;-log(p[j])+L[,j]
  }

  tempmax&lt;-apply(clustlike,1,max)
  for(j in 1:K) {
    clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
  }
  tempsum&lt;-apply(clustlike,1,sum)

  for(j in 1:K) {
    clustlike[,j]&lt;-clustlike[,j]/tempsum
  }

  p.post&lt;-matrix(0,xrow,xcol)
  for(j in 1:K) {
    for(i in 1:xcol) {
      if(patid[j,i] &gt; 0.5) {
        p.post[,i]&lt;-p.post[,i]+clustlike[,j]
      }
    }
  }

  ## return
  #calculate back loglikelihood
  loglike.old&lt;-loglike.old-sum(log(p))/xrow
  loglike.old&lt;-loglike.old*xrow
  result&lt;-list(p.post=p.post, motif.prior=p, loglike=loglike.old)
}

generatetype&lt;-function(limfitted)
{
  jtype&lt;-list()
  df&lt;-limfitted$g1num+limfitted$g2num-2+limfitted$df0
  for(j in 1:limfitted$compnum)
  {
    jtype[[j]]&lt;-list(f0=modt.f0.loglike, f0.param=df[j], f1=modt.f1.loglike, f1.param=c(df[j],limfitted$g1num[j],limfitted$g2num[j],limfitted$v0[j]))
  }
  jtype
}

cormotiffit &lt;- function(exprs, groupid=NULL, compid=NULL, K=1, tol=1e-3,
                        max.iter=100, BIC=TRUE, norm.factor.method=&quot;TMM&quot;,
                        voom.normalize.method = &quot;none&quot;, runtype=c(&quot;logCPM&quot;,&quot;counts&quot;,&quot;limmafits&quot;), each=3)
{
  # first I want to do some typechecking. Input can be either a normalized
  # matrix, a count matrix, or a list of limma fits. Dispatch the correct
  # limmafit accordingly.
  # todo: add some typechecking here
  limfitted &lt;- list()
  if (runtype==&quot;counts&quot;) {
    limfitted &lt;- limmafit.counts(exprs,groupid,compid, norm.factor.method, voom.normalize.method)
  } else if (runtype==&quot;logCPM&quot;) {
    limfitted &lt;- limmafit.default(exprs,groupid,compid)
  } else if (runtype==&quot;limmafits&quot;) {
    limfitted &lt;- limmafit.list(exprs)
  } else {
    stop(&quot;runtype must be one of &#39;logCPM&#39;, &#39;counts&#39;, or &#39;limmafits&#39;&quot;)
  }


  jtype&lt;-generatetype(limfitted)
  fitresult&lt;-list()
  ks &lt;- rep(K, each = each)
  fitresult &lt;- bplapply(1:length(ks), function(i, x, type, ks, tol, max.iter) {
    cmfit.X(x, type, K = ks[i], tol = tol, max.iter = max.iter)
  }, x=limfitted$t, type=jtype, ks=ks, tol=tol, max.iter=max.iter)

  best.fitresults &lt;- list()
  for (i in 1:length(K)) {
    w.k &lt;- which(ks==K[i])
    this.bic &lt;- c()
    for (j in w.k) this.bic[j] &lt;- -2 * fitresult[[j]]$loglike + (K[i] - 1 + K[i] * limfitted$compnum) * log(dim(limfitted$t)[1])
    w.min &lt;- which(this.bic == min(this.bic, na.rm = TRUE))[1]
    best.fitresults[[i]] &lt;- fitresult[[w.min]]
  }
  fitresult &lt;- best.fitresults

  bic &lt;- rep(0, length(K))
  aic &lt;- rep(0, length(K))
  loglike &lt;- rep(0, length(K))
  for (i in 1:length(K)) loglike[i] &lt;- fitresult[[i]]$loglike
  for (i in 1:length(K)) bic[i] &lt;- -2 * fitresult[[i]]$loglike + (K[i] - 1 + K[i] * limfitted$compnum) * log(dim(limfitted$t)[1])
  for (i in 1:length(K)) aic[i] &lt;- -2 * fitresult[[i]]$loglike + 2 * (K[i] - 1 + K[i] * limfitted$compnum)
  if(BIC==TRUE) {
    bestflag=which(bic==min(bic))
  }
  else {
    bestflag=which(aic==min(aic))
  }
  result&lt;-list(bestmotif=fitresult[[bestflag]],bic=cbind(K,bic),
               aic=cbind(K,aic),loglike=cbind(K,loglike), allmotifs=fitresult)

}

cormotiffitall&lt;-function(exprs,groupid,compid, tol=1e-3, max.iter=100)
{
  limfitted&lt;-limmafit(exprs,groupid,compid)
  jtype&lt;-generatetype(limfitted)
  fitresult&lt;-cmfitall(limfitted$t,type=jtype,tol=1e-3,max.iter=max.iter)
}

cormotiffitsep&lt;-function(exprs,groupid,compid, tol=1e-3, max.iter=100)
{
  limfitted&lt;-limmafit(exprs,groupid,compid)
  jtype&lt;-generatetype(limfitted)
  fitresult&lt;-cmfitsep(limfitted$t,type=jtype,tol=1e-3,max.iter=max.iter)
}

cormotiffitfull&lt;-function(exprs,groupid,compid, tol=1e-3, max.iter=100)
{
  limfitted&lt;-limmafit(exprs,groupid,compid)
  jtype&lt;-generatetype(limfitted)
  fitresult&lt;-cmfitfull(limfitted$t,type=jtype,tol=1e-3,max.iter=max.iter)
}

plotIC&lt;-function(fitted_cormotif)
{
  oldpar&lt;-par(mfrow=c(1,2))
  plot(fitted_cormotif$bic[,1], fitted_cormotif$bic[,2], type=&quot;b&quot;,xlab=&quot;Motif Number&quot;, ylab=&quot;BIC&quot;, main=&quot;BIC&quot;)
  plot(fitted_cormotif$aic[,1], fitted_cormotif$aic[,2], type=&quot;b&quot;,xlab=&quot;Motif Number&quot;, ylab=&quot;AIC&quot;, main=&quot;AIC&quot;)
}

plotMotif&lt;-function(fitted_cormotif,title=&quot;&quot;)
{
  layout(matrix(1:2,ncol=2))
  u&lt;-1:dim(fitted_cormotif$bestmotif$motif.q)[2]
  v&lt;-1:dim(fitted_cormotif$bestmotif$motif.q)[1]
  image(u,v,t(fitted_cormotif$bestmotif$motif.q),
        col=gray(seq(from=1,to=0,by=-0.1)),xlab=&quot;Study&quot;,yaxt = &quot;n&quot;,
        ylab=&quot;Corr. Motifs&quot;,main=paste(title,&quot;pattern&quot;,sep=&quot; &quot;))
  axis(2,at=1:length(v))
  for(i in 1:(length(u)+1))
  {
    abline(v=(i-0.5))
  }
  for(i in 1:(length(v)+1))
  {
    abline(h=(i-0.5))
  }
  Ng=10000
  if(is.null(fitted_cormotif$bestmotif$p.post)!=TRUE)
    Ng=nrow(fitted_cormotif$bestmotif$p.post)
  genecount=floor(fitted_cormotif$bestmotif$motif.p*Ng)
  NK=nrow(fitted_cormotif$bestmotif$motif.q)
  plot(0,0.7,pch=&quot;.&quot;,xlim=c(0,1.2),ylim=c(0.75,NK+0.25),
       frame.plot=FALSE,axes=FALSE,xlab=&quot;No. of genes&quot;,ylab=&quot;&quot;, main=paste(title,&quot;frequency&quot;,sep=&quot; &quot;))
  segments(0,0.7,fitted_cormotif$bestmotif$motif.p[1],0.7)
  rect(0,1:NK-0.3,fitted_cormotif$bestmotif$motif.p,1:NK+0.3,
       col=&quot;dark grey&quot;)
  mtext(1:NK,at=1:NK,side=2,cex=0.8)
  text(fitted_cormotif$bestmotif$motif.p+0.15,1:NK,
       labels=floor(fitted_cormotif$bestmotif$motif.p*Ng))
}</code></pre>
</div>
<div id="load-required-libraries" class="section level2">
<h2><strong>📌 Load Required Libraries</strong></h2>
<pre class="r"><code>library(Cormotif)
library(Rfast)
library(dplyr)
library(BiocParallel)
library(gprofiler2)
library(ggplot2)</code></pre>
</div>
<div id="corrmotif-model-0.1-micromolar" class="section level2">
<h2><strong>📌 Corrmotif Model 0.1 Micromolar</strong></h2>
</div>
<div id="load-corrmotif-data" class="section level2">
<h2><strong>📌 Load Corrmotif Data</strong></h2>
<pre class="r"><code># Read the Corrmotif Results
Corrmotif &lt;- read.csv(&quot;data/Corrmotif/CX5461.csv&quot;)
Corrmotif_df &lt;- data.frame(Corrmotif)
rownames(Corrmotif_df) &lt;- Corrmotif_df$Gene

# Filter for 0.1 Concentration Only
exprs.corrmotif &lt;- as.matrix(Corrmotif_df[, grep(&quot;0.1&quot;, colnames(Corrmotif_df))])


# Read group and comparison IDs
groupid &lt;- read.csv(&quot;data/Corrmotif/groupid.csv&quot;)
groupid_df &lt;- data.frame(groupid[, grep(&quot;0.1&quot;, colnames(groupid))])


compid &lt;- read.csv(&quot;data/Corrmotif/Compid.csv&quot;)
compid_df &lt;- compid[compid$Cond1 %in% unique(as.numeric(groupid_df)) &amp; compid$Cond2 %in% unique(as.numeric(groupid_df)), ]</code></pre>
</div>
<div id="corrmotif-model-0.1-micromolar-k18" class="section level2">
<h2><strong>📌 Corrmotif Model 0.1 Micromolar (K=1:8)</strong></h2>
</div>
<div id="fit-corrmotif-model-k18-0.1-micromolar" class="section level2">
<h2><strong>📌 Fit Corrmotif Model (K=1:8) (0.1
Micromolar)</strong></h2>
<pre class="r"><code>set.seed(11111)
# Fit Corrmotif Model (K = 1 to 8)
set.seed(11111)
motif.fitted_0.1 &lt;- cormotiffit(
  exprs = exprs.corrmotif,
  groupid = groupid_df,
  compid = compid_df,
  K = 1:8,
  max.iter = 1000,
  BIC = TRUE,
  runtype = &quot;logCPM&quot;
)

gene_prob_0.1 &lt;- motif.fitted_0.1$bestmotif$p.post
rownames(gene_prob_0.1) &lt;- rownames(Corrmotif_df)
motif_prob_0.1 &lt;- motif.fitted_0.1$bestmotif$clustlike
rownames(motif_prob_0.1) &lt;- rownames(gene_prob_0.1)
write.csv(motif_prob_0.1,&quot;data/cormotif_probability_genelist_0.1.csv&quot;)</code></pre>
</div>
<div id="plot-motif-0.1-micromolar" class="section level2">
<h2><strong>📌 Plot motif (0.1 Micromolar)</strong></h2>
<pre class="r"><code>cormotif_0.1 &lt;- readRDS(&quot;data/Corrmotif/cormotif_0.1.RDS&quot;)

cormotif_0.1$bic</code></pre>
<pre><code>     K      bic
[1,] 1 291696.5
[2,] 2 284585.3
[3,] 3 283482.9
[4,] 4 283551.8
[5,] 5 283620.7
[6,] 6 283689.6
[7,] 7 283758.5
[8,] 8 283827.4</code></pre>
<pre class="r"><code>plotIC(cormotif_0.1)</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Model1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Model1-1">
Past versions of Model1-1.png
</button>
</p>
<div id="fig-Model1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/41cd1be487c4309fb8ab2f2d36b18ad63cdc31e4/docs/figure/Corrmotif_Conc.Rmd/Model1-1.png" target="_blank">41cd1be</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ce4b32557fc3c7ee693369db9edf0a9d22968203/docs/figure/Corrmotif_Conc.Rmd/Model1-1.png" target="_blank">ce4b325</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plotMotif(cormotif_0.1)</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Model1-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Model1-2">
Past versions of Model1-2.png
</button>
</p>
<div id="fig-Model1-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/41cd1be487c4309fb8ab2f2d36b18ad63cdc31e4/docs/figure/Corrmotif_Conc.Rmd/Model1-2.png" target="_blank">41cd1be</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ce4b32557fc3c7ee693369db9edf0a9d22968203/docs/figure/Corrmotif_Conc.Rmd/Model1-2.png" target="_blank">ce4b325</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="extract-gene-probabilities-0.1-micromolar"
class="section level2">
<h2><strong>📌 Extract Gene Probabilities (0.1 Micromolar)</strong></h2>
<pre class="r"><code># Extract posterior probabilities for genes
gene_prob_tran_0.1 &lt;- cormotif_0.1$bestmotif$p.post

rownames(gene_prob_tran_0.1) &lt;- rownames(Corrmotif_df)

# Define gene probability groups
prob_1_0.1  &lt;- rownames(gene_prob_tran_0.1[(gene_prob_tran_0.1[,1] &lt;0.5 &amp; gene_prob_tran_0.1[,2] &lt;0.5 &amp; gene_prob_tran_0.1[,3] &lt;0.5 &amp; gene_prob_tran_0.1[,4] &lt;0.5 &amp; gene_prob_tran_0.1[,5] &lt; 0.5 &amp; gene_prob_tran_0.1[,6]&lt;0.5),])
length(prob_1_0.1)</code></pre>
<pre><code>[1] 12308</code></pre>
<pre class="r"><code>prob_2_0.1  &lt;- rownames(gene_prob_tran_0.1[(gene_prob_tran_0.1[,1] &lt;0.5 &amp; gene_prob_tran_0.1[,2] &gt;0.5 &amp; gene_prob_tran_0.1[,3] &gt;0.5 &amp; gene_prob_tran_0.1[,4] &lt;0.5 &amp; gene_prob_tran_0.1[,5] &gt; 0.5 &amp; gene_prob_tran_0.1[,6]&gt;0.5),])
length(prob_2_0.1)</code></pre>
<pre><code>[1] 415</code></pre>
<pre class="r"><code>prob_3_0.1  &lt;- rownames(gene_prob_tran_0.1[(gene_prob_tran_0.1[,1] &lt;0.5 &amp; gene_prob_tran_0.1[,2] &lt;0.5 &amp; gene_prob_tran_0.1[,3] &lt;0.5 &amp; gene_prob_tran_0.1[,4] &lt;0.5 &amp; gene_prob_tran_0.1[,5] &gt; 0.5 &amp; gene_prob_tran_0.1[,6]&gt;0.5),])
length(prob_3_0.1)</code></pre>
<pre><code>[1] 1551</code></pre>
</div>
<div
id="distribution-of-gene-clusters-identified-by-corrmotif-0.1-micromolar"
class="section level2">
<h2><strong>📌 Distribution of Gene Clusters Identified by Corrmotif
(0.1 micromolar)</strong></h2>
<pre class="r"><code># Load necessary library
library(ggplot2)

# Data
data &lt;- data.frame(
  Category = c(&quot;Non response (0.1 µM)&quot;, &quot;CX-DOX mid-late response (0.1 µM)&quot;, &quot;DOX only mid-late (0.1 µM)&quot;),
  Value = c(12308, 415, 1551)
)

# Define custom colors
custom_colors &lt;- c(&quot;Non response (0.1 µM)&quot; = &quot;#FF9999&quot;,
                   &quot;DOX only mid-late (0.1 µM)&quot; = &quot;#99FF99&quot;,
                   &quot;CX-DOX mid-late response (0.1 µM)&quot; = &quot;#66B2FF&quot;)

# Create pie chart
ggplot(data, aes(x = &quot;&quot;, y = Value, fill = Category)) +
  geom_bar(width = 1, stat = &quot;identity&quot;) +
  coord_polar(&quot;y&quot;, start = 0) +
  geom_text(aes(label = Value),
            position = position_stack(vjust = 0.5),
            size = 4, color = &quot;black&quot;) +
  labs(title = &quot;Pie Chart (0.1 micromolar Corrmotif)&quot;, x = NULL, y = NULL) +
  theme_void() +
  scale_fill_manual(values = custom_colors)</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/pie-chart_0.1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pie-chart_0.1-1">
Past versions of pie-chart_0.1-1.png
</button>
</p>
<div id="fig-pie-chart_0.1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2de67e6a82f82edbc78941b11ba72ed8aba9964e/docs/figure/Corrmotif_Conc.Rmd/pie-chart_0.1-1.png" target="_blank">2de67e6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="save-corr-motif-datasets-for-gene-ontology-analysis-0.1-micromolar"
class="section level2">
<h2><strong>📌 Save Corr-motif datasets for Gene Ontology analysis (0.1
Micromolar)</strong></h2>
<pre class="r"><code>write.csv(data.frame(Entrez_ID = prob_1_0.1), &quot;data/prob_1_0.1.csv&quot;, row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_2_0.1), &quot;data/prob_2_0.1.csv&quot;, row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_3_0.1), &quot;data/prob_3_0.1.csv&quot;, row.names = FALSE)</code></pre>
</div>
<div id="micromolar-abs-logfc" class="section level2">
<h2><strong>📌 0.1 Micromolar abs logFC</strong></h2>
<pre class="r"><code># Load Required Libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# Load Datasets (Only 0.1 Micromolar)
CX_0.1_3  &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;)
CX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;)
CX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;)

DOX_0.1_3  &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;)
DOX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;)
DOX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;)

# Combine All 0.1 Micromolar Datasets into a Single Dataframe
all_toptables_0.1 &lt;- bind_rows(
  CX_0.1_3  %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  CX_0.1_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  CX_0.1_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;),
  DOX_0.1_3  %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  DOX_0.1_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  DOX_0.1_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;)
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.1$Entrez_ID &lt;- as.character(all_toptables_0.1$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.1 &lt;- all_toptables_0.1 %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 micromolar)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response\n(0.1 micromolar)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response\n(0.1 micromolar)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group))

# Compute Absolute logFC
all_toptables_0.1 &lt;- all_toptables_0.1 %&gt;%
  mutate(absFC = abs(logFC))

# Convert Factors for Proper Ordering (Reversed Order for Response Groups)
all_toptables_0.1 &lt;- all_toptables_0.1 %&gt;%
  mutate(
    Drug = factor(Drug, levels = c(&quot;CX.5461&quot;, &quot;DOX&quot;)),
    Timepoint = factor(Timepoint, levels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Response_Group = factor(Response_Group,
                            levels = c(
                              &quot;CX-DOX mid-late response\n(0.1 micromolar)&quot;,
                              &quot;DOX only mid-late response\n(0.1 micromolar)&quot;,
                              &quot;Non response\n(0.1 micromolar)&quot;  # Reversed Order
                            ))
  )

# **Plot the Boxplot with Faceted Labels Wrapping Correctly**
ggplot(all_toptables_0.1, aes(x = Drug, y = absFC, fill = Drug)) +
  geom_boxplot() +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +  # Custom color palette
  facet_grid(Response_Group ~ Timepoint, labeller = label_wrap_gen(width = 20)) +  # Ensure Proper Wrapping
  theme_bw() +
  labs(
    x = &quot;Drugs&quot;,
    y = &quot;|Log Fold Change|&quot;,
    title = &quot;|Log Fold| for 0.1 micromolar&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.line = element_line(linewidth = 1.5),
    strip.background = element_rect(fill = &quot;gray&quot;),  # Gray background for facet labels
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),  # Bold styling for facet labels
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 15)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.1abs%20logFC-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-0.1abs-logFC-1">
Past versions of "0.1abs logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-0.1abs-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.1abs logFC-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.1abs logFC-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="micromolar-mean-abs-logfc-across-timepoints"
class="section level2">
<h2><strong>📌 0.1 Micromolar mean (Abs logFC) across
timepoints</strong></h2>
<pre class="r"><code># Load Required Libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# Load Datasets (Only 0.1 Micromolar)
CX_0.1_3  &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;)
CX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;)
CX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;)

DOX_0.1_3  &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;)
DOX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;)
DOX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;)

# Combine All 0.1 Micromolar Datasets into a Single Dataframe
all_toptables_0.1 &lt;- bind_rows(
  CX_0.1_3  %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;3&quot;),
  CX_0.1_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;24&quot;),
  CX_0.1_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;48&quot;),
  DOX_0.1_3  %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;3&quot;),
  DOX_0.1_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;24&quot;),
  DOX_0.1_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;48&quot;)
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.1$Entrez_ID &lt;- as.character(all_toptables_0.1$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.1 &lt;- all_toptables_0.1 %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 micromolar)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response\n(0.1 micromolar)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response\n(0.1 micromolar)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group))

# Compute Mean Absolute logFC for Line Plot
data_summary &lt;- all_toptables_0.1 %&gt;%
  mutate(abs_logFC = abs(logFC)) %&gt;%  # Take absolute log fold change
  group_by(Response_Group, Drug, Timepoint) %&gt;%
  dplyr::summarize(mean_abs_logFC = mean(abs_logFC, na.rm = TRUE), .groups = &quot;drop&quot;) %&gt;%
  as.data.frame()

# **Ensure all timepoints exist in the summary**
timepoints_full &lt;- expand.grid(
  Response_Group = unique(all_toptables_0.1$Response_Group),
  Drug = unique(all_toptables_0.1$Drug),
  Timepoint = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;)
)

# **Merge to keep missing timepoints**
data_summary &lt;- full_join(timepoints_full, data_summary, by = c(&quot;Response_Group&quot;, &quot;Drug&quot;, &quot;Timepoint&quot;))

# **Replace NA mean_abs_logFC with 0 if no genes were present**
data_summary$mean_abs_logFC[is.na(data_summary$mean_abs_logFC)] &lt;- 0

# Convert Factors for Proper Ordering (Reversed Order for Response Groups)
data_summary &lt;- data_summary %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;), labels = c(&quot;3 hours&quot;, &quot;24 hours&quot;, &quot;48 hours&quot;)),
    Response_Group = factor(Response_Group, levels = c(
      &quot;CX-DOX mid-late response\n(0.1 micromolar)&quot;,
      &quot;DOX only mid-late response\n(0.1 micromolar)&quot;,
      &quot;Non response\n(0.1 micromolar)&quot;  # Reversed Order
    ))
  )

# Define custom drug palette
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)

# **Plot the Line Plot for Absolute logFC**
ggplot(data_summary, aes(x = Timepoint, y = mean_abs_logFC, group = Drug, color = Drug)) +
  geom_point(size = 3) +
  geom_line(size = 1.2) +
  scale_color_manual(values = drug_palette) +
  ylim(0, 2.5) +  # Adjust the Y-axis for better visualization
  facet_wrap(~ Response_Group, ncol = 1) +  # Facet by Response Group (Reversed Order)
  theme_bw() +
  labs(
    x = &quot;Timepoints&quot;,
    y = &quot;Mean |Log Fold Change|&quot;,
    title = &quot;Mean |Log Fold Change| Across Response Groups (0.1 micromolar)&quot;,
    color = &quot;Drug&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text = element_text(size = 12, color = &quot;black&quot;),
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.1%20mean-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-0.1-mean-1">
Past versions of "0.1 mean-1.png"
</button>
</p>
<div id="fig-no-spaces-0.1-mean-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.1 mean-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.1 mean-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="micromolar-logfc" class="section level2">
<h2><strong>📌 0.1 Micromolar logFC</strong></h2>
<pre class="r"><code># Load required libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# Load Datasets (Only 0.1 Micromolar)
CX_0.1_3  &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;)
CX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;)
CX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;)

DOX_0.1_3  &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;)
DOX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;)
DOX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;)

# Combine All 0.1 Micromolar Datasets into a Single Dataframe
all_toptables_0.1 &lt;- bind_rows(
  CX_0.1_3  %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  CX_0.1_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  CX_0.1_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;),
  DOX_0.1_3  %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  DOX_0.1_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  DOX_0.1_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;)
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.1$Entrez_ID &lt;- as.character(all_toptables_0.1$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.1 &lt;- all_toptables_0.1 %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 micromolar)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response\n(0.1 micromolar)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response\n(0.1 micromolar)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group))

# Convert factors to ensure correct ordering (Reversed Order for Response Groups)
all_toptables_0.1 &lt;- all_toptables_0.1 %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Response_Group = factor(Response_Group, levels = c(
      &quot;CX-DOX mid-late response\n(0.1 micromolar)&quot;,
      &quot;DOX only mid-late response\n(0.1 micromolar)&quot;,
      &quot;Non response\n(0.1 micromolar)&quot;  # Reversed Order
    ))
  )

# **Plot the Boxplot**
ggplot(all_toptables_0.1, aes(x = Drug, y = logFC, fill = Drug)) +
  geom_boxplot() +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  facet_grid(Response_Group ~ Timepoint) +
  theme_bw() +
  labs(x = &quot;Drugs&quot;, y = &quot;Log Fold Change&quot;, title = &quot;Log Fold Change for 0.1 Micromolar&quot;) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.1%20logFC-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-0.1-logFC-1">
Past versions of "0.1 logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-0.1-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.1 logFC-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.1 logFC-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="micromolar-mean-logfc-across-timepoints"
class="section level2">
<h2><strong>📌 0.1 Micromolar mean logFC across timepoints</strong></h2>
<pre class="r"><code># Compute Mean logFC for Line Plot
data_summary &lt;- all_toptables_0.1 %&gt;%
  group_by(Response_Group, Drug, Timepoint) %&gt;%
  dplyr::summarize(mean_logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;) %&gt;%
  as.data.frame()

# Convert factors to ensure correct ordering (Reversed Order for Response Groups)
data_summary &lt;- data_summary %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Response_Group = factor(Response_Group, levels = c(
      &quot;CX-DOX mid-late response\n(0.1 micromolar)&quot;,
      &quot;DOX only mid-late response\n(0.1 micromolar)&quot;,
      &quot;Non response\n(0.1 micromolar)&quot;  # Reversed Order
    ))
  )

# **Plot the Line Plot**
ggplot(data_summary, aes(x = Timepoint, y = mean_logFC, group = Drug, color = Drug)) +
  geom_point(size = 3) +
  geom_line(size = 1.2) +
  scale_color_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  ylim(-2, 1.5) +  # Adjust the Y-axis for better visualization
  facet_wrap(~ Response_Group, ncol = 1) +  # Facet by Response Group (Reversed Order)
  theme_bw() +
  labs(
    x = &quot;Timepoints&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    title = &quot;Mean Log Fold Change Across Response Groups (0.1 micromolar)&quot;,
    color = &quot;Drug&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text = element_text(size = 12, color = &quot;black&quot;),
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.1%20mean2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-0.1-mean2-1">
Past versions of "0.1 mean2-1.png"
</button>
</p>
<div id="fig-no-spaces-0.1-mean2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.1 mean2-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.1 mean2-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="corrmotif-model-0.5-micromolar" class="section level2">
<h2><strong>📌 Corrmotif Model 0.5 Micromolar</strong></h2>
</div>
<div id="load-corrmotif-data-1" class="section level2">
<h2><strong>📌 Load Corrmotif Data</strong></h2>
<pre class="r"><code># Read the Corrmotif Results
Corrmotif &lt;- read.csv(&quot;data/Corrmotif/CX5461.csv&quot;)
Corrmotif_df &lt;- data.frame(Corrmotif)
rownames(Corrmotif_df) &lt;- Corrmotif_df$Gene

# Filter for 0.5 Concentration Only
exprs.corrmotif &lt;- as.matrix(Corrmotif_df[, grep(&quot;0.5&quot;, colnames(Corrmotif_df))])


# Read group and comparison IDs
groupid &lt;- read.csv(&quot;data/Corrmotif/groupid.csv&quot;)
groupid_df &lt;- data.frame(groupid[, grep(&quot;0.5&quot;, colnames(groupid))])


compid &lt;- read.csv(&quot;data/Corrmotif/Compid.csv&quot;)
compid_df &lt;- compid[compid$Cond1 %in% unique(as.numeric(groupid_df)) &amp; compid$Cond2 %in% unique(as.numeric(groupid_df)), ]</code></pre>
</div>
<div id="corrmotif-model-0.5-micromolar-k18" class="section level2">
<h2><strong>📌 Corrmotif Model 0.5 Micromolar (K=1:8)</strong></h2>
</div>
<div id="fit-corrmotif-model-k18-0.5-micromolar" class="section level2">
<h2><strong>📌 Fit Corrmotif Model (K=1:8) (0.5
Micromolar)</strong></h2>
<pre class="r"><code># Fit Corrmotif Model (K = 1 to 8)
set.seed(11111)
motif.fitted_0.5 &lt;- cormotiffit(
  exprs = exprs.corrmotif,
  groupid = groupid_df,
  compid = compid_df,
  K = 1:8,
  max.iter = 1000,
  BIC = TRUE,
  runtype = &quot;logCPM&quot;
)

gene_prob_0.5 &lt;- motif.fitted_0.5$bestmotif$p.post
rownames(gene_prob_0.5) &lt;- rownames(Corrmotif_df)
motif_prob_0.5 &lt;- motif.fitted_0.5$bestmotif$clustlike
rownames(motif_prob_0.5) &lt;- rownames(gene_prob_0.5)
write.csv(motif_prob_0.5,&quot;data/cormotif_probability_genelist_0.5.csv&quot;)</code></pre>
</div>
<div id="plot-motif-0.5-micromolar" class="section level2">
<h2><strong>📌 Plot motif (0.5 Micromolar)</strong></h2>
<pre class="r"><code>cormotif_0.5 &lt;- readRDS(&quot;data/Corrmotif/cormotif_0.5.RDS&quot;)

cormotif_0.5$bic</code></pre>
<pre><code>     K      bic
[1,] 1 352140.7
[2,] 2 346785.8
[3,] 3 344812.9
[4,] 4 344860.1
[5,] 5 344751.9
[6,] 6 344820.8
[7,] 7 344889.7
[8,] 8 344966.6</code></pre>
<pre class="r"><code>plotIC(cormotif_0.5)</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Motif_0.5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Motif_0.5-1">
Past versions of Motif_0.5-1.png
</button>
</p>
<div id="fig-Motif_0.5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/41cd1be487c4309fb8ab2f2d36b18ad63cdc31e4/docs/figure/Corrmotif_Conc.Rmd/Motif_0.5-1.png" target="_blank">41cd1be</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plotMotif(cormotif_0.5)</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Motif_0.5-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Motif_0.5-2">
Past versions of Motif_0.5-2.png
</button>
</p>
<div id="fig-Motif_0.5-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/41cd1be487c4309fb8ab2f2d36b18ad63cdc31e4/docs/figure/Corrmotif_Conc.Rmd/Motif_0.5-2.png" target="_blank">41cd1be</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="extract-gene-probabilities-0.5-micromolar"
class="section level2">
<h2><strong>📌 Extract Gene Probabilities (0.5 Micromolar)</strong></h2>
<pre class="r"><code># Extract posterior probabilities for genes
gene_prob_tran_0.5 &lt;- cormotif_0.5$bestmotif$p.post

rownames(gene_prob_tran_0.5) &lt;- rownames(Corrmotif_df)

# Define gene probability groups
prob_1_0.5  &lt;- rownames(gene_prob_tran_0.5[(gene_prob_tran_0.5[,1] &lt;0.5 &amp; gene_prob_tran_0.5[,2] &lt;0.5 &amp; gene_prob_tran_0.5[,3] &lt;0.5 &amp; gene_prob_tran_0.5[,4] &lt;0.5 &amp; gene_prob_tran_0.5[,5] &lt; 0.5 &amp; gene_prob_tran_0.5[,6]&lt;0.5),])
length(prob_1_0.5)</code></pre>
<pre><code>[1] 7134</code></pre>
<pre class="r"><code>prob_2_0.5  &lt;- rownames(gene_prob_tran_0.5[(gene_prob_tran_0.5[,1] &lt;0.5 &amp; gene_prob_tran_0.5[,2] &lt;0.5 &amp; gene_prob_tran_0.5[,3] &lt;0.5 &amp; gene_prob_tran_0.5[,4] &gt;0.5 &amp; gene_prob_tran_0.5[,5] &gt; 0.5 &amp; gene_prob_tran_0.5[,6]&gt;=0.02),])
length(prob_2_0.5)</code></pre>
<pre><code>[1] 179</code></pre>
<pre class="r"><code>prob_3_0.5  &lt;- rownames(gene_prob_tran_0.5[(gene_prob_tran_0.5[,1] &lt;0.5 &amp; gene_prob_tran_0.5[,2] &lt;0.5 &amp; gene_prob_tran_0.5[,3] &lt;0.5 &amp; gene_prob_tran_0.5[,4] &lt;0.5 &amp; gene_prob_tran_0.5[,5] &gt; 0.5 &amp; gene_prob_tran_0.5[,6]&gt;0.5),])
length(prob_3_0.5)</code></pre>
<pre><code>[1] 6450</code></pre>
<pre class="r"><code>prob_4_0.5  &lt;- rownames(gene_prob_tran_0.5[(gene_prob_tran_0.5[,1] &gt;= 0.1 &amp; gene_prob_tran_0.5[,2] &gt; 0.5 &amp; gene_prob_tran_0.5[,3] &gt; 0.5 &amp; gene_prob_tran_0.5[,4] &gt;= 0.02 &amp; gene_prob_tran_0.5[,5] &lt; 0.5 &amp; gene_prob_tran_0.5[,6] &lt; 0.5),]) 

length(prob_4_0.5)</code></pre>
<pre><code>[1] 142</code></pre>
<pre class="r"><code>prob_5_0.5  &lt;- rownames(gene_prob_tran_0.5[(gene_prob_tran_0.5[,1] &lt;0.5 &amp; gene_prob_tran_0.5[,2] &gt;0.5 &amp; gene_prob_tran_0.5[,3] &gt;0.5 &amp; gene_prob_tran_0.5[,4] &gt;=0.02 &amp; gene_prob_tran_0.5[,4] &lt;0.5 &amp; gene_prob_tran_0.5[,5] &gt; 0.5 &amp; gene_prob_tran_0.5[,6]&gt;0.5),])
length(prob_5_0.5)</code></pre>
<pre><code>[1] 221</code></pre>
</div>
<div
id="distribution-of-gene-clusters-identified-by-corrmotif-0.5-micromolar"
class="section level2">
<h2><strong>📌 Distribution of Gene Clusters Identified by Corrmotif
(0.5 micromolar)</strong></h2>
<pre class="r"><code># Load necessary library
library(ggplot2)

# Data
data &lt;- data.frame(
  Category = c(&quot;Non response (0.5)&quot;, &quot;DOX-only early-mid response (0.5 µM)&quot;, &quot;DOX only mid-late response (0.5 µM)&quot;, &quot;CX-only mid-late response (0.5 µM)&quot;, &quot;CX-DOX mid-late response (0.5 µM)&quot;),
  Value = c(7134,179,6450,142,221)
)

# Add values to category names (to be displayed in the legend)
data$Category &lt;- paste0(data$Category, &quot; (&quot;, data$Value, &quot;)&quot;)

# Define custom colors with updated category names
custom_colors &lt;- setNames(
  c(&quot;#FF9999&quot;, &quot;#FF66CC&quot;, &quot;#66B2FF&quot;, &quot;#99FF99&quot;, &quot;#FFD700&quot;),
  data$Category  # Ensures color names match updated categories
)

# Create pie chart without number labels inside
ggplot(data, aes(x = &quot;&quot;, y = Value, fill = Category)) +
  geom_bar(width = 1, stat = &quot;identity&quot;) +
  coord_polar(&quot;y&quot;, start = 0) +
  labs(title = &quot;Pie Chart (0.5 micromolar Corrmotif)&quot;, x = NULL, y = NULL) +
  theme_void() +
  scale_fill_manual(values = custom_colors)</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/pie-chart_0.5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pie-chart_0.5-1">
Past versions of pie-chart_0.5-1.png
</button>
</p>
<div id="fig-pie-chart_0.5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/pie-chart_0.5-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2de67e6a82f82edbc78941b11ba72ed8aba9964e/docs/figure/Corrmotif_Conc.Rmd/pie-chart_0.5-1.png" target="_blank">2de67e6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="combined-pie-charts-for-concentrations" class="section level2">
<h2><strong>📌 Combined pie charts for concentrations</strong></h2>
<pre class="r"><code># Load necessary libraries
library(ggplot2)
library(dplyr)

# Data for 0.1 µM
data_0.1 &lt;- data.frame(
  Category = c(&quot;Non response&quot;, &quot;CX-DOX mid-late response&quot;, &quot;DOX only mid-late&quot;),
  Value = c(12308, 415, 1551),
  Concentration = &quot;0.1 µM&quot;
)

# Data for 0.5 µM
data_0.5 &lt;- data.frame(
  Category = c(&quot;Non response&quot;, &quot;DOX specific response&quot;, &quot;DOX only mid-late response&quot;, 
               &quot;CX + DOX (early) response&quot;, &quot;DOX + CX (mid-late) response&quot;),
  Value = c(7134, 179, 6450, 142, 221),
  Concentration = &quot;0.5 µM&quot;
)

# Combine both datasets
combined_data &lt;- bind_rows(data_0.1, data_0.5)

# Add values to category names (for legend only)
combined_data$Category_Legend &lt;- paste0(combined_data$Category, &quot; (&quot;, combined_data$Value, &quot;)&quot;)

# Define custom colors for updated categories
custom_colors &lt;- c(
  &quot;Non response (12308)&quot; = &quot;#FF9999&quot;,
  &quot;CX-DOX mid-late response (415)&quot; = &quot;#66B2FF&quot;,
  &quot;DOX only mid-late (1551)&quot; = &quot;#99FF99&quot;,
  &quot;Non response (7134)&quot; = &quot;#FF9999&quot;,
  &quot;DOX specific response (179)&quot; = &quot;#FF66CC&quot;,
  &quot;DOX only mid-late response (6450)&quot; = &quot;#99FF99&quot;,
  &quot;CX + DOX (early) response (142)&quot; = &quot;#FFD700&quot;,
  &quot;DOX + CX (mid-late) response (221)&quot; = &quot;#8A2BE2&quot;
)

# Ensure categories appear in a specific order
combined_data$Category_Legend &lt;- factor(combined_data$Category_Legend, levels = names(custom_colors))

# Create faceted pie charts without numbers inside the slices
pie_chart &lt;- ggplot(combined_data, aes(x = &quot;&quot;, y = Value, fill = Category_Legend)) +
  geom_bar(width = 1, stat = &quot;identity&quot;) +
  coord_polar(&quot;y&quot;, start = 0) +
  facet_wrap(~ Concentration) +  # Facet by concentration (side-by-side)
  labs(title = &quot;Corrmotif Pie Charts for 0.1 µM and 0.5 µM&quot;, x = NULL, y = NULL, fill = &quot;Category&quot;) +
  theme_void() +
  theme(
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.5),  # Adds box around facets
    strip.background = element_rect(fill = &quot;white&quot;, color = &quot;black&quot;, linewidth = 1),  # Box for facet titles
    strip.text = element_text(size = 12, face = &quot;bold&quot;, color = &quot;black&quot;),
    legend.title = element_text(size = 12, face = &quot;bold&quot;),  # Bold legend title
    legend.text = element_text(size = 10)  # Adjust legend text size
  ) +
  scale_fill_manual(values = custom_colors)

# Display the plot
print(pie_chart)</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/pie-chart_comb-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div
id="save-corr-motif-datasets-for-gene-ontology-analysis-0.5-micromolar"
class="section level2">
<h2><strong>📌 Save Corr-motif datasets for Gene Ontology analysis (0.5
Micromolar)</strong></h2>
<pre class="r"><code>write.csv(data.frame(Entrez_ID = prob_1_0.5), &quot;data/prob_1_0.5.csv&quot;, row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_2_0.5), &quot;data/prob_2_0.5.csv&quot;, row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_3_0.5), &quot;data/prob_3_0.5.csv&quot;, row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_4_0.5), &quot;data/prob_4_0.5.csv&quot;, row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_5_0.5), &quot;data/prob_5_0.5.csv&quot;, row.names = FALSE)</code></pre>
</div>
<div id="micromolar-abs-logfc-1" class="section level2">
<h2><strong>📌 0.5 Micromolar abs logFC</strong></h2>
<pre class="r"><code># Load Response Groups from CSV Files
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# Load Datasets (Only 0.5 Micromolar)
CX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)

DOX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

# Convert datasets to DataFrames
Toptable_CX_0.5_3_df  &lt;- data.frame(CX_0.5_3)
Toptable_CX_0.5_24_df &lt;- data.frame(CX_0.5_24)
Toptable_CX_0.5_48_df &lt;- data.frame(CX_0.5_48)

Toptable_DOX_0.5_3_df  &lt;- data.frame(DOX_0.5_3)
Toptable_DOX_0.5_24_df &lt;- data.frame(DOX_0.5_24)
Toptable_DOX_0.5_48_df &lt;- data.frame(DOX_0.5_48)

# Combine All 0.5 Micromolar Datasets into a Single Dataframe
all_toptables_0.5 &lt;- bind_rows(
  Toptable_CX_0.5_3_df  %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;3&quot;),
  Toptable_CX_0.5_24_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;24&quot;),
  Toptable_CX_0.5_48_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;48&quot;),
  Toptable_DOX_0.5_3_df  %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;3&quot;),
  Toptable_DOX_0.5_24_df %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;24&quot;),
  Toptable_DOX_0.5_48_df %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;48&quot;)
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.5$Entrez_ID &lt;- as.character(all_toptables_0.5$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-only early-mid response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX-only mid-late response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;CX-DOX mid-late response\n(0.5 micromolar)&quot;,
      TRUE ~ NA_character_
    )
  )

# Remove NA Values (Genes Not in Response Groups)
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;% filter(!is.na(Response_Group))

# Compute Absolute logFC
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;%
  mutate(absFC = abs(logFC))

# Convert Factors for Proper Ordering (Reversed Order for Response Groups)
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;%
  mutate(
    Drug = factor(Drug, levels = c(&quot;CX.5461&quot;, &quot;DOX&quot;)),
    Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;),
                       labels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Response_Group = factor(Response_Group,
                            levels = c(&quot;CX-DOX mid-late response\n(0.5 micromolar)&quot;,
                                       &quot;CX-only mid-late response\n(0.5 micromolar)&quot;,
                                       &quot;DOX only mid-late response\n(0.5 micromolar)&quot;,
                                       &quot;DOX-only early-mid response\n(0.5 micromolar)&quot;,
                                       &quot;Non response\n(0.5 micromolar)&quot;)) # Reversed Order
  )

# **Plot the Boxplot with Faceted Labels Wrapping Correctly**
ggplot(all_toptables_0.5, aes(x = Drug, y = absFC, fill = Drug)) +
  geom_boxplot() +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +  # Custom color palette
  facet_grid(Response_Group ~ Timepoint, labeller = label_wrap_gen(width = 20)) +  # Ensure Proper Wrapping
  theme_bw() +
  labs(
    x = &quot;Drugs&quot;,
    y = &quot;|Log Fold Change|&quot;,
    title = &quot;|Log Fold| for 0.5 micromolar&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.line = element_line(linewidth = 1.5),
    strip.background = element_rect(fill = &quot;gray&quot;),  # Gray background for facet labels
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),  # Bold styling for facet labels
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 15)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.5abs%20logFC-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-0.5abs-logFC-1">
Past versions of "0.5abs logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-0.5abs-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.5abs logFC-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.5abs logFC-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="micromolar-mean-abs-logfc-across-timepoints-1"
class="section level2">
<h2><strong>📌 0.5 Micromolar mean (Abs logFC) across
timepoints</strong></h2>
<pre class="r"><code># Load required libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# Load Datasets (Only 0.5 Micromolar)
CX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)

DOX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

# Combine All 0.5 Micromolar Datasets into a Single Dataframe
all_toptables_0.5 &lt;- bind_rows(
  CX_0.5_3  %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;3&quot;),
  CX_0.5_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;24&quot;),
  CX_0.5_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;48&quot;),
  DOX_0.5_3  %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;3&quot;),
  DOX_0.5_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;24&quot;),
  DOX_0.5_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;48&quot;)
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.5$Entrez_ID &lt;- as.character(all_toptables_0.5$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-only early-mid response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX-only mid-late response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX-only mid-late response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;CX-DOX mid-late response\n(0.5 micromolar)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group))  # Remove NA values

# Compute Mean Absolute logFC for Line Plot
data_summary &lt;- all_toptables_0.5 %&gt;%
  mutate(abs_logFC = abs(logFC)) %&gt;%
  group_by(Response_Group, Drug, Timepoint) %&gt;%
  dplyr::summarize(mean_abs_logFC = mean(abs_logFC, na.rm = TRUE), .groups = &quot;drop&quot;) %&gt;%
  as.data.frame()

# **Ensure all timepoints exist in the summary**
timepoints_full &lt;- expand.grid(
  Response_Group = unique(all_toptables_0.5$Response_Group),
  Drug = unique(all_toptables_0.5$Drug),
  Timepoint = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;)
)

# **Merge to keep missing timepoints**
data_summary &lt;- full_join(timepoints_full, data_summary, by = c(&quot;Response_Group&quot;, &quot;Drug&quot;, &quot;Timepoint&quot;))

# **Replace NA mean_abs_logFC with 0 if no genes were present**
data_summary$mean_abs_logFC[is.na(data_summary$mean_abs_logFC)] &lt;- 0

# Convert Factors for Proper Ordering (Reversed Order for Response Groups)
data_summary &lt;- data_summary %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;), labels = c(&quot;3 hours&quot;, &quot;24 hours&quot;, &quot;48 hours&quot;)),
    Response_Group = factor(Response_Group, levels = c(
      &quot;CX-DOX mid-late response\n(0.5 micromolar)&quot;,
      &quot;CX-only mid-late response\n(0.5 micromolar)&quot;,
      &quot;DOX-only mid-late response\n(0.5 micromolar)&quot;,
      &quot;DOX-only early-mid response\n(0.5 micromolar)&quot;,
      &quot;Non response\n(0.5 micromolar)&quot;  # Reversed order
    ))
  )

# Define custom drug palette
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)

# **Plot the Line Plot for Mean Absolute logFC**
ggplot(data_summary, aes(x = Timepoint, y = mean_abs_logFC, group = Drug, color = Drug)) +
  geom_point(size = 3) +
  geom_line(size = 1.2) +
  scale_color_manual(values = drug_palette) +
  ylim(0, 2.5) +  # Adjust the Y-axis for better visualization
  facet_wrap(~ Response_Group, ncol = 1) +  # Facet by Response Group (Reversed Order)
  theme_bw() +
  labs(
    x = &quot;Timepoints&quot;,
    y = &quot;Mean |Log Fold Change|&quot;,
    title = &quot;Mean |Log Fold Change| Across Response Groups (0.5 micromolar)&quot;,
    color = &quot;Drug&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text = element_text(size = 12, color = &quot;black&quot;),
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.5%20mean-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-0.5-mean-1">
Past versions of "0.5 mean-1.png"
</button>
</p>
<div id="fig-no-spaces-0.5-mean-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.5 mean-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.5 mean-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="micromolar-logfc-1" class="section level2">
<h2><strong>📌 0.5 Micromolar logFC</strong></h2>
<pre class="r"><code># Load required libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# Load Datasets (Only 0.5 Micromolar)
CX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)

DOX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

# Combine All 0.5 Micromolar Datasets into a Single Dataframe
all_toptables_0.5 &lt;- bind_rows(
  CX_0.5_3  %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  CX_0.5_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  CX_0.5_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;),
  DOX_0.5_3  %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  DOX_0.5_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  DOX_0.5_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;)
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.5$Entrez_ID &lt;- as.character(all_toptables_0.5$Entrez_ID)

# Assign Response Groups
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-only early-mid response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX-only mid-late response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX-only mid-late response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;CX-DOX mid-late response\n(0.5 micromolar)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group))

# Convert factors to ensure correct ordering (Reversed Order for Response Groups)
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Response_Group = factor(Response_Group, levels = c(
      &quot;CX-DOX mid-late response\n(0.5 micromolar)&quot;,
      &quot;CX-only mid-late response\n(0.5 micromolar)&quot;,
      &quot;DOX-only mid-late response\n(0.5 micromolar)&quot;,
      &quot;DOX-only early-mid response\n(0.5 micromolar)&quot;,
      &quot;Non response\n(0.5 micromolar)&quot;  # Reversed Order
    ))
  )

# **Plot the Boxplot**
ggplot(all_toptables_0.5, aes(x = Drug, y = logFC, fill = Drug)) +
  geom_boxplot() +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  facet_grid(Response_Group ~ Timepoint) +
  theme_bw() +
  labs(x = &quot;Drugs&quot;, y = &quot;Log Fold Change&quot;, title = &quot;Log Fold Change for 0.5 Micromolar&quot;) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.5logFC-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-0.5logFC-1">
Past versions of 0.5logFC-1.png
</button>
</p>
<div id="fig-0.5logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.5logFC-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.5logFC-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="micromolar-mean-logfc-across-timepoints-1"
class="section level2">
<h2><strong>📌 0.5 Micromolar mean logFC across timepoints</strong></h2>
<pre class="r"><code># Load Required Libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# Load Datasets (Only 0.5 Micromolar)
CX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)

DOX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

# Combine All 0.5 Micromolar Datasets into a Single Dataframe
all_toptables_0.5 &lt;- bind_rows(
  CX_0.5_3  %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  CX_0.5_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  CX_0.5_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;),
  DOX_0.5_3  %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  DOX_0.5_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  DOX_0.5_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;)
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.5$Entrez_ID &lt;- as.character(all_toptables_0.5$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-only early-mid response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX-only mid-late response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX-only mid-late response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;CX-DOX mid-late response\n(0.5 micromolar)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group))

# Compute Mean logFC for Line Plot
data_summary_0.5 &lt;- all_toptables_0.5 %&gt;%
  group_by(Response_Group, Drug, Timepoint) %&gt;%
  dplyr::summarize(mean_logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;) %&gt;%
  as.data.frame()

# **Ensure all timepoints exist in the summary**
timepoints_full &lt;- expand.grid(
  Response_Group = unique(all_toptables_0.5$Response_Group),
  Drug = unique(all_toptables_0.5$Drug),
  Timepoint = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)
)

# **Merge to keep missing timepoints**
data_summary_0.5 &lt;- full_join(timepoints_full, data_summary_0.5, by = c(&quot;Response_Group&quot;, &quot;Drug&quot;, &quot;Timepoint&quot;))

# **Replace NA mean_logFC with 0 if no genes were present**
data_summary_0.5$mean_logFC[is.na(data_summary_0.5$mean_logFC)] &lt;- 0

# Convert Factors for Proper Ordering (Reversed Order for Response Groups)
data_summary_0.5 &lt;- data_summary_0.5 %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Response_Group = factor(Response_Group, levels = c(
      &quot;CX-DOX mid-late response\n(0.5 micromolar)&quot;,
      &quot;CX-only mid-late response\n(0.5 micromolar)&quot;,
      &quot;DOX-only mid-late response\n(0.5 micromolar)&quot;,
      &quot;DOX-only early-mid response\n(0.5 micromolar)&quot;,
      &quot;Non response\n(0.5 micromolar)&quot;  # Reversed Order
    ))
  )

# Define custom drug palette
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)

# **Plot the Line Plot for Mean logFC**
ggplot(data_summary_0.5, aes(x = Timepoint, y = mean_logFC, group = Drug, color = Drug)) +
  geom_point(size = 3) +
  geom_line(size = 1.2) +
  scale_color_manual(values = drug_palette) +
  ylim(-2, 1.5) +  # Adjust the Y-axis for better visualization
  facet_wrap(~ Response_Group, ncol = 1) +  # Facet by Response Group (Reversed Order)
  theme_bw() +
  labs(
    x = &quot;Timepoints&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    title = &quot;Mean Log Fold Change Across Response Groups (0.5 micromolar)&quot;,
    color = &quot;Drug&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text = element_text(size = 12, color = &quot;black&quot;),
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.5%20mean2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-0.5-mean2-1">
Past versions of "0.5 mean2-1.png"
</button>
</p>
<div id="fig-no-spaces-0.5-mean2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.5 mean2-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.5 mean2-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="proportion-of-dna-damage-repair-genes-0.1-micromolar"
class="section level2">
<h2><strong>📌 Proportion of DNA Damage Repair Genes (0.1
micromolar)</strong></h2>
<pre class="r"><code># Load necessary libraries
library(dplyr)
library(ggplot2)
library(tidyr)</code></pre>
<pre><code>Warning: package &#39;tidyr&#39; was built under R version 4.3.3</code></pre>
<pre class="r"><code>library(org.Hs.eg.db)</code></pre>
<pre><code>Warning: package &#39;AnnotationDbi&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;IRanges&#39; was built under R version 4.3.1</code></pre>
<pre><code>Warning: package &#39;S4Vectors&#39; was built under R version 4.3.1</code></pre>
<pre class="r"><code># **🔹 Read DNA Damage Repair Gene List**
DNA_damage &lt;- read.csv(&quot;data/DNA_Damage.csv&quot;, stringsAsFactors = FALSE)

# Convert gene symbols to Entrez IDs
DNA_damage &lt;- DNA_damage %&gt;%
  mutate(Entrez_ID = mapIds(org.Hs.eg.db,
                            keys = DNA_damage$Symbol,
                            column = &quot;ENTREZID&quot;,
                            keytype = &quot;SYMBOL&quot;,
                            multiVals = &quot;first&quot;))

DNA_damage_genes &lt;- na.omit(DNA_damage$Entrez_ID)

# **🔹 Load Corrmotif Groups for 0.1 Concentration**
prob_groups_0.1 &lt;- list(
  &quot;Non Response (0.1)&quot; = read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID,
   &quot;CX_DOX mid-late (0.1)&quot; = read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID,
  &quot;DOX only mid-late (0.1)&quot;= read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID
)

# **🔹 Create Dataframe for Corrmotif Groups**
corrmotif_df_0.1 &lt;- bind_rows(
  lapply(prob_groups_0.1, function(ids) {
    data.frame(Entrez_ID = ids)
  }),
  .id = &quot;Response_Group&quot;
)

# **🔹 Match Entrez_IDs with DNA Damage Repair Genes**
corrmotif_df_0.1 &lt;- corrmotif_df_0.1 %&gt;%
  mutate(DNA_Damage = ifelse(Entrez_ID %in% DNA_damage_genes, &quot;Yes&quot;, &quot;No&quot;))

# **🔹 Count DNA Damage Repair Genes in Each Response Group**
proportion_data &lt;- corrmotif_df_0.1 %&gt;%
  group_by(Response_Group, DNA_Damage) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Response_Group) %&gt;%
  mutate(Percentage = (Count / sum(Count)) * 100)

# **🔹 Ensure &quot;Yes&quot; is at the Bottom and &quot;No&quot; at the Top**
proportion_data$DNA_Damage &lt;- factor(proportion_data$DNA_Damage, levels = c(&quot;Yes&quot;, &quot;No&quot;))

# **🔹 Set Order of Response Groups for X-axis**
response_order &lt;- c(&quot;Non Response (0.1)&quot;, &quot;CX_DOX mid-late (0.1)&quot;,&quot;DOX only mid-late (0.1)&quot;)
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = response_order)

# **🔹 Perform Chi-Square Tests for &quot;DOX only mid-late (0.1)&quot; and &quot;CX_DOX mid-late (0.1)&quot; vs &quot;Non Response (0.1)&quot;**
non_response_counts &lt;- proportion_data %&gt;%
  filter(Response_Group == &quot;Non Response (0.1)&quot;) %&gt;%
  dplyr::select(DNA_Damage, Count) %&gt;%
  {setNames(.$Count, .$DNA_Damage)}  # Convert to named vector

chi_results &lt;- proportion_data %&gt;%
  filter(Response_Group %in% c(&quot;DOX only mid-late (0.1)&quot;, &quot;CX_DOX mid-late (0.1)&quot;)) %&gt;%
  group_by(Response_Group) %&gt;%
  summarise(
    p_value = {
      group_counts &lt;- Count[DNA_Damage %in% c(&quot;Yes&quot;, &quot;No&quot;)]
      if (!&quot;Yes&quot; %in% DNA_Damage) group_counts &lt;- c(group_counts, 0)
      if (!&quot;No&quot; %in% DNA_Damage) group_counts &lt;- c(0, group_counts)

      contingency_table &lt;- matrix(c(
        group_counts[1], group_counts[2],
        non_response_counts[&quot;Yes&quot;], non_response_counts[&quot;No&quot;]
      ), nrow = 2, byrow = TRUE)

      # Perform chi-square test if all values are valid
      if (all(contingency_table &gt;= 0 &amp; is.finite(contingency_table))) {
        chisq.test(contingency_table)$p.value
      } else {
        NA
      }
    },
    .groups = &quot;drop&quot;
  ) %&gt;%
  mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))

# **🔹 Merge Chi-Square Results into Proportion Data**
proportion_data &lt;- proportion_data %&gt;%
  left_join(chi_results %&gt;% dplyr::select(Response_Group, Significance), by = &quot;Response_Group&quot;)

# **🔹 Set Star Position Uniform Across Groups at 105%**
star_positions &lt;- data.frame(
  Response_Group = c(&quot;CX_DOX mid-late (0.1)&quot;, &quot;DOX only mid-late (0.1)&quot;),
  y_pos = 105,  # Fixed at 105% of Y-axis
  Significance = chi_results$Significance
)

# **🔹 Generate Proportion Plot with Chi-Square Stars**
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = DNA_Damage)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +  # Stacked bars
  geom_text(
    data = star_positions,
    aes(x = Response_Group, y = y_pos, label = Significance),  # Place stars at fixed 105%
    inherit.aes = FALSE,
    size = 6, color = &quot;black&quot;, fontface = &quot;bold&quot;, vjust = 0  # Keeps stars aligned
  ) +
  scale_y_continuous(labels = scales::percent_format(scale = 1), limits = c(0, 110)) +  # Fixed Y-axis to 100%
  scale_fill_manual(values = c(&quot;Yes&quot; = &quot;#e41a1c&quot;, &quot;No&quot; = &quot;#377eb8&quot;)) +  # Yes (Red), No (Blue)
  labs(
    title = &quot;Proportion of DNA Damage Repair Genes in\n0.1 Corrmotif Response Groups&quot;,
    x = &quot;Response Groups (0.1 Concentration)&quot;,
    y = &quot;Percentage&quot;,
    fill = &quot;DNA Damage Repair&quot;
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
    legend.title = element_blank(),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    strip.background = element_blank(),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/prop%20DNA-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-prop-DNA-1">
Past versions of "prop DNA-1.png"
</button>
</p>
<div id="fig-no-spaces-prop-DNA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/81247f64f08df43c2b40a386707fc8e9cc240143/docs/figure/Corrmotif_Conc.Rmd/prop DNA-1.png" target="_blank">81247f6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/96d0db145cf68e6db0d39e70bad9af0f75793aae/docs/figure/Corrmotif_Conc.Rmd/prop DNA-1.png" target="_blank">96d0db1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="proportion-of-dna-damage-repair-genes-0.5-micromolar"
class="section level2">
<h2><strong>📌 Proportion of DNA Damage Repair Genes (0.5 micromolar)
</strong></h2>
<pre class="r"><code># Load necessary libraries
library(dplyr)
library(ggplot2)
library(tidyr)
library(org.Hs.eg.db)

# **🔹 Read DNA Damage Repair Gene List**
DNA_damage &lt;- read.csv(&quot;data/DNA_Damage.csv&quot;, stringsAsFactors = FALSE)

# Convert gene symbols to Entrez IDs
DNA_damage &lt;- DNA_damage %&gt;%
  mutate(Entrez_ID = mapIds(org.Hs.eg.db,
                            keys = DNA_damage$Symbol,
                            column = &quot;ENTREZID&quot;,
                            keytype = &quot;SYMBOL&quot;,
                            multiVals = &quot;first&quot;))

DNA_damage_genes &lt;- na.omit(DNA_damage$Entrez_ID)

# **🔹 Load Corrmotif Groups for 0.5 Concentration**
prob_groups_0.5 &lt;- list(
  &quot;Non Response (0.5)&quot; = read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID,
  &quot;DOX-specific response (0.5)&quot; = read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID,
  &quot;DOX only mid-late response (0.5)&quot; = read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID,
  &quot;CX DOX (early) response (0.5)&quot; = read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID,
  &quot;DOX + CX (mid-late) response (0.5)&quot; = read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID
)

# **🔹 Create Dataframe for Corrmotif Groups**
corrmotif_df_0.5 &lt;- bind_rows(
  lapply(prob_groups_0.5, function(ids) {
    data.frame(Entrez_ID = ids)
  }),
  .id = &quot;Response_Group&quot;
)

# **🔹 Match Entrez_IDs with DNA Damage Repair Genes**
corrmotif_df_0.5 &lt;- corrmotif_df_0.5 %&gt;%
  mutate(DNA_Damage = ifelse(Entrez_ID %in% DNA_damage_genes, &quot;Yes&quot;, &quot;No&quot;))

# **🔹 Count DNA Damage Repair Genes in Each Response Group**
proportion_data &lt;- corrmotif_df_0.5 %&gt;%
  group_by(Response_Group, DNA_Damage) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Response_Group) %&gt;%
  mutate(Percentage = (Count / sum(Count)) * 100)

# **🔹 Ensure &quot;Yes&quot; is at the Bottom and &quot;No&quot; at the Top**
proportion_data$DNA_Damage &lt;- factor(proportion_data$DNA_Damage, levels = c(&quot;Yes&quot;, &quot;No&quot;))

# **🔹 Set Order of Response Groups for X-axis**
response_order &lt;- c(&quot;Non Response (0.5)&quot;, &quot;DOX-specific response (0.5)&quot;, &quot;DOX only mid-late response (0.5)&quot;, 
                    &quot;CX DOX (early) response (0.5)&quot;, &quot;DOX + CX (mid-late) response (0.5)&quot;)
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = response_order)

# **🔹 Perform Chi-Square Tests for Each Response Group vs Non-Response**
non_response_counts &lt;- proportion_data %&gt;%
  filter(Response_Group == &quot;Non Response (0.5)&quot;) %&gt;%
  dplyr::select(DNA_Damage, Count) %&gt;%
  {setNames(.$Count, .$DNA_Damage)}  # Convert to named vector

# **Comparing Each Group Against &quot;Non Response (0.5)&quot;**
chi_results &lt;- proportion_data %&gt;%
  filter(Response_Group %in% c(&quot;DOX-specific response (0.5)&quot;, &quot;DOX only mid-late response (0.5)&quot;, 
                               &quot;CX DOX (early) response (0.5)&quot;, &quot;DOX + CX (mid-late) response (0.5)&quot;)) %&gt;%
  group_by(Response_Group) %&gt;%
  summarise(
    p_value = {
      group_counts &lt;- Count[DNA_Damage %in% c(&quot;Yes&quot;, &quot;No&quot;)]
      if (!&quot;Yes&quot; %in% DNA_Damage) group_counts &lt;- c(group_counts, 0)
      if (!&quot;No&quot; %in% DNA_Damage) group_counts &lt;- c(0, group_counts)

      contingency_table &lt;- matrix(c(
        group_counts[1], group_counts[2],  # Response group counts
        non_response_counts[&quot;Yes&quot;], non_response_counts[&quot;No&quot;]  # Non-response counts
      ), nrow = 2, byrow = TRUE)

      # Perform chi-square test if all values are valid
      if (all(contingency_table &gt;= 0 &amp; is.finite(contingency_table))) {
        chisq.test(contingency_table)$p.value
      } else {
        NA
      }
    },
    .groups = &quot;drop&quot;
  ) %&gt;%
  mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))

# **🔹 Merge Chi-Square Results into Proportion Data**
proportion_data &lt;- proportion_data %&gt;%
  left_join(chi_results %&gt;% dplyr::select(Response_Group, Significance), by = &quot;Response_Group&quot;)

# **🔹 Set Star Position Uniform Across Groups at 105%**
star_positions &lt;- data.frame(
  Response_Group = c(&quot;DOX-specific response (0.5)&quot;, &quot;DOX only mid-late response (0.5)&quot;, 
                     &quot;CX DOX (early) response (0.5)&quot;, &quot;DOX + CX (mid-late) response (0.5)&quot;),
  y_pos = 105,  # Fixed at 105% of Y-axis
  Significance = chi_results$Significance
)

# **🔹 Generate Proportion Plot with Chi-Square Stars**
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = DNA_Damage)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +  # Stacked bars
  geom_text(
    data = star_positions,
    aes(x = Response_Group, y = y_pos, label = Significance),  # Place stars at fixed 105%
    inherit.aes = FALSE,
    size = 6, color = &quot;black&quot;, fontface = &quot;bold&quot;, vjust = 0  # Keeps stars aligned
  ) +
  scale_y_continuous(labels = scales::percent_format(scale = 1), limits = c(0, 110)) +  # **Y-axis now limited to 110% for visibility**
  scale_fill_manual(values = c(&quot;Yes&quot; = &quot;#e41a1c&quot;, &quot;No&quot; = &quot;#377eb8&quot;)) +  # Yes (Red), No (Blue)
  labs(
    title = &quot;Proportion of DNA Damage Repair Genes in\n0.5 Corrmotif Response Groups&quot;,
    x = &quot;Response Groups (0.5 Concentration)&quot;,
    y = &quot;Percentage&quot;,
    fill = &quot;DNA Damage Repair Genes&quot;
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
    legend.title = element_blank(),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    strip.background = element_blank(),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/prop%20DNA%200.5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-prop-DNA-0.5-1">
Past versions of "prop DNA 0.5-1.png"
</button>
</p>
<div id="fig-no-spaces-prop-DNA-0.5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/96d0db145cf68e6db0d39e70bad9af0f75793aae/docs/figure/Corrmotif_Conc.Rmd/prop DNA 0.5-1.png" target="_blank">96d0db1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="proportion-of-p53-target-genes-genes-0.1-micromolar"
class="section level2">
<h2><strong>📌 Proportion of P53 target Genes Genes (0.1
micromolar)</strong></h2>
<pre class="r"><code># Load necessary libraries
library(dplyr)
library(ggplot2)
library(tidyr)
library(org.Hs.eg.db)

# **🔹 Read P53 Target Gene List**
P53_Target &lt;- read.csv(&quot;data/P53_Target.csv&quot;, stringsAsFactors = FALSE)

# Convert gene symbols to Entrez IDs
P53_Target &lt;- P53_Target %&gt;%
  mutate(Entrez_ID = mapIds(org.Hs.eg.db,
                            keys = P53_Target$Symbol,
                            column = &quot;ENTREZID&quot;,
                            keytype = &quot;SYMBOL&quot;,
                            multiVals = &quot;first&quot;))

P53_Target_genes &lt;- na.omit(P53_Target$Entrez_ID)

# **🔹 Load Corrmotif Groups for 0.1 Concentration**
prob_groups_0.1 &lt;- list(
  &quot;Non Response (0.1)&quot; = read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID,
  &quot;CX_DOX mid-late (0.1)&quot; = read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID,
 &quot;DOX only mid-late (0.1)&quot;= read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID
)

# **🔹 Create Dataframe for Corrmotif Groups**
corrmotif_df_0.1 &lt;- bind_rows(
  lapply(prob_groups_0.1, function(ids) {
    data.frame(Entrez_ID = ids)
  }),
  .id = &quot;Response_Group&quot;
)

# **🔹 Match Entrez_IDs with P53 Target Genes**
corrmotif_df_0.1 &lt;- corrmotif_df_0.1 %&gt;%
  mutate(P53_Target = ifelse(Entrez_ID %in% P53_Target_genes, &quot;Yes&quot;, &quot;No&quot;))

# **🔹 Count P53 Target Genes in Each Response Group**
proportion_data &lt;- corrmotif_df_0.1 %&gt;%
  group_by(Response_Group, P53_Target) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Response_Group) %&gt;%
  mutate(Percentage = (Count / sum(Count)) * 100)

# **🔹 Ensure &quot;Yes&quot; is at the Bottom and &quot;No&quot; at the Top**
proportion_data$P53_Target &lt;- factor(proportion_data$P53_Target, levels = c(&quot;Yes&quot;, &quot;No&quot;))

# **🔹 Set Order of Response Groups for X-axis**
response_order &lt;- c(&quot;Non Response (0.1)&quot;, &quot;CX_DOX mid-late (0.1)&quot;, &quot;DOX only mid-late (0.1)&quot;)
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = response_order)

# **🔹 Perform Chi-Square Tests for &quot;DOX only mid-late (0.1)&quot; and &quot;CX_DOX mid-late (0.1)&quot; vs &quot;Non Response (0.1)&quot;**
non_response_counts &lt;- proportion_data %&gt;%
  filter(Response_Group == &quot;Non Response (0.1)&quot;) %&gt;%
  dplyr::select(P53_Target, Count) %&gt;%
  {setNames(.$Count, .$P53_Target)}  # Convert to named vector

chi_results &lt;- proportion_data %&gt;%
  filter(Response_Group %in% c(&quot;CX_DOX mid-late (0.1)&quot;,&quot;DOX only mid-late (0.1)&quot;)) %&gt;%
  group_by(Response_Group) %&gt;%
  summarise(
    p_value = {
      group_counts &lt;- Count[P53_Target %in% c(&quot;Yes&quot;, &quot;No&quot;)]
      if (!&quot;Yes&quot; %in% P53_Target) group_counts &lt;- c(group_counts, 0)
      if (!&quot;No&quot; %in% P53_Target) group_counts &lt;- c(0, group_counts)

      contingency_table &lt;- matrix(c(
        group_counts[1], group_counts[2],
        non_response_counts[&quot;Yes&quot;], non_response_counts[&quot;No&quot;]
      ), nrow = 2, byrow = TRUE)

      # Perform chi-square test if all values are valid
      if (all(contingency_table &gt;= 0 &amp; is.finite(contingency_table))) {
        chisq.test(contingency_table)$p.value
      } else {
        NA
      }
    },
    .groups = &quot;drop&quot;
  ) %&gt;%
  mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))

# **🔹 Merge Chi-Square Results into Proportion Data**
proportion_data &lt;- proportion_data %&gt;%
  left_join(chi_results %&gt;% dplyr::select(Response_Group, Significance), by = &quot;Response_Group&quot;)

# **🔹 Set Star Position Uniform Across Groups at 105%**
star_positions &lt;- data.frame(
  Response_Group = c(&quot;CX_DOX mid-late (0.1)&quot;,&quot;DOX only mid-late (0.1)&quot;),
  y_pos = 105,  # Fixed at 105% of Y-axis
  Significance = chi_results$Significance
)

# **🔹 Generate Proportion Plot with Chi-Square Stars**
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = P53_Target)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +  # Stacked bars
  geom_text(
    data = star_positions,
    aes(x = Response_Group, y = y_pos, label = Significance),  # Place stars at fixed 105%
    inherit.aes = FALSE,
    size = 6, color = &quot;black&quot;, fontface = &quot;bold&quot;, vjust = 0  # Keeps stars aligned
  ) +
  scale_y_continuous(labels = scales::percent_format(scale = 1), limits = c(0, 110)) +  # Fixed Y-axis to 100%
  scale_fill_manual(values = c(&quot;Yes&quot; = &quot;#e41a1c&quot;, &quot;No&quot; = &quot;#377eb8&quot;)) +  # Yes (Red), No (Blue)
  labs(
    title = &quot;Proportion of P53 Target Genes in\n0.1 Corrmotif Response Groups&quot;,
    x = &quot;Response Groups (0.1 Concentration)&quot;,
    y = &quot;Percentage&quot;,
    fill = &quot;P53 Target Genes&quot;
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
    legend.title = element_blank(),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    strip.background = element_blank(),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/prop%20P53%200.1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-prop-P53-0.1-1">
Past versions of "prop P53 0.1-1.png"
</button>
</p>
<div id="fig-no-spaces-prop-P53-0.1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/81247f64f08df43c2b40a386707fc8e9cc240143/docs/figure/Corrmotif_Conc.Rmd/prop P53 0.1-1.png" target="_blank">81247f6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/96d0db145cf68e6db0d39e70bad9af0f75793aae/docs/figure/Corrmotif_Conc.Rmd/prop P53 0.1-1.png" target="_blank">96d0db1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="proportion-of-p53-target-genes-0.5-micromolar"
class="section level2">
<h2><strong>📌 Proportion of P53 target Genes (0.5 micromolar)
</strong></h2>
<pre class="r"><code># Load necessary libraries
library(dplyr)
library(ggplot2)
library(tidyr)
library(org.Hs.eg.db)

# **🔹 Read P53 Target Gene List**
P53_Target &lt;- read.csv(&quot;data/P53_Target.csv&quot;, stringsAsFactors = FALSE)

# Convert gene symbols to Entrez IDs
P53_Target &lt;- P53_Target %&gt;%
  mutate(Entrez_ID = mapIds(org.Hs.eg.db,
                            keys = P53_Target$Symbol,
                            column = &quot;ENTREZID&quot;,
                            keytype = &quot;SYMBOL&quot;,
                            multiVals = &quot;first&quot;))

P53_Target_genes &lt;- na.omit(P53_Target$Entrez_ID)

# **🔹 Load Corrmotif Groups for 0.5 Concentration**
prob_groups_0.5 &lt;- list(
  &quot;Non Response (0.5)&quot; = read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID,
  &quot;DOX-specific response (0.5)&quot; = read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID,
  &quot;DOX only mid-late response (0.5)&quot; = read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID,
  &quot;CX DOX (early) response (0.5)&quot; = read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID,
  &quot;DOX + CX (mid-late) response (0.5)&quot; = read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID
)

# **🔹 Create Dataframe for Corrmotif Groups**
corrmotif_df_0.5 &lt;- bind_rows(
  lapply(prob_groups_0.5, function(ids) {
    data.frame(Entrez_ID = ids)
  }),
  .id = &quot;Response_Group&quot;
)

# **🔹 Match Entrez_IDs with P53 Target Genes**
corrmotif_df_0.5 &lt;- corrmotif_df_0.5 %&gt;%
  mutate(P53_Target = ifelse(Entrez_ID %in% P53_Target_genes, &quot;Yes&quot;, &quot;No&quot;))

# **🔹 Count P53 Target Genes in Each Response Group**
proportion_data &lt;- corrmotif_df_0.5 %&gt;%
  group_by(Response_Group, P53_Target) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Response_Group) %&gt;%
  mutate(Percentage = (Count / sum(Count)) * 100)

# **🔹 Ensure &quot;Yes&quot; is at the Bottom and &quot;No&quot; at the Top**
proportion_data$P53_Target &lt;- factor(proportion_data$P53_Target, levels = c(&quot;Yes&quot;, &quot;No&quot;))

# **🔹 Set Order of Response Groups for X-axis**
response_order &lt;- c(&quot;Non Response (0.5)&quot;, &quot;DOX-specific response (0.5)&quot;, &quot;DOX only mid-late response (0.5)&quot;, 
                    &quot;CX DOX (early) response (0.5)&quot;, &quot;DOX + CX (mid-late) response (0.5)&quot;)
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = response_order)

# **🔹 Perform Chi-Square Tests for Each Response Group vs Non-Response**
non_response_counts &lt;- proportion_data %&gt;%
  filter(Response_Group == &quot;Non Response (0.5)&quot;) %&gt;%
  dplyr::select(P53_Target, Count) %&gt;%
  {setNames(.$Count, .$P53_Target)}  # Convert to named vector

# **Comparing Each Group Against &quot;Non Response (0.5)&quot;**
chi_results &lt;- proportion_data %&gt;%
  filter(Response_Group %in% c(&quot;DOX-specific response (0.5)&quot;, &quot;DOX only mid-late response (0.5)&quot;, 
                               &quot;CX DOX (early) response (0.5)&quot;, &quot;DOX + CX (mid-late) response (0.5)&quot;)) %&gt;%
  group_by(Response_Group) %&gt;%
  summarise(
    p_value = {
      group_counts &lt;- Count[P53_Target %in% c(&quot;Yes&quot;, &quot;No&quot;)]
      if (!&quot;Yes&quot; %in% P53_Target) group_counts &lt;- c(group_counts, 0)
      if (!&quot;No&quot; %in% P53_Target) group_counts &lt;- c(0, group_counts)

      contingency_table &lt;- matrix(c(
        group_counts[1], group_counts[2],  # Response group counts
        non_response_counts[&quot;Yes&quot;], non_response_counts[&quot;No&quot;]  # Non-response counts
      ), nrow = 2, byrow = TRUE)

      # Perform chi-square test if all values are valid
      if (all(contingency_table &gt;= 0 &amp; is.finite(contingency_table))) {
        chisq.test(contingency_table)$p.value
      } else {
        NA
      }
    },
    .groups = &quot;drop&quot;
  ) %&gt;%
  mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))</code></pre>
<pre><code>Warning: There was 1 warning in `summarise()`.
ℹ In argument: `p_value = { ... }`.
ℹ In group 3: `Response_Group = CX DOX (early) response (0.5)`.
Caused by warning in `chisq.test()`:
! Chi-squared approximation may be incorrect</code></pre>
<pre class="r"><code># **🔹 Merge Chi-Square Results into Proportion Data**
proportion_data &lt;- proportion_data %&gt;%
  left_join(chi_results %&gt;% dplyr::select(Response_Group, Significance), by = &quot;Response_Group&quot;)

# **🔹 Set Star Position Uniform Across Groups at 105%**
star_positions &lt;- data.frame(
  Response_Group = c(&quot;DOX-specific response (0.5)&quot;, &quot;DOX only mid-late response (0.5)&quot;, 
                     &quot;CX DOX (early) response (0.5)&quot;, &quot;DOX + CX (mid-late) response (0.5)&quot;),
  y_pos = 105,  # Fixed at 105% of Y-axis
  Significance = chi_results$Significance
)

# **🔹 Generate Proportion Plot with Chi-Square Stars**
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = P53_Target)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +  # Stacked bars
  geom_text(
    data = star_positions,
    aes(x = Response_Group, y = y_pos, label = Significance),  # Place stars at fixed 105%
    inherit.aes = FALSE,
    size = 6, color = &quot;black&quot;, fontface = &quot;bold&quot;, vjust = 0  # Keeps stars aligned
  ) +
  scale_y_continuous(labels = scales::percent_format(scale = 1), limits = c(0, 110)) +  # **Y-axis now limited to 110% for visibility**
  scale_fill_manual(values = c(&quot;Yes&quot; = &quot;#e41a1c&quot;, &quot;No&quot; = &quot;#377eb8&quot;)) +  # Yes (Red), No (Blue)
  labs(
    title = &quot;Proportion of P53 Target Genes in\n0.5 Corrmotif Response Groups&quot;,
    x = &quot;Response Groups (0.5 Concentration)&quot;,
    y = &quot;Percentage&quot;,
    fill = &quot;P53 Target Genes&quot;
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
    legend.title = element_blank(),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    strip.background = element_blank(),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/prop%20P53%200.5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-prop-P53-0.5-1">
Past versions of "prop P53 0.5-1.png"
</button>
</p>
<div id="fig-no-spaces-prop-P53-0.5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/96d0db145cf68e6db0d39e70bad9af0f75793aae/docs/figure/Corrmotif_Conc.Rmd/prop P53 0.5-1.png" target="_blank">96d0db1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 22631)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] org.Hs.eg.db_3.18.0  AnnotationDbi_1.64.1 IRanges_2.36.0      
 [4] S4Vectors_0.40.1     tidyr_1.3.1          ggplot2_3.5.1       
 [7] gprofiler2_0.2.3     BiocParallel_1.36.0  dplyr_1.1.4         
[10] Rfast_2.1.0          RcppParallel_5.1.9   RcppZiggurat_0.1.6  
[13] Rcpp_1.0.12          Cormotif_1.48.0      limma_3.58.1        
[16] affy_1.80.0          Biobase_2.62.0       BiocGenerics_0.48.1 
[19] workflowr_1.7.1     

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.1        viridisLite_0.4.2       farver_2.1.2           
 [4] blob_1.2.4              bitops_1.0-7            Biostrings_2.70.1      
 [7] RCurl_1.98-1.13         fastmap_1.1.1           lazyeval_0.2.2         
[10] promises_1.3.0          digest_0.6.34           lifecycle_1.0.4        
[13] statmod_1.5.0           processx_3.8.5          KEGGREST_1.42.0        
[16] RSQLite_2.3.3           magrittr_2.0.3          compiler_4.3.0         
[19] rlang_1.1.3             sass_0.4.9              tools_4.3.0            
[22] yaml_2.3.10             data.table_1.14.10      knitr_1.49             
[25] labeling_0.4.3          htmlwidgets_1.6.4       bit_4.0.5              
[28] withr_3.0.2             purrr_1.0.2             grid_4.3.0             
[31] preprocessCore_1.64.0   git2r_0.35.0            colorspace_2.1-0       
[34] scales_1.3.0            cli_3.6.1               crayon_1.5.3           
[37] rmarkdown_2.29          generics_0.1.3          rstudioapi_0.17.1      
[40] httr_1.4.7              DBI_1.2.3               cachem_1.0.8           
[43] stringr_1.5.1           zlibbioc_1.48.0         parallel_4.3.0         
[46] XVector_0.42.0          BiocManager_1.30.25     vctrs_0.6.5            
[49] jsonlite_1.8.9          callr_3.7.6             bit64_4.0.5            
[52] plotly_4.10.4           jquerylib_0.1.4         affyio_1.72.0          
[55] glue_1.7.0              codetools_0.2-20        ps_1.8.1               
[58] stringi_1.8.3           gtable_0.3.6            GenomeInfoDb_1.38.8    
[61] later_1.3.2             munsell_0.5.1           tibble_3.2.1           
[64] pillar_1.10.1           htmltools_0.5.8.1       GenomeInfoDbData_1.2.11
[67] R6_2.5.1                rprojroot_2.0.4         evaluate_1.0.3         
[70] png_0.1-8               memoise_2.0.1           httpuv_1.6.15          
[73] bslib_0.8.0             whisker_0.4.1           xfun_0.50              
[76] fs_1.6.3                getPass_0.2-4           pkgconfig_2.0.3        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
