<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Corrmotif Conc Analysis</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CX5461_Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Corrmotif Conc Analysis</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-05-15
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>CX5461_Project/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree01a423feca180bf7e8669c85a3c03e54357c1b38targetblank01a423fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/01a423feca180bf7e8669c85a3c03e54357c1b38" target="_blank">01a423f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree01a423feca180bf7e8669c85a3c03e54357c1b38targetblank01a423fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/01a423feca180bf7e8669c85a3c03e54357c1b38" target="_blank">01a423f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    0.1 box.svg
    Ignored:    Rplot04.svg
    Ignored:    analysis/DGE_Analysis.html

Untracked files:
    Untracked:  0.1 density.svg
    Untracked:  0.1.emf
    Untracked:  0.1.svg
    Untracked:  0.5 box.svg
    Untracked:  0.5 density.svg
    Untracked:  0.5.svg
    Untracked:  Additional/
    Untracked:  CX_5461_Pattern_Genes_24hr.csv
    Untracked:  CX_5461_Pattern_Genes_3hr.csv
    Untracked:  Cell viability box plot.svg
    Untracked:  DRC1.svg
    Untracked:  Figure 1.jpeg
    Untracked:  Figure 1.pdf
    Untracked:  Figure_CM_Purity.pdf
    Untracked:  Rplot.svg
    Untracked:  Rplot01.svg
    Untracked:  Rplot02.svg
    Untracked:  Rplot03.svg
    Untracked:  Rplot05.svg
    Untracked:  Rplot06.svg
    Untracked:  Rplot07.svg
    Untracked:  Rplot08.jpeg
    Untracked:  Rplot08.svg
    Untracked:  Rplot09.svg
    Untracked:  Rplot10.svg
    Untracked:  Rplot11.svg
    Untracked:  TS HPA (Violin).svg
    Untracked:  TS HPA.svg
    Untracked:  TS_HA.svg
    Untracked:  TS_HV.svg
    Untracked:  Violin HA.svg
    Untracked:  Violin HV (CX vs DOX).svg
    Untracked:  Violin HV.svg
    Untracked:  data/AF.csv
    Untracked:  data/AF_Mapped.csv
    Untracked:  data/AF_genes.csv
    Untracked:  data/Annotated_DOX_Gene_Table.csv
    Untracked:  data/CAD_genes.csv
    Untracked:  data/Cardiotox.csv
    Untracked:  data/Cardiotox_mapped.csv
    Untracked:  data/DOX_Vald.csv
    Untracked:  data/DOX_Vald_Mapped.csv
    Untracked:  data/Entrez_Cardiotox.csv
    Untracked:  data/Entrez_Cardiotox_Mapped.csv
    Untracked:  data/GWAS.xlsx
    Untracked:  data/GWAS_SNPs.bed
    Untracked:  data/HF.csv
    Untracked:  data/HF_Mapped.csv
    Untracked:  data/HF_genes.csv
    Untracked:  data/Hypertension_genes.csv
    Untracked:  data/MI_genes.csv
    Untracked:  data/P53_Target_mapped.csv
    Untracked:  data/Sample_annotated.csv
    Untracked:  data/Samples.csv
    Untracked:  data/Samples.xlsx
    Untracked:  data/TOP2A.bed
    Untracked:  data/TOP2A_target.csv
    Untracked:  data/TOP2A_target_lit.csv
    Untracked:  data/TOP2A_target_lit_mapped.csv
    Untracked:  data/TOP2A_target_mapped.csv
    Untracked:  data/TOP2B_target.csv
    Untracked:  data/TOP2B_target_heatmap.csv
    Untracked:  data/TOP2B_target_heatmap_mapped.csv
    Untracked:  data/TOP2B_target_mapped.csv
    Untracked:  data/TS.csv
    Untracked:  data/TS_HPA.csv
    Untracked:  data/TS_HPA_mapped.csv
    Untracked:  data/Toptable_CX_0.1_24.csv
    Untracked:  data/Toptable_CX_0.1_3.csv
    Untracked:  data/Toptable_CX_0.1_48.csv
    Untracked:  data/Toptable_CX_0.5_24.csv
    Untracked:  data/Toptable_CX_0.5_3.csv
    Untracked:  data/Toptable_CX_0.5_48.csv
    Untracked:  data/Toptable_DOX_0.1_24.csv
    Untracked:  data/Toptable_DOX_0.1_3.csv
    Untracked:  data/Toptable_DOX_0.1_48.csv
    Untracked:  data/Toptable_DOX_0.5_24.csv
    Untracked:  data/Toptable_DOX_0.5_3.csv
    Untracked:  data/Toptable_DOX_0.5_48.csv
    Untracked:  data/count.tsv
    Untracked:  data/ts_data_mapped
    Untracked:  results/
    Untracked:  run_bedtools.bat

Unstaged changes:
    Deleted:    analysis/Actox.Rmd
    Modified:   data/count.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Corrmotif_Conc.Rmd</code>) and
HTML (<code>docs/Corrmotif_Conc.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/analysis/Corrmotif_Conc.Rmd" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/Corrmotif_Conc.html" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/1661001828f9b43008c4958fd25156f288811169/analysis/Corrmotif_Conc.Rmd" target="_blank">1661001</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/1661001828f9b43008c4958fd25156f288811169/docs/Corrmotif_Conc.html" target="_blank">1661001</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/b9a286768620bdf4d87f2a63ddc247e5316445d0/analysis/Corrmotif_Conc.Rmd" target="_blank">b9a2867</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-05
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/b9a286768620bdf4d87f2a63ddc247e5316445d0/docs/Corrmotif_Conc.html" target="_blank">b9a2867</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-05
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/10ab0f3506d3aaa233e23ec14f7671cce1563e28/analysis/Corrmotif_Conc.Rmd" target="_blank">10ab0f3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-05
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/10ab0f3506d3aaa233e23ec14f7671cce1563e28/docs/Corrmotif_Conc.html" target="_blank">10ab0f3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-05
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/da5bc51ec43f9e4e8d0c046a5388f15106eaa1c2/analysis/Corrmotif_Conc.Rmd" target="_blank">da5bc51</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-04
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/da5bc51ec43f9e4e8d0c046a5388f15106eaa1c2/docs/Corrmotif_Conc.html" target="_blank">da5bc51</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-04
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/6f9e594c722182678bafc5ddb7c9cdeb3f27dfbf/analysis/Corrmotif_Conc.Rmd" target="_blank">6f9e594</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-24
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/6f9e594c722182678bafc5ddb7c9cdeb3f27dfbf/docs/Corrmotif_Conc.html" target="_blank">6f9e594</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-24
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d6226a09c4a6b3b6c2280e9c4f41ba17acb41045/analysis/Corrmotif_Conc.Rmd" target="_blank">d6226a0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-24
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/d6226a09c4a6b3b6c2280e9c4f41ba17acb41045/docs/Corrmotif_Conc.html" target="_blank">d6226a0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-24
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/062a73abf4aa51016b5a82ec16868e5950354cdf/analysis/Corrmotif_Conc.Rmd" target="_blank">062a73a</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-24
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/062a73abf4aa51016b5a82ec16868e5950354cdf/docs/Corrmotif_Conc.html" target="_blank">062a73a</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-24
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/07b9670bf922129e5d7a932d22711efc033fde34/docs/Corrmotif_Conc.html" target="_blank">07b9670</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-21
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/9618a00ad12263a31ed79f64344c82a444c6d5c1/docs/Corrmotif_Conc.html" target="_blank">9618a00</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-21
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/008117ab97e9c8b972ffcc1bad9c90b0946274be/docs/Corrmotif_Conc.html" target="_blank">008117a</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/1433d3784ab1f1829ee095be49b4a2de6ab57ca5/docs/Corrmotif_Conc.html" target="_blank">1433d37</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-14
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/4bad905ad26e9ebac9188a323b7f5fe10aa56ce6/analysis/Corrmotif_Conc.Rmd" target="_blank">4bad905</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-14
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/4bad905ad26e9ebac9188a323b7f5fe10aa56ce6/docs/Corrmotif_Conc.html" target="_blank">4bad905</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-14
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/6ef5f61785e28f0c6385c185a7d54db2620ea914/analysis/Corrmotif_Conc.Rmd" target="_blank">6ef5f61</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-14
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/6ef5f61785e28f0c6385c185a7d54db2620ea914/docs/Corrmotif_Conc.html" target="_blank">6ef5f61</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-14
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d8a2f25dc1beb5fef19f7a7a03802eb5999393cd/analysis/Corrmotif_Conc.Rmd" target="_blank">d8a2f25</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-14
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/d8a2f25dc1beb5fef19f7a7a03802eb5999393cd/docs/Corrmotif_Conc.html" target="_blank">d8a2f25</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-14
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/a56887202d8df9b0816df4b966d55f223263e84a/analysis/Corrmotif_Conc.Rmd" target="_blank">a568872</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-10
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/a56887202d8df9b0816df4b966d55f223263e84a/docs/Corrmotif_Conc.html" target="_blank">a568872</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-10
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/4e40091300781c5ddf17c486457d658c5619f25d/analysis/Corrmotif_Conc.Rmd" target="_blank">4e40091</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/4e40091300781c5ddf17c486457d658c5619f25d/docs/Corrmotif_Conc.html" target="_blank">4e40091</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/analysis/Corrmotif_Conc.Rmd" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/Corrmotif_Conc.html" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2990b21e59ad6434e6e7b4b92ae6cad8ff37f997/analysis/Corrmotif_Conc.Rmd" target="_blank">2990b21</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-10
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/2990b21e59ad6434e6e7b4b92ae6cad8ff37f997/docs/Corrmotif_Conc.html" target="_blank">2990b21</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-10
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ef3a95166a7a18770eaa7786b80994b40bba9973/analysis/Corrmotif_Conc.Rmd" target="_blank">ef3a951</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-09
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/ef3a95166a7a18770eaa7786b80994b40bba9973/docs/Corrmotif_Conc.html" target="_blank">ef3a951</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-09
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/81247f64f08df43c2b40a386707fc8e9cc240143/analysis/Corrmotif_Conc.Rmd" target="_blank">81247f6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-06
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/81247f64f08df43c2b40a386707fc8e9cc240143/docs/Corrmotif_Conc.html" target="_blank">81247f6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-06
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/96d0db145cf68e6db0d39e70bad9af0f75793aae/analysis/Corrmotif_Conc.Rmd" target="_blank">96d0db1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/96d0db145cf68e6db0d39e70bad9af0f75793aae/docs/Corrmotif_Conc.html" target="_blank">96d0db1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/analysis/Corrmotif_Conc.Rmd" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/Corrmotif_Conc.html" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/analysis/Corrmotif_Conc.Rmd" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/Corrmotif_Conc.html" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2de67e6a82f82edbc78941b11ba72ed8aba9964e/analysis/Corrmotif_Conc.Rmd" target="_blank">2de67e6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/2de67e6a82f82edbc78941b11ba72ed8aba9964e/docs/Corrmotif_Conc.html" target="_blank">2de67e6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/41cd1be487c4309fb8ab2f2d36b18ad63cdc31e4/analysis/Corrmotif_Conc.Rmd" target="_blank">41cd1be</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/41cd1be487c4309fb8ab2f2d36b18ad63cdc31e4/docs/Corrmotif_Conc.html" target="_blank">41cd1be</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/f84821f0421d3805726f2ccfce5b5ca2275c6756/analysis/Corrmotif_Conc.Rmd" target="_blank">f84821f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-26
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/b6e38a14f7f28ac96f0b16857be1f89d71989e58/analysis/Corrmotif_Conc.Rmd" target="_blank">b6e38a1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-25
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/b6e38a14f7f28ac96f0b16857be1f89d71989e58/docs/Corrmotif_Conc.html" target="_blank">b6e38a1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-25
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ce4b32557fc3c7ee693369db9edf0a9d22968203/analysis/Corrmotif_Conc.Rmd" target="_blank">ce4b325</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-25
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/ce4b32557fc3c7ee693369db9edf0a9d22968203/docs/Corrmotif_Conc.html" target="_blank">ce4b325</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-25
</td>
<td>
Commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="micromolar" class="section level2">
<h2><strong>📌 0.1 Micromolar</strong></h2>
</div>
<div id="fit-limma-model-functions" class="section level2">
<h2><strong>📌 Fit Limma Model Functions</strong></h2>
<pre class="r"><code>## Fit limma model using code as it is found in the original cormotif code. It has
## only been modified to add names to the matrix of t values, as well as the
## limma fits

limmafit.default &lt;- function(exprs,groupid,compid) {
  limmafits  &lt;- list()
  compnum    &lt;- nrow(compid)
  genenum    &lt;- nrow(exprs)
  limmat     &lt;- matrix(0,genenum,compnum)
  limmas2    &lt;- rep(0,compnum)
  limmadf    &lt;- rep(0,compnum)
  limmav0    &lt;- rep(0,compnum)
  limmag1num &lt;- rep(0,compnum)
  limmag2num &lt;- rep(0,compnum)

  rownames(limmat)  &lt;- rownames(exprs)
  colnames(limmat)  &lt;- rownames(compid)
  names(limmas2)    &lt;- rownames(compid)
  names(limmadf)    &lt;- rownames(compid)
  names(limmav0)    &lt;- rownames(compid)
  names(limmag1num) &lt;- rownames(compid)
  names(limmag2num) &lt;- rownames(compid)

  for(i in 1:compnum) {
    selid1 &lt;- which(groupid == compid[i,1])
    selid2 &lt;- which(groupid == compid[i,2])
    eset   &lt;- new(&quot;ExpressionSet&quot;, exprs=cbind(exprs[,selid1],exprs[,selid2]))
    g1num  &lt;- length(selid1)
    g2num  &lt;- length(selid2)
    designmat &lt;- cbind(base=rep(1,(g1num+g2num)), delta=c(rep(0,g1num),rep(1,g2num)))
    fit &lt;- lmFit(eset,designmat)
    fit &lt;- eBayes(fit)
    limmat[,i] &lt;- fit$t[,2]
    limmas2[i] &lt;- fit$s2.prior
    limmadf[i] &lt;- fit$df.prior
    limmav0[i] &lt;- fit$var.prior[2]
    limmag1num[i] &lt;- g1num
    limmag2num[i] &lt;- g2num
    limmafits[[i]] &lt;- fit

    # log odds
    # w&lt;-sqrt(1+fit$var.prior[2]/(1/g1num+1/g2num))
    # log(0.99)+dt(fit$t[1,2],g1num+g2num-2+fit$df.prior,log=TRUE)-log(0.01)-dt(fit$t[1,2]/w, g1num+g2num-2+fit$df.prior, log=TRUE)+log(w)
  }
  names(limmafits) &lt;- rownames(compid)
  limmacompnum&lt;-nrow(compid)
  result&lt;-list(t       = limmat,
               v0      = limmav0,
               df0     = limmadf,
               s20     = limmas2,
               g1num   = limmag1num,
               g2num   = limmag2num,
               compnum = limmacompnum,
               fits    = limmafits)
}

limmafit.counts &lt;-
  function (exprs, groupid, compid, norm.factor.method = &quot;TMM&quot;, voom.normalize.method = &quot;none&quot;)
  {
    limmafits  &lt;- list()
    compnum    &lt;- nrow(compid)
    genenum    &lt;- nrow(exprs)
    limmat     &lt;- matrix(NA,genenum,compnum)
    limmas2    &lt;- rep(0,compnum)
    limmadf    &lt;- rep(0,compnum)
    limmav0    &lt;- rep(0,compnum)
    limmag1num &lt;- rep(0,compnum)
    limmag2num &lt;- rep(0,compnum)

    rownames(limmat)  &lt;- rownames(exprs)
    colnames(limmat)  &lt;- rownames(compid)
    names(limmas2)    &lt;- rownames(compid)
    names(limmadf)    &lt;- rownames(compid)
    names(limmav0)    &lt;- rownames(compid)
    names(limmag1num) &lt;- rownames(compid)
    names(limmag2num) &lt;- rownames(compid)

    for (i in 1:compnum) {
      message(paste(&quot;Running limma for comparision&quot;,i,&quot;/&quot;,compnum))
      selid1 &lt;- which(groupid == compid[i, 1])
      selid2 &lt;- which(groupid == compid[i, 2])
      # make a new count data frame
      counts &lt;- cbind(exprs[, selid1], exprs[, selid2])

      # remove NAs
      not.nas &lt;- which(apply(counts, 1, function(x) !any(is.na(x))) == TRUE)

      # runn voom/limma
      d &lt;- DGEList(counts[not.nas,])
      d &lt;- calcNormFactors(d, method = norm.factor.method)
      g1num &lt;- length(selid1)
      g2num &lt;- length(selid2)
      designmat &lt;- cbind(base = rep(1, (g1num + g2num)), delta = c(rep(0,
                                                                       g1num), rep(1, g2num)))

      y &lt;- voom(d, designmat, normalize.method = voom.normalize.method)
      fit &lt;- lmFit(y, designmat)
      fit &lt;- eBayes(fit)

      limmafits[[i]] &lt;- fit
      limmat[not.nas, i] &lt;- fit$t[, 2]
      limmas2[i] &lt;- fit$s2.prior
      limmadf[i] &lt;- fit$df.prior
      limmav0[i] &lt;- fit$var.prior[2]
      limmag1num[i] &lt;- g1num
      limmag2num[i] &lt;- g2num
    }
    limmacompnum &lt;- nrow(compid)
    names(limmafits) &lt;- rownames(compid)
    result &lt;- list(t       = limmat,
                   v0      = limmav0,
                   df0     = limmadf,
                   s20     = limmas2,
                   g1num   = limmag1num,
                   g2num   = limmag2num,
                   compnum = limmacompnum,
                   fits    = limmafits)
  }

limmafit.list &lt;-
  function (fitlist, cmp.idx=2)
  {
    compnum    &lt;- length(fitlist)

    genes &lt;- c()
    for (i in 1:compnum) genes &lt;- unique(c(genes, rownames(fitlist[[i]])))

    genenum    &lt;- length(genes)
    limmat     &lt;- matrix(NA,genenum,compnum)
    limmas2    &lt;- rep(0,compnum)
    limmadf    &lt;- rep(0,compnum)
    limmav0    &lt;- rep(0,compnum)
    limmag1num &lt;- rep(0,compnum)
    limmag2num &lt;- rep(0,compnum)

    rownames(limmat)  &lt;- genes
    colnames(limmat)  &lt;- names(fitlist)
    names(limmas2)    &lt;- names(fitlist)
    names(limmadf)    &lt;- names(fitlist)
    names(limmav0)    &lt;- names(fitlist)
    names(limmag1num) &lt;- names(fitlist)
    names(limmag2num) &lt;- names(fitlist)

    for (i in 1:compnum) {
      this.t &lt;- fitlist[[i]]$t[,cmp.idx]
      limmat[names(this.t),i] &lt;- this.t

      limmas2[i]    &lt;- fitlist[[i]]$s2.prior
      limmadf[i]    &lt;- fitlist[[i]]$df.prior
      limmav0[i]    &lt;- fitlist[[i]]$var.prior[cmp.idx]
      limmag1num[i] &lt;- sum(fitlist[[i]]$design[,cmp.idx]==0)
      limmag2num[i] &lt;- sum(fitlist[[i]]$design[,cmp.idx]==1)
    }

    limmacompnum &lt;- compnum
    result &lt;- list(t       = limmat,
                   v0      = limmav0,
                   df0     = limmadf,
                   s20     = limmas2,
                   g1num   = limmag1num,
                   g2num   = limmag2num,
                   compnum = limmacompnum,
                   fits    = limmafits)

  }

## Rank genes based on statistics
generank&lt;-function(x) {
  xcol&lt;-ncol(x)
  xrow&lt;-nrow(x)
  result&lt;-matrix(0,xrow,xcol)
  z&lt;-(1:1:xrow)
  for(i in 1:xcol) {
    y&lt;-sort(x[,i],decreasing=TRUE,na.last=TRUE)
    result[,i]&lt;-match(x[,i],y)
    result[,i]&lt;-order(result[,i])
  }
  result
}

## Log-likelihood for moderated t under H0
modt.f0.loglike&lt;-function(x,df) {
  a&lt;-dt(x, df, log=TRUE)
  result&lt;-as.vector(a)
  flag&lt;-which(is.na(result)==TRUE)
  result[flag]&lt;-0
  result
}

## Log-likelihood for moderated t under H1
## param=c(df,g1num,g2num,v0)
modt.f1.loglike&lt;-function(x,param) {
  df&lt;-param[1]
  g1num&lt;-param[2]
  g2num&lt;-param[3]
  v0&lt;-param[4]
  w&lt;-sqrt(1+v0/(1/g1num+1/g2num))
  dt(x/w, df, log=TRUE)-log(w)
  a&lt;-dt(x/w, df, log=TRUE)-log(w)
  result&lt;-as.vector(a)
  flag&lt;-which(is.na(result)==TRUE)
  result[flag]&lt;-0
  result
}

## Correlation Motif Fit
cmfit.X&lt;-function(x, type, K=1, tol=1e-3, max.iter=100) {
  ## initialize
  xrow &lt;- nrow(x)
  xcol &lt;- ncol(x)
  loglike0 &lt;- list()
  loglike1 &lt;- list()
  p &lt;- rep(1, K)/K
  q &lt;- matrix(runif(K * xcol), K, xcol)
  q[1, ] &lt;- rep(0.01, xcol)
  for (i in 1:xcol) {
    f0 &lt;- type[[i]][[1]]
    f0param &lt;- type[[i]][[2]]
    f1 &lt;- type[[i]][[3]]
    f1param &lt;- type[[i]][[4]]
    loglike0[[i]] &lt;- f0(x[, i], f0param)
    loglike1[[i]] &lt;- f1(x[, i], f1param)
  }
  condlike &lt;- list()
  for (i in 1:xcol) {
    condlike[[i]] &lt;- matrix(0, xrow, K)
  }
  loglike.old &lt;- -1e+10
  for (i.iter in 1:max.iter) {
    if ((i.iter%%50) == 0) {
      print(paste(&quot;We have run the first &quot;, i.iter, &quot; iterations for K=&quot;,
                  K, sep = &quot;&quot;))
    }
    err &lt;- tol + 1
    clustlike &lt;- matrix(0, xrow, K)
    #templike &lt;- matrix(0, xrow, 2)
    templike1 &lt;- rep(0, xrow)
    templike2 &lt;- rep(0, xrow)
    for (j in 1:K) {
      for (i in 1:xcol) {
        templike1 &lt;- log(q[j, i]) + loglike1[[i]]
        templike2 &lt;- log(1 - q[j, i]) + loglike0[[i]]
        tempmax &lt;- Rfast::Pmax(templike1, templike2)

        templike1 &lt;- exp(templike1 - tempmax)
        templike2 &lt;- exp(templike2 - tempmax)

        tempsum &lt;- templike1 + templike2
        clustlike[, j] &lt;- clustlike[, j] + tempmax +
          log(tempsum)
        condlike[[i]][, j] &lt;- templike1/tempsum
      }
      clustlike[, j] &lt;- clustlike[, j] + log(p[j])
    }
    #tempmax &lt;- apply(clustlike, 1, max)
    tempmax &lt;- Rfast::rowMaxs(clustlike, value=TRUE)
    for (j in 1:K) {
      clustlike[, j] &lt;- exp(clustlike[, j] - tempmax)
    }
    #tempsum &lt;- apply(clustlike, 1, sum)
    tempsum &lt;- Rfast::rowsums(clustlike)
    for (j in 1:K) {
      clustlike[, j] &lt;- clustlike[, j]/tempsum
    }
    #p.new &lt;- (apply(clustlike, 2, sum) + 1)/(xrow + K)
    p.new &lt;- (Rfast::colsums(clustlike) + 1)/(xrow + K)
    q.new &lt;- matrix(0, K, xcol)
    for (j in 1:K) {
      clustpsum &lt;- sum(clustlike[, j])
      for (i in 1:xcol) {
        q.new[j, i] &lt;- (sum(clustlike[, j] * condlike[[i]][,
                                                           j]) + 1)/(clustpsum + 2)
      }
    }
    err.p &lt;- max(abs(p.new - p)/p)
    err.q &lt;- max(abs(q.new - q)/q)
    err &lt;- max(err.p, err.q)
    loglike.new &lt;- (sum(tempmax + log(tempsum)) + sum(log(p.new)) +
                      sum(log(q.new) + log(1 - q.new)))/xrow
    p &lt;- p.new
    q &lt;- q.new
    loglike.old &lt;- loglike.new
    if (err &lt; tol) {
      break
    }
  }
  clustlike &lt;- matrix(0, xrow, K)
  for (j in 1:K) {
    for (i in 1:xcol) {
      templike1 &lt;- log(q[j, i]) + loglike1[[i]]
      templike2 &lt;- log(1 - q[j, i]) + loglike0[[i]]
      tempmax &lt;- Rfast::Pmax(templike1, templike2)

      templike1 &lt;- exp(templike1 - tempmax)
      templike2 &lt;- exp(templike2 - tempmax)

      tempsum &lt;- templike1 + templike2
      clustlike[, j] &lt;- clustlike[, j] + tempmax + log(tempsum)
      condlike[[i]][, j] &lt;- templike1/tempsum
    }
    clustlike[, j] &lt;- clustlike[, j] + log(p[j])
  }
  #tempmax &lt;- apply(clustlike, 1, max)
  tempmax &lt;- Rfast::rowMaxs(clustlike, value=TRUE)
  for (j in 1:K) {
    clustlike[, j] &lt;- exp(clustlike[, j] - tempmax)
  }
  #tempsum &lt;- apply(clustlike, 1, sum)
  tempsum &lt;- Rfast::rowsums(clustlike)
  for (j in 1:K) {
    clustlike[, j] &lt;- clustlike[, j]/tempsum
  }
  p.post &lt;- matrix(0, xrow, xcol)
  for (j in 1:K) {
    for (i in 1:xcol) {
      p.post[, i] &lt;- p.post[, i] + clustlike[, j] * condlike[[i]][,
                                                                  j]
    }
  }
  loglike.old &lt;- loglike.old - (sum(log(p)) + sum(log(q) +
                                                    log(1 - q)))/xrow
  loglike.old &lt;- loglike.old * xrow
  result &lt;- list(p.post = p.post, motif.prior = p, motif.q = q,
                 loglike = loglike.old, clustlike=clustlike, condlike=condlike)
}

## Fit using (0,0,...,0) and (1,1,...,1)
cmfitall&lt;-function(x, type, tol=1e-3, max.iter=100) {
  ## initialize
  xrow&lt;-nrow(x)
  xcol&lt;-ncol(x)
  loglike0&lt;-list()
  loglike1&lt;-list()
  p&lt;-0.01

  ## compute loglikelihood
  L0&lt;-matrix(0,xrow,1)
  L1&lt;-matrix(0,xrow,1)
  for(i in 1:xcol) {
    f0&lt;-type[[i]][[1]]
    f0param&lt;-type[[i]][[2]]
    f1&lt;-type[[i]][[3]]
    f1param&lt;-type[[i]][[4]]
    loglike0[[i]]&lt;-f0(x[,i],f0param)
    loglike1[[i]]&lt;-f1(x[,i],f1param)
    L0&lt;-L0+loglike0[[i]]
    L1&lt;-L1+loglike1[[i]]
  }


  ## EM algorithm to get MLE of p and q
  loglike.old &lt;- -1e10
  for(i.iter in 1:max.iter) {
    if((i.iter%%50) == 0) {
      print(paste(&quot;We have run the first &quot;, i.iter, &quot; iterations&quot;,sep=&quot;&quot;))
    }
    err&lt;-tol+1

    ## compute posterior cluster membership
    clustlike&lt;-matrix(0,xrow,2)
    clustlike[,1]&lt;-log(1-p)+L0
    clustlike[,2]&lt;-log(p)+L1

    tempmax&lt;-apply(clustlike,1,max)
    for(j in 1:2) {
      clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
    }
    tempsum&lt;-apply(clustlike,1,sum)

    ## update motif occurrence rate
    for(j in 1:2) {
      clustlike[,j]&lt;-clustlike[,j]/tempsum
    }

    p.new&lt;-(sum(clustlike[,2])+1)/(xrow+2)

    ## evaluate convergence
    err&lt;-abs(p.new-p)/p

    ## evaluate whether the log.likelihood increases
    loglike.new&lt;-(sum(tempmax+log(tempsum))+log(p.new)+log(1-p.new))/xrow

    loglike.old&lt;-loglike.new
    p&lt;-p.new

    if(err&lt;tol) {
      break;
    }
  }

  ## compute posterior p
  clustlike&lt;-matrix(0,xrow,2)
  clustlike[,1]&lt;-log(1-p)+L0
  clustlike[,2]&lt;-log(p)+L1

  tempmax&lt;-apply(clustlike,1,max)
  for(j in 1:2) {
    clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
  }
  tempsum&lt;-apply(clustlike,1,sum)

  for(j in 1:2) {
    clustlike[,j]&lt;-clustlike[,j]/tempsum
  }

  p.post&lt;-matrix(0,xrow,xcol)
  for(i in 1:xcol) {
    p.post[,i]&lt;-clustlike[,2]
  }

  ## return

  #calculate back loglikelihood
  loglike.old&lt;-loglike.old-(log(p)+log(1-p))/xrow
  loglike.old&lt;-loglike.old*xrow
  result&lt;-list(p.post=p.post, motif.prior=p, loglike=loglike.old)
}

## Fit each dataset separately
cmfitsep&lt;-function(x, type, tol=1e-3, max.iter=100) {
  ## initialize
  xrow&lt;-nrow(x)
  xcol&lt;-ncol(x)
  loglike0&lt;-list()
  loglike1&lt;-list()
  p&lt;-0.01*rep(1,xcol)
  loglike.final&lt;-rep(0,xcol)

  ## compute loglikelihood
  for(i in 1:xcol) {
    f0&lt;-type[[i]][[1]]
    f0param&lt;-type[[i]][[2]]
    f1&lt;-type[[i]][[3]]
    f1param&lt;-type[[i]][[4]]
    loglike0[[i]]&lt;-f0(x[,i],f0param)
    loglike1[[i]]&lt;-f1(x[,i],f1param)
  }

  p.post&lt;-matrix(0,xrow,xcol)

  ## EM algorithm to get MLE of p
  for(coli in 1:xcol) {
    loglike.old &lt;- -1e10
    for(i.iter in 1:max.iter) {
      if((i.iter%%50) == 0) {
        print(paste(&quot;We have run the first &quot;, i.iter, &quot; iterations&quot;,sep=&quot;&quot;))
      }
      err&lt;-tol+1

      ## compute posterior cluster membership
      clustlike&lt;-matrix(0,xrow,2)
      clustlike[,1]&lt;-log(1-p[coli])+loglike0[[coli]]
      clustlike[,2]&lt;-log(p[coli])+loglike1[[coli]]

      tempmax&lt;-apply(clustlike,1,max)
      for(j in 1:2) {
        clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
      }
      tempsum&lt;-apply(clustlike,1,sum)

      ## evaluate whether the log.likelihood increases
      loglike.new&lt;-sum(tempmax+log(tempsum))/xrow

      ## update motif occurrence rate
      for(j in 1:2) {
        clustlike[,j]&lt;-clustlike[,j]/tempsum
      }

      p.new&lt;-(sum(clustlike[,2]))/(xrow)

      ## evaluate convergence
      err&lt;-abs(p.new-p[coli])/p[coli]
      loglike.old&lt;-loglike.new
      p[coli]&lt;-p.new

      if(err&lt;tol) {
        break;
      }
    }

    ## compute posterior p
    clustlike&lt;-matrix(0,xrow,2)
    clustlike[,1]&lt;-log(1-p[coli])+loglike0[[coli]]
    clustlike[,2]&lt;-log(p[coli])+loglike1[[coli]]

    tempmax&lt;-apply(clustlike,1,max)
    for(j in 1:2) {
      clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
    }
    tempsum&lt;-apply(clustlike,1,sum)

    for(j in 1:2) {
      clustlike[,j]&lt;-clustlike[,j]/tempsum
    }

    p.post[,coli]&lt;-clustlike[,2]
    loglike.final[coli]&lt;-loglike.old
  }


  ## return
  loglike.final&lt;-loglike.final*xrow
  result&lt;-list(p.post=p.post, motif.prior=p, loglike=loglike.final)
}

## Fit the full model
cmfitfull&lt;-function(x, type, tol=1e-3, max.iter=100) {
  ## initialize
  xrow&lt;-nrow(x)
  xcol&lt;-ncol(x)
  loglike0&lt;-list()
  loglike1&lt;-list()
  K&lt;-2^xcol
  p&lt;-rep(1,K)/K
  pattern&lt;-rep(0,xcol)
  patid&lt;-matrix(0,K,xcol)

  ## compute loglikelihood
  for(i in 1:xcol) {
    f0&lt;-type[[i]][[1]]
    f0param&lt;-type[[i]][[2]]
    f1&lt;-type[[i]][[3]]
    f1param&lt;-type[[i]][[4]]
    loglike0[[i]]&lt;-f0(x[,i],f0param)
    loglike1[[i]]&lt;-f1(x[,i],f1param)
  }
  L&lt;-matrix(0,xrow,K)
  for(i in 1:K)
  {
    patid[i,]&lt;-pattern
    for(j in 1:xcol) {
      if(pattern[j] &lt; 0.5) {
        L[,i]&lt;-L[,i]+loglike0[[j]]
      } else {
        L[,i]&lt;-L[,i]+loglike1[[j]]
      }
    }

    if(i &lt; K) {
      pattern[xcol]&lt;-pattern[xcol]+1
      j&lt;-xcol
      while(pattern[j] &gt; 1) {
        pattern[j]&lt;-0
        j&lt;-j-1
        pattern[j]&lt;-pattern[j]+1
      }
    }
  }

  ## EM algorithm to get MLE of p and q
  loglike.old &lt;- -1e10
  for(i.iter in 1:max.iter) {
    if((i.iter%%50) == 0) {
      print(paste(&quot;We have run the first &quot;, i.iter, &quot; iterations&quot;,sep=&quot;&quot;))
    }
    err&lt;-tol+1

    ## compute posterior cluster membership
    clustlike&lt;-matrix(0,xrow,K)
    for(j in 1:K) {
      clustlike[,j]&lt;-log(p[j])+L[,j]
    }

    tempmax&lt;-apply(clustlike,1,max)
    for(j in 1:K) {
      clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
    }
    tempsum&lt;-apply(clustlike,1,sum)

    ## update motif occurrence rate
    for(j in 1:K) {
      clustlike[,j]&lt;-clustlike[,j]/tempsum
    }

    p.new&lt;-(apply(clustlike,2,sum)+1)/(xrow+K)

    ## evaluate convergence
    err&lt;-max(abs(p.new-p)/p)

    ## evaluate whether the log.likelihood increases
    loglike.new&lt;-(sum(tempmax+log(tempsum))+sum(log(p.new)))/xrow

    loglike.old&lt;-loglike.new
    p&lt;-p.new

    if(err&lt;tol) {
      break;
    }
  }

  ## compute posterior p
  clustlike&lt;-matrix(0,xrow,K)
  for(j in 1:K) {
    clustlike[,j]&lt;-log(p[j])+L[,j]
  }

  tempmax&lt;-apply(clustlike,1,max)
  for(j in 1:K) {
    clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
  }
  tempsum&lt;-apply(clustlike,1,sum)

  for(j in 1:K) {
    clustlike[,j]&lt;-clustlike[,j]/tempsum
  }

  p.post&lt;-matrix(0,xrow,xcol)
  for(j in 1:K) {
    for(i in 1:xcol) {
      if(patid[j,i] &gt; 0.5) {
        p.post[,i]&lt;-p.post[,i]+clustlike[,j]
      }
    }
  }

  ## return
  #calculate back loglikelihood
  loglike.old&lt;-loglike.old-sum(log(p))/xrow
  loglike.old&lt;-loglike.old*xrow
  result&lt;-list(p.post=p.post, motif.prior=p, loglike=loglike.old)
}

generatetype&lt;-function(limfitted)
{
  jtype&lt;-list()
  df&lt;-limfitted$g1num+limfitted$g2num-2+limfitted$df0
  for(j in 1:limfitted$compnum)
  {
    jtype[[j]]&lt;-list(f0=modt.f0.loglike, f0.param=df[j], f1=modt.f1.loglike, f1.param=c(df[j],limfitted$g1num[j],limfitted$g2num[j],limfitted$v0[j]))
  }
  jtype
}

cormotiffit &lt;- function(exprs, groupid=NULL, compid=NULL, K=1, tol=1e-3,
                        max.iter=100, BIC=TRUE, norm.factor.method=&quot;TMM&quot;,
                        voom.normalize.method = &quot;none&quot;, runtype=c(&quot;logCPM&quot;,&quot;counts&quot;,&quot;limmafits&quot;), each=3)
{
  # first I want to do some typechecking. Input can be either a normalized
  # matrix, a count matrix, or a list of limma fits. Dispatch the correct
  # limmafit accordingly.
  # todo: add some typechecking here
  limfitted &lt;- list()
  if (runtype==&quot;counts&quot;) {
    limfitted &lt;- limmafit.counts(exprs,groupid,compid, norm.factor.method, voom.normalize.method)
  } else if (runtype==&quot;logCPM&quot;) {
    limfitted &lt;- limmafit.default(exprs,groupid,compid)
  } else if (runtype==&quot;limmafits&quot;) {
    limfitted &lt;- limmafit.list(exprs)
  } else {
    stop(&quot;runtype must be one of &#39;logCPM&#39;, &#39;counts&#39;, or &#39;limmafits&#39;&quot;)
  }


  jtype&lt;-generatetype(limfitted)
  fitresult&lt;-list()
  ks &lt;- rep(K, each = each)
  fitresult &lt;- bplapply(1:length(ks), function(i, x, type, ks, tol, max.iter) {
    cmfit.X(x, type, K = ks[i], tol = tol, max.iter = max.iter)
  }, x=limfitted$t, type=jtype, ks=ks, tol=tol, max.iter=max.iter)

  best.fitresults &lt;- list()
  for (i in 1:length(K)) {
    w.k &lt;- which(ks==K[i])
    this.bic &lt;- c()
    for (j in w.k) this.bic[j] &lt;- -2 * fitresult[[j]]$loglike + (K[i] - 1 + K[i] * limfitted$compnum) * log(dim(limfitted$t)[1])
    w.min &lt;- which(this.bic == min(this.bic, na.rm = TRUE))[1]
    best.fitresults[[i]] &lt;- fitresult[[w.min]]
  }
  fitresult &lt;- best.fitresults

  bic &lt;- rep(0, length(K))
  aic &lt;- rep(0, length(K))
  loglike &lt;- rep(0, length(K))
  for (i in 1:length(K)) loglike[i] &lt;- fitresult[[i]]$loglike
  for (i in 1:length(K)) bic[i] &lt;- -2 * fitresult[[i]]$loglike + (K[i] - 1 + K[i] * limfitted$compnum) * log(dim(limfitted$t)[1])
  for (i in 1:length(K)) aic[i] &lt;- -2 * fitresult[[i]]$loglike + 2 * (K[i] - 1 + K[i] * limfitted$compnum)
  if(BIC==TRUE) {
    bestflag=which(bic==min(bic))
  }
  else {
    bestflag=which(aic==min(aic))
  }
  result&lt;-list(bestmotif=fitresult[[bestflag]],bic=cbind(K,bic),
               aic=cbind(K,aic),loglike=cbind(K,loglike), allmotifs=fitresult)

}

cormotiffitall&lt;-function(exprs,groupid,compid, tol=1e-3, max.iter=100)
{
  limfitted&lt;-limmafit(exprs,groupid,compid)
  jtype&lt;-generatetype(limfitted)
  fitresult&lt;-cmfitall(limfitted$t,type=jtype,tol=1e-3,max.iter=max.iter)
}

cormotiffitsep&lt;-function(exprs,groupid,compid, tol=1e-3, max.iter=100)
{
  limfitted&lt;-limmafit(exprs,groupid,compid)
  jtype&lt;-generatetype(limfitted)
  fitresult&lt;-cmfitsep(limfitted$t,type=jtype,tol=1e-3,max.iter=max.iter)
}

cormotiffitfull&lt;-function(exprs,groupid,compid, tol=1e-3, max.iter=100)
{
  limfitted&lt;-limmafit(exprs,groupid,compid)
  jtype&lt;-generatetype(limfitted)
  fitresult&lt;-cmfitfull(limfitted$t,type=jtype,tol=1e-3,max.iter=max.iter)
}

plotIC&lt;-function(fitted_cormotif)
{
  oldpar&lt;-par(mfrow=c(1,2))
  plot(fitted_cormotif$bic[,1], fitted_cormotif$bic[,2], type=&quot;b&quot;,xlab=&quot;Motif Number&quot;, ylab=&quot;BIC&quot;, main=&quot;BIC&quot;)
  plot(fitted_cormotif$aic[,1], fitted_cormotif$aic[,2], type=&quot;b&quot;,xlab=&quot;Motif Number&quot;, ylab=&quot;AIC&quot;, main=&quot;AIC&quot;)
}

plotMotif&lt;-function(fitted_cormotif,title=&quot;&quot;)
{
  layout(matrix(1:2,ncol=2))
  u&lt;-1:dim(fitted_cormotif$bestmotif$motif.q)[2]
  v&lt;-1:dim(fitted_cormotif$bestmotif$motif.q)[1]
  image(u,v,t(fitted_cormotif$bestmotif$motif.q),
        col=gray(seq(from=1,to=0,by=-0.1)),xlab=&quot;Study&quot;,yaxt = &quot;n&quot;,
        ylab=&quot;Corr. Motifs&quot;,main=paste(title,&quot;pattern&quot;,sep=&quot; &quot;))
  axis(2,at=1:length(v))
  for(i in 1:(length(u)+1))
  {
    abline(v=(i-0.5))
  }
  for(i in 1:(length(v)+1))
  {
    abline(h=(i-0.5))
  }
  Ng=10000
  if(is.null(fitted_cormotif$bestmotif$p.post)!=TRUE)
    Ng=nrow(fitted_cormotif$bestmotif$p.post)
  genecount=floor(fitted_cormotif$bestmotif$motif.p*Ng)
  NK=nrow(fitted_cormotif$bestmotif$motif.q)
  plot(0,0.7,pch=&quot;.&quot;,xlim=c(0,1.2),ylim=c(0.75,NK+0.25),
       frame.plot=FALSE,axes=FALSE,xlab=&quot;No. of genes&quot;,ylab=&quot;&quot;, main=paste(title,&quot;frequency&quot;,sep=&quot; &quot;))
  segments(0,0.7,fitted_cormotif$bestmotif$motif.p[1],0.7)
  rect(0,1:NK-0.3,fitted_cormotif$bestmotif$motif.p,1:NK+0.3,
       col=&quot;dark grey&quot;)
  mtext(1:NK,at=1:NK,side=2,cex=0.8)
  text(fitted_cormotif$bestmotif$motif.p+0.15,1:NK,
       labels=floor(fitted_cormotif$bestmotif$motif.p*Ng))
}</code></pre>
</div>
<div id="load-required-libraries" class="section level2">
<h2><strong>📌 Load Required Libraries</strong></h2>
<pre class="r"><code>library(Cormotif)
library(Rfast)
library(dplyr)
library(BiocParallel)
library(gprofiler2)
library(ggplot2)</code></pre>
</div>
<div id="corrmotif-model-0.1-micromolar" class="section level2">
<h2><strong>📌 Corrmotif Model 0.1 Micromolar</strong></h2>
</div>
<div id="load-corrmotif-data" class="section level2">
<h2><strong>📌 Load Corrmotif Data</strong></h2>
<pre class="r"><code># Read the Corrmotif Results
Corrmotif &lt;- read.csv(&quot;data/Corrmotif/CX5461.csv&quot;)
Corrmotif_df &lt;- data.frame(Corrmotif)
rownames(Corrmotif_df) &lt;- Corrmotif_df$Gene

# Filter for 0.1 Concentration Only
exprs.corrmotif &lt;- as.matrix(Corrmotif_df[, grep(&quot;0.1&quot;, colnames(Corrmotif_df))])


# Read group and comparison IDs
groupid &lt;- read.csv(&quot;data/Corrmotif/groupid.csv&quot;)
groupid_df &lt;- data.frame(groupid[, grep(&quot;0.1&quot;, colnames(groupid))])


compid &lt;- read.csv(&quot;data/Corrmotif/Compid.csv&quot;)
compid_df &lt;- compid[compid$Cond1 %in% unique(as.numeric(groupid_df)) &amp; compid$Cond2 %in% unique(as.numeric(groupid_df)), ]</code></pre>
</div>
<div id="corrmotif-model-0.1-micromolar-k18" class="section level2">
<h2><strong>📌 Corrmotif Model 0.1 Micromolar (K=1:8)</strong></h2>
</div>
<div id="fit-corrmotif-model-k18-0.1-micromolar" class="section level2">
<h2><strong>📌 Fit Corrmotif Model (K=1:8) (0.1
Micromolar)</strong></h2>
<pre class="r"><code>set.seed(11111)
# Fit Corrmotif Model (K = 1 to 8)
set.seed(11111)
motif.fitted_0.1 &lt;- cormotiffit(
  exprs = exprs.corrmotif,
  groupid = groupid_df,
  compid = compid_df,
  K = 1:8,
  max.iter = 1000,
  BIC = TRUE,
  runtype = &quot;logCPM&quot;
)

gene_prob_0.1 &lt;- motif.fitted_0.1$bestmotif$p.post
rownames(gene_prob_0.1) &lt;- rownames(Corrmotif_df)
motif_prob_0.1 &lt;- motif.fitted_0.1$bestmotif$clustlike
rownames(motif_prob_0.1) &lt;- rownames(gene_prob_0.1)
write.csv(motif_prob_0.1,&quot;data/cormotif_probability_genelist_0.1.csv&quot;)</code></pre>
</div>
<div id="plot-motif-0.1-micromolar" class="section level2">
<h2><strong>📌 Plot motif (0.1 Micromolar)</strong></h2>
<pre class="r"><code>cormotif_0.1 &lt;- readRDS(&quot;data/Corrmotif/cormotif_0.1.RDS&quot;)

cormotif_0.1$bic</code></pre>
<pre><code>     K      bic
[1,] 1 291696.5
[2,] 2 284585.3
[3,] 3 283482.9
[4,] 4 283551.8
[5,] 5 283620.7
[6,] 6 283689.6
[7,] 7 283758.5
[8,] 8 283827.4</code></pre>
<pre class="r"><code>plotIC(cormotif_0.1)</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Model1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Model1-1">
Past versions of Model1-1.png
</button>
</p>
<div id="fig-Model1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/Model1-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/41cd1be487c4309fb8ab2f2d36b18ad63cdc31e4/docs/figure/Corrmotif_Conc.Rmd/Model1-1.png" target="_blank">41cd1be</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ce4b32557fc3c7ee693369db9edf0a9d22968203/docs/figure/Corrmotif_Conc.Rmd/Model1-1.png" target="_blank">ce4b325</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plotMotif(cormotif_0.1)</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Model1-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Model1-2">
Past versions of Model1-2.png
</button>
</p>
<div id="fig-Model1-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/Model1-2.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/41cd1be487c4309fb8ab2f2d36b18ad63cdc31e4/docs/figure/Corrmotif_Conc.Rmd/Model1-2.png" target="_blank">41cd1be</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ce4b32557fc3c7ee693369db9edf0a9d22968203/docs/figure/Corrmotif_Conc.Rmd/Model1-2.png" target="_blank">ce4b325</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="extract-gene-probabilities-0.1-micromolar"
class="section level2">
<h2><strong>📌 Extract Gene Probabilities (0.1 Micromolar)</strong></h2>
<pre class="r"><code># Extract posterior probabilities for genes
gene_prob_tran_0.1 &lt;- cormotif_0.1$bestmotif$p.post

rownames(gene_prob_tran_0.1) &lt;- rownames(Corrmotif_df)

# Define gene probability groups
prob_1_0.1  &lt;- rownames(gene_prob_tran_0.1[(gene_prob_tran_0.1[,1] &lt;0.5 &amp; gene_prob_tran_0.1[,2] &lt;0.5 &amp; gene_prob_tran_0.1[,3] &lt;0.5 &amp; gene_prob_tran_0.1[,4] &lt;0.5 &amp; gene_prob_tran_0.1[,5] &lt; 0.5 &amp; gene_prob_tran_0.1[,6]&lt;0.5),])
length(prob_1_0.1)</code></pre>
<pre><code>[1] 12308</code></pre>
<pre class="r"><code>prob_2_0.1  &lt;- rownames(gene_prob_tran_0.1[(gene_prob_tran_0.1[,1] &lt;0.5 &amp; gene_prob_tran_0.1[,2] &gt;0.5 &amp; gene_prob_tran_0.1[,3] &gt;0.5 &amp; gene_prob_tran_0.1[,4] &lt;0.5 &amp; gene_prob_tran_0.1[,5] &gt; 0.5 &amp; gene_prob_tran_0.1[,6]&gt;0.5),])
length(prob_2_0.1)</code></pre>
<pre><code>[1] 415</code></pre>
<pre class="r"><code>prob_3_0.1  &lt;- rownames(gene_prob_tran_0.1[(gene_prob_tran_0.1[,1] &lt;0.5 &amp; gene_prob_tran_0.1[,2] &lt;0.5 &amp; gene_prob_tran_0.1[,3] &lt;0.5 &amp; gene_prob_tran_0.1[,4] &lt;0.5 &amp; gene_prob_tran_0.1[,5] &gt; 0.5 &amp; gene_prob_tran_0.1[,6]&gt;0.5),])
length(prob_3_0.1)</code></pre>
<pre><code>[1] 1551</code></pre>
</div>
<div
id="distribution-of-gene-clusters-identified-by-corrmotif-0.1-micromolar"
class="section level2">
<h2><strong>📌 Distribution of Gene Clusters Identified by Corrmotif
(0.1 micromolar)</strong></h2>
<pre class="r"><code># Load necessary library
library(ggplot2)

# Data
data &lt;- data.frame(
  Category = c(&quot;Non response (0.1 µM)&quot;, &quot;CX-DOX mid-late response (0.1 µM)&quot;, &quot;DOX only mid-late (0.1 µM)&quot;),
  Value = c(12308, 415, 1551)
)

# Define custom colors
custom_colors &lt;- c(&quot;Non response (0.1 µM)&quot; = &quot;#FF9999&quot;,
                   &quot;CX-DOX mid-late response (0.1 µM)&quot; = &quot;#66B2FF&quot;,
                   &quot;DOX only mid-late (0.1 µM)&quot; = &quot;#99FF99&quot;)

# Create pie chart
ggplot(data, aes(x = &quot;&quot;, y = Value, fill = Category)) +
  geom_bar(width = 1, stat = &quot;identity&quot;) +
  coord_polar(&quot;y&quot;, start = 0) +
  geom_text(aes(label = Value),
            position = position_stack(vjust = 0.5),
            size = 4, color = &quot;black&quot;) +
  labs(title = &quot;Pie Chart (0.1 micromolar Corrmotif)&quot;, x = NULL, y = NULL) +
  theme_void() +
  scale_fill_manual(values = custom_colors)</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/pie-chart_0.1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pie-chart_0.1-1">
Past versions of pie-chart_0.1-1.png
</button>
</p>
<div id="fig-pie-chart_0.1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/pie-chart_0.1-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2de67e6a82f82edbc78941b11ba72ed8aba9964e/docs/figure/Corrmotif_Conc.Rmd/pie-chart_0.1-1.png" target="_blank">2de67e6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="save-corr-motif-datasets-for-gene-ontology-analysis-0.1-micromolar"
class="section level2">
<h2><strong>📌 Save Corr-motif datasets for Gene Ontology analysis (0.1
Micromolar)</strong></h2>
<pre class="r"><code>write.csv(data.frame(Entrez_ID = prob_1_0.1), &quot;data/prob_1_0.1.csv&quot;, row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_2_0.1), &quot;data/prob_2_0.1.csv&quot;, row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_3_0.1), &quot;data/prob_3_0.1.csv&quot;, row.names = FALSE)</code></pre>
</div>
<div id="micromolar-abs-logfc" class="section level2">
<h2><strong>📌 0.1 Micromolar abs logFC</strong></h2>
<pre class="r"><code># Load Required Libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# Load Datasets (Only 0.1 Micromolar)
CX_0.1_3  &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;)
CX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;)
CX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;)

DOX_0.1_3  &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;)
DOX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;)
DOX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;)

# Combine All 0.1 Micromolar Datasets into a Single Dataframe
all_toptables_0.1 &lt;- bind_rows(
  CX_0.1_3  %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  CX_0.1_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  CX_0.1_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;),
  DOX_0.1_3  %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  DOX_0.1_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  DOX_0.1_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;)
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.1$Entrez_ID &lt;- as.character(all_toptables_0.1$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.1 &lt;- all_toptables_0.1 %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 micromolar)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response\n(0.1 micromolar)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response\n(0.1 micromolar)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group))

# Compute Absolute logFC
all_toptables_0.1 &lt;- all_toptables_0.1 %&gt;%
  mutate(absFC = abs(logFC))

# Convert Factors for Proper Ordering (Reversed Order for Response Groups)
all_toptables_0.1 &lt;- all_toptables_0.1 %&gt;%
  mutate(
    Drug = factor(Drug, levels = c(&quot;CX.5461&quot;, &quot;DOX&quot;)),
    Timepoint = factor(Timepoint, levels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Response_Group = factor(Response_Group,
                            levels = c(
                              &quot;DOX only mid-late response\n(0.1 micromolar)&quot;,
                              &quot;CX-DOX mid-late response\n(0.1 micromolar)&quot;,
                              &quot;Non response\n(0.1 micromolar)&quot;  # Reversed Order
                            ))
  )

# **Plot the Boxplot with Faceted Labels Wrapping Correctly**
ggplot(all_toptables_0.1, aes(x = Drug, y = absFC, fill = Drug)) +
  geom_boxplot() +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +  # Custom color palette
  facet_grid(Response_Group ~ Timepoint, labeller = label_wrap_gen(width = 20)) +  # Ensure Proper Wrapping
  theme_bw() +
  labs(
    x = &quot;Drugs&quot;,
    y = &quot;|Log Fold Change|&quot;,
    title = &quot;|Log Fold| for 0.1 micromolar&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.line = element_line(linewidth = 1.5),
    strip.background = element_rect(fill = &quot;gray&quot;),  # Gray background for facet labels
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),  # Bold styling for facet labels
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 15)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.1abs%20logFC-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-0.1abs-logFC-1">
Past versions of "0.1abs logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-0.1abs-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/0.1abs logFC-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2990b21e59ad6434e6e7b4b92ae6cad8ff37f997/docs/figure/Corrmotif_Conc.Rmd/0.1abs logFC-1.png" target="_blank">2990b21</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.1abs logFC-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.1abs logFC-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="micromolar-mean-abs-logfc-across-timepoints"
class="section level2">
<h2><strong>📌 0.1 Micromolar mean (Abs logFC) across
timepoints</strong></h2>
<pre class="r"><code># Load Required Libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# Load Datasets (Only 0.1 Micromolar)
CX_0.1_3  &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;)
CX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;)
CX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;)

DOX_0.1_3  &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;)
DOX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;)
DOX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;)

# Combine All 0.1 Micromolar Datasets into a Single Dataframe
all_toptables_0.1 &lt;- bind_rows(
  CX_0.1_3  %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;3&quot;),
  CX_0.1_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;24&quot;),
  CX_0.1_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;48&quot;),
  DOX_0.1_3  %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;3&quot;),
  DOX_0.1_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;24&quot;),
  DOX_0.1_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;48&quot;)
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.1$Entrez_ID &lt;- as.character(all_toptables_0.1$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.1 &lt;- all_toptables_0.1 %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 micromolar)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response\n(0.1 micromolar)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response\n(0.1 micromolar)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group))

# Compute Mean Absolute logFC for Line Plot
data_summary &lt;- all_toptables_0.1 %&gt;%
  mutate(abs_logFC = abs(logFC)) %&gt;%  # Take absolute log fold change
  group_by(Response_Group, Drug, Timepoint) %&gt;%
  dplyr::summarize(mean_abs_logFC = mean(abs_logFC, na.rm = TRUE), .groups = &quot;drop&quot;) %&gt;%
  as.data.frame()

# **Ensure all timepoints exist in the summary**
timepoints_full &lt;- expand.grid(
  Response_Group = unique(all_toptables_0.1$Response_Group),
  Drug = unique(all_toptables_0.1$Drug),
  Timepoint = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;)
)

# **Merge to keep missing timepoints**
data_summary &lt;- full_join(timepoints_full, data_summary, by = c(&quot;Response_Group&quot;, &quot;Drug&quot;, &quot;Timepoint&quot;))

# **Replace NA mean_abs_logFC with 0 if no genes were present**
data_summary$mean_abs_logFC[is.na(data_summary$mean_abs_logFC)] &lt;- 0

# Convert Factors for Proper Ordering (Reversed Order for Response Groups)
data_summary &lt;- data_summary %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;), labels = c(&quot;3 hours&quot;, &quot;24 hours&quot;, &quot;48 hours&quot;)),
    Response_Group = factor(Response_Group, levels = c(
      &quot;DOX only mid-late response\n(0.1 micromolar)&quot;,
      &quot;CX-DOX mid-late response\n(0.1 micromolar)&quot;,
      &quot;Non response\n(0.1 micromolar)&quot;  # Reversed Order
    ))
  )

# Define custom drug palette
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)

# **Plot the Line Plot for Absolute logFC**
ggplot(data_summary, aes(x = Timepoint, y = mean_abs_logFC, group = Drug, color = Drug)) +
  geom_point(size = 3) +
  geom_line(size = 1.2) +
  scale_color_manual(values = drug_palette) +
  ylim(0, 2.5) +  # Adjust the Y-axis for better visualization
  facet_wrap(~ Response_Group, ncol = 1) +  # Facet by Response Group (Reversed Order)
  theme_bw() +
  labs(
    x = &quot;Timepoints&quot;,
    y = &quot;Mean |Log Fold Change|&quot;,
    title = &quot;Mean |Log Fold Change| Across Response Groups (0.1 micromolar)&quot;,
    color = &quot;Drug&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text = element_text(size = 12, color = &quot;black&quot;),
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.1%20mean-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-0.1-mean-1">
Past versions of "0.1 mean-1.png"
</button>
</p>
<div id="fig-no-spaces-0.1-mean-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/0.1 mean-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2990b21e59ad6434e6e7b4b92ae6cad8ff37f997/docs/figure/Corrmotif_Conc.Rmd/0.1 mean-1.png" target="_blank">2990b21</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.1 mean-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.1 mean-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="micromolar-logfc" class="section level2">
<h2><strong>📌 0.1 Micromolar logFC</strong></h2>
<pre class="r"><code># Load required libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# Load Datasets (Only 0.1 Micromolar)
CX_0.1_3  &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;)
CX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;)
CX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;)

DOX_0.1_3  &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;)
DOX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;)
DOX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;)

# Combine All 0.1 Micromolar Datasets into a Single Dataframe
all_toptables_0.1 &lt;- bind_rows(
  CX_0.1_3  %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  CX_0.1_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  CX_0.1_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;),
  DOX_0.1_3  %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  DOX_0.1_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  DOX_0.1_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;)
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.1$Entrez_ID &lt;- as.character(all_toptables_0.1$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.1 &lt;- all_toptables_0.1 %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 micromolar)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response\n(0.1 micromolar)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response\n(0.1 micromolar)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group))

# Convert factors to ensure correct ordering (Reversed Order for Response Groups)
all_toptables_0.1 &lt;- all_toptables_0.1 %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Response_Group = factor(Response_Group, levels = c(
      &quot;DOX only mid-late response\n(0.1 micromolar)&quot;,
      &quot;CX-DOX mid-late response\n(0.1 micromolar)&quot;,
      &quot;Non response\n(0.1 micromolar)&quot;  # Reversed Order
    ))
  )

# **Plot the Boxplot**
ggplot(all_toptables_0.1, aes(x = Drug, y = logFC, fill = Drug)) +
  geom_boxplot() +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  facet_grid(Response_Group ~ Timepoint) +
  theme_bw() +
  labs(x = &quot;Drugs&quot;, y = &quot;Log Fold Change&quot;, title = &quot;Log Fold Change for 0.1 Micromolar&quot;) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.1%20logFC-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-0.1-logFC-1">
Past versions of "0.1 logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-0.1-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/0.1 logFC-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2990b21e59ad6434e6e7b4b92ae6cad8ff37f997/docs/figure/Corrmotif_Conc.Rmd/0.1 logFC-1.png" target="_blank">2990b21</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.1 logFC-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.1 logFC-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="micromolar-mean-logfc-across-timepoints"
class="section level2">
<h2><strong>📌 0.1 Micromolar mean logFC across timepoints</strong></h2>
<pre class="r"><code># Compute Mean logFC for Line Plot
data_summary &lt;- all_toptables_0.1 %&gt;%
  group_by(Response_Group, Drug, Timepoint) %&gt;%
  dplyr::summarize(mean_logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;) %&gt;%
  as.data.frame()

# Convert factors to ensure correct ordering (Reversed Order for Response Groups)
data_summary &lt;- data_summary %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Response_Group = factor(Response_Group, levels = c(
      &quot;DOX only mid-late response\n(0.1 micromolar)&quot;,
      &quot;CX-DOX mid-late response\n(0.1 micromolar)&quot;,
      &quot;Non response\n(0.1 micromolar)&quot;  # Reversed Order
    ))
  )

# **Plot the Line Plot**
ggplot(data_summary, aes(x = Timepoint, y = mean_logFC, group = Drug, color = Drug)) +
  geom_point(size = 3) +
  geom_line(size = 1.2) +
  scale_color_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  ylim(-2, 1.5) +  # Adjust the Y-axis for better visualization
  facet_wrap(~ Response_Group, ncol = 1) +  # Facet by Response Group (Reversed Order)
  theme_bw() +
  labs(
    x = &quot;Timepoints&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    title = &quot;Mean Log Fold Change Across Response Groups (0.1 micromolar)&quot;,
    color = &quot;Drug&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text = element_text(size = 12, color = &quot;black&quot;),
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.1%20mean2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-0.1-mean2-1">
Past versions of "0.1 mean2-1.png"
</button>
</p>
<div id="fig-no-spaces-0.1-mean2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/0.1 mean2-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2990b21e59ad6434e6e7b4b92ae6cad8ff37f997/docs/figure/Corrmotif_Conc.Rmd/0.1 mean2-1.png" target="_blank">2990b21</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.1 mean2-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.1 mean2-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="corrmotif-model-0.5-micromolar" class="section level2">
<h2><strong>📌 Corrmotif Model 0.5 Micromolar</strong></h2>
</div>
<div id="load-corrmotif-data-1" class="section level2">
<h2><strong>📌 Load Corrmotif Data</strong></h2>
<pre class="r"><code># Read the Corrmotif Results
Corrmotif &lt;- read.csv(&quot;data/Corrmotif/CX5461.csv&quot;)
Corrmotif_df &lt;- data.frame(Corrmotif)
rownames(Corrmotif_df) &lt;- Corrmotif_df$Gene

# Filter for 0.5 Concentration Only
exprs.corrmotif &lt;- as.matrix(Corrmotif_df[, grep(&quot;0.5&quot;, colnames(Corrmotif_df))])


# Read group and comparison IDs
groupid &lt;- read.csv(&quot;data/Corrmotif/groupid.csv&quot;)
groupid_df &lt;- data.frame(groupid[, grep(&quot;0.5&quot;, colnames(groupid))])


compid &lt;- read.csv(&quot;data/Corrmotif/Compid.csv&quot;)
compid_df &lt;- compid[compid$Cond1 %in% unique(as.numeric(groupid_df)) &amp; compid$Cond2 %in% unique(as.numeric(groupid_df)), ]</code></pre>
</div>
<div id="corrmotif-model-0.5-micromolar-k18" class="section level2">
<h2><strong>📌 Corrmotif Model 0.5 Micromolar (K=1:8)</strong></h2>
</div>
<div id="fit-corrmotif-model-k18-0.5-micromolar" class="section level2">
<h2><strong>📌 Fit Corrmotif Model (K=1:8) (0.5
Micromolar)</strong></h2>
<pre class="r"><code># Fit Corrmotif Model (K = 1 to 8)
set.seed(11111)
motif.fitted_0.5 &lt;- cormotiffit(
  exprs = exprs.corrmotif,
  groupid = groupid_df,
  compid = compid_df,
  K = 1:8,
  max.iter = 1000,
  BIC = TRUE,
  runtype = &quot;logCPM&quot;
)

gene_prob_0.5 &lt;- motif.fitted_0.5$bestmotif$p.post
rownames(gene_prob_0.5) &lt;- rownames(Corrmotif_df)
motif_prob_0.5 &lt;- motif.fitted_0.5$bestmotif$clustlike
rownames(motif_prob_0.5) &lt;- rownames(gene_prob_0.5)
write.csv(motif_prob_0.5,&quot;data/cormotif_probability_genelist_0.5.csv&quot;)</code></pre>
</div>
<div id="plot-motif-0.5-micromolar" class="section level2">
<h2><strong>📌 Plot motif (0.5 Micromolar)</strong></h2>
<pre class="r"><code>cormotif_0.5 &lt;- readRDS(&quot;data/Corrmotif/cormotif_0.5.RDS&quot;)

cormotif_0.5$bic</code></pre>
<pre><code>     K      bic
[1,] 1 352140.7
[2,] 2 346785.8
[3,] 3 344812.9
[4,] 4 344860.1
[5,] 5 344751.9
[6,] 6 344820.8
[7,] 7 344889.7
[8,] 8 344966.6</code></pre>
<pre class="r"><code>plotIC(cormotif_0.5)</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Motif_0.5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Motif_0.5-1">
Past versions of Motif_0.5-1.png
</button>
</p>
<div id="fig-Motif_0.5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/Motif_0.5-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/41cd1be487c4309fb8ab2f2d36b18ad63cdc31e4/docs/figure/Corrmotif_Conc.Rmd/Motif_0.5-1.png" target="_blank">41cd1be</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plotMotif(cormotif_0.5)</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Motif_0.5-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Motif_0.5-2">
Past versions of Motif_0.5-2.png
</button>
</p>
<div id="fig-Motif_0.5-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/Motif_0.5-2.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/41cd1be487c4309fb8ab2f2d36b18ad63cdc31e4/docs/figure/Corrmotif_Conc.Rmd/Motif_0.5-2.png" target="_blank">41cd1be</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="extract-gene-probabilities-0.5-micromolar"
class="section level2">
<h2><strong>📌 Extract Gene Probabilities (0.5 Micromolar)</strong></h2>
<pre class="r"><code># Extract posterior probabilities for genes
gene_prob_tran_0.5 &lt;- cormotif_0.5$bestmotif$p.post

rownames(gene_prob_tran_0.5) &lt;- rownames(Corrmotif_df)

# Define gene probability groups
prob_1_0.5  &lt;- rownames(gene_prob_tran_0.5[(gene_prob_tran_0.5[,1] &lt;0.5 &amp; gene_prob_tran_0.5[,2] &lt;0.5 &amp; gene_prob_tran_0.5[,3] &lt;0.5 &amp; gene_prob_tran_0.5[,4] &lt;0.5 &amp; gene_prob_tran_0.5[,5] &lt; 0.5 &amp; gene_prob_tran_0.5[,6]&lt;0.5),])
length(prob_1_0.5)</code></pre>
<pre><code>[1] 7134</code></pre>
<pre class="r"><code>prob_2_0.5  &lt;- rownames(gene_prob_tran_0.5[(gene_prob_tran_0.5[,1] &lt;0.5 &amp; gene_prob_tran_0.5[,2] &lt;0.5 &amp; gene_prob_tran_0.5[,3] &lt;0.5 &amp; gene_prob_tran_0.5[,4] &gt;0.5 &amp; gene_prob_tran_0.5[,5] &gt; 0.5 &amp; gene_prob_tran_0.5[,6]&gt;=0.02),])
length(prob_2_0.5)</code></pre>
<pre><code>[1] 179</code></pre>
<pre class="r"><code>prob_3_0.5  &lt;- rownames(gene_prob_tran_0.5[(gene_prob_tran_0.5[,1] &lt;0.5 &amp; gene_prob_tran_0.5[,2] &lt;0.5 &amp; gene_prob_tran_0.5[,3] &lt;0.5 &amp; gene_prob_tran_0.5[,4] &lt;0.5 &amp; gene_prob_tran_0.5[,5] &gt; 0.5 &amp; gene_prob_tran_0.5[,6]&gt;0.5),])
length(prob_3_0.5)</code></pre>
<pre><code>[1] 6450</code></pre>
<pre class="r"><code>prob_4_0.5  &lt;- rownames(gene_prob_tran_0.5[(gene_prob_tran_0.5[,1] &gt;= 0.1 &amp; gene_prob_tran_0.5[,2] &gt; 0.5 &amp; gene_prob_tran_0.5[,3] &gt; 0.5 &amp; gene_prob_tran_0.5[,4] &gt;= 0.02 &amp; gene_prob_tran_0.5[,5] &lt; 0.5 &amp; gene_prob_tran_0.5[,6] &lt; 0.5),]) 

length(prob_4_0.5)</code></pre>
<pre><code>[1] 142</code></pre>
<pre class="r"><code>prob_5_0.5  &lt;- rownames(gene_prob_tran_0.5[(gene_prob_tran_0.5[,1] &lt;0.5 &amp; gene_prob_tran_0.5[,2] &gt;0.5 &amp; gene_prob_tran_0.5[,3] &gt;0.5 &amp; gene_prob_tran_0.5[,4] &gt;=0.02 &amp; gene_prob_tran_0.5[,4] &lt;0.5 &amp; gene_prob_tran_0.5[,5] &gt; 0.5 &amp; gene_prob_tran_0.5[,6]&gt;0.5),])
length(prob_5_0.5)</code></pre>
<pre><code>[1] 221</code></pre>
</div>
<div
id="distribution-of-gene-clusters-identified-by-corrmotif-0.5-micromolar"
class="section level2">
<h2><strong>📌 Distribution of Gene Clusters Identified by Corrmotif
(0.5 micromolar)</strong></h2>
<pre class="r"><code># Load necessary library
library(ggplot2)

# Data
data &lt;- data.frame(
  Category = c(&quot;Non response (0.5)&quot;, &quot;DOX-specific response (0.5 µM)&quot;, &quot;DOX only mid-late response (0.5 µM)&quot;, &quot;CX + DOX (early) response (0.5 µM)&quot;, &quot;DOX + CX (mid-late) response (0.5 µM)&quot;),
  Value = c(7134,179,6450,142,221)
)

# Add values to category names (to be displayed in the legend)
data$Category &lt;- paste0(data$Category, &quot; (&quot;, data$Value, &quot;)&quot;)

# Define custom colors with updated category names
custom_colors &lt;- setNames(
  c(&quot;#FF9999&quot;, &quot;#FF66CC&quot;, &quot;#66B2FF&quot;, &quot;#99FF99&quot;, &quot;#FFD700&quot;),
  data$Category  # Ensures color names match updated categories
)

# Create pie chart without number labels inside
ggplot(data, aes(x = &quot;&quot;, y = Value, fill = Category)) +
  geom_bar(width = 1, stat = &quot;identity&quot;) +
  coord_polar(&quot;y&quot;, start = 0) +
  labs(title = &quot;Pie Chart (0.5 micromolar Corrmotif)&quot;, x = NULL, y = NULL) +
  theme_void() +
  scale_fill_manual(values = custom_colors)</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/pie-chart_0.5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pie-chart_0.5-1">
Past versions of pie-chart_0.5-1.png
</button>
</p>
<div id="fig-pie-chart_0.5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/pie-chart_0.5-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2990b21e59ad6434e6e7b4b92ae6cad8ff37f997/docs/figure/Corrmotif_Conc.Rmd/pie-chart_0.5-1.png" target="_blank">2990b21</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/pie-chart_0.5-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2de67e6a82f82edbc78941b11ba72ed8aba9964e/docs/figure/Corrmotif_Conc.Rmd/pie-chart_0.5-1.png" target="_blank">2de67e6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="combined-pie-charts-for-concentrations" class="section level2">
<h2><strong>📌 Combined pie charts for concentrations</strong></h2>
<pre class="r"><code># Load necessary libraries
library(ggplot2)
library(dplyr)

# Data for 0.1 µM
data_0.1 &lt;- data.frame(
  Category = c(&quot;Non response&quot;, &quot;CX-DOX mid-late response&quot;, &quot;DOX only mid-late&quot;),
  Value = c(12308, 415, 1551),
  Concentration = &quot;0.1 µM&quot;
)

# Data for 0.5 µM
data_0.5 &lt;- data.frame(
  Category = c(&quot;Non response&quot;, &quot;DOX specific response&quot;, &quot;DOX only mid-late response&quot;, 
               &quot;CX + DOX (early) response&quot;, &quot;DOX + CX (mid-late) response&quot;),
  Value = c(7134, 179, 6450, 142, 221),
  Concentration = &quot;0.5 µM&quot;
)

# Combine both datasets
combined_data &lt;- bind_rows(data_0.1, data_0.5)

# Add values to category names (for legend only)
combined_data$Category_Legend &lt;- paste0(combined_data$Category, &quot; (&quot;, combined_data$Value, &quot;)&quot;)

# Define custom colors for updated categories
custom_colors &lt;- c(
  &quot;Non response (12308)&quot; = &quot;#FF9999&quot;,
  &quot;CX-DOX mid-late response (415)&quot; = &quot;#66B2FF&quot;,
  &quot;DOX only mid-late (1551)&quot; = &quot;#99FF99&quot;,
  &quot;Non response (7134)&quot; = &quot;#FF9999&quot;,
  &quot;DOX specific response (179)&quot; = &quot;#FF66CC&quot;,
  &quot;DOX only mid-late response (6450)&quot; = &quot;#99FF99&quot;,
  &quot;CX + DOX (early) response (142)&quot; = &quot;#FFD700&quot;,
  &quot;DOX + CX (mid-late) response (221)&quot; = &quot;#8A2BE2&quot;
)

# Ensure categories appear in a specific order
combined_data$Category_Legend &lt;- factor(combined_data$Category_Legend, levels = names(custom_colors))

# Create faceted pie charts without numbers inside the slices
pie_chart &lt;- ggplot(combined_data, aes(x = &quot;&quot;, y = Value, fill = Category_Legend)) +
  geom_bar(width = 1, stat = &quot;identity&quot;) +
  coord_polar(&quot;y&quot;, start = 0) +
  facet_wrap(~ Concentration) +  # Facet by concentration (side-by-side)
  labs(title = &quot;Corrmotif Pie Charts for 0.1 µM and 0.5 µM&quot;, x = NULL, y = NULL, fill = &quot;Category&quot;) +
  theme_void() +
  theme(
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.5),  # Adds box around facets
    strip.background = element_rect(fill = &quot;white&quot;, color = &quot;black&quot;, linewidth = 1),  # Box for facet titles
    strip.text = element_text(size = 12, face = &quot;bold&quot;, color = &quot;black&quot;),
    legend.title = element_text(size = 12, face = &quot;bold&quot;),  # Bold legend title
    legend.text = element_text(size = 10)  # Adjust legend text size
  ) +
  scale_fill_manual(values = custom_colors)

# Display the plot
print(pie_chart)</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/pie-chart_comb-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pie-chart_comb-1">
Past versions of pie-chart_comb-1.png
</button>
</p>
<div id="fig-pie-chart_comb-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/pie-chart_comb-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ef3a95166a7a18770eaa7786b80994b40bba9973/docs/figure/Corrmotif_Conc.Rmd/pie-chart_comb-1.png" target="_blank">ef3a951</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="save-corr-motif-datasets-for-gene-ontology-analysis-0.5-micromolar"
class="section level2">
<h2><strong>📌 Save Corr-motif datasets for Gene Ontology analysis (0.5
Micromolar)</strong></h2>
<pre class="r"><code>write.csv(data.frame(Entrez_ID = prob_1_0.5), &quot;data/prob_1_0.5.csv&quot;, row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_2_0.5), &quot;data/prob_2_0.5.csv&quot;, row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_3_0.5), &quot;data/prob_3_0.5.csv&quot;, row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_4_0.5), &quot;data/prob_4_0.5.csv&quot;, row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_5_0.5), &quot;data/prob_5_0.5.csv&quot;, row.names = FALSE)</code></pre>
</div>
<div id="micromolar-abs-logfc-1" class="section level2">
<h2><strong>📌 0.5 Micromolar abs logFC</strong></h2>
<pre class="r"><code># Load Response Groups from CSV Files
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# Load Datasets (Only 0.5 Micromolar)
CX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)

DOX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

# Convert datasets to DataFrames
Toptable_CX_0.5_3_df  &lt;- data.frame(CX_0.5_3)
Toptable_CX_0.5_24_df &lt;- data.frame(CX_0.5_24)
Toptable_CX_0.5_48_df &lt;- data.frame(CX_0.5_48)

Toptable_DOX_0.5_3_df  &lt;- data.frame(DOX_0.5_3)
Toptable_DOX_0.5_24_df &lt;- data.frame(DOX_0.5_24)
Toptable_DOX_0.5_48_df &lt;- data.frame(DOX_0.5_48)

# Combine All 0.5 Micromolar Datasets into a Single Dataframe
all_toptables_0.5 &lt;- bind_rows(
  Toptable_CX_0.5_3_df  %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;3&quot;),
  Toptable_CX_0.5_24_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;24&quot;),
  Toptable_CX_0.5_48_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;48&quot;),
  Toptable_DOX_0.5_3_df  %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;3&quot;),
  Toptable_DOX_0.5_24_df %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;24&quot;),
  Toptable_DOX_0.5_48_df %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;48&quot;)
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.5$Entrez_ID &lt;- as.character(all_toptables_0.5$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX + DOX (early) response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX + CX (mid-late) response\n(0.5 micromolar)&quot;,
      TRUE ~ NA_character_
    )
  )

# Remove NA Values (Genes Not in Response Groups)
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;% filter(!is.na(Response_Group))

# Compute Absolute logFC
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;%
  mutate(absFC = abs(logFC))

# Convert Factors for Proper Ordering (Reversed Order for Response Groups)
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;%
  mutate(
    Drug = factor(Drug, levels = c(&quot;CX.5461&quot;, &quot;DOX&quot;)),
    Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;),
                       labels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Response_Group = factor(Response_Group,
                            levels = c(&quot;DOX + CX (mid-late) response\n(0.5 micromolar)&quot;,
                                       &quot;CX + DOX (early) response\n(0.5 micromolar)&quot;,
                                       &quot;DOX only mid-late response\n(0.5 micromolar)&quot;,
                                       &quot;DOX-specific response\n(0.5 micromolar)&quot;,
                                       &quot;Non response\n(0.5 micromolar)&quot;)) # Reversed Order
  )

# **Plot the Boxplot with Faceted Labels Wrapping Correctly**
ggplot(all_toptables_0.5, aes(x = Drug, y = absFC, fill = Drug)) +
  geom_boxplot() +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +  # Custom color palette
  facet_grid(Response_Group ~ Timepoint, labeller = label_wrap_gen(width = 20)) +  # Ensure Proper Wrapping
  theme_bw() +
  labs(
    x = &quot;Drugs&quot;,
    y = &quot;|Log Fold Change|&quot;,
    title = &quot;|Log Fold| for 0.5 micromolar&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.line = element_line(linewidth = 1.5),
    strip.background = element_rect(fill = &quot;gray&quot;),  # Gray background for facet labels
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),  # Bold styling for facet labels
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 15)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.5abs%20logFC-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-0.5abs-logFC-1">
Past versions of "0.5abs logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-0.5abs-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/0.5abs logFC-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2990b21e59ad6434e6e7b4b92ae6cad8ff37f997/docs/figure/Corrmotif_Conc.Rmd/0.5abs logFC-1.png" target="_blank">2990b21</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.5abs logFC-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.5abs logFC-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="micromolar-mean-abs-logfc-across-timepoints-1"
class="section level2">
<h2><strong>📌 0.5 Micromolar mean (Abs logFC) across
timepoints</strong></h2>
<pre class="r"><code># Load required libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# Load Datasets (Only 0.5 Micromolar)
CX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)

DOX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

# Combine All 0.5 Micromolar Datasets into a Single Dataframe
all_toptables_0.5 &lt;- bind_rows(
  CX_0.5_3  %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;3&quot;),
  CX_0.5_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;24&quot;),
  CX_0.5_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;48&quot;),
  DOX_0.5_3  %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;3&quot;),
  DOX_0.5_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;24&quot;),
  DOX_0.5_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;48&quot;)
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.5$Entrez_ID &lt;- as.character(all_toptables_0.5$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX + DOX (early) response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX + CX (mid-late) response\n(0.5 micromolar)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group))  # Remove NA values

# Compute Mean Absolute logFC for Line Plot
data_summary &lt;- all_toptables_0.5 %&gt;%
  mutate(abs_logFC = abs(logFC)) %&gt;%
  group_by(Response_Group, Drug, Timepoint) %&gt;%
  dplyr::summarize(mean_abs_logFC = mean(abs_logFC, na.rm = TRUE), .groups = &quot;drop&quot;) %&gt;%
  as.data.frame()

# **Ensure all timepoints exist in the summary**
timepoints_full &lt;- expand.grid(
  Response_Group = unique(all_toptables_0.5$Response_Group),
  Drug = unique(all_toptables_0.5$Drug),
  Timepoint = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;)
)

# **Merge to keep missing timepoints**
data_summary &lt;- full_join(timepoints_full, data_summary, by = c(&quot;Response_Group&quot;, &quot;Drug&quot;, &quot;Timepoint&quot;))

# **Replace NA mean_abs_logFC with 0 if no genes were present**
data_summary$mean_abs_logFC[is.na(data_summary$mean_abs_logFC)] &lt;- 0

# Convert Factors for Proper Ordering (Reversed Order for Response Groups)
data_summary &lt;- data_summary %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;), labels = c(&quot;3 hours&quot;, &quot;24 hours&quot;, &quot;48 hours&quot;)),
    Response_Group = factor(Response_Group, levels = c(
      &quot;DOX + CX (mid-late) response\n(0.5 micromolar)&quot;,
                                       &quot;CX + DOX (early) response\n(0.5 micromolar)&quot;,
                                       &quot;DOX only mid-late response\n(0.5 micromolar)&quot;,
                                       &quot;DOX-specific response\n(0.5 micromolar)&quot;,
                                       &quot;Non response\n(0.5 micromolar)&quot;  # Reversed order
    ))
  )

# Define custom drug palette
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)

# **Plot the Line Plot for Mean Absolute logFC**
ggplot(data_summary, aes(x = Timepoint, y = mean_abs_logFC, group = Drug, color = Drug)) +
  geom_point(size = 3) +
  geom_line(size = 1.2) +
  scale_color_manual(values = drug_palette) +
  ylim(0, 2.5) +  # Adjust the Y-axis for better visualization
  facet_wrap(~ Response_Group, ncol = 1) +  # Facet by Response Group (Reversed Order)
  theme_bw() +
  labs(
    x = &quot;Timepoints&quot;,
    y = &quot;Mean |Log Fold Change|&quot;,
    title = &quot;Mean |Log Fold Change| Across Response Groups (0.5 micromolar)&quot;,
    color = &quot;Drug&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text = element_text(size = 12, color = &quot;black&quot;),
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.5%20mean-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-0.5-mean-1">
Past versions of "0.5 mean-1.png"
</button>
</p>
<div id="fig-no-spaces-0.5-mean-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/0.5 mean-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2990b21e59ad6434e6e7b4b92ae6cad8ff37f997/docs/figure/Corrmotif_Conc.Rmd/0.5 mean-1.png" target="_blank">2990b21</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.5 mean-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.5 mean-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="micromolar-logfc-1" class="section level2">
<h2><strong>📌 0.5 Micromolar logFC</strong></h2>
<pre class="r"><code># Load required libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# Load Datasets (Only 0.5 Micromolar)
CX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)

DOX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

# Combine All 0.5 Micromolar Datasets into a Single Dataframe
all_toptables_0.5 &lt;- bind_rows(
  CX_0.5_3  %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  CX_0.5_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  CX_0.5_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;),
  DOX_0.5_3  %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  DOX_0.5_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  DOX_0.5_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;)
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.5$Entrez_ID &lt;- as.character(all_toptables_0.5$Entrez_ID)

# Assign Response Groups
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX + DOX (early) response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX + CX (mid-late) response\n(0.5 micromolar)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group))

# Convert factors to ensure correct ordering (Reversed Order for Response Groups)
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Response_Group = factor(Response_Group, levels = c(
      &quot;DOX + CX (mid-late) response\n(0.5 micromolar)&quot;,
                                       &quot;CX + DOX (early) response\n(0.5 micromolar)&quot;,
                                       &quot;DOX only mid-late response\n(0.5 micromolar)&quot;,
                                       &quot;DOX-specific response\n(0.5 micromolar)&quot;,
                                       &quot;Non response\n(0.5 micromolar)&quot;  # Reversed Order
    ))
  )

# **Plot the Boxplot**
ggplot(all_toptables_0.5, aes(x = Drug, y = logFC, fill = Drug)) +
  geom_boxplot() +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  facet_grid(Response_Group ~ Timepoint) +
  theme_bw() +
  labs(x = &quot;Drugs&quot;, y = &quot;Log Fold Change&quot;, title = &quot;Log Fold Change for 0.5 Micromolar&quot;) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.5logFC-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-0.5logFC-1">
Past versions of 0.5logFC-1.png
</button>
</p>
<div id="fig-0.5logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/0.5logFC-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2990b21e59ad6434e6e7b4b92ae6cad8ff37f997/docs/figure/Corrmotif_Conc.Rmd/0.5logFC-1.png" target="_blank">2990b21</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.5logFC-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.5logFC-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="micromolar-mean-logfc-across-timepoints-1"
class="section level2">
<h2><strong>📌 0.5 Micromolar mean logFC across timepoints</strong></h2>
<pre class="r"><code># Load Required Libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# Load Datasets (Only 0.5 Micromolar)
CX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)

DOX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

# Combine All 0.5 Micromolar Datasets into a Single Dataframe
all_toptables_0.5 &lt;- bind_rows(
  CX_0.5_3  %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  CX_0.5_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  CX_0.5_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;),
  DOX_0.5_3  %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  DOX_0.5_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  DOX_0.5_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;)
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.5$Entrez_ID &lt;- as.character(all_toptables_0.5$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX + DOX (early) response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX + CX (mid-late) response\n(0.5 micromolar)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group))

# Compute Mean logFC for Line Plot
data_summary_0.5 &lt;- all_toptables_0.5 %&gt;%
  group_by(Response_Group, Drug, Timepoint) %&gt;%
  dplyr::summarize(mean_logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;) %&gt;%
  as.data.frame()

# **Ensure all timepoints exist in the summary**
timepoints_full &lt;- expand.grid(
  Response_Group = unique(all_toptables_0.5$Response_Group),
  Drug = unique(all_toptables_0.5$Drug),
  Timepoint = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)
)

# **Merge to keep missing timepoints**
data_summary_0.5 &lt;- full_join(timepoints_full, data_summary_0.5, by = c(&quot;Response_Group&quot;, &quot;Drug&quot;, &quot;Timepoint&quot;))

# **Replace NA mean_logFC with 0 if no genes were present**
data_summary_0.5$mean_logFC[is.na(data_summary_0.5$mean_logFC)] &lt;- 0

# Convert Factors for Proper Ordering (Reversed Order for Response Groups)
data_summary_0.5 &lt;- data_summary_0.5 %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Response_Group = factor(Response_Group, levels = c(
      &quot;DOX + CX (mid-late) response\n(0.5 micromolar)&quot;,
                                       &quot;CX + DOX (early) response\n(0.5 micromolar)&quot;,
                                       &quot;DOX only mid-late response\n(0.5 micromolar)&quot;,
                                       &quot;DOX-specific response\n(0.5 micromolar)&quot;,
                                       &quot;Non response\n(0.5 micromolar)&quot;  # Reversed Order
    ))
  )

# Define custom drug palette
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)

# **Plot the Line Plot for Mean logFC**
ggplot(data_summary_0.5, aes(x = Timepoint, y = mean_logFC, group = Drug, color = Drug)) +
  geom_point(size = 3) +
  geom_line(size = 1.2) +
  scale_color_manual(values = drug_palette) +
  ylim(-2, 1.5) +  # Adjust the Y-axis for better visualization
  facet_wrap(~ Response_Group, ncol = 1) +  # Facet by Response Group (Reversed Order)
  theme_bw() +
  labs(
    x = &quot;Timepoints&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    title = &quot;Mean Log Fold Change Across Response Groups (0.5 micromolar)&quot;,
    color = &quot;Drug&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text = element_text(size = 12, color = &quot;black&quot;),
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.5%20mean2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-0.5-mean2-1">
Past versions of "0.5 mean2-1.png"
</button>
</p>
<div id="fig-no-spaces-0.5-mean2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/0.5 mean2-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2990b21e59ad6434e6e7b4b92ae6cad8ff37f997/docs/figure/Corrmotif_Conc.Rmd/0.5 mean2-1.png" target="_blank">2990b21</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.5 mean2-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.5 mean2-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="proportion-of-dna-damage-repair-genes-0.1-micromolar-md-anderson-dataset"
class="section level2">
<h2><strong>📌 Proportion of DNA Damage Repair Genes (0.1 micromolar)
(MD Anderson dataset)</strong></h2>
<pre class="r"><code># Load necessary libraries
library(dplyr)
library(ggplot2)
library(tidyr)</code></pre>
<pre><code>Warning: package &#39;tidyr&#39; was built under R version 4.3.3</code></pre>
<pre class="r"><code>library(org.Hs.eg.db)</code></pre>
<pre><code>Warning: package &#39;AnnotationDbi&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;IRanges&#39; was built under R version 4.3.1</code></pre>
<pre><code>Warning: package &#39;S4Vectors&#39; was built under R version 4.3.2</code></pre>
<pre class="r"><code># **🔹 Read DNA Damage Repair Gene List**
DNA_damage &lt;- read.csv(&quot;data/DNA_Damage.csv&quot;, stringsAsFactors = FALSE)

# Convert gene symbols to Entrez IDs
DNA_damage &lt;- DNA_damage %&gt;%
  mutate(Entrez_ID = mapIds(org.Hs.eg.db,
                            keys = DNA_damage$Symbol,
                            column = &quot;ENTREZID&quot;,
                            keytype = &quot;SYMBOL&quot;,
                            multiVals = &quot;first&quot;))

DNA_damage_genes &lt;- na.omit(DNA_damage$Entrez_ID)

# **🔹 Load Corrmotif Groups for 0.1 Concentration**
prob_groups_0.1 &lt;- list(
  &quot;Non Response (0.1)&quot; = read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID,
   &quot;CX_DOX mid-late (0.1)&quot; = read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID,
  &quot;DOX only mid-late (0.1)&quot;= read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID
)

# **🔹 Create Dataframe for Corrmotif Groups**
corrmotif_df_0.1 &lt;- bind_rows(
  lapply(prob_groups_0.1, function(ids) {
    data.frame(Entrez_ID = ids)
  }),
  .id = &quot;Response_Group&quot;
)

# **🔹 Match Entrez_IDs with DNA Damage Repair Genes**
corrmotif_df_0.1 &lt;- corrmotif_df_0.1 %&gt;%
  mutate(DNA_Damage = ifelse(Entrez_ID %in% DNA_damage_genes, &quot;Yes&quot;, &quot;No&quot;))

# **🔹 Count DNA Damage Repair Genes in Each Response Group**
proportion_data &lt;- corrmotif_df_0.1 %&gt;%
  group_by(Response_Group, DNA_Damage) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Response_Group) %&gt;%
  mutate(Percentage = (Count / sum(Count)) * 100)

# **🔹 Ensure &quot;Yes&quot; is at the Bottom and &quot;No&quot; at the Top**
proportion_data$DNA_Damage &lt;- factor(proportion_data$DNA_Damage, levels = c(&quot;Yes&quot;, &quot;No&quot;))

# **🔹 Set Order of Response Groups for X-axis**
response_order &lt;- c(&quot;Non Response (0.1)&quot;, &quot;CX_DOX mid-late (0.1)&quot;,&quot;DOX only mid-late (0.1)&quot;)
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = response_order)

# **🔹 Perform Chi-Square Tests for &quot;DOX only mid-late (0.1)&quot; and &quot;CX_DOX mid-late (0.1)&quot; vs &quot;Non Response (0.1)&quot;**
non_response_counts &lt;- proportion_data %&gt;%
  filter(Response_Group == &quot;Non Response (0.1)&quot;) %&gt;%
  dplyr::select(DNA_Damage, Count) %&gt;%
  {setNames(.$Count, .$DNA_Damage)}  # Convert to named vector

chi_results &lt;- proportion_data %&gt;%
  filter(Response_Group %in% c(&quot;CX_DOX mid-late (0.1)&quot;,&quot;DOX only mid-late (0.1)&quot;)) %&gt;%
  group_by(Response_Group) %&gt;%
  summarise(
    p_value = {
      group_counts &lt;- Count[DNA_Damage %in% c(&quot;Yes&quot;, &quot;No&quot;)]
      if (!&quot;Yes&quot; %in% DNA_Damage) group_counts &lt;- c(group_counts, 0)
      if (!&quot;No&quot; %in% DNA_Damage) group_counts &lt;- c(0, group_counts)

      contingency_table &lt;- matrix(c(
        group_counts[1], group_counts[2],
        non_response_counts[&quot;Yes&quot;], non_response_counts[&quot;No&quot;]
      ), nrow = 2, byrow = TRUE)

      # Perform chi-square test if all values are valid
      if (all(contingency_table &gt;= 0 &amp; is.finite(contingency_table))) {
        chisq.test(contingency_table)$p.value
      } else {
        NA
      }
    },
    .groups = &quot;drop&quot;
  ) %&gt;%
  mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))

# **🔹 Merge Chi-Square Results into Proportion Data**
proportion_data &lt;- proportion_data %&gt;%
  left_join(chi_results %&gt;% dplyr::select(Response_Group, Significance), by = &quot;Response_Group&quot;)

# **🔹 Set Star Position Uniform Across Groups at 105%**
star_positions &lt;- data.frame(
  Response_Group = c(&quot;CX_DOX mid-late (0.1)&quot;, &quot;DOX only mid-late (0.1)&quot;),
  y_pos = 105,  # Fixed at 105% of Y-axis
  Significance = chi_results$Significance
)

# **🔹 Generate Proportion Plot with Chi-Square Stars**
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = DNA_Damage)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +  # Stacked bars
  geom_text(
    data = star_positions,
    aes(x = Response_Group, y = y_pos, label = Significance),  # Place stars at fixed 105%
    inherit.aes = FALSE,
    size = 6, color = &quot;black&quot;, fontface = &quot;bold&quot;, vjust = 0  # Keeps stars aligned
  ) +
  scale_y_continuous(labels = scales::percent_format(scale = 1), limits = c(0, 110)) +  # Fixed Y-axis to 100%
  scale_fill_manual(values = c(&quot;Yes&quot; = &quot;#e41a1c&quot;, &quot;No&quot; = &quot;#377eb8&quot;)) +  # Yes (Red), No (Blue)
  labs(
    title = &quot;Proportion of DNA Damage Repair Genes in\n0.1 Corrmotif Response Groups&quot;,
    x = &quot;Response Groups (0.1 Concentration)&quot;,
    y = &quot;Percentage&quot;,
    fill = &quot;DNA Damage Repair&quot;
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
    legend.title = element_blank(),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    strip.background = element_blank(),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/prop%20DNA-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-prop-DNA-1">
Past versions of "prop DNA-1.png"
</button>
</p>
<div id="fig-no-spaces-prop-DNA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/prop DNA-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/81247f64f08df43c2b40a386707fc8e9cc240143/docs/figure/Corrmotif_Conc.Rmd/prop DNA-1.png" target="_blank">81247f6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/96d0db145cf68e6db0d39e70bad9af0f75793aae/docs/figure/Corrmotif_Conc.Rmd/prop DNA-1.png" target="_blank">96d0db1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="proportion-of-dna-damage-repair-genes-0.5-micromolar-md-anderson-dataset"
class="section level2">
<h2><strong>📌 Proportion of DNA Damage Repair Genes (0.5 micromolar)
(MD Anderson dataset)</strong></h2>
<pre class="r"><code># Load necessary libraries
library(dplyr)
library(ggplot2)
library(tidyr)
library(org.Hs.eg.db)

# **🔹 Read DNA Damage Repair Gene List**
DNA_damage &lt;- read.csv(&quot;data/DNA_Damage.csv&quot;, stringsAsFactors = FALSE)

# Convert gene symbols to Entrez IDs
DNA_damage &lt;- DNA_damage %&gt;%
  mutate(Entrez_ID = mapIds(org.Hs.eg.db,
                            keys = DNA_damage$Symbol,
                            column = &quot;ENTREZID&quot;,
                            keytype = &quot;SYMBOL&quot;,
                            multiVals = &quot;first&quot;))

DNA_damage_genes &lt;- na.omit(DNA_damage$Entrez_ID)

# **🔹 Load Corrmotif Groups for 0.5 Concentration**
prob_groups_0.5 &lt;- list(
  &quot;Non Response (0.5)&quot; = read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID,
  &quot;DOX-specific response (0.5)&quot; = read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID,
  &quot;DOX only mid-late response (0.5)&quot; = read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID,
  &quot;CX DOX (early) response (0.5)&quot; = read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID,
  &quot;DOX + CX (mid-late) response (0.5)&quot; = read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID
)

# **🔹 Create Dataframe for Corrmotif Groups**
corrmotif_df_0.5 &lt;- bind_rows(
  lapply(prob_groups_0.5, function(ids) {
    data.frame(Entrez_ID = ids)
  }),
  .id = &quot;Response_Group&quot;
)

# **🔹 Match Entrez_IDs with DNA Damage Repair Genes**
corrmotif_df_0.5 &lt;- corrmotif_df_0.5 %&gt;%
  mutate(DNA_Damage = ifelse(Entrez_ID %in% DNA_damage_genes, &quot;Yes&quot;, &quot;No&quot;))

# **🔹 Count DNA Damage Repair Genes in Each Response Group**
proportion_data &lt;- corrmotif_df_0.5 %&gt;%
  group_by(Response_Group, DNA_Damage) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Response_Group) %&gt;%
  mutate(Percentage = (Count / sum(Count)) * 100)

# **🔹 Ensure &quot;Yes&quot; is at the Bottom and &quot;No&quot; at the Top**
proportion_data$DNA_Damage &lt;- factor(proportion_data$DNA_Damage, levels = c(&quot;Yes&quot;, &quot;No&quot;))

# **🔹 Set Order of Response Groups for X-axis**
response_order &lt;- c(&quot;Non Response (0.5)&quot;, &quot;DOX-specific response (0.5)&quot;, &quot;DOX only mid-late response (0.5)&quot;, 
                    &quot;CX DOX (early) response (0.5)&quot;, &quot;DOX + CX (mid-late) response (0.5)&quot;)
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = response_order)

# **🔹 Perform Chi-Square Tests for Each Response Group vs Non-Response**
non_response_counts &lt;- proportion_data %&gt;%
  filter(Response_Group == &quot;Non Response (0.5)&quot;) %&gt;%
  dplyr::select(DNA_Damage, Count) %&gt;%
  {setNames(.$Count, .$DNA_Damage)}  # Convert to named vector

# **Comparing Each Group Against &quot;Non Response (0.5)&quot;**
chi_results &lt;- proportion_data %&gt;%
  filter(Response_Group %in% c(&quot;DOX-specific response (0.5)&quot;, &quot;DOX only mid-late response (0.5)&quot;, 
                               &quot;CX DOX (early) response (0.5)&quot;, &quot;DOX + CX (mid-late) response (0.5)&quot;)) %&gt;%
  group_by(Response_Group) %&gt;%
  summarise(
    p_value = {
      group_counts &lt;- Count[DNA_Damage %in% c(&quot;Yes&quot;, &quot;No&quot;)]
      if (!&quot;Yes&quot; %in% DNA_Damage) group_counts &lt;- c(group_counts, 0)
      if (!&quot;No&quot; %in% DNA_Damage) group_counts &lt;- c(0, group_counts)

      contingency_table &lt;- matrix(c(
        group_counts[1], group_counts[2],  # Response group counts
        non_response_counts[&quot;Yes&quot;], non_response_counts[&quot;No&quot;]  # Non-response counts
      ), nrow = 2, byrow = TRUE)

      # Perform chi-square test if all values are valid
      if (all(contingency_table &gt;= 0 &amp; is.finite(contingency_table))) {
        chisq.test(contingency_table)$p.value
      } else {
        NA
      }
    },
    .groups = &quot;drop&quot;
  ) %&gt;%
  mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))

# **🔹 Merge Chi-Square Results into Proportion Data**
proportion_data &lt;- proportion_data %&gt;%
  left_join(chi_results %&gt;% dplyr::select(Response_Group, Significance), by = &quot;Response_Group&quot;)

# **🔹 Set Star Position Uniform Across Groups at 105%**
star_positions &lt;- data.frame(
  Response_Group = c(&quot;DOX-specific response (0.5)&quot;, &quot;DOX only mid-late response (0.5)&quot;, 
                     &quot;CX DOX (early) response (0.5)&quot;, &quot;DOX + CX (mid-late) response (0.5)&quot;),
  y_pos = 105,  # Fixed at 105% of Y-axis
  Significance = chi_results$Significance
)

# **🔹 Generate Proportion Plot with Chi-Square Stars**
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = DNA_Damage)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +  # Stacked bars
  geom_text(
    data = star_positions,
    aes(x = Response_Group, y = y_pos, label = Significance),  # Place stars at fixed 105%
    inherit.aes = FALSE,
    size = 6, color = &quot;black&quot;, fontface = &quot;bold&quot;, vjust = 0  # Keeps stars aligned
  ) +
  scale_y_continuous(labels = scales::percent_format(scale = 1), limits = c(0, 110)) +  # **Y-axis now limited to 110% for visibility**
  scale_fill_manual(values = c(&quot;Yes&quot; = &quot;#e41a1c&quot;, &quot;No&quot; = &quot;#377eb8&quot;)) +  # Yes (Red), No (Blue)
  labs(
    title = &quot;Proportion of DNA Damage Repair Genes in\n0.5 Corrmotif Response Groups&quot;,
    x = &quot;Response Groups (0.5 Concentration)&quot;,
    y = &quot;Percentage&quot;,
    fill = &quot;DNA Damage Repair Genes&quot;
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
    legend.title = element_blank(),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    strip.background = element_blank(),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/prop%20DNA%200.5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-prop-DNA-0.5-1">
Past versions of "prop DNA 0.5-1.png"
</button>
</p>
<div id="fig-no-spaces-prop-DNA-0.5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/prop DNA 0.5-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/96d0db145cf68e6db0d39e70bad9af0f75793aae/docs/figure/Corrmotif_Conc.Rmd/prop DNA 0.5-1.png" target="_blank">96d0db1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="proportion-of-dna-damage-repair-genes-combined-md-anderson-dataset"
class="section level2">
<h2><strong>📌 Proportion of DNA Damage Repair Genes (Combined) (MD
Anderson dataset)</strong></h2>
<pre class="r"><code># Load Required Libraries
library(dplyr)
library(ggplot2)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load DNA Damage Genes -----------------
DNA_damage &lt;- read.csv(&quot;data/DNA_Damage.csv&quot;, stringsAsFactors = FALSE)

DNA_damage$Entrez_ID &lt;- mapIds(
  org.Hs.eg.db,
  keys = DNA_damage$Symbol,
  column = &quot;ENTREZID&quot;,
  keytype = &quot;SYMBOL&quot;,
  multiVals = &quot;first&quot;
)
dna_damage_ids &lt;- na.omit(DNA_damage$Entrez_ID)

# ----------------- Load CorrMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% dna_damage_ids, &quot;Yes&quot;, &quot;No&quot;),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% dna_damage_ids, &quot;Yes&quot;, &quot;No&quot;),
    Concentration = &quot;0.5 µM&quot;
  )

# ----------------- Combine All -----------------
df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Calculate Proportions -----------------
proportion_data &lt;- df_combined %&gt;%
  group_by(Concentration, Response_Group, Category) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Set Order: &quot;Yes&quot; at bottom -----------------
proportion_data$Category &lt;- factor(proportion_data$Category, levels = c(&quot;Yes&quot;, &quot;No&quot;))

# ----------------- Chi-square or Fisher&#39;s Test -----------------
get_chi_or_fisher &lt;- function(df, ref_group) {
  ref_counts &lt;- df %&gt;%
    ungroup() %&gt;%
    filter(Response_Group == ref_group) %&gt;%
    dplyr::select(Category, Count) %&gt;%
    {setNames(.$Count, .$Category)}

  df %&gt;%
    filter(Response_Group != ref_group) %&gt;%
    group_by(Response_Group) %&gt;%
    summarise(
      p_value = {
        group_counts &lt;- Count[Category %in% c(&quot;Yes&quot;, &quot;No&quot;)]
        if (length(group_counts) &lt; 2) group_counts &lt;- c(group_counts, 0)
        contingency_table &lt;- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts[&quot;Yes&quot;], ref_counts[&quot;No&quot;]
        ), nrow = 2)

        if (any(contingency_table &lt; 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = &quot;drop&quot;
    ) %&gt;%
    mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))
}

# Perform tests
chi_0.1 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.1 µM&quot;), &quot;Non response&quot;)
chi_0.5 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.5 µM&quot;), &quot;Non response&quot;)

chi_all &lt;- bind_rows(
  chi_0.1 %&gt;% mutate(Concentration = &quot;0.1 µM&quot;),
  chi_0.5 %&gt;% mutate(Concentration = &quot;0.5 µM&quot;)
)

proportion_data &lt;- proportion_data %&gt;%
  left_join(chi_all, by = c(&quot;Concentration&quot;, &quot;Response_Group&quot;))

# ----------------- Factor Levels -----------------
proportion_data$Concentration &lt;- factor(proportion_data$Concentration, levels = c(&quot;0.1 µM&quot;, &quot;0.5 µM&quot;))
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
))

# ----------------- Star Positions -----------------
label_data &lt;- proportion_data %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  summarise(Significance = dplyr::first(Significance), .groups = &quot;drop&quot;) %&gt;%
  filter(Significance == &quot;*&quot;) %&gt;%
  mutate(y_pos = 105)

# ----------------- Stack Order -----------------
proportion_data &lt;- proportion_data %&gt;%
  arrange(Concentration, Response_Group, Category)

# ----------------- Plot -----------------
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(
    data = label_data,
    aes(x = Response_Group, y = y_pos, label = Significance),
    inherit.aes = FALSE,
    size = 5, color = &quot;black&quot;
  ) +
  facet_wrap(~ Concentration, scales = &quot;free_x&quot;) +
  scale_fill_manual(values = c(&quot;Yes&quot; = &quot;#e41a1c&quot;, &quot;No&quot; = &quot;#377eb8&quot;)) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +
  labs(
    title = &quot;DNA Damage Gene Proportions Across\nCormotif Clusters (0.1 and 0.5 µM)&quot;,
    x = &quot;Response Groups&quot;,
    y = &quot;Percentage of Genes&quot;,
    fill = &quot;DNA Damage Gene&quot;
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = &quot;bold&quot;),
    axis.title.x = element_text(size = 14, face = &quot;bold&quot;),
    axis.title.y = element_text(size = 14, face = &quot;bold&quot;),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    panel.spacing = unit(1.2, &quot;lines&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/prop%20DNA%20combined-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-prop-DNA-combined-1">
Past versions of "prop DNA combined-1.png"
</button>
</p>
<div id="fig-no-spaces-prop-DNA-combined-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/07b9670bf922129e5d7a932d22711efc033fde34/docs/figure/Corrmotif_Conc.Rmd/prop DNA combined-1.png" target="_blank">07b9670</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/9618a00ad12263a31ed79f64344c82a444c6d5c1/docs/figure/Corrmotif_Conc.Rmd/prop DNA combined-1.png" target="_blank">9618a00</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="proportion-of-dna-damage-response-genes-molecular-signature-dataset"
class="section level2">
<h2><strong>📌 Proportion of DNA Damage Response Genes (Molecular
signature dataset)</strong></h2>
<pre class="r"><code># ----------------- Load Required Libraries -----------------
library(dplyr)
library(ggplot2)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- DDR Entrez IDs -----------------
ddr_entrez_ids &lt;- as.character(c(
  10111, 1017, 1019, 1020, 1021, 1026, 1027, 10912, 11011, 1111,
  11200, 1385, 1643, 1647, 1869, 207, 2177, 25, 27113, 27244,
  3014, 317, 355, 4193, 4292, 4361, 4609, 4616, 4683, 472, 50484,
  5366, 5371, 54205, 545, 55367, 5591, 581, 5810, 5883, 5884,
  5888, 5893, 5925, 595, 5981, 6118, 637, 672, 7157, 7799,
  8243, 836, 841, 84126, 842, 8795, 891, 894, 896, 898,
  9133, 9134, 983, 9874, 993, 995, 5916
))

# ----------------- Load CorrMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% ddr_entrez_ids, &quot;DDR Genes&quot;, &quot;Non-DDR Genes&quot;),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% ddr_entrez_ids, &quot;DDR Genes&quot;, &quot;Non-DDR Genes&quot;),
    Concentration = &quot;0.5 µM&quot;
  )

# ----------------- Combine Data -----------------
df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Calculate Proportions -----------------
proportion_data &lt;- df_combined %&gt;%
  group_by(Concentration, Response_Group, Category) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Chi-square or Fisher&#39;s Test -----------------
get_chi_or_fisher &lt;- function(df, ref_group) {
  ref_counts &lt;- df %&gt;%
    ungroup() %&gt;%
    filter(Response_Group == ref_group) %&gt;%
    { ref_tbl &lt;- .[ , c(&quot;Category&quot;, &quot;Count&quot;)]; setNames(ref_tbl$Count, ref_tbl$Category) }
  
  df %&gt;%
    filter(Response_Group != ref_group) %&gt;%
    group_by(Response_Group) %&gt;%
    summarise(
      p_value = {
        group_counts &lt;- Count[Category %in% c(&quot;DDR Genes&quot;, &quot;Non-DDR Genes&quot;)]
        if (length(group_counts) &lt; 2) group_counts &lt;- c(group_counts, 0)
        contingency_table &lt;- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts[&quot;DDR Genes&quot;], ref_counts[&quot;Non-DDR Genes&quot;]
        ), nrow = 2)
        if (any(contingency_table &lt; 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = &quot;drop&quot;
    ) %&gt;%
    mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))
}

# Run test for each concentration
chi_0.1 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.1 µM&quot;), &quot;Non response&quot;)</code></pre>
<pre><code>Warning: There was 1 warning in `summarise()`.
ℹ In argument: `p_value = { ... }`.
ℹ In group 1: `Response_Group = &quot;CX-DOX mid-late response&quot;`.
Caused by warning in `chisq.test()`:
! Chi-squared approximation may be incorrect</code></pre>
<pre class="r"><code>chi_0.5 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.5 µM&quot;), &quot;Non response&quot;)</code></pre>
<pre><code>Warning: There were 2 warnings in `summarise()`.
The first warning was:
ℹ In argument: `p_value = { ... }`.
ℹ In group 1: `Response_Group = &quot;CX total + DOX early response&quot;`.
Caused by warning in `chisq.test()`:
! Chi-squared approximation may be incorrect
ℹ Run `dplyr::last_dplyr_warnings()` to see the 1 remaining warning.</code></pre>
<pre class="r"><code>chi_all &lt;- bind_rows(
  chi_0.1 %&gt;% mutate(Concentration = &quot;0.1 µM&quot;),
  chi_0.5 %&gt;% mutate(Concentration = &quot;0.5 µM&quot;)
)

# Merge test results into proportions
proportion_data &lt;- left_join(proportion_data, chi_all, by = c(&quot;Concentration&quot;, &quot;Response_Group&quot;))

# ----------------- Reorder Factor Levels -----------------
proportion_data$Concentration &lt;- factor(proportion_data$Concentration, levels = c(&quot;0.1 µM&quot;, &quot;0.5 µM&quot;))
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
))

# ----------------- Star Label Position -----------------
label_data &lt;- proportion_data %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  summarise(Significance = dplyr::first(Significance), .groups = &quot;drop&quot;) %&gt;%
  filter(!is.na(Significance) &amp; Significance != &quot;&quot;) %&gt;%
  mutate(y_pos = 105)

# ----------------- Final Plot -----------------
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(
    data = label_data,
    aes(x = Response_Group, y = y_pos, label = Significance),
    inherit.aes = FALSE,
    size = 5,
    color = &quot;black&quot;
  ) +
  facet_wrap(~ Concentration, scales = &quot;free_x&quot;) +
  scale_fill_manual(values = c(
    &quot;DDR Genes&quot; = &quot;#E41A1C&quot;,
    &quot;Non-DDR Genes&quot; = &quot;#377EB8&quot;
  )) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +
  labs(
    title = &quot;DNA Damage Response Gene Proportions Across\nCormotif Clusters (0.1 and 0.5)&quot;,
    x = &quot;Response Groups&quot;,
    y = &quot;Percentage of Genes&quot;,
    fill = &quot;Gene Category&quot;
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = &quot;bold&quot;),
    axis.title.x = element_text(size = 14, face = &quot;bold&quot;),
    axis.title.y = element_text(size = 14, face = &quot;bold&quot;),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    panel.spacing = unit(1.2, &quot;lines&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/prop%20DDR-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-prop-DDR-1">
Past versions of "prop DDR-1.png"
</button>
</p>
<div id="fig-no-spaces-prop-DDR-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/prop DDR-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-dna-damage-response-genes-across-corrmotifs-molecular-signature-dataset-per-concentration"
class="section level2">
<h2><strong>📌 Mean logFC of DNA damage response genes across corrmotifs
(Molecular signature dataset) (per concentration)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)</code></pre>
<pre><code>Warning: package &#39;tidyverse&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;readr&#39; was built under R version 4.3.3</code></pre>
<pre><code>Warning: package &#39;purrr&#39; was built under R version 4.3.3</code></pre>
<pre><code>Warning: package &#39;stringr&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;lubridate&#39; was built under R version 4.3.3</code></pre>
<pre class="r"><code>library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- DDR Entrez IDs -----------------
entrez_ids &lt;- c(
  10111, 1017, 1019, 1020, 1021, 1026, 1027, 10912, 11011, 1111,
  11200, 1385, 1643, 1647, 1869, 207, 2177, 25, 27113, 27244,
  3014, 317, 355, 4193, 4292, 4361, 4609, 4616, 4683, 472, 50484,
  5366, 5371, 54205, 545, 55367, 5591, 581, 5810, 5883, 5884,
  5888, 5893, 5925, 595, 5981, 6118, 637, 672, 7157, 7799,
  8243, 836, 841, 84126, 842, 8795, 891, 894, 896, 898,
  9133, 9134, 983, 9874, 993, 995, 5916
)

# ----------------- Load CorMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate Response Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% as.character(entrez_ids)) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% as.character(entrez_ids)) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Aggregate DEG Data -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;,
      TRUE ~ NA_character_
    )
  )

deg_mean &lt;- deg_all %&gt;%
  group_by(Entrez_ID, Concentration) %&gt;%
  summarise(
    logFC = mean(logFC, na.rm = TRUE),
    .groups = &quot;drop&quot;
  )

# ----------------- Merge with CorMotif DDR Genes -----------------
ddr_logfc &lt;- inner_join(df_combined, deg_mean, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Group = paste(Response_Group, Concentration)
  )

# ----------------- Define order -----------------
group_levels &lt;- c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
)

ddr_logfc$Response_Group &lt;- factor(ddr_logfc$Response_Group, levels = group_levels)

# ----------------- Wilcoxon Test -----------------
comparison_map &lt;- list(
  &quot;CX-DOX mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX only mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX-specific response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX only mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;CX total + DOX early response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX early + CX-DOX mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;
)

star_df &lt;- lapply(names(comparison_map), function(resp_group) {
  control_group &lt;- comparison_map[[resp_group]]
  resp_vals &lt;- ddr_logfc$logFC[ddr_logfc$Group == resp_group]
  control_vals &lt;- ddr_logfc$logFC[ddr_logfc$Group == control_group]

  if (length(resp_vals) &gt;= 2 &amp;&amp; length(control_vals) &gt;= 2) {
    test_result &lt;- wilcox.test(resp_vals, control_vals)
    pval &lt;- test_result$p.value
    if (pval &lt; 0.05) {
      label &lt;- case_when(
        pval &lt; 0.001 ~ &quot;***&quot;,
        pval &lt; 0.01 ~ &quot;**&quot;,
        TRUE ~ &quot;*&quot;
      )
      y_pos &lt;- max(resp_vals, na.rm = TRUE) + 0.5
      return(data.frame(Group = resp_group, y_pos = y_pos, label = label, P_Value = signif(pval, 4)))
    }
  }
  return(NULL)
}) %&gt;% bind_rows()

# ----------------- Prepare Annotation -----------------
label_data &lt;- star_df %&gt;%
  separate(Group, into = c(&quot;Response_Group&quot;, &quot;Concentration&quot;), sep = &quot; (?=0\\.)&quot;) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = group_levels))

# ----------------- Plot -----------------
ggplot(ddr_logfc, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~Concentration, scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Mean logFC of DDR Genes\nAcross CorMotif Response Groups&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20DDR%20logFC-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-DDR-logFC-1">
Past versions of "Mean DDR logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-DDR-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean DDR logFC-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-dna-damage-response-genes-across-corrmotifs-molecular-signature-dataset-faceted-by-concentration-timepoint"
class="section level2">
<h2><strong>📌 Mean logFC of DNA damage response genes across corrmotifs
(Molecular signature dataset) (Faceted by concentration ×
timepoint)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- DDR Entrez IDs -----------------
entrez_ids &lt;- c(
  10111, 1017, 1019, 1020, 1021, 1026, 1027, 10912, 11011, 1111,
  11200, 1385, 1643, 1647, 1869, 207, 2177, 25, 27113, 27244,
  3014, 317, 355, 4193, 4292, 4361, 4609, 4616, 4683, 472, 50484,
  5366, 5371, 54205, 545, 55367, 5591, 581, 5810, 5883, 5884,
  5888, 5893, 5925, 595, 5981, 6118, 637, 672, 7157, 7799,
  8243, 836, 841, 84126, 842, 8795, 891, 894, 896, 898,
  9133, 9134, 983, 9874, 993, 995, 5916
) %&gt;% as.character()

# ----------------- Load CorMotif Groups -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate Response Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Annotate DEG Data -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    ),
    Timepoint = factor(case_when(
      str_detect(Condition, &quot;_3$&quot;) ~ &quot;3 hr&quot;,
      str_detect(Condition, &quot;_24$&quot;) ~ &quot;24 hr&quot;,
      str_detect(Condition, &quot;_48$&quot;) ~ &quot;48 hr&quot;
    ), levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))
  )

# ----------------- Merge with CorMotif DDR Genes -----------------
ddr_all &lt;- inner_join(df_combined, deg_all, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response&quot;, &quot;CX-DOX mid-late response&quot;, &quot;DOX-specific response&quot;,
      &quot;DOX only mid-late response&quot;, &quot;CX total + DOX early response&quot;, &quot;DOX early + CX-DOX mid-late response&quot;
    ))
  )

# ----------------- Wilcoxon Test -----------------
comparison_groups &lt;- levels(ddr_all$Response_Group)[-1]  # exclude &quot;Non response&quot;
stat_results &lt;- list()

for (conc in unique(ddr_all$Concentration)) {
  for (tp in levels(ddr_all$Timepoint)) {
    sub &lt;- ddr_all %&gt;% filter(Concentration == conc, Timepoint == tp)
    for (grp in comparison_groups) {
      if (grp %in% unique(sub$Response_Group)) {
        test &lt;- tryCatch({
          wilcox.test(
            sub$logFC[sub$Response_Group == grp],
            sub$logFC[sub$Response_Group == &quot;Non response&quot;]
          )
        }, error = function(e) NULL)
        if (!is.null(test) &amp;&amp; !is.nan(test$p.value)) {
          stat_results[[length(stat_results) + 1]] &lt;- data.frame(
            Response_Group = grp,
            Concentration = conc,
            Timepoint = tp,
            p_value = signif(test$p.value, 3),
            label = case_when(
              test$p.value &lt; 0.001 ~ &quot;***&quot;,
              test$p.value &lt; 0.01 ~ &quot;**&quot;,
              test$p.value &lt; 0.05 ~ &quot;*&quot;,
              TRUE ~ &quot;&quot;
            ),
            y_pos = max(sub$logFC[sub$Response_Group == grp], na.rm = TRUE) + 0.5
          )
        }
      }
    }
  }
}

label_data &lt;- bind_rows(stat_results) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = levels(ddr_all$Response_Group)))

# ----------------- Plot -----------------
ggplot(ddr_all, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7, na.rm = TRUE) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 4, vjust = 0) +
  facet_grid(rows = vars(factor(Timepoint, levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))),
             cols = vars(Concentration), scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;DDR Gene logFC by CorMotif Group, Timepoint, and Concentration&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20DDR%20logFC1-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-DDR-logFC1-1">
Past versions of "Mean DDR logFC1-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-DDR-logFC1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean DDR logFC1-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-dna-damage-response-genes-across-corrmotifs-molecular-signature-dataset-faceted-by-concentration-timepoint-drug-cx-vs-dox-side-by-side"
class="section level2">
<h2><strong>📌 Mean logFC of DNA damage response genes across corrmotifs
(Molecular signature dataset) (Faceted by concentration × timepoint ×
drug (CX vs DOX side-by-side))</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- DDR Entrez IDs -----------------
ddr_ids &lt;- c(
  10111, 1017, 1019, 1020, 1021, 1026, 1027, 10912, 11011, 1111,
  11200, 1385, 1643, 1647, 1869, 207, 2177, 25, 27113, 27244,
  3014, 317, 355, 4193, 4292, 4361, 4609, 4616, 4683, 472, 50484,
  5366, 5371, 54205, 545, 55367, 5591, 581, 5810, 5883, 5884,
  5888, 5893, 5925, 595, 5981, 6118, 637, 672, 7157, 7799,
  8243, 836, 841, 84126, 842, 8795, 891, 894, 896, 898,
  9133, 9134, 983, 9874, 993, 995, 5916
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.1 µM) -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.1 µM DEG Files -----------------
deg_0.1 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.1 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late\nresponse (0.1 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% ddr_ids)

# ----------------- Set Factor Levels -----------------
deg_0.1 &lt;- deg_0.1 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.1 µM)&quot;,
      &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.1 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.1 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.1 µM DDR) -----------------
ggplot(deg_0.1, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;DDR Gene logFC by Drug and Response Group (0.1 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20DDR%20logFC3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-DDR-logFC3-1">
Past versions of "Mean DDR logFC3-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-DDR-logFC3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean DDR logFC3-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>################### DDR Genes — 0.5 µM ###################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- DDR Entrez IDs -----------------
ddr_ids &lt;- c(
  10111, 1017, 1019, 1020, 1021, 1026, 1027, 10912, 11011, 1111,
  11200, 1385, 1643, 1647, 1869, 207, 2177, 25, 27113, 27244,
  3014, 317, 355, 4193, 4292, 4361, 4609, 4616, 4683, 472, 50484,
  5366, 5371, 54205, 545, 55367, 5591, 581, 5810, 5883, 5884,
  5888, 5893, 5925, 595, 5981, 6118, 637, 672, 7157, 7799,
  8243, 836, 841, 84126, 842, 8795, 891, 894, 896, 898,
  9133, 9134, 983, 9874, 993, 995, 5916
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.5 µM) -----------------
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.5 µM DEG Files -----------------
deg_0.5 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.5 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 µM)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% ddr_ids)

# ----------------- Set Factor Levels -----------------
deg_0.5 &lt;- deg_0.5 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.5 µM)&quot;,
      &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.5 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.5 µM DDR) -----------------
ggplot(deg_0.5, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;DDR Gene logFC by Drug and Response Group (0.5 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20DDR%20logFC3-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-DDR-logFC3-2">
Past versions of "Mean DDR logFC3-2.png"
</button>
</p>
<div id="fig-no-spaces-Mean-DDR-logFC3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean DDR logFC3-2.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="proportion-of-p53-target-genes-across-corrmotif-clusters"
class="section level2">
<h2><strong>📌 Proportion of P53 target Genes across corrmotif
clusters</strong></h2>
<pre class="r"><code># ----------------- Load Required Libraries -----------------
library(dplyr)
library(ggplot2)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- P53 Entrez IDs -----------------
p53_entrez_ids &lt;- as.character(c(
  1026, 50484, 4193, 9766, 9518, 7832, 1643, 1647, 1263, 57103, 51065, 8795, 51499, 64393, 581, 5228, 5429, 8493, 55959,
  7508, 64782, 282991, 355, 53836, 4814, 10769, 9050, 27244, 9540, 94241, 26154, 57763, 900, 26999, 55332, 26263, 23479,
  23612, 29950, 9618, 10346, 8824, 134147, 55294, 22824, 4254, 6560, 467, 27113, 60492, 8444, 60401, 1969, 220965, 2232,
  3976, 55191, 84284, 93129, 5564, 7803, 83667, 7779, 132671, 7039, 51768, 137695, 93134, 7633, 10973, 340485, 307,
  27350, 23245, 3732, 29965, 1363, 1435, 196513, 8507, 8061, 2517, 51278, 53354, 54858, 23228, 5366, 5912, 6236, 51222,
  26152, 59, 1907, 50650, 91012, 780, 9249, 11072, 144455, 64787, 116151, 27165, 2876, 57822, 55733, 57722, 121457,
  375449, 85377, 4851, 5875, 127544, 29901, 84958, 8797, 8793, 441631, 220001, 54541, 5889, 5054, 25816, 25987, 5111,
  98, 317, 598, 604, 10904, 1294, 80315, 53944, 1606, 2770, 3628, 3675, 3985, 4035, 4163, 84552, 29085, 55367, 5371,
  5791, 54884, 5980, 8794, 1462, 50808, 220, 583, 694, 1056, 9076, 10978, 54677, 1612, 55040, 114907, 2274, 127707,
  4000, 8079, 4646, 4747, 27445, 5143, 80055, 79156, 5360, 5364, 23654, 5565, 5613, 5625, 10076, 56963, 6004, 390,
  255488, 6326, 6330, 23513, 7869, 283130, 204962, 83959, 6548, 6774, 9263, 10228, 22954, 10475, 85363, 494514, 10142,
  79714, 1006, 8446, 9648, 79828, 5507, 55240, 63874, 25841, 9289, 84883, 154810, 51321, 421, 8553, 655, 119032, 84280,
  10950, 824, 839, 57828, 857, 8812, 8837, 94027, 113189, 22837, 132864, 10898, 3300, 81704, 1847, 1849, 1947, 9538,
  24139, 5168, 147965, 115548, 9873, 23768, 2632, 2817, 3280, 3265, 23308, 3490, 51477, 182, 3856, 8844, 144811, 9404,
  4043, 9848, 2872, 23041, 740, 343263, 4638, 26509, 4792, 22861, 57523, 55214, 80025, 164091, 57060, 64065, 51090,
  5453, 8496, 333926, 55671, 5900, 55544, 23179, 8601, 389, 6223, 55800, 6385, 4088, 6643, 122809, 257397, 285343,
  7011, 54790, 374618, 55362, 51754, 7157, 9537, 22906, 7205, 80705, 219699, 55245, 83719, 7748, 25946, 118738
))

# ----------------- Load CorrMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% p53_entrez_ids, &quot;P53 Target Genes&quot;, &quot;Non-P53 Genes&quot;),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% p53_entrez_ids, &quot;P53 Target Genes&quot;, &quot;Non-P53 Genes&quot;),
    Concentration = &quot;0.5 µM&quot;
  )

# ----------------- Combine Data -----------------
df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Set Category Order (P53 first) -----------------
df_combined$Category &lt;- factor(df_combined$Category, levels = c(&quot;P53 Target Genes&quot;, &quot;Non-P53 Genes&quot;))

# ----------------- Calculate Proportions -----------------
proportion_data &lt;- df_combined %&gt;%
  group_by(Concentration, Response_Group, Category) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Chi-square or Fisher&#39;s Test -----------------
get_chi_or_fisher &lt;- function(df, ref_group) {
  ref_counts &lt;- df %&gt;%
    ungroup() %&gt;%
    filter(Response_Group == ref_group) %&gt;%
    { ref_tbl &lt;- .[ , c(&quot;Category&quot;, &quot;Count&quot;)]; setNames(ref_tbl$Count, ref_tbl$Category) }
  
  df %&gt;%
    filter(Response_Group != ref_group) %&gt;%
    group_by(Response_Group) %&gt;%
    summarise(
      p_value = {
        group_counts &lt;- Count[Category %in% c(&quot;P53 Target Genes&quot;, &quot;Non-P53 Genes&quot;)]
        if (length(group_counts) &lt; 2) group_counts &lt;- c(group_counts, 0)
        contingency_table &lt;- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts[&quot;P53 Target Genes&quot;], ref_counts[&quot;Non-P53 Genes&quot;]
        ), nrow = 2)
        if (any(contingency_table &lt; 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = &quot;drop&quot;
    ) %&gt;%
    mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))
}

# Run test for each concentration
chi_0.1 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.1 µM&quot;), &quot;Non response&quot;)
chi_0.5 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.5 µM&quot;), &quot;Non response&quot;)</code></pre>
<pre><code>Warning: There was 1 warning in `summarise()`.
ℹ In argument: `p_value = { ... }`.
ℹ In group 2: `Response_Group = &quot;DOX early + CX-DOX mid-late response&quot;`.
Caused by warning in `chisq.test()`:
! Chi-squared approximation may be incorrect</code></pre>
<pre class="r"><code>chi_all &lt;- bind_rows(
  chi_0.1 %&gt;% mutate(Concentration = &quot;0.1 µM&quot;),
  chi_0.5 %&gt;% mutate(Concentration = &quot;0.5 µM&quot;)
)

# Merge test results into proportions
proportion_data &lt;- left_join(proportion_data, chi_all, by = c(&quot;Concentration&quot;, &quot;Response_Group&quot;))

# ----------------- Reorder Factor Levels -----------------
proportion_data$Concentration &lt;- factor(proportion_data$Concentration, levels = c(&quot;0.1 µM&quot;, &quot;0.5 µM&quot;))
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
))

# ----------------- Star Label Position -----------------
label_data &lt;- proportion_data %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  summarise(Significance = dplyr::first(Significance), .groups = &quot;drop&quot;) %&gt;%
  filter(!is.na(Significance) &amp; Significance != &quot;&quot;) %&gt;%
  mutate(y_pos = 105)

# ----------------- Final Plot -----------------
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(
    data = label_data,
    aes(x = Response_Group, y = y_pos, label = Significance),
    inherit.aes = FALSE,
    size = 5,
    color = &quot;black&quot;
  ) +
  facet_wrap(~ Concentration, scales = &quot;free_x&quot;) +
  scale_fill_manual(values = c(
    &quot;P53 Target Genes&quot; = &quot;#E41A1C&quot;,
    &quot;Non-P53 Genes&quot; = &quot;#377EB8&quot;
  )) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +
  labs(
    title = &quot;P53 Target Gene Proportions Across\nCormotif Clusters (0.1 and 0.5)&quot;,
    x = &quot;Response Groups&quot;,
    y = &quot;Percentage of Genes&quot;,
    fill = &quot;Gene Category&quot;
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = &quot;bold&quot;),
    axis.title.x = element_text(size = 14, face = &quot;bold&quot;),
    axis.title.y = element_text(size = 14, face = &quot;bold&quot;),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    panel.spacing = unit(1.2, &quot;lines&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/prop%20P53%200.1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-prop-P53-0.1-1">
Past versions of "prop P53 0.1-1.png"
</button>
</p>
<div id="fig-no-spaces-prop-P53-0.1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/prop P53 0.1-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/prop P53 0.1-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/81247f64f08df43c2b40a386707fc8e9cc240143/docs/figure/Corrmotif_Conc.Rmd/prop P53 0.1-1.png" target="_blank">81247f6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/96d0db145cf68e6db0d39e70bad9af0f75793aae/docs/figure/Corrmotif_Conc.Rmd/prop P53 0.1-1.png" target="_blank">96d0db1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-p53-target-genes-across-corrmotifs-per-concentration"
class="section level2">
<h2><strong>📌 Mean logFC of P53 target genes across corrmotifs (per
concentration)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- P53 Entrez IDs -----------------
entrez_ids &lt;- c(
  1026, 50484, 4193, 9766, 9518, 7832, 1643, 1647, 1263, 57103, 51065, 8795, 51499, 64393, 581, 5228, 5429, 8493, 55959,
  7508, 64782, 282991, 355, 53836, 4814, 10769, 9050, 27244, 9540, 94241, 26154, 57763, 900, 26999, 55332, 26263, 23479,
  23612, 29950, 9618, 10346, 8824, 134147, 55294, 22824, 4254, 6560, 467, 27113, 60492, 8444, 60401, 1969, 220965, 2232,
  3976, 55191, 84284, 93129, 5564, 7803, 83667, 7779, 132671, 7039, 51768, 137695, 93134, 7633, 10973, 340485, 307,
  27350, 23245, 3732, 29965, 1363, 1435, 196513, 8507, 8061, 2517, 51278, 53354, 54858, 23228, 5366, 5912, 6236, 51222,
  26152, 59, 1907, 50650, 91012, 780, 9249, 11072, 144455, 64787, 116151, 27165, 2876, 57822, 55733, 57722, 121457,
  375449, 85377, 4851, 5875, 127544, 29901, 84958, 8797, 8793, 441631, 220001, 54541, 5889, 5054, 25816, 25987, 5111,
  98, 317, 598, 604, 10904, 1294, 80315, 53944, 1606, 2770, 3628, 3675, 3985, 4035, 4163, 84552, 29085, 55367, 5371,
  5791, 54884, 5980, 8794, 1462, 50808, 220, 583, 694, 1056, 9076, 10978, 54677, 1612, 55040, 114907, 2274, 127707,
  4000, 8079, 4646, 4747, 27445, 5143, 80055, 79156, 5360, 5364, 23654, 5565, 5613, 5625, 10076, 56963, 6004, 390,
  255488, 6326, 6330, 23513, 7869, 283130, 204962, 83959, 6548, 6774, 9263, 10228, 22954, 10475, 85363, 494514, 10142,
  79714, 1006, 8446, 9648, 79828, 5507, 55240, 63874, 25841, 9289, 84883, 154810, 51321, 421, 8553, 655, 119032, 84280,
  10950, 824, 839, 57828, 857, 8812, 8837, 94027, 113189, 22837, 132864, 10898, 3300, 81704, 1847, 1849, 1947, 9538,
  24139, 5168, 147965, 115548, 9873, 23768, 2632, 2817, 3280, 3265, 23308, 3490, 51477, 182, 3856, 8844, 144811, 9404,
  4043, 9848, 2872, 23041, 740, 343263, 4638, 26509, 4792, 22861, 57523, 55214, 80025, 164091, 57060, 64065, 51090,
  5453, 8496, 333926, 55671, 5900, 55544, 23179, 8601, 389, 6223, 55800, 6385, 4088, 6643, 122809, 257397, 285343,
  7011, 54790, 374618, 55362, 51754, 7157, 9537, 22906, 7205, 80705, 219699, 55245, 83719, 7748, 25946, 118738
)

# ----------------- Load CorMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate Response Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% as.character(entrez_ids)) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% as.character(entrez_ids)) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Aggregate DEG Data -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;,
      TRUE ~ NA_character_
    )
  )

deg_mean &lt;- deg_all %&gt;%
  group_by(Entrez_ID, Concentration) %&gt;%
  summarise(logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;)

# ----------------- Merge with CorMotif P53 Genes -----------------
p53_logfc &lt;- inner_join(df_combined, deg_mean, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Group = paste(Response_Group, Concentration)
  )

# ----------------- Define order -----------------
group_levels &lt;- c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
)

p53_logfc$Response_Group &lt;- factor(p53_logfc$Response_Group, levels = group_levels)

# ----------------- Wilcoxon Test -----------------
comparison_map &lt;- list(
  &quot;CX-DOX mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX only mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX-specific response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX only mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;CX total + DOX early response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX early + CX-DOX mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;
)

star_df &lt;- lapply(names(comparison_map), function(resp_group) {
  control_group &lt;- comparison_map[[resp_group]]
  resp_vals &lt;- p53_logfc$logFC[p53_logfc$Group == resp_group]
  control_vals &lt;- p53_logfc$logFC[p53_logfc$Group == control_group]

  if (length(resp_vals) &gt;= 2 &amp;&amp; length(control_vals) &gt;= 2) {
    test_result &lt;- wilcox.test(resp_vals, control_vals)
    pval &lt;- test_result$p.value
    if (pval &lt; 0.05) {
      label &lt;- case_when(
        pval &lt; 0.001 ~ &quot;***&quot;,
        pval &lt; 0.01 ~ &quot;**&quot;,
        TRUE ~ &quot;*&quot;
      )
      y_pos &lt;- max(resp_vals, na.rm = TRUE) + 0.5
      return(data.frame(Group = resp_group, y_pos = y_pos, label = label, P_Value = signif(pval, 4)))
    }
  }
  return(NULL)
}) %&gt;% bind_rows()


# ----------------- Prepare Annotation -----------------
label_data &lt;- star_df %&gt;%
  separate(Group, into = c(&quot;Response_Group&quot;, &quot;Concentration&quot;), sep = &quot; (?=0\\.)&quot;) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = group_levels))




# ----------------- Plot -----------------
ggplot(p53_logfc, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~Concentration, scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Mean logFC of P53 Target Genes\nAcross CorMotif Response Groups&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20P53%20logFC-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-P53-logFC-1">
Past versions of "Mean P53 logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-P53-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean P53 logFC-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-p53-target-genes-across-corrmotifs-faceted-by-concentration-timepoint"
class="section level2">
<h2><strong>📌 Mean logFC of P53 target genes across corrmotifs (Faceted
by concentration × timepoint)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- P53 Entrez IDs -----------------
entrez_ids &lt;- c(
  1026, 50484, 4193, 9766, 9518, 7832, 1643, 1647, 1263, 57103, 51065, 8795, 51499, 64393, 581, 5228, 5429, 8493, 55959,
  7508, 64782, 282991, 355, 53836, 4814, 10769, 9050, 27244, 9540, 94241, 26154, 57763, 900, 26999, 55332, 26263, 23479,
  23612, 29950, 9618, 10346, 8824, 134147, 55294, 22824, 4254, 6560, 467, 27113, 60492, 8444, 60401, 1969, 220965, 2232,
  3976, 55191, 84284, 93129, 5564, 7803, 83667, 7779, 132671, 7039, 51768, 137695, 93134, 7633, 10973, 340485, 307,
  27350, 23245, 3732, 29965, 1363, 1435, 196513, 8507, 8061, 2517, 51278, 53354, 54858, 23228, 5366, 5912, 6236, 51222,
  26152, 59, 1907, 50650, 91012, 780, 9249, 11072, 144455, 64787, 116151, 27165, 2876, 57822, 55733, 57722, 121457,
  375449, 85377, 4851, 5875, 127544, 29901, 84958, 8797, 8793, 441631, 220001, 54541, 5889, 5054, 25816, 25987, 5111,
  98, 317, 598, 604, 10904, 1294, 80315, 53944, 1606, 2770, 3628, 3675, 3985, 4035, 4163, 84552, 29085, 55367, 5371,
  5791, 54884, 5980, 8794, 1462, 50808, 220, 583, 694, 1056, 9076, 10978, 54677, 1612, 55040, 114907, 2274, 127707,
  4000, 8079, 4646, 4747, 27445, 5143, 80055, 79156, 5360, 5364, 23654, 5565, 5613, 5625, 10076, 56963, 6004, 390,
  255488, 6326, 6330, 23513, 7869, 283130, 204962, 83959, 6548, 6774, 9263, 10228, 22954, 10475, 85363, 494514, 10142,
  79714, 1006, 8446, 9648, 79828, 5507, 55240, 63874, 25841, 9289, 84883, 154810, 51321, 421, 8553, 655, 119032, 84280,
  10950, 824, 839, 57828, 857, 8812, 8837, 94027, 113189, 22837, 132864, 10898, 3300, 81704, 1847, 1849, 1947, 9538,
  24139, 5168, 147965, 115548, 9873, 23768, 2632, 2817, 3280, 3265, 23308, 3490, 51477, 182, 3856, 8844, 144811, 9404,
  4043, 9848, 2872, 23041, 740, 343263, 4638, 26509, 4792, 22861, 57523, 55214, 80025, 164091, 57060, 64065, 51090,
  5453, 8496, 333926, 55671, 5900, 55544, 23179, 8601, 389, 6223, 55800, 6385, 4088, 6643, 122809, 257397, 285343,
  7011, 54790, 374618, 55362, 51754, 7157, 9537, 22906, 7205, 80705, 219699, 55245, 83719, 7748, 25946, 118738
) %&gt;% as.character()

# ----------------- Load CorMotif Response Group Data -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorMotif Group Membership -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Annotate DEG Files -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    ),
    Timepoint = factor(case_when(
      str_detect(Condition, &quot;_3$&quot;) ~ &quot;3 hr&quot;,
      str_detect(Condition, &quot;_24$&quot;) ~ &quot;24 hr&quot;,
      str_detect(Condition, &quot;_48$&quot;) ~ &quot;48 hr&quot;
    ), levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))
  )

# ----------------- Merge with P53 Targets and Annotate -----------------
p53_all &lt;- inner_join(df_combined, deg_all, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID, column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response&quot;, &quot;CX-DOX mid-late response&quot;, &quot;DOX-specific response&quot;,
      &quot;DOX only mid-late response&quot;, &quot;CX total + DOX early response&quot;, &quot;DOX early + CX-DOX mid-late response&quot;
    ))
  )

# ----------------- Wilcoxon Test -----------------
comparison_groups &lt;- levels(p53_all$Response_Group)[-1]
stat_results &lt;- list()

for (conc in unique(p53_all$Concentration)) {
  for (tp in levels(p53_all$Timepoint)) {
    sub &lt;- p53_all %&gt;% filter(Concentration == conc, Timepoint == tp)
    for (grp in comparison_groups) {
      if (grp %in% unique(sub$Response_Group)) {
        test &lt;- tryCatch({
          wilcox.test(
            sub$logFC[sub$Response_Group == grp],
            sub$logFC[sub$Response_Group == &quot;Non response&quot;]
          )
        }, error = function(e) NULL)
        if (!is.null(test) &amp;&amp; !is.nan(test$p.value)) {
          stat_results[[length(stat_results) + 1]] &lt;- data.frame(
            Response_Group = grp,
            Concentration = conc,
            Timepoint = tp,
            p_value = signif(test$p.value, 3),
            label = case_when(
              test$p.value &lt; 0.001 ~ &quot;***&quot;,
              test$p.value &lt; 0.01 ~ &quot;**&quot;,
              test$p.value &lt; 0.05 ~ &quot;*&quot;,
              TRUE ~ &quot;&quot;
            ),
            y_pos = max(sub$logFC[sub$Response_Group == grp], na.rm = TRUE) + 0.5
          )
        }
      }
    }
  }
}

label_data &lt;- bind_rows(stat_results) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = levels(p53_all$Response_Group)))

# ----------------- Plot -----------------
ggplot(p53_all, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7, na.rm = TRUE) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 4, vjust = 0) +
  facet_grid(rows = vars(factor(Timepoint, levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))),
             cols = vars(Concentration), scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;P53 Target Gene logFC by\nCorMotif Group, Timepoint, and Concentration&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20P53%20logFC1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-P53-logFC1-1">
Past versions of "Mean P53 logFC1-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-P53-logFC1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean P53 logFC1-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-p53-target-genes-across-corrmotifs-faceted-by-concentration-timepoint-drug-cx-vs-dox-side-by-side"
class="section level2">
<h2><strong>📌 Mean logFC of P53 target genes across corrmotifs (Faceted
by concentration × timepoint × drug (CX vs DOX
side-by-side))</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- P53 Target Entrez IDs -----------------
p53_ids &lt;- c(
  1026, 50484, 4193, 9766, 9518, 7832, 1643, 1647, 1263, 57103, 51065, 8795, 51499, 64393, 581, 5228, 5429, 8493, 55959,
  7508, 64782, 282991, 355, 53836, 4814, 10769, 9050, 27244, 9540, 94241, 26154, 57763, 900, 26999, 55332, 26263, 23479,
  23612, 29950, 9618, 10346, 8824, 134147, 55294, 22824, 4254, 6560, 467, 27113, 60492, 8444, 60401, 1969, 220965, 2232,
  3976, 55191, 84284, 93129, 5564, 7803, 83667, 7779, 132671, 7039, 51768, 137695, 93134, 7633, 10973, 340485, 307,
  27350, 23245, 3732, 29965, 1363, 1435, 196513, 8507, 8061, 2517, 51278, 53354, 54858, 23228, 5366, 5912, 6236, 51222,
  26152, 59, 1907, 50650, 91012, 780, 9249, 11072, 144455, 64787, 116151, 27165, 2876, 57822, 55733, 57722, 121457,
  375449, 85377, 4851, 5875, 127544, 29901, 84958, 8797, 8793, 441631, 220001, 54541, 5889, 5054, 25816, 25987, 5111,
  98, 317, 598, 604, 10904, 1294, 80315, 53944, 1606, 2770, 3628, 3675, 3985, 4035, 4163, 84552, 29085, 55367, 5371,
  5791, 54884, 5980, 8794, 1462, 50808, 220, 583, 694, 1056, 9076, 10978, 54677, 1612, 55040, 114907, 2274, 127707,
  4000, 8079, 4646, 4747, 27445, 5143, 80055, 79156, 5360, 5364, 23654, 5565, 5613, 5625, 10076, 56963, 6004, 390,
  255488, 6326, 6330, 23513, 7869, 283130, 204962, 83959, 6548, 6774, 9263, 10228, 22954, 10475, 85363, 494514, 10142,
  79714, 1006, 8446, 9648, 79828, 5507, 55240, 63874, 25841, 9289, 84883, 154810, 51321, 421, 8553, 655, 119032, 84280,
  10950, 824, 839, 57828, 857, 8812, 8837, 94027, 113189, 22837, 132864, 10898, 3300, 81704, 1847, 1849, 1947, 9538,
  24139, 5168, 147965, 115548, 9873, 23768, 2632, 2817, 3280, 3265, 23308, 3490, 51477, 182, 3856, 8844, 144811, 9404,
  4043, 9848, 2872, 23041, 740, 343263, 4638, 26509, 4792, 22861, 57523, 55214, 80025, 164091, 57060, 64065, 51090,
  5453, 8496, 333926, 55671, 5900, 55544, 23179, 8601, 389, 6223, 55800, 6385, 4088, 6643, 122809, 257397, 285343,
  7011, 54790, 374618, 55362, 51754, 7157, 9537, 22906, 7205, 80705, 219699, 55245, 83719, 7748, 25946, 118738
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.1 µM) -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.1 µM DEG Files -----------------
deg_0.1 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.1 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late\nresponse (0.1 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% p53_ids)

# ----------------- Set Factor Levels -----------------
deg_0.1 &lt;- deg_0.1 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.1 µM)&quot;,
      &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.1 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.1 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.1 µM P53) -----------------
ggplot(deg_0.1, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;P53 Target Gene logFC by Drug and Response Group (0.1 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20P53%20logFC3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-P53-logFC3-1">
Past versions of "Mean P53 logFC3-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-P53-logFC3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean P53 logFC3-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>################### P53 Target Genes — 0.5 µM ###################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- P53 Target Entrez IDs -----------------
p53_ids &lt;- c(
  1026, 50484, 4193, 9766, 9518, 7832, 1643, 1647, 1263, 57103, 51065, 8795, 51499, 64393, 581, 5228, 5429, 8493, 55959,
  7508, 64782, 282991, 355, 53836, 4814, 10769, 9050, 27244, 9540, 94241, 26154, 57763, 900, 26999, 55332, 26263, 23479,
  23612, 29950, 9618, 10346, 8824, 134147, 55294, 22824, 4254, 6560, 467, 27113, 60492, 8444, 60401, 1969, 220965, 2232,
  3976, 55191, 84284, 93129, 5564, 7803, 83667, 7779, 132671, 7039, 51768, 137695, 93134, 7633, 10973, 340485, 307,
  27350, 23245, 3732, 29965, 1363, 1435, 196513, 8507, 8061, 2517, 51278, 53354, 54858, 23228, 5366, 5912, 6236, 51222,
  26152, 59, 1907, 50650, 91012, 780, 9249, 11072, 144455, 64787, 116151, 27165, 2876, 57822, 55733, 57722, 121457,
  375449, 85377, 4851, 5875, 127544, 29901, 84958, 8797, 8793, 441631, 220001, 54541, 5889, 5054, 25816, 25987, 5111,
  98, 317, 598, 604, 10904, 1294, 80315, 53944, 1606, 2770, 3628, 3675, 3985, 4035, 4163, 84552, 29085, 55367, 5371,
  5791, 54884, 5980, 8794, 1462, 50808, 220, 583, 694, 1056, 9076, 10978, 54677, 1612, 55040, 114907, 2274, 127707,
  4000, 8079, 4646, 4747, 27445, 5143, 80055, 79156, 5360, 5364, 23654, 5565, 5613, 5625, 10076, 56963, 6004, 390,
  255488, 6326, 6330, 23513, 7869, 283130, 204962, 83959, 6548, 6774, 9263, 10228, 22954, 10475, 85363, 494514, 10142,
  79714, 1006, 8446, 9648, 79828, 5507, 55240, 63874, 25841, 9289, 84883, 154810, 51321, 421, 8553, 655, 119032, 84280,
  10950, 824, 839, 57828, 857, 8812, 8837, 94027, 113189, 22837, 132864, 10898, 3300, 81704, 1847, 1849, 1947, 9538,
  24139, 5168, 147965, 115548, 9873, 23768, 2632, 2817, 3280, 3265, 23308, 3490, 51477, 182, 3856, 8844, 144811, 9404,
  4043, 9848, 2872, 23041, 740, 343263, 4638, 26509, 4792, 22861, 57523, 55214, 80025, 164091, 57060, 64065, 51090,
  5453, 8496, 333926, 55671, 5900, 55544, 23179, 8601, 389, 6223, 55800, 6385, 4088, 6643, 122809, 257397, 285343,
  7011, 54790, 374618, 55362, 51754, 7157, 9537, 22906, 7205, 80705, 219699, 55245, 83719, 7748, 25946, 118738
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.5 µM) -----------------
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.5 µM DEG Files -----------------
deg_0.5 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.5 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 µM)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% p53_ids)

# ----------------- Set Factor Levels -----------------
deg_0.5 &lt;- deg_0.5 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.5 µM)&quot;,
      &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.5 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.5 µM P53) -----------------
ggplot(deg_0.5, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;P53 Target Gene logFC by Drug and Response Group (0.5 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20P53%20logFC3-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-P53-logFC3-2">
Past versions of "Mean P53 logFC3-2.png"
</button>
</p>
<div id="fig-no-spaces-Mean-P53-logFC3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean P53 logFC3-2.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="top2b-target-genes-proportions-across-corrmotifs"
class="section level2">
<h2><strong>📌 TOP2B target genes proportions across
corrmotifs</strong></h2>
<pre class="r"><code># ----------------- Load Required Libraries -----------------
library(dplyr)
library(ggplot2)
library(readr)

# ----------------- Load TOP2B Entrez IDs -----------------
top2b_entrez_ids &lt;- read_csv(&quot;data/TOP2B_target_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorrMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% top2b_entrez_ids, &quot;TOP2B Target Genes&quot;, &quot;Non-TOP2B Genes&quot;),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% top2b_entrez_ids, &quot;TOP2B Target Genes&quot;, &quot;Non-TOP2B Genes&quot;),
    Concentration = &quot;0.5 µM&quot;
  )

# ----------------- Combine Data -----------------
df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Set Category Order (TOP2B first) -----------------
df_combined$Category &lt;- factor(df_combined$Category, levels = c(&quot;TOP2B Target Genes&quot;, &quot;Non-TOP2B Genes&quot;))

# ----------------- Calculate Proportions -----------------
proportion_data &lt;- df_combined %&gt;%
  group_by(Concentration, Response_Group, Category) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Chi-square or Fisher&#39;s Test -----------------
get_chi_or_fisher &lt;- function(df, ref_group) {
  ref_counts &lt;- df %&gt;%
    ungroup() %&gt;%
    filter(Response_Group == ref_group) %&gt;%
    { ref_tbl &lt;- .[ , c(&quot;Category&quot;, &quot;Count&quot;)]; setNames(ref_tbl$Count, ref_tbl$Category) }
  
  df %&gt;%
    filter(Response_Group != ref_group) %&gt;%
    group_by(Response_Group) %&gt;%
    summarise(
      p_value = {
        group_counts &lt;- Count[Category %in% c(&quot;TOP2B Target Genes&quot;, &quot;Non-TOP2B Genes&quot;)]
        if (length(group_counts) &lt; 2) group_counts &lt;- c(group_counts, 0)
        contingency_table &lt;- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts[&quot;TOP2B Target Genes&quot;], ref_counts[&quot;Non-TOP2B Genes&quot;]
        ), nrow = 2)
        if (any(contingency_table &lt; 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = &quot;drop&quot;
    ) %&gt;%
    mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))
}

# Run test for each concentration
chi_0.1 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.1 µM&quot;), &quot;Non response&quot;)
chi_0.5 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.5 µM&quot;), &quot;Non response&quot;)

chi_all &lt;- bind_rows(
  chi_0.1 %&gt;% mutate(Concentration = &quot;0.1 µM&quot;),
  chi_0.5 %&gt;% mutate(Concentration = &quot;0.5 µM&quot;)
)

# Merge test results into proportions
proportion_data &lt;- left_join(proportion_data, chi_all, by = c(&quot;Concentration&quot;, &quot;Response_Group&quot;))

# ----------------- Reorder Factor Levels -----------------
proportion_data$Concentration &lt;- factor(proportion_data$Concentration, levels = c(&quot;0.1 µM&quot;, &quot;0.5 µM&quot;))
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
))

# ----------------- Star Label Position -----------------
label_data &lt;- proportion_data %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  summarise(Significance = dplyr::first(Significance), .groups = &quot;drop&quot;) %&gt;%
  filter(!is.na(Significance) &amp; Significance != &quot;&quot;) %&gt;%
  mutate(y_pos = 105)

# ----------------- Final Plot -----------------
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(
    data = label_data,
    aes(x = Response_Group, y = y_pos, label = Significance),
    inherit.aes = FALSE,
    size = 5,
    color = &quot;black&quot;
  ) +
  facet_wrap(~ Concentration, scales = &quot;free_x&quot;) +
  scale_fill_manual(values = c(
    &quot;TOP2B Target Genes&quot; = &quot;#E41A1C&quot;,
    &quot;Non-TOP2B Genes&quot; = &quot;#377EB8&quot;
  )) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +
  labs(
    title = &quot;TOP2B Target Gene Proportions Across\nCormotif Clusters (0.1 and 0.5)&quot;,
    x = &quot;Response Groups&quot;,
    y = &quot;Percentage of Genes&quot;,
    fill = &quot;Gene Category&quot;
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = &quot;bold&quot;),
    axis.title.x = element_text(size = 14, face = &quot;bold&quot;),
    axis.title.y = element_text(size = 14, face = &quot;bold&quot;),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    panel.spacing = unit(1.2, &quot;lines&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/TOP2B%20Prop-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-TOP2B-Prop-1">
Past versions of "TOP2B Prop-1.png"
</button>
</p>
<div id="fig-no-spaces-TOP2B-Prop-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/TOP2B Prop-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-top2b-target-genes-across-corrmotifs-per-concentration"
class="section level2">
<h2><strong>📌 Mean logFC of TOP2B target genes across corrmotifs (per
concentration)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- TOP2B Entrez IDs -----------------
top2b_entrez_ids &lt;- read_csv(&quot;data/TOP2B_target_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate Response Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% top2b_entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% top2b_entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Aggregate DEG Data -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;,
      TRUE ~ NA_character_
    )
  )

deg_mean &lt;- deg_all %&gt;%
  group_by(Entrez_ID, Concentration) %&gt;%
  summarise(logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;)

# ----------------- Merge with CorMotif TOP2B Genes -----------------
top2b_logfc &lt;- inner_join(df_combined, deg_mean, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Group = paste(Response_Group, Concentration)
  )

# ----------------- Define order -----------------
group_levels &lt;- c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
)

top2b_logfc$Response_Group &lt;- factor(top2b_logfc$Response_Group, levels = group_levels)

# ----------------- Wilcoxon Test -----------------
comparison_map &lt;- list(
  &quot;CX-DOX mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX only mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX-specific response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX only mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;CX total + DOX early response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX early + CX-DOX mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;
)

star_df &lt;- lapply(names(comparison_map), function(resp_group) {
  control_group &lt;- comparison_map[[resp_group]]
  resp_vals &lt;- top2b_logfc$logFC[top2b_logfc$Group == resp_group]
  control_vals &lt;- top2b_logfc$logFC[top2b_logfc$Group == control_group]

  if (length(resp_vals) &gt;= 2 &amp;&amp; length(control_vals) &gt;= 2) {
    test_result &lt;- wilcox.test(resp_vals, control_vals)
    pval &lt;- test_result$p.value
    if (pval &lt; 0.05) {
      label &lt;- case_when(
        pval &lt; 0.001 ~ &quot;***&quot;,
        pval &lt; 0.01 ~ &quot;**&quot;,
        TRUE ~ &quot;*&quot;
      )
      y_pos &lt;- max(resp_vals, na.rm = TRUE) + 0.5
      return(data.frame(Group = resp_group, y_pos = y_pos, label = label, P_Value = signif(pval, 4)))
    }
  }
  return(NULL)
}) %&gt;% bind_rows()

# ----------------- Prepare Annotation -----------------
label_data &lt;- star_df %&gt;%
  separate(Group, into = c(&quot;Response_Group&quot;, &quot;Concentration&quot;), sep = &quot; (?=0\\.)&quot;) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = group_levels))

# ----------------- Plot -----------------
ggplot(top2b_logfc, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~Concentration, scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Mean logFC of TOP2B Target Genes\nAcross CorMotif Response Groups&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2B%20logFC-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2B-logFC-1">
Past versions of "Mean TOP2B logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2B-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2B logFC-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-top2b-target-genes-across-corrmotifs-faceted-by-concentration-timepoint"
class="section level2">
<h2><strong>📌 Mean logFC of TOP2B target genes across corrmotifs
(Faceted by concentration × timepoint)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- TOP2B Entrez IDs -----------------
top2b_entrez_ids &lt;- read_csv(&quot;data/TOP2B_target_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Response Group Data -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorMotif Group Membership -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% top2b_entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% top2b_entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Annotate DEG Files -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    ),
    Timepoint = factor(case_when(
      str_detect(Condition, &quot;_3$&quot;) ~ &quot;3 hr&quot;,
      str_detect(Condition, &quot;_24$&quot;) ~ &quot;24 hr&quot;,
      str_detect(Condition, &quot;_48$&quot;) ~ &quot;48 hr&quot;
    ), levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))
  )

# ----------------- Merge with TOP2B Targets and Annotate -----------------
top2b_all &lt;- inner_join(df_combined, deg_all, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  filter(Entrez_ID %in% top2b_entrez_ids) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID, column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response&quot;, &quot;CX-DOX mid-late response&quot;, &quot;DOX-specific response&quot;,
      &quot;DOX only mid-late response&quot;, &quot;CX total + DOX early response&quot;, &quot;DOX early + CX-DOX mid-late response&quot;
    ))
  )

# ----------------- Wilcoxon Test -----------------
comparison_groups &lt;- levels(top2b_all$Response_Group)[-1]
stat_results &lt;- list()

for (conc in unique(top2b_all$Concentration)) {
  for (tp in levels(top2b_all$Timepoint)) {
    sub &lt;- top2b_all %&gt;% filter(Concentration == conc, Timepoint == tp)
    for (grp in comparison_groups) {
      if (grp %in% unique(sub$Response_Group)) {
        test &lt;- tryCatch({
          wilcox.test(
            sub$logFC[sub$Response_Group == grp],
            sub$logFC[sub$Response_Group == &quot;Non response&quot;]
          )
        }, error = function(e) NULL)
        if (!is.null(test) &amp;&amp; !is.nan(test$p.value)) {
          stat_results[[length(stat_results) + 1]] &lt;- data.frame(
            Response_Group = grp,
            Concentration = conc,
            Timepoint = tp,
            p_value = signif(test$p.value, 3),
            label = case_when(
              test$p.value &lt; 0.001 ~ &quot;***&quot;,
              test$p.value &lt; 0.01 ~ &quot;**&quot;,
              test$p.value &lt; 0.05 ~ &quot;*&quot;,
              TRUE ~ &quot;&quot;
            ),
            y_pos = max(sub$logFC[sub$Response_Group == grp], na.rm = TRUE) + 0.5
          )
        }
      }
    }
  }
}

label_data &lt;- bind_rows(stat_results) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = levels(top2b_all$Response_Group)))

# ----------------- Plot -----------------
ggplot(top2b_all, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7, na.rm = TRUE) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 4, vjust = 0) +
  facet_grid(rows = vars(factor(Timepoint, levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))),
             cols = vars(Concentration), scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;TOP2B Target Gene logFC by\nCorMotif Group, Timepoint, and Concentration&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2B%20logFC1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2B-logFC1-1">
Past versions of "Mean TOP2B logFC1-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2B-logFC1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2B logFC1-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-top2b-target-genes-across-corrmotifs-faceted-by-concentration-timepoint-drug-cx-vs-dox-side-by-side"
class="section level2">
<h2><strong>📌 Mean logFC of TOP2B target genes across corrmotifs
(Faceted by concentration × timepoint × drug (CX vs DOX
side-by-side))</strong></h2>
<pre class="r"><code>################### TOP2B Target Genes — 0.1 µM ###################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load TOP2B Target Entrez IDs -----------------
top2b_entrez_ids &lt;- read_csv(&quot;data/TOP2B_target_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Groups (0.1 µM) -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.1 µM DEG Files -----------------
deg_0.1 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.1 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late\nresponse (0.1 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% top2b_entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.1 &lt;- deg_0.1 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.1 µM)&quot;,
      &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.1 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.1 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.1 µM TOP2B) -----------------
ggplot(deg_0.1, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;TOP2B Target Gene logFC by Drug and Response Group (0.1 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2B%20logFC3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2B-logFC3-1">
Past versions of "Mean TOP2B logFC3-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2B-logFC3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2B logFC3-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>################### TOP2B Target Genes — 0.5 µM ###################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load TOP2B Target Entrez IDs -----------------
top2b_entrez_ids &lt;- read_csv(&quot;data/TOP2B_target_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Groups (0.5 µM) -----------------
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.5 µM DEG Files -----------------
deg_0.5 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.5 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 µM)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% top2b_entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.5 &lt;- deg_0.5 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.5 µM)&quot;,
      &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.5 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.5 µM TOP2B) -----------------
ggplot(deg_0.5, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;TOP2B Target Gene logFC by Drug and Response Group (0.5 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2B%20logFC3-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2B-logFC3-2">
Past versions of "Mean TOP2B logFC3-2.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2B-logFC3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2B logFC3-2.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="top2a-target-genes-chip_atlas-proportion-across-corrmotifs"
class="section level2">
<h2><strong>📌 TOP2A target genes (ChIP_Atlas) proportion across
corrmotifs</strong></h2>
<pre class="r"><code># ----------------- Load Required Libraries -----------------
library(dplyr)
library(ggplot2)
library(readr)

# ----------------- Load TOP2A Entrez IDs -----------------
top2a_entrez_ids &lt;- read_csv(&quot;data/TOP2A_target_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorrMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% top2a_entrez_ids, &quot;TOP2A Target Genes&quot;, &quot;Non-TOP2A Genes&quot;),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% top2a_entrez_ids, &quot;TOP2A Target Genes&quot;, &quot;Non-TOP2A Genes&quot;),
    Concentration = &quot;0.5 µM&quot;
  )

# ----------------- Combine Data -----------------
df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Set Category Order -----------------
df_combined$Category &lt;- factor(df_combined$Category, levels = c(&quot;TOP2A Target Genes&quot;, &quot;Non-TOP2A Genes&quot;))

# ----------------- Calculate Proportions -----------------
proportion_data &lt;- df_combined %&gt;%
  group_by(Concentration, Response_Group, Category) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Chi-square or Fisher&#39;s Test -----------------
get_chi_or_fisher &lt;- function(df, ref_group) {
  ref_counts &lt;- df %&gt;%
    ungroup() %&gt;%
    filter(Response_Group == ref_group) %&gt;%
    { ref_tbl &lt;- .[ , c(&quot;Category&quot;, &quot;Count&quot;)]; setNames(ref_tbl$Count, ref_tbl$Category) }
  
  df %&gt;%
    filter(Response_Group != ref_group) %&gt;%
    group_by(Response_Group) %&gt;%
    summarise(
      p_value = {
        group_counts &lt;- Count[Category %in% c(&quot;TOP2A Target Genes&quot;, &quot;Non-TOP2A Genes&quot;)]
        if (length(group_counts) &lt; 2) group_counts &lt;- c(group_counts, 0)
        contingency_table &lt;- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts[&quot;TOP2A Target Genes&quot;], ref_counts[&quot;Non-TOP2A Genes&quot;]
        ), nrow = 2)
        if (any(contingency_table &lt; 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = &quot;drop&quot;
    ) %&gt;%
    mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))
}

# Run test for each concentration
chi_0.1 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.1 µM&quot;), &quot;Non response&quot;)
chi_0.5 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.5 µM&quot;), &quot;Non response&quot;)</code></pre>
<pre><code>Warning: There was 1 warning in `summarise()`.
ℹ In argument: `p_value = { ... }`.
ℹ In group 4: `Response_Group = &quot;DOX-specific response&quot;`.
Caused by warning in `chisq.test()`:
! Chi-squared approximation may be incorrect</code></pre>
<pre class="r"><code>chi_all &lt;- bind_rows(
  chi_0.1 %&gt;% mutate(Concentration = &quot;0.1 µM&quot;),
  chi_0.5 %&gt;% mutate(Concentration = &quot;0.5 µM&quot;)
)

# Merge test results into proportions
proportion_data &lt;- left_join(proportion_data, chi_all, by = c(&quot;Concentration&quot;, &quot;Response_Group&quot;))

# ----------------- Reorder Factor Levels -----------------
proportion_data$Concentration &lt;- factor(proportion_data$Concentration, levels = c(&quot;0.1 µM&quot;, &quot;0.5 µM&quot;))
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
))

# ----------------- Star Label Position -----------------
label_data &lt;- proportion_data %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  summarise(Significance = dplyr::first(Significance), .groups = &quot;drop&quot;) %&gt;%
  filter(!is.na(Significance) &amp; Significance != &quot;&quot;) %&gt;%
  mutate(y_pos = 105)

# ----------------- Final Plot -----------------
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(
    data = label_data,
    aes(x = Response_Group, y = y_pos, label = Significance),
    inherit.aes = FALSE,
    size = 5,
    color = &quot;black&quot;
  ) +
  facet_wrap(~ Concentration, scales = &quot;free_x&quot;) +
  scale_fill_manual(values = c(
    &quot;TOP2A Target Genes&quot; = &quot;#E41A1C&quot;,
    &quot;Non-TOP2A Genes&quot; = &quot;#377EB8&quot;
  )) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +
  labs(
    title = &quot;TOP2A Target Gene Proportions Across\nCormotif Clusters (0.1 and 0.5)&quot;,
    x = &quot;Response Groups&quot;,
    y = &quot;Percentage of Genes&quot;,
    fill = &quot;Gene Category&quot;
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = &quot;bold&quot;),
    axis.title.x = element_text(size = 14, face = &quot;bold&quot;),
    axis.title.y = element_text(size = 14, face = &quot;bold&quot;),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    panel.spacing = unit(1.2, &quot;lines&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2A_Prop-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2A_Prop-1">
Past versions of "Mean TOP2A_Prop-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2A_Prop-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2A_Prop-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-top2a-target-genes-chip_atlas-across-corrmotifs-per-concentration"
class="section level2">
<h2><strong>📌 Mean logFC of TOP2A target genes (ChIP_Atlas) across
corrmotifs (per concentration)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load TOP2A Entrez IDs -----------------
top2a_entrez_ids &lt;- read_csv(&quot;data/TOP2A_target_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate Response Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% top2a_entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% top2a_entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Aggregate DEG Data -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;,
      TRUE ~ NA_character_
    )
  )

deg_mean &lt;- deg_all %&gt;%
  group_by(Entrez_ID, Concentration) %&gt;%
  summarise(logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;)

# ----------------- Merge with CorMotif TOP2A Genes -----------------
top2a_logfc &lt;- inner_join(df_combined, deg_mean, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Group = paste(Response_Group, Concentration)
  )

# ----------------- Define Response Group Order -----------------
group_levels &lt;- c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
)

top2a_logfc$Response_Group &lt;- factor(top2a_logfc$Response_Group, levels = group_levels)

# ----------------- Wilcoxon Test with Output -----------------
comparison_map &lt;- list(
  &quot;CX-DOX mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX only mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX-specific response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX only mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;CX total + DOX early response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX early + CX-DOX mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;
)

wilcoxon_results &lt;- lapply(names(comparison_map), function(resp_group) {
  control_group &lt;- comparison_map[[resp_group]]
  resp_vals &lt;- top2a_logfc$logFC[top2a_logfc$Group == resp_group]
  control_vals &lt;- top2a_logfc$logFC[top2a_logfc$Group == control_group]

  if (length(resp_vals) &gt;= 2 &amp;&amp; length(control_vals) &gt;= 2) {
    test_result &lt;- wilcox.test(resp_vals, control_vals)
    pval &lt;- test_result$p.value
    stat &lt;- test_result$statistic
    label &lt;- case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01  ~ &quot;**&quot;,
      pval &lt; 0.05  ~ &quot;*&quot;,
      TRUE ~ &quot;&quot;
    )
    y_pos &lt;- max(resp_vals, na.rm = TRUE) + 0.5
    return(data.frame(
      Response_Group = resp_group,
      Control_Group = control_group,
      W_statistic = as.numeric(stat),
      P_value = signif(pval, 4),
      Significance = label,
      y_pos = y_pos
    ))
  } else {
    return(data.frame(
      Response_Group = resp_group,
      Control_Group = control_group,
      W_statistic = NA,
      P_value = NA,
      Significance = &quot;&quot;,
      y_pos = NA
    ))
  }
}) %&gt;% bind_rows()

# ----------------- Prepare Annotation -----------------
label_data &lt;- wilcoxon_results %&gt;%
  separate(Response_Group, into = c(&quot;Response_Group&quot;, &quot;Concentration&quot;), sep = &quot; (?=0\\.)&quot;) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = group_levels))

# ----------------- Plot -----------------
ggplot(top2a_logfc, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = Significance),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~Concentration, scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Mean logFC of TOP2A Target Genes\nAcross CorMotif Response Groups (ChIP atlas)&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2A_ChIP%20logFC-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2A_ChIP-logFC-1">
Past versions of "Mean TOP2A_ChIP logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2A_ChIP-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2A_ChIP logFC-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-top2a-target-genes-chip_atlas-across-corrmotifs-faceted-by-concentration-timepoint"
class="section level2">
<h2><strong>📌 Mean logFC of TOP2A target genes (ChIP_Atlas) across
corrmotifs (Faceted by concentration × timepoint)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load TOP2A Entrez IDs -----------------
top2a_entrez_ids &lt;- read_csv(&quot;data/TOP2A_target_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Response Group Data -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorMotif Group Membership -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% top2a_entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% top2a_entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Annotate DEG Files -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    ),
    Timepoint = factor(case_when(
      str_detect(Condition, &quot;_3$&quot;) ~ &quot;3 hr&quot;,
      str_detect(Condition, &quot;_24$&quot;) ~ &quot;24 hr&quot;,
      str_detect(Condition, &quot;_48$&quot;) ~ &quot;48 hr&quot;
    ), levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))
  )

# ----------------- Merge with TOP2A Targets and Annotate -----------------
top2a_all &lt;- inner_join(df_combined, deg_all, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  filter(Entrez_ID %in% top2a_entrez_ids) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID, column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response&quot;, &quot;CX-DOX mid-late response&quot;, &quot;DOX-specific response&quot;,
      &quot;DOX only mid-late response&quot;, &quot;CX total + DOX early response&quot;, &quot;DOX early + CX-DOX mid-late response&quot;
    ))
  )

# ----------------- Wilcoxon Test -----------------
comparison_groups &lt;- levels(top2a_all$Response_Group)[-1]
stat_results &lt;- list()

for (conc in unique(top2a_all$Concentration)) {
  for (tp in levels(top2a_all$Timepoint)) {
    sub &lt;- top2a_all %&gt;% filter(Concentration == conc, Timepoint == tp)
    for (grp in comparison_groups) {
      if (grp %in% unique(sub$Response_Group)) {
        test &lt;- tryCatch({
          wilcox.test(
            sub$logFC[sub$Response_Group == grp],
            sub$logFC[sub$Response_Group == &quot;Non response&quot;]
          )
        }, error = function(e) NULL)
        if (!is.null(test) &amp;&amp; !is.nan(test$p.value)) {
          stat_results[[length(stat_results) + 1]] &lt;- data.frame(
            Response_Group = grp,
            Concentration = conc,
            Timepoint = tp,
            p_value = signif(test$p.value, 3),
            label = case_when(
              test$p.value &lt; 0.001 ~ &quot;***&quot;,
              test$p.value &lt; 0.01 ~ &quot;**&quot;,
              test$p.value &lt; 0.05 ~ &quot;*&quot;,
              TRUE ~ &quot;&quot;
            ),
            y_pos = max(sub$logFC[sub$Response_Group == grp], na.rm = TRUE) + 0.5
          )
        }
      }
    }
  }
}

label_data &lt;- bind_rows(stat_results) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = levels(top2a_all$Response_Group)))

# ----------------- Plot -----------------
ggplot(top2a_all, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7, na.rm = TRUE) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 4, vjust = 0) +
  facet_grid(rows = vars(factor(Timepoint, levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))),
             cols = vars(Concentration), scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;TOP2A Target Gene logFC by\nCorMotif Group, Timepoint, and Concentration (ChIP Atlas)&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2A_ChIP_Atlas%20logFC1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2A_ChIP_Atlas-logFC1-1">
Past versions of "Mean TOP2A_ChIP_Atlas logFC1-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2A_ChIP_Atlas-logFC1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2A_ChIP_Atlas logFC1-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-top2a-target-genes-chip_atlas-across-corrmotifs-faceted-by-concentration-timepoint-drug-cx-vs-dox-side-by-side"
class="section level2">
<h2><strong>📌 Mean logFC of TOP2A target genes (ChIP_Atlas) across
corrmotifs (Faceted by concentration × timepoint × drug (CX vs DOX
side-by-side))</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load TOP2A ChIP-Atlas Entrez IDs -----------------
top2a_entrez_ids &lt;- read_csv(&quot;data/TOP2A_target_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Groups (0.1 µM) -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.1 µM DEG Files -----------------
deg_0.1 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.1 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late\nresponse (0.1 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% top2a_entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.1 &lt;- deg_0.1 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.1 µM)&quot;,
      &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.1 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.1 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.1 µM TOP2A) -----------------
ggplot(deg_0.1, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;TOP2A Target Gene logFC by Drug and Response Group (0.1 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2A_ChIP_Atlas%20logFC3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2A_ChIP_Atlas-logFC3-1">
Past versions of "Mean TOP2A_ChIP_Atlas logFC3-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2A_ChIP_Atlas-logFC3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2A_ChIP_Atlas logFC3-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>################### TOP2A Target Genes (ChIP-Atlas) — 0.5 µM ###################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load TOP2A Entrez IDs -----------------
top2a_entrez_ids &lt;- read_csv(&quot;data/TOP2A_target_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Groups (0.5 µM) -----------------
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.5 µM DEG Files -----------------
deg_0.5 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.5 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 µM)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% top2a_entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.5 &lt;- deg_0.5 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.5 µM)&quot;,
      &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.5 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.5 µM TOP2A ChIP-Atlas) -----------------
ggplot(deg_0.5, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;TOP2A (ChIP-Atlas) Target Gene logFC by Drug and Response Group (0.5 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2A_ChIP_Atlas%20logFC3-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2A_ChIP_Atlas-logFC3-2">
Past versions of "Mean TOP2A_ChIP_Atlas logFC3-2.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2A_ChIP_Atlas-logFC3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2A_ChIP_Atlas logFC3-2.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="proportion-of-top2a-target-genes-rpe-1-across-corrmotifs"
class="section level2">
<h2><strong>📌 Proportion of TOP2A target genes (RPE-1) across
corrmotifs</strong></h2>
<pre class="r"><code># ----------------- Load Required Libraries -----------------
library(dplyr)
library(ggplot2)
library(readr)

# ----------------- Load TOP2A (RPE-1) Entrez IDs -----------------
top2a_lit_ids &lt;- read_csv(&quot;data/TOP2A_target_lit_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorrMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% top2a_lit_ids, &quot;TOP2A (RPE-1) Targets&quot;, &quot;Non-TOP2A Genes&quot;),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% top2a_lit_ids, &quot;TOP2A (RPE-1) Targets&quot;, &quot;Non-TOP2A Genes&quot;),
    Concentration = &quot;0.5 µM&quot;
  )

# ----------------- Combine Data -----------------
df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Set Category Order -----------------
df_combined$Category &lt;- factor(df_combined$Category, levels = c(&quot;TOP2A (RPE-1) Targets&quot;, &quot;Non-TOP2A Genes&quot;))

# ----------------- Calculate Proportions -----------------
proportion_data &lt;- df_combined %&gt;%
  group_by(Concentration, Response_Group, Category) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Chi-square or Fisher&#39;s Test -----------------
get_chi_or_fisher &lt;- function(df, ref_group) {
  ref_counts &lt;- df %&gt;%
    ungroup() %&gt;%
    filter(Response_Group == ref_group) %&gt;%
    { ref_tbl &lt;- .[ , c(&quot;Category&quot;, &quot;Count&quot;)]; setNames(ref_tbl$Count, ref_tbl$Category) }
  
  df %&gt;%
    filter(Response_Group != ref_group) %&gt;%
    group_by(Response_Group) %&gt;%
    summarise(
      p_value = {
        group_counts &lt;- Count[Category %in% c(&quot;TOP2A (RPE-1) Targets&quot;, &quot;Non-TOP2A Genes&quot;)]
        if (length(group_counts) &lt; 2) group_counts &lt;- c(group_counts, 0)
        contingency_table &lt;- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts[&quot;TOP2A (RPE-1) Targets&quot;], ref_counts[&quot;Non-TOP2A Genes&quot;]
        ), nrow = 2)
        if (any(contingency_table &lt; 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = &quot;drop&quot;
    ) %&gt;%
    mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))
}

# Run test for each concentration
chi_0.1 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.1 µM&quot;), &quot;Non response&quot;)
chi_0.5 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.5 µM&quot;), &quot;Non response&quot;)

chi_all &lt;- bind_rows(
  chi_0.1 %&gt;% mutate(Concentration = &quot;0.1 µM&quot;),
  chi_0.5 %&gt;% mutate(Concentration = &quot;0.5 µM&quot;)
)

# Merge test results into proportions
proportion_data &lt;- left_join(proportion_data, chi_all, by = c(&quot;Concentration&quot;, &quot;Response_Group&quot;))

# ----------------- Reorder Factor Levels -----------------
proportion_data$Concentration &lt;- factor(proportion_data$Concentration, levels = c(&quot;0.1 µM&quot;, &quot;0.5 µM&quot;))
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
))

# ----------------- Star Label Position -----------------
label_data &lt;- proportion_data %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  summarise(Significance = dplyr::first(Significance), .groups = &quot;drop&quot;) %&gt;%
  filter(!is.na(Significance) &amp; Significance != &quot;&quot;) %&gt;%
  mutate(y_pos = 105)

# ----------------- Final Plot -----------------
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(
    data = label_data,
    aes(x = Response_Group, y = y_pos, label = Significance),
    inherit.aes = FALSE,
    size = 5,
    color = &quot;black&quot;
  ) +
  facet_wrap(~ Concentration, scales = &quot;free_x&quot;) +
  scale_fill_manual(values = c(
    &quot;TOP2A (RPE-1) Targets&quot; = &quot;#E41A1C&quot;,
    &quot;Non-TOP2A Genes&quot; = &quot;#377EB8&quot;
  )) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +
  labs(
    title = &quot;TOP2A (RPE-1) Target Gene Proportions Across\nCormotif Clusters (0.1 and 0.5)&quot;,
    x = &quot;Response Groups&quot;,
    y = &quot;Percentage of Genes&quot;,
    fill = &quot;Gene Category&quot;
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = &quot;bold&quot;),
    axis.title.x = element_text(size = 14, face = &quot;bold&quot;),
    axis.title.y = element_text(size = 14, face = &quot;bold&quot;),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    panel.spacing = unit(1.2, &quot;lines&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/TOP2A_lit_prop%20logFC-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-TOP2A_lit_prop-logFC-1">
Past versions of "TOP2A_lit_prop logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-TOP2A_lit_prop-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/TOP2A_lit_prop logFC-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-top2a-target-genes-rpe-1-across-corrmotifs-per-concentration"
class="section level2">
<h2><strong>📌 Mean logFC of TOP2A target genes (RPE-1) across
corrmotifs (per concentration)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load TOP2A (RPE-1) Entrez IDs -----------------
top2a_lit_ids &lt;- read_csv(&quot;data/TOP2A_target_lit_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate Response Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% top2a_lit_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% top2a_lit_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Aggregate DEG Data -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;,
      TRUE ~ NA_character_
    )
  )

deg_mean &lt;- deg_all %&gt;%
  group_by(Entrez_ID, Concentration) %&gt;%
  summarise(logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;)

# ----------------- Merge with CorMotif Genes -----------------
top2a_lit_logfc &lt;- inner_join(df_combined, deg_mean, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Group = paste(Response_Group, Concentration)
  )

# ----------------- Define Response Group Order -----------------
group_levels &lt;- c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
)

top2a_lit_logfc$Response_Group &lt;- factor(top2a_lit_logfc$Response_Group, levels = group_levels)

# ----------------- Wilcoxon Test with Output -----------------
comparison_map &lt;- list(
  &quot;CX-DOX mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX only mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX-specific response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX only mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;CX total + DOX early response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX early + CX-DOX mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;
)

wilcoxon_results &lt;- lapply(names(comparison_map), function(resp_group) {
  control_group &lt;- comparison_map[[resp_group]]
  resp_vals &lt;- top2a_lit_logfc$logFC[top2a_lit_logfc$Group == resp_group]
  control_vals &lt;- top2a_lit_logfc$logFC[top2a_lit_logfc$Group == control_group]

  if (length(resp_vals) &gt;= 2 &amp;&amp; length(control_vals) &gt;= 2) {
    test_result &lt;- wilcox.test(resp_vals, control_vals)
    pval &lt;- test_result$p.value
    stat &lt;- test_result$statistic
    label &lt;- case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01  ~ &quot;**&quot;,
      pval &lt; 0.05  ~ &quot;*&quot;,
      TRUE ~ &quot;&quot;
    )
    y_pos &lt;- max(resp_vals, na.rm = TRUE) + 0.5
    return(data.frame(
      Response_Group = resp_group,
      Control_Group = control_group,
      W_statistic = as.numeric(stat),
      P_value = signif(pval, 4),
      Significance = label,
      y_pos = y_pos
    ))
  } else {
    return(data.frame(
      Response_Group = resp_group,
      Control_Group = control_group,
      W_statistic = NA,
      P_value = NA,
      Significance = &quot;&quot;,
      y_pos = NA
    ))
  }
}) %&gt;% bind_rows()

# ----------------- Prepare Annotation -----------------
label_data &lt;- wilcoxon_results %&gt;%
  separate(Response_Group, into = c(&quot;Response_Group&quot;, &quot;Concentration&quot;), sep = &quot; (?=0\\.)&quot;) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = group_levels))

# ----------------- Plot -----------------
ggplot(top2a_lit_logfc, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = Significance),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~Concentration, scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Mean logFC of Literature-Based TOP2A Targets (RPE-1)\nAcross CorMotif Response Groups&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2A_lit_1%20logFC-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2A_lit_1-logFC-1">
Past versions of "Mean TOP2A_lit_1 logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2A_lit_1-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2A_lit_1 logFC-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-top2a-target-genes-rpe-1-across-corrmotifs-faceted-by-concentration-timepoint"
class="section level2">
<h2><strong>📌 Mean logFC of TOP2A target genes (RPE-1) across
corrmotifs (Faceted by concentration × timepoint)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load TOP2A (RPE-1) Entrez IDs -----------------
top2a_lit_ids &lt;- read_csv(&quot;data/TOP2A_target_lit_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Response Group Data -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorMotif Group Membership -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% top2a_lit_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% top2a_lit_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Annotate DEG Files -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    ),
    Timepoint = factor(case_when(
      str_detect(Condition, &quot;_3$&quot;) ~ &quot;3 hr&quot;,
      str_detect(Condition, &quot;_24$&quot;) ~ &quot;24 hr&quot;,
      str_detect(Condition, &quot;_48$&quot;) ~ &quot;48 hr&quot;
    ), levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))
  )

# ----------------- Merge with TOP2A (Lit) Targets and Annotate -----------------
top2a_lit_all &lt;- inner_join(df_combined, deg_all, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  filter(Entrez_ID %in% top2a_lit_ids) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID, column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response&quot;, &quot;CX-DOX mid-late response&quot;, &quot;DOX-specific response&quot;,
      &quot;DOX only mid-late response&quot;, &quot;CX total + DOX early response&quot;, &quot;DOX early + CX-DOX mid-late response&quot;
    ))
  )

# ----------------- Wilcoxon Test -----------------
comparison_groups &lt;- levels(top2a_lit_all$Response_Group)[-1]
stat_results &lt;- list()

for (conc in unique(top2a_lit_all$Concentration)) {
  for (tp in levels(top2a_lit_all$Timepoint)) {
    sub &lt;- top2a_lit_all %&gt;% filter(Concentration == conc, Timepoint == tp)
    for (grp in comparison_groups) {
      if (grp %in% unique(sub$Response_Group)) {
        test &lt;- tryCatch({
          wilcox.test(
            sub$logFC[sub$Response_Group == grp],
            sub$logFC[sub$Response_Group == &quot;Non response&quot;]
          )
        }, error = function(e) NULL)
        if (!is.null(test) &amp;&amp; !is.nan(test$p.value)) {
          stat_results[[length(stat_results) + 1]] &lt;- data.frame(
            Response_Group = grp,
            Concentration = conc,
            Timepoint = tp,
            p_value = signif(test$p.value, 3),
            label = case_when(
              test$p.value &lt; 0.001 ~ &quot;***&quot;,
              test$p.value &lt; 0.01 ~ &quot;**&quot;,
              test$p.value &lt; 0.05 ~ &quot;*&quot;,
              TRUE ~ &quot;&quot;
            ),
            y_pos = max(sub$logFC[sub$Response_Group == grp], na.rm = TRUE) + 0.5
          )
        }
      }
    }
  }
}

label_data &lt;- bind_rows(stat_results) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = levels(top2a_lit_all$Response_Group)))

# ----------------- Plot -----------------
ggplot(top2a_lit_all, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7, na.rm = TRUE) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 4, vjust = 0) +
  facet_grid(rows = vars(factor(Timepoint, levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))),
             cols = vars(Concentration), scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;TOP2A Target Gene (RPE-1 cell) logFC by\nCorMotif Group, Timepoint, and Concentration&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2A_lit_2%20logFC1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2A_lit_2-logFC1-1">
Past versions of "Mean TOP2A_lit_2 logFC1-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2A_lit_2-logFC1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2A_lit_2 logFC1-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-top2a-target-genes-rpe-1-across-corrmotifs-faceted-by-concentration-timepoint-drug-cx-vs-dox-side-by-side"
class="section level2">
<h2><strong>📌 Mean logFC of TOP2A target genes (RPE-1) across
corrmotifs (Faceted by concentration × timepoint × drug (CX vs DOX
side-by-side))</strong></h2>
<pre class="r"><code>################### TOP2A Target Genes (RPE-1 Literature) — 0.1 µM ###################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load TOP2A (RPE-1) Entrez IDs -----------------
top2a_lit_ids &lt;- read_csv(&quot;data/TOP2A_target_lit_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Groups (0.1 µM) -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.1 µM DEG Files -----------------
deg_0.1 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.1 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late\nresponse (0.1 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% top2a_lit_ids)

# ----------------- Set Factor Levels -----------------
deg_0.1 &lt;- deg_0.1 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.1 µM)&quot;,
      &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.1 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.1 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.1 µM TOP2A RPE-1) -----------------
ggplot(deg_0.1, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;TOP2A (RPE-1) Target Gene logFC by Drug and Response Group (0.1 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2A_lit_3%20logFC3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2A_lit_3-logFC3-1">
Past versions of "Mean TOP2A_lit_3 logFC3-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2A_lit_3-logFC3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2A_lit_3 logFC3-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>################### TOP2A Target Genes (RPE-1 Literature) — 0.5 µM ###################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load TOP2A (RPE-1) Entrez IDs -----------------
top2a_lit_ids &lt;- read_csv(&quot;data/TOP2A_target_lit_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Groups (0.5 µM) -----------------
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.5 µM DEG Files -----------------
deg_0.5 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.5 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 µM)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% top2a_lit_ids)

# ----------------- Set Factor Levels -----------------
deg_0.5 &lt;- deg_0.5 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.5 µM)&quot;,
      &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.5 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.5 µM TOP2A RPE-1) -----------------
ggplot(deg_0.5, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;TOP2A (RPE-1) Target Gene logFC by Drug and Response Group (0.5 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2A_lit_3%20logFC3-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2A_lit_3-logFC3-2">
Past versions of "Mean TOP2A_lit_3 logFC3-2.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2A_lit_3-logFC3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2A_lit_3 logFC3-2.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="proportion-of-anthracycline-cardiotoxicity-genes-across-corrmotifs"
class="section level2">
<h2><strong>📌 Proportion of Anthracycline Cardiotoxicity genes across
corrmotifs</strong></h2>
<pre class="r"><code># ----------------- Load Required Libraries -----------------
library(dplyr)
library(ggplot2)

# ----------------- AC Cardiotoxicity Entrez IDs -----------------
ac_cardio_ids &lt;- c(
  6272, 8029, 11128, 79899, 54477, 121665, 5095, 22863, 57161, 4692,
  8214, 23151, 56606, 108, 22999, 56895, 9603, 3181, 4023, 10499,
  92949, 4363, 10057, 5243, 5244, 5880, 1535, 2950, 847, 5447,
  3038, 3077, 4846, 3958, 23327, 29899, 23155, 80856, 55020, 78996,
  23262, 150383, 9620, 79730, 344595, 5066, 6251, 3482, 9588, 339416,
  7292, 55157, 87769, 23409, 720, 3107, 54535, 1590, 80059, 7991,
  57110, 8803, 323, 54826, 5916, 23371, 283337, 64078, 80010, 1933,
  10818, 51020
) %&gt;% as.character()

# ----------------- Load CorrMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% ac_cardio_ids, &quot;AC-Toxicity Genes&quot;, &quot;Non-AC Genes&quot;),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% ac_cardio_ids, &quot;AC-Toxicity Genes&quot;, &quot;Non-AC Genes&quot;),
    Concentration = &quot;0.5 µM&quot;
  )

# ----------------- Combine Data -----------------
df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Set Category Order -----------------
df_combined$Category &lt;- factor(df_combined$Category, levels = c(&quot;AC-Toxicity Genes&quot;, &quot;Non-AC Genes&quot;))

# ----------------- Calculate Proportions -----------------
proportion_data &lt;- df_combined %&gt;%
  group_by(Concentration, Response_Group, Category) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Chi-square or Fisher&#39;s Test -----------------
get_chi_or_fisher &lt;- function(df, ref_group) {
  ref_counts &lt;- df %&gt;%
    ungroup() %&gt;%
    filter(Response_Group == ref_group) %&gt;%
    { ref_tbl &lt;- .[ , c(&quot;Category&quot;, &quot;Count&quot;)]; setNames(ref_tbl$Count, ref_tbl$Category) }
  
  df %&gt;%
    filter(Response_Group != ref_group) %&gt;%
    group_by(Response_Group) %&gt;%
    summarise(
      p_value = {
        group_counts &lt;- Count[Category %in% c(&quot;AC-Toxicity Genes&quot;, &quot;Non-AC Genes&quot;)]
        if (length(group_counts) &lt; 2) group_counts &lt;- c(group_counts, 0)
        contingency_table &lt;- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts[&quot;AC-Toxicity Genes&quot;], ref_counts[&quot;Non-AC Genes&quot;]
        ), nrow = 2)
        if (any(contingency_table &lt; 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = &quot;drop&quot;
    ) %&gt;%
    mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))
}

# Run test for each concentration
chi_0.1 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.1 µM&quot;), &quot;Non response&quot;)
chi_0.5 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.5 µM&quot;), &quot;Non response&quot;)

chi_all &lt;- bind_rows(
  chi_0.1 %&gt;% mutate(Concentration = &quot;0.1 µM&quot;),
  chi_0.5 %&gt;% mutate(Concentration = &quot;0.5 µM&quot;)
)

# Merge test results into proportions
proportion_data &lt;- left_join(proportion_data, chi_all, by = c(&quot;Concentration&quot;, &quot;Response_Group&quot;))

# ----------------- Reorder Factor Levels -----------------
proportion_data$Concentration &lt;- factor(proportion_data$Concentration, levels = c(&quot;0.1 µM&quot;, &quot;0.5 µM&quot;))
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
))

# ----------------- Star Label Position -----------------
label_data &lt;- proportion_data %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  summarise(Significance = dplyr::first(Significance), .groups = &quot;drop&quot;) %&gt;%
  filter(!is.na(Significance) &amp; Significance != &quot;&quot;) %&gt;%
  mutate(y_pos = 105)

# ----------------- Final Plot -----------------
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(
    data = label_data,
    aes(x = Response_Group, y = y_pos, label = Significance),
    inherit.aes = FALSE,
    size = 5,
    color = &quot;black&quot;
  ) +
  facet_wrap(~ Concentration, scales = &quot;free_x&quot;) +
  scale_fill_manual(values = c(
    &quot;AC-Toxicity Genes&quot; = &quot;#E41A1C&quot;,
    &quot;Non-AC Genes&quot; = &quot;#377EB8&quot;
  )) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +
  labs(
    title = &quot;AC Cardiotoxicity Gene Proportions Across\nCormotif Clusters (0.1 and 0.5)&quot;,
    x = &quot;Response Groups&quot;,
    y = &quot;Percentage of Genes&quot;,
    fill = &quot;Gene Category&quot;
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = &quot;bold&quot;),
    axis.title.x = element_text(size = 14, face = &quot;bold&quot;),
    axis.title.y = element_text(size = 14, face = &quot;bold&quot;),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    panel.spacing = unit(1.2, &quot;lines&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Ac_tox%20prop-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Ac_tox-prop-1">
Past versions of "Ac_tox prop-1.png"
</button>
</p>
<div id="fig-no-spaces-Ac_tox-prop-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Ac_tox prop-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-anthracycline-cardiotoxicity-genes-across-corrmotifs-per-concentration"
class="section level2">
<h2><strong>📌 Mean logFC of Anthracycline Cardiotoxicity genes across
corrmotifs (per concentration)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- AC Cardiotoxicity Entrez IDs -----------------
entrez_ids &lt;- c(
  6272, 8029, 11128, 79899, 54477, 121665, 5095, 22863, 57161, 4692,
  8214, 23151, 56606, 108, 22999, 56895, 9603, 3181, 4023, 10499,
  92949, 4363, 10057, 5243, 5244, 5880, 1535, 2950, 847, 5447,
  3038, 3077, 4846, 3958, 23327, 29899, 23155, 80856, 55020, 78996,
  23262, 150383, 9620, 79730, 344595, 5066, 6251, 3482, 9588, 339416,
  7292, 55157, 87769, 23409, 720, 3107, 54535, 1590, 80059, 7991,
  57110, 8803, 323, 54826, 5916, 23371, 283337, 64078, 80010, 1933,
  10818, 51020
) %&gt;% as.character()

# ----------------- Load CorMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate Response Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Aggregate DEG Data -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;,
      TRUE ~ NA_character_
    )
  )

deg_mean &lt;- deg_all %&gt;%
  group_by(Entrez_ID, Concentration) %&gt;%
  summarise(logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;)

# ----------------- Merge with CorMotif Genes -----------------
ac_logfc &lt;- inner_join(df_combined, deg_mean, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Group = paste(Response_Group, Concentration)
  )

# ----------------- Define Group Order -----------------
group_levels &lt;- c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
)

ac_logfc$Response_Group &lt;- factor(ac_logfc$Response_Group, levels = group_levels)

# ----------------- Wilcoxon Test -----------------
comparison_map &lt;- list(
  &quot;CX-DOX mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX only mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX-specific response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX only mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;CX total + DOX early response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX early + CX-DOX mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;
)

wilcoxon_results &lt;- lapply(names(comparison_map), function(resp_group) {
  control_group &lt;- comparison_map[[resp_group]]
  resp_vals &lt;- ac_logfc$logFC[ac_logfc$Group == resp_group]
  control_vals &lt;- ac_logfc$logFC[ac_logfc$Group == control_group]

  if (length(resp_vals) &gt;= 2 &amp;&amp; length(control_vals) &gt;= 2) {
    test_result &lt;- wilcox.test(resp_vals, control_vals)
    pval &lt;- test_result$p.value
    stat &lt;- test_result$statistic
    label &lt;- case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01  ~ &quot;**&quot;,
      pval &lt; 0.05  ~ &quot;*&quot;,
      TRUE ~ &quot;&quot;
    )
    y_pos &lt;- max(resp_vals, na.rm = TRUE) + 0.5
    return(data.frame(
      Group = resp_group,
      y_pos = y_pos,
      label = label,
      P_Value = signif(pval, 4)
    ))
  } else {
    return(NULL)
  }
}) %&gt;% bind_rows()

# ----------------- Prepare Annotation -----------------
label_data &lt;- wilcoxon_results %&gt;%
  separate(Group, into = c(&quot;Response_Group&quot;, &quot;Concentration&quot;), sep = &quot; (?=0\\.)&quot;) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = group_levels))

# ----------------- Plot -----------------
ggplot(ac_logfc, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~Concentration, scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Mean logFC of AC Cardiotoxicity Genes\nAcross CorMotif Response Groups&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20Ac_tox%20logFC-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-Ac_tox-logFC-1">
Past versions of "Mean Ac_tox logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-Ac_tox-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean Ac_tox logFC-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-anthracycline-cardiotoxicity-genes-across-corrmotifs-faceted-by-concentration-timepoint"
class="section level2">
<h2><strong>📌 Mean logFC of Anthracycline Cardiotoxicity genes across
corrmotifs (Faceted by concentration × timepoint)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- AC Cardiotoxicity Entrez IDs -----------------
entrez_ids &lt;- c(
  6272, 8029, 11128, 79899, 54477, 121665, 5095, 22863, 57161, 4692,
  8214, 23151, 56606, 108, 22999, 56895, 9603, 3181, 4023, 10499,
  92949, 4363, 10057, 5243, 5244, 5880, 1535, 2950, 847, 5447,
  3038, 3077, 4846, 3958, 23327, 29899, 23155, 80856, 55020, 78996,
  23262, 150383, 9620, 79730, 344595, 5066, 6251, 3482, 9588, 339416,
  7292, 55157, 87769, 23409, 720, 3107, 54535, 1590, 80059, 7991,
  57110, 8803, 323, 54826, 5916, 23371, 283337, 64078, 80010, 1933,
  10818, 51020
) %&gt;% as.character()

# ----------------- Load CorMotif Response Group Data -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorMotif Group Membership -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Annotate DEG Files -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    ),
    Timepoint = factor(case_when(
      str_detect(Condition, &quot;_3$&quot;) ~ &quot;3 hr&quot;,
      str_detect(Condition, &quot;_24$&quot;) ~ &quot;24 hr&quot;,
      str_detect(Condition, &quot;_48$&quot;) ~ &quot;48 hr&quot;
    ), levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))
  )

# ----------------- Merge with AC Cardiotoxicity Targets -----------------
ac_all &lt;- inner_join(df_combined, deg_all, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID, column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response&quot;, &quot;CX-DOX mid-late response&quot;, &quot;DOX-specific response&quot;,
      &quot;DOX only mid-late response&quot;, &quot;CX total + DOX early response&quot;, &quot;DOX early + CX-DOX mid-late response&quot;
    ))
  )

# ----------------- Wilcoxon Test -----------------
comparison_groups &lt;- levels(ac_all$Response_Group)[-1]
stat_results &lt;- list()

for (conc in unique(ac_all$Concentration)) {
  for (tp in levels(ac_all$Timepoint)) {
    sub &lt;- ac_all %&gt;% filter(Concentration == conc, Timepoint == tp)
    for (grp in comparison_groups) {
      if (grp %in% unique(sub$Response_Group)) {
        test &lt;- tryCatch({
          wilcox.test(
            sub$logFC[sub$Response_Group == grp],
            sub$logFC[sub$Response_Group == &quot;Non response&quot;]
          )
        }, error = function(e) NULL)
        if (!is.null(test) &amp;&amp; !is.nan(test$p.value)) {
          stat_results[[length(stat_results) + 1]] &lt;- data.frame(
            Response_Group = grp,
            Concentration = conc,
            Timepoint = tp,
            p_value = signif(test$p.value, 3),
            label = case_when(
              test$p.value &lt; 0.001 ~ &quot;***&quot;,
              test$p.value &lt; 0.01 ~ &quot;**&quot;,
              test$p.value &lt; 0.05 ~ &quot;*&quot;,
              TRUE ~ &quot;&quot;
            ),
            y_pos = max(sub$logFC[sub$Response_Group == grp], na.rm = TRUE) + 0.5
          )
        }
      }
    }
  }
}

label_data &lt;- bind_rows(stat_results) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = levels(ac_all$Response_Group)))

# ----------------- Plot -----------------
ggplot(ac_all, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7, na.rm = TRUE) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 4, vjust = 0) +
  facet_grid(rows = vars(factor(Timepoint, levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))),
             cols = vars(Concentration), scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;AC Cardiotoxicity Gene logFC by\nCorMotif Group, Timepoint, and Concentration&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20Ac_tox_2%20logFC1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-Ac_tox_2-logFC1-1">
Past versions of "Mean Ac_tox_2 logFC1-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-Ac_tox_2-logFC1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean Ac_tox_2 logFC1-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-anthracycline-cardiotoxicity-genes-across-corrmotifs-faceted-by-concentration-timepoint-drug-cx-vs-dox-side-by-side"
class="section level2">
<h2><strong>📌 Mean logFC of Anthracycline Cardiotoxicity genes across
corrmotifs (Faceted by concentration × timepoint × drug (CX vs DOX
side-by-side))</strong></h2>
<pre class="r"><code>################### AC Cardiotoxicity Genes — 0.1 µM ###################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- AC Cardiotoxicity Entrez IDs -----------------
ac_ids &lt;- c(
  6272, 8029, 11128, 79899, 54477, 121665, 5095, 22863, 57161, 4692,
  8214, 23151, 56606, 108, 22999, 56895, 9603, 3181, 4023, 10499,
  92949, 4363, 10057, 5243, 5244, 5880, 1535, 2950, 847, 5447,
  3038, 3077, 4846, 3958, 23327, 29899, 23155, 80856, 55020, 78996,
  23262, 150383, 9620, 79730, 344595, 5066, 6251, 3482, 9588, 339416,
  7292, 55157, 87769, 23409, 720, 3107, 54535, 1590, 80059, 7991,
  57110, 8803, 323, 54826, 5916, 23371, 283337, 64078, 80010, 1933,
  10818, 51020
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.1 µM) -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.1 µM DEG Files -----------------
deg_0.1 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.1 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late\nresponse (0.1 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% ac_ids)

# ----------------- Set Factor Levels -----------------
deg_0.1 &lt;- deg_0.1 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.1 µM)&quot;,
      &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.1 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.1 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.1 µM AC Cardiotoxicity) -----------------
ggplot(deg_0.1, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;AC Cardiotoxicity Gene logFC by Drug and Response Group (0.1 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20Ac_tox_3%20logFC3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-Ac_tox_3-logFC3-1">
Past versions of "Mean Ac_tox_3 logFC3-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-Ac_tox_3-logFC3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean Ac_tox_3 logFC3-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>################### AC Genes — 0.5 µM ###################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- AC Cardiotoxicity Entrez IDs -----------------
ac_ids &lt;- c(
  6272, 8029, 11128, 79899, 54477, 121665, 5095, 22863, 57161, 4692,
  8214, 23151, 56606, 108, 22999, 56895, 9603, 3181, 4023, 10499,
  92949, 4363, 10057, 5243, 5244, 5880, 1535, 2950, 847, 5447,
  3038, 3077, 4846, 3958, 23327, 29899, 23155, 80856, 55020, 78996,
  23262, 150383, 9620, 79730, 344595, 5066, 6251, 3482, 9588, 339416,
  7292, 55157, 87769, 23409, 720, 3107, 54535, 1590, 80059, 7991,
  57110, 8803, 323, 54826, 5916, 23371, 283337, 64078, 80010, 1933,
  10818, 51020
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.5 µM) -----------------
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.5 µM DEG Files -----------------
deg_0.5 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.5 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 µM)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% ac_ids)

# ----------------- Set Factor Levels -----------------
deg_0.5 &lt;- deg_0.5 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.5 µM)&quot;,
      &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.5 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.5 µM AC Genes) -----------------
ggplot(deg_0.5, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;AC Gene logFC by Drug and Response Group (0.5 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20Ac_tox_3%20logFC3-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-Ac_tox_3-logFC3-2">
Past versions of "Mean Ac_tox_3 logFC3-2.png"
</button>
</p>
<div id="fig-no-spaces-Mean-Ac_tox_3-logFC3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean Ac_tox_3 logFC3-2.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="proportion-of-dox-cardiotoxicity-genes-across-corrmotifs"
class="section level2">
<h2><strong>📌 Proportion of DOX Cardiotoxicity genes across
corrmotifs</strong></h2>
<pre class="r"><code># ----------------- Load Required Libraries -----------------
library(dplyr)
library(ggplot2)

# ----------------- DOX Cardiotoxicity Entrez IDs -----------------
dox_cardio_ids &lt;- c(
  847, 873, 2064, 2878, 2944, 3038, 4846, 51196, 5880, 6687,
  7799, 4292, 5916, 3077, 51310, 9154, 64078, 5244, 10057, 10060,
  89845, 56853, 4625, 1573, 79890
) %&gt;% as.character()

# ----------------- Load CorrMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% dox_cardio_ids, &quot;DOX-Toxicity Genes&quot;, &quot;Non-DOX Genes&quot;),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% dox_cardio_ids, &quot;DOX-Toxicity Genes&quot;, &quot;Non-DOX Genes&quot;),
    Concentration = &quot;0.5 µM&quot;
  )

# ----------------- Combine Data -----------------
df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Set Category Order -----------------
df_combined$Category &lt;- factor(df_combined$Category, levels = c(&quot;DOX-Toxicity Genes&quot;, &quot;Non-DOX Genes&quot;))

# ----------------- Calculate Proportions -----------------
proportion_data &lt;- df_combined %&gt;%
  group_by(Concentration, Response_Group, Category) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Chi-square or Fisher&#39;s Test -----------------
get_chi_or_fisher &lt;- function(df, ref_group) {
  ref_counts &lt;- df %&gt;%
    ungroup() %&gt;%
    filter(Response_Group == ref_group) %&gt;%
    { ref_tbl &lt;- .[ , c(&quot;Category&quot;, &quot;Count&quot;)]; setNames(ref_tbl$Count, ref_tbl$Category) }

  df %&gt;%
    filter(Response_Group != ref_group) %&gt;%
    group_by(Response_Group) %&gt;%
    summarise(
      p_value = {
        group_counts &lt;- Count[Category %in% c(&quot;DOX-Toxicity Genes&quot;, &quot;Non-DOX Genes&quot;)]
        if (length(group_counts) &lt; 2) group_counts &lt;- c(group_counts, 0)
        contingency_table &lt;- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts[&quot;DOX-Toxicity Genes&quot;], ref_counts[&quot;Non-DOX Genes&quot;]
        ), nrow = 2)
        if (any(contingency_table &lt; 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = &quot;drop&quot;
    ) %&gt;%
    mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))
}

# Run test for each concentration
chi_0.1 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.1 µM&quot;), &quot;Non response&quot;)</code></pre>
<pre><code>Warning: There was 1 warning in `summarise()`.
ℹ In argument: `p_value = { ... }`.
ℹ In group 2: `Response_Group = &quot;DOX only mid-late response&quot;`.
Caused by warning in `chisq.test()`:
! Chi-squared approximation may be incorrect</code></pre>
<pre class="r"><code>chi_0.5 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.5 µM&quot;), &quot;Non response&quot;)

chi_all &lt;- bind_rows(
  chi_0.1 %&gt;% mutate(Concentration = &quot;0.1 µM&quot;),
  chi_0.5 %&gt;% mutate(Concentration = &quot;0.5 µM&quot;)
)

# Merge test results into proportions
proportion_data &lt;- left_join(proportion_data, chi_all, by = c(&quot;Concentration&quot;, &quot;Response_Group&quot;))

# ----------------- Reorder Factor Levels -----------------
proportion_data$Concentration &lt;- factor(proportion_data$Concentration, levels = c(&quot;0.1 µM&quot;, &quot;0.5 µM&quot;))
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
))

# ----------------- Star Label Position -----------------
label_data &lt;- proportion_data %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  summarise(Significance = dplyr::first(Significance), .groups = &quot;drop&quot;) %&gt;%
  filter(!is.na(Significance) &amp; Significance != &quot;&quot;) %&gt;%
  mutate(y_pos = 105)

# ----------------- Final Plot -----------------
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(
    data = label_data,
    aes(x = Response_Group, y = y_pos, label = Significance),
    inherit.aes = FALSE,
    size = 5,
    color = &quot;black&quot;
  ) +
  facet_wrap(~ Concentration, scales = &quot;free_x&quot;) +
  scale_fill_manual(values = c(
    &quot;DOX-Toxicity Genes&quot; = &quot;#E41A1C&quot;,
    &quot;Non-DOX Genes&quot; = &quot;#377EB8&quot;
  )) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +
  labs(
    title = &quot;DOX Cardiotoxicity Gene Proportions Across\nCormotif Clusters (0.1 and 0.5)&quot;,
    x = &quot;Response Groups&quot;,
    y = &quot;Percentage of Genes&quot;,
    fill = &quot;Gene Category&quot;
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = &quot;bold&quot;),
    axis.title.x = element_text(size = 14, face = &quot;bold&quot;),
    axis.title.y = element_text(size = 14, face = &quot;bold&quot;),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    panel.spacing = unit(1.2, &quot;lines&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/DOX_tox%20prop-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-DOX_tox-prop-1">
Past versions of "DOX_tox prop-1.png"
</button>
</p>
<div id="fig-no-spaces-DOX_tox-prop-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/DOX_tox prop-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-dox-cardiotoxicity-genes-across-corrmotifs-per-concentration"
class="section level2">
<h2><strong>📌 Mean logFC of DOX Cardiotoxicity genes across corrmotifs
(per concentration)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- DOX Cardiotoxicity Entrez IDs -----------------
entrez_ids &lt;- c(
  847, 873, 2064, 2878, 2944, 3038, 4846, 51196, 5880, 6687,
  7799, 4292, 5916, 3077, 51310, 9154, 64078, 5244, 10057, 10060,
  89845, 56853, 4625, 1573, 79890
) %&gt;% as.character()

# ----------------- Load CorMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate Response Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Aggregate DEG Data -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;,
      TRUE ~ NA_character_
    )
  )

deg_mean &lt;- deg_all %&gt;%
  group_by(Entrez_ID, Concentration) %&gt;%
  summarise(logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;)

# ----------------- Merge with CorMotif Genes -----------------
dox_cardio_logfc &lt;- inner_join(df_combined, deg_mean, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Group = paste(Response_Group, Concentration)
  )

# ----------------- Define Response Group Order -----------------
group_levels &lt;- c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
)

dox_cardio_logfc$Response_Group &lt;- factor(dox_cardio_logfc$Response_Group, levels = group_levels)

# ----------------- Wilcoxon Test -----------------
comparison_map &lt;- list(
  &quot;CX-DOX mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX only mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX-specific response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX only mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;CX total + DOX early response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX early + CX-DOX mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;
)

wilcoxon_results &lt;- lapply(names(comparison_map), function(resp_group) {
  control_group &lt;- comparison_map[[resp_group]]
  resp_vals &lt;- dox_cardio_logfc$logFC[dox_cardio_logfc$Group == resp_group]
  control_vals &lt;- dox_cardio_logfc$logFC[dox_cardio_logfc$Group == control_group]

  if (length(resp_vals) &gt;= 2 &amp;&amp; length(control_vals) &gt;= 2) {
    test_result &lt;- wilcox.test(resp_vals, control_vals)
    pval &lt;- test_result$p.value
    stat &lt;- test_result$statistic
    label &lt;- case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01  ~ &quot;**&quot;,
      pval &lt; 0.05  ~ &quot;*&quot;,
      TRUE ~ &quot;&quot;
    )
    y_pos &lt;- max(resp_vals, na.rm = TRUE) + 0.5
    return(data.frame(
      Group = resp_group,
      y_pos = y_pos,
      label = label,
      P_Value = signif(pval, 4)
    ))
  } else {
    return(NULL)
  }
}) %&gt;% bind_rows()

# ----------------- Prepare Annotation -----------------
label_data &lt;- wilcoxon_results %&gt;%
  separate(Group, into = c(&quot;Response_Group&quot;, &quot;Concentration&quot;), sep = &quot; (?=0\\.)&quot;) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = group_levels))

# ----------------- Plot -----------------
ggplot(dox_cardio_logfc, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~Concentration, scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Mean logFC of DOX Cardiotoxicity Genes\nAcross CorMotif Response Groups&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20DOX_tox%20logFC-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-DOX_tox-logFC-1">
Past versions of "Mean DOX_tox logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-DOX_tox-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean DOX_tox logFC-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-dox-cardiotoxicity-genes-across-corrmotifs-faceted-by-concentration-timepoint"
class="section level2">
<h2><strong>📌 Mean logFC of DOX Cardiotoxicity genes across corrmotifs
(Faceted by concentration × timepoint)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- DOX Cardiotoxicity Entrez IDs -----------------
entrez_ids &lt;- c(
  847, 873, 2064, 2878, 2944, 3038, 4846, 51196, 5880, 6687,
  7799, 4292, 5916, 3077, 51310, 9154, 64078, 5244, 10057, 10060,
  89845, 56853, 4625, 1573, 79890
) %&gt;% as.character()

# ----------------- Load CorMotif Response Group Data -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorMotif Group Membership -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Annotate DEG Files -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    ),
    Timepoint = factor(case_when(
      str_detect(Condition, &quot;_3$&quot;) ~ &quot;3 hr&quot;,
      str_detect(Condition, &quot;_24$&quot;) ~ &quot;24 hr&quot;,
      str_detect(Condition, &quot;_48$&quot;) ~ &quot;48 hr&quot;
    ), levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))
  )

# ----------------- Merge with DOX Cardiotoxicity Targets -----------------
dox_all &lt;- inner_join(df_combined, deg_all, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID, column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response&quot;, &quot;CX-DOX mid-late response&quot;, &quot;DOX-specific response&quot;,
      &quot;DOX only mid-late response&quot;, &quot;CX total + DOX early response&quot;, &quot;DOX early + CX-DOX mid-late response&quot;
    ))
  )

# ----------------- Wilcoxon Test -----------------
comparison_groups &lt;- levels(dox_all$Response_Group)[-1]
stat_results &lt;- list()

for (conc in unique(dox_all$Concentration)) {
  for (tp in levels(dox_all$Timepoint)) {
    sub &lt;- dox_all %&gt;% filter(Concentration == conc, Timepoint == tp)
    for (grp in comparison_groups) {
      if (grp %in% unique(sub$Response_Group)) {
        test &lt;- tryCatch({
          wilcox.test(
            sub$logFC[sub$Response_Group == grp],
            sub$logFC[sub$Response_Group == &quot;Non response&quot;]
          )
        }, error = function(e) NULL)
        if (!is.null(test) &amp;&amp; !is.nan(test$p.value)) {
          stat_results[[length(stat_results) + 1]] &lt;- data.frame(
            Response_Group = grp,
            Concentration = conc,
            Timepoint = tp,
            p_value = signif(test$p.value, 3),
            label = case_when(
              test$p.value &lt; 0.001 ~ &quot;***&quot;,
              test$p.value &lt; 0.01 ~ &quot;**&quot;,
              test$p.value &lt; 0.05 ~ &quot;*&quot;,
              TRUE ~ &quot;&quot;
            ),
            y_pos = max(sub$logFC[sub$Response_Group == grp], na.rm = TRUE) + 0.5
          )
        }
      }
    }
  }
}

label_data &lt;- bind_rows(stat_results) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = levels(dox_all$Response_Group)))

# ----------------- Plot -----------------
ggplot(dox_all, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7, na.rm = TRUE) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 4, vjust = 0) +
  facet_grid(rows = vars(factor(Timepoint, levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))),
             cols = vars(Concentration), scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;DOX Cardiotoxicity Gene logFC by\nCorMotif Group, Timepoint, and Concentration&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20DOX_tox_2%20logFC1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-DOX_tox_2-logFC1-1">
Past versions of "Mean DOX_tox_2 logFC1-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-DOX_tox_2-logFC1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean DOX_tox_2 logFC1-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-dox-cardiotoxicity-genes-across-corrmotifs-faceted-by-concentration-timepoint-drug-cx-vs-dox-side-by-side"
class="section level2">
<h2><strong>📌 Mean logFC of DOX Cardiotoxicity genes across corrmotifs
(Faceted by concentration × timepoint × drug (CX vs DOX
side-by-side))</strong></h2>
<pre class="r"><code>################### DOX Cardiotoxicity Genes — 0.1 µM ###################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- DOX Cardiotoxicity Entrez IDs -----------------
entrez_ids &lt;- c(
  847, 873, 2064, 2878, 2944, 3038, 4846, 51196, 5880, 6687,
  7799, 4292, 5916, 3077, 51310, 9154, 64078, 5244, 10057, 10060,
  89845, 56853, 4625, 1573, 79890
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.1 µM) -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.1 µM DEG Files -----------------
deg_0.1 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.1 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late\nresponse (0.1 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.1 &lt;- deg_0.1 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.1 µM)&quot;,
      &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.1 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.1 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.1 µM DOX Cardiotoxicity Genes) -----------------
ggplot(deg_0.1, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;DOX Cardiotoxicity Gene logFC by Drug and Response Group (0.1 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20DOX_tox_3%20logFC3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-DOX_tox_3-logFC3-1">
Past versions of "Mean DOX_tox_3 logFC3-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-DOX_tox_3-logFC3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean DOX_tox_3 logFC3-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>################### DOX Cardiotoxicity Genes — 0.5 µM ###################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- DOX Cardiotoxicity Entrez IDs -----------------
entrez_ids &lt;- c(
  847, 873, 2064, 2878, 2944, 3038, 4846, 51196, 5880, 6687,
  7799, 4292, 5916, 3077, 51310, 9154, 64078, 5244, 10057, 10060,
  89845, 56853, 4625, 1573, 79890
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.5 µM) -----------------
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.5 µM DEG Files -----------------
deg_0.5 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.5 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 µM)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.5 &lt;- deg_0.5 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.5 µM)&quot;,
      &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.5 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.5 µM DOX Cardiotoxicity Genes) -----------------
ggplot(deg_0.5, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;DOX Cardiotoxicity Gene logFC by Drug and Response Group (0.5 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20DOX_tox_3%20logFC3-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-DOX_tox_3-logFC3-2">
Past versions of "Mean DOX_tox_3 logFC3-2.png"
</button>
</p>
<div id="fig-no-spaces-Mean-DOX_tox_3-logFC3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean DOX_tox_3 logFC3-2.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="proportion-of-heart-specific-genes-combined"
class="section level2">
<h2><strong>📌 Proportion of Heart Specific Genes
(Combined)</strong></h2>
<pre class="r"><code># Load Required Libraries
library(dplyr)
library(ggplot2)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load Heart-Specific Genes -----------------
heart_genes &lt;- read.csv(&quot;data/Human_Heart_Genes.csv&quot;, stringsAsFactors = FALSE)
heart_genes$Entrez_ID &lt;- mapIds(
  org.Hs.eg.db,
  keys = heart_genes$Gene,
  column = &quot;ENTREZID&quot;,
  keytype = &quot;SYMBOL&quot;,
  multiVals = &quot;first&quot;
)
heart_entrez_ids &lt;- na.omit(heart_genes$Entrez_ID)

# ----------------- Load CorrMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% heart_entrez_ids, &quot;Heart-specific Genes&quot;, &quot;Non-Heart-specific Genes&quot;),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% heart_entrez_ids, &quot;Heart-specific Genes&quot;, &quot;Non-Heart-specific Genes&quot;),
    Concentration = &quot;0.5 µM&quot;
  )

# ----------------- Combine Data -----------------
df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Calculate Proportions -----------------
proportion_data &lt;- df_combined %&gt;%
  dplyr::group_by(Concentration, Response_Group, Category) %&gt;%
  dplyr::summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  dplyr::group_by(Concentration, Response_Group) %&gt;%
  dplyr::mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Chi-square or Fisher&#39;s Test -----------------
get_chi_or_fisher &lt;- function(df, ref_group) {
  ref_counts &lt;- df %&gt;%
    dplyr::ungroup() %&gt;%
    dplyr::filter(Response_Group == ref_group) %&gt;%
    dplyr::select(Category, Count) %&gt;%
    {setNames(.$Count, .$Category)}
  
  df %&gt;%
    dplyr::filter(Response_Group != ref_group) %&gt;%
    dplyr::group_by(Response_Group) %&gt;%
    dplyr::summarise(
      p_value = {
        group_counts &lt;- Count[Category %in% c(&quot;Heart-specific Genes&quot;, &quot;Non-Heart-specific Genes&quot;)]
        if (length(group_counts) &lt; 2) group_counts &lt;- c(group_counts, 0)
        contingency_table &lt;- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts[&quot;Heart-specific Genes&quot;], ref_counts[&quot;Non-Heart-specific Genes&quot;]
        ), nrow = 2)
        if (any(contingency_table &lt; 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = &quot;drop&quot;
    ) %&gt;%
    dplyr::mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))
}

# Run test for each concentration
chi_0.1 &lt;- get_chi_or_fisher(proportion_data %&gt;% dplyr::filter(Concentration == &quot;0.1 µM&quot;), &quot;Non response&quot;)
chi_0.5 &lt;- get_chi_or_fisher(proportion_data %&gt;% dplyr::filter(Concentration == &quot;0.5 µM&quot;), &quot;Non response&quot;)

chi_all &lt;- dplyr::bind_rows(
  chi_0.1 %&gt;% dplyr::mutate(Concentration = &quot;0.1 µM&quot;),
  chi_0.5 %&gt;% dplyr::mutate(Concentration = &quot;0.5 µM&quot;)
)

# Merge test results into proportions
proportion_data &lt;- proportion_data %&gt;%
  dplyr::left_join(chi_all, by = c(&quot;Concentration&quot;, &quot;Response_Group&quot;))

# ----------------- Reorder Factor Levels -----------------
proportion_data$Concentration &lt;- factor(proportion_data$Concentration, levels = c(&quot;0.1 µM&quot;, &quot;0.5 µM&quot;))
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
))

# ----------------- Compute Y-position for Significance Labels -----------------
label_data &lt;- proportion_data %&gt;%
  dplyr::group_by(Concentration, Response_Group) %&gt;%
  dplyr::summarise(Significance = dplyr::first(Significance),
                   ymax = sum(Percentage), .groups = &quot;drop&quot;) %&gt;%
  dplyr::filter(!is.na(Significance)) %&gt;%
  dplyr::mutate(y_pos = ymax + 3)

# ----------------- Final Plot -----------------
# --- Position stars above 100
label_data &lt;- proportion_data %&gt;%
  dplyr::group_by(Concentration, Response_Group) %&gt;%
  dplyr::summarise(Significance = dplyr::first(Significance), .groups = &quot;drop&quot;) %&gt;%
  dplyr::filter(!is.na(Significance)) %&gt;%
  dplyr::mutate(y_pos = 105)  # stars go at 105

# --- Plot
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(
    data = label_data,
    aes(x = Response_Group, y = y_pos, label = Significance),
    inherit.aes = FALSE,
    size = 5,
    color = &quot;black&quot;
  ) +
  facet_wrap(~ Concentration, scales = &quot;free_x&quot;) +
  scale_fill_manual(values = c(
    &quot;Heart-specific Genes&quot; = &quot;#4daf4a&quot;,
    &quot;Non-Heart-specific Genes&quot; = &quot;#377eb8&quot;
  )) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +  # allow space for stars and border
  labs(
    title = &quot;Heart-Specific Gene Proportions Across\nCormotif Clusters (0.1 and 0.5)&quot;,
    x = &quot;Response Groups&quot;,
    y = &quot;Percentage of Genes&quot;,
    fill = &quot;Gene Category&quot;
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = &quot;bold&quot;),
    axis.title.x = element_text(size = 14, face = &quot;bold&quot;),
    axis.title.y = element_text(size = 14, face = &quot;bold&quot;),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),  # draws full box
    panel.spacing = unit(1.2, &quot;lines&quot;)  # adds vertical spacing to avoid cramped look
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/prop%20heart%20comb-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-prop-heart-comb-1">
Past versions of "prop heart comb-1.png"
</button>
</p>
<div id="fig-no-spaces-prop-heart-comb-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/4bad905ad26e9ebac9188a323b7f5fe10aa56ce6/docs/figure/Corrmotif_Conc.Rmd/prop heart comb-1.png" target="_blank">4bad905</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/6ef5f61785e28f0c6385c185a7d54db2620ea914/docs/figure/Corrmotif_Conc.Rmd/prop heart comb-1.png" target="_blank">6ef5f61</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-heart-specific-genes-across-corrmotifs-per-concentration"
class="section level2">
<h2><strong>📌 Mean logFC of Heart specific genes across corrmotifs (per
concentration)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load Heart-Specific Genes -----------------
heart_genes &lt;- read.csv(&quot;data/Human_Heart_Genes.csv&quot;, stringsAsFactors = FALSE)
heart_genes$Entrez_ID &lt;- mapIds(
  org.Hs.eg.db,
  keys = heart_genes$Gene,
  column = &quot;ENTREZID&quot;,
  keytype = &quot;SYMBOL&quot;,
  multiVals = &quot;first&quot;
)
entrez_ids &lt;- na.omit(heart_genes$Entrez_ID) %&gt;% as.character()

# ----------------- Load CorMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate Response Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Aggregate DEG Data -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;% mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    )
  )

deg_mean &lt;- deg_all %&gt;%
  group_by(Entrez_ID, Concentration) %&gt;%
  summarise(logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;)

# ----------------- Merge with Group Annotations -----------------
heart_logfc &lt;- inner_join(df_combined, deg_mean, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Group = paste(Response_Group, Concentration)
  )

# ----------------- Define Response Group Order -----------------
group_levels &lt;- c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
)

heart_logfc$Response_Group &lt;- factor(heart_logfc$Response_Group, levels = group_levels)

# ----------------- Wilcoxon Test -----------------
comparison_map &lt;- list(
  &quot;CX-DOX mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX only mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX-specific response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX only mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;CX total + DOX early response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX early + CX-DOX mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;
)

wilcoxon_results &lt;- lapply(names(comparison_map), function(resp_group) {
  control_group &lt;- comparison_map[[resp_group]]
  resp_vals &lt;- heart_logfc$logFC[heart_logfc$Group == resp_group]
  control_vals &lt;- heart_logfc$logFC[heart_logfc$Group == control_group]

  if (length(resp_vals) &gt;= 2 &amp;&amp; length(control_vals) &gt;= 2) {
    test_result &lt;- wilcox.test(resp_vals, control_vals)
    pval &lt;- test_result$p.value
    label &lt;- case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01  ~ &quot;**&quot;,
      pval &lt; 0.05  ~ &quot;*&quot;,
      TRUE ~ &quot;&quot;
    )
    y_pos &lt;- max(resp_vals, na.rm = TRUE) + 0.5
    return(data.frame(
      Group = resp_group,
      y_pos = y_pos,
      label = label,
      P_Value = signif(pval, 4)
    ))
  }
  return(NULL)
}) %&gt;% bind_rows()

# ----------------- Annotation Data -----------------
label_data &lt;- wilcoxon_results %&gt;%
  separate(Group, into = c(&quot;Response_Group&quot;, &quot;Concentration&quot;), sep = &quot; (?=0\\.)&quot;) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = group_levels))

# ----------------- Plot -----------------
ggplot(heart_logfc, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~Concentration, scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Mean logFC of Heart-Specific Genes\nAcross CorMotif Response Groups&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20HS%20logFC-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-HS-logFC-1">
Past versions of "Mean HS logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-HS-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean HS logFC-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-heart-specific-genes-across-corrmotifs-faceted-by-concentration-timepoint"
class="section level2">
<h2><strong>📌 Mean logFC of Heart specific genes across corrmotifs
(Faceted by concentration × timepoint)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load Heart-Specific Genes -----------------
heart_genes &lt;- read.csv(&quot;data/Human_Heart_Genes.csv&quot;, stringsAsFactors = FALSE)
heart_genes$Entrez_ID &lt;- mapIds(
  org.Hs.eg.db,
  keys = heart_genes$Gene,
  column = &quot;ENTREZID&quot;,
  keytype = &quot;SYMBOL&quot;,
  multiVals = &quot;first&quot;
)
entrez_ids &lt;- na.omit(heart_genes$Entrez_ID) %&gt;% as.character()

# ----------------- Load CorMotif Response Group Data -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorMotif Group Membership -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Annotate DEG Files -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    ),
    Timepoint = factor(case_when(
      str_detect(Condition, &quot;_3$&quot;) ~ &quot;3 hr&quot;,
      str_detect(Condition, &quot;_24$&quot;) ~ &quot;24 hr&quot;,
      str_detect(Condition, &quot;_48$&quot;) ~ &quot;48 hr&quot;
    ), levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))
  )

# ----------------- Merge with Heart-Specific Targets -----------------
heart_all &lt;- inner_join(df_combined, deg_all, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID, column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response&quot;, &quot;CX-DOX mid-late response&quot;, &quot;DOX-specific response&quot;,
      &quot;DOX only mid-late response&quot;, &quot;CX total + DOX early response&quot;, &quot;DOX early + CX-DOX mid-late response&quot;
    ))
  )

# ----------------- Wilcoxon Test -----------------
comparison_groups &lt;- levels(heart_all$Response_Group)[-1]
stat_results &lt;- list()

for (conc in unique(heart_all$Concentration)) {
  for (tp in levels(heart_all$Timepoint)) {
    sub &lt;- heart_all %&gt;% filter(Concentration == conc, Timepoint == tp)
    for (grp in comparison_groups) {
      if (grp %in% unique(sub$Response_Group)) {
        test &lt;- tryCatch({
          wilcox.test(
            sub$logFC[sub$Response_Group == grp],
            sub$logFC[sub$Response_Group == &quot;Non response&quot;]
          )
        }, error = function(e) NULL)
        if (!is.null(test) &amp;&amp; !is.nan(test$p.value)) {
          stat_results[[length(stat_results) + 1]] &lt;- data.frame(
            Response_Group = grp,
            Concentration = conc,
            Timepoint = tp,
            p_value = signif(test$p.value, 3),
            label = case_when(
              test$p.value &lt; 0.001 ~ &quot;***&quot;,
              test$p.value &lt; 0.01 ~ &quot;**&quot;,
              test$p.value &lt; 0.05 ~ &quot;*&quot;,
              TRUE ~ &quot;&quot;
            ),
            y_pos = max(sub$logFC[sub$Response_Group == grp], na.rm = TRUE) + 0.5
          )
        }
      }
    }
  }
}

label_data &lt;- bind_rows(stat_results) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = levels(heart_all$Response_Group)))

# ----------------- Plot -----------------
ggplot(heart_all, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7, na.rm = TRUE) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 4, vjust = 0) +
  facet_grid(rows = vars(factor(Timepoint, levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))),
             cols = vars(Concentration), scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Heart-Specific Gene logFC by\nCorMotif Group, Timepoint, and Concentration&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20HS_2%20logFC1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-HS_2-logFC1-1">
Past versions of "Mean HS_2 logFC1-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-HS_2-logFC1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean HS_2 logFC1-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-heart-specific-genes-across-corrmotifs-faceted-by-concentration-timepoint-drug-cx-vs-dox-side-by-side"
class="section level2">
<h2><strong>📌 Mean logFC of Heart specific genes across corrmotifs
(Faceted by concentration × timepoint × drug (CX vs DOX
side-by-side))</strong></h2>
<pre class="r"><code>############ HEART-SPECIFIC GENES — 0.1 µM #######################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load Heart-Specific Genes -----------------
heart_genes &lt;- read.csv(&quot;data/Human_Heart_Genes.csv&quot;, stringsAsFactors = FALSE)
heart_genes$Entrez_ID &lt;- mapIds(
  org.Hs.eg.db,
  keys = heart_genes$Gene,
  column = &quot;ENTREZID&quot;,
  keytype = &quot;SYMBOL&quot;,
  multiVals = &quot;first&quot;
)
entrez_ids &lt;- na.omit(heart_genes$Entrez_ID) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.1 µM) -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.1 µM DEG Files -----------------
deg_0.1 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.1 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late\nresponse (0.1 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.1 &lt;- deg_0.1 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.1 µM)&quot;,
      &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.1 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.1 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.1 µM Heart-Specific Genes) -----------------
ggplot(deg_0.1, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;Heart-Specific Gene logFC by Drug and Response Group (0.1 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20HS_3%20logFC3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-HS_3-logFC3-1">
Past versions of "Mean HS_3 logFC3-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-HS_3-logFC3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean HS_3 logFC3-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>############ HEART-SPECIFIC GENES — 0.5 µM #######################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load Heart-Specific Genes -----------------
heart_genes &lt;- read.csv(&quot;data/Human_Heart_Genes.csv&quot;, stringsAsFactors = FALSE)
heart_genes$Entrez_ID &lt;- mapIds(
  org.Hs.eg.db,
  keys = heart_genes$Gene,
  column = &quot;ENTREZID&quot;,
  keytype = &quot;SYMBOL&quot;,
  multiVals = &quot;first&quot;
)
entrez_ids &lt;- na.omit(heart_genes$Entrez_ID) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.5 µM) -----------------
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.5 µM DEG Files -----------------
deg_0.5 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.5 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 µM)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.5 &lt;- deg_0.5 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.5 µM)&quot;,
      &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.5 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.5 µM Heart-Specific Genes) -----------------
ggplot(deg_0.5, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;Heart-Specific Gene logFC by Drug and Response Group (0.5 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20HS_3%20logFC3-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-HS_3-logFC3-2">
Past versions of "Mean HS_3 logFC3-2.png"
</button>
</p>
<div id="fig-no-spaces-Mean-HS_3-logFC3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean HS_3 logFC3-2.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="proportion-of-atrial-fibrillation-gwas-genes-across-corrmotifs"
class="section level2">
<h2><strong>📌 Proportion of Atrial Fibrillation GWAS genes across
corrmotifs</strong></h2>
<pre class="r"><code># ----------------- Load Required Libraries -----------------
library(dplyr)
library(ggplot2)

# ----------------- AF Entrez IDs -----------------
af_entrez_ids &lt;- c(
  3425, 53834, 9464, 79658, 55876, 3781, 9472, 11017, 64395, 23353,
  57794, 79591, 2702, 51306, 857, 10418, 9644, 115509, 89797, 10728,
  282996, 8497, 5695, 9026, 57666, 143384, 5883, 90102, 84909, 221037,
  29119, 79933, 391, 81575, 8738, 5396, 26010, 744, 56981, 80724,
  5781, 6910, 4598, 79068, 2250, 56978, 3752, 5208, 9736, 26778,
  80315, 64374, 79159, 11078, 5308, 3782, 10654, 79776, 6310, 8082,
  2995, 23224, 79741, 5708, 26136, 9967, 84033, 9148, 29982, 81488,
  1788, 9208, 2908, 1021, 5339, 5727, 23118, 10090, 57585, 221154,
  1073, 29998, 817, 858, 3486, 5747, 2104, 55105, 4194, 3570,
  1607, 6586, 5570, 2070, 27145, 4646, 3159, 9748, 79991, 221035,
  5033, 80212, 196385, 79600, 11046, 54014, 151636, 64778, 92597, 3983,
  7090, 9267, 7273, 26018, 9948, 2131, 55870, 5523, 5318, 6239,
  3480, 55777, 2066, 11165, 2028, 57619, 2690, 23414, 55818, 84700,
  28965, 80204, 463, 5108, 222553, 387119, 28981, 11113, 10301, 995,
  23030, 2697, 996, 339500, 54805, 9960, 91404, 145781, 100820829, 84542,
  2176, 51684, 9513, 7473, 4666, 23150, 5915, 5062, 4016, 23039,
  159686, 1839, 5201, 93166, 64753, 29959, 5496, 23245, 5069, 56916,
  92344, 23092, 3992, 9415, 10554, 7456, 9570, 57178, 23143, 161176,
  5424, 2034, 10277, 11278, 79803, 6653, 4756, 132660, 5430, 9031,
  57158, 285761, 8110, 387700, 1829, 4126, 7323, 51308, 7332, 6598,
  3757, 6187, 6660, 10529, 6920, 115286, 8451, 8943, 4137, 7514,
  4801, 9709, 23177, 8671, 29915, 26207, 3680, 490, 493856, 58489,
  54897, 4625, 22955, 84952, 10221, 2263, 84641, 4892, 1026, 84650,
  8382, 221656, 2969, 144453, 117177, 2626, 8476, 161882, 51807, 4624,
  9612, 55795, 51043, 144348, 51232, 8729, 3899, 11155, 23316, 79006,
  146330, 6403, 6331, 4300, 427, 845, 3313, 113622, 5789, 376132,
  29841, 8462, 203859, 401397, 5506, 55521, 5819, 4059, 6934, 57727,
  23066, 79568, 83478, 10087, 9586, 222194, 10466, 10499, 58499, 79720,
  4772, 10794, 125919, 602, 27332, 59345, 340359, 3705, 64710, 57801,
  10818, 143684, 149281, 55013, 23095, 93649, 84034, 23347, 440926, 11124,
  23293, 51426, 832, 7068, 57646, 152002, 7531, 1398, 100101267, 221937,
  26873, 3709, 10576, 27040, 201176, 284403, 307, 6525, 5387, 1808,
  114907, 1952, 4091, 5096, 23451, 51207, 142891, 4084, 3797, 1185,
  5334, 217, 2042, 7781, 253461, 152404, 2992, 153478, 6695, 26249,
  662, 23036, 22852, 23411, 23387, 6786, 7690, 10021, 8925, 5595,
  63892, 23466, 11149, 139411, 755, 55347, 22820, 55111, 84444, 6258,
  57623, 1387, 3762, 5861, 3339, 114822, 129787, 196528, 54437, 10395,
  4023, 5095, 894, 3156, 1877, 283871, 2673, 23157, 10512, 2258,
  79750, 84665, 26091, 5978, 8751, 79695, 2768, 155382, 84163, 7709,
  4092, 1837, 2064, 4629, 55122, 8882, 150962, 23013, 8742, 489,
  201134, 55114, 84264, 64428, 4089, 861, 7422
) %&gt;% as.character()

# ----------------- Load CorrMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% af_entrez_ids, &quot;AF Genes&quot;, &quot;Non-AF Genes&quot;),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% af_entrez_ids, &quot;AF Genes&quot;, &quot;Non-AF Genes&quot;),
    Concentration = &quot;0.5 µM&quot;
  )

# ----------------- Combine Data -----------------
df_combined &lt;- bind_rows(df_0.1, df_0.5)
df_combined$Category &lt;- factor(df_combined$Category, levels = c(&quot;AF Genes&quot;, &quot;Non-AF Genes&quot;))

# ----------------- Calculate Proportions -----------------
proportion_data &lt;- df_combined %&gt;%
  group_by(Concentration, Response_Group, Category) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Chi-square or Fisher&#39;s Test -----------------
get_chi_or_fisher &lt;- function(df, ref_group) {
  ref_counts &lt;- df %&gt;%
    ungroup() %&gt;%
    filter(Response_Group == ref_group) %&gt;%
    { ref_tbl &lt;- .[ , c(&quot;Category&quot;, &quot;Count&quot;)]; setNames(ref_tbl$Count, ref_tbl$Category) }
  
  df %&gt;%
    filter(Response_Group != ref_group) %&gt;%
    group_by(Response_Group) %&gt;%
    summarise(
      p_value = {
        group_counts &lt;- Count[Category %in% c(&quot;AF Genes&quot;, &quot;Non-AF Genes&quot;)]
        if (length(group_counts) &lt; 2) group_counts &lt;- c(group_counts, 0)
        contingency_table &lt;- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts[&quot;AF Genes&quot;], ref_counts[&quot;Non-AF Genes&quot;]
        ), nrow = 2)
        if (any(contingency_table &lt; 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = &quot;drop&quot;
    ) %&gt;%
    mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))
}

# ----------------- Run test -----------------
chi_0.1 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.1 µM&quot;), &quot;Non response&quot;)
chi_0.5 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.5 µM&quot;), &quot;Non response&quot;)</code></pre>
<pre><code>Warning: There was 1 warning in `summarise()`.
ℹ In argument: `p_value = { ... }`.
ℹ In group 4: `Response_Group = &quot;DOX-specific response&quot;`.
Caused by warning in `chisq.test()`:
! Chi-squared approximation may be incorrect</code></pre>
<pre class="r"><code>chi_all &lt;- bind_rows(
  chi_0.1 %&gt;% mutate(Concentration = &quot;0.1 µM&quot;),
  chi_0.5 %&gt;% mutate(Concentration = &quot;0.5 µM&quot;)
)
proportion_data &lt;- left_join(proportion_data, chi_all, by = c(&quot;Concentration&quot;, &quot;Response_Group&quot;))

# ----------------- Factor Order and Star Label -----------------
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
))
proportion_data$Concentration &lt;- factor(proportion_data$Concentration, levels = c(&quot;0.1 µM&quot;, &quot;0.5 µM&quot;))
label_data &lt;- proportion_data %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  summarise(Significance = dplyr::first(Significance), .groups = &quot;drop&quot;) %&gt;%
  filter(!is.na(Significance) &amp; Significance != &quot;&quot;) %&gt;%
  mutate(y_pos = 105)

# ----------------- Final Plot -----------------
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = Significance),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~ Concentration, scales = &quot;free_x&quot;) +
  scale_fill_manual(values = c(&quot;AF Genes&quot; = &quot;#E41A1C&quot;, &quot;Non-AF Genes&quot; = &quot;#377EB8&quot;)) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +
  labs(
    title = &quot;AF-Associated Gene Proportions Across\nCormotif Clusters (0.1 and 0.5 µM)&quot;,
    x = &quot;Response Groups&quot;, y = &quot;Percentage of Genes&quot;, fill = &quot;Gene Category&quot;
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = &quot;bold&quot;),
    axis.title.x = element_text(size = 14, face = &quot;bold&quot;),
    axis.title.y = element_text(size = 14, face = &quot;bold&quot;),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    panel.spacing = unit(1.2, &quot;lines&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/AF_Prop-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-AF_Prop-1">
Past versions of AF_Prop-1.png
</button>
</p>
<div id="fig-AF_Prop-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/AF_Prop-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-atrial-fibrillation-gwas-genes-across-corrmotifs-per-concentration"
class="section level2">
<h2><strong>📌 Mean logFC of Atrial Fibrillation GWAS genes across
corrmotifs (per concentration)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- AF Entrez IDs -----------------
entrez_ids &lt;- c(
  3425, 53834, 9464, 79658, 55876, 3781, 9472, 11017, 64395, 23353,
  57794, 79591, 2702, 51306, 857, 10418, 9644, 115509, 89797, 10728,
  282996, 8497, 5695, 9026, 57666, 143384, 5883, 90102, 84909, 221037,
  29119, 79933, 391, 81575, 8738, 5396, 26010, 744, 56981, 80724,
  5781, 6910, 4598, 79068, 2250, 56978, 3752, 5208, 9736, 26778,
  80315, 64374, 79159, 11078, 5308, 3782, 10654, 79776, 6310, 8082,
  2995, 23224, 79741, 5708, 26136, 9967, 84033, 9148, 29982, 81488,
  1788, 9208, 2908, 1021, 5339, 5727, 23118, 10090, 57585, 221154,
  1073, 29998, 817, 858, 3486, 5747, 2104, 55105, 4194, 3570,
  1607, 6586, 5570, 2070, 27145, 4646, 3159, 9748, 79991, 221035,
  5033, 80212, 196385, 79600, 11046, 54014, 151636, 64778, 92597, 3983,
  7090, 9267, 7273, 26018, 9948, 2131, 55870, 5523, 5318, 6239,
  3480, 55777, 2066, 11165, 2028, 57619, 2690, 23414, 55818, 84700,
  28965, 80204, 463, 5108, 222553, 387119, 28981, 11113, 10301, 995,
  23030, 2697, 996, 339500, 54805, 9960, 91404, 145781, 100820829, 84542,
  2176, 51684, 9513, 7473, 4666, 23150, 5915, 5062, 4016, 23039,
  159686, 1839, 5201, 93166, 64753, 29959, 5496, 23245, 5069, 56916,
  92344, 23092, 3992, 9415, 10554, 7456, 9570, 57178, 23143, 161176,
  5424, 2034, 10277, 11278, 79803, 6653, 4756, 132660, 5430, 9031,
  57158, 285761, 8110, 387700, 1829, 4126, 7323, 51308, 7332, 6598,
  3757, 6187, 6660, 10529, 6920, 115286, 8451, 8943, 4137, 7514,
  4801, 9709, 23177, 8671, 29915, 26207, 3680, 490, 493856, 58489,
  54897, 4625, 22955, 84952, 10221, 2263, 84641, 4892, 1026, 84650,
  8382, 221656, 2969, 144453, 117177, 2626, 8476, 161882, 51807, 4624,
  9612, 55795, 51043, 144348, 51232, 8729, 3899, 11155, 23316, 79006,
  146330, 6403, 6331, 4300, 427, 845, 3313, 113622, 5789, 376132,
  29841, 8462, 203859, 401397, 5506, 55521, 5819, 4059, 6934, 57727,
  23066, 79568, 83478, 10087, 9586, 222194, 10466, 10499, 58499, 79720,
  4772, 10794, 125919, 602, 27332, 59345, 340359, 3705, 64710, 57801,
  10818, 143684, 149281, 55013, 23095, 93649, 84034, 23347, 440926, 11124,
  23293, 51426, 832, 7068, 57646, 152002, 7531, 1398, 100101267, 221937,
  26873, 3709, 10576, 27040, 201176, 284403, 307, 6525, 5387, 1808,
  114907, 1952, 4091, 5096, 23451, 51207, 142891, 4084, 3797, 1185,
  5334, 217, 2042, 7781, 253461, 152404, 2992, 153478, 6695, 26249,
  662, 23036, 22852, 23411, 23387, 6786, 7690, 10021, 8925, 5595,
  63892, 23466, 11149, 139411, 755, 55347, 22820, 55111, 84444, 6258,
  57623, 1387, 3762, 5861, 3339, 114822, 129787, 196528, 54437, 10395,
  4023, 5095, 894, 3156, 1877, 283871, 2673, 23157, 10512, 2258,
  79750, 84665, 26091, 5978, 8751, 79695, 2768, 155382, 84163, 7709,
  4092, 1837, 2064, 4629, 55122, 8882, 150962, 23013, 8742, 489,
  201134, 55114, 84264, 64428, 4089, 861, 7422
) %&gt;% as.character()

# ----------------- Load CorMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate Response Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Aggregate DEG Data -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;% mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    )
  )

deg_mean &lt;- deg_all %&gt;%
  group_by(Entrez_ID, Concentration) %&gt;%
  summarise(logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;)

# ----------------- Merge and Annotate -----------------
af_logfc &lt;- inner_join(df_combined, deg_mean, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Group = paste(Response_Group, Concentration)
  )

# ----------------- Response Group Order -----------------
group_levels &lt;- c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
)
af_logfc$Response_Group &lt;- factor(af_logfc$Response_Group, levels = group_levels)

# ----------------- Wilcoxon Test -----------------
comparison_map &lt;- list(
  &quot;CX-DOX mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX only mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX-specific response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX only mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;CX total + DOX early response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX early + CX-DOX mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;
)

wilcoxon_results &lt;- lapply(names(comparison_map), function(resp_group) {
  control_group &lt;- comparison_map[[resp_group]]
  resp_vals &lt;- af_logfc$logFC[af_logfc$Group == resp_group]
  control_vals &lt;- af_logfc$logFC[af_logfc$Group == control_group]

  if (length(resp_vals) &gt;= 2 &amp;&amp; length(control_vals) &gt;= 2) {
    test_result &lt;- wilcox.test(resp_vals, control_vals)
    pval &lt;- test_result$p.value
    label &lt;- case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01  ~ &quot;**&quot;,
      pval &lt; 0.05  ~ &quot;*&quot;,
      TRUE ~ &quot;&quot;
    )
    y_pos &lt;- max(resp_vals, na.rm = TRUE) + 0.5
    return(data.frame(Group = resp_group, y_pos = y_pos, label = label, P_Value = signif(pval, 4)))
  }
  return(NULL)
}) %&gt;% bind_rows()

label_data &lt;- wilcoxon_results %&gt;%
  separate(Group, into = c(&quot;Response_Group&quot;, &quot;Concentration&quot;), sep = &quot; (?=0\\.)&quot;) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = group_levels))

# ----------------- Plot -----------------
ggplot(af_logfc, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~Concentration, scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Mean logFC of Atrial Fibrillation Genes\nAcross CorMotif Response Groups&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20AF%20logFC-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-AF-logFC-1">
Past versions of "Mean AF logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-AF-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean AF logFC-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-atrial-fibrillation-gwas-genes-across-corrmotifs-faceted-by-concentration-timepoint"
class="section level2">
<h2><strong>📌 Mean logFC of Atrial Fibrillation GWAS genes across
corrmotifs (Faceted by concentration × timepoint)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load Heart-Specific Genes -----------------
heart_genes &lt;- read.csv(&quot;data/Human_Heart_Genes.csv&quot;, stringsAsFactors = FALSE)
heart_genes$Entrez_ID &lt;- mapIds(
  org.Hs.eg.db,
  keys = heart_genes$Gene,
  column = &quot;ENTREZID&quot;,
  keytype = &quot;SYMBOL&quot;,
  multiVals = &quot;first&quot;
)
entrez_ids &lt;- na.omit(heart_genes$Entrez_ID) %&gt;% as.character()

# ----------------- Load CorMotif Response Group Data -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorMotif Group Membership -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Annotate DEG Files -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    ),
    Timepoint = factor(case_when(
      str_detect(Condition, &quot;_3$&quot;) ~ &quot;3 hr&quot;,
      str_detect(Condition, &quot;_24$&quot;) ~ &quot;24 hr&quot;,
      str_detect(Condition, &quot;_48$&quot;) ~ &quot;48 hr&quot;
    ), levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))
  )

# ----------------- Merge with Heart-Specific Targets -----------------
heart_all &lt;- inner_join(df_combined, deg_all, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID, column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response&quot;, &quot;CX-DOX mid-late response&quot;, &quot;DOX-specific response&quot;,
      &quot;DOX only mid-late response&quot;, &quot;CX total + DOX early response&quot;, &quot;DOX early + CX-DOX mid-late response&quot;
    ))
  )

# ----------------- Wilcoxon Test -----------------
comparison_groups &lt;- levels(heart_all$Response_Group)[-1]
stat_results &lt;- list()

for (conc in unique(heart_all$Concentration)) {
  for (tp in levels(heart_all$Timepoint)) {
    sub &lt;- heart_all %&gt;% filter(Concentration == conc, Timepoint == tp)
    for (grp in comparison_groups) {
      if (grp %in% unique(sub$Response_Group)) {
        test &lt;- tryCatch({
          wilcox.test(
            sub$logFC[sub$Response_Group == grp],
            sub$logFC[sub$Response_Group == &quot;Non response&quot;]
          )
        }, error = function(e) NULL)
        if (!is.null(test) &amp;&amp; !is.nan(test$p.value)) {
          stat_results[[length(stat_results) + 1]] &lt;- data.frame(
            Response_Group = grp,
            Concentration = conc,
            Timepoint = tp,
            p_value = signif(test$p.value, 3),
            label = case_when(
              test$p.value &lt; 0.001 ~ &quot;***&quot;,
              test$p.value &lt; 0.01 ~ &quot;**&quot;,
              test$p.value &lt; 0.05 ~ &quot;*&quot;,
              TRUE ~ &quot;&quot;
            ),
            y_pos = max(sub$logFC[sub$Response_Group == grp], na.rm = TRUE) + 0.5
          )
        }
      }
    }
  }
}

label_data &lt;- bind_rows(stat_results) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = levels(heart_all$Response_Group)))

# ----------------- Plot -----------------
ggplot(heart_all, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7, na.rm = TRUE) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 4, vjust = 0) +
  facet_grid(rows = vars(factor(Timepoint, levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))),
             cols = vars(Concentration), scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Heart-Specific Gene logFC by\nCorMotif Group, Timepoint, and Concentration&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20AF%20logFC_2%20logFC1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-AF-logFC_2-logFC1-1">
Past versions of "Mean AF logFC_2 logFC1-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-AF-logFC_2-logFC1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean AF logFC_2 logFC1-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-atrial-fibrillation-gwas-genes-across-corrmotifs-faceted-by-concentration-timepoint-drug-cx-vs-dox-side-by-side"
class="section level2">
<h2><strong>📌 Mean logFC of Atrial Fibrillation GWAS genes across
corrmotifs (Faceted by concentration × timepoint × drug (CX vs DOX
side-by-side))</strong></h2>
<pre class="r"><code>############ AF GENES — 0.1 µM #######################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- AF Entrez IDs -----------------
entrez_ids &lt;- c(
  3425, 53834, 9464, 79658, 55876, 3781, 9472, 11017, 64395, 23353,
  57794, 79591, 2702, 51306, 857, 10418, 9644, 115509, 89797, 10728,
  282996, 8497, 5695, 9026, 57666, 143384, 5883, 90102, 84909, 221037,
  29119, 79933, 391, 81575, 8738, 5396, 26010, 744, 56981, 80724,
  5781, 6910, 4598, 79068, 2250, 56978, 3752, 5208, 9736, 26778,
  80315, 64374, 79159, 11078, 5308, 3782, 10654, 79776, 6310, 8082,
  2995, 23224, 79741, 5708, 26136, 9967, 84033, 9148, 29982, 81488,
  1788, 9208, 2908, 1021, 5339, 5727, 23118, 10090, 57585, 221154,
  1073, 29998, 817, 858, 3486, 5747, 2104, 55105, 4194, 3570,
  1607, 6586, 5570, 2070, 27145, 4646, 3159, 9748, 79991, 221035,
  5033, 80212, 196385, 79600, 11046, 54014, 151636, 64778, 92597, 3983,
  7090, 9267, 7273, 26018, 9948, 2131, 55870, 5523, 5318, 6239,
  3480, 55777, 2066, 11165, 2028, 57619, 2690, 23414, 55818, 84700,
  28965, 80204, 463, 5108, 222553, 387119, 28981, 11113, 10301, 995,
  23030, 2697, 996, 339500, 54805, 9960, 91404, 145781, 100820829, 84542,
  2176, 51684, 9513, 7473, 4666, 23150, 5915, 5062, 4016, 23039,
  159686, 1839, 5201, 93166, 64753, 29959, 5496, 23245, 5069, 56916,
  92344, 23092, 3992, 9415, 10554, 7456, 9570, 57178, 23143, 161176,
  5424, 2034, 10277, 11278, 79803, 6653, 4756, 132660, 5430, 9031,
  57158, 285761, 8110, 387700, 1829, 4126, 7323, 51308, 7332, 6598,
  3757, 6187, 6660, 10529, 6920, 115286, 8451, 8943, 4137, 7514,
  4801, 9709, 23177, 8671, 29915, 26207, 3680, 490, 493856, 58489,
  54897, 4625, 22955, 84952, 10221, 2263, 84641, 4892, 1026, 84650,
  8382, 221656, 2969, 144453, 117177, 2626, 8476, 161882, 51807, 4624,
  9612, 55795, 51043, 144348, 51232, 8729, 3899, 11155, 23316, 79006,
  146330, 6403, 6331, 4300, 427, 845, 3313, 113622, 5789, 376132,
  29841, 8462, 203859, 401397, 5506, 55521, 5819, 4059, 6934, 57727,
  23066, 79568, 83478, 10087, 9586, 222194, 10466, 10499, 58499, 79720,
  4772, 10794, 125919, 602, 27332, 59345, 340359, 3705, 64710, 57801,
  10818, 143684, 149281, 55013, 23095, 93649, 84034, 23347, 440926, 11124,
  23293, 51426, 832, 7068, 57646, 152002, 7531, 1398, 100101267, 221937,
  26873, 3709, 10576, 27040, 201176, 284403, 307, 6525, 5387, 1808,
  114907, 1952, 4091, 5096, 23451, 51207, 142891, 4084, 3797, 1185,
  5334, 217, 2042, 7781, 253461, 152404, 2992, 153478, 6695, 26249,
  662, 23036, 22852, 23411, 23387, 6786, 7690, 10021, 8925, 5595,
  63892, 23466, 11149, 139411, 755, 55347, 22820, 55111, 84444, 6258,
  57623, 1387, 3762, 5861, 3339, 114822, 129787, 196528, 54437, 10395,
  4023, 5095, 894, 3156, 1877, 283871, 2673, 23157, 10512, 2258,
  79750, 84665, 26091, 5978, 8751, 79695, 2768, 155382, 84163, 7709,
  4092, 1837, 2064, 4629, 55122, 8882, 150962, 23013, 8742, 489,
  201134, 55114, 84264, 64428, 4089, 861, 7422
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.1 µM) -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.1 µM DEG Files -----------------
deg_0.1 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.1 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late\nresponse (0.1 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.1 &lt;- deg_0.1 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.1 µM)&quot;,
      &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.1 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.1 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.1 µM AF Genes) -----------------
ggplot(deg_0.1, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;AF-Associated Gene logFC by Drug and Response Group (0.1 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20AF%20logFC_3%20logFC3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-AF-logFC_3-logFC3-1">
Past versions of "Mean AF logFC_3 logFC3-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-AF-logFC_3-logFC3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean AF logFC_3 logFC3-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>############ AF GENES — 0.5 µM #######################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- AF Entrez IDs -----------------
entrez_ids &lt;- c(
  3425, 53834, 9464, 79658, 55876, 3781, 9472, 11017, 64395, 23353,
  57794, 79591, 2702, 51306, 857, 10418, 9644, 115509, 89797, 10728,
  282996, 8497, 5695, 9026, 57666, 143384, 5883, 90102, 84909, 221037,
  29119, 79933, 391, 81575, 8738, 5396, 26010, 744, 56981, 80724,
  5781, 6910, 4598, 79068, 2250, 56978, 3752, 5208, 9736, 26778,
  80315, 64374, 79159, 11078, 5308, 3782, 10654, 79776, 6310, 8082,
  2995, 23224, 79741, 5708, 26136, 9967, 84033, 9148, 29982, 81488,
  1788, 9208, 2908, 1021, 5339, 5727, 23118, 10090, 57585, 221154,
  1073, 29998, 817, 858, 3486, 5747, 2104, 55105, 4194, 3570,
  1607, 6586, 5570, 2070, 27145, 4646, 3159, 9748, 79991, 221035,
  5033, 80212, 196385, 79600, 11046, 54014, 151636, 64778, 92597, 3983,
  7090, 9267, 7273, 26018, 9948, 2131, 55870, 5523, 5318, 6239,
  3480, 55777, 2066, 11165, 2028, 57619, 2690, 23414, 55818, 84700,
  28965, 80204, 463, 5108, 222553, 387119, 28981, 11113, 10301, 995,
  23030, 2697, 996, 339500, 54805, 9960, 91404, 145781, 100820829, 84542,
  2176, 51684, 9513, 7473, 4666, 23150, 5915, 5062, 4016, 23039,
  159686, 1839, 5201, 93166, 64753, 29959, 5496, 23245, 5069, 56916,
  92344, 23092, 3992, 9415, 10554, 7456, 9570, 57178, 23143, 161176,
  5424, 2034, 10277, 11278, 79803, 6653, 4756, 132660, 5430, 9031,
  57158, 285761, 8110, 387700, 1829, 4126, 7323, 51308, 7332, 6598,
  3757, 6187, 6660, 10529, 6920, 115286, 8451, 8943, 4137, 7514,
  4801, 9709, 23177, 8671, 29915, 26207, 3680, 490, 493856, 58489,
  54897, 4625, 22955, 84952, 10221, 2263, 84641, 4892, 1026, 84650,
  8382, 221656, 2969, 144453, 117177, 2626, 8476, 161882, 51807, 4624,
  9612, 55795, 51043, 144348, 51232, 8729, 3899, 11155, 23316, 79006,
  146330, 6403, 6331, 4300, 427, 845, 3313, 113622, 5789, 376132,
  29841, 8462, 203859, 401397, 5506, 55521, 5819, 4059, 6934, 57727,
  23066, 79568, 83478, 10087, 9586, 222194, 10466, 10499, 58499, 79720,
  4772, 10794, 125919, 602, 27332, 59345, 340359, 3705, 64710, 57801,
  10818, 143684, 149281, 55013, 23095, 93649, 84034, 23347, 440926, 11124,
  23293, 51426, 832, 7068, 57646, 152002, 7531, 1398, 100101267, 221937,
  26873, 3709, 10576, 27040, 201176, 284403, 307, 6525, 5387, 1808,
  114907, 1952, 4091, 5096, 23451, 51207, 142891, 4084, 3797, 1185,
  5334, 217, 2042, 7781, 253461, 152404, 2992, 153478, 6695, 26249,
  662, 23036, 22852, 23411, 23387, 6786, 7690, 10021, 8925, 5595,
  63892, 23466, 11149, 139411, 755, 55347, 22820, 55111, 84444, 6258,
  57623, 1387, 3762, 5861, 3339, 114822, 129787, 196528, 54437, 10395,
  4023, 5095, 894, 3156, 1877, 283871, 2673, 23157, 10512, 2258,
  79750, 84665, 26091, 5978, 8751, 79695, 2768, 155382, 84163, 7709,
  4092, 1837, 2064, 4629, 55122, 8882, 150962, 23013, 8742, 489,
  201134, 55114, 84264, 64428, 4089, 861, 7422
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.5 µM) -----------------
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.5 µM DEG Files -----------------
deg_0.5 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.5 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 µM)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.5 &lt;- deg_0.5 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.5 µM)&quot;,
      &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.5 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.5 µM AF Genes) -----------------
ggplot(deg_0.5, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;AF-Associated Gene logFC by Drug and Response Group (0.5 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20AF%20logFC_3%20logFC3-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-AF-logFC_3-logFC3-2">
Past versions of "Mean AF logFC_3 logFC3-2.png"
</button>
</p>
<div id="fig-no-spaces-Mean-AF-logFC_3-logFC3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean AF logFC_3 logFC3-2.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="proportion-of-heart-failure-gwas-genes-across-corrmotifs-per-concentration"
class="section level2">
<h2><strong>📌 Proportion of Heart Failure GWAS genes across corrmotifs
(per concentration)</strong></h2>
<pre class="r"><code># ----------------- Load Required Libraries -----------------
library(dplyr)
library(ggplot2)

# ----------------- Heart Failure Entrez IDs -----------------
hf_entrez_ids &lt;- c(
  9709, 8882, 4023, 29959, 5496, 3992, 9415, 5308, 1026, 54437, 79068, 10221,
  9031, 1187, 1952, 3705, 84722, 7273, 23293, 155382, 9531, 602, 27258, 84163,
  81846, 79933, 56911, 64753, 93210, 1021, 283450, 5998, 57602, 114991, 7073,
  3156, 100101267, 22996, 285025, 11080, 11124, 54810, 7531, 27241, 4774, 57794,
  463, 91319, 6598, 9640, 2186, 26010, 80816, 571, 88, 51652, 64788, 90523, 2969,
  7781, 80777, 10725, 23387, 817, 134728, 8842, 949, 6934, 129787, 10327, 202052,
  2318, 5578, 6801, 6311, 10019, 80724, 217, 84909, 388591, 55101, 9839, 27161,
  5310, 387119, 4641, 5587, 55188, 222553, 9960, 22852, 10087, 9570, 54497,
  200942, 26249, 4137, 375056, 5409, 64116, 8291, 22876, 339855, 4864, 5142,
  221692, 55023, 51426, 6146, 84251, 8189, 27332, 57099, 1869, 1112, 23327,
  11264, 6001
) %&gt;% as.character()

# ----------------- Load CorrMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% hf_entrez_ids, &quot;HF Genes&quot;, &quot;Non-HF Genes&quot;),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% hf_entrez_ids, &quot;HF Genes&quot;, &quot;Non-HF Genes&quot;),
    Concentration = &quot;0.5 µM&quot;
  )

# ----------------- Combine Data -----------------
df_combined &lt;- bind_rows(df_0.1, df_0.5)
df_combined$Category &lt;- factor(df_combined$Category, levels = c(&quot;HF Genes&quot;, &quot;Non-HF Genes&quot;))

# ----------------- Calculate Proportions -----------------
proportion_data &lt;- df_combined %&gt;%
  group_by(Concentration, Response_Group, Category) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Chi-square or Fisher&#39;s Test -----------------
get_chi_or_fisher &lt;- function(df, ref_group) {
  ref_counts &lt;- df %&gt;%
    ungroup() %&gt;%
    filter(Response_Group == ref_group) %&gt;%
    { ref_tbl &lt;- .[ , c(&quot;Category&quot;, &quot;Count&quot;)]; setNames(ref_tbl$Count, ref_tbl$Category) }

  df %&gt;%
    filter(Response_Group != ref_group) %&gt;%
    group_by(Response_Group) %&gt;%
    summarise(
      p_value = {
        group_counts &lt;- Count[Category %in% c(&quot;HF Genes&quot;, &quot;Non-HF Genes&quot;)]
        if (length(group_counts) &lt; 2) group_counts &lt;- c(group_counts, 0)
        contingency_table &lt;- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts[&quot;HF Genes&quot;], ref_counts[&quot;Non-HF Genes&quot;]
        ), nrow = 2)
        if (any(contingency_table &lt; 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = &quot;drop&quot;
    ) %&gt;%
    mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))
}

# ----------------- Run test -----------------
chi_0.1 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.1 µM&quot;), &quot;Non response&quot;)</code></pre>
<pre><code>Warning: There was 1 warning in `summarise()`.
ℹ In argument: `p_value = { ... }`.
ℹ In group 1: `Response_Group = &quot;CX-DOX mid-late response&quot;`.
Caused by warning in `chisq.test()`:
! Chi-squared approximation may be incorrect</code></pre>
<pre class="r"><code>chi_0.5 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.5 µM&quot;), &quot;Non response&quot;)
chi_all &lt;- bind_rows(
  chi_0.1 %&gt;% mutate(Concentration = &quot;0.1 µM&quot;),
  chi_0.5 %&gt;% mutate(Concentration = &quot;0.5 µM&quot;)
)
proportion_data &lt;- left_join(proportion_data, chi_all, by = c(&quot;Concentration&quot;, &quot;Response_Group&quot;))

# ----------------- Factor Order and Star Label -----------------
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
))
proportion_data$Concentration &lt;- factor(proportion_data$Concentration, levels = c(&quot;0.1 µM&quot;, &quot;0.5 µM&quot;))
label_data &lt;- proportion_data %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  summarise(Significance = dplyr::first(Significance), .groups = &quot;drop&quot;) %&gt;%
  filter(!is.na(Significance) &amp; Significance != &quot;&quot;) %&gt;%
  mutate(y_pos = 105)

# ----------------- Final Plot -----------------
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = Significance),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~ Concentration, scales = &quot;free_x&quot;) +
  scale_fill_manual(values = c(&quot;HF Genes&quot; = &quot;#E41A1C&quot;, &quot;Non-HF Genes&quot; = &quot;#377EB8&quot;)) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +
  labs(
    title = &quot;Heart Failure Gene Proportions Across\nCormotif Clusters (0.1 and 0.5 µM)&quot;,
    x = &quot;Response Groups&quot;, y = &quot;Percentage of Genes&quot;, fill = &quot;Gene Category&quot;
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = &quot;bold&quot;),
    axis.title.x = element_text(size = 14, face = &quot;bold&quot;),
    axis.title.y = element_text(size = 14, face = &quot;bold&quot;),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    panel.spacing = unit(1.2, &quot;lines&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/HF%20Prop-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-HF-Prop-1">
Past versions of "HF Prop-1.png"
</button>
</p>
<div id="fig-no-spaces-HF-Prop-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/HF Prop-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-heart-failure-gwas-genes-across-corrmotifs-per-concentration"
class="section level2">
<h2><strong>📌 Mean logFC of Heart Failure GWAS genes across corrmotifs
(per concentration)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Heart Failure Entrez IDs -----------------
entrez_ids &lt;- c(
  9709, 8882, 4023, 29959, 5496, 3992, 9415, 5308, 1026, 54437, 79068, 10221,
  9031, 1187, 1952, 3705, 84722, 7273, 23293, 155382, 9531, 602, 27258, 84163,
  81846, 79933, 56911, 64753, 93210, 1021, 283450, 5998, 57602, 114991, 7073,
  3156, 100101267, 22996, 285025, 11080, 11124, 54810, 7531, 27241, 4774, 57794,
  463, 91319, 6598, 9640, 2186, 26010, 80816, 571, 88, 51652, 64788, 90523, 2969,
  7781, 80777, 10725, 23387, 817, 134728, 8842, 949, 6934, 129787, 10327, 202052,
  2318, 5578, 6801, 6311, 10019, 80724, 217, 84909, 388591, 55101, 9839, 27161,
  5310, 387119, 4641, 5587, 55188, 222553, 9960, 22852, 10087, 9570, 54497,
  200942, 26249, 4137, 375056, 5409, 64116, 8291, 22876, 339855, 4864, 5142,
  221692, 55023, 51426, 6146, 84251, 8189, 27332, 57099, 1869, 1112, 23327,
  11264, 6001
) %&gt;% as.character()

# ----------------- Load CorMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate Response Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Aggregate DEG Data -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    )
  )

deg_mean &lt;- deg_all %&gt;%
  group_by(Entrez_ID, Concentration) %&gt;%
  summarise(logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;)

# ----------------- Merge and Annotate -----------------
hf_logfc &lt;- inner_join(df_combined, deg_mean, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Group = paste(Response_Group, Concentration)
  )

# ----------------- Define Group Order -----------------
group_levels &lt;- c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
)
hf_logfc$Response_Group &lt;- factor(hf_logfc$Response_Group, levels = group_levels)

# ----------------- Wilcoxon Test -----------------
comparison_map &lt;- list(
  &quot;CX-DOX mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX only mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX-specific response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX only mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;CX total + DOX early response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX early + CX-DOX mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;
)

wilcoxon_results &lt;- lapply(names(comparison_map), function(resp_group) {
  control_group &lt;- comparison_map[[resp_group]]
  resp_vals &lt;- hf_logfc$logFC[hf_logfc$Group == resp_group]
  control_vals &lt;- hf_logfc$logFC[hf_logfc$Group == control_group]

  if (length(resp_vals) &gt;= 2 &amp;&amp; length(control_vals) &gt;= 2) {
    test_result &lt;- wilcox.test(resp_vals, control_vals)
    pval &lt;- test_result$p.value
    label &lt;- case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01  ~ &quot;**&quot;,
      pval &lt; 0.05  ~ &quot;*&quot;,
      TRUE ~ &quot;&quot;
    )
    y_pos &lt;- max(resp_vals, na.rm = TRUE) + 0.5
    return(data.frame(Group = resp_group, y_pos = y_pos, label = label, P_Value = signif(pval, 4)))
  }
  return(NULL)
}) %&gt;% bind_rows()

label_data &lt;- wilcoxon_results %&gt;%
  separate(Group, into = c(&quot;Response_Group&quot;, &quot;Concentration&quot;), sep = &quot; (?=0\\.)&quot;) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = group_levels))

# ----------------- Plot -----------------
ggplot(hf_logfc, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~Concentration, scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Mean logFC of Heart Failure Genes\nAcross CorMotif Response Groups&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20HF%20logFC-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-HF-logFC-1">
Past versions of "Mean HF logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-HF-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean HF logFC-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-heart-failure-gwas-genes-across-corrmotifs-faceted-by-concentration-timepoint"
class="section level2">
<h2><strong>📌 Mean logFC of Heart Failure GWAS genes across corrmotifs
(Faceted by concentration × timepoint)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Heart Failure Entrez IDs -----------------
entrez_ids &lt;- c(
  9709, 8882, 4023, 29959, 5496, 3992, 9415, 5308, 1026, 54437, 79068, 10221,
  9031, 1187, 1952, 3705, 84722, 7273, 23293, 155382, 9531, 602, 27258, 84163,
  81846, 79933, 56911, 64753, 93210, 1021, 283450, 5998, 57602, 114991, 7073,
  3156, 100101267, 22996, 285025, 11080, 11124, 54810, 7531, 27241, 4774, 57794,
  463, 91319, 6598, 9640, 2186, 26010, 80816, 571, 88, 51652, 64788, 90523, 2969,
  7781, 80777, 10725, 23387, 817, 134728, 8842, 949, 6934, 129787, 10327, 202052,
  2318, 5578, 6801, 6311, 10019, 80724, 217, 84909, 388591, 55101, 9839, 27161,
  5310, 387119, 4641, 5587, 55188, 222553, 9960, 22852, 10087, 9570, 54497,
  200942, 26249, 4137, 375056, 5409, 64116, 8291, 22876, 339855, 4864, 5142,
  221692, 55023, 51426, 6146, 84251, 8189, 27332, 57099, 1869, 1112, 23327,
  11264, 6001
) %&gt;% as.character()

# ----------------- Load CorMotif Response Group Data -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorMotif Group Membership -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load DEG Files -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    ),
    Timepoint = factor(case_when(
      str_detect(Condition, &quot;_3$&quot;) ~ &quot;3 hr&quot;,
      str_detect(Condition, &quot;_24$&quot;) ~ &quot;24 hr&quot;,
      str_detect(Condition, &quot;_48$&quot;) ~ &quot;48 hr&quot;
    ), levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))
  )

# ----------------- Merge and Annotate -----------------
hf_all &lt;- inner_join(df_combined, deg_all, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response&quot;, &quot;CX-DOX mid-late response&quot;, &quot;DOX-specific response&quot;,
      &quot;DOX only mid-late response&quot;, &quot;CX total + DOX early response&quot;, &quot;DOX early + CX-DOX mid-late response&quot;
    ))
  )

# ----------------- Wilcoxon Tests -----------------
comparison_groups &lt;- levels(hf_all$Response_Group)[-1]
stat_results &lt;- list()

for (conc in unique(hf_all$Concentration)) {
  for (tp in levels(hf_all$Timepoint)) {
    sub &lt;- hf_all %&gt;% filter(Concentration == conc, Timepoint == tp)
    for (grp in comparison_groups) {
      if (grp %in% unique(sub$Response_Group)) {
        test &lt;- tryCatch({
          wilcox.test(
            sub$logFC[sub$Response_Group == grp],
            sub$logFC[sub$Response_Group == &quot;Non response&quot;]
          )
        }, error = function(e) NULL)
        if (!is.null(test) &amp;&amp; !is.nan(test$p.value)) {
          stat_results[[length(stat_results) + 1]] &lt;- data.frame(
            Response_Group = grp,
            Concentration = conc,
            Timepoint = tp,
            p_value = signif(test$p.value, 3),
            label = case_when(
              test$p.value &lt; 0.001 ~ &quot;***&quot;,
              test$p.value &lt; 0.01 ~ &quot;**&quot;,
              test$p.value &lt; 0.05 ~ &quot;*&quot;,
              TRUE ~ &quot;&quot;
            ),
            y_pos = max(sub$logFC[sub$Response_Group == grp], na.rm = TRUE) + 0.5
          )
        }
      }
    }
  }
}

label_data &lt;- bind_rows(stat_results) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = levels(hf_all$Response_Group)))

# ----------------- Plot -----------------
ggplot(hf_all, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7, na.rm = TRUE) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 4, vjust = 0) +
  facet_grid(rows = vars(factor(Timepoint, levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))),
             cols = vars(Concentration), scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Heart Failure Gene logFC by\nCorMotif Group, Timepoint, and Concentration&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20HF%20logFC_2%20logFC1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-HF-logFC_2-logFC1-1">
Past versions of "Mean HF logFC_2 logFC1-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-HF-logFC_2-logFC1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean HF logFC_2 logFC1-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-heart-failure-gwas-genes-across-corrmotifs-faceted-by-concentration-timepoint-drug-cx-vs-dox-side-by-side"
class="section level2">
<h2><strong>📌 Mean logFC of Heart Failure GWAS genes across corrmotifs
(Faceted by concentration × timepoint × drug (CX vs DOX
side-by-side))</strong></h2>
<pre class="r"><code>############ HEART FAILURE GENES — 0.1 µM #######################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Heart Failure Entrez IDs -----------------
entrez_ids &lt;- c(
  9709, 8882, 4023, 29959, 5496, 3992, 9415, 5308, 1026, 54437, 79068, 10221,
  9031, 1187, 1952, 3705, 84722, 7273, 23293, 155382, 9531, 602, 27258, 84163,
  81846, 79933, 56911, 64753, 93210, 1021, 283450, 5998, 57602, 114991, 7073,
  3156, 100101267, 22996, 285025, 11080, 11124, 54810, 7531, 27241, 4774, 57794,
  463, 91319, 6598, 9640, 2186, 26010, 80816, 571, 88, 51652, 64788, 90523, 2969,
  7781, 80777, 10725, 23387, 817, 134728, 8842, 949, 6934, 129787, 10327, 202052,
  2318, 5578, 6801, 6311, 10019, 80724, 217, 84909, 388591, 55101, 9839, 27161,
  5310, 387119, 4641, 5587, 55188, 222553, 9960, 22852, 10087, 9570, 54497,
  200942, 26249, 4137, 375056, 5409, 64116, 8291, 22876, 339855, 4864, 5142,
  221692, 55023, 51426, 6146, 84251, 8189, 27332, 57099, 1869, 1112, 23327,
  11264, 6001
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.1 µM) -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.1 µM DEG Files -----------------
deg_0.1 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.1 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late\nresponse (0.1 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.1 &lt;- deg_0.1 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.1 µM)&quot;,
      &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.1 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.1 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.1 µM HF Genes) -----------------
ggplot(deg_0.1, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;Heart Failure Gene logFC by Drug and Response Group (0.1 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20HF%20logFC_3%20logFC3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-HF-logFC_3-logFC3-1">
Past versions of "Mean HF logFC_3 logFC3-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-HF-logFC_3-logFC3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean HF logFC_3 logFC3-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>############ HEART FAILURE GENES — 0.5 µM #######################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Heart Failure Entrez IDs -----------------
entrez_ids &lt;- c(
  9709, 8882, 4023, 29959, 5496, 3992, 9415, 5308, 1026, 54437, 79068, 10221,
  9031, 1187, 1952, 3705, 84722, 7273, 23293, 155382, 9531, 602, 27258, 84163,
  81846, 79933, 56911, 64753, 93210, 1021, 283450, 5998, 57602, 114991, 7073,
  3156, 100101267, 22996, 285025, 11080, 11124, 54810, 7531, 27241, 4774, 57794,
  463, 91319, 6598, 9640, 2186, 26010, 80816, 571, 88, 51652, 64788, 90523, 2969,
  7781, 80777, 10725, 23387, 817, 134728, 8842, 949, 6934, 129787, 10327, 202052,
  2318, 5578, 6801, 6311, 10019, 80724, 217, 84909, 388591, 55101, 9839, 27161,
  5310, 387119, 4641, 5587, 55188, 222553, 9960, 22852, 10087, 9570, 54497,
  200942, 26249, 4137, 375056, 5409, 64116, 8291, 22876, 339855, 4864, 5142,
  221692, 55023, 51426, 6146, 84251, 8189, 27332, 57099, 1869, 1112, 23327,
  11264, 6001
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.5 µM) -----------------
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.5 µM DEG Files -----------------
deg_0.5 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.5 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 µM)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.5 &lt;- deg_0.5 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.5 µM)&quot;,
      &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.5 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.5 µM HF Genes) -----------------
ggplot(deg_0.5, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;Heart Failure Gene logFC by Drug and Response Group (0.5 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20HF%20logFC_3%20logFC3-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-HF-logFC_3-logFC3-2">
Past versions of "Mean HF logFC_3 logFC3-2.png"
</button>
</p>
<div id="fig-no-spaces-Mean-HF-logFC_3-logFC3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean HF logFC_3 logFC3-2.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="corr-motif-boxplots" class="section level2">
<h2><strong>📌 Corr-motif Boxplots</strong></h2>
</div>
<div id="non-response-0.1-micromolar" class="section level2">
<h2><strong>📌 Non response (0.1 micromolar)</strong></h2>
<pre class="r"><code># Load libraries
library(dplyr)
library(ggplot2)
library(reshape2)

# Load response group files
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)

# Load expression dataset
boxplot1 &lt;- read.csv(&quot;data/boxplot1.csv&quot;, check.names = FALSE)
boxplot1 &lt;- as.data.frame(boxplot1)

# Choose a gene from the non-response group
target_entrez &lt;- 92342  # Ensure this is from prob_1_0.1

# Stop if the gene is not in the non-response group
if (!(target_entrez %in% prob_1_0.1)) {
  stop(&quot;Selected gene is not in the non-response group for 0.1 micromolar.&quot;)
}

# Filter for selected gene
gene_data &lt;- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if(nrow(gene_data) == 0) stop(&quot;No data found for selected ENTIREZID.&quot;)

# Reshape to long format
gene_data_long &lt;- melt(gene_data,
                       id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                       variable.name = &quot;Sample&quot;,
                       value.name = &quot;log2CPM&quot;)

# Extract metadata from sample names
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)
  )

# Filter for 0.1 micromolar only
gene_data_long &lt;- gene_data_long %&gt;% filter(Concentration == &quot;0.1&quot;)

# Convert to factors
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = &quot;0.1&quot;)

# Map individual IDs
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, 
                  &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], &quot;Unknown&quot;))

# Define color palette for drugs
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;#08306B&quot;, &quot;DOX&quot; = &quot;#E7298A&quot;, &quot;VEH&quot; = &quot;green&quot;)

# Extract gene symbol for labeling
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +  # Only facets by Time now
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;Non response (0.1 micromolar)&quot;) +
  labs(
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/box1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box1-1">
Past versions of box1-1.png
</button>
</p>
<div id="fig-box1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/box1-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cx-dox-mid-late-response-0.1-micromolar"
class="section level2">
<h2><strong>📌 CX-DOX Mid-Late Response (0.1 micromolar)</strong></h2>
<pre class="r"><code># Load libraries
library(dplyr)
library(ggplot2)
library(reshape2)

# Load response group files
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)

prob_2_0.1</code></pre>
<pre><code>  [1] &quot;1870&quot;      &quot;79000&quot;     &quot;3925&quot;      &quot;83667&quot;     &quot;63967&quot;     &quot;55143&quot;    
  [7] &quot;991&quot;       &quot;11004&quot;     &quot;4678&quot;      &quot;8438&quot;      &quot;6491&quot;      &quot;1031&quot;     
 [13] &quot;4998&quot;      &quot;55706&quot;     &quot;51668&quot;     &quot;7398&quot;      &quot;55635&quot;     &quot;8317&quot;     
 [19] &quot;343099&quot;    &quot;163786&quot;    &quot;29899&quot;     &quot;57535&quot;     &quot;84722&quot;     &quot;64858&quot;    
 [25] &quot;653820&quot;    &quot;554282&quot;    &quot;728833&quot;    &quot;8370&quot;      &quot;23632&quot;     &quot;54964&quot;    
 [31] &quot;1163&quot;      &quot;128239&quot;    &quot;3068&quot;      &quot;83540&quot;     &quot;55732&quot;     &quot;91687&quot;    
 [37] &quot;259266&quot;    &quot;9928&quot;      &quot;29089&quot;     &quot;729533&quot;    &quot;4751&quot;      &quot;51514&quot;    
 [43] &quot;1063&quot;      &quot;3930&quot;      &quot;9156&quot;      &quot;10276&quot;     &quot;55388&quot;     &quot;79723&quot;    
 [49] &quot;414149&quot;    &quot;84930&quot;     &quot;11130&quot;     &quot;983&quot;       &quot;219790&quot;    &quot;1763&quot;     
 [55] &quot;79933&quot;     &quot;9585&quot;      &quot;3832&quot;      &quot;55165&quot;     &quot;3070&quot;      &quot;84986&quot;    
 [61] &quot;10613&quot;     &quot;4288&quot;      &quot;6240&quot;      &quot;10346&quot;     &quot;113174&quot;    &quot;258010&quot;   
 [67] &quot;81930&quot;     &quot;1643&quot;      &quot;374393&quot;    &quot;2237&quot;      &quot;3619&quot;      &quot;113130&quot;   
 [73] &quot;23649&quot;     &quot;283131&quot;    &quot;9633&quot;      &quot;10714&quot;     &quot;220042&quot;    &quot;143888&quot;   
 [79] &quot;3014&quot;      &quot;53340&quot;     &quot;1111&quot;      &quot;23310&quot;     &quot;2305&quot;      &quot;10635&quot;    
 [85] &quot;9918&quot;      &quot;83461&quot;     &quot;440081&quot;    &quot;441631&quot;    &quot;79022&quot;     &quot;10376&quot;    
 [91] &quot;84790&quot;     &quot;10024&quot;     &quot;29127&quot;     &quot;9700&quot;      &quot;23468&quot;     &quot;1017&quot;     
 [97] &quot;8914&quot;      &quot;5557&quot;      &quot;23306&quot;     &quot;57122&quot;     &quot;4193&quot;      &quot;8089&quot;     
[103] &quot;6636&quot;      &quot;100128191&quot; &quot;7112&quot;      &quot;283431&quot;    &quot;55010&quot;     &quot;7374&quot;     
[109] &quot;5985&quot;      &quot;11113&quot;     &quot;9735&quot;      &quot;387893&quot;    &quot;140707&quot;    &quot;102659353&quot;
[115] &quot;5426&quot;      &quot;221150&quot;    &quot;55835&quot;     &quot;3146&quot;      &quot;675&quot;       &quot;5983&quot;     
[121] &quot;11340&quot;     &quot;10301&quot;     &quot;440145&quot;    &quot;79866&quot;     &quot;10426&quot;     &quot;7027&quot;     
[127] &quot;650669&quot;    &quot;55632&quot;     &quot;55320&quot;     &quot;5427&quot;      &quot;64841&quot;     &quot;1033&quot;     
[133] &quot;11169&quot;     &quot;9787&quot;      &quot;4522&quot;      &quot;161145&quot;    &quot;122953&quot;    &quot;145508&quot;   
[139] &quot;90141&quot;     &quot;7443&quot;      &quot;89839&quot;     &quot;100288637&quot; &quot;9824&quot;      &quot;701&quot;      
[145] &quot;90417&quot;     &quot;57082&quot;     &quot;5888&quot;      &quot;11339&quot;     &quot;51203&quot;     &quot;79968&quot;    
[151] &quot;1854&quot;      &quot;22995&quot;     &quot;55329&quot;     &quot;9133&quot;      &quot;9768&quot;      &quot;80119&quot;    
[157] &quot;54962&quot;     &quot;55055&quot;     &quot;9493&quot;      &quot;64782&quot;     &quot;55215&quot;     &quot;90381&quot;    
[163] &quot;641&quot;       &quot;9055&quot;      &quot;1106&quot;      &quot;63922&quot;     &quot;899&quot;       &quot;124222&quot;   
[169] &quot;9088&quot;      &quot;23589&quot;     &quot;116028&quot;    &quot;54820&quot;     &quot;23204&quot;     &quot;5347&quot;     
[175] &quot;3835&quot;      &quot;8479&quot;      &quot;79801&quot;     &quot;23594&quot;     &quot;64785&quot;     &quot;6560&quot;     
[181] &quot;55159&quot;     &quot;55839&quot;     &quot;51659&quot;     &quot;81620&quot;     &quot;2175&quot;      &quot;359845&quot;   
[187] &quot;727910&quot;    &quot;83903&quot;     &quot;54478&quot;     &quot;116840&quot;    &quot;9212&quot;      &quot;6470&quot;     
[193] &quot;27346&quot;     &quot;10615&quot;     &quot;79915&quot;     &quot;990&quot;       &quot;7153&quot;      &quot;29893&quot;    
[199] &quot;672&quot;       &quot;78995&quot;     &quot;146909&quot;    &quot;10951&quot;     &quot;146956&quot;    &quot;348235&quot;   
[205] &quot;55771&quot;     &quot;83990&quot;     &quot;3838&quot;      &quot;9382&quot;      &quot;2232&quot;      &quot;7083&quot;     
[211] &quot;332&quot;       &quot;10403&quot;     &quot;162655&quot;    &quot;6632&quot;      &quot;220134&quot;    &quot;162681&quot;   
[217] &quot;148229&quot;    &quot;84823&quot;     &quot;10036&quot;     &quot;126282&quot;    &quot;29128&quot;     &quot;162993&quot;   
[223] &quot;1786&quot;      &quot;147841&quot;    &quot;10535&quot;     &quot;55723&quot;     &quot;93323&quot;     &quot;54858&quot;    
[229] &quot;81931&quot;     &quot;284403&quot;    &quot;8193&quot;      &quot;7733&quot;      &quot;27113&quot;     &quot;3978&quot;     
[235] &quot;5424&quot;      &quot;84798&quot;     &quot;27338&quot;     &quot;6241&quot;      &quot;348654&quot;    &quot;79172&quot;    
[241] &quot;1058&quot;      &quot;4436&quot;      &quot;6637&quot;      &quot;23397&quot;     &quot;699&quot;       &quot;150468&quot;   
[247] &quot;4175&quot;      &quot;57405&quot;     &quot;83879&quot;     &quot;129401&quot;    &quot;284992&quot;    &quot;151246&quot;   
[253] &quot;79411&quot;     &quot;55355&quot;     &quot;23178&quot;     &quot;1841&quot;      &quot;994&quot;       &quot;5111&quot;     
[259] &quot;84515&quot;     &quot;140733&quot;    &quot;9837&quot;      &quot;22974&quot;     &quot;1869&quot;      &quot;79980&quot;    
[265] &quot;5933&quot;      &quot;81610&quot;     &quot;4605&quot;      &quot;11065&quot;     &quot;6790&quot;      &quot;54069&quot;    
[271] &quot;8208&quot;      &quot;54058&quot;     &quot;8318&quot;      &quot;421&quot;       &quot;5902&quot;      &quot;11200&quot;    
[277] &quot;4174&quot;      &quot;8398&quot;      &quot;5905&quot;      &quot;79019&quot;     &quot;51512&quot;     &quot;2177&quot;     
[283] &quot;27258&quot;     &quot;151648&quot;    &quot;56992&quot;     &quot;993&quot;       &quot;10293&quot;     &quot;11186&quot;    
[289] &quot;25886&quot;     &quot;57650&quot;     &quot;10721&quot;     &quot;4171&quot;      &quot;8930&quot;      &quot;11073&quot;    
[295] &quot;10051&quot;     &quot;286827&quot;    &quot;1894&quot;      &quot;254827&quot;    &quot;5984&quot;      &quot;116832&quot;   
[301] &quot;7884&quot;      &quot;10460&quot;     &quot;7468&quot;      &quot;64151&quot;     &quot;132671&quot;    &quot;9662&quot;     
[307] &quot;1633&quot;      &quot;3015&quot;      &quot;1062&quot;      &quot;55345&quot;     &quot;4085&quot;      &quot;890&quot;      
[313] &quot;10733&quot;     &quot;132320&quot;    &quot;84057&quot;     &quot;201725&quot;    &quot;3148&quot;      &quot;55247&quot;    
[319] &quot;79682&quot;     &quot;55722&quot;     &quot;9319&quot;      &quot;375444&quot;    &quot;55789&quot;     &quot;64105&quot;    
[325] &quot;891&quot;       &quot;64946&quot;     &quot;100129716&quot; &quot;100996485&quot; &quot;10112&quot;     &quot;995&quot;      
[331] &quot;9232&quot;      &quot;134492&quot;    &quot;3161&quot;      &quot;54908&quot;     &quot;375484&quot;    &quot;83463&quot;    
[337] &quot;7913&quot;      &quot;51053&quot;     &quot;8339&quot;      &quot;7738&quot;      &quot;11270&quot;     &quot;9656&quot;     
[343] &quot;203068&quot;    &quot;6941&quot;      &quot;3833&quot;      &quot;1026&quot;      &quot;55166&quot;     &quot;4172&quot;     
[349] &quot;5558&quot;      &quot;7272&quot;      &quot;9994&quot;      &quot;253714&quot;    &quot;387103&quot;    &quot;2037&quot;     
[355] &quot;113115&quot;    &quot;26271&quot;     &quot;4521&quot;      &quot;29887&quot;     &quot;23658&quot;     &quot;54443&quot;    
[361] &quot;7378&quot;      &quot;10793&quot;     &quot;5982&quot;      &quot;55972&quot;     &quot;10926&quot;     &quot;4176&quot;     
[367] &quot;23165&quot;     &quot;2146&quot;      &quot;7516&quot;      &quot;54892&quot;     &quot;389610&quot;    &quot;8795&quot;     
[373] &quot;8794&quot;      &quot;157313&quot;    &quot;1808&quot;      &quot;157570&quot;    &quot;55872&quot;     &quot;84296&quot;    
[379] &quot;4173&quot;      &quot;8836&quot;      &quot;4603&quot;      &quot;1875&quot;      &quot;25788&quot;     &quot;9134&quot;     
[385] &quot;50484&quot;     &quot;79075&quot;     &quot;27085&quot;     &quot;29028&quot;     &quot;9401&quot;      &quot;347240&quot;   
[391] &quot;2189&quot;      &quot;84904&quot;     &quot;9833&quot;      &quot;100289137&quot; &quot;80010&quot;     &quot;1164&quot;     
[397] &quot;2176&quot;      &quot;195828&quot;    &quot;10541&quot;     &quot;10592&quot;     &quot;26147&quot;     &quot;11064&quot;    
[403] &quot;7633&quot;      &quot;29941&quot;     &quot;89958&quot;     &quot;10383&quot;     &quot;441478&quot;    &quot;6839&quot;     
[409] &quot;60401&quot;     &quot;24137&quot;     &quot;54821&quot;     &quot;79366&quot;     &quot;2491&quot;      &quot;11013&quot;    
[415] &quot;3149&quot;     </code></pre>
<pre class="r"><code># Load expression dataset
boxplot1 &lt;- read.csv(&quot;data/boxplot1.csv&quot;, check.names = FALSE)
boxplot1 &lt;- as.data.frame(boxplot1)

# Choose a gene from the CX-DOX mid-late response group
target_entrez &lt;- 4521  # Replace with your desired Entrez ID from prob_2_0.1

# Stop if the gene is not in the CX-DOX mid-late response group
if (!(target_entrez %in% prob_2_0.1)) {
  stop(&quot;Selected gene is not in the CX-DOX mid-late response group for 0.1 micromolar.&quot;)
}

# Filter for selected gene
gene_data &lt;- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if(nrow(gene_data) == 0) stop(&quot;No data found for selected ENTREZID.&quot;)

# Reshape to long format
gene_data_long &lt;- melt(gene_data,
                       id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                       variable.name = &quot;Sample&quot;,
                       value.name = &quot;log2CPM&quot;)

# Extract metadata from sample names
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)
  )

# Filter for 0.1 micromolar only
gene_data_long &lt;- gene_data_long %&gt;% filter(Concentration == &quot;0.1&quot;)

# Convert to factors
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = &quot;0.1&quot;)

# Map individual IDs
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, 
                  &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], &quot;Unknown&quot;))

# Define color palette for drugs
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;#08306B&quot;, &quot;DOX&quot; = &quot;#E7298A&quot;, &quot;VEH&quot; = &quot;green&quot;)

# Extract gene symbol for labeling
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +  # Only facets by Time
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;CX-DOX mid-late\nresponse (0.1 micromolar)&quot;) +
  labs(
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/box2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box2-1">
Past versions of box2-1.png
</button>
</p>
<div id="fig-box2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/box2-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dox-only-mid-late-response-0.1-micromolar"
class="section level2">
<h2><strong>📌 DOX-only mid-late response (0.1 micromolar)</strong></h2>
<pre class="r"><code># Load libraries
library(dplyr)
library(ggplot2)
library(reshape2)

# Load response group files
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# Print gene list if needed
prob_3_0.1</code></pre>
<pre><code>   [1] &quot;643837&quot;    &quot;5192&quot;      &quot;63976&quot;     &quot;83715&quot;     &quot;8863&quot;      &quot;50651&quot;    
   [7] &quot;80045&quot;     &quot;5293&quot;      &quot;64802&quot;     &quot;10277&quot;     &quot;26232&quot;     &quot;729614&quot;   
  [13] &quot;1188&quot;      &quot;1969&quot;      &quot;11240&quot;     &quot;100861548&quot; &quot;57576&quot;     &quot;400745&quot;   
  [19] &quot;84224&quot;     &quot;5909&quot;      &quot;10236&quot;     &quot;3399&quot;      &quot;11313&quot;     &quot;2582&quot;     
  [25] &quot;3155&quot;      &quot;57822&quot;     &quot;57185&quot;     &quot;164091&quot;    &quot;149420&quot;    &quot;64793&quot;    
  [31] &quot;6195&quot;      &quot;55650&quot;     &quot;84065&quot;     &quot;84958&quot;     &quot;9064&quot;      &quot;2537&quot;     
  [37] &quot;6118&quot;      &quot;8420&quot;      &quot;9672&quot;      &quot;9410&quot;      &quot;79140&quot;     &quot;339483&quot;   
  [43] &quot;3065&quot;      &quot;81493&quot;     &quot;8565&quot;      &quot;252995&quot;    &quot;127544&quot;    &quot;6421&quot;     
  [49] &quot;5453&quot;      &quot;353500&quot;    &quot;10465&quot;     &quot;149465&quot;    &quot;55929&quot;     &quot;1263&quot;     
  [55] &quot;343521&quot;    &quot;149478&quot;    &quot;8643&quot;      &quot;10327&quot;     &quot;51249&quot;     &quot;10103&quot;    
  [61] &quot;55624&quot;     &quot;22996&quot;     &quot;5865&quot;      &quot;9372&quot;      &quot;200014&quot;    &quot;1376&quot;     
  [67] &quot;606495&quot;    &quot;23648&quot;     &quot;100507634&quot; &quot;163782&quot;    &quot;23421&quot;     &quot;84455&quot;    
  [73] &quot;1647&quot;      &quot;100526835&quot; &quot;8790&quot;      &quot;10026&quot;     &quot;55283&quot;     &quot;126820&quot;   
  [79] &quot;105378828&quot; &quot;7049&quot;      &quot;253152&quot;    &quot;11146&quot;     &quot;50999&quot;     &quot;100131564&quot;
  [85] &quot;178&quot;       &quot;7412&quot;      &quot;2135&quot;      &quot;101928370&quot; &quot;1435&quot;      &quot;10390&quot;    
  [91] &quot;100129269&quot; &quot;11218&quot;     &quot;389&quot;       &quot;333926&quot;    &quot;4803&quot;      &quot;84852&quot;    
  [97] &quot;9398&quot;      &quot;54855&quot;     &quot;26227&quot;     &quot;5565&quot;      &quot;440689&quot;    &quot;2209&quot;     
 [103] &quot;105371405&quot; &quot;8337&quot;      &quot;337873&quot;    &quot;723790&quot;    &quot;554313&quot;    &quot;8349&quot;     
 [109] &quot;9900&quot;      &quot;1893&quot;      &quot;56882&quot;     &quot;117145&quot;    &quot;148327&quot;    &quot;3782&quot;     
 [115] &quot;55974&quot;     &quot;23623&quot;     &quot;7818&quot;      &quot;4000&quot;      &quot;64218&quot;     &quot;79957&quot;    
 [121] &quot;63827&quot;     &quot;51093&quot;     &quot;57549&quot;     &quot;23385&quot;     &quot;50848&quot;     &quot;400793&quot;   
 [127] &quot;5087&quot;      &quot;92241&quot;     &quot;92346&quot;     &quot;100302401&quot; &quot;26092&quot;     &quot;148756&quot;   
 [133] &quot;10228&quot;     &quot;51278&quot;     &quot;23127&quot;     &quot;779&quot;       &quot;23612&quot;     &quot;6051&quot;     
 [139] &quot;4660&quot;      &quot;59349&quot;     &quot;7832&quot;      &quot;6635&quot;      &quot;55224&quot;     &quot;9911&quot;     
 [145] &quot;81788&quot;     &quot;5129&quot;      &quot;85414&quot;     &quot;1604&quot;      &quot;27042&quot;     &quot;574036&quot;   
 [151] &quot;55733&quot;     &quot;25896&quot;     &quot;5525&quot;      &quot;467&quot;       &quot;56950&quot;     &quot;7042&quot;     
 [157] &quot;54996&quot;     &quot;11221&quot;     &quot;10637&quot;     &quot;7044&quot;      &quot;286826&quot;    &quot;142&quot;      
 [163] &quot;339500&quot;    &quot;7483&quot;      &quot;126731&quot;    &quot;57568&quot;     &quot;80003&quot;     &quot;359948&quot;   
 [169] &quot;7107&quot;      &quot;6262&quot;      &quot;55083&quot;     &quot;84838&quot;     &quot;79754&quot;     &quot;80760&quot;    
 [175] &quot;254427&quot;    &quot;11164&quot;     &quot;83643&quot;     &quot;22929&quot;     &quot;338596&quot;    &quot;387640&quot;   
 [181] &quot;643236&quot;    &quot;9721&quot;      &quot;143162&quot;    &quot;259230&quot;    &quot;8030&quot;      &quot;9886&quot;     
 [187] &quot;1959&quot;      &quot;84989&quot;     &quot;29119&quot;     &quot;219743&quot;    &quot;27143&quot;     &quot;5092&quot;     
 [193] &quot;5532&quot;      &quot;58529&quot;     &quot;8509&quot;      &quot;5328&quot;      &quot;219654&quot;    &quot;9060&quot;     
 [199] &quot;100144748&quot; &quot;355&quot;       &quot;3434&quot;      &quot;51196&quot;     &quot;953&quot;       &quot;10023&quot;    
 [205] &quot;80019&quot;     &quot;6425&quot;      &quot;55118&quot;     &quot;84171&quot;     &quot;282991&quot;    &quot;81621&quot;    
 [211] &quot;27343&quot;     &quot;9221&quot;      &quot;84108&quot;     &quot;9446&quot;      &quot;120&quot;       &quot;9126&quot;     
 [217] &quot;9937&quot;      &quot;259217&quot;    &quot;196051&quot;    &quot;54780&quot;     &quot;80007&quot;     &quot;9679&quot;     
 [223] &quot;399818&quot;    &quot;642938&quot;    &quot;80313&quot;     &quot;196743&quot;    &quot;60626&quot;     &quot;3265&quot;     
 [229] &quot;64787&quot;     &quot;55367&quot;     &quot;402778&quot;    &quot;3481&quot;      &quot;3784&quot;      &quot;4928&quot;     
 [235] &quot;6786&quot;      &quot;10612&quot;     &quot;6881&quot;      &quot;1200&quot;      &quot;143425&quot;    &quot;56672&quot;    
 [241] &quot;56675&quot;     &quot;23258&quot;     &quot;81846&quot;     &quot;406&quot;       &quot;84280&quot;     &quot;84188&quot;    
 [247] &quot;8048&quot;      &quot;100126784&quot; &quot;10553&quot;     &quot;55366&quot;     &quot;196294&quot;    &quot;966&quot;      
 [253] &quot;6506&quot;      &quot;9537&quot;      &quot;56981&quot;     &quot;9409&quot;      &quot;85456&quot;     &quot;6749&quot;     
 [259] &quot;219539&quot;    &quot;25921&quot;     &quot;9404&quot;      &quot;79073&quot;     &quot;220001&quot;    &quot;9066&quot;     
 [265] &quot;6094&quot;      &quot;374395&quot;    &quot;283248&quot;    &quot;79829&quot;     &quot;83707&quot;     &quot;7536&quot;     
 [271] &quot;55561&quot;     &quot;29901&quot;     &quot;7108&quot;      &quot;741&quot;       &quot;823&quot;       &quot;83786&quot;    
 [277] &quot;101927789&quot; &quot;91056&quot;     &quot;11007&quot;     &quot;8061&quot;      &quot;83638&quot;     &quot;55690&quot;    
 [283] &quot;64837&quot;     &quot;9973&quot;      &quot;57804&quot;     &quot;57571&quot;     &quot;5790&quot;      &quot;80194&quot;    
 [289] &quot;9600&quot;      &quot;10263&quot;     &quot;2950&quot;      &quot;55191&quot;     &quot;3636&quot;      &quot;10809&quot;    
 [295] &quot;58473&quot;     &quot;100287896&quot; &quot;100507050&quot; &quot;408&quot;       &quot;5612&quot;      &quot;100506127&quot;
 [301] &quot;25987&quot;     &quot;60492&quot;     &quot;10873&quot;     &quot;11098&quot;     &quot;1075&quot;      &quot;85459&quot;    
 [307] &quot;4361&quot;      &quot;53942&quot;     &quot;114908&quot;    &quot;2977&quot;      &quot;1662&quot;      &quot;23235&quot;    
 [313] &quot;54970&quot;     &quot;57646&quot;     &quot;25996&quot;     &quot;51092&quot;     &quot;100652768&quot; &quot;23621&quot;    
 [319] &quot;22897&quot;     &quot;53826&quot;     &quot;6330&quot;      &quot;55823&quot;     &quot;64137&quot;     &quot;79671&quot;    
 [325] &quot;4162&quot;      &quot;9638&quot;      &quot;9538&quot;      &quot;219844&quot;    &quot;50937&quot;     &quot;283174&quot;   
 [331] &quot;196513&quot;    &quot;57103&quot;     &quot;1108&quot;      &quot;9746&quot;      &quot;374443&quot;    &quot;4973&quot;     
 [337] &quot;8531&quot;      &quot;55766&quot;     &quot;51768&quot;     &quot;65981&quot;     &quot;1663&quot;      &quot;84920&quot;    
 [343] &quot;55605&quot;     &quot;283463&quot;    &quot;196528&quot;    &quot;54407&quot;     &quot;100288798&quot; &quot;41&quot;       
 [349] &quot;81566&quot;     &quot;23371&quot;     &quot;84926&quot;     &quot;7786&quot;      &quot;57658&quot;     &quot;25946&quot;    
 [355] &quot;5502&quot;      &quot;29095&quot;     &quot;1606&quot;      &quot;64375&quot;     &quot;2065&quot;      &quot;10193&quot;    
 [361] &quot;27165&quot;     &quot;9880&quot;      &quot;79837&quot;     &quot;10956&quot;     &quot;1019&quot;      &quot;9194&quot;     
 [367] &quot;29110&quot;     &quot;1368&quot;      &quot;144453&quot;    &quot;27345&quot;     &quot;5787&quot;      &quot;8549&quot;     
 [373] &quot;22822&quot;     &quot;79738&quot;     &quot;144455&quot;    &quot;160335&quot;    &quot;25834&quot;     &quot;4254&quot;     
 [379] &quot;1848&quot;      &quot;694&quot;       &quot;51134&quot;     &quot;121457&quot;    &quot;121601&quot;    &quot;55332&quot;    
 [385] &quot;51559&quot;     &quot;121053&quot;    &quot;121551&quot;    &quot;23479&quot;     &quot;121642&quot;    &quot;89910&quot;    
 [391] &quot;4598&quot;      &quot;51228&quot;     &quot;84260&quot;     &quot;400073&quot;    &quot;28981&quot;     &quot;5501&quot;     
 [397] &quot;113675&quot;    &quot;160777&quot;    &quot;5564&quot;      &quot;51499&quot;     &quot;5025&quot;      &quot;144404&quot;   
 [403] &quot;144406&quot;    &quot;10198&quot;     &quot;5901&quot;      &quot;283383&quot;    &quot;100996573&quot; &quot;5827&quot;     
 [409] &quot;7637&quot;      &quot;51761&quot;     &quot;387496&quot;    &quot;432369&quot;    &quot;23281&quot;     &quot;9201&quot;     
 [415] &quot;5994&quot;      &quot;55068&quot;     &quot;81617&quot;     &quot;79621&quot;     &quot;220108&quot;    &quot;100507398&quot;
 [421] &quot;341676&quot;    &quot;55901&quot;     &quot;26586&quot;     &quot;5100&quot;      &quot;688&quot;       &quot;79596&quot;    
 [427] &quot;10082&quot;     &quot;160897&quot;    &quot;100861541&quot; &quot;100289373&quot; &quot;8660&quot;      &quot;283487&quot;   
 [433] &quot;23250&quot;     &quot;3916&quot;      &quot;113622&quot;    &quot;2621&quot;      &quot;10038&quot;     &quot;7011&quot;     
 [439] &quot;63874&quot;     &quot;26020&quot;     &quot;54930&quot;     &quot;4625&quot;      &quot;161247&quot;    &quot;5720&quot;     
 [445] &quot;51292&quot;     &quot;5875&quot;      &quot;4857&quot;      &quot;11154&quot;     &quot;9472&quot;      &quot;64067&quot;    
 [451] &quot;8487&quot;      &quot;122525&quot;    &quot;57697&quot;     &quot;122769&quot;    &quot;79944&quot;     &quot;5836&quot;     
 [457] &quot;114088&quot;    &quot;6815&quot;      &quot;9786&quot;      &quot;6252&quot;      &quot;23224&quot;     &quot;7597&quot;     
 [463] &quot;57452&quot;     &quot;2079&quot;      &quot;6547&quot;      &quot;26037&quot;     &quot;53349&quot;     &quot;91748&quot;    
 [469] &quot;23093&quot;     &quot;7043&quot;      &quot;2954&quot;      &quot;85439&quot;     &quot;55812&quot;     &quot;11099&quot;    
 [475] &quot;79882&quot;     &quot;400236&quot;    &quot;78990&quot;     &quot;122509&quot;    &quot;5891&quot;      &quot;7517&quot;     
 [481] &quot;388021&quot;    &quot;283638&quot;    &quot;647310&quot;    &quot;283643&quot;    &quot;7082&quot;      &quot;342184&quot;   
 [487] &quot;254531&quot;    &quot;23015&quot;     &quot;440270&quot;    &quot;5330&quot;      &quot;729082&quot;    &quot;80237&quot;    
 [493] &quot;50506&quot;     &quot;80031&quot;     &quot;399694&quot;    &quot;23312&quot;     &quot;145788&quot;    &quot;64864&quot;    
 [499] &quot;54629&quot;     &quot;83660&quot;     &quot;51065&quot;     &quot;83464&quot;     &quot;53944&quot;     &quot;54956&quot;    
 [505] &quot;51495&quot;     &quot;22801&quot;     &quot;8125&quot;      &quot;7090&quot;      &quot;60677&quot;     &quot;3073&quot;     
 [511] &quot;5371&quot;      &quot;440288&quot;    &quot;1464&quot;      &quot;26263&quot;     &quot;10099&quot;     &quot;55466&quot;    
 [517] &quot;11173&quot;     &quot;5923&quot;      &quot;283687&quot;    &quot;57214&quot;     &quot;53346&quot;     &quot;6457&quot;     
 [523] &quot;9154&quot;      &quot;5151&quot;      &quot;11214&quot;     &quot;3669&quot;      &quot;145864&quot;    &quot;4240&quot;     
 [529] &quot;342132&quot;    &quot;64784&quot;     &quot;26276&quot;     &quot;28232&quot;     &quot;56963&quot;     &quot;23336&quot;    
 [535] &quot;170691&quot;    &quot;22856&quot;     &quot;6627&quot;      &quot;51728&quot;     &quot;79622&quot;     &quot;64285&quot;    
 [541] &quot;4833&quot;      &quot;84572&quot;     &quot;283871&quot;    &quot;80178&quot;     &quot;83886&quot;     &quot;84256&quot;    
 [547] &quot;54985&quot;     &quot;84464&quot;     &quot;10131&quot;     &quot;79585&quot;     &quot;29965&quot;     &quot;23295&quot;    
 [553] &quot;79641&quot;     &quot;2013&quot;      &quot;8651&quot;      &quot;9516&quot;      &quot;729993&quot;    &quot;89927&quot;    
 [559] &quot;368&quot;       &quot;64131&quot;     &quot;653190&quot;    &quot;79905&quot;     &quot;51760&quot;     &quot;123879&quot;   
 [565] &quot;57149&quot;     &quot;3566&quot;      &quot;728734&quot;    &quot;100507607&quot; &quot;6693&quot;      &quot;4150&quot;     
 [571] &quot;253980&quot;    &quot;83719&quot;     &quot;79077&quot;     &quot;283932&quot;    &quot;93129&quot;     &quot;2521&quot;     
 [577] &quot;440359&quot;    &quot;23090&quot;     &quot;23322&quot;     &quot;4502&quot;      &quot;4493&quot;      &quot;9688&quot;     
 [583] &quot;23568&quot;     &quot;1006&quot;      &quot;51192&quot;     &quot;8883&quot;      &quot;8824&quot;      &quot;23491&quot;    
 [589] &quot;283849&quot;    &quot;29800&quot;     &quot;1506&quot;      &quot;3931&quot;      &quot;146198&quot;    &quot;3038&quot;     
 [595] &quot;16&quot;        &quot;23450&quot;     &quot;794&quot;       &quot;55565&quot;     &quot;79583&quot;     &quot;100130958&quot;
 [601] &quot;51741&quot;     &quot;4094&quot;      &quot;123775&quot;    &quot;9100&quot;      &quot;23199&quot;     &quot;2300&quot;     
 [607] &quot;333929&quot;    &quot;124045&quot;    &quot;4157&quot;      &quot;79007&quot;     &quot;51763&quot;     &quot;23293&quot;    
 [613] &quot;23108&quot;     &quot;162514&quot;    &quot;7442&quot;      &quot;23729&quot;     &quot;3682&quot;      &quot;101928000&quot;
 [619] &quot;708&quot;       &quot;728392&quot;    &quot;432&quot;       &quot;57659&quot;     &quot;5430&quot;      &quot;968&quot;      
 [625] &quot;6462&quot;      &quot;482&quot;       &quot;55135&quot;     &quot;1949&quot;      &quot;92162&quot;     &quot;5198&quot;     
 [631] &quot;388335&quot;    &quot;201164&quot;    &quot;201163&quot;    &quot;91647&quot;     &quot;7156&quot;      &quot;5636&quot;     
 [637] &quot;125206&quot;    &quot;644815&quot;    &quot;4239&quot;      &quot;26118&quot;     &quot;8844&quot;      &quot;7126&quot;     
 [643] &quot;147007&quot;    &quot;116238&quot;    &quot;284086&quot;    &quot;9618&quot;      &quot;147015&quot;    &quot;116236&quot;   
 [649] &quot;84940&quot;     &quot;374786&quot;    &quot;105371730&quot; &quot;5892&quot;      &quot;31&quot;        &quot;57636&quot;    
 [655] &quot;2064&quot;      &quot;84299&quot;     &quot;7067&quot;      &quot;125111&quot;    &quot;3880&quot;      &quot;9001&quot;     
 [661] &quot;535&quot;       &quot;79990&quot;     &quot;84313&quot;     &quot;339201&quot;    &quot;2896&quot;      &quot;284071&quot;   
 [667] &quot;124790&quot;    &quot;9842&quot;      &quot;4137&quot;      &quot;996&quot;       &quot;3837&quot;      &quot;80279&quot;    
 [673] &quot;10241&quot;     &quot;65264&quot;     &quot;10642&quot;     &quot;81558&quot;     &quot;11143&quot;     &quot;284080&quot;   
 [679] &quot;3675&quot;      &quot;4831&quot;      &quot;8165&quot;      &quot;6426&quot;      &quot;54894&quot;     &quot;56155&quot;    
 [685] &quot;5889&quot;      &quot;81671&quot;     &quot;8493&quot;      &quot;388407&quot;    &quot;64750&quot;     &quot;146880&quot;   
 [691] &quot;440456&quot;    &quot;201283&quot;    &quot;201134&quot;    &quot;23461&quot;     &quot;5608&quot;      &quot;400617&quot;   
 [697] &quot;6662&quot;      &quot;94015&quot;     &quot;9121&quot;      &quot;100287042&quot; &quot;3021&quot;      &quot;8811&quot;     
 [703] &quot;23265&quot;     &quot;114804&quot;    &quot;8877&quot;      &quot;100507246&quot; &quot;6427&quot;      &quot;2548&quot;     
 [709] &quot;79092&quot;     &quot;284129&quot;    &quot;440465&quot;    &quot;284184&quot;    &quot;25794&quot;     &quot;1468&quot;     
 [715] &quot;4097&quot;      &quot;92659&quot;     &quot;255275&quot;    &quot;201254&quot;    &quot;201255&quot;    &quot;23347&quot;    
 [721] &quot;103910&quot;    &quot;649446&quot;    &quot;84777&quot;     &quot;84617&quot;     &quot;10939&quot;     &quot;55125&quot;    
 [727] &quot;753&quot;       &quot;26256&quot;     &quot;25925&quot;     &quot;284252&quot;    &quot;51444&quot;     &quot;80206&quot;    
 [733] &quot;57536&quot;     &quot;9050&quot;      &quot;201501&quot;    &quot;4645&quot;      &quot;80323&quot;     &quot;5366&quot;     
 [739] &quot;8710&quot;      &quot;5271&quot;      &quot;92126&quot;     &quot;25914&quot;     &quot;101927989&quot; &quot;440498&quot;   
 [745] &quot;84552&quot;     &quot;610&quot;       &quot;22904&quot;     &quot;4946&quot;      &quot;7064&quot;      &quot;4782&quot;     
 [751] &quot;729359&quot;    &quot;4902&quot;      &quot;6813&quot;      &quot;388499&quot;    &quot;4701&quot;      &quot;256949&quot;   
 [757] &quot;4670&quot;      &quot;374879&quot;    &quot;93134&quot;     &quot;93145&quot;     &quot;3383&quot;      &quot;3386&quot;     
 [763] &quot;5141&quot;      &quot;3609&quot;      &quot;126075&quot;    &quot;57139&quot;     &quot;147837&quot;    &quot;4125&quot;     
 [769] &quot;199699&quot;    &quot;9592&quot;      &quot;65249&quot;     &quot;9524&quot;      &quot;84449&quot;     &quot;376497&quot;   
 [775] &quot;5296&quot;      &quot;5143&quot;      &quot;9518&quot;      &quot;51477&quot;     &quot;10715&quot;     &quot;729991&quot;   
 [781] &quot;284440&quot;    &quot;7639&quot;      &quot;171392&quot;    &quot;79156&quot;     &quot;898&quot;       &quot;5348&quot;     
 [787] &quot;100128682&quot; &quot;115703&quot;    &quot;333&quot;       &quot;84911&quot;     &quot;374899&quot;    &quot;101927720&quot;
 [793] &quot;22835&quot;     &quot;57828&quot;     &quot;399473&quot;    &quot;147965&quot;    &quot;126432&quot;    &quot;115290&quot;   
 [799] &quot;57622&quot;     &quot;4294&quot;      &quot;126526&quot;    &quot;29950&quot;     &quot;645&quot;       &quot;57731&quot;    
 [805] &quot;80271&quot;     &quot;6626&quot;      &quot;7040&quot;      &quot;641649&quot;    &quot;6223&quot;      &quot;10567&quot;    
 [811] &quot;478&quot;       &quot;5452&quot;      &quot;3991&quot;      &quot;653583&quot;    &quot;56006&quot;     &quot;7596&quot;     
 [817] &quot;56971&quot;     &quot;79090&quot;     &quot;8189&quot;      &quot;808&quot;       &quot;147906&quot;    &quot;100506068&quot;
 [823] &quot;29888&quot;     &quot;10055&quot;     &quot;22900&quot;     &quot;9266&quot;      &quot;770&quot;       &quot;23645&quot;    
 [829] &quot;581&quot;       &quot;10856&quot;     &quot;8463&quot;      &quot;83596&quot;     &quot;23636&quot;     &quot;25888&quot;    
 [835] &quot;94030&quot;     &quot;79986&quot;     &quot;126017&quot;    &quot;55422&quot;     &quot;147807&quot;    &quot;284307&quot;   
 [841] &quot;348327&quot;    &quot;284309&quot;    &quot;10782&quot;     &quot;100131691&quot; &quot;7593&quot;      &quot;9687&quot;     
 [847] &quot;6382&quot;      &quot;64342&quot;     &quot;9540&quot;      &quot;391356&quot;    &quot;10890&quot;     &quot;60509&quot;    
 [853] &quot;3795&quot;      &quot;7349&quot;      &quot;4358&quot;      &quot;9913&quot;      &quot;2355&quot;      &quot;253558&quot;   
 [859] &quot;440854&quot;    &quot;130589&quot;    &quot;728730&quot;    &quot;6546&quot;      &quot;79823&quot;     &quot;2956&quot;     
 [865] &quot;129285&quot;    &quot;98&quot;        &quot;100129434&quot; &quot;53335&quot;     &quot;23301&quot;     &quot;119&quot;      
 [871] &quot;94097&quot;     &quot;150726&quot;    &quot;7840&quot;      &quot;51002&quot;     &quot;116540&quot;    &quot;84695&quot;    
 [877] &quot;10618&quot;     &quot;54884&quot;     &quot;112597&quot;    &quot;150572&quot;    &quot;30818&quot;     &quot;56910&quot;    
 [883] &quot;55654&quot;     &quot;10865&quot;     &quot;10120&quot;     &quot;2274&quot;      &quot;165055&quot;    &quot;96626&quot;    
 [889] &quot;64682&quot;     &quot;10461&quot;     &quot;84910&quot;     &quot;79134&quot;     &quot;91409&quot;     &quot;90557&quot;    
 [895] &quot;116372&quot;    &quot;4249&quot;      &quot;100129961&quot; &quot;84083&quot;     &quot;23518&quot;     &quot;11249&quot;    
 [901] &quot;390&quot;       &quot;55183&quot;     &quot;114793&quot;    &quot;11227&quot;     &quot;22925&quot;     &quot;55137&quot;    
 [907] &quot;22837&quot;     &quot;6326&quot;      &quot;2571&quot;      &quot;9874&quot;      &quot;8520&quot;      &quot;29789&quot;    
 [913] &quot;1123&quot;      &quot;51661&quot;     &quot;64172&quot;     &quot;5378&quot;      &quot;3628&quot;      &quot;54842&quot;    
 [919] &quot;4430&quot;      &quot;64859&quot;     &quot;91526&quot;     &quot;3329&quot;      &quot;3336&quot;      &quot;23314&quot;    
 [925] &quot;60491&quot;     &quot;7855&quot;      &quot;3417&quot;      &quot;4133&quot;      &quot;33&quot;        &quot;1373&quot;     
 [931] &quot;2066&quot;      &quot;580&quot;       &quot;26154&quot;     &quot;55825&quot;     &quot;64114&quot;     &quot;64320&quot;    
 [937] &quot;10058&quot;     &quot;1674&quot;      &quot;79586&quot;     &quot;130367&quot;    &quot;130340&quot;    &quot;57590&quot;    
 [943] &quot;5270&quot;      &quot;80309&quot;     &quot;80210&quot;     &quot;10123&quot;     &quot;64284&quot;     &quot;51665&quot;    
 [949] &quot;2859&quot;      &quot;140809&quot;    &quot;64773&quot;     &quot;3704&quot;      &quot;83959&quot;     &quot;146&quot;      
 [955] &quot;149840&quot;    &quot;63926&quot;     &quot;6616&quot;      &quot;182&quot;       &quot;55304&quot;     &quot;1471&quot;     
 [961] &quot;26152&quot;     &quot;11264&quot;     &quot;2937&quot;      &quot;25939&quot;     &quot;10904&quot;     &quot;9935&quot;     
 [967] &quot;128486&quot;    &quot;11142&quot;     &quot;10953&quot;     &quot;27296&quot;     &quot;51604&quot;     &quot;8202&quot;     
 [973] &quot;55959&quot;     &quot;23394&quot;     &quot;8675&quot;      &quot;2778&quot;      &quot;1522&quot;      &quot;81928&quot;    
 [979] &quot;55738&quot;     &quot;102723996&quot; &quot;10950&quot;     &quot;54149&quot;     &quot;29104&quot;     &quot;30811&quot;    
 [985] &quot;2618&quot;      &quot;29980&quot;     &quot;1827&quot;      &quot;861&quot;       &quot;9980&quot;      &quot;6450&quot;     
 [991] &quot;54020&quot;     &quot;5316&quot;      &quot;23308&quot;     &quot;54082&quot;     &quot;728039&quot;    &quot;5116&quot;     
 [997] &quot;83733&quot;     &quot;529&quot;       &quot;5625&quot;      &quot;25786&quot;     &quot;439931&quot;    &quot;23331&quot;    
[1003] &quot;25807&quot;     &quot;3976&quot;      &quot;6948&quot;      &quot;23762&quot;     &quot;3985&quot;      &quot;6523&quot;     
[1009] &quot;10042&quot;     &quot;23780&quot;     &quot;80020&quot;     &quot;7263&quot;      &quot;29775&quot;     &quot;85377&quot;    
[1015] &quot;84645&quot;     &quot;23764&quot;     &quot;102800317&quot; &quot;11144&quot;     &quot;27350&quot;     &quot;5155&quot;     
[1021] &quot;79640&quot;     &quot;10762&quot;     &quot;5465&quot;      &quot;55020&quot;     &quot;5600&quot;      &quot;23654&quot;    
[1027] &quot;410&quot;       &quot;284942&quot;    &quot;11158&quot;     &quot;859&quot;       &quot;9901&quot;      &quot;55209&quot;    
[1033] &quot;80746&quot;     &quot;55287&quot;     &quot;9922&quot;      &quot;7476&quot;      &quot;7508&quot;      &quot;6533&quot;     
[1039] &quot;80852&quot;     &quot;7155&quot;      &quot;10217&quot;     &quot;5333&quot;      &quot;9942&quot;      &quot;9941&quot;     
[1045] &quot;6331&quot;      &quot;199223&quot;    &quot;64651&quot;     &quot;25907&quot;     &quot;79442&quot;     &quot;4634&quot;     
[1051] &quot;5745&quot;      &quot;54859&quot;     &quot;11277&quot;     &quot;54681&quot;     &quot;2876&quot;      &quot;1605&quot;     
[1057] &quot;8927&quot;      &quot;7318&quot;      &quot;10991&quot;     &quot;7866&quot;      &quot;1795&quot;      &quot;57060&quot;    
[1063] &quot;5580&quot;      &quot;7086&quot;      &quot;8309&quot;      &quot;56999&quot;     &quot;9223&quot;      &quot;84541&quot;    
[1069] &quot;27086&quot;     &quot;1371&quot;      &quot;84319&quot;     &quot;344595&quot;    &quot;90102&quot;     &quot;79691&quot;    
[1075] &quot;116064&quot;    &quot;3836&quot;      &quot;64770&quot;     &quot;84303&quot;     &quot;57501&quot;     &quot;339942&quot;   
[1081] &quot;100507032&quot; &quot;729375&quot;    &quot;339855&quot;    &quot;10274&quot;     &quot;92369&quot;     &quot;9435&quot;     
[1087] &quot;6596&quot;      &quot;53836&quot;     &quot;5912&quot;      &quot;79674&quot;     &quot;51319&quot;     &quot;79740&quot;    
[1093] &quot;5274&quot;      &quot;5584&quot;      &quot;200916&quot;    &quot;23043&quot;     &quot;5337&quot;      &quot;64393&quot;    
[1099] &quot;51193&quot;     &quot;59345&quot;     &quot;5437&quot;      &quot;6434&quot;      &quot;2119&quot;      &quot;9076&quot;     
[1105] &quot;2257&quot;      &quot;401106&quot;    &quot;93109&quot;     &quot;131583&quot;    &quot;152687&quot;    &quot;54872&quot;    
[1111] &quot;4636&quot;      &quot;84179&quot;     &quot;317648&quot;    &quot;285489&quot;    &quot;4043&quot;      &quot;5522&quot;     
[1117] &quot;23324&quot;     &quot;768239&quot;    &quot;56606&quot;     &quot;5860&quot;      &quot;23231&quot;     &quot;5981&quot;     
[1123] &quot;23244&quot;     &quot;386617&quot;    &quot;10699&quot;     &quot;3815&quot;      &quot;5471&quot;      &quot;10606&quot;    
[1129] &quot;23284&quot;     &quot;1060&quot;      &quot;8671&quot;      &quot;374&quot;       &quot;3184&quot;      &quot;22872&quot;    
[1135] &quot;51191&quot;     &quot;10144&quot;     &quot;285513&quot;    &quot;23173&quot;     &quot;100507053&quot; &quot;285556&quot;   
[1141] &quot;115265&quot;    &quot;90024&quot;     &quot;79807&quot;     &quot;79071&quot;     &quot;92610&quot;     &quot;287&quot;      
[1147] &quot;8492&quot;      &quot;79625&quot;     &quot;5393&quot;      &quot;166378&quot;    &quot;22824&quot;     &quot;25819&quot;    
[1153] &quot;84709&quot;     &quot;166785&quot;    &quot;1909&quot;      &quot;987&quot;       &quot;4886&quot;      &quot;1363&quot;     
[1159] &quot;23022&quot;     &quot;57630&quot;     &quot;116966&quot;    &quot;201973&quot;    &quot;256309&quot;    &quot;27295&quot;    
[1165] &quot;10016&quot;     &quot;11076&quot;     &quot;85409&quot;     &quot;134147&quot;    &quot;1767&quot;      &quot;643401&quot;   
[1171] &quot;23037&quot;     &quot;81792&quot;     &quot;51151&quot;     &quot;79925&quot;     &quot;2668&quot;      &quot;345557&quot;   
[1177] &quot;6414&quot;      &quot;23530&quot;     &quot;53918&quot;     &quot;4338&quot;      &quot;112574&quot;    &quot;493869&quot;   
[1183] &quot;345651&quot;    &quot;10769&quot;     &quot;285671&quot;    &quot;11174&quot;     &quot;64087&quot;     &quot;84327&quot;    
[1189] &quot;345778&quot;    &quot;100505994&quot; &quot;84250&quot;     &quot;83890&quot;     &quot;22836&quot;     &quot;90355&quot;    
[1195] &quot;64839&quot;     &quot;9315&quot;      &quot;28966&quot;     &quot;57507&quot;     &quot;501&quot;       &quot;115123&quot;   
[1201] &quot;84466&quot;     &quot;2201&quot;      &quot;51015&quot;     &quot;3094&quot;      &quot;51735&quot;     &quot;6583&quot;     
[1207] &quot;23176&quot;     &quot;134549&quot;    &quot;51780&quot;     &quot;51308&quot;     &quot;202051&quot;    &quot;202052&quot;   
[1213] &quot;340061&quot;    &quot;51523&quot;     &quot;404734&quot;    &quot;10011&quot;     &quot;285613&quot;    &quot;10007&quot;    
[1219] &quot;81848&quot;     &quot;2246&quot;      &quot;10915&quot;     &quot;389336&quot;    &quot;79628&quot;     &quot;78991&quot;    
[1225] &quot;22993&quot;     &quot;6949&quot;      &quot;26112&quot;     &quot;206358&quot;    &quot;2890&quot;      &quot;134353&quot;   
[1231] &quot;6586&quot;      &quot;64901&quot;     &quot;100268168&quot; &quot;4488&quot;      &quot;3274&quot;      &quot;2870&quot;     
[1237] &quot;11285&quot;     &quot;85007&quot;     &quot;8878&quot;      &quot;94234&quot;     &quot;2296&quot;      &quot;648791&quot;   
[1243] &quot;2651&quot;      &quot;51522&quot;     &quot;221687&quot;    &quot;6310&quot;      &quot;401236&quot;    &quot;10486&quot;    
[1249] &quot;57380&quot;     &quot;81688&quot;     &quot;3006&quot;      &quot;8347&quot;      &quot;8334&quot;      &quot;3017&quot;     
[1255] &quot;8367&quot;      &quot;3012&quot;      &quot;8365&quot;      &quot;493812&quot;    &quot;8970&quot;      &quot;85236&quot;    
[1261] &quot;64288&quot;     &quot;114821&quot;    &quot;3133&quot;      &quot;8870&quot;      &quot;780&quot;       &quot;1460&quot;     
[1267] &quot;57819&quot;     &quot;4758&quot;      &quot;221527&quot;    &quot;6890&quot;      &quot;1616&quot;      &quot;578&quot;      
[1273] &quot;221491&quot;    &quot;5603&quot;      &quot;27154&quot;     &quot;6428&quot;      &quot;105375039&quot; &quot;5292&quot;     
[1279] &quot;221468&quot;    &quot;9025&quot;      &quot;55809&quot;     &quot;441150&quot;    &quot;116138&quot;    &quot;65989&quot;    
[1285] &quot;57510&quot;     &quot;5429&quot;      &quot;55362&quot;     &quot;2030&quot;      &quot;4794&quot;      &quot;202500&quot;   
[1291] &quot;10231&quot;     &quot;51302&quot;     &quot;27242&quot;     &quot;442213&quot;    &quot;85315&quot;     &quot;90523&quot;    
[1297] &quot;51557&quot;     &quot;7803&quot;      &quot;26503&quot;     &quot;135228&quot;    &quot;27145&quot;     &quot;4646&quot;     
[1303] &quot;55023&quot;     &quot;9324&quot;      &quot;7162&quot;      &quot;22832&quot;     &quot;2570&quot;      &quot;58528&quot;    
[1309] &quot;79694&quot;     &quot;10973&quot;     &quot;64208&quot;     &quot;57673&quot;     &quot;55084&quot;     &quot;11231&quot;    
[1315] &quot;27244&quot;     &quot;285753&quot;    &quot;9896&quot;      &quot;101927768&quot; &quot;222537&quot;    &quot;23493&quot;    
[1321] &quot;81847&quot;     &quot;5796&quot;      &quot;116843&quot;    &quot;4217&quot;      &quot;57224&quot;     &quot;441172&quot;   
[1327] &quot;51390&quot;     &quot;348995&quot;    &quot;80329&quot;     &quot;9590&quot;      &quot;26034&quot;     &quot;8871&quot;     
[1333] &quot;39&quot;        &quot;5071&quot;      &quot;9444&quot;      &quot;2852&quot;      &quot;79778&quot;     &quot;114796&quot;   
[1339] &quot;29886&quot;     &quot;23288&quot;     &quot;55698&quot;     &quot;80028&quot;     &quot;221955&quot;    &quot;54753&quot;    
[1345] &quot;6119&quot;      &quot;9678&quot;      &quot;55536&quot;     &quot;10457&quot;     &quot;1687&quot;      &quot;26031&quot;    
[1351] &quot;3181&quot;      &quot;11335&quot;     &quot;221895&quot;    &quot;79017&quot;     &quot;5137&quot;      &quot;51251&quot;    
[1357] &quot;57057&quot;     &quot;9844&quot;      &quot;349114&quot;    &quot;136647&quot;    &quot;79783&quot;     &quot;285954&quot;   
[1363] &quot;9263&quot;      &quot;58498&quot;     &quot;94239&quot;     &quot;1956&quot;      &quot;81552&quot;     &quot;51427&quot;    
[1369] &quot;168374&quot;    &quot;154881&quot;    &quot;378108&quot;    &quot;100101148&quot; &quot;9031&quot;      &quot;541473&quot;   
[1375] &quot;375593&quot;    &quot;3092&quot;      &quot;222183&quot;    &quot;5218&quot;      &quot;1595&quot;      &quot;64921&quot;    
[1381] &quot;55971&quot;     &quot;10898&quot;     &quot;255374&quot;    &quot;55262&quot;     &quot;79690&quot;     &quot;402682&quot;   
[1387] &quot;43&quot;        &quot;1174&quot;      &quot;8985&quot;      &quot;222236&quot;    &quot;55973&quot;     &quot;4897&quot;     
[1393] &quot;858&quot;       &quot;93653&quot;     &quot;7982&quot;      &quot;401399&quot;    &quot;4899&quot;      &quot;51200&quot;    
[1399] &quot;91584&quot;     &quot;29062&quot;     &quot;100507421&quot; &quot;9715&quot;      &quot;285966&quot;    &quot;84626&quot;    
[1405] &quot;155066&quot;    &quot;9311&quot;      &quot;10049&quot;     &quot;55112&quot;     &quot;101927595&quot; &quot;2055&quot;     
[1411] &quot;157285&quot;    &quot;640&quot;       &quot;101929490&quot; &quot;10395&quot;     &quot;51201&quot;     &quot;23362&quot;    
[1417] &quot;55790&quot;     &quot;55806&quot;     &quot;80346&quot;     &quot;9796&quot;      &quot;23516&quot;     &quot;8793&quot;     
[1423] &quot;8797&quot;      &quot;4741&quot;      &quot;23087&quot;     &quot;51435&quot;     &quot;79618&quot;     &quot;1846&quot;     
[1429] &quot;7486&quot;      &quot;3084&quot;      &quot;286&quot;       &quot;5327&quot;      &quot;137695&quot;    &quot;55636&quot;    
[1435] &quot;5150&quot;      &quot;79776&quot;     &quot;5828&quot;      &quot;619279&quot;    &quot;85444&quot;     &quot;4683&quot;     
[1441] &quot;862&quot;       &quot;137392&quot;    &quot;54704&quot;     &quot;94241&quot;     &quot;51001&quot;     &quot;9791&quot;     
[1447] &quot;83988&quot;     &quot;23414&quot;     &quot;137735&quot;    &quot;114788&quot;    &quot;5885&quot;      &quot;90390&quot;    
[1453] &quot;4982&quot;      &quot;4856&quot;      &quot;114907&quot;    &quot;5820&quot;      &quot;114&quot;       &quot;286077&quot;   
[1459] &quot;100128338&quot; &quot;83481&quot;     &quot;84875&quot;     &quot;8733&quot;      &quot;81704&quot;     &quot;5991&quot;     
[1465] &quot;403313&quot;    &quot;4781&quot;      &quot;203238&quot;    &quot;6456&quot;      &quot;54801&quot;     &quot;340485&quot;   
[1471] &quot;4300&quot;      &quot;360&quot;       &quot;441459&quot;    &quot;9853&quot;      &quot;51754&quot;     &quot;64425&quot;    
[1477] &quot;219&quot;       &quot;347252&quot;    &quot;687&quot;       &quot;7091&quot;      &quot;7088&quot;      &quot;80318&quot;    
[1483] &quot;2619&quot;      &quot;79048&quot;     &quot;54769&quot;     &quot;4920&quot;      &quot;4814&quot;      &quot;445577&quot;   
[1489] &quot;23196&quot;     &quot;55357&quot;     &quot;54566&quot;     &quot;114299&quot;    &quot;445815&quot;    &quot;158399&quot;   
[1495] &quot;9991&quot;      &quot;158405&quot;    &quot;169834&quot;    &quot;85301&quot;     &quot;25861&quot;     &quot;23245&quot;    
[1501] &quot;55755&quot;     &quot;286204&quot;    &quot;169611&quot;    &quot;1759&quot;      &quot;4957&quot;      &quot;389792&quot;   
[1507] &quot;100506119&quot; &quot;100506190&quot; &quot;140459&quot;    &quot;83543&quot;     &quot;11092&quot;     &quot;6834&quot;     
[1513] &quot;6256&quot;      &quot;138162&quot;    &quot;4851&quot;      &quot;101928483&quot; &quot;29085&quot;     &quot;158056&quot;   
[1519] &quot;7186&quot;      &quot;100289341&quot; &quot;55344&quot;     &quot;283981&quot;    &quot;3563&quot;      &quot;25878&quot;    
[1525] &quot;5613&quot;      &quot;5277&quot;      &quot;22866&quot;     &quot;5422&quot;      &quot;11141&quot;     &quot;6990&quot;     
[1531] &quot;80258&quot;     &quot;55593&quot;     &quot;5355&quot;      &quot;89801&quot;     &quot;2245&quot;      &quot;55906&quot;    
[1537] &quot;367&quot;       &quot;27112&quot;     &quot;1896&quot;      &quot;54413&quot;     &quot;340526&quot;    &quot;9452&quot;     
[1543] &quot;695&quot;       &quot;105373311&quot; &quot;90293&quot;     &quot;286467&quot;    &quot;347475&quot;    &quot;55559&quot;    
[1549] &quot;5365&quot;      &quot;1736&quot;      &quot;4354&quot;     </code></pre>
<pre class="r"><code># Load expression dataset
boxplot1 &lt;- read.csv(&quot;data/boxplot1.csv&quot;, check.names = FALSE)
boxplot1 &lt;- as.data.frame(boxplot1)

# Choose a gene from the DOX-only mid-late response group
target_entrez &lt;- 1674  # Replace with your desired Entrez ID from prob_3_0.1

# Stop if the gene is not in the DOX-only mid-late response group
if (!(target_entrez %in% prob_3_0.1)) {
  stop(&quot;Selected gene is not in the DOX-only mid-late response group for 0.1 micromolar.&quot;)
}

# Filter for selected gene
gene_data &lt;- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if(nrow(gene_data) == 0) stop(&quot;No data found for selected ENTREZID.&quot;)

# Reshape to long format
gene_data_long &lt;- reshape2::melt(gene_data,
                                 id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                                 variable.name = &quot;Sample&quot;,
                                 value.name = &quot;log2CPM&quot;)

# Extract metadata from sample names
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)
  )

# Filter for 0.1 micromolar only
gene_data_long &lt;- gene_data_long %&gt;% filter(Concentration == &quot;0.1&quot;)

# Convert to factors
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = &quot;0.1&quot;)

# Map individual IDs
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, 
                  &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], &quot;Unknown&quot;))

# Define color palette for drugs
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;#08306B&quot;, &quot;DOX&quot; = &quot;#E7298A&quot;, &quot;VEH&quot; = &quot;green&quot;)

# Extract gene symbol for labeling
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;DOX-only mid-late\nresponse (0.1 micromolar)&quot;) +
  labs(
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/box3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box3-1">
Past versions of box3-1.png
</button>
</p>
<div id="fig-box3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/box3-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="non-response-0.5-micromolar" class="section level2">
<h2><strong>📌 Non response (0.5 micromolar)</strong></h2>
<pre class="r"><code># Load libraries
library(dplyr)
library(ggplot2)
library(reshape2)

# Load response group: Non response (0.5 µM)
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)

# Print genes (optional)
prob_1_0.5</code></pre>
<pre><code>   [1] &quot;653635&quot;    &quot;102723897&quot; &quot;102465432&quot; &quot;339451&quot;    &quot;57801&quot;     &quot;54973&quot;    
   [7] &quot;1855&quot;      &quot;54998&quot;     &quot;148413&quot;    &quot;55052&quot;     &quot;64856&quot;     &quot;219293&quot;   
  [13] &quot;83858&quot;     &quot;29101&quot;     &quot;643988&quot;    &quot;984&quot;       &quot;8511&quot;      &quot;728642&quot;   
  [19] &quot;9906&quot;      &quot;65220&quot;     &quot;2782&quot;      &quot;100506504&quot; &quot;199990&quot;    &quot;5192&quot;     
  [25] &quot;55229&quot;     &quot;440556&quot;    &quot;27237&quot;     &quot;1953&quot;      &quot;127262&quot;    &quot;49856&quot;    
  [31] &quot;57212&quot;     &quot;148870&quot;    &quot;57470&quot;     &quot;1677&quot;      &quot;261734&quot;    &quot;6146&quot;     
  [37] &quot;23463&quot;     &quot;8718&quot;      &quot;79707&quot;     &quot;3104&quot;      &quot;9903&quot;      &quot;148479&quot;   
  [43] &quot;90326&quot;     &quot;55735&quot;     &quot;23261&quot;     &quot;9341&quot;      &quot;11315&quot;     &quot;50651&quot;    
  [49] &quot;473&quot;       &quot;2023&quot;      &quot;80045&quot;     &quot;84275&quot;     &quot;199953&quot;    &quot;22883&quot;    
  [55] &quot;84328&quot;     &quot;64802&quot;     &quot;23095&quot;     &quot;378708&quot;    &quot;1676&quot;      &quot;5195&quot;     
  [61] &quot;54897&quot;     &quot;148345&quot;    &quot;23435&quot;     &quot;6723&quot;      &quot;2475&quot;      &quot;29914&quot;    
  [67] &quot;26270&quot;     &quot;10459&quot;     &quot;4524&quot;      &quot;1185&quot;      &quot;4878&quot;      &quot;55187&quot;    
  [73] &quot;126755&quot;    &quot;7799&quot;      &quot;79180&quot;     &quot;842&quot;       &quot;23341&quot;     &quot;84301&quot;    
  [79] &quot;6248&quot;      &quot;23207&quot;     &quot;284723&quot;    &quot;440567&quot;    &quot;23013&quot;     &quot;7709&quot;     
  [85] &quot;348487&quot;    &quot;128272&quot;    &quot;54455&quot;     &quot;55707&quot;     &quot;114819&quot;    &quot;55672&quot;    
  [91] &quot;84809&quot;     &quot;11209&quot;     &quot;4237&quot;      &quot;23400&quot;     &quot;6390&quot;      &quot;127707&quot;   
  [97] &quot;8659&quot;      &quot;23352&quot;     &quot;23065&quot;     &quot;100506730&quot; &quot;832&quot;       &quot;440574&quot;   
 [103] &quot;644068&quot;    &quot;255104&quot;    &quot;79594&quot;     &quot;50809&quot;     &quot;1889&quot;      &quot;5909&quot;     
 [109] &quot;401944&quot;    &quot;3339&quot;      &quot;29092&quot;     &quot;998&quot;       &quot;9923&quot;      &quot;23028&quot;    
 [115] &quot;7798&quot;      &quot;3352&quot;      &quot;80818&quot;     &quot;6920&quot;      &quot;55616&quot;     &quot;6135&quot;     
 [121] &quot;55629&quot;     &quot;57185&quot;     &quot;100506985&quot; &quot;10250&quot;     &quot;23585&quot;     &quot;55219&quot;    
 [127] &quot;56181&quot;     &quot;3925&quot;      &quot;5051&quot;      &quot;84676&quot;     &quot;149420&quot;    &quot;79927&quot;    
 [133] &quot;51042&quot;     &quot;10256&quot;     &quot;83442&quot;     &quot;79947&quot;     &quot;8289&quot;      &quot;63906&quot;    
 [139] &quot;10726&quot;     &quot;388610&quot;    &quot;23038&quot;     &quot;644961&quot;    &quot;2827&quot;      &quot;10163&quot;    
 [145] &quot;5511&quot;      &quot;6118&quot;      &quot;27293&quot;     &quot;55113&quot;     &quot;22826&quot;     &quot;93974&quot;    
 [151] &quot;65979&quot;     &quot;6080&quot;      &quot;54952&quot;     &quot;677838&quot;    &quot;10691&quot;     &quot;51441&quot;    
 [157] &quot;2035&quot;      &quot;6429&quot;      &quot;51102&quot;     &quot;9698&quot;      &quot;64129&quot;     &quot;90853&quot;    
 [163] &quot;8073&quot;      &quot;56063&quot;     &quot;65108&quot;     &quot;339487&quot;    &quot;5928&quot;      &quot;81493&quot;    
 [169] &quot;57648&quot;     &quot;64766&quot;     &quot;204&quot;       &quot;149076&quot;    &quot;1912&quot;      &quot;7579&quot;     
 [175] &quot;114784&quot;    &quot;113444&quot;    &quot;653160&quot;    &quot;79932&quot;     &quot;23154&quot;     &quot;5690&quot;     
 [181] &quot;127703&quot;    &quot;27095&quot;     &quot;55700&quot;     &quot;9967&quot;      &quot;79729&quot;     &quot;55194&quot;    
 [187] &quot;64960&quot;     &quot;728431&quot;    &quot;64769&quot;     &quot;54955&quot;     &quot;3633&quot;      &quot;10946&quot;    
 [193] &quot;51118&quot;     &quot;4725&quot;      &quot;643314&quot;    &quot;728448&quot;    &quot;8761&quot;      &quot;84879&quot;    
 [199] &quot;10487&quot;     &quot;6018&quot;      &quot;10269&quot;     &quot;127396&quot;    &quot;100130557&quot; &quot;4802&quot;     
 [205] &quot;163732&quot;    &quot;1503&quot;      &quot;100507178&quot; &quot;100132074&quot; &quot;1907&quot;      &quot;22887&quot;    
 [211] &quot;79717&quot;     &quot;728621&quot;    &quot;10465&quot;     &quot;114625&quot;    &quot;6513&quot;      &quot;10969&quot;    
 [217] &quot;64834&quot;     &quot;112950&quot;    &quot;23334&quot;     &quot;9682&quot;      &quot;9670&quot;      &quot;1802&quot;     
 [223] &quot;149473&quot;    &quot;55929&quot;     &quot;79033&quot;     &quot;79639&quot;     &quot;339541&quot;    &quot;6202&quot;     
 [229] &quot;7389&quot;      &quot;57643&quot;     &quot;84842&quot;     &quot;10327&quot;     &quot;4678&quot;      &quot;3652&quot;     
 [235] &quot;8503&quot;      &quot;55624&quot;     &quot;541468&quot;    &quot;7388&quot;      &quot;8569&quot;      &quot;148932&quot;   
 [241] &quot;64756&quot;     &quot;9813&quot;      &quot;51727&quot;     &quot;388630&quot;    &quot;54558&quot;     &quot;22996&quot;    
 [247] &quot;4898&quot;      &quot;51060&quot;     &quot;91408&quot;     &quot;200014&quot;    &quot;84950&quot;     &quot;2882&quot;     
 [253] &quot;79699&quot;     &quot;55268&quot;     &quot;6342&quot;      &quot;127435&quot;    &quot;54987&quot;     &quot;4116&quot;     
 [259] &quot;7804&quot;      &quot;54432&quot;     &quot;51668&quot;     &quot;115353&quot;    &quot;127428&quot;    &quot;606495&quot;   
 [265] &quot;51253&quot;     &quot;23648&quot;     &quot;619518&quot;    &quot;7268&quot;      &quot;25973&quot;     &quot;1718&quot;     
 [271] &quot;23358&quot;     &quot;5563&quot;      &quot;1600&quot;      &quot;114803&quot;    &quot;3725&quot;      &quot;4774&quot;     
 [277] &quot;83941&quot;     &quot;10207&quot;     &quot;163782&quot;    &quot;7398&quot;      &quot;85440&quot;     &quot;29929&quot;    
 [283] &quot;23421&quot;     &quot;55225&quot;     &quot;205&quot;       &quot;9829&quot;      &quot;54741&quot;     &quot;3953&quot;     
 [289] &quot;84251&quot;     &quot;57708&quot;     &quot;26135&quot;     &quot;55970&quot;     &quot;100289178&quot; &quot;79971&quot;    
 [295] &quot;57554&quot;     &quot;55631&quot;     &quot;9295&quot;      &quot;81573&quot;     &quot;1491&quot;      &quot;5733&quot;     
 [301] &quot;9406&quot;      &quot;257194&quot;    &quot;127253&quot;    &quot;204962&quot;    &quot;34&quot;        &quot;1744&quot;     
 [307] &quot;81849&quot;     &quot;23032&quot;     &quot;374986&quot;    &quot;79739&quot;     &quot;5567&quot;      &quot;1486&quot;     
 [313] &quot;23566&quot;     &quot;84144&quot;     &quot;57489&quot;     &quot;51100&quot;     &quot;9653&quot;      &quot;8543&quot;     
 [319] &quot;56267&quot;     &quot;23507&quot;     &quot;400761&quot;    &quot;84146&quot;     &quot;7049&quot;      &quot;646817&quot;   
 [325] &quot;79871&quot;     &quot;7813&quot;      &quot;6125&quot;      &quot;22823&quot;     &quot;50999&quot;     &quot;100131564&quot;
 [331] &quot;9411&quot;      &quot;199857&quot;    &quot;148534&quot;    &quot;100527978&quot; &quot;58155&quot;     &quot;51375&quot;    
 [337] &quot;391059&quot;    &quot;23443&quot;     &quot;64645&quot;     &quot;1629&quot;      &quot;100506007&quot; &quot;8634&quot;     
 [343] &quot;2135&quot;      &quot;1301&quot;      &quot;55599&quot;     &quot;280&quot;       &quot;55119&quot;     &quot;6301&quot;     
 [349] &quot;343263&quot;    &quot;127002&quot;    &quot;284613&quot;    &quot;83873&quot;     &quot;2773&quot;      &quot;271&quot;      
 [355] &quot;2948&quot;      &quot;2946&quot;      &quot;2944&quot;      &quot;2947&quot;      &quot;85369&quot;     &quot;388662&quot;   
 [361] &quot;9122&quot;      &quot;55791&quot;     &quot;10390&quot;     &quot;128344&quot;    &quot;128346&quot;    &quot;5906&quot;     
 [367] &quot;3752&quot;      &quot;55917&quot;     &quot;54879&quot;     &quot;829&quot;       &quot;9860&quot;      &quot;260425&quot;   
 [373] &quot;10717&quot;     &quot;56944&quot;     &quot;51592&quot;     &quot;10286&quot;     &quot;7812&quot;      &quot;80143&quot;    
 [379] &quot;10100&quot;     &quot;81839&quot;     &quot;845&quot;       &quot;55356&quot;     &quot;476&quot;       &quot;84852&quot;    
 [385] &quot;3321&quot;      &quot;9398&quot;      &quot;8458&quot;      &quot;80263&quot;     &quot;10885&quot;     &quot;10352&quot;    
 [391] &quot;101929147&quot; &quot;90874&quot;     &quot;4853&quot;      &quot;728841&quot;    &quot;101929796&quot; &quot;101060684&quot;
 [397] &quot;728855&quot;    &quot;103091866&quot; &quot;100288142&quot; &quot;101929780&quot; &quot;653519&quot;    &quot;11126&quot;    
 [403] &quot;10623&quot;     &quot;200035&quot;    &quot;10401&quot;     &quot;8515&quot;      &quot;10628&quot;     &quot;100132406&quot;
 [409] &quot;388677&quot;    &quot;149013&quot;    &quot;171423&quot;    &quot;607&quot;       &quot;200030&quot;    &quot;25832&quot;    
 [415] &quot;100996763&quot; &quot;400818&quot;    &quot;101060226&quot; &quot;57234&quot;     &quot;51027&quot;     &quot;10262&quot;    
 [421] &quot;10903&quot;     &quot;51177&quot;     &quot;81611&quot;     &quot;23632&quot;     &quot;51107&quot;     &quot;79630&quot;    
 [427] &quot;54460&quot;     &quot;9129&quot;      &quot;54507&quot;     &quot;2029&quot;      &quot;1513&quot;      &quot;29956&quot;    
 [433] &quot;56882&quot;     &quot;10500&quot;     &quot;388695&quot;    &quot;79005&quot;     &quot;8394&quot;      &quot;5710&quot;     
 [439] &quot;7286&quot;      &quot;81609&quot;     &quot;65005&quot;     &quot;11022&quot;     &quot;6281&quot;      &quot;6282&quot;     
 [445] &quot;6277&quot;      &quot;6275&quot;      &quot;23557&quot;     &quot;11000&quot;     &quot;9909&quot;      &quot;27173&quot;    
 [451] &quot;148327&quot;    &quot;10899&quot;     &quot;5872&quot;      &quot;6232&quot;      &quot;7170&quot;      &quot;25912&quot;    
 [457] &quot;9898&quot;      &quot;103&quot;       &quot;10654&quot;     &quot;57326&quot;     &quot;90780&quot;     &quot;6464&quot;     
 [463] &quot;1945&quot;      &quot;1942&quot;      &quot;54344&quot;     &quot;7059&quot;      &quot;4580&quot;      &quot;2630&quot;     
 [469] &quot;2629&quot;      &quot;1196&quot;      &quot;2224&quot;      &quot;55870&quot;     &quot;55249&quot;     &quot;23208&quot;    
 [475] &quot;6746&quot;      &quot;92312&quot;     &quot;23381&quot;     &quot;84283&quot;     &quot;112770&quot;    &quot;128240&quot;   
 [481] &quot;54865&quot;     &quot;63827&quot;     &quot;81875&quot;     &quot;5546&quot;      &quot;375033&quot;    &quot;9826&quot;     
 [487] &quot;3428&quot;      &quot;54935&quot;     &quot;8407&quot;      &quot;93185&quot;     &quot;8682&quot;      &quot;50717&quot;    
 [493] &quot;5824&quot;      &quot;1314&quot;      &quot;474338&quot;    &quot;23385&quot;     &quot;55600&quot;     &quot;50848&quot;    
 [499] &quot;100131187&quot; &quot;5202&quot;      &quot;4817&quot;      &quot;9191&quot;      &quot;8703&quot;      &quot;4720&quot;     
 [505] &quot;6391&quot;      &quot;257177&quot;    &quot;11266&quot;     &quot;25903&quot;     &quot;9722&quot;      &quot;400793&quot;   
 [511] &quot;223&quot;       &quot;54499&quot;     &quot;117143&quot;    &quot;387597&quot;    &quot;9019&quot;      &quot;25874&quot;    
 [517] &quot;55827&quot;     &quot;261726&quot;    &quot;2153&quot;      &quot;6403&quot;      &quot;92342&quot;     &quot;57147&quot;    
 [523] &quot;22920&quot;     &quot;23215&quot;     &quot;92346&quot;     &quot;5279&quot;      &quot;7292&quot;      &quot;27252&quot;    
 [529] &quot;55157&quot;     &quot;100506046&quot; &quot;26802&quot;     &quot;84614&quot;     &quot;27101&quot;     &quot;63931&quot;    
 [535] &quot;64326&quot;     &quot;60676&quot;     &quot;460&quot;       &quot;9462&quot;      &quot;55103&quot;     &quot;9917&quot;     
 [541] &quot;27&quot;        &quot;6646&quot;      &quot;163590&quot;    &quot;26092&quot;     &quot;9857&quot;      &quot;5768&quot;     
 [547] &quot;200058&quot;    &quot;100527964&quot; &quot;84320&quot;     &quot;9213&quot;      &quot;6041&quot;      &quot;1660&quot;     
 [553] &quot;3915&quot;      &quot;3918&quot;      &quot;23057&quot;     &quot;9887&quot;      &quot;23179&quot;     &quot;23127&quot;    
 [559] &quot;81563&quot;     &quot;6045&quot;      &quot;81627&quot;     &quot;83872&quot;     &quot;54953&quot;     &quot;100616345&quot;
 [565] &quot;51377&quot;     &quot;6738&quot;      &quot;343450&quot;    &quot;163486&quot;    &quot;140609&quot;    &quot;23046&quot;    
 [571] &quot;252839&quot;    &quot;3898&quot;      &quot;7135&quot;      &quot;23612&quot;     &quot;10440&quot;     &quot;51094&quot;    
 [577] &quot;9877&quot;      &quot;100381270&quot; &quot;6635&quot;      &quot;3814&quot;      &quot;84919&quot;     &quot;4194&quot;     
 [583] &quot;10446&quot;     &quot;23114&quot;     &quot;5929&quot;      &quot;64710&quot;     &quot;254428&quot;    &quot;284581&quot;   
 [589] &quot;103021295&quot; &quot;9641&quot;      &quot;83593&quot;     &quot;1939&quot;      &quot;8444&quot;      &quot;9261&quot;     
 [595] &quot;5362&quot;      &quot;50486&quot;     &quot;3664&quot;      &quot;27042&quot;     &quot;56256&quot;     &quot;55758&quot;    
 [601] &quot;7188&quot;      &quot;84791&quot;     &quot;100131205&quot; &quot;5525&quot;      &quot;25936&quot;     &quot;642946&quot;   
 [607] &quot;28982&quot;     &quot;79805&quot;     &quot;90806&quot;     &quot;100505832&quot; &quot;5629&quot;      &quot;5784&quot;     
 [613] &quot;51133&quot;     &quot;2104&quot;      &quot;7042&quot;      &quot;127018&quot;    &quot;55699&quot;     &quot;25782&quot;    
 [619] &quot;6791&quot;      &quot;4139&quot;      &quot;79762&quot;     &quot;9015&quot;      &quot;375056&quot;    &quot;64853&quot;    
 [625] &quot;148362&quot;    &quot;84976&quot;     &quot;55061&quot;     &quot;824&quot;       &quot;105369195&quot; &quot;4931&quot;     
 [631] &quot;3930&quot;      &quot;55740&quot;     &quot;6726&quot;      &quot;9725&quot;      &quot;10637&quot;     &quot;7044&quot;     
 [637] &quot;64746&quot;     &quot;83881&quot;     &quot;286826&quot;    &quot;5664&quot;      &quot;65094&quot;     &quot;116841&quot;   
 [643] &quot;79169&quot;     &quot;128308&quot;    &quot;81559&quot;     &quot;149603&quot;    &quot;5867&quot;      &quot;126731&quot;   
 [649] &quot;55746&quot;     &quot;23456&quot;     &quot;27097&quot;     &quot;9816&quot;      &quot;2590&quot;      &quot;79605&quot;    
 [655] &quot;183&quot;       &quot;84886&quot;     &quot;64801&quot;     &quot;128061&quot;    &quot;8443&quot;      &quot;54583&quot;    
 [661] &quot;7257&quot;      &quot;84284&quot;     &quot;3775&quot;      &quot;101927765&quot; &quot;388753&quot;    &quot;9453&quot;     
 [667] &quot;4811&quot;      &quot;128178&quot;    &quot;3964&quot;      &quot;55127&quot;     &quot;1131&quot;      &quot;2271&quot;     
 [673] &quot;23596&quot;     &quot;1122&quot;      &quot;200150&quot;    &quot;731275&quot;    &quot;10806&quot;     &quot;10000&quot;    
 [679] &quot;159&quot;       &quot;51029&quot;     &quot;116228&quot;    &quot;3192&quot;      &quot;101928068&quot; &quot;163882&quot;   
 [685] &quot;100130097&quot; &quot;25909&quot;     &quot;93474&quot;     &quot;79862&quot;     &quot;388759&quot;    &quot;84838&quot;    
 [691] &quot;25893&quot;     &quot;343171&quot;    &quot;80851&quot;     &quot;79894&quot;     &quot;267002&quot;    &quot;10771&quot;    
 [697] &quot;91734&quot;     &quot;3422&quot;      &quot;22884&quot;     &quot;5214&quot;      &quot;1316&quot;      &quot;1645&quot;     
 [703] &quot;10276&quot;     &quot;54906&quot;     &quot;2665&quot;      &quot;54522&quot;     &quot;84893&quot;     &quot;84991&quot;    
 [709] &quot;5209&quot;      &quot;22944&quot;     &quot;10659&quot;     &quot;9712&quot;      &quot;79746&quot;     &quot;254427&quot;   
 [715] &quot;55526&quot;     &quot;55176&quot;     &quot;11164&quot;     &quot;8872&quot;      &quot;57118&quot;     &quot;22929&quot;    
 [721] &quot;55691&quot;     &quot;51182&quot;     &quot;64421&quot;     &quot;221060&quot;    &quot;10557&quot;     &quot;221061&quot;   
 [727] &quot;8516&quot;      &quot;8029&quot;      &quot;7431&quot;      &quot;8027&quot;      &quot;221079&quot;    &quot;84898&quot;    
 [733] &quot;399726&quot;    &quot;387640&quot;    &quot;64215&quot;     &quot;23412&quot;     &quot;648&quot;       &quot;100499489&quot;
 [739] &quot;22921&quot;     &quot;56243&quot;     &quot;56952&quot;     &quot;57512&quot;     &quot;23590&quot;     &quot;10006&quot;    
 [745] &quot;91452&quot;     &quot;22931&quot;     &quot;143098&quot;    &quot;220906&quot;    &quot;51322&quot;     &quot;102724316&quot;
 [751] &quot;220929&quot;    &quot;6935&quot;      &quot;94134&quot;     &quot;3799&quot;      &quot;80314&quot;     &quot;79741&quot;    
 [757] &quot;3688&quot;      &quot;8829&quot;      &quot;56288&quot;     &quot;8453&quot;      &quot;219771&quot;    &quot;219749&quot;   
 [763] &quot;399744&quot;    &quot;441666&quot;    &quot;100129482&quot; &quot;221002&quot;    &quot;8187&quot;      &quot;220992&quot;   
 [769] &quot;414208&quot;    &quot;6387&quot;      &quot;11067&quot;     &quot;220979&quot;    &quot;93550&quot;     &quot;253725&quot;   
 [775] &quot;119016&quot;    &quot;728407&quot;    &quot;100287932&quot; &quot;653268&quot;    &quot;728113&quot;    &quot;643650&quot;   
 [781] &quot;5540&quot;      &quot;83849&quot;     &quot;414241&quot;    &quot;399761&quot;    &quot;653145&quot;    &quot;642517&quot;   
 [787] &quot;642826&quot;    &quot;439965&quot;    &quot;102724488&quot; &quot;100996758&quot; &quot;414224&quot;    &quot;143162&quot;   
 [793] &quot;2074&quot;      &quot;8505&quot;      &quot;100652748&quot; &quot;414189&quot;    &quot;387680&quot;    &quot;56624&quot;    
 [799] &quot;259230&quot;    &quot;653308&quot;    &quot;5592&quot;      &quot;23283&quot;     &quot;22943&quot;     &quot;7019&quot;     
 [805] &quot;8030&quot;      &quot;288&quot;       &quot;84159&quot;     &quot;221035&quot;    &quot;347731&quot;    &quot;26091&quot;    
 [811] &quot;84665&quot;     &quot;3189&quot;      &quot;55680&quot;     &quot;8034&quot;      &quot;80312&quot;     &quot;55749&quot;    
 [817] &quot;79009&quot;     &quot;9188&quot;      &quot;6832&quot;      &quot;80201&quot;     &quot;3098&quot;      &quot;6865&quot;     
 [823] &quot;55506&quot;     &quot;84883&quot;     &quot;56681&quot;     &quot;5464&quot;      &quot;140766&quot;    &quot;219699&quot;   
 [829] &quot;728978&quot;    &quot;55315&quot;     &quot;64115&quot;     &quot;9469&quot;      &quot;9806&quot;      &quot;51008&quot;    
 [835] &quot;119504&quot;    &quot;54541&quot;     &quot;54788&quot;     &quot;10367&quot;     &quot;90550&quot;     &quot;5033&quot;     
 [841] &quot;25961&quot;     &quot;11319&quot;     &quot;317662&quot;    &quot;23234&quot;     &quot;414245&quot;    &quot;51021&quot;    
 [847] &quot;118491&quot;    &quot;310&quot;       &quot;5532&quot;      &quot;159195&quot;    &quot;58529&quot;     &quot;729092&quot;   
 [853] &quot;729096&quot;    &quot;9632&quot;      &quot;118487&quot;    &quot;8509&quot;      &quot;818&quot;       &quot;7414&quot;     
 [859] &quot;26985&quot;     &quot;23522&quot;     &quot;51207&quot;     &quot;7417&quot;      &quot;118881&quot;    &quot;84858&quot;    
 [865] &quot;100131213&quot; &quot;3778&quot;      &quot;9231&quot;      &quot;11128&quot;     &quot;57178&quot;     &quot;10105&quot;    
 [871] &quot;219654&quot;    &quot;143244&quot;    &quot;650623&quot;    &quot;101060691&quot; &quot;311&quot;       &quot;84293&quot;    
 [877] &quot;10718&quot;     &quot;92211&quot;     &quot;2894&quot;      &quot;23063&quot;     &quot;79812&quot;     &quot;6623&quot;     
 [883] &quot;728118&quot;    &quot;728190&quot;    &quot;9562&quot;      &quot;84896&quot;     &quot;57559&quot;     &quot;59&quot;       
 [889] &quot;3988&quot;      &quot;84333&quot;     &quot;22849&quot;     &quot;3416&quot;      &quot;54536&quot;     &quot;26509&quot;    
 [895] &quot;118924&quot;    &quot;266971&quot;    &quot;64318&quot;     &quot;9124&quot;      &quot;10580&quot;     &quot;26123&quot;    
 [901] &quot;54619&quot;     &quot;9849&quot;      &quot;118788&quot;    &quot;728658&quot;    &quot;26148&quot;     &quot;84986&quot;    
 [907] &quot;644215&quot;    &quot;5223&quot;      &quot;51013&quot;     &quot;84287&quot;     &quot;64210&quot;     &quot;26287&quot;    
 [913] &quot;112817&quot;    &quot;83742&quot;     &quot;27291&quot;     &quot;84795&quot;     &quot;26507&quot;     &quot;81894&quot;    
 [919] &quot;1355&quot;      &quot;23268&quot;     &quot;1147&quot;      &quot;282991&quot;    &quot;6319&quot;      &quot;25956&quot;    
 [925] &quot;4714&quot;      &quot;55662&quot;     &quot;57715&quot;     &quot;84545&quot;     &quot;81855&quot;     &quot;81621&quot;    
 [931] &quot;8945&quot;      &quot;27343&quot;     &quot;25911&quot;     &quot;10360&quot;     &quot;10724&quot;     &quot;79803&quot;    
 [937] &quot;23082&quot;     &quot;9221&quot;      &quot;8729&quot;      &quot;5662&quot;      &quot;79176&quot;     &quot;79004&quot;    
 [943] &quot;79847&quot;     &quot;10121&quot;     &quot;51684&quot;     &quot;403&quot;       &quot;118980&quot;    &quot;54838&quot;    
 [949] &quot;729020&quot;    &quot;84833&quot;     &quot;51063&quot;     &quot;79991&quot;     &quot;9748&quot;      &quot;119392&quot;   
 [955] &quot;9446&quot;      &quot;159686&quot;    &quot;114815&quot;    &quot;7511&quot;      &quot;120&quot;       &quot;10285&quot;    
 [961] &quot;9126&quot;      &quot;143187&quot;    &quot;6934&quot;      &quot;840&quot;       &quot;374354&quot;    &quot;153&quot;      
 [967] &quot;55088&quot;     &quot;3983&quot;      &quot;57700&quot;     &quot;142940&quot;    &quot;57698&quot;     &quot;63877&quot;    
 [973] &quot;8661&quot;      &quot;10935&quot;     &quot;2869&quot;      &quot;6001&quot;      &quot;22876&quot;     &quot;79892&quot;    
 [979] &quot;55717&quot;     &quot;11101&quot;     &quot;5654&quot;      &quot;118672&quot;    &quot;36&quot;        &quot;9184&quot;     
 [985] &quot;119587&quot;    &quot;9679&quot;      &quot;23172&quot;     &quot;54764&quot;     &quot;26098&quot;     &quot;7390&quot;     
 [991] &quot;56647&quot;     &quot;55760&quot;     &quot;8038&quot;      &quot;5791&quot;      &quot;4255&quot;      &quot;387723&quot;   
 [997] &quot;119437&quot;    &quot;10539&quot;     &quot;55844&quot;     &quot;664&quot;       &quot;282974&quot;    &quot;170394&quot;   
[1003] &quot;3632&quot;      &quot;85442&quot;     &quot;10844&quot;     &quot;118472&quot;    &quot;282969&quot;    &quot;1892&quot;     
[1009] &quot;196743&quot;    &quot;92170&quot;     &quot;503542&quot;    &quot;51272&quot;     &quot;80162&quot;     &quot;10581&quot;    
[1015] &quot;8519&quot;      &quot;10410&quot;     &quot;81490&quot;     &quot;3265&quot;      &quot;8045&quot;      &quot;57661&quot;    
[1021] &quot;10522&quot;     &quot;283232&quot;    &quot;64787&quot;     &quot;347862&quot;    &quot;105369190&quot; &quot;79751&quot;    
[1027] &quot;6181&quot;      &quot;283229&quot;    &quot;977&quot;       &quot;66005&quot;     &quot;161&quot;       &quot;9024&quot;     
[1033] &quot;81532&quot;     &quot;1850&quot;      &quot;338651&quot;    &quot;1509&quot;      &quot;6150&quot;      &quot;283120&quot;   
[1039] &quot;3481&quot;      &quot;51214&quot;     &quot;975&quot;       &quot;10984&quot;     &quot;833&quot;       &quot;7748&quot;     
[1045] &quot;650368&quot;    &quot;116969&quot;    &quot;391&quot;       &quot;6737&quot;      &quot;85363&quot;     &quot;84067&quot;    
[1051] &quot;112464&quot;    &quot;6609&quot;      &quot;322&quot;       &quot;23647&quot;     &quot;26515&quot;     &quot;23378&quot;    
[1057] &quot;1200&quot;      &quot;8495&quot;      &quot;51700&quot;     &quot;8665&quot;      &quot;9866&quot;      &quot;6157&quot;     
[1063] &quot;6764&quot;      &quot;56674&quot;     &quot;493900&quot;    &quot;283102&quot;    &quot;10527&quot;     &quot;644656&quot;   
[1069] &quot;7465&quot;      &quot;440028&quot;    &quot;133&quot;       &quot;100463486&quot; &quot;50862&quot;     &quot;10335&quot;    
[1075] &quot;58486&quot;     &quot;729013&quot;    &quot;55031&quot;     &quot;9645&quot;      &quot;84953&quot;     &quot;55742&quot;    
[1081] &quot;7003&quot;      &quot;10418&quot;     &quot;22800&quot;     &quot;5140&quot;      &quot;55553&quot;     &quot;10944&quot;    
[1087] &quot;6207&quot;      &quot;4925&quot;      &quot;11234&quot;     &quot;2965&quot;      &quot;3939&quot;      &quot;7251&quot;     
[1093] &quot;55293&quot;     &quot;8048&quot;      &quot;79733&quot;     &quot;89797&quot;     &quot;100874014&quot; &quot;10196&quot;    
[1099] &quot;203859&quot;    &quot;2188&quot;      &quot;258010&quot;    &quot;159963&quot;    &quot;627&quot;       &quot;120534&quot;   
[1105] &quot;120526&quot;    &quot;5954&quot;      &quot;79832&quot;     &quot;55346&quot;     &quot;283267&quot;    &quot;25758&quot;    
[1111] &quot;4005&quot;      &quot;55226&quot;     &quot;847&quot;       &quot;8050&quot;      &quot;960&quot;       &quot;6506&quot;     
[1117] &quot;24147&quot;     &quot;54765&quot;     &quot;143458&quot;    &quot;29099&quot;     &quot;7189&quot;      &quot;5896&quot;     
[1123] &quot;8539&quot;      &quot;55761&quot;     &quot;51144&quot;     &quot;84680&quot;     &quot;2132&quot;      &quot;90139&quot;    
[1129] &quot;56981&quot;     &quot;57586&quot;     &quot;8534&quot;      &quot;55343&quot;     &quot;1408&quot;      &quot;9479&quot;     
[1135] &quot;51317&quot;     &quot;55626&quot;     &quot;283254&quot;    &quot;392&quot;       &quot;4038&quot;      &quot;84364&quot;    
[1141] &quot;29763&quot;     &quot;53&quot;        &quot;10062&quot;     &quot;91252&quot;     &quot;5702&quot;      &quot;55709&quot;    
[1147] &quot;4722&quot;      &quot;114900&quot;    &quot;23360&quot;     &quot;5795&quot;      &quot;2346&quot;      &quot;29015&quot;    
[1153] &quot;8501&quot;      &quot;26519&quot;     &quot;9246&quot;      &quot;710&quot;       &quot;25921&quot;     &quot;280636&quot;   
[1159] &quot;63901&quot;     &quot;5007&quot;      &quot;219988&quot;    &quot;27339&quot;     &quot;54972&quot;     &quot;55048&quot;    
[1165] &quot;26007&quot;     &quot;79869&quot;     &quot;54949&quot;     &quot;9066&quot;      &quot;746&quot;       &quot;3992&quot;     
[1171] &quot;3995&quot;      &quot;5866&quot;      &quot;7439&quot;      &quot;80150&quot;     &quot;79026&quot;     &quot;1937&quot;     
[1177] &quot;256364&quot;    &quot;23193&quot;     &quot;102288414&quot; &quot;790955&quot;    &quot;51035&quot;     &quot;26580&quot;    
[1183] &quot;221092&quot;    &quot;283237&quot;    &quot;79842&quot;     &quot;5436&quot;      &quot;374395&quot;    &quot;9303&quot;     
[1189] &quot;11145&quot;     &quot;25923&quot;     &quot;144097&quot;    &quot;1351&quot;      &quot;28992&quot;     &quot;10963&quot;    
[1195] &quot;83707&quot;     &quot;3338&quot;      &quot;7423&quot;      &quot;2286&quot;      &quot;26472&quot;     &quot;5331&quot;     
[1201] &quot;56834&quot;     &quot;572&quot;       &quot;2101&quot;      &quot;25824&quot;     &quot;283234&quot;    &quot;8986&quot;     
[1207] &quot;4221&quot;      &quot;402&quot;       &quot;7542&quot;      &quot;7108&quot;      &quot;728975&quot;    &quot;283131&quot;   
[1213] &quot;101927789&quot; &quot;378938&quot;    &quot;57410&quot;     &quot;4054&quot;      &quot;254102&quot;    &quot;4296&quot;     
[1219] &quot;399909&quot;    &quot;5970&quot;      &quot;84153&quot;     &quot;9158&quot;      &quot;83638&quot;     &quot;8815&quot;     
[1225] &quot;10992&quot;     &quot;55690&quot;     &quot;81876&quot;     &quot;9610&quot;      &quot;65003&quot;     &quot;10072&quot;    
[1231] &quot;254359&quot;    &quot;89&quot;        &quot;8722&quot;      &quot;55231&quot;     &quot;10432&quot;     &quot;5936&quot;     
[1237] &quot;83759&quot;     &quot;79703&quot;     &quot;9986&quot;      &quot;29984&quot;     &quot;57804&quot;     &quot;5883&quot;     
[1243] &quot;2950&quot;      &quot;4723&quot;      &quot;390213&quot;    &quot;347853&quot;    &quot;221&quot;       &quot;1119&quot;     
[1249] &quot;51111&quot;     &quot;55291&quot;     &quot;219927&quot;    &quot;3508&quot;      &quot;8500&quot;      &quot;55199&quot;    
[1255] &quot;55298&quot;     &quot;4926&quot;      &quot;55004&quot;     &quot;25906&quot;     &quot;2348&quot;      &quot;116985&quot;   
[1261] &quot;9873&quot;      &quot;9828&quot;      &quot;23201&quot;     &quot;58473&quot;     &quot;5870&quot;      &quot;51642&quot;    
[1267] &quot;80227&quot;     &quot;7351&quot;      &quot;283209&quot;    &quot;387787&quot;    &quot;143570&quot;    &quot;10825&quot;    
[1273] &quot;6188&quot;      &quot;283212&quot;    &quot;871&quot;       &quot;84649&quot;     &quot;7481&quot;      &quot;5612&quot;     
[1279] &quot;56946&quot;     &quot;2615&quot;      &quot;726&quot;       &quot;5058&quot;      &quot;1207&quot;      &quot;28971&quot;    
[1285] &quot;92105&quot;     &quot;79053&quot;     &quot;57558&quot;     &quot;9846&quot;      &quot;5547&quot;      &quot;27314&quot;    
[1291] &quot;100506233&quot; &quot;51585&quot;     &quot;338699&quot;    &quot;1740&quot;      &quot;55863&quot;     &quot;58487&quot;    
[1297] &quot;54843&quot;     &quot;8301&quot;      &quot;8322&quot;      &quot;1075&quot;      &quot;50507&quot;     &quot;10003&quot;    
[1303] &quot;26973&quot;     &quot;684959&quot;    &quot;692063&quot;    &quot;654320&quot;    &quot;677805&quot;    &quot;677822&quot;   
[1309] &quot;79101&quot;     &quot;9440&quot;      &quot;24145&quot;     &quot;4361&quot;      &quot;2526&quot;      &quot;51503&quot;    
[1315] &quot;9702&quot;      &quot;8898&quot;      &quot;8690&quot;      &quot;143872&quot;    &quot;5241&quot;      &quot;329&quot;      
[1321] &quot;114908&quot;    &quot;4319&quot;      &quot;100288077&quot; &quot;4312&quot;      &quot;84259&quot;     &quot;79659&quot;    
[1327] &quot;837&quot;       &quot;143879&quot;    &quot;60496&quot;     &quot;55531&quot;     &quot;8065&quot;      &quot;38&quot;       
[1333] &quot;4863&quot;      &quot;5962&quot;      &quot;143903&quot;    &quot;23235&quot;     &quot;5519&quot;      &quot;79796&quot;    
[1339] &quot;91893&quot;     &quot;64776&quot;     &quot;1410&quot;      &quot;3316&quot;      &quot;85458&quot;     &quot;6392&quot;     
[1345] &quot;5805&quot;      &quot;54970&quot;     &quot;1813&quot;      &quot;54494&quot;     &quot;10179&quot;     &quot;335&quot;      
[1351] &quot;5049&quot;      &quot;100652768&quot; &quot;9159&quot;      &quot;53826&quot;     &quot;196264&quot;    &quot;9354&quot;     
[1357] &quot;10632&quot;     &quot;4297&quot;      &quot;372&quot;       &quot;23187&quot;     &quot;1656&quot;      &quot;283149&quot;   
[1363] &quot;338657&quot;    &quot;6230&quot;      &quot;51399&quot;     &quot;10525&quot;     &quot;1798&quot;      &quot;25988&quot;    
[1369] &quot;79102&quot;     &quot;5818&quot;      &quot;23365&quot;     &quot;7007&quot;      &quot;6309&quot;      &quot;399959&quot;   
[1375] &quot;3312&quot;      &quot;57476&quot;     &quot;84897&quot;     &quot;53340&quot;     &quot;219844&quot;    &quot;50937&quot;    
[1381] &quot;84881&quot;     &quot;79607&quot;     &quot;114609&quot;    &quot;2313&quot;      &quot;100507392&quot; &quot;9743&quot;     
[1387] &quot;8538&quot;      &quot;4798&quot;      &quot;29068&quot;     &quot;103611081&quot; &quot;399979&quot;    &quot;50863&quot;    
[1393] &quot;4978&quot;      &quot;22997&quot;     &quot;83700&quot;     &quot;27034&quot;     &quot;100288778&quot; &quot;6540&quot;     
[1399] &quot;5927&quot;      &quot;84318&quot;     &quot;4815&quot;      &quot;65125&quot;     &quot;5893&quot;      &quot;81029&quot;    
[1405] &quot;79602&quot;     &quot;55846&quot;     &quot;83714&quot;     &quot;83695&quot;     &quot;7289&quot;      &quot;7004&quot;     
[1411] &quot;57097&quot;     &quot;894&quot;       &quot;57102&quot;     &quot;10566&quot;     &quot;4908&quot;      &quot;7132&quot;     
[1417] &quot;678655&quot;    &quot;51258&quot;     &quot;2597&quot;      &quot;25900&quot;     &quot;4839&quot;      &quot;51147&quot;    
[1423] &quot;171017&quot;    &quot;196500&quot;    &quot;50813&quot;     &quot;5763&quot;      &quot;10536&quot;     &quot;2784&quot;     
[1429] &quot;8078&quot;      &quot;7167&quot;      &quot;84727&quot;     &quot;171220&quot;    &quot;10233&quot;     &quot;2026&quot;     
[1435] &quot;1822&quot;      &quot;113246&quot;    &quot;5777&quot;      &quot;11331&quot;     &quot;677777&quot;    &quot;10436&quot;    
[1441] &quot;716&quot;       &quot;715&quot;       &quot;51279&quot;     &quot;5830&quot;      &quot;144195&quot;    &quot;6515&quot;     
[1447] &quot;55810&quot;     &quot;440078&quot;    &quot;8076&quot;      &quot;57494&quot;     &quot;1911&quot;      &quot;144571&quot;   
[1453] &quot;2&quot;         &quot;100499405&quot; &quot;29121&quot;     &quot;23710&quot;     &quot;10748&quot;     &quot;100129361&quot;
[1459] &quot;100506248&quot; &quot;503693&quot;    &quot;1389&quot;      &quot;1027&quot;      &quot;81575&quot;     &quot;51202&quot;    
[1465] &quot;9052&quot;      &quot;693199&quot;    &quot;50865&quot;     &quot;57613&quot;     &quot;83445&quot;     &quot;55729&quot;    
[1471] &quot;79887&quot;     &quot;144608&quot;    &quot;51729&quot;     &quot;11171&quot;     &quot;51071&quot;     &quot;4257&quot;     
[1477] &quot;121536&quot;    &quot;5139&quot;      &quot;79912&quot;     &quot;5965&quot;      &quot;51026&quot;     &quot;80763&quot;    
[1483] &quot;2998&quot;      &quot;3945&quot;      &quot;3764&quot;      &quot;10060&quot;     &quot;55907&quot;     &quot;9847&quot;     
[1489] &quot;144363&quot;    &quot;3845&quot;      &quot;8082&quot;      &quot;51768&quot;     &quot;9412&quot;      &quot;23012&quot;    
[1495] &quot;56938&quot;     &quot;8496&quot;      &quot;387849&quot;    &quot;55297&quot;     &quot;100506606&quot; &quot;83857&quot;    
[1501] &quot;55196&quot;     &quot;51067&quot;     &quot;144245&quot;    &quot;114134&quot;    &quot;1272&quot;      &quot;283464&quot;   
[1507] &quot;5756&quot;      &quot;400027&quot;    &quot;196528&quot;    &quot;81539&quot;     &quot;54407&quot;     &quot;100288798&quot;
[1513] &quot;347902&quot;    &quot;79657&quot;     &quot;10411&quot;     &quot;55652&quot;     &quot;1280&quot;      &quot;140461&quot;   
[1519] &quot;121274&quot;    &quot;784&quot;       &quot;9416&quot;      &quot;85478&quot;     &quot;51303&quot;     &quot;7480&quot;     
[1525] &quot;5571&quot;      &quot;8085&quot;      &quot;55716&quot;     &quot;79962&quot;     &quot;65244&quot;     &quot;84070&quot;    
[1531] &quot;25766&quot;     &quot;91010&quot;     &quot;7009&quot;      &quot;57701&quot;     &quot;6602&quot;      &quot;2819&quot;     
[1537] &quot;84987&quot;     &quot;51474&quot;     &quot;113251&quot;    &quot;25840&quot;     &quot;7024&quot;      &quot;6334&quot;     
[1543] &quot;401720&quot;    &quot;105369971&quot; &quot;91&quot;        &quot;160622&quot;    &quot;60673&quot;     &quot;144501&quot;   
[1549] &quot;3856&quot;      &quot;1975&quot;      &quot;283335&quot;    &quot;84926&quot;     &quot;3489&quot;      &quot;84975&quot;    
[1555] &quot;5204&quot;      &quot;5094&quot;      &quot;7786&quot;      &quot;6895&quot;      &quot;23583&quot;     &quot;22818&quot;    
[1561] &quot;3679&quot;      &quot;2647&quot;      &quot;967&quot;       &quot;10220&quot;     &quot;84324&quot;     &quot;4327&quot;     
[1567] &quot;84305&quot;     &quot;102724112&quot; &quot;1606&quot;      &quot;6490&quot;      &quot;5869&quot;      &quot;6171&quot;     
[1573] &quot;23344&quot;     &quot;140465&quot;    &quot;4637&quot;      &quot;6601&quot;      &quot;79035&quot;     &quot;93058&quot;    
[1579] &quot;10330&quot;     &quot;9924&quot;      &quot;283377&quot;    &quot;11176&quot;     &quot;506&quot;       &quot;10728&quot;    
[1585] &quot;4666&quot;      &quot;4665&quot;      &quot;6778&quot;      &quot;4035&quot;      &quot;246329&quot;    &quot;22864&quot;    
[1591] &quot;114785&quot;    &quot;10540&quot;     &quot;3798&quot;      &quot;10956&quot;     &quot;4234&quot;      &quot;121227&quot;   
[1597] &quot;338811&quot;    &quot;23041&quot;     &quot;57460&quot;     &quot;283417&quot;    &quot;29110&quot;     &quot;23592&quot;    
[1603] &quot;8091&quot;      &quot;84298&quot;     &quot;11213&quot;     &quot;23426&quot;     &quot;55832&quot;     &quot;8445&quot;     
[1609] &quot;56890&quot;     &quot;57122&quot;     &quot;55508&quot;     &quot;11052&quot;     &quot;10576&quot;     &quot;144453&quot;   
[1615] &quot;196446&quot;    &quot;101928062&quot; &quot;196441&quot;    &quot;64786&quot;     &quot;552889&quot;    &quot;11010&quot;    
[1621] &quot;4673&quot;      &quot;79738&quot;     &quot;1466&quot;      &quot;89795&quot;     &quot;6857&quot;      &quot;4659&quot;     
[1627] &quot;283310&quot;    &quot;8825&quot;      &quot;79611&quot;     &quot;84190&quot;     &quot;160335&quot;    &quot;400058&quot;   
[1633] &quot;91298&quot;     &quot;160418&quot;    &quot;4060&quot;      &quot;1634&quot;      &quot;440107&quot;    &quot;8411&quot;     
[1639] &quot;7334&quot;      &quot;8835&quot;      &quot;8738&quot;      &quot;10154&quot;     &quot;57458&quot;     &quot;7181&quot;     
[1645] &quot;10988&quot;     &quot;59277&quot;     &quot;121441&quot;    &quot;121456&quot;    &quot;55592&quot;     &quot;55681&quot;    
[1651] &quot;27340&quot;     &quot;400&quot;       &quot;79158&quot;     &quot;51019&quot;     &quot;79023&quot;     &quot;7184&quot;     
[1657] &quot;728568&quot;    &quot;50515&quot;     &quot;160428&quot;    &quot;23325&quot;     &quot;55198&quot;     &quot;9891&quot;     
[1663] &quot;10970&quot;     &quot;90488&quot;     &quot;80298&quot;     &quot;11137&quot;     &quot;11153&quot;     &quot;6404&quot;     
[1669] &quot;54434&quot;     &quot;84749&quot;     &quot;121642&quot;    &quot;9815&quot;      &quot;400073&quot;    &quot;10094&quot;    
[1675] &quot;29902&quot;     &quot;144715&quot;    &quot;160760&quot;    &quot;79600&quot;     &quot;160762&quot;    &quot;4633&quot;     
[1681] &quot;23316&quot;     &quot;6311&quot;      &quot;217&quot;       &quot;51275&quot;     &quot;8550&quot;      &quot;89894&quot;    
[1687] &quot;8759&quot;      &quot;10961&quot;     &quot;283450&quot;    &quot;6128&quot;      &quot;5781&quot;      &quot;53373&quot;    
[1693] &quot;80024&quot;     &quot;196463&quot;    &quot;113675&quot;    &quot;54997&quot;     &quot;23014&quot;     &quot;54621&quot;    
[1699] &quot;5037&quot;      &quot;64426&quot;     &quot;26353&quot;     &quot;10985&quot;     &quot;6175&quot;      &quot;5829&quot;     
[1705] &quot;23409&quot;     &quot;51499&quot;     &quot;8683&quot;      &quot;100506668&quot; &quot;84274&quot;     &quot;9921&quot;     
[1711] &quot;51367&quot;     &quot;35&quot;        &quot;121665&quot;    &quot;5025&quot;      &quot;10645&quot;     &quot;80196&quot;    
[1717] &quot;23067&quot;     &quot;5715&quot;      &quot;144406&quot;    &quot;605&quot;       &quot;56616&quot;     &quot;65082&quot;    
[1723] &quot;55596&quot;     &quot;65117&quot;     &quot;8562&quot;      &quot;79720&quot;     &quot;23457&quot;     &quot;51329&quot;    
[1729] &quot;57605&quot;     &quot;10198&quot;     &quot;91574&quot;     &quot;196383&quot;    &quot;353116&quot;    &quot;10959&quot;    
[1735] &quot;57696&quot;     &quot;23545&quot;     &quot;144348&quot;    &quot;7316&quot;      &quot;114795&quot;    &quot;121260&quot;   
[1741] &quot;2054&quot;      &quot;5901&quot;      &quot;283383&quot;    &quot;6433&quot;      &quot;317781&quot;    &quot;79050&quot;    
[1747] &quot;100996573&quot; &quot;23141&quot;     &quot;55743&quot;     &quot;7574&quot;      &quot;7637&quot;      &quot;7699&quot;     
[1753] &quot;101060200&quot; &quot;7556&quot;      &quot;10795&quot;     &quot;55269&quot;     &quot;7750&quot;      &quot;2700&quot;     
[1759] &quot;51084&quot;     &quot;8100&quot;      &quot;53342&quot;     &quot;253832&quot;    &quot;221154&quot;    &quot;2254&quot;     
[1765] &quot;6445&quot;      &quot;26278&quot;     &quot;4285&quot;      &quot;5042&quot;      &quot;9818&quot;      &quot;10810&quot;    
[1771] &quot;6144&quot;      &quot;2971&quot;      &quot;219402&quot;    &quot;255967&quot;    &quot;2321&quot;      &quot;51371&quot;    
[1777] &quot;3146&quot;      &quot;23111&quot;     &quot;29880&quot;     &quot;55578&quot;     &quot;122011&quot;    &quot;10631&quot;    
[1783] &quot;341640&quot;    &quot;80209&quot;     &quot;89890&quot;     &quot;9617&quot;      &quot;11215&quot;     &quot;94240&quot;    
[1789] &quot;29103&quot;     &quot;26747&quot;     &quot;55425&quot;     &quot;2963&quot;      &quot;7178&quot;      &quot;100190939&quot;
[1795] &quot;253512&quot;    &quot;83548&quot;     &quot;23091&quot;     &quot;8803&quot;      &quot;55270&quot;     &quot;29079&quot;    
[1801] &quot;22862&quot;     &quot;81602&quot;     &quot;83852&quot;     &quot;3839&quot;      &quot;26512&quot;     &quot;79758&quot;    
[1807] &quot;440138&quot;    &quot;9724&quot;      &quot;55901&quot;     &quot;10910&quot;     &quot;81550&quot;     &quot;64881&quot;    
[1813] &quot;9882&quot;      &quot;170622&quot;    &quot;1203&quot;      &quot;23077&quot;     &quot;122060&quot;    &quot;1910&quot;     
[1819] &quot;64062&quot;     &quot;54602&quot;     &quot;26050&quot;     &quot;407975&quot;    &quot;23483&quot;     &quot;10257&quot;    
[1825] &quot;5611&quot;      &quot;55757&quot;     &quot;5911&quot;      &quot;23348&quot;     &quot;9375&quot;      &quot;171425&quot;   
[1831] &quot;84899&quot;     &quot;79070&quot;     &quot;54841&quot;     &quot;3981&quot;      &quot;84945&quot;     &quot;1282&quot;     
[1837] &quot;55647&quot;     &quot;79587&quot;     &quot;8874&quot;      &quot;2159&quot;      &quot;102724474&quot; &quot;8451&quot;     
[1843] &quot;55208&quot;     &quot;55002&quot;     &quot;7027&quot;      &quot;2621&quot;      &quot;8881&quot;      &quot;65110&quot;    
[1849] &quot;283489&quot;    &quot;57820&quot;     &quot;85495&quot;     &quot;6038&quot;      &quot;6035&quot;      &quot;101929718&quot;
[1855] &quot;57447&quot;     &quot;55701&quot;     &quot;51222&quot;     &quot;11198&quot;     &quot;57680&quot;     &quot;9878&quot;     
[1861] &quot;56339&quot;     &quot;6297&quot;      &quot;5018&quot;      &quot;9056&quot;      &quot;122704&quot;    &quot;4323&quot;     
[1867] &quot;55147&quot;     &quot;10419&quot;     &quot;84962&quot;     &quot;22985&quot;     &quot;90673&quot;     &quot;105378176&quot;
[1873] &quot;8106&quot;      &quot;51310&quot;     &quot;10278&quot;     &quot;79178&quot;     &quot;8906&quot;      &quot;10901&quot;    
[1879] &quot;317749&quot;    &quot;728635&quot;    &quot;4901&quot;      &quot;80344&quot;     &quot;5720&quot;      &quot;51016&quot;    
[1885] &quot;55072&quot;     &quot;10379&quot;     &quot;10548&quot;     &quot;283629&quot;    &quot;145553&quot;    &quot;51292&quot;    
[1891] &quot;7051&quot;      &quot;161424&quot;    &quot;57523&quot;     &quot;23351&quot;     &quot;643866&quot;    &quot;29091&quot;    
[1897] &quot;4857&quot;      &quot;5587&quot;      &quot;23256&quot;     &quot;1690&quot;      &quot;29966&quot;     &quot;25938&quot;    
[1903] &quot;112487&quot;    &quot;80224&quot;     &quot;394&quot;       &quot;101954278&quot; &quot;112399&quot;    &quot;171546&quot;   
[1909] &quot;55837&quot;     &quot;1073&quot;      &quot;6729&quot;      &quot;283635&quot;    &quot;5687&quot;      &quot;4792&quot;     
[1915] &quot;51562&quot;     &quot;10484&quot;     &quot;122553&quot;    &quot;5411&quot;      &quot;4253&quot;      &quot;23116&quot;    
[1921] &quot;2287&quot;      &quot;57697&quot;     &quot;6235&quot;      &quot;6029&quot;      &quot;122769&quot;    &quot;6166&quot;     
[1927] &quot;4247&quot;      &quot;55172&quot;     &quot;122773&quot;    &quot;378706&quot;    &quot;382&quot;       &quot;6655&quot;     
[1933] &quot;27109&quot;     &quot;11183&quot;     &quot;51062&quot;     &quot;51199&quot;     &quot;114088&quot;    &quot;81542&quot;    
[1939] &quot;51637&quot;     &quot;30001&quot;     &quot;5706&quot;      &quot;6815&quot;      &quot;80821&quot;     &quot;23034&quot;    
[1945] &quot;93487&quot;     &quot;100129075&quot; &quot;54916&quot;     &quot;10640&quot;     &quot;122830&quot;    &quot;145407&quot;   
[1951] &quot;55860&quot;     &quot;5926&quot;      &quot;51339&quot;     &quot;112849&quot;    &quot;51528&quot;     &quot;6252&quot;     
[1957] &quot;51635&quot;     &quot;57570&quot;     &quot;145389&quot;    &quot;3091&quot;      &quot;81537&quot;     &quot;23224&quot;    
[1963] &quot;9495&quot;      &quot;22890&quot;     &quot;3306&quot;      &quot;91612&quot;     &quot;376267&quot;    &quot;2530&quot;     
[1969] &quot;10243&quot;     &quot;64398&quot;     &quot;1965&quot;      &quot;26499&quot;     &quot;57475&quot;     &quot;384&quot;      
[1975] &quot;10490&quot;     &quot;23503&quot;     &quot;677&quot;       &quot;87&quot;        &quot;8816&quot;      &quot;57452&quot;    
[1981] &quot;2079&quot;      &quot;55334&quot;     &quot;51241&quot;     &quot;10001&quot;     &quot;23508&quot;     &quot;4293&quot;     
[1987] &quot;22990&quot;     &quot;26094&quot;     &quot;58517&quot;     &quot;89932&quot;     &quot;79697&quot;     &quot;641371&quot;   
[1993] &quot;10965&quot;     &quot;83544&quot;     &quot;9240&quot;      &quot;91748&quot;     &quot;100506476&quot; &quot;145482&quot;   
[1999] &quot;57862&quot;     &quot;145483&quot;    &quot;51004&quot;     &quot;4329&quot;      &quot;91750&quot;     &quot;10577&quot;    
[2005] &quot;9870&quot;      &quot;1743&quot;      &quot;97&quot;        &quot;79696&quot;     &quot;91754&quot;     &quot;10972&quot;    
[2011] &quot;55640&quot;     &quot;7043&quot;      &quot;112752&quot;    &quot;2103&quot;      &quot;23357&quot;     &quot;85457&quot;    
[2017] &quot;57156&quot;     &quot;63894&quot;     &quot;81892&quot;     &quot;57143&quot;     &quot;6400&quot;      &quot;23768&quot;    
[2023] &quot;2581&quot;      &quot;55812&quot;     &quot;79882&quot;     &quot;161436&quot;    &quot;1112&quot;      &quot;55775&quot;    
[2029] &quot;56659&quot;     &quot;5700&quot;      &quot;801&quot;       &quot;9321&quot;      &quot;4287&quot;      &quot;4707&quot;     
[2035] &quot;53981&quot;     &quot;79890&quot;     &quot;5641&quot;      &quot;26175&quot;     &quot;84520&quot;     &quot;55148&quot;    
[2041] &quot;55727&quot;     &quot;78990&quot;     &quot;122509&quot;    &quot;5265&quot;      &quot;23405&quot;     &quot;79789&quot;    
[2047] &quot;51218&quot;     &quot;56967&quot;     &quot;55102&quot;     &quot;51527&quot;     &quot;64919&quot;     &quot;84193&quot;    
[2053] &quot;317762&quot;    &quot;84439&quot;     &quot;2009&quot;      &quot;51466&quot;     &quot;123096&quot;    &quot;7453&quot;     
[2059] &quot;79446&quot;     &quot;8788&quot;      &quot;55384&quot;     &quot;1778&quot;      &quot;3320&quot;      &quot;91833&quot;    
[2065] &quot;122416&quot;    &quot;7187&quot;      &quot;9578&quot;      &quot;4140&quot;      &quot;1152&quot;      &quot;84334&quot;    
[2071] &quot;3831&quot;      &quot;23368&quot;     &quot;9556&quot;      &quot;374569&quot;    &quot;26153&quot;     &quot;10572&quot;    
[2077] &quot;113146&quot;    &quot;3714&quot;      &quot;256281&quot;    &quot;2972&quot;      &quot;90135&quot;     &quot;647310&quot;   
[2083] &quot;100507437&quot; &quot;283755&quot;    &quot;646243&quot;    &quot;100132101&quot; &quot;339005&quot;    &quot;81614&quot;    
[2089] &quot;23191&quot;     &quot;114791&quot;    &quot;4692&quot;      &quot;347746&quot;    &quot;8123&quot;      &quot;100033416&quot;
[2095] &quot;100033432&quot; &quot;3653&quot;      &quot;7337&quot;      &quot;440248&quot;    &quot;440253&quot;    &quot;646278&quot;   
[2101] &quot;321&quot;       &quot;56160&quot;     &quot;653075&quot;    &quot;101059918&quot; &quot;728498&quot;    &quot;54893&quot;    
[2107] &quot;51621&quot;     &quot;653125&quot;    &quot;728047&quot;    &quot;100288615&quot; &quot;100996255&quot; &quot;643699&quot;   
[2113] &quot;342184&quot;    &quot;6263&quot;      &quot;57099&quot;     &quot;56851&quot;     &quot;161779&quot;    &quot;51234&quot;    
[2119] &quot;55505&quot;     &quot;254531&quot;    &quot;101928174&quot; &quot;9716&quot;      &quot;54989&quot;     &quot;283742&quot;   
[2125] &quot;10125&quot;     &quot;440275&quot;    &quot;6727&quot;      &quot;100131244&quot; &quot;100505573&quot; &quot;85455&quot;    
[2131] &quot;3712&quot;      &quot;90416&quot;     &quot;27079&quot;     &quot;100505648&quot; &quot;643338&quot;    &quot;84936&quot;    
[2137] &quot;54617&quot;     &quot;11261&quot;     &quot;51103&quot;     &quot;4058&quot;      &quot;23269&quot;     &quot;23005&quot;    
[2143] &quot;51332&quot;     &quot;825&quot;       &quot;255252&quot;    &quot;55142&quot;     &quot;57519&quot;     &quot;146059&quot;   
[2149] &quot;197131&quot;    &quot;23582&quot;     &quot;9836&quot;      &quot;161823&quot;    &quot;146050&quot;    &quot;27229&quot;    
[2155] &quot;9677&quot;      &quot;1159&quot;      &quot;117155&quot;    &quot;548596&quot;    &quot;2923&quot;      &quot;10169&quot;    
[2161] &quot;4236&quot;      &quot;84978&quot;     &quot;728758&quot;    &quot;113201&quot;    &quot;51496&quot;     &quot;645212&quot;   
[2167] &quot;8669&quot;      &quot;80208&quot;     &quot;100419583&quot; &quot;6652&quot;      &quot;2628&quot;      &quot;7782&quot;     
[2173] &quot;26258&quot;     &quot;58472&quot;     &quot;80031&quot;     &quot;1854&quot;      &quot;2200&quot;      &quot;23741&quot;    
[2179] &quot;9318&quot;      &quot;2585&quot;      &quot;2252&quot;      &quot;2553&quot;      &quot;9101&quot;      &quot;54822&quot;    
[2185] &quot;84888&quot;     &quot;23431&quot;     &quot;123169&quot;    &quot;10681&quot;     &quot;55930&quot;     &quot;51187&quot;    
[2191] &quot;5873&quot;      &quot;145788&quot;    &quot;4734&quot;      &quot;55329&quot;     &quot;6938&quot;      &quot;81488&quot;    
[2197] &quot;102&quot;       &quot;79811&quot;     &quot;54778&quot;     &quot;145773&quot;    &quot;2958&quot;      &quot;663&quot;      
[2203] &quot;302&quot;       &quot;6095&quot;      &quot;54832&quot;     &quot;83464&quot;     &quot;9960&quot;      &quot;23604&quot;    
[2209] &quot;84191&quot;     &quot;6642&quot;      &quot;79856&quot;     &quot;5479&quot;      &quot;23060&quot;     &quot;4947&quot;     
[2215] &quot;348093&quot;    &quot;51324&quot;     &quot;123263&quot;    &quot;390594&quot;    &quot;10845&quot;     &quot;54956&quot;    
[2221] &quot;9543&quot;      &quot;54878&quot;     &quot;51495&quot;     &quot;10260&quot;     &quot;8766&quot;      &quot;115752&quot;   
[2227] &quot;5604&quot;      &quot;10302&quot;     &quot;6124&quot;      &quot;79719&quot;     &quot;64799&quot;     &quot;100506686&quot;
[2233] &quot;145853&quot;    &quot;5607&quot;      &quot;8554&quot;      &quot;91860&quot;     &quot;10116&quot;     &quot;22801&quot;    
[2239] &quot;26035&quot;     &quot;54852&quot;     &quot;6176&quot;      &quot;55075&quot;     &quot;55323&quot;     &quot;79875&quot;    
[2245] &quot;4649&quot;      &quot;123228&quot;    &quot;56965&quot;     &quot;25820&quot;     &quot;83440&quot;     &quot;4756&quot;     
[2251] &quot;10021&quot;     &quot;27020&quot;     &quot;4016&quot;      &quot;64220&quot;     &quot;1583&quot;      &quot;8482&quot;     
[2257] &quot;84993&quot;     &quot;440288&quot;    &quot;80153&quot;     &quot;1543&quot;      &quot;1445&quot;      &quot;25989&quot;    
[2263] &quot;10066&quot;     &quot;4351&quot;      &quot;60490&quot;     &quot;56905&quot;     &quot;54939&quot;     &quot;79661&quot;    
[2269] &quot;4123&quot;      &quot;25942&quot;     &quot;10073&quot;     &quot;257364&quot;    &quot;2108&quot;      &quot;5955&quot;     
[2275] &quot;10099&quot;     &quot;79834&quot;     &quot;10363&quot;     &quot;23102&quot;     &quot;10518&quot;     &quot;55466&quot;    
[2281] &quot;80349&quot;     &quot;3658&quot;      &quot;1138&quot;      &quot;1136&quot;      &quot;10933&quot;     &quot;1512&quot;     
[2287] &quot;23423&quot;     &quot;23251&quot;     &quot;2184&quot;      &quot;9915&quot;      &quot;57214&quot;     &quot;79631&quot;    
[2293] &quot;647042&quot;    &quot;100134869&quot; &quot;440295&quot;    &quot;80154&quot;     &quot;642402&quot;    &quot;6218&quot;     
[2299] &quot;100505616&quot; &quot;9455&quot;      &quot;123207&quot;    &quot;53339&quot;     &quot;53346&quot;     &quot;57188&quot;    
[2305] &quot;440300&quot;    &quot;388152&quot;    &quot;643707&quot;    &quot;374650&quot;    &quot;388165&quot;    &quot;54993&quot;    
[2311] &quot;54581&quot;     &quot;84942&quot;     &quot;23478&quot;     &quot;9640&quot;      &quot;57538&quot;     &quot;642423&quot;   
[2317] &quot;101929479&quot; &quot;100133220&quot; &quot;4916&quot;      &quot;26589&quot;     &quot;64963&quot;     &quot;55070&quot;    
[2323] &quot;8800&quot;      &quot;290&quot;       &quot;348110&quot;    &quot;374655&quot;    &quot;3418&quot;      &quot;10509&quot;    
[2329] &quot;390637&quot;    &quot;8826&quot;      &quot;2242&quot;      &quot;4122&quot;      &quot;55898&quot;     &quot;26276&quot;    
[2335] &quot;8128&quot;      &quot;100507217&quot; &quot;440311&quot;    &quot;7026&quot;      &quot;91947&quot;     &quot;23336&quot;    
[2341] &quot;123355&quot;    &quot;7190&quot;      &quot;4205&quot;      &quot;145748&quot;    &quot;441733&quot;    &quot;55180&quot;    
[2347] &quot;55829&quot;     &quot;80213&quot;     &quot;123283&quot;    &quot;100128108&quot; &quot;55692&quot;     &quot;83986&quot;    
[2353] &quot;398&quot;       &quot;64714&quot;     &quot;9727&quot;      &quot;6650&quot;      &quot;9091&quot;      &quot;84326&quot;    
[2359] &quot;84331&quot;     &quot;197335&quot;    &quot;9028&quot;      &quot;10273&quot;     &quot;339123&quot;    &quot;84219&quot;    
[2365] &quot;146330&quot;    &quot;124093&quot;    &quot;64428&quot;     &quot;8912&quot;      &quot;8938&quot;      &quot;64718&quot;    
[2371] &quot;283951&quot;    &quot;9894&quot;      &quot;9742&quot;      &quot;90861&quot;     &quot;23162&quot;     &quot;197342&quot;   
[2377] &quot;90864&quot;     &quot;3029&quot;      &quot;6187&quot;      &quot;10607&quot;     &quot;2671&quot;      &quot;9143&quot;     
[2383] &quot;90850&quot;     &quot;7249&quot;      &quot;5310&quot;      &quot;25837&quot;     &quot;100507303&quot; &quot;84231&quot;    
[2389] &quot;57524&quot;     &quot;283870&quot;    &quot;1775&quot;      &quot;4917&quot;      &quot;57465&quot;     &quot;51005&quot;    
[2395] &quot;752014&quot;    &quot;83886&quot;     &quot;23524&quot;     &quot;6923&quot;      &quot;9074&quot;      &quot;64386&quot;    
[2401] &quot;100507419&quot; &quot;9235&quot;      &quot;7752&quot;      &quot;91151&quot;     &quot;7727&quot;      &quot;146434&quot;   
[2407] &quot;197358&quot;    &quot;1387&quot;      &quot;115&quot;       &quot;7023&quot;      &quot;84662&quot;     &quot;79585&quot;    
[2413] &quot;9093&quot;      &quot;57407&quot;     &quot;3163&quot;      &quot;342346&quot;    &quot;23295&quot;     &quot;84309&quot;    
[2419] &quot;124401&quot;    &quot;26048&quot;     &quot;29855&quot;     &quot;5493&quot;      &quot;54715&quot;     &quot;79091&quot;    
[2425] &quot;25880&quot;     &quot;23589&quot;     &quot;7874&quot;      &quot;2903&quot;      &quot;80063&quot;     &quot;780776&quot;   
[2431] &quot;23274&quot;     &quot;29066&quot;     &quot;2935&quot;      &quot;729993&quot;    &quot;2072&quot;      &quot;57496&quot;    
[2437] &quot;51283&quot;     &quot;642778&quot;    &quot;23420&quot;     &quot;100288162&quot; &quot;23042&quot;     &quot;54700&quot;    
[2443] &quot;353511&quot;    &quot;105369154&quot; &quot;100288332&quot; &quot;89927&quot;     &quot;4629&quot;      &quot;4363&quot;     
[2449] &quot;408050&quot;    &quot;102465683&quot; &quot;102723728&quot; &quot;283820&quot;    &quot;6210&quot;      &quot;23049&quot;    
[2455] &quot;10229&quot;     &quot;162073&quot;    &quot;51573&quot;     &quot;57020&quot;     &quot;124152&quot;    &quot;51704&quot;    
[2461] &quot;55623&quot;     &quot;57146&quot;     &quot;23117&quot;     &quot;100190986&quot; &quot;440345&quot;    &quot;7385&quot;     
[2467] &quot;255762&quot;    &quot;730094&quot;    &quot;29904&quot;     &quot;55718&quot;     &quot;1039&quot;      &quot;100131998&quot;
[2473] &quot;641298&quot;    &quot;100132247&quot; &quot;91949&quot;     &quot;23062&quot;     &quot;124454&quot;    &quot;4706&quot;     
[2479] &quot;79728&quot;     &quot;5930&quot;      &quot;27327&quot;     &quot;55114&quot;     &quot;102723510&quot; &quot;51451&quot;    
[2485] &quot;342357&quot;    &quot;79831&quot;     &quot;197370&quot;    &quot;2975&quot;      &quot;23247&quot;     &quot;728689&quot;   
[2491] &quot;112869&quot;    &quot;8663&quot;      &quot;11273&quot;     &quot;25970&quot;     &quot;100289092&quot; &quot;84901&quot;    
[2497] &quot;83985&quot;     &quot;27040&quot;     &quot;552900&quot;    &quot;440354&quot;    &quot;4150&quot;      &quot;112476&quot;   
[2503] &quot;9961&quot;      &quot;283899&quot;    &quot;8448&quot;      &quot;226&quot;       &quot;5531&quot;      &quot;83719&quot;    
[2509] &quot;654483&quot;    &quot;548593&quot;    &quot;100526830&quot; &quot;613037&quot;    &quot;595101&quot;    &quot;105371169&quot;
[2515] &quot;1731&quot;      &quot;197407&quot;    &quot;51333&quot;     &quot;79724&quot;     &quot;65988&quot;     &quot;115509&quot;   
[2521] &quot;730183&quot;    &quot;9810&quot;      &quot;9274&quot;      &quot;1489&quot;      &quot;93129&quot;     &quot;112755&quot;   
[2527] &quot;6810&quot;      &quot;339105&quot;    &quot;79001&quot;     &quot;84148&quot;     &quot;5652&quot;      &quot;2521&quot;     
[2533] &quot;283933&quot;    &quot;64755&quot;     &quot;124411&quot;    &quot;606551&quot;    &quot;84706&quot;     &quot;10294&quot;    
[2539] &quot;81533&quot;     &quot;83752&quot;     &quot;100507577&quot; &quot;6477&quot;      &quot;9683&quot;      &quot;869&quot;      
[2545] &quot;255919&quot;    &quot;113&quot;       &quot;29117&quot;     &quot;80205&quot;     &quot;5934&quot;      &quot;64400&quot;    
[2551] &quot;643911&quot;    &quot;4313&quot;      &quot;54947&quot;     &quot;1066&quot;      &quot;2775&quot;      &quot;267&quot;      
[2557] &quot;11051&quot;     &quot;583&quot;       &quot;9688&quot;      &quot;84166&quot;     &quot;221184&quot;    &quot;89970&quot;    
[2563] &quot;51090&quot;     &quot;6376&quot;      &quot;57019&quot;     &quot;57017&quot;     &quot;5432&quot;      &quot;55715&quot;    
[2569] &quot;10300&quot;     &quot;3801&quot;      &quot;29105&quot;     &quot;65009&quot;     &quot;23019&quot;     &quot;55238&quot;    
[2575] &quot;644649&quot;    &quot;123920&quot;    &quot;1783&quot;      &quot;57546&quot;     &quot;51647&quot;     &quot;283848&quot;   
[2581] &quot;80262&quot;     &quot;84752&quot;     &quot;8717&quot;      &quot;55336&quot;     &quot;3299&quot;      &quot;1874&quot;     
[2587] &quot;79767&quot;     &quot;29109&quot;     &quot;25894&quot;     &quot;79567&quot;     &quot;146206&quot;    &quot;50855&quot;    
[2593] &quot;388284&quot;    &quot;81577&quot;     &quot;57215&quot;     &quot;10204&quot;     &quot;5681&quot;      &quot;3931&quot;     
[2599] &quot;6560&quot;      &quot;9057&quot;      &quot;84138&quot;     &quot;79613&quot;     &quot;84916&quot;     &quot;84342&quot;    
[2605] &quot;64146&quot;     &quot;7014&quot;      &quot;80777&quot;     &quot;283970&quot;    &quot;55066&quot;     &quot;100506060&quot;
[2611] &quot;118460&quot;    &quot;11269&quot;     &quot;100506083&quot; &quot;55308&quot;     &quot;6483&quot;      &quot;25839&quot;    
[2617] &quot;92154&quot;     &quot;55697&quot;     &quot;55783&quot;     &quot;794&quot;       &quot;164&quot;       &quot;463&quot;      
[2623] &quot;5713&quot;      &quot;283922&quot;    &quot;440348&quot;    &quot;2734&quot;      &quot;55159&quot;     &quot;79726&quot;    
[2629] &quot;9564&quot;      &quot;124491&quot;    &quot;4166&quot;      &quot;3735&quot;      &quot;170692&quot;    &quot;283927&quot;   
[2635] &quot;57687&quot;     &quot;123775&quot;    &quot;2653&quot;      &quot;114780&quot;    &quot;5336&quot;      &quot;1012&quot;     
[2641] &quot;105371371&quot; &quot;29948&quot;     &quot;146167&quot;    &quot;23406&quot;     &quot;79786&quot;     &quot;55625&quot;    
[2647] &quot;23199&quot;     &quot;404550&quot;    &quot;64779&quot;     &quot;2303&quot;      &quot;100506581&quot; &quot;79791&quot;    
[2653] &quot;57338&quot;     &quot;54758&quot;     &quot;105371397&quot; &quot;8140&quot;      &quot;54971&quot;     &quot;84627&quot;    
[2659] &quot;161882&quot;    &quot;197187&quot;    &quot;348180&quot;    &quot;353&quot;       &quot;2588&quot;      &quot;863&quot;      
[2665] &quot;283860&quot;    &quot;197320&quot;    &quot;29123&quot;     &quot;6687&quot;      &quot;6137&quot;      &quot;5119&quot;     
[2671] &quot;8558&quot;      &quot;124044&quot;    &quot;9605&quot;      &quot;100128881&quot; &quot;92822&quot;     &quot;22980&quot;    
[2677] &quot;92806&quot;     &quot;172&quot;       &quot;2622&quot;      &quot;9501&quot;      &quot;400566&quot;    &quot;51031&quot;    
[2683] &quot;55178&quot;     &quot;64359&quot;     &quot;29928&quot;     &quot;29&quot;        &quot;7531&quot;      &quot;100306951&quot;
[2689] &quot;8578&quot;      &quot;83547&quot;     &quot;10594&quot;     &quot;84981&quot;     &quot;5176&quot;      &quot;114826&quot;   
[2695] &quot;124641&quot;    &quot;63826&quot;     &quot;55720&quot;     &quot;9905&quot;      &quot;79066&quot;     &quot;5048&quot;     
[2701] &quot;23277&quot;     &quot;23108&quot;     &quot;1497&quot;      &quot;55421&quot;     &quot;84254&quot;     &quot;23140&quot;    
[2707] &quot;124936&quot;    &quot;7326&quot;      &quot;342527&quot;    &quot;246&quot;       &quot;27043&quot;     &quot;409&quot;      
[2713] &quot;400569&quot;    &quot;5694&quot;      &quot;5338&quot;      &quot;50488&quot;     &quot;2811&quot;      &quot;26001&quot;    
[2719] &quot;5216&quot;      &quot;9552&quot;      &quot;23125&quot;     &quot;10749&quot;     &quot;124961&quot;    &quot;7775&quot;     
[2725] &quot;84622&quot;     &quot;9135&quot;      &quot;84268&quot;     &quot;56919&quot;     &quot;51009&quot;     &quot;22861&quot;    
[2731] &quot;23302&quot;     &quot;83394&quot;     &quot;84817&quot;     &quot;51003&quot;     &quot;388327&quot;    &quot;201232&quot;   
[2737] &quot;432&quot;       &quot;1856&quot;      &quot;79142&quot;     &quot;23399&quot;     &quot;23587&quot;     &quot;1984&quot;     
[2743] &quot;9744&quot;      &quot;8711&quot;      &quot;8742&quot;      &quot;26168&quot;     &quot;26781&quot;     &quot;102724164&quot;
[2749] &quot;9526&quot;      &quot;9513&quot;      &quot;55135&quot;     &quot;84316&quot;     &quot;1107&quot;      &quot;9196&quot;     
[2755] &quot;116840&quot;    &quot;6844&quot;      &quot;84314&quot;     &quot;6154&quot;      &quot;4628&quot;      &quot;8522&quot;     
[2761] &quot;4621&quot;      &quot;100128006&quot; &quot;60528&quot;     &quot;1352&quot;      &quot;201158&quot;    &quot;54902&quot;    
[2767] &quot;9611&quot;      &quot;125144&quot;    &quot;26800&quot;     &quot;388341&quot;    &quot;57336&quot;     &quot;56953&quot;    
[2773] &quot;10743&quot;     &quot;6720&quot;      &quot;146691&quot;    &quot;2314&quot;      &quot;125170&quot;    &quot;7156&quot;     
[2779] &quot;2310&quot;      &quot;51030&quot;     &quot;27077&quot;     &quot;5598&quot;      &quot;4239&quot;      &quot;9706&quot;     
[2785] &quot;11216&quot;     &quot;440416&quot;    &quot;105369205&quot; &quot;3768&quot;      &quot;339263&quot;    &quot;100462977&quot;
[2791] &quot;26118&quot;     &quot;7448&quot;      &quot;113235&quot;    &quot;9094&quot;      &quot;94005&quot;     &quot;124923&quot;   
[2797] &quot;6388&quot;      &quot;147011&quot;    &quot;83871&quot;     &quot;55731&quot;     &quot;26284&quot;     &quot;2319&quot;     
[2803] &quot;147015&quot;    &quot;57649&quot;     &quot;124925&quot;    &quot;51268&quot;     &quot;57532&quot;     &quot;57551&quot;    
[2809] &quot;84940&quot;     &quot;642&quot;       &quot;1362&quot;      &quot;9527&quot;      &quot;147172&quot;    &quot;440423&quot;   
[2815] &quot;79736&quot;     &quot;55813&quot;     &quot;23512&quot;     &quot;7756&quot;      &quot;5717&quot;      &quot;4642&quot;     
[2821] &quot;26022&quot;     &quot;10693&quot;     &quot;91603&quot;     &quot;117584&quot;    &quot;55106&quot;     &quot;163&quot;      
[2827] &quot;8148&quot;      &quot;414062&quot;    &quot;102724862&quot; &quot;101060321&quot; &quot;9326&quot;      &quot;80179&quot;    
[2833] &quot;31&quot;        &quot;6871&quot;      &quot;11072&quot;     &quot;11056&quot;     &quot;101060376&quot; &quot;101060389&quot;
[2839] &quot;414060&quot;    &quot;101929950&quot; &quot;102723859&quot; &quot;729873&quot;    &quot;440434&quot;    &quot;84311&quot;    
[2845] &quot;80725&quot;     &quot;4302&quot;      &quot;9349&quot;      &quot;3927&quot;      &quot;100505576&quot; &quot;6143&quot;     
[2851] &quot;5469&quot;      &quot;84152&quot;     &quot;10948&quot;     &quot;5409&quot;      &quot;93210&quot;     &quot;84299&quot;    
[2857] &quot;2886&quot;      &quot;94103&quot;     &quot;339287&quot;    &quot;22794&quot;     &quot;51195&quot;     &quot;147179&quot;   
[2863] &quot;6605&quot;      &quot;3858&quot;      &quot;147184&quot;    &quot;3880&quot;      &quot;10209&quot;     &quot;55175&quot;    
[2869] &quot;83538&quot;     &quot;28511&quot;     &quot;2648&quot;      &quot;5878&quot;      &quot;6777&quot;      &quot;6776&quot;     
[2875] &quot;284119&quot;    &quot;535&quot;       &quot;4669&quot;      &quot;80347&quot;     &quot;7283&quot;      &quot;79990&quot;    
[2881] &quot;2826&quot;      &quot;84313&quot;     &quot;65266&quot;     &quot;10197&quot;     &quot;146923&quot;    &quot;6155&quot;     
[2887] &quot;3430&quot;      &quot;10493&quot;     &quot;8153&quot;      &quot;10230&quot;     &quot;101929767&quot; &quot;4077&quot;     
[2893] &quot;379&quot;       &quot;1845&quot;      &quot;4355&quot;      &quot;84336&quot;     &quot;124801&quot;    &quot;92579&quot;    
[2899] &quot;10014&quot;     &quot;92591&quot;     &quot;56970&quot;     &quot;101926996&quot; &quot;10900&quot;     &quot;284069&quot;   
[2905] &quot;23131&quot;     &quot;124808&quot;    &quot;80174&quot;     &quot;4185&quot;      &quot;79877&quot;     &quot;79777&quot;    
[2911] &quot;9020&quot;      &quot;9842&quot;      &quot;55073&quot;     &quot;401884&quot;    &quot;147081&quot;    &quot;4137&quot;     
[2917] &quot;284058&quot;    &quot;9884&quot;      &quot;100506084&quot; &quot;51326&quot;     &quot;474170&quot;    &quot;9570&quot;     
[2923] &quot;653479&quot;    &quot;9755&quot;      &quot;124995&quot;    &quot;102724532&quot; &quot;51226&quot;     &quot;4779&quot;     
[2929] &quot;10951&quot;     &quot;29916&quot;     &quot;3213&quot;      &quot;3214&quot;      &quot;11267&quot;     &quot;10642&quot;    
[2935] &quot;22834&quot;     &quot;5245&quot;      &quot;4804&quot;      &quot;10237&quot;     &quot;3675&quot;      &quot;201191&quot;   
[2941] &quot;1277&quot;      &quot;51264&quot;     &quot;55379&quot;     &quot;64847&quot;     &quot;8913&quot;      &quot;91369&quot;    
[2947] &quot;10140&quot;     &quot;9043&quot;      &quot;4830&quot;      &quot;54799&quot;     &quot;1353&quot;      &quot;252983&quot;   
[2953] &quot;8161&quot;      &quot;59342&quot;     &quot;124540&quot;    &quot;51649&quot;     &quot;101929185&quot; &quot;140735&quot;   
[2959] &quot;9110&quot;      &quot;5414&quot;      &quot;55181&quot;     &quot;388403&quot;    &quot;79665&quot;     &quot;51651&quot;    
[2965] &quot;81671&quot;     &quot;6198&quot;      &quot;101927755&quot; &quot;653645&quot;    &quot;63897&quot;     &quot;653653&quot;   
[2971] &quot;342538&quot;    &quot;57508&quot;     &quot;9902&quot;      &quot;51204&quot;     &quot;92335&quot;     &quot;57003&quot;    
[2977] &quot;11325&quot;     &quot;117246&quot;    &quot;5705&quot;      &quot;2081&quot;      &quot;692227&quot;    &quot;64750&quot;    
[2983] &quot;5578&quot;      &quot;27092&quot;     &quot;5718&quot;      &quot;25926&quot;     &quot;284018&quot;    &quot;51321&quot;    
[2989] &quot;22901&quot;     &quot;5573&quot;      &quot;54757&quot;     &quot;201266&quot;    &quot;6169&quot;      &quot;94015&quot;    
[2995] &quot;26151&quot;     &quot;54868&quot;     &quot;3396&quot;      &quot;6613&quot;      &quot;51081&quot;     &quot;57409&quot;    
[3001] &quot;100287042&quot; &quot;60386&quot;     &quot;2885&quot;      &quot;9772&quot;      &quot;283989&quot;    &quot;3993&quot;     
[3007] &quot;80022&quot;     &quot;9400&quot;      &quot;643008&quot;    &quot;3691&quot;      &quot;2584&quot;      &quot;23558&quot;    
[3013] &quot;91107&quot;     &quot;64978&quot;     &quot;85302&quot;     &quot;51&quot;        &quot;6730&quot;      &quot;114804&quot;   
[3019] &quot;5635&quot;      &quot;63893&quot;     &quot;100507246&quot; &quot;439921&quot;    &quot;23210&quot;     &quot;79157&quot;    
[3025] &quot;6397&quot;      &quot;10801&quot;     &quot;100131096&quot; &quot;11322&quot;     &quot;9144&quot;      &quot;9489&quot;     
[3031] &quot;7077&quot;      &quot;124583&quot;    &quot;114897&quot;    &quot;64772&quot;     &quot;84733&quot;     &quot;57332&quot;    
[3037] &quot;8535&quot;      &quot;125058&quot;    &quot;55036&quot;     &quot;6448&quot;      &quot;284131&quot;    &quot;100129503&quot;
[3043] &quot;284184&quot;    &quot;124565&quot;    &quot;57597&quot;     &quot;25794&quot;     &quot;80233&quot;     &quot;55666&quot;    
[3049] &quot;83882&quot;     &quot;339231&quot;    &quot;348262&quot;    &quot;5034&quot;      &quot;396&quot;       &quot;51529&quot;    
[3055] &quot;5833&quot;      &quot;79058&quot;     &quot;201254&quot;    &quot;2873&quot;      &quot;64118&quot;     &quot;9123&quot;     
[3061] &quot;284004&quot;    &quot;79415&quot;     &quot;3607&quot;      &quot;56270&quot;     &quot;10966&quot;     &quot;6904&quot;     
[3067] &quot;284207&quot;    &quot;727758&quot;    &quot;81035&quot;     &quot;84034&quot;     &quot;727896&quot;    &quot;10627&quot;    
[3073] &quot;339290&quot;    &quot;7541&quot;      &quot;23136&quot;     &quot;645369&quot;    &quot;79822&quot;     &quot;5797&quot;     
[3079] &quot;57045&quot;     &quot;10928&quot;     &quot;9989&quot;      &quot;9218&quot;      &quot;147495&quot;    &quot;63895&quot;    
[3085] &quot;65258&quot;     &quot;84617&quot;     &quot;10650&quot;     &quot;56907&quot;     &quot;56984&quot;     &quot;79959&quot;    
[3091] &quot;5771&quot;      &quot;81929&quot;     &quot;8731&quot;      &quot;440482&quot;    &quot;6093&quot;      &quot;80000&quot;    
[3097] &quot;114799&quot;    &quot;100128893&quot; &quot;64693&quot;     &quot;5932&quot;      &quot;85019&quot;     &quot;147463&quot;   
[3103] &quot;3909&quot;      &quot;6875&quot;      &quot;1829&quot;      &quot;9331&quot;      &quot;22878&quot;     &quot;54941&quot;    
[3109] &quot;51444&quot;     &quot;64762&quot;     &quot;441818&quot;    &quot;10982&quot;     &quot;7572&quot;      &quot;2589&quot;     
[3115] &quot;83608&quot;     &quot;25800&quot;     &quot;55250&quot;     &quot;55034&quot;     &quot;57724&quot;     &quot;115106&quot;   
[3121] &quot;494470&quot;    &quot;29906&quot;     &quot;9063&quot;      &quot;83473&quot;     &quot;84064&quot;     &quot;51124&quot;    
[3127] &quot;4087&quot;      &quot;9811&quot;      &quot;54808&quot;     &quot;497661&quot;    &quot;6139&quot;      &quot;10449&quot;    
[3133] &quot;4152&quot;      &quot;5596&quot;      &quot;4200&quot;      &quot;55520&quot;     &quot;4089&quot;      &quot;8932&quot;     
[3139] &quot;5874&quot;      &quot;80323&quot;     &quot;6925&quot;      &quot;9352&quot;      &quot;2235&quot;      &quot;100505549&quot;
[3145] &quot;5205&quot;      &quot;23327&quot;     &quot;10892&quot;     &quot;55205&quot;     &quot;90701&quot;     &quot;3998&quot;     
[3151] &quot;147372&quot;    &quot;23556&quot;     &quot;57614&quot;     &quot;8792&quot;      &quot;596&quot;       &quot;5055&quot;     
[3157] &quot;54495&quot;     &quot;220164&quot;    &quot;25914&quot;     &quot;81832&quot;     &quot;400657&quot;    &quot;284273&quot;   
[3163] &quot;10194&quot;     &quot;7776&quot;      &quot;4772&quot;      &quot;80148&quot;     &quot;440498&quot;    &quot;10907&quot;    
[3169] &quot;22850&quot;     &quot;375690&quot;    &quot;54531&quot;     &quot;91978&quot;     &quot;997&quot;       &quot;682&quot;      
[3175] &quot;55658&quot;     &quot;5064&quot;      &quot;79948&quot;     &quot;10025&quot;     &quot;91300&quot;     &quot;1820&quot;     
[3181] &quot;116444&quot;    &quot;91304&quot;     &quot;103625684&quot; &quot;10347&quot;     &quot;5434&quot;      &quot;2879&quot;     
[3187] &quot;22904&quot;     &quot;255057&quot;    &quot;90007&quot;     &quot;148046&quot;    &quot;2593&quot;      &quot;148223&quot;   
[3193] &quot;54760&quot;     &quot;399664&quot;    &quot;10975&quot;     &quot;6929&quot;      &quot;100288123&quot; &quot;81926&quot;    
[3199] &quot;113178&quot;    &quot;84444&quot;     &quot;8175&quot;      &quot;4946&quot;      &quot;51690&quot;     &quot;56928&quot;    
[3205] &quot;26517&quot;     &quot;6449&quot;      &quot;115196&quot;    &quot;58492&quot;     &quot;56926&quot;     &quot;51343&quot;    
[3211] &quot;10362&quot;     &quot;9546&quot;      &quot;116541&quot;    &quot;23217&quot;     &quot;693222&quot;    &quot;1938&quot;     
[3217] &quot;51588&quot;     &quot;5605&quot;      &quot;51548&quot;     &quot;170961&quot;    &quot;55702&quot;     &quot;79187&quot;    
[3223] &quot;55620&quot;     &quot;84954&quot;     &quot;6455&quot;      &quot;80700&quot;     &quot;84717&quot;     &quot;56005&quot;    
[3229] &quot;91039&quot;     &quot;100131094&quot; &quot;10226&quot;     &quot;23030&quot;     &quot;9667&quot;      &quot;6294&quot;     
[3235] &quot;374875&quot;    &quot;25873&quot;     &quot;9361&quot;      &quot;163154&quot;    &quot;56931&quot;     &quot;126328&quot;   
[3241] &quot;400673&quot;    &quot;828&quot;       &quot;8498&quot;      &quot;5990&quot;      &quot;4298&quot;      &quot;2962&quot;     
[3247] &quot;8570&quot;      &quot;79085&quot;     &quot;56927&quot;     &quot;9322&quot;      &quot;140467&quot;    &quot;57662&quot;    
[3253] &quot;126003&quot;    &quot;115704&quot;    &quot;404217&quot;    &quot;79603&quot;     &quot;51293&quot;     &quot;4701&quot;     
[3259] &quot;6234&quot;      &quot;51129&quot;     &quot;100507567&quot; &quot;9230&quot;      &quot;4670&quot;      &quot;148156&quot;   
[3265] &quot;57693&quot;     &quot;10781&quot;     &quot;79088&quot;     &quot;101928238&quot; &quot;54811&quot;     &quot;162993&quot;   
[3271] &quot;5300&quot;      &quot;55337&quot;     &quot;56342&quot;     &quot;5032&quot;      &quot;8666&quot;      &quot;9294&quot;     
[3277] &quot;125950&quot;    &quot;3385&quot;      &quot;7297&quot;      &quot;9817&quot;      &quot;84971&quot;     &quot;65095&quot;    
[3283] &quot;3609&quot;      &quot;1785&quot;      &quot;25959&quot;     &quot;57572&quot;     &quot;126075&quot;    &quot;2057&quot;     
[3289] &quot;5589&quot;      &quot;199692&quot;    &quot;55552&quot;     &quot;126068&quot;    &quot;126069&quot;    &quot;90594&quot;    
[3295] &quot;90592&quot;     &quot;388507&quot;    &quot;7695&quot;      &quot;51710&quot;     &quot;10224&quot;     &quot;163051&quot;   
[3301] &quot;163049&quot;    &quot;84292&quot;     &quot;1725&quot;      &quot;30000&quot;     &quot;29911&quot;     &quot;7001&quot;     
[3307] &quot;22983&quot;     &quot;1777&quot;      &quot;2639&quot;      &quot;256126&quot;    &quot;2193&quot;      &quot;90480&quot;    
[3313] &quot;4784&quot;      &quot;55621&quot;     &quot;81576&quot;     &quot;84245&quot;     &quot;28974&quot;     &quot;284454&quot;   
[3319] &quot;54862&quot;     &quot;90379&quot;     &quot;113230&quot;    &quot;90378&quot;     &quot;100507373&quot; &quot;22859&quot;    
[3325] &quot;10212&quot;     &quot;3337&quot;      &quot;4713&quot;      &quot;4854&quot;      &quot;26993&quot;     &quot;8907&quot;     
[3331] &quot;10365&quot;     &quot;58513&quot;     &quot;10523&quot;     &quot;79939&quot;     &quot;79041&quot;     &quot;2063&quot;     
[3337] &quot;29086&quot;     &quot;64981&quot;     &quot;79016&quot;     &quot;57719&quot;     &quot;25796&quot;     &quot;79709&quot;    
[3343] &quot;55201&quot;     &quot;6142&quot;      &quot;115098&quot;    &quot;3780&quot;      &quot;23031&quot;     &quot;84769&quot;    
[3349] &quot;5864&quot;      &quot;3727&quot;      &quot;25804&quot;     &quot;54858&quot;     &quot;170463&quot;    &quot;23770&quot;    
[3355] &quot;79036&quot;     &quot;25789&quot;     &quot;5976&quot;      &quot;9454&quot;      &quot;10147&quot;     &quot;93436&quot;    
[3361] &quot;284439&quot;    &quot;729991&quot;    &quot;126382&quot;    &quot;57794&quot;     &quot;23383&quot;     &quot;51079&quot;    
[3367] &quot;148113&quot;    &quot;9170&quot;      &quot;51291&quot;     &quot;7561&quot;      &quot;284440&quot;    &quot;56242&quot;    
[3373] &quot;91120&quot;     &quot;90649&quot;     &quot;100129842&quot; &quot;199777&quot;    &quot;7639&quot;      &quot;80264&quot;    
[3379] &quot;148206&quot;    &quot;170959&quot;    &quot;7562&quot;      &quot;148203&quot;    &quot;353088&quot;    &quot;163227&quot;   
[3385] &quot;7757&quot;      &quot;7644&quot;      &quot;148213&quot;    &quot;388524&quot;    &quot;101929206&quot; &quot;148189&quot;   
[3391] &quot;101927151&quot; &quot;7386&quot;      &quot;10775&quot;     &quot;83636&quot;     &quot;898&quot;       &quot;8725&quot;     
[3397] &quot;22847&quot;     &quot;147991&quot;    &quot;84079&quot;     &quot;85415&quot;     &quot;126248&quot;    &quot;4037&quot;     
[3403] &quot;1054&quot;      &quot;5184&quot;      &quot;79047&quot;     &quot;10054&quot;     &quot;126374&quot;    &quot;55900&quot;    
[3409] &quot;339318&quot;    &quot;148103&quot;    &quot;90075&quot;     &quot;126375&quot;    &quot;57655&quot;     &quot;3249&quot;     
[3415] &quot;163175&quot;    &quot;5348&quot;      &quot;53827&quot;     &quot;100128682&quot; &quot;93099&quot;     &quot;100506469&quot;
[3421] &quot;1340&quot;      &quot;55957&quot;     &quot;126393&quot;    &quot;84807&quot;     &quot;644096&quot;    &quot;199745&quot;   
[3427] &quot;1155&quot;      &quot;826&quot;       &quot;84924&quot;     &quot;101927599&quot; &quot;92283&quot;     &quot;284408&quot;   
[3433] &quot;388536&quot;    &quot;25850&quot;     &quot;147923&quot;    &quot;199704&quot;    &quot;92285&quot;     &quot;163087&quot;   
[3439] &quot;284459&quot;    &quot;148266&quot;    &quot;148268&quot;    &quot;101927720&quot; &quot;100507433&quot; &quot;163255&quot;   
[3445] &quot;51276&quot;     &quot;22835&quot;     &quot;84775&quot;     &quot;23094&quot;     &quot;94274&quot;     &quot;10653&quot;    
[3451] &quot;9424&quot;      &quot;27335&quot;     &quot;81&quot;        &quot;147968&quot;    &quot;1891&quot;      &quot;6183&quot;     
[3457] &quot;342897&quot;    &quot;55095&quot;     &quot;54623&quot;     &quot;6217&quot;      &quot;6829&quot;      &quot;92609&quot;    
[3463] &quot;10683&quot;     &quot;126272&quot;    &quot;163126&quot;    &quot;9149&quot;      &quot;2091&quot;      &quot;5704&quot;     
[3469] &quot;339327&quot;    &quot;163131&quot;    &quot;284323&quot;    &quot;208&quot;       &quot;23646&quot;     &quot;147746&quot;   
[3475] &quot;29946&quot;     &quot;8425&quot;      &quot;9253&quot;      &quot;79934&quot;     &quot;284325&quot;    &quot;6626&quot;     
[3481] &quot;112398&quot;    &quot;558&quot;       &quot;11100&quot;     &quot;90324&quot;     &quot;641649&quot;    &quot;593&quot;      
[3487] &quot;55101&quot;     &quot;64763&quot;     &quot;116115&quot;    &quot;2077&quot;      &quot;23152&quot;     &quot;100996307&quot;
[3493] &quot;3991&quot;      &quot;23474&quot;     &quot;126298&quot;    &quot;79177&quot;     &quot;199731&quot;    &quot;5329&quot;     
[3499] &quot;284349&quot;    &quot;7711&quot;      &quot;7773&quot;      &quot;7673&quot;      &quot;7766&quot;      &quot;342909&quot;   
[3505] &quot;7767&quot;      &quot;100379224&quot; &quot;7768&quot;      &quot;10780&quot;     &quot;7769&quot;      &quot;7772&quot;     
[3511] &quot;7733&quot;      &quot;5817&quot;      &quot;56971&quot;     &quot;602&quot;       &quot;4059&quot;      &quot;10452&quot;    
[3517] &quot;348&quot;       &quot;11129&quot;     &quot;79760&quot;     &quot;57787&quot;     &quot;147700&quot;    &quot;10848&quot;    
[3523] &quot;10849&quot;     &quot;7408&quot;      &quot;24139&quot;     &quot;6633&quot;      &quot;54814&quot;     &quot;23403&quot;    
[3529] &quot;147912&quot;    &quot;1760&quot;      &quot;339344&quot;    &quot;808&quot;       &quot;147906&quot;    &quot;100506068&quot;
[3535] &quot;25865&quot;     &quot;1175&quot;      &quot;2909&quot;      &quot;54958&quot;     &quot;10055&quot;     &quot;26093&quot;    
[3541] &quot;11133&quot;     &quot;29998&quot;     &quot;29997&quot;     &quot;6415&quot;      &quot;8605&quot;      &quot;100505812&quot;
[3547] &quot;10945&quot;     &quot;2906&quot;      &quot;6141&quot;      &quot;770&quot;       &quot;587&quot;       &quot;51171&quot;    
[3553] &quot;4924&quot;      &quot;581&quot;       &quot;2512&quot;      &quot;2997&quot;      &quot;10856&quot;     &quot;6625&quot;     
[3559] &quot;3270&quot;      &quot;28968&quot;     &quot;951&quot;       &quot;55011&quot;     &quot;126133&quot;    &quot;23521&quot;    
[3565] &quot;6205&quot;      &quot;2217&quot;      &quot;57333&quot;     &quot;6237&quot;      &quot;58506&quot;     &quot;126129&quot;   
[3571] &quot;160&quot;       &quot;80199&quot;     &quot;53635&quot;     &quot;101928378&quot; &quot;11284&quot;     &quot;84335&quot;    
[3577] &quot;79735&quot;     &quot;22809&quot;     &quot;51231&quot;     &quot;79784&quot;     &quot;3748&quot;      &quot;7376&quot;     
[3583] &quot;112703&quot;    &quot;284361&quot;    &quot;126119&quot;    &quot;6320&quot;      &quot;84798&quot;     &quot;90353&quot;    
[3589] &quot;147645&quot;    &quot;7728&quot;      &quot;147650&quot;    &quot;84765&quot;     &quot;65251&quot;     &quot;59348&quot;    
[3595] &quot;80110&quot;     &quot;9668&quot;      &quot;284371&quot;    &quot;5518&quot;      &quot;162963&quot;    &quot;102724105&quot;
[3601] &quot;84436&quot;     &quot;147660&quot;    &quot;55762&quot;     &quot;7696&quot;      &quot;55769&quot;     &quot;81856&quot;    
[3607] &quot;90338&quot;     &quot;342926&quot;    &quot;91661&quot;     &quot;388561&quot;    &quot;55422&quot;     &quot;59283&quot;    
[3613] &quot;4696&quot;      &quot;29844&quot;     &quot;26121&quot;     &quot;4849&quot;      &quot;79042&quot;     &quot;6203&quot;     
[3619] &quot;104355426&quot; &quot;114823&quot;    &quot;94059&quot;     &quot;55655&quot;     &quot;54869&quot;     &quot;54776&quot;    
[3625] &quot;7138&quot;      &quot;352909&quot;    &quot;5794&quot;      &quot;23640&quot;     &quot;3589&quot;      &quot;6158&quot;     
[3631] &quot;89887&quot;     &quot;100507290&quot; &quot;51157&quot;     &quot;29924&quot;     &quot;126208&quot;    &quot;147948&quot;   
[3637] &quot;386758&quot;    &quot;63934&quot;     &quot;388566&quot;    &quot;9422&quot;      &quot;390980&quot;    &quot;10794&quot;    
[3643] &quot;125919&quot;    &quot;57343&quot;     &quot;284306&quot;    &quot;10597&quot;     &quot;7565&quot;      &quot;388567&quot;   
[3649] &quot;400720&quot;    &quot;79744&quot;     &quot;374928&quot;    &quot;256051&quot;    &quot;55659&quot;     &quot;284307&quot;   
[3655] &quot;348327&quot;    &quot;7693&quot;      &quot;10520&quot;     &quot;7710&quot;      &quot;54807&quot;     &quot;730051&quot;   
[3661] &quot;147687&quot;    &quot;10172&quot;     &quot;147685&quot;    &quot;80095&quot;     &quot;100128398&quot; &quot;65982&quot;    
[3667] &quot;79673&quot;     &quot;27300&quot;     &quot;7554&quot;      &quot;105372483&quot; &quot;6193&quot;      &quot;7691&quot;     
[3673] &quot;10998&quot;     &quot;10155&quot;     &quot;27243&quot;     &quot;9040&quot;      &quot;65996&quot;     &quot;642273&quot;   
[3679] &quot;52&quot;        &quot;7173&quot;      &quot;7260&quot;      &quot;51112&quot;     &quot;55256&quot;     &quot;100506054&quot;
[3685] &quot;6201&quot;      &quot;78989&quot;     &quot;57498&quot;     &quot;129642&quot;    &quot;8853&quot;      &quot;9270&quot;     
[3691] &quot;6868&quot;      &quot;10971&quot;     &quot;9014&quot;      &quot;79954&quot;     &quot;10130&quot;     &quot;130814&quot;   
[3697] &quot;9475&quot;      &quot;1876&quot;      &quot;9687&quot;      &quot;23175&quot;     &quot;28951&quot;     &quot;4613&quot;     
[3703] &quot;7447&quot;      &quot;79677&quot;     &quot;57665&quot;     &quot;130502&quot;    &quot;57539&quot;     &quot;9741&quot;     
[3709] &quot;23369&quot;     &quot;388&quot;       &quot;64342&quot;     &quot;54454&quot;     &quot;165324&quot;    &quot;80304&quot;    
[3715] &quot;51639&quot;     &quot;375190&quot;    &quot;391356&quot;    &quot;51277&quot;     &quot;1788&quot;      &quot;1838&quot;     
[3721] &quot;3797&quot;      &quot;3030&quot;      &quot;85465&quot;     &quot;54978&quot;     &quot;22924&quot;     &quot;54867&quot;    
[3727] &quot;11117&quot;     &quot;10669&quot;     &quot;10113&quot;     &quot;8884&quot;      &quot;51374&quot;     &quot;285126&quot;   
[3733] &quot;2976&quot;      &quot;8890&quot;      &quot;9784&quot;      &quot;130557&quot;    &quot;29959&quot;     &quot;84226&quot;    
[3739] &quot;84450&quot;     &quot;11321&quot;     &quot;9913&quot;      &quot;22950&quot;     &quot;105374378&quot; &quot;64080&quot;    
[3745] &quot;9577&quot;      &quot;5500&quot;      &quot;55006&quot;     &quot;81606&quot;     &quot;253558&quot;    &quot;30845&quot;    
[3751] &quot;51072&quot;     &quot;6683&quot;      &quot;58484&quot;     &quot;84272&quot;     &quot;25780&quot;     &quot;51232&quot;    
[3757] &quot;6801&quot;      &quot;54497&quot;     &quot;5610&quot;      &quot;100505876&quot; &quot;10153&quot;     &quot;55471&quot;    
[3763] &quot;23683&quot;     &quot;25797&quot;     &quot;10602&quot;     &quot;64225&quot;     &quot;92906&quot;     &quot;90957&quot;    
[3769] &quot;729967&quot;    &quot;8491&quot;      &quot;63892&quot;     &quot;79823&quot;     &quot;5581&quot;      &quot;23433&quot;    
[3775] &quot;9419&quot;      &quot;9655&quot;      &quot;805&quot;       &quot;618&quot;       &quot;4072&quot;      &quot;2956&quot;     
[3781] &quot;80204&quot;     &quot;129285&quot;    &quot;9378&quot;      &quot;51130&quot;     &quot;494143&quot;    &quot;27248&quot;    
[3787] &quot;10936&quot;     &quot;23198&quot;     &quot;6711&quot;      &quot;130162&quot;    &quot;6233&quot;      &quot;4528&quot;     
[3793] &quot;344405&quot;    &quot;55704&quot;     &quot;112942&quot;    &quot;57223&quot;     &quot;87178&quot;     &quot;2202&quot;     
[3799] &quot;7444&quot;      &quot;55120&quot;     &quot;53335&quot;     &quot;64895&quot;     &quot;5966&quot;      &quot;150962&quot;   
[3805] &quot;5194&quot;      &quot;84542&quot;     &quot;130872&quot;    &quot;7514&quot;      &quot;84140&quot;     &quot;10575&quot;    
[3811] &quot;150684&quot;    &quot;51057&quot;     &quot;7360&quot;      &quot;6509&quot;      &quot;5861&quot;      &quot;10097&quot;    
[3817] &quot;4211&quot;      &quot;54465&quot;     &quot;10438&quot;     &quot;56902&quot;     &quot;5534&quot;      &quot;25927&quot;    
[3823] &quot;554251&quot;    &quot;27302&quot;     &quot;84168&quot;     &quot;2673&quot;      &quot;27247&quot;     &quot;22848&quot;    
[3829] &quot;5093&quot;      &quot;51449&quot;     &quot;6637&quot;      &quot;84908&quot;     &quot;7039&quot;      &quot;119&quot;      
[3835] &quot;113419&quot;    &quot;55577&quot;     &quot;84693&quot;     &quot;400961&quot;    &quot;23233&quot;     &quot;6697&quot;     
[3841] &quot;84279&quot;     &quot;10574&quot;     &quot;8446&quot;      &quot;1716&quot;      &quot;200424&quot;    &quot;10797&quot;    
[3847] &quot;57835&quot;     &quot;1639&quot;      &quot;84058&quot;     &quot;6242&quot;      &quot;84865&quot;     &quot;84695&quot;    
[3853] &quot;1796&quot;      &quot;3099&quot;      &quot;56655&quot;     &quot;6936&quot;      &quot;80059&quot;     &quot;9168&quot;     
[3859] &quot;56888&quot;     &quot;83439&quot;     &quot;84173&quot;     &quot;822&quot;       &quot;2677&quot;      &quot;10791&quot;    
[3865] &quot;129303&quot;    &quot;10713&quot;     &quot;388969&quot;    &quot;84913&quot;     &quot;8869&quot;      &quot;90784&quot;    
[3871] &quot;25885&quot;     &quot;51652&quot;     &quot;7844&quot;      &quot;400966&quot;    &quot;5343&quot;      &quot;285074&quot;   
[3877] &quot;102724642&quot; &quot;5342&quot;      &quot;729857&quot;    &quot;55258&quot;     &quot;654342&quot;    &quot;64969&quot;    
[3883] &quot;7549&quot;      &quot;51011&quot;     &quot;285033&quot;    &quot;9391&quot;      &quot;23020&quot;     &quot;55683&quot;    
[3889] &quot;90342&quot;     &quot;51239&quot;     &quot;54910&quot;     &quot;1329&quot;      &quot;23505&quot;     &quot;3631&quot;     
[3895] &quot;493753&quot;    &quot;25972&quot;     &quot;51601&quot;     &quot;129531&quot;    &quot;10190&quot;     &quot;9669&quot;     
[3901] &quot;51455&quot;     &quot;3899&quot;      &quot;4862&quot;      &quot;6160&quot;      &quot;11138&quot;     &quot;100506286&quot;
[3907] &quot;9448&quot;      &quot;84804&quot;     &quot;130827&quot;    &quot;79074&quot;     &quot;8440&quot;      &quot;653489&quot;   
[3913] &quot;84620&quot;     &quot;285190&quot;    &quot;27233&quot;     &quot;5903&quot;      &quot;165055&quot;    &quot;100287216&quot;
[3919] &quot;151011&quot;    &quot;84220&quot;     &quot;96626&quot;     &quot;440894&quot;    &quot;7851&quot;      &quot;205251&quot;   
[3925] &quot;151009&quot;    &quot;729540&quot;    &quot;84524&quot;     &quot;727851&quot;    &quot;6574&quot;      &quot;150472&quot;   
[3931] &quot;729468&quot;    &quot;84771&quot;     &quot;118433&quot;    &quot;80255&quot;     &quot;10096&quot;     &quot;8886&quot;     
[3937] &quot;55240&quot;     &quot;140738&quot;    &quot;200373&quot;    &quot;2736&quot;      &quot;7247&quot;      &quot;2995&quot;     
[3943] &quot;274&quot;       &quot;2071&quot;      &quot;55677&quot;     &quot;4648&quot;      &quot;84826&quot;     &quot;83607&quot;    
[3949] &quot;56886&quot;     &quot;84084&quot;     &quot;440905&quot;    &quot;728378&quot;    &quot;285103&quot;    &quot;80097&quot;    
[3955] &quot;84317&quot;     &quot;92856&quot;     &quot;26469&quot;     &quot;50649&quot;     &quot;130074&quot;    &quot;150786&quot;   
[3961] &quot;653784&quot;    &quot;90557&quot;     &quot;29798&quot;     &quot;2863&quot;      &quot;344148&quot;    &quot;81615&quot;    
[3967] &quot;905&quot;       &quot;22930&quot;     &quot;84083&quot;     &quot;23190&quot;     &quot;1615&quot;      &quot;3176&quot;     
[3973] &quot;339745&quot;    &quot;9839&quot;      &quot;26122&quot;     &quot;130576&quot;    &quot;130574&quot;    &quot;27249&quot;    
[3979] &quot;55183&quot;     &quot;4703&quot;      &quot;10254&quot;     &quot;55660&quot;     &quot;151188&quot;    &quot;3760&quot;     
[3985] &quot;2820&quot;      &quot;728066&quot;    &quot;90&quot;        &quot;29994&quot;     &quot;22925&quot;     &quot;5937&quot;     
[3991] &quot;10010&quot;     &quot;100996579&quot; &quot;285116&quot;    &quot;25801&quot;     &quot;2888&quot;      &quot;6326&quot;     
[3997] &quot;79809&quot;     &quot;6335&quot;      &quot;253782&quot;    &quot;10324&quot;     &quot;9360&quot;      &quot;493911&quot;   
[4003] &quot;6741&quot;      &quot;130507&quot;    &quot;79901&quot;     &quot;1781&quot;      &quot;8520&quot;      &quot;3655&quot;     
[4009] &quot;5163&quot;      &quot;11069&quot;     &quot;51776&quot;     &quot;79634&quot;     &quot;151556&quot;    &quot;7456&quot;     
[4015] &quot;1134&quot;      &quot;1386&quot;      &quot;10651&quot;     &quot;220988&quot;    &quot;8540&quot;      &quot;65977&quot;    
[4021] &quot;91404&quot;     &quot;151126&quot;    &quot;10477&quot;     &quot;6744&quot;      &quot;151242&quot;    &quot;2487&quot;     
[4027] &quot;142679&quot;    &quot;55854&quot;     &quot;3685&quot;      &quot;165215&quot;    &quot;7035&quot;      &quot;1281&quot;     
[4033] &quot;1290&quot;      &quot;84128&quot;     &quot;30061&quot;     &quot;150709&quot;    &quot;94101&quot;     &quot;84281&quot;    
[4039] &quot;100131211&quot; &quot;4664&quot;      &quot;8436&quot;      &quot;57181&quot;     &quot;80055&quot;     &quot;91526&quot;    
[4045] &quot;23451&quot;     &quot;80219&quot;     &quot;3329&quot;      &quot;3336&quot;      &quot;25843&quot;     &quot;5334&quot;     
[4051] &quot;129450&quot;    &quot;285172&quot;    &quot;4709&quot;      &quot;8837&quot;      &quot;841&quot;       &quot;130540&quot;   
[4057] &quot;66008&quot;     &quot;55437&quot;     &quot;65062&quot;     &quot;57679&quot;     &quot;7341&quot;      &quot;51602&quot;    
[4063] &quot;659&quot;       &quot;150864&quot;    &quot;130026&quot;    &quot;57404&quot;     &quot;10152&quot;     &quot;65059&quot;    
[4069] &quot;117583&quot;    &quot;54891&quot;     &quot;1933&quot;      &quot;57683&quot;     &quot;8745&quot;      &quot;8609&quot;     
[4075] &quot;3417&quot;      &quot;200576&quot;    &quot;4133&quot;      &quot;6120&quot;      &quot;151050&quot;    &quot;10314&quot;    
[4081] &quot;79582&quot;     &quot;471&quot;       &quot;2335&quot;      &quot;7520&quot;      &quot;50485&quot;     &quot;6168&quot;     
[4087] &quot;3485&quot;      &quot;3488&quot;      &quot;10109&quot;     &quot;58190&quot;     &quot;57695&quot;     &quot;84812&quot;    
[4093] &quot;64320&quot;     &quot;27148&quot;     &quot;1593&quot;      &quot;27013&quot;     &quot;130617&quot;    &quot;79065&quot;    
[4099] &quot;8576&quot;      &quot;3300&quot;      &quot;5798&quot;      &quot;23549&quot;     &quot;130612&quot;    &quot;3623&quot;     
[4105] &quot;2043&quot;      &quot;2181&quot;      &quot;7857&quot;      &quot;130340&quot;    &quot;57590&quot;     &quot;8452&quot;     
[4111] &quot;84236&quot;     &quot;1286&quot;      &quot;1285&quot;      &quot;56947&quot;     &quot;3267&quot;      &quot;164781&quot;   
[4117] &quot;92737&quot;     &quot;6672&quot;      &quot;81618&quot;     &quot;5707&quot;      &quot;80210&quot;     &quot;5147&quot;     
[4123] &quot;64708&quot;     &quot;9470&quot;      &quot;26058&quot;     &quot;3635&quot;      &quot;8527&quot;      &quot;55230&quot;    
[4129] &quot;23677&quot;     &quot;116987&quot;    &quot;79781&quot;     &quot;10920&quot;     &quot;1293&quot;      &quot;64284&quot;    
[4135] &quot;10267&quot;     &quot;140739&quot;    &quot;339768&quot;    &quot;80895&quot;     &quot;643387&quot;    &quot;26146&quot;    
[4141] &quot;4705&quot;      &quot;2817&quot;      &quot;57140&quot;     &quot;50636&quot;     &quot;3069&quot;      &quot;9855&quot;     
[4147] &quot;666&quot;       &quot;51078&quot;     &quot;23192&quot;     &quot;84289&quot;     &quot;728294&quot;    &quot;80023&quot;    
[4153] &quot;1457&quot;      &quot;140809&quot;    &quot;9491&quot;      &quot;55321&quot;     &quot;55968&quot;     &quot;10326&quot;    
[4159] &quot;140885&quot;    &quot;6628&quot;      &quot;79175&quot;     &quot;3420&quot;      &quot;56265&quot;     &quot;64601&quot;    
[4165] &quot;5786&quot;      &quot;64949&quot;     &quot;60493&quot;     &quot;9762&quot;      &quot;8455&quot;      &quot;80332&quot;    
[4171] &quot;54976&quot;     &quot;1059&quot;      &quot;57506&quot;     &quot;11237&quot;     &quot;9962&quot;      &quot;29058&quot;    
[4177] &quot;149837&quot;    &quot;149840&quot;    &quot;1114&quot;      &quot;55612&quot;     &quot;650&quot;       &quot;23236&quot;    
[4183] &quot;128710&quot;    &quot;182&quot;       &quot;55617&quot;     &quot;51575&quot;     &quot;11034&quot;     &quot;6238&quot;     
[4189] &quot;27131&quot;     &quot;7692&quot;      &quot;10621&quot;     &quot;10483&quot;     &quot;92675&quot;     &quot;100270804&quot;
[4195] &quot;51126&quot;     &quot;22803&quot;     &quot;7056&quot;      &quot;29107&quot;     &quot;64412&quot;     &quot;79953&quot;    
[4201] &quot;57136&quot;     &quot;101926889&quot; &quot;5834&quot;      &quot;22981&quot;     &quot;26152&quot;     &quot;100134868&quot;
[4207] &quot;100289097&quot; &quot;284802&quot;    &quot;81502&quot;     &quot;598&quot;       &quot;128853&quot;    &quot;81572&quot;    
[4213] &quot;9777&quot;      &quot;5326&quot;      &quot;9371&quot;      &quot;171023&quot;    &quot;149951&quot;    &quot;1789&quot;     
[4219] &quot;51654&quot;     &quot;63941&quot;     &quot;84905&quot;     &quot;22913&quot;     &quot;8894&quot;      &quot;191&quot;      
[4225] &quot;83737&quot;     &quot;83658&quot;     &quot;23054&quot;     &quot;2937&quot;      &quot;26133&quot;     &quot;55741&quot;    
[4231] &quot;10893&quot;     &quot;101410538&quot; &quot;11190&quot;     &quot;51614&quot;     &quot;80307&quot;     &quot;6676&quot;     
[4237] &quot;8904&quot;      &quot;10137&quot;     &quot;9054&quot;      &quot;140823&quot;    &quot;9584&quot;      &quot;51230&quot;    
[4243] &quot;51282&quot;     &quot;647979&quot;    &quot;22839&quot;     &quot;10398&quot;     &quot;60436&quot;     &quot;55969&quot;    
[4249] &quot;140710&quot;    &quot;140711&quot;    &quot;10904&quot;     &quot;4826&quot;      &quot;56259&quot;     &quot;128434&quot;   
[4255] &quot;9675&quot;      &quot;58490&quot;     &quot;57148&quot;     &quot;7150&quot;      &quot;5335&quot;      &quot;26013&quot;    
[4261] &quot;10955&quot;     &quot;7529&quot;      &quot;80336&quot;     &quot;90196&quot;     &quot;116092&quot;    &quot;10005&quot;    
[4267] &quot;140831&quot;    &quot;90204&quot;     &quot;128497&quot;    &quot;140825&quot;    &quot;5360&quot;      &quot;63935&quot;    
[4273] &quot;57468&quot;     &quot;51006&quot;     &quot;112858&quot;    &quot;23613&quot;     &quot;57580&quot;     &quot;10564&quot;    
[4279] &quot;1434&quot;      &quot;6780&quot;      &quot;55661&quot;     &quot;441951&quot;    &quot;9334&quot;      &quot;55905&quot;    
[4285] &quot;7335&quot;      &quot;101927559&quot; &quot;140876&quot;    &quot;84612&quot;     &quot;55653&quot;     &quot;101927631&quot;
[4291] &quot;3755&quot;      &quot;4773&quot;      &quot;10079&quot;     &quot;5203&quot;      &quot;55816&quot;     &quot;1477&quot;     
[4297] &quot;55370&quot;     &quot;9217&quot;      &quot;164284&quot;    &quot;100534593&quot; &quot;79716&quot;     &quot;1522&quot;     
[4303] &quot;81027&quot;     &quot;514&quot;       &quot;51012&quot;     &quot;116154&quot;    &quot;6874&quot;      &quot;149986&quot;   
[4309] &quot;26039&quot;     &quot;9885&quot;      &quot;6227&quot;      &quot;1299&quot;      &quot;11083&quot;     &quot;54915&quot;    
[4315] &quot;85441&quot;     &quot;26205&quot;     &quot;50861&quot;     &quot;10139&quot;     &quot;56731&quot;     &quot;140685&quot;   
[4321] &quot;140701&quot;    &quot;80331&quot;     &quot;100113386&quot; &quot;24148&quot;     &quot;6919&quot;      &quot;4987&quot;     
[4327] &quot;102724023&quot; &quot;102724159&quot; &quot;102724219&quot; &quot;102724441&quot; &quot;102724594&quot; &quot;102724843&quot;
[4333] &quot;102724951&quot; &quot;102723360&quot; &quot;100500862&quot; &quot;100500815&quot; &quot;103504731&quot; &quot;6782&quot;     
[4339] &quot;8204&quot;      &quot;1525&quot;      &quot;4685&quot;      &quot;54148&quot;     &quot;58494&quot;     &quot;522&quot;      
[4345] &quot;351&quot;       &quot;116159&quot;    &quot;9510&quot;      &quot;26046&quot;     &quot;10600&quot;     &quot;56911&quot;    
[4351] &quot;2897&quot;      &quot;7074&quot;      &quot;30811&quot;     &quot;9875&quot;      &quot;140290&quot;    &quot;56683&quot;    
[4357] &quot;94104&quot;     &quot;3588&quot;      &quot;3454&quot;      &quot;3460&quot;      &quot;757&quot;       &quot;54943&quot;    
[4363] &quot;2618&quot;      &quot;6651&quot;      &quot;6453&quot;      &quot;1827&quot;      &quot;54093&quot;     &quot;873&quot;      
[4369] &quot;100506428&quot; &quot;6493&quot;      &quot;51227&quot;     &quot;7267&quot;      &quot;1859&quot;      &quot;2114&quot;     
[4375] &quot;8624&quot;      &quot;54014&quot;     &quot;150082&quot;    &quot;25825&quot;     &quot;25966&quot;     &quot;10785&quot;    
[4381] &quot;4731&quot;      &quot;7307&quot;      &quot;8566&quot;      &quot;1476&quot;      &quot;8568&quot;      &quot;5822&quot;     
[4387] &quot;8209&quot;      &quot;5211&quot;      &quot;755&quot;       &quot;7327&quot;      &quot;754&quot;       &quot;104&quot;      
[4393] &quot;23275&quot;     &quot;642852&quot;    &quot;80781&quot;     &quot;54039&quot;     &quot;1292&quot;      &quot;8888&quot;     
[4399] &quot;54059&quot;     &quot;5116&quot;      &quot;51816&quot;     &quot;27443&quot;     &quot;23786&quot;     &quot;55670&quot;    
[4405] &quot;11274&quot;     &quot;8214&quot;      &quot;9993&quot;      &quot;100652736&quot; &quot;6576&quot;      &quot;8218&quot;     
[4411] &quot;7290&quot;      &quot;7353&quot;      &quot;5413&quot;      &quot;100526833&quot; &quot;10587&quot;     &quot;1312&quot;     
[4417] &quot;27037&quot;     &quot;29801&quot;     &quot;65078&quot;     &quot;85359&quot;     &quot;91179&quot;     &quot;84861&quot;    
[4423] &quot;51586&quot;     &quot;5297&quot;      &quot;9342&quot;      &quot;8216&quot;      &quot;80764&quot;     &quot;9127&quot;     
[4429] &quot;728441&quot;    &quot;7332&quot;      &quot;29799&quot;     &quot;5594&quot;      &quot;9647&quot;      &quot;8940&quot;     
[4435] &quot;96610&quot;     &quot;91227&quot;     &quot;9609&quot;      &quot;613&quot;       &quot;26226&quot;     &quot;91316&quot;    
[4441] &quot;266747&quot;    &quot;7621&quot;      &quot;4320&quot;      &quot;6598&quot;      &quot;66035&quot;     &quot;653689&quot;   
[4447] &quot;100037417&quot; &quot;1652&quot;      &quot;2953&quot;      &quot;56241&quot;     &quot;23384&quot;     &quot;6634&quot;     
[4453] &quot;2678&quot;      &quot;388886&quot;    &quot;85379&quot;     &quot;91355&quot;     &quot;157&quot;       &quot;89781&quot;    
[4459] &quot;8459&quot;      &quot;4330&quot;      &quot;23760&quot;     &quot;284900&quot;    &quot;150274&quot;    &quot;129080&quot;   
[4465] &quot;2130&quot;      &quot;10634&quot;     &quot;162&quot;       &quot;8563&quot;      &quot;29796&quot;     &quot;84164&quot;    
[4471] &quot;8897&quot;      &quot;652968&quot;    &quot;83874&quot;     &quot;10291&quot;     &quot;85371&quot;     &quot;200312&quot;   
[4477] &quot;51537&quot;     &quot;23481&quot;     &quot;339665&quot;    &quot;150290&quot;    &quot;22880&quot;     &quot;55000&quot;    
[4483] &quot;6525&quot;      &quot;113791&quot;    &quot;54944&quot;     &quot;56478&quot;     &quot;51493&quot;     &quot;25793&quot;    
[4489] &quot;9215&quot;      &quot;10043&quot;     &quot;23551&quot;     &quot;4151&quot;      &quot;23543&quot;     &quot;4627&quot;     
[4495] &quot;25828&quot;     &quot;8664&quot;      &quot;114904&quot;    &quot;5880&quot;      &quot;11135&quot;     &quot;26088&quot;    
[4501] &quot;57026&quot;     &quot;11078&quot;     &quot;3005&quot;      &quot;23464&quot;     &quot;129138&quot;    &quot;51386&quot;    
[4507] &quot;5435&quot;      &quot;9463&quot;      &quot;23539&quot;     &quot;25829&quot;     &quot;1454&quot;      &quot;3761&quot;     
[4513] &quot;11144&quot;     &quot;646851&quot;    &quot;25776&quot;     &quot;56993&quot;     &quot;25777&quot;     &quot;10126&quot;    
[4519] &quot;23467&quot;     &quot;23466&quot;     &quot;9582&quot;      &quot;200316&quot;    &quot;60489&quot;     &quot;6122&quot;     
[4525] &quot;4248&quot;      &quot;54471&quot;     &quot;91582&quot;     &quot;23112&quot;     &quot;158&quot;       &quot;27352&quot;    
[4531] &quot;57591&quot;     &quot;63929&quot;     &quot;9978&quot;      &quot;83746&quot;     &quot;5905&quot;      &quot;23264&quot;    
[4537] &quot;7008&quot;      &quot;10766&quot;     &quot;50&quot;        &quot;171568&quot;    &quot;5372&quot;      &quot;4809&quot;     
[4543] &quot;79879&quot;     &quot;91689&quot;     &quot;1565&quot;      &quot;1564&quot;      &quot;6942&quot;      &quot;94009&quot;    
[4549] &quot;253190&quot;    &quot;91695&quot;     &quot;1727&quot;      &quot;27349&quot;     &quot;25830&quot;     &quot;25813&quot;    
[4555] &quot;29780&quot;     &quot;64098&quot;     &quot;85352&quot;     &quot;100506714&quot; &quot;10762&quot;     &quot;2192&quot;     
[4561] &quot;25814&quot;     &quot;5465&quot;      &quot;64781&quot;     &quot;25771&quot;     &quot;23774&quot;     &quot;79174&quot;    
[4567] &quot;415116&quot;    &quot;56666&quot;     &quot;85378&quot;     &quot;83933&quot;     &quot;23654&quot;     &quot;414918&quot;   
[4573] &quot;9701&quot;      &quot;9997&quot;      &quot;1375&quot;      &quot;23542&quot;     &quot;284942&quot;    &quot;27255&quot;    
[4579] &quot;152330&quot;    &quot;51095&quot;     &quot;51185&quot;     &quot;285362&quot;    &quot;6419&quot;      &quot;8553&quot;     
[4585] &quot;55207&quot;     &quot;29995&quot;     &quot;5021&quot;      &quot;440944&quot;    &quot;10474&quot;     &quot;10093&quot;    
[4591] &quot;84522&quot;     &quot;132014&quot;    &quot;78987&quot;     &quot;55831&quot;     &quot;442075&quot;    &quot;401052&quot;   
[4597] &quot;152302&quot;    &quot;55845&quot;     &quot;7428&quot;      &quot;3656&quot;      &quot;9797&quot;      &quot;491&quot;      
[4603] &quot;10533&quot;     &quot;9686&quot;      &quot;5894&quot;      &quot;55287&quot;     &quot;6161&quot;      &quot;79885&quot;    
[4609] &quot;2199&quot;      &quot;131474&quot;    &quot;79188&quot;     &quot;6533&quot;      &quot;51244&quot;     &quot;84077&quot;    
[4615] &quot;100505641&quot; &quot;7182&quot;      &quot;23473&quot;     &quot;100505696&quot; &quot;9467&quot;      &quot;131965&quot;   
[4621] &quot;26061&quot;     &quot;23180&quot;     &quot;6304&quot;      &quot;79750&quot;     &quot;7324&quot;      &quot;28512&quot;    
[4627] &quot;55768&quot;     &quot;54995&quot;     &quot;152110&quot;    &quot;9497&quot;      &quot;152100&quot;    &quot;152098&quot;   
[4633] &quot;100873979&quot; &quot;7048&quot;      &quot;152189&quot;    &quot;51143&quot;     &quot;25904&quot;     &quot;2720&quot;     
[4639] &quot;643853&quot;    &quot;10491&quot;     &quot;25827&quot;     &quot;23122&quot;     &quot;10015&quot;     &quot;9881&quot;     
[4645] &quot;9852&quot;      &quot;9209&quot;      &quot;2803&quot;      &quot;3680&quot;      &quot;101928153&quot; &quot;10217&quot;    
[4651] &quot;30&quot;        &quot;4615&quot;      &quot;93&quot;        &quot;64689&quot;     &quot;3921&quot;      &quot;9045&quot;     
[4657] &quot;253639&quot;    &quot;1499&quot;      &quot;54986&quot;     &quot;22906&quot;     &quot;7433&quot;      &quot;51188&quot;    
[4663] &quot;4820&quot;      &quot;25994&quot;     &quot;84892&quot;     &quot;55129&quot;     &quot;285343&quot;    &quot;353274&quot;   
[4669] &quot;10168&quot;     &quot;7584&quot;      &quot;101928592&quot; &quot;91392&quot;     &quot;57456&quot;     &quot;131616&quot;   
[4675] &quot;23016&quot;     &quot;64866&quot;     &quot;23395&quot;     &quot;8994&quot;      &quot;22908&quot;     &quot;54585&quot;    
[4681] &quot;79443&quot;     &quot;6997&quot;      &quot;259173&quot;    &quot;29122&quot;     &quot;4634&quot;      &quot;5745&quot;     
[4687] &quot;151903&quot;    &quot;23218&quot;     &quot;23276&quot;     &quot;6599&quot;      &quot;22907&quot;     &quot;4134&quot;     
[4693] &quot;51385&quot;     &quot;51246&quot;     &quot;5210&quot;      &quot;1294&quot;      &quot;7384&quot;      &quot;65010&quot;    
[4699] &quot;1951&quot;      &quot;51447&quot;     &quot;5576&quot;      &quot;788&quot;       &quot;11180&quot;     &quot;55152&quot;    
[4705] &quot;25915&quot;     &quot;3615&quot;      &quot;54870&quot;     &quot;5859&quot;      &quot;64925&quot;     &quot;200942&quot;   
[4711] &quot;387&quot;       &quot;6988&quot;      &quot;84276&quot;     &quot;327&quot;       &quot;4485&quot;      &quot;63891&quot;    
[4717] &quot;29925&quot;     &quot;9807&quot;      &quot;10180&quot;     &quot;10181&quot;     &quot;100775107&quot; &quot;101928931&quot;
[4723] &quot;7869&quot;      &quot;24142&quot;     &quot;3373&quot;      &quot;8692&quot;      &quot;11068&quot;     &quot;11070&quot;    
[4729] &quot;9254&quot;      &quot;51409&quot;     &quot;1154&quot;      &quot;7867&quot;      &quot;84836&quot;     &quot;25864&quot;    
[4735] &quot;6159&quot;      &quot;11344&quot;     &quot;132160&quot;    &quot;80335&quot;     &quot;132158&quot;    &quot;25981&quot;    
[4741] &quot;8314&quot;      &quot;11188&quot;     &quot;440957&quot;    &quot;26354&quot;     &quot;28972&quot;     &quot;6787&quot;     
[4747] &quot;3699&quot;      &quot;3700&quot;      &quot;51460&quot;     &quot;55802&quot;     &quot;776&quot;       &quot;55349&quot;    
[4753] &quot;23272&quot;     &quot;54756&quot;     &quot;201626&quot;    &quot;2317&quot;      &quot;5162&quot;      &quot;200845&quot;   
[4759] &quot;5793&quot;      &quot;80145&quot;     &quot;9861&quot;      &quot;115286&quot;    &quot;26018&quot;     &quot;8801&quot;     
[4765] &quot;285203&quot;    &quot;7110&quot;      &quot;9039&quot;      &quot;10550&quot;     &quot;4286&quot;      &quot;2850&quot;     
[4771] &quot;727936&quot;    &quot;151987&quot;    &quot;55096&quot;     &quot;23024&quot;     &quot;101927249&quot; &quot;401074&quot;   
[4777] &quot;100131827&quot; &quot;285237&quot;    &quot;2042&quot;      &quot;5627&quot;      &quot;200894&quot;    &quot;131544&quot;   
[4783] &quot;56650&quot;     &quot;1371&quot;      &quot;100874207&quot; &quot;10402&quot;     &quot;131566&quot;    &quot;55773&quot;    
[4789] &quot;56954&quot;     &quot;9868&quot;      &quot;348801&quot;    &quot;55076&quot;     &quot;10342&quot;     &quot;54931&quot;    
[4795] &quot;57092&quot;     &quot;100009676&quot; &quot;6152&quot;      &quot;105374013&quot; &quot;285359&quot;    &quot;79598&quot;    
[4801] &quot;91775&quot;     &quot;214&quot;       &quot;344595&quot;    &quot;55081&quot;     &quot;9666&quot;      &quot;25945&quot;    
[4807] &quot;79669&quot;     &quot;257144&quot;    &quot;4345&quot;      &quot;64422&quot;     &quot;55032&quot;     &quot;100506621&quot;
[4813] &quot;151887&quot;    &quot;29083&quot;     &quot;91653&quot;     &quot;55779&quot;     &quot;152185&quot;    &quot;205717&quot;   
[4819] &quot;254887&quot;    &quot;57577&quot;     &quot;26137&quot;     &quot;152404&quot;    &quot;7348&quot;      &quot;8702&quot;     
[4825] &quot;57514&quot;     &quot;55254&quot;     &quot;51300&quot;     &quot;141&quot;       &quot;10063&quot;     &quot;2932&quot;     
[4831] &quot;11167&quot;     &quot;285282&quot;    &quot;2960&quot;      &quot;2804&quot;      &quot;9657&quot;      &quot;6565&quot;     
[4837] &quot;26355&quot;     &quot;54554&quot;     &quot;3836&quot;      &quot;83666&quot;     &quot;10954&quot;     &quot;26984&quot;    
[4843] &quot;64770&quot;     &quot;8997&quot;      &quot;3693&quot;      &quot;57493&quot;     &quot;84561&quot;     &quot;7707&quot;     
[4849] &quot;8723&quot;      &quot;114885&quot;    &quot;54946&quot;     &quot;166012&quot;    &quot;131601&quot;    &quot;50512&quot;    
[4855] &quot;11343&quot;     &quot;29927&quot;     &quot;8607&quot;      &quot;60678&quot;     &quot;6184&quot;      &quot;28976&quot;    
[4861] &quot;7555&quot;      &quot;22820&quot;     &quot;8971&quot;      &quot;339942&quot;    &quot;55764&quot;     &quot;23129&quot;    
[4867] &quot;7200&quot;      &quot;646300&quot;    &quot;30849&quot;     &quot;28990&quot;     &quot;84129&quot;     &quot;79876&quot;    
[4873] &quot;27031&quot;     &quot;7018&quot;      &quot;51560&quot;     &quot;6259&quot;      &quot;51421&quot;     &quot;339855&quot;   
[4879] &quot;5523&quot;      &quot;5096&quot;      &quot;4690&quot;      &quot;199221&quot;    &quot;51163&quot;     &quot;22808&quot;    
[4885] &quot;55179&quot;     &quot;56945&quot;     &quot;9276&quot;      &quot;100507291&quot; &quot;64084&quot;     &quot;55186&quot;    
[4891] &quot;92369&quot;     &quot;92370&quot;     &quot;253461&quot;    &quot;9616&quot;      &quot;483&quot;       &quot;5357&quot;     
[4897] &quot;7220&quot;      &quot;23350&quot;     &quot;205428&quot;    &quot;5352&quot;      &quot;2992&quot;      &quot;25937&quot;    
[4903] &quot;9819&quot;      &quot;27230&quot;     &quot;51714&quot;     &quot;6478&quot;      &quot;26084&quot;     &quot;170506&quot;   
[4909] &quot;4311&quot;      &quot;8833&quot;      &quot;7881&quot;      &quot;25976&quot;     &quot;730091&quot;    &quot;339894&quot;   
[4915] &quot;85476&quot;     &quot;56925&quot;     &quot;64747&quot;     &quot;57560&quot;     &quot;3840&quot;      &quot;151742&quot;   
[4921] &quot;51068&quot;     &quot;79740&quot;     &quot;11235&quot;     &quot;93556&quot;     &quot;2122&quot;      &quot;84517&quot;    
[4927] &quot;100128164&quot; &quot;7095&quot;      &quot;80012&quot;     &quot;6498&quot;      &quot;56648&quot;     &quot;64778&quot;    
[4933] &quot;5290&quot;      &quot;55669&quot;     &quot;86&quot;        &quot;57129&quot;     &quot;4711&quot;      &quot;339829&quot;   
[4939] &quot;8087&quot;      &quot;23200&quot;     &quot;54165&quot;     &quot;23101&quot;     &quot;84002&quot;     &quot;54800&quot;    
[4945] &quot;55689&quot;     &quot;55486&quot;     &quot;10057&quot;     &quot;8893&quot;      &quot;1857&quot;      &quot;1173&quot;     
[4951] &quot;1981&quot;      &quot;23355&quot;     &quot;285382&quot;    &quot;1962&quot;      &quot;200879&quot;    &quot;59343&quot;    
[4957] &quot;6434&quot;      &quot;51726&quot;     &quot;5648&quot;      &quot;101929130&quot; &quot;339929&quot;    &quot;55214&quot;    
[4963] &quot;152137&quot;    &quot;151963&quot;    &quot;57110&quot;     &quot;84789&quot;     &quot;4976&quot;      &quot;93109&quot;    
[4969] &quot;131583&quot;    &quot;152002&quot;    &quot;23527&quot;     &quot;5504&quot;      &quot;727956&quot;    &quot;440993&quot;   
[4975] &quot;255812&quot;    &quot;200931&quot;    &quot;5130&quot;      &quot;255758&quot;    &quot;165918&quot;    &quot;348793&quot;   
[4981] &quot;200933&quot;    &quot;54965&quot;     &quot;220729&quot;    &quot;84248&quot;     &quot;84223&quot;     &quot;6165&quot;     
[4987] &quot;152687&quot;    &quot;7700&quot;      &quot;170960&quot;    &quot;521&quot;       &quot;100129917&quot; &quot;10815&quot;    
[4993] &quot;2580&quot;      &quot;84286&quot;     &quot;1609&quot;      &quot;10861&quot;     &quot;3425&quot;      &quot;285463&quot;   
[4999] &quot;1487&quot;      &quot;10296&quot;     &quot;101928621&quot; &quot;57654&quot;     &quot;285464&quot;    &quot;152877&quot;   
[5005] &quot;92305&quot;     &quot;3954&quot;      &quot;7469&quot;      &quot;401115&quot;    &quot;339983&quot;    &quot;79441&quot;    
[5011] &quot;57732&quot;     &quot;6047&quot;      &quot;79155&quot;     &quot;6452&quot;      &quot;118&quot;       &quot;10227&quot;    
[5017] &quot;8602&quot;      &quot;2868&quot;      &quot;3064&quot;      &quot;6002&quot;      &quot;3083&quot;      &quot;85013&quot;    
[5023] &quot;55646&quot;     &quot;166793&quot;    &quot;53407&quot;     &quot;100507266&quot; &quot;1400&quot;      &quot;7466&quot;     
[5029] &quot;93621&quot;     &quot;93622&quot;     &quot;9778&quot;      &quot;93624&quot;     &quot;57537&quot;     &quot;768239&quot;   
[5035] &quot;94031&quot;     &quot;152992&quot;    &quot;8532&quot;      &quot;56606&quot;     &quot;9948&quot;      &quot;85460&quot;    
[5041] &quot;9364&quot;      &quot;26234&quot;     &quot;285550&quot;    &quot;683&quot;       &quot;8842&quot;      &quot;254251&quot;   
[5047] &quot;9353&quot;      &quot;133015&quot;    &quot;80333&quot;     &quot;6649&quot;      &quot;51091&quot;     &quot;55300&quot;    
[5053] &quot;29945&quot;     &quot;10568&quot;     &quot;389203&quot;    &quot;768211&quot;    &quot;55276&quot;     &quot;23216&quot;    
[5059] &quot;51088&quot;     &quot;57728&quot;     &quot;11019&quot;     &quot;7358&quot;      &quot;100885776&quot; &quot;23244&quot;    
[5065] &quot;55728&quot;     &quot;79730&quot;     &quot;323&quot;       &quot;7345&quot;      &quot;22998&quot;     &quot;55161&quot;    
[5071] &quot;10463&quot;     &quot;10396&quot;     &quot;60558&quot;     &quot;132789&quot;    &quot;57205&quot;     &quot;152518&quot;   
[5077] &quot;7006&quot;      &quot;285527&quot;    &quot;54940&quot;     &quot;132299&quot;    &quot;23142&quot;     &quot;6443&quot;     
[5083] &quot;57291&quot;     &quot;81608&quot;     &quot;3791&quot;      &quot;55858&quot;     &quot;9575&quot;      &quot;57482&quot;    
[5089] &quot;5471&quot;      &quot;10606&quot;     &quot;5978&quot;      &quot;3490&quot;      &quot;2044&quot;      &quot;55236&quot;    
[5095] &quot;550112&quot;    &quot;91746&quot;     &quot;6783&quot;      &quot;10117&quot;     &quot;2926&quot;      &quot;9508&quot;     
[5101] &quot;285521&quot;    &quot;174&quot;       &quot;3576&quot;      &quot;441024&quot;    &quot;374&quot;       &quot;25849&quot;    
[5107] &quot;25898&quot;     &quot;152815&quot;    &quot;8999&quot;      &quot;8615&quot;      &quot;27163&quot;     &quot;419&quot;      
[5113] &quot;53371&quot;     &quot;57619&quot;     &quot;10983&quot;     &quot;306&quot;       &quot;152559&quot;    &quot;118429&quot;   
[5119] &quot;441027&quot;    &quot;22872&quot;     &quot;79725&quot;     &quot;51138&quot;     &quot;27235&quot;     &quot;51023&quot;    
[5125] &quot;84142&quot;     &quot;84803&quot;     &quot;23001&quot;     &quot;5602&quot;      &quot;5783&quot;      &quot;100506746&quot;
[5131] &quot;57563&quot;     &quot;6696&quot;      &quot;5311&quot;      &quot;9429&quot;      &quot;55008&quot;     &quot;8916&quot;     
[5137] &quot;285512&quot;    &quot;166815&quot;    &quot;6622&quot;      &quot;401145&quot;    &quot;8633&quot;      &quot;1977&quot;     
[5143] &quot;285556&quot;    &quot;79982&quot;     &quot;5530&quot;      &quot;4790&quot;      &quot;4126&quot;      &quot;7323&quot;     
[5149] &quot;133308&quot;    &quot;56898&quot;     &quot;80319&quot;     &quot;54790&quot;     &quot;57117&quot;     &quot;79807&quot;    
[5155] &quot;255743&quot;    &quot;9255&quot;      &quot;9061&quot;      &quot;113612&quot;    &quot;6164&quot;      &quot;84570&quot;    
[5161] &quot;100533182&quot; &quot;10427&quot;     &quot;55013&quot;     &quot;81579&quot;     &quot;3426&quot;      &quot;1950&quot;     
[5167] &quot;79071&quot;     &quot;2028&quot;      &quot;132720&quot;    &quot;55435&quot;     &quot;7368&quot;      &quot;100093630&quot;
[5173] &quot;57721&quot;     &quot;54532&quot;     &quot;401152&quot;    &quot;645513&quot;    &quot;8654&quot;      &quot;11107&quot;    
[5179] &quot;79625&quot;     &quot;7222&quot;      &quot;11162&quot;     &quot;166378&quot;    &quot;10252&quot;     &quot;79960&quot;    
[5185] &quot;54510&quot;     &quot;4717&quot;      &quot;4258&quot;      &quot;55534&quot;     &quot;57484&quot;     &quot;64399&quot;    
[5191] &quot;10393&quot;     &quot;152485&quot;    &quot;11157&quot;     &quot;84068&quot;     &quot;55751&quot;     &quot;90826&quot;    
[5197] &quot;79658&quot;     &quot;4306&quot;      &quot;152503&quot;    &quot;729830&quot;    &quot;54553&quot;     &quot;23321&quot;    
[5203] &quot;303&quot;       &quot;23240&quot;     &quot;5356&quot;      &quot;2983&quot;      &quot;1519&quot;      &quot;56034&quot;    
[5209] &quot;2110&quot;      &quot;56884&quot;     &quot;55319&quot;     &quot;55016&quot;     &quot;152756&quot;    &quot;11275&quot;    
[5215] &quot;7092&quot;      &quot;55601&quot;     &quot;84869&quot;     &quot;51809&quot;     &quot;8819&quot;      &quot;79804&quot;    
[5221] &quot;80817&quot;     &quot;2823&quot;      &quot;60559&quot;     &quot;7424&quot;      &quot;175&quot;       &quot;55714&quot;    
[5227] &quot;1635&quot;      &quot;152641&quot;    &quot;80014&quot;     &quot;55602&quot;     &quot;201965&quot;    &quot;56977&quot;    
[5233] &quot;201973&quot;    &quot;57587&quot;     &quot;353322&quot;    &quot;27295&quot;     &quot;285440&quot;    &quot;2195&quot;     
[5239] &quot;2483&quot;      &quot;6389&quot;      &quot;57491&quot;     &quot;11336&quot;     &quot;11076&quot;     &quot;653082&quot;   
[5245] &quot;79844&quot;     &quot;65980&quot;     &quot;85409&quot;     &quot;79888&quot;     &quot;728609&quot;    &quot;728613&quot;   
[5251] &quot;64979&quot;     &quot;4726&quot;      &quot;50805&quot;     &quot;153572&quot;    &quot;153571&quot;    &quot;79192&quot;    
[5257] &quot;84246&quot;     &quot;108&quot;       &quot;79072&quot;     &quot;100505738&quot; &quot;22948&quot;     &quot;10299&quot;    
[5263] &quot;1611&quot;      &quot;56172&quot;     &quot;90441&quot;     &quot;728411&quot;    &quot;1008&quot;      &quot;643401&quot;   
[5269] &quot;1004&quot;      &quot;64083&quot;     &quot;54545&quot;     &quot;51663&quot;     &quot;4883&quot;      &quot;81792&quot;    
[5275] &quot;55299&quot;     &quot;167127&quot;    &quot;6502&quot;      &quot;646719&quot;    &quot;65250&quot;     &quot;9631&quot;     
[5281] &quot;3977&quot;      &quot;9180&quot;      &quot;253260&quot;    &quot;5562&quot;      &quot;100506548&quot; &quot;6167&quot;     
[5287] &quot;730&quot;       &quot;5019&quot;      &quot;285636&quot;    &quot;100132356&quot; &quot;100506639&quot; &quot;64417&quot;    
[5293] &quot;100652772&quot; &quot;100506674&quot; &quot;10884&quot;     &quot;348980&quot;    &quot;79668&quot;     &quot;3670&quot;     
[5299] &quot;53918&quot;     &quot;3672&quot;      &quot;3673&quot;      &quot;257396&quot;    &quot;54622&quot;     &quot;112574&quot;   
[5305] &quot;23517&quot;     &quot;8611&quot;      &quot;379013&quot;    &quot;153129&quot;    &quot;133396&quot;    &quot;3572&quot;     
[5311] &quot;4214&quot;      &quot;133383&quot;    &quot;166968&quot;    &quot;65056&quot;     &quot;5144&quot;      &quot;91942&quot;    
[5317] &quot;643155&quot;    &quot;3796&quot;      &quot;27292&quot;     &quot;51194&quot;     &quot;401190&quot;    &quot;285672&quot;   
[5323] &quot;373&quot;       &quot;80006&quot;     &quot;57486&quot;     &quot;55914&quot;     &quot;375449&quot;    &quot;5295&quot;     
[5329] &quot;1022&quot;      &quot;202243&quot;    &quot;102157402&quot; &quot;6880&quot;      &quot;100506658&quot; &quot;6607&quot;     
[5335] &quot;11039&quot;     &quot;11042&quot;     &quot;6606&quot;      &quot;4671&quot;      &quot;55814&quot;     &quot;64087&quot;    
[5341] &quot;4131&quot;      &quot;689&quot;       &quot;84135&quot;     &quot;8507&quot;      &quot;84340&quot;     &quot;26049&quot;    
[5347] &quot;3156&quot;      &quot;10087&quot;     &quot;2151&quot;      &quot;2149&quot;      &quot;1393&quot;      &quot;55109&quot;    
[5353] &quot;6902&quot;      &quot;8546&quot;      &quot;9522&quot;      &quot;10184&quot;     &quot;9456&quot;      &quot;167153&quot;   
[5359] &quot;345778&quot;    &quot;7060&quot;      &quot;441089&quot;    &quot;9765&quot;      &quot;340120&quot;    &quot;100462981&quot;
[5365] &quot;100131067&quot; &quot;23635&quot;     &quot;6228&quot;      &quot;7518&quot;      &quot;10085&quot;     &quot;1350&quot;     
[5371] &quot;5921&quot;      &quot;902&quot;       &quot;4208&quot;      &quot;1070&quot;      &quot;153364&quot;    &quot;10622&quot;    
[5377] &quot;116068&quot;    &quot;57561&quot;     &quot;83989&quot;     &quot;285600&quot;    &quot;22836&quot;     &quot;2745&quot;     
[5383] &quot;831&quot;       &quot;64167&quot;     &quot;4012&quot;      &quot;285704&quot;    &quot;1105&quot;      &quot;100289230&quot;
[5389] &quot;345757&quot;    &quot;5066&quot;      &quot;23262&quot;     &quot;90355&quot;     &quot;1946&quot;      &quot;64839&quot;    
[5395] &quot;9867&quot;      &quot;4124&quot;      &quot;642987&quot;    &quot;91137&quot;     &quot;114915&quot;    &quot;6728&quot;     
[5401] &quot;7905&quot;      &quot;4163&quot;      &quot;3781&quot;      &quot;5229&quot;      &quot;51014&quot;     &quot;51397&quot;    
[5407] &quot;100422885&quot; &quot;1657&quot;      &quot;51334&quot;     &quot;153443&quot;    &quot;4015&quot;      &quot;9627&quot;     
[5413] &quot;6643&quot;      &quot;5480&quot;      &quot;93166&quot;     &quot;1456&quot;      &quot;501&quot;       &quot;115123&quot;   
[5419] &quot;644873&quot;    &quot;337876&quot;    &quot;3094&quot;      &quot;56990&quot;     &quot;51735&quot;     &quot;96459&quot;    
[5425] &quot;8974&quot;      &quot;8572&quot;      &quot;553103&quot;    &quot;6584&quot;      &quot;11127&quot;     &quot;23176&quot;    
[5431] &quot;27089&quot;     &quot;27125&quot;     &quot;56951&quot;     &quot;7416&quot;      &quot;5515&quot;      &quot;7320&quot;     
[5437] &quot;91368&quot;     &quot;10802&quot;     &quot;819&quot;       &quot;9879&quot;      &quot;79770&quot;     &quot;84105&quot;    
[5443] &quot;100616209&quot; &quot;9547&quot;      &quot;7045&quot;      &quot;6695&quot;      &quot;10949&quot;     &quot;9499&quot;     
[5449] &quot;51306&quot;     &quot;10902&quot;     &quot;8697&quot;      &quot;51308&quot;     &quot;1495&quot;      &quot;64374&quot;    
[5455] &quot;724102&quot;    &quot;9782&quot;      &quot;51247&quot;     &quot;389333&quot;    &quot;202051&quot;    &quot;202052&quot;   
[5461] &quot;340061&quot;    &quot;7322&quot;      &quot;51523&quot;     &quot;492311&quot;    &quot;84418&quot;     &quot;5201&quot;     
[5467] &quot;1839&quot;      &quot;54882&quot;     &quot;10011&quot;     &quot;3550&quot;      &quot;373863&quot;    &quot;3035&quot;     
[5473] &quot;56143&quot;     &quot;56141&quot;     &quot;9752&quot;      &quot;56135&quot;     &quot;56134&quot;     &quot;56131&quot;    
[5479] &quot;26167&quot;     &quot;57717&quot;     &quot;56127&quot;     &quot;56126&quot;     &quot;56125&quot;     &quot;56124&quot;    
[5485] &quot;56121&quot;     &quot;8641&quot;      &quot;9708&quot;      &quot;56106&quot;     &quot;56099&quot;     &quot;56105&quot;    
[5491] &quot;5098&quot;      &quot;56098&quot;     &quot;8841&quot;      &quot;89848&quot;     &quot;9604&quot;      &quot;81848&quot;    
[5497] &quot;23092&quot;     &quot;81555&quot;     &quot;153768&quot;    &quot;153769&quot;    &quot;54439&quot;     &quot;5521&quot;     
[5503] &quot;1809&quot;      &quot;389336&quot;    &quot;81545&quot;     &quot;154&quot;       &quot;22885&quot;     &quot;134266&quot;   
[5509] &quot;1452&quot;      &quot;133522&quot;    &quot;644762&quot;    &quot;1836&quot;      &quot;5159&quot;      &quot;815&quot;      
[5515] &quot;340075&quot;    &quot;6208&quot;      &quot;11346&quot;     &quot;85027&quot;     &quot;6678&quot;      &quot;10146&quot;    
[5521] &quot;2890&quot;      &quot;4238&quot;      &quot;55568&quot;     &quot;79685&quot;     &quot;10826&quot;     &quot;25929&quot;    
[5527] &quot;153743&quot;    &quot;134353&quot;    &quot;9685&quot;      &quot;7265&quot;      &quot;114825&quot;    &quot;2561&quot;     
[5533] &quot;900&quot;       &quot;5917&quot;      &quot;345630&quot;    &quot;4869&quot;      &quot;6793&quot;      &quot;92181&quot;    
[5539] &quot;54492&quot;     &quot;57222&quot;     &quot;51121&quot;     &quot;8992&quot;      &quot;8614&quot;      &quot;91272&quot;    
[5545] &quot;80315&quot;     &quot;1812&quot;      &quot;94081&quot;     &quot;84321&quot;     &quot;51491&quot;     &quot;192286&quot;   
[5551] &quot;1212&quot;      &quot;23197&quot;     &quot;22838&quot;     &quot;114787&quot;    &quot;51720&quot;     &quot;23567&quot;    
[5557] &quot;64324&quot;     &quot;27166&quot;     &quot;10636&quot;     &quot;9260&quot;      &quot;79930&quot;     &quot;51428&quot;    
[5563] &quot;54540&quot;     &quot;54732&quot;     &quot;11285&quot;     &quot;202181&quot;    &quot;728554&quot;    &quot;23138&quot;    
[5569] &quot;64777&quot;     &quot;85007&quot;     &quot;57396&quot;     &quot;117608&quot;    &quot;80108&quot;     &quot;80230&quot;    
[5575] &quot;821&quot;       &quot;9794&quot;      &quot;51149&quot;     &quot;23061&quot;     &quot;55819&quot;     &quot;5601&quot;     
[5581] &quot;100859930&quot; &quot;729678&quot;    &quot;101928649&quot; &quot;84851&quot;     &quot;100507602&quot; &quot;100132062&quot;
[5587] &quot;340156&quot;    &quot;5269&quot;      &quot;401232&quot;    &quot;4835&quot;      &quot;670&quot;       &quot;7280&quot;     
[5593] &quot;347733&quot;    &quot;389362&quot;    &quot;57128&quot;     &quot;10667&quot;     &quot;6239&quot;      &quot;83732&quot;    
[5599] &quot;1832&quot;      &quot;654&quot;       &quot;81567&quot;     &quot;55003&quot;     &quot;51522&quot;     &quot;81853&quot;    
[5605] &quot;54898&quot;     &quot;221710&quot;    &quot;4739&quot;      &quot;100113407&quot; &quot;100130357&quot; &quot;51256&quot;    
[5611] &quot;54438&quot;     &quot;23408&quot;     &quot;51406&quot;     &quot;63933&quot;     &quot;3720&quot;      &quot;84062&quot;    
[5617] &quot;2766&quot;      &quot;221662&quot;    &quot;9972&quot;      &quot;7172&quot;      &quot;221656&quot;    &quot;154141&quot;   
[5623] &quot;2822&quot;      &quot;7915&quot;      &quot;9856&quot;      &quot;51567&quot;     &quot;55604&quot;     &quot;10475&quot;    
[5629] &quot;3077&quot;      &quot;8353&quot;      &quot;54718&quot;     &quot;11120&quot;     &quot;29777&quot;     &quot;100129195&quot;
[5635] &quot;80345&quot;     &quot;7745&quot;      &quot;651302&quot;    &quot;64288&quot;     &quot;5987&quot;      &quot;3135&quot;     
[5641] &quot;3136&quot;      &quot;3105&quot;      &quot;100507399&quot; &quot;30834&quot;     &quot;6992&quot;      &quot;3139&quot;     
[5647] &quot;414777&quot;    &quot;23&quot;        &quot;5514&quot;      &quot;79969&quot;     &quot;8449&quot;      &quot;10211&quot;    
[5653] &quot;2968&quot;      &quot;57176&quot;     &quot;54535&quot;     &quot;100507436&quot; &quot;4277&quot;      &quot;692199&quot;   
[5659] &quot;7917&quot;      &quot;7918&quot;      &quot;58496&quot;     &quot;80737&quot;     &quot;7407&quot;      &quot;57819&quot;    
[5665] &quot;3305&quot;      &quot;3303&quot;      &quot;50854&quot;     &quot;26797&quot;     &quot;7936&quot;      &quot;6499&quot;     
[5671] &quot;8859&quot;      &quot;720&quot;       &quot;1590&quot;      &quot;7146&quot;      &quot;721&quot;       &quot;7148&quot;     
[5677] &quot;1388&quot;      &quot;10554&quot;     &quot;6048&quot;      &quot;177&quot;       &quot;63940&quot;     &quot;1302&quot;     
[5683] &quot;6257&quot;      &quot;7923&quot;      &quot;6015&quot;      &quot;6293&quot;      &quot;6222&quot;      &quot;8705&quot;     
[5689] &quot;6892&quot;      &quot;1616&quot;      &quot;5252&quot;      &quot;51596&quot;     &quot;221504&quot;    &quot;578&quot;      
[5695] &quot;221496&quot;    &quot;6631&quot;      &quot;222663&quot;    &quot;7629&quot;      &quot;5467&quot;      &quot;4736&quot;     
[5701] &quot;7005&quot;      &quot;57699&quot;     &quot;221477&quot;    &quot;221476&quot;    &quot;9025&quot;      &quot;23070&quot;    
[5707] &quot;154467&quot;    &quot;60685&quot;     &quot;2739&quot;      &quot;2740&quot;      &quot;101929555&quot; &quot;10930&quot;    
[5713] &quot;221442&quot;    &quot;4188&quot;      &quot;7942&quot;      &quot;10817&quot;     &quot;9477&quot;      &quot;55173&quot;    
[5719] &quot;55809&quot;     &quot;23304&quot;     &quot;401261&quot;    &quot;285855&quot;    &quot;10695&quot;     &quot;4201&quot;     
[5725] &quot;5528&quot;      &quot;9820&quot;      &quot;51069&quot;     &quot;89953&quot;     &quot;5754&quot;      &quot;6722&quot;     
[5731] &quot;23113&quot;     &quot;24149&quot;     &quot;89845&quot;     &quot;221424&quot;    &quot;57510&quot;     &quot;9587&quot;     
[5737] &quot;55168&quot;     &quot;7422&quot;      &quot;101929705&quot; &quot;64928&quot;     &quot;3326&quot;      &quot;988&quot;      
[5743] &quot;8464&quot;      &quot;53405&quot;     &quot;51302&quot;     &quot;9481&quot;      &quot;442213&quot;    &quot;135398&quot;   
[5749] &quot;85315&quot;     &quot;114327&quot;    &quot;9697&quot;      &quot;125050&quot;    &quot;22858&quot;     &quot;26268&quot;    
[5755] &quot;60481&quot;     &quot;55227&quot;     &quot;221336&quot;    &quot;57691&quot;     &quot;9532&quot;      &quot;375513&quot;   
[5761] &quot;100463487&quot; &quot;23469&quot;     &quot;441155&quot;    &quot;1310&quot;      &quot;1297&quot;      &quot;57579&quot;    
[5767] &quot;135154&quot;    &quot;60682&quot;     &quot;79940&quot;     &quot;80759&quot;     &quot;25821&quot;     &quot;1915&quot;     
[5773] &quot;26503&quot;     &quot;1303&quot;      &quot;1347&quot;      &quot;55754&quot;     &quot;26054&quot;     &quot;3351&quot;     
[5779] &quot;83699&quot;     &quot;594&quot;       &quot;5238&quot;      &quot;112611&quot;    &quot;4199&quot;      &quot;9892&quot;     
[5785] &quot;51167&quot;     &quot;22832&quot;     &quot;9096&quot;      &quot;4907&quot;      &quot;10492&quot;     &quot;387066&quot;   
[5791] &quot;57038&quot;     &quot;55122&quot;     &quot;8732&quot;      &quot;10957&quot;     &quot;135295&quot;    &quot;2570&quot;     
[5797] &quot;23195&quot;     &quot;60468&quot;     &quot;81491&quot;     &quot;29078&quot;     &quot;26235&quot;     &quot;51805&quot;    
[5803] &quot;25957&quot;     &quot;85015&quot;     &quot;892&quot;       &quot;10973&quot;     &quot;57531&quot;     &quot;389421&quot;   
[5809] &quot;9474&quot;      &quot;100133941&quot; &quot;57673&quot;     &quot;57107&quot;     &quot;55084&quot;     &quot;8724&quot;     
[5815] &quot;246269&quot;    &quot;84071&quot;     &quot;285755&quot;    &quot;6610&quot;      &quot;64780&quot;     &quot;9841&quot;     
[5821] &quot;221264&quot;    &quot;9896&quot;      &quot;84154&quot;     &quot;442245&quot;    &quot;91749&quot;     &quot;10758&quot;    
[5827] &quot;2534&quot;      &quot;3910&quot;      &quot;221294&quot;    &quot;23270&quot;     &quot;7259&quot;      &quot;29940&quot;    
[5833] &quot;51389&quot;     &quot;345895&quot;    &quot;221302&quot;    &quot;3841&quot;      &quot;285761&quot;    &quot;57120&quot;    
[5839] &quot;116150&quot;    &quot;254394&quot;    &quot;25842&quot;     &quot;3298&quot;      &quot;57515&quot;     &quot;10924&quot;    
[5845] &quot;7164&quot;      &quot;135112&quot;    &quot;135114&quot;    &quot;84870&quot;     &quot;55862&quot;     &quot;5796&quot;     
[5851] &quot;3908&quot;      &quot;93663&quot;     &quot;9465&quot;      &quot;9439&quot;      &quot;26002&quot;     &quot;8417&quot;     
[5857] &quot;2070&quot;      &quot;154091&quot;    &quot;10767&quot;     &quot;54806&quot;     &quot;9774&quot;      &quot;9053&quot;     
[5863] &quot;340146&quot;    &quot;53832&quot;     &quot;3459&quot;      &quot;7128&quot;      &quot;23593&quot;     &quot;25901&quot;    
[5869] &quot;57211&quot;     &quot;153910&quot;    &quot;84946&quot;     &quot;5325&quot;      &quot;83443&quot;     &quot;8676&quot;     
[5875] &quot;7957&quot;      &quot;84085&quot;     &quot;10090&quot;     &quot;23118&quot;     &quot;85313&quot;     &quot;116254&quot;   
[5881] &quot;645958&quot;    &quot;11104&quot;     &quot;9113&quot;      &quot;348995&quot;    &quot;5110&quot;      &quot;84918&quot;    
[5887] &quot;79465&quot;     &quot;81706&quot;     &quot;57480&quot;     &quot;25902&quot;     &quot;55005&quot;     &quot;23345&quot;    
[5893] &quot;22828&quot;     &quot;57492&quot;     &quot;79683&quot;     &quot;51429&quot;     &quot;84947&quot;     &quot;56995&quot;    
[5899] &quot;6993&quot;      &quot;94120&quot;     &quot;83861&quot;     &quot;6648&quot;      &quot;39&quot;        &quot;6950&quot;     
[5905] &quot;6581&quot;      &quot;4216&quot;      &quot;79992&quot;     &quot;113402&quot;    &quot;6196&quot;      &quot;8635&quot;     
[5911] &quot;4301&quot;      &quot;7058&quot;      &quot;6991&quot;      &quot;55780&quot;     &quot;84498&quot;     &quot;5134&quot;     
[5917] &quot;56975&quot;     &quot;5154&quot;      &quot;54919&quot;     &quot;51608&quot;     &quot;90639&quot;     &quot;84310&quot;    
[5923] &quot;26173&quot;     &quot;7975&quot;      &quot;114796&quot;    &quot;8379&quot;      &quot;4521&quot;      &quot;29886&quot;    
[5929] &quot;55501&quot;     &quot;101927181&quot; &quot;221927&quot;    &quot;23288&quot;     &quot;80727&quot;     &quot;2768&quot;     
[5935] &quot;57786&quot;     &quot;222962&quot;    &quot;84629&quot;     &quot;51622&quot;     &quot;5395&quot;      &quot;7965&quot;     
[5941] &quot;27102&quot;     &quot;5879&quot;      &quot;79034&quot;     &quot;7559&quot;      &quot;441194&quot;    &quot;56913&quot;    
[5947] &quot;101927391&quot; &quot;54468&quot;     &quot;729852&quot;    &quot;113263&quot;    &quot;4697&quot;      &quot;221806&quot;   
[5953] &quot;2115&quot;      &quot;729920&quot;    &quot;28969&quot;     &quot;27075&quot;     &quot;222901&quot;    &quot;6671&quot;     
[5959] &quot;8701&quot;      &quot;54543&quot;     &quot;84668&quot;     &quot;55975&quot;     &quot;115416&quot;    &quot;256355&quot;   
[5965] &quot;29896&quot;     &quot;1197&quot;      &quot;90693&quot;     &quot;51678&quot;     &quot;54205&quot;     &quot;136895&quot;   
[5971] &quot;9603&quot;      &quot;11335&quot;     &quot;8935&quot;      &quot;11112&quot;     &quot;9586&quot;      &quot;54504&quot;    
[5977] &quot;1124&quot;      &quot;646762&quot;    &quot;9805&quot;      &quot;55033&quot;     &quot;222166&quot;    &quot;223082&quot;   
[5983] &quot;222161&quot;    &quot;23658&quot;     &quot;23080&quot;     &quot;100129460&quot; &quot;441208&quot;    &quot;102724668&quot;
[5989] &quot;25948&quot;     &quot;11328&quot;     &quot;51251&quot;     &quot;989&quot;       &quot;23366&quot;     &quot;54749&quot;    
[5995] &quot;273&quot;       &quot;57002&quot;     &quot;349114&quot;    &quot;8621&quot;      &quot;285954&quot;    &quot;2737&quot;     
[6001] &quot;79020&quot;     &quot;64983&quot;     &quot;644&quot;       &quot;64951&quot;     &quot;51619&quot;     &quot;84820&quot;    
[6007] &quot;285955&quot;    &quot;401331&quot;    &quot;28988&quot;     &quot;10652&quot;     &quot;816&quot;       &quot;23386&quot;    
[6013] &quot;54606&quot;     &quot;4967&quot;      &quot;83637&quot;     &quot;94239&quot;     &quot;5814&quot;      &quot;285958&quot;   
[6019] &quot;9238&quot;      &quot;64759&quot;     &quot;168507&quot;    &quot;168448&quot;    &quot;375567&quot;    &quot;2887&quot;     
[6025] &quot;23480&quot;     &quot;1956&quot;      &quot;360132&quot;    &quot;2631&quot;      &quot;5723&quot;      &quot;51142&quot;    
[6031] &quot;728927&quot;    &quot;649395&quot;    &quot;340252&quot;    &quot;7697&quot;      &quot;10793&quot;     &quot;51351&quot;    
[6037] &quot;2086&quot;      &quot;168374&quot;    &quot;441242&quot;    &quot;2990&quot;      &quot;27297&quot;     &quot;285908&quot;   
[6043] &quot;100289098&quot; &quot;493754&quot;    &quot;154881&quot;    &quot;644794&quot;    &quot;55069&quot;     &quot;64940&quot;    
[6049] &quot;441250&quot;    &quot;260294&quot;    &quot;442578&quot;    &quot;55695&quot;     &quot;9275&quot;      &quot;84277&quot;    
[6055] &quot;6804&quot;      &quot;83451&quot;     &quot;1365&quot;      &quot;1364&quot;      &quot;3984&quot;      &quot;7458&quot;     
[6061] &quot;84163&quot;     &quot;5383&quot;      &quot;389524&quot;    &quot;5380&quot;      &quot;54441&quot;     &quot;155400&quot;   
[6067] &quot;5387&quot;      &quot;83862&quot;     &quot;51657&quot;     &quot;4191&quot;      &quot;3315&quot;      &quot;7532&quot;     
[6073] &quot;113878&quot;    &quot;105375355&quot; &quot;441263&quot;    &quot;57639&quot;     &quot;54103&quot;     &quot;5782&quot;     
[6079] &quot;57157&quot;     &quot;9863&quot;      &quot;10512&quot;     &quot;781&quot;       &quot;9723&quot;      &quot;223117&quot;   
[6085] &quot;222223&quot;    &quot;9988&quot;      &quot;11257&quot;     &quot;5244&quot;      &quot;53616&quot;     &quot;442523&quot;   
[6091] &quot;261729&quot;    &quot;79846&quot;     &quot;85865&quot;     &quot;8321&quot;      &quot;889&quot;       &quot;54467&quot;    
[6097] &quot;257415&quot;    &quot;1021&quot;      &quot;10282&quot;     &quot;1278&quot;      &quot;64921&quot;     &quot;23089&quot;    
[6103] &quot;5445&quot;      &quot;10165&quot;     &quot;7979&quot;      &quot;57001&quot;     &quot;389538&quot;    &quot;22853&quot;    
[6109] &quot;25851&quot;     &quot;25798&quot;     &quot;8295&quot;      &quot;101927550&quot; &quot;10552&quot;     &quot;11333&quot;    
[6115] &quot;8896&quot;      &quot;9551&quot;      &quot;285989&quot;    &quot;79027&quot;     &quot;221785&quot;    &quot;89122&quot;    
[6121] &quot;7586&quot;      &quot;7589&quot;      &quot;7551&quot;      &quot;10980&quot;     &quot;9179&quot;      &quot;6878&quot;     
[6127] &quot;245812&quot;    &quot;389541&quot;    &quot;221914&quot;    &quot;79037&quot;     &quot;441272&quot;    &quot;5379&quot;     
[6133] &quot;101752399&quot; &quot;55063&quot;     &quot;56257&quot;     &quot;221908&quot;    &quot;81628&quot;     &quot;222950&quot;   
[6139] &quot;3268&quot;      &quot;100316904&quot; &quot;5118&quot;      &quot;2783&quot;      &quot;64599&quot;     &quot;10248&quot;    
[6145] &quot;2050&quot;      &quot;56996&quot;     &quot;7205&quot;      &quot;81844&quot;     &quot;5054&quot;      &quot;7425&quot;     
[6151] &quot;51024&quot;     &quot;1523&quot;      &quot;729597&quot;    &quot;79706&quot;     &quot;80228&quot;     &quot;5439&quot;     
[6157] &quot;100271927&quot; &quot;548644&quot;    &quot;441273&quot;    &quot;10156&quot;     &quot;246721&quot;    &quot;100134938&quot;
[6163] &quot;100310812&quot; &quot;100129424&quot; &quot;27000&quot;     &quot;5701&quot;      &quot;5649&quot;      &quot;54517&quot;    
[6169] &quot;168451&quot;    &quot;4189&quot;      &quot;83943&quot;     &quot;9732&quot;      &quot;11179&quot;     &quot;401397&quot;   
[6175] &quot;26136&quot;     &quot;858&quot;       &quot;857&quot;       &quot;830&quot;       &quot;93653&quot;     &quot;51691&quot;    
[6181] &quot;23554&quot;     &quot;54556&quot;     &quot;79974&quot;     &quot;93664&quot;     &quot;4698&quot;      &quot;8976&quot;     
[6187] &quot;168850&quot;    &quot;401399&quot;    &quot;3614&quot;      &quot;402483&quot;    &quot;346653&quot;    &quot;813&quot;      
[6193] &quot;2318&quot;      &quot;23534&quot;     &quot;286016&quot;    &quot;407835&quot;    &quot;340348&quot;    &quot;6608&quot;     
[6199] &quot;57464&quot;     &quot;105375620&quot; &quot;7328&quot;      &quot;51530&quot;     &quot;51200&quot;     &quot;4232&quot;     
[6205] &quot;26958&quot;     &quot;5420&quot;      &quot;100500804&quot; &quot;231&quot;       &quot;800&quot;       &quot;340351&quot;   
[6211] &quot;78996&quot;     &quot;4850&quot;      &quot;647087&quot;    &quot;26266&quot;     &quot;5764&quot;      &quot;9162&quot;     
[6217] &quot;64764&quot;     &quot;8805&quot;      &quot;57670&quot;     &quot;92092&quot;     &quot;51631&quot;     &quot;154791&quot;   
[6223] &quot;28996&quot;     &quot;64761&quot;     &quot;84255&quot;     &quot;27147&quot;     &quot;90956&quot;     &quot;4708&quot;     
[6229] &quot;673&quot;       &quot;51650&quot;     &quot;57189&quot;     &quot;6742&quot;      &quot;100463482&quot; &quot;3792&quot;     
[6235] &quot;7791&quot;      &quot;728377&quot;    &quot;7984&quot;      &quot;26047&quot;     &quot;8454&quot;      &quot;9601&quot;     
[6241] &quot;57541&quot;     &quot;8427&quot;      &quot;155060&quot;    &quot;155061&quot;    &quot;79970&quot;     &quot;23145&quot;    
[6247] &quot;155066&quot;    &quot;65999&quot;     &quot;113763&quot;    &quot;5919&quot;      &quot;29803&quot;     &quot;285971&quot;   
[6253] &quot;728743&quot;    &quot;28959&quot;     &quot;55365&quot;     &quot;285973&quot;    &quot;11194&quot;     &quot;83590&quot;    
[6259] &quot;54480&quot;     &quot;51667&quot;     &quot;349136&quot;    &quot;63917&quot;     &quot;58508&quot;     &quot;100128822&quot;
[6265] &quot;202781&quot;    &quot;3638&quot;      &quot;100506302&quot; &quot;129790&quot;    &quot;140545&quot;    &quot;9690&quot;     
[6271] &quot;10049&quot;     &quot;7434&quot;      &quot;644128&quot;    &quot;101927595&quot; &quot;157695&quot;    &quot;157697&quot;   
[6277] &quot;2055&quot;      &quot;9639&quot;      &quot;9920&quot;      &quot;9172&quot;      &quot;79648&quot;     &quot;55326&quot;    
[6283] &quot;90459&quot;     &quot;8658&quot;      &quot;4482&quot;      &quot;54984&quot;     &quot;286046&quot;    &quot;66036&quot;    
[6289] &quot;100506990&quot; &quot;7991&quot;      &quot;29883&quot;     &quot;5157&quot;      &quot;57509&quot;     &quot;5108&quot;     
[6295] &quot;427&quot;       &quot;101929066&quot; &quot;63898&quot;     &quot;55174&quot;     &quot;11178&quot;     &quot;80346&quot;    
[6301] &quot;649&quot;       &quot;9796&quot;      &quot;661&quot;       &quot;5533&quot;      &quot;541565&quot;    &quot;23221&quot;    
[6307] &quot;9583&quot;      &quot;51312&quot;     &quot;6781&quot;      &quot;80005&quot;     &quot;54793&quot;     &quot;665&quot;      
[6313] &quot;10687&quot;     &quot;1808&quot;      &quot;51435&quot;     &quot;55246&quot;     &quot;55140&quot;     &quot;7976&quot;     
[6319] &quot;2137&quot;      &quot;55756&quot;     &quot;51669&quot;     &quot;10671&quot;     &quot;2961&quot;      &quot;29942&quot;    
[6325] &quot;79845&quot;     &quot;100507420&quot; &quot;11160&quot;     &quot;728024&quot;    &quot;11212&quot;     &quot;25960&quot;    
[6331] &quot;55290&quot;     &quot;80223&quot;     &quot;9070&quot;      &quot;27257&quot;     &quot;9530&quot;      &quot;84513&quot;    
[6337] &quot;54904&quot;     &quot;2260&quot;      &quot;286140&quot;    &quot;59339&quot;     &quot;83877&quot;     &quot;8754&quot;     
[6343] &quot;79698&quot;     &quot;51125&quot;     &quot;10947&quot;     &quot;5423&quot;      &quot;7419&quot;      &quot;81790&quot;    
[6349] &quot;2339&quot;      &quot;84197&quot;     &quot;138050&quot;    &quot;5591&quot;      &quot;7336&quot;      &quot;6591&quot;     
[6355] &quot;10434&quot;     &quot;64321&quot;     &quot;96764&quot;     &quot;4067&quot;      &quot;5324&quot;      &quot;54928&quot;    
[6361] &quot;90362&quot;     &quot;137886&quot;    &quot;8439&quot;      &quot;9760&quot;      &quot;5862&quot;      &quot;55636&quot;    
[6367] &quot;444&quot;       &quot;9420&quot;      &quot;55156&quot;     &quot;9650&quot;      &quot;84675&quot;     &quot;137872&quot;   
[6373] &quot;56260&quot;     &quot;641638&quot;    &quot;79848&quot;     &quot;10565&quot;     &quot;23213&quot;     &quot;10499&quot;    
[6379] &quot;157869&quot;    &quot;6129&quot;      &quot;27067&quot;     &quot;55284&quot;     &quot;54968&quot;     &quot;56704&quot;    
[6385] &quot;54332&quot;     &quot;83690&quot;     &quot;101805492&quot; &quot;79776&quot;     &quot;5569&quot;      &quot;51101&quot;    
[6391] &quot;11075&quot;     &quot;23462&quot;     &quot;28957&quot;     &quot;619279&quot;    &quot;3612&quot;      &quot;92421&quot;    
[6397] &quot;401466&quot;    &quot;11059&quot;     &quot;51115&quot;     &quot;8895&quot;      &quot;8767&quot;      &quot;1666&quot;     
[6403] &quot;55529&quot;     &quot;286144&quot;    &quot;137392&quot;    &quot;55472&quot;     &quot;2669&quot;      &quot;25962&quot;    
[6409] &quot;286148&quot;    &quot;55656&quot;     &quot;7381&quot;      &quot;6383&quot;      &quot;10404&quot;     &quot;55353&quot;    
[6415] &quot;4147&quot;      &quot;6156&quot;      &quot;10247&quot;     &quot;10940&quot;     &quot;6788&quot;      &quot;1345&quot;     
[6421] &quot;25897&quot;     &quot;26986&quot;     &quot;83955&quot;     &quot;528&quot;       &quot;8323&quot;      &quot;115908&quot;   
[6427] &quot;81034&quot;     &quot;29967&quot;     &quot;55074&quot;     &quot;137735&quot;    &quot;3646&quot;      &quot;7227&quot;     
[6433] &quot;5885&quot;      &quot;441376&quot;    &quot;90390&quot;     &quot;6873&quot;      &quot;7373&quot;      &quot;28998&quot;    
[6439] &quot;22882&quot;     &quot;93594&quot;     &quot;11244&quot;     &quot;55093&quot;     &quot;157378&quot;    &quot;55039&quot;    
[6445] &quot;101927612&quot; &quot;11236&quot;     &quot;83940&quot;     &quot;4715&quot;      &quot;137209&quot;    &quot;9897&quot;     
[6451] &quot;286053&quot;    &quot;157638&quot;    &quot;101867536&quot; &quot;4609&quot;      &quot;51571&quot;     &quot;23167&quot;    
[6457] &quot;23639&quot;     &quot;51105&quot;     &quot;6482&quot;      &quot;57623&quot;     &quot;169044&quot;    &quot;54108&quot;    
[6463] &quot;27161&quot;     &quot;5747&quot;      &quot;22898&quot;     &quot;11156&quot;     &quot;389690&quot;    &quot;203062&quot;   
[6469] &quot;4061&quot;      &quot;286128&quot;    &quot;2738&quot;      &quot;100507316&quot; &quot;116447&quot;    &quot;114822&quot;   
[6475] &quot;23144&quot;     &quot;79792&quot;     &quot;93100&quot;     &quot;84948&quot;     &quot;286075&quot;    &quot;100128338&quot;
[6481] &quot;22827&quot;     &quot;83481&quot;     &quot;2907&quot;      &quot;81858&quot;     &quot;51236&quot;     &quot;727957&quot;   
[6487] &quot;23246&quot;     &quot;642658&quot;    &quot;3297&quot;      &quot;340393&quot;    &quot;26233&quot;     &quot;79581&quot;    
[6493] &quot;29894&quot;     &quot;55630&quot;     &quot;84988&quot;     &quot;2875&quot;      &quot;113655&quot;    &quot;9684&quot;     
[6499] &quot;80728&quot;     &quot;90987&quot;     &quot;6132&quot;      &quot;7553&quot;      &quot;28991&quot;     &quot;58500&quot;    
[6505] &quot;7564&quot;      &quot;286101&quot;    &quot;286102&quot;    &quot;100287171&quot; &quot;55871&quot;     &quot;9933&quot;     
[6511] &quot;169792&quot;    &quot;10171&quot;     &quot;3717&quot;      &quot;55848&quot;     &quot;29126&quot;     &quot;79956&quot;    
[6517] &quot;26953&quot;     &quot;115426&quot;    &quot;23081&quot;     &quot;90871&quot;     &quot;7306&quot;      &quot;286343&quot;   
[6523] &quot;8777&quot;      &quot;4781&quot;      &quot;6619&quot;      &quot;54801&quot;     &quot;55667&quot;     &quot;6194&quot;     
[6529] &quot;1029&quot;      &quot;1030&quot;      &quot;1993&quot;      &quot;286319&quot;    &quot;79886&quot;     &quot;9373&quot;     
[6535] &quot;80173&quot;     &quot;203228&quot;    &quot;48&quot;        &quot;23586&quot;     &quot;10210&quot;     &quot;4712&quot;     
[6541] &quot;138474&quot;    &quot;54840&quot;     &quot;3301&quot;      &quot;4799&quot;      &quot;65083&quot;     &quot;441459&quot;   
[6547] &quot;304&quot;       &quot;54926&quot;     &quot;25853&quot;     &quot;51271&quot;     &quot;57462&quot;     &quot;84688&quot;    
[6553] &quot;138716&quot;    &quot;11258&quot;     &quot;10280&quot;     &quot;3590&quot;      &quot;23349&quot;     &quot;25822&quot;    
[6559] &quot;7415&quot;      &quot;30968&quot;     &quot;7169&quot;      &quot;9827&quot;      &quot;4882&quot;      &quot;26206&quot;    
[6565] &quot;1211&quot;      &quot;10020&quot;     &quot;152006&quot;    &quot;84186&quot;     &quot;9380&quot;      &quot;9925&quot;     
[6571] &quot;64425&quot;     &quot;51010&quot;     &quot;79269&quot;     &quot;92014&quot;     &quot;100289137&quot; &quot;253650&quot;   
[6577] &quot;79937&quot;     &quot;401509&quot;    &quot;389741&quot;    &quot;105376047&quot; &quot;100132352&quot; &quot;595135&quot;   
[6583] &quot;644019&quot;    &quot;105376066&quot; &quot;728577&quot;    &quot;728673&quot;    &quot;220869&quot;    &quot;26149&quot;    
[6589] &quot;643827&quot;    &quot;100131997&quot; &quot;445571&quot;    &quot;116224&quot;    &quot;320&quot;       &quot;256691&quot;   
[6595] &quot;23137&quot;     &quot;687&quot;       &quot;7763&quot;      &quot;301&quot;       &quot;138199&quot;    &quot;54981&quot;    
[6601] &quot;26578&quot;     &quot;653162&quot;    &quot;2650&quot;      &quot;158471&quot;    &quot;23230&quot;     &quot;29968&quot;    
[6607] &quot;414328&quot;    &quot;29979&quot;     &quot;80318&quot;     &quot;55582&quot;     &quot;3190&quot;      &quot;80010&quot;    
[6613] &quot;23287&quot;     &quot;60560&quot;     &quot;51280&quot;     &quot;81689&quot;     &quot;79670&quot;     &quot;440173&quot;   
[6619] &quot;1612&quot;      &quot;23552&quot;     &quot;286223&quot;    &quot;1903&quot;      &quot;53358&quot;     &quot;10507&quot;    
[6625] &quot;10912&quot;     &quot;549&quot;       &quot;3376&quot;      &quot;55035&quot;     &quot;65268&quot;     &quot;158293&quot;   
[6631] &quot;5253&quot;      &quot;138639&quot;    &quot;54754&quot;     &quot;84641&quot;     &quot;84909&quot;     &quot;5727&quot;     
[6637] &quot;100128782&quot; &quot;195827&quot;    &quot;22869&quot;     &quot;158431&quot;    &quot;441457&quot;    &quot;84278&quot;    
[6643] &quot;23424&quot;     &quot;158427&quot;    &quot;4686&quot;      &quot;7507&quot;      &quot;51531&quot;     &quot;10541&quot;    
[6649] &quot;54187&quot;     &quot;203286&quot;    &quot;79695&quot;     &quot;7046&quot;      &quot;85365&quot;     &quot;10952&quot;    
[6655] &quot;55014&quot;     &quot;27130&quot;     &quot;91283&quot;     &quot;8577&quot;      &quot;54534&quot;     &quot;7743&quot;     
[6661] &quot;84302&quot;     &quot;56254&quot;     &quot;116443&quot;    &quot;25934&quot;     &quot;55335&quot;     &quot;83856&quot;    
[6667] &quot;55151&quot;     &quot;58499&quot;     &quot;5887&quot;      &quot;23731&quot;     &quot;11217&quot;     &quot;79987&quot;    
[6673] &quot;23392&quot;     &quot;158399&quot;    &quot;22949&quot;     &quot;7357&quot;      &quot;64420&quot;     &quot;158405&quot;   
[6679] &quot;1318&quot;      &quot;23307&quot;     &quot;1317&quot;      &quot;246184&quot;    &quot;9128&quot;      &quot;114987&quot;   
[6685] &quot;54836&quot;     &quot;5998&quot;      &quot;114991&quot;    &quot;80709&quot;     &quot;203197&quot;    &quot;3371&quot;     
[6691] &quot;5069&quot;      &quot;22954&quot;     &quot;1955&quot;      &quot;26190&quot;     &quot;5711&quot;      &quot;253039&quot;   
[6697] &quot;7185&quot;      &quot;727&quot;       &quot;51552&quot;     &quot;57000&quot;     &quot;153090&quot;    &quot;158135&quot;   
[6703] &quot;92400&quot;     &quot;5082&quot;      &quot;54542&quot;     &quot;10773&quot;     &quot;57684&quot;     &quot;23637&quot;    
[6709] &quot;81571&quot;     &quot;55342&quot;     &quot;57706&quot;     &quot;10783&quot;     &quot;5695&quot;      &quot;2649&quot;     
[6715] &quot;100379345&quot; &quot;11224&quot;     &quot;2800&quot;      &quot;5537&quot;      &quot;10244&quot;     &quot;26130&quot;    
[6721] &quot;79109&quot;     &quot;5090&quot;      &quot;89853&quot;     &quot;84253&quot;     &quot;64855&quot;     &quot;286207&quot;   
[6727] &quot;138428&quot;    &quot;27433&quot;     &quot;2356&quot;      &quot;203&quot;       &quot;30815&quot;     &quot;8818&quot;     
[6733] &quot;100289019&quot; &quot;25792&quot;     &quot;1759&quot;      &quot;375757&quot;    &quot;51117&quot;     &quot;10999&quot;    
[6739] &quot;81605&quot;     &quot;51148&quot;     &quot;89891&quot;     &quot;6418&quot;      &quot;2021&quot;      &quot;51490&quot;    
[6745] &quot;883&quot;       &quot;56262&quot;     &quot;22845&quot;     &quot;23511&quot;     &quot;57171&quot;     &quot;9536&quot;     
[6751] &quot;51759&quot;     &quot;23048&quot;     &quot;57720&quot;     &quot;25&quot;        &quot;10319&quot;     &quot;83543&quot;    
[6757] &quot;8021&quot;      &quot;84726&quot;     &quot;83549&quot;     &quot;2889&quot;      &quot;9442&quot;      &quot;64794&quot;    
[6763] &quot;11092&quot;     &quot;7248&quot;      &quot;1056&quot;      &quot;6838&quot;      &quot;6130&quot;      &quot;26815&quot;    
[6769] &quot;6835&quot;      &quot;6836&quot;      &quot;57109&quot;     &quot;11093&quot;     &quot;11094&quot;     &quot;1757&quot;     
[6775] &quot;7410&quot;      &quot;8019&quot;      &quot;11091&quot;     &quot;10439&quot;     &quot;9858&quot;      &quot;138162&quot;   
[6781] &quot;57582&quot;     &quot;157922&quot;    &quot;10422&quot;     &quot;90120&quot;     &quot;728489&quot;    &quot;6621&quot;     
[6787] &quot;56623&quot;     &quot;9919&quot;      &quot;51162&quot;     &quot;10555&quot;     &quot;84973&quot;     &quot;100131193&quot;
[6793] &quot;55684&quot;     &quot;389813&quot;    &quot;158056&quot;    &quot;8721&quot;      &quot;286256&quot;    &quot;5730&quot;     
[6799] &quot;20&quot;        &quot;56654&quot;     &quot;91373&quot;     &quot;11253&quot;     &quot;29952&quot;     &quot;29882&quot;    
[6805] &quot;286262&quot;    &quot;94107&quot;     &quot;727800&quot;    &quot;375791&quot;    &quot;25920&quot;     &quot;54863&quot;    
[6811] &quot;54932&quot;     &quot;10811&quot;     &quot;375775&quot;    &quot;116225&quot;    &quot;85026&quot;     &quot;774&quot;      
[6817] &quot;8225&quot;      &quot;3563&quot;      &quot;293&quot;       &quot;105373102&quot; &quot;751580&quot;    &quot;80161&quot;    
[6823] &quot;8623&quot;      &quot;9189&quot;      &quot;401577&quot;    &quot;4267&quot;      &quot;414&quot;       &quot;415&quot;      
[6829] &quot;57502&quot;     &quot;8226&quot;      &quot;8228&quot;      &quot;3730&quot;      &quot;357&quot;       &quot;1183&quot;     
[6835] &quot;3052&quot;      &quot;395&quot;       &quot;10943&quot;     &quot;7114&quot;      &quot;9367&quot;      &quot;6399&quot;     
[6841] &quot;8481&quot;      &quot;2824&quot;      &quot;54960&quot;     &quot;286480&quot;    &quot;158747&quot;    &quot;59272&quot;    
[6847] &quot;340591&quot;    &quot;11238&quot;     &quot;8233&quot;      &quot;8905&quot;      &quot;55787&quot;     &quot;5931&quot;     
[6853] &quot;9185&quot;      &quot;10389&quot;     &quot;6792&quot;      &quot;5256&quot;      &quot;5160&quot;      &quot;389840&quot;   
[6859] &quot;30011&quot;     &quot;256643&quot;    &quot;256714&quot;    &quot;1964&quot;      &quot;6197&quot;      &quot;404281&quot;   
[6865] &quot;6611&quot;      &quot;139411&quot;    &quot;10549&quot;     &quot;1968&quot;      &quot;7543&quot;      &quot;5165&quot;     
[6871] &quot;5422&quot;      &quot;2710&quot;      &quot;5638&quot;      &quot;7504&quot;      &quot;94122&quot;     &quot;8406&quot;     
[6877] &quot;7102&quot;      &quot;58526&quot;     &quot;54880&quot;     &quot;159013&quot;    &quot;9282&quot;      &quot;8239&quot;     
[6883] &quot;1654&quot;      &quot;4128&quot;      &quot;4129&quot;      &quot;139341&quot;    &quot;7403&quot;      &quot;101927574&quot;
[6889] &quot;55634&quot;     &quot;641339&quot;    &quot;401588&quot;    &quot;84679&quot;     &quot;7317&quot;      &quot;8552&quot;     
[6895] &quot;5127&quot;      &quot;203414&quot;    &quot;6853&quot;      &quot;7076&quot;      &quot;2002&quot;      &quot;8409&quot;     
[6901] &quot;347344&quot;    &quot;7569&quot;      &quot;57232&quot;     &quot;24140&quot;     &quot;64840&quot;     &quot;5935&quot;     
[6907] &quot;10013&quot;     &quot;27344&quot;     &quot;7355&quot;      &quot;3750&quot;      &quot;56850&quot;     &quot;11230&quot;    
[6913] &quot;11152&quot;     &quot;27238&quot;     &quot;4007&quot;      &quot;28952&quot;     &quot;1184&quot;      &quot;9500&quot;     
[6919] &quot;8242&quot;      &quot;23096&quot;     &quot;8243&quot;      &quot;3028&quot;      &quot;10075&quot;     &quot;2245&quot;     
[6925] &quot;54552&quot;     &quot;10916&quot;     &quot;90736&quot;     &quot;158880&quot;    &quot;11279&quot;     &quot;550643&quot;   
[6931] &quot;442454&quot;    &quot;474343&quot;    &quot;54466&quot;     &quot;158586&quot;    &quot;7789&quot;      &quot;55613&quot;    
[6937] &quot;340554&quot;    &quot;81887&quot;     &quot;9843&quot;      &quot;286451&quot;    &quot;9754&quot;      &quot;64219&quot;    
[6943] &quot;27112&quot;     &quot;51248&quot;     &quot;1741&quot;      &quot;4303&quot;      &quot;54413&quot;     &quot;6872&quot;     
[6949] &quot;8473&quot;      &quot;93953&quot;     &quot;340527&quot;    &quot;441502&quot;    &quot;392490&quot;    &quot;5303&quot;     
[6955] &quot;6191&quot;      &quot;5255&quot;      &quot;4674&quot;      &quot;7503&quot;      &quot;554203&quot;    &quot;100302692&quot;
[6961] &quot;51132&quot;     &quot;51260&quot;     &quot;57692&quot;     &quot;546&quot;       &quot;84061&quot;     &quot;1349&quot;     
[6967] &quot;441531&quot;    &quot;5230&quot;      &quot;51616&quot;     &quot;254065&quot;    &quot;79366&quot;     &quot;6451&quot;     
[6973] &quot;139324&quot;    &quot;139322&quot;    &quot;7552&quot;      &quot;1121&quot;      &quot;56062&quot;     &quot;4675&quot;     
[6979] &quot;286499&quot;    &quot;1730&quot;      &quot;64102&quot;     &quot;7105&quot;      &quot;27286&quot;     &quot;402415&quot;   
[6985] &quot;59353&quot;     &quot;695&quot;       &quot;3188&quot;      &quot;51566&quot;     &quot;84460&quot;     &quot;11013&quot;    
[6991] &quot;9737&quot;      &quot;100287765&quot; &quot;55859&quot;     &quot;90843&quot;     &quot;56849&quot;     &quot;51186&quot;    
[6997] &quot;27018&quot;     &quot;79921&quot;     &quot;85012&quot;     &quot;9338&quot;      &quot;286527&quot;    &quot;100101478&quot;
[7003] &quot;401612&quot;    &quot;644353&quot;    &quot;139231&quot;    &quot;203447&quot;    &quot;79589&quot;     &quot;55285&quot;    
[7009] &quot;5631&quot;      &quot;1831&quot;      &quot;11043&quot;     &quot;5716&quot;      &quot;1288&quot;      &quot;1287&quot;     
[7015] &quot;2182&quot;      &quot;84187&quot;     &quot;91851&quot;     &quot;401613&quot;    &quot;79868&quot;     &quot;154796&quot;   
[7021] &quot;57631&quot;     &quot;101927352&quot; &quot;5358&quot;      &quot;54521&quot;     &quot;139818&quot;    &quot;100303728&quot;
[7027] &quot;292&quot;       &quot;63932&quot;     &quot;55922&quot;     &quot;23157&quot;     &quot;6170&quot;      &quot;65109&quot;    
[7033] &quot;7737&quot;      &quot;4694&quot;      &quot;79576&quot;     &quot;10009&quot;     &quot;55026&quot;     &quot;23439&quot;    
[7039] &quot;3920&quot;      &quot;8450&quot;      &quot;28985&quot;     &quot;29071&quot;     &quot;331&quot;       &quot;10735&quot;    
[7045] &quot;10178&quot;     &quot;6594&quot;      &quot;4952&quot;      &quot;10813&quot;     &quot;63035&quot;     &quot;9363&quot;     
[7051] &quot;55609&quot;     &quot;10495&quot;     &quot;3547&quot;      &quot;51765&quot;     &quot;57826&quot;     &quot;101928578&quot;
[7057] &quot;90161&quot;     &quot;2719&quot;      &quot;3251&quot;      &quot;84848&quot;     &quot;441518&quot;    &quot;8933&quot;     
[7063] &quot;26071&quot;     &quot;644596&quot;    &quot;7626&quot;      &quot;203522&quot;    &quot;93380&quot;     &quot;10479&quot;    
[7069] &quot;79649&quot;     &quot;27336&quot;     &quot;9459&quot;      &quot;158696&quot;    &quot;2258&quot;      &quot;286410&quot;   
[7075] &quot;23641&quot;     &quot;139065&quot;    &quot;2334&quot;      &quot;100131434&quot; &quot;91966&quot;     &quot;100272228&quot;
[7081] &quot;83692&quot;     &quot;3149&quot;      &quot;203547&quot;    &quot;79057&quot;     &quot;1069&quot;      &quot;29944&quot;    
[7087] &quot;92002&quot;     &quot;139728&quot;    &quot;215&quot;       &quot;5365&quot;      &quot;3421&quot;      &quot;3897&quot;     
[7093] &quot;393&quot;       &quot;8260&quot;      &quot;3054&quot;      &quot;8269&quot;      &quot;3654&quot;      &quot;4204&quot;     
[7099] &quot;2316&quot;      &quot;2010&quot;      &quot;6134&quot;      &quot;26778&quot;     &quot;1774&quot;      &quot;158960&quot;   
[7105] &quot;537&quot;       &quot;2664&quot;      &quot;9130&quot;      &quot;55558&quot;     &quot;8270&quot;      &quot;60343&quot;    
[7111] &quot;139716&quot;    &quot;8263&quot;      &quot;65991&quot;     &quot;100272147&quot; &quot;4515&quot;      &quot;79184&quot;    
[7117] &quot;7411&quot;      &quot;116442&quot;    &quot;474383&quot;    &quot;474384&quot;    &quot;10251&quot;     &quot;6845&quot;     
[7123] &quot;105379416&quot; &quot;6192&quot;      &quot;7544&quot;      &quot;5616&quot;      &quot;64595&quot;     &quot;8287&quot;     
[7129] &quot;8653&quot;      &quot;7404&quot;      &quot;22829&quot;     &quot;246126&quot;    &quot;8284&quot;      &quot;9086&quot;     </code></pre>
<pre class="r"><code># Load expression dataset
boxplot1 &lt;- read.csv(&quot;data/boxplot1.csv&quot;, check.names = FALSE)
boxplot1 &lt;- as.data.frame(boxplot1)

# Choose a gene from the non-response group (0.5 µM)
target_entrez &lt;- 9446  # Replace with your desired Entrez ID from prob_1_0.5

# Check if it&#39;s in the group
if (!(target_entrez %in% prob_1_0.5)) {
  stop(&quot;Selected gene is not in the Non response group for 0.5 micromolar.&quot;)
}

# Filter for selected gene
gene_data &lt;- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if(nrow(gene_data) == 0) stop(&quot;No data found for selected ENTREZID.&quot;)

# Reshape to long format
gene_data_long &lt;- melt(gene_data,
                       id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                       variable.name = &quot;Sample&quot;,
                       value.name = &quot;log2CPM&quot;)

# Extract metadata from sample names
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)
  )

# Filter for 0.5 micromolar only
gene_data_long &lt;- gene_data_long %&gt;% filter(Concentration == &quot;0.5&quot;)

# Convert to factors
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = &quot;0.5&quot;)

# Map individual IDs
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, 
                  &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], &quot;Unknown&quot;))

# Define color palette for drugs
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;#08306B&quot;, &quot;DOX&quot; = &quot;#E7298A&quot;, &quot;VEH&quot; = &quot;green&quot;)

# Extract gene symbol for labeling
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;Non response (0.5 micromolar)&quot;) +
  labs(
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/box4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box4-1">
Past versions of box4-1.png
</button>
</p>
<div id="fig-box4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/box4-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dox-specific-response-0.5-µm" class="section level2">
<h2><strong>📌 DOX-specific response (0.5 µM)</strong></h2>
<pre class="r"><code># Load libraries
library(dplyr)
library(ggplot2)
library(reshape2)

# Load response group: DOX-specific response (0.5 µM)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)

# View gene list (optional)
prob_2_0.5</code></pre>
<pre><code>  [1] &quot;55160&quot;     &quot;27245&quot;     &quot;9202&quot;      &quot;5453&quot;      &quot;4610&quot;      &quot;25974&quot;    
  [7] &quot;65260&quot;     &quot;26009&quot;     &quot;2959&quot;      &quot;100506343&quot; &quot;105371405&quot; &quot;23248&quot;    
 [13] &quot;9869&quot;      &quot;51043&quot;     &quot;55154&quot;     &quot;27005&quot;     &quot;10560&quot;     &quot;92344&quot;    
 [19] &quot;6004&quot;      &quot;338382&quot;    &quot;26750&quot;     &quot;51742&quot;     &quot;27063&quot;     &quot;9937&quot;     
 [25] &quot;51259&quot;     &quot;10482&quot;     &quot;23642&quot;     &quot;9304&quot;      &quot;9298&quot;      &quot;254225&quot;   
 [31] &quot;282679&quot;    &quot;283219&quot;    &quot;84441&quot;     &quot;91801&quot;     &quot;257160&quot;    &quot;3741&quot;     
 [37] &quot;6843&quot;      &quot;254013&quot;    &quot;29843&quot;     &quot;85406&quot;     &quot;84872&quot;     &quot;10193&quot;    
 [43] &quot;23389&quot;     &quot;387496&quot;    &quot;5994&quot;      &quot;84078&quot;     &quot;10253&quot;     &quot;3621&quot;     
 [49] &quot;283624&quot;    &quot;85446&quot;     &quot;2643&quot;      &quot;112840&quot;    &quot;2957&quot;      &quot;283596&quot;   
 [55] &quot;64784&quot;     &quot;84464&quot;     &quot;5373&quot;      &quot;92595&quot;     &quot;10847&quot;     &quot;57659&quot;    
 [61] &quot;284023&quot;    &quot;100128288&quot; &quot;124751&quot;    &quot;100874058&quot; &quot;57547&quot;     &quot;90313&quot;    
 [67] &quot;5193&quot;      &quot;284098&quot;    &quot;5914&quot;      &quot;3759&quot;      &quot;79637&quot;     &quot;8811&quot;     
 [73] &quot;8877&quot;      &quot;57521&quot;     &quot;84307&quot;     &quot;4092&quot;      &quot;5609&quot;      &quot;84167&quot;    
 [79] &quot;339324&quot;    &quot;7596&quot;      &quot;7770&quot;      &quot;353355&quot;    &quot;7771&quot;      &quot;26974&quot;    
 [85] &quot;83987&quot;     &quot;57479&quot;     &quot;23636&quot;     &quot;25888&quot;     &quot;90317&quot;     &quot;147657&quot;   
 [91] &quot;84787&quot;     &quot;147949&quot;    &quot;284309&quot;    &quot;84914&quot;     &quot;29841&quot;     &quot;4953&quot;     
 [97] &quot;8648&quot;      &quot;4084&quot;      &quot;3554&quot;      &quot;55777&quot;     &quot;57703&quot;     &quot;1385&quot;     
[103] &quot;55502&quot;     &quot;55317&quot;     &quot;9935&quot;      &quot;23394&quot;     &quot;6526&quot;      &quot;23076&quot;    
[109] &quot;402055&quot;    &quot;84133&quot;     &quot;23598&quot;     &quot;100506472&quot; &quot;150383&quot;    &quot;3269&quot;     
[115] &quot;29072&quot;     &quot;8927&quot;      &quot;9730&quot;      &quot;23132&quot;     &quot;200895&quot;    &quot;79691&quot;    
[121] &quot;23023&quot;     &quot;57018&quot;     &quot;1181&quot;      &quot;5437&quot;      &quot;2049&quot;      &quot;10644&quot;    
[127] &quot;55171&quot;     &quot;3280&quot;      &quot;10336&quot;     &quot;348926&quot;    &quot;114932&quot;    &quot;4299&quot;     
[133] &quot;9464&quot;      &quot;133957&quot;    &quot;54491&quot;     &quot;25836&quot;     &quot;2668&quot;      &quot;2966&quot;     
[139] &quot;54826&quot;     &quot;324&quot;       &quot;9542&quot;      &quot;79616&quot;     &quot;6310&quot;      &quot;378884&quot;   
[145] &quot;79692&quot;     &quot;114821&quot;    &quot;170954&quot;    &quot;117283&quot;    &quot;23294&quot;     &quot;114781&quot;   
[151] &quot;129685&quot;    &quot;8871&quot;      &quot;100128653&quot; &quot;221937&quot;    &quot;729438&quot;    &quot;5001&quot;     
[157] &quot;55904&quot;     &quot;222255&quot;    &quot;4899&quot;      &quot;135932&quot;    &quot;835&quot;       &quot;2626&quot;     
[163] &quot;606553&quot;    &quot;5520&quot;      &quot;79145&quot;     &quot;1875&quot;      &quot;862&quot;       &quot;23414&quot;    
[169] &quot;340359&quot;    &quot;10221&quot;     &quot;8629&quot;      &quot;2619&quot;      &quot;286336&quot;    &quot;6256&quot;     
[175] &quot;170082&quot;    &quot;11040&quot;     &quot;90060&quot;     &quot;286495&quot;    &quot;101928747&quot;</code></pre>
<pre class="r"><code># Load expression dataset
boxplot1 &lt;- read.csv(&quot;data/boxplot1.csv&quot;, check.names = FALSE)
boxplot1 &lt;- as.data.frame(boxplot1)

# Choose a gene from the DOX-specific response group (0.5 µM)
target_entrez &lt;- 114821  # Replace with your desired Entrez ID from prob_2_0.5

# Check membership
if (!(target_entrez %in% prob_2_0.5)) {
  stop(&quot;Selected gene is not in the DOX-specific response group for 0.5 micromolar.&quot;)
}

# Filter for selected gene
gene_data &lt;- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if(nrow(gene_data) == 0) stop(&quot;No data found for selected ENTREZID.&quot;)

# Reshape to long format
gene_data_long &lt;- melt(gene_data,
                       id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                       variable.name = &quot;Sample&quot;,
                       value.name = &quot;log2CPM&quot;)

# Extract metadata from sample names
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)
  )

# Filter for 0.5 micromolar only
gene_data_long &lt;- gene_data_long %&gt;% filter(Concentration == &quot;0.5&quot;)

# Convert to factors
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = &quot;0.5&quot;)

# Map individual IDs
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, 
                  &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], &quot;Unknown&quot;))

# Define color palette for drugs
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;#08306B&quot;, &quot;DOX&quot; = &quot;#E7298A&quot;, &quot;VEH&quot; = &quot;green&quot;)

# Extract gene symbol for labeling
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;DOX-specific\nresponse (0.5 micromolar)&quot;) +
  labs(
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/box5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box5-1">
Past versions of box5-1.png
</button>
</p>
<div id="fig-box5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/box5-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dox-only-mid-late-response-0.5-µm" class="section level2">
<h2><strong>📌 DOX only mid-late response (0.5 µM)</strong></h2>
<pre class="r"><code># Load libraries
library(dplyr)
library(ggplot2)
library(reshape2)

# Load response group: DOX only mid-late response (0.5 µM)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)

# View gene list (optional)
prob_3_0.5</code></pre>
<pre><code>   [1] &quot;729737&quot;    &quot;100132287&quot; &quot;100133331&quot; &quot;100288069&quot; &quot;79854&quot;     &quot;643837&quot;   
   [7] &quot;100130417&quot; &quot;148398&quot;    &quot;26155&quot;     &quot;9636&quot;      &quot;375790&quot;    &quot;54991&quot;    
  [13] &quot;51150&quot;     &quot;126792&quot;    &quot;388581&quot;    &quot;118424&quot;    &quot;6339&quot;      &quot;116983&quot;   
  [19] &quot;126789&quot;    &quot;80772&quot;     &quot;83756&quot;     &quot;54587&quot;     &quot;81669&quot;     &quot;55210&quot;    
  [25] &quot;339453&quot;    &quot;142678&quot;    &quot;8510&quot;      &quot;728661&quot;    &quot;105378949&quot; &quot;5590&quot;     
  [31] &quot;6497&quot;      &quot;79906&quot;     &quot;100129534&quot; &quot;11079&quot;     &quot;127281&quot;    &quot;63976&quot;    
  [37] &quot;388588&quot;    &quot;9731&quot;      &quot;339448&quot;    &quot;8514&quot;      &quot;102724450&quot; &quot;388591&quot;   
  [43] &quot;387509&quot;    &quot;11332&quot;     &quot;83715&quot;     &quot;57449&quot;     &quot;8863&quot;      &quot;3604&quot;     
  [49] &quot;54206&quot;     &quot;9563&quot;      &quot;80176&quot;     &quot;5293&quot;      &quot;56998&quot;     &quot;10277&quot;    
  [55] &quot;5226&quot;      &quot;10747&quot;     &quot;5394&quot;      &quot;26232&quot;     &quot;93611&quot;     &quot;57085&quot;    
  [61] &quot;4879&quot;      &quot;90231&quot;     &quot;5351&quot;      &quot;9927&quot;      &quot;60672&quot;     &quot;9249&quot;     
  [67] &quot;10630&quot;     &quot;23254&quot;     &quot;55092&quot;     &quot;54751&quot;     &quot;729614&quot;    &quot;27129&quot;    
  [73] &quot;1187&quot;      &quot;1188&quot;      &quot;1969&quot;      &quot;79363&quot;     &quot;26099&quot;     &quot;11223&quot;    
  [79] &quot;9696&quot;      &quot;11240&quot;     &quot;55920&quot;     &quot;126917&quot;    &quot;51154&quot;     &quot;8574&quot;     
  [85] &quot;54896&quot;     &quot;23252&quot;     &quot;55450&quot;     &quot;163933&quot;    &quot;65018&quot;     &quot;100861548&quot;
  [91] &quot;1650&quot;      &quot;57576&quot;     &quot;400745&quot;    &quot;8672&quot;      &quot;84224&quot;     &quot;249&quot;      
  [97] &quot;84196&quot;     &quot;54361&quot;     &quot;2048&quot;      &quot;100996511&quot; &quot;10236&quot;     &quot;148898&quot;   
 [103] &quot;3399&quot;      &quot;6924&quot;      &quot;57095&quot;     &quot;11313&quot;     &quot;2582&quot;      &quot;3155&quot;     
 [109] &quot;2517&quot;      &quot;163702&quot;    &quot;57822&quot;     &quot;90529&quot;     &quot;11123&quot;     &quot;25949&quot;    
 [115] &quot;57035&quot;     &quot;26119&quot;     &quot;57134&quot;     &quot;57190&quot;     &quot;646471&quot;    &quot;164091&quot;   
 [121] &quot;2134&quot;      &quot;7780&quot;      &quot;64793&quot;     &quot;91544&quot;     &quot;6195&quot;      &quot;101928728&quot;
 [127] &quot;55650&quot;     &quot;84243&quot;     &quot;54707&quot;     &quot;115572&quot;    &quot;6548&quot;      &quot;84065&quot;    
 [133] &quot;84958&quot;     &quot;9064&quot;      &quot;2537&quot;      &quot;199870&quot;    &quot;23673&quot;     &quot;2140&quot;     
 [139] &quot;54797&quot;     &quot;1104&quot;      &quot;8420&quot;      &quot;85028&quot;     &quot;6883&quot;      &quot;399474&quot;   
 [145] &quot;10076&quot;     &quot;9672&quot;      &quot;79570&quot;     &quot;9410&quot;      &quot;51538&quot;     &quot;2170&quot;     
 [151] &quot;347735&quot;    &quot;553115&quot;    &quot;1307&quot;      &quot;576&quot;       &quot;10657&quot;     &quot;23633&quot;    
 [157] &quot;200081&quot;    &quot;79140&quot;     &quot;55721&quot;     &quot;8668&quot;      &quot;339483&quot;    &quot;3065&quot;     
 [163] &quot;100128071&quot; &quot;81629&quot;     &quot;55108&quot;     &quot;728116&quot;    &quot;653121&quot;    &quot;8565&quot;     
 [169] &quot;252995&quot;    &quot;113452&quot;    &quot;127544&quot;    &quot;113451&quot;    &quot;55223&quot;     &quot;100506144&quot;
 [175] &quot;9204&quot;      &quot;79830&quot;     &quot;6421&quot;      &quot;192670&quot;    &quot;26523&quot;     &quot;192669&quot;   
 [181] &quot;54936&quot;     &quot;83931&quot;     &quot;84967&quot;     &quot;127700&quot;    &quot;80149&quot;     &quot;79753&quot;    
 [187] &quot;7802&quot;      &quot;29889&quot;     &quot;79693&quot;     &quot;127687&quot;    &quot;4520&quot;      &quot;2275&quot;     
 [193] &quot;64121&quot;     &quot;79647&quot;     &quot;23499&quot;     &quot;353500&quot;    &quot;26508&quot;     &quot;10450&quot;    
 [199] &quot;54802&quot;     &quot;5538&quot;      &quot;1298&quot;      &quot;64744&quot;     &quot;64789&quot;     &quot;9783&quot;     
 [205] &quot;9132&quot;      &quot;22955&quot;     &quot;59269&quot;     &quot;284716&quot;    &quot;4904&quot;      &quot;64175&quot;    
 [211] &quot;79078&quot;     &quot;374969&quot;    &quot;51058&quot;     &quot;149465&quot;    &quot;128218&quot;    &quot;7075&quot;     
 [217] &quot;5792&quot;      &quot;6487&quot;      &quot;533&quot;       &quot;8704&quot;      &quot;6536&quot;      &quot;55182&quot;    
 [223] &quot;1263&quot;      &quot;343521&quot;    &quot;149478&quot;    &quot;8643&quot;      &quot;8891&quot;      &quot;79654&quot;    
 [229] &quot;4595&quot;      &quot;114034&quot;    &quot;10420&quot;     &quot;5052&quot;      &quot;149483&quot;    &quot;60313&quot;    
 [235] &quot;51249&quot;     &quot;23139&quot;     &quot;10103&quot;     &quot;10489&quot;     &quot;387338&quot;    &quot;2166&quot;     
 [241] &quot;79656&quot;     &quot;11124&quot;     &quot;26994&quot;     &quot;2060&quot;      &quot;5865&quot;      &quot;9372&quot;     
 [247] &quot;23318&quot;     &quot;1376&quot;      &quot;9528&quot;      &quot;26027&quot;     &quot;338094&quot;    &quot;374977&quot;   
 [253] &quot;100527960&quot; &quot;100507634&quot; &quot;8613&quot;      &quot;115209&quot;    &quot;55277&quot;     &quot;51361&quot;    
 [259] &quot;1573&quot;      &quot;84938&quot;     &quot;84455&quot;     &quot;5236&quot;      &quot;4919&quot;      &quot;57685&quot;    
 [265] &quot;3716&quot;      &quot;5142&quot;      &quot;79819&quot;     &quot;23169&quot;     &quot;1647&quot;      &quot;9077&quot;     
 [271] &quot;11147&quot;     &quot;100132618&quot; &quot;100526835&quot; &quot;8790&quot;      &quot;51086&quot;     &quot;1429&quot;     
 [277] &quot;5876&quot;      &quot;256435&quot;    &quot;374987&quot;    &quot;91624&quot;     &quot;8880&quot;      &quot;11080&quot;    
 [283] &quot;54810&quot;     &quot;64123&quot;     &quot;23266&quot;     &quot;80135&quot;     &quot;2787&quot;      &quot;100505741&quot;
 [289] &quot;117178&quot;    &quot;55283&quot;     &quot;148423&quot;    &quot;8915&quot;      &quot;23576&quot;     &quot;3491&quot;     
 [295] &quot;54680&quot;     &quot;105378828&quot; &quot;9403&quot;      &quot;5586&quot;      &quot;494115&quot;    &quot;2635&quot;     
 [301] &quot;2634&quot;      &quot;84230&quot;     &quot;55144&quot;     &quot;284695&quot;    &quot;164045&quot;    &quot;253152&quot;   
 [307] &quot;23285&quot;     &quot;11146&quot;     &quot;388650&quot;    &quot;1810&quot;      &quot;54874&quot;     &quot;8412&quot;     
 [313] &quot;30836&quot;     &quot;2730&quot;      &quot;5825&quot;      &quot;2152&quot;      &quot;126969&quot;    &quot;1266&quot;     
 [319] &quot;105378862&quot; &quot;729970&quot;    &quot;25950&quot;     &quot;1806&quot;      &quot;178&quot;       &quot;54482&quot;    
 [325] &quot;127495&quot;    &quot;8556&quot;      &quot;7412&quot;      &quot;148867&quot;    &quot;51611&quot;     &quot;102606465&quot;
 [331] &quot;101928370&quot; &quot;1901&quot;      &quot;55170&quot;     &quot;29957&quot;     &quot;284611&quot;    &quot;113802&quot;   
 [337] &quot;6814&quot;      &quot;23155&quot;     &quot;22911&quot;     &quot;6884&quot;      &quot;56900&quot;     &quot;1952&quot;     
 [343] &quot;6272&quot;      &quot;5686&quot;      &quot;284612&quot;    &quot;57463&quot;     &quot;2780&quot;      &quot;1435&quot;     
 [349] &quot;10768&quot;     &quot;3749&quot;      &quot;64783&quot;     &quot;10542&quot;     &quot;128338&quot;    &quot;5016&quot;     
 [355] &quot;79084&quot;     &quot;515&quot;       &quot;100129269&quot; &quot;55924&quot;     &quot;11218&quot;     &quot;7482&quot;     
 [361] &quot;4343&quot;      &quot;389&quot;       &quot;333926&quot;    &quot;6566&quot;      &quot;100506392&quot; &quot;10745&quot;    
 [367] &quot;204851&quot;    &quot;163259&quot;    &quot;4803&quot;      &quot;126868&quot;    &quot;965&quot;       &quot;5738&quot;     
 [373] &quot;10905&quot;     &quot;54855&quot;     &quot;54834&quot;     &quot;9554&quot;      &quot;100996724&quot; &quot;653464&quot;   
 [379] &quot;100996712&quot; &quot;284565&quot;    &quot;647135&quot;    &quot;101060254&quot; &quot;27246&quot;     &quot;8799&quot;     
 [385] &quot;9939&quot;      &quot;128077&quot;    &quot;84265&quot;     &quot;284615&quot;    &quot;148738&quot;    &quot;170688&quot;   
 [391] &quot;728989&quot;    &quot;5565&quot;      &quot;2330&quot;      &quot;51205&quot;     &quot;2702&quot;      &quot;51463&quot;    
 [397] &quot;100132057&quot; &quot;388685&quot;    &quot;644591&quot;    &quot;440672&quot;    &quot;9659&quot;      &quot;653513&quot;   
 [403] &quot;100996761&quot; &quot;440689&quot;    &quot;2209&quot;      &quot;8370&quot;      &quot;8337&quot;      &quot;337873&quot;   
 [409] &quot;723790&quot;    &quot;554313&quot;    &quot;8349&quot;      &quot;9900&quot;      &quot;56957&quot;     &quot;11311&quot;    
 [415] &quot;148523&quot;    &quot;80222&quot;     &quot;1893&quot;      &quot;4170&quot;      &quot;55204&quot;     &quot;405&quot;      
 [421] &quot;8416&quot;      &quot;55793&quot;     &quot;58497&quot;     &quot;10962&quot;     &quot;126626&quot;    &quot;6944&quot;     
 [427] &quot;57592&quot;     &quot;5298&quot;      &quot;5993&quot;      &quot;8991&quot;      &quot;5692&quot;      &quot;23126&quot;    
 [433] &quot;57530&quot;     &quot;117145&quot;    &quot;140576&quot;    &quot;6284&quot;      &quot;26097&quot;     &quot;3608&quot;     
 [439] &quot;65123&quot;     &quot;200186&quot;    &quot;10456&quot;     &quot;89872&quot;     &quot;57198&quot;     &quot;3570&quot;     
 [445] &quot;55585&quot;     &quot;3782&quot;      &quot;101928120&quot; &quot;8751&quot;      &quot;1944&quot;      &quot;55974&quot;    
 [451] &quot;200185&quot;    &quot;80128&quot;     &quot;4582&quot;      &quot;10712&quot;     &quot;10067&quot;     &quot;57657&quot;    
 [457] &quot;23623&quot;     &quot;645676&quot;    &quot;7818&quot;      &quot;100129405&quot; &quot;54856&quot;     &quot;6016&quot;     
 [463] &quot;22889&quot;     &quot;9181&quot;      &quot;56893&quot;     &quot;28956&quot;     &quot;4000&quot;      &quot;64218&quot;    
 [469] &quot;9673&quot;      &quot;11243&quot;     &quot;79957&quot;     &quot;7203&quot;      &quot;4209&quot;      &quot;60484&quot;    
 [475] &quot;10763&quot;     &quot;1382&quot;      &quot;51093&quot;     &quot;79590&quot;     &quot;3068&quot;      &quot;2117&quot;     
 [481] &quot;55243&quot;     &quot;912&quot;       &quot;57549&quot;     &quot;93183&quot;     &quot;477&quot;       &quot;844&quot;      
 [487] &quot;57216&quot;     &quot;7391&quot;      &quot;126823&quot;    &quot;51506&quot;     &quot;5498&quot;      &quot;84134&quot;    
 [493] &quot;3310&quot;      &quot;22926&quot;     &quot;127933&quot;    &quot;6675&quot;      &quot;4921&quot;      &quot;51478&quot;    
 [499] &quot;5999&quot;      &quot;8490&quot;      &quot;5087&quot;      &quot;6258&quot;      &quot;400794&quot;    &quot;4259&quot;     
 [505] &quot;7371&quot;      &quot;57645&quot;     &quot;92235&quot;     &quot;5451&quot;      &quot;8804&quot;      &quot;92241&quot;    
 [511] &quot;23432&quot;     &quot;375035&quot;    &quot;84832&quot;     &quot;101928596&quot; &quot;481&quot;       &quot;29922&quot;    
 [517] &quot;8548&quot;      &quot;57821&quot;     &quot;149281&quot;    &quot;5396&quot;      &quot;2329&quot;      &quot;8674&quot;     
 [523] &quot;51603&quot;     &quot;26052&quot;     &quot;51430&quot;     &quot;9588&quot;      &quot;339416&quot;    &quot;60674&quot;    
 [529] &quot;26774&quot;     &quot;149041&quot;    &quot;9910&quot;      &quot;9674&quot;      &quot;100302401&quot; &quot;400798&quot;   
 [535] &quot;64222&quot;     &quot;148756&quot;    &quot;57710&quot;     &quot;10228&quot;     &quot;2752&quot;      &quot;80896&quot;    
 [541] &quot;10092&quot;     &quot;116461&quot;    &quot;80267&quot;     &quot;116496&quot;    &quot;54823&quot;     &quot;10625&quot;    
 [547] &quot;7175&quot;      &quot;339479&quot;    &quot;101929120&quot; &quot;5997&quot;      &quot;51022&quot;     &quot;79577&quot;    
 [553] &quot;8707&quot;      &quot;360023&quot;    &quot;388722&quot;    &quot;23528&quot;     &quot;83479&quot;     &quot;23271&quot;    
 [559] &quot;779&quot;       &quot;7139&quot;      &quot;1465&quot;      &quot;89796&quot;     &quot;55705&quot;     &quot;149345&quot;   
 [565] &quot;25802&quot;     &quot;6051&quot;      &quot;1999&quot;      &quot;127829&quot;    &quot;4660&quot;      &quot;127833&quot;   
 [571] &quot;10765&quot;     &quot;100506696&quot; &quot;5877&quot;      &quot;59349&quot;     &quot;51706&quot;     &quot;92703&quot;    
 [577] &quot;8497&quot;      &quot;134&quot;       &quot;493&quot;       &quot;84719&quot;     &quot;9580&quot;      &quot;55224&quot;    
 [583] &quot;22874&quot;     &quot;5287&quot;      &quot;388730&quot;    &quot;25778&quot;     &quot;9911&quot;      &quot;81788&quot;    
 [589] &quot;55220&quot;     &quot;5129&quot;      &quot;2005&quot;      &quot;85414&quot;     &quot;8934&quot;      &quot;115019&quot;   
 [595] &quot;23380&quot;     &quot;5208&quot;      &quot;55432&quot;     &quot;1604&quot;      &quot;4179&quot;      &quot;3914&quot;     
 [601] &quot;148304&quot;    &quot;574036&quot;    &quot;55733&quot;     &quot;7779&quot;      &quot;9926&quot;      &quot;25896&quot;    
 [607] &quot;55248&quot;     &quot;29937&quot;     &quot;467&quot;       &quot;128387&quot;    &quot;56950&quot;     &quot;55105&quot;    
 [613] &quot;51018&quot;     &quot;2058&quot;      &quot;10380&quot;     &quot;54996&quot;     &quot;64757&quot;     &quot;3142&quot;     
 [619] &quot;11221&quot;     &quot;7159&quot;      &quot;23219&quot;     &quot;8560&quot;      &quot;29097&quot;     &quot;80232&quot;    
 [625] &quot;149111&quot;    &quot;127602&quot;    &quot;2052&quot;      &quot;29920&quot;     &quot;163859&quot;    &quot;3020&quot;     
 [631] &quot;142&quot;       &quot;375057&quot;    &quot;3707&quot;      &quot;56997&quot;     &quot;8476&quot;      &quot;339500&quot;   
 [637] &quot;100130093&quot; &quot;7483&quot;      &quot;375&quot;       &quot;2987&quot;      &quot;200205&quot;    &quot;84033&quot;    
 [643] &quot;101927401&quot; &quot;51127&quot;     &quot;92815&quot;     &quot;58480&quot;     &quot;58&quot;        &quot;22796&quot;    
 [649] &quot;101927604&quot; &quot;79573&quot;     &quot;375061&quot;    &quot;149371&quot;    &quot;83932&quot;     &quot;27184&quot;    
 [655] &quot;57568&quot;     &quot;54627&quot;     &quot;80003&quot;     &quot;84451&quot;     &quot;6894&quot;      &quot;359948&quot;   
 [661] &quot;9804&quot;      &quot;23029&quot;     &quot;6905&quot;      &quot;148789&quot;    &quot;1130&quot;      &quot;7107&quot;     
 [667] &quot;56605&quot;     &quot;88&quot;        &quot;4548&quot;      &quot;6262&quot;      &quot;56776&quot;     &quot;9859&quot;     
 [673] &quot;10472&quot;     &quot;284702&quot;    &quot;84288&quot;     &quot;55083&quot;     &quot;64754&quot;     &quot;64216&quot;    
 [679] &quot;51097&quot;     &quot;7678&quot;      &quot;55657&quot;     &quot;22982&quot;     &quot;23185&quot;     &quot;23560&quot;    
 [685] &quot;101927824&quot; &quot;10531&quot;     &quot;100507034&quot; &quot;79754&quot;     &quot;5588&quot;      &quot;439949&quot;   
 [691] &quot;57713&quot;     &quot;80760&quot;     &quot;509&quot;       &quot;83860&quot;     &quot;414196&quot;    &quot;26019&quot;    
 [697] &quot;83643&quot;     &quot;10133&quot;     &quot;5264&quot;      &quot;222389&quot;    &quot;8559&quot;      &quot;83641&quot;    
 [703] &quot;9397&quot;      &quot;80013&quot;     &quot;9317&quot;      &quot;389941&quot;    &quot;6251&quot;      &quot;1787&quot;     
 [709] &quot;100507347&quot; &quot;338596&quot;    &quot;9200&quot;      &quot;102723166&quot; &quot;783&quot;       &quot;221078&quot;   
 [715] &quot;10529&quot;     &quot;8028&quot;      &quot;5305&quot;      &quot;256815&quot;    &quot;220213&quot;    &quot;57584&quot;    
 [721] &quot;79896&quot;     &quot;22852&quot;     &quot;10730&quot;     &quot;25805&quot;     &quot;6840&quot;      &quot;57608&quot;    
 [727] &quot;55149&quot;     &quot;1326&quot;      &quot;645954&quot;    &quot;220930&quot;    &quot;1390&quot;      &quot;8325&quot;     
 [733] &quot;7581&quot;      &quot;7587&quot;      &quot;7582&quot;      &quot;3185&quot;      &quot;7580&quot;      &quot;643236&quot;   
 [739] &quot;83937&quot;     &quot;7570&quot;      &quot;220972&quot;    &quot;8031&quot;      &quot;9721&quot;      &quot;2658&quot;     
 [745] &quot;118738&quot;    &quot;5599&quot;      &quot;58504&quot;     &quot;196740&quot;    &quot;118461&quot;    &quot;55753&quot;    
 [751] &quot;100421577&quot; &quot;104355295&quot; &quot;253430&quot;    &quot;55847&quot;     &quot;7321&quot;      &quot;80114&quot;    
 [757] &quot;220965&quot;    &quot;220963&quot;    &quot;9886&quot;      &quot;84890&quot;     &quot;1959&quot;      &quot;29982&quot;    
 [763] &quot;84989&quot;     &quot;29119&quot;     &quot;23411&quot;     &quot;64081&quot;     &quot;219736&quot;    &quot;26128&quot;    
 [769] &quot;9559&quot;      &quot;23555&quot;     &quot;219738&quot;    &quot;1305&quot;      &quot;219743&quot;    &quot;55222&quot;    
 [775] &quot;1979&quot;      &quot;27143&quot;     &quot;8879&quot;      &quot;5092&quot;      &quot;5660&quot;      &quot;118490&quot;   
 [781] &quot;2749&quot;      &quot;170384&quot;    &quot;23053&quot;     &quot;100507331&quot; &quot;5328&quot;      &quot;132&quot;      
 [787] &quot;142891&quot;    &quot;100128292&quot; &quot;6229&quot;      &quot;729262&quot;    &quot;642361&quot;    &quot;80195&quot;    
 [793] &quot;81619&quot;     &quot;27069&quot;     &quot;54462&quot;     &quot;11155&quot;     &quot;657&quot;       &quot;2746&quot;     
 [799] &quot;54537&quot;     &quot;439994&quot;    &quot;728130&quot;    &quot;9060&quot;      &quot;142913&quot;    &quot;5728&quot;     
 [805] &quot;355&quot;       &quot;3434&quot;      &quot;24138&quot;     &quot;387700&quot;    &quot;53354&quot;     &quot;3363&quot;     
 [811] &quot;10556&quot;     &quot;119369&quot;    &quot;143279&quot;    &quot;5507&quot;      &quot;80351&quot;     &quot;143282&quot;   
 [817] &quot;9044&quot;      &quot;54708&quot;     &quot;159371&quot;    &quot;51196&quot;     &quot;23232&quot;     &quot;5832&quot;     
 [823] &quot;953&quot;       &quot;7093&quot;      &quot;56889&quot;     &quot;102723665&quot; &quot;84458&quot;     &quot;10023&quot;    
 [829] &quot;23401&quot;     &quot;23223&quot;     &quot;80019&quot;     &quot;118812&quot;    &quot;55361&quot;     &quot;60370&quot;    
 [835] &quot;118813&quot;    &quot;6425&quot;      &quot;55118&quot;     &quot;84171&quot;     &quot;3257&quot;      &quot;2805&quot;     
 [841] &quot;57089&quot;     &quot;51076&quot;     &quot;55280&quot;     &quot;90271&quot;     &quot;55719&quot;     &quot;56652&quot;    
 [847] &quot;84445&quot;     &quot;6468&quot;      &quot;30819&quot;     &quot;8861&quot;      &quot;4791&quot;      &quot;100505761&quot;
 [853] &quot;81603&quot;     &quot;119032&quot;    &quot;57412&quot;     &quot;54805&quot;     &quot;22978&quot;     &quot;9118&quot;     
 [859] &quot;84108&quot;     &quot;6877&quot;      &quot;22984&quot;     &quot;9148&quot;      &quot;9644&quot;      &quot;119391&quot;   
 [865] &quot;85450&quot;     &quot;22986&quot;     &quot;4601&quot;      &quot;1847&quot;      &quot;282996&quot;    &quot;282997&quot;   
 [871] &quot;27250&quot;     &quot;92482&quot;     &quot;8036&quot;      &quot;150&quot;       &quot;57678&quot;     &quot;64429&quot;    
 [877] &quot;4892&quot;      &quot;84632&quot;     &quot;26033&quot;     &quot;2674&quot;      &quot;259217&quot;    &quot;6571&quot;     
 [883] &quot;118987&quot;    &quot;22841&quot;     &quot;143384&quot;    &quot;340719&quot;    &quot;404636&quot;    &quot;119559&quot;   
 [889] &quot;7073&quot;      &quot;9531&quot;      &quot;11196&quot;     &quot;196051&quot;    &quot;2263&quot;      &quot;10579&quot;    
 [895] &quot;59338&quot;     &quot;80007&quot;     &quot;64376&quot;     &quot;51363&quot;     &quot;4942&quot;      &quot;64077&quot;    
 [901] &quot;399818&quot;    &quot;1488&quot;      &quot;92565&quot;     &quot;1793&quot;      &quot;642938&quot;    &quot;282973&quot;   
 [907] &quot;10570&quot;     &quot;80313&quot;     &quot;54777&quot;     &quot;101&quot;       &quot;100133161&quot; &quot;60626&quot;    
 [913] &quot;23410&quot;     &quot;5719&quot;      &quot;338707&quot;    &quot;59307&quot;     &quot;6050&quot;      &quot;115399&quot;   
 [919] &quot;256329&quot;    &quot;100506211&quot; &quot;3665&quot;      &quot;6888&quot;      &quot;171391&quot;    &quot;51286&quot;    
 [925] &quot;101927423&quot; &quot;55367&quot;     &quot;57104&quot;     &quot;5441&quot;      &quot;7106&quot;      &quot;54472&quot;    
 [931] &quot;402778&quot;    &quot;29125&quot;     &quot;10077&quot;     &quot;10078&quot;     &quot;3784&quot;      &quot;1028&quot;     
 [937] &quot;5002&quot;      &quot;7262&quot;      &quot;4676&quot;      &quot;114879&quot;    &quot;4928&quot;      &quot;27315&quot;    
 [943] &quot;6786&quot;      &quot;55128&quot;     &quot;117854&quot;    &quot;10612&quot;     &quot;144132&quot;    &quot;3611&quot;     
 [949] &quot;6881&quot;      &quot;8642&quot;      &quot;63875&quot;     &quot;7761&quot;      &quot;143425&quot;    &quot;7275&quot;     
 [955] &quot;79608&quot;     &quot;65975&quot;     &quot;56672&quot;     &quot;56675&quot;     &quot;57758&quot;     &quot;23258&quot;    
 [961] &quot;440026&quot;    &quot;7702&quot;      &quot;23075&quot;     &quot;81846&quot;     &quot;272&quot;       &quot;9646&quot;     
 [967] &quot;1982&quot;      &quot;374378&quot;    &quot;283106&quot;    &quot;27122&quot;     &quot;406&quot;       &quot;84280&quot;    
 [973] &quot;84188&quot;     &quot;1315&quot;      &quot;5682&quot;      &quot;120227&quot;    &quot;144100&quot;    &quot;5286&quot;     
 [979] &quot;374383&quot;    &quot;3767&quot;      &quot;10083&quot;     &quot;26297&quot;     &quot;113174&quot;    &quot;144108&quot;   
 [985] &quot;144110&quot;    &quot;54503&quot;     &quot;100126784&quot; &quot;10553&quot;     &quot;2620&quot;      &quot;387758&quot;   
 [991] &quot;91057&quot;     &quot;55366&quot;     &quot;55327&quot;     &quot;497258&quot;    &quot;196074&quot;    &quot;3739&quot;     
 [997] &quot;744&quot;       &quot;196294&quot;    &quot;26610&quot;     &quot;10480&quot;     &quot;91614&quot;     &quot;1479&quot;     
[1003] &quot;10114&quot;     &quot;966&quot;       &quot;26273&quot;     &quot;4076&quot;      &quot;25841&quot;     &quot;51074&quot;    
[1009] &quot;79899&quot;     &quot;119710&quot;    &quot;387763&quot;    &quot;3732&quot;      &quot;9537&quot;      &quot;9409&quot;     
[1015] &quot;90993&quot;     &quot;8525&quot;      &quot;4192&quot;      &quot;1132&quot;      &quot;9776&quot;      &quot;79797&quot;    
[1021] &quot;9793&quot;      &quot;79096&quot;     &quot;1643&quot;      &quot;8567&quot;      &quot;4607&quot;      &quot;10658&quot;    
[1027] &quot;114971&quot;    &quot;23788&quot;     &quot;23279&quot;     &quot;85456&quot;     &quot;6749&quot;      &quot;219539&quot;   
[1033] &quot;10978&quot;     &quot;219541&quot;    &quot;51075&quot;     &quot;1500&quot;      &quot;9404&quot;      &quot;80829&quot;    
[1039] &quot;23220&quot;     &quot;6809&quot;      &quot;54948&quot;     &quot;79080&quot;     &quot;220001&quot;    &quot;1642&quot;     
[1045] &quot;220002&quot;    &quot;51524&quot;     &quot;113157&quot;    &quot;747&quot;       &quot;745&quot;       &quot;9415&quot;     
[1051] &quot;2495&quot;      &quot;64852&quot;     &quot;9219&quot;      &quot;6094&quot;      &quot;26229&quot;     &quot;80789&quot;    
[1057] &quot;79081&quot;     &quot;751071&quot;    &quot;10629&quot;     &quot;79064&quot;     &quot;6811&quot;      &quot;54663&quot;    
[1063] &quot;6520&quot;      &quot;10313&quot;     &quot;65998&quot;     &quot;283248&quot;    &quot;79829&quot;     &quot;55611&quot;    
[1069] &quot;83706&quot;     &quot;84304&quot;     &quot;51504&quot;     &quot;10235&quot;     &quot;5837&quot;      &quot;7536&quot;     
[1075] &quot;5871&quot;      &quot;55561&quot;     &quot;10938&quot;     &quot;23130&quot;     &quot;5526&quot;      &quot;29901&quot;    
[1081] &quot;738&quot;       &quot;741&quot;       &quot;2197&quot;      &quot;740&quot;       &quot;84447&quot;     &quot;823&quot;      
[1087] &quot;10435&quot;     &quot;5977&quot;      &quot;283130&quot;    &quot;83786&quot;     &quot;254100&quot;    &quot;10534&quot;    
[1093] &quot;23625&quot;     &quot;6494&quot;      &quot;91056&quot;     &quot;254122&quot;    &quot;1072&quot;      &quot;80198&quot;    
[1099] &quot;30008&quot;     &quot;8061&quot;      &quot;10589&quot;     &quot;9092&quot;      &quot;84285&quot;     &quot;89792&quot;    
[1105] &quot;64837&quot;     &quot;254263&quot;    &quot;10897&quot;     &quot;256472&quot;    &quot;57124&quot;     &quot;25855&quot;    
[1111] &quot;11041&quot;     &quot;3177&quot;      &quot;246330&quot;    &quot;582&quot;       &quot;9973&quot;      &quot;6712&quot;     
[1117] &quot;5091&quot;      &quot;78999&quot;     &quot;91683&quot;     &quot;22992&quot;     &quot;156&quot;       &quot;338692&quot;   
[1123] &quot;54961&quot;     &quot;23529&quot;     &quot;5499&quot;      &quot;374403&quot;    &quot;57571&quot;     &quot;6199&quot;     
[1129] &quot;5790&quot;      &quot;57175&quot;     &quot;80194&quot;     &quot;9049&quot;      &quot;9600&quot;      &quot;10263&quot;    
[1135] &quot;254552&quot;    &quot;81622&quot;     &quot;4728&quot;      &quot;10312&quot;     &quot;53838&quot;     &quot;4041&quot;     
[1141] &quot;51083&quot;     &quot;1374&quot;      &quot;219931&quot;    &quot;595&quot;       &quot;220064&quot;    &quot;8772&quot;     
[1147] &quot;2017&quot;      &quot;1717&quot;      &quot;55191&quot;     &quot;10068&quot;     &quot;220074&quot;    &quot;3636&quot;     
[1153] &quot;81570&quot;     &quot;10809&quot;     &quot;89849&quot;     &quot;5029&quot;      &quot;84957&quot;     &quot;51287&quot;    
[1159] &quot;26005&quot;     &quot;51400&quot;     &quot;283208&quot;    &quot;100287896&quot; &quot;9789&quot;      &quot;100507050&quot;
[1165] &quot;408&quot;       &quot;81544&quot;     &quot;4135&quot;      &quot;7405&quot;      &quot;100506127&quot; &quot;25987&quot;    
[1171] &quot;55331&quot;     &quot;51773&quot;     &quot;4718&quot;      &quot;79731&quot;     &quot;26011&quot;     &quot;60492&quot;    
[1177] &quot;84233&quot;     &quot;8726&quot;      &quot;51501&quot;     &quot;10873&quot;     &quot;11098&quot;     &quot;65084&quot;    
[1183] &quot;120114&quot;    &quot;120103&quot;    &quot;56935&quot;     &quot;85459&quot;     &quot;28970&quot;     &quot;54851&quot;    
[1189] &quot;143689&quot;    &quot;154810&quot;    &quot;10929&quot;     &quot;23052&quot;     &quot;143686&quot;    &quot;143684&quot;   
[1195] &quot;79780&quot;     &quot;53942&quot;     &quot;83935&quot;     &quot;57562&quot;     &quot;10413&quot;     &quot;330&quot;      
[1201] &quot;80310&quot;     &quot;84437&quot;     &quot;2977&quot;      &quot;143884&quot;    &quot;54733&quot;     &quot;472&quot;      
[1207] &quot;23086&quot;     &quot;1662&quot;      &quot;85463&quot;     &quot;2230&quot;      &quot;1737&quot;      &quot;120379&quot;   
[1213] &quot;55216&quot;     &quot;26521&quot;     &quot;83875&quot;     &quot;101928847&quot; &quot;4684&quot;      &quot;100288346&quot;
[1219] &quot;9183&quot;      &quot;57646&quot;     &quot;4837&quot;      &quot;25996&quot;     &quot;23705&quot;     &quot;84811&quot;    
[1225] &quot;8882&quot;      &quot;23387&quot;     &quot;51092&quot;     &quot;6876&quot;      &quot;23621&quot;     &quot;100379571&quot;
[1231] &quot;22897&quot;     &quot;6330&quot;      &quot;6327&quot;      &quot;84866&quot;     &quot;56912&quot;     &quot;2542&quot;     
[1237] &quot;55823&quot;     &quot;3145&quot;      &quot;9854&quot;      &quot;64137&quot;     &quot;79671&quot;     &quot;867&quot;      
[1243] &quot;4162&quot;      &quot;9099&quot;      &quot;7070&quot;      &quot;220323&quot;    &quot;219902&quot;    &quot;219899&quot;   
[1249] &quot;6653&quot;      &quot;84959&quot;     &quot;79864&quot;     &quot;55800&quot;     &quot;7753&quot;      &quot;4013&quot;     
[1255] &quot;54414&quot;     &quot;79684&quot;     &quot;100507283&quot; &quot;219854&quot;    &quot;9638&quot;      &quot;9538&quot;     
[1261] &quot;3703&quot;      &quot;83480&quot;     &quot;6734&quot;      &quot;55572&quot;     &quot;28960&quot;     &quot;399972&quot;   
[1267] &quot;6484&quot;      &quot;2113&quot;      &quot;3762&quot;      &quot;219833&quot;    &quot;56980&quot;     &quot;334&quot;      
[1273] &quot;6768&quot;      &quot;11095&quot;     &quot;170689&quot;    &quot;283174&quot;    &quot;112936&quot;    &quot;29087&quot;    
[1279] &quot;89944&quot;     &quot;27087&quot;     &quot;283358&quot;    &quot;100049716&quot; &quot;23085&quot;     &quot;144699&quot;   
[1285] &quot;196513&quot;    &quot;775&quot;       &quot;2288&quot;      &quot;10867&quot;     &quot;103752584&quot; &quot;57103&quot;    
[1291] &quot;8798&quot;      &quot;4704&quot;      &quot;3742&quot;      &quot;928&quot;       &quot;4055&quot;      &quot;55080&quot;    
[1297] &quot;1108&quot;      &quot;8079&quot;      &quot;27239&quot;     &quot;10162&quot;     &quot;9746&quot;      &quot;25977&quot;    
[1303] &quot;4074&quot;      &quot;10219&quot;     &quot;642846&quot;    &quot;440081&quot;    &quot;374443&quot;    &quot;4973&quot;     
[1309] &quot;55110&quot;     &quot;8531&quot;      &quot;2120&quot;      &quot;4040&quot;      &quot;54682&quot;     &quot;118426&quot;   
[1315] &quot;80824&quot;     &quot;387841&quot;    &quot;93164&quot;     &quot;2012&quot;      &quot;105369669&quot; &quot;55766&quot;    
[1321] &quot;397&quot;       &quot;85004&quot;     &quot;2059&quot;      &quot;55885&quot;     &quot;728622&quot;    &quot;55500&quot;    
[1327] &quot;6660&quot;      &quot;586&quot;       &quot;100506451&quot; &quot;11228&quot;     &quot;79365&quot;     &quot;3709&quot;     
[1333] &quot;55726&quot;     &quot;26127&quot;     &quot;60488&quot;     &quot;57542&quot;     &quot;5744&quot;      &quot;55711&quot;    
[1339] &quot;51290&quot;     &quot;10526&quot;     &quot;65981&quot;     &quot;1663&quot;      &quot;58516&quot;     &quot;160518&quot;   
[1345] &quot;100874249&quot; &quot;196394&quot;    &quot;440093&quot;    &quot;121512&quot;    &quot;10059&quot;     &quot;5318&quot;     
[1351] &quot;84920&quot;     &quot;144402&quot;    &quot;55605&quot;     &quot;283463&quot;    &quot;10138&quot;     &quot;85437&quot;    
[1357] &quot;51535&quot;     &quot;144165&quot;    &quot;83448&quot;     &quot;51135&quot;     &quot;84216&quot;     &quot;4753&quot;     
[1363] &quot;51054&quot;     &quot;196527&quot;    &quot;9169&quot;      &quot;55089&quot;     &quot;51564&quot;     &quot;5213&quot;     
[1369] &quot;54934&quot;     &quot;904&quot;       &quot;112&quot;       &quot;27289&quot;     &quot;377&quot;       &quot;23109&quot;    
[1375] &quot;7846&quot;      &quot;10445&quot;     &quot;100286844&quot; &quot;144233&quot;    &quot;41&quot;        &quot;91012&quot;    
[1381] &quot;57609&quot;     &quot;466&quot;       &quot;4891&quot;      &quot;25875&quot;     &quot;81566&quot;     &quot;5463&quot;     
[1387] &quot;9802&quot;      &quot;11226&quot;     &quot;9498&quot;      &quot;3164&quot;      &quot;3855&quot;      &quot;3875&quot;     
[1393] &quot;23371&quot;     &quot;51380&quot;     &quot;283337&quot;    &quot;5916&quot;      &quot;60314&quot;     &quot;8086&quot;     
[1399] &quot;6667&quot;      &quot;269&quot;       &quot;54458&quot;     &quot;517&quot;       &quot;57658&quot;     &quot;3178&quot;     
[1405] &quot;25946&quot;     &quot;3678&quot;      &quot;5502&quot;      &quot;29095&quot;     &quot;440104&quot;    &quot;6821&quot;     
[1411] &quot;64375&quot;     &quot;6231&quot;      &quot;2065&quot;      &quot;5036&quot;      &quot;283373&quot;    &quot;1431&quot;     
[1417] &quot;6773&quot;      &quot;27165&quot;     &quot;5939&quot;      &quot;9880&quot;      &quot;11247&quot;     &quot;6472&quot;     
[1423] &quot;83729&quot;     &quot;4141&quot;      &quot;1649&quot;      &quot;79837&quot;     &quot;196403&quot;    &quot;115557&quot;   
[1429] &quot;6302&quot;      &quot;1019&quot;      &quot;92979&quot;     &quot;25895&quot;     &quot;10102&quot;     &quot;10677&quot;    
[1435] &quot;10106&quot;     &quot;100506844&quot; &quot;91419&quot;     &quot;9194&quot;      &quot;9958&quot;      &quot;10329&quot;    
[1441] &quot;144577&quot;    &quot;11260&quot;     &quot;283349&quot;    &quot;2799&quot;      &quot;253827&quot;    &quot;100129940&quot;
[1447] &quot;51643&quot;     &quot;92797&quot;     &quot;5908&quot;      &quot;1368&quot;      &quot;8089&quot;      &quot;10818&quot;    
[1453] &quot;376132&quot;    &quot;117177&quot;    &quot;4848&quot;      &quot;27345&quot;     &quot;5787&quot;      &quot;7103&quot;     
[1459] &quot;8549&quot;      &quot;83591&quot;     &quot;55266&quot;     &quot;23011&quot;     &quot;29953&quot;     &quot;84698&quot;    
[1465] &quot;11103&quot;     &quot;114882&quot;    &quot;23390&quot;     &quot;144455&quot;    &quot;5074&quot;      &quot;29080&quot;    
[1471] &quot;55117&quot;     &quot;9182&quot;      &quot;25834&quot;     &quot;80184&quot;     &quot;4254&quot;      &quot;1848&quot;     
[1477] &quot;105369889&quot; &quot;282809&quot;    &quot;490&quot;       &quot;338758&quot;    &quot;694&quot;       &quot;11163&quot;    
[1483] &quot;51134&quot;     &quot;55967&quot;     &quot;55785&quot;     &quot;55591&quot;     &quot;6636&quot;      &quot;4048&quot;     
[1489] &quot;2004&quot;      &quot;5128&quot;      &quot;5250&quot;      &quot;121457&quot;    &quot;317&quot;       &quot;23074&quot;    
[1495] &quot;64431&quot;     &quot;121601&quot;    &quot;56994&quot;     &quot;55332&quot;     &quot;51559&quot;     &quot;6996&quot;     
[1501] &quot;83468&quot;     &quot;29915&quot;     &quot;4801&quot;      &quot;7296&quot;      &quot;84102&quot;     &quot;121053&quot;   
[1507] &quot;387882&quot;    &quot;255394&quot;    &quot;55703&quot;     &quot;55188&quot;     &quot;1407&quot;      &quot;121551&quot;   
[1513] &quot;11108&quot;     &quot;9671&quot;      &quot;1240&quot;      &quot;9733&quot;      &quot;23479&quot;     &quot;23603&quot;    
[1519] &quot;32&quot;        &quot;89910&quot;     &quot;326625&quot;    &quot;4598&quot;      &quot;84915&quot;     &quot;59341&quot;    
[1525] &quot;51228&quot;     &quot;84260&quot;     &quot;88455&quot;     &quot;28981&quot;     &quot;488&quot;       &quot;51184&quot;    
[1531] &quot;51699&quot;     &quot;5501&quot;      &quot;100131138&quot; &quot;144717&quot;    &quot;10019&quot;     &quot;8315&quot;     
[1537] &quot;80724&quot;     &quot;10906&quot;     &quot;79039&quot;     &quot;84934&quot;     &quot;9904&quot;      &quot;6910&quot;     
[1543] &quot;255480&quot;    &quot;6926&quot;      &quot;643246&quot;    &quot;79794&quot;     &quot;84900&quot;     &quot;8739&quot;     
[1549] &quot;26259&quot;     &quot;55884&quot;     &quot;160777&quot;    &quot;5564&quot;      &quot;92558&quot;     &quot;11021&quot;    
[1555] &quot;100506649&quot; &quot;1337&quot;      &quot;283459&quot;    &quot;8655&quot;      &quot;9761&quot;      &quot;84747&quot;    
[1561] &quot;51433&quot;     &quot;84678&quot;     &quot;84876&quot;     &quot;144404&quot;    &quot;54509&quot;     &quot;338799&quot;   
[1567] &quot;22877&quot;     &quot;6249&quot;      &quot;9026&quot;      &quot;79676&quot;     &quot;8099&quot;      &quot;55206&quot;    
[1573] &quot;100293704&quot; &quot;387893&quot;    &quot;11066&quot;     &quot;1967&quot;      &quot;79867&quot;     &quot;196385&quot;   
[1579] &quot;80212&quot;     &quot;9612&quot;      &quot;949&quot;       &quot;57647&quot;     &quot;65985&quot;     &quot;8408&quot;     
[1585] &quot;80324&quot;     &quot;57634&quot;     &quot;347918&quot;    &quot;57666&quot;     &quot;5827&quot;      &quot;192111&quot;   
[1591] &quot;2802&quot;      &quot;100289635&quot; &quot;284232&quot;    &quot;54737&quot;     &quot;9205&quot;      &quot;2706&quot;     
[1597] &quot;221143&quot;    &quot;64328&quot;     &quot;26524&quot;     &quot;10284&quot;     &quot;650794&quot;    &quot;55504&quot;    
[1603] &quot;221178&quot;    &quot;143&quot;       &quot;9107&quot;      &quot;51761&quot;     &quot;387914&quot;    &quot;6049&quot;     
[1609] &quot;1024&quot;      &quot;219333&quot;    &quot;222484&quot;    &quot;51082&quot;     &quot;432369&quot;    &quot;283537&quot;   
[1615] &quot;23281&quot;     &quot;6541&quot;      &quot;5412&quot;      &quot;84056&quot;     &quot;10208&quot;     &quot;10808&quot;    
[1621] &quot;145173&quot;    &quot;10129&quot;     &quot;90634&quot;     &quot;10443&quot;     &quot;23047&quot;     &quot;90627&quot;    
[1627] &quot;26960&quot;     &quot;9201&quot;      &quot;8900&quot;      &quot;400120&quot;    &quot;4093&quot;      &quot;7223&quot;     
[1633] &quot;51569&quot;     &quot;387921&quot;    &quot;10186&quot;     &quot;2308&quot;      &quot;10240&quot;     &quot;10166&quot;    
[1639] &quot;1997&quot;      &quot;11193&quot;     &quot;79612&quot;     &quot;28984&quot;     &quot;23078&quot;     &quot;160851&quot;   
[1645] &quot;55068&quot;     &quot;160857&quot;    &quot;144811&quot;    &quot;8848&quot;      &quot;641467&quot;    &quot;23143&quot;    
[1651] &quot;2098&quot;      &quot;9445&quot;      &quot;5925&quot;      &quot;1102&quot;      &quot;81617&quot;     &quot;51131&quot;    
[1657] &quot;55213&quot;     &quot;84650&quot;     &quot;57213&quot;     &quot;10206&quot;     &quot;10301&quot;     &quot;145165&quot;   
[1663] &quot;79621&quot;     &quot;220108&quot;    &quot;100507398&quot; &quot;115825&quot;    &quot;540&quot;       &quot;341676&quot;   
[1669] &quot;4752&quot;      &quot;100887750&quot; &quot;51028&quot;     &quot;26586&quot;     &quot;144983&quot;    &quot;5100&quot;     
[1675] &quot;27253&quot;     &quot;81624&quot;     &quot;5101&quot;      &quot;1602&quot;      &quot;22894&quot;     &quot;10464&quot;    
[1681] &quot;688&quot;       &quot;11278&quot;     &quot;7347&quot;      &quot;4008&quot;      &quot;115207&quot;    &quot;26224&quot;    
[1687] &quot;5457&quot;      &quot;79596&quot;     &quot;100505538&quot; &quot;693207&quot;    &quot;10082&quot;     &quot;160897&quot;   
[1693] &quot;22873&quot;     &quot;266722&quot;    &quot;10150&quot;     &quot;3843&quot;      &quot;10160&quot;     &quot;8428&quot;     
[1699] &quot;100861541&quot; &quot;100289373&quot; &quot;337867&quot;    &quot;1880&quot;      &quot;5095&quot;      &quot;87769&quot;    
[1705] &quot;7174&quot;      &quot;93081&quot;     &quot;2073&quot;      &quot;1948&quot;      &quot;55082&quot;     &quot;728215&quot;   
[1711] &quot;8660&quot;      &quot;1284&quot;      &quot;55739&quot;     &quot;105376684&quot; &quot;283487&quot;    &quot;55608&quot;    
[1717] &quot;23250&quot;     &quot;23263&quot;     &quot;100289410&quot; &quot;55795&quot;     &quot;3916&quot;      &quot;79774&quot;    
[1723] &quot;101928841&quot; &quot;113622&quot;    &quot;100506394&quot; &quot;22821&quot;     &quot;91875&quot;     &quot;10038&quot;    
[1729] &quot;7011&quot;      &quot;55644&quot;     &quot;328&quot;       &quot;90809&quot;     &quot;4860&quot;      &quot;64745&quot;    
[1735] &quot;3183&quot;      &quot;84932&quot;     &quot;1603&quot;      &quot;63874&quot;     &quot;26020&quot;     &quot;54930&quot;    
[1741] &quot;60686&quot;     &quot;5693&quot;      &quot;64403&quot;     &quot;55017&quot;     &quot;23428&quot;     &quot;57594&quot;    
[1747] &quot;599&quot;       &quot;116173&quot;    &quot;4624&quot;      &quot;4625&quot;      &quot;25983&quot;     &quot;102724814&quot;
[1753] &quot;55449&quot;     &quot;5106&quot;      &quot;161247&quot;    &quot;5721&quot;      &quot;9985&quot;      &quot;79711&quot;    
[1759] &quot;4738&quot;      &quot;26277&quot;     &quot;5875&quot;      &quot;115817&quot;    &quot;1241&quot;      &quot;4776&quot;     
[1765] &quot;56948&quot;     &quot;11154&quot;     &quot;25831&quot;     &quot;84837&quot;     &quot;9472&quot;      &quot;64067&quot;    
[1771] &quot;58533&quot;     &quot;11177&quot;     &quot;55012&quot;     &quot;9692&quot;      &quot;253959&quot;    &quot;84312&quot;    
[1777] &quot;89874&quot;     &quot;100129794&quot; &quot;145282&quot;    &quot;8487&quot;      &quot;254170&quot;    &quot;122525&quot;   
[1783] &quot;54813&quot;     &quot;55015&quot;     &quot;161357&quot;    &quot;100506433&quot; &quot;23588&quot;     &quot;9147&quot;     
[1789] &quot;79609&quot;     &quot;79944&quot;     &quot;8814&quot;      &quot;60485&quot;     &quot;5836&quot;      &quot;122786&quot;   
[1795] &quot;145438&quot;    &quot;54331&quot;     &quot;22795&quot;     &quot;57544&quot;     &quot;283554&quot;    &quot;10979&quot;    
[1801] &quot;652&quot;       &quot;10175&quot;     &quot;2764&quot;      &quot;10668&quot;     &quot;122809&quot;    &quot;3958&quot;     
[1807] &quot;55030&quot;     &quot;22863&quot;     &quot;3895&quot;      &quot;57161&quot;     &quot;55745&quot;     &quot;5684&quot;     
[1813] &quot;379025&quot;    &quot;26520&quot;     &quot;9786&quot;      &quot;23002&quot;     &quot;64582&quot;     &quot;64430&quot;    
[1819] &quot;5494&quot;      &quot;51804&quot;     &quot;4331&quot;      &quot;5583&quot;      &quot;6617&quot;      &quot;5529&quot;     
[1825] &quot;7597&quot;      &quot;26030&quot;     &quot;6710&quot;      &quot;2342&quot;      &quot;4149&quot;      &quot;51382&quot;    
[1831] &quot;161145&quot;    &quot;5283&quot;      &quot;51109&quot;     &quot;5890&quot;      &quot;55218&quot;     &quot;9766&quot;     
[1837] &quot;6430&quot;      &quot;6547&quot;      &quot;55333&quot;     &quot;26037&quot;     &quot;9628&quot;      &quot;8110&quot;     
[1843] &quot;53349&quot;     &quot;5663&quot;      &quot;8650&quot;      &quot;957&quot;       &quot;80127&quot;     &quot;5826&quot;     
[1849] &quot;122961&quot;    &quot;4053&quot;      &quot;51077&quot;     &quot;56252&quot;     &quot;83694&quot;     &quot;5228&quot;     
[1855] &quot;8892&quot;      &quot;27030&quot;     &quot;2353&quot;      &quot;101928377&quot; &quot;122953&quot;    &quot;11161&quot;    
[1861] &quot;23093&quot;     &quot;55668&quot;     &quot;22846&quot;     &quot;100506603&quot; &quot;64207&quot;     &quot;29954&quot;    
[1867] &quot;2954&quot;      &quot;283578&quot;    &quot;161394&quot;    &quot;10598&quot;     &quot;9517&quot;      &quot;8846&quot;     
[1873] &quot;22938&quot;     &quot;9369&quot;      &quot;85439&quot;     &quot;11099&quot;     &quot;123016&quot;    &quot;400236&quot;   
[1879] &quot;90141&quot;     &quot;55051&quot;     &quot;145567&quot;    &quot;9252&quot;      &quot;80017&quot;     &quot;440193&quot;   
[1885] &quot;55671&quot;     &quot;10516&quot;     &quot;9950&quot;      &quot;3705&quot;      &quot;64112&quot;     &quot;57578&quot;    
[1891] &quot;51676&quot;     &quot;57062&quot;     &quot;83982&quot;     &quot;12&quot;        &quot;400242&quot;    &quot;161176&quot;   
[1897] &quot;624&quot;       &quot;730202&quot;    &quot;10914&quot;     &quot;8812&quot;      &quot;10858&quot;     &quot;7528&quot;     
[1903] &quot;5527&quot;      &quot;5891&quot;      &quot;55778&quot;     &quot;51550&quot;     &quot;9895&quot;      &quot;23186&quot;    
[1909] &quot;7127&quot;      &quot;1983&quot;      &quot;115708&quot;    &quot;9529&quot;      &quot;79038&quot;     &quot;400258&quot;   
[1915] &quot;388021&quot;    &quot;64423&quot;     &quot;122622&quot;    &quot;207&quot;       &quot;100128927&quot; &quot;283638&quot;   
[1921] &quot;122616&quot;    &quot;23241&quot;     &quot;9112&quot;      &quot;1397&quot;      &quot;1396&quot;      &quot;283643&quot;   
[1927] &quot;80757&quot;     &quot;646214&quot;    &quot;123606&quot;    &quot;54551&quot;     &quot;6638&quot;      &quot;2562&quot;     
[1933] &quot;8924&quot;      &quot;23359&quot;     &quot;7082&quot;      &quot;22909&quot;     &quot;102725022&quot; &quot;6447&quot;     
[1939] &quot;26585&quot;     &quot;79768&quot;     &quot;9990&quot;      &quot;23015&quot;     &quot;440270&quot;    &quot;57369&quot;    
[1945] &quot;70&quot;        &quot;723972&quot;    &quot;89978&quot;     &quot;84529&quot;     &quot;4212&quot;      &quot;161742&quot;   
[1951] &quot;7057&quot;      &quot;11245&quot;     &quot;100131089&quot; &quot;90427&quot;     &quot;56924&quot;     &quot;5330&quot;     
[1957] &quot;388115&quot;    &quot;22893&quot;     &quot;113189&quot;    &quot;55177&quot;     &quot;2644&quot;      &quot;55192&quot;    
[1963] &quot;6692&quot;      &quot;57617&quot;     &quot;79094&quot;     &quot;729082&quot;    &quot;23168&quot;     &quot;26015&quot;    
[1969] &quot;7301&quot;      &quot;100137047&quot; &quot;30844&quot;     &quot;23339&quot;     &quot;25963&quot;     &quot;2595&quot;     
[1975] &quot;64397&quot;     &quot;8773&quot;      &quot;146057&quot;    &quot;80021&quot;     &quot;7158&quot;      &quot;4130&quot;     
[1981] &quot;80237&quot;     &quot;619189&quot;    &quot;567&quot;       &quot;140691&quot;    &quot;50506&quot;     &quot;90525&quot;    
[1987] &quot;79029&quot;     &quot;50804&quot;     &quot;399694&quot;    &quot;9728&quot;      &quot;196951&quot;    &quot;56986&quot;    
[1993] &quot;55056&quot;     &quot;100129387&quot; &quot;388121&quot;    &quot;23312&quot;     &quot;256586&quot;    &quot;29767&quot;    
[1999] &quot;29766&quot;     &quot;5597&quot;      &quot;4644&quot;      &quot;10776&quot;     &quot;56204&quot;     &quot;9488&quot;     
[2005] &quot;9236&quot;      &quot;26108&quot;     &quot;283659&quot;    &quot;64864&quot;     &quot;374618&quot;    &quot;54816&quot;    
[2011] &quot;145783&quot;    &quot;84952&quot;     &quot;145781&quot;    &quot;100820829&quot; &quot;54629&quot;     &quot;4643&quot;     
[2017] &quot;79664&quot;     &quot;83660&quot;     &quot;7168&quot;      &quot;114294&quot;    &quot;51065&quot;     &quot;51762&quot;    
[2023] &quot;283807&quot;    &quot;8925&quot;      &quot;53944&quot;     &quot;9325&quot;      &quot;80301&quot;     &quot;348094&quot;   
[2029] &quot;51285&quot;     &quot;10081&quot;     &quot;8483&quot;      &quot;57722&quot;     &quot;81556&quot;     &quot;9187&quot;     
[2035] &quot;4088&quot;      &quot;10391&quot;     &quot;8125&quot;      &quot;7090&quot;      &quot;54839&quot;     &quot;56906&quot;    
[2041] &quot;5315&quot;      &quot;60677&quot;     &quot;3073&quot;      &quot;585&quot;       &quot;101241892&quot; &quot;80381&quot;    
[2047] &quot;388135&quot;    &quot;100287616&quot; &quot;9399&quot;      &quot;5371&quot;      &quot;10620&quot;     &quot;1198&quot;     
[2053] &quot;102723750&quot; &quot;57184&quot;     &quot;9377&quot;      &quot;54913&quot;     &quot;192683&quot;    &quot;5780&quot;     
[2059] &quot;55272&quot;     &quot;1464&quot;      &quot;92912&quot;     &quot;145957&quot;    &quot;49855&quot;     &quot;84894&quot;    
[2065] &quot;3419&quot;      &quot;5685&quot;      &quot;11173&quot;     &quot;5923&quot;      &quot;10588&quot;     &quot;400410&quot;   
[2071] &quot;283687&quot;    &quot;54469&quot;     &quot;58489&quot;     &quot;23184&quot;     &quot;59274&quot;     &quot;84206&quot;    
[2077] &quot;390660&quot;    &quot;64506&quot;     &quot;123722&quot;    &quot;123720&quot;    &quot;83640&quot;     &quot;50810&quot;    
[2083] &quot;646&quot;       &quot;6457&quot;      &quot;4828&quot;      &quot;9154&quot;      &quot;5151&quot;      &quot;11214&quot;    
[2089] &quot;64410&quot;     &quot;3669&quot;      &quot;145864&quot;    &quot;4240&quot;      &quot;11057&quot;     &quot;5428&quot;     
[2095] &quot;374654&quot;    &quot;10239&quot;     &quot;10519&quot;     &quot;51335&quot;     &quot;342132&quot;    &quot;5045&quot;     
[2101] &quot;374659&quot;    &quot;91433&quot;     &quot;28232&quot;     &quot;400451&quot;    &quot;56963&quot;     &quot;104472848&quot;
[2107] &quot;3480&quot;      &quot;64927&quot;     &quot;170691&quot;    &quot;140460&quot;    &quot;220&quot;       &quot;22856&quot;    
[2113] &quot;6627&quot;      &quot;5046&quot;      &quot;105376734&quot; &quot;101927628&quot; &quot;374666&quot;    &quot;51728&quot;    
[2119] &quot;79622&quot;     &quot;64285&quot;     &quot;4350&quot;      &quot;8131&quot;      &quot;8786&quot;      &quot;8312&quot;     
[2125] &quot;10573&quot;     &quot;58986&quot;     &quot;4833&quot;      &quot;26063&quot;     &quot;57799&quot;     &quot;89941&quot;    
[2131] &quot;79006&quot;     &quot;65990&quot;     &quot;84264&quot;     &quot;113000&quot;    &quot;64788&quot;     &quot;7329&quot;     
[2137] &quot;115939&quot;    &quot;84572&quot;     &quot;645811&quot;    &quot;1186&quot;      &quot;57585&quot;     &quot;4832&quot;     
[2143] &quot;65993&quot;     &quot;10101&quot;     &quot;81889&quot;     &quot;51734&quot;     &quot;4716&quot;      &quot;735301&quot;   
[2149] &quot;124056&quot;    &quot;283869&quot;    &quot;9351&quot;      &quot;4913&quot;      &quot;64223&quot;     &quot;283871&quot;   
[2155] &quot;1877&quot;      &quot;1632&quot;      &quot;10921&quot;     &quot;21&quot;        &quot;80178&quot;     &quot;527&quot;      
[2161] &quot;5170&quot;      &quot;652276&quot;    &quot;645644&quot;    &quot;100507321&quot; &quot;54442&quot;     &quot;114984&quot;   
[2167] &quot;84256&quot;     &quot;79412&quot;     &quot;9080&quot;      &quot;51330&quot;     &quot;54985&quot;     &quot;79228&quot;    
[2173] &quot;7755&quot;      &quot;100507458&quot; &quot;7760&quot;      &quot;10127&quot;     &quot;7627&quot;      &quot;79903&quot;    
[2179] &quot;23059&quot;     &quot;1773&quot;      &quot;10131&quot;     &quot;6345&quot;      &quot;100507501&quot; &quot;51025&quot;    
[2185] &quot;114990&quot;    &quot;29965&quot;     &quot;124402&quot;    &quot;79641&quot;     &quot;84656&quot;     &quot;9717&quot;     
[2191] &quot;51172&quot;     &quot;56052&quot;     &quot;196483&quot;    &quot;18&quot;        &quot;29035&quot;     &quot;2013&quot;     
[2197] &quot;4682&quot;      &quot;28955&quot;     &quot;8651&quot;      &quot;9516&quot;      &quot;8303&quot;      &quot;51061&quot;    
[2203] &quot;26156&quot;     &quot;92017&quot;     &quot;55313&quot;     &quot;5073&quot;      &quot;642799&quot;    &quot;9284&quot;     
[2209] &quot;123803&quot;    &quot;9665&quot;      &quot;123811&quot;    &quot;368&quot;       &quot;339044&quot;    &quot;101059938&quot;
[2215] &quot;64131&quot;     &quot;101059953&quot; &quot;105376752&quot; &quot;653190&quot;    &quot;79905&quot;     &quot;51760&quot;    
[2221] &quot;9738&quot;      &quot;400506&quot;    &quot;112479&quot;    &quot;123879&quot;    &quot;57149&quot;     &quot;1428&quot;     
[2227] &quot;100271836&quot; &quot;387254&quot;    &quot;51108&quot;     &quot;730092&quot;    &quot;57478&quot;     &quot;56061&quot;    
[2233] &quot;84516&quot;     &quot;5579&quot;      &quot;3566&quot;      &quot;23214&quot;     &quot;388228&quot;    &quot;728741&quot;   
[2239] &quot;1201&quot;      &quot;6799&quot;      &quot;6817&quot;      &quot;728734&quot;    &quot;100507607&quot; &quot;7284&quot;     
[2245] &quot;487&quot;       &quot;79874&quot;     &quot;653390&quot;    &quot;728888&quot;    &quot;6693&quot;      &quot;23475&quot;    
[2251] &quot;79447&quot;     &quot;10423&quot;     &quot;26470&quot;     &quot;253982&quot;    &quot;253980&quot;    &quot;124446&quot;   
[2257] &quot;9344&quot;      &quot;5595&quot;      &quot;10421&quot;     &quot;26000&quot;     &quot;79077&quot;     &quot;22928&quot;    
[2263] &quot;146540&quot;    &quot;78994&quot;     &quot;64319&quot;     &quot;100862671&quot; &quot;5261&quot;      &quot;90835&quot;    
[2269] &quot;23361&quot;     &quot;101928736&quot; &quot;283932&quot;    &quot;54620&quot;     &quot;9739&quot;      &quot;80270&quot;    
[2275] &quot;79759&quot;     &quot;9726&quot;      &quot;10295&quot;     &quot;1339&quot;      &quot;79798&quot;     &quot;7041&quot;     
[2281] &quot;6524&quot;      &quot;440359&quot;    &quot;100132341&quot; &quot;10308&quot;     &quot;55737&quot;     &quot;91807&quot;    
[2287] &quot;388272&quot;    &quot;81831&quot;     &quot;5257&quot;      &quot;23090&quot;     &quot;55027&quot;     &quot;64282&quot;    
[2293] &quot;85407&quot;     &quot;1540&quot;      &quot;6299&quot;      &quot;23322&quot;     &quot;79068&quot;     &quot;79191&quot;    
[2299] &quot;10265&quot;     &quot;6530&quot;      &quot;55239&quot;     &quot;4502&quot;      &quot;4493&quot;      &quot;4501&quot;     
[2305] &quot;9709&quot;      &quot;80011&quot;     &quot;23568&quot;     &quot;92922&quot;     &quot;9289&quot;      &quot;57567&quot;    
[2311] &quot;79650&quot;     &quot;4324&quot;      &quot;1459&quot;      &quot;29070&quot;     &quot;79918&quot;     &quot;2806&quot;     
[2317] &quot;1006&quot;      &quot;1009&quot;      &quot;146227&quot;    &quot;7084&quot;      &quot;51192&quot;     &quot;146223&quot;   
[2323] &quot;8883&quot;      &quot;6236&quot;      &quot;8824&quot;      &quot;23491&quot;     &quot;865&quot;       &quot;8996&quot;     
[2329] &quot;653319&quot;    &quot;283849&quot;    &quot;29100&quot;     &quot;6553&quot;      &quot;29800&quot;     &quot;9114&quot;     
[2335] &quot;100505942&quot; &quot;10664&quot;     &quot;84080&quot;     &quot;57610&quot;     &quot;55815&quot;     &quot;80152&quot;    
[2341] &quot;1506&quot;      &quot;5699&quot;      &quot;55794&quot;     &quot;54920&quot;     &quot;4775&quot;      &quot;80004&quot;    
[2347] &quot;23659&quot;     &quot;54496&quot;     &quot;55512&quot;     &quot;146198&quot;    &quot;1001&quot;      &quot;999&quot;      
[2353] &quot;3038&quot;      &quot;54921&quot;     &quot;6645&quot;      &quot;27183&quot;     &quot;51388&quot;     &quot;10725&quot;    
[2359] &quot;1728&quot;      &quot;11060&quot;     &quot;16&quot;        &quot;197258&quot;    &quot;23450&quot;     &quot;7571&quot;     
[2365] &quot;7567&quot;      &quot;91862&quot;     &quot;23035&quot;     &quot;342371&quot;    &quot;55565&quot;     &quot;9798&quot;     
[2371] &quot;1723&quot;      &quot;54957&quot;     &quot;9785&quot;      &quot;197259&quot;    &quot;84937&quot;     &quot;197257&quot;   
[2377] &quot;162239&quot;    &quot;10428&quot;     &quot;79583&quot;     &quot;11345&quot;     &quot;23536&quot;     &quot;22879&quot;    
[2383] &quot;100130958&quot; &quot;51741&quot;     &quot;4094&quot;      &quot;124359&quot;    &quot;56942&quot;     &quot;23300&quot;    
[2389] &quot;8139&quot;      &quot;80790&quot;     &quot;10200&quot;     &quot;3281&quot;      &quot;23417&quot;     &quot;54550&quot;    
[2395] &quot;8720&quot;      &quot;83693&quot;     &quot;9013&quot;      &quot;57707&quot;     &quot;9100&quot;      &quot;83716&quot;    
[2401] &quot;9764&quot;      &quot;10328&quot;     &quot;1327&quot;      &quot;2300&quot;      &quot;81631&quot;     &quot;23174&quot;    
[2407] &quot;124245&quot;    &quot;1535&quot;      &quot;4597&quot;      &quot;333929&quot;    &quot;115992&quot;    &quot;9780&quot;     
[2413] &quot;51693&quot;     &quot;197322&quot;    &quot;105371414&quot; &quot;27132&quot;     &quot;124045&quot;    &quot;4157&quot;     
[2419] &quot;10381&quot;     &quot;54849&quot;     &quot;79007&quot;     &quot;359845&quot;    &quot;55275&quot;     &quot;79850&quot;    
[2425] &quot;50628&quot;     &quot;1398&quot;      &quot;4641&quot;      &quot;51763&quot;     &quot;5306&quot;      &quot;124935&quot;   
[2431] &quot;727910&quot;    &quot;124997&quot;    &quot;6117&quot;      &quot;146760&quot;    &quot;1801&quot;      &quot;3090&quot;     
[2437] &quot;23293&quot;     &quot;4335&quot;      &quot;105371592&quot; &quot;162514&quot;    &quot;7442&quot;      &quot;23729&quot;    
[2443] &quot;30851&quot;     &quot;83460&quot;     &quot;5026&quot;      &quot;3682&quot;      &quot;5023&quot;      &quot;489&quot;      
[2449] &quot;51479&quot;     &quot;124976&quot;    &quot;10514&quot;     &quot;58191&quot;     &quot;388323&quot;    &quot;8402&quot;     
[2455] &quot;2027&quot;      &quot;388324&quot;    &quot;101928000&quot; &quot;9098&quot;      &quot;4927&quot;      &quot;708&quot;      
[2461] &quot;79003&quot;     &quot;728392&quot;    &quot;9851&quot;      &quot;100506713&quot; &quot;239&quot;       &quot;440400&quot;   
[2467] &quot;1742&quot;      &quot;37&quot;        &quot;442898&quot;    &quot;11337&quot;     &quot;6517&quot;      &quot;51087&quot;    
[2473] &quot;2874&quot;      &quot;84461&quot;     &quot;147040&quot;    &quot;254863&quot;    &quot;57555&quot;     &quot;284114&quot;   
[2479] &quot;2256&quot;      &quot;1140&quot;      &quot;5430&quot;      &quot;968&quot;       &quot;6462&quot;      &quot;112483&quot;   
[2485] &quot;482&quot;       &quot;7157&quot;      &quot;1949&quot;      &quot;23135&quot;     &quot;92162&quot;     &quot;124637&quot;   
[2491] &quot;58485&quot;     &quot;84667&quot;     &quot;5187&quot;      &quot;54785&quot;     &quot;284029&quot;    &quot;80169&quot;    
[2497] &quot;5198&quot;      &quot;399512&quot;    &quot;29098&quot;     &quot;22899&quot;     &quot;81565&quot;     &quot;9423&quot;     
[2503] &quot;9482&quot;      &quot;124739&quot;    &quot;6341&quot;      &quot;56985&quot;     &quot;388335&quot;    &quot;6416&quot;     
[2509] &quot;93649&quot;     &quot;9912&quot;      &quot;9955&quot;      &quot;9953&quot;      &quot;84815&quot;     &quot;5376&quot;     
[2515] &quot;10626&quot;     &quot;57335&quot;     &quot;4213&quot;      &quot;136&quot;       &quot;125150&quot;    &quot;9487&quot;     
[2521] &quot;201161&quot;    &quot;7314&quot;      &quot;51393&quot;     &quot;23164&quot;     &quot;201164&quot;    &quot;201163&quot;   
[2527] &quot;8533&quot;      &quot;55090&quot;     &quot;51655&quot;     &quot;10400&quot;     &quot;100507131&quot; &quot;83450&quot;    
[2533] &quot;91647&quot;     &quot;79018&quot;     &quot;1819&quot;      &quot;54890&quot;     &quot;3996&quot;      &quot;140775&quot;   
[2539] &quot;6470&quot;      &quot;729288&quot;    &quot;147166&quot;    &quot;5636&quot;      &quot;125206&quot;    &quot;644815&quot;   
[2545] &quot;22905&quot;     &quot;55244&quot;     &quot;224&quot;       &quot;92521&quot;     &quot;339184&quot;    &quot;23326&quot;    
[2551] &quot;25979&quot;     &quot;256302&quot;    &quot;5606&quot;      &quot;8844&quot;      &quot;201229&quot;    &quot;51701&quot;    
[2557] &quot;90410&quot;     &quot;7126&quot;      &quot;26073&quot;     &quot;147007&quot;    &quot;23098&quot;     &quot;230&quot;      
[2563] &quot;9703&quot;      &quot;6830&quot;      &quot;6147&quot;      &quot;116238&quot;    &quot;284086&quot;    &quot;9618&quot;     
[2569] &quot;399687&quot;    &quot;9220&quot;      &quot;105371931&quot; &quot;116236&quot;    &quot;28964&quot;     &quot;124930&quot;   
[2575] &quot;85464&quot;     &quot;374786&quot;    &quot;6532&quot;      &quot;51379&quot;     &quot;4763&quot;      &quot;84440&quot;    
[2581] &quot;55352&quot;     &quot;114659&quot;    &quot;105371730&quot; &quot;55288&quot;     &quot;64149&quot;     &quot;8851&quot;     
[2587] &quot;40&quot;        &quot;6347&quot;      &quot;3980&quot;      &quot;5892&quot;      &quot;54475&quot;     &quot;146862&quot;   
[2593] &quot;105371932&quot; &quot;162394&quot;    &quot;91608&quot;     &quot;79893&quot;     &quot;79154&quot;     &quot;79922&quot;    
[2599] &quot;3975&quot;      &quot;26574&quot;     &quot;11276&quot;     &quot;101060351&quot; &quot;30837&quot;     &quot;57636&quot;    
[2605] &quot;100170841&quot; &quot;284106&quot;    &quot;7703&quot;      &quot;100287808&quot; &quot;5691&quot;      &quot;8396&quot;     
[2611] &quot;54883&quot;     &quot;782&quot;       &quot;84961&quot;     &quot;8557&quot;      &quot;2064&quot;      &quot;55876&quot;    
[2617] &quot;5709&quot;      &quot;9862&quot;      &quot;7067&quot;      &quot;9572&quot;      &quot;125111&quot;    &quot;3487&quot;     
[2623] &quot;9001&quot;      &quot;3728&quot;      &quot;10609&quot;     &quot;60681&quot;     &quot;115024&quot;    &quot;47&quot;       
[2629] &quot;1267&quot;      &quot;7266&quot;      &quot;201181&quot;    &quot;84514&quot;     &quot;6774&quot;      &quot;3292&quot;     
[2635] &quot;162427&quot;    &quot;27175&quot;     &quot;8506&quot;      &quot;2145&quot;      &quot;100190938&quot; &quot;10266&quot;    
[2641] &quot;28958&quot;     &quot;8678&quot;      &quot;314&quot;       &quot;100885850&quot; &quot;100885848&quot; &quot;113277&quot;   
[2647] &quot;1659&quot;      &quot;2118&quot;      &quot;4356&quot;      &quot;162417&quot;    &quot;105371789&quot; &quot;339201&quot;   
[2653] &quot;79089&quot;     &quot;7343&quot;      &quot;51629&quot;     &quot;2896&quot;      &quot;2535&quot;      &quot;284071&quot;   
[2659] &quot;10052&quot;     &quot;9343&quot;      &quot;388389&quot;    &quot;2670&quot;      &quot;10882&quot;     &quot;4836&quot;     
[2665] &quot;113026&quot;    &quot;10614&quot;     &quot;124790&quot;    &quot;752&quot;       &quot;201176&quot;    &quot;728806&quot;   
[2671] &quot;4905&quot;      &quot;7473&quot;      &quot;996&quot;       &quot;4635&quot;      &quot;3690&quot;      &quot;102724508&quot;
[2677] &quot;124989&quot;    &quot;9520&quot;      &quot;3837&quot;      &quot;114881&quot;    &quot;90507&quot;     &quot;80320&quot;    
[2683] &quot;6668&quot;      &quot;55163&quot;     &quot;80279&quot;     &quot;100874362&quot; &quot;10241&quot;     &quot;516&quot;      
[2689] &quot;65264&quot;     &quot;11248&quot;     &quot;8405&quot;      &quot;81558&quot;     &quot;11143&quot;     &quot;284080&quot;   
[2695] &quot;5164&quot;      &quot;84687&quot;     &quot;6442&quot;      &quot;64132&quot;     &quot;80221&quot;     &quot;55316&quot;    
[2701] &quot;55040&quot;     &quot;8714&quot;      &quot;51747&quot;     &quot;400604&quot;    &quot;4831&quot;      &quot;51096&quot;    
[2707] &quot;10040&quot;     &quot;23531&quot;     &quot;58488&quot;     &quot;9241&quot;      &quot;339210&quot;    &quot;8526&quot;     
[2713] &quot;7706&quot;      &quot;8165&quot;      &quot;404093&quot;    &quot;6426&quot;      &quot;54903&quot;     &quot;9256&quot;     
[2719] &quot;100506779&quot; &quot;6827&quot;      &quot;54894&quot;     &quot;56155&quot;     &quot;22843&quot;     &quot;4591&quot;     
[2725] &quot;348235&quot;    &quot;284161&quot;    &quot;1213&quot;      &quot;51174&quot;     &quot;84669&quot;     &quot;10513&quot;    
[2731] &quot;8493&quot;      &quot;54828&quot;     &quot;103689912&quot; &quot;6909&quot;      &quot;388407&quot;    &quot;9969&quot;     
[2737] &quot;339175&quot;    &quot;11011&quot;     &quot;26115&quot;     &quot;1534&quot;      &quot;1636&quot;      &quot;10238&quot;    
[2743] &quot;4215&quot;      &quot;80774&quot;     &quot;729683&quot;    &quot;6603&quot;      &quot;55852&quot;     &quot;11232&quot;    
[2749] &quot;1655&quot;      &quot;100616387&quot; &quot;100616408&quot; &quot;90799&quot;     &quot;146880&quot;    &quot;440456&quot;   
[2755] &quot;374819&quot;    &quot;201283&quot;    &quot;10672&quot;     &quot;8787&quot;      &quot;8313&quot;      &quot;201134&quot;   
[2761] &quot;9931&quot;      &quot;26207&quot;     &quot;100499466&quot; &quot;9120&quot;      &quot;55062&quot;     &quot;693220&quot;   
[2767] &quot;10351&quot;     &quot;23461&quot;     &quot;5608&quot;      &quot;400617&quot;    &quot;6662&quot;      &quot;9382&quot;     
[2773] &quot;84923&quot;     &quot;55028&quot;     &quot;23580&quot;     &quot;54549&quot;     &quot;55890&quot;     &quot;9368&quot;     
[2779] &quot;283987&quot;    &quot;30850&quot;     &quot;23510&quot;     &quot;10476&quot;     &quot;9121&quot;      &quot;30833&quot;    
[2785] &quot;51155&quot;     &quot;79902&quot;     &quot;23163&quot;     &quot;57513&quot;     &quot;29115&quot;     &quot;3021&quot;     
[2791] &quot;85451&quot;     &quot;201294&quot;    &quot;201292&quot;    &quot;23265&quot;     &quot;283991&quot;    &quot;84074&quot;    
[2797] &quot;79651&quot;     &quot;114757&quot;    &quot;768206&quot;    &quot;10610&quot;     &quot;105274304&quot; &quot;105376847&quot;
[2803] &quot;124512&quot;    &quot;6427&quot;      &quot;654434&quot;    &quot;57690&quot;     &quot;147138&quot;    &quot;125061&quot;   
[2809] &quot;9021&quot;      &quot;8632&quot;      &quot;9267&quot;      &quot;57602&quot;     &quot;3959&quot;      &quot;100507410&quot;
[2815] &quot;2548&quot;      &quot;9775&quot;      &quot;79092&quot;     &quot;284129&quot;    &quot;57674&quot;     &quot;79643&quot;    
[2821] &quot;440465&quot;    &quot;10458&quot;     &quot;9625&quot;      &quot;22994&quot;     &quot;146705&quot;    &quot;71&quot;       
[2827] &quot;339229&quot;    &quot;339230&quot;    &quot;9146&quot;      &quot;6182&quot;      &quot;1468&quot;      &quot;10189&quot;    
[2833] &quot;256933&quot;    &quot;51547&quot;     &quot;4097&quot;      &quot;92659&quot;     &quot;5831&quot;      &quot;255275&quot;   
[2839] &quot;5881&quot;      &quot;51181&quot;     &quot;5986&quot;      &quot;2194&quot;      &quot;284001&quot;    &quot;1453&quot;     
[2845] &quot;79701&quot;     &quot;26502&quot;     &quot;79672&quot;     &quot;64122&quot;     &quot;146712&quot;    &quot;9097&quot;     
[2851] &quot;9984&quot;      &quot;27098&quot;     &quot;494514&quot;    &quot;55556&quot;     &quot;7525&quot;      &quot;64863&quot;    
[2857] &quot;23347&quot;     &quot;9663&quot;      &quot;8736&quot;      &quot;103910&quot;    &quot;7050&quot;      &quot;9229&quot;     
[2863] &quot;649446&quot;    &quot;84777&quot;     &quot;147525&quot;    &quot;91133&quot;     &quot;284217&quot;    &quot;201475&quot;   
[2869] &quot;23255&quot;     &quot;4729&quot;      &quot;101927275&quot; &quot;23253&quot;     &quot;11031&quot;     &quot;8774&quot;     
[2875] &quot;2774&quot;      &quot;57132&quot;     &quot;10939&quot;     &quot;55125&quot;     &quot;753&quot;       &quot;125228&quot;   
[2881] &quot;6632&quot;      &quot;171586&quot;    &quot;57534&quot;     &quot;102723167&quot; &quot;91768&quot;     &quot;8780&quot;     
[2887] &quot;29919&quot;     &quot;4864&quot;      &quot;125488&quot;    &quot;26256&quot;     &quot;114876&quot;    &quot;55364&quot;    
[2893] &quot;25925&quot;     &quot;6760&quot;      &quot;284252&quot;    &quot;1000&quot;      &quot;1825&quot;      &quot;1824&quot;     
[2899] &quot;80816&quot;     &quot;1837&quot;      &quot;100101467&quot; &quot;10778&quot;     &quot;252884&quot;    &quot;125476&quot;   
[2905] &quot;55197&quot;     &quot;80206&quot;     &quot;25941&quot;     &quot;57536&quot;     &quot;56853&quot;     &quot;641516&quot;   
[2911] &quot;5289&quot;      &quot;9050&quot;      &quot;498&quot;       &quot;147339&quot;    &quot;201501&quot;    &quot;9388&quot;     
[2917] &quot;4645&quot;      &quot;220136&quot;    &quot;30827&quot;     &quot;51320&quot;     &quot;11201&quot;     &quot;23335&quot;    
[2923] &quot;4677&quot;      &quot;115701&quot;    &quot;5366&quot;      &quot;220441&quot;    &quot;54877&quot;     &quot;23239&quot;    
[2929] &quot;2531&quot;      &quot;9525&quot;      &quot;8710&quot;      &quot;5271&quot;      &quot;92126&quot;     &quot;10666&quot;    
[2935] &quot;9306&quot;      &quot;147381&quot;    &quot;29090&quot;     &quot;1528&quot;      &quot;9658&quot;      &quot;101927989&quot;
[2941] &quot;4155&quot;      &quot;374868&quot;    &quot;9150&quot;      &quot;79863&quot;     &quot;84552&quot;     &quot;399844&quot;   
[2947] &quot;126567&quot;    &quot;25759&quot;     &quot;610&quot;       &quot;5442&quot;      &quot;10272&quot;     &quot;5725&quot;     
[2953] &quot;1675&quot;      &quot;57418&quot;     &quot;1265&quot;      &quot;23526&quot;     &quot;6794&quot;      &quot;513&quot;      
[2959] &quot;1153&quot;      &quot;55009&quot;     &quot;1943&quot;      &quot;84939&quot;     &quot;374291&quot;    &quot;26528&quot;    
[2965] &quot;6209&quot;      &quot;10297&quot;     &quot;92840&quot;     &quot;339366&quot;    &quot;53615&quot;     &quot;57455&quot;    
[2971] &quot;83855&quot;     &quot;113179&quot;    &quot;1455&quot;      &quot;55643&quot;     &quot;2872&quot;      &quot;126308&quot;   
[2977] &quot;113177&quot;    &quot;8943&quot;      &quot;55111&quot;     &quot;268&quot;       &quot;4616&quot;      &quot;2788&quot;     
[2983] &quot;148252&quot;    &quot;29985&quot;     &quot;7064&quot;      &quot;148254&quot;    &quot;126295&quot;    &quot;79816&quot;    
[2989] &quot;7089&quot;      &quot;166&quot;       &quot;2767&quot;      &quot;60680&quot;     &quot;4782&quot;      &quot;83475&quot;    
[2995] &quot;126321&quot;    &quot;100128569&quot; &quot;126326&quot;    &quot;6915&quot;      &quot;58509&quot;     &quot;23396&quot;    
[3001] &quot;27231&quot;     &quot;1613&quot;      &quot;51341&quot;     &quot;56961&quot;     &quot;729359&quot;    &quot;440503&quot;   
[3007] &quot;10501&quot;     &quot;55527&quot;     &quot;148022&quot;    &quot;5802&quot;      &quot;257000&quot;    &quot;125988&quot;   
[3013] &quot;4902&quot;      &quot;8192&quot;      &quot;84266&quot;     &quot;10382&quot;     &quot;10045&quot;     &quot;79230&quot;    
[3019] &quot;3643&quot;      &quot;23370&quot;     &quot;92960&quot;     &quot;57192&quot;     &quot;10908&quot;     &quot;56949&quot;    
[3025] &quot;100131801&quot; &quot;6813&quot;      &quot;80164&quot;     &quot;388499&quot;    &quot;80131&quot;     &quot;100507588&quot;
[3031] &quot;6618&quot;      &quot;10469&quot;     &quot;1994&quot;      &quot;256949&quot;    &quot;51257&quot;     &quot;84330&quot;    
[3037] &quot;81794&quot;     &quot;374879&quot;    &quot;84527&quot;     &quot;7675&quot;      &quot;93134&quot;     &quot;284385&quot;   
[3043] &quot;54850&quot;     &quot;59286&quot;     &quot;93145&quot;     &quot;51073&quot;     &quot;3383&quot;      &quot;3386&quot;     
[3049] &quot;7087&quot;      &quot;100125288&quot; &quot;112812&quot;    &quot;11140&quot;     &quot;5141&quot;      &quot;1032&quot;     
[3055] &quot;57153&quot;     &quot;147727&quot;    &quot;81890&quot;     &quot;11018&quot;     &quot;10498&quot;     &quot;78992&quot;    
[3061] &quot;90580&quot;     &quot;6597&quot;      &quot;3949&quot;      &quot;9545&quot;      &quot;374882&quot;    &quot;64748&quot;    
[3067] &quot;126074&quot;    &quot;57139&quot;     &quot;115950&quot;    &quot;51295&quot;     &quot;1264&quot;      &quot;84337&quot;    
[3073] &quot;664709&quot;    &quot;126070&quot;    &quot;7620&quot;      &quot;284390&quot;    &quot;163059&quot;    &quot;284391&quot;   
[3079] &quot;90589&quot;     &quot;147837&quot;    &quot;79973&quot;     &quot;90576&quot;     &quot;163050&quot;    &quot;57474&quot;    
[3085] &quot;4125&quot;      &quot;51398&quot;     &quot;84261&quot;     &quot;79002&quot;     &quot;439&quot;       &quot;3726&quot;     
[3091] &quot;811&quot;       &quot;5886&quot;      &quot;199699&quot;    &quot;4066&quot;      &quot;112939&quot;    &quot;8677&quot;     
[3097] &quot;9592&quot;      &quot;65249&quot;     &quot;5989&quot;      &quot;9466&quot;      &quot;5566&quot;      &quot;976&quot;      
[3103] &quot;5585&quot;      &quot;10755&quot;     &quot;9524&quot;      &quot;84449&quot;     &quot;85360&quot;     &quot;10994&quot;    
[3109] &quot;10270&quot;     &quot;58525&quot;     &quot;7171&quot;      &quot;4218&quot;      &quot;26017&quot;     &quot;9441&quot;     
[3115] &quot;79086&quot;     &quot;23309&quot;     &quot;9002&quot;      &quot;27151&quot;     &quot;4650&quot;      &quot;55850&quot;    
[3121] &quot;79629&quot;     &quot;79575&quot;     &quot;84705&quot;     &quot;93343&quot;     &quot;376497&quot;    &quot;3718&quot;     
[3127] &quot;27106&quot;     &quot;5296&quot;      &quot;10437&quot;     &quot;5143&quot;      &quot;9518&quot;      &quot;51477&quot;    
[3133] &quot;8178&quot;      &quot;7311&quot;      &quot;55049&quot;     &quot;55295&quot;     &quot;10715&quot;     &quot;11316&quot;    
[3139] &quot;54555&quot;     &quot;54929&quot;     &quot;8625&quot;      &quot;404037&quot;    &quot;53345&quot;     &quot;54815&quot;    
[3145] &quot;83983&quot;     &quot;374887&quot;    &quot;57130&quot;     &quot;94039&quot;     &quot;440515&quot;    &quot;81931&quot;    
[3151] &quot;284443&quot;    &quot;7594&quot;      &quot;100132815&quot; &quot;171392&quot;    &quot;9534&quot;      &quot;79156&quot;    
[3157] &quot;9745&quot;      &quot;57616&quot;     &quot;9141&quot;      &quot;390916&quot;    &quot;84902&quot;     &quot;55094&quot;    
[3163] &quot;1050&quot;      &quot;80054&quot;     &quot;9710&quot;      &quot;2821&quot;      &quot;84306&quot;     &quot;6324&quot;     
[3169] &quot;51599&quot;     &quot;7392&quot;      &quot;26330&quot;     &quot;10430&quot;     &quot;79171&quot;     &quot;9757&quot;     
[3175] &quot;79713&quot;     &quot;199746&quot;    &quot;55851&quot;     &quot;148137&quot;    &quot;333&quot;       &quot;79414&quot;    
[3181] &quot;84964&quot;     &quot;25999&quot;     &quot;5438&quot;      &quot;7705&quot;      &quot;100506930&quot; &quot;57677&quot;    
[3187] &quot;284406&quot;    &quot;57711&quot;     &quot;84911&quot;     &quot;163081&quot;    &quot;342892&quot;    &quot;728485&quot;   
[3193] &quot;374899&quot;    &quot;374900&quot;    &quot;84503&quot;     &quot;390927&quot;    &quot;163115&quot;    &quot;126231&quot;   
[3199] &quot;90522&quot;     &quot;57828&quot;     &quot;5714&quot;      &quot;199720&quot;    &quot;399473&quot;    &quot;147965&quot;   
[3205] &quot;3191&quot;      &quot;126432&quot;    &quot;22933&quot;     &quot;4793&quot;      &quot;54938&quot;     &quot;115290&quot;   
[3211] &quot;126433&quot;    &quot;10298&quot;     &quot;57622&quot;     &quot;55588&quot;     &quot;7538&quot;      &quot;64857&quot;    
[3217] &quot;4294&quot;      &quot;126526&quot;    &quot;29950&quot;     &quot;645&quot;       &quot;57731&quot;     &quot;92799&quot;    
[3223] &quot;80271&quot;     &quot;29785&quot;     &quot;7040&quot;      &quot;80776&quot;     &quot;56915&quot;     &quot;374907&quot;   
[3229] &quot;6223&quot;      &quot;9138&quot;      &quot;10567&quot;     &quot;478&quot;       &quot;2901&quot;      &quot;5452&quot;     
[3235] &quot;162989&quot;    &quot;2931&quot;      &quot;5050&quot;      &quot;1954&quot;      &quot;653583&quot;    &quot;7515&quot;     
[3241] &quot;126299&quot;    &quot;56006&quot;     &quot;342908&quot;    &quot;100505715&quot; &quot;7638&quot;      &quot;9310&quot;     
[3247] &quot;5819&quot;      &quot;1209&quot;      &quot;5971&quot;      &quot;284352&quot;    &quot;79090&quot;     &quot;388552&quot;   
[3253] &quot;90332&quot;     &quot;1158&quot;      &quot;2068&quot;      &quot;2067&quot;      &quot;2354&quot;      &quot;6253&quot;     
[3259] &quot;80207&quot;     &quot;2696&quot;      &quot;1762&quot;      &quot;8189&quot;      &quot;26145&quot;     &quot;729440&quot;   
[3265] &quot;64344&quot;     &quot;5536&quot;      &quot;55228&quot;     &quot;57469&quot;     &quot;29888&quot;     &quot;79147&quot;    
[3271] &quot;6510&quot;      &quot;4861&quot;      &quot;255783&quot;    &quot;9704&quot;      &quot;56917&quot;     &quot;8775&quot;     
[3277] &quot;30846&quot;     &quot;22900&quot;     &quot;163071&quot;    &quot;93233&quot;     &quot;2014&quot;      &quot;55260&quot;    
[3283] &quot;83743&quot;     &quot;3770&quot;      &quot;9266&quot;      &quot;114783&quot;    &quot;56848&quot;     &quot;1628&quot;     
[3289] &quot;54922&quot;     &quot;57664&quot;     &quot;23645&quot;     &quot;94027&quot;     &quot;4909&quot;      &quot;3743&quot;     
[3295] &quot;64130&quot;     &quot;8541&quot;      &quot;54795&quot;     &quot;8463&quot;      &quot;100507003&quot; &quot;51070&quot;    
[3301] &quot;3661&quot;      &quot;3276&quot;      &quot;81857&quot;     &quot;94030&quot;     &quot;84258&quot;     &quot;342918&quot;   
[3307] &quot;50944&quot;     &quot;2109&quot;      &quot;125875&quot;    &quot;79898&quot;     &quot;284370&quot;    &quot;90321&quot;    
[3313] &quot;400713&quot;    &quot;162966&quot;    &quot;7576&quot;      &quot;90333&quot;     &quot;162967&quot;    &quot;388559&quot;   
[3319] &quot;125893&quot;    &quot;79986&quot;     &quot;55786&quot;     &quot;84671&quot;     &quot;79788&quot;     &quot;390963&quot;   
[3325] &quot;91664&quot;     &quot;170958&quot;    &quot;147804&quot;    &quot;91663&quot;     &quot;59285&quot;     &quot;79165&quot;    
[3331] &quot;79143&quot;     &quot;148170&quot;    &quot;112724&quot;    &quot;7137&quot;      &quot;255043&quot;    &quot;22870&quot;    
[3337] &quot;84446&quot;     &quot;79763&quot;     &quot;57106&quot;     &quot;284297&quot;    &quot;163033&quot;    &quot;147807&quot;   
[3343] &quot;147808&quot;    &quot;51545&quot;     &quot;29903&quot;     &quot;11338&quot;     &quot;55311&quot;     &quot;79149&quot;    
[3349] &quot;147947&quot;    &quot;100128252&quot; &quot;57573&quot;     &quot;140612&quot;    &quot;58491&quot;     &quot;147694&quot;   
[3355] &quot;57191&quot;     &quot;162972&quot;    &quot;90233&quot;     &quot;79891&quot;     &quot;79818&quot;     &quot;2282&quot;     
[3361] &quot;100293516&quot; &quot;147686&quot;    &quot;7694&quot;      &quot;10782&quot;     &quot;342945&quot;    &quot;1&quot;        
[3367] &quot;503538&quot;    &quot;162968&quot;    &quot;116412&quot;    &quot;201514&quot;    &quot;388569&quot;    &quot;25799&quot;    
[3373] &quot;55663&quot;     &quot;84878&quot;     &quot;100131691&quot; &quot;7593&quot;      &quot;26751&quot;     &quot;129787&quot;   
[3379] &quot;7837&quot;      &quot;246243&quot;    &quot;6664&quot;      &quot;9781&quot;      &quot;100506299&quot; &quot;3398&quot;     
[3385] &quot;51692&quot;     &quot;285148&quot;    &quot;8462&quot;      &quot;3241&quot;      &quot;245973&quot;    &quot;3754&quot;     
[3391] &quot;51594&quot;     &quot;1653&quot;      &quot;10408&quot;     &quot;130497&quot;    &quot;4148&quot;      &quot;6382&quot;     
[3397] &quot;60526&quot;     &quot;114818&quot;    &quot;2281&quot;      &quot;9540&quot;      &quot;375189&quot;    &quot;50618&quot;    
[3403] &quot;109&quot;       &quot;22979&quot;     &quot;55252&quot;     &quot;10890&quot;     &quot;3777&quot;      &quot;56896&quot;    
[3409] &quot;60509&quot;     &quot;100128731&quot; &quot;3795&quot;      &quot;790&quot;       &quot;7781&quot;      &quot;57159&quot;    
[3415] &quot;7349&quot;      &quot;4358&quot;      &quot;5496&quot;      &quot;2498&quot;      &quot;26160&quot;     &quot;64838&quot;    
[3421] &quot;79635&quot;     &quot;9553&quot;      &quot;2355&quot;      &quot;151056&quot;    &quot;23160&quot;     &quot;388939&quot;   
[3427] &quot;79745&quot;     &quot;51646&quot;     &quot;79623&quot;     &quot;440854&quot;    &quot;84661&quot;     &quot;55676&quot;    
[3433] &quot;57448&quot;     &quot;55622&quot;     &quot;4052&quot;      &quot;25940&quot;     &quot;100288911&quot; &quot;9637&quot;     
[3439] &quot;151393&quot;    &quot;1545&quot;      &quot;130589&quot;    &quot;6432&quot;      &quot;79833&quot;     &quot;6654&quot;     
[3445] &quot;728730&quot;    &quot;130733&quot;    &quot;6546&quot;      &quot;91461&quot;     &quot;27436&quot;     &quot;9167&quot;     
[3451] &quot;57504&quot;     &quot;678&quot;       &quot;51626&quot;     &quot;10128&quot;     &quot;5495&quot;      &quot;9581&quot;     
[3457] &quot;55133&quot;     &quot;102724965&quot; &quot;2034&quot;      &quot;90423&quot;     &quot;5281&quot;      &quot;90411&quot;    
[3463] &quot;57217&quot;     &quot;3344&quot;      &quot;11037&quot;     &quot;98&quot;        &quot;400954&quot;    &quot;57142&quot;    
[3469] &quot;100129434&quot; &quot;114800&quot;    &quot;339803&quot;    &quot;339804&quot;    &quot;9736&quot;      &quot;10678&quot;    
[3475] &quot;200728&quot;    &quot;23301&quot;     &quot;4190&quot;      &quot;51542&quot;     &quot;57162&quot;     &quot;29094&quot;    
[3481] &quot;54812&quot;     &quot;9792&quot;      &quot;23177&quot;     &quot;200734&quot;    &quot;116143&quot;    &quot;200558&quot;   
[3487] &quot;307&quot;       &quot;64395&quot;     &quot;11017&quot;     &quot;151516&quot;    &quot;400960&quot;    &quot;54980&quot;    
[3493] &quot;7072&quot;      &quot;10199&quot;     &quot;27332&quot;     &quot;8291&quot;      &quot;56603&quot;     &quot;94097&quot;    
[3499] &quot;26056&quot;     &quot;10322&quot;     &quot;150726&quot;    &quot;7840&quot;      &quot;51002&quot;     &quot;10617&quot;    
[3505] &quot;72&quot;        &quot;388962&quot;    &quot;100507171&quot; &quot;55233&quot;     &quot;388963&quot;    &quot;23559&quot;    
[3511] &quot;7841&quot;      &quot;116540&quot;    &quot;64427&quot;     &quot;84759&quot;     &quot;550&quot;       &quot;27429&quot;    
[3517] &quot;10505&quot;     &quot;84141&quot;     &quot;9801&quot;      &quot;347730&quot;    &quot;8802&quot;      &quot;10618&quot;    
[3523] &quot;54884&quot;     &quot;100630918&quot; &quot;4144&quot;      &quot;8673&quot;      &quot;51255&quot;     &quot;55037&quot;    
[3529] &quot;10989&quot;     &quot;51318&quot;     &quot;65055&quot;     &quot;55818&quot;     &quot;64795&quot;     &quot;112597&quot;   
[3535] &quot;730268&quot;    &quot;51315&quot;     &quot;150572&quot;    &quot;9451&quot;      &quot;101928403&quot; &quot;22934&quot;    
[3541] &quot;150696&quot;    &quot;30818&quot;     &quot;150759&quot;    &quot;400986&quot;    &quot;729234&quot;    &quot;56910&quot;    
[3547] &quot;55654&quot;     &quot;10865&quot;     &quot;81562&quot;     &quot;26504&quot;     &quot;26505&quot;     &quot;200539&quot;   
[3553] &quot;151313&quot;    &quot;375248&quot;    &quot;100506076&quot; &quot;100506123&quot; &quot;57730&quot;     &quot;10120&quot;    
[3559] &quot;728537&quot;    &quot;11320&quot;     &quot;343990&quot;    &quot;80705&quot;     &quot;51263&quot;     &quot;164832&quot;   
[3565] &quot;9486&quot;      &quot;79031&quot;     &quot;55571&quot;     &quot;284996&quot;    &quot;6549&quot;      &quot;64965&quot;    
[3571] &quot;9392&quot;      &quot;100506473&quot; &quot;2274&quot;      &quot;80146&quot;     &quot;9648&quot;      &quot;3987&quot;     
[3577] &quot;344558&quot;    &quot;65124&quot;     &quot;4867&quot;      &quot;100288695&quot; &quot;55289&quot;     &quot;10018&quot;    
[3583] &quot;541471&quot;    &quot;64682&quot;     &quot;10461&quot;     &quot;84910&quot;     &quot;129804&quot;    &quot;376940&quot;   
[3589] &quot;150465&quot;    &quot;84172&quot;     &quot;84269&quot;     &quot;23550&quot;     &quot;375260&quot;    &quot;11159&quot;    
[3595] &quot;54520&quot;     &quot;51141&quot;     &quot;130355&quot;    &quot;1622&quot;      &quot;80775&quot;     &quot;5775&quot;     
[3601] &quot;57669&quot;     &quot;79134&quot;     &quot;5899&quot;      &quot;23332&quot;     &quot;254128&quot;    &quot;84365&quot;    
[3607] &quot;10746&quot;     &quot;100506922&quot; &quot;55679&quot;     &quot;2840&quot;      &quot;55339&quot;     &quot;5433&quot;     
[3613] &quot;79595&quot;     &quot;9394&quot;      &quot;91409&quot;     &quot;55627&quot;     &quot;653269&quot;    &quot;653781&quot;   
[3619] &quot;55041&quot;     &quot;445582&quot;    &quot;401010&quot;    &quot;150776&quot;    &quot;116372&quot;    &quot;4249&quot;     
[3625] &quot;100129961&quot; &quot;23518&quot;     &quot;7852&quot;      &quot;11249&quot;     &quot;79712&quot;     &quot;92&quot;       
[3631] &quot;5000&quot;      &quot;3800&quot;      &quot;390&quot;       &quot;26225&quot;     &quot;114793&quot;    &quot;114805&quot;   
[3637] &quot;4929&quot;      &quot;11227&quot;     &quot;8502&quot;      &quot;85461&quot;     &quot;151525&quot;    &quot;643072&quot;   
[3643] &quot;64844&quot;     &quot;10213&quot;     &quot;1803&quot;      &quot;64135&quot;     &quot;90134&quot;     &quot;55137&quot;    
[3649] &quot;22837&quot;     &quot;80034&quot;     &quot;129446&quot;    &quot;27347&quot;     &quot;115677&quot;    &quot;10170&quot;    
[3655] &quot;4036&quot;      &quot;129880&quot;    &quot;79675&quot;     &quot;29081&quot;     &quot;285141&quot;    &quot;2571&quot;     
[3661] &quot;26003&quot;     &quot;9874&quot;      &quot;79828&quot;     &quot;80067&quot;     &quot;8604&quot;      &quot;254042&quot;   
[3667] &quot;6670&quot;      &quot;29789&quot;     &quot;9541&quot;      &quot;440926&quot;    &quot;1123&quot;      &quot;518&quot;      
[3673] &quot;80856&quot;     &quot;4780&quot;      &quot;100130691&quot; &quot;150737&quot;    &quot;92104&quot;     &quot;129831&quot;   
[3679] &quot;114880&quot;    &quot;101927027&quot; &quot;8575&quot;      &quot;494513&quot;    &quot;51661&quot;     &quot;7273&quot;     
[3685] &quot;285025&quot;    &quot;3676&quot;      &quot;375298&quot;    &quot;54431&quot;     &quot;10787&quot;     &quot;129401&quot;   
[3691] &quot;401024&quot;    &quot;10203&quot;     &quot;51454&quot;     &quot;54529&quot;     &quot;64172&quot;     &quot;5378&quot;     
[3697] &quot;26275&quot;     &quot;3628&quot;      &quot;54842&quot;     &quot;105373801&quot; &quot;2744&quot;      &quot;6772&quot;     
[3703] &quot;6775&quot;      &quot;4430&quot;      &quot;64859&quot;     &quot;9262&quot;      &quot;57520&quot;     &quot;9330&quot;     
[3709] &quot;92935&quot;     &quot;23314&quot;     &quot;205327&quot;    &quot;79568&quot;     &quot;26010&quot;     &quot;130535&quot;   
[3715] &quot;9689&quot;      &quot;1195&quot;      &quot;53938&quot;     &quot;60491&quot;     &quot;4999&quot;      &quot;8324&quot;     
[3721] &quot;55759&quot;     &quot;79800&quot;     &quot;65065&quot;     &quot;8828&quot;      &quot;100329109&quot; &quot;4719&quot;     
[3727] &quot;2825&quot;      &quot;22868&quot;     &quot;151194&quot;    &quot;151195&quot;    &quot;7855&quot;      &quot;33&quot;       
[3733] &quot;1373&quot;      &quot;2066&quot;      &quot;22807&quot;     &quot;580&quot;       &quot;26154&quot;     &quot;55686&quot;    
[3739] &quot;55825&quot;     &quot;150967&quot;    &quot;57574&quot;     &quot;7145&quot;      &quot;14&quot;        &quot;25953&quot;    
[3745] &quot;64114&quot;     &quot;9125&quot;      &quot;7701&quot;      &quot;617&quot;       &quot;9654&quot;      &quot;79840&quot;    
[3751] &quot;79137&quot;     &quot;10058&quot;     &quot;55139&quot;     &quot;7277&quot;      &quot;1674&quot;      &quot;10290&quot;    
[3757] &quot;100996693&quot; &quot;79586&quot;     &quot;23363&quot;     &quot;114790&quot;    &quot;6508&quot;      &quot;130367&quot;   
[3763] &quot;10056&quot;     &quot;23704&quot;     &quot;65080&quot;     &quot;5270&quot;      &quot;80309&quot;     &quot;55022&quot;    
[3769] &quot;9320&quot;      &quot;130888&quot;    &quot;151473&quot;    &quot;3431&quot;      &quot;51719&quot;     &quot;257407&quot;   
[3775] &quot;93010&quot;     &quot;105373926&quot; &quot;4691&quot;      &quot;5757&quot;      &quot;129563&quot;    &quot;9427&quot;     
[3781] &quot;200765&quot;    &quot;80303&quot;     &quot;55054&quot;     &quot;151507&quot;    &quot;10123&quot;     &quot;57007&quot;    
[3787] &quot;9208&quot;      &quot;377007&quot;    &quot;8864&quot;      &quot;51665&quot;     &quot;150678&quot;    &quot;51281&quot;    
[3793] &quot;285193&quot;    &quot;101752400&quot; &quot;11132&quot;     &quot;2859&quot;      &quot;547&quot;       &quot;25992&quot;    
[3799] &quot;130916&quot;    &quot;5510&quot;      &quot;4735&quot;      &quot;10494&quot;     &quot;728323&quot;    &quot;140680&quot;   
[3805] &quot;85364&quot;     &quot;100507459&quot; &quot;6666&quot;      &quot;57761&quot;     &quot;10616&quot;     &quot;128637&quot;   
[3811] &quot;83541&quot;     &quot;9751&quot;      &quot;2280&quot;      &quot;140901&quot;    &quot;10528&quot;     &quot;57593&quot;    
[3817] &quot;64773&quot;     &quot;65992&quot;     &quot;3704&quot;      &quot;83959&quot;     &quot;25943&quot;     &quot;80025&quot;    
[3823] &quot;54498&quot;     &quot;146&quot;       &quot;5621&quot;      &quot;9770&quot;      &quot;5111&quot;      &quot;8760&quot;     
[3829] &quot;56261&quot;     &quot;51605&quot;     &quot;54675&quot;     &quot;164312&quot;    &quot;56255&quot;     &quot;5332&quot;     
[3835] &quot;63926&quot;     &quot;6616&quot;      &quot;8195&quot;      &quot;22903&quot;     &quot;55304&quot;     &quot;79133&quot;    
[3841] &quot;140733&quot;    &quot;23767&quot;     &quot;55614&quot;     &quot;6629&quot;      &quot;631&quot;       &quot;92667&quot;    
[3847] &quot;100303755&quot; &quot;57325&quot;     &quot;55184&quot;     &quot;10741&quot;     &quot;388789&quot;    &quot;57419&quot;    
[3853] &quot;54453&quot;     &quot;51340&quot;     &quot;57186&quot;     &quot;55857&quot;     &quot;63908&quot;     &quot;1471&quot;     
[3859] &quot;955&quot;       &quot;26090&quot;     &quot;140838&quot;    &quot;102724826&quot; &quot;100101490&quot; &quot;100874042&quot;
[3865] &quot;3397&quot;      &quot;128854&quot;    &quot;23509&quot;     &quot;22919&quot;     &quot;6640&quot;      &quot;9139&quot;     
[3871] &quot;11264&quot;     &quot;128866&quot;    &quot;84557&quot;     &quot;128869&quot;    &quot;58476&quot;     &quot;2686&quot;     
[3877] &quot;55902&quot;     &quot;57644&quot;     &quot;10544&quot;     &quot;3692&quot;      &quot;55245&quot;     &quot;2036&quot;     
[3883] &quot;25980&quot;     &quot;57446&quot;     &quot;25939&quot;     &quot;140699&quot;    &quot;6185&quot;      &quot;63905&quot;    
[3889] &quot;6714&quot;      &quot;7052&quot;      &quot;388796&quot;    &quot;128439&quot;    &quot;79913&quot;     &quot;60625&quot;    
[3895] &quot;64900&quot;     &quot;84181&quot;     &quot;6431&quot;      &quot;51098&quot;     &quot;57158&quot;     &quot;51526&quot;    
[3901] &quot;100505783&quot; &quot;128486&quot;    &quot;79183&quot;     &quot;11142&quot;     &quot;100&quot;       &quot;140730&quot;   
[3907] &quot;10953&quot;     &quot;6789&quot;      &quot;6385&quot;      &quot;27296&quot;     &quot;55861&quot;     &quot;51604&quot;    
[3913] &quot;90203&quot;     &quot;5476&quot;      &quot;63925&quot;     &quot;57727&quot;     &quot;63916&quot;     &quot;55713&quot;    
[3919] &quot;64849&quot;     &quot;81031&quot;     &quot;8202&quot;      &quot;55959&quot;     &quot;57169&quot;     &quot;5740&quot;     
[3925] &quot;23315&quot;     &quot;9825&quot;      &quot;387521&quot;    &quot;1051&quot;      &quot;5770&quot;      &quot;8813&quot;     
[3931] &quot;27304&quot;     &quot;57167&quot;     &quot;55734&quot;     &quot;128553&quot;    &quot;7764&quot;      &quot;116151&quot;   
[3937] &quot;51507&quot;     &quot;655&quot;       &quot;8480&quot;      &quot;55544&quot;     &quot;56937&quot;     &quot;57403&quot;    
[3943] &quot;8675&quot;      &quot;2778&quot;      &quot;51497&quot;     &quot;63939&quot;     &quot;5509&quot;      &quot;1002&quot;     
[3949] &quot;5688&quot;      &quot;26164&quot;     &quot;11047&quot;     &quot;3911&quot;      &quot;81928&quot;     &quot;140628&quot;   
[3955] &quot;253868&quot;    &quot;128826&quot;    &quot;55257&quot;     &quot;11054&quot;     &quot;54994&quot;     &quot;55738&quot;    
[3961] &quot;1917&quot;      &quot;79144&quot;     &quot;51750&quot;     &quot;84619&quot;     &quot;54923&quot;     &quot;7165&quot;     
[3967] &quot;54963&quot;     &quot;57473&quot;     &quot;140700&quot;    &quot;10287&quot;     &quot;55251&quot;     &quot;102723996&quot;
[3973] &quot;102724200&quot; &quot;102724428&quot; &quot;102724560&quot; &quot;102723475&quot; &quot;29761&quot;     &quot;388815&quot;   
[3979] &quot;10950&quot;     &quot;54149&quot;     &quot;140578&quot;    &quot;2551&quot;      &quot;11096&quot;     &quot;29104&quot;    
[3985] &quot;10069&quot;     &quot;10694&quot;     &quot;571&quot;       &quot;6647&quot;      &quot;57466&quot;     &quot;8867&quot;     
[3991] &quot;100506215&quot; &quot;3455&quot;      &quot;9946&quot;      &quot;539&quot;       &quot;64968&quot;     &quot;54065&quot;    
[3997] &quot;3753&quot;      &quot;861&quot;       &quot;9980&quot;      &quot;23515&quot;     &quot;3141&quot;      &quot;10311&quot;    
[4003] &quot;3150&quot;      &quot;7485&quot;      &quot;6450&quot;      &quot;54101&quot;     &quot;63977&quot;     &quot;49854&quot;    
[4009] &quot;9619&quot;      &quot;54020&quot;     &quot;5152&quot;      &quot;5316&quot;      &quot;875&quot;       &quot;150094&quot;   
[4015] &quot;56894&quot;     &quot;7109&quot;      &quot;23308&quot;     &quot;81543&quot;     &quot;54082&quot;     &quot;6612&quot;     
[4021] &quot;85395&quot;     &quot;728039&quot;    &quot;6573&quot;      &quot;1291&quot;      &quot;10841&quot;     &quot;84221&quot;    
[4027] &quot;4047&quot;      &quot;114044&quot;    &quot;23181&quot;     &quot;23765&quot;     &quot;27439&quot;     &quot;27440&quot;    
[4033] &quot;83733&quot;     &quot;529&quot;       &quot;637&quot;       &quot;57553&quot;     &quot;51807&quot;     &quot;728229&quot;   
[4039] &quot;728233&quot;    &quot;85376&quot;     &quot;5625&quot;      &quot;25786&quot;     &quot;8220&quot;      &quot;128977&quot;   
[4045] &quot;54584&quot;     &quot;79680&quot;     &quot;421&quot;       &quot;128989&quot;    &quot;54487&quot;     &quot;5902&quot;     
[4051] &quot;7625&quot;      &quot;84222&quot;     &quot;3053&quot;      &quot;1399&quot;      &quot;439931&quot;    &quot;6545&quot;     
[4057] &quot;440804&quot;    &quot;23119&quot;     &quot;645426&quot;    &quot;375133&quot;    &quot;150223&quot;    &quot;23753&quot;    
[4063] &quot;23759&quot;     &quot;140883&quot;    &quot;2781&quot;      &quot;400916&quot;    &quot;91319&quot;     &quot;284889&quot;   
[4069] &quot;23523&quot;     &quot;83606&quot;     &quot;1416&quot;      &quot;84700&quot;     &quot;57168&quot;     &quot;24144&quot;    
[4075] &quot;23331&quot;     &quot;150275&quot;    &quot;7494&quot;      &quot;83999&quot;     &quot;25807&quot;     &quot;10633&quot;    
[4081] &quot;4744&quot;      &quot;8508&quot;      &quot;4771&quot;      &quot;55954&quot;     &quot;550631&quot;    &quot;23541&quot;    
[4087] &quot;6948&quot;      &quot;23762&quot;     &quot;140606&quot;    &quot;27124&quot;     &quot;91445&quot;     &quot;3985&quot;     
[4093] &quot;4733&quot;      &quot;9814&quot;      &quot;23761&quot;     &quot;253143&quot;    &quot;7533&quot;      &quot;6523&quot;     
[4099] &quot;7078&quot;      &quot;3162&quot;      &quot;23780&quot;     &quot;80020&quot;     &quot;11020&quot;     &quot;7263&quot;     
[4105] &quot;4357&quot;      &quot;79734&quot;     &quot;114794&quot;    &quot;29775&quot;     &quot;23616&quot;     &quot;3956&quot;     
[4111] &quot;79159&quot;     &quot;85377&quot;     &quot;84645&quot;     &quot;80115&quot;     &quot;8398&quot;      &quot;23764&quot;    
[4117] &quot;102800317&quot; &quot;400927&quot;    &quot;11015&quot;     &quot;10521&quot;     &quot;9929&quot;      &quot;9567&quot;     
[4123] &quot;27350&quot;     &quot;23492&quot;     &quot;5155&quot;      &quot;9145&quot;      &quot;10454&quot;     &quot;468&quot;      
[4129] &quot;10478&quot;     &quot;6767&quot;      &quot;84844&quot;     &quot;27351&quot;     &quot;2547&quot;      &quot;6721&quot;     
[4135] &quot;55964&quot;     &quot;4668&quot;      &quot;4700&quot;      &quot;100132273&quot; &quot;27341&quot;     &quot;84271&quot;    
[4141] &quot;53947&quot;     &quot;26286&quot;     &quot;11252&quot;     &quot;25809&quot;     &quot;638&quot;       &quot;706&quot;      
[4147] &quot;23170&quot;     &quot;80339&quot;     &quot;84247&quot;     &quot;55615&quot;     &quot;112885&quot;    &quot;23313&quot;    
[4153] &quot;55020&quot;     &quot;55687&quot;     &quot;9620&quot;      &quot;23151&quot;     &quot;9889&quot;      &quot;79087&quot;    
[4159] &quot;80305&quot;     &quot;83642&quot;     &quot;6300&quot;      &quot;5600&quot;      &quot;6305&quot;      &quot;91289&quot;    
[4165] &quot;1890&quot;      &quot;100144603&quot; &quot;410&quot;       &quot;85358&quot;     &quot;11158&quot;     &quot;57633&quot;    
[4171] &quot;100996539&quot; &quot;3708&quot;      &quot;9695&quot;      &quot;859&quot;       &quot;56852&quot;     &quot;9901&quot;     
[4177] &quot;25917&quot;     &quot;55209&quot;     &quot;64419&quot;     &quot;7862&quot;      &quot;4968&quot;      &quot;8536&quot;     
[4183] &quot;26140&quot;     &quot;285367&quot;    &quot;84818&quot;     &quot;285368&quot;    &quot;6396&quot;      &quot;6529&quot;     
[4189] &quot;132001&quot;    &quot;5468&quot;      &quot;80746&quot;     &quot;23609&quot;     &quot;23066&quot;     &quot;9922&quot;     
[4195] &quot;23225&quot;     &quot;7476&quot;      &quot;7508&quot;      &quot;27258&quot;     &quot;80852&quot;     &quot;64432&quot;    
[4201] &quot;64145&quot;     &quot;85403&quot;     &quot;8292&quot;      &quot;686&quot;       &quot;23243&quot;     &quot;285381&quot;   
[4207] &quot;92106&quot;     &quot;23228&quot;     &quot;9779&quot;      &quot;5868&quot;      &quot;8850&quot;      &quot;7325&quot;     
[4213] &quot;6138&quot;      &quot;9975&quot;      &quot;7068&quot;      &quot;5915&quot;      &quot;7155&quot;      &quot;64343&quot;    
[4219] &quot;27303&quot;     &quot;201595&quot;    &quot;114884&quot;    &quot;23171&quot;     &quot;112616&quot;    &quot;54918&quot;    
[4225] &quot;105377021&quot; &quot;131405&quot;    &quot;26032&quot;     &quot;7342&quot;      &quot;6769&quot;      &quot;4292&quot;     
[4231] &quot;5333&quot;      &quot;9943&quot;      &quot;100128640&quot; &quot;9941&quot;      &quot;6331&quot;      &quot;57599&quot;    
[4237] &quot;199223&quot;    &quot;64651&quot;     &quot;165904&quot;    &quot;54977&quot;     &quot;10289&quot;     &quot;285267&quot;   
[4243] &quot;285268&quot;    &quot;131375&quot;    &quot;9117&quot;      &quot;92999&quot;     &quot;57467&quot;     &quot;1238&quot;     
[4249] &quot;54861&quot;     &quot;51099&quot;     &quot;285346&quot;    &quot;55888&quot;     &quot;285349&quot;    &quot;115560&quot;   
[4255] &quot;51304&quot;     &quot;25907&quot;     &quot;54716&quot;     &quot;79442&quot;     &quot;339906&quot;    &quot;285352&quot;   
[4261] &quot;64147&quot;     &quot;25930&quot;     &quot;22937&quot;     &quot;54859&quot;     &quot;10675&quot;     &quot;10201&quot;    
[4267] &quot;5364&quot;      &quot;79714&quot;     &quot;51372&quot;     &quot;84126&quot;     &quot;11277&quot;     &quot;51517&quot;    
[4273] &quot;100506637&quot; &quot;646450&quot;    &quot;10425&quot;     &quot;54681&quot;     &quot;10869&quot;     &quot;3913&quot;     
[4279] &quot;375341&quot;    &quot;7375&quot;      &quot;2876&quot;      &quot;275&quot;       &quot;1605&quot;      &quot;386724&quot;   
[4285] &quot;389119&quot;    &quot;7318&quot;      &quot;84315&quot;     &quot;6405&quot;      &quot;10991&quot;     &quot;2771&quot;     
[4291] &quot;132228&quot;    &quot;7866&quot;      &quot;8372&quot;      &quot;11334&quot;     &quot;11186&quot;     &quot;10641&quot;    
[4297] &quot;51161&quot;     &quot;1795&quot;      &quot;7873&quot;      &quot;29890&quot;     &quot;51368&quot;     &quot;9136&quot;     
[4303] &quot;10039&quot;     &quot;57060&quot;     &quot;95&quot;        &quot;1849&quot;      &quot;211&quot;       &quot;51533&quot;    
[4309] &quot;56920&quot;     &quot;7134&quot;      &quot;64943&quot;     &quot;55193&quot;     &quot;55830&quot;     &quot;375346&quot;   
[4315] &quot;91869&quot;     &quot;5580&quot;      &quot;7086&quot;      &quot;55540&quot;     &quot;93973&quot;     &quot;58515&quot;    
[4321] &quot;55799&quot;     &quot;7474&quot;      &quot;26059&quot;     &quot;285331&quot;    &quot;50650&quot;     &quot;26060&quot;    
[4327] &quot;378&quot;       &quot;7871&quot;      &quot;57406&quot;     &quot;11102&quot;     &quot;54899&quot;     &quot;8309&quot;     
[4333] &quot;2272&quot;      &quot;57415&quot;     &quot;8618&quot;      &quot;6314&quot;      &quot;100507062&quot; &quot;166336&quot;   
[4339] &quot;56999&quot;     &quot;9223&quot;      &quot;84541&quot;     &quot;56203&quot;     &quot;23150&quot;     &quot;27086&quot;    
[4345] &quot;105377206&quot; &quot;317649&quot;    &quot;23429&quot;     &quot;55164&quot;     &quot;100846999&quot; &quot;692099&quot;   
[4351] &quot;6092&quot;      &quot;6091&quot;      &quot;2632&quot;      &quot;389136&quot;    &quot;25978&quot;     &quot;8545&quot;     
[4357] &quot;105377200&quot; &quot;55279&quot;     &quot;63899&quot;     &quot;84100&quot;     &quot;84864&quot;     &quot;1295&quot;     
[4363] &quot;84319&quot;     &quot;11259&quot;     &quot;57337&quot;     &quot;27107&quot;     &quot;64332&quot;     &quot;868&quot;      
[4369] &quot;56987&quot;     &quot;961&quot;       &quot;90102&quot;     &quot;55347&quot;     &quot;25871&quot;     &quot;80218&quot;    
[4375] &quot;523&quot;       &quot;2596&quot;      &quot;4045&quot;      &quot;56983&quot;     &quot;51365&quot;     &quot;64091&quot;    
[4381] &quot;116064&quot;    &quot;4710&quot;      &quot;51725&quot;     &quot;131076&quot;    &quot;151636&quot;    &quot;54625&quot;    
[4387] &quot;79663&quot;     &quot;84925&quot;     &quot;100129550&quot; &quot;54437&quot;     &quot;111&quot;       &quot;201562&quot;   
[4393] &quot;100506826&quot; &quot;4638&quot;      &quot;7372&quot;      &quot;10840&quot;     &quot;28999&quot;     &quot;79364&quot;    
[4399] &quot;114112&quot;    &quot;84303&quot;     &quot;5361&quot;      &quot;2624&quot;      &quot;7879&quot;      &quot;57501&quot;    
[4405] &quot;339122&quot;    &quot;57461&quot;     &quot;56941&quot;     &quot;132241&quot;    &quot;100507032&quot; &quot;729375&quot;   
[4411] &quot;27032&quot;     &quot;79858&quot;     &quot;152195&quot;    &quot;131870&quot;    &quot;11222&quot;     &quot;23317&quot;    
[4417] &quot;100532724&quot; &quot;66000&quot;     &quot;55573&quot;     &quot;58477&quot;     &quot;6578&quot;      &quot;25847&quot;    
[4423] &quot;80254&quot;     &quot;55167&quot;     &quot;10274&quot;     &quot;80723&quot;     &quot;101927597&quot; &quot;25852&quot;    
[4429] &quot;83850&quot;     &quot;80321&quot;     &quot;5291&quot;      &quot;5947&quot;      &quot;349565&quot;    &quot;5922&quot;     
[4435] &quot;7029&quot;      &quot;256356&quot;    &quot;54464&quot;     &quot;545&quot;       &quot;26577&quot;     &quot;344838&quot;   
[4441] &quot;100289361&quot; &quot;9435&quot;      &quot;285195&quot;    &quot;57088&quot;     &quot;5359&quot;      &quot;6596&quot;     
[4447] &quot;84343&quot;     &quot;4071&quot;      &quot;51122&quot;     &quot;11342&quot;     &quot;5217&quot;      &quot;646903&quot;   
[4453] &quot;653659&quot;    &quot;83939&quot;     &quot;116931&quot;    &quot;9934&quot;      &quot;53836&quot;     &quot;285313&quot;   
[4459] &quot;4154&quot;      &quot;401093&quot;    &quot;5028&quot;      &quot;5912&quot;      &quot;285315&quot;    &quot;9197&quot;     
[4465] &quot;6747&quot;      &quot;647033&quot;    &quot;100498859&quot; &quot;79674&quot;     &quot;5806&quot;      &quot;51319&quot;    
[4471] &quot;100996447&quot; &quot;4291&quot;      &quot;8706&quot;      &quot;101243545&quot; &quot;590&quot;       &quot;5274&quot;     
[4477] &quot;27333&quot;     &quot;55892&quot;     &quot;5584&quot;      &quot;5010&quot;      &quot;200916&quot;    &quot;6514&quot;     
[4483] &quot;23043&quot;     &quot;5337&quot;      &quot;57552&quot;     &quot;22871&quot;     &quot;254827&quot;    &quot;79718&quot;    
[4489] &quot;27094&quot;     &quot;51193&quot;     &quot;59345&quot;     &quot;8975&quot;      &quot;151613&quot;    &quot;131118&quot;   
[4495] &quot;56922&quot;     &quot;27074&quot;     &quot;100505687&quot; &quot;79929&quot;     &quot;55324&quot;     &quot;10195&quot;    
[4501] &quot;9718&quot;      &quot;5708&quot;      &quot;131408&quot;    &quot;8646&quot;      &quot;64110&quot;     &quot;9175&quot;     
[4507] &quot;90407&quot;     &quot;2119&quot;      &quot;1974&quot;      &quot;6480&quot;      &quot;604&quot;       &quot;4026&quot;     
[4513] &quot;9076&quot;      &quot;3556&quot;      &quot;2257&quot;      &quot;105374278&quot; &quot;131578&quot;    &quot;79572&quot;    
[4519] &quot;401106&quot;    &quot;100507297&quot; &quot;55341&quot;     &quot;105374297&quot; &quot;200958&quot;    &quot;10188&quot;    
[4525] &quot;7037&quot;      &quot;26043&quot;     &quot;255798&quot;    &quot;84984&quot;     &quot;5062&quot;      &quot;205564&quot;   
[4531] &quot;22916&quot;     &quot;152217&quot;    &quot;80235&quot;     &quot;4241&quot;      &quot;100507057&quot; &quot;1739&quot;     
[4537] &quot;622&quot;       &quot;9711&quot;      &quot;84859&quot;     &quot;89782&quot;     &quot;255403&quot;    &quot;79963&quot;    
[4543] &quot;54872&quot;     &quot;5158&quot;      &quot;4636&quot;      &quot;84179&quot;     &quot;53834&quot;     &quot;10417&quot;    
[4549] &quot;92070&quot;     &quot;7884&quot;      &quot;2261&quot;      &quot;353497&quot;    &quot;10608&quot;     &quot;317648&quot;   
[4555] &quot;285489&quot;    &quot;4043&quot;      &quot;152&quot;       &quot;27065&quot;     &quot;4487&quot;      &quot;54360&quot;    
[4561] &quot;55351&quot;     &quot;132884&quot;    &quot;2121&quot;      &quot;5522&quot;      &quot;23324&quot;     &quot;55330&quot;    
[4567] &quot;57533&quot;     &quot;257236&quot;    &quot;80273&quot;     &quot;60312&quot;     &quot;84448&quot;     &quot;54436&quot;    
[4573] &quot;8310&quot;      &quot;259282&quot;    &quot;132864&quot;    &quot;57545&quot;     &quot;202018&quot;    &quot;202020&quot;   
[4579] &quot;5860&quot;      &quot;51056&quot;     &quot;80306&quot;     &quot;27146&quot;     &quot;54876&quot;     &quot;166647&quot;   
[4585] &quot;10891&quot;     &quot;1665&quot;      &quot;91050&quot;     &quot;55203&quot;     &quot;29063&quot;     &quot;23231&quot;    
[4591] &quot;3516&quot;      &quot;55296&quot;     &quot;57620&quot;     &quot;5099&quot;      &quot;116984&quot;    &quot;51274&quot;    
[4597] &quot;92689&quot;     &quot;5981&quot;      &quot;6133&quot;      &quot;201895&quot;    &quot;3093&quot;      &quot;386617&quot;   
[4603] &quot;54951&quot;     &quot;10699&quot;     &quot;152519&quot;    &quot;57606&quot;     &quot;339977&quot;    &quot;64854&quot;    
[4609] &quot;643783&quot;    &quot;100288413&quot; &quot;65997&quot;     &quot;152579&quot;    &quot;26511&quot;     &quot;5156&quot;     
[4615] &quot;3815&quot;      &quot;79644&quot;     &quot;55763&quot;     &quot;9662&quot;      &quot;132949&quot;    &quot;6731&quot;     
[4621] &quot;84525&quot;     &quot;84273&quot;     &quot;5431&quot;      &quot;23284&quot;     &quot;253017&quot;    &quot;1060&quot;     
[4627] &quot;57050&quot;     &quot;22902&quot;     &quot;92597&quot;     &quot;8671&quot;      &quot;255324&quot;    &quot;9908&quot;     
[4633] &quot;55153&quot;     &quot;950&quot;       &quot;8987&quot;      &quot;55752&quot;     &quot;901&quot;       &quot;246175&quot;   
[4639] &quot;65008&quot;     &quot;80144&quot;     &quot;55589&quot;     &quot;56978&quot;     &quot;2250&quot;      &quot;153020&quot;   
[4645] &quot;3184&quot;      &quot;9987&quot;      &quot;58478&quot;     &quot;79966&quot;     &quot;100499177&quot; &quot;132660&quot;   
[4651] &quot;10855&quot;     &quot;113510&quot;    &quot;1040&quot;      &quot;404201&quot;    &quot;51170&quot;     &quot;53343&quot;    
[4657] &quot;152926&quot;    &quot;51191&quot;     &quot;100873748&quot; &quot;266812&quot;    &quot;10144&quot;     &quot;285513&quot;   
[4663] &quot;56916&quot;     &quot;10611&quot;     &quot;5910&quot;      &quot;10098&quot;     &quot;23173&quot;     &quot;128&quot;      
[4669] &quot;100507053&quot; &quot;93587&quot;     &quot;8649&quot;      &quot;115265&quot;    &quot;90024&quot;     &quot;64116&quot;    
[4675] &quot;493856&quot;    &quot;27068&quot;     &quot;93627&quot;     &quot;166929&quot;    &quot;3033&quot;      &quot;51176&quot;    
[4681] &quot;58505&quot;     &quot;839&quot;       &quot;54433&quot;     &quot;5308&quot;      &quot;92610&quot;     &quot;51574&quot;    
[4687] &quot;287&quot;       &quot;817&quot;       &quot;79642&quot;     &quot;133022&quot;    &quot;8492&quot;      &quot;645455&quot;   
[4693] &quot;729218&quot;    &quot;9871&quot;      &quot;171024&quot;    &quot;51778&quot;     &quot;401149&quot;    &quot;308&quot;      
[4699] &quot;55212&quot;     &quot;84162&quot;     &quot;166379&quot;    &quot;2247&quot;      &quot;57182&quot;     &quot;79633&quot;    
[4705] &quot;27152&quot;     &quot;22824&quot;     &quot;256471&quot;    &quot;80167&quot;     &quot;55132&quot;     &quot;10424&quot;    
[4711] &quot;100507487&quot; &quot;132321&quot;    &quot;57575&quot;     &quot;132430&quot;    &quot;23657&quot;     &quot;25819&quot;    
[4717] &quot;1998&quot;      &quot;84709&quot;     &quot;80155&quot;     &quot;83452&quot;     &quot;80854&quot;     &quot;60592&quot;    
[4723] &quot;1047&quot;      &quot;255520&quot;    &quot;23158&quot;     &quot;27309&quot;     &quot;8821&quot;      &quot;84640&quot;    
[4729] &quot;2549&quot;      &quot;8467&quot;      &quot;646576&quot;    &quot;6059&quot;      &quot;54726&quot;     &quot;4086&quot;     
[4735] &quot;166785&quot;    &quot;1909&quot;      &quot;166614&quot;    &quot;987&quot;       &quot;10586&quot;     &quot;6189&quot;     
[4741] &quot;5188&quot;      &quot;55294&quot;     &quot;27236&quot;     &quot;6423&quot;      &quot;79884&quot;     &quot;2982&quot;     
[4747] &quot;2743&quot;      &quot;51313&quot;     &quot;55314&quot;     &quot;5481&quot;      &quot;57600&quot;     &quot;9693&quot;     
[4753] &quot;92345&quot;     &quot;4886&quot;      &quot;23520&quot;     &quot;201931&quot;    &quot;6307&quot;      &quot;1363&quot;     
[4759] &quot;91351&quot;     &quot;23022&quot;     &quot;57630&quot;     &quot;4750&quot;      &quot;1182&quot;      &quot;54969&quot;    
[4765] &quot;9848&quot;      &quot;26269&quot;     &quot;116966&quot;    &quot;140458&quot;    &quot;90768&quot;     &quot;3622&quot;     
[4771] &quot;60684&quot;     &quot;3660&quot;      &quot;836&quot;       &quot;2180&quot;      &quot;291&quot;       &quot;83891&quot;    
[4777] &quot;55805&quot;     &quot;55325&quot;     &quot;256309&quot;    &quot;8470&quot;      &quot;25854&quot;     &quot;153478&quot;   
[4783] &quot;389257&quot;    &quot;10016&quot;     &quot;116349&quot;    &quot;6550&quot;      &quot;100288152&quot; &quot;100506688&quot;
[4789] &quot;10723&quot;     &quot;101928793&quot; &quot;81037&quot;     &quot;23379&quot;     &quot;134111&quot;    &quot;54888&quot;    
[4795] &quot;6715&quot;      &quot;11044&quot;     &quot;4552&quot;      &quot;9037&quot;      &quot;100505806&quot; &quot;134145&quot;   
[4801] &quot;134147&quot;    &quot;651746&quot;    &quot;1501&quot;      &quot;1767&quot;      &quot;90268&quot;     &quot;100130744&quot;
[4807] &quot;23194&quot;     &quot;441061&quot;    &quot;54463&quot;     &quot;4651&quot;      &quot;10409&quot;     &quot;29102&quot;    
[4813] &quot;55322&quot;     &quot;23037&quot;     &quot;10923&quot;     &quot;6897&quot;      &quot;51151&quot;     &quot;23600&quot;    
[4819] &quot;26064&quot;     &quot;5810&quot;      &quot;134218&quot;    &quot;79925&quot;     &quot;92255&quot;     &quot;133686&quot;   
[4825] &quot;6507&quot;      &quot;55100&quot;     &quot;100506495&quot; &quot;1601&quot;      &quot;23548&quot;     &quot;345557&quot;   
[4831] &quot;26272&quot;     &quot;2690&quot;      &quot;6414&quot;      &quot;648987&quot;    &quot;7690&quot;      &quot;3157&quot;     
[4837] &quot;10605&quot;     &quot;23530&quot;     &quot;2255&quot;      &quot;133418&quot;    &quot;642366&quot;    &quot;4338&quot;     
[4843] &quot;10468&quot;     &quot;4724&quot;      &quot;8988&quot;      &quot;493869&quot;    &quot;54505&quot;     &quot;345651&quot;   
[4849] &quot;10769&quot;     &quot;79993&quot;     &quot;1161&quot;      &quot;57688&quot;     &quot;285671&quot;    &quot;10283&quot;    
[4855] &quot;11174&quot;     &quot;23398&quot;     &quot;54557&quot;     &quot;140890&quot;    &quot;64924&quot;     &quot;92259&quot;    
[4861] &quot;5884&quot;      &quot;153562&quot;    &quot;728492&quot;    &quot;653238&quot;    &quot;8293&quot;      &quot;23107&quot;    
[4867] &quot;79810&quot;     &quot;3842&quot;      &quot;115548&quot;    &quot;2297&quot;      &quot;57763&quot;     &quot;64283&quot;    
[4873] &quot;3074&quot;      &quot;10412&quot;     &quot;51426&quot;     &quot;134359&quot;    &quot;22987&quot;     &quot;10788&quot;    
[4879] &quot;2150&quot;      &quot;84327&quot;     &quot;728723&quot;    &quot;8622&quot;      &quot;55255&quot;     &quot;728769&quot;   
[4885] &quot;411&quot;       &quot;133746&quot;    &quot;105379046&quot; &quot;202333&quot;    &quot;256987&quot;    &quot;644936&quot;   
[4891] &quot;167555&quot;    &quot;1719&quot;      &quot;4437&quot;      &quot;5924&quot;      &quot;1160&quot;      &quot;84240&quot;    
[4897] &quot;83734&quot;     &quot;92270&quot;     &quot;153339&quot;    &quot;1462&quot;      &quot;105379054&quot; &quot;1404&quot;     
[4903] &quot;153396&quot;    &quot;100505894&quot; &quot;84059&quot;     &quot;100505994&quot; &quot;100129716&quot; &quot;84250&quot;    
[4909] &quot;79772&quot;     &quot;9652&quot;      &quot;153642&quot;    &quot;83890&quot;     &quot;22936&quot;     &quot;51752&quot;    
[4915] &quot;167410&quot;    &quot;55781&quot;     &quot;7903&quot;      &quot;83594&quot;     &quot;2241&quot;      &quot;134430&quot;   
[4921] &quot;134429&quot;    &quot;100505678&quot; &quot;9315&quot;      &quot;64097&quot;     &quot;54508&quot;     &quot;167227&quot;   
[4927] &quot;64848&quot;     &quot;55521&quot;     &quot;153733&quot;    &quot;56929&quot;     &quot;1036&quot;      &quot;9140&quot;     
[4933] &quot;1176&quot;      &quot;57556&quot;     &quot;285605&quot;    &quot;25816&quot;     &quot;3295&quot;      &quot;28966&quot;    
[4939] &quot;153241&quot;    &quot;57507&quot;     &quot;65983&quot;     &quot;51808&quot;     &quot;84466&quot;     &quot;133619&quot;   
[4945] &quot;6558&quot;      &quot;2201&quot;      &quot;28965&quot;     &quot;51015&quot;     &quot;171019&quot;    &quot;90624&quot;    
[4951] &quot;6583&quot;      &quot;441108&quot;    &quot;3659&quot;      &quot;10111&quot;     &quot;134549&quot;    &quot;2661&quot;     
[4957] &quot;116842&quot;    &quot;54819&quot;     &quot;3308&quot;      &quot;6932&quot;      &quot;6500&quot;      &quot;51265&quot;    
[4963] &quot;23338&quot;     &quot;51128&quot;     &quot;134553&quot;    &quot;9555&quot;      &quot;4090&quot;      &quot;26249&quot;    
[4969] &quot;4888&quot;      &quot;8382&quot;      &quot;2676&quot;      &quot;51307&quot;     &quot;51780&quot;     &quot;1958&quot;     
[4975] &quot;2107&quot;      &quot;3313&quot;      &quot;100505636&quot; &quot;5813&quot;      &quot;404734&quot;    &quot;10307&quot;    
[4981] &quot;113829&quot;    &quot;55374&quot;     &quot;4695&quot;      &quot;23438&quot;     &quot;153527&quot;    &quot;56146&quot;    
[4987] &quot;56145&quot;     &quot;56144&quot;     &quot;56142&quot;     &quot;56139&quot;     &quot;56138&quot;     &quot;56137&quot;    
[4993] &quot;56133&quot;     &quot;56132&quot;     &quot;56130&quot;     &quot;56129&quot;     &quot;56128&quot;     &quot;56123&quot;    
[4999] &quot;56122&quot;     &quot;6879&quot;      &quot;56103&quot;     &quot;56100&quot;     &quot;1729&quot;      &quot;285613&quot;   
[5005] &quot;64411&quot;     &quot;5097&quot;      &quot;9812&quot;      &quot;10007&quot;     &quot;80762&quot;     &quot;2246&quot;     
[5011] &quot;2908&quot;      &quot;57528&quot;     &quot;51520&quot;     &quot;10915&quot;     &quot;79628&quot;     &quot;134265&quot;   
[5017] &quot;78991&quot;     &quot;27190&quot;     &quot;389337&quot;    &quot;5145&quot;      &quot;81789&quot;     &quot;22993&quot;    
[5023] &quot;6949&quot;      &quot;972&quot;       &quot;3340&quot;      &quot;91977&quot;     &quot;55696&quot;     &quot;51164&quot;    
[5029] &quot;91975&quot;     &quot;2878&quot;      &quot;10318&quot;     &quot;309&quot;       &quot;26112&quot;     &quot;2760&quot;     
[5035] &quot;206358&quot;    &quot;475&quot;       &quot;100652758&quot; &quot;386627&quot;    &quot;9421&quot;      &quot;23367&quot;    
[5041] &quot;9337&quot;      &quot;29093&quot;     &quot;6444&quot;      &quot;9443&quot;      &quot;26999&quot;     &quot;348938&quot;   
[5047] &quot;8728&quot;      &quot;54974&quot;     &quot;153830&quot;    &quot;134510&quot;    &quot;147&quot;       &quot;63920&quot;    
[5053] &quot;10569&quot;     &quot;134492&quot;    &quot;27430&quot;     &quot;57451&quot;     &quot;23286&quot;     &quot;79646&quot;    
[5059] &quot;6586&quot;      &quot;3779&quot;      &quot;64901&quot;     &quot;8817&quot;      &quot;23291&quot;     &quot;285590&quot;   
[5065] &quot;1843&quot;      &quot;100268168&quot; &quot;153222&quot;    &quot;662&quot;       &quot;1482&quot;      &quot;4488&quot;     
[5071] &quot;3274&quot;      &quot;375484&quot;    &quot;57179&quot;     &quot;285598&quot;    &quot;54825&quot;     &quot;26262&quot;    
[5077] &quot;53917&quot;     &quot;10960&quot;     &quot;2161&quot;      &quot;2870&quot;      &quot;80758&quot;     &quot;1627&quot;     
[5083] &quot;55651&quot;     &quot;3182&quot;      &quot;91522&quot;     &quot;6940&quot;      &quot;345462&quot;    &quot;30832&quot;    
[5089] &quot;9509&quot;      &quot;3187&quot;      &quot;11282&quot;     &quot;8878&quot;      &quot;9945&quot;      &quot;57472&quot;    
[5095] &quot;4245&quot;      &quot;643836&quot;    &quot;81786&quot;     &quot;90933&quot;     &quot;10399&quot;     &quot;619571&quot;   
[5101] &quot;56940&quot;     &quot;55770&quot;     &quot;94234&quot;     &quot;2296&quot;      &quot;100508120&quot; &quot;56897&quot;    
[5107] &quot;1992&quot;      &quot;5272&quot;      &quot;401233&quot;    &quot;101927759&quot; &quot;8737&quot;      &quot;63027&quot;    
[5113] &quot;221749&quot;    &quot;26240&quot;     &quot;8899&quot;      &quot;10455&quot;     &quot;9425&quot;      &quot;10799&quot;    
[5119] &quot;648791&quot;    &quot;6745&quot;      &quot;154007&quot;    &quot;9521&quot;      &quot;51000&quot;     &quot;2651&quot;     
[5125] &quot;4117&quot;      &quot;3096&quot;      &quot;1906&quot;      &quot;221692&quot;    &quot;10048&quot;     &quot;221687&quot;   
[5131] &quot;9308&quot;      &quot;29116&quot;     &quot;101928433&quot; &quot;401236&quot;    &quot;10486&quot;     &quot;51439&quot;    
[5137] &quot;63971&quot;     &quot;255488&quot;    &quot;3400&quot;      &quot;1871&quot;      &quot;54901&quot;     &quot;6659&quot;     
[5143] &quot;57380&quot;     &quot;55856&quot;     &quot;81688&quot;     &quot;9750&quot;      &quot;8418&quot;      &quot;3006&quot;     
[5149] &quot;8347&quot;      &quot;8334&quot;      &quot;3017&quot;      &quot;8367&quot;      &quot;8339&quot;      &quot;3012&quot;     
[5155] &quot;8365&quot;      &quot;11118&quot;     &quot;10385&quot;     &quot;11119&quot;     &quot;493812&quot;    &quot;10473&quot;    
[5161] &quot;8970&quot;      &quot;8969&quot;      &quot;85236&quot;     &quot;7754&quot;      &quot;346157&quot;    &quot;7738&quot;     
[5167] &quot;8330&quot;      &quot;8341&quot;      &quot;7718&quot;      &quot;7746&quot;      &quot;222699&quot;    &quot;387032&quot;   
[5173] &quot;7741&quot;      &quot;84547&quot;     &quot;80317&quot;     &quot;9753&quot;      &quot;222696&quot;    &quot;2550&quot;     
[5179] &quot;4340&quot;      &quot;346171&quot;    &quot;3134&quot;      &quot;3137&quot;      &quot;7726&quot;      &quot;56658&quot;    
[5185] &quot;79897&quot;     &quot;3133&quot;      &quot;2794&quot;      &quot;80742&quot;     &quot;28973&quot;     &quot;221545&quot;   
[5191] &quot;203068&quot;    &quot;780&quot;       &quot;3107&quot;      &quot;3106&quot;      &quot;534&quot;       &quot;4795&quot;     
[5197] &quot;7916&quot;      &quot;55937&quot;     &quot;57827&quot;     &quot;1460&quot;      &quot;7920&quot;      &quot;23564&quot;    
[5203] &quot;1192&quot;      &quot;3304&quot;      &quot;4758&quot;      &quot;10919&quot;     &quot;221527&quot;    &quot;63943&quot;    
[5209] &quot;9374&quot;      &quot;5089&quot;      &quot;4855&quot;      &quot;6891&quot;      &quot;5696&quot;      &quot;6890&quot;     
[5215] &quot;100294145&quot; &quot;6046&quot;      &quot;3115&quot;      &quot;7922&quot;      &quot;9277&quot;      &quot;10471&quot;    
[5221] &quot;5863&quot;      &quot;8831&quot;      &quot;3710&quot;      &quot;84300&quot;     &quot;3159&quot;      &quot;221491&quot;   
[5227] &quot;6204&quot;      &quot;29993&quot;     &quot;64771&quot;     &quot;54887&quot;     &quot;6882&quot;      &quot;2178&quot;     
[5233] &quot;2289&quot;      &quot;6732&quot;      &quot;1432&quot;      &quot;5603&quot;      &quot;27154&quot;     &quot;222658&quot;   
[5239] &quot;11329&quot;     &quot;6428&quot;      &quot;101154753&quot; &quot;105375039&quot; &quot;51645&quot;     &quot;23787&quot;    
[5245] &quot;5292&quot;      &quot;221468&quot;    &quot;55633&quot;     &quot;266727&quot;    &quot;55776&quot;     &quot;23500&quot;    
[5251] &quot;4337&quot;      &quot;222643&quot;    &quot;221443&quot;    &quot;116113&quot;    &quot;29964&quot;     &quot;100188893&quot;
[5257] &quot;25862&quot;     &quot;705&quot;       &quot;896&quot;       &quot;647024&quot;    &quot;2979&quot;      &quot;5961&quot;     
[5263] &quot;6903&quot;      &quot;441150&quot;    &quot;27232&quot;     &quot;5190&quot;      &quot;116138&quot;    &quot;88745&quot;    
[5269] &quot;10591&quot;     &quot;65989&quot;     &quot;93643&quot;     &quot;25844&quot;     &quot;9533&quot;      &quot;5429&quot;     
[5275] &quot;54676&quot;     &quot;55362&quot;     &quot;2030&quot;      &quot;347734&quot;    &quot;4794&quot;      &quot;202500&quot;   
[5281] &quot;57505&quot;     &quot;22875&quot;     &quot;59084&quot;     &quot;10231&quot;     &quot;221395&quot;    &quot;27242&quot;    
[5287] &quot;23607&quot;     &quot;221391&quot;    &quot;4594&quot;      &quot;401264&quot;    &quot;730101&quot;    &quot;28978&quot;    
[5293] &quot;2941&quot;      &quot;2729&quot;      &quot;401265&quot;    &quot;90523&quot;     &quot;222584&quot;    &quot;54511&quot;    
[5299] &quot;653&quot;       &quot;81578&quot;     &quot;667&quot;       &quot;26036&quot;     &quot;51715&quot;     &quot;51557&quot;    
[5305] &quot;7803&quot;      &quot;55788&quot;     &quot;79627&quot;     &quot;22999&quot;     &quot;56479&quot;     &quot;135228&quot;   
[5311] &quot;27145&quot;     &quot;4646&quot;      &quot;134728&quot;    &quot;55023&quot;     &quot;9324&quot;      &quot;100288198&quot;
[5317] &quot;167691&quot;    &quot;6785&quot;      &quot;55603&quot;     &quot;25998&quot;     &quot;7162&quot;      &quot;90025&quot;    
[5323] &quot;23033&quot;     &quot;167681&quot;    &quot;112609&quot;    &quot;57231&quot;     &quot;23036&quot;     &quot;57150&quot;    
[5329] &quot;206412&quot;    &quot;63914&quot;     &quot;10559&quot;     &quot;23595&quot;     &quot;135293&quot;    &quot;51465&quot;    
[5335] &quot;58528&quot;     &quot;22881&quot;     &quot;6885&quot;      &quot;2045&quot;      &quot;79694&quot;     &quot;23376&quot;    
[5341] &quot;84553&quot;     &quot;101927365&quot; &quot;100130890&quot; &quot;2898&quot;      &quot;11149&quot;     &quot;154442&quot;   
[5347] &quot;64208&quot;     &quot;5550&quot;      &quot;84816&quot;     &quot;55278&quot;     &quot;51250&quot;     &quot;11231&quot;    
[5353] &quot;28962&quot;     &quot;2309&quot;      &quot;285753&quot;    &quot;8763&quot;      &quot;8936&quot;      &quot;51362&quot;    
[5359] &quot;23097&quot;     &quot;262&quot;       &quot;117247&quot;    &quot;5980&quot;      &quot;643749&quot;    &quot;51175&quot;    
[5365] &quot;619208&quot;    &quot;4082&quot;      &quot;3066&quot;      &quot;101927768&quot; &quot;222537&quot;    &quot;254228&quot;   
[5371] &quot;222553&quot;    &quot;387119&quot;    &quot;5350&quot;      &quot;79632&quot;     &quot;4121&quot;      &quot;221322&quot;   
[5377] &quot;2697&quot;      &quot;5570&quot;      &quot;10345&quot;     &quot;101927990&quot; &quot;7955&quot;      &quot;154214&quot;   
[5383] &quot;51020&quot;     &quot;23493&quot;     &quot;60487&quot;     &quot;81847&quot;     &quot;84456&quot;     &quot;114801&quot;   
[5389] &quot;5167&quot;      &quot;1490&quot;      &quot;116843&quot;    &quot;9519&quot;      &quot;6446&quot;      &quot;4217&quot;     
[5395] &quot;5191&quot;      &quot;64065&quot;     &quot;57221&quot;     &quot;57224&quot;     &quot;441172&quot;    &quot;345930&quot;   
[5401] &quot;85021&quot;     &quot;58527&quot;     &quot;51696&quot;     &quot;167838&quot;    &quot;10370&quot;     &quot;51534&quot;    
[5407] &quot;3097&quot;      &quot;51390&quot;     &quot;134637&quot;    &quot;8504&quot;      &quot;2519&quot;      &quot;9749&quot;     
[5413] &quot;7402&quot;      &quot;257218&quot;    &quot;10981&quot;     &quot;729176&quot;    &quot;134957&quot;    &quot;389432&quot;   
[5419] &quot;23328&quot;     &quot;80328&quot;     &quot;80329&quot;     &quot;9590&quot;      &quot;57621&quot;     &quot;79624&quot;    
[5425] &quot;54516&quot;     &quot;26034&quot;     &quot;154043&quot;    &quot;26230&quot;     &quot;51106&quot;     &quot;729515&quot;   
[5431] &quot;404672&quot;    &quot;57583&quot;     &quot;7430&quot;      &quot;9589&quot;      &quot;29074&quot;     &quot;3482&quot;     
[5437] &quot;729603&quot;    &quot;56895&quot;     &quot;5071&quot;      &quot;135138&quot;    &quot;9444&quot;      &quot;10846&quot;    
[5443] &quot;100289495&quot; &quot;51660&quot;     &quot;11116&quot;     &quot;253769&quot;    &quot;387263&quot;    &quot;55274&quot;    
[5449] &quot;28514&quot;     &quot;5689&quot;      &quot;6908&quot;      &quot;441307&quot;    &quot;5575&quot;      &quot;23353&quot;    
[5455] &quot;102723743&quot; &quot;11033&quot;     &quot;115330&quot;    &quot;2852&quot;      &quot;90637&quot;     &quot;101927021&quot;
[5461] &quot;79778&quot;     &quot;202915&quot;    &quot;84262&quot;     &quot;392617&quot;    &quot;29960&quot;     &quot;8662&quot;     
[5467] &quot;3955&quot;      &quot;84433&quot;     &quot;9907&quot;      &quot;55698&quot;     &quot;441191&quot;    &quot;26100&quot;    
[5473] &quot;80028&quot;     &quot;60&quot;        &quot;6624&quot;      &quot;54476&quot;     &quot;9265&quot;      &quot;84792&quot;    
[5479] &quot;221955&quot;    &quot;11014&quot;     &quot;55146&quot;     &quot;54753&quot;     &quot;100131017&quot; &quot;221960&quot;   
[5485] &quot;6119&quot;      &quot;3382&quot;      &quot;100505938&quot; &quot;9678&quot;      &quot;221981&quot;    &quot;54664&quot;    
[5491] &quot;10124&quot;     &quot;1607&quot;      &quot;57037&quot;     &quot;196&quot;       &quot;23161&quot;     &quot;9734&quot;     
[5497] &quot;7291&quot;      &quot;221830&quot;    &quot;3696&quot;      &quot;55536&quot;     &quot;11097&quot;     &quot;10457&quot;    
[5503] &quot;10643&quot;     &quot;340277&quot;    &quot;4852&quot;      &quot;1687&quot;      &quot;26031&quot;     &quot;3181&quot;     
[5509] &quot;8887&quot;      &quot;221895&quot;    &quot;9865&quot;      &quot;222171&quot;    &quot;644150&quot;    &quot;84725&quot;    
[5515] &quot;10392&quot;     &quot;79017&quot;     &quot;401320&quot;    &quot;2617&quot;      &quot;358&quot;       &quot;5137&quot;     
[5521] &quot;401321&quot;    &quot;441212&quot;    &quot;6100&quot;      &quot;27241&quot;     &quot;168667&quot;    &quot;23333&quot;    
[5527] &quot;554236&quot;    &quot;57057&quot;     &quot;64224&quot;     &quot;101930085&quot; &quot;100506725&quot; &quot;80820&quot;    
[5533] &quot;9844&quot;      &quot;6424&quot;      &quot;83930&quot;     &quot;27072&quot;     &quot;5898&quot;      &quot;136647&quot;   
[5539] &quot;79783&quot;     &quot;3624&quot;      &quot;5683&quot;      &quot;9263&quot;      &quot;55744&quot;     &quot;55665&quot;    
[5545] &quot;255031&quot;    &quot;5224&quot;      &quot;27434&quot;     &quot;165&quot;       &quot;5425&quot;      &quot;58498&quot;    
[5551] &quot;2645&quot;      &quot;222068&quot;    &quot;5478&quot;      &quot;83605&quot;     &quot;23148&quot;     &quot;107&quot;      
[5557] &quot;641977&quot;    &quot;3486&quot;      &quot;3364&quot;      &quot;23242&quot;     &quot;55915&quot;     &quot;81552&quot;    
[5563] &quot;349075&quot;    &quot;51373&quot;     &quot;908&quot;       &quot;25870&quot;     &quot;51427&quot;     &quot;643180&quot;   
[5569] &quot;644619&quot;    &quot;643253&quot;    &quot;154807&quot;    &quot;435&quot;       &quot;8460&quot;      &quot;100996437&quot;
[5575] &quot;27342&quot;     &quot;51119&quot;     &quot;55253&quot;     &quot;5382&quot;      &quot;26053&quot;     &quot;64409&quot;    
[5581] &quot;155370&quot;    &quot;9883&quot;      &quot;378108&quot;    &quot;100101148&quot; &quot;100093631&quot; &quot;8326&quot;     
[5587] &quot;9031&quot;      &quot;26608&quot;     &quot;155382&quot;    &quot;114049&quot;    &quot;7462&quot;      &quot;7461&quot;     
[5593] &quot;9569&quot;      &quot;2969&quot;      &quot;442582&quot;    &quot;81554&quot;     &quot;2970&quot;      &quot;541473&quot;   
[5599] &quot;375593&quot;    &quot;100101267&quot; &quot;442590&quot;    &quot;3092&quot;      &quot;57414&quot;     &quot;5447&quot;     
[5605] &quot;222183&quot;    &quot;136853&quot;    &quot;7784&quot;      &quot;100133091&quot; &quot;22932&quot;     &quot;10875&quot;    
[5611] &quot;100505854&quot; &quot;222194&quot;    &quot;85025&quot;     &quot;100505881&quot; &quot;2770&quot;      &quot;948&quot;      
[5617] &quot;27445&quot;     &quot;10371&quot;     &quot;79161&quot;     &quot;54677&quot;     &quot;5243&quot;      &quot;55972&quot;    
[5623] &quot;6717&quot;      &quot;9069&quot;      &quot;5218&quot;      &quot;7978&quot;      &quot;10142&quot;     &quot;1595&quot;     
[5629] &quot;57798&quot;     &quot;5189&quot;      &quot;84060&quot;     &quot;55610&quot;     &quot;7980&quot;      &quot;2791&quot;     
[5635] &quot;8910&quot;      &quot;55607&quot;     &quot;5166&quot;      &quot;1780&quot;      &quot;440&quot;       &quot;55971&quot;    
[5641] &quot;10095&quot;     &quot;10898&quot;     &quot;84124&quot;     &quot;23660&quot;     &quot;221786&quot;    &quot;255374&quot;   
[5647] &quot;55262&quot;     &quot;79690&quot;     &quot;10734&quot;     &quot;352954&quot;    &quot;4034&quot;      &quot;64598&quot;    
[5653] &quot;51593&quot;     &quot;402682&quot;    &quot;43&quot;        &quot;1174&quot;      &quot;8985&quot;      &quot;10467&quot;    
[5659] &quot;24146&quot;     &quot;64792&quot;     &quot;136227&quot;    &quot;10603&quot;     &quot;54784&quot;     &quot;222229&quot;   
[5665] &quot;222234&quot;    &quot;222235&quot;    &quot;10234&quot;     &quot;83787&quot;     &quot;222236&quot;    &quot;9512&quot;     
[5671] &quot;375611&quot;    &quot;100216546&quot; &quot;100216545&quot; &quot;6733&quot;      &quot;60561&quot;     &quot;6856&quot;     
[5677] &quot;10135&quot;     &quot;168455&quot;    &quot;26959&quot;     &quot;10466&quot;     &quot;2845&quot;      &quot;11062&quot;    
[5683] &quot;55973&quot;     &quot;1738&quot;      &quot;3912&quot;      &quot;4897&quot;      &quot;50640&quot;     &quot;3475&quot;     
[5689] &quot;286006&quot;    &quot;64418&quot;     &quot;154743&quot;    &quot;101928036&quot; &quot;54329&quot;     &quot;5506&quot;     
[5695] &quot;29969&quot;     &quot;4233&quot;      &quot;7982&quot;      &quot;83992&quot;     &quot;3751&quot;      &quot;10447&quot;    
[5701] &quot;5803&quot;      &quot;10157&quot;     &quot;442721&quot;    &quot;2861&quot;      &quot;25913&quot;     &quot;2918&quot;     
[5707] &quot;79571&quot;     &quot;381&quot;       &quot;27044&quot;     &quot;64101&quot;     &quot;157247&quot;    &quot;55131&quot;    
[5713] &quot;29923&quot;     &quot;64753&quot;     &quot;9296&quot;      &quot;100130705&quot; &quot;23382&quot;     &quot;23008&quot;    
[5719] &quot;84928&quot;     &quot;378805&quot;    &quot;4289&quot;      &quot;100506881&quot; &quot;91584&quot;     &quot;54927&quot;    
[5725] &quot;60412&quot;     &quot;84912&quot;     &quot;669&quot;       &quot;55281&quot;     &quot;29062&quot;     &quot;23165&quot;    
[5731] &quot;1129&quot;      &quot;56829&quot;     &quot;79989&quot;     &quot;254048&quot;    &quot;80853&quot;     &quot;100134229&quot;
[5737] &quot;23608&quot;     &quot;100134713&quot; &quot;100507421&quot; &quot;55750&quot;     &quot;2051&quot;      &quot;373156&quot;   
[5743] &quot;9715&quot;      &quot;285966&quot;    &quot;9747&quot;      &quot;27010&quot;     &quot;136051&quot;    &quot;155054&quot;   
[5749] &quot;7988&quot;      &quot;100289678&quot; &quot;27153&quot;     &quot;84626&quot;     &quot;168544&quot;    &quot;643641&quot;   
[5755] &quot;401431&quot;    &quot;3757&quot;      &quot;4846&quot;      &quot;9311&quot;      &quot;1020&quot;      &quot;6522&quot;     
[5761] &quot;10922&quot;     &quot;116988&quot;    &quot;10061&quot;     &quot;6604&quot;      &quot;6009&quot;      &quot;51422&quot;    
[5767] &quot;57180&quot;     &quot;22976&quot;     &quot;155435&quot;    &quot;64327&quot;     &quot;64434&quot;     &quot;57488&quot;    
[5773] &quot;55112&quot;     &quot;169270&quot;    &quot;26260&quot;     &quot;100287015&quot; &quot;286042&quot;    &quot;157285&quot;   
[5779] &quot;137075&quot;    &quot;9258&quot;      &quot;79660&quot;     &quot;83648&quot;     &quot;640&quot;       &quot;101929490&quot;
[5785] &quot;252969&quot;    &quot;2222&quot;      &quot;1508&quot;      &quot;85002&quot;     &quot;91694&quot;     &quot;57604&quot;    
[5791] &quot;10395&quot;     &quot;157773&quot;    &quot;286097&quot;    &quot;51201&quot;     &quot;137492&quot;    &quot;9108&quot;     
[5797] &quot;6542&quot;      &quot;9&quot;         &quot;23362&quot;     &quot;55790&quot;     &quot;4023&quot;      &quot;526&quot;      
[5803] &quot;23039&quot;     &quot;2039&quot;      &quot;64760&quot;     &quot;79873&quot;     &quot;55806&quot;     &quot;23516&quot;    
[5809] &quot;10174&quot;     &quot;64236&quot;     &quot;57805&quot;     &quot;55909&quot;     &quot;1960&quot;      &quot;8793&quot;     
[5815] &quot;8797&quot;      &quot;91782&quot;     &quot;203069&quot;    &quot;4017&quot;      &quot;4824&quot;      &quot;137814&quot;   
[5821] &quot;4741&quot;      &quot;4747&quot;      &quot;2796&quot;      &quot;23087&quot;     &quot;2185&quot;      &quot;2053&quot;     
[5827] &quot;1191&quot;      &quot;55893&quot;     &quot;157574&quot;    &quot;79618&quot;     &quot;23303&quot;     &quot;1846&quot;     
[5833] &quot;23484&quot;     &quot;11030&quot;     &quot;2936&quot;      &quot;7993&quot;      &quot;5516&quot;      &quot;7486&quot;     
[5839] &quot;3084&quot;      &quot;84750&quot;     &quot;84549&quot;     &quot;80185&quot;     &quot;78986&quot;     &quot;137970&quot;   
[5845] &quot;80139&quot;     &quot;1978&quot;      &quot;23259&quot;     &quot;137994&quot;    &quot;6867&quot;      &quot;6422&quot;     
[5851] &quot;137964&quot;    &quot;286&quot;       &quot;7994&quot;      &quot;5327&quot;      &quot;3551&quot;      &quot;6575&quot;     
[5857] &quot;114926&quot;    &quot;55145&quot;     &quot;84376&quot;     &quot;23514&quot;     &quot;1052&quot;      &quot;115294&quot;   
[5863] &quot;9821&quot;      &quot;51606&quot;     &quot;8601&quot;      &quot;6917&quot;      &quot;29088&quot;     &quot;137695&quot;   
[5869] &quot;6224&quot;      &quot;6386&quot;      &quot;767&quot;       &quot;253943&quot;    &quot;5150&quot;      &quot;23212&quot;    
[5875] &quot;80124&quot;     &quot;81796&quot;     &quot;23471&quot;     &quot;51110&quot;     &quot;7013&quot;      &quot;157506&quot;   
[5881] &quot;6921&quot;      &quot;5828&quot;      &quot;3574&quot;      &quot;7163&quot;      &quot;65986&quot;     &quot;55824&quot;    
[5887] &quot;2171&quot;      &quot;79752&quot;     &quot;64089&quot;     &quot;377677&quot;    &quot;760&quot;       &quot;4325&quot;     
[5893] &quot;101929709&quot; &quot;734&quot;       &quot;4683&quot;      &quot;793&quot;       &quot;169200&quot;    &quot;64168&quot;    
[5899] &quot;100127983&quot; &quot;100506365&quot; &quot;51633&quot;     &quot;389677&quot;    &quot;91147&quot;     &quot;54704&quot;    
[5905] &quot;54845&quot;     &quot;137682&quot;    &quot;94241&quot;     &quot;79666&quot;     &quot;157657&quot;    &quot;392255&quot;   
[5911] &quot;51001&quot;     &quot;9791&quot;      &quot;85453&quot;     &quot;92140&quot;     &quot;203111&quot;    &quot;79815&quot;    
[5917] &quot;157680&quot;    &quot;5440&quot;      &quot;6674&quot;      &quot;157567&quot;    &quot;7534&quot;      &quot;51123&quot;    
[5923] &quot;83988&quot;     &quot;51366&quot;     &quot;7071&quot;      &quot;51582&quot;     &quot;100506753&quot; &quot;25879&quot;    
[5929] &quot;9699&quot;      &quot;284&quot;       &quot;9694&quot;      &quot;157753&quot;    &quot;84955&quot;     &quot;56943&quot;    
[5935] &quot;93035&quot;     &quot;27012&quot;     &quot;114788&quot;    &quot;8667&quot;      &quot;84294&quot;     &quot;2131&quot;     
[5941] &quot;401474&quot;    &quot;4982&quot;      &quot;114569&quot;    &quot;4856&quot;      &quot;5168&quot;      &quot;64798&quot;    
[5947] &quot;6641&quot;      &quot;3037&quot;      &quot;79139&quot;     &quot;84933&quot;     &quot;114907&quot;    &quot;157769&quot;   
[5953] &quot;9788&quot;      &quot;6713&quot;      &quot;5820&quot;      &quot;50807&quot;     &quot;114&quot;       &quot;3786&quot;     
[5959] &quot;137835&quot;    &quot;7038&quot;      &quot;10397&quot;     &quot;10656&quot;     &quot;83696&quot;     &quot;57210&quot;    
[5965] &quot;2843&quot;      &quot;51337&quot;     &quot;66004&quot;     &quot;79943&quot;     &quot;642475&quot;    &quot;1936&quot;     
[5971] &quot;65263&quot;     &quot;7264&quot;      &quot;9831&quot;      &quot;225689&quot;    &quot;286077&quot;    &quot;23513&quot;    
[5977] &quot;340371&quot;    &quot;5339&quot;      &quot;84875&quot;     &quot;26873&quot;     &quot;54512&quot;     &quot;8733&quot;     
[5983] &quot;1537&quot;      &quot;84232&quot;     &quot;340390&quot;    &quot;8694&quot;      &quot;203054&quot;    &quot;51160&quot;    
[5989] &quot;4796&quot;      &quot;100287098&quot; &quot;50626&quot;     &quot;90990&quot;     &quot;414919&quot;    &quot;80778&quot;    
[5995] &quot;340385&quot;    &quot;65265&quot;     &quot;81704&quot;     &quot;23189&quot;     &quot;6595&quot;      &quot;7436&quot;     
[6001] &quot;5991&quot;      &quot;6505&quot;      &quot;55064&quot;     &quot;403313&quot;    &quot;55664&quot;     &quot;50808&quot;    
[6007] &quot;57589&quot;     &quot;158358&quot;    &quot;2731&quot;      &quot;5789&quot;      &quot;101929407&quot; &quot;340481&quot;   
[6013] &quot;158326&quot;    &quot;158219&quot;    &quot;203238&quot;    &quot;54796&quot;     &quot;54875&quot;     &quot;6456&quot;     
[6019] &quot;92949&quot;     &quot;10670&quot;     &quot;123&quot;       &quot;340485&quot;    &quot;4300&quot;      &quot;54914&quot;    
[6025] &quot;55958&quot;     &quot;4507&quot;      &quot;63951&quot;     &quot;7010&quot;      &quot;79817&quot;     &quot;158038&quot;   
[6031] &quot;100129250&quot; &quot;55234&quot;     &quot;2683&quot;      &quot;573&quot;       &quot;51510&quot;     &quot;360&quot;      
[6037] &quot;441394&quot;    &quot;11191&quot;     &quot;55833&quot;     &quot;318&quot;       &quot;203259&quot;    &quot;1271&quot;     
[6043] &quot;2592&quot;      &quot;84720&quot;     &quot;10497&quot;     &quot;158381&quot;    &quot;9853&quot;      &quot;7016&quot;     
[6049] &quot;6023&quot;      &quot;203260&quot;    &quot;7094&quot;      &quot;10488&quot;     &quot;57704&quot;     &quot;84681&quot;    
[6055] &quot;51754&quot;     &quot;92973&quot;     &quot;8434&quot;      &quot;152007&quot;    &quot;100506710&quot; &quot;26267&quot;    
[6061] &quot;401505&quot;    &quot;22844&quot;     &quot;6461&quot;      &quot;219&quot;       &quot;347252&quot;    &quot;101929959&quot;
[6067] &quot;5239&quot;      &quot;169693&quot;    &quot;8395&quot;      &quot;2395&quot;      &quot;9414&quot;      &quot;100885775&quot;
[6073] &quot;9413&quot;      &quot;375743&quot;    &quot;23670&quot;     &quot;51104&quot;     &quot;138241&quot;    &quot;55071&quot;    
[6079] &quot;5125&quot;      &quot;55312&quot;     &quot;2776&quot;      &quot;84131&quot;     &quot;7091&quot;      &quot;7088&quot;     
[6085] &quot;158158&quot;    &quot;102723989&quot; &quot;257019&quot;    &quot;105376114&quot; &quot;64078&quot;     &quot;4915&quot;     
[6091] &quot;389765&quot;    &quot;1514&quot;      &quot;79048&quot;     &quot;54769&quot;     &quot;4783&quot;      &quot;4920&quot;     
[6097] &quot;10558&quot;     &quot;54829&quot;     &quot;64768&quot;     &quot;23299&quot;     &quot;138649&quot;    &quot;83744&quot;    
[6103] &quot;84270&quot;     &quot;4814&quot;      &quot;445577&quot;    &quot;23196&quot;     &quot;158257&quot;    &quot;169841&quot;   
[6109] &quot;101928099&quot; &quot;375748&quot;    &quot;11046&quot;     &quot;22927&quot;     &quot;8555&quot;      &quot;1515&quot;     
[6115] &quot;7111&quot;      &quot;9830&quot;      &quot;7464&quot;      &quot;55357&quot;     &quot;8013&quot;      &quot;23071&quot;    
[6121] &quot;54881&quot;     &quot;347273&quot;    &quot;19&quot;        &quot;23446&quot;     &quot;26134&quot;     &quot;2218&quot;     
[6127] &quot;105376204&quot; &quot;9314&quot;      &quot;8518&quot;      &quot;54942&quot;     &quot;8727&quot;      &quot;54566&quot;    
[6133] &quot;5774&quot;      &quot;114299&quot;    &quot;445815&quot;    &quot;7295&quot;      &quot;548645&quot;    &quot;2790&quot;     
[6139] &quot;9991&quot;      &quot;84263&quot;     &quot;58493&quot;     &quot;401548&quot;    &quot;169834&quot;    &quot;7539&quot;     
[6145] &quot;81932&quot;     &quot;210&quot;       &quot;54107&quot;     &quot;85301&quot;     &quot;25861&quot;     &quot;9550&quot;     
[6151] &quot;23245&quot;     &quot;7099&quot;      &quot;1620&quot;      &quot;55755&quot;     &quot;2934&quot;      &quot;2040&quot;     
[6157] &quot;4702&quot;      &quot;92399&quot;     &quot;5742&quot;      &quot;286204&quot;    &quot;100129034&quot; &quot;169611&quot;   
[6163] &quot;81873&quot;     &quot;286205&quot;    &quot;3309&quot;      &quot;23099&quot;     &quot;403341&quot;    &quot;9649&quot;     
[6169] &quot;23452&quot;     &quot;29988&quot;     &quot;7633&quot;      &quot;6136&quot;      &quot;90678&quot;     &quot;6812&quot;     
[6175] &quot;1025&quot;      &quot;2022&quot;      &quot;27090&quot;     &quot;399665&quot;    &quot;203245&quot;    &quot;114789&quot;   
[6181] &quot;80142&quot;     &quot;79095&quot;     &quot;2801&quot;      &quot;2733&quot;      &quot;6709&quot;      &quot;84885&quot;    
[6187] &quot;10444&quot;     &quot;54662&quot;     &quot;254295&quot;    &quot;56904&quot;     &quot;84895&quot;     &quot;1384&quot;     
[6193] &quot;5524&quot;      &quot;389792&quot;    &quot;100506119&quot; &quot;100506190&quot; &quot;28989&quot;     &quot;140459&quot;   
[6199] &quot;27348&quot;     &quot;1861&quot;      &quot;10868&quot;     &quot;23413&quot;     &quot;445&quot;       &quot;100272217&quot;
[6205] &quot;8939&quot;      &quot;23404&quot;     &quot;84814&quot;     &quot;10585&quot;     &quot;23064&quot;     &quot;7270&quot;     
[6211] &quot;9329&quot;      &quot;9328&quot;      &quot;5900&quot;      &quot;26301&quot;     &quot;6837&quot;      &quot;6834&quot;     
[6217] &quot;11182&quot;     &quot;9719&quot;      &quot;266655&quot;    &quot;1289&quot;      &quot;100506599&quot; &quot;51116&quot;    
[6223] &quot;138151&quot;    &quot;169714&quot;    &quot;26086&quot;     &quot;64170&quot;     &quot;10807&quot;     &quot;23203&quot;    
[6229] &quot;158055&quot;    &quot;4851&quot;      &quot;101928483&quot; &quot;138311&quot;    &quot;85014&quot;     &quot;84960&quot;    
[6235] &quot;29085&quot;     &quot;7186&quot;      &quot;54461&quot;     &quot;286257&quot;    &quot;100289341&quot; &quot;8636&quot;     
[6241] &quot;27158&quot;     &quot;26012&quot;     &quot;64975&quot;     &quot;92715&quot;     &quot;92714&quot;     &quot;55344&quot;    
[6247] &quot;283981&quot;    &quot;28227&quot;     &quot;8227&quot;      &quot;207063&quot;    &quot;8908&quot;      &quot;25878&quot;    
[6253] &quot;5613&quot;      &quot;389906&quot;    &quot;412&quot;       &quot;6907&quot;      &quot;4935&quot;      &quot;55841&quot;    
[6259] &quot;4281&quot;      &quot;5634&quot;      &quot;140456&quot;    &quot;5277&quot;      &quot;8544&quot;      &quot;57393&quot;    
[6265] &quot;56474&quot;     &quot;94056&quot;     &quot;4810&quot;      &quot;6322&quot;      &quot;10742&quot;     &quot;10149&quot;    
[6271] &quot;22866&quot;     &quot;23676&quot;     &quot;51360&quot;     &quot;23597&quot;     &quot;6303&quot;      &quot;79135&quot;    
[6277] &quot;9468&quot;      &quot;11141&quot;     &quot;257397&quot;    &quot;1756&quot;      &quot;83604&quot;     &quot;6990&quot;     
[6283] &quot;6103&quot;      &quot;10159&quot;     &quot;8573&quot;      &quot;80258&quot;     &quot;56548&quot;     &quot;6102&quot;     
[6289] &quot;9767&quot;      &quot;9104&quot;      &quot;54539&quot;     &quot;8241&quot;      &quot;8237&quot;      &quot;7592&quot;     
[6295] &quot;369&quot;       &quot;10682&quot;     &quot;4943&quot;      &quot;64743&quot;     &quot;10245&quot;     &quot;10084&quot;    
[6301] &quot;55593&quot;     &quot;7030&quot;      &quot;79917&quot;     &quot;5355&quot;      &quot;89801&quot;     &quot;389856&quot;   
[6307] &quot;139189&quot;    &quot;57477&quot;     &quot;170685&quot;    &quot;55190&quot;     &quot;23708&quot;     &quot;81557&quot;    
[6313] &quot;728239&quot;    &quot;727866&quot;    &quot;29057&quot;     &quot;54328&quot;     &quot;64061&quot;     &quot;23133&quot;    
[6319] &quot;54954&quot;     &quot;65267&quot;     &quot;90121&quot;     &quot;7216&quot;      &quot;27301&quot;     &quot;100463488&quot;
[6325] &quot;28986&quot;     &quot;10325&quot;     &quot;29978&quot;     &quot;169981&quot;    &quot;92249&quot;     &quot;23229&quot;    
[6331] &quot;139285&quot;    &quot;55906&quot;     &quot;83957&quot;     &quot;4478&quot;      &quot;367&quot;       &quot;4983&quot;     
[6337] &quot;1947&quot;      &quot;1896&quot;      &quot;3476&quot;      &quot;54857&quot;     &quot;105373244&quot; &quot;29934&quot;    
[6343] &quot;9968&quot;      &quot;9203&quot;      &quot;4841&quot;      &quot;26548&quot;     &quot;340526&quot;    &quot;55869&quot;    
[6349] &quot;139201&quot;    &quot;53344&quot;     &quot;6567&quot;      &quot;22&quot;        &quot;139596&quot;    &quot;646127&quot;   
[6355] &quot;158866&quot;    &quot;538&quot;       &quot;9452&quot;      &quot;27330&quot;     &quot;140886&quot;    &quot;57526&quot;    
[6361] &quot;94121&quot;     &quot;1478&quot;      &quot;27035&quot;     &quot;79979&quot;     &quot;1821&quot;      &quot;54457&quot;    
[6367] &quot;1678&quot;      &quot;6173&quot;      &quot;2717&quot;      &quot;100131755&quot; &quot;51309&quot;     &quot;54470&quot;    
[6373] &quot;9823&quot;      &quot;64860&quot;     &quot;80823&quot;     &quot;56271&quot;     &quot;84707&quot;     &quot;9643&quot;     
[6379] &quot;51209&quot;     &quot;286437&quot;    &quot;139221&quot;    &quot;55086&quot;     &quot;54885&quot;     &quot;79710&quot;    
[6385] &quot;54830&quot;     &quot;84443&quot;     &quot;115201&quot;    &quot;55916&quot;     &quot;105373311&quot; &quot;9949&quot;     
[6391] &quot;90293&quot;     &quot;3597&quot;      &quot;170261&quot;    &quot;10857&quot;     &quot;203427&quot;    &quot;7319&quot;     
[6397] &quot;2747&quot;      &quot;2892&quot;      &quot;57187&quot;     &quot;8862&quot;      &quot;51114&quot;     &quot;2000&quot;     
[6403] &quot;9131&quot;      &quot;9016&quot;      &quot;51634&quot;     &quot;55855&quot;     &quot;286467&quot;    &quot;55796&quot;    
[6409] &quot;2239&quot;      &quot;347475&quot;    &quot;84295&quot;     &quot;159090&quot;    &quot;159091&quot;    &quot;56180&quot;    
[6415] &quot;644538&quot;    &quot;203523&quot;    &quot;399668&quot;    &quot;2273&quot;      &quot;27316&quot;     &quot;100129662&quot;
[6421] &quot;4168&quot;      &quot;2332&quot;      &quot;3423&quot;      &quot;100130086&quot; &quot;84548&quot;     &quot;100506164&quot;
[6427] &quot;541578&quot;    &quot;10046&quot;     &quot;4534&quot;      &quot;8776&quot;      &quot;2564&quot;      &quot;50814&quot;    
[6433] &quot;7739&quot;      &quot;84968&quot;     &quot;10838&quot;     &quot;633&quot;       &quot;6535&quot;      &quot;10134&quot;    
[6439] &quot;26576&quot;     &quot;6748&quot;      &quot;57595&quot;     &quot;5973&quot;      &quot;6901&quot;      &quot;8266&quot;     
[6445] &quot;8273&quot;      &quot;2539&quot;      &quot;8517&quot;      &quot;1736&quot;      &quot;2157&quot;      &quot;55217&quot;    </code></pre>
<pre class="r"><code># Load expression dataset
boxplot1 &lt;- read.csv(&quot;data/boxplot1.csv&quot;, check.names = FALSE)
boxplot1 &lt;- as.data.frame(boxplot1)

# Choose a gene from the DOX only mid-late response group (0.5 µM)
target_entrez &lt;- 4892  # Replace with your desired Entrez ID from prob_3_0.5

# Check membership
if (!(target_entrez %in% prob_3_0.5)) {
  stop(&quot;Selected gene is not in the DOX only mid-late response group for 0.5 micromolar.&quot;)
}

# Filter for selected gene
gene_data &lt;- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if(nrow(gene_data) == 0) stop(&quot;No data found for selected ENTREZID.&quot;)

# Reshape to long format
gene_data_long &lt;- melt(gene_data,
                       id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                       variable.name = &quot;Sample&quot;,
                       value.name = &quot;log2CPM&quot;)

# Extract metadata from sample names
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)
  )

# Filter for 0.5 micromolar only
gene_data_long &lt;- gene_data_long %&gt;% filter(Concentration == &quot;0.5&quot;)

# Convert to factors
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = &quot;0.5&quot;)

# Map individual IDs
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, 
                  &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], &quot;Unknown&quot;))

# Define color palette for drugs
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;#08306B&quot;, &quot;DOX&quot; = &quot;#E7298A&quot;, &quot;VEH&quot; = &quot;green&quot;)

# Extract gene symbol for labeling
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;DOX only mid-late\nresponse (0.5 micromolar)&quot;) +
  labs(
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/box6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box6-1">
Past versions of box6-1.png
</button>
</p>
<div id="fig-box6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/box6-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cx-total-dox-early-response-0.5-micromolar"
class="section level2">
<h2><strong>📌 CX total + DOX early response (0.5
micromolar)</strong></h2>
<pre class="r"><code># Load libraries
library(dplyr)
library(ggplot2)
library(reshape2)

# Load response group: CX total + DOX early response (0.5 µM)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)

# View gene list (optional)
prob_4_0.5</code></pre>
<pre><code>  [1] &quot;1870&quot;      &quot;79000&quot;     &quot;63967&quot;     &quot;55143&quot;     &quot;8438&quot;      &quot;6491&quot;     
  [7] &quot;1031&quot;      &quot;4998&quot;      &quot;55706&quot;     &quot;8317&quot;      &quot;343099&quot;    &quot;64858&quot;    
 [13] &quot;653820&quot;    &quot;554282&quot;    &quot;728833&quot;    &quot;1163&quot;      &quot;91687&quot;     &quot;79723&quot;    
 [19] &quot;414149&quot;    &quot;84930&quot;     &quot;983&quot;       &quot;219790&quot;    &quot;1763&quot;      &quot;6240&quot;     
 [25] &quot;2237&quot;      &quot;113130&quot;    &quot;23649&quot;     &quot;220042&quot;    &quot;2305&quot;      &quot;9918&quot;     
 [31] &quot;83461&quot;     &quot;9700&quot;      &quot;1017&quot;      &quot;100128191&quot; &quot;7112&quot;      &quot;283431&quot;   
 [37] &quot;5985&quot;      &quot;9735&quot;      &quot;140707&quot;    &quot;102659353&quot; &quot;5426&quot;      &quot;675&quot;      
 [43] &quot;440145&quot;    &quot;650669&quot;    &quot;11169&quot;     &quot;7443&quot;      &quot;7517&quot;      &quot;400322&quot;   
 [49] &quot;11339&quot;     &quot;51203&quot;     &quot;22995&quot;     &quot;9768&quot;      &quot;54982&quot;     &quot;90381&quot;    
 [55] &quot;1106&quot;      &quot;899&quot;       &quot;124222&quot;    &quot;9088&quot;      &quot;116028&quot;    &quot;54820&quot;    
 [61] &quot;8479&quot;      &quot;79801&quot;     &quot;23594&quot;     &quot;64785&quot;     &quot;55839&quot;     &quot;81620&quot;    
 [67] &quot;2175&quot;      &quot;83903&quot;     &quot;54478&quot;     &quot;79915&quot;     &quot;146857&quot;    &quot;29893&quot;    
 [73] &quot;78995&quot;     &quot;146956&quot;    &quot;3838&quot;      &quot;7083&quot;      &quot;201255&quot;    &quot;126282&quot;   
 [79] &quot;29128&quot;     &quot;10535&quot;     &quot;348654&quot;    &quot;79172&quot;     &quot;4175&quot;      &quot;83879&quot;    
 [85] &quot;151246&quot;    &quot;23178&quot;     &quot;1841&quot;      &quot;994&quot;       &quot;84515&quot;     &quot;81610&quot;    
 [91] &quot;6790&quot;      &quot;10732&quot;     &quot;29980&quot;     &quot;8208&quot;      &quot;54058&quot;     &quot;151648&quot;   
 [97] &quot;56992&quot;     &quot;993&quot;       &quot;25886&quot;     &quot;5984&quot;      &quot;10460&quot;     &quot;7468&quot;     
[103] &quot;1633&quot;      &quot;3015&quot;      &quot;890&quot;       &quot;201725&quot;    &quot;3148&quot;      &quot;79682&quot;    
[109] &quot;375444&quot;    &quot;55789&quot;     &quot;64946&quot;     &quot;4001&quot;      &quot;100996485&quot; &quot;995&quot;      
[115] &quot;3161&quot;      &quot;83463&quot;     &quot;51053&quot;     &quot;4800&quot;      &quot;55166&quot;     &quot;5558&quot;     
[121] &quot;9994&quot;      &quot;253714&quot;    &quot;387103&quot;    &quot;113115&quot;    &quot;26271&quot;     &quot;63979&quot;    
[127] &quot;5982&quot;      &quot;10926&quot;     &quot;4176&quot;      &quot;157313&quot;    &quot;157570&quot;    &quot;4173&quot;     
[133] &quot;8836&quot;      &quot;4603&quot;      &quot;9134&quot;      &quot;27085&quot;     &quot;9401&quot;      &quot;4957&quot;     
[139] &quot;29941&quot;     &quot;89958&quot;     &quot;10383&quot;     &quot;54821&quot;    </code></pre>
<pre class="r"><code># Load expression dataset
boxplot1 &lt;- read.csv(&quot;data/boxplot1.csv&quot;, check.names = FALSE)
boxplot1 &lt;- as.data.frame(boxplot1)

# Choose a gene from the CX total + DOX early response group (0.5 µM)
target_entrez &lt;- 983  # Replace with your desired Entrez ID from prob_4_0.5

# Check membership
if (!(target_entrez %in% prob_4_0.5)) {
  stop(&quot;Selected gene is not in the CX total + DOX early response group for 0.5 micromolar.&quot;)
}

# Filter for selected gene
gene_data &lt;- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if(nrow(gene_data) == 0) stop(&quot;No data found for selected ENTREZID.&quot;)

# Reshape to long format
gene_data_long &lt;- melt(gene_data,
                       id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                       variable.name = &quot;Sample&quot;,
                       value.name = &quot;log2CPM&quot;)

# Extract metadata from sample names
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)
  )

# Filter for 0.5 micromolar only
gene_data_long &lt;- gene_data_long %&gt;% filter(Concentration == &quot;0.5&quot;)

# Convert to factors
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = &quot;0.5&quot;)

# Map individual IDs
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, 
                  &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], &quot;Unknown&quot;))

# Define color palette for drugs
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;#08306B&quot;, &quot;DOX&quot; = &quot;#E7298A&quot;, &quot;VEH&quot; = &quot;green&quot;)

# Extract gene symbol for labeling
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;CX total+ DOX early\nresponse (0.5 micromolar)&quot;) +
  labs(
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/box7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box7-1">
Past versions of box7-1.png
</button>
</p>
<div id="fig-box7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/box7-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dox-early-cx-dox-mid-late-response-0.5-µm"
class="section level2">
<h2><strong>📌 DOX early + CX-DOX mid-late response (0.5
µM)</strong></h2>
<pre class="r"><code># Load libraries
library(dplyr)
library(ggplot2)
library(reshape2)

# Load response group: DOX early + CX-DOX mid-late response (0.5 µM)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# View gene list (optional)
prob_5_0.5</code></pre>
<pre><code>  [1] &quot;3151&quot;      &quot;83667&quot;     &quot;149175&quot;    &quot;991&quot;       &quot;11004&quot;     &quot;55635&quot;    
  [7] &quot;126820&quot;    &quot;163786&quot;    &quot;29899&quot;     &quot;57535&quot;     &quot;26227&quot;     &quot;9557&quot;     
 [13] &quot;54964&quot;     &quot;128239&quot;    &quot;83540&quot;     &quot;55732&quot;     &quot;51278&quot;     &quot;259266&quot;   
 [19] &quot;9928&quot;      &quot;29089&quot;     &quot;7832&quot;      &quot;729533&quot;    &quot;4751&quot;      &quot;51514&quot;    
 [25] &quot;1063&quot;      &quot;9156&quot;      &quot;55388&quot;     &quot;11130&quot;     &quot;79933&quot;     &quot;100144748&quot;
 [31] &quot;9585&quot;      &quot;3832&quot;      &quot;55165&quot;     &quot;3070&quot;      &quot;10613&quot;     &quot;54780&quot;    
 [37] &quot;4288&quot;      &quot;10346&quot;     &quot;81930&quot;     &quot;374393&quot;    &quot;79073&quot;     &quot;3619&quot;     
 [43] &quot;9633&quot;      &quot;10714&quot;     &quot;143888&quot;    &quot;3014&quot;      &quot;1111&quot;      &quot;23310&quot;    
 [49] &quot;10635&quot;     &quot;441631&quot;    &quot;10376&quot;     &quot;84790&quot;     &quot;10024&quot;     &quot;29127&quot;    
 [55] &quot;23468&quot;     &quot;8914&quot;      &quot;5557&quot;      &quot;23306&quot;     &quot;4193&quot;      &quot;55010&quot;    
 [61] &quot;7374&quot;      &quot;11113&quot;     &quot;221150&quot;    &quot;55835&quot;     &quot;5983&quot;      &quot;11340&quot;    
 [67] &quot;79866&quot;     &quot;10426&quot;     &quot;55632&quot;     &quot;55320&quot;     &quot;5427&quot;      &quot;64841&quot;    
 [73] &quot;1033&quot;      &quot;9787&quot;      &quot;4522&quot;      &quot;145508&quot;    &quot;89839&quot;     &quot;100288637&quot;
 [79] &quot;9824&quot;      &quot;701&quot;       &quot;90417&quot;     &quot;57082&quot;     &quot;5888&quot;      &quot;79968&quot;    
 [85] &quot;9133&quot;      &quot;80119&quot;     &quot;54962&quot;     &quot;55055&quot;     &quot;9493&quot;      &quot;26263&quot;    
 [91] &quot;64782&quot;     &quot;55215&quot;     &quot;641&quot;       &quot;9055&quot;      &quot;63922&quot;     &quot;23204&quot;    
 [97] &quot;5347&quot;      &quot;3835&quot;      &quot;65057&quot;     &quot;51659&quot;     &quot;9212&quot;      &quot;10615&quot;    
[103] &quot;990&quot;       &quot;7153&quot;      &quot;672&quot;       &quot;146909&quot;    &quot;55771&quot;     &quot;83990&quot;    
[109] &quot;105371937&quot; &quot;2232&quot;      &quot;332&quot;       &quot;7298&quot;      &quot;10403&quot;     &quot;3613&quot;     
[115] &quot;162655&quot;    &quot;220134&quot;    &quot;162681&quot;    &quot;148229&quot;    &quot;84823&quot;     &quot;10036&quot;    
[121] &quot;1786&quot;      &quot;147841&quot;    &quot;23354&quot;     &quot;115703&quot;    &quot;284403&quot;    &quot;8193&quot;     
[127] &quot;27113&quot;     &quot;3978&quot;      &quot;5424&quot;      &quot;27338&quot;     &quot;6241&quot;      &quot;1058&quot;     
[133] &quot;4436&quot;      &quot;23397&quot;     &quot;699&quot;       &quot;150468&quot;    &quot;57405&quot;     &quot;284992&quot;   
[139] &quot;55355&quot;     &quot;9837&quot;      &quot;22974&quot;     &quot;1869&quot;      &quot;79980&quot;     &quot;5933&quot;     
[145] &quot;4605&quot;      &quot;11065&quot;     &quot;54069&quot;     &quot;8318&quot;      &quot;11200&quot;     &quot;3976&quot;     
[151] &quot;4174&quot;      &quot;79640&quot;     &quot;79019&quot;     &quot;51512&quot;     &quot;2177&quot;      &quot;9942&quot;     
[157] &quot;10293&quot;     &quot;57650&quot;     &quot;10721&quot;     &quot;4171&quot;      &quot;8930&quot;      &quot;10051&quot;    
[163] &quot;1894&quot;      &quot;64393&quot;     &quot;116832&quot;    &quot;132671&quot;    &quot;1062&quot;      &quot;55345&quot;    
[169] &quot;4085&quot;      &quot;5393&quot;      &quot;10733&quot;     &quot;132320&quot;    &quot;84057&quot;     &quot;51166&quot;    
[175] &quot;55247&quot;     &quot;9319&quot;      &quot;64105&quot;     &quot;891&quot;       &quot;10112&quot;     &quot;9232&quot;     
[181] &quot;54908&quot;     &quot;7913&quot;      &quot;11270&quot;     &quot;9656&quot;      &quot;6941&quot;      &quot;1026&quot;     
[187] &quot;4172&quot;      &quot;7272&quot;      &quot;27244&quot;     &quot;2037&quot;      &quot;29887&quot;     &quot;54443&quot;    
[193] &quot;7378&quot;      &quot;349152&quot;    &quot;7516&quot;      &quot;54892&quot;     &quot;389610&quot;    &quot;8795&quot;     
[199] &quot;8794&quot;      &quot;55872&quot;     &quot;84296&quot;     &quot;25788&quot;     &quot;50484&quot;     &quot;79075&quot;    
[205] &quot;29028&quot;     &quot;11168&quot;     &quot;347240&quot;    &quot;2189&quot;      &quot;84904&quot;     &quot;9833&quot;     
[211] &quot;401541&quot;    &quot;100128361&quot; &quot;10592&quot;     &quot;26147&quot;     &quot;6839&quot;      &quot;139886&quot;   
[217] &quot;60401&quot;     &quot;24137&quot;     &quot;2491&quot;      &quot;55559&quot;     &quot;4354&quot;     </code></pre>
<pre class="r"><code># Load expression dataset
boxplot1 &lt;- read.csv(&quot;data/boxplot1.csv&quot;, check.names = FALSE)
boxplot1 &lt;- as.data.frame(boxplot1)

# Choose a gene from the DOX early + CX-DOX mid-late response group (0.5 µM)
target_entrez &lt;- 672  # Replace with your desired Entrez ID from prob_5_0.5

# Check membership
if (!(target_entrez %in% prob_5_0.5)) {
  stop(&quot;Selected gene is not in the DOX early + CX-DOX mid-late response group for 0.5 micromolar.&quot;)
}

# Filter for selected gene
gene_data &lt;- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if(nrow(gene_data) == 0) stop(&quot;No data found for selected ENTREZID.&quot;)

# Reshape to long format
gene_data_long &lt;- melt(gene_data,
                       id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                       variable.name = &quot;Sample&quot;,
                       value.name = &quot;log2CPM&quot;)

# Extract metadata from sample names
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)
  )

# Filter for 0.5 micromolar only
gene_data_long &lt;- gene_data_long %&gt;% filter(Concentration == &quot;0.5&quot;)

# Convert to factors
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = &quot;0.5&quot;)

# Map individual IDs
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, 
                  &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], &quot;Unknown&quot;))

# Define color palette for drugs
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;#08306B&quot;, &quot;DOX&quot; = &quot;#E7298A&quot;, &quot;VEH&quot; = &quot;green&quot;)

# Extract gene symbol for labeling
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;DOX early+ CX-DOX mid-late\nresponse (0.5 micromolar)&quot;) +
  labs(
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/box8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box8-1">
Past versions of box8-1.png
</button>
</p>
<div id="fig-box8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/box8-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="corrmotif-conc-boxplot-for-manuscript-0.1"
class="section level2">
<h2><strong>📌 Corrmotif conc boxplot for manuscript 0.1</strong></h2>
<pre class="r"><code># Load Required Libraries
library(dplyr)
library(ggplot2)

# ----------------- Load Response Groups -----------------

# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# ----------------- Load DEG Data -----------------

# Helper function to load and label
load_deg &lt;- function(path, drug, time) {
  read.csv(path) %&gt;%
    mutate(Drug = drug, Timepoint = time)
}

# 0.1 µM DEG data
deg_0.1 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;, &quot;CX.5461&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;, &quot;CX.5461&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;, &quot;CX.5461&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response\n(0.1 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group))

# ----------------- Set Factor Levels -----------------

deg_0.1 &lt;- deg_0.1 %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(3, 24, 48)),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.1 µM)&quot;,
      &quot;CX-DOX mid-late response\n(0.1 µM)&quot;,
      &quot;DOX only mid-late response\n(0.1 µM)&quot;
    ))
  )

# ----------------- Plot Boxplot (Grouped by Timepoint, Faceted by Response Group) -----------------

ggplot(deg_0.1, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75)) +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  facet_grid(. ~ Response_Group, scales = &quot;free_x&quot;) +
  theme_bw() +
  labs(
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;Log Fold Change&quot;,
    title = &quot;Log Fold Change Distribution by Drug and Response Group (0.1 µM)&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/box9-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box9-1">
Past versions of box9-1.png
</button>
</p>
<div id="fig-box9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/a56887202d8df9b0816df4b966d55f223263e84a/docs/figure/Corrmotif_Conc.Rmd/box9-1.png" target="_blank">a568872</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="corrmotif-conc-boxplot-for-manuscript-0.5"
class="section level2">
<h2><strong>📌 Corrmotif conc boxplot for manuscript 0.5</strong></h2>
<pre class="r"><code># Load Required Libraries
library(dplyr)
library(ggplot2)

# ----------------- Load Response Groups -----------------

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Load DEG Data -----------------

# Helper function to load and label
load_deg &lt;- function(path, drug, time) {
  read.csv(path) %&gt;%
    mutate(Drug = drug, Timepoint = time)
}

# 0.5 µM DEG data
deg_0.5 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;, &quot;CX.5461&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;, &quot;CX.5461&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;, &quot;CX.5461&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 µM)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX + DOX (early)\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX + CX (mid-late)\nresponse (0.5 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group))

# ----------------- Set Factor Levels -----------------

deg_0.5 &lt;- deg_0.5 %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(3, 24, 48)),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.5 µM)&quot;,
      &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      &quot;CX + DOX (early)\nresponse (0.5 µM)&quot;,
      &quot;DOX + CX (mid-late)\nresponse (0.5 µM)&quot;
    ))
  )

# ----------------- Plot Boxplot (Grouped by Timepoint, Faceted by Response Group) -----------------

ggplot(deg_0.5, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75)) +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  facet_grid(. ~ Response_Group, scales = &quot;free_x&quot;) +
  theme_bw() +
  labs(
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;Log Fold Change&quot;,
    title = &quot;Log Fold Change Distribution by Drug and Response Group (0.5 µM)&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/box10-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box10-1">
Past versions of box10-1.png
</button>
</p>
<div id="fig-box10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/a56887202d8df9b0816df4b966d55f223263e84a/docs/figure/Corrmotif_Conc.Rmd/box10-1.png" target="_blank">a568872</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 26100)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] reshape2_1.4.4       lubridate_1.9.4      forcats_1.0.0       
 [4] stringr_1.5.1        purrr_1.0.4          readr_2.1.5         
 [7] tibble_3.2.1         tidyverse_2.0.0      org.Hs.eg.db_3.18.0 
[10] AnnotationDbi_1.64.1 IRanges_2.36.0       S4Vectors_0.40.2    
[13] tidyr_1.3.1          ggplot2_3.5.2        gprofiler2_0.2.3    
[16] BiocParallel_1.36.0  dplyr_1.1.4          Rfast_2.1.5.1       
[19] RcppParallel_5.1.10  zigg_0.0.2           Rcpp_1.0.12         
[22] Cormotif_1.48.0      limma_3.58.1         affy_1.80.0         
[25] Biobase_2.62.0       BiocGenerics_0.48.1  workflowr_1.7.1     

loaded via a namespace (and not attached):
 [1] DBI_1.2.3               bitops_1.0-9            rlang_1.1.3            
 [4] magrittr_2.0.3          git2r_0.36.2            compiler_4.3.0         
 [7] RSQLite_2.3.9           getPass_0.2-4           png_0.1-8              
[10] callr_3.7.6             vctrs_0.6.5             pkgconfig_2.0.3        
[13] crayon_1.5.3            fastmap_1.2.0           XVector_0.42.0         
[16] labeling_0.4.3          promises_1.3.2          rmarkdown_2.29         
[19] tzdb_0.5.0              ps_1.8.1                preprocessCore_1.64.0  
[22] bit_4.6.0               xfun_0.52               zlibbioc_1.48.2        
[25] cachem_1.1.0            GenomeInfoDb_1.38.8     jsonlite_2.0.0         
[28] blob_1.2.4              later_1.3.2             parallel_4.3.0         
[31] R6_2.6.1                RColorBrewer_1.1-3      bslib_0.9.0            
[34] stringi_1.8.3           jquerylib_0.1.4         knitr_1.50             
[37] timechange_0.3.0        httpuv_1.6.15           tidyselect_1.2.1       
[40] rstudioapi_0.17.1       yaml_2.3.10             codetools_0.2-20       
[43] processx_3.8.6          plyr_1.8.9              withr_3.0.2            
[46] KEGGREST_1.42.0         evaluate_1.0.3          Biostrings_2.70.3      
[49] pillar_1.10.2           affyio_1.72.0           BiocManager_1.30.25    
[52] whisker_0.4.1           plotly_4.10.4           generics_0.1.3         
[55] vroom_1.6.5             rprojroot_2.0.4         RCurl_1.98-1.17        
[58] hms_1.1.3               munsell_0.5.1           scales_1.3.0           
[61] glue_1.7.0              lazyeval_0.2.2          tools_4.3.0            
[64] data.table_1.17.0       fs_1.6.3                grid_4.3.0             
[67] colorspace_2.1-0        GenomeInfoDbData_1.2.11 cli_3.6.1              
[70] viridisLite_0.4.2       gtable_0.3.6            sass_0.4.10            
[73] digest_0.6.34           htmlwidgets_1.6.4       farver_2.1.2           
[76] memoise_2.0.1           htmltools_0.5.8.1       lifecycle_1.0.4        
[79] httr_1.4.7              statmod_1.5.0           bit64_4.6.0-1          </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
