<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Corrmotif Conc Analysis</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CX5461_Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Corrmotif Conc Analysis</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-06
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>CX5461_Project/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttreedf65b82ed20ee529ee980c6798b831c117e5159btargetblankdf65b82a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/df65b82ed20ee529ee980c6798b831c117e5159b" target="_blank">df65b82</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttreedf65b82ed20ee529ee980c6798b831c117e5159btargetblankdf65b82a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/df65b82ed20ee529ee980c6798b831c117e5159b" target="_blank">df65b82</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    0.1 box.svg
    Ignored:    Rplot04.svg
    Ignored:    analysis/Corrmotif_Conc.html

Untracked files:
    Untracked:  0.1 density.svg
    Untracked:  0.1.emf
    Untracked:  0.1.svg
    Untracked:  0.5 box.svg
    Untracked:  0.5 density.svg
    Untracked:  0.5.svg
    Untracked:  Additional/
    Untracked:  Autosome factors.svg
    Untracked:  CX_5461_Pattern_Genes_24hr.csv
    Untracked:  CX_5461_Pattern_Genes_3hr.csv
    Untracked:  Cell viability box plot.svg
    Untracked:  DEG GO terms.svg
    Untracked:  DNA damage associated GO terms.svg
    Untracked:  DRC1.svg
    Untracked:  Figure 1.jpeg
    Untracked:  Figure 1.pdf
    Untracked:  Figure_CM_Purity.pdf
    Untracked:  G Quadruplex DEGs.svg
    Untracked:  PC2 Vs PC3 Autosome.svg
    Untracked:  PCA autosome.svg
    Untracked:  Rplot 18.svg
    Untracked:  Rplot.svg
    Untracked:  Rplot01.svg
    Untracked:  Rplot02.svg
    Untracked:  Rplot03.svg
    Untracked:  Rplot05.svg
    Untracked:  Rplot06.svg
    Untracked:  Rplot07.svg
    Untracked:  Rplot08.jpeg
    Untracked:  Rplot08.svg
    Untracked:  Rplot09.svg
    Untracked:  Rplot10.svg
    Untracked:  Rplot11.svg
    Untracked:  Rplot12.svg
    Untracked:  Rplot13.svg
    Untracked:  Rplot14.svg
    Untracked:  Rplot15.svg
    Untracked:  Rplot16.svg
    Untracked:  Rplot17.svg
    Untracked:  Rplot18.svg
    Untracked:  Rplot19.svg
    Untracked:  Rplot20.svg
    Untracked:  Rplot21.svg
    Untracked:  Rplot22.svg
    Untracked:  Rplot23.svg
    Untracked:  Rplot24.svg
    Untracked:  TOP2B.bed
    Untracked:  TS HPA (Violin).svg
    Untracked:  TS HPA.svg
    Untracked:  TS_HA.svg
    Untracked:  TS_HV.svg
    Untracked:  Violin HA.svg
    Untracked:  Violin HV (CX vs DOX).svg
    Untracked:  Violin HV.svg
    Untracked:  data/AF.csv
    Untracked:  data/AF_Mapped.csv
    Untracked:  data/AF_genes.csv
    Untracked:  data/Annotated_DOX_Gene_Table.csv
    Untracked:  data/BP/
    Untracked:  data/CAD_genes.csv
    Untracked:  data/Cardiotox.csv
    Untracked:  data/Cardiotox_mapped.csv
    Untracked:  data/Corrmotif_GO/
    Untracked:  data/DOX_Vald.csv
    Untracked:  data/DOX_Vald_Mapped.csv
    Untracked:  data/DOX_alt.csv
    Untracked:  data/Entrez_Cardiotox.csv
    Untracked:  data/Entrez_Cardiotox_Mapped.csv
    Untracked:  data/GWAS.xlsx
    Untracked:  data/GWAS_SNPs.bed
    Untracked:  data/HF.csv
    Untracked:  data/HF_Mapped.csv
    Untracked:  data/HF_genes.csv
    Untracked:  data/Hypertension_genes.csv
    Untracked:  data/MI_genes.csv
    Untracked:  data/P53_Target_mapped.csv
    Untracked:  data/Sample_annotated.csv
    Untracked:  data/Samples.csv
    Untracked:  data/Samples.xlsx
    Untracked:  data/TOP2A.bed
    Untracked:  data/TOP2A_target.csv
    Untracked:  data/TOP2A_target_lit.csv
    Untracked:  data/TOP2A_target_lit_mapped.csv
    Untracked:  data/TOP2A_target_mapped.csv
    Untracked:  data/TOP2B.bed
    Untracked:  data/TOP2B_target.csv
    Untracked:  data/TOP2B_target_heatmap.csv
    Untracked:  data/TOP2B_target_heatmap_mapped.csv
    Untracked:  data/TOP2B_target_mapped.csv
    Untracked:  data/TS.csv
    Untracked:  data/TS_HPA.csv
    Untracked:  data/TS_HPA_mapped.csv
    Untracked:  data/Toptable_CX_0.1_24.csv
    Untracked:  data/Toptable_CX_0.1_3.csv
    Untracked:  data/Toptable_CX_0.1_48.csv
    Untracked:  data/Toptable_CX_0.5_24.csv
    Untracked:  data/Toptable_CX_0.5_3.csv
    Untracked:  data/Toptable_CX_0.5_48.csv
    Untracked:  data/Toptable_DOX_0.1_24.csv
    Untracked:  data/Toptable_DOX_0.1_3.csv
    Untracked:  data/Toptable_DOX_0.1_48.csv
    Untracked:  data/Toptable_DOX_0.5_24.csv
    Untracked:  data/Toptable_DOX_0.5_3.csv
    Untracked:  data/Toptable_DOX_0.5_48.csv
    Untracked:  data/count.tsv
    Untracked:  data/ts_data_mapped
    Untracked:  results/
    Untracked:  run_bedtools.bat

Unstaged changes:
    Deleted:    analysis/Actox.Rmd
    Modified:   analysis/Corrmotif_Conc.Rmd
    Modified:   data/DOX_0.5_48 (Combined).csv
    Modified:   data/Human_Heart_Genes.csv
    Modified:   data/Total_number_of_Mapped_reads_by_Individuals.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Corrmotif_Conc.Rmd</code>) and
HTML (<code>docs/Corrmotif_Conc.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/a77ea961c598cd4f1f5109f61945452408af492b/analysis/Corrmotif_Conc.Rmd" target="_blank">a77ea96</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-16
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/a77ea961c598cd4f1f5109f61945452408af492b/docs/Corrmotif_Conc.html" target="_blank">a77ea96</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-16
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/e29b0caae160d0ce9cd0ddd4b2fd512739ad5cf9/docs/Corrmotif_Conc.html" target="_blank">e29b0ca</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/analysis/Corrmotif_Conc.Rmd" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/Corrmotif_Conc.html" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/1661001828f9b43008c4958fd25156f288811169/analysis/Corrmotif_Conc.Rmd" target="_blank">1661001</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/1661001828f9b43008c4958fd25156f288811169/docs/Corrmotif_Conc.html" target="_blank">1661001</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/b9a286768620bdf4d87f2a63ddc247e5316445d0/analysis/Corrmotif_Conc.Rmd" target="_blank">b9a2867</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-05
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/b9a286768620bdf4d87f2a63ddc247e5316445d0/docs/Corrmotif_Conc.html" target="_blank">b9a2867</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-05
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/10ab0f3506d3aaa233e23ec14f7671cce1563e28/analysis/Corrmotif_Conc.Rmd" target="_blank">10ab0f3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-05
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/10ab0f3506d3aaa233e23ec14f7671cce1563e28/docs/Corrmotif_Conc.html" target="_blank">10ab0f3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-05
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/da5bc51ec43f9e4e8d0c046a5388f15106eaa1c2/analysis/Corrmotif_Conc.Rmd" target="_blank">da5bc51</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-04
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/da5bc51ec43f9e4e8d0c046a5388f15106eaa1c2/docs/Corrmotif_Conc.html" target="_blank">da5bc51</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-04
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/6f9e594c722182678bafc5ddb7c9cdeb3f27dfbf/analysis/Corrmotif_Conc.Rmd" target="_blank">6f9e594</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-24
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/6f9e594c722182678bafc5ddb7c9cdeb3f27dfbf/docs/Corrmotif_Conc.html" target="_blank">6f9e594</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-24
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d6226a09c4a6b3b6c2280e9c4f41ba17acb41045/analysis/Corrmotif_Conc.Rmd" target="_blank">d6226a0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-24
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/d6226a09c4a6b3b6c2280e9c4f41ba17acb41045/docs/Corrmotif_Conc.html" target="_blank">d6226a0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-24
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/062a73abf4aa51016b5a82ec16868e5950354cdf/analysis/Corrmotif_Conc.Rmd" target="_blank">062a73a</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-24
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/062a73abf4aa51016b5a82ec16868e5950354cdf/docs/Corrmotif_Conc.html" target="_blank">062a73a</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-24
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/07b9670bf922129e5d7a932d22711efc033fde34/docs/Corrmotif_Conc.html" target="_blank">07b9670</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-21
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/9618a00ad12263a31ed79f64344c82a444c6d5c1/docs/Corrmotif_Conc.html" target="_blank">9618a00</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-21
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/008117ab97e9c8b972ffcc1bad9c90b0946274be/docs/Corrmotif_Conc.html" target="_blank">008117a</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/1433d3784ab1f1829ee095be49b4a2de6ab57ca5/docs/Corrmotif_Conc.html" target="_blank">1433d37</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-14
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/4bad905ad26e9ebac9188a323b7f5fe10aa56ce6/analysis/Corrmotif_Conc.Rmd" target="_blank">4bad905</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-14
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/4bad905ad26e9ebac9188a323b7f5fe10aa56ce6/docs/Corrmotif_Conc.html" target="_blank">4bad905</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-14
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/6ef5f61785e28f0c6385c185a7d54db2620ea914/analysis/Corrmotif_Conc.Rmd" target="_blank">6ef5f61</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-14
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/6ef5f61785e28f0c6385c185a7d54db2620ea914/docs/Corrmotif_Conc.html" target="_blank">6ef5f61</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-14
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d8a2f25dc1beb5fef19f7a7a03802eb5999393cd/analysis/Corrmotif_Conc.Rmd" target="_blank">d8a2f25</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-14
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/d8a2f25dc1beb5fef19f7a7a03802eb5999393cd/docs/Corrmotif_Conc.html" target="_blank">d8a2f25</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-14
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/a56887202d8df9b0816df4b966d55f223263e84a/analysis/Corrmotif_Conc.Rmd" target="_blank">a568872</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-10
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/a56887202d8df9b0816df4b966d55f223263e84a/docs/Corrmotif_Conc.html" target="_blank">a568872</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-10
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/4e40091300781c5ddf17c486457d658c5619f25d/analysis/Corrmotif_Conc.Rmd" target="_blank">4e40091</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/4e40091300781c5ddf17c486457d658c5619f25d/docs/Corrmotif_Conc.html" target="_blank">4e40091</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/analysis/Corrmotif_Conc.Rmd" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/Corrmotif_Conc.html" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2990b21e59ad6434e6e7b4b92ae6cad8ff37f997/analysis/Corrmotif_Conc.Rmd" target="_blank">2990b21</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-10
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/2990b21e59ad6434e6e7b4b92ae6cad8ff37f997/docs/Corrmotif_Conc.html" target="_blank">2990b21</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-10
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ef3a95166a7a18770eaa7786b80994b40bba9973/analysis/Corrmotif_Conc.Rmd" target="_blank">ef3a951</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-09
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/ef3a95166a7a18770eaa7786b80994b40bba9973/docs/Corrmotif_Conc.html" target="_blank">ef3a951</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-09
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/81247f64f08df43c2b40a386707fc8e9cc240143/analysis/Corrmotif_Conc.Rmd" target="_blank">81247f6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-06
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/81247f64f08df43c2b40a386707fc8e9cc240143/docs/Corrmotif_Conc.html" target="_blank">81247f6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-06
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/96d0db145cf68e6db0d39e70bad9af0f75793aae/analysis/Corrmotif_Conc.Rmd" target="_blank">96d0db1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/96d0db145cf68e6db0d39e70bad9af0f75793aae/docs/Corrmotif_Conc.html" target="_blank">96d0db1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/analysis/Corrmotif_Conc.Rmd" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/Corrmotif_Conc.html" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/analysis/Corrmotif_Conc.Rmd" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/Corrmotif_Conc.html" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2de67e6a82f82edbc78941b11ba72ed8aba9964e/analysis/Corrmotif_Conc.Rmd" target="_blank">2de67e6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/2de67e6a82f82edbc78941b11ba72ed8aba9964e/docs/Corrmotif_Conc.html" target="_blank">2de67e6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/41cd1be487c4309fb8ab2f2d36b18ad63cdc31e4/analysis/Corrmotif_Conc.Rmd" target="_blank">41cd1be</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/41cd1be487c4309fb8ab2f2d36b18ad63cdc31e4/docs/Corrmotif_Conc.html" target="_blank">41cd1be</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/f84821f0421d3805726f2ccfce5b5ca2275c6756/analysis/Corrmotif_Conc.Rmd" target="_blank">f84821f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-26
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/b6e38a14f7f28ac96f0b16857be1f89d71989e58/analysis/Corrmotif_Conc.Rmd" target="_blank">b6e38a1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-25
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/b6e38a14f7f28ac96f0b16857be1f89d71989e58/docs/Corrmotif_Conc.html" target="_blank">b6e38a1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-25
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ce4b32557fc3c7ee693369db9edf0a9d22968203/analysis/Corrmotif_Conc.Rmd" target="_blank">ce4b325</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-25
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/ce4b32557fc3c7ee693369db9edf0a9d22968203/docs/Corrmotif_Conc.html" target="_blank">ce4b325</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-25
</td>
<td>
Commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="micromolar" class="section level2">
<h2><strong>📌 0.1 Micromolar</strong></h2>
</div>
<div id="fit-limma-model-functions" class="section level2">
<h2><strong>📌 Fit Limma Model Functions</strong></h2>
<pre class="r"><code>## Fit limma model using code as it is found in the original cormotif code. It has
## only been modified to add names to the matrix of t values, as well as the
## limma fits

limmafit.default &lt;- function(exprs,groupid,compid) {
  limmafits  &lt;- list()
  compnum    &lt;- nrow(compid)
  genenum    &lt;- nrow(exprs)
  limmat     &lt;- matrix(0,genenum,compnum)
  limmas2    &lt;- rep(0,compnum)
  limmadf    &lt;- rep(0,compnum)
  limmav0    &lt;- rep(0,compnum)
  limmag1num &lt;- rep(0,compnum)
  limmag2num &lt;- rep(0,compnum)

  rownames(limmat)  &lt;- rownames(exprs)
  colnames(limmat)  &lt;- rownames(compid)
  names(limmas2)    &lt;- rownames(compid)
  names(limmadf)    &lt;- rownames(compid)
  names(limmav0)    &lt;- rownames(compid)
  names(limmag1num) &lt;- rownames(compid)
  names(limmag2num) &lt;- rownames(compid)

  for(i in 1:compnum) {
    selid1 &lt;- which(groupid == compid[i,1])
    selid2 &lt;- which(groupid == compid[i,2])
    eset   &lt;- new(&quot;ExpressionSet&quot;, exprs=cbind(exprs[,selid1],exprs[,selid2]))
    g1num  &lt;- length(selid1)
    g2num  &lt;- length(selid2)
    designmat &lt;- cbind(base=rep(1,(g1num+g2num)), delta=c(rep(0,g1num),rep(1,g2num)))
    fit &lt;- lmFit(eset,designmat)
    fit &lt;- eBayes(fit)
    limmat[,i] &lt;- fit$t[,2]
    limmas2[i] &lt;- fit$s2.prior
    limmadf[i] &lt;- fit$df.prior
    limmav0[i] &lt;- fit$var.prior[2]
    limmag1num[i] &lt;- g1num
    limmag2num[i] &lt;- g2num
    limmafits[[i]] &lt;- fit

    # log odds
    # w&lt;-sqrt(1+fit$var.prior[2]/(1/g1num+1/g2num))
    # log(0.99)+dt(fit$t[1,2],g1num+g2num-2+fit$df.prior,log=TRUE)-log(0.01)-dt(fit$t[1,2]/w, g1num+g2num-2+fit$df.prior, log=TRUE)+log(w)
  }
  names(limmafits) &lt;- rownames(compid)
  limmacompnum&lt;-nrow(compid)
  result&lt;-list(t       = limmat,
               v0      = limmav0,
               df0     = limmadf,
               s20     = limmas2,
               g1num   = limmag1num,
               g2num   = limmag2num,
               compnum = limmacompnum,
               fits    = limmafits)
}

limmafit.counts &lt;-
  function (exprs, groupid, compid, norm.factor.method = &quot;TMM&quot;, voom.normalize.method = &quot;none&quot;)
  {
    limmafits  &lt;- list()
    compnum    &lt;- nrow(compid)
    genenum    &lt;- nrow(exprs)
    limmat     &lt;- matrix(NA,genenum,compnum)
    limmas2    &lt;- rep(0,compnum)
    limmadf    &lt;- rep(0,compnum)
    limmav0    &lt;- rep(0,compnum)
    limmag1num &lt;- rep(0,compnum)
    limmag2num &lt;- rep(0,compnum)

    rownames(limmat)  &lt;- rownames(exprs)
    colnames(limmat)  &lt;- rownames(compid)
    names(limmas2)    &lt;- rownames(compid)
    names(limmadf)    &lt;- rownames(compid)
    names(limmav0)    &lt;- rownames(compid)
    names(limmag1num) &lt;- rownames(compid)
    names(limmag2num) &lt;- rownames(compid)

    for (i in 1:compnum) {
      message(paste(&quot;Running limma for comparision&quot;,i,&quot;/&quot;,compnum))
      selid1 &lt;- which(groupid == compid[i, 1])
      selid2 &lt;- which(groupid == compid[i, 2])
      # make a new count data frame
      counts &lt;- cbind(exprs[, selid1], exprs[, selid2])

      # remove NAs
      not.nas &lt;- which(apply(counts, 1, function(x) !any(is.na(x))) == TRUE)

      # runn voom/limma
      d &lt;- DGEList(counts[not.nas,])
      d &lt;- calcNormFactors(d, method = norm.factor.method)
      g1num &lt;- length(selid1)
      g2num &lt;- length(selid2)
      designmat &lt;- cbind(base = rep(1, (g1num + g2num)), delta = c(rep(0,
                                                                       g1num), rep(1, g2num)))

      y &lt;- voom(d, designmat, normalize.method = voom.normalize.method)
      fit &lt;- lmFit(y, designmat)
      fit &lt;- eBayes(fit)

      limmafits[[i]] &lt;- fit
      limmat[not.nas, i] &lt;- fit$t[, 2]
      limmas2[i] &lt;- fit$s2.prior
      limmadf[i] &lt;- fit$df.prior
      limmav0[i] &lt;- fit$var.prior[2]
      limmag1num[i] &lt;- g1num
      limmag2num[i] &lt;- g2num
    }
    limmacompnum &lt;- nrow(compid)
    names(limmafits) &lt;- rownames(compid)
    result &lt;- list(t       = limmat,
                   v0      = limmav0,
                   df0     = limmadf,
                   s20     = limmas2,
                   g1num   = limmag1num,
                   g2num   = limmag2num,
                   compnum = limmacompnum,
                   fits    = limmafits)
  }

limmafit.list &lt;-
  function (fitlist, cmp.idx=2)
  {
    compnum    &lt;- length(fitlist)

    genes &lt;- c()
    for (i in 1:compnum) genes &lt;- unique(c(genes, rownames(fitlist[[i]])))

    genenum    &lt;- length(genes)
    limmat     &lt;- matrix(NA,genenum,compnum)
    limmas2    &lt;- rep(0,compnum)
    limmadf    &lt;- rep(0,compnum)
    limmav0    &lt;- rep(0,compnum)
    limmag1num &lt;- rep(0,compnum)
    limmag2num &lt;- rep(0,compnum)

    rownames(limmat)  &lt;- genes
    colnames(limmat)  &lt;- names(fitlist)
    names(limmas2)    &lt;- names(fitlist)
    names(limmadf)    &lt;- names(fitlist)
    names(limmav0)    &lt;- names(fitlist)
    names(limmag1num) &lt;- names(fitlist)
    names(limmag2num) &lt;- names(fitlist)

    for (i in 1:compnum) {
      this.t &lt;- fitlist[[i]]$t[,cmp.idx]
      limmat[names(this.t),i] &lt;- this.t

      limmas2[i]    &lt;- fitlist[[i]]$s2.prior
      limmadf[i]    &lt;- fitlist[[i]]$df.prior
      limmav0[i]    &lt;- fitlist[[i]]$var.prior[cmp.idx]
      limmag1num[i] &lt;- sum(fitlist[[i]]$design[,cmp.idx]==0)
      limmag2num[i] &lt;- sum(fitlist[[i]]$design[,cmp.idx]==1)
    }

    limmacompnum &lt;- compnum
    result &lt;- list(t       = limmat,
                   v0      = limmav0,
                   df0     = limmadf,
                   s20     = limmas2,
                   g1num   = limmag1num,
                   g2num   = limmag2num,
                   compnum = limmacompnum,
                   fits    = limmafits)

  }

## Rank genes based on statistics
generank&lt;-function(x) {
  xcol&lt;-ncol(x)
  xrow&lt;-nrow(x)
  result&lt;-matrix(0,xrow,xcol)
  z&lt;-(1:1:xrow)
  for(i in 1:xcol) {
    y&lt;-sort(x[,i],decreasing=TRUE,na.last=TRUE)
    result[,i]&lt;-match(x[,i],y)
    result[,i]&lt;-order(result[,i])
  }
  result
}

## Log-likelihood for moderated t under H0
modt.f0.loglike&lt;-function(x,df) {
  a&lt;-dt(x, df, log=TRUE)
  result&lt;-as.vector(a)
  flag&lt;-which(is.na(result)==TRUE)
  result[flag]&lt;-0
  result
}

## Log-likelihood for moderated t under H1
## param=c(df,g1num,g2num,v0)
modt.f1.loglike&lt;-function(x,param) {
  df&lt;-param[1]
  g1num&lt;-param[2]
  g2num&lt;-param[3]
  v0&lt;-param[4]
  w&lt;-sqrt(1+v0/(1/g1num+1/g2num))
  dt(x/w, df, log=TRUE)-log(w)
  a&lt;-dt(x/w, df, log=TRUE)-log(w)
  result&lt;-as.vector(a)
  flag&lt;-which(is.na(result)==TRUE)
  result[flag]&lt;-0
  result
}

## Correlation Motif Fit
cmfit.X&lt;-function(x, type, K=1, tol=1e-3, max.iter=100) {
  ## initialize
  xrow &lt;- nrow(x)
  xcol &lt;- ncol(x)
  loglike0 &lt;- list()
  loglike1 &lt;- list()
  p &lt;- rep(1, K)/K
  q &lt;- matrix(runif(K * xcol), K, xcol)
  q[1, ] &lt;- rep(0.01, xcol)
  for (i in 1:xcol) {
    f0 &lt;- type[[i]][[1]]
    f0param &lt;- type[[i]][[2]]
    f1 &lt;- type[[i]][[3]]
    f1param &lt;- type[[i]][[4]]
    loglike0[[i]] &lt;- f0(x[, i], f0param)
    loglike1[[i]] &lt;- f1(x[, i], f1param)
  }
  condlike &lt;- list()
  for (i in 1:xcol) {
    condlike[[i]] &lt;- matrix(0, xrow, K)
  }
  loglike.old &lt;- -1e+10
  for (i.iter in 1:max.iter) {
    if ((i.iter%%50) == 0) {
      print(paste(&quot;We have run the first &quot;, i.iter, &quot; iterations for K=&quot;,
                  K, sep = &quot;&quot;))
    }
    err &lt;- tol + 1
    clustlike &lt;- matrix(0, xrow, K)
    #templike &lt;- matrix(0, xrow, 2)
    templike1 &lt;- rep(0, xrow)
    templike2 &lt;- rep(0, xrow)
    for (j in 1:K) {
      for (i in 1:xcol) {
        templike1 &lt;- log(q[j, i]) + loglike1[[i]]
        templike2 &lt;- log(1 - q[j, i]) + loglike0[[i]]
        tempmax &lt;- Rfast::Pmax(templike1, templike2)

        templike1 &lt;- exp(templike1 - tempmax)
        templike2 &lt;- exp(templike2 - tempmax)

        tempsum &lt;- templike1 + templike2
        clustlike[, j] &lt;- clustlike[, j] + tempmax +
          log(tempsum)
        condlike[[i]][, j] &lt;- templike1/tempsum
      }
      clustlike[, j] &lt;- clustlike[, j] + log(p[j])
    }
    #tempmax &lt;- apply(clustlike, 1, max)
    tempmax &lt;- Rfast::rowMaxs(clustlike, value=TRUE)
    for (j in 1:K) {
      clustlike[, j] &lt;- exp(clustlike[, j] - tempmax)
    }
    #tempsum &lt;- apply(clustlike, 1, sum)
    tempsum &lt;- Rfast::rowsums(clustlike)
    for (j in 1:K) {
      clustlike[, j] &lt;- clustlike[, j]/tempsum
    }
    #p.new &lt;- (apply(clustlike, 2, sum) + 1)/(xrow + K)
    p.new &lt;- (Rfast::colsums(clustlike) + 1)/(xrow + K)
    q.new &lt;- matrix(0, K, xcol)
    for (j in 1:K) {
      clustpsum &lt;- sum(clustlike[, j])
      for (i in 1:xcol) {
        q.new[j, i] &lt;- (sum(clustlike[, j] * condlike[[i]][,
                                                           j]) + 1)/(clustpsum + 2)
      }
    }
    err.p &lt;- max(abs(p.new - p)/p)
    err.q &lt;- max(abs(q.new - q)/q)
    err &lt;- max(err.p, err.q)
    loglike.new &lt;- (sum(tempmax + log(tempsum)) + sum(log(p.new)) +
                      sum(log(q.new) + log(1 - q.new)))/xrow
    p &lt;- p.new
    q &lt;- q.new
    loglike.old &lt;- loglike.new
    if (err &lt; tol) {
      break
    }
  }
  clustlike &lt;- matrix(0, xrow, K)
  for (j in 1:K) {
    for (i in 1:xcol) {
      templike1 &lt;- log(q[j, i]) + loglike1[[i]]
      templike2 &lt;- log(1 - q[j, i]) + loglike0[[i]]
      tempmax &lt;- Rfast::Pmax(templike1, templike2)

      templike1 &lt;- exp(templike1 - tempmax)
      templike2 &lt;- exp(templike2 - tempmax)

      tempsum &lt;- templike1 + templike2
      clustlike[, j] &lt;- clustlike[, j] + tempmax + log(tempsum)
      condlike[[i]][, j] &lt;- templike1/tempsum
    }
    clustlike[, j] &lt;- clustlike[, j] + log(p[j])
  }
  #tempmax &lt;- apply(clustlike, 1, max)
  tempmax &lt;- Rfast::rowMaxs(clustlike, value=TRUE)
  for (j in 1:K) {
    clustlike[, j] &lt;- exp(clustlike[, j] - tempmax)
  }
  #tempsum &lt;- apply(clustlike, 1, sum)
  tempsum &lt;- Rfast::rowsums(clustlike)
  for (j in 1:K) {
    clustlike[, j] &lt;- clustlike[, j]/tempsum
  }
  p.post &lt;- matrix(0, xrow, xcol)
  for (j in 1:K) {
    for (i in 1:xcol) {
      p.post[, i] &lt;- p.post[, i] + clustlike[, j] * condlike[[i]][,
                                                                  j]
    }
  }
  loglike.old &lt;- loglike.old - (sum(log(p)) + sum(log(q) +
                                                    log(1 - q)))/xrow
  loglike.old &lt;- loglike.old * xrow
  result &lt;- list(p.post = p.post, motif.prior = p, motif.q = q,
                 loglike = loglike.old, clustlike=clustlike, condlike=condlike)
}

## Fit using (0,0,...,0) and (1,1,...,1)
cmfitall&lt;-function(x, type, tol=1e-3, max.iter=100) {
  ## initialize
  xrow&lt;-nrow(x)
  xcol&lt;-ncol(x)
  loglike0&lt;-list()
  loglike1&lt;-list()
  p&lt;-0.01

  ## compute loglikelihood
  L0&lt;-matrix(0,xrow,1)
  L1&lt;-matrix(0,xrow,1)
  for(i in 1:xcol) {
    f0&lt;-type[[i]][[1]]
    f0param&lt;-type[[i]][[2]]
    f1&lt;-type[[i]][[3]]
    f1param&lt;-type[[i]][[4]]
    loglike0[[i]]&lt;-f0(x[,i],f0param)
    loglike1[[i]]&lt;-f1(x[,i],f1param)
    L0&lt;-L0+loglike0[[i]]
    L1&lt;-L1+loglike1[[i]]
  }


  ## EM algorithm to get MLE of p and q
  loglike.old &lt;- -1e10
  for(i.iter in 1:max.iter) {
    if((i.iter%%50) == 0) {
      print(paste(&quot;We have run the first &quot;, i.iter, &quot; iterations&quot;,sep=&quot;&quot;))
    }
    err&lt;-tol+1

    ## compute posterior cluster membership
    clustlike&lt;-matrix(0,xrow,2)
    clustlike[,1]&lt;-log(1-p)+L0
    clustlike[,2]&lt;-log(p)+L1

    tempmax&lt;-apply(clustlike,1,max)
    for(j in 1:2) {
      clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
    }
    tempsum&lt;-apply(clustlike,1,sum)

    ## update motif occurrence rate
    for(j in 1:2) {
      clustlike[,j]&lt;-clustlike[,j]/tempsum
    }

    p.new&lt;-(sum(clustlike[,2])+1)/(xrow+2)

    ## evaluate convergence
    err&lt;-abs(p.new-p)/p

    ## evaluate whether the log.likelihood increases
    loglike.new&lt;-(sum(tempmax+log(tempsum))+log(p.new)+log(1-p.new))/xrow

    loglike.old&lt;-loglike.new
    p&lt;-p.new

    if(err&lt;tol) {
      break;
    }
  }

  ## compute posterior p
  clustlike&lt;-matrix(0,xrow,2)
  clustlike[,1]&lt;-log(1-p)+L0
  clustlike[,2]&lt;-log(p)+L1

  tempmax&lt;-apply(clustlike,1,max)
  for(j in 1:2) {
    clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
  }
  tempsum&lt;-apply(clustlike,1,sum)

  for(j in 1:2) {
    clustlike[,j]&lt;-clustlike[,j]/tempsum
  }

  p.post&lt;-matrix(0,xrow,xcol)
  for(i in 1:xcol) {
    p.post[,i]&lt;-clustlike[,2]
  }

  ## return

  #calculate back loglikelihood
  loglike.old&lt;-loglike.old-(log(p)+log(1-p))/xrow
  loglike.old&lt;-loglike.old*xrow
  result&lt;-list(p.post=p.post, motif.prior=p, loglike=loglike.old)
}

## Fit each dataset separately
cmfitsep&lt;-function(x, type, tol=1e-3, max.iter=100) {
  ## initialize
  xrow&lt;-nrow(x)
  xcol&lt;-ncol(x)
  loglike0&lt;-list()
  loglike1&lt;-list()
  p&lt;-0.01*rep(1,xcol)
  loglike.final&lt;-rep(0,xcol)

  ## compute loglikelihood
  for(i in 1:xcol) {
    f0&lt;-type[[i]][[1]]
    f0param&lt;-type[[i]][[2]]
    f1&lt;-type[[i]][[3]]
    f1param&lt;-type[[i]][[4]]
    loglike0[[i]]&lt;-f0(x[,i],f0param)
    loglike1[[i]]&lt;-f1(x[,i],f1param)
  }

  p.post&lt;-matrix(0,xrow,xcol)

  ## EM algorithm to get MLE of p
  for(coli in 1:xcol) {
    loglike.old &lt;- -1e10
    for(i.iter in 1:max.iter) {
      if((i.iter%%50) == 0) {
        print(paste(&quot;We have run the first &quot;, i.iter, &quot; iterations&quot;,sep=&quot;&quot;))
      }
      err&lt;-tol+1

      ## compute posterior cluster membership
      clustlike&lt;-matrix(0,xrow,2)
      clustlike[,1]&lt;-log(1-p[coli])+loglike0[[coli]]
      clustlike[,2]&lt;-log(p[coli])+loglike1[[coli]]

      tempmax&lt;-apply(clustlike,1,max)
      for(j in 1:2) {
        clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
      }
      tempsum&lt;-apply(clustlike,1,sum)

      ## evaluate whether the log.likelihood increases
      loglike.new&lt;-sum(tempmax+log(tempsum))/xrow

      ## update motif occurrence rate
      for(j in 1:2) {
        clustlike[,j]&lt;-clustlike[,j]/tempsum
      }

      p.new&lt;-(sum(clustlike[,2]))/(xrow)

      ## evaluate convergence
      err&lt;-abs(p.new-p[coli])/p[coli]
      loglike.old&lt;-loglike.new
      p[coli]&lt;-p.new

      if(err&lt;tol) {
        break;
      }
    }

    ## compute posterior p
    clustlike&lt;-matrix(0,xrow,2)
    clustlike[,1]&lt;-log(1-p[coli])+loglike0[[coli]]
    clustlike[,2]&lt;-log(p[coli])+loglike1[[coli]]

    tempmax&lt;-apply(clustlike,1,max)
    for(j in 1:2) {
      clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
    }
    tempsum&lt;-apply(clustlike,1,sum)

    for(j in 1:2) {
      clustlike[,j]&lt;-clustlike[,j]/tempsum
    }

    p.post[,coli]&lt;-clustlike[,2]
    loglike.final[coli]&lt;-loglike.old
  }


  ## return
  loglike.final&lt;-loglike.final*xrow
  result&lt;-list(p.post=p.post, motif.prior=p, loglike=loglike.final)
}

## Fit the full model
cmfitfull&lt;-function(x, type, tol=1e-3, max.iter=100) {
  ## initialize
  xrow&lt;-nrow(x)
  xcol&lt;-ncol(x)
  loglike0&lt;-list()
  loglike1&lt;-list()
  K&lt;-2^xcol
  p&lt;-rep(1,K)/K
  pattern&lt;-rep(0,xcol)
  patid&lt;-matrix(0,K,xcol)

  ## compute loglikelihood
  for(i in 1:xcol) {
    f0&lt;-type[[i]][[1]]
    f0param&lt;-type[[i]][[2]]
    f1&lt;-type[[i]][[3]]
    f1param&lt;-type[[i]][[4]]
    loglike0[[i]]&lt;-f0(x[,i],f0param)
    loglike1[[i]]&lt;-f1(x[,i],f1param)
  }
  L&lt;-matrix(0,xrow,K)
  for(i in 1:K)
  {
    patid[i,]&lt;-pattern
    for(j in 1:xcol) {
      if(pattern[j] &lt; 0.5) {
        L[,i]&lt;-L[,i]+loglike0[[j]]
      } else {
        L[,i]&lt;-L[,i]+loglike1[[j]]
      }
    }

    if(i &lt; K) {
      pattern[xcol]&lt;-pattern[xcol]+1
      j&lt;-xcol
      while(pattern[j] &gt; 1) {
        pattern[j]&lt;-0
        j&lt;-j-1
        pattern[j]&lt;-pattern[j]+1
      }
    }
  }

  ## EM algorithm to get MLE of p and q
  loglike.old &lt;- -1e10
  for(i.iter in 1:max.iter) {
    if((i.iter%%50) == 0) {
      print(paste(&quot;We have run the first &quot;, i.iter, &quot; iterations&quot;,sep=&quot;&quot;))
    }
    err&lt;-tol+1

    ## compute posterior cluster membership
    clustlike&lt;-matrix(0,xrow,K)
    for(j in 1:K) {
      clustlike[,j]&lt;-log(p[j])+L[,j]
    }

    tempmax&lt;-apply(clustlike,1,max)
    for(j in 1:K) {
      clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
    }
    tempsum&lt;-apply(clustlike,1,sum)

    ## update motif occurrence rate
    for(j in 1:K) {
      clustlike[,j]&lt;-clustlike[,j]/tempsum
    }

    p.new&lt;-(apply(clustlike,2,sum)+1)/(xrow+K)

    ## evaluate convergence
    err&lt;-max(abs(p.new-p)/p)

    ## evaluate whether the log.likelihood increases
    loglike.new&lt;-(sum(tempmax+log(tempsum))+sum(log(p.new)))/xrow

    loglike.old&lt;-loglike.new
    p&lt;-p.new

    if(err&lt;tol) {
      break;
    }
  }

  ## compute posterior p
  clustlike&lt;-matrix(0,xrow,K)
  for(j in 1:K) {
    clustlike[,j]&lt;-log(p[j])+L[,j]
  }

  tempmax&lt;-apply(clustlike,1,max)
  for(j in 1:K) {
    clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
  }
  tempsum&lt;-apply(clustlike,1,sum)

  for(j in 1:K) {
    clustlike[,j]&lt;-clustlike[,j]/tempsum
  }

  p.post&lt;-matrix(0,xrow,xcol)
  for(j in 1:K) {
    for(i in 1:xcol) {
      if(patid[j,i] &gt; 0.5) {
        p.post[,i]&lt;-p.post[,i]+clustlike[,j]
      }
    }
  }

  ## return
  #calculate back loglikelihood
  loglike.old&lt;-loglike.old-sum(log(p))/xrow
  loglike.old&lt;-loglike.old*xrow
  result&lt;-list(p.post=p.post, motif.prior=p, loglike=loglike.old)
}

generatetype&lt;-function(limfitted)
{
  jtype&lt;-list()
  df&lt;-limfitted$g1num+limfitted$g2num-2+limfitted$df0
  for(j in 1:limfitted$compnum)
  {
    jtype[[j]]&lt;-list(f0=modt.f0.loglike, f0.param=df[j], f1=modt.f1.loglike, f1.param=c(df[j],limfitted$g1num[j],limfitted$g2num[j],limfitted$v0[j]))
  }
  jtype
}

cormotiffit &lt;- function(exprs, groupid=NULL, compid=NULL, K=1, tol=1e-3,
                        max.iter=100, BIC=TRUE, norm.factor.method=&quot;TMM&quot;,
                        voom.normalize.method = &quot;none&quot;, runtype=c(&quot;logCPM&quot;,&quot;counts&quot;,&quot;limmafits&quot;), each=3)
{
  # first I want to do some typechecking. Input can be either a normalized
  # matrix, a count matrix, or a list of limma fits. Dispatch the correct
  # limmafit accordingly.
  # todo: add some typechecking here
  limfitted &lt;- list()
  if (runtype==&quot;counts&quot;) {
    limfitted &lt;- limmafit.counts(exprs,groupid,compid, norm.factor.method, voom.normalize.method)
  } else if (runtype==&quot;logCPM&quot;) {
    limfitted &lt;- limmafit.default(exprs,groupid,compid)
  } else if (runtype==&quot;limmafits&quot;) {
    limfitted &lt;- limmafit.list(exprs)
  } else {
    stop(&quot;runtype must be one of &#39;logCPM&#39;, &#39;counts&#39;, or &#39;limmafits&#39;&quot;)
  }


  jtype&lt;-generatetype(limfitted)
  fitresult&lt;-list()
  ks &lt;- rep(K, each = each)
  fitresult &lt;- bplapply(1:length(ks), function(i, x, type, ks, tol, max.iter) {
    cmfit.X(x, type, K = ks[i], tol = tol, max.iter = max.iter)
  }, x=limfitted$t, type=jtype, ks=ks, tol=tol, max.iter=max.iter)

  best.fitresults &lt;- list()
  for (i in 1:length(K)) {
    w.k &lt;- which(ks==K[i])
    this.bic &lt;- c()
    for (j in w.k) this.bic[j] &lt;- -2 * fitresult[[j]]$loglike + (K[i] - 1 + K[i] * limfitted$compnum) * log(dim(limfitted$t)[1])
    w.min &lt;- which(this.bic == min(this.bic, na.rm = TRUE))[1]
    best.fitresults[[i]] &lt;- fitresult[[w.min]]
  }
  fitresult &lt;- best.fitresults

  bic &lt;- rep(0, length(K))
  aic &lt;- rep(0, length(K))
  loglike &lt;- rep(0, length(K))
  for (i in 1:length(K)) loglike[i] &lt;- fitresult[[i]]$loglike
  for (i in 1:length(K)) bic[i] &lt;- -2 * fitresult[[i]]$loglike + (K[i] - 1 + K[i] * limfitted$compnum) * log(dim(limfitted$t)[1])
  for (i in 1:length(K)) aic[i] &lt;- -2 * fitresult[[i]]$loglike + 2 * (K[i] - 1 + K[i] * limfitted$compnum)
  if(BIC==TRUE) {
    bestflag=which(bic==min(bic))
  }
  else {
    bestflag=which(aic==min(aic))
  }
  result&lt;-list(bestmotif=fitresult[[bestflag]],bic=cbind(K,bic),
               aic=cbind(K,aic),loglike=cbind(K,loglike), allmotifs=fitresult)

}

cormotiffitall&lt;-function(exprs,groupid,compid, tol=1e-3, max.iter=100)
{
  limfitted&lt;-limmafit(exprs,groupid,compid)
  jtype&lt;-generatetype(limfitted)
  fitresult&lt;-cmfitall(limfitted$t,type=jtype,tol=1e-3,max.iter=max.iter)
}

cormotiffitsep&lt;-function(exprs,groupid,compid, tol=1e-3, max.iter=100)
{
  limfitted&lt;-limmafit(exprs,groupid,compid)
  jtype&lt;-generatetype(limfitted)
  fitresult&lt;-cmfitsep(limfitted$t,type=jtype,tol=1e-3,max.iter=max.iter)
}

cormotiffitfull&lt;-function(exprs,groupid,compid, tol=1e-3, max.iter=100)
{
  limfitted&lt;-limmafit(exprs,groupid,compid)
  jtype&lt;-generatetype(limfitted)
  fitresult&lt;-cmfitfull(limfitted$t,type=jtype,tol=1e-3,max.iter=max.iter)
}

plotIC&lt;-function(fitted_cormotif)
{
  oldpar&lt;-par(mfrow=c(1,2))
  plot(fitted_cormotif$bic[,1], fitted_cormotif$bic[,2], type=&quot;b&quot;,xlab=&quot;Motif Number&quot;, ylab=&quot;BIC&quot;, main=&quot;BIC&quot;)
  plot(fitted_cormotif$aic[,1], fitted_cormotif$aic[,2], type=&quot;b&quot;,xlab=&quot;Motif Number&quot;, ylab=&quot;AIC&quot;, main=&quot;AIC&quot;)
}

plotMotif&lt;-function(fitted_cormotif,title=&quot;&quot;)
{
  layout(matrix(1:2,ncol=2))
  u&lt;-1:dim(fitted_cormotif$bestmotif$motif.q)[2]
  v&lt;-1:dim(fitted_cormotif$bestmotif$motif.q)[1]
  image(u,v,t(fitted_cormotif$bestmotif$motif.q),
        col=gray(seq(from=1,to=0,by=-0.1)),xlab=&quot;Study&quot;,yaxt = &quot;n&quot;,
        ylab=&quot;Corr. Motifs&quot;,main=paste(title,&quot;pattern&quot;,sep=&quot; &quot;))
  axis(2,at=1:length(v))
  for(i in 1:(length(u)+1))
  {
    abline(v=(i-0.5))
  }
  for(i in 1:(length(v)+1))
  {
    abline(h=(i-0.5))
  }
  Ng=10000
  if(is.null(fitted_cormotif$bestmotif$p.post)!=TRUE)
    Ng=nrow(fitted_cormotif$bestmotif$p.post)
  genecount=floor(fitted_cormotif$bestmotif$motif.p*Ng)
  NK=nrow(fitted_cormotif$bestmotif$motif.q)
  plot(0,0.7,pch=&quot;.&quot;,xlim=c(0,1.2),ylim=c(0.75,NK+0.25),
       frame.plot=FALSE,axes=FALSE,xlab=&quot;No. of genes&quot;,ylab=&quot;&quot;, main=paste(title,&quot;frequency&quot;,sep=&quot; &quot;))
  segments(0,0.7,fitted_cormotif$bestmotif$motif.p[1],0.7)
  rect(0,1:NK-0.3,fitted_cormotif$bestmotif$motif.p,1:NK+0.3,
       col=&quot;dark grey&quot;)
  mtext(1:NK,at=1:NK,side=2,cex=0.8)
  text(fitted_cormotif$bestmotif$motif.p+0.15,1:NK,
       labels=floor(fitted_cormotif$bestmotif$motif.p*Ng))
}</code></pre>
</div>
<div id="load-required-libraries" class="section level2">
<h2><strong>📌 Load Required Libraries</strong></h2>
<pre class="r"><code>library(Cormotif)
library(Rfast)
library(dplyr)
library(BiocParallel)
library(gprofiler2)
library(ggplot2)</code></pre>
</div>
<div id="corrmotif-model-0.1-micromolar" class="section level2">
<h2><strong>📌 Corrmotif Model 0.1 Micromolar</strong></h2>
</div>
<div id="load-corrmotif-data" class="section level2">
<h2><strong>📌 Load Corrmotif Data</strong></h2>
<pre class="r"><code># Read the Corrmotif Results
Corrmotif &lt;- read.csv(&quot;data/Corrmotif/CX5461.csv&quot;)
Corrmotif_df &lt;- data.frame(Corrmotif)
rownames(Corrmotif_df) &lt;- Corrmotif_df$Gene

# Filter for 0.1 Concentration Only
exprs.corrmotif &lt;- as.matrix(Corrmotif_df[, grep(&quot;0.1&quot;, colnames(Corrmotif_df))])


# Read group and comparison IDs
groupid &lt;- read.csv(&quot;data/Corrmotif/groupid.csv&quot;)
groupid_df &lt;- data.frame(groupid[, grep(&quot;0.1&quot;, colnames(groupid))])


compid &lt;- read.csv(&quot;data/Corrmotif/Compid.csv&quot;)
compid_df &lt;- compid[compid$Cond1 %in% unique(as.numeric(groupid_df)) &amp; compid$Cond2 %in% unique(as.numeric(groupid_df)), ]</code></pre>
</div>
<div id="corrmotif-model-0.1-micromolar-k18" class="section level2">
<h2><strong>📌 Corrmotif Model 0.1 Micromolar (K=1:8)</strong></h2>
</div>
<div id="fit-corrmotif-model-k18-0.1-micromolar" class="section level2">
<h2><strong>📌 Fit Corrmotif Model (K=1:8) (0.1
Micromolar)</strong></h2>
<pre class="r"><code>set.seed(11111)
# Fit Corrmotif Model (K = 1 to 8)
set.seed(11111)
motif.fitted_0.1 &lt;- cormotiffit(
  exprs = exprs.corrmotif,
  groupid = groupid_df,
  compid = compid_df,
  K = 1:8,
  max.iter = 1000,
  BIC = TRUE,
  runtype = &quot;logCPM&quot;
)

gene_prob_0.1 &lt;- motif.fitted_0.1$bestmotif$p.post
rownames(gene_prob_0.1) &lt;- rownames(Corrmotif_df)
motif_prob_0.1 &lt;- motif.fitted_0.1$bestmotif$clustlike
rownames(motif_prob_0.1) &lt;- rownames(gene_prob_0.1)
write.csv(motif_prob_0.1,&quot;data/cormotif_probability_genelist_0.1.csv&quot;)</code></pre>
</div>
<div id="plot-motif-0.1-micromolar" class="section level2">
<h2><strong>📌 Plot motif (0.1 Micromolar)</strong></h2>
<pre class="r"><code>cormotif_0.1 &lt;- readRDS(&quot;data/Corrmotif/cormotif_0.1.RDS&quot;)

cormotif_0.1$bic</code></pre>
<pre><code>     K      bic
[1,] 1 291696.5
[2,] 2 284585.3
[3,] 3 283482.9
[4,] 4 283551.8
[5,] 5 283620.7
[6,] 6 283689.6
[7,] 7 283758.5
[8,] 8 283827.4</code></pre>
<pre class="r"><code>plotIC(cormotif_0.1)</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Model1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Model1-1">
Past versions of Model1-1.png
</button>
</p>
<div id="fig-Model1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/Model1-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/41cd1be487c4309fb8ab2f2d36b18ad63cdc31e4/docs/figure/Corrmotif_Conc.Rmd/Model1-1.png" target="_blank">41cd1be</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ce4b32557fc3c7ee693369db9edf0a9d22968203/docs/figure/Corrmotif_Conc.Rmd/Model1-1.png" target="_blank">ce4b325</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plotMotif(cormotif_0.1)</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Model1-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Model1-2">
Past versions of Model1-2.png
</button>
</p>
<div id="fig-Model1-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/Model1-2.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/41cd1be487c4309fb8ab2f2d36b18ad63cdc31e4/docs/figure/Corrmotif_Conc.Rmd/Model1-2.png" target="_blank">41cd1be</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ce4b32557fc3c7ee693369db9edf0a9d22968203/docs/figure/Corrmotif_Conc.Rmd/Model1-2.png" target="_blank">ce4b325</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="extract-gene-probabilities-0.1-micromolar"
class="section level2">
<h2><strong>📌 Extract Gene Probabilities (0.1 Micromolar)</strong></h2>
<pre class="r"><code># Extract posterior probabilities for genes
gene_prob_tran_0.1 &lt;- cormotif_0.1$bestmotif$p.post

rownames(gene_prob_tran_0.1) &lt;- rownames(Corrmotif_df)

# Define gene probability groups
prob_1_0.1  &lt;- rownames(gene_prob_tran_0.1[(gene_prob_tran_0.1[,1] &lt;0.5 &amp; gene_prob_tran_0.1[,2] &lt;0.5 &amp; gene_prob_tran_0.1[,3] &lt;0.5 &amp; gene_prob_tran_0.1[,4] &lt;0.5 &amp; gene_prob_tran_0.1[,5] &lt; 0.5 &amp; gene_prob_tran_0.1[,6]&lt;0.5),])
length(prob_1_0.1)</code></pre>
<pre><code>[1] 12308</code></pre>
<pre class="r"><code>prob_2_0.1  &lt;- rownames(gene_prob_tran_0.1[(gene_prob_tran_0.1[,1] &lt;0.5 &amp; gene_prob_tran_0.1[,2] &gt;0.5 &amp; gene_prob_tran_0.1[,3] &gt;0.5 &amp; gene_prob_tran_0.1[,4] &lt;0.5 &amp; gene_prob_tran_0.1[,5] &gt; 0.5 &amp; gene_prob_tran_0.1[,6]&gt;0.5),])
length(prob_2_0.1)</code></pre>
<pre><code>[1] 415</code></pre>
<pre class="r"><code>prob_3_0.1  &lt;- rownames(gene_prob_tran_0.1[(gene_prob_tran_0.1[,1] &lt;0.5 &amp; gene_prob_tran_0.1[,2] &lt;0.5 &amp; gene_prob_tran_0.1[,3] &lt;0.5 &amp; gene_prob_tran_0.1[,4] &lt;0.5 &amp; gene_prob_tran_0.1[,5] &gt; 0.5 &amp; gene_prob_tran_0.1[,6]&gt;0.5),])
length(prob_3_0.1)</code></pre>
<pre><code>[1] 1551</code></pre>
</div>
<div
id="distribution-of-gene-clusters-identified-by-corrmotif-0.1-micromolar"
class="section level2">
<h2><strong>📌 Distribution of Gene Clusters Identified by Corrmotif
(0.1 micromolar)</strong></h2>
<pre class="r"><code># Load necessary library
library(ggplot2)

# Data
data &lt;- data.frame(
  Category = c(&quot;Non response (0.1 µM)&quot;, &quot;CX-DOX mid-late response (0.1 µM)&quot;, &quot;DOX only mid-late (0.1 µM)&quot;),
  Value = c(12308, 415, 1551)
)

# Define custom colors
custom_colors &lt;- c(&quot;Non response (0.1 µM)&quot; = &quot;#FF9999&quot;,
                   &quot;CX-DOX mid-late response (0.1 µM)&quot; = &quot;#66B2FF&quot;,
                   &quot;DOX only mid-late (0.1 µM)&quot; = &quot;#99FF99&quot;)

# Create pie chart
ggplot(data, aes(x = &quot;&quot;, y = Value, fill = Category)) +
  geom_bar(width = 1, stat = &quot;identity&quot;) +
  coord_polar(&quot;y&quot;, start = 0) +
  geom_text(aes(label = Value),
            position = position_stack(vjust = 0.5),
            size = 4, color = &quot;black&quot;) +
  labs(title = &quot;Pie Chart (0.1 micromolar Corrmotif)&quot;, x = NULL, y = NULL) +
  theme_void() +
  scale_fill_manual(values = custom_colors)</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/pie-chart_0.1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pie-chart_0.1-1">
Past versions of pie-chart_0.1-1.png
</button>
</p>
<div id="fig-pie-chart_0.1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/pie-chart_0.1-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2de67e6a82f82edbc78941b11ba72ed8aba9964e/docs/figure/Corrmotif_Conc.Rmd/pie-chart_0.1-1.png" target="_blank">2de67e6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="save-corr-motif-datasets-for-gene-ontology-analysis-0.1-micromolar"
class="section level2">
<h2><strong>📌 Save Corr-motif datasets for Gene Ontology analysis (0.1
Micromolar)</strong></h2>
<pre class="r"><code>write.csv(data.frame(Entrez_ID = prob_1_0.1), &quot;data/prob_1_0.1.csv&quot;, row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_2_0.1), &quot;data/prob_2_0.1.csv&quot;, row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_3_0.1), &quot;data/prob_3_0.1.csv&quot;, row.names = FALSE)</code></pre>
</div>
<div id="micromolar-abs-logfc" class="section level2">
<h2><strong>📌 0.1 Micromolar abs logFC</strong></h2>
<pre class="r"><code># Load Required Libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# Load Datasets (Only 0.1 Micromolar)
CX_0.1_3  &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;)
CX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;)
CX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;)

DOX_0.1_3  &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;)
DOX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;)
DOX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;)

# Combine All 0.1 Micromolar Datasets into a Single Dataframe
all_toptables_0.1 &lt;- bind_rows(
  CX_0.1_3  %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  CX_0.1_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  CX_0.1_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;),
  DOX_0.1_3  %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  DOX_0.1_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  DOX_0.1_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;)
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.1$Entrez_ID &lt;- as.character(all_toptables_0.1$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.1 &lt;- all_toptables_0.1 %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 micromolar)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response\n(0.1 micromolar)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response\n(0.1 micromolar)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group))

# Compute Absolute logFC
all_toptables_0.1 &lt;- all_toptables_0.1 %&gt;%
  mutate(absFC = abs(logFC))

# Convert Factors for Proper Ordering (Reversed Order for Response Groups)
all_toptables_0.1 &lt;- all_toptables_0.1 %&gt;%
  mutate(
    Drug = factor(Drug, levels = c(&quot;CX.5461&quot;, &quot;DOX&quot;)),
    Timepoint = factor(Timepoint, levels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Response_Group = factor(Response_Group,
                            levels = c(
                              &quot;DOX only mid-late response\n(0.1 micromolar)&quot;,
                              &quot;CX-DOX mid-late response\n(0.1 micromolar)&quot;,
                              &quot;Non response\n(0.1 micromolar)&quot;  # Reversed Order
                            ))
  )

# **Plot the Boxplot with Faceted Labels Wrapping Correctly**
ggplot(all_toptables_0.1, aes(x = Drug, y = absFC, fill = Drug)) +
  geom_boxplot() +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +  # Custom color palette
  facet_grid(Response_Group ~ Timepoint, labeller = label_wrap_gen(width = 20)) +  # Ensure Proper Wrapping
  theme_bw() +
  labs(
    x = &quot;Drugs&quot;,
    y = &quot;|Log Fold Change|&quot;,
    title = &quot;|Log Fold| for 0.1 micromolar&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.line = element_line(linewidth = 1.5),
    strip.background = element_rect(fill = &quot;gray&quot;),  # Gray background for facet labels
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),  # Bold styling for facet labels
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 15)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.1abs%20logFC-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-0.1abs-logFC-1">
Past versions of "0.1abs logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-0.1abs-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/0.1abs logFC-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2990b21e59ad6434e6e7b4b92ae6cad8ff37f997/docs/figure/Corrmotif_Conc.Rmd/0.1abs logFC-1.png" target="_blank">2990b21</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.1abs logFC-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.1abs logFC-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="micromolar-mean-abs-logfc-across-timepoints"
class="section level2">
<h2><strong>📌 0.1 Micromolar mean (Abs logFC) across
timepoints</strong></h2>
<pre class="r"><code># Load Required Libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# Load Datasets (Only 0.1 Micromolar)
CX_0.1_3  &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;)
CX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;)
CX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;)

DOX_0.1_3  &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;)
DOX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;)
DOX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;)

# Combine All 0.1 Micromolar Datasets into a Single Dataframe
all_toptables_0.1 &lt;- bind_rows(
  CX_0.1_3  %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;3&quot;),
  CX_0.1_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;24&quot;),
  CX_0.1_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;48&quot;),
  DOX_0.1_3  %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;3&quot;),
  DOX_0.1_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;24&quot;),
  DOX_0.1_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;48&quot;)
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.1$Entrez_ID &lt;- as.character(all_toptables_0.1$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.1 &lt;- all_toptables_0.1 %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 micromolar)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response\n(0.1 micromolar)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response\n(0.1 micromolar)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group))

# Compute Mean Absolute logFC for Line Plot
data_summary &lt;- all_toptables_0.1 %&gt;%
  mutate(abs_logFC = abs(logFC)) %&gt;%  # Take absolute log fold change
  group_by(Response_Group, Drug, Timepoint) %&gt;%
  dplyr::summarize(mean_abs_logFC = mean(abs_logFC, na.rm = TRUE), .groups = &quot;drop&quot;) %&gt;%
  as.data.frame()

# **Ensure all timepoints exist in the summary**
timepoints_full &lt;- expand.grid(
  Response_Group = unique(all_toptables_0.1$Response_Group),
  Drug = unique(all_toptables_0.1$Drug),
  Timepoint = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;)
)

# **Merge to keep missing timepoints**
data_summary &lt;- full_join(timepoints_full, data_summary, by = c(&quot;Response_Group&quot;, &quot;Drug&quot;, &quot;Timepoint&quot;))

# **Replace NA mean_abs_logFC with 0 if no genes were present**
data_summary$mean_abs_logFC[is.na(data_summary$mean_abs_logFC)] &lt;- 0

# Convert Factors for Proper Ordering (Reversed Order for Response Groups)
data_summary &lt;- data_summary %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;), labels = c(&quot;3 hours&quot;, &quot;24 hours&quot;, &quot;48 hours&quot;)),
    Response_Group = factor(Response_Group, levels = c(
      &quot;DOX only mid-late response\n(0.1 micromolar)&quot;,
      &quot;CX-DOX mid-late response\n(0.1 micromolar)&quot;,
      &quot;Non response\n(0.1 micromolar)&quot;  # Reversed Order
    ))
  )

# Define custom drug palette
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)

# **Plot the Line Plot for Absolute logFC**
ggplot(data_summary, aes(x = Timepoint, y = mean_abs_logFC, group = Drug, color = Drug)) +
  geom_point(size = 3) +
  geom_line(size = 1.2) +
  scale_color_manual(values = drug_palette) +
  ylim(0, 2.5) +  # Adjust the Y-axis for better visualization
  facet_wrap(~ Response_Group, ncol = 1) +  # Facet by Response Group (Reversed Order)
  theme_bw() +
  labs(
    x = &quot;Timepoints&quot;,
    y = &quot;Mean |Log Fold Change|&quot;,
    title = &quot;Mean |Log Fold Change| Across Response Groups (0.1 micromolar)&quot;,
    color = &quot;Drug&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text = element_text(size = 12, color = &quot;black&quot;),
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.1%20mean-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-0.1-mean-1">
Past versions of "0.1 mean-1.png"
</button>
</p>
<div id="fig-no-spaces-0.1-mean-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/0.1 mean-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2990b21e59ad6434e6e7b4b92ae6cad8ff37f997/docs/figure/Corrmotif_Conc.Rmd/0.1 mean-1.png" target="_blank">2990b21</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.1 mean-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.1 mean-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="micromolar-logfc" class="section level2">
<h2><strong>📌 0.1 Micromolar logFC</strong></h2>
<pre class="r"><code># Load required libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# Load Datasets (Only 0.1 Micromolar)
CX_0.1_3  &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;)
CX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;)
CX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;)

DOX_0.1_3  &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;)
DOX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;)
DOX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;)

# Combine All 0.1 Micromolar Datasets into a Single Dataframe
all_toptables_0.1 &lt;- bind_rows(
  CX_0.1_3  %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  CX_0.1_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  CX_0.1_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;),
  DOX_0.1_3  %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  DOX_0.1_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  DOX_0.1_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;)
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.1$Entrez_ID &lt;- as.character(all_toptables_0.1$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.1 &lt;- all_toptables_0.1 %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 micromolar)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response\n(0.1 micromolar)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response\n(0.1 micromolar)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group))

# Convert factors to ensure correct ordering (Reversed Order for Response Groups)
all_toptables_0.1 &lt;- all_toptables_0.1 %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Response_Group = factor(Response_Group, levels = c(
      &quot;DOX only mid-late response\n(0.1 micromolar)&quot;,
      &quot;CX-DOX mid-late response\n(0.1 micromolar)&quot;,
      &quot;Non response\n(0.1 micromolar)&quot;  # Reversed Order
    ))
  )

# **Plot the Boxplot**
ggplot(all_toptables_0.1, aes(x = Drug, y = logFC, fill = Drug)) +
  geom_boxplot() +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  facet_grid(Response_Group ~ Timepoint) +
  theme_bw() +
  labs(x = &quot;Drugs&quot;, y = &quot;Log Fold Change&quot;, title = &quot;Log Fold Change for 0.1 Micromolar&quot;) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.1%20logFC-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-0.1-logFC-1">
Past versions of "0.1 logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-0.1-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/0.1 logFC-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2990b21e59ad6434e6e7b4b92ae6cad8ff37f997/docs/figure/Corrmotif_Conc.Rmd/0.1 logFC-1.png" target="_blank">2990b21</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.1 logFC-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.1 logFC-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="micromolar-mean-logfc-across-timepoints"
class="section level2">
<h2><strong>📌 0.1 Micromolar mean logFC across timepoints</strong></h2>
<pre class="r"><code># Compute Mean logFC for Line Plot
data_summary &lt;- all_toptables_0.1 %&gt;%
  group_by(Response_Group, Drug, Timepoint) %&gt;%
  dplyr::summarize(mean_logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;) %&gt;%
  as.data.frame()

# Convert factors to ensure correct ordering (Reversed Order for Response Groups)
data_summary &lt;- data_summary %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Response_Group = factor(Response_Group, levels = c(
      &quot;DOX only mid-late response\n(0.1 micromolar)&quot;,
      &quot;CX-DOX mid-late response\n(0.1 micromolar)&quot;,
      &quot;Non response\n(0.1 micromolar)&quot;  # Reversed Order
    ))
  )

# **Plot the Line Plot**
ggplot(data_summary, aes(x = Timepoint, y = mean_logFC, group = Drug, color = Drug)) +
  geom_point(size = 3) +
  geom_line(size = 1.2) +
  scale_color_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  ylim(-2, 1.5) +  # Adjust the Y-axis for better visualization
  facet_wrap(~ Response_Group, ncol = 1) +  # Facet by Response Group (Reversed Order)
  theme_bw() +
  labs(
    x = &quot;Timepoints&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    title = &quot;Mean Log Fold Change Across Response Groups (0.1 micromolar)&quot;,
    color = &quot;Drug&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text = element_text(size = 12, color = &quot;black&quot;),
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.1%20mean2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-0.1-mean2-1">
Past versions of "0.1 mean2-1.png"
</button>
</p>
<div id="fig-no-spaces-0.1-mean2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/0.1 mean2-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2990b21e59ad6434e6e7b4b92ae6cad8ff37f997/docs/figure/Corrmotif_Conc.Rmd/0.1 mean2-1.png" target="_blank">2990b21</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.1 mean2-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.1 mean2-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="corrmotif-model-0.5-micromolar" class="section level2">
<h2><strong>📌 Corrmotif Model 0.5 Micromolar</strong></h2>
</div>
<div id="load-corrmotif-data-1" class="section level2">
<h2><strong>📌 Load Corrmotif Data</strong></h2>
<pre class="r"><code># Read the Corrmotif Results
Corrmotif &lt;- read.csv(&quot;data/Corrmotif/CX5461.csv&quot;)
Corrmotif_df &lt;- data.frame(Corrmotif)
rownames(Corrmotif_df) &lt;- Corrmotif_df$Gene

# Filter for 0.5 Concentration Only
exprs.corrmotif &lt;- as.matrix(Corrmotif_df[, grep(&quot;0.5&quot;, colnames(Corrmotif_df))])


# Read group and comparison IDs
groupid &lt;- read.csv(&quot;data/Corrmotif/groupid.csv&quot;)
groupid_df &lt;- data.frame(groupid[, grep(&quot;0.5&quot;, colnames(groupid))])


compid &lt;- read.csv(&quot;data/Corrmotif/Compid.csv&quot;)
compid_df &lt;- compid[compid$Cond1 %in% unique(as.numeric(groupid_df)) &amp; compid$Cond2 %in% unique(as.numeric(groupid_df)), ]</code></pre>
</div>
<div id="corrmotif-model-0.5-micromolar-k18" class="section level2">
<h2><strong>📌 Corrmotif Model 0.5 Micromolar (K=1:8)</strong></h2>
</div>
<div id="fit-corrmotif-model-k18-0.5-micromolar" class="section level2">
<h2><strong>📌 Fit Corrmotif Model (K=1:8) (0.5
Micromolar)</strong></h2>
<pre class="r"><code># Fit Corrmotif Model (K = 1 to 8)
set.seed(11111)
motif.fitted_0.5 &lt;- cormotiffit(
  exprs = exprs.corrmotif,
  groupid = groupid_df,
  compid = compid_df,
  K = 1:8,
  max.iter = 1000,
  BIC = TRUE,
  runtype = &quot;logCPM&quot;
)

gene_prob_0.5 &lt;- motif.fitted_0.5$bestmotif$p.post
rownames(gene_prob_0.5) &lt;- rownames(Corrmotif_df)
motif_prob_0.5 &lt;- motif.fitted_0.5$bestmotif$clustlike
rownames(motif_prob_0.5) &lt;- rownames(gene_prob_0.5)
write.csv(motif_prob_0.5,&quot;data/cormotif_probability_genelist_0.5.csv&quot;)</code></pre>
</div>
<div id="plot-motif-0.5-micromolar" class="section level2">
<h2><strong>📌 Plot motif (0.5 Micromolar)</strong></h2>
<pre class="r"><code>cormotif_0.5 &lt;- readRDS(&quot;data/Corrmotif/cormotif_0.5.RDS&quot;)

cormotif_0.5$bic</code></pre>
<pre><code>     K      bic
[1,] 1 352140.7
[2,] 2 346785.8
[3,] 3 344812.9
[4,] 4 344860.1
[5,] 5 344751.9
[6,] 6 344820.8
[7,] 7 344889.7
[8,] 8 344966.6</code></pre>
<pre class="r"><code>plotIC(cormotif_0.5)</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Motif_0.5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Motif_0.5-1">
Past versions of Motif_0.5-1.png
</button>
</p>
<div id="fig-Motif_0.5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/Motif_0.5-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/41cd1be487c4309fb8ab2f2d36b18ad63cdc31e4/docs/figure/Corrmotif_Conc.Rmd/Motif_0.5-1.png" target="_blank">41cd1be</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plotMotif(cormotif_0.5)</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Motif_0.5-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Motif_0.5-2">
Past versions of Motif_0.5-2.png
</button>
</p>
<div id="fig-Motif_0.5-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/Motif_0.5-2.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/41cd1be487c4309fb8ab2f2d36b18ad63cdc31e4/docs/figure/Corrmotif_Conc.Rmd/Motif_0.5-2.png" target="_blank">41cd1be</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="extract-gene-probabilities-0.5-micromolar"
class="section level2">
<h2><strong>📌 Extract Gene Probabilities (0.5 Micromolar)</strong></h2>
<pre class="r"><code># Extract posterior probabilities for genes
gene_prob_tran_0.5 &lt;- cormotif_0.5$bestmotif$p.post

rownames(gene_prob_tran_0.5) &lt;- rownames(Corrmotif_df)

# Define gene probability groups
prob_1_0.5  &lt;- rownames(gene_prob_tran_0.5[(gene_prob_tran_0.5[,1] &lt;0.5 &amp; gene_prob_tran_0.5[,2] &lt;0.5 &amp; gene_prob_tran_0.5[,3] &lt;0.5 &amp; gene_prob_tran_0.5[,4] &lt;0.5 &amp; gene_prob_tran_0.5[,5] &lt; 0.5 &amp; gene_prob_tran_0.5[,6]&lt;0.5),])
length(prob_1_0.5)</code></pre>
<pre><code>[1] 7134</code></pre>
<pre class="r"><code>prob_2_0.5  &lt;- rownames(gene_prob_tran_0.5[(gene_prob_tran_0.5[,1] &lt;0.5 &amp; gene_prob_tran_0.5[,2] &lt;0.5 &amp; gene_prob_tran_0.5[,3] &lt;0.5 &amp; gene_prob_tran_0.5[,4] &gt;0.5 &amp; gene_prob_tran_0.5[,5] &gt; 0.5 &amp; gene_prob_tran_0.5[,6]&gt;=0.02),])
length(prob_2_0.5)</code></pre>
<pre><code>[1] 179</code></pre>
<pre class="r"><code>prob_3_0.5  &lt;- rownames(gene_prob_tran_0.5[(gene_prob_tran_0.5[,1] &lt;0.5 &amp; gene_prob_tran_0.5[,2] &lt;0.5 &amp; gene_prob_tran_0.5[,3] &lt;0.5 &amp; gene_prob_tran_0.5[,4] &lt;0.5 &amp; gene_prob_tran_0.5[,5] &gt; 0.5 &amp; gene_prob_tran_0.5[,6]&gt;0.5),])
length(prob_3_0.5)</code></pre>
<pre><code>[1] 6450</code></pre>
<pre class="r"><code>prob_4_0.5  &lt;- rownames(gene_prob_tran_0.5[(gene_prob_tran_0.5[,1] &gt;= 0.1 &amp; gene_prob_tran_0.5[,2] &gt; 0.5 &amp; gene_prob_tran_0.5[,3] &gt; 0.5 &amp; gene_prob_tran_0.5[,4] &gt;= 0.02 &amp; gene_prob_tran_0.5[,5] &lt; 0.5 &amp; gene_prob_tran_0.5[,6] &lt; 0.5),]) 

length(prob_4_0.5)</code></pre>
<pre><code>[1] 142</code></pre>
<pre class="r"><code>prob_5_0.5  &lt;- rownames(gene_prob_tran_0.5[(gene_prob_tran_0.5[,1] &lt;0.5 &amp; gene_prob_tran_0.5[,2] &gt;0.5 &amp; gene_prob_tran_0.5[,3] &gt;0.5 &amp; gene_prob_tran_0.5[,4] &gt;=0.02 &amp; gene_prob_tran_0.5[,4] &lt;0.5 &amp; gene_prob_tran_0.5[,5] &gt; 0.5 &amp; gene_prob_tran_0.5[,6]&gt;0.5),])
length(prob_5_0.5)</code></pre>
<pre><code>[1] 221</code></pre>
</div>
<div
id="distribution-of-gene-clusters-identified-by-corrmotif-0.5-micromolar"
class="section level2">
<h2><strong>📌 Distribution of Gene Clusters Identified by Corrmotif
(0.5 micromolar)</strong></h2>
<pre class="r"><code># Load necessary library
library(ggplot2)

# Data
data &lt;- data.frame(
  Category = c(&quot;Non response (0.5)&quot;, &quot;DOX-specific response (0.5 µM)&quot;, &quot;DOX only mid-late response (0.5 µM)&quot;, &quot;CX + DOX (early) response (0.5 µM)&quot;, &quot;DOX + CX (mid-late) response (0.5 µM)&quot;),
  Value = c(7134,179,6450,142,221)
)

# Add values to category names (to be displayed in the legend)
data$Category &lt;- paste0(data$Category, &quot; (&quot;, data$Value, &quot;)&quot;)

# Define custom colors with updated category names
custom_colors &lt;- setNames(
  c(&quot;#FF9999&quot;, &quot;#FF66CC&quot;, &quot;#66B2FF&quot;, &quot;#99FF99&quot;, &quot;#FFD700&quot;),
  data$Category  # Ensures color names match updated categories
)

# Create pie chart without number labels inside
ggplot(data, aes(x = &quot;&quot;, y = Value, fill = Category)) +
  geom_bar(width = 1, stat = &quot;identity&quot;) +
  coord_polar(&quot;y&quot;, start = 0) +
  labs(title = &quot;Pie Chart (0.5 micromolar Corrmotif)&quot;, x = NULL, y = NULL) +
  theme_void() +
  scale_fill_manual(values = custom_colors)</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/pie-chart_0.5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pie-chart_0.5-1">
Past versions of pie-chart_0.5-1.png
</button>
</p>
<div id="fig-pie-chart_0.5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/pie-chart_0.5-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2990b21e59ad6434e6e7b4b92ae6cad8ff37f997/docs/figure/Corrmotif_Conc.Rmd/pie-chart_0.5-1.png" target="_blank">2990b21</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/pie-chart_0.5-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2de67e6a82f82edbc78941b11ba72ed8aba9964e/docs/figure/Corrmotif_Conc.Rmd/pie-chart_0.5-1.png" target="_blank">2de67e6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="combined-pie-charts-for-concentrations" class="section level2">
<h2><strong>📌 Combined pie charts for concentrations</strong></h2>
<pre class="r"><code># Load necessary libraries
library(ggplot2)
library(dplyr)

# Data for 0.1 µM
data_0.1 &lt;- data.frame(
  Category = c(&quot;Non response&quot;, &quot;CX-DOX mid-late response&quot;, &quot;DOX only mid-late&quot;),
  Value = c(12308, 415, 1551),
  Concentration = &quot;0.1 µM&quot;
)

# Data for 0.5 µM
data_0.5 &lt;- data.frame(
  Category = c(&quot;Non response&quot;, &quot;DOX specific response&quot;, &quot;DOX only mid-late response&quot;, 
               &quot;CX + DOX (early) response&quot;, &quot;DOX + CX (mid-late) response&quot;),
  Value = c(7134, 179, 6450, 142, 221),
  Concentration = &quot;0.5 µM&quot;
)

# Combine both datasets
combined_data &lt;- bind_rows(data_0.1, data_0.5)

# Add values to category names (for legend only)
combined_data$Category_Legend &lt;- paste0(combined_data$Category, &quot; (&quot;, combined_data$Value, &quot;)&quot;)

# Define custom colors for updated categories
custom_colors &lt;- c(
  &quot;Non response (12308)&quot; = &quot;#FF9999&quot;,
  &quot;CX-DOX mid-late response (415)&quot; = &quot;#66B2FF&quot;,
  &quot;DOX only mid-late (1551)&quot; = &quot;#99FF99&quot;,
  &quot;Non response (7134)&quot; = &quot;#FF9999&quot;,
  &quot;DOX specific response (179)&quot; = &quot;#FF66CC&quot;,
  &quot;DOX only mid-late response (6450)&quot; = &quot;#99FF99&quot;,
  &quot;CX + DOX (early) response (142)&quot; = &quot;#FFD700&quot;,
  &quot;DOX + CX (mid-late) response (221)&quot; = &quot;#8A2BE2&quot;
)

# Ensure categories appear in a specific order
combined_data$Category_Legend &lt;- factor(combined_data$Category_Legend, levels = names(custom_colors))

# Create faceted pie charts without numbers inside the slices
pie_chart &lt;- ggplot(combined_data, aes(x = &quot;&quot;, y = Value, fill = Category_Legend)) +
  geom_bar(width = 1, stat = &quot;identity&quot;) +
  coord_polar(&quot;y&quot;, start = 0) +
  facet_wrap(~ Concentration) +  # Facet by concentration (side-by-side)
  labs(title = &quot;Corrmotif Pie Charts for 0.1 µM and 0.5 µM&quot;, x = NULL, y = NULL, fill = &quot;Category&quot;) +
  theme_void() +
  theme(
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.5),  # Adds box around facets
    strip.background = element_rect(fill = &quot;white&quot;, color = &quot;black&quot;, linewidth = 1),  # Box for facet titles
    strip.text = element_text(size = 12, face = &quot;bold&quot;, color = &quot;black&quot;),
    legend.title = element_text(size = 12, face = &quot;bold&quot;),  # Bold legend title
    legend.text = element_text(size = 10)  # Adjust legend text size
  ) +
  scale_fill_manual(values = custom_colors)

# Display the plot
print(pie_chart)</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/pie-chart_comb-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pie-chart_comb-1">
Past versions of pie-chart_comb-1.png
</button>
</p>
<div id="fig-pie-chart_comb-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/pie-chart_comb-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ef3a95166a7a18770eaa7786b80994b40bba9973/docs/figure/Corrmotif_Conc.Rmd/pie-chart_comb-1.png" target="_blank">ef3a951</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="save-corr-motif-datasets-for-gene-ontology-analysis-0.5-micromolar"
class="section level2">
<h2><strong>📌 Save Corr-motif datasets for Gene Ontology analysis (0.5
Micromolar)</strong></h2>
<pre class="r"><code>write.csv(data.frame(Entrez_ID = prob_1_0.5), &quot;data/prob_1_0.5.csv&quot;, row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_2_0.5), &quot;data/prob_2_0.5.csv&quot;, row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_3_0.5), &quot;data/prob_3_0.5.csv&quot;, row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_4_0.5), &quot;data/prob_4_0.5.csv&quot;, row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_5_0.5), &quot;data/prob_5_0.5.csv&quot;, row.names = FALSE)</code></pre>
</div>
<div id="micromolar-abs-logfc-1" class="section level2">
<h2><strong>📌 0.5 Micromolar abs logFC</strong></h2>
<pre class="r"><code># Load Response Groups from CSV Files
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# Load Datasets (Only 0.5 Micromolar)
CX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)

DOX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

# Convert datasets to DataFrames
Toptable_CX_0.5_3_df  &lt;- data.frame(CX_0.5_3)
Toptable_CX_0.5_24_df &lt;- data.frame(CX_0.5_24)
Toptable_CX_0.5_48_df &lt;- data.frame(CX_0.5_48)

Toptable_DOX_0.5_3_df  &lt;- data.frame(DOX_0.5_3)
Toptable_DOX_0.5_24_df &lt;- data.frame(DOX_0.5_24)
Toptable_DOX_0.5_48_df &lt;- data.frame(DOX_0.5_48)

# Combine All 0.5 Micromolar Datasets into a Single Dataframe
all_toptables_0.5 &lt;- bind_rows(
  Toptable_CX_0.5_3_df  %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;3&quot;),
  Toptable_CX_0.5_24_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;24&quot;),
  Toptable_CX_0.5_48_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;48&quot;),
  Toptable_DOX_0.5_3_df  %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;3&quot;),
  Toptable_DOX_0.5_24_df %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;24&quot;),
  Toptable_DOX_0.5_48_df %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;48&quot;)
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.5$Entrez_ID &lt;- as.character(all_toptables_0.5$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX + DOX (early) response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX + CX (mid-late) response\n(0.5 micromolar)&quot;,
      TRUE ~ NA_character_
    )
  )

# Remove NA Values (Genes Not in Response Groups)
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;% filter(!is.na(Response_Group))

# Compute Absolute logFC
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;%
  mutate(absFC = abs(logFC))

# Convert Factors for Proper Ordering (Reversed Order for Response Groups)
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;%
  mutate(
    Drug = factor(Drug, levels = c(&quot;CX.5461&quot;, &quot;DOX&quot;)),
    Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;),
                       labels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Response_Group = factor(Response_Group,
                            levels = c(&quot;DOX + CX (mid-late) response\n(0.5 micromolar)&quot;,
                                       &quot;CX + DOX (early) response\n(0.5 micromolar)&quot;,
                                       &quot;DOX only mid-late response\n(0.5 micromolar)&quot;,
                                       &quot;DOX-specific response\n(0.5 micromolar)&quot;,
                                       &quot;Non response\n(0.5 micromolar)&quot;)) # Reversed Order
  )

# **Plot the Boxplot with Faceted Labels Wrapping Correctly**
ggplot(all_toptables_0.5, aes(x = Drug, y = absFC, fill = Drug)) +
  geom_boxplot() +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +  # Custom color palette
  facet_grid(Response_Group ~ Timepoint, labeller = label_wrap_gen(width = 20)) +  # Ensure Proper Wrapping
  theme_bw() +
  labs(
    x = &quot;Drugs&quot;,
    y = &quot;|Log Fold Change|&quot;,
    title = &quot;|Log Fold| for 0.5 micromolar&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.line = element_line(linewidth = 1.5),
    strip.background = element_rect(fill = &quot;gray&quot;),  # Gray background for facet labels
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),  # Bold styling for facet labels
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 15)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.5abs%20logFC-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-0.5abs-logFC-1">
Past versions of "0.5abs logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-0.5abs-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/0.5abs logFC-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2990b21e59ad6434e6e7b4b92ae6cad8ff37f997/docs/figure/Corrmotif_Conc.Rmd/0.5abs logFC-1.png" target="_blank">2990b21</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.5abs logFC-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.5abs logFC-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="micromolar-mean-abs-logfc-across-timepoints-1"
class="section level2">
<h2><strong>📌 0.5 Micromolar mean (Abs logFC) across
timepoints</strong></h2>
<pre class="r"><code># Load required libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# Load Datasets (Only 0.5 Micromolar)
CX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)

DOX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

# Combine All 0.5 Micromolar Datasets into a Single Dataframe
all_toptables_0.5 &lt;- bind_rows(
  CX_0.5_3  %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;3&quot;),
  CX_0.5_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;24&quot;),
  CX_0.5_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;48&quot;),
  DOX_0.5_3  %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;3&quot;),
  DOX_0.5_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;24&quot;),
  DOX_0.5_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;48&quot;)
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.5$Entrez_ID &lt;- as.character(all_toptables_0.5$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX + DOX (early) response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX + CX (mid-late) response\n(0.5 micromolar)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group))  # Remove NA values

# Compute Mean Absolute logFC for Line Plot
data_summary &lt;- all_toptables_0.5 %&gt;%
  mutate(abs_logFC = abs(logFC)) %&gt;%
  group_by(Response_Group, Drug, Timepoint) %&gt;%
  dplyr::summarize(mean_abs_logFC = mean(abs_logFC, na.rm = TRUE), .groups = &quot;drop&quot;) %&gt;%
  as.data.frame()

# **Ensure all timepoints exist in the summary**
timepoints_full &lt;- expand.grid(
  Response_Group = unique(all_toptables_0.5$Response_Group),
  Drug = unique(all_toptables_0.5$Drug),
  Timepoint = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;)
)

# **Merge to keep missing timepoints**
data_summary &lt;- full_join(timepoints_full, data_summary, by = c(&quot;Response_Group&quot;, &quot;Drug&quot;, &quot;Timepoint&quot;))

# **Replace NA mean_abs_logFC with 0 if no genes were present**
data_summary$mean_abs_logFC[is.na(data_summary$mean_abs_logFC)] &lt;- 0

# Convert Factors for Proper Ordering (Reversed Order for Response Groups)
data_summary &lt;- data_summary %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;), labels = c(&quot;3 hours&quot;, &quot;24 hours&quot;, &quot;48 hours&quot;)),
    Response_Group = factor(Response_Group, levels = c(
      &quot;DOX + CX (mid-late) response\n(0.5 micromolar)&quot;,
                                       &quot;CX + DOX (early) response\n(0.5 micromolar)&quot;,
                                       &quot;DOX only mid-late response\n(0.5 micromolar)&quot;,
                                       &quot;DOX-specific response\n(0.5 micromolar)&quot;,
                                       &quot;Non response\n(0.5 micromolar)&quot;  # Reversed order
    ))
  )

# Define custom drug palette
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)

# **Plot the Line Plot for Mean Absolute logFC**
ggplot(data_summary, aes(x = Timepoint, y = mean_abs_logFC, group = Drug, color = Drug)) +
  geom_point(size = 3) +
  geom_line(size = 1.2) +
  scale_color_manual(values = drug_palette) +
  ylim(0, 2.5) +  # Adjust the Y-axis for better visualization
  facet_wrap(~ Response_Group, ncol = 1) +  # Facet by Response Group (Reversed Order)
  theme_bw() +
  labs(
    x = &quot;Timepoints&quot;,
    y = &quot;Mean |Log Fold Change|&quot;,
    title = &quot;Mean |Log Fold Change| Across Response Groups (0.5 micromolar)&quot;,
    color = &quot;Drug&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text = element_text(size = 12, color = &quot;black&quot;),
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.5%20mean-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-0.5-mean-1">
Past versions of "0.5 mean-1.png"
</button>
</p>
<div id="fig-no-spaces-0.5-mean-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/0.5 mean-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2990b21e59ad6434e6e7b4b92ae6cad8ff37f997/docs/figure/Corrmotif_Conc.Rmd/0.5 mean-1.png" target="_blank">2990b21</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.5 mean-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.5 mean-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="micromolar-logfc-1" class="section level2">
<h2><strong>📌 0.5 Micromolar logFC</strong></h2>
<pre class="r"><code># Load required libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# Load Datasets (Only 0.5 Micromolar)
CX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)

DOX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

# Combine All 0.5 Micromolar Datasets into a Single Dataframe
all_toptables_0.5 &lt;- bind_rows(
  CX_0.5_3  %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  CX_0.5_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  CX_0.5_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;),
  DOX_0.5_3  %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  DOX_0.5_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  DOX_0.5_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;)
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.5$Entrez_ID &lt;- as.character(all_toptables_0.5$Entrez_ID)

# Assign Response Groups
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX + DOX (early) response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX + CX (mid-late) response\n(0.5 micromolar)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group))

# Convert factors to ensure correct ordering (Reversed Order for Response Groups)
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Response_Group = factor(Response_Group, levels = c(
      &quot;DOX + CX (mid-late) response\n(0.5 micromolar)&quot;,
                                       &quot;CX + DOX (early) response\n(0.5 micromolar)&quot;,
                                       &quot;DOX only mid-late response\n(0.5 micromolar)&quot;,
                                       &quot;DOX-specific response\n(0.5 micromolar)&quot;,
                                       &quot;Non response\n(0.5 micromolar)&quot;  # Reversed Order
    ))
  )

# **Plot the Boxplot**
ggplot(all_toptables_0.5, aes(x = Drug, y = logFC, fill = Drug)) +
  geom_boxplot() +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  facet_grid(Response_Group ~ Timepoint) +
  theme_bw() +
  labs(x = &quot;Drugs&quot;, y = &quot;Log Fold Change&quot;, title = &quot;Log Fold Change for 0.5 Micromolar&quot;) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.5logFC-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-0.5logFC-1">
Past versions of 0.5logFC-1.png
</button>
</p>
<div id="fig-0.5logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/0.5logFC-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2990b21e59ad6434e6e7b4b92ae6cad8ff37f997/docs/figure/Corrmotif_Conc.Rmd/0.5logFC-1.png" target="_blank">2990b21</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.5logFC-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.5logFC-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="micromolar-mean-logfc-across-timepoints-1"
class="section level2">
<h2><strong>📌 0.5 Micromolar mean logFC across timepoints</strong></h2>
<pre class="r"><code># Load Required Libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# Load Datasets (Only 0.5 Micromolar)
CX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)

DOX_0.5_3  &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

# Combine All 0.5 Micromolar Datasets into a Single Dataframe
all_toptables_0.5 &lt;- bind_rows(
  CX_0.5_3  %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  CX_0.5_24 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  CX_0.5_48 %&gt;% mutate(Drug = &quot;CX.5461&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;),
  DOX_0.5_3  %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 3 hours&quot;),
  DOX_0.5_24 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 24 hours&quot;),
  DOX_0.5_48 %&gt;% mutate(Drug = &quot;DOX&quot;, Timepoint = &quot;Timepoint: 48 hours&quot;)
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.5$Entrez_ID &lt;- as.character(all_toptables_0.5$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.5 &lt;- all_toptables_0.5 %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX + DOX (early) response\n(0.5 micromolar)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX + CX (mid-late) response\n(0.5 micromolar)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group))

# Compute Mean logFC for Line Plot
data_summary_0.5 &lt;- all_toptables_0.5 %&gt;%
  group_by(Response_Group, Drug, Timepoint) %&gt;%
  dplyr::summarize(mean_logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;) %&gt;%
  as.data.frame()

# **Ensure all timepoints exist in the summary**
timepoints_full &lt;- expand.grid(
  Response_Group = unique(all_toptables_0.5$Response_Group),
  Drug = unique(all_toptables_0.5$Drug),
  Timepoint = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)
)

# **Merge to keep missing timepoints**
data_summary_0.5 &lt;- full_join(timepoints_full, data_summary_0.5, by = c(&quot;Response_Group&quot;, &quot;Drug&quot;, &quot;Timepoint&quot;))

# **Replace NA mean_logFC with 0 if no genes were present**
data_summary_0.5$mean_logFC[is.na(data_summary_0.5$mean_logFC)] &lt;- 0

# Convert Factors for Proper Ordering (Reversed Order for Response Groups)
data_summary_0.5 &lt;- data_summary_0.5 %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Response_Group = factor(Response_Group, levels = c(
      &quot;DOX + CX (mid-late) response\n(0.5 micromolar)&quot;,
                                       &quot;CX + DOX (early) response\n(0.5 micromolar)&quot;,
                                       &quot;DOX only mid-late response\n(0.5 micromolar)&quot;,
                                       &quot;DOX-specific response\n(0.5 micromolar)&quot;,
                                       &quot;Non response\n(0.5 micromolar)&quot;  # Reversed Order
    ))
  )

# Define custom drug palette
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)

# **Plot the Line Plot for Mean logFC**
ggplot(data_summary_0.5, aes(x = Timepoint, y = mean_logFC, group = Drug, color = Drug)) +
  geom_point(size = 3) +
  geom_line(size = 1.2) +
  scale_color_manual(values = drug_palette) +
  ylim(-2, 1.5) +  # Adjust the Y-axis for better visualization
  facet_wrap(~ Response_Group, ncol = 1) +  # Facet by Response Group (Reversed Order)
  theme_bw() +
  labs(
    x = &quot;Timepoints&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    title = &quot;Mean Log Fold Change Across Response Groups (0.5 micromolar)&quot;,
    color = &quot;Drug&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text = element_text(size = 12, color = &quot;black&quot;),
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/0.5%20mean2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-0.5-mean2-1">
Past versions of "0.5 mean2-1.png"
</button>
</p>
<div id="fig-no-spaces-0.5-mean2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/0.5 mean2-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/2990b21e59ad6434e6e7b4b92ae6cad8ff37f997/docs/figure/Corrmotif_Conc.Rmd/0.5 mean2-1.png" target="_blank">2990b21</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91a7ce4cfdf8b7a23eb6abfbe1464ed273d6e346/docs/figure/Corrmotif_Conc.Rmd/0.5 mean2-1.png" target="_blank">91a7ce4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d9ff8536d5445d0044cb44f5641eaa1d86e5c8a7/docs/figure/Corrmotif_Conc.Rmd/0.5 mean2-1.png" target="_blank">d9ff853</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="proportion-of-dna-damage-repair-genes-0.1-micromolar-md-anderson-dataset"
class="section level2">
<h2><strong>📌 Proportion of DNA Damage Repair Genes (0.1 micromolar)
(MD Anderson dataset)</strong></h2>
<pre class="r"><code># Load necessary libraries
library(dplyr)
library(ggplot2)
library(tidyr)</code></pre>
<pre><code>Warning: package &#39;tidyr&#39; was built under R version 4.3.3</code></pre>
<pre class="r"><code>library(org.Hs.eg.db)</code></pre>
<pre><code>Warning: package &#39;AnnotationDbi&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;IRanges&#39; was built under R version 4.3.1</code></pre>
<pre><code>Warning: package &#39;S4Vectors&#39; was built under R version 4.3.2</code></pre>
<pre class="r"><code># **🔹 Read DNA Damage Repair Gene List**
DNA_damage &lt;- read.csv(&quot;data/DNA_Damage.csv&quot;, stringsAsFactors = FALSE)

# Convert gene symbols to Entrez IDs
DNA_damage &lt;- DNA_damage %&gt;%
  mutate(Entrez_ID = mapIds(org.Hs.eg.db,
                            keys = DNA_damage$Symbol,
                            column = &quot;ENTREZID&quot;,
                            keytype = &quot;SYMBOL&quot;,
                            multiVals = &quot;first&quot;))

DNA_damage_genes &lt;- na.omit(DNA_damage$Entrez_ID)

# **🔹 Load Corrmotif Groups for 0.1 Concentration**
prob_groups_0.1 &lt;- list(
  &quot;Non Response (0.1)&quot; = read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID,
   &quot;CX_DOX mid-late (0.1)&quot; = read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID,
  &quot;DOX only mid-late (0.1)&quot;= read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID
)

# **🔹 Create Dataframe for Corrmotif Groups**
corrmotif_df_0.1 &lt;- bind_rows(
  lapply(prob_groups_0.1, function(ids) {
    data.frame(Entrez_ID = ids)
  }),
  .id = &quot;Response_Group&quot;
)

# **🔹 Match Entrez_IDs with DNA Damage Repair Genes**
corrmotif_df_0.1 &lt;- corrmotif_df_0.1 %&gt;%
  mutate(DNA_Damage = ifelse(Entrez_ID %in% DNA_damage_genes, &quot;Yes&quot;, &quot;No&quot;))

# **🔹 Count DNA Damage Repair Genes in Each Response Group**
proportion_data &lt;- corrmotif_df_0.1 %&gt;%
  group_by(Response_Group, DNA_Damage) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Response_Group) %&gt;%
  mutate(Percentage = (Count / sum(Count)) * 100)

# **🔹 Ensure &quot;Yes&quot; is at the Bottom and &quot;No&quot; at the Top**
proportion_data$DNA_Damage &lt;- factor(proportion_data$DNA_Damage, levels = c(&quot;Yes&quot;, &quot;No&quot;))

# **🔹 Set Order of Response Groups for X-axis**
response_order &lt;- c(&quot;Non Response (0.1)&quot;, &quot;CX_DOX mid-late (0.1)&quot;,&quot;DOX only mid-late (0.1)&quot;)
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = response_order)

# **🔹 Perform Chi-Square Tests for &quot;DOX only mid-late (0.1)&quot; and &quot;CX_DOX mid-late (0.1)&quot; vs &quot;Non Response (0.1)&quot;**
non_response_counts &lt;- proportion_data %&gt;%
  filter(Response_Group == &quot;Non Response (0.1)&quot;) %&gt;%
  dplyr::select(DNA_Damage, Count) %&gt;%
  {setNames(.$Count, .$DNA_Damage)}  # Convert to named vector

chi_results &lt;- proportion_data %&gt;%
  filter(Response_Group %in% c(&quot;CX_DOX mid-late (0.1)&quot;,&quot;DOX only mid-late (0.1)&quot;)) %&gt;%
  group_by(Response_Group) %&gt;%
  summarise(
    p_value = {
      group_counts &lt;- Count[DNA_Damage %in% c(&quot;Yes&quot;, &quot;No&quot;)]
      if (!&quot;Yes&quot; %in% DNA_Damage) group_counts &lt;- c(group_counts, 0)
      if (!&quot;No&quot; %in% DNA_Damage) group_counts &lt;- c(0, group_counts)

      contingency_table &lt;- matrix(c(
        group_counts[1], group_counts[2],
        non_response_counts[&quot;Yes&quot;], non_response_counts[&quot;No&quot;]
      ), nrow = 2, byrow = TRUE)

      # Perform chi-square test if all values are valid
      if (all(contingency_table &gt;= 0 &amp; is.finite(contingency_table))) {
        chisq.test(contingency_table)$p.value
      } else {
        NA
      }
    },
    .groups = &quot;drop&quot;
  ) %&gt;%
  mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))

# **🔹 Merge Chi-Square Results into Proportion Data**
proportion_data &lt;- proportion_data %&gt;%
  left_join(chi_results %&gt;% dplyr::select(Response_Group, Significance), by = &quot;Response_Group&quot;)

# **🔹 Set Star Position Uniform Across Groups at 105%**
star_positions &lt;- data.frame(
  Response_Group = c(&quot;CX_DOX mid-late (0.1)&quot;, &quot;DOX only mid-late (0.1)&quot;),
  y_pos = 105,  # Fixed at 105% of Y-axis
  Significance = chi_results$Significance
)

# **🔹 Generate Proportion Plot with Chi-Square Stars**
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = DNA_Damage)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +  # Stacked bars
  geom_text(
    data = star_positions,
    aes(x = Response_Group, y = y_pos, label = Significance),  # Place stars at fixed 105%
    inherit.aes = FALSE,
    size = 6, color = &quot;black&quot;, fontface = &quot;bold&quot;, vjust = 0  # Keeps stars aligned
  ) +
  scale_y_continuous(labels = scales::percent_format(scale = 1), limits = c(0, 110)) +  # Fixed Y-axis to 100%
  scale_fill_manual(values = c(&quot;Yes&quot; = &quot;#e41a1c&quot;, &quot;No&quot; = &quot;#377eb8&quot;)) +  # Yes (Red), No (Blue)
  labs(
    title = &quot;Proportion of DNA Damage Repair Genes in\n0.1 Corrmotif Response Groups&quot;,
    x = &quot;Response Groups (0.1 Concentration)&quot;,
    y = &quot;Percentage&quot;,
    fill = &quot;DNA Damage Repair&quot;
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
    legend.title = element_blank(),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    strip.background = element_blank(),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/prop%20DNA-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-prop-DNA-1">
Past versions of "prop DNA-1.png"
</button>
</p>
<div id="fig-no-spaces-prop-DNA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/prop DNA-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/81247f64f08df43c2b40a386707fc8e9cc240143/docs/figure/Corrmotif_Conc.Rmd/prop DNA-1.png" target="_blank">81247f6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/96d0db145cf68e6db0d39e70bad9af0f75793aae/docs/figure/Corrmotif_Conc.Rmd/prop DNA-1.png" target="_blank">96d0db1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="proportion-of-dna-damage-repair-genes-0.5-micromolar-md-anderson-dataset"
class="section level2">
<h2><strong>📌 Proportion of DNA Damage Repair Genes (0.5 micromolar)
(MD Anderson dataset)</strong></h2>
<pre class="r"><code># Load necessary libraries
library(dplyr)
library(ggplot2)
library(tidyr)
library(org.Hs.eg.db)

# **🔹 Read DNA Damage Repair Gene List**
DNA_damage &lt;- read.csv(&quot;data/DNA_Damage.csv&quot;, stringsAsFactors = FALSE)

# Convert gene symbols to Entrez IDs
DNA_damage &lt;- DNA_damage %&gt;%
  mutate(Entrez_ID = mapIds(org.Hs.eg.db,
                            keys = DNA_damage$Symbol,
                            column = &quot;ENTREZID&quot;,
                            keytype = &quot;SYMBOL&quot;,
                            multiVals = &quot;first&quot;))

DNA_damage_genes &lt;- na.omit(DNA_damage$Entrez_ID)

# **🔹 Load Corrmotif Groups for 0.5 Concentration**
prob_groups_0.5 &lt;- list(
  &quot;Non Response (0.5)&quot; = read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID,
  &quot;DOX-specific response (0.5)&quot; = read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID,
  &quot;DOX only mid-late response (0.5)&quot; = read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID,
  &quot;CX DOX (early) response (0.5)&quot; = read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID,
  &quot;DOX + CX (mid-late) response (0.5)&quot; = read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID
)

# **🔹 Create Dataframe for Corrmotif Groups**
corrmotif_df_0.5 &lt;- bind_rows(
  lapply(prob_groups_0.5, function(ids) {
    data.frame(Entrez_ID = ids)
  }),
  .id = &quot;Response_Group&quot;
)

# **🔹 Match Entrez_IDs with DNA Damage Repair Genes**
corrmotif_df_0.5 &lt;- corrmotif_df_0.5 %&gt;%
  mutate(DNA_Damage = ifelse(Entrez_ID %in% DNA_damage_genes, &quot;Yes&quot;, &quot;No&quot;))

# **🔹 Count DNA Damage Repair Genes in Each Response Group**
proportion_data &lt;- corrmotif_df_0.5 %&gt;%
  group_by(Response_Group, DNA_Damage) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Response_Group) %&gt;%
  mutate(Percentage = (Count / sum(Count)) * 100)

# **🔹 Ensure &quot;Yes&quot; is at the Bottom and &quot;No&quot; at the Top**
proportion_data$DNA_Damage &lt;- factor(proportion_data$DNA_Damage, levels = c(&quot;Yes&quot;, &quot;No&quot;))

# **🔹 Set Order of Response Groups for X-axis**
response_order &lt;- c(&quot;Non Response (0.5)&quot;, &quot;DOX-specific response (0.5)&quot;, &quot;DOX only mid-late response (0.5)&quot;, 
                    &quot;CX DOX (early) response (0.5)&quot;, &quot;DOX + CX (mid-late) response (0.5)&quot;)
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = response_order)

# **🔹 Perform Chi-Square Tests for Each Response Group vs Non-Response**
non_response_counts &lt;- proportion_data %&gt;%
  filter(Response_Group == &quot;Non Response (0.5)&quot;) %&gt;%
  dplyr::select(DNA_Damage, Count) %&gt;%
  {setNames(.$Count, .$DNA_Damage)}  # Convert to named vector

# **Comparing Each Group Against &quot;Non Response (0.5)&quot;**
chi_results &lt;- proportion_data %&gt;%
  filter(Response_Group %in% c(&quot;DOX-specific response (0.5)&quot;, &quot;DOX only mid-late response (0.5)&quot;, 
                               &quot;CX DOX (early) response (0.5)&quot;, &quot;DOX + CX (mid-late) response (0.5)&quot;)) %&gt;%
  group_by(Response_Group) %&gt;%
  summarise(
    p_value = {
      group_counts &lt;- Count[DNA_Damage %in% c(&quot;Yes&quot;, &quot;No&quot;)]
      if (!&quot;Yes&quot; %in% DNA_Damage) group_counts &lt;- c(group_counts, 0)
      if (!&quot;No&quot; %in% DNA_Damage) group_counts &lt;- c(0, group_counts)

      contingency_table &lt;- matrix(c(
        group_counts[1], group_counts[2],  # Response group counts
        non_response_counts[&quot;Yes&quot;], non_response_counts[&quot;No&quot;]  # Non-response counts
      ), nrow = 2, byrow = TRUE)

      # Perform chi-square test if all values are valid
      if (all(contingency_table &gt;= 0 &amp; is.finite(contingency_table))) {
        chisq.test(contingency_table)$p.value
      } else {
        NA
      }
    },
    .groups = &quot;drop&quot;
  ) %&gt;%
  mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))

# **🔹 Merge Chi-Square Results into Proportion Data**
proportion_data &lt;- proportion_data %&gt;%
  left_join(chi_results %&gt;% dplyr::select(Response_Group, Significance), by = &quot;Response_Group&quot;)

# **🔹 Set Star Position Uniform Across Groups at 105%**
star_positions &lt;- data.frame(
  Response_Group = c(&quot;DOX-specific response (0.5)&quot;, &quot;DOX only mid-late response (0.5)&quot;, 
                     &quot;CX DOX (early) response (0.5)&quot;, &quot;DOX + CX (mid-late) response (0.5)&quot;),
  y_pos = 105,  # Fixed at 105% of Y-axis
  Significance = chi_results$Significance
)

# **🔹 Generate Proportion Plot with Chi-Square Stars**
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = DNA_Damage)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +  # Stacked bars
  geom_text(
    data = star_positions,
    aes(x = Response_Group, y = y_pos, label = Significance),  # Place stars at fixed 105%
    inherit.aes = FALSE,
    size = 6, color = &quot;black&quot;, fontface = &quot;bold&quot;, vjust = 0  # Keeps stars aligned
  ) +
  scale_y_continuous(labels = scales::percent_format(scale = 1), limits = c(0, 110)) +  # **Y-axis now limited to 110% for visibility**
  scale_fill_manual(values = c(&quot;Yes&quot; = &quot;#e41a1c&quot;, &quot;No&quot; = &quot;#377eb8&quot;)) +  # Yes (Red), No (Blue)
  labs(
    title = &quot;Proportion of DNA Damage Repair Genes in\n0.5 Corrmotif Response Groups&quot;,
    x = &quot;Response Groups (0.5 Concentration)&quot;,
    y = &quot;Percentage&quot;,
    fill = &quot;DNA Damage Repair Genes&quot;
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
    legend.title = element_blank(),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    strip.background = element_blank(),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/prop%20DNA%200.5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-prop-DNA-0.5-1">
Past versions of "prop DNA 0.5-1.png"
</button>
</p>
<div id="fig-no-spaces-prop-DNA-0.5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/prop DNA 0.5-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/96d0db145cf68e6db0d39e70bad9af0f75793aae/docs/figure/Corrmotif_Conc.Rmd/prop DNA 0.5-1.png" target="_blank">96d0db1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="proportion-of-dna-damage-repair-genes-combined-md-anderson-dataset"
class="section level2">
<h2><strong>📌 Proportion of DNA Damage Repair Genes (Combined) (MD
Anderson dataset)</strong></h2>
<pre class="r"><code># Load Required Libraries
library(dplyr)
library(ggplot2)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load DNA Damage Genes -----------------
DNA_damage &lt;- read.csv(&quot;data/DNA_Damage.csv&quot;, stringsAsFactors = FALSE)

DNA_damage$Entrez_ID &lt;- mapIds(
  org.Hs.eg.db,
  keys = DNA_damage$Symbol,
  column = &quot;ENTREZID&quot;,
  keytype = &quot;SYMBOL&quot;,
  multiVals = &quot;first&quot;
)
dna_damage_ids &lt;- na.omit(DNA_damage$Entrez_ID)

# ----------------- Load CorrMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% dna_damage_ids, &quot;Yes&quot;, &quot;No&quot;),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% dna_damage_ids, &quot;Yes&quot;, &quot;No&quot;),
    Concentration = &quot;0.5 µM&quot;
  )

# ----------------- Combine All -----------------
df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Calculate Proportions -----------------
proportion_data &lt;- df_combined %&gt;%
  group_by(Concentration, Response_Group, Category) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Set Order: &quot;Yes&quot; at bottom -----------------
proportion_data$Category &lt;- factor(proportion_data$Category, levels = c(&quot;Yes&quot;, &quot;No&quot;))

# ----------------- Chi-square or Fisher&#39;s Test -----------------
get_chi_or_fisher &lt;- function(df, ref_group) {
  ref_counts &lt;- df %&gt;%
    ungroup() %&gt;%
    filter(Response_Group == ref_group) %&gt;%
    dplyr::select(Category, Count) %&gt;%
    {setNames(.$Count, .$Category)}

  df %&gt;%
    filter(Response_Group != ref_group) %&gt;%
    group_by(Response_Group) %&gt;%
    summarise(
      p_value = {
        group_counts &lt;- Count[Category %in% c(&quot;Yes&quot;, &quot;No&quot;)]
        if (length(group_counts) &lt; 2) group_counts &lt;- c(group_counts, 0)
        contingency_table &lt;- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts[&quot;Yes&quot;], ref_counts[&quot;No&quot;]
        ), nrow = 2)

        if (any(contingency_table &lt; 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = &quot;drop&quot;
    ) %&gt;%
    mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))
}

# Perform tests
chi_0.1 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.1 µM&quot;), &quot;Non response&quot;)
chi_0.5 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.5 µM&quot;), &quot;Non response&quot;)

chi_all &lt;- bind_rows(
  chi_0.1 %&gt;% mutate(Concentration = &quot;0.1 µM&quot;),
  chi_0.5 %&gt;% mutate(Concentration = &quot;0.5 µM&quot;)
)

proportion_data &lt;- proportion_data %&gt;%
  left_join(chi_all, by = c(&quot;Concentration&quot;, &quot;Response_Group&quot;))

# ----------------- Factor Levels -----------------
proportion_data$Concentration &lt;- factor(proportion_data$Concentration, levels = c(&quot;0.1 µM&quot;, &quot;0.5 µM&quot;))
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
))

# ----------------- Star Positions -----------------
label_data &lt;- proportion_data %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  summarise(Significance = dplyr::first(Significance), .groups = &quot;drop&quot;) %&gt;%
  filter(Significance == &quot;*&quot;) %&gt;%
  mutate(y_pos = 105)

# ----------------- Stack Order -----------------
proportion_data &lt;- proportion_data %&gt;%
  arrange(Concentration, Response_Group, Category)

# ----------------- Plot -----------------
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(
    data = label_data,
    aes(x = Response_Group, y = y_pos, label = Significance),
    inherit.aes = FALSE,
    size = 5, color = &quot;black&quot;
  ) +
  facet_wrap(~ Concentration, scales = &quot;free_x&quot;) +
  scale_fill_manual(values = c(&quot;Yes&quot; = &quot;#e41a1c&quot;, &quot;No&quot; = &quot;#377eb8&quot;)) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +
  labs(
    title = &quot;DNA Damage Gene Proportions Across\nCormotif Clusters (0.1 and 0.5 µM)&quot;,
    x = &quot;Response Groups&quot;,
    y = &quot;Percentage of Genes&quot;,
    fill = &quot;DNA Damage Gene&quot;
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = &quot;bold&quot;),
    axis.title.x = element_text(size = 14, face = &quot;bold&quot;),
    axis.title.y = element_text(size = 14, face = &quot;bold&quot;),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    panel.spacing = unit(1.2, &quot;lines&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/prop%20DNA%20combined-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-prop-DNA-combined-1">
Past versions of "prop DNA combined-1.png"
</button>
</p>
<div id="fig-no-spaces-prop-DNA-combined-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/07b9670bf922129e5d7a932d22711efc033fde34/docs/figure/Corrmotif_Conc.Rmd/prop DNA combined-1.png" target="_blank">07b9670</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/9618a00ad12263a31ed79f64344c82a444c6d5c1/docs/figure/Corrmotif_Conc.Rmd/prop DNA combined-1.png" target="_blank">9618a00</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="proportion-of-dna-damage-response-genes-molecular-signature-dataset"
class="section level2">
<h2><strong>📌 Proportion of DNA Damage Response Genes (Molecular
signature dataset)</strong></h2>
<pre class="r"><code># ----------------- Load Required Libraries -----------------
library(dplyr)
library(ggplot2)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- DDR Entrez IDs -----------------
ddr_entrez_ids &lt;- as.character(c(
  10111, 1017, 1019, 1020, 1021, 1026, 1027, 10912, 11011, 1111,
  11200, 1385, 1643, 1647, 1869, 207, 2177, 25, 27113, 27244,
  3014, 317, 355, 4193, 4292, 4361, 4609, 4616, 4683, 472, 50484,
  5366, 5371, 54205, 545, 55367, 5591, 581, 5810, 5883, 5884,
  5888, 5893, 5925, 595, 5981, 6118, 637, 672, 7157, 7799,
  8243, 836, 841, 84126, 842, 8795, 891, 894, 896, 898,
  9133, 9134, 983, 9874, 993, 995, 5916
))

# ----------------- Load CorrMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% ddr_entrez_ids, &quot;DDR Genes&quot;, &quot;Non-DDR Genes&quot;),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% ddr_entrez_ids, &quot;DDR Genes&quot;, &quot;Non-DDR Genes&quot;),
    Concentration = &quot;0.5 µM&quot;
  )

# ----------------- Combine Data -----------------
df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Calculate Proportions -----------------
proportion_data &lt;- df_combined %&gt;%
  group_by(Concentration, Response_Group, Category) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Chi-square or Fisher&#39;s Test -----------------
get_chi_or_fisher &lt;- function(df, ref_group) {
  ref_counts &lt;- df %&gt;%
    ungroup() %&gt;%
    filter(Response_Group == ref_group) %&gt;%
    { ref_tbl &lt;- .[ , c(&quot;Category&quot;, &quot;Count&quot;)]; setNames(ref_tbl$Count, ref_tbl$Category) }
  
  df %&gt;%
    filter(Response_Group != ref_group) %&gt;%
    group_by(Response_Group) %&gt;%
    summarise(
      p_value = {
        group_counts &lt;- Count[Category %in% c(&quot;DDR Genes&quot;, &quot;Non-DDR Genes&quot;)]
        if (length(group_counts) &lt; 2) group_counts &lt;- c(group_counts, 0)
        contingency_table &lt;- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts[&quot;DDR Genes&quot;], ref_counts[&quot;Non-DDR Genes&quot;]
        ), nrow = 2)
        if (any(contingency_table &lt; 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = &quot;drop&quot;
    ) %&gt;%
    mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))
}

# Run test for each concentration
chi_0.1 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.1 µM&quot;), &quot;Non response&quot;)</code></pre>
<pre><code>Warning: There was 1 warning in `summarise()`.
ℹ In argument: `p_value = { ... }`.
ℹ In group 1: `Response_Group = &quot;CX-DOX mid-late response&quot;`.
Caused by warning in `chisq.test()`:
! Chi-squared approximation may be incorrect</code></pre>
<pre class="r"><code>chi_0.5 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.5 µM&quot;), &quot;Non response&quot;)</code></pre>
<pre><code>Warning: There were 2 warnings in `summarise()`.
The first warning was:
ℹ In argument: `p_value = { ... }`.
ℹ In group 1: `Response_Group = &quot;CX total + DOX early response&quot;`.
Caused by warning in `chisq.test()`:
! Chi-squared approximation may be incorrect
ℹ Run `dplyr::last_dplyr_warnings()` to see the 1 remaining warning.</code></pre>
<pre class="r"><code>chi_all &lt;- bind_rows(
  chi_0.1 %&gt;% mutate(Concentration = &quot;0.1 µM&quot;),
  chi_0.5 %&gt;% mutate(Concentration = &quot;0.5 µM&quot;)
)

# Merge test results into proportions
proportion_data &lt;- left_join(proportion_data, chi_all, by = c(&quot;Concentration&quot;, &quot;Response_Group&quot;))

# ----------------- Reorder Factor Levels -----------------
proportion_data$Concentration &lt;- factor(proportion_data$Concentration, levels = c(&quot;0.1 µM&quot;, &quot;0.5 µM&quot;))
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
))

# ----------------- Star Label Position -----------------
label_data &lt;- proportion_data %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  summarise(Significance = dplyr::first(Significance), .groups = &quot;drop&quot;) %&gt;%
  filter(!is.na(Significance) &amp; Significance != &quot;&quot;) %&gt;%
  mutate(y_pos = 105)

# ----------------- Final Plot -----------------
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(
    data = label_data,
    aes(x = Response_Group, y = y_pos, label = Significance),
    inherit.aes = FALSE,
    size = 5,
    color = &quot;black&quot;
  ) +
  facet_wrap(~ Concentration, scales = &quot;free_x&quot;) +
  scale_fill_manual(values = c(
    &quot;DDR Genes&quot; = &quot;#E41A1C&quot;,
    &quot;Non-DDR Genes&quot; = &quot;#377EB8&quot;
  )) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +
  labs(
    title = &quot;DNA Damage Response Gene Proportions Across\nCormotif Clusters (0.1 and 0.5)&quot;,
    x = &quot;Response Groups&quot;,
    y = &quot;Percentage of Genes&quot;,
    fill = &quot;Gene Category&quot;
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = &quot;bold&quot;),
    axis.title.x = element_text(size = 14, face = &quot;bold&quot;),
    axis.title.y = element_text(size = 14, face = &quot;bold&quot;),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    panel.spacing = unit(1.2, &quot;lines&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/prop%20DDR-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-prop-DDR-1">
Past versions of "prop DDR-1.png"
</button>
</p>
<div id="fig-no-spaces-prop-DDR-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/prop DDR-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-dna-damage-response-genes-across-corrmotifs-molecular-signature-dataset-per-concentration"
class="section level2">
<h2><strong>📌 Mean logFC of DNA damage response genes across corrmotifs
(Molecular signature dataset) (per concentration)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)</code></pre>
<pre><code>Warning: package &#39;tidyverse&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;readr&#39; was built under R version 4.3.3</code></pre>
<pre><code>Warning: package &#39;purrr&#39; was built under R version 4.3.3</code></pre>
<pre><code>Warning: package &#39;stringr&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;lubridate&#39; was built under R version 4.3.3</code></pre>
<pre class="r"><code>library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- DDR Entrez IDs -----------------
entrez_ids &lt;- c(
  10111, 1017, 1019, 1020, 1021, 1026, 1027, 10912, 11011, 1111,
  11200, 1385, 1643, 1647, 1869, 207, 2177, 25, 27113, 27244,
  3014, 317, 355, 4193, 4292, 4361, 4609, 4616, 4683, 472, 50484,
  5366, 5371, 54205, 545, 55367, 5591, 581, 5810, 5883, 5884,
  5888, 5893, 5925, 595, 5981, 6118, 637, 672, 7157, 7799,
  8243, 836, 841, 84126, 842, 8795, 891, 894, 896, 898,
  9133, 9134, 983, 9874, 993, 995, 5916
)

# ----------------- Load CorMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate Response Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% as.character(entrez_ids)) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% as.character(entrez_ids)) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Aggregate DEG Data -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;,
      TRUE ~ NA_character_
    )
  )

deg_mean &lt;- deg_all %&gt;%
  group_by(Entrez_ID, Concentration) %&gt;%
  summarise(
    logFC = mean(logFC, na.rm = TRUE),
    .groups = &quot;drop&quot;
  )

# ----------------- Merge with CorMotif DDR Genes -----------------
ddr_logfc &lt;- inner_join(df_combined, deg_mean, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Group = paste(Response_Group, Concentration)
  )

# ----------------- Define order -----------------
group_levels &lt;- c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
)

ddr_logfc$Response_Group &lt;- factor(ddr_logfc$Response_Group, levels = group_levels)

# ----------------- Wilcoxon Test -----------------
comparison_map &lt;- list(
  &quot;CX-DOX mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX only mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX-specific response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX only mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;CX total + DOX early response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX early + CX-DOX mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;
)

star_df &lt;- lapply(names(comparison_map), function(resp_group) {
  control_group &lt;- comparison_map[[resp_group]]
  resp_vals &lt;- ddr_logfc$logFC[ddr_logfc$Group == resp_group]
  control_vals &lt;- ddr_logfc$logFC[ddr_logfc$Group == control_group]

  if (length(resp_vals) &gt;= 2 &amp;&amp; length(control_vals) &gt;= 2) {
    test_result &lt;- wilcox.test(resp_vals, control_vals)
    pval &lt;- test_result$p.value
    if (pval &lt; 0.05) {
      label &lt;- case_when(
        pval &lt; 0.001 ~ &quot;***&quot;,
        pval &lt; 0.01 ~ &quot;**&quot;,
        TRUE ~ &quot;*&quot;
      )
      y_pos &lt;- max(resp_vals, na.rm = TRUE) + 0.5
      return(data.frame(Group = resp_group, y_pos = y_pos, label = label, P_Value = signif(pval, 4)))
    }
  }
  return(NULL)
}) %&gt;% bind_rows()

# ----------------- Prepare Annotation -----------------
label_data &lt;- star_df %&gt;%
  separate(Group, into = c(&quot;Response_Group&quot;, &quot;Concentration&quot;), sep = &quot; (?=0\\.)&quot;) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = group_levels))

# ----------------- Plot -----------------
ggplot(ddr_logfc, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~Concentration, scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Mean logFC of DDR Genes\nAcross CorMotif Response Groups&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20DDR%20logFC-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-DDR-logFC-1">
Past versions of "Mean DDR logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-DDR-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean DDR logFC-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-dna-damage-response-genes-across-corrmotifs-molecular-signature-dataset-faceted-by-concentration-timepoint"
class="section level2">
<h2><strong>📌 Mean logFC of DNA damage response genes across corrmotifs
(Molecular signature dataset) (Faceted by concentration ×
timepoint)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- DDR Entrez IDs -----------------
entrez_ids &lt;- c(
  10111, 1017, 1019, 1020, 1021, 1026, 1027, 10912, 11011, 1111,
  11200, 1385, 1643, 1647, 1869, 207, 2177, 25, 27113, 27244,
  3014, 317, 355, 4193, 4292, 4361, 4609, 4616, 4683, 472, 50484,
  5366, 5371, 54205, 545, 55367, 5591, 581, 5810, 5883, 5884,
  5888, 5893, 5925, 595, 5981, 6118, 637, 672, 7157, 7799,
  8243, 836, 841, 84126, 842, 8795, 891, 894, 896, 898,
  9133, 9134, 983, 9874, 993, 995, 5916
) %&gt;% as.character()

# ----------------- Load CorMotif Groups -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate Response Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Annotate DEG Data -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    ),
    Timepoint = factor(case_when(
      str_detect(Condition, &quot;_3$&quot;) ~ &quot;3 hr&quot;,
      str_detect(Condition, &quot;_24$&quot;) ~ &quot;24 hr&quot;,
      str_detect(Condition, &quot;_48$&quot;) ~ &quot;48 hr&quot;
    ), levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))
  )

# ----------------- Merge with CorMotif DDR Genes -----------------
ddr_all &lt;- inner_join(df_combined, deg_all, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response&quot;, &quot;CX-DOX mid-late response&quot;, &quot;DOX-specific response&quot;,
      &quot;DOX only mid-late response&quot;, &quot;CX total + DOX early response&quot;, &quot;DOX early + CX-DOX mid-late response&quot;
    ))
  )

# ----------------- Wilcoxon Test -----------------
comparison_groups &lt;- levels(ddr_all$Response_Group)[-1]  # exclude &quot;Non response&quot;
stat_results &lt;- list()

for (conc in unique(ddr_all$Concentration)) {
  for (tp in levels(ddr_all$Timepoint)) {
    sub &lt;- ddr_all %&gt;% filter(Concentration == conc, Timepoint == tp)
    for (grp in comparison_groups) {
      if (grp %in% unique(sub$Response_Group)) {
        test &lt;- tryCatch({
          wilcox.test(
            sub$logFC[sub$Response_Group == grp],
            sub$logFC[sub$Response_Group == &quot;Non response&quot;]
          )
        }, error = function(e) NULL)
        if (!is.null(test) &amp;&amp; !is.nan(test$p.value)) {
          stat_results[[length(stat_results) + 1]] &lt;- data.frame(
            Response_Group = grp,
            Concentration = conc,
            Timepoint = tp,
            p_value = signif(test$p.value, 3),
            label = case_when(
              test$p.value &lt; 0.001 ~ &quot;***&quot;,
              test$p.value &lt; 0.01 ~ &quot;**&quot;,
              test$p.value &lt; 0.05 ~ &quot;*&quot;,
              TRUE ~ &quot;&quot;
            ),
            y_pos = max(sub$logFC[sub$Response_Group == grp], na.rm = TRUE) + 0.5
          )
        }
      }
    }
  }
}

label_data &lt;- bind_rows(stat_results) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = levels(ddr_all$Response_Group)))

# ----------------- Plot -----------------
ggplot(ddr_all, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7, na.rm = TRUE) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 4, vjust = 0) +
  facet_grid(rows = vars(factor(Timepoint, levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))),
             cols = vars(Concentration), scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;DDR Gene logFC by CorMotif Group, Timepoint, and Concentration&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20DDR%20logFC1-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-DDR-logFC1-1">
Past versions of "Mean DDR logFC1-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-DDR-logFC1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean DDR logFC1-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-dna-damage-response-genes-across-corrmotifs-molecular-signature-dataset-faceted-by-concentration-timepoint-drug-cx-vs-dox-side-by-side"
class="section level2">
<h2><strong>📌 Mean logFC of DNA damage response genes across corrmotifs
(Molecular signature dataset) (Faceted by concentration × timepoint ×
drug (CX vs DOX side-by-side))</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- DDR Entrez IDs -----------------
ddr_ids &lt;- c(
  10111, 1017, 1019, 1020, 1021, 1026, 1027, 10912, 11011, 1111,
  11200, 1385, 1643, 1647, 1869, 207, 2177, 25, 27113, 27244,
  3014, 317, 355, 4193, 4292, 4361, 4609, 4616, 4683, 472, 50484,
  5366, 5371, 54205, 545, 55367, 5591, 581, 5810, 5883, 5884,
  5888, 5893, 5925, 595, 5981, 6118, 637, 672, 7157, 7799,
  8243, 836, 841, 84126, 842, 8795, 891, 894, 896, 898,
  9133, 9134, 983, 9874, 993, 995, 5916
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.1 µM) -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.1 µM DEG Files -----------------
deg_0.1 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.1 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late\nresponse (0.1 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% ddr_ids)

# ----------------- Set Factor Levels -----------------
deg_0.1 &lt;- deg_0.1 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.1 µM)&quot;,
      &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.1 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.1 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.1 µM DDR) -----------------
ggplot(deg_0.1, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;DDR Gene logFC by Drug and Response Group (0.1 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20DDR%20logFC3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-DDR-logFC3-1">
Past versions of "Mean DDR logFC3-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-DDR-logFC3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean DDR logFC3-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>################### DDR Genes — 0.5 µM ###################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- DDR Entrez IDs -----------------
ddr_ids &lt;- c(
  10111, 1017, 1019, 1020, 1021, 1026, 1027, 10912, 11011, 1111,
  11200, 1385, 1643, 1647, 1869, 207, 2177, 25, 27113, 27244,
  3014, 317, 355, 4193, 4292, 4361, 4609, 4616, 4683, 472, 50484,
  5366, 5371, 54205, 545, 55367, 5591, 581, 5810, 5883, 5884,
  5888, 5893, 5925, 595, 5981, 6118, 637, 672, 7157, 7799,
  8243, 836, 841, 84126, 842, 8795, 891, 894, 896, 898,
  9133, 9134, 983, 9874, 993, 995, 5916
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.5 µM) -----------------
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.5 µM DEG Files -----------------
deg_0.5 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.5 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 µM)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% ddr_ids)

# ----------------- Set Factor Levels -----------------
deg_0.5 &lt;- deg_0.5 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.5 µM)&quot;,
      &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.5 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.5 µM DDR) -----------------
ggplot(deg_0.5, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;DDR Gene logFC by Drug and Response Group (0.5 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20DDR%20logFC3-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-DDR-logFC3-2">
Past versions of "Mean DDR logFC3-2.png"
</button>
</p>
<div id="fig-no-spaces-Mean-DDR-logFC3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean DDR logFC3-2.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="proportion-of-p53-target-genes-across-corrmotif-clusters"
class="section level2">
<h2><strong>📌 Proportion of P53 target Genes across corrmotif
clusters</strong></h2>
<pre class="r"><code># ----------------- Load Required Libraries -----------------
library(dplyr)
library(ggplot2)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- P53 Entrez IDs -----------------
p53_entrez_ids &lt;- as.character(c(
  1026, 50484, 4193, 9766, 9518, 7832, 1643, 1647, 1263, 57103, 51065, 8795, 51499, 64393, 581, 5228, 5429, 8493, 55959,
  7508, 64782, 282991, 355, 53836, 4814, 10769, 9050, 27244, 9540, 94241, 26154, 57763, 900, 26999, 55332, 26263, 23479,
  23612, 29950, 9618, 10346, 8824, 134147, 55294, 22824, 4254, 6560, 467, 27113, 60492, 8444, 60401, 1969, 220965, 2232,
  3976, 55191, 84284, 93129, 5564, 7803, 83667, 7779, 132671, 7039, 51768, 137695, 93134, 7633, 10973, 340485, 307,
  27350, 23245, 3732, 29965, 1363, 1435, 196513, 8507, 8061, 2517, 51278, 53354, 54858, 23228, 5366, 5912, 6236, 51222,
  26152, 59, 1907, 50650, 91012, 780, 9249, 11072, 144455, 64787, 116151, 27165, 2876, 57822, 55733, 57722, 121457,
  375449, 85377, 4851, 5875, 127544, 29901, 84958, 8797, 8793, 441631, 220001, 54541, 5889, 5054, 25816, 25987, 5111,
  98, 317, 598, 604, 10904, 1294, 80315, 53944, 1606, 2770, 3628, 3675, 3985, 4035, 4163, 84552, 29085, 55367, 5371,
  5791, 54884, 5980, 8794, 1462, 50808, 220, 583, 694, 1056, 9076, 10978, 54677, 1612, 55040, 114907, 2274, 127707,
  4000, 8079, 4646, 4747, 27445, 5143, 80055, 79156, 5360, 5364, 23654, 5565, 5613, 5625, 10076, 56963, 6004, 390,
  255488, 6326, 6330, 23513, 7869, 283130, 204962, 83959, 6548, 6774, 9263, 10228, 22954, 10475, 85363, 494514, 10142,
  79714, 1006, 8446, 9648, 79828, 5507, 55240, 63874, 25841, 9289, 84883, 154810, 51321, 421, 8553, 655, 119032, 84280,
  10950, 824, 839, 57828, 857, 8812, 8837, 94027, 113189, 22837, 132864, 10898, 3300, 81704, 1847, 1849, 1947, 9538,
  24139, 5168, 147965, 115548, 9873, 23768, 2632, 2817, 3280, 3265, 23308, 3490, 51477, 182, 3856, 8844, 144811, 9404,
  4043, 9848, 2872, 23041, 740, 343263, 4638, 26509, 4792, 22861, 57523, 55214, 80025, 164091, 57060, 64065, 51090,
  5453, 8496, 333926, 55671, 5900, 55544, 23179, 8601, 389, 6223, 55800, 6385, 4088, 6643, 122809, 257397, 285343,
  7011, 54790, 374618, 55362, 51754, 7157, 9537, 22906, 7205, 80705, 219699, 55245, 83719, 7748, 25946, 118738
))

# ----------------- Load CorrMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% p53_entrez_ids, &quot;P53 Target Genes&quot;, &quot;Non-P53 Genes&quot;),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% p53_entrez_ids, &quot;P53 Target Genes&quot;, &quot;Non-P53 Genes&quot;),
    Concentration = &quot;0.5 µM&quot;
  )

# ----------------- Combine Data -----------------
df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Set Category Order (P53 first) -----------------
df_combined$Category &lt;- factor(df_combined$Category, levels = c(&quot;P53 Target Genes&quot;, &quot;Non-P53 Genes&quot;))

# ----------------- Calculate Proportions -----------------
proportion_data &lt;- df_combined %&gt;%
  group_by(Concentration, Response_Group, Category) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Chi-square or Fisher&#39;s Test -----------------
get_chi_or_fisher &lt;- function(df, ref_group) {
  ref_counts &lt;- df %&gt;%
    ungroup() %&gt;%
    filter(Response_Group == ref_group) %&gt;%
    { ref_tbl &lt;- .[ , c(&quot;Category&quot;, &quot;Count&quot;)]; setNames(ref_tbl$Count, ref_tbl$Category) }
  
  df %&gt;%
    filter(Response_Group != ref_group) %&gt;%
    group_by(Response_Group) %&gt;%
    summarise(
      p_value = {
        group_counts &lt;- Count[Category %in% c(&quot;P53 Target Genes&quot;, &quot;Non-P53 Genes&quot;)]
        if (length(group_counts) &lt; 2) group_counts &lt;- c(group_counts, 0)
        contingency_table &lt;- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts[&quot;P53 Target Genes&quot;], ref_counts[&quot;Non-P53 Genes&quot;]
        ), nrow = 2)
        if (any(contingency_table &lt; 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = &quot;drop&quot;
    ) %&gt;%
    mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))
}

# Run test for each concentration
chi_0.1 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.1 µM&quot;), &quot;Non response&quot;)
chi_0.5 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.5 µM&quot;), &quot;Non response&quot;)</code></pre>
<pre><code>Warning: There was 1 warning in `summarise()`.
ℹ In argument: `p_value = { ... }`.
ℹ In group 2: `Response_Group = &quot;DOX early + CX-DOX mid-late response&quot;`.
Caused by warning in `chisq.test()`:
! Chi-squared approximation may be incorrect</code></pre>
<pre class="r"><code>chi_all &lt;- bind_rows(
  chi_0.1 %&gt;% mutate(Concentration = &quot;0.1 µM&quot;),
  chi_0.5 %&gt;% mutate(Concentration = &quot;0.5 µM&quot;)
)

# Merge test results into proportions
proportion_data &lt;- left_join(proportion_data, chi_all, by = c(&quot;Concentration&quot;, &quot;Response_Group&quot;))

# ----------------- Reorder Factor Levels -----------------
proportion_data$Concentration &lt;- factor(proportion_data$Concentration, levels = c(&quot;0.1 µM&quot;, &quot;0.5 µM&quot;))
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
))

# ----------------- Star Label Position -----------------
label_data &lt;- proportion_data %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  summarise(Significance = dplyr::first(Significance), .groups = &quot;drop&quot;) %&gt;%
  filter(!is.na(Significance) &amp; Significance != &quot;&quot;) %&gt;%
  mutate(y_pos = 105)

# ----------------- Final Plot -----------------
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(
    data = label_data,
    aes(x = Response_Group, y = y_pos, label = Significance),
    inherit.aes = FALSE,
    size = 5,
    color = &quot;black&quot;
  ) +
  facet_wrap(~ Concentration, scales = &quot;free_x&quot;) +
  scale_fill_manual(values = c(
    &quot;P53 Target Genes&quot; = &quot;#E41A1C&quot;,
    &quot;Non-P53 Genes&quot; = &quot;#377EB8&quot;
  )) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +
  labs(
    title = &quot;P53 Target Gene Proportions Across\nCormotif Clusters (0.1 and 0.5)&quot;,
    x = &quot;Response Groups&quot;,
    y = &quot;Percentage of Genes&quot;,
    fill = &quot;Gene Category&quot;
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = &quot;bold&quot;),
    axis.title.x = element_text(size = 14, face = &quot;bold&quot;),
    axis.title.y = element_text(size = 14, face = &quot;bold&quot;),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    panel.spacing = unit(1.2, &quot;lines&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/prop%20P53%200.1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-prop-P53-0.1-1">
Past versions of "prop P53 0.1-1.png"
</button>
</p>
<div id="fig-no-spaces-prop-P53-0.1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/prop P53 0.1-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/prop P53 0.1-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/81247f64f08df43c2b40a386707fc8e9cc240143/docs/figure/Corrmotif_Conc.Rmd/prop P53 0.1-1.png" target="_blank">81247f6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/96d0db145cf68e6db0d39e70bad9af0f75793aae/docs/figure/Corrmotif_Conc.Rmd/prop P53 0.1-1.png" target="_blank">96d0db1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-p53-target-genes-across-corrmotifs-per-concentration"
class="section level2">
<h2><strong>📌 Mean logFC of P53 target genes across corrmotifs (per
concentration)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- P53 Entrez IDs -----------------
entrez_ids &lt;- c(
  1026, 50484, 4193, 9766, 9518, 7832, 1643, 1647, 1263, 57103, 51065, 8795, 51499, 64393, 581, 5228, 5429, 8493, 55959,
  7508, 64782, 282991, 355, 53836, 4814, 10769, 9050, 27244, 9540, 94241, 26154, 57763, 900, 26999, 55332, 26263, 23479,
  23612, 29950, 9618, 10346, 8824, 134147, 55294, 22824, 4254, 6560, 467, 27113, 60492, 8444, 60401, 1969, 220965, 2232,
  3976, 55191, 84284, 93129, 5564, 7803, 83667, 7779, 132671, 7039, 51768, 137695, 93134, 7633, 10973, 340485, 307,
  27350, 23245, 3732, 29965, 1363, 1435, 196513, 8507, 8061, 2517, 51278, 53354, 54858, 23228, 5366, 5912, 6236, 51222,
  26152, 59, 1907, 50650, 91012, 780, 9249, 11072, 144455, 64787, 116151, 27165, 2876, 57822, 55733, 57722, 121457,
  375449, 85377, 4851, 5875, 127544, 29901, 84958, 8797, 8793, 441631, 220001, 54541, 5889, 5054, 25816, 25987, 5111,
  98, 317, 598, 604, 10904, 1294, 80315, 53944, 1606, 2770, 3628, 3675, 3985, 4035, 4163, 84552, 29085, 55367, 5371,
  5791, 54884, 5980, 8794, 1462, 50808, 220, 583, 694, 1056, 9076, 10978, 54677, 1612, 55040, 114907, 2274, 127707,
  4000, 8079, 4646, 4747, 27445, 5143, 80055, 79156, 5360, 5364, 23654, 5565, 5613, 5625, 10076, 56963, 6004, 390,
  255488, 6326, 6330, 23513, 7869, 283130, 204962, 83959, 6548, 6774, 9263, 10228, 22954, 10475, 85363, 494514, 10142,
  79714, 1006, 8446, 9648, 79828, 5507, 55240, 63874, 25841, 9289, 84883, 154810, 51321, 421, 8553, 655, 119032, 84280,
  10950, 824, 839, 57828, 857, 8812, 8837, 94027, 113189, 22837, 132864, 10898, 3300, 81704, 1847, 1849, 1947, 9538,
  24139, 5168, 147965, 115548, 9873, 23768, 2632, 2817, 3280, 3265, 23308, 3490, 51477, 182, 3856, 8844, 144811, 9404,
  4043, 9848, 2872, 23041, 740, 343263, 4638, 26509, 4792, 22861, 57523, 55214, 80025, 164091, 57060, 64065, 51090,
  5453, 8496, 333926, 55671, 5900, 55544, 23179, 8601, 389, 6223, 55800, 6385, 4088, 6643, 122809, 257397, 285343,
  7011, 54790, 374618, 55362, 51754, 7157, 9537, 22906, 7205, 80705, 219699, 55245, 83719, 7748, 25946, 118738
)

# ----------------- Load CorMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate Response Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% as.character(entrez_ids)) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% as.character(entrez_ids)) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Aggregate DEG Data -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;,
      TRUE ~ NA_character_
    )
  )

deg_mean &lt;- deg_all %&gt;%
  group_by(Entrez_ID, Concentration) %&gt;%
  summarise(logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;)

# ----------------- Merge with CorMotif P53 Genes -----------------
p53_logfc &lt;- inner_join(df_combined, deg_mean, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Group = paste(Response_Group, Concentration)
  )

# ----------------- Define order -----------------
group_levels &lt;- c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
)

p53_logfc$Response_Group &lt;- factor(p53_logfc$Response_Group, levels = group_levels)

# ----------------- Wilcoxon Test -----------------
comparison_map &lt;- list(
  &quot;CX-DOX mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX only mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX-specific response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX only mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;CX total + DOX early response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX early + CX-DOX mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;
)

star_df &lt;- lapply(names(comparison_map), function(resp_group) {
  control_group &lt;- comparison_map[[resp_group]]
  resp_vals &lt;- p53_logfc$logFC[p53_logfc$Group == resp_group]
  control_vals &lt;- p53_logfc$logFC[p53_logfc$Group == control_group]

  if (length(resp_vals) &gt;= 2 &amp;&amp; length(control_vals) &gt;= 2) {
    test_result &lt;- wilcox.test(resp_vals, control_vals)
    pval &lt;- test_result$p.value
    if (pval &lt; 0.05) {
      label &lt;- case_when(
        pval &lt; 0.001 ~ &quot;***&quot;,
        pval &lt; 0.01 ~ &quot;**&quot;,
        TRUE ~ &quot;*&quot;
      )
      y_pos &lt;- max(resp_vals, na.rm = TRUE) + 0.5
      return(data.frame(Group = resp_group, y_pos = y_pos, label = label, P_Value = signif(pval, 4)))
    }
  }
  return(NULL)
}) %&gt;% bind_rows()


# ----------------- Prepare Annotation -----------------
label_data &lt;- star_df %&gt;%
  separate(Group, into = c(&quot;Response_Group&quot;, &quot;Concentration&quot;), sep = &quot; (?=0\\.)&quot;) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = group_levels))




# ----------------- Plot -----------------
ggplot(p53_logfc, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~Concentration, scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Mean logFC of P53 Target Genes\nAcross CorMotif Response Groups&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20P53%20logFC-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-P53-logFC-1">
Past versions of "Mean P53 logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-P53-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean P53 logFC-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-p53-target-genes-across-corrmotifs-faceted-by-concentration-timepoint"
class="section level2">
<h2><strong>📌 Mean logFC of P53 target genes across corrmotifs (Faceted
by concentration × timepoint)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- P53 Entrez IDs -----------------
entrez_ids &lt;- c(
  1026, 50484, 4193, 9766, 9518, 7832, 1643, 1647, 1263, 57103, 51065, 8795, 51499, 64393, 581, 5228, 5429, 8493, 55959,
  7508, 64782, 282991, 355, 53836, 4814, 10769, 9050, 27244, 9540, 94241, 26154, 57763, 900, 26999, 55332, 26263, 23479,
  23612, 29950, 9618, 10346, 8824, 134147, 55294, 22824, 4254, 6560, 467, 27113, 60492, 8444, 60401, 1969, 220965, 2232,
  3976, 55191, 84284, 93129, 5564, 7803, 83667, 7779, 132671, 7039, 51768, 137695, 93134, 7633, 10973, 340485, 307,
  27350, 23245, 3732, 29965, 1363, 1435, 196513, 8507, 8061, 2517, 51278, 53354, 54858, 23228, 5366, 5912, 6236, 51222,
  26152, 59, 1907, 50650, 91012, 780, 9249, 11072, 144455, 64787, 116151, 27165, 2876, 57822, 55733, 57722, 121457,
  375449, 85377, 4851, 5875, 127544, 29901, 84958, 8797, 8793, 441631, 220001, 54541, 5889, 5054, 25816, 25987, 5111,
  98, 317, 598, 604, 10904, 1294, 80315, 53944, 1606, 2770, 3628, 3675, 3985, 4035, 4163, 84552, 29085, 55367, 5371,
  5791, 54884, 5980, 8794, 1462, 50808, 220, 583, 694, 1056, 9076, 10978, 54677, 1612, 55040, 114907, 2274, 127707,
  4000, 8079, 4646, 4747, 27445, 5143, 80055, 79156, 5360, 5364, 23654, 5565, 5613, 5625, 10076, 56963, 6004, 390,
  255488, 6326, 6330, 23513, 7869, 283130, 204962, 83959, 6548, 6774, 9263, 10228, 22954, 10475, 85363, 494514, 10142,
  79714, 1006, 8446, 9648, 79828, 5507, 55240, 63874, 25841, 9289, 84883, 154810, 51321, 421, 8553, 655, 119032, 84280,
  10950, 824, 839, 57828, 857, 8812, 8837, 94027, 113189, 22837, 132864, 10898, 3300, 81704, 1847, 1849, 1947, 9538,
  24139, 5168, 147965, 115548, 9873, 23768, 2632, 2817, 3280, 3265, 23308, 3490, 51477, 182, 3856, 8844, 144811, 9404,
  4043, 9848, 2872, 23041, 740, 343263, 4638, 26509, 4792, 22861, 57523, 55214, 80025, 164091, 57060, 64065, 51090,
  5453, 8496, 333926, 55671, 5900, 55544, 23179, 8601, 389, 6223, 55800, 6385, 4088, 6643, 122809, 257397, 285343,
  7011, 54790, 374618, 55362, 51754, 7157, 9537, 22906, 7205, 80705, 219699, 55245, 83719, 7748, 25946, 118738
) %&gt;% as.character()

# ----------------- Load CorMotif Response Group Data -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorMotif Group Membership -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Annotate DEG Files -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    ),
    Timepoint = factor(case_when(
      str_detect(Condition, &quot;_3$&quot;) ~ &quot;3 hr&quot;,
      str_detect(Condition, &quot;_24$&quot;) ~ &quot;24 hr&quot;,
      str_detect(Condition, &quot;_48$&quot;) ~ &quot;48 hr&quot;
    ), levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))
  )

# ----------------- Merge with P53 Targets and Annotate -----------------
p53_all &lt;- inner_join(df_combined, deg_all, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID, column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response&quot;, &quot;CX-DOX mid-late response&quot;, &quot;DOX-specific response&quot;,
      &quot;DOX only mid-late response&quot;, &quot;CX total + DOX early response&quot;, &quot;DOX early + CX-DOX mid-late response&quot;
    ))
  )

# ----------------- Wilcoxon Test -----------------
comparison_groups &lt;- levels(p53_all$Response_Group)[-1]
stat_results &lt;- list()

for (conc in unique(p53_all$Concentration)) {
  for (tp in levels(p53_all$Timepoint)) {
    sub &lt;- p53_all %&gt;% filter(Concentration == conc, Timepoint == tp)
    for (grp in comparison_groups) {
      if (grp %in% unique(sub$Response_Group)) {
        test &lt;- tryCatch({
          wilcox.test(
            sub$logFC[sub$Response_Group == grp],
            sub$logFC[sub$Response_Group == &quot;Non response&quot;]
          )
        }, error = function(e) NULL)
        if (!is.null(test) &amp;&amp; !is.nan(test$p.value)) {
          stat_results[[length(stat_results) + 1]] &lt;- data.frame(
            Response_Group = grp,
            Concentration = conc,
            Timepoint = tp,
            p_value = signif(test$p.value, 3),
            label = case_when(
              test$p.value &lt; 0.001 ~ &quot;***&quot;,
              test$p.value &lt; 0.01 ~ &quot;**&quot;,
              test$p.value &lt; 0.05 ~ &quot;*&quot;,
              TRUE ~ &quot;&quot;
            ),
            y_pos = max(sub$logFC[sub$Response_Group == grp], na.rm = TRUE) + 0.5
          )
        }
      }
    }
  }
}

label_data &lt;- bind_rows(stat_results) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = levels(p53_all$Response_Group)))

# ----------------- Plot -----------------
ggplot(p53_all, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7, na.rm = TRUE) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 4, vjust = 0) +
  facet_grid(rows = vars(factor(Timepoint, levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))),
             cols = vars(Concentration), scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;P53 Target Gene logFC by\nCorMotif Group, Timepoint, and Concentration&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20P53%20logFC1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-P53-logFC1-1">
Past versions of "Mean P53 logFC1-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-P53-logFC1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean P53 logFC1-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-p53-target-genes-across-corrmotifs-faceted-by-concentration-timepoint-drug-cx-vs-dox-side-by-side"
class="section level2">
<h2><strong>📌 Mean logFC of P53 target genes across corrmotifs (Faceted
by concentration × timepoint × drug (CX vs DOX
side-by-side))</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- P53 Target Entrez IDs -----------------
p53_ids &lt;- c(
  1026, 50484, 4193, 9766, 9518, 7832, 1643, 1647, 1263, 57103, 51065, 8795, 51499, 64393, 581, 5228, 5429, 8493, 55959,
  7508, 64782, 282991, 355, 53836, 4814, 10769, 9050, 27244, 9540, 94241, 26154, 57763, 900, 26999, 55332, 26263, 23479,
  23612, 29950, 9618, 10346, 8824, 134147, 55294, 22824, 4254, 6560, 467, 27113, 60492, 8444, 60401, 1969, 220965, 2232,
  3976, 55191, 84284, 93129, 5564, 7803, 83667, 7779, 132671, 7039, 51768, 137695, 93134, 7633, 10973, 340485, 307,
  27350, 23245, 3732, 29965, 1363, 1435, 196513, 8507, 8061, 2517, 51278, 53354, 54858, 23228, 5366, 5912, 6236, 51222,
  26152, 59, 1907, 50650, 91012, 780, 9249, 11072, 144455, 64787, 116151, 27165, 2876, 57822, 55733, 57722, 121457,
  375449, 85377, 4851, 5875, 127544, 29901, 84958, 8797, 8793, 441631, 220001, 54541, 5889, 5054, 25816, 25987, 5111,
  98, 317, 598, 604, 10904, 1294, 80315, 53944, 1606, 2770, 3628, 3675, 3985, 4035, 4163, 84552, 29085, 55367, 5371,
  5791, 54884, 5980, 8794, 1462, 50808, 220, 583, 694, 1056, 9076, 10978, 54677, 1612, 55040, 114907, 2274, 127707,
  4000, 8079, 4646, 4747, 27445, 5143, 80055, 79156, 5360, 5364, 23654, 5565, 5613, 5625, 10076, 56963, 6004, 390,
  255488, 6326, 6330, 23513, 7869, 283130, 204962, 83959, 6548, 6774, 9263, 10228, 22954, 10475, 85363, 494514, 10142,
  79714, 1006, 8446, 9648, 79828, 5507, 55240, 63874, 25841, 9289, 84883, 154810, 51321, 421, 8553, 655, 119032, 84280,
  10950, 824, 839, 57828, 857, 8812, 8837, 94027, 113189, 22837, 132864, 10898, 3300, 81704, 1847, 1849, 1947, 9538,
  24139, 5168, 147965, 115548, 9873, 23768, 2632, 2817, 3280, 3265, 23308, 3490, 51477, 182, 3856, 8844, 144811, 9404,
  4043, 9848, 2872, 23041, 740, 343263, 4638, 26509, 4792, 22861, 57523, 55214, 80025, 164091, 57060, 64065, 51090,
  5453, 8496, 333926, 55671, 5900, 55544, 23179, 8601, 389, 6223, 55800, 6385, 4088, 6643, 122809, 257397, 285343,
  7011, 54790, 374618, 55362, 51754, 7157, 9537, 22906, 7205, 80705, 219699, 55245, 83719, 7748, 25946, 118738
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.1 µM) -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.1 µM DEG Files -----------------
deg_0.1 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.1 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late\nresponse (0.1 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% p53_ids)

# ----------------- Set Factor Levels -----------------
deg_0.1 &lt;- deg_0.1 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.1 µM)&quot;,
      &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.1 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.1 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.1 µM P53) -----------------
ggplot(deg_0.1, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;P53 Target Gene logFC by Drug and Response Group (0.1 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20P53%20logFC3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-P53-logFC3-1">
Past versions of "Mean P53 logFC3-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-P53-logFC3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean P53 logFC3-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>################### P53 Target Genes — 0.5 µM ###################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- P53 Target Entrez IDs -----------------
p53_ids &lt;- c(
  1026, 50484, 4193, 9766, 9518, 7832, 1643, 1647, 1263, 57103, 51065, 8795, 51499, 64393, 581, 5228, 5429, 8493, 55959,
  7508, 64782, 282991, 355, 53836, 4814, 10769, 9050, 27244, 9540, 94241, 26154, 57763, 900, 26999, 55332, 26263, 23479,
  23612, 29950, 9618, 10346, 8824, 134147, 55294, 22824, 4254, 6560, 467, 27113, 60492, 8444, 60401, 1969, 220965, 2232,
  3976, 55191, 84284, 93129, 5564, 7803, 83667, 7779, 132671, 7039, 51768, 137695, 93134, 7633, 10973, 340485, 307,
  27350, 23245, 3732, 29965, 1363, 1435, 196513, 8507, 8061, 2517, 51278, 53354, 54858, 23228, 5366, 5912, 6236, 51222,
  26152, 59, 1907, 50650, 91012, 780, 9249, 11072, 144455, 64787, 116151, 27165, 2876, 57822, 55733, 57722, 121457,
  375449, 85377, 4851, 5875, 127544, 29901, 84958, 8797, 8793, 441631, 220001, 54541, 5889, 5054, 25816, 25987, 5111,
  98, 317, 598, 604, 10904, 1294, 80315, 53944, 1606, 2770, 3628, 3675, 3985, 4035, 4163, 84552, 29085, 55367, 5371,
  5791, 54884, 5980, 8794, 1462, 50808, 220, 583, 694, 1056, 9076, 10978, 54677, 1612, 55040, 114907, 2274, 127707,
  4000, 8079, 4646, 4747, 27445, 5143, 80055, 79156, 5360, 5364, 23654, 5565, 5613, 5625, 10076, 56963, 6004, 390,
  255488, 6326, 6330, 23513, 7869, 283130, 204962, 83959, 6548, 6774, 9263, 10228, 22954, 10475, 85363, 494514, 10142,
  79714, 1006, 8446, 9648, 79828, 5507, 55240, 63874, 25841, 9289, 84883, 154810, 51321, 421, 8553, 655, 119032, 84280,
  10950, 824, 839, 57828, 857, 8812, 8837, 94027, 113189, 22837, 132864, 10898, 3300, 81704, 1847, 1849, 1947, 9538,
  24139, 5168, 147965, 115548, 9873, 23768, 2632, 2817, 3280, 3265, 23308, 3490, 51477, 182, 3856, 8844, 144811, 9404,
  4043, 9848, 2872, 23041, 740, 343263, 4638, 26509, 4792, 22861, 57523, 55214, 80025, 164091, 57060, 64065, 51090,
  5453, 8496, 333926, 55671, 5900, 55544, 23179, 8601, 389, 6223, 55800, 6385, 4088, 6643, 122809, 257397, 285343,
  7011, 54790, 374618, 55362, 51754, 7157, 9537, 22906, 7205, 80705, 219699, 55245, 83719, 7748, 25946, 118738
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.5 µM) -----------------
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.5 µM DEG Files -----------------
deg_0.5 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.5 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 µM)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% p53_ids)

# ----------------- Set Factor Levels -----------------
deg_0.5 &lt;- deg_0.5 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.5 µM)&quot;,
      &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.5 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.5 µM P53) -----------------
ggplot(deg_0.5, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;P53 Target Gene logFC by Drug and Response Group (0.5 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20P53%20logFC3-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-P53-logFC3-2">
Past versions of "Mean P53 logFC3-2.png"
</button>
</p>
<div id="fig-no-spaces-Mean-P53-logFC3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean P53 logFC3-2.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="top2b-target-genes-proportions-across-corrmotifs"
class="section level2">
<h2><strong>📌 TOP2B target genes proportions across
corrmotifs</strong></h2>
<pre class="r"><code># ----------------- Load Required Libraries -----------------
library(dplyr)
library(ggplot2)
library(readr)

# ----------------- Load TOP2B Entrez IDs -----------------
top2b_entrez_ids &lt;- read_csv(&quot;data/TOP2B_target_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorrMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% top2b_entrez_ids, &quot;TOP2B Target Genes&quot;, &quot;Non-TOP2B Genes&quot;),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% top2b_entrez_ids, &quot;TOP2B Target Genes&quot;, &quot;Non-TOP2B Genes&quot;),
    Concentration = &quot;0.5 µM&quot;
  )

# ----------------- Combine Data -----------------
df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Set Category Order (TOP2B first) -----------------
df_combined$Category &lt;- factor(df_combined$Category, levels = c(&quot;TOP2B Target Genes&quot;, &quot;Non-TOP2B Genes&quot;))

# ----------------- Calculate Proportions -----------------
proportion_data &lt;- df_combined %&gt;%
  group_by(Concentration, Response_Group, Category) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Chi-square or Fisher&#39;s Test -----------------
get_chi_or_fisher &lt;- function(df, ref_group) {
  ref_counts &lt;- df %&gt;%
    ungroup() %&gt;%
    filter(Response_Group == ref_group) %&gt;%
    { ref_tbl &lt;- .[ , c(&quot;Category&quot;, &quot;Count&quot;)]; setNames(ref_tbl$Count, ref_tbl$Category) }
  
  df %&gt;%
    filter(Response_Group != ref_group) %&gt;%
    group_by(Response_Group) %&gt;%
    summarise(
      p_value = {
        group_counts &lt;- Count[Category %in% c(&quot;TOP2B Target Genes&quot;, &quot;Non-TOP2B Genes&quot;)]
        if (length(group_counts) &lt; 2) group_counts &lt;- c(group_counts, 0)
        contingency_table &lt;- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts[&quot;TOP2B Target Genes&quot;], ref_counts[&quot;Non-TOP2B Genes&quot;]
        ), nrow = 2)
        if (any(contingency_table &lt; 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = &quot;drop&quot;
    ) %&gt;%
    mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))
}

# Run test for each concentration
chi_0.1 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.1 µM&quot;), &quot;Non response&quot;)
chi_0.5 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.5 µM&quot;), &quot;Non response&quot;)

chi_all &lt;- bind_rows(
  chi_0.1 %&gt;% mutate(Concentration = &quot;0.1 µM&quot;),
  chi_0.5 %&gt;% mutate(Concentration = &quot;0.5 µM&quot;)
)

# Merge test results into proportions
proportion_data &lt;- left_join(proportion_data, chi_all, by = c(&quot;Concentration&quot;, &quot;Response_Group&quot;))

# ----------------- Reorder Factor Levels -----------------
proportion_data$Concentration &lt;- factor(proportion_data$Concentration, levels = c(&quot;0.1 µM&quot;, &quot;0.5 µM&quot;))
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
))

# ----------------- Star Label Position -----------------
label_data &lt;- proportion_data %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  summarise(Significance = dplyr::first(Significance), .groups = &quot;drop&quot;) %&gt;%
  filter(!is.na(Significance) &amp; Significance != &quot;&quot;) %&gt;%
  mutate(y_pos = 105)

# ----------------- Final Plot -----------------
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(
    data = label_data,
    aes(x = Response_Group, y = y_pos, label = Significance),
    inherit.aes = FALSE,
    size = 5,
    color = &quot;black&quot;
  ) +
  facet_wrap(~ Concentration, scales = &quot;free_x&quot;) +
  scale_fill_manual(values = c(
    &quot;TOP2B Target Genes&quot; = &quot;#E41A1C&quot;,
    &quot;Non-TOP2B Genes&quot; = &quot;#377EB8&quot;
  )) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +
  labs(
    title = &quot;TOP2B Target Gene Proportions Across\nCormotif Clusters (0.1 and 0.5)&quot;,
    x = &quot;Response Groups&quot;,
    y = &quot;Percentage of Genes&quot;,
    fill = &quot;Gene Category&quot;
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = &quot;bold&quot;),
    axis.title.x = element_text(size = 14, face = &quot;bold&quot;),
    axis.title.y = element_text(size = 14, face = &quot;bold&quot;),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    panel.spacing = unit(1.2, &quot;lines&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/TOP2B%20Prop-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-TOP2B-Prop-1">
Past versions of "TOP2B Prop-1.png"
</button>
</p>
<div id="fig-no-spaces-TOP2B-Prop-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/TOP2B Prop-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-top2b-target-genes-across-corrmotifs-per-concentration"
class="section level2">
<h2><strong>📌 Mean logFC of TOP2B target genes across corrmotifs (per
concentration)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- TOP2B Entrez IDs -----------------
top2b_entrez_ids &lt;- read_csv(&quot;data/TOP2B_target_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate Response Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% top2b_entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% top2b_entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Aggregate DEG Data -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;,
      TRUE ~ NA_character_
    )
  )

deg_mean &lt;- deg_all %&gt;%
  group_by(Entrez_ID, Concentration) %&gt;%
  summarise(logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;)

# ----------------- Merge with CorMotif TOP2B Genes -----------------
top2b_logfc &lt;- inner_join(df_combined, deg_mean, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Group = paste(Response_Group, Concentration)
  )

# ----------------- Define order -----------------
group_levels &lt;- c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
)

top2b_logfc$Response_Group &lt;- factor(top2b_logfc$Response_Group, levels = group_levels)

# ----------------- Wilcoxon Test -----------------
comparison_map &lt;- list(
  &quot;CX-DOX mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX only mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX-specific response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX only mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;CX total + DOX early response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX early + CX-DOX mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;
)

star_df &lt;- lapply(names(comparison_map), function(resp_group) {
  control_group &lt;- comparison_map[[resp_group]]
  resp_vals &lt;- top2b_logfc$logFC[top2b_logfc$Group == resp_group]
  control_vals &lt;- top2b_logfc$logFC[top2b_logfc$Group == control_group]

  if (length(resp_vals) &gt;= 2 &amp;&amp; length(control_vals) &gt;= 2) {
    test_result &lt;- wilcox.test(resp_vals, control_vals)
    pval &lt;- test_result$p.value
    if (pval &lt; 0.05) {
      label &lt;- case_when(
        pval &lt; 0.001 ~ &quot;***&quot;,
        pval &lt; 0.01 ~ &quot;**&quot;,
        TRUE ~ &quot;*&quot;
      )
      y_pos &lt;- max(resp_vals, na.rm = TRUE) + 0.5
      return(data.frame(Group = resp_group, y_pos = y_pos, label = label, P_Value = signif(pval, 4)))
    }
  }
  return(NULL)
}) %&gt;% bind_rows()

# ----------------- Prepare Annotation -----------------
label_data &lt;- star_df %&gt;%
  separate(Group, into = c(&quot;Response_Group&quot;, &quot;Concentration&quot;), sep = &quot; (?=0\\.)&quot;) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = group_levels))

# ----------------- Plot -----------------
ggplot(top2b_logfc, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~Concentration, scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Mean logFC of TOP2B Target Genes\nAcross CorMotif Response Groups&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2B%20logFC-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2B-logFC-1">
Past versions of "Mean TOP2B logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2B-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2B logFC-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-top2b-target-genes-across-corrmotifs-faceted-by-concentration-timepoint"
class="section level2">
<h2><strong>📌 Mean logFC of TOP2B target genes across corrmotifs
(Faceted by concentration × timepoint)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- TOP2B Entrez IDs -----------------
top2b_entrez_ids &lt;- read_csv(&quot;data/TOP2B_target_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Response Group Data -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorMotif Group Membership -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% top2b_entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% top2b_entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Annotate DEG Files -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    ),
    Timepoint = factor(case_when(
      str_detect(Condition, &quot;_3$&quot;) ~ &quot;3 hr&quot;,
      str_detect(Condition, &quot;_24$&quot;) ~ &quot;24 hr&quot;,
      str_detect(Condition, &quot;_48$&quot;) ~ &quot;48 hr&quot;
    ), levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))
  )

# ----------------- Merge with TOP2B Targets and Annotate -----------------
top2b_all &lt;- inner_join(df_combined, deg_all, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  filter(Entrez_ID %in% top2b_entrez_ids) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID, column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response&quot;, &quot;CX-DOX mid-late response&quot;, &quot;DOX-specific response&quot;,
      &quot;DOX only mid-late response&quot;, &quot;CX total + DOX early response&quot;, &quot;DOX early + CX-DOX mid-late response&quot;
    ))
  )

# ----------------- Wilcoxon Test -----------------
comparison_groups &lt;- levels(top2b_all$Response_Group)[-1]
stat_results &lt;- list()

for (conc in unique(top2b_all$Concentration)) {
  for (tp in levels(top2b_all$Timepoint)) {
    sub &lt;- top2b_all %&gt;% filter(Concentration == conc, Timepoint == tp)
    for (grp in comparison_groups) {
      if (grp %in% unique(sub$Response_Group)) {
        test &lt;- tryCatch({
          wilcox.test(
            sub$logFC[sub$Response_Group == grp],
            sub$logFC[sub$Response_Group == &quot;Non response&quot;]
          )
        }, error = function(e) NULL)
        if (!is.null(test) &amp;&amp; !is.nan(test$p.value)) {
          stat_results[[length(stat_results) + 1]] &lt;- data.frame(
            Response_Group = grp,
            Concentration = conc,
            Timepoint = tp,
            p_value = signif(test$p.value, 3),
            label = case_when(
              test$p.value &lt; 0.001 ~ &quot;***&quot;,
              test$p.value &lt; 0.01 ~ &quot;**&quot;,
              test$p.value &lt; 0.05 ~ &quot;*&quot;,
              TRUE ~ &quot;&quot;
            ),
            y_pos = max(sub$logFC[sub$Response_Group == grp], na.rm = TRUE) + 0.5
          )
        }
      }
    }
  }
}

label_data &lt;- bind_rows(stat_results) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = levels(top2b_all$Response_Group)))

# ----------------- Plot -----------------
ggplot(top2b_all, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7, na.rm = TRUE) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 4, vjust = 0) +
  facet_grid(rows = vars(factor(Timepoint, levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))),
             cols = vars(Concentration), scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;TOP2B Target Gene logFC by\nCorMotif Group, Timepoint, and Concentration&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2B%20logFC1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2B-logFC1-1">
Past versions of "Mean TOP2B logFC1-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2B-logFC1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2B logFC1-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-top2b-target-genes-across-corrmotifs-faceted-by-concentration-timepoint-drug-cx-vs-dox-side-by-side"
class="section level2">
<h2><strong>📌 Mean logFC of TOP2B target genes across corrmotifs
(Faceted by concentration × timepoint × drug (CX vs DOX
side-by-side))</strong></h2>
<pre class="r"><code>################### TOP2B Target Genes — 0.1 µM ###################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load TOP2B Target Entrez IDs -----------------
top2b_entrez_ids &lt;- read_csv(&quot;data/TOP2B_target_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Groups (0.1 µM) -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.1 µM DEG Files -----------------
deg_0.1 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.1 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late\nresponse (0.1 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% top2b_entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.1 &lt;- deg_0.1 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.1 µM)&quot;,
      &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.1 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.1 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.1 µM TOP2B) -----------------
ggplot(deg_0.1, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;TOP2B Target Gene logFC by Drug and Response Group (0.1 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2B%20logFC3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2B-logFC3-1">
Past versions of "Mean TOP2B logFC3-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2B-logFC3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2B logFC3-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>################### TOP2B Target Genes — 0.5 µM ###################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load TOP2B Target Entrez IDs -----------------
top2b_entrez_ids &lt;- read_csv(&quot;data/TOP2B_target_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Groups (0.5 µM) -----------------
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.5 µM DEG Files -----------------
deg_0.5 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.5 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 µM)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% top2b_entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.5 &lt;- deg_0.5 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.5 µM)&quot;,
      &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.5 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.5 µM TOP2B) -----------------
ggplot(deg_0.5, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;TOP2B Target Gene logFC by Drug and Response Group (0.5 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2B%20logFC3-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2B-logFC3-2">
Past versions of "Mean TOP2B logFC3-2.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2B-logFC3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2B logFC3-2.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="top2a-target-genes-chip_atlas-proportion-across-corrmotifs"
class="section level2">
<h2><strong>📌 TOP2A target genes (ChIP_Atlas) proportion across
corrmotifs</strong></h2>
<pre class="r"><code># ----------------- Load Required Libraries -----------------
library(dplyr)
library(ggplot2)
library(readr)

# ----------------- Load TOP2A Entrez IDs -----------------
top2a_entrez_ids &lt;- read_csv(&quot;data/TOP2A_target_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorrMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% top2a_entrez_ids, &quot;TOP2A Target Genes&quot;, &quot;Non-TOP2A Genes&quot;),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% top2a_entrez_ids, &quot;TOP2A Target Genes&quot;, &quot;Non-TOP2A Genes&quot;),
    Concentration = &quot;0.5 µM&quot;
  )

# ----------------- Combine Data -----------------
df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Set Category Order -----------------
df_combined$Category &lt;- factor(df_combined$Category, levels = c(&quot;TOP2A Target Genes&quot;, &quot;Non-TOP2A Genes&quot;))

# ----------------- Calculate Proportions -----------------
proportion_data &lt;- df_combined %&gt;%
  group_by(Concentration, Response_Group, Category) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Chi-square or Fisher&#39;s Test -----------------
get_chi_or_fisher &lt;- function(df, ref_group) {
  ref_counts &lt;- df %&gt;%
    ungroup() %&gt;%
    filter(Response_Group == ref_group) %&gt;%
    { ref_tbl &lt;- .[ , c(&quot;Category&quot;, &quot;Count&quot;)]; setNames(ref_tbl$Count, ref_tbl$Category) }
  
  df %&gt;%
    filter(Response_Group != ref_group) %&gt;%
    group_by(Response_Group) %&gt;%
    summarise(
      p_value = {
        group_counts &lt;- Count[Category %in% c(&quot;TOP2A Target Genes&quot;, &quot;Non-TOP2A Genes&quot;)]
        if (length(group_counts) &lt; 2) group_counts &lt;- c(group_counts, 0)
        contingency_table &lt;- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts[&quot;TOP2A Target Genes&quot;], ref_counts[&quot;Non-TOP2A Genes&quot;]
        ), nrow = 2)
        if (any(contingency_table &lt; 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = &quot;drop&quot;
    ) %&gt;%
    mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))
}

# Run test for each concentration
chi_0.1 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.1 µM&quot;), &quot;Non response&quot;)
chi_0.5 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.5 µM&quot;), &quot;Non response&quot;)</code></pre>
<pre><code>Warning: There was 1 warning in `summarise()`.
ℹ In argument: `p_value = { ... }`.
ℹ In group 4: `Response_Group = &quot;DOX-specific response&quot;`.
Caused by warning in `chisq.test()`:
! Chi-squared approximation may be incorrect</code></pre>
<pre class="r"><code>chi_all &lt;- bind_rows(
  chi_0.1 %&gt;% mutate(Concentration = &quot;0.1 µM&quot;),
  chi_0.5 %&gt;% mutate(Concentration = &quot;0.5 µM&quot;)
)

# Merge test results into proportions
proportion_data &lt;- left_join(proportion_data, chi_all, by = c(&quot;Concentration&quot;, &quot;Response_Group&quot;))

# ----------------- Reorder Factor Levels -----------------
proportion_data$Concentration &lt;- factor(proportion_data$Concentration, levels = c(&quot;0.1 µM&quot;, &quot;0.5 µM&quot;))
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
))

# ----------------- Star Label Position -----------------
label_data &lt;- proportion_data %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  summarise(Significance = dplyr::first(Significance), .groups = &quot;drop&quot;) %&gt;%
  filter(!is.na(Significance) &amp; Significance != &quot;&quot;) %&gt;%
  mutate(y_pos = 105)

# ----------------- Final Plot -----------------
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(
    data = label_data,
    aes(x = Response_Group, y = y_pos, label = Significance),
    inherit.aes = FALSE,
    size = 5,
    color = &quot;black&quot;
  ) +
  facet_wrap(~ Concentration, scales = &quot;free_x&quot;) +
  scale_fill_manual(values = c(
    &quot;TOP2A Target Genes&quot; = &quot;#E41A1C&quot;,
    &quot;Non-TOP2A Genes&quot; = &quot;#377EB8&quot;
  )) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +
  labs(
    title = &quot;TOP2A Target Gene Proportions Across\nCormotif Clusters (0.1 and 0.5)&quot;,
    x = &quot;Response Groups&quot;,
    y = &quot;Percentage of Genes&quot;,
    fill = &quot;Gene Category&quot;
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = &quot;bold&quot;),
    axis.title.x = element_text(size = 14, face = &quot;bold&quot;),
    axis.title.y = element_text(size = 14, face = &quot;bold&quot;),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    panel.spacing = unit(1.2, &quot;lines&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2A_Prop-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2A_Prop-1">
Past versions of "Mean TOP2A_Prop-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2A_Prop-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2A_Prop-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-top2a-target-genes-chip_atlas-across-corrmotifs-per-concentration"
class="section level2">
<h2><strong>📌 Mean logFC of TOP2A target genes (ChIP_Atlas) across
corrmotifs (per concentration)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load TOP2A Entrez IDs -----------------
top2a_entrez_ids &lt;- read_csv(&quot;data/TOP2A_target_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate Response Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% top2a_entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% top2a_entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Aggregate DEG Data -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;,
      TRUE ~ NA_character_
    )
  )

deg_mean &lt;- deg_all %&gt;%
  group_by(Entrez_ID, Concentration) %&gt;%
  summarise(logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;)

# ----------------- Merge with CorMotif TOP2A Genes -----------------
top2a_logfc &lt;- inner_join(df_combined, deg_mean, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Group = paste(Response_Group, Concentration)
  )

# ----------------- Define Response Group Order -----------------
group_levels &lt;- c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
)

top2a_logfc$Response_Group &lt;- factor(top2a_logfc$Response_Group, levels = group_levels)

# ----------------- Wilcoxon Test with Output -----------------
comparison_map &lt;- list(
  &quot;CX-DOX mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX only mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX-specific response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX only mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;CX total + DOX early response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX early + CX-DOX mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;
)

wilcoxon_results &lt;- lapply(names(comparison_map), function(resp_group) {
  control_group &lt;- comparison_map[[resp_group]]
  resp_vals &lt;- top2a_logfc$logFC[top2a_logfc$Group == resp_group]
  control_vals &lt;- top2a_logfc$logFC[top2a_logfc$Group == control_group]

  if (length(resp_vals) &gt;= 2 &amp;&amp; length(control_vals) &gt;= 2) {
    test_result &lt;- wilcox.test(resp_vals, control_vals)
    pval &lt;- test_result$p.value
    stat &lt;- test_result$statistic
    label &lt;- case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01  ~ &quot;**&quot;,
      pval &lt; 0.05  ~ &quot;*&quot;,
      TRUE ~ &quot;&quot;
    )
    y_pos &lt;- max(resp_vals, na.rm = TRUE) + 0.5
    return(data.frame(
      Response_Group = resp_group,
      Control_Group = control_group,
      W_statistic = as.numeric(stat),
      P_value = signif(pval, 4),
      Significance = label,
      y_pos = y_pos
    ))
  } else {
    return(data.frame(
      Response_Group = resp_group,
      Control_Group = control_group,
      W_statistic = NA,
      P_value = NA,
      Significance = &quot;&quot;,
      y_pos = NA
    ))
  }
}) %&gt;% bind_rows()

# ----------------- Prepare Annotation -----------------
label_data &lt;- wilcoxon_results %&gt;%
  separate(Response_Group, into = c(&quot;Response_Group&quot;, &quot;Concentration&quot;), sep = &quot; (?=0\\.)&quot;) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = group_levels))

# ----------------- Plot -----------------
ggplot(top2a_logfc, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = Significance),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~Concentration, scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Mean logFC of TOP2A Target Genes\nAcross CorMotif Response Groups (ChIP atlas)&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2A_ChIP%20logFC-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2A_ChIP-logFC-1">
Past versions of "Mean TOP2A_ChIP logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2A_ChIP-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2A_ChIP logFC-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-top2a-target-genes-chip_atlas-across-corrmotifs-faceted-by-concentration-timepoint"
class="section level2">
<h2><strong>📌 Mean logFC of TOP2A target genes (ChIP_Atlas) across
corrmotifs (Faceted by concentration × timepoint)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load TOP2A Entrez IDs -----------------
top2a_entrez_ids &lt;- read_csv(&quot;data/TOP2A_target_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Response Group Data -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorMotif Group Membership -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% top2a_entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% top2a_entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Annotate DEG Files -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    ),
    Timepoint = factor(case_when(
      str_detect(Condition, &quot;_3$&quot;) ~ &quot;3 hr&quot;,
      str_detect(Condition, &quot;_24$&quot;) ~ &quot;24 hr&quot;,
      str_detect(Condition, &quot;_48$&quot;) ~ &quot;48 hr&quot;
    ), levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))
  )

# ----------------- Merge with TOP2A Targets and Annotate -----------------
top2a_all &lt;- inner_join(df_combined, deg_all, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  filter(Entrez_ID %in% top2a_entrez_ids) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID, column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response&quot;, &quot;CX-DOX mid-late response&quot;, &quot;DOX-specific response&quot;,
      &quot;DOX only mid-late response&quot;, &quot;CX total + DOX early response&quot;, &quot;DOX early + CX-DOX mid-late response&quot;
    ))
  )

# ----------------- Wilcoxon Test -----------------
comparison_groups &lt;- levels(top2a_all$Response_Group)[-1]
stat_results &lt;- list()

for (conc in unique(top2a_all$Concentration)) {
  for (tp in levels(top2a_all$Timepoint)) {
    sub &lt;- top2a_all %&gt;% filter(Concentration == conc, Timepoint == tp)
    for (grp in comparison_groups) {
      if (grp %in% unique(sub$Response_Group)) {
        test &lt;- tryCatch({
          wilcox.test(
            sub$logFC[sub$Response_Group == grp],
            sub$logFC[sub$Response_Group == &quot;Non response&quot;]
          )
        }, error = function(e) NULL)
        if (!is.null(test) &amp;&amp; !is.nan(test$p.value)) {
          stat_results[[length(stat_results) + 1]] &lt;- data.frame(
            Response_Group = grp,
            Concentration = conc,
            Timepoint = tp,
            p_value = signif(test$p.value, 3),
            label = case_when(
              test$p.value &lt; 0.001 ~ &quot;***&quot;,
              test$p.value &lt; 0.01 ~ &quot;**&quot;,
              test$p.value &lt; 0.05 ~ &quot;*&quot;,
              TRUE ~ &quot;&quot;
            ),
            y_pos = max(sub$logFC[sub$Response_Group == grp], na.rm = TRUE) + 0.5
          )
        }
      }
    }
  }
}

label_data &lt;- bind_rows(stat_results) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = levels(top2a_all$Response_Group)))

# ----------------- Plot -----------------
ggplot(top2a_all, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7, na.rm = TRUE) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 4, vjust = 0) +
  facet_grid(rows = vars(factor(Timepoint, levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))),
             cols = vars(Concentration), scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;TOP2A Target Gene logFC by\nCorMotif Group, Timepoint, and Concentration (ChIP Atlas)&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2A_ChIP_Atlas%20logFC1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2A_ChIP_Atlas-logFC1-1">
Past versions of "Mean TOP2A_ChIP_Atlas logFC1-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2A_ChIP_Atlas-logFC1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2A_ChIP_Atlas logFC1-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-top2a-target-genes-chip_atlas-across-corrmotifs-faceted-by-concentration-timepoint-drug-cx-vs-dox-side-by-side"
class="section level2">
<h2><strong>📌 Mean logFC of TOP2A target genes (ChIP_Atlas) across
corrmotifs (Faceted by concentration × timepoint × drug (CX vs DOX
side-by-side))</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load TOP2A ChIP-Atlas Entrez IDs -----------------
top2a_entrez_ids &lt;- read_csv(&quot;data/TOP2A_target_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Groups (0.1 µM) -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.1 µM DEG Files -----------------
deg_0.1 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.1 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late\nresponse (0.1 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% top2a_entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.1 &lt;- deg_0.1 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.1 µM)&quot;,
      &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.1 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.1 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.1 µM TOP2A) -----------------
ggplot(deg_0.1, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;TOP2A Target Gene logFC by Drug and Response Group (0.1 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2A_ChIP_Atlas%20logFC3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2A_ChIP_Atlas-logFC3-1">
Past versions of "Mean TOP2A_ChIP_Atlas logFC3-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2A_ChIP_Atlas-logFC3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2A_ChIP_Atlas logFC3-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>################### TOP2A Target Genes (ChIP-Atlas) — 0.5 µM ###################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load TOP2A Entrez IDs -----------------
top2a_entrez_ids &lt;- read_csv(&quot;data/TOP2A_target_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Groups (0.5 µM) -----------------
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.5 µM DEG Files -----------------
deg_0.5 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.5 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 µM)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% top2a_entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.5 &lt;- deg_0.5 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.5 µM)&quot;,
      &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.5 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.5 µM TOP2A ChIP-Atlas) -----------------
ggplot(deg_0.5, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;TOP2A (ChIP-Atlas) Target Gene logFC by Drug and Response Group (0.5 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2A_ChIP_Atlas%20logFC3-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2A_ChIP_Atlas-logFC3-2">
Past versions of "Mean TOP2A_ChIP_Atlas logFC3-2.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2A_ChIP_Atlas-logFC3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2A_ChIP_Atlas logFC3-2.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="proportion-of-top2a-target-genes-rpe-1-across-corrmotifs"
class="section level2">
<h2><strong>📌 Proportion of TOP2A target genes (RPE-1) across
corrmotifs</strong></h2>
<pre class="r"><code># ----------------- Load Required Libraries -----------------
library(dplyr)
library(ggplot2)
library(readr)

# ----------------- Load TOP2A (RPE-1) Entrez IDs -----------------
top2a_lit_ids &lt;- read_csv(&quot;data/TOP2A_target_lit_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorrMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% top2a_lit_ids, &quot;TOP2A (RPE-1) Targets&quot;, &quot;Non-TOP2A Genes&quot;),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% top2a_lit_ids, &quot;TOP2A (RPE-1) Targets&quot;, &quot;Non-TOP2A Genes&quot;),
    Concentration = &quot;0.5 µM&quot;
  )

# ----------------- Combine Data -----------------
df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Set Category Order -----------------
df_combined$Category &lt;- factor(df_combined$Category, levels = c(&quot;TOP2A (RPE-1) Targets&quot;, &quot;Non-TOP2A Genes&quot;))

# ----------------- Calculate Proportions -----------------
proportion_data &lt;- df_combined %&gt;%
  group_by(Concentration, Response_Group, Category) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Chi-square or Fisher&#39;s Test -----------------
get_chi_or_fisher &lt;- function(df, ref_group) {
  ref_counts &lt;- df %&gt;%
    ungroup() %&gt;%
    filter(Response_Group == ref_group) %&gt;%
    { ref_tbl &lt;- .[ , c(&quot;Category&quot;, &quot;Count&quot;)]; setNames(ref_tbl$Count, ref_tbl$Category) }
  
  df %&gt;%
    filter(Response_Group != ref_group) %&gt;%
    group_by(Response_Group) %&gt;%
    summarise(
      p_value = {
        group_counts &lt;- Count[Category %in% c(&quot;TOP2A (RPE-1) Targets&quot;, &quot;Non-TOP2A Genes&quot;)]
        if (length(group_counts) &lt; 2) group_counts &lt;- c(group_counts, 0)
        contingency_table &lt;- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts[&quot;TOP2A (RPE-1) Targets&quot;], ref_counts[&quot;Non-TOP2A Genes&quot;]
        ), nrow = 2)
        if (any(contingency_table &lt; 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = &quot;drop&quot;
    ) %&gt;%
    mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))
}

# Run test for each concentration
chi_0.1 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.1 µM&quot;), &quot;Non response&quot;)
chi_0.5 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.5 µM&quot;), &quot;Non response&quot;)

chi_all &lt;- bind_rows(
  chi_0.1 %&gt;% mutate(Concentration = &quot;0.1 µM&quot;),
  chi_0.5 %&gt;% mutate(Concentration = &quot;0.5 µM&quot;)
)

# Merge test results into proportions
proportion_data &lt;- left_join(proportion_data, chi_all, by = c(&quot;Concentration&quot;, &quot;Response_Group&quot;))

# ----------------- Reorder Factor Levels -----------------
proportion_data$Concentration &lt;- factor(proportion_data$Concentration, levels = c(&quot;0.1 µM&quot;, &quot;0.5 µM&quot;))
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
))

# ----------------- Star Label Position -----------------
label_data &lt;- proportion_data %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  summarise(Significance = dplyr::first(Significance), .groups = &quot;drop&quot;) %&gt;%
  filter(!is.na(Significance) &amp; Significance != &quot;&quot;) %&gt;%
  mutate(y_pos = 105)

# ----------------- Final Plot -----------------
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(
    data = label_data,
    aes(x = Response_Group, y = y_pos, label = Significance),
    inherit.aes = FALSE,
    size = 5,
    color = &quot;black&quot;
  ) +
  facet_wrap(~ Concentration, scales = &quot;free_x&quot;) +
  scale_fill_manual(values = c(
    &quot;TOP2A (RPE-1) Targets&quot; = &quot;#E41A1C&quot;,
    &quot;Non-TOP2A Genes&quot; = &quot;#377EB8&quot;
  )) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +
  labs(
    title = &quot;TOP2A (RPE-1) Target Gene Proportions Across\nCormotif Clusters (0.1 and 0.5)&quot;,
    x = &quot;Response Groups&quot;,
    y = &quot;Percentage of Genes&quot;,
    fill = &quot;Gene Category&quot;
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = &quot;bold&quot;),
    axis.title.x = element_text(size = 14, face = &quot;bold&quot;),
    axis.title.y = element_text(size = 14, face = &quot;bold&quot;),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    panel.spacing = unit(1.2, &quot;lines&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/TOP2A_lit_prop%20logFC-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-TOP2A_lit_prop-logFC-1">
Past versions of "TOP2A_lit_prop logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-TOP2A_lit_prop-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/TOP2A_lit_prop logFC-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-top2a-target-genes-rpe-1-across-corrmotifs-per-concentration"
class="section level2">
<h2><strong>📌 Mean logFC of TOP2A target genes (RPE-1) across
corrmotifs (per concentration)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load TOP2A (RPE-1) Entrez IDs -----------------
top2a_lit_ids &lt;- read_csv(&quot;data/TOP2A_target_lit_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate Response Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% top2a_lit_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% top2a_lit_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Aggregate DEG Data -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;,
      TRUE ~ NA_character_
    )
  )

deg_mean &lt;- deg_all %&gt;%
  group_by(Entrez_ID, Concentration) %&gt;%
  summarise(logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;)

# ----------------- Merge with CorMotif Genes -----------------
top2a_lit_logfc &lt;- inner_join(df_combined, deg_mean, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Group = paste(Response_Group, Concentration)
  )

# ----------------- Define Response Group Order -----------------
group_levels &lt;- c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
)

top2a_lit_logfc$Response_Group &lt;- factor(top2a_lit_logfc$Response_Group, levels = group_levels)

# ----------------- Wilcoxon Test with Output -----------------
comparison_map &lt;- list(
  &quot;CX-DOX mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX only mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX-specific response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX only mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;CX total + DOX early response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX early + CX-DOX mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;
)

wilcoxon_results &lt;- lapply(names(comparison_map), function(resp_group) {
  control_group &lt;- comparison_map[[resp_group]]
  resp_vals &lt;- top2a_lit_logfc$logFC[top2a_lit_logfc$Group == resp_group]
  control_vals &lt;- top2a_lit_logfc$logFC[top2a_lit_logfc$Group == control_group]

  if (length(resp_vals) &gt;= 2 &amp;&amp; length(control_vals) &gt;= 2) {
    test_result &lt;- wilcox.test(resp_vals, control_vals)
    pval &lt;- test_result$p.value
    stat &lt;- test_result$statistic
    label &lt;- case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01  ~ &quot;**&quot;,
      pval &lt; 0.05  ~ &quot;*&quot;,
      TRUE ~ &quot;&quot;
    )
    y_pos &lt;- max(resp_vals, na.rm = TRUE) + 0.5
    return(data.frame(
      Response_Group = resp_group,
      Control_Group = control_group,
      W_statistic = as.numeric(stat),
      P_value = signif(pval, 4),
      Significance = label,
      y_pos = y_pos
    ))
  } else {
    return(data.frame(
      Response_Group = resp_group,
      Control_Group = control_group,
      W_statistic = NA,
      P_value = NA,
      Significance = &quot;&quot;,
      y_pos = NA
    ))
  }
}) %&gt;% bind_rows()

# ----------------- Prepare Annotation -----------------
label_data &lt;- wilcoxon_results %&gt;%
  separate(Response_Group, into = c(&quot;Response_Group&quot;, &quot;Concentration&quot;), sep = &quot; (?=0\\.)&quot;) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = group_levels))

# ----------------- Plot -----------------
ggplot(top2a_lit_logfc, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = Significance),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~Concentration, scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Mean logFC of Literature-Based TOP2A Targets (RPE-1)\nAcross CorMotif Response Groups&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2A_lit_1%20logFC-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2A_lit_1-logFC-1">
Past versions of "Mean TOP2A_lit_1 logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2A_lit_1-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2A_lit_1 logFC-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-top2a-target-genes-rpe-1-across-corrmotifs-faceted-by-concentration-timepoint"
class="section level2">
<h2><strong>📌 Mean logFC of TOP2A target genes (RPE-1) across
corrmotifs (Faceted by concentration × timepoint)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load TOP2A (RPE-1) Entrez IDs -----------------
top2a_lit_ids &lt;- read_csv(&quot;data/TOP2A_target_lit_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Response Group Data -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorMotif Group Membership -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% top2a_lit_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% top2a_lit_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Annotate DEG Files -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    ),
    Timepoint = factor(case_when(
      str_detect(Condition, &quot;_3$&quot;) ~ &quot;3 hr&quot;,
      str_detect(Condition, &quot;_24$&quot;) ~ &quot;24 hr&quot;,
      str_detect(Condition, &quot;_48$&quot;) ~ &quot;48 hr&quot;
    ), levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))
  )

# ----------------- Merge with TOP2A (Lit) Targets and Annotate -----------------
top2a_lit_all &lt;- inner_join(df_combined, deg_all, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  filter(Entrez_ID %in% top2a_lit_ids) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID, column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response&quot;, &quot;CX-DOX mid-late response&quot;, &quot;DOX-specific response&quot;,
      &quot;DOX only mid-late response&quot;, &quot;CX total + DOX early response&quot;, &quot;DOX early + CX-DOX mid-late response&quot;
    ))
  )

# ----------------- Wilcoxon Test -----------------
comparison_groups &lt;- levels(top2a_lit_all$Response_Group)[-1]
stat_results &lt;- list()

for (conc in unique(top2a_lit_all$Concentration)) {
  for (tp in levels(top2a_lit_all$Timepoint)) {
    sub &lt;- top2a_lit_all %&gt;% filter(Concentration == conc, Timepoint == tp)
    for (grp in comparison_groups) {
      if (grp %in% unique(sub$Response_Group)) {
        test &lt;- tryCatch({
          wilcox.test(
            sub$logFC[sub$Response_Group == grp],
            sub$logFC[sub$Response_Group == &quot;Non response&quot;]
          )
        }, error = function(e) NULL)
        if (!is.null(test) &amp;&amp; !is.nan(test$p.value)) {
          stat_results[[length(stat_results) + 1]] &lt;- data.frame(
            Response_Group = grp,
            Concentration = conc,
            Timepoint = tp,
            p_value = signif(test$p.value, 3),
            label = case_when(
              test$p.value &lt; 0.001 ~ &quot;***&quot;,
              test$p.value &lt; 0.01 ~ &quot;**&quot;,
              test$p.value &lt; 0.05 ~ &quot;*&quot;,
              TRUE ~ &quot;&quot;
            ),
            y_pos = max(sub$logFC[sub$Response_Group == grp], na.rm = TRUE) + 0.5
          )
        }
      }
    }
  }
}

label_data &lt;- bind_rows(stat_results) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = levels(top2a_lit_all$Response_Group)))

# ----------------- Plot -----------------
ggplot(top2a_lit_all, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7, na.rm = TRUE) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 4, vjust = 0) +
  facet_grid(rows = vars(factor(Timepoint, levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))),
             cols = vars(Concentration), scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;TOP2A Target Gene (RPE-1 cell) logFC by\nCorMotif Group, Timepoint, and Concentration&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2A_lit_2%20logFC1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2A_lit_2-logFC1-1">
Past versions of "Mean TOP2A_lit_2 logFC1-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2A_lit_2-logFC1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2A_lit_2 logFC1-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-top2a-target-genes-rpe-1-across-corrmotifs-faceted-by-concentration-timepoint-drug-cx-vs-dox-side-by-side"
class="section level2">
<h2><strong>📌 Mean logFC of TOP2A target genes (RPE-1) across
corrmotifs (Faceted by concentration × timepoint × drug (CX vs DOX
side-by-side))</strong></h2>
<pre class="r"><code>################### TOP2A Target Genes (RPE-1 Literature) — 0.1 µM ###################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load TOP2A (RPE-1) Entrez IDs -----------------
top2a_lit_ids &lt;- read_csv(&quot;data/TOP2A_target_lit_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Groups (0.1 µM) -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.1 µM DEG Files -----------------
deg_0.1 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.1 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late\nresponse (0.1 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% top2a_lit_ids)

# ----------------- Set Factor Levels -----------------
deg_0.1 &lt;- deg_0.1 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.1 µM)&quot;,
      &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.1 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.1 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.1 µM TOP2A RPE-1) -----------------
ggplot(deg_0.1, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;TOP2A (RPE-1) Target Gene logFC by Drug and Response Group (0.1 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2A_lit_3%20logFC3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2A_lit_3-logFC3-1">
Past versions of "Mean TOP2A_lit_3 logFC3-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2A_lit_3-logFC3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2A_lit_3 logFC3-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>################### TOP2A Target Genes (RPE-1 Literature) — 0.5 µM ###################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load TOP2A (RPE-1) Entrez IDs -----------------
top2a_lit_ids &lt;- read_csv(&quot;data/TOP2A_target_lit_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load CorMotif Groups (0.5 µM) -----------------
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.5 µM DEG Files -----------------
deg_0.5 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.5 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 µM)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% top2a_lit_ids)

# ----------------- Set Factor Levels -----------------
deg_0.5 &lt;- deg_0.5 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.5 µM)&quot;,
      &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.5 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.5 µM TOP2A RPE-1) -----------------
ggplot(deg_0.5, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;TOP2A (RPE-1) Target Gene logFC by Drug and Response Group (0.5 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20TOP2A_lit_3%20logFC3-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-TOP2A_lit_3-logFC3-2">
Past versions of "Mean TOP2A_lit_3 logFC3-2.png"
</button>
</p>
<div id="fig-no-spaces-Mean-TOP2A_lit_3-logFC3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean TOP2A_lit_3 logFC3-2.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="proportion-of-anthracycline-cardiotoxicity-genes-across-corrmotifs"
class="section level2">
<h2><strong>📌 Proportion of Anthracycline Cardiotoxicity genes across
corrmotifs</strong></h2>
<pre class="r"><code># ----------------- Load Required Libraries -----------------
library(dplyr)
library(ggplot2)

# ----------------- AC Cardiotoxicity Entrez IDs -----------------
ac_cardio_ids &lt;- c(
  6272, 8029, 11128, 79899, 54477, 121665, 5095, 22863, 57161, 4692,
  8214, 23151, 56606, 108, 22999, 56895, 9603, 3181, 4023, 10499,
  92949, 4363, 10057, 5243, 5244, 5880, 1535, 2950, 847, 5447,
  3038, 3077, 4846, 3958, 23327, 29899, 23155, 80856, 55020, 78996,
  23262, 150383, 9620, 79730, 344595, 5066, 6251, 3482, 9588, 339416,
  7292, 55157, 87769, 23409, 720, 3107, 54535, 1590, 80059, 7991,
  57110, 8803, 323, 54826, 5916, 23371, 283337, 64078, 80010, 1933,
  10818, 51020
) %&gt;% as.character()

# ----------------- Load CorrMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% ac_cardio_ids, &quot;AC-Toxicity Genes&quot;, &quot;Non-AC Genes&quot;),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% ac_cardio_ids, &quot;AC-Toxicity Genes&quot;, &quot;Non-AC Genes&quot;),
    Concentration = &quot;0.5 µM&quot;
  )

# ----------------- Combine Data -----------------
df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Set Category Order -----------------
df_combined$Category &lt;- factor(df_combined$Category, levels = c(&quot;AC-Toxicity Genes&quot;, &quot;Non-AC Genes&quot;))

# ----------------- Calculate Proportions -----------------
proportion_data &lt;- df_combined %&gt;%
  group_by(Concentration, Response_Group, Category) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Chi-square or Fisher&#39;s Test -----------------
get_chi_or_fisher &lt;- function(df, ref_group) {
  ref_counts &lt;- df %&gt;%
    ungroup() %&gt;%
    filter(Response_Group == ref_group) %&gt;%
    { ref_tbl &lt;- .[ , c(&quot;Category&quot;, &quot;Count&quot;)]; setNames(ref_tbl$Count, ref_tbl$Category) }
  
  df %&gt;%
    filter(Response_Group != ref_group) %&gt;%
    group_by(Response_Group) %&gt;%
    summarise(
      p_value = {
        group_counts &lt;- Count[Category %in% c(&quot;AC-Toxicity Genes&quot;, &quot;Non-AC Genes&quot;)]
        if (length(group_counts) &lt; 2) group_counts &lt;- c(group_counts, 0)
        contingency_table &lt;- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts[&quot;AC-Toxicity Genes&quot;], ref_counts[&quot;Non-AC Genes&quot;]
        ), nrow = 2)
        if (any(contingency_table &lt; 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = &quot;drop&quot;
    ) %&gt;%
    mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))
}

# Run test for each concentration
chi_0.1 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.1 µM&quot;), &quot;Non response&quot;)
chi_0.5 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.5 µM&quot;), &quot;Non response&quot;)

chi_all &lt;- bind_rows(
  chi_0.1 %&gt;% mutate(Concentration = &quot;0.1 µM&quot;),
  chi_0.5 %&gt;% mutate(Concentration = &quot;0.5 µM&quot;)
)

# Merge test results into proportions
proportion_data &lt;- left_join(proportion_data, chi_all, by = c(&quot;Concentration&quot;, &quot;Response_Group&quot;))

# ----------------- Reorder Factor Levels -----------------
proportion_data$Concentration &lt;- factor(proportion_data$Concentration, levels = c(&quot;0.1 µM&quot;, &quot;0.5 µM&quot;))
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
))

# ----------------- Star Label Position -----------------
label_data &lt;- proportion_data %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  summarise(Significance = dplyr::first(Significance), .groups = &quot;drop&quot;) %&gt;%
  filter(!is.na(Significance) &amp; Significance != &quot;&quot;) %&gt;%
  mutate(y_pos = 105)

# ----------------- Final Plot -----------------
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(
    data = label_data,
    aes(x = Response_Group, y = y_pos, label = Significance),
    inherit.aes = FALSE,
    size = 5,
    color = &quot;black&quot;
  ) +
  facet_wrap(~ Concentration, scales = &quot;free_x&quot;) +
  scale_fill_manual(values = c(
    &quot;AC-Toxicity Genes&quot; = &quot;#E41A1C&quot;,
    &quot;Non-AC Genes&quot; = &quot;#377EB8&quot;
  )) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +
  labs(
    title = &quot;AC Cardiotoxicity Gene Proportions Across\nCormotif Clusters (0.1 and 0.5)&quot;,
    x = &quot;Response Groups&quot;,
    y = &quot;Percentage of Genes&quot;,
    fill = &quot;Gene Category&quot;
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = &quot;bold&quot;),
    axis.title.x = element_text(size = 14, face = &quot;bold&quot;),
    axis.title.y = element_text(size = 14, face = &quot;bold&quot;),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    panel.spacing = unit(1.2, &quot;lines&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Ac_tox%20prop-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Ac_tox-prop-1">
Past versions of "Ac_tox prop-1.png"
</button>
</p>
<div id="fig-no-spaces-Ac_tox-prop-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Ac_tox prop-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-anthracycline-cardiotoxicity-genes-across-corrmotifs-per-concentration"
class="section level2">
<h2><strong>📌 Mean logFC of Anthracycline Cardiotoxicity genes across
corrmotifs (per concentration)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- AC Cardiotoxicity Entrez IDs -----------------
entrez_ids &lt;- c(
  6272, 8029, 11128, 79899, 54477, 121665, 5095, 22863, 57161, 4692,
  8214, 23151, 56606, 108, 22999, 56895, 9603, 3181, 4023, 10499,
  92949, 4363, 10057, 5243, 5244, 5880, 1535, 2950, 847, 5447,
  3038, 3077, 4846, 3958, 23327, 29899, 23155, 80856, 55020, 78996,
  23262, 150383, 9620, 79730, 344595, 5066, 6251, 3482, 9588, 339416,
  7292, 55157, 87769, 23409, 720, 3107, 54535, 1590, 80059, 7991,
  57110, 8803, 323, 54826, 5916, 23371, 283337, 64078, 80010, 1933,
  10818, 51020
) %&gt;% as.character()

# ----------------- Load CorMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate Response Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Aggregate DEG Data -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;,
      TRUE ~ NA_character_
    )
  )

deg_mean &lt;- deg_all %&gt;%
  group_by(Entrez_ID, Concentration) %&gt;%
  summarise(logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;)

# ----------------- Merge with CorMotif Genes -----------------
ac_logfc &lt;- inner_join(df_combined, deg_mean, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Group = paste(Response_Group, Concentration)
  )

# ----------------- Define Group Order -----------------
group_levels &lt;- c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
)

ac_logfc$Response_Group &lt;- factor(ac_logfc$Response_Group, levels = group_levels)

# ----------------- Wilcoxon Test -----------------
comparison_map &lt;- list(
  &quot;CX-DOX mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX only mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX-specific response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX only mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;CX total + DOX early response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX early + CX-DOX mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;
)

wilcoxon_results &lt;- lapply(names(comparison_map), function(resp_group) {
  control_group &lt;- comparison_map[[resp_group]]
  resp_vals &lt;- ac_logfc$logFC[ac_logfc$Group == resp_group]
  control_vals &lt;- ac_logfc$logFC[ac_logfc$Group == control_group]

  if (length(resp_vals) &gt;= 2 &amp;&amp; length(control_vals) &gt;= 2) {
    test_result &lt;- wilcox.test(resp_vals, control_vals)
    pval &lt;- test_result$p.value
    stat &lt;- test_result$statistic
    label &lt;- case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01  ~ &quot;**&quot;,
      pval &lt; 0.05  ~ &quot;*&quot;,
      TRUE ~ &quot;&quot;
    )
    y_pos &lt;- max(resp_vals, na.rm = TRUE) + 0.5
    return(data.frame(
      Group = resp_group,
      y_pos = y_pos,
      label = label,
      P_Value = signif(pval, 4)
    ))
  } else {
    return(NULL)
  }
}) %&gt;% bind_rows()

# ----------------- Prepare Annotation -----------------
label_data &lt;- wilcoxon_results %&gt;%
  separate(Group, into = c(&quot;Response_Group&quot;, &quot;Concentration&quot;), sep = &quot; (?=0\\.)&quot;) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = group_levels))

# ----------------- Plot -----------------
ggplot(ac_logfc, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~Concentration, scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Mean logFC of AC Cardiotoxicity Genes\nAcross CorMotif Response Groups&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20Ac_tox%20logFC-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-Ac_tox-logFC-1">
Past versions of "Mean Ac_tox logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-Ac_tox-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean Ac_tox logFC-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-anthracycline-cardiotoxicity-genes-across-corrmotifs-faceted-by-concentration-timepoint"
class="section level2">
<h2><strong>📌 Mean logFC of Anthracycline Cardiotoxicity genes across
corrmotifs (Faceted by concentration × timepoint)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- AC Cardiotoxicity Entrez IDs -----------------
entrez_ids &lt;- c(
  6272, 8029, 11128, 79899, 54477, 121665, 5095, 22863, 57161, 4692,
  8214, 23151, 56606, 108, 22999, 56895, 9603, 3181, 4023, 10499,
  92949, 4363, 10057, 5243, 5244, 5880, 1535, 2950, 847, 5447,
  3038, 3077, 4846, 3958, 23327, 29899, 23155, 80856, 55020, 78996,
  23262, 150383, 9620, 79730, 344595, 5066, 6251, 3482, 9588, 339416,
  7292, 55157, 87769, 23409, 720, 3107, 54535, 1590, 80059, 7991,
  57110, 8803, 323, 54826, 5916, 23371, 283337, 64078, 80010, 1933,
  10818, 51020
) %&gt;% as.character()

# ----------------- Load CorMotif Response Group Data -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorMotif Group Membership -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Annotate DEG Files -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    ),
    Timepoint = factor(case_when(
      str_detect(Condition, &quot;_3$&quot;) ~ &quot;3 hr&quot;,
      str_detect(Condition, &quot;_24$&quot;) ~ &quot;24 hr&quot;,
      str_detect(Condition, &quot;_48$&quot;) ~ &quot;48 hr&quot;
    ), levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))
  )

# ----------------- Merge with AC Cardiotoxicity Targets -----------------
ac_all &lt;- inner_join(df_combined, deg_all, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID, column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response&quot;, &quot;CX-DOX mid-late response&quot;, &quot;DOX-specific response&quot;,
      &quot;DOX only mid-late response&quot;, &quot;CX total + DOX early response&quot;, &quot;DOX early + CX-DOX mid-late response&quot;
    ))
  )

# ----------------- Wilcoxon Test -----------------
comparison_groups &lt;- levels(ac_all$Response_Group)[-1]
stat_results &lt;- list()

for (conc in unique(ac_all$Concentration)) {
  for (tp in levels(ac_all$Timepoint)) {
    sub &lt;- ac_all %&gt;% filter(Concentration == conc, Timepoint == tp)
    for (grp in comparison_groups) {
      if (grp %in% unique(sub$Response_Group)) {
        test &lt;- tryCatch({
          wilcox.test(
            sub$logFC[sub$Response_Group == grp],
            sub$logFC[sub$Response_Group == &quot;Non response&quot;]
          )
        }, error = function(e) NULL)
        if (!is.null(test) &amp;&amp; !is.nan(test$p.value)) {
          stat_results[[length(stat_results) + 1]] &lt;- data.frame(
            Response_Group = grp,
            Concentration = conc,
            Timepoint = tp,
            p_value = signif(test$p.value, 3),
            label = case_when(
              test$p.value &lt; 0.001 ~ &quot;***&quot;,
              test$p.value &lt; 0.01 ~ &quot;**&quot;,
              test$p.value &lt; 0.05 ~ &quot;*&quot;,
              TRUE ~ &quot;&quot;
            ),
            y_pos = max(sub$logFC[sub$Response_Group == grp], na.rm = TRUE) + 0.5
          )
        }
      }
    }
  }
}

label_data &lt;- bind_rows(stat_results) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = levels(ac_all$Response_Group)))

# ----------------- Plot -----------------
ggplot(ac_all, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7, na.rm = TRUE) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 4, vjust = 0) +
  facet_grid(rows = vars(factor(Timepoint, levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))),
             cols = vars(Concentration), scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;AC Cardiotoxicity Gene logFC by\nCorMotif Group, Timepoint, and Concentration&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20Ac_tox_2%20logFC1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-Ac_tox_2-logFC1-1">
Past versions of "Mean Ac_tox_2 logFC1-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-Ac_tox_2-logFC1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean Ac_tox_2 logFC1-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-anthracycline-cardiotoxicity-genes-across-corrmotifs-faceted-by-concentration-timepoint-drug-cx-vs-dox-side-by-side"
class="section level2">
<h2><strong>📌 Mean logFC of Anthracycline Cardiotoxicity genes across
corrmotifs (Faceted by concentration × timepoint × drug (CX vs DOX
side-by-side))</strong></h2>
<pre class="r"><code>################### AC Cardiotoxicity Genes — 0.1 µM ###################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- AC Cardiotoxicity Entrez IDs -----------------
ac_ids &lt;- c(
  6272, 8029, 11128, 79899, 54477, 121665, 5095, 22863, 57161, 4692,
  8214, 23151, 56606, 108, 22999, 56895, 9603, 3181, 4023, 10499,
  92949, 4363, 10057, 5243, 5244, 5880, 1535, 2950, 847, 5447,
  3038, 3077, 4846, 3958, 23327, 29899, 23155, 80856, 55020, 78996,
  23262, 150383, 9620, 79730, 344595, 5066, 6251, 3482, 9588, 339416,
  7292, 55157, 87769, 23409, 720, 3107, 54535, 1590, 80059, 7991,
  57110, 8803, 323, 54826, 5916, 23371, 283337, 64078, 80010, 1933,
  10818, 51020
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.1 µM) -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.1 µM DEG Files -----------------
deg_0.1 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.1 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late\nresponse (0.1 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% ac_ids)

# ----------------- Set Factor Levels -----------------
deg_0.1 &lt;- deg_0.1 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.1 µM)&quot;,
      &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.1 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.1 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.1 µM AC Cardiotoxicity) -----------------
ggplot(deg_0.1, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;AC Cardiotoxicity Gene logFC by Drug and Response Group (0.1 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20Ac_tox_3%20logFC3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-Ac_tox_3-logFC3-1">
Past versions of "Mean Ac_tox_3 logFC3-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-Ac_tox_3-logFC3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean Ac_tox_3 logFC3-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>################### AC Genes — 0.5 µM ###################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- AC Cardiotoxicity Entrez IDs -----------------
ac_ids &lt;- c(
  6272, 8029, 11128, 79899, 54477, 121665, 5095, 22863, 57161, 4692,
  8214, 23151, 56606, 108, 22999, 56895, 9603, 3181, 4023, 10499,
  92949, 4363, 10057, 5243, 5244, 5880, 1535, 2950, 847, 5447,
  3038, 3077, 4846, 3958, 23327, 29899, 23155, 80856, 55020, 78996,
  23262, 150383, 9620, 79730, 344595, 5066, 6251, 3482, 9588, 339416,
  7292, 55157, 87769, 23409, 720, 3107, 54535, 1590, 80059, 7991,
  57110, 8803, 323, 54826, 5916, 23371, 283337, 64078, 80010, 1933,
  10818, 51020
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.5 µM) -----------------
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.5 µM DEG Files -----------------
deg_0.5 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.5 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 µM)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% ac_ids)

# ----------------- Set Factor Levels -----------------
deg_0.5 &lt;- deg_0.5 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.5 µM)&quot;,
      &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.5 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.5 µM AC Genes) -----------------
ggplot(deg_0.5, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;AC Gene logFC by Drug and Response Group (0.5 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20Ac_tox_3%20logFC3-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-Ac_tox_3-logFC3-2">
Past versions of "Mean Ac_tox_3 logFC3-2.png"
</button>
</p>
<div id="fig-no-spaces-Mean-Ac_tox_3-logFC3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean Ac_tox_3 logFC3-2.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="proportion-of-dox-cardiotoxicity-genes-across-corrmotifs"
class="section level2">
<h2><strong>📌 Proportion of DOX Cardiotoxicity genes across
corrmotifs</strong></h2>
<pre class="r"><code># ----------------- Load Required Libraries -----------------
library(dplyr)
library(ggplot2)

# ----------------- DOX Cardiotoxicity Entrez IDs -----------------
dox_cardio_ids &lt;- c(
  847, 873, 2064, 2878, 2944, 3038, 4846, 51196, 5880, 6687,
  7799, 4292, 5916, 3077, 51310, 9154, 64078, 5244, 10057, 10060,
  89845, 56853, 4625, 1573, 79890
) %&gt;% as.character()

# ----------------- Load CorrMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% dox_cardio_ids, &quot;DOX-Toxicity Genes&quot;, &quot;Non-DOX Genes&quot;),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% dox_cardio_ids, &quot;DOX-Toxicity Genes&quot;, &quot;Non-DOX Genes&quot;),
    Concentration = &quot;0.5 µM&quot;
  )

# ----------------- Combine Data -----------------
df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Set Category Order -----------------
df_combined$Category &lt;- factor(df_combined$Category, levels = c(&quot;DOX-Toxicity Genes&quot;, &quot;Non-DOX Genes&quot;))

# ----------------- Calculate Proportions -----------------
proportion_data &lt;- df_combined %&gt;%
  group_by(Concentration, Response_Group, Category) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Chi-square or Fisher&#39;s Test -----------------
get_chi_or_fisher &lt;- function(df, ref_group) {
  ref_counts &lt;- df %&gt;%
    ungroup() %&gt;%
    filter(Response_Group == ref_group) %&gt;%
    { ref_tbl &lt;- .[ , c(&quot;Category&quot;, &quot;Count&quot;)]; setNames(ref_tbl$Count, ref_tbl$Category) }

  df %&gt;%
    filter(Response_Group != ref_group) %&gt;%
    group_by(Response_Group) %&gt;%
    summarise(
      p_value = {
        group_counts &lt;- Count[Category %in% c(&quot;DOX-Toxicity Genes&quot;, &quot;Non-DOX Genes&quot;)]
        if (length(group_counts) &lt; 2) group_counts &lt;- c(group_counts, 0)
        contingency_table &lt;- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts[&quot;DOX-Toxicity Genes&quot;], ref_counts[&quot;Non-DOX Genes&quot;]
        ), nrow = 2)
        if (any(contingency_table &lt; 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = &quot;drop&quot;
    ) %&gt;%
    mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))
}

# Run test for each concentration
chi_0.1 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.1 µM&quot;), &quot;Non response&quot;)</code></pre>
<pre><code>Warning: There was 1 warning in `summarise()`.
ℹ In argument: `p_value = { ... }`.
ℹ In group 2: `Response_Group = &quot;DOX only mid-late response&quot;`.
Caused by warning in `chisq.test()`:
! Chi-squared approximation may be incorrect</code></pre>
<pre class="r"><code>chi_0.5 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.5 µM&quot;), &quot;Non response&quot;)

chi_all &lt;- bind_rows(
  chi_0.1 %&gt;% mutate(Concentration = &quot;0.1 µM&quot;),
  chi_0.5 %&gt;% mutate(Concentration = &quot;0.5 µM&quot;)
)

# Merge test results into proportions
proportion_data &lt;- left_join(proportion_data, chi_all, by = c(&quot;Concentration&quot;, &quot;Response_Group&quot;))

# ----------------- Reorder Factor Levels -----------------
proportion_data$Concentration &lt;- factor(proportion_data$Concentration, levels = c(&quot;0.1 µM&quot;, &quot;0.5 µM&quot;))
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
))

# ----------------- Star Label Position -----------------
label_data &lt;- proportion_data %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  summarise(Significance = dplyr::first(Significance), .groups = &quot;drop&quot;) %&gt;%
  filter(!is.na(Significance) &amp; Significance != &quot;&quot;) %&gt;%
  mutate(y_pos = 105)

# ----------------- Final Plot -----------------
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(
    data = label_data,
    aes(x = Response_Group, y = y_pos, label = Significance),
    inherit.aes = FALSE,
    size = 5,
    color = &quot;black&quot;
  ) +
  facet_wrap(~ Concentration, scales = &quot;free_x&quot;) +
  scale_fill_manual(values = c(
    &quot;DOX-Toxicity Genes&quot; = &quot;#E41A1C&quot;,
    &quot;Non-DOX Genes&quot; = &quot;#377EB8&quot;
  )) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +
  labs(
    title = &quot;DOX Cardiotoxicity Gene Proportions Across\nCormotif Clusters (0.1 and 0.5)&quot;,
    x = &quot;Response Groups&quot;,
    y = &quot;Percentage of Genes&quot;,
    fill = &quot;Gene Category&quot;
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = &quot;bold&quot;),
    axis.title.x = element_text(size = 14, face = &quot;bold&quot;),
    axis.title.y = element_text(size = 14, face = &quot;bold&quot;),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    panel.spacing = unit(1.2, &quot;lines&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/DOX_tox%20prop-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-DOX_tox-prop-1">
Past versions of "DOX_tox prop-1.png"
</button>
</p>
<div id="fig-no-spaces-DOX_tox-prop-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/DOX_tox prop-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-dox-cardiotoxicity-genes-across-corrmotifs-per-concentration"
class="section level2">
<h2><strong>📌 Mean logFC of DOX Cardiotoxicity genes across corrmotifs
(per concentration)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- DOX Cardiotoxicity Entrez IDs -----------------
entrez_ids &lt;- c(
  847, 873, 2064, 2878, 2944, 3038, 4846, 51196, 5880, 6687,
  7799, 4292, 5916, 3077, 51310, 9154, 64078, 5244, 10057, 10060,
  89845, 56853, 4625, 1573, 79890
) %&gt;% as.character()

# ----------------- Load CorMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate Response Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Aggregate DEG Data -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;,
      TRUE ~ NA_character_
    )
  )

deg_mean &lt;- deg_all %&gt;%
  group_by(Entrez_ID, Concentration) %&gt;%
  summarise(logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;)

# ----------------- Merge with CorMotif Genes -----------------
dox_cardio_logfc &lt;- inner_join(df_combined, deg_mean, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Group = paste(Response_Group, Concentration)
  )

# ----------------- Define Response Group Order -----------------
group_levels &lt;- c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
)

dox_cardio_logfc$Response_Group &lt;- factor(dox_cardio_logfc$Response_Group, levels = group_levels)

# ----------------- Wilcoxon Test -----------------
comparison_map &lt;- list(
  &quot;CX-DOX mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX only mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX-specific response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX only mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;CX total + DOX early response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX early + CX-DOX mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;
)

wilcoxon_results &lt;- lapply(names(comparison_map), function(resp_group) {
  control_group &lt;- comparison_map[[resp_group]]
  resp_vals &lt;- dox_cardio_logfc$logFC[dox_cardio_logfc$Group == resp_group]
  control_vals &lt;- dox_cardio_logfc$logFC[dox_cardio_logfc$Group == control_group]

  if (length(resp_vals) &gt;= 2 &amp;&amp; length(control_vals) &gt;= 2) {
    test_result &lt;- wilcox.test(resp_vals, control_vals)
    pval &lt;- test_result$p.value
    stat &lt;- test_result$statistic
    label &lt;- case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01  ~ &quot;**&quot;,
      pval &lt; 0.05  ~ &quot;*&quot;,
      TRUE ~ &quot;&quot;
    )
    y_pos &lt;- max(resp_vals, na.rm = TRUE) + 0.5
    return(data.frame(
      Group = resp_group,
      y_pos = y_pos,
      label = label,
      P_Value = signif(pval, 4)
    ))
  } else {
    return(NULL)
  }
}) %&gt;% bind_rows()

# ----------------- Prepare Annotation -----------------
label_data &lt;- wilcoxon_results %&gt;%
  separate(Group, into = c(&quot;Response_Group&quot;, &quot;Concentration&quot;), sep = &quot; (?=0\\.)&quot;) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = group_levels))

# ----------------- Plot -----------------
ggplot(dox_cardio_logfc, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~Concentration, scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Mean logFC of DOX Cardiotoxicity Genes\nAcross CorMotif Response Groups&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20DOX_tox%20logFC-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-DOX_tox-logFC-1">
Past versions of "Mean DOX_tox logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-DOX_tox-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean DOX_tox logFC-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-dox-cardiotoxicity-genes-across-corrmotifs-faceted-by-concentration-timepoint"
class="section level2">
<h2><strong>📌 Mean logFC of DOX Cardiotoxicity genes across corrmotifs
(Faceted by concentration × timepoint)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- DOX Cardiotoxicity Entrez IDs -----------------
entrez_ids &lt;- c(
  847, 873, 2064, 2878, 2944, 3038, 4846, 51196, 5880, 6687,
  7799, 4292, 5916, 3077, 51310, 9154, 64078, 5244, 10057, 10060,
  89845, 56853, 4625, 1573, 79890
) %&gt;% as.character()

# ----------------- Load CorMotif Response Group Data -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorMotif Group Membership -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Annotate DEG Files -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    ),
    Timepoint = factor(case_when(
      str_detect(Condition, &quot;_3$&quot;) ~ &quot;3 hr&quot;,
      str_detect(Condition, &quot;_24$&quot;) ~ &quot;24 hr&quot;,
      str_detect(Condition, &quot;_48$&quot;) ~ &quot;48 hr&quot;
    ), levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))
  )

# ----------------- Merge with DOX Cardiotoxicity Targets -----------------
dox_all &lt;- inner_join(df_combined, deg_all, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID, column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response&quot;, &quot;CX-DOX mid-late response&quot;, &quot;DOX-specific response&quot;,
      &quot;DOX only mid-late response&quot;, &quot;CX total + DOX early response&quot;, &quot;DOX early + CX-DOX mid-late response&quot;
    ))
  )

# ----------------- Wilcoxon Test -----------------
comparison_groups &lt;- levels(dox_all$Response_Group)[-1]
stat_results &lt;- list()

for (conc in unique(dox_all$Concentration)) {
  for (tp in levels(dox_all$Timepoint)) {
    sub &lt;- dox_all %&gt;% filter(Concentration == conc, Timepoint == tp)
    for (grp in comparison_groups) {
      if (grp %in% unique(sub$Response_Group)) {
        test &lt;- tryCatch({
          wilcox.test(
            sub$logFC[sub$Response_Group == grp],
            sub$logFC[sub$Response_Group == &quot;Non response&quot;]
          )
        }, error = function(e) NULL)
        if (!is.null(test) &amp;&amp; !is.nan(test$p.value)) {
          stat_results[[length(stat_results) + 1]] &lt;- data.frame(
            Response_Group = grp,
            Concentration = conc,
            Timepoint = tp,
            p_value = signif(test$p.value, 3),
            label = case_when(
              test$p.value &lt; 0.001 ~ &quot;***&quot;,
              test$p.value &lt; 0.01 ~ &quot;**&quot;,
              test$p.value &lt; 0.05 ~ &quot;*&quot;,
              TRUE ~ &quot;&quot;
            ),
            y_pos = max(sub$logFC[sub$Response_Group == grp], na.rm = TRUE) + 0.5
          )
        }
      }
    }
  }
}

label_data &lt;- bind_rows(stat_results) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = levels(dox_all$Response_Group)))

# ----------------- Plot -----------------
ggplot(dox_all, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7, na.rm = TRUE) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 4, vjust = 0) +
  facet_grid(rows = vars(factor(Timepoint, levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))),
             cols = vars(Concentration), scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;DOX Cardiotoxicity Gene logFC by\nCorMotif Group, Timepoint, and Concentration&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20DOX_tox_2%20logFC1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-DOX_tox_2-logFC1-1">
Past versions of "Mean DOX_tox_2 logFC1-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-DOX_tox_2-logFC1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean DOX_tox_2 logFC1-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-dox-cardiotoxicity-genes-across-corrmotifs-faceted-by-concentration-timepoint-drug-cx-vs-dox-side-by-side"
class="section level2">
<h2><strong>📌 Mean logFC of DOX Cardiotoxicity genes across corrmotifs
(Faceted by concentration × timepoint × drug (CX vs DOX
side-by-side))</strong></h2>
<pre class="r"><code>################### DOX Cardiotoxicity Genes — 0.1 µM ###################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- DOX Cardiotoxicity Entrez IDs -----------------
entrez_ids &lt;- c(
  847, 873, 2064, 2878, 2944, 3038, 4846, 51196, 5880, 6687,
  7799, 4292, 5916, 3077, 51310, 9154, 64078, 5244, 10057, 10060,
  89845, 56853, 4625, 1573, 79890
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.1 µM) -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.1 µM DEG Files -----------------
deg_0.1 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.1 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late\nresponse (0.1 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.1 &lt;- deg_0.1 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.1 µM)&quot;,
      &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.1 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.1 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.1 µM DOX Cardiotoxicity Genes) -----------------
ggplot(deg_0.1, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;DOX Cardiotoxicity Gene logFC by Drug and Response Group (0.1 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20DOX_tox_3%20logFC3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-DOX_tox_3-logFC3-1">
Past versions of "Mean DOX_tox_3 logFC3-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-DOX_tox_3-logFC3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean DOX_tox_3 logFC3-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>################### DOX Cardiotoxicity Genes — 0.5 µM ###################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- DOX Cardiotoxicity Entrez IDs -----------------
entrez_ids &lt;- c(
  847, 873, 2064, 2878, 2944, 3038, 4846, 51196, 5880, 6687,
  7799, 4292, 5916, 3077, 51310, 9154, 64078, 5244, 10057, 10060,
  89845, 56853, 4625, 1573, 79890
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.5 µM) -----------------
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.5 µM DEG Files -----------------
deg_0.5 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.5 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 µM)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.5 &lt;- deg_0.5 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.5 µM)&quot;,
      &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.5 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.5 µM DOX Cardiotoxicity Genes) -----------------
ggplot(deg_0.5, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;DOX Cardiotoxicity Gene logFC by Drug and Response Group (0.5 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20DOX_tox_3%20logFC3-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-DOX_tox_3-logFC3-2">
Past versions of "Mean DOX_tox_3 logFC3-2.png"
</button>
</p>
<div id="fig-no-spaces-Mean-DOX_tox_3-logFC3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean DOX_tox_3 logFC3-2.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="proportion-of-heart-specific-genes-combined"
class="section level2">
<h2><strong>📌 Proportion of Heart Specific Genes
(Combined)</strong></h2>
<pre class="r"><code># ----------------- Load Required Libraries -----------------
library(dplyr)
library(ggplot2)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load Heart-Specific Genes -----------------
heart_genes &lt;- read.csv(&quot;data/Human_Heart_Genes.csv&quot;, stringsAsFactors = FALSE)
heart_genes$Entrez_ID &lt;- mapIds(
  org.Hs.eg.db,
  keys = heart_genes$Gene,
  column = &quot;ENTREZID&quot;,
  keytype = &quot;SYMBOL&quot;,
  multiVals = &quot;first&quot;
)
heart_entrez_ids &lt;- na.omit(heart_genes$Entrez_ID)

# ----------------- Load CorrMotif Groups -----------------
# 0.1
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% heart_entrez_ids, &quot;Heart-specific Genes&quot;, &quot;Non-Heart-specific Genes&quot;),
    Concentration = &quot;0.1&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% heart_entrez_ids, &quot;Heart-specific Genes&quot;, &quot;Non-Heart-specific Genes&quot;),
    Concentration = &quot;0.5&quot;
  )

# ----------------- Combine Data -----------------
df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Calculate Proportions -----------------
proportion_data &lt;- df_combined %&gt;%
  dplyr::group_by(Concentration, Response_Group, Category) %&gt;%
  dplyr::summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  dplyr::group_by(Concentration, Response_Group) %&gt;%
  dplyr::mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Chi-square or Fisher&#39;s Test -----------------
get_chi_or_fisher &lt;- function(df, ref_group) {
  ref_counts &lt;- df %&gt;%
    dplyr::ungroup() %&gt;%
    dplyr::filter(Response_Group == ref_group) %&gt;%
    dplyr::select(Category, Count) %&gt;%
    {setNames(.$Count, .$Category)}
  
  df %&gt;%
    dplyr::filter(Response_Group != ref_group) %&gt;%
    dplyr::group_by(Response_Group) %&gt;%
    dplyr::summarise(
      p_value = {
        group_counts &lt;- Count[Category %in% c(&quot;Heart-specific Genes&quot;, &quot;Non-Heart-specific Genes&quot;)]
        if (length(group_counts) &lt; 2) group_counts &lt;- c(group_counts, 0)
        contingency_table &lt;- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts[&quot;Heart-specific Genes&quot;], ref_counts[&quot;Non-Heart-specific Genes&quot;]
        ), nrow = 2)
        if (any(contingency_table &lt; 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = &quot;drop&quot;
    ) %&gt;%
    dplyr::mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))
}

# Run tests
chi_0.1 &lt;- get_chi_or_fisher(proportion_data %&gt;% dplyr::filter(Concentration == &quot;0.1&quot;), &quot;Non response&quot;)
chi_0.5 &lt;- get_chi_or_fisher(proportion_data %&gt;% dplyr::filter(Concentration == &quot;0.5&quot;), &quot;Non response&quot;)

chi_all &lt;- bind_rows(
  chi_0.1 %&gt;% mutate(Concentration = &quot;0.1&quot;),
  chi_0.5 %&gt;% mutate(Concentration = &quot;0.5&quot;)
)

# Merge test results
proportion_data &lt;- proportion_data %&gt;%
  left_join(chi_all, by = c(&quot;Concentration&quot;, &quot;Response_Group&quot;))

# ----------------- Reorder Factor Levels -----------------
proportion_data$Concentration &lt;- factor(proportion_data$Concentration, levels = c(&quot;0.1&quot;, &quot;0.5&quot;))
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
))

# ----------------- Position Significance Stars -----------------
label_data &lt;- proportion_data %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  summarise(Significance = dplyr::first(Significance), .groups = &quot;drop&quot;) %&gt;%
  filter(!is.na(Significance)) %&gt;%
  mutate(y_pos = 105)


# ----------------- Final Plot -----------------
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(
    data = label_data,
    aes(x = Response_Group, y = y_pos, label = Significance),
    inherit.aes = FALSE,
    size = 5,
    color = &quot;black&quot;
  ) +
  facet_wrap(~ Concentration, scales = &quot;free_x&quot;) +
  scale_fill_manual(values = c(
    &quot;Heart-specific Genes&quot; = &quot;#4daf4a&quot;,
    &quot;Non-Heart-specific Genes&quot; = &quot;#377eb8&quot;
  )) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +
  labs(
    title = &quot;Heart-Specific Gene Proportions Across\nCormotif Clusters (0.1 and 0.5)&quot;,
    x = &quot;Response Groups&quot;,
    y = &quot;Percentage of Genes&quot;,
    fill = &quot;Gene Category&quot;
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = &quot;bold&quot;),
    axis.title.x = element_text(size = 14, face = &quot;bold&quot;),
    axis.title.y = element_text(size = 14, face = &quot;bold&quot;),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    panel.spacing = unit(1.2, &quot;lines&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/prop%20heart%20comb-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-prop-heart-comb-1">
Past versions of "prop heart comb-1.png"
</button>
</p>
<div id="fig-no-spaces-prop-heart-comb-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/4bad905ad26e9ebac9188a323b7f5fe10aa56ce6/docs/figure/Corrmotif_Conc.Rmd/prop heart comb-1.png" target="_blank">4bad905</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/6ef5f61785e28f0c6385c185a7d54db2620ea914/docs/figure/Corrmotif_Conc.Rmd/prop heart comb-1.png" target="_blank">6ef5f61</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-heart-specific-genes-across-corrmotifs-per-concentration"
class="section level2">
<h2><strong>📌 Mean logFC of Heart specific genes across corrmotifs (per
concentration)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load Heart-Specific Genes -----------------
heart_genes &lt;- read.csv(&quot;data/Human_Heart_Genes.csv&quot;, stringsAsFactors = FALSE)
heart_genes$Entrez_ID &lt;- mapIds(
  org.Hs.eg.db,
  keys = heart_genes$Gene,
  column = &quot;ENTREZID&quot;,
  keytype = &quot;SYMBOL&quot;,
  multiVals = &quot;first&quot;
)
entrez_ids &lt;- na.omit(heart_genes$Entrez_ID) %&gt;% as.character()

# ----------------- Load CorMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate Response Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Aggregate DEG Data -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;% mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    )
  )

deg_mean &lt;- deg_all %&gt;%
  group_by(Entrez_ID, Concentration) %&gt;%
  summarise(logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;)

# ----------------- Merge with Group Annotations -----------------
heart_logfc &lt;- inner_join(df_combined, deg_mean, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Group = paste(Response_Group, Concentration)
  )

# ----------------- Define Response Group Order -----------------
group_levels &lt;- c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
)

heart_logfc$Response_Group &lt;- factor(heart_logfc$Response_Group, levels = group_levels)

# ----------------- Wilcoxon Test -----------------
comparison_map &lt;- list(
  &quot;CX-DOX mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX only mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX-specific response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX only mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;CX total + DOX early response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX early + CX-DOX mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;
)

wilcoxon_results &lt;- lapply(names(comparison_map), function(resp_group) {
  control_group &lt;- comparison_map[[resp_group]]
  resp_vals &lt;- heart_logfc$logFC[heart_logfc$Group == resp_group]
  control_vals &lt;- heart_logfc$logFC[heart_logfc$Group == control_group]

  if (length(resp_vals) &gt;= 2 &amp;&amp; length(control_vals) &gt;= 2) {
    test_result &lt;- wilcox.test(resp_vals, control_vals)
    pval &lt;- test_result$p.value
    label &lt;- case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01  ~ &quot;**&quot;,
      pval &lt; 0.05  ~ &quot;*&quot;,
      TRUE ~ &quot;&quot;
    )
    y_pos &lt;- max(resp_vals, na.rm = TRUE) + 0.5
    return(data.frame(
      Group = resp_group,
      y_pos = y_pos,
      label = label,
      P_Value = signif(pval, 4)
    ))
  }
  return(NULL)
}) %&gt;% bind_rows()

# ----------------- Annotation Data -----------------
label_data &lt;- wilcoxon_results %&gt;%
  separate(Group, into = c(&quot;Response_Group&quot;, &quot;Concentration&quot;), sep = &quot; (?=0\\.)&quot;) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = group_levels))

# ----------------- Plot -----------------
ggplot(heart_logfc, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~Concentration, scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Mean logFC of Heart-Specific Genes\nAcross CorMotif Response Groups&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20HS%20logFC-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-HS-logFC-1">
Past versions of "Mean HS logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-HS-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean HS logFC-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-heart-specific-genes-across-corrmotifs-faceted-by-concentration-timepoint"
class="section level2">
<h2><strong>📌 Mean logFC of Heart specific genes across corrmotifs
(Faceted by concentration × timepoint)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load Heart-Specific Genes -----------------
heart_genes &lt;- read.csv(&quot;data/Human_Heart_Genes.csv&quot;, stringsAsFactors = FALSE)
heart_genes$Entrez_ID &lt;- mapIds(
  org.Hs.eg.db,
  keys = heart_genes$Gene,
  column = &quot;ENTREZID&quot;,
  keytype = &quot;SYMBOL&quot;,
  multiVals = &quot;first&quot;
)
entrez_ids &lt;- na.omit(heart_genes$Entrez_ID) %&gt;% as.character()

# ----------------- Load CorMotif Response Group Data -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorMotif Group Membership -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Annotate DEG Files -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    ),
    Timepoint = factor(case_when(
      str_detect(Condition, &quot;_3$&quot;) ~ &quot;3 hr&quot;,
      str_detect(Condition, &quot;_24$&quot;) ~ &quot;24 hr&quot;,
      str_detect(Condition, &quot;_48$&quot;) ~ &quot;48 hr&quot;
    ), levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))
  )

# ----------------- Merge with Heart-Specific Targets -----------------
heart_all &lt;- inner_join(df_combined, deg_all, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID, column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response&quot;, &quot;CX-DOX mid-late response&quot;, &quot;DOX-specific response&quot;,
      &quot;DOX only mid-late response&quot;, &quot;CX total + DOX early response&quot;, &quot;DOX early + CX-DOX mid-late response&quot;
    ))
  )

# ----------------- Wilcoxon Test -----------------
comparison_groups &lt;- levels(heart_all$Response_Group)[-1]
stat_results &lt;- list()

for (conc in unique(heart_all$Concentration)) {
  for (tp in levels(heart_all$Timepoint)) {
    sub &lt;- heart_all %&gt;% filter(Concentration == conc, Timepoint == tp)
    for (grp in comparison_groups) {
      if (grp %in% unique(sub$Response_Group)) {
        test &lt;- tryCatch({
          wilcox.test(
            sub$logFC[sub$Response_Group == grp],
            sub$logFC[sub$Response_Group == &quot;Non response&quot;]
          )
        }, error = function(e) NULL)
        if (!is.null(test) &amp;&amp; !is.nan(test$p.value)) {
          stat_results[[length(stat_results) + 1]] &lt;- data.frame(
            Response_Group = grp,
            Concentration = conc,
            Timepoint = tp,
            p_value = signif(test$p.value, 3),
            label = case_when(
              test$p.value &lt; 0.001 ~ &quot;***&quot;,
              test$p.value &lt; 0.01 ~ &quot;**&quot;,
              test$p.value &lt; 0.05 ~ &quot;*&quot;,
              TRUE ~ &quot;&quot;
            ),
            y_pos = max(sub$logFC[sub$Response_Group == grp], na.rm = TRUE) + 0.5
          )
        }
      }
    }
  }
}

label_data &lt;- bind_rows(stat_results) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = levels(heart_all$Response_Group)))

# ----------------- Plot -----------------
ggplot(heart_all, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7, na.rm = TRUE) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 4, vjust = 0) +
  facet_grid(rows = vars(factor(Timepoint, levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))),
             cols = vars(Concentration), scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Heart-Specific Gene logFC by\nCorMotif Group, Timepoint, and Concentration&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20HS_2%20logFC1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-HS_2-logFC1-1">
Past versions of "Mean HS_2 logFC1-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-HS_2-logFC1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean HS_2 logFC1-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-heart-specific-genes-across-corrmotifs-faceted-by-concentration-timepoint-drug-cx-vs-dox-side-by-side"
class="section level2">
<h2><strong>📌 Mean logFC of Heart specific genes across corrmotifs
(Faceted by concentration × timepoint × drug (CX vs DOX
side-by-side))</strong></h2>
<pre class="r"><code>############ HEART-SPECIFIC GENES — 0.1 µM #######################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load Heart-Specific Genes -----------------
heart_genes &lt;- read.csv(&quot;data/Human_Heart_Genes.csv&quot;, stringsAsFactors = FALSE)
heart_genes$Entrez_ID &lt;- mapIds(
  org.Hs.eg.db,
  keys = heart_genes$Gene,
  column = &quot;ENTREZID&quot;,
  keytype = &quot;SYMBOL&quot;,
  multiVals = &quot;first&quot;
)
entrez_ids &lt;- na.omit(heart_genes$Entrez_ID) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.1 µM) -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.1 µM DEG Files -----------------
deg_0.1 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.1 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late\nresponse (0.1 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.1 &lt;- deg_0.1 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.1 µM)&quot;,
      &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.1 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.1 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.1 µM Heart-Specific Genes) -----------------
ggplot(deg_0.1, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;Heart-Specific Gene logFC by Drug and Response Group (0.1 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20HS_3%20logFC3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-HS_3-logFC3-1">
Past versions of "Mean HS_3 logFC3-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-HS_3-logFC3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean HS_3 logFC3-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>############ HEART-SPECIFIC GENES — 0.5 µM #######################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load Heart-Specific Genes -----------------
heart_genes &lt;- read.csv(&quot;data/Human_Heart_Genes.csv&quot;, stringsAsFactors = FALSE)
heart_genes$Entrez_ID &lt;- mapIds(
  org.Hs.eg.db,
  keys = heart_genes$Gene,
  column = &quot;ENTREZID&quot;,
  keytype = &quot;SYMBOL&quot;,
  multiVals = &quot;first&quot;
)
entrez_ids &lt;- na.omit(heart_genes$Entrez_ID) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.5 µM) -----------------
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.5 µM DEG Files -----------------
deg_0.5 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.5 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 µM)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.5 &lt;- deg_0.5 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.5 µM)&quot;,
      &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.5 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.5 µM Heart-Specific Genes) -----------------
ggplot(deg_0.5, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;Heart-Specific Gene logFC by Drug and Response Group (0.5 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20HS_3%20logFC3-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-HS_3-logFC3-2">
Past versions of "Mean HS_3 logFC3-2.png"
</button>
</p>
<div id="fig-no-spaces-Mean-HS_3-logFC3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean HS_3 logFC3-2.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="proportion-of-atrial-fibrillation-gwas-genes-across-corrmotifs"
class="section level2">
<h2><strong>📌 Proportion of Atrial Fibrillation GWAS genes across
corrmotifs</strong></h2>
<pre class="r"><code># ----------------- Load Required Libraries -----------------
library(dplyr)
library(ggplot2)

# ----------------- AF Entrez IDs -----------------
af_entrez_ids &lt;- c(
  3425, 53834, 9464, 79658, 55876, 3781, 9472, 11017, 64395, 23353,
  57794, 79591, 2702, 51306, 857, 10418, 9644, 115509, 89797, 10728,
  282996, 8497, 5695, 9026, 57666, 143384, 5883, 90102, 84909, 221037,
  29119, 79933, 391, 81575, 8738, 5396, 26010, 744, 56981, 80724,
  5781, 6910, 4598, 79068, 2250, 56978, 3752, 5208, 9736, 26778,
  80315, 64374, 79159, 11078, 5308, 3782, 10654, 79776, 6310, 8082,
  2995, 23224, 79741, 5708, 26136, 9967, 84033, 9148, 29982, 81488,
  1788, 9208, 2908, 1021, 5339, 5727, 23118, 10090, 57585, 221154,
  1073, 29998, 817, 858, 3486, 5747, 2104, 55105, 4194, 3570,
  1607, 6586, 5570, 2070, 27145, 4646, 3159, 9748, 79991, 221035,
  5033, 80212, 196385, 79600, 11046, 54014, 151636, 64778, 92597, 3983,
  7090, 9267, 7273, 26018, 9948, 2131, 55870, 5523, 5318, 6239,
  3480, 55777, 2066, 11165, 2028, 57619, 2690, 23414, 55818, 84700,
  28965, 80204, 463, 5108, 222553, 387119, 28981, 11113, 10301, 995,
  23030, 2697, 996, 339500, 54805, 9960, 91404, 145781, 100820829, 84542,
  2176, 51684, 9513, 7473, 4666, 23150, 5915, 5062, 4016, 23039,
  159686, 1839, 5201, 93166, 64753, 29959, 5496, 23245, 5069, 56916,
  92344, 23092, 3992, 9415, 10554, 7456, 9570, 57178, 23143, 161176,
  5424, 2034, 10277, 11278, 79803, 6653, 4756, 132660, 5430, 9031,
  57158, 285761, 8110, 387700, 1829, 4126, 7323, 51308, 7332, 6598,
  3757, 6187, 6660, 10529, 6920, 115286, 8451, 8943, 4137, 7514,
  4801, 9709, 23177, 8671, 29915, 26207, 3680, 490, 493856, 58489,
  54897, 4625, 22955, 84952, 10221, 2263, 84641, 4892, 1026, 84650,
  8382, 221656, 2969, 144453, 117177, 2626, 8476, 161882, 51807, 4624,
  9612, 55795, 51043, 144348, 51232, 8729, 3899, 11155, 23316, 79006,
  146330, 6403, 6331, 4300, 427, 845, 3313, 113622, 5789, 376132,
  29841, 8462, 203859, 401397, 5506, 55521, 5819, 4059, 6934, 57727,
  23066, 79568, 83478, 10087, 9586, 222194, 10466, 10499, 58499, 79720,
  4772, 10794, 125919, 602, 27332, 59345, 340359, 3705, 64710, 57801,
  10818, 143684, 149281, 55013, 23095, 93649, 84034, 23347, 440926, 11124,
  23293, 51426, 832, 7068, 57646, 152002, 7531, 1398, 100101267, 221937,
  26873, 3709, 10576, 27040, 201176, 284403, 307, 6525, 5387, 1808,
  114907, 1952, 4091, 5096, 23451, 51207, 142891, 4084, 3797, 1185,
  5334, 217, 2042, 7781, 253461, 152404, 2992, 153478, 6695, 26249,
  662, 23036, 22852, 23411, 23387, 6786, 7690, 10021, 8925, 5595,
  63892, 23466, 11149, 139411, 755, 55347, 22820, 55111, 84444, 6258,
  57623, 1387, 3762, 5861, 3339, 114822, 129787, 196528, 54437, 10395,
  4023, 5095, 894, 3156, 1877, 283871, 2673, 23157, 10512, 2258,
  79750, 84665, 26091, 5978, 8751, 79695, 2768, 155382, 84163, 7709,
  4092, 1837, 2064, 4629, 55122, 8882, 150962, 23013, 8742, 489,
  201134, 55114, 84264, 64428, 4089, 861, 7422
) %&gt;% as.character()

# ----------------- Load CorrMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% af_entrez_ids, &quot;AF Genes&quot;, &quot;Non-AF Genes&quot;),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% af_entrez_ids, &quot;AF Genes&quot;, &quot;Non-AF Genes&quot;),
    Concentration = &quot;0.5 µM&quot;
  )

# ----------------- Combine Data -----------------
df_combined &lt;- bind_rows(df_0.1, df_0.5)
df_combined$Category &lt;- factor(df_combined$Category, levels = c(&quot;AF Genes&quot;, &quot;Non-AF Genes&quot;))

# ----------------- Calculate Proportions -----------------
proportion_data &lt;- df_combined %&gt;%
  group_by(Concentration, Response_Group, Category) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Chi-square or Fisher&#39;s Test -----------------
get_chi_or_fisher &lt;- function(df, ref_group) {
  ref_counts &lt;- df %&gt;%
    ungroup() %&gt;%
    filter(Response_Group == ref_group) %&gt;%
    { ref_tbl &lt;- .[ , c(&quot;Category&quot;, &quot;Count&quot;)]; setNames(ref_tbl$Count, ref_tbl$Category) }
  
  df %&gt;%
    filter(Response_Group != ref_group) %&gt;%
    group_by(Response_Group) %&gt;%
    summarise(
      p_value = {
        group_counts &lt;- Count[Category %in% c(&quot;AF Genes&quot;, &quot;Non-AF Genes&quot;)]
        if (length(group_counts) &lt; 2) group_counts &lt;- c(group_counts, 0)
        contingency_table &lt;- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts[&quot;AF Genes&quot;], ref_counts[&quot;Non-AF Genes&quot;]
        ), nrow = 2)
        if (any(contingency_table &lt; 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = &quot;drop&quot;
    ) %&gt;%
    mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))
}

# ----------------- Run test -----------------
chi_0.1 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.1 µM&quot;), &quot;Non response&quot;)
chi_0.5 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.5 µM&quot;), &quot;Non response&quot;)</code></pre>
<pre><code>Warning: There was 1 warning in `summarise()`.
ℹ In argument: `p_value = { ... }`.
ℹ In group 4: `Response_Group = &quot;DOX-specific response&quot;`.
Caused by warning in `chisq.test()`:
! Chi-squared approximation may be incorrect</code></pre>
<pre class="r"><code>chi_all &lt;- bind_rows(
  chi_0.1 %&gt;% mutate(Concentration = &quot;0.1 µM&quot;),
  chi_0.5 %&gt;% mutate(Concentration = &quot;0.5 µM&quot;)
)
proportion_data &lt;- left_join(proportion_data, chi_all, by = c(&quot;Concentration&quot;, &quot;Response_Group&quot;))

# ----------------- Factor Order and Star Label -----------------
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
))
proportion_data$Concentration &lt;- factor(proportion_data$Concentration, levels = c(&quot;0.1 µM&quot;, &quot;0.5 µM&quot;))
label_data &lt;- proportion_data %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  summarise(Significance = dplyr::first(Significance), .groups = &quot;drop&quot;) %&gt;%
  filter(!is.na(Significance) &amp; Significance != &quot;&quot;) %&gt;%
  mutate(y_pos = 105)

# ----------------- Final Plot -----------------
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = Significance),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~ Concentration, scales = &quot;free_x&quot;) +
  scale_fill_manual(values = c(&quot;AF Genes&quot; = &quot;#E41A1C&quot;, &quot;Non-AF Genes&quot; = &quot;#377EB8&quot;)) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +
  labs(
    title = &quot;AF-Associated Gene Proportions Across\nCormotif Clusters (0.1 and 0.5 µM)&quot;,
    x = &quot;Response Groups&quot;, y = &quot;Percentage of Genes&quot;, fill = &quot;Gene Category&quot;
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = &quot;bold&quot;),
    axis.title.x = element_text(size = 14, face = &quot;bold&quot;),
    axis.title.y = element_text(size = 14, face = &quot;bold&quot;),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    panel.spacing = unit(1.2, &quot;lines&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/AF_Prop-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-AF_Prop-1">
Past versions of AF_Prop-1.png
</button>
</p>
<div id="fig-AF_Prop-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/AF_Prop-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-atrial-fibrillation-gwas-genes-across-corrmotifs-per-concentration"
class="section level2">
<h2><strong>📌 Mean logFC of Atrial Fibrillation GWAS genes across
corrmotifs (per concentration)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- AF Entrez IDs -----------------
entrez_ids &lt;- c(
  3425, 53834, 9464, 79658, 55876, 3781, 9472, 11017, 64395, 23353,
  57794, 79591, 2702, 51306, 857, 10418, 9644, 115509, 89797, 10728,
  282996, 8497, 5695, 9026, 57666, 143384, 5883, 90102, 84909, 221037,
  29119, 79933, 391, 81575, 8738, 5396, 26010, 744, 56981, 80724,
  5781, 6910, 4598, 79068, 2250, 56978, 3752, 5208, 9736, 26778,
  80315, 64374, 79159, 11078, 5308, 3782, 10654, 79776, 6310, 8082,
  2995, 23224, 79741, 5708, 26136, 9967, 84033, 9148, 29982, 81488,
  1788, 9208, 2908, 1021, 5339, 5727, 23118, 10090, 57585, 221154,
  1073, 29998, 817, 858, 3486, 5747, 2104, 55105, 4194, 3570,
  1607, 6586, 5570, 2070, 27145, 4646, 3159, 9748, 79991, 221035,
  5033, 80212, 196385, 79600, 11046, 54014, 151636, 64778, 92597, 3983,
  7090, 9267, 7273, 26018, 9948, 2131, 55870, 5523, 5318, 6239,
  3480, 55777, 2066, 11165, 2028, 57619, 2690, 23414, 55818, 84700,
  28965, 80204, 463, 5108, 222553, 387119, 28981, 11113, 10301, 995,
  23030, 2697, 996, 339500, 54805, 9960, 91404, 145781, 100820829, 84542,
  2176, 51684, 9513, 7473, 4666, 23150, 5915, 5062, 4016, 23039,
  159686, 1839, 5201, 93166, 64753, 29959, 5496, 23245, 5069, 56916,
  92344, 23092, 3992, 9415, 10554, 7456, 9570, 57178, 23143, 161176,
  5424, 2034, 10277, 11278, 79803, 6653, 4756, 132660, 5430, 9031,
  57158, 285761, 8110, 387700, 1829, 4126, 7323, 51308, 7332, 6598,
  3757, 6187, 6660, 10529, 6920, 115286, 8451, 8943, 4137, 7514,
  4801, 9709, 23177, 8671, 29915, 26207, 3680, 490, 493856, 58489,
  54897, 4625, 22955, 84952, 10221, 2263, 84641, 4892, 1026, 84650,
  8382, 221656, 2969, 144453, 117177, 2626, 8476, 161882, 51807, 4624,
  9612, 55795, 51043, 144348, 51232, 8729, 3899, 11155, 23316, 79006,
  146330, 6403, 6331, 4300, 427, 845, 3313, 113622, 5789, 376132,
  29841, 8462, 203859, 401397, 5506, 55521, 5819, 4059, 6934, 57727,
  23066, 79568, 83478, 10087, 9586, 222194, 10466, 10499, 58499, 79720,
  4772, 10794, 125919, 602, 27332, 59345, 340359, 3705, 64710, 57801,
  10818, 143684, 149281, 55013, 23095, 93649, 84034, 23347, 440926, 11124,
  23293, 51426, 832, 7068, 57646, 152002, 7531, 1398, 100101267, 221937,
  26873, 3709, 10576, 27040, 201176, 284403, 307, 6525, 5387, 1808,
  114907, 1952, 4091, 5096, 23451, 51207, 142891, 4084, 3797, 1185,
  5334, 217, 2042, 7781, 253461, 152404, 2992, 153478, 6695, 26249,
  662, 23036, 22852, 23411, 23387, 6786, 7690, 10021, 8925, 5595,
  63892, 23466, 11149, 139411, 755, 55347, 22820, 55111, 84444, 6258,
  57623, 1387, 3762, 5861, 3339, 114822, 129787, 196528, 54437, 10395,
  4023, 5095, 894, 3156, 1877, 283871, 2673, 23157, 10512, 2258,
  79750, 84665, 26091, 5978, 8751, 79695, 2768, 155382, 84163, 7709,
  4092, 1837, 2064, 4629, 55122, 8882, 150962, 23013, 8742, 489,
  201134, 55114, 84264, 64428, 4089, 861, 7422
) %&gt;% as.character()

# ----------------- Load CorMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate Response Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Aggregate DEG Data -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;% mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    )
  )

deg_mean &lt;- deg_all %&gt;%
  group_by(Entrez_ID, Concentration) %&gt;%
  summarise(logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;)

# ----------------- Merge and Annotate -----------------
af_logfc &lt;- inner_join(df_combined, deg_mean, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Group = paste(Response_Group, Concentration)
  )

# ----------------- Response Group Order -----------------
group_levels &lt;- c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
)
af_logfc$Response_Group &lt;- factor(af_logfc$Response_Group, levels = group_levels)

# ----------------- Wilcoxon Test -----------------
comparison_map &lt;- list(
  &quot;CX-DOX mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX only mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX-specific response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX only mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;CX total + DOX early response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX early + CX-DOX mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;
)

wilcoxon_results &lt;- lapply(names(comparison_map), function(resp_group) {
  control_group &lt;- comparison_map[[resp_group]]
  resp_vals &lt;- af_logfc$logFC[af_logfc$Group == resp_group]
  control_vals &lt;- af_logfc$logFC[af_logfc$Group == control_group]

  if (length(resp_vals) &gt;= 2 &amp;&amp; length(control_vals) &gt;= 2) {
    test_result &lt;- wilcox.test(resp_vals, control_vals)
    pval &lt;- test_result$p.value
    label &lt;- case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01  ~ &quot;**&quot;,
      pval &lt; 0.05  ~ &quot;*&quot;,
      TRUE ~ &quot;&quot;
    )
    y_pos &lt;- max(resp_vals, na.rm = TRUE) + 0.5
    return(data.frame(Group = resp_group, y_pos = y_pos, label = label, P_Value = signif(pval, 4)))
  }
  return(NULL)
}) %&gt;% bind_rows()

label_data &lt;- wilcoxon_results %&gt;%
  separate(Group, into = c(&quot;Response_Group&quot;, &quot;Concentration&quot;), sep = &quot; (?=0\\.)&quot;) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = group_levels))

# ----------------- Plot -----------------
ggplot(af_logfc, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~Concentration, scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Mean logFC of Atrial Fibrillation Genes\nAcross CorMotif Response Groups&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20AF%20logFC-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-AF-logFC-1">
Past versions of "Mean AF logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-AF-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean AF logFC-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-atrial-fibrillation-gwas-genes-across-corrmotifs-faceted-by-concentration-timepoint"
class="section level2">
<h2><strong>📌 Mean logFC of Atrial Fibrillation GWAS genes across
corrmotifs (Faceted by concentration × timepoint)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load Heart-Specific Genes -----------------
heart_genes &lt;- read.csv(&quot;data/Human_Heart_Genes.csv&quot;, stringsAsFactors = FALSE)
heart_genes$Entrez_ID &lt;- mapIds(
  org.Hs.eg.db,
  keys = heart_genes$Gene,
  column = &quot;ENTREZID&quot;,
  keytype = &quot;SYMBOL&quot;,
  multiVals = &quot;first&quot;
)
entrez_ids &lt;- na.omit(heart_genes$Entrez_ID) %&gt;% as.character()

# ----------------- Load CorMotif Response Group Data -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorMotif Group Membership -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Annotate DEG Files -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    ),
    Timepoint = factor(case_when(
      str_detect(Condition, &quot;_3$&quot;) ~ &quot;3 hr&quot;,
      str_detect(Condition, &quot;_24$&quot;) ~ &quot;24 hr&quot;,
      str_detect(Condition, &quot;_48$&quot;) ~ &quot;48 hr&quot;
    ), levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))
  )

# ----------------- Merge with Heart-Specific Targets -----------------
heart_all &lt;- inner_join(df_combined, deg_all, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID, column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response&quot;, &quot;CX-DOX mid-late response&quot;, &quot;DOX-specific response&quot;,
      &quot;DOX only mid-late response&quot;, &quot;CX total + DOX early response&quot;, &quot;DOX early + CX-DOX mid-late response&quot;
    ))
  )

# ----------------- Wilcoxon Test -----------------
comparison_groups &lt;- levels(heart_all$Response_Group)[-1]
stat_results &lt;- list()

for (conc in unique(heart_all$Concentration)) {
  for (tp in levels(heart_all$Timepoint)) {
    sub &lt;- heart_all %&gt;% filter(Concentration == conc, Timepoint == tp)
    for (grp in comparison_groups) {
      if (grp %in% unique(sub$Response_Group)) {
        test &lt;- tryCatch({
          wilcox.test(
            sub$logFC[sub$Response_Group == grp],
            sub$logFC[sub$Response_Group == &quot;Non response&quot;]
          )
        }, error = function(e) NULL)
        if (!is.null(test) &amp;&amp; !is.nan(test$p.value)) {
          stat_results[[length(stat_results) + 1]] &lt;- data.frame(
            Response_Group = grp,
            Concentration = conc,
            Timepoint = tp,
            p_value = signif(test$p.value, 3),
            label = case_when(
              test$p.value &lt; 0.001 ~ &quot;***&quot;,
              test$p.value &lt; 0.01 ~ &quot;**&quot;,
              test$p.value &lt; 0.05 ~ &quot;*&quot;,
              TRUE ~ &quot;&quot;
            ),
            y_pos = max(sub$logFC[sub$Response_Group == grp], na.rm = TRUE) + 0.5
          )
        }
      }
    }
  }
}

label_data &lt;- bind_rows(stat_results) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = levels(heart_all$Response_Group)))

# ----------------- Plot -----------------
ggplot(heart_all, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7, na.rm = TRUE) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 4, vjust = 0) +
  facet_grid(rows = vars(factor(Timepoint, levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))),
             cols = vars(Concentration), scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Heart-Specific Gene logFC by\nCorMotif Group, Timepoint, and Concentration&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20AF%20logFC_2%20logFC1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-AF-logFC_2-logFC1-1">
Past versions of "Mean AF logFC_2 logFC1-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-AF-logFC_2-logFC1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean AF logFC_2 logFC1-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-atrial-fibrillation-gwas-genes-across-corrmotifs-faceted-by-concentration-timepoint-drug-cx-vs-dox-side-by-side"
class="section level2">
<h2><strong>📌 Mean logFC of Atrial Fibrillation GWAS genes across
corrmotifs (Faceted by concentration × timepoint × drug (CX vs DOX
side-by-side))</strong></h2>
<pre class="r"><code>############ AF GENES — 0.1 µM #######################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- AF Entrez IDs -----------------
entrez_ids &lt;- c(
  3425, 53834, 9464, 79658, 55876, 3781, 9472, 11017, 64395, 23353,
  57794, 79591, 2702, 51306, 857, 10418, 9644, 115509, 89797, 10728,
  282996, 8497, 5695, 9026, 57666, 143384, 5883, 90102, 84909, 221037,
  29119, 79933, 391, 81575, 8738, 5396, 26010, 744, 56981, 80724,
  5781, 6910, 4598, 79068, 2250, 56978, 3752, 5208, 9736, 26778,
  80315, 64374, 79159, 11078, 5308, 3782, 10654, 79776, 6310, 8082,
  2995, 23224, 79741, 5708, 26136, 9967, 84033, 9148, 29982, 81488,
  1788, 9208, 2908, 1021, 5339, 5727, 23118, 10090, 57585, 221154,
  1073, 29998, 817, 858, 3486, 5747, 2104, 55105, 4194, 3570,
  1607, 6586, 5570, 2070, 27145, 4646, 3159, 9748, 79991, 221035,
  5033, 80212, 196385, 79600, 11046, 54014, 151636, 64778, 92597, 3983,
  7090, 9267, 7273, 26018, 9948, 2131, 55870, 5523, 5318, 6239,
  3480, 55777, 2066, 11165, 2028, 57619, 2690, 23414, 55818, 84700,
  28965, 80204, 463, 5108, 222553, 387119, 28981, 11113, 10301, 995,
  23030, 2697, 996, 339500, 54805, 9960, 91404, 145781, 100820829, 84542,
  2176, 51684, 9513, 7473, 4666, 23150, 5915, 5062, 4016, 23039,
  159686, 1839, 5201, 93166, 64753, 29959, 5496, 23245, 5069, 56916,
  92344, 23092, 3992, 9415, 10554, 7456, 9570, 57178, 23143, 161176,
  5424, 2034, 10277, 11278, 79803, 6653, 4756, 132660, 5430, 9031,
  57158, 285761, 8110, 387700, 1829, 4126, 7323, 51308, 7332, 6598,
  3757, 6187, 6660, 10529, 6920, 115286, 8451, 8943, 4137, 7514,
  4801, 9709, 23177, 8671, 29915, 26207, 3680, 490, 493856, 58489,
  54897, 4625, 22955, 84952, 10221, 2263, 84641, 4892, 1026, 84650,
  8382, 221656, 2969, 144453, 117177, 2626, 8476, 161882, 51807, 4624,
  9612, 55795, 51043, 144348, 51232, 8729, 3899, 11155, 23316, 79006,
  146330, 6403, 6331, 4300, 427, 845, 3313, 113622, 5789, 376132,
  29841, 8462, 203859, 401397, 5506, 55521, 5819, 4059, 6934, 57727,
  23066, 79568, 83478, 10087, 9586, 222194, 10466, 10499, 58499, 79720,
  4772, 10794, 125919, 602, 27332, 59345, 340359, 3705, 64710, 57801,
  10818, 143684, 149281, 55013, 23095, 93649, 84034, 23347, 440926, 11124,
  23293, 51426, 832, 7068, 57646, 152002, 7531, 1398, 100101267, 221937,
  26873, 3709, 10576, 27040, 201176, 284403, 307, 6525, 5387, 1808,
  114907, 1952, 4091, 5096, 23451, 51207, 142891, 4084, 3797, 1185,
  5334, 217, 2042, 7781, 253461, 152404, 2992, 153478, 6695, 26249,
  662, 23036, 22852, 23411, 23387, 6786, 7690, 10021, 8925, 5595,
  63892, 23466, 11149, 139411, 755, 55347, 22820, 55111, 84444, 6258,
  57623, 1387, 3762, 5861, 3339, 114822, 129787, 196528, 54437, 10395,
  4023, 5095, 894, 3156, 1877, 283871, 2673, 23157, 10512, 2258,
  79750, 84665, 26091, 5978, 8751, 79695, 2768, 155382, 84163, 7709,
  4092, 1837, 2064, 4629, 55122, 8882, 150962, 23013, 8742, 489,
  201134, 55114, 84264, 64428, 4089, 861, 7422
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.1 µM) -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.1 µM DEG Files -----------------
deg_0.1 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.1 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late\nresponse (0.1 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.1 &lt;- deg_0.1 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.1 µM)&quot;,
      &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.1 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.1 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.1 µM AF Genes) -----------------
ggplot(deg_0.1, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;AF-Associated Gene logFC by Drug and Response Group (0.1 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20AF%20logFC_3%20logFC3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-AF-logFC_3-logFC3-1">
Past versions of "Mean AF logFC_3 logFC3-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-AF-logFC_3-logFC3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean AF logFC_3 logFC3-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>############ AF GENES — 0.5 µM #######################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- AF Entrez IDs -----------------
entrez_ids &lt;- c(
  3425, 53834, 9464, 79658, 55876, 3781, 9472, 11017, 64395, 23353,
  57794, 79591, 2702, 51306, 857, 10418, 9644, 115509, 89797, 10728,
  282996, 8497, 5695, 9026, 57666, 143384, 5883, 90102, 84909, 221037,
  29119, 79933, 391, 81575, 8738, 5396, 26010, 744, 56981, 80724,
  5781, 6910, 4598, 79068, 2250, 56978, 3752, 5208, 9736, 26778,
  80315, 64374, 79159, 11078, 5308, 3782, 10654, 79776, 6310, 8082,
  2995, 23224, 79741, 5708, 26136, 9967, 84033, 9148, 29982, 81488,
  1788, 9208, 2908, 1021, 5339, 5727, 23118, 10090, 57585, 221154,
  1073, 29998, 817, 858, 3486, 5747, 2104, 55105, 4194, 3570,
  1607, 6586, 5570, 2070, 27145, 4646, 3159, 9748, 79991, 221035,
  5033, 80212, 196385, 79600, 11046, 54014, 151636, 64778, 92597, 3983,
  7090, 9267, 7273, 26018, 9948, 2131, 55870, 5523, 5318, 6239,
  3480, 55777, 2066, 11165, 2028, 57619, 2690, 23414, 55818, 84700,
  28965, 80204, 463, 5108, 222553, 387119, 28981, 11113, 10301, 995,
  23030, 2697, 996, 339500, 54805, 9960, 91404, 145781, 100820829, 84542,
  2176, 51684, 9513, 7473, 4666, 23150, 5915, 5062, 4016, 23039,
  159686, 1839, 5201, 93166, 64753, 29959, 5496, 23245, 5069, 56916,
  92344, 23092, 3992, 9415, 10554, 7456, 9570, 57178, 23143, 161176,
  5424, 2034, 10277, 11278, 79803, 6653, 4756, 132660, 5430, 9031,
  57158, 285761, 8110, 387700, 1829, 4126, 7323, 51308, 7332, 6598,
  3757, 6187, 6660, 10529, 6920, 115286, 8451, 8943, 4137, 7514,
  4801, 9709, 23177, 8671, 29915, 26207, 3680, 490, 493856, 58489,
  54897, 4625, 22955, 84952, 10221, 2263, 84641, 4892, 1026, 84650,
  8382, 221656, 2969, 144453, 117177, 2626, 8476, 161882, 51807, 4624,
  9612, 55795, 51043, 144348, 51232, 8729, 3899, 11155, 23316, 79006,
  146330, 6403, 6331, 4300, 427, 845, 3313, 113622, 5789, 376132,
  29841, 8462, 203859, 401397, 5506, 55521, 5819, 4059, 6934, 57727,
  23066, 79568, 83478, 10087, 9586, 222194, 10466, 10499, 58499, 79720,
  4772, 10794, 125919, 602, 27332, 59345, 340359, 3705, 64710, 57801,
  10818, 143684, 149281, 55013, 23095, 93649, 84034, 23347, 440926, 11124,
  23293, 51426, 832, 7068, 57646, 152002, 7531, 1398, 100101267, 221937,
  26873, 3709, 10576, 27040, 201176, 284403, 307, 6525, 5387, 1808,
  114907, 1952, 4091, 5096, 23451, 51207, 142891, 4084, 3797, 1185,
  5334, 217, 2042, 7781, 253461, 152404, 2992, 153478, 6695, 26249,
  662, 23036, 22852, 23411, 23387, 6786, 7690, 10021, 8925, 5595,
  63892, 23466, 11149, 139411, 755, 55347, 22820, 55111, 84444, 6258,
  57623, 1387, 3762, 5861, 3339, 114822, 129787, 196528, 54437, 10395,
  4023, 5095, 894, 3156, 1877, 283871, 2673, 23157, 10512, 2258,
  79750, 84665, 26091, 5978, 8751, 79695, 2768, 155382, 84163, 7709,
  4092, 1837, 2064, 4629, 55122, 8882, 150962, 23013, 8742, 489,
  201134, 55114, 84264, 64428, 4089, 861, 7422
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.5 µM) -----------------
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.5 µM DEG Files -----------------
deg_0.5 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.5 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 µM)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.5 &lt;- deg_0.5 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.5 µM)&quot;,
      &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.5 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.5 µM AF Genes) -----------------
ggplot(deg_0.5, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;AF-Associated Gene logFC by Drug and Response Group (0.5 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20AF%20logFC_3%20logFC3-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-AF-logFC_3-logFC3-2">
Past versions of "Mean AF logFC_3 logFC3-2.png"
</button>
</p>
<div id="fig-no-spaces-Mean-AF-logFC_3-logFC3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean AF logFC_3 logFC3-2.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="proportion-of-heart-failure-gwas-genes-across-corrmotifs-per-concentration"
class="section level2">
<h2><strong>📌 Proportion of Heart Failure GWAS genes across corrmotifs
(per concentration)</strong></h2>
<pre class="r"><code># ----------------- Load Required Libraries -----------------
library(dplyr)
library(ggplot2)

# ----------------- Heart Failure Entrez IDs -----------------
hf_entrez_ids &lt;- c(
  9709, 8882, 4023, 29959, 5496, 3992, 9415, 5308, 1026, 54437, 79068, 10221,
  9031, 1187, 1952, 3705, 84722, 7273, 23293, 155382, 9531, 602, 27258, 84163,
  81846, 79933, 56911, 64753, 93210, 1021, 283450, 5998, 57602, 114991, 7073,
  3156, 100101267, 22996, 285025, 11080, 11124, 54810, 7531, 27241, 4774, 57794,
  463, 91319, 6598, 9640, 2186, 26010, 80816, 571, 88, 51652, 64788, 90523, 2969,
  7781, 80777, 10725, 23387, 817, 134728, 8842, 949, 6934, 129787, 10327, 202052,
  2318, 5578, 6801, 6311, 10019, 80724, 217, 84909, 388591, 55101, 9839, 27161,
  5310, 387119, 4641, 5587, 55188, 222553, 9960, 22852, 10087, 9570, 54497,
  200942, 26249, 4137, 375056, 5409, 64116, 8291, 22876, 339855, 4864, 5142,
  221692, 55023, 51426, 6146, 84251, 8189, 27332, 57099, 1869, 1112, 23327,
  11264, 6001
) %&gt;% as.character()

# ----------------- Load CorrMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% hf_entrez_ids, &quot;HF Genes&quot;, &quot;Non-HF Genes&quot;),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Category = ifelse(Entrez_ID %in% hf_entrez_ids, &quot;HF Genes&quot;, &quot;Non-HF Genes&quot;),
    Concentration = &quot;0.5 µM&quot;
  )

# ----------------- Combine Data -----------------
df_combined &lt;- bind_rows(df_0.1, df_0.5)
df_combined$Category &lt;- factor(df_combined$Category, levels = c(&quot;HF Genes&quot;, &quot;Non-HF Genes&quot;))

# ----------------- Calculate Proportions -----------------
proportion_data &lt;- df_combined %&gt;%
  group_by(Concentration, Response_Group, Category) %&gt;%
  summarise(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Chi-square or Fisher&#39;s Test -----------------
get_chi_or_fisher &lt;- function(df, ref_group) {
  ref_counts &lt;- df %&gt;%
    ungroup() %&gt;%
    filter(Response_Group == ref_group) %&gt;%
    { ref_tbl &lt;- .[ , c(&quot;Category&quot;, &quot;Count&quot;)]; setNames(ref_tbl$Count, ref_tbl$Category) }

  df %&gt;%
    filter(Response_Group != ref_group) %&gt;%
    group_by(Response_Group) %&gt;%
    summarise(
      p_value = {
        group_counts &lt;- Count[Category %in% c(&quot;HF Genes&quot;, &quot;Non-HF Genes&quot;)]
        if (length(group_counts) &lt; 2) group_counts &lt;- c(group_counts, 0)
        contingency_table &lt;- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts[&quot;HF Genes&quot;], ref_counts[&quot;Non-HF Genes&quot;]
        ), nrow = 2)
        if (any(contingency_table &lt; 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = &quot;drop&quot;
    ) %&gt;%
    mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))
}

# ----------------- Run test -----------------
chi_0.1 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.1 µM&quot;), &quot;Non response&quot;)</code></pre>
<pre><code>Warning: There was 1 warning in `summarise()`.
ℹ In argument: `p_value = { ... }`.
ℹ In group 1: `Response_Group = &quot;CX-DOX mid-late response&quot;`.
Caused by warning in `chisq.test()`:
! Chi-squared approximation may be incorrect</code></pre>
<pre class="r"><code>chi_0.5 &lt;- get_chi_or_fisher(proportion_data %&gt;% filter(Concentration == &quot;0.5 µM&quot;), &quot;Non response&quot;)
chi_all &lt;- bind_rows(
  chi_0.1 %&gt;% mutate(Concentration = &quot;0.1 µM&quot;),
  chi_0.5 %&gt;% mutate(Concentration = &quot;0.5 µM&quot;)
)
proportion_data &lt;- left_join(proportion_data, chi_all, by = c(&quot;Concentration&quot;, &quot;Response_Group&quot;))

# ----------------- Factor Order and Star Label -----------------
proportion_data$Response_Group &lt;- factor(proportion_data$Response_Group, levels = c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
))
proportion_data$Concentration &lt;- factor(proportion_data$Concentration, levels = c(&quot;0.1 µM&quot;, &quot;0.5 µM&quot;))
label_data &lt;- proportion_data %&gt;%
  group_by(Concentration, Response_Group) %&gt;%
  summarise(Significance = dplyr::first(Significance), .groups = &quot;drop&quot;) %&gt;%
  filter(!is.na(Significance) &amp; Significance != &quot;&quot;) %&gt;%
  mutate(y_pos = 105)

# ----------------- Final Plot -----------------
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = Significance),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~ Concentration, scales = &quot;free_x&quot;) +
  scale_fill_manual(values = c(&quot;HF Genes&quot; = &quot;#E41A1C&quot;, &quot;Non-HF Genes&quot; = &quot;#377EB8&quot;)) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +
  labs(
    title = &quot;Heart Failure Gene Proportions Across\nCormotif Clusters (0.1 and 0.5 µM)&quot;,
    x = &quot;Response Groups&quot;, y = &quot;Percentage of Genes&quot;, fill = &quot;Gene Category&quot;
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = &quot;bold&quot;),
    axis.title.x = element_text(size = 14, face = &quot;bold&quot;),
    axis.title.y = element_text(size = 14, face = &quot;bold&quot;),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    panel.spacing = unit(1.2, &quot;lines&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/HF%20Prop-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-HF-Prop-1">
Past versions of "HF Prop-1.png"
</button>
</p>
<div id="fig-no-spaces-HF-Prop-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/HF Prop-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-heart-failure-gwas-genes-across-corrmotifs-per-concentration"
class="section level2">
<h2><strong>📌 Mean logFC of Heart Failure GWAS genes across corrmotifs
(per concentration)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Heart Failure Entrez IDs -----------------
entrez_ids &lt;- c(
  9709, 8882, 4023, 29959, 5496, 3992, 9415, 5308, 1026, 54437, 79068, 10221,
  9031, 1187, 1952, 3705, 84722, 7273, 23293, 155382, 9531, 602, 27258, 84163,
  81846, 79933, 56911, 64753, 93210, 1021, 283450, 5998, 57602, 114991, 7073,
  3156, 100101267, 22996, 285025, 11080, 11124, 54810, 7531, 27241, 4774, 57794,
  463, 91319, 6598, 9640, 2186, 26010, 80816, 571, 88, 51652, 64788, 90523, 2969,
  7781, 80777, 10725, 23387, 817, 134728, 8842, 949, 6934, 129787, 10327, 202052,
  2318, 5578, 6801, 6311, 10019, 80724, 217, 84909, 388591, 55101, 9839, 27161,
  5310, 387119, 4641, 5587, 55188, 222553, 9960, 22852, 10087, 9570, 54497,
  200942, 26249, 4137, 375056, 5409, 64116, 8291, 22876, 339855, 4864, 5142,
  221692, 55023, 51426, 6146, 84251, 8189, 27332, 57099, 1869, 1112, 23327,
  11264, 6001
) %&gt;% as.character()

# ----------------- Load CorMotif Groups -----------------
# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate Response Groups -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load and Aggregate DEG Data -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    )
  )

deg_mean &lt;- deg_all %&gt;%
  group_by(Entrez_ID, Concentration) %&gt;%
  summarise(logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;)

# ----------------- Merge and Annotate -----------------
hf_logfc &lt;- inner_join(df_combined, deg_mean, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Group = paste(Response_Group, Concentration)
  )

# ----------------- Define Group Order -----------------
group_levels &lt;- c(
  &quot;Non response&quot;,
  &quot;CX-DOX mid-late response&quot;,
  &quot;DOX-specific response&quot;,
  &quot;DOX only mid-late response&quot;,
  &quot;CX total + DOX early response&quot;,
  &quot;DOX early + CX-DOX mid-late response&quot;
)
hf_logfc$Response_Group &lt;- factor(hf_logfc$Response_Group, levels = group_levels)

# ----------------- Wilcoxon Test -----------------
comparison_map &lt;- list(
  &quot;CX-DOX mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX only mid-late response 0.1 µM&quot; = &quot;Non response 0.1 µM&quot;,
  &quot;DOX-specific response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX only mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;CX total + DOX early response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;,
  &quot;DOX early + CX-DOX mid-late response 0.5 µM&quot; = &quot;Non response 0.5 µM&quot;
)

wilcoxon_results &lt;- lapply(names(comparison_map), function(resp_group) {
  control_group &lt;- comparison_map[[resp_group]]
  resp_vals &lt;- hf_logfc$logFC[hf_logfc$Group == resp_group]
  control_vals &lt;- hf_logfc$logFC[hf_logfc$Group == control_group]

  if (length(resp_vals) &gt;= 2 &amp;&amp; length(control_vals) &gt;= 2) {
    test_result &lt;- wilcox.test(resp_vals, control_vals)
    pval &lt;- test_result$p.value
    label &lt;- case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01  ~ &quot;**&quot;,
      pval &lt; 0.05  ~ &quot;*&quot;,
      TRUE ~ &quot;&quot;
    )
    y_pos &lt;- max(resp_vals, na.rm = TRUE) + 0.5
    return(data.frame(Group = resp_group, y_pos = y_pos, label = label, P_Value = signif(pval, 4)))
  }
  return(NULL)
}) %&gt;% bind_rows()

label_data &lt;- wilcoxon_results %&gt;%
  separate(Group, into = c(&quot;Response_Group&quot;, &quot;Concentration&quot;), sep = &quot; (?=0\\.)&quot;) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = group_levels))

# ----------------- Plot -----------------
ggplot(hf_logfc, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 5, color = &quot;black&quot;) +
  facet_wrap(~Concentration, scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Mean logFC of Heart Failure Genes\nAcross CorMotif Response Groups&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20HF%20logFC-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-HF-logFC-1">
Past versions of "Mean HF logFC-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-HF-logFC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean HF logFC-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-heart-failure-gwas-genes-across-corrmotifs-faceted-by-concentration-timepoint"
class="section level2">
<h2><strong>📌 Mean logFC of Heart Failure GWAS genes across corrmotifs
(Faceted by concentration × timepoint)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Heart Failure Entrez IDs -----------------
entrez_ids &lt;- c(
  9709, 8882, 4023, 29959, 5496, 3992, 9415, 5308, 1026, 54437, 79068, 10221,
  9031, 1187, 1952, 3705, 84722, 7273, 23293, 155382, 9531, 602, 27258, 84163,
  81846, 79933, 56911, 64753, 93210, 1021, 283450, 5998, 57602, 114991, 7073,
  3156, 100101267, 22996, 285025, 11080, 11124, 54810, 7531, 27241, 4774, 57794,
  463, 91319, 6598, 9640, 2186, 26010, 80816, 571, 88, 51652, 64788, 90523, 2969,
  7781, 80777, 10725, 23387, 817, 134728, 8842, 949, 6934, 129787, 10327, 202052,
  2318, 5578, 6801, 6311, 10019, 80724, 217, 84909, 388591, 55101, 9839, 27161,
  5310, 387119, 4641, 5587, 55188, 222553, 9960, 22852, 10087, 9570, 54497,
  200942, 26249, 4137, 375056, 5409, 64116, 8291, 22876, 339855, 4864, 5142,
  221692, 55023, 51426, 6146, 84251, 8189, 27332, 57099, 1869, 1112, 23327,
  11264, 6001
) %&gt;% as.character()

# ----------------- Load CorMotif Response Group Data -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Annotate CorMotif Group Membership -----------------
df_0.1 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response&quot;
    ),
    Concentration = &quot;0.1 µM&quot;
  )

df_0.5 &lt;- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific response&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late response&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early response&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late response&quot;
    ),
    Concentration = &quot;0.5 µM&quot;
  )

df_combined &lt;- bind_rows(df_0.1, df_0.5)

# ----------------- Load DEG Files -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

deg_all &lt;- map2_dfr(deg_files, names(deg_files), ~{
  read_csv(.x, show_col_types = FALSE) %&gt;%
    mutate(Condition = .y)
}) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Concentration = case_when(
      str_detect(Condition, &quot;0.1&quot;) ~ &quot;0.1 µM&quot;,
      str_detect(Condition, &quot;0.5&quot;) ~ &quot;0.5 µM&quot;
    ),
    Timepoint = factor(case_when(
      str_detect(Condition, &quot;_3$&quot;) ~ &quot;3 hr&quot;,
      str_detect(Condition, &quot;_24$&quot;) ~ &quot;24 hr&quot;,
      str_detect(Condition, &quot;_48$&quot;) ~ &quot;48 hr&quot;
    ), levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))
  )

# ----------------- Merge and Annotate -----------------
hf_all &lt;- inner_join(df_combined, deg_all, by = c(&quot;Entrez_ID&quot;, &quot;Concentration&quot;)) %&gt;%
  filter(Entrez_ID %in% entrez_ids) %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response&quot;, &quot;CX-DOX mid-late response&quot;, &quot;DOX-specific response&quot;,
      &quot;DOX only mid-late response&quot;, &quot;CX total + DOX early response&quot;, &quot;DOX early + CX-DOX mid-late response&quot;
    ))
  )

# ----------------- Wilcoxon Tests -----------------
comparison_groups &lt;- levels(hf_all$Response_Group)[-1]
stat_results &lt;- list()

for (conc in unique(hf_all$Concentration)) {
  for (tp in levels(hf_all$Timepoint)) {
    sub &lt;- hf_all %&gt;% filter(Concentration == conc, Timepoint == tp)
    for (grp in comparison_groups) {
      if (grp %in% unique(sub$Response_Group)) {
        test &lt;- tryCatch({
          wilcox.test(
            sub$logFC[sub$Response_Group == grp],
            sub$logFC[sub$Response_Group == &quot;Non response&quot;]
          )
        }, error = function(e) NULL)
        if (!is.null(test) &amp;&amp; !is.nan(test$p.value)) {
          stat_results[[length(stat_results) + 1]] &lt;- data.frame(
            Response_Group = grp,
            Concentration = conc,
            Timepoint = tp,
            p_value = signif(test$p.value, 3),
            label = case_when(
              test$p.value &lt; 0.001 ~ &quot;***&quot;,
              test$p.value &lt; 0.01 ~ &quot;**&quot;,
              test$p.value &lt; 0.05 ~ &quot;*&quot;,
              TRUE ~ &quot;&quot;
            ),
            y_pos = max(sub$logFC[sub$Response_Group == grp], na.rm = TRUE) + 0.5
          )
        }
      }
    }
  }
}

label_data &lt;- bind_rows(stat_results) %&gt;%
  mutate(Response_Group = factor(Response_Group, levels = levels(hf_all$Response_Group)))

# ----------------- Plot -----------------
ggplot(hf_all, aes(x = Response_Group, y = logFC, fill = Response_Group)) +
  geom_boxplot(outlier.size = 0.6, width = 0.7, na.rm = TRUE) +
  geom_text(data = label_data, aes(x = Response_Group, y = y_pos, label = label),
            inherit.aes = FALSE, size = 4, vjust = 0) +
  facet_grid(rows = vars(factor(Timepoint, levels = c(&quot;3 hr&quot;, &quot;24 hr&quot;, &quot;48 hr&quot;))),
             cols = vars(Concentration), scales = &quot;free_x&quot;) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(
    title = &quot;Heart Failure Gene logFC by\nCorMotif Group, Timepoint, and Concentration&quot;,
    x = &quot;Response Group&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Response Group&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;, size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title = element_text(size = 14),
    legend.title = element_text(size = 13),
    strip.text = element_text(size = 13, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20HF%20logFC_2%20logFC1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-HF-logFC_2-logFC1-1">
Past versions of "Mean HF logFC_2 logFC1-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-HF-logFC_2-logFC1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean HF logFC_2 logFC1-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="mean-logfc-of-heart-failure-gwas-genes-across-corrmotifs-faceted-by-concentration-timepoint-drug-cx-vs-dox-side-by-side"
class="section level2">
<h2><strong>📌 Mean logFC of Heart Failure GWAS genes across corrmotifs
(Faceted by concentration × timepoint × drug (CX vs DOX
side-by-side))</strong></h2>
<pre class="r"><code>############ HEART FAILURE GENES — 0.1 µM #######################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Heart Failure Entrez IDs -----------------
entrez_ids &lt;- c(
  9709, 8882, 4023, 29959, 5496, 3992, 9415, 5308, 1026, 54437, 79068, 10221,
  9031, 1187, 1952, 3705, 84722, 7273, 23293, 155382, 9531, 602, 27258, 84163,
  81846, 79933, 56911, 64753, 93210, 1021, 283450, 5998, 57602, 114991, 7073,
  3156, 100101267, 22996, 285025, 11080, 11124, 54810, 7531, 27241, 4774, 57794,
  463, 91319, 6598, 9640, 2186, 26010, 80816, 571, 88, 51652, 64788, 90523, 2969,
  7781, 80777, 10725, 23387, 817, 134728, 8842, 949, 6934, 129787, 10327, 202052,
  2318, 5578, 6801, 6311, 10019, 80724, 217, 84909, 388591, 55101, 9839, 27161,
  5310, 387119, 4641, 5587, 55188, 222553, 9960, 22852, 10087, 9570, 54497,
  200942, 26249, 4137, 375056, 5409, 64116, 8291, 22876, 339855, 4864, 5142,
  221692, 55023, 51426, 6146, 84251, 8189, 27332, 57099, 1869, 1112, 23327,
  11264, 6001
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.1 µM) -----------------
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.1 µM DEG Files -----------------
deg_0.1 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.1 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late\nresponse (0.1 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.1 &lt;- deg_0.1 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.1 µM)&quot;,
      &quot;CX-DOX mid-late\nresponse (0.1 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.1 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.1 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.1 µM HF Genes) -----------------
ggplot(deg_0.1, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;Heart Failure Gene logFC by Drug and Response Group (0.1 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20HF%20logFC_3%20logFC3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-HF-logFC_3-logFC3-1">
Past versions of "Mean HF logFC_3 logFC3-1.png"
</button>
</p>
<div id="fig-no-spaces-Mean-HF-logFC_3-logFC3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean HF logFC_3 logFC3-1.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>############ HEART FAILURE GENES — 0.5 µM #######################

# ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Heart Failure Entrez IDs -----------------
entrez_ids &lt;- c(
  9709, 8882, 4023, 29959, 5496, 3992, 9415, 5308, 1026, 54437, 79068, 10221,
  9031, 1187, 1952, 3705, 84722, 7273, 23293, 155382, 9531, 602, 27258, 84163,
  81846, 79933, 56911, 64753, 93210, 1021, 283450, 5998, 57602, 114991, 7073,
  3156, 100101267, 22996, 285025, 11080, 11124, 54810, 7531, 27241, 4774, 57794,
  463, 91319, 6598, 9640, 2186, 26010, 80816, 571, 88, 51652, 64788, 90523, 2969,
  7781, 80777, 10725, 23387, 817, 134728, 8842, 949, 6934, 129787, 10327, 202052,
  2318, 5578, 6801, 6311, 10019, 80724, 217, 84909, 388591, 55101, 9839, 27161,
  5310, 387119, 4641, 5587, 55188, 222553, 9960, 22852, 10087, 9570, 54497,
  200942, 26249, 4137, 375056, 5409, 64116, 8291, 22876, 339855, 4864, 5142,
  221692, 55023, 51426, 6146, 84251, 8189, 27332, 57099, 1869, 1112, 23327,
  11264, 6001
) %&gt;% as.character()

# ----------------- Load CorMotif Groups (0.5 µM) -----------------
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Helper: Load and Label DEG -----------------
load_deg &lt;- function(file, drug, time_hr) {
  read.csv(file) %&gt;%
    mutate(
      Drug = drug,
      Timepoint = factor(time_hr, levels = c(3, 24, 48)),
      Entrez_ID = as.character(Entrez_ID)
    )
}

# ----------------- Load All 0.5 µM DEG Files -----------------
deg_0.5 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;, &quot;CX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;, &quot;CX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;, &quot;CX&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Concentration = &quot;0.5 µM&quot;,
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 µM)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group)) %&gt;%
  filter(Entrez_ID %in% entrez_ids)

# ----------------- Set Factor Levels -----------------
deg_0.5 &lt;- deg_0.5 %&gt;%
  mutate(
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.5 µM)&quot;,
      &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      &quot;CX total + DOX early\nresponse (0.5 µM)&quot;,
      &quot;DOX early + CX-DOX mid-late\nresponse (0.5 µM)&quot;
    ))
  )

# ----------------- Wilcoxon Test (CX and DOX separate) -----------------
wilcox_data &lt;- deg_0.5 %&gt;%
  group_by(Drug, Timepoint) %&gt;%
  group_modify(~{
    df &lt;- .x
    ref &lt;- df %&gt;% filter(str_detect(Response_Group, &quot;Non response&quot;))
    other &lt;- df %&gt;% filter(!str_detect(Response_Group, &quot;Non response&quot;))

    if (nrow(ref) == 0 || nrow(other) == 0) return(tibble())

    map_dfr(unique(other$Response_Group), function(grp) {
      g_data &lt;- other %&gt;% filter(Response_Group == grp)
      if (nrow(g_data) &gt; 0) {
        test &lt;- tryCatch(
          wilcox.test(g_data$logFC, ref$logFC),
          error = function(e) NULL
        )
        tibble(
          Response_Group = grp,
          p_value = if (!is.null(test)) signif(test$p.value, 3) else NA,
          label = if (!is.null(test)) case_when(
            test$p.value &lt; 0.001 ~ &quot;***&quot;,
            test$p.value &lt; 0.01 ~ &quot;**&quot;,
            test$p.value &lt; 0.05 ~ &quot;*&quot;,
            TRUE ~ &quot;&quot;
          ) else &quot;&quot;,
          y_pos = max(g_data$logFC, na.rm = TRUE) + 0.5
        )
      } else {
        tibble()
      }
    })
  }) %&gt;%
  ungroup()

# ----------------- Final Boxplot (0.5 µM HF Genes) -----------------
ggplot(deg_0.5, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(0.75), outlier.size = 0.6, width = 0.7) +
  geom_text(data = wilcox_data,
            aes(x = Timepoint, y = y_pos, label = label, group = Drug),
            position = position_dodge(0.75),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  facet_grid(. ~ Response_Group, scales = &quot;free_y&quot;) +
  scale_fill_manual(values = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;Heart Failure Gene logFC by Drug and Response Group (0.5 µM)&quot;,
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/Mean%20HF%20logFC_3%20logFC3-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Mean-HF-logFC_3-logFC3-2">
Past versions of "Mean HF logFC_3 logFC3-2.png"
</button>
</p>
<div id="fig-no-spaces-Mean-HF-logFC_3-logFC3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/01a423feca180bf7e8669c85a3c03e54357c1b38/docs/figure/Corrmotif_Conc.Rmd/Mean HF logFC_3 logFC3-2.png" target="_blank">01a423f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="corr-motif-boxplots" class="section level2">
<h2><strong>📌 Corr-motif Boxplots</strong></h2>
</div>
<div id="non-response-0.1-micromolar" class="section level2">
<h2><strong>📌 Non response (0.1 micromolar)</strong></h2>
<pre class="r"><code># Load libraries
library(dplyr)
library(ggplot2)
library(reshape2)

# Load response group files
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)

# Load expression dataset
boxplot1 &lt;- read.csv(&quot;data/boxplot1.csv&quot;, check.names = FALSE)
boxplot1 &lt;- as.data.frame(boxplot1)

# Choose a gene from the non-response group
target_entrez &lt;- 92342  # Ensure this is from prob_1_0.1

# Stop if the gene is not in the non-response group
if (!(target_entrez %in% prob_1_0.1)) {
  stop(&quot;Selected gene is not in the non-response group for 0.1 micromolar.&quot;)
}

# Filter for selected gene
gene_data &lt;- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if(nrow(gene_data) == 0) stop(&quot;No data found for selected ENTIREZID.&quot;)

# Reshape to long format
gene_data_long &lt;- melt(gene_data,
                       id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                       variable.name = &quot;Sample&quot;,
                       value.name = &quot;log2CPM&quot;)

# Extract metadata from sample names
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)
  )

# Filter for 0.1 micromolar only
gene_data_long &lt;- gene_data_long %&gt;% filter(Concentration == &quot;0.1&quot;)

# Convert to factors
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = &quot;0.1&quot;)

# Map individual IDs
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, 
                  &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], &quot;Unknown&quot;))

# Define color palette for drugs
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;#08306B&quot;, &quot;DOX&quot; = &quot;#E7298A&quot;, &quot;VEH&quot; = &quot;green&quot;)

# Extract gene symbol for labeling
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +  # Only facets by Time now
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;Non response (0.1 micromolar)&quot;) +
  labs(
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/box1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box1-1">
Past versions of box1-1.png
</button>
</p>
<div id="fig-box1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/box1-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cx-dox-mid-late-response-0.1-micromolar"
class="section level2">
<h2><strong>📌 CX-DOX Mid-Late Response (0.1 micromolar)</strong></h2>
<pre class="r"><code># Load libraries
library(dplyr)
library(ggplot2)
library(reshape2)

# Load response group files
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)


# Load expression dataset
boxplot1 &lt;- read.csv(&quot;data/boxplot1.csv&quot;, check.names = FALSE)
boxplot1 &lt;- as.data.frame(boxplot1)

# Choose a gene from the CX-DOX mid-late response group
target_entrez &lt;- 4521  # Replace with your desired Entrez ID from prob_2_0.1

# Stop if the gene is not in the CX-DOX mid-late response group
if (!(target_entrez %in% prob_2_0.1)) {
  stop(&quot;Selected gene is not in the CX-DOX mid-late response group for 0.1 micromolar.&quot;)
}

# Filter for selected gene
gene_data &lt;- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if(nrow(gene_data) == 0) stop(&quot;No data found for selected ENTREZID.&quot;)

# Reshape to long format
gene_data_long &lt;- melt(gene_data,
                       id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                       variable.name = &quot;Sample&quot;,
                       value.name = &quot;log2CPM&quot;)

# Extract metadata from sample names
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)
  )

# Filter for 0.1 micromolar only
gene_data_long &lt;- gene_data_long %&gt;% filter(Concentration == &quot;0.1&quot;)

# Convert to factors
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = &quot;0.1&quot;)

# Map individual IDs
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, 
                  &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], &quot;Unknown&quot;))

# Define color palette for drugs
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;#08306B&quot;, &quot;DOX&quot; = &quot;#E7298A&quot;, &quot;VEH&quot; = &quot;green&quot;)

# Extract gene symbol for labeling
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +  # Only facets by Time
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;CX-DOX mid-late\nresponse (0.1 micromolar)&quot;) +
  labs(
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/box2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box2-1">
Past versions of box2-1.png
</button>
</p>
<div id="fig-box2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/box2-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dox-only-mid-late-response-0.1-micromolar"
class="section level2">
<h2><strong>📌 DOX-only mid-late response (0.1 micromolar)</strong></h2>
<pre class="r"><code># Load libraries
library(dplyr)
library(ggplot2)
library(reshape2)

# Load response group files
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)


# Load expression dataset
boxplot1 &lt;- read.csv(&quot;data/boxplot1.csv&quot;, check.names = FALSE)
boxplot1 &lt;- as.data.frame(boxplot1)

# Choose a gene from the DOX-only mid-late response group
target_entrez &lt;- 1674  # Replace with your desired Entrez ID from prob_3_0.1

# Stop if the gene is not in the DOX-only mid-late response group
if (!(target_entrez %in% prob_3_0.1)) {
  stop(&quot;Selected gene is not in the DOX-only mid-late response group for 0.1 micromolar.&quot;)
}

# Filter for selected gene
gene_data &lt;- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if(nrow(gene_data) == 0) stop(&quot;No data found for selected ENTREZID.&quot;)

# Reshape to long format
gene_data_long &lt;- reshape2::melt(gene_data,
                                 id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                                 variable.name = &quot;Sample&quot;,
                                 value.name = &quot;log2CPM&quot;)

# Extract metadata from sample names
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)
  )

# Filter for 0.1 micromolar only
gene_data_long &lt;- gene_data_long %&gt;% filter(Concentration == &quot;0.1&quot;)

# Convert to factors
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = &quot;0.1&quot;)

# Map individual IDs
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, 
                  &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], &quot;Unknown&quot;))

# Define color palette for drugs
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;#08306B&quot;, &quot;DOX&quot; = &quot;#E7298A&quot;, &quot;VEH&quot; = &quot;green&quot;)

# Extract gene symbol for labeling
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;DOX-only mid-late\nresponse (0.1 micromolar)&quot;) +
  labs(
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/box3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box3-1">
Past versions of box3-1.png
</button>
</p>
<div id="fig-box3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/box3-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="non-response-0.5-micromolar" class="section level2">
<h2><strong>📌 Non response (0.5 micromolar)</strong></h2>
<pre class="r"><code># Load libraries
library(dplyr)
library(ggplot2)
library(reshape2)

# Load response group: Non response (0.5 µM)
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)


# Load expression dataset
boxplot1 &lt;- read.csv(&quot;data/boxplot1.csv&quot;, check.names = FALSE)
boxplot1 &lt;- as.data.frame(boxplot1)

# Choose a gene from the non-response group (0.5 µM)
target_entrez &lt;- 9446  # Replace with your desired Entrez ID from prob_1_0.5

# Check if it&#39;s in the group
if (!(target_entrez %in% prob_1_0.5)) {
  stop(&quot;Selected gene is not in the Non response group for 0.5 micromolar.&quot;)
}

# Filter for selected gene
gene_data &lt;- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if(nrow(gene_data) == 0) stop(&quot;No data found for selected ENTREZID.&quot;)

# Reshape to long format
gene_data_long &lt;- melt(gene_data,
                       id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                       variable.name = &quot;Sample&quot;,
                       value.name = &quot;log2CPM&quot;)

# Extract metadata from sample names
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)
  )

# Filter for 0.5 micromolar only
gene_data_long &lt;- gene_data_long %&gt;% filter(Concentration == &quot;0.5&quot;)

# Convert to factors
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = &quot;0.5&quot;)

# Map individual IDs
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, 
                  &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], &quot;Unknown&quot;))

# Define color palette for drugs
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;#08306B&quot;, &quot;DOX&quot; = &quot;#E7298A&quot;, &quot;VEH&quot; = &quot;green&quot;)

# Extract gene symbol for labeling
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;Non response (0.5 micromolar)&quot;) +
  labs(
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/box4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box4-1">
Past versions of box4-1.png
</button>
</p>
<div id="fig-box4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/box4-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dox-specific-response-0.5-µm" class="section level2">
<h2><strong>📌 DOX-specific response (0.5 µM)</strong></h2>
<pre class="r"><code># Load libraries
library(dplyr)
library(ggplot2)
library(reshape2)

# Load response group: DOX-specific response (0.5 µM)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)


# Load expression dataset
boxplot1 &lt;- read.csv(&quot;data/boxplot1.csv&quot;, check.names = FALSE)
boxplot1 &lt;- as.data.frame(boxplot1)

# Choose a gene from the DOX-specific response group (0.5 µM)
target_entrez &lt;- 114821  # Replace with your desired Entrez ID from prob_2_0.5

# Check membership
if (!(target_entrez %in% prob_2_0.5)) {
  stop(&quot;Selected gene is not in the DOX-specific response group for 0.5 micromolar.&quot;)
}

# Filter for selected gene
gene_data &lt;- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if(nrow(gene_data) == 0) stop(&quot;No data found for selected ENTREZID.&quot;)

# Reshape to long format
gene_data_long &lt;- melt(gene_data,
                       id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                       variable.name = &quot;Sample&quot;,
                       value.name = &quot;log2CPM&quot;)

# Extract metadata from sample names
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)
  )

# Filter for 0.5 micromolar only
gene_data_long &lt;- gene_data_long %&gt;% filter(Concentration == &quot;0.5&quot;)

# Convert to factors
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = &quot;0.5&quot;)

# Map individual IDs
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, 
                  &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], &quot;Unknown&quot;))

# Define color palette for drugs
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;#08306B&quot;, &quot;DOX&quot; = &quot;#E7298A&quot;, &quot;VEH&quot; = &quot;green&quot;)

# Extract gene symbol for labeling
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;DOX-specific\nresponse (0.5 micromolar)&quot;) +
  labs(
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/box5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box5-1">
Past versions of box5-1.png
</button>
</p>
<div id="fig-box5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/box5-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dox-only-mid-late-response-0.5-µm" class="section level2">
<h2><strong>📌 DOX only mid-late response (0.5 µM)</strong></h2>
<pre class="r"><code># Load libraries
library(dplyr)
library(ggplot2)
library(reshape2)

# Load response group: DOX only mid-late response (0.5 µM)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)


# Load expression dataset
boxplot1 &lt;- read.csv(&quot;data/boxplot1.csv&quot;, check.names = FALSE)
boxplot1 &lt;- as.data.frame(boxplot1)

# Choose a gene from the DOX only mid-late response group (0.5 µM)
target_entrez &lt;- 4892  # Replace with your desired Entrez ID from prob_3_0.5

# Check membership
if (!(target_entrez %in% prob_3_0.5)) {
  stop(&quot;Selected gene is not in the DOX only mid-late response group for 0.5 micromolar.&quot;)
}

# Filter for selected gene
gene_data &lt;- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if(nrow(gene_data) == 0) stop(&quot;No data found for selected ENTREZID.&quot;)

# Reshape to long format
gene_data_long &lt;- melt(gene_data,
                       id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                       variable.name = &quot;Sample&quot;,
                       value.name = &quot;log2CPM&quot;)

# Extract metadata from sample names
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)
  )

# Filter for 0.5 micromolar only
gene_data_long &lt;- gene_data_long %&gt;% filter(Concentration == &quot;0.5&quot;)

# Convert to factors
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = &quot;0.5&quot;)

# Map individual IDs
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, 
                  &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], &quot;Unknown&quot;))

# Define color palette for drugs
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;#08306B&quot;, &quot;DOX&quot; = &quot;#E7298A&quot;, &quot;VEH&quot; = &quot;green&quot;)

# Extract gene symbol for labeling
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;DOX only mid-late\nresponse (0.5 micromolar)&quot;) +
  labs(
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/box6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box6-1">
Past versions of box6-1.png
</button>
</p>
<div id="fig-box6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/box6-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cx-total-dox-early-response-0.5-micromolar"
class="section level2">
<h2><strong>📌 CX total + DOX early response (0.5
micromolar)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(dplyr)
library(ggplot2)
library(reshape2)

# ----------------- Load Response Group -----------------
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)


# ----------------- Load Expression Dataset -----------------
boxplot1 &lt;- read.csv(&quot;data/boxplot1.csv&quot;, check.names = FALSE)
boxplot1 &lt;- as.data.frame(boxplot1)

# ----------------- Choose a Target Gene -----------------
target_entrez &lt;- 100996485  # Replace this with any valid Entrez ID from prob_4_0.5

# Check if gene is valid
if (!(target_entrez %in% prob_4_0.5)) {
  stop(&quot;Selected gene is not in the CX total + DOX early response group for 0.5 micromolar.&quot;)
}

# ----------------- Filter Data for Selected Gene -----------------
gene_data &lt;- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if (nrow(gene_data) == 0) stop(&quot;No data found for selected ENTREZID.&quot;)

# ----------------- Reshape to Long Format -----------------
gene_data_long &lt;- melt(gene_data,
                       id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                       variable.name = &quot;Sample&quot;,
                       value.name = &quot;log2CPM&quot;)

# ----------------- Extract Metadata -----------------
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)
  )

# ----------------- Filter for 0.5 µM Only -----------------
gene_data_long &lt;- gene_data_long %&gt;% filter(Concentration == &quot;0.5&quot;)

# ----------------- Convert to Factors -----------------
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = &quot;0.5&quot;)

# ----------------- Map Individual IDs -----------------
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, 
                  &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], &quot;Unknown&quot;))

# ----------------- Define Drug Color Palette -----------------
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;#08306B&quot;, &quot;DOX&quot; = &quot;#E7298A&quot;, &quot;VEH&quot; = &quot;green&quot;)

# ----------------- Extract Gene Symbol -----------------
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# ----------------- Plot -----------------
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;CX total+ DOX early\nresponse (0.5 micromolar)&quot;) +
  labs(
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/box7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box7-1">
Past versions of box7-1.png
</button>
</p>
<div id="fig-box7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/a77ea961c598cd4f1f5109f61945452408af492b/docs/figure/Corrmotif_Conc.Rmd/box7-1.png" target="_blank">a77ea96</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/box7-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dox-early-cx-dox-mid-late-response-0.5-µm"
class="section level2">
<h2><strong>📌 DOX early + CX-DOX mid-late response (0.5
µM)</strong></h2>
<pre class="r"><code># Load libraries
library(dplyr)
library(ggplot2)
library(reshape2)

# Load response group: DOX early + CX-DOX mid-late response (0.5 µM)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)


# Load expression dataset
boxplot1 &lt;- read.csv(&quot;data/boxplot1.csv&quot;, check.names = FALSE)
boxplot1 &lt;- as.data.frame(boxplot1)

# Choose a gene from the DOX early + CX-DOX mid-late response group (0.5 µM)
target_entrez &lt;- 672  # Replace with your desired Entrez ID from prob_5_0.5

# Check membership
if (!(target_entrez %in% prob_5_0.5)) {
  stop(&quot;Selected gene is not in the DOX early + CX-DOX mid-late response group for 0.5 micromolar.&quot;)
}

# Filter for selected gene
gene_data &lt;- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if(nrow(gene_data) == 0) stop(&quot;No data found for selected ENTREZID.&quot;)

# Reshape to long format
gene_data_long &lt;- melt(gene_data,
                       id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                       variable.name = &quot;Sample&quot;,
                       value.name = &quot;log2CPM&quot;)

# Extract metadata from sample names
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)
  )

# Filter for 0.5 micromolar only
gene_data_long &lt;- gene_data_long %&gt;% filter(Concentration == &quot;0.5&quot;)

# Convert to factors
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = &quot;0.5&quot;)

# Map individual IDs
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, 
                  &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], &quot;Unknown&quot;))

# Define color palette for drugs
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;#08306B&quot;, &quot;DOX&quot; = &quot;#E7298A&quot;, &quot;VEH&quot; = &quot;green&quot;)

# Extract gene symbol for labeling
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;DOX early+ CX-DOX mid-late\nresponse (0.5 micromolar)&quot;) +
  labs(
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/box8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box8-1">
Past versions of box8-1.png
</button>
</p>
<div id="fig-box8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aab6b9f3f2edcffc45770d376b789be169cfe44e/docs/figure/Corrmotif_Conc.Rmd/box8-1.png" target="_blank">aab6b9f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="corrmotif-conc-boxplot-for-manuscript-0.1"
class="section level2">
<h2><strong>📌 Corrmotif conc boxplot for manuscript 0.1</strong></h2>
<pre class="r"><code># Load Required Libraries
library(dplyr)
library(ggplot2)

# ----------------- Load Response Groups -----------------

# 0.1 µM
prob_1_0.1 &lt;- as.character(read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID)
prob_2_0.1 &lt;- as.character(read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID)
prob_3_0.1 &lt;- as.character(read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID)

# ----------------- Load DEG Data -----------------

# Helper function to load and label
load_deg &lt;- function(path, drug, time) {
  read.csv(path) %&gt;%
    mutate(Drug = drug, Timepoint = time)
}

# 0.1 µM DEG data
deg_0.1 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;, &quot;CX.5461&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;, &quot;CX.5461&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;, &quot;CX.5461&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ &quot;Non response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_2_0.1 ~ &quot;CX-DOX mid-late response\n(0.1 µM)&quot;,
      Entrez_ID %in% prob_3_0.1 ~ &quot;DOX only mid-late response\n(0.1 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group))

# ----------------- Set Factor Levels -----------------

deg_0.1 &lt;- deg_0.1 %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(3, 24, 48)),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.1 µM)&quot;,
      &quot;CX-DOX mid-late response\n(0.1 µM)&quot;,
      &quot;DOX only mid-late response\n(0.1 µM)&quot;
    ))
  )

# ----------------- Plot Boxplot (Grouped by Timepoint, Faceted by Response Group) -----------------

ggplot(deg_0.1, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75)) +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  facet_grid(. ~ Response_Group, scales = &quot;free_x&quot;) +
  theme_bw() +
  labs(
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;Log Fold Change&quot;,
    title = &quot;Log Fold Change Distribution by Drug and Response Group (0.1 µM)&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/box9-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box9-1">
Past versions of box9-1.png
</button>
</p>
<div id="fig-box9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/a56887202d8df9b0816df4b966d55f223263e84a/docs/figure/Corrmotif_Conc.Rmd/box9-1.png" target="_blank">a568872</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="corrmotif-conc-boxplot-for-manuscript-0.5"
class="section level2">
<h2><strong>📌 Corrmotif conc boxplot for manuscript 0.5</strong></h2>
<pre class="r"><code># Load Required Libraries
library(dplyr)
library(ggplot2)

# ----------------- Load Response Groups -----------------

# 0.5 µM
prob_1_0.5 &lt;- as.character(read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID)
prob_2_0.5 &lt;- as.character(read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID)
prob_3_0.5 &lt;- as.character(read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID)
prob_4_0.5 &lt;- as.character(read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID)
prob_5_0.5 &lt;- as.character(read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID)

# ----------------- Load DEG Data -----------------

# Helper function to load and label
load_deg &lt;- function(path, drug, time) {
  read.csv(path) %&gt;%
    mutate(Drug = drug, Timepoint = time)
}

# 0.5 µM DEG data
deg_0.5 &lt;- bind_rows(
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;, &quot;CX.5461&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;, &quot;CX.5461&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;, &quot;CX.5461&quot;, 48),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;, &quot;DOX&quot;, 3),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;, &quot;DOX&quot;, 24),
  load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;, &quot;DOX&quot;, 48)
) %&gt;%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ &quot;Non response\n(0.5 µM)&quot;,
      Entrez_ID %in% prob_2_0.5 ~ &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_3_0.5 ~ &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_4_0.5 ~ &quot;CX + DOX (early)\nresponse (0.5 µM)&quot;,
      Entrez_ID %in% prob_5_0.5 ~ &quot;DOX + CX (mid-late)\nresponse (0.5 µM)&quot;,
      TRUE ~ NA_character_
    )
  ) %&gt;%
  filter(!is.na(Response_Group))

# ----------------- Set Factor Levels -----------------

deg_0.5 &lt;- deg_0.5 %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(3, 24, 48)),
    Response_Group = factor(Response_Group, levels = c(
      &quot;Non response\n(0.5 µM)&quot;,
      &quot;DOX-specific\nresponse (0.5 µM)&quot;,
      &quot;DOX only mid-late\nresponse (0.5 µM)&quot;,
      &quot;CX + DOX (early)\nresponse (0.5 µM)&quot;,
      &quot;DOX + CX (mid-late)\nresponse (0.5 µM)&quot;
    ))
  )

# ----------------- Plot Boxplot (Grouped by Timepoint, Faceted by Response Group) -----------------

ggplot(deg_0.5, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75)) +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  facet_grid(. ~ Response_Group, scales = &quot;free_x&quot;) +
  theme_bw() +
  labs(
    x = &quot;Timepoint (hours)&quot;,
    y = &quot;Log Fold Change&quot;,
    title = &quot;Log Fold Change Distribution by Drug and Response Group (0.5 µM)&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_Conc.Rmd/box10-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box10-1">
Past versions of box10-1.png
</button>
</p>
<div id="fig-box10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/a56887202d8df9b0816df4b966d55f223263e84a/docs/figure/Corrmotif_Conc.Rmd/box10-1.png" target="_blank">a568872</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 26100)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] reshape2_1.4.4       lubridate_1.9.4      forcats_1.0.0       
 [4] stringr_1.5.1        purrr_1.0.4          readr_2.1.5         
 [7] tibble_3.2.1         tidyverse_2.0.0      org.Hs.eg.db_3.18.0 
[10] AnnotationDbi_1.64.1 IRanges_2.36.0       S4Vectors_0.40.2    
[13] tidyr_1.3.1          ggplot2_3.5.2        gprofiler2_0.2.3    
[16] BiocParallel_1.36.0  dplyr_1.1.4          Rfast_2.1.5.1       
[19] RcppParallel_5.1.10  zigg_0.0.2           Rcpp_1.0.12         
[22] Cormotif_1.48.0      limma_3.58.1         affy_1.80.0         
[25] Biobase_2.62.0       BiocGenerics_0.48.1  workflowr_1.7.1     

loaded via a namespace (and not attached):
 [1] DBI_1.2.3               bitops_1.0-9            rlang_1.1.3            
 [4] magrittr_2.0.3          git2r_0.36.2            compiler_4.3.0         
 [7] RSQLite_2.3.9           getPass_0.2-4           png_0.1-8              
[10] callr_3.7.6             vctrs_0.6.5             pkgconfig_2.0.3        
[13] crayon_1.5.3            fastmap_1.2.0           XVector_0.42.0         
[16] labeling_0.4.3          promises_1.3.2          rmarkdown_2.29         
[19] tzdb_0.5.0              ps_1.8.1                preprocessCore_1.64.0  
[22] bit_4.6.0               xfun_0.52               zlibbioc_1.48.2        
[25] cachem_1.1.0            GenomeInfoDb_1.38.8     jsonlite_2.0.0         
[28] blob_1.2.4              later_1.3.2             parallel_4.3.0         
[31] R6_2.6.1                RColorBrewer_1.1-3      bslib_0.9.0            
[34] stringi_1.8.3           jquerylib_0.1.4         knitr_1.50             
[37] timechange_0.3.0        httpuv_1.6.15           tidyselect_1.2.1       
[40] rstudioapi_0.17.1       yaml_2.3.10             codetools_0.2-20       
[43] processx_3.8.6          plyr_1.8.9              withr_3.0.2            
[46] KEGGREST_1.42.0         evaluate_1.0.3          Biostrings_2.70.3      
[49] pillar_1.10.2           affyio_1.72.0           BiocManager_1.30.25    
[52] whisker_0.4.1           plotly_4.10.4           generics_0.1.3         
[55] vroom_1.6.5             rprojroot_2.0.4         RCurl_1.98-1.17        
[58] hms_1.1.3               munsell_0.5.1           scales_1.3.0           
[61] glue_1.7.0              lazyeval_0.2.2          tools_4.3.0            
[64] data.table_1.17.0       fs_1.6.3                grid_4.3.0             
[67] colorspace_2.1-0        GenomeInfoDbData_1.2.11 cli_3.6.1              
[70] viridisLite_0.4.2       gtable_0.3.6            sass_0.4.10            
[73] digest_0.6.34           htmlwidgets_1.6.4       farver_2.1.2           
[76] memoise_2.0.1           htmltools_0.5.8.1       lifecycle_1.0.4        
[79] httr_1.4.7              statmod_1.5.0           bit64_4.6.0-1          </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
