<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Corrmotif overlap</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CX5461_Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Corrmotif overlap</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-03-04
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>CX5461_Project/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree56e1527e2bafdcf8961406466d74b2890e9aa04ftargetblank56e1527a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/56e1527e2bafdcf8961406466d74b2890e9aa04f" target="_blank">56e1527</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree56e1527e2bafdcf8961406466d74b2890e9aa04ftargetblank56e1527a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/56e1527e2bafdcf8961406466d74b2890e9aa04f" target="_blank">56e1527</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Unstaged changes:
    Modified:   analysis/Corrmotif_overlap.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Corrmotif_overlap.Rmd</code>) and
HTML (<code>docs/Corrmotif_overlap.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/56e1527e2bafdcf8961406466d74b2890e9aa04f/analysis/Corrmotif_overlap.Rmd" target="_blank">56e1527</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/56e1527e2bafdcf8961406466d74b2890e9aa04f/docs/Corrmotif_overlap.html" target="_blank">56e1527</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/63ae929740c542c542320c42ac26e7d93dbe842e/analysis/Corrmotif_overlap.Rmd" target="_blank">63ae929</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/63ae929740c542c542320c42ac26e7d93dbe842e/docs/Corrmotif_overlap.html" target="_blank">63ae929</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
<td>
Commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div
id="overlap-of-go-functions-between-corrmotif-all-and-corrmotif-conc."
class="section level2">
<h2><strong>📌 Overlap of GO functions between Corrmotif all and
corrmotif Conc.</strong></h2>
<pre class="r"><code>library(UpSetR)
library(dplyr)
library(tools)
library(GO.db)

# Set the folder path
folder_path &lt;- &quot;data/all_GO&quot;

# Get a list of all CSV files in the folder
csv_files &lt;- list.files(folder_path, pattern = &quot;\\.csv$&quot;, full.names = TRUE)

# Loop through each file and assign it as a variable in the global environment
for (file in csv_files) {
  # Generate a valid R variable name from the file name (remove extension and replace spaces)
  file_name &lt;- tools::file_path_sans_ext(basename(file))
  file_name &lt;- gsub(&quot; &quot;, &quot;_&quot;, file_name)  # Replace spaces with underscores
  file_name &lt;- make.names(file_name)  # Ensure the name is valid in R
  
  # Assign the CSV file as a variable in the environment
  assign(file_name, read.csv(file, stringsAsFactors = FALSE))
}

# Define datasets (lists of Entrez Gene IDs)
sets &lt;- list(
  &quot;Non response all&quot; = prob_all_1$ID,
  &quot;CX_DOX shared late response all&quot; = prob_all_2$ID,
  &quot;Dox specific response all&quot; = prob_all_3$ID,
  &quot;Late high dose DOX specific response all&quot; = prob_all_4$ID,
  
  &quot;Non response (0.1)&quot; = prob_1_0.1$ID,
  &quot;DOX only mid-late (0.1)&quot; = prob_2_0.1$ID,
  &quot;CX_DOX mid-late (0.1)&quot; = prob_3_0.1$ID,
  
  &quot;Non response (0.5)&quot; = prob_1_0.5$ID,
  &quot;DOX only early-mid (0.5)&quot; = prob_2_0.5$ID,
  &quot;DOX only mid-late (0.5)&quot; = prob_3_0.5$ID,
  &quot;CX only mid-late (0.5)&quot; = prob_4_0.5$ID,
  &quot;CX_DOX mid-late (0.5)&quot; = prob_5_0.5$ID
)

# Create a binary matrix for UpSet plot
all_genes &lt;- unique(unlist(sets))  # Get all unique Entrez Gene IDs
binary_matrix &lt;- data.frame(Gene_ID = all_genes)  # Initialize DataFrame

# Convert gene lists into a presence/absence matrix (1 = present, 0 = absent)
for (set_name in names(sets)) {
  binary_matrix[[set_name]] &lt;- as.integer(all_genes %in% sets[[set_name]])
}

# Remove Gene_ID column as UpSetR only needs the binary matrix
binary_matrix &lt;- binary_matrix[, -1]

upset(binary_matrix,
      sets = names(sets),
      order.by = &quot;freq&quot;,
      sets.bar.color = &quot;#56B4E9&quot;,  # Blue bars for set sizes
      mainbar.y.label = &quot;Number of Shared Functions&quot;,
      sets.x.label = &quot;GO terms per set&quot;,
      text.scale = 1.2,
      nintersects = 30)</code></pre>
<p><img src="figure/Corrmotif_overlap.Rmd/0verlap-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-0verlap-1">
Past versions of 0verlap-1.png
</button>
</p>
<div id="fig-0verlap-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/63ae929740c542c542320c42ac26e7d93dbe842e/docs/figure/Corrmotif_overlap.Rmd/0verlap-1.png" target="_blank">63ae929</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="identify-unique-go-terms-for-each-response-group"
class="section level2">
<h2><strong>📌 Identify Unique GO Terms for Each Response
Group</strong></h2>
<pre class="r"><code># Create a list to store unique GO terms per category
unique_go_terms &lt;- list()

# Loop through each set to find unique GO terms
for (set_name in names(sets)) {
  # Get the GO terms for the current set
  current_go_terms &lt;- sets[[set_name]]
  
  # Find GO terms that appear **only** in this set and not in others
  unique_terms &lt;- current_go_terms[!(current_go_terms %in% unlist(sets[names(sets) != set_name]))]
  
  # Store in the list if there are any unique terms
  if (length(unique_terms) &gt; 0) {
    unique_go_terms[[set_name]] &lt;- unique_terms
  }
}

# Display unique GO terms for each category
unique_go_terms</code></pre>
<pre><code>$`Non response all`
[1] &quot;GO:0071339&quot; &quot;GO:1990204&quot; &quot;GO:0101031&quot; &quot;GO:0070469&quot; &quot;GO:0070971&quot;
[6] &quot;GO:0044665&quot;

$`CX_DOX shared late response all`
[1] &quot;GO:0009132&quot; &quot;GO:0097421&quot; &quot;GO:0140719&quot; &quot;GO:0140299&quot; &quot;GO:0035064&quot;
[6] &quot;GO:0140034&quot; &quot;GO:1990498&quot; &quot;GO:0005682&quot; &quot;GO:0030532&quot;

$`Dox specific response all`
[1] &quot;GO:0045177&quot; &quot;GO:0016323&quot; &quot;GO:0009925&quot;

$`Non response (0.1)`
 [1] &quot;GO:0034470&quot; &quot;GO:0042254&quot; &quot;GO:0022613&quot; &quot;GO:0140053&quot; &quot;GO:0006413&quot;
 [6] &quot;GO:0006364&quot; &quot;GO:0032543&quot; &quot;GO:0048193&quot; &quot;GO:0033108&quot; &quot;GO:0016072&quot;
[11] &quot;GO:0008135&quot; &quot;GO:0090079&quot; &quot;GO:0030684&quot; &quot;GO:0005759&quot; &quot;GO:0098800&quot;
[16] &quot;GO:0098803&quot; &quot;GO:0010494&quot; &quot;GO:0000313&quot; &quot;GO:0005761&quot; &quot;GO:0035770&quot;

$`DOX only mid-late (0.1)`
 [1] &quot;GO:0009162&quot; &quot;GO:0009130&quot; &quot;GO:0006999&quot; &quot;GO:0006221&quot; &quot;GO:0007096&quot;
 [6] &quot;GO:0045859&quot; &quot;GO:0006978&quot; &quot;GO:0009129&quot; &quot;GO:0046785&quot; &quot;GO:0002562&quot;
[11] &quot;GO:0016444&quot; &quot;GO:0042772&quot; &quot;GO:0010458&quot; &quot;GO:0071900&quot; &quot;GO:0051292&quot;
[16] &quot;GO:0006289&quot; &quot;GO:0008584&quot; &quot;GO:0008406&quot; &quot;GO:0046546&quot; &quot;GO:0045739&quot;
[21] &quot;GO:0045137&quot; &quot;GO:0043549&quot; &quot;GO:0009124&quot; &quot;GO:0048144&quot; &quot;GO:0008301&quot;
[26] &quot;GO:0017056&quot; &quot;GO:0000803&quot; &quot;GO:0043240&quot;

$`CX_DOX mid-late (0.1)`
[1] &quot;GO:0005402&quot;

$`DOX only early-mid (0.5)`
[1] &quot;GO:0140297&quot; &quot;GO:0061629&quot; &quot;GO:0090575&quot; &quot;GO:0005667&quot; &quot;GO:0097550&quot;
[6] &quot;GO:0005669&quot;

$`DOX only mid-late (0.5)`
 [1] &quot;GO:0007186&quot; &quot;GO:0048738&quot; &quot;GO:0014706&quot; &quot;GO:0099084&quot; &quot;GO:0099173&quot;
 [6] &quot;GO:0045598&quot; &quot;GO:0046486&quot; &quot;GO:0003013&quot; &quot;GO:0045444&quot; &quot;GO:0046620&quot;
[11] &quot;GO:0016236&quot;

$`CX only mid-late (0.5)`
 [1] &quot;GO:0048599&quot; &quot;GO:0090305&quot; &quot;GO:0009994&quot; &quot;GO:2000243&quot; &quot;GO:0018105&quot;
 [6] &quot;GO:0018209&quot; &quot;GO:0035561&quot; &quot;GO:0090657&quot; &quot;GO:0048477&quot; &quot;GO:0071732&quot;
[11] &quot;GO:0006264&quot; &quot;GO:0000722&quot; &quot;GO:0071731&quot; &quot;GO:1902170&quot; &quot;GO:0004518&quot;
[16] &quot;GO:0019205&quot;

$`CX_DOX mid-late (0.5)`
 [1] &quot;GO:0048146&quot; &quot;GO:0030865&quot; &quot;GO:0051493&quot; &quot;GO:0051782&quot; &quot;GO:0090399&quot;
 [6] &quot;GO:2000279&quot; &quot;GO:0030010&quot; &quot;GO:1901875&quot; &quot;GO:0032147&quot; &quot;GO:0031398&quot;
[11] &quot;GO:0007163&quot; &quot;GO:0051972&quot; &quot;GO:0000235&quot; &quot;GO:0005818&quot; &quot;GO:0101019&quot;</code></pre>
</div>
<div id="map-go-ids-to-function-names-using-go.db"
class="section level2">
<h2><strong>📌 Map GO IDs to Function Names Using GO.db</strong></h2>
<pre class="r"><code>map_go_terms_local &lt;- function(go_ids) {
  go_names &lt;- unlist(mget(go_ids, GOTERM, ifnotfound = NA))  # Retrieve function names
  go_descriptions &lt;- sapply(go_names, function(x) if (!is.na(x)) Term(x) else NA)
  
  # Create a dataframe
  go_mapping &lt;- data.frame(GO_ID = go_ids, Function = go_descriptions, stringsAsFactors = FALSE)
  
  # Remove NAs (unrecognized GO IDs)
  go_mapping &lt;- go_mapping[!is.na(go_mapping$Function), ]
  
  return(go_mapping)
}
mapped_unique_go_terms &lt;- list()

for (set_name in names(unique_go_terms)) {
  if (length(unique_go_terms[[set_name]]) &gt; 0) {
    go_data &lt;- map_go_terms_local(unique_go_terms[[set_name]])
    mapped_unique_go_terms[[set_name]] &lt;- go_data
  }
}</code></pre>
<pre><code>Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;</code></pre>
<pre class="r"><code># Display mapped GO terms
mapped_unique_go_terms</code></pre>
<pre><code>$`Non response all`
                GO_ID                          Function
GO:0071339 GO:0071339                      MLL1 complex
GO:1990204 GO:1990204            oxidoreductase complex
GO:0101031 GO:0101031 protein folding chaperone complex
GO:0070469 GO:0070469                       respirasome
GO:0070971 GO:0070971   endoplasmic reticulum exit site
GO:0044665 GO:0044665                    MLL1/2 complex

$`CX_DOX shared late response all`
                GO_ID                                 Function
GO:0009132 GO:0009132 nucleoside diphosphate metabolic process
GO:0097421 GO:0097421                       liver regeneration
GO:0140719 GO:0140719   constitutive heterochromatin formation
GO:0140299 GO:0140299           small molecule sensor activity
GO:0035064 GO:0035064               methylated histone binding
GO:0140034 GO:0140034    methylation-dependent protein binding
GO:1990498 GO:1990498              mitotic spindle microtubule
GO:0005682 GO:0005682                                 U5 snRNP
GO:0030532 GO:0030532  small nuclear ribonucleoprotein complex

$`Dox specific response all`
                GO_ID                    Function
GO:0045177 GO:0045177         apical part of cell
GO:0016323 GO:0016323 basolateral plasma membrane
GO:0009925 GO:0009925       basal plasma membrane

$`Non response (0.1)`
                GO_ID                                             Function
GO:0034470 GO:0034470                                     ncRNA processing
GO:0042254 GO:0042254                                  ribosome biogenesis
GO:0022613 GO:0022613                 ribonucleoprotein complex biogenesis
GO:0140053 GO:0140053                        mitochondrial gene expression
GO:0006413 GO:0006413                             translational initiation
GO:0006364 GO:0006364                                      rRNA processing
GO:0032543 GO:0032543                            mitochondrial translation
GO:0048193 GO:0048193                              Golgi vesicle transport
GO:0033108 GO:0033108     mitochondrial respiratory chain complex assembly
GO:0016072 GO:0016072                               rRNA metabolic process
GO:0008135 GO:0008135             translation factor activity, RNA binding
GO:0090079 GO:0090079 translation regulator activity, nucleic acid binding
GO:0030684 GO:0030684                                          preribosome
GO:0005759 GO:0005759                                 mitochondrial matrix
GO:0098800 GO:0098800         inner mitochondrial membrane protein complex
GO:0098803 GO:0098803                            respiratory chain complex
GO:0010494 GO:0010494                           cytoplasmic stress granule
GO:0000313 GO:0000313                                  organellar ribosome
GO:0005761 GO:0005761                               mitochondrial ribosome
GO:0035770 GO:0035770                            ribonucleoprotein granule

$`DOX only mid-late (0.1)`
                GO_ID
GO:0009162 GO:0009162
GO:0009130 GO:0009130
GO:0006999 GO:0006999
GO:0006221 GO:0006221
GO:0007096 GO:0007096
GO:0045859 GO:0045859
GO:0006978 GO:0006978
GO:0009129 GO:0009129
GO:0046785 GO:0046785
GO:0002562 GO:0002562
GO:0016444 GO:0016444
GO:0042772 GO:0042772
GO:0010458 GO:0010458
GO:0071900 GO:0071900
GO:0051292 GO:0051292
GO:0006289 GO:0006289
GO:0008584 GO:0008584
GO:0008406 GO:0008406
GO:0046546 GO:0046546
GO:0045739 GO:0045739
GO:0045137 GO:0045137
GO:0043549 GO:0043549
GO:0009124 GO:0009124
GO:0048144 GO:0048144
GO:0008301 GO:0008301
GO:0017056 GO:0017056
GO:0000803 GO:0000803
GO:0043240 GO:0043240
                                                                                                                  Function
GO:0009162                                                             deoxyribonucleoside monophosphate metabolic process
GO:0009130                                                        pyrimidine nucleoside monophosphate biosynthetic process
GO:0006999                                                                                       nuclear pore organization
GO:0006221                                                                      pyrimidine nucleotide biosynthetic process
GO:0007096                                                                                 regulation of exit from mitosis
GO:0045859                                                                           regulation of protein kinase activity
GO:0006978 DNA damage response, signal transduction by p53 class mediator resulting in transcription of p21 class mediator
GO:0009129                                                           pyrimidine nucleoside monophosphate metabolic process
GO:0046785                                                                                      microtubule polymerization
GO:0002562                    somatic diversification of immune receptors via germline recombination within a single locus
GO:0016444                                                                                  somatic cell DNA recombination
GO:0042772                                             DNA damage response, signal transduction resulting in transcription
GO:0010458                                                                                               exit from mitosis
GO:0071900                                                          regulation of protein serine/threonine kinase activity
GO:0051292                                                                                   nuclear pore complex assembly
GO:0006289                                                                                      nucleotide-excision repair
GO:0008584                                                                                          male gonad development
GO:0008406                                                                                               gonad development
GO:0046546                                                              development of primary male sexual characteristics
GO:0045739                                                                               positive regulation of DNA repair
GO:0045137                                                                   development of primary sexual characteristics
GO:0043549                                                                                   regulation of kinase activity
GO:0009124                                                                   nucleoside monophosphate biosynthetic process
GO:0048144                                                                                        fibroblast proliferation
GO:0008301                                                                                            DNA binding, bending
GO:0017056                                                                          structural constituent of nuclear pore
GO:0000803                                                                                                  sex chromosome
GO:0043240                                                                                 Fanconi anaemia nuclear complex

$`CX_DOX mid-late (0.1)`
                GO_ID                                          Function
GO:0005402 GO:0005402 carbohydrate:monoatomic cation symporter activity

$`DOX only early-mid (0.5)`
                GO_ID
GO:0140297 GO:0140297
GO:0061629 GO:0061629
GO:0090575 GO:0090575
GO:0005667 GO:0005667
GO:0097550 GO:0097550
GO:0005669 GO:0005669
                                                                      Function
GO:0140297                            DNA-binding transcription factor binding
GO:0061629 RNA polymerase II-specific DNA-binding transcription factor binding
GO:0090575                   RNA polymerase II transcription regulator complex
GO:0005667                                     transcription regulator complex
GO:0097550                                 transcription preinitiation complex
GO:0005669                                  transcription factor TFIID complex

$`DOX only mid-late (0.5)`
                GO_ID                                     Function
GO:0007186 GO:0007186 G protein-coupled receptor signaling pathway
GO:0048738 GO:0048738            cardiac muscle tissue development
GO:0014706 GO:0014706           striated muscle tissue development
GO:0099084 GO:0099084     postsynaptic specialization organization
GO:0099173 GO:0099173                     postsynapse organization
GO:0045598 GO:0045598       regulation of fat cell differentiation
GO:0046486 GO:0046486               glycerolipid metabolic process
GO:0003013 GO:0003013                   circulatory system process
GO:0045444 GO:0045444                     fat cell differentiation
GO:0046620 GO:0046620                   regulation of organ growth
GO:0016236 GO:0016236                               macroautophagy

$`CX only mid-late (0.5)`
                GO_ID                                       Function
GO:0048599 GO:0048599                             oocyte development
GO:0090305 GO:0090305    nucleic acid phosphodiester bond hydrolysis
GO:0009994 GO:0009994                         oocyte differentiation
GO:2000243 GO:2000243    positive regulation of reproductive process
GO:0018105 GO:0018105                peptidyl-serine phosphorylation
GO:0018209 GO:0018209                   peptidyl-serine modification
GO:0035561 GO:0035561                regulation of chromatin binding
GO:0090657 GO:0090657                     telomeric loop disassembly
GO:0048477 GO:0048477                                      oogenesis
GO:0071732 GO:0071732              cellular response to nitric oxide
GO:0006264 GO:0006264                  mitochondrial DNA replication
GO:0000722 GO:0000722         telomere maintenance via recombination
GO:0071731 GO:0071731                       response to nitric oxide
GO:1902170 GO:1902170 cellular response to reactive nitrogen species
GO:0004518 GO:0004518                              nuclease activity
GO:0019205 GO:0019205 nucleobase-containing compound kinase activity

$`CX_DOX mid-late (0.5)`
                GO_ID
GO:0048146 GO:0048146
GO:0030865 GO:0030865
GO:0051493 GO:0051493
GO:0051782 GO:0051782
GO:0090399 GO:0090399
GO:2000279 GO:2000279
GO:0030010 GO:0030010
GO:1901875 GO:1901875
GO:0032147 GO:0032147
GO:0031398 GO:0031398
GO:0007163 GO:0007163
GO:0051972 GO:0051972
GO:0000235 GO:0000235
GO:0005818 GO:0005818
GO:0101019 GO:0101019
                                                                 Function
GO:0048146                positive regulation of fibroblast proliferation
GO:0030865                             cortical cytoskeleton organization
GO:0051493                        regulation of cytoskeleton organization
GO:0051782                           negative regulation of cell division
GO:0090399                                         replicative senescence
GO:2000279                negative regulation of DNA biosynthetic process
GO:0030010                                 establishment of cell polarity
GO:1901875 positive regulation of post-translational protein modification
GO:0032147                          activation of protein kinase activity
GO:0031398                  positive regulation of protein ubiquitination
GO:0007163                  establishment or maintenance of cell polarity
GO:0051972                              regulation of telomerase activity
GO:0000235                                             astral microtubule
GO:0005818                                                          aster
GO:0101019                              nucleolar exosome (RNase complex)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 22631)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats4    tools     stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
[1] GO.db_3.18.0         AnnotationDbi_1.64.1 IRanges_2.36.0      
[4] S4Vectors_0.40.1     Biobase_2.62.0       BiocGenerics_0.48.1 
[7] dplyr_1.1.4          UpSetR_1.4.0        

loaded via a namespace (and not attached):
 [1] KEGGREST_1.42.0         gtable_0.3.6            xfun_0.50              
 [4] bslib_0.8.0             ggplot2_3.5.1           vctrs_0.6.5            
 [7] bitops_1.0-7            generics_0.1.3          tibble_3.2.1           
[10] RSQLite_2.3.3           blob_1.2.4              pkgconfig_2.0.3        
[13] lifecycle_1.0.4         GenomeInfoDbData_1.2.11 farver_2.1.2           
[16] compiler_4.3.0          stringr_1.5.1           git2r_0.35.0           
[19] Biostrings_2.70.1       munsell_0.5.1           httpuv_1.6.15          
[22] GenomeInfoDb_1.38.8     htmltools_0.5.8.1       sass_0.4.9             
[25] RCurl_1.98-1.13         yaml_2.3.10             later_1.3.2            
[28] pillar_1.10.1           crayon_1.5.3            jquerylib_0.1.4        
[31] whisker_0.4.1           cachem_1.0.8            tidyselect_1.2.1       
[34] digest_0.6.34           stringi_1.8.3           labeling_0.4.3         
[37] rprojroot_2.0.4         fastmap_1.1.1           grid_4.3.0             
[40] colorspace_2.1-0        cli_3.6.1               magrittr_2.0.3         
[43] withr_3.0.2             scales_1.3.0            promises_1.3.0         
[46] bit64_4.0.5             rmarkdown_2.29          XVector_0.42.0         
[49] httr_1.4.7              bit_4.0.5               gridExtra_2.3          
[52] workflowr_1.7.1         png_0.1-8               memoise_2.0.1          
[55] evaluate_1.0.3          knitr_1.49              rlang_1.1.3            
[58] Rcpp_1.0.12             glue_1.7.0              DBI_1.2.3              
[61] rstudioapi_0.17.1       jsonlite_1.8.9          R6_2.5.1               
[64] plyr_1.8.9              fs_1.6.3                zlibbioc_1.48.0        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
