<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Correlation between CX and DOX logFC</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CX5461_Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Correlation between CX and DOX logFC</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-03-30
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>CX5461_Project/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree2f71a7fa63393244ef41553d1c63b775067ca949targetblank2f71a7fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/2f71a7fa63393244ef41553d1c63b775067ca949" target="_blank">2f71a7f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree2f71a7fa63393244ef41553d1c63b775067ca949targetblank2f71a7fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/2f71a7fa63393244ef41553d1c63b775067ca949" target="_blank">2f71a7f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  0.1 box.svg
    Untracked:  0.1 density.svg
    Untracked:  0.1.emf
    Untracked:  0.1.svg
    Untracked:  0.5 box.svg
    Untracked:  0.5 density.svg
    Untracked:  0.5.svg
    Untracked:  Rplot.svg
    Untracked:  Rplot01.svg
    Untracked:  Rplot02.svg
    Untracked:  Rplot03.svg

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/CX_DOX.Rmd</code>) and HTML
(<code>docs/CX_DOX.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/2f71a7fa63393244ef41553d1c63b775067ca949/docs/CX_DOX.html" target="_blank">2f71a7f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-30
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/7f99e6853f3e7f92acb27e732f083cdd9d7781c3/analysis/CX_DOX.Rmd" target="_blank">7f99e68</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-30
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/7f99e6853f3e7f92acb27e732f083cdd9d7781c3/docs/CX_DOX.html" target="_blank">7f99e68</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-30
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/61329dfa00e657eb011e5a9aef5444048e1d5c46/analysis/CX_DOX.Rmd" target="_blank">61329df</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-16
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/61329dfa00e657eb011e5a9aef5444048e1d5c46/docs/CX_DOX.html" target="_blank">61329df</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-16
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aec1debb213d99b1dbc743d343c40a7fe213fdad/analysis/CX_DOX.Rmd" target="_blank">aec1deb</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-02
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/aec1debb213d99b1dbc743d343c40a7fe213fdad/docs/CX_DOX.html" target="_blank">aec1deb</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-02
</td>
<td>
Commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="correlation-between-cx-and-dox-logfc" class="section level2">
<h2><strong>📌 Correlation between CX and DOX logFC</strong></h2>
</div>
<div id="load-required-libraries" class="section level2">
<h2><strong>📌 Load Required Libraries</strong></h2>
<pre class="r"><code># Load necessary libraries
library(dplyr)</code></pre>
<pre><code>Warning: package &#39;dplyr&#39; was built under R version 4.3.2</code></pre>
<pre class="r"><code>library(ggplot2)</code></pre>
<pre><code>Warning: package &#39;ggplot2&#39; was built under R version 4.3.3</code></pre>
</div>
<div id="correlation-scatter-plot" class="section level2">
<h2><strong>📌 Correlation Scatter plot</strong></h2>
<pre class="r"><code># Load DEGs Data
CX_0.1_3 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;)
CX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;)
CX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;)
CX_0.5_3 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)

DOX_0.1_3 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;)
DOX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;)
DOX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;)
DOX_0.5_3 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

# Extract Significant DEGs
DEG1 &lt;- as.character(CX_0.1_3$Entrez_ID[CX_0.1_3$adj.P.Val &lt; 0.05])
DEG2 &lt;- as.character(CX_0.1_24$Entrez_ID[CX_0.1_24$adj.P.Val &lt; 0.05])
DEG3 &lt;- as.character(CX_0.1_48$Entrez_ID[CX_0.1_48$adj.P.Val &lt; 0.05])
DEG4 &lt;- as.character(CX_0.5_3$Entrez_ID[CX_0.5_3$adj.P.Val &lt; 0.05])
DEG5 &lt;- as.character(CX_0.5_24$Entrez_ID[CX_0.5_24$adj.P.Val &lt; 0.05])
DEG6 &lt;- as.character(CX_0.5_48$Entrez_ID[CX_0.5_48$adj.P.Val &lt; 0.05])
DEG7 &lt;- as.character(DOX_0.1_3$Entrez_ID[DOX_0.1_3$adj.P.Val &lt; 0.05])
DEG8 &lt;- as.character(DOX_0.1_24$Entrez_ID[DOX_0.1_24$adj.P.Val &lt; 0.05])
DEG9 &lt;- as.character(DOX_0.1_48$Entrez_ID[DOX_0.1_48$adj.P.Val &lt; 0.05])
DEG10 &lt;- as.character(DOX_0.5_3$Entrez_ID[DOX_0.5_3$adj.P.Val &lt; 0.05])
DEG11 &lt;- as.character(DOX_0.5_24$Entrez_ID[DOX_0.5_24$adj.P.Val &lt; 0.05])
DEG12 &lt;- as.character(DOX_0.5_48$Entrez_ID[DOX_0.5_48$adj.P.Val &lt; 0.05])

# Ensure Entrez_ID is a character across all datasets
datasets &lt;- list(CX_0.1_3, CX_0.1_24, CX_0.1_48, CX_0.5_3, CX_0.5_24, CX_0.5_48, 
                 DOX_0.1_3, DOX_0.1_24, DOX_0.1_48, DOX_0.5_3, DOX_0.5_24, DOX_0.5_48)

for (i in seq_along(datasets)) {
  datasets[[i]]$Entrez_ID &lt;- as.character(datasets[[i]]$Entrez_ID)
}

# Define dataset pairs for correlation analysis
dataset_pairs &lt;- list(
  list(&quot;CX_0.1_3&quot;, CX_0.1_3, &quot;DOX_0.1_3&quot;, DOX_0.1_3, &quot;3 hours&quot;, &quot;0.1 micromolar&quot;),
  list(&quot;CX_0.1_24&quot;, CX_0.1_24, &quot;DOX_0.1_24&quot;, DOX_0.1_24, &quot;24 hours&quot;, &quot;0.1 micromolar&quot;),
  list(&quot;CX_0.1_48&quot;, CX_0.1_48, &quot;DOX_0.1_48&quot;, DOX_0.1_48, &quot;48 hours&quot;, &quot;0.1 micromolar&quot;),
  list(&quot;CX_0.5_3&quot;, CX_0.5_3, &quot;DOX_0.5_3&quot;, DOX_0.5_3, &quot;3 hours&quot;, &quot;0.5 micromolar&quot;),
  list(&quot;CX_0.5_24&quot;, CX_0.5_24, &quot;DOX_0.5_24&quot;, DOX_0.5_24, &quot;24 hours&quot;, &quot;0.5 micromolar&quot;),
  list(&quot;CX_0.5_48&quot;, CX_0.5_48, &quot;DOX_0.5_48&quot;, DOX_0.5_48, &quot;48 hours&quot;, &quot;0.5 micromolar&quot;)
)

# Create an empty list to store merged data
merged_data_list &lt;- list()

# Loop through dataset pairs and merge based on Entrez_ID
for (pair in dataset_pairs) {
  cx_name &lt;- pair[[1]]
  cx_data &lt;- pair[[2]]
  dox_name &lt;- pair[[3]]
  dox_data &lt;- pair[[4]]
  timepoint &lt;- pair[[5]]
  concentration &lt;- pair[[6]]
  
  merged_data &lt;- merge(cx_data, dox_data, by = &quot;Entrez_ID&quot;, suffixes = c(&quot;_CX&quot;, &quot;_DOX&quot;))
  merged_data$Timepoint &lt;- timepoint
  merged_data$Concentration &lt;- concentration
  
  merged_data_list[[paste(cx_name, dox_name, sep = &quot;_vs_&quot;)]] &lt;- merged_data
}

# Combine all merged datasets into a single dataframe
combined_data &lt;- do.call(rbind, merged_data_list)

# Select necessary columns and rename them
combined_data &lt;- combined_data %&gt;%
  dplyr::select(Entrez_ID, logFC_CX = logFC_CX, logFC_DOX = logFC_DOX, Timepoint, Concentration)

# Ensure timepoints and concentrations are in the correct order
combined_data$Timepoint &lt;- factor(combined_data$Timepoint, levels = c(&quot;3 hours&quot;, &quot;24 hours&quot;, &quot;48 hours&quot;))
combined_data$Concentration &lt;- factor(combined_data$Concentration, levels = c(&quot;0.1 micromolar&quot;, &quot;0.5 micromolar&quot;))

# **Step 1: Compute global min and max for y-axis scale**
y_min &lt;- min(combined_data$logFC_DOX, na.rm = TRUE)
y_max &lt;- max(combined_data$logFC_DOX, na.rm = TRUE)

# **Step 2: Compute correlations for each dataset with exact p-values**
correlations &lt;- combined_data %&gt;%
  group_by(Concentration, Timepoint) %&gt;%
  summarise(
    r_value = cor(logFC_CX, logFC_DOX, method = &quot;pearson&quot;),
    p_value = cor.test(logFC_CX, logFC_DOX, method = &quot;pearson&quot;)$p.value,
    .groups = &quot;drop&quot;
  )

# **Step 3: Display only r-value and whether p &lt; 0.05 or p &gt; 0.05**
correlations &lt;- correlations %&gt;%
  mutate(
    significance = ifelse(p_value &lt; 0.05, &quot;p &lt; 0.05&quot;, &quot;p &gt; 0.05&quot;),  # Mark significant comparisons
    label = paste0(&quot;r = &quot;, round(r_value, 3), &quot;\n&quot;, significance)
  )

# **Step 4: Create scatter plots faceted by timepoints and concentration**
scatter_plot &lt;- ggplot(combined_data, aes(x = logFC_CX, y = logFC_DOX)) +
  geom_point(alpha = 0.6, color = &quot;black&quot;) +  # Black scatter points
  geom_smooth(method = &quot;lm&quot;, color = &quot;black&quot;, se = FALSE) +  # Black regression line
  scale_y_continuous(limits = c(y_min, y_max)) +  # Fixed Y-axis across all facets
  labs(
    title = &quot;Correlation between CX and DOX logFC&quot;,
    x = &quot;logFC (CX)&quot;,
    y = &quot;logFC (DOX)&quot;
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 2),
    strip.background = element_rect(fill = &quot;white&quot;, color = &quot;black&quot;, linewidth = 1.5),
    strip.text = element_text(size = 12, face = &quot;bold&quot;, color = &quot;black&quot;)
  ) +
  facet_grid(Timepoint ~ Concentration, scales = &quot;fixed&quot;) +  # Ensure same y-axis scale for all facets
  geom_text(data = correlations,
            aes(x = 1.5, y = y_max * 0.9, label = label),
            inherit.aes = FALSE, size = 3, fontface = &quot;bold&quot;)

# **Step 5: Display the plot**
print(scatter_plot)</code></pre>
<p><img src="figure/CX_DOX.Rmd/load_File%20Paths-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-load_File-Paths-1">
Past versions of "load_File Paths-1.png"
</button>
</p>
<div id="fig-no-spaces-load_File-Paths-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/7f99e6853f3e7f92acb27e732f083cdd9d7781c3/docs/figure/CX_DOX.Rmd/load_File Paths-1.png" target="_blank">7f99e68</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/aec1debb213d99b1dbc743d343c40a7fe213fdad/docs/figure/CX_DOX.Rmd/load_File Paths-1.png" target="_blank">aec1deb</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="correlation-heatmap" class="section level2">
<h2><strong>📌 Correlation Heatmap</strong></h2>
<pre class="r"><code># Load necessary libraries
library(ggplot2)
library(reshape2)
library(dplyr)

# Define dataset pairs for correlation analysis
dataset_pairs &lt;- list(
  list(&quot;CX_0.1_3&quot;, CX_0.1_3, &quot;DOX_0.1_3&quot;, DOX_0.1_3),
  list(&quot;CX_0.1_24&quot;, CX_0.1_24, &quot;DOX_0.1_24&quot;, DOX_0.1_24),
  list(&quot;CX_0.1_48&quot;, CX_0.1_48, &quot;DOX_0.1_48&quot;, DOX_0.1_48),
  list(&quot;CX_0.5_3&quot;, CX_0.5_3, &quot;DOX_0.5_3&quot;, DOX_0.5_3),
  list(&quot;CX_0.5_24&quot;, CX_0.5_24, &quot;DOX_0.5_24&quot;, DOX_0.5_24),
  list(&quot;CX_0.5_48&quot;, CX_0.5_48, &quot;DOX_0.5_48&quot;, DOX_0.5_48)
)

# Create an empty data frame to store correlations
correlation_data &lt;- data.frame(CX_Sample = character(), Correlation = numeric())

# Compute correlations for each CX vs. DOX dataset pair
for (pair in dataset_pairs) {
  cx_name &lt;- pair[[1]]
  cx_data &lt;- pair[[2]]
  dox_name &lt;- pair[[3]]
  dox_data &lt;- pair[[4]]
  
  # Merge datasets on Entrez_ID
  merged_data &lt;- merge(cx_data, dox_data, by = &quot;Entrez_ID&quot;, suffixes = c(&quot;_CX&quot;, &quot;_DOX&quot;))
  
  # Compute Pearson correlation
  r_value &lt;- cor(merged_data$logFC_CX, merged_data$logFC_DOX, method = &quot;pearson&quot;, use = &quot;complete.obs&quot;)
  
  # Clamp between 0 and 1
  r_value &lt;- max(0, min(1, r_value))
  
  # Store the result
  correlation_data &lt;- rbind(correlation_data, data.frame(CX_Sample = cx_name, Correlation = r_value))
}

# Add a single column category for labeling
correlation_data$Comparison &lt;- &quot;DOX&quot;

# Convert to long format for ggplot
heatmap_data_long &lt;- melt(correlation_data, id.vars = c(&quot;CX_Sample&quot;, &quot;Comparison&quot;))

# Ensure the Y-axis follows the correct ordering (CX samples top to bottom)
heatmap_data_long$CX_Sample &lt;- factor(heatmap_data_long$CX_Sample, levels = rev(c(
  &quot;CX_0.1_3&quot;, &quot;CX_0.5_3&quot;,
  &quot;CX_0.1_24&quot;, &quot;CX_0.5_24&quot;,
  &quot;CX_0.1_48&quot;, &quot;CX_0.5_48&quot;
)))

# Create the CX vs. DOX correlation heatmap (0 to 1, white to red)
ggplot(heatmap_data_long, aes(x = Comparison, y = CX_Sample, fill = value)) +
  geom_tile(color = &quot;white&quot;) +
  scale_fill_gradient(low = &quot;white&quot;, high = &quot;red&quot;, limits = c(0, 1)) +
  geom_text(aes(label = round(value, 3)), color = &quot;black&quot;, size = 5, fontface = &quot;bold&quot;) +
  labs(
    x = &quot;&quot;, y = &quot;CX Samples&quot;, fill = &quot;Correlation (r)&quot;,
    title = &quot;Correlation Heatmap: CX-5461 vs. DOX Gene Expression&quot;
  ) +
  theme_minimal() +
  theme(
    axis.text.x = element_text(face = &quot;bold&quot;, size = 14),
    axis.text.y = element_text(face = &quot;bold&quot;, size = 12),
    axis.title.y = element_text(face = &quot;bold&quot;, size = 14),
    plot.title = element_text(face = &quot;bold&quot;, hjust = 0.5, size = 16),
    legend.title = element_text(face = &quot;bold&quot;),
    panel.grid = element_blank()
  )</code></pre>
<p><img src="figure/CX_DOX.Rmd/corr-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-corr-1">
Past versions of corr-1.png
</button>
</p>
<div id="fig-corr-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/7f99e6853f3e7f92acb27e732f083cdd9d7781c3/docs/figure/CX_DOX.Rmd/corr-1.png" target="_blank">7f99e68</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/61329dfa00e657eb011e5a9aef5444048e1d5c46/docs/figure/CX_DOX.Rmd/corr-1.png" target="_blank">61329df</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 26100)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] reshape2_1.4.4  ggplot2_3.5.1   dplyr_1.1.4     workflowr_1.7.1

loaded via a namespace (and not attached):
 [1] sass_0.4.9        generics_0.1.3    stringi_1.8.3     lattice_0.22-5   
 [5] digest_0.6.34     magrittr_2.0.3    evaluate_1.0.3    grid_4.3.0       
 [9] fastmap_1.1.1     plyr_1.8.9        rprojroot_2.0.4   jsonlite_1.8.9   
[13] Matrix_1.6-1.1    processx_3.8.5    whisker_0.4.1     ps_1.8.1         
[17] promises_1.3.0    httr_1.4.7        mgcv_1.9-1        scales_1.3.0     
[21] jquerylib_0.1.4   cli_3.6.1         rlang_1.1.3       munsell_0.5.1    
[25] splines_4.3.0     withr_3.0.2       cachem_1.0.8      yaml_2.3.10      
[29] tools_4.3.0       colorspace_2.1-0  httpuv_1.6.15     vctrs_0.6.5      
[33] R6_2.5.1          lifecycle_1.0.4   git2r_0.35.0      stringr_1.5.1    
[37] fs_1.6.3          pkgconfig_2.0.3   callr_3.7.6       pillar_1.10.1    
[41] bslib_0.8.0       later_1.3.2       gtable_0.3.6      glue_1.7.0       
[45] Rcpp_1.0.12       xfun_0.50         tibble_3.2.1      tidyselect_1.2.1 
[49] rstudioapi_0.17.1 knitr_1.49        farver_2.1.2      htmltools_0.5.8.1
[53] nlme_3.1-166      rmarkdown_2.29    labeling_0.4.3    compiler_4.3.0   
[57] getPass_0.2-4    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
