<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Specific GO terms enrichment in Cormotifs</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Paul_CX_2025</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Specific GO terms enrichment in
Cormotifs</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-07-03
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>CX5461_Project/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree98c546964aac1b29220c891e2a4ad037b8b48ecctargetblank98c5469a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/98c546964aac1b29220c891e2a4ad037b8b48ecc" target="_blank">98c5469</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree98c546964aac1b29220c891e2a4ad037b8b48ecctargetblank98c5469a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/98c546964aac1b29220c891e2a4ad037b8b48ecc" target="_blank">98c5469</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    0.1 box.svg
    Ignored:    Rplot04.svg

Untracked files:
    Untracked:  0.1 density.svg
    Untracked:  0.1.emf
    Untracked:  0.1.svg
    Untracked:  0.5 box.svg
    Untracked:  0.5 density.svg
    Untracked:  0.5.svg
    Untracked:  Additional/
    Untracked:  Autosome factors.svg
    Untracked:  CX_5461_Pattern_Genes_24hr.csv
    Untracked:  CX_5461_Pattern_Genes_3hr.csv
    Untracked:  Cell viability box plot.svg
    Untracked:  DEG GO terms.svg
    Untracked:  DNA damage associated GO terms.svg
    Untracked:  DRC1.svg
    Untracked:  Figure 1.jpeg
    Untracked:  Figure 1.pdf
    Untracked:  Figure_CM_Purity.pdf
    Untracked:  G Quadruplex DEGs.svg
    Untracked:  PC2 Vs PC3 Autosome.svg
    Untracked:  PCA autosome.svg
    Untracked:  Rplot 18.svg
    Untracked:  Rplot.svg
    Untracked:  Rplot01.svg
    Untracked:  Rplot02.svg
    Untracked:  Rplot03.svg
    Untracked:  Rplot05.svg
    Untracked:  Rplot06.svg
    Untracked:  Rplot07.svg
    Untracked:  Rplot08.jpeg
    Untracked:  Rplot08.svg
    Untracked:  Rplot09.svg
    Untracked:  Rplot10.svg
    Untracked:  Rplot11.svg
    Untracked:  Rplot12.svg
    Untracked:  Rplot13.svg
    Untracked:  Rplot14.svg
    Untracked:  Rplot15.svg
    Untracked:  Rplot16.svg
    Untracked:  Rplot17.svg
    Untracked:  Rplot18.svg
    Untracked:  Rplot19.svg
    Untracked:  Rplot20.svg
    Untracked:  Rplot21.svg
    Untracked:  Rplot22.svg
    Untracked:  Rplot23.svg
    Untracked:  Rplot24.svg
    Untracked:  S15.pdf
    Untracked:  TOP2B.bed
    Untracked:  TS HPA (Violin).svg
    Untracked:  TS HPA.svg
    Untracked:  TS_HA.svg
    Untracked:  TS_HV.svg
    Untracked:  Violin HA.svg
    Untracked:  Violin HV (CX vs DOX).svg
    Untracked:  Violin HV.svg
    Untracked:  data/AF.csv
    Untracked:  data/AF_Mapped.csv
    Untracked:  data/AF_genes.csv
    Untracked:  data/Annotated_DOX_Gene_Table.csv
    Untracked:  data/BP/
    Untracked:  data/CAD_genes.csv
    Untracked:  data/Cardiotox.csv
    Untracked:  data/Cardiotox_mapped.csv
    Untracked:  data/Col_DEG_proportion_data.csv
    Untracked:  data/Col_DEGs.csv
    Untracked:  data/Corrmotif_GO/
    Untracked:  data/DOX_Vald.csv
    Untracked:  data/DOX_Vald_Mapped.csv
    Untracked:  data/DOX_alt.csv
    Untracked:  data/Entrez_Cardiotox.csv
    Untracked:  data/Entrez_Cardiotox_Mapped.csv
    Untracked:  data/GWAS.xlsx
    Untracked:  data/GWAS_SNPs.bed
    Untracked:  data/HF.csv
    Untracked:  data/HF_Mapped.csv
    Untracked:  data/HF_genes.csv
    Untracked:  data/Hypertension_genes.csv
    Untracked:  data/MI_genes.csv
    Untracked:  data/P53_Target_mapped.csv
    Untracked:  data/Sample_annotated.csv
    Untracked:  data/Samples.csv
    Untracked:  data/Samples.xlsx
    Untracked:  data/TOP2A.bed
    Untracked:  data/TOP2A_target.csv
    Untracked:  data/TOP2A_target_lit.csv
    Untracked:  data/TOP2A_target_lit_mapped.csv
    Untracked:  data/TOP2A_target_mapped.csv
    Untracked:  data/TOP2B.bed
    Untracked:  data/TOP2B_target.csv
    Untracked:  data/TOP2B_target_heatmap.csv
    Untracked:  data/TOP2B_target_heatmap_mapped.csv
    Untracked:  data/TOP2B_target_mapped.csv
    Untracked:  data/TS.csv
    Untracked:  data/TS_HPA.csv
    Untracked:  data/TS_HPA_mapped.csv
    Untracked:  data/Toptable_CX_0.1_24.csv
    Untracked:  data/Toptable_CX_0.1_3.csv
    Untracked:  data/Toptable_CX_0.1_48.csv
    Untracked:  data/Toptable_CX_0.5_24.csv
    Untracked:  data/Toptable_CX_0.5_3.csv
    Untracked:  data/Toptable_CX_0.5_48.csv
    Untracked:  data/Toptable_DOX_0.1_24.csv
    Untracked:  data/Toptable_DOX_0.1_3.csv
    Untracked:  data/Toptable_DOX_0.1_48.csv
    Untracked:  data/Toptable_DOX_0.5_24.csv
    Untracked:  data/Toptable_DOX_0.5_3.csv
    Untracked:  data/Toptable_DOX_0.5_48.csv
    Untracked:  data/count.tsv
    Untracked:  data/heatmap.csv
    Untracked:  data/ts_data_mapped
    Untracked:  results/
    Untracked:  run_bedtools.bat

Unstaged changes:
    Deleted:    analysis/Actox.Rmd
    Modified:   analysis/_site.yml
    Modified:   data/DEGs/Toptable_DOX_0.5_3.csv
    Modified:   data/DOX_0.5_48 (Combined).csv
    Modified:   data/Total_number_of_Mapped_reads_by_Individuals.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/DDR_Gquad_Corr.Rmd</code>) and
HTML (<code>docs/DDR_Gquad_Corr.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/75100b050d64e89fc8400cff10fa0b7864b6f3a9/docs/DDR_Gquad_Corr.html" target="_blank">75100b0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-06-03
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/2fdef0f6dbaf512c861629337bbbcd30d63eb25f/docs/DDR_Gquad_Corr.html" target="_blank">2fdef0f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-06-03
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/fce174d8ae14c51fa490eedb0a9f5deac9c98f76/analysis/DDR_Gquad_Corr.Rmd" target="_blank">fce174d</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-06-03
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/fce174d8ae14c51fa490eedb0a9f5deac9c98f76/docs/DDR_Gquad_Corr.html" target="_blank">fce174d</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-06-03
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/af36d284d81aec64e765ae5fd47dcb8e935d6f9b/analysis/DDR_Gquad_Corr.Rmd" target="_blank">af36d28</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-06-01
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/af36d284d81aec64e765ae5fd47dcb8e935d6f9b/docs/DDR_Gquad_Corr.html" target="_blank">af36d28</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-06-01
</td>
<td>
Commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="dna-damage-response-go-terms" class="section level2">
<h2><strong>📌 DNA damage response GO terms</strong></h2>
<pre class="r"><code>library(tidyverse)
library(data.table)
library(ComplexHeatmap)
library(circlize)
library(grid)

### 📁 Define CorMotif GO Enrichment Files
go_files &lt;- list(
  &quot;Non response (0.1)&quot;                         = &quot;data/BP/CorMotif_Terms/GO_BP_Non_response_(0.1).csv&quot;,
  &quot;CX-DOX mid-late response (0.1)&quot;             = &quot;data/BP/CorMotif_Terms/GO_BP_CX-DOX_mid-late_response_(0.1).csv&quot;,
  &quot;DOX only mid-late (0.1)&quot;                    = &quot;data/BP/CorMotif_Terms/GO_BP_DOX_only_mid-late_(0.1).csv&quot;,
  &quot;Non response (0.5)&quot;                         = &quot;data/BP/CorMotif_Terms/GO_BP_Non_response_(0.5).csv&quot;,
  &quot;DOX specific response (0.5)&quot;                = &quot;data/BP/CorMotif_Terms/GO_BP_DOX_specific_response_(0.5).csv&quot;,
  &quot;DOX only mid-late response (0.5)&quot;           = &quot;data/BP/CorMotif_Terms/GO_BP_DOX_only_mid-late_response_(0.5).csv&quot;,
  &quot;CX total + DOX early response (0.5)&quot;        = &quot;data/BP/CorMotif_Terms/GO_BP_CX_total_+_DOX_early_response_(0.5).csv&quot;,
  &quot;DOX early + CX-DOX mid-late response (0.5)&quot; = &quot;data/BP/CorMotif_Terms/GO_BP_DOX_early_+_CX-DOX_mid-late_response_(0.5).csv&quot;
)

### 🧬 Define GO parent terms and map their children
parent_terms &lt;- list(
  &quot;GO:0006974&quot; = &quot;DNA damage response&quot;,
  &quot;GO:0141112&quot; = &quot;broken chromosome clustering&quot;,
  &quot;GO:0006281&quot; = &quot;DNA repair&quot;,
  &quot;GO:0140861&quot; = &quot;DNA repair-dependent chromatin remodeling&quot;,
  &quot;GO:0008630&quot; = &quot;intrinsic apoptotic signaling pathway in response to DNA damage&quot;,
  &quot;GO:0042770&quot; = &quot;signal transduction in response to DNA damage&quot;,
  &quot;GO:0009432&quot; = &quot;SOS response&quot;,
  &quot;GO:0043247&quot; = &quot;telomere maintenance in response to DNA damage&quot;
)

child_map &lt;- list(
  &quot;GO:0006281&quot; = c(&quot;GO:0006284&quot;, &quot;GO:0006307&quot;, &quot;GO:0006302&quot;, &quot;GO:0006298&quot;, &quot;GO:0043504&quot;,
                   &quot;GO:0006289&quot;, &quot;GO:0006301&quot;, &quot;GO:0006290&quot;, &quot;GO:0000725&quot;, &quot;GO:0000012&quot;),
  &quot;GO:0008630&quot; = c(&quot;GO:0042771&quot;, &quot;GO:1902230&quot;, &quot;GO:1902231&quot;, &quot;GO:1902229&quot;),
  &quot;GO:0042770&quot; = c(&quot;GO:0000077&quot;, &quot;GO:0030330&quot;, &quot;GO:0042772&quot;, &quot;GO:0044773&quot;,
                   &quot;GO:2000002&quot;, &quot;GO:2000003&quot;, &quot;GO:2000001&quot;),
  &quot;GO:0043247&quot; = c(&quot;GO:1904506&quot;, &quot;GO:1904507&quot;, &quot;GO:0031848&quot;, &quot;GO:1904505&quot;, &quot;GO:0097698&quot;)
)

ddr_descriptions &lt;- unname(unlist(parent_terms))

### 🔁 Step 1: Extract best p-values from GO enrichment results
go_matrix_df &lt;- map_dfr(names(go_files), function(cond) {
  file &lt;- go_files[[cond]]
  df &lt;- tryCatch(fread(file), error = function(e) return(data.table()))
  
  if (nrow(df) == 0 || !all(c(&quot;ID&quot;, &quot;Description&quot;, &quot;pvalue&quot;, &quot;p.adjust&quot;) %in% colnames(df))) {
    message(&quot;⚠️ Skipping or padding malformed file: &quot;, cond)
    return(tibble(Description = ddr_descriptions, pvalue = NA, p.adjust = NA, log10p = NA, Condition = cond))
  }
  
  df &lt;- as_tibble(df) %&gt;% dplyr::select(ID, Description, pvalue, p.adjust)
  
  results &lt;- lapply(names(parent_terms), function(parent_id) {
    all_ids &lt;- c(parent_id, child_map[[parent_id]])
    df_sub &lt;- df %&gt;% filter(ID %in% all_ids)
    
    if (nrow(df_sub) == 0) {
      tibble(Description = parent_terms[[parent_id]], pvalue = NA, p.adjust = NA, log10p = NA, Condition = cond)
    } else {
      best_row &lt;- df_sub %&gt;% slice_min(pvalue, n = 1)
      tibble(Description = parent_terms[[parent_id]],
             pvalue = best_row$pvalue,
             p.adjust = best_row$p.adjust,
             log10p = -log10(best_row$pvalue),
             Condition = cond)
    }
  })
  
  bind_rows(results)
})

### 🧱 Step 2: Construct log10(p) and raw p-value matrices
heatmap_data &lt;- go_matrix_df %&gt;%
  dplyr::select(Description, Condition, log10p) %&gt;%
  tidyr::pivot_wider(names_from = Condition, values_from = log10p) %&gt;%
  tibble::column_to_rownames(&quot;Description&quot;) %&gt;%
  as.matrix()

pval_matrix &lt;- go_matrix_df %&gt;%
  dplyr::select(Description, Condition, pvalue) %&gt;%
  tidyr::pivot_wider(names_from = Condition, values_from = pvalue) %&gt;%
  tibble::column_to_rownames(&quot;Description&quot;) %&gt;%
  as.matrix()

p_adj_matrix &lt;- go_matrix_df %&gt;%
  dplyr::select(Description, Condition, p.adjust) %&gt;%
  tidyr::pivot_wider(names_from = Condition, values_from = p.adjust) %&gt;%
  tibble::column_to_rownames(&quot;Description&quot;) %&gt;%
  as.matrix()

### ✅ Step 3: Pad missing columns
all_conditions &lt;- names(go_files)
missing_cols &lt;- setdiff(all_conditions, colnames(heatmap_data))
if (length(missing_cols) &gt; 0) {
  for (cond in missing_cols) {
    heatmap_data[, cond] &lt;- NA
    pval_matrix[, cond] &lt;- NA
  }
  heatmap_data &lt;- heatmap_data[, all_conditions]
  pval_matrix &lt;- pval_matrix[, all_conditions]
}

### 🎨 Step 4: Define color palette
breaks &lt;- seq(0, 20, by = 2.5)
palette &lt;- colorRampPalette(c(&quot;white&quot;, &quot;#fde0dd&quot;, &quot;#fa9fb5&quot;, &quot;#f768a1&quot;, &quot;#c51b8a&quot;, &quot;#7a0177&quot;, &quot;#49006a&quot;))(length(breaks))
col_fun &lt;- colorRamp2(breaks, palette)

### 🔥 Step 5: Plot heatmap with stars for p.adjust &lt; 0.05
ht &lt;- Heatmap(
  heatmap_data,
  name = &quot;-log10(p)&quot;,
  col = col_fun,
  na_col = &quot;white&quot;,
  rect_gp = gpar(col = &quot;black&quot;, lwd = 0.5),
  cluster_rows = FALSE,
  cluster_columns = FALSE,
  row_names_gp = gpar(fontsize = 9),
  column_names_gp = gpar(fontsize = 9),
  column_names_rot = 45,
  row_names_max_width = max_text_width(rownames(heatmap_data), gp = gpar(fontsize = 9)),
  cell_fun = function(j, i, x, y, width, height, fill) {
    adj_p &lt;- p_adj_matrix[i, j]  # Use p.adjust matrix for significance check
    if (!is.na(adj_p) &amp;&amp; adj_p &lt; 0.05) {
      grid.text(&quot;*&quot;, x, y, gp = gpar(fontsize = 12))
    }
  },
  heatmap_legend_param = list(
    title = &quot;-log10(p value)&quot;,
    at = breaks,
    labels = as.character(breaks),
    legend_width = unit(5, &quot;cm&quot;),
    direction = &quot;horizontal&quot;,
    title_gp = gpar(fontsize = 10, fontface = &quot;bold&quot;),
    labels_gp = gpar(fontsize = 9)
  )
)

### 🖼 Final Draw
draw(ht, heatmap_legend_side = &quot;top&quot;)</code></pre>
<p><img src="figure/DDR_Gquad_Corr.Rmd/load_libraries-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-load_libraries-1">
Past versions of load_libraries-1.png
</button>
</p>
<div id="fig-load_libraries-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/fce174d8ae14c51fa490eedb0a9f5deac9c98f76/docs/figure/DDR_Gquad_Corr.Rmd/load_libraries-1.png" target="_blank">fce174d</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/af36d284d81aec64e765ae5fd47dcb8e935d6f9b/docs/figure/DDR_Gquad_Corr.Rmd/load_libraries-1.png" target="_blank">af36d28</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-06-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ribosome-biogenesis-go-terms" class="section level2">
<h2><strong>📌 Ribosome biogenesis GO terms</strong></h2>
<pre class="r"><code>library(tidyverse)
library(data.table)
library(ComplexHeatmap)
library(circlize)
library(grid)

### 📁 Define CorMotif GO Enrichment Files
go_files &lt;- list(
  &quot;Non response (0.1)&quot;                         = &quot;data/BP/CorMotif_Terms/GO_BP_Non_response_(0.1).csv&quot;,
  &quot;CX-DOX mid-late response (0.1)&quot;             = &quot;data/BP/CorMotif_Terms/GO_BP_CX-DOX_mid-late_response_(0.1).csv&quot;,
  &quot;DOX only mid-late (0.1)&quot;                    = &quot;data/BP/CorMotif_Terms/GO_BP_DOX_only_mid-late_(0.1).csv&quot;,
  &quot;Non response (0.5)&quot;                         = &quot;data/BP/CorMotif_Terms/GO_BP_Non_response_(0.5).csv&quot;,
  &quot;DOX specific response (0.5)&quot;                = &quot;data/BP/CorMotif_Terms/GO_BP_DOX_specific_response_(0.5).csv&quot;,
  &quot;DOX only mid-late response (0.5)&quot;           = &quot;data/BP/CorMotif_Terms/GO_BP_DOX_only_mid-late_response_(0.5).csv&quot;,
  &quot;CX total + DOX early response (0.5)&quot;        = &quot;data/BP/CorMotif_Terms/GO_BP_CX_total_+_DOX_early_response_(0.5).csv&quot;,
  &quot;DOX early + CX-DOX mid-late response (0.5)&quot; = &quot;data/BP/CorMotif_Terms/GO_BP_DOX_early_+_CX-DOX_mid-late_response_(0.5).csv&quot;
)

### 🧬 Ribosome Biogenesis GO Terms
parent_terms &lt;- list(
  &quot;GO:0042254&quot; = &quot;ribosome biogenesis&quot;,
  &quot;GO:0090071&quot; = &quot;negative regulation of ribosome biogenesis&quot;,
  &quot;GO:0042273&quot; = &quot;ribosomal large subunit biogenesis&quot;,
  &quot;GO:0042274&quot; = &quot;ribosomal small subunit biogenesis&quot;,
  &quot;GO:0000054&quot; = &quot;ribosomal subunit export from nucleus&quot;,
  &quot;GO:0042255&quot; = &quot;ribosome assembly&quot;,
  &quot;GO:0006364&quot; = &quot;rRNA processing&quot;
)

child_map &lt;- list(
  &quot;GO:0090071&quot; = c(&quot;GO:2000201&quot;, &quot;GO:2000233&quot;),
  &quot;GO:0042273&quot; = c(&quot;GO:0000470&quot;, &quot;GO:0000027&quot;),
  &quot;GO:0042274&quot; = c(&quot;GO:0030490&quot;, &quot;GO:0000028&quot;, &quot;GO:0140638&quot;),
  &quot;GO:0000054&quot; = c(&quot;GO:2000201&quot;, &quot;GO:2000202&quot;, &quot;GO:2000200&quot;, &quot;GO:0000055&quot;, &quot;GO:0000056&quot;),
  &quot;GO:0042255&quot; = c(&quot;GO:0042256&quot;, &quot;GO:0061668&quot;, &quot;GO:0000027&quot;, &quot;GO:0000028&quot;),
  &quot;GO:0006364&quot; = c(
    &quot;GO:1901259&quot;, &quot;GO:0000450&quot;, &quot;GO:0000483&quot;, &quot;GO:0002103&quot;, &quot;GO:0000479&quot;, &quot;GO:0000449&quot;,
    &quot;GO:0000475&quot;, &quot;GO:0000476&quot;, &quot;GO:0000460&quot;, &quot;GO:0000481&quot;, &quot;GO:0000470&quot;, &quot;GO:0030490&quot;,
    &quot;GO:2000233&quot;, &quot;GO:2000234&quot;, &quot;GO:2000232&quot;, &quot;GO:0031125&quot;, &quot;GO:0000967&quot;, &quot;GO:0000154&quot;, &quot;GO:1990417&quot;
  )
)

rbio_descriptions &lt;- unname(unlist(parent_terms))

### 🔁 Step 1: Extract best p-values from GO enrichment results
go_matrix_df &lt;- purrr::map_dfr(names(go_files), function(cond) {
  file &lt;- go_files[[cond]]
  df &lt;- tryCatch(fread(file), error = function(e) return(data.table()))

  if (nrow(df) == 0 || !all(c(&quot;ID&quot;, &quot;Description&quot;, &quot;pvalue&quot;, &quot;p.adjust&quot;) %in% colnames(df))) {
    message(&quot;⚠️ Skipping or padding malformed file: &quot;, cond)
    return(tibble(Description = rbio_descriptions, pvalue = NA, p.adjust = NA, log10p = NA, Condition = cond))
  }

  df &lt;- as_tibble(df) %&gt;% dplyr::select(ID, Description, pvalue, p.adjust)

  results &lt;- lapply(names(parent_terms), function(parent_id) {
    all_ids &lt;- c(parent_id, child_map[[parent_id]])
    df_sub &lt;- df %&gt;% dplyr::filter(ID %in% all_ids)

    if (nrow(df_sub) == 0) {
      tibble(Description = parent_terms[[parent_id]], pvalue = NA, p.adjust = NA, log10p = NA, Condition = cond)
    } else {
      best_row &lt;- df_sub %&gt;% dplyr::slice_min(pvalue, n = 1)
      tibble(
        Description = parent_terms[[parent_id]],
        pvalue = best_row$pvalue,
        p.adjust = best_row$p.adjust,
        log10p = -log10(best_row$pvalue),
        Condition = cond
      )
    }
  })

  bind_rows(results)
})

### 🧱 Step 2: Construct log10(p) and raw p-value matrices
heatmap_data &lt;- go_matrix_df %&gt;%
  dplyr::select(Description, Condition, log10p) %&gt;%
  tidyr::pivot_wider(names_from = Condition, values_from = log10p) %&gt;%
  tibble::column_to_rownames(&quot;Description&quot;) %&gt;%
  as.matrix()

pval_matrix &lt;- go_matrix_df %&gt;%
  dplyr::select(Description, Condition, pvalue) %&gt;%
  tidyr::pivot_wider(names_from = Condition, values_from = pvalue) %&gt;%
  tibble::column_to_rownames(&quot;Description&quot;) %&gt;%
  as.matrix()

p_adj_matrix &lt;- go_matrix_df %&gt;%
  dplyr::select(Description, Condition, p.adjust) %&gt;%
  tidyr::pivot_wider(names_from = Condition, values_from = p.adjust) %&gt;%
  tibble::column_to_rownames(&quot;Description&quot;) %&gt;%
  as.matrix()

### ✅ Step 3: Pad missing columns
all_conditions &lt;- names(go_files)
missing_cols &lt;- setdiff(all_conditions, colnames(heatmap_data))
if (length(missing_cols) &gt; 0) {
  for (cond in missing_cols) {
    heatmap_data[, cond] &lt;- NA
    pval_matrix[, cond] &lt;- NA
  }
  heatmap_data &lt;- heatmap_data[, all_conditions]
  pval_matrix &lt;- pval_matrix[, all_conditions]
}

### 🎨 Step 4: Define color palette
breaks &lt;- seq(0, 20, by = 2.5)
palette &lt;- colorRampPalette(c(&quot;white&quot;, &quot;#fde0dd&quot;, &quot;#fa9fb5&quot;, &quot;#f768a1&quot;, &quot;#c51b8a&quot;, &quot;#7a0177&quot;, &quot;#49006a&quot;))(length(breaks))
col_fun &lt;- colorRamp2(breaks, palette)

### 🔥 Step 5: Plot heatmap with stars for p.adjust &lt; 0.05
ht &lt;- Heatmap(
  heatmap_data,
  name = &quot;-log10(p)&quot;,
  col = col_fun,
  na_col = &quot;white&quot;,
  rect_gp = gpar(col = &quot;black&quot;, lwd = 0.5),
  cluster_rows = FALSE,
  cluster_columns = FALSE,
  row_names_gp = gpar(fontsize = 9),
  column_names_gp = gpar(fontsize = 9),
  column_names_rot = 45,
  row_names_max_width = max_text_width(rownames(heatmap_data), gp = gpar(fontsize = 9)),
  cell_fun = function(j, i, x, y, width, height, fill) {
    adj_p &lt;- p_adj_matrix[i, j]  # Use p.adjust matrix for significance check
    if (!is.na(adj_p) &amp;&amp; adj_p &lt; 0.05) {
      grid.text(&quot;*&quot;, x, y, gp = gpar(fontsize = 12))
    }
  },
  heatmap_legend_param = list(
    title = &quot;-log10(p value)&quot;,
    at = breaks,
    labels = as.character(breaks),
    legend_width = unit(5, &quot;cm&quot;),
    direction = &quot;horizontal&quot;,
    title_gp = gpar(fontsize = 10, fontface = &quot;bold&quot;),
    labels_gp = gpar(fontsize = 9)
  )
)

### 🖼 Final Draw
draw(ht, heatmap_legend_side = &quot;top&quot;)</code></pre>
<p><img src="figure/DDR_Gquad_Corr.Rmd/RB-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RB-1">
Past versions of RB-1.png
</button>
</p>
<div id="fig-RB-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/fce174d8ae14c51fa490eedb0a9f5deac9c98f76/docs/figure/DDR_Gquad_Corr.Rmd/RB-1.png" target="_blank">fce174d</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/af36d284d81aec64e765ae5fd47dcb8e935d6f9b/docs/figure/DDR_Gquad_Corr.Rmd/RB-1.png" target="_blank">af36d28</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-06-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="g-quadruplex-go-terms" class="section level2">
<h2><strong>📌 G-Quadruplex GO terms</strong></h2>
<pre class="r"><code>### 📦 Load Required Libraries
# 📦 Load Required Libraries
library(tidyverse)
library(data.table)
library(ComplexHeatmap)
library(circlize)
library(grid)

# 📁 Define GO Enrichment Files
go_files &lt;- list(
  &quot;Non response (0.1)&quot;                         = &quot;data/BP/CorMotif_Terms_Combined/GO_All_Non_response_(0.1).csv&quot;,
  &quot;CX-DOX mid-late response (0.1)&quot;             = &quot;data/BP/CorMotif_Terms_Combined/GO_All_CX-DOX_mid-late_response_(0.1).csv&quot;,
  &quot;DOX only mid-late (0.1)&quot;                    = &quot;data/BP/CorMotif_Terms_Combined/GO_All_DOX_only_mid-late_(0.1).csv&quot;,
  &quot;Non response (0.5)&quot;                         = &quot;data/BP/CorMotif_Terms_Combined/GO_All_Non_response_(0.5).csv&quot;,
  &quot;DOX specific response (0.5)&quot;                = &quot;data/BP/CorMotif_Terms_Combined/GO_All_DOX_specific_response_(0.5).csv&quot;,
  &quot;DOX only mid-late response (0.5)&quot;           = &quot;data/BP/CorMotif_Terms_Combined/GO_All_DOX_only_mid-late_response_(0.5).csv&quot;,
  &quot;CX total + DOX early response (0.5)&quot;        = &quot;data/BP/CorMotif_Terms_Combined/GO_All_CX_total_+_DOX_early_response_(0.5).csv&quot;,
  &quot;DOX early + CX-DOX mid-late response (0.5)&quot; = &quot;data/BP/CorMotif_Terms_Combined/GO_All_DOX_early_+_CX-DOX_mid-late_response_(0.5).csv&quot;
)

# 🧬 G4-Related GO Terms
g4_terms &lt;- list(
  &quot;GO:0051880&quot; = &quot;G-quadruplex DNA binding&quot;,
  &quot;GO:0002151&quot; = &quot;G-quadruplex RNA binding&quot;,
  &quot;GO:0071919&quot; = &quot;G-quadruplex DNA formation&quot;,
  &quot;GO:1905493&quot; = &quot;Regulation of G-quadruplex DNA binding&quot;,
  &quot;GO:0160225&quot; = &quot;G-quadruplex unwinding activity&quot;,
  &quot;GO:0061849&quot; = &quot;Telomeric G-quadruplex DNA binding&quot;
)

g4_descriptions &lt;- unname(unlist(g4_terms))

# 🔁 Step 1: Extract best p-values and p.adjust values
go_matrix_df &lt;- purrr::map_dfr(names(go_files), function(cond) {
  file &lt;- go_files[[cond]]
  df &lt;- tryCatch(fread(file), error = function(e) return(data.table()))

  if (nrow(df) == 0 || !all(c(&quot;ID&quot;, &quot;Description&quot;, &quot;pvalue&quot;, &quot;p.adjust&quot;) %in% colnames(df))) {
    message(&quot;⚠️ Skipping or padding malformed file: &quot;, cond)
    return(tibble(Description = g4_descriptions, pvalue = NA, padj = NA, log10p = NA, Condition = cond))
  }

  df &lt;- as_tibble(df) %&gt;% dplyr::select(ID, Description, pvalue, p.adjust)

  results &lt;- lapply(names(g4_terms), function(go_id) {
    df_sub &lt;- df %&gt;% dplyr::filter(ID == go_id)
    if (nrow(df_sub) == 0) {
      tibble(Description = g4_terms[[go_id]], pvalue = NA, padj = NA, log10p = NA, Condition = cond)
    } else {
      best_row &lt;- df_sub %&gt;% dplyr::slice_min(pvalue, n = 1)
      tibble(
        Description = g4_terms[[go_id]],
        pvalue = best_row$pvalue,
        padj = best_row$p.adjust,
        log10p = -log10(best_row$pvalue),
        Condition = cond
      )
    }
  })

  bind_rows(results)
})

# 🧱 Step 2: Create heatmap and p-value matrices
heatmap_data &lt;- go_matrix_df %&gt;%
  dplyr::select(Description, Condition, log10p) %&gt;%
  tidyr::pivot_wider(names_from = Condition, values_from = log10p) %&gt;%
  tibble::column_to_rownames(&quot;Description&quot;) %&gt;%
  as.matrix()

pval_matrix &lt;- go_matrix_df %&gt;%
  dplyr::select(Description, Condition, pvalue) %&gt;%
  tidyr::pivot_wider(names_from = Condition, values_from = pvalue) %&gt;%
  tibble::column_to_rownames(&quot;Description&quot;) %&gt;%
  as.matrix()

padj_matrix &lt;- go_matrix_df %&gt;%
  dplyr::select(Description, Condition, padj) %&gt;%
  tidyr::pivot_wider(names_from = Condition, values_from = padj) %&gt;%
  tibble::column_to_rownames(&quot;Description&quot;) %&gt;%
  as.matrix()

# ✅ Step 3: Pad missing columns
all_conditions &lt;- names(go_files)
missing_cols &lt;- setdiff(all_conditions, colnames(heatmap_data))
if (length(missing_cols) &gt; 0) {
  for (cond in missing_cols) {
    heatmap_data[, cond] &lt;- NA
    pval_matrix[, cond] &lt;- NA
    padj_matrix[, cond] &lt;- NA
  }
  heatmap_data &lt;- heatmap_data[, all_conditions]
  pval_matrix &lt;- pval_matrix[, all_conditions]
  padj_matrix &lt;- padj_matrix[, all_conditions]
}

# 🎨 Step 4: Color palette
breaks &lt;- seq(0, 20, by = 2.5)
palette &lt;- colorRampPalette(c(&quot;white&quot;, &quot;#fde0dd&quot;, &quot;#fa9fb5&quot;, &quot;#f768a1&quot;, &quot;#c51b8a&quot;, &quot;#7a0177&quot;, &quot;#49006a&quot;))(length(breaks))
col_fun &lt;- colorRamp2(breaks, palette)

# 🔥 Step 5: Plot heatmap
ht &lt;- Heatmap(
  heatmap_data,
  name = &quot;-log10(p)&quot;,
  col = col_fun,
  na_col = &quot;white&quot;,
  rect_gp = gpar(col = &quot;black&quot;, lwd = 0.5),
  cluster_rows = FALSE,
  cluster_columns = FALSE,
  row_names_gp = gpar(fontsize = 9),
  column_names_gp = gpar(fontsize = 9),
  column_names_rot = 45,
  row_names_max_width = max_text_width(rownames(heatmap_data), gp = gpar(fontsize = 9)),
  cell_fun = function(j, i, x, y, width, height, fill) {
    raw_padj &lt;- padj_matrix[i, j]
    if (!is.na(raw_padj) &amp;&amp; raw_padj &lt; 0.05) {
      grid.text(&quot;*&quot;, x, y, gp = gpar(fontsize = 12))
    }
  },
  heatmap_legend_param = list(
    title = &quot;-log10(p value)&quot;,
    at = breaks,
    labels = as.character(breaks),
    legend_width = unit(5, &quot;cm&quot;),
    direction = &quot;horizontal&quot;,
    title_gp = gpar(fontsize = 10, fontface = &quot;bold&quot;),
    labels_gp = gpar(fontsize = 9)
  )
)

# 🖼 Final Draw
draw(ht, heatmap_legend_side = &quot;top&quot;)</code></pre>
<p><img src="figure/DDR_Gquad_Corr.Rmd/GQ-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GQ-1">
Past versions of GQ-1.png
</button>
</p>
<div id="fig-GQ-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/fce174d8ae14c51fa490eedb0a9f5deac9c98f76/docs/figure/DDR_Gquad_Corr.Rmd/GQ-1.png" target="_blank">fce174d</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/af36d284d81aec64e765ae5fd47dcb8e935d6f9b/docs/figure/DDR_Gquad_Corr.Rmd/GQ-1.png" target="_blank">af36d28</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-06-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="transcription-by-rna-polymerase-i" class="section level2">
<h2><strong>📌 Transcription by RNA polymerase I</strong></h2>
<pre class="r"><code>### 📦 Load Required Libraries
library(tidyverse)
library(data.table)
library(ComplexHeatmap)
library(circlize)
library(grid)

### 📁 Define GO Enrichment Files
go_files &lt;- list(
  &quot;Non response (0.1)&quot;                         = &quot;data/BP/CorMotif_Terms_Combined/GO_All_Non_response_(0.1).csv&quot;,
  &quot;CX-DOX mid-late response (0.1)&quot;             = &quot;data/BP/CorMotif_Terms_Combined/GO_All_CX-DOX_mid-late_response_(0.1).csv&quot;,
  &quot;DOX only mid-late (0.1)&quot;                    = &quot;data/BP/CorMotif_Terms_Combined/GO_All_DOX_only_mid-late_(0.1).csv&quot;,
  &quot;Non response (0.5)&quot;                         = &quot;data/BP/CorMotif_Terms_Combined/GO_All_Non_response_(0.5).csv&quot;,
  &quot;DOX specific response (0.5)&quot;                = &quot;data/BP/CorMotif_Terms_Combined/GO_All_DOX_specific_response_(0.5).csv&quot;,
  &quot;DOX only mid-late response (0.5)&quot;           = &quot;data/BP/CorMotif_Terms_Combined/GO_All_DOX_only_mid-late_response_(0.5).csv&quot;,
  &quot;CX total + DOX early response (0.5)&quot;        = &quot;data/BP/CorMotif_Terms_Combined/GO_All_CX_total_+_DOX_early_response_(0.5).csv&quot;,
  &quot;DOX early + CX-DOX mid-late response (0.5)&quot; = &quot;data/BP/CorMotif_Terms_Combined/GO_All_DOX_early_+_CX-DOX_mid-late_response_(0.5).csv&quot;
)

### 🔧 Parent GO Terms and their Children
parent_terms &lt;- list(
  &quot;GO:0006360&quot; = &quot;transcription by RNA polymerase I&quot;,
  &quot;GO:0016479&quot; = &quot;negative regulation of transcription by RNA polymerase I&quot;,
  &quot;GO:0042790&quot; = &quot;nucleolar large rRNA transcription by RNA polymerase I&quot;,
  &quot;GO:0045943&quot; = &quot;positive regulation of transcription by RNA polymerase I&quot;,
  &quot;GO:0006356&quot; = &quot;regulation of transcription by RNA polymerase I&quot;,
  &quot;GO:0006363&quot; = &quot;termination of RNA polymerase I transcription&quot;,
  &quot;GO:0006361&quot; = &quot;transcription initiation at RNA polymerase I promoter&quot;,
  &quot;GO:0005736&quot; = &quot;RNA polymerase I complex&quot;
)

rnapolI_descriptions &lt;- unname(unlist(parent_terms))

### 🔁 Step 1: Extract best p-values
go_matrix_df &lt;- purrr::map_dfr(names(go_files), function(cond) {
  file &lt;- go_files[[cond]]
  df &lt;- tryCatch(fread(file), error = function(e) return(data.table()))

  if (nrow(df) == 0 || !all(c(&quot;ID&quot;, &quot;Description&quot;, &quot;pvalue&quot;, &quot;p.adjust&quot;) %in% colnames(df))) {
    message(&quot;⚠️ Skipping or padding malformed file: &quot;, cond)
    return(tibble(Description = rnapolI_descriptions, pvalue = NA, padj = NA, log10p = NA, Condition = cond))
  }

  df &lt;- as_tibble(df) %&gt;% dplyr::select(ID, Description, pvalue, p.adjust)

  results &lt;- lapply(names(parent_terms), function(go_id) {
    df_sub &lt;- df %&gt;% dplyr::filter(ID == go_id)
    if (nrow(df_sub) == 0) {
      tibble(Description = parent_terms[[go_id]], pvalue = NA, padj = NA, log10p = NA, Condition = cond)
    } else {
      best_row &lt;- df_sub %&gt;% dplyr::slice_min(pvalue, n = 1)
      tibble(
        Description = parent_terms[[go_id]],
        pvalue = best_row$pvalue,
        padj = best_row$p.adjust,
        log10p = -log10(best_row$pvalue),
        Condition = cond
      )
    }
  })

  bind_rows(results)
})

### 🧱 Step 2: Create heatmap and p-value matrices
heatmap_data &lt;- go_matrix_df %&gt;%
  dplyr::select(Description, Condition, log10p) %&gt;%
  tidyr::pivot_wider(names_from = Condition, values_from = log10p) %&gt;%
  tibble::column_to_rownames(&quot;Description&quot;) %&gt;%
  as.matrix()

padj_matrix &lt;- go_matrix_df %&gt;%
  dplyr::select(Description, Condition, padj) %&gt;%
  tidyr::pivot_wider(names_from = Condition, values_from = padj) %&gt;%
  tibble::column_to_rownames(&quot;Description&quot;) %&gt;%
  as.matrix()

### ✅ Step 3: Pad missing columns
all_conditions &lt;- names(go_files)
missing_cols &lt;- setdiff(all_conditions, colnames(heatmap_data))
if (length(missing_cols) &gt; 0) {
  for (cond in missing_cols) {
    heatmap_data[, cond] &lt;- NA
    padj_matrix[, cond] &lt;- NA
  }
  heatmap_data &lt;- heatmap_data[, all_conditions]
  padj_matrix &lt;- padj_matrix[, all_conditions]
}

### 🎨 Step 4: Color palette
breaks &lt;- seq(0, 20, by = 2.5)
palette &lt;- colorRampPalette(c(&quot;white&quot;, &quot;#fde0dd&quot;, &quot;#fa9fb5&quot;, &quot;#f768a1&quot;, &quot;#c51b8a&quot;, &quot;#7a0177&quot;, &quot;#49006a&quot;))(length(breaks))
col_fun &lt;- colorRamp2(breaks, palette)

### 🔥 Step 5: Plot heatmap
ht &lt;- Heatmap(
  heatmap_data,
  name = &quot;-log10(p)&quot;,
  col = col_fun,
  na_col = &quot;white&quot;,
  rect_gp = gpar(col = &quot;black&quot;, lwd = 0.5),
  cluster_rows = FALSE,
  cluster_columns = FALSE,
  row_names_gp = gpar(fontsize = 9),
  column_names_gp = gpar(fontsize = 9),
  column_names_rot = 45,
  row_names_max_width = max_text_width(rownames(heatmap_data), gp = gpar(fontsize = 9)),
  cell_fun = function(j, i, x, y, width, height, fill) {
    adj_p &lt;- padj_matrix[i, j]
    if (!is.na(adj_p) &amp;&amp; adj_p &lt; 0.05) {
      grid.text(&quot;*&quot;, x, y, gp = gpar(fontsize = 12))
    }
  },
  heatmap_legend_param = list(
    title = &quot;-log10(p value)&quot;,
    at = breaks,
    labels = as.character(breaks),
    legend_width = unit(5, &quot;cm&quot;),
    direction = &quot;horizontal&quot;,
    title_gp = gpar(fontsize = 10, fontface = &quot;bold&quot;),
    labels_gp = gpar(fontsize = 9)
  )
)

### 🖼 Final Draw
draw(ht, heatmap_legend_side = &quot;top&quot;)</code></pre>
<p><img src="figure/DDR_Gquad_Corr.Rmd/RNA_POl-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RNA_POl-1">
Past versions of RNA_POl-1.png
</button>
</p>
<div id="fig-RNA_POl-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/af36d284d81aec64e765ae5fd47dcb8e935d6f9b/docs/figure/DDR_Gquad_Corr.Rmd/RNA_POl-1.png" target="_blank">af36d28</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-06-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 26100)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] circlize_0.4.16       ComplexHeatmap_2.18.0 data.table_1.17.0    
 [4] lubridate_1.9.4       forcats_1.0.0         stringr_1.5.1        
 [7] dplyr_1.1.4           purrr_1.0.4           readr_2.1.5          
[10] tidyr_1.3.1           tibble_3.2.1          ggplot2_3.5.2        
[13] tidyverse_2.0.0       workflowr_1.7.1      

loaded via a namespace (and not attached):
 [1] shape_1.4.6.1       gtable_0.3.6        rjson_0.2.23       
 [4] xfun_0.52           bslib_0.9.0         GlobalOptions_0.1.2
 [7] processx_3.8.6      callr_3.7.6         tzdb_0.5.0         
[10] Cairo_1.6-2         vctrs_0.6.5         tools_4.3.0        
[13] ps_1.8.1            generics_0.1.3      stats4_4.3.0       
[16] parallel_4.3.0      cluster_2.1.8.1     pkgconfig_2.0.3    
[19] RColorBrewer_1.1-3  S4Vectors_0.40.2    lifecycle_1.0.4    
[22] compiler_4.3.0      git2r_0.36.2        munsell_0.5.1      
[25] getPass_0.2-4       codetools_0.2-20    clue_0.3-66        
[28] httpuv_1.6.15       htmltools_0.5.8.1   sass_0.4.10        
[31] yaml_2.3.10         later_1.3.2         pillar_1.10.2      
[34] crayon_1.5.3        jquerylib_0.1.4     whisker_0.4.1      
[37] cachem_1.1.0        magick_2.8.6        iterators_1.0.14   
[40] foreach_1.5.2       tidyselect_1.2.1    digest_0.6.34      
[43] stringi_1.8.3       rprojroot_2.0.4     fastmap_1.2.0      
[46] colorspace_2.1-0    cli_3.6.1           magrittr_2.0.3     
[49] withr_3.0.2         scales_1.3.0        promises_1.3.2     
[52] timechange_0.3.0    rmarkdown_2.29      httr_1.4.7         
[55] matrixStats_1.5.0   png_0.1-8           GetoptLong_1.0.5   
[58] hms_1.1.3           evaluate_1.0.3      knitr_1.50         
[61] IRanges_2.36.0      doParallel_1.0.17   rlang_1.1.3        
[64] Rcpp_1.0.12         glue_1.7.0          BiocGenerics_0.48.1
[67] rstudioapi_0.17.1   jsonlite_2.0.0      R6_2.6.1           
[70] fs_1.6.3           </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
