<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Differential Gene Expression Analysis</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CX5461_Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Differential Gene Expression Analysis</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-05-13
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>CX5461_Project/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree5a1db8113750a17ebafa6fec8521e12848f7e4fatargetblank5a1db81a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/5a1db8113750a17ebafa6fec8521e12848f7e4fa" target="_blank">5a1db81</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree5a1db8113750a17ebafa6fec8521e12848f7e4fatargetblank5a1db81a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/5a1db8113750a17ebafa6fec8521e12848f7e4fa" target="_blank">5a1db81</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    0.1 box.svg
    Ignored:    Rplot04.svg

Untracked files:
    Untracked:  0.1 density.svg
    Untracked:  0.1.emf
    Untracked:  0.1.svg
    Untracked:  0.5 box.svg
    Untracked:  0.5 density.svg
    Untracked:  0.5.svg
    Untracked:  Additional/
    Untracked:  CX_5461_Pattern_Genes_24hr.csv
    Untracked:  CX_5461_Pattern_Genes_3hr.csv
    Untracked:  Cell viability box plot.svg
    Untracked:  DRC1.svg
    Untracked:  Figure 1.jpeg
    Untracked:  Figure 1.pdf
    Untracked:  Figure_CM_Purity.pdf
    Untracked:  Rplot.svg
    Untracked:  Rplot01.svg
    Untracked:  Rplot02.svg
    Untracked:  Rplot03.svg
    Untracked:  Rplot05.svg
    Untracked:  Rplot06.svg
    Untracked:  Rplot07.svg
    Untracked:  Rplot08.jpeg
    Untracked:  Rplot08.svg
    Untracked:  Rplot09.svg
    Untracked:  Rplot10.svg
    Untracked:  Rplot11.svg
    Untracked:  TS HPA (Violin).svg
    Untracked:  TS HPA.svg
    Untracked:  TS_HA.svg
    Untracked:  TS_HV.svg
    Untracked:  Violin HA.svg
    Untracked:  Violin HV (CX vs DOX).svg
    Untracked:  Violin HV.svg
    Untracked:  data/AF.csv
    Untracked:  data/AF_Mapped.csv
    Untracked:  data/AF_genes.csv
    Untracked:  data/Annotated_DOX_Gene_Table.csv
    Untracked:  data/CAD_genes.csv
    Untracked:  data/Cardiotox.csv
    Untracked:  data/Cardiotox_mapped.csv
    Untracked:  data/DOX_Vald.csv
    Untracked:  data/DOX_Vald_Mapped.csv
    Untracked:  data/Entrez_Cardiotox.csv
    Untracked:  data/Entrez_Cardiotox_Mapped.csv
    Untracked:  data/GWAS.xlsx
    Untracked:  data/GWAS_SNPs.bed
    Untracked:  data/HF.csv
    Untracked:  data/HF_Mapped.csv
    Untracked:  data/HF_genes.csv
    Untracked:  data/Hypertension_genes.csv
    Untracked:  data/MI_genes.csv
    Untracked:  data/P53_Target_mapped.csv
    Untracked:  data/TOP2A.bed
    Untracked:  data/TOP2A_target.csv
    Untracked:  data/TOP2A_target_lit.csv
    Untracked:  data/TOP2A_target_lit_mapped.csv
    Untracked:  data/TOP2A_target_mapped.csv
    Untracked:  data/TOP2B_target.csv
    Untracked:  data/TOP2B_target_heatmap.csv
    Untracked:  data/TOP2B_target_heatmap_mapped.csv
    Untracked:  data/TOP2B_target_mapped.csv
    Untracked:  data/TS.csv
    Untracked:  data/TS_HPA.csv
    Untracked:  data/TS_HPA_mapped.csv
    Untracked:  data/Toptable_CX_0.1_24.csv
    Untracked:  data/Toptable_CX_0.1_3.csv
    Untracked:  data/Toptable_CX_0.1_48.csv
    Untracked:  data/Toptable_CX_0.5_24.csv
    Untracked:  data/Toptable_CX_0.5_3.csv
    Untracked:  data/Toptable_CX_0.5_48.csv
    Untracked:  data/Toptable_DOX_0.1_24.csv
    Untracked:  data/Toptable_DOX_0.1_3.csv
    Untracked:  data/Toptable_DOX_0.1_48.csv
    Untracked:  data/Toptable_DOX_0.5_24.csv
    Untracked:  data/Toptable_DOX_0.5_3.csv
    Untracked:  data/Toptable_DOX_0.5_48.csv
    Untracked:  data/ts_data_mapped
    Untracked:  results/
    Untracked:  run_bedtools.bat

Unstaged changes:
    Deleted:    analysis/Actox.Rmd
    Modified:   analysis/DGE_Analysis.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/DGE_Analysis.Rmd</code>) and HTML
(<code>docs/DGE_Analysis.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/5a1db8113750a17ebafa6fec8521e12848f7e4fa/analysis/DGE_Analysis.Rmd" target="_blank">5a1db81</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-09
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/5a1db8113750a17ebafa6fec8521e12848f7e4fa/docs/DGE_Analysis.html" target="_blank">5a1db81</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-09
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/dcf1e50fd654002fbd5ada235cd7e5956b9da7a8/docs/DGE_Analysis.html" target="_blank">dcf1e50</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-09
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3d33304d85bf2b6b87a3d324bc3a9d5e93720619/analysis/DGE_Analysis.Rmd" target="_blank">3d33304</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-09
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/3d33304d85bf2b6b87a3d324bc3a9d5e93720619/docs/DGE_Analysis.html" target="_blank">3d33304</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-09
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/72226f194b3d254ad4121b6ddbafec94f6adb811/analysis/DGE_Analysis.Rmd" target="_blank">72226f1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-09
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/72226f194b3d254ad4121b6ddbafec94f6adb811/docs/DGE_Analysis.html" target="_blank">72226f1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-09
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/328c60f4df9618213e2b4ce7339b3810e00c835e/analysis/DGE_Analysis.Rmd" target="_blank">328c60f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/328c60f4df9618213e2b4ce7339b3810e00c835e/docs/DGE_Analysis.html" target="_blank">328c60f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d11a2ea01708a96de53fa57d625c8ec8b2d13700/analysis/DGE_Analysis.Rmd" target="_blank">d11a2ea</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/d11a2ea01708a96de53fa57d625c8ec8b2d13700/docs/DGE_Analysis.html" target="_blank">d11a2ea</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/711098f70cf35739ae12795badc60904e4505b4b/analysis/DGE_Analysis.Rmd" target="_blank">711098f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/711098f70cf35739ae12795badc60904e4505b4b/docs/DGE_Analysis.html" target="_blank">711098f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/72b1144a683ea60eae33b338cce0b8f025d95b8a/docs/DGE_Analysis.html" target="_blank">72b1144</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/db482f63e9d4185545bdd38644f38bfdc441f6f9/analysis/DGE_Analysis.Rmd" target="_blank">db482f6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/db482f63e9d4185545bdd38644f38bfdc441f6f9/docs/DGE_Analysis.html" target="_blank">db482f6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/43e348d0bc195290fd54602de2e15faafc29c815/analysis/DGE_Analysis.Rmd" target="_blank">43e348d</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/43e348d0bc195290fd54602de2e15faafc29c815/docs/DGE_Analysis.html" target="_blank">43e348d</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/92099e7a5fa068d2357cb772c18c0e40b0953661/analysis/DGE_Analysis.Rmd" target="_blank">92099e7</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/92099e7a5fa068d2357cb772c18c0e40b0953661/docs/DGE_Analysis.html" target="_blank">92099e7</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/a6d66b90210019cfb38d0ec5de6773e6336904a7/docs/DGE_Analysis.html" target="_blank">a6d66b9</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/918f852228f1179da3bf3f39795d013d73a5aeaa/analysis/DGE_Analysis.Rmd" target="_blank">918f852</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-07
</td>
<td>
Save Toptables for LogFC correlation
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/7698b5e21c0acb45e7b3bf6586aa8b4bea489eb5/docs/DGE_Analysis.html" target="_blank">7698b5e</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/1c03f6037b4de5c6f8a8b9efccc352dfcdb2b1c6/docs/DGE_Analysis.html" target="_blank">1c03f60</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0a87308bc928450da5a2f7c3fea4e0f1f9da7086/analysis/DGE_Analysis.Rmd" target="_blank">0a87308</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
<td>
Save Toptables for LogFC correlation
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/9ba22e3aa55034dcd7d1f2f13b35956b8e87abbf/docs/DGE_Analysis.html" target="_blank">9ba22e3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/9f3ac074d486430a1a32b24765acb844105a2147/analysis/DGE_Analysis.Rmd" target="_blank">9f3ac07</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
<td>
Added Volcano Plots for all DEGs
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/705bc0beb1901215cb45444b35067e229b579056/docs/DGE_Analysis.html" target="_blank">705bc0b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/8bfc3198e17cf4191ef896fdfd42d8f8562fcd95/analysis/DGE_Analysis.Rmd" target="_blank">8bfc319</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
<td>
wflow_publish("analysis/DGE_Analysis.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/12a9a6b4d1b105c8eb555ad00ec4f7b5c9e7c9a2/docs/DGE_Analysis.html" target="_blank">12a9a6b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/eebe7d21cf0f3ca2f52d16c96959ced94418b6a1/analysis/DGE_Analysis.Rmd" target="_blank">eebe7d2</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
<td>
wflow_publish("analysis/DGE_Analysis.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/1ab310a2868484e3894cf13650eefbd3753c6e8a/docs/DGE_Analysis.html" target="_blank">1ab310a</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/4306834016806a55ae92ee460303be80c5c8d3d0/analysis/DGE_Analysis.Rmd" target="_blank">4306834</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
<td>
Added mean-variance trend plot
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/f59d02b117be50c4b08155f889f593bf515a81c5/docs/DGE_Analysis.html" target="_blank">f59d02b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f764eec17b1e639d7cc260f1f9541ddc0d21ceb/analysis/DGE_Analysis.Rmd" target="_blank">3f764ee</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
<td>
Added DGE Analysis with structured chunks
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/d6850ef5338055d6359fa111d4d8140bf7eb511f/docs/DGE_Analysis.html" target="_blank">d6850ef</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d31730d64c2d509cb4e48ac77ce13ca85cf5283e/analysis/DGE_Analysis.Rmd" target="_blank">d31730d</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
<td>
Added DGE Analysis with structured chunks
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="differential-gene-expression-analysis" class="section level2">
<h2><strong>📌 Differential Gene Expression Analysis </strong></h2>
<div id="load-required-libraries" class="section level3">
<h3><strong>📌 Load Required Libraries</strong></h3>
<pre class="r"><code># Load necessary packages
library(edgeR)</code></pre>
<pre><code>Warning: package &#39;edgeR&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;limma&#39; was built under R version 4.3.1</code></pre>
<pre class="r"><code>library(limma)
library(data.table)</code></pre>
<pre><code>Warning: package &#39;data.table&#39; was built under R version 4.3.3</code></pre>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>Warning: package &#39;tidyverse&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;tidyr&#39; was built under R version 4.3.3</code></pre>
<pre><code>Warning: package &#39;readr&#39; was built under R version 4.3.3</code></pre>
<pre><code>Warning: package &#39;purrr&#39; was built under R version 4.3.3</code></pre>
<pre><code>Warning: package &#39;dplyr&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;stringr&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;lubridate&#39; was built under R version 4.3.3</code></pre>
<pre class="r"><code>library(ggplot2)
library(dplyr)
library(scales)</code></pre>
<pre><code>Warning: package &#39;scales&#39; was built under R version 4.3.2</code></pre>
<pre class="r"><code>library(biomaRt)</code></pre>
<pre><code>Warning: package &#39;biomaRt&#39; was built under R version 4.3.2</code></pre>
<pre class="r"><code>library(Homo.sapiens)</code></pre>
<pre><code>Warning: package &#39;AnnotationDbi&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;BiocGenerics&#39; was built under R version 4.3.1</code></pre>
<pre><code>Warning: package &#39;Biobase&#39; was built under R version 4.3.1</code></pre>
<pre><code>Warning: package &#39;IRanges&#39; was built under R version 4.3.1</code></pre>
<pre><code>Warning: package &#39;S4Vectors&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;OrganismDbi&#39; was built under R version 4.3.1</code></pre>
<pre><code>Warning: package &#39;GenomicFeatures&#39; was built under R version 4.3.3</code></pre>
<pre><code>Warning: package &#39;GenomeInfoDb&#39; was built under R version 4.3.3</code></pre>
<pre><code>Warning: package &#39;GenomicRanges&#39; was built under R version 4.3.1</code></pre>
<p>📍 Load Data</p>
</div>
</div>
<div id="normalize-and-filter-counts" class="section level2">
<h2><strong>📌 Normalize and Filter Counts</strong></h2>
</div>
<div id="prepare-dgelist-object" class="section level2">
<h2><strong>📌 Prepare DGEList Object</strong></h2>
<pre class="r"><code># Subset count matrix based on filtered CPM matrix
x &lt;- counts_matrix[row.names(filcpm_matrix),]
dim(x)</code></pre>
<pre><code>[1] 14279   108</code></pre>
<pre class="r"><code># Modify Metadata
Metadata_2 &lt;- Metadata
rownames(Metadata_2) &lt;- Metadata_2$Sample_bam
colnames(x) &lt;- Metadata_2$Sample_ID
rownames(Metadata_2) &lt;- Metadata_2$Sample_ID

Metadata_2$Drug_time &lt;- make.names(Metadata_2$Drug_time)
Metadata_2$Ind &lt;- as.character(Metadata_2$Ind)

# ✅ Create DGEList object
dge &lt;- DGEList(counts = x)
dge$samples$group &lt;- factor(Metadata_2$Drug_time)
dge &lt;- calcNormFactors(dge, method = &quot;TMM&quot;)

# ✅ Check normalization factors
dge$samples</code></pre>
<pre><code>                             group lib.size norm.factors
CX-5461_0.1_3_17-3   CX.5461_0.1_3 21200035    1.0102693
DOX_0.5_24_17-3         DOX_0.5_24 16563771    1.1076182
DOX_0.5_24_87-1         DOX_0.5_24 21656653    0.9933568
VEH_0.1_24_87-1         VEH_0.1_24 22426842    0.9766783
VEH_0.5_24_87-1         VEH_0.5_24 22162275    0.9924549
CX-5461_0.1_48_87-1 CX.5461_0.1_48 22187620    0.9683497
CX-5461_0.5_48_87-1 CX.5461_0.5_48 20782167    0.9814665
DOX_0.1_48_87-1         DOX_0.1_48 25382841    1.0011030
DOX_0.5_48_87-1         DOX_0.5_48 19863531    1.0541528
VEH_0.1_48_87-1         VEH_0.1_48 21473253    1.0138011
VEH_0.5_48_87-1         VEH_0.5_48 20337337    1.0219756
VEH_0.1_24_17-3         VEH_0.1_24 15750538    0.9952313
VEH_0.5_24_17-3         VEH_0.5_24 15638967    1.0367909
CX-5461_0.1_48_17-3 CX.5461_0.1_48 18465522    0.9543636
CX-5461_0.5_48_17-3 CX.5461_0.5_48 16876134    0.9636074
DOX_0.1_48_17-3         DOX_0.1_48 20609601    0.9839708
DOX_0.5_48_17-3         DOX_0.5_48 18829944    1.1876478
VEH_0.1_48_17-3         VEH_0.1_48 24712537    0.9804184
VEH_0.5_48_17-3         VEH_0.5_48 17006648    0.9707617
CX-5461_0.1_3_84-1   CX.5461_0.1_3 17178171    1.0219387
CX-5461_0.5_3_17-3   CX.5461_0.5_3 17694385    1.0103837
CX-5461_0.5_3_84-1   CX.5461_0.5_3 17261114    1.0060170
DOX_0.1_3_84-1           DOX_0.1_3 17352493    0.9911466
DOX_0.5_3_84-1           DOX_0.5_3 17757477    1.0034479
VEH_0.1_3_84-1           VEH_0.1_3 19305213    1.0354377
VEH_0.5_3_84-1           VEH_0.5_3 17297185    1.0285524
CX-5461_0.1_24_84-1 CX.5461_0.1_24 23860498    0.9857321
CX-5461_0.5_24_84-1 CX.5461_0.5_24 19482159    0.9638391
DOX_0.1_24_84-1         DOX_0.1_24 19659504    0.9565317
DOX_0.5_24_84-1         DOX_0.5_24 17875612    1.0212021
VEH_0.1_24_84-1         VEH_0.1_24 16449210    0.9998393
DOX_0.1_3_17-3           DOX_0.1_3 16400599    0.9962197
VEH_0.5_24_84-1         VEH_0.5_24 18783630    0.9616033
CX-5461_0.1_48_84-1 CX.5461_0.1_48 13781326    0.9400278
CX-5461_0.5_48_84-1 CX.5461_0.5_48 20232591    0.9563694
DOX_0.1_48_84-1         DOX_0.1_48 16028063    0.9651474
DOX_0.5_48_84-1         DOX_0.5_48 16269098    1.0652791
VEH_0.1_48_84-1         VEH_0.1_48 15276408    0.9612979
VEH_0.5_48_84-1         VEH_0.5_48 17447290    1.0050272
CX-5461_0.1_3_90-1   CX.5461_0.1_3 15807726    0.9991915
CX-5461_0.5_3_90-1   CX.5461_0.5_3 17946649    0.9956913
DOX_0.1_3_90-1           DOX_0.1_3 20294777    0.9830679
DOX_0.5_3_17-3           DOX_0.5_3 18191217    0.9080939
DOX_0.5_3_90-1           DOX_0.5_3 20699125    0.9833929
VEH_0.1_3_90-1           VEH_0.1_3 20027541    0.9953796
VEH_0.5_3_90-1           VEH_0.5_3 17833214    1.0003025
CX-5461_0.1_24_90-1 CX.5461_0.1_24 15094639    1.0102577
CX-5461_0.5_24_90-1 CX.5461_0.5_24 17758047    1.0114258
DOX_0.1_24_90-1         DOX_0.1_24 16921727    0.9900805
DOX_0.5_24_90-1         DOX_0.5_24 19271264    1.0068767
VEH_0.1_24_90-1         VEH_0.1_24 20601703    0.9898691
VEH_0.5_24_90-1         VEH_0.5_24 19875553    1.0060950
CX-5461_0.1_48_90-1 CX.5461_0.1_48 23817456    1.0255425
VEH_0.1_3_17-3           VEH_0.1_3 26353255    0.9919794
CX-5461_0.5_48_90-1 CX.5461_0.5_48 18651093    1.0370010
DOX_0.1_48_90-1         DOX_0.1_48 17767662    1.0233343
DOX_0.5_48_90-1         DOX_0.5_48 20031141    1.1189284
VEH_0.1_48_90-1         VEH_0.1_48 19523120    1.0484497
VEH_0.5_48_90-1         VEH_0.5_48 17657193    1.0550052
CX-5461_0.1_3_75-1   CX.5461_0.1_3 20046785    0.9894881
CX-5461_0.5_3_75-1   CX.5461_0.5_3 19885297    0.9767025
DOX_0.1_3_75-1           DOX_0.1_3 18575114    0.9755788
DOX_0.5_3_75-1           DOX_0.5_3 15260497    0.9562336
VEH_0.1_3_75-1           VEH_0.1_3 19040007    0.9839769
VEH_0.5_3_17-3           VEH_0.5_3 18217503    0.9735069
VEH_0.5_3_75-1           VEH_0.5_3 21311999    0.9663251
CX-5461_0.1_24_75-1 CX.5461_0.1_24 19256517    1.0184093
CX-5461_0.5_24_75-1 CX.5461_0.5_24 14304255    1.0186309
DOX_0.1_24_75-1         DOX_0.1_24 21875105    1.0386302
DOX_0.5_24_75-1         DOX_0.5_24 17032433    0.9686987
VEH_0.1_24_75-1         VEH_0.1_24 19796422    1.0295019
VEH_0.5_24_75-1         VEH_0.5_24 21524523    1.0376940
CX-5461_0.1_48_75-1 CX.5461_0.1_48 20955873    0.9752971
CX-5461_0.5_48_75-1 CX.5461_0.5_48 19924867    0.9768371
DOX_0.1_48_75-1         DOX_0.1_48 17698471    1.0369925
CX-5461_0.1_24_17-3 CX.5461_0.1_24 18239213    0.9808618
DOX_0.5_48_75-1         DOX_0.5_48 15422887    1.0296304
VEH_0.1_48_75-1         VEH_0.1_48 21014387    1.0100812
VEH_0.5_48_75-1         VEH_0.5_48 21346438    1.0015670
CX-5461_0.1_3_78-1   CX.5461_0.1_3 20824890    0.9686975
CX-5461_0.5_3_78-1   CX.5461_0.5_3 19425405    1.0100551
DOX_0.1_3_78-1           DOX_0.1_3 18950843    0.9933529
DOX_0.5_3_78-1           DOX_0.5_3 21654404    0.9808038
VEH_0.1_3_78-1           VEH_0.1_3 20054872    0.9802655
VEH_0.5_3_78-1           VEH_0.5_3 17661233    1.0220629
CX-5461_0.1_24_78-1 CX.5461_0.1_24 16754941    0.9785599
CX-5461_0.5_24_17-3 CX.5461_0.5_24 16014473    1.0010257
CX-5461_0.5_24_78-1 CX.5461_0.5_24 16077548    0.9819166
DOX_0.1_24_78-1         DOX_0.1_24 27842910    0.9788383
DOX_0.5_24_78-1         DOX_0.5_24  6326261    1.0007627
VEH_0.1_24_78-1         VEH_0.1_24 18032196    0.9767098
VEH_0.5_24_78-1         VEH_0.5_24 18727398    0.9931695
CX-5461_0.1_48_78-1 CX.5461_0.1_48 21292146    1.0036183
CX-5461_0.5_48_78-1 CX.5461_0.5_48 18392707    0.9988905
DOX_0.1_48_78-1         DOX_0.1_48 18735925    1.0433629
DOX_0.5_48_78-1         DOX_0.5_48 16079343    1.1063695
VEH_0.1_48_78-1         VEH_0.1_48 20500742    1.0149206
DOX_0.1_24_17-3         DOX_0.1_24 23900805    1.0243090
VEH_0.5_48_78-1         VEH_0.5_48 18433692    1.0396358
CX-5461_0.1_3_87-1   CX.5461_0.1_3 19664754    0.9573708
CX-5461_0.5_3_87-1   CX.5461_0.5_3 21451819    1.0192062
DOX_0.1_3_87-1           DOX_0.1_3 20624889    0.9539861
DOX_0.5_3_87-1           DOX_0.5_3 22838835    0.9919424
VEH_0.1_3_87-1           VEH_0.1_3 17616767    0.9841881
VEH_0.5_3_87-1           VEH_0.5_3 22148086    1.0203028
CX-5461_0.1_24_87-1 CX.5461_0.1_24 30093416    0.9439250
CX-5461_0.5_24_87-1 CX.5461_0.5_24 22093483    0.9589794
DOX_0.1_24_87-1         DOX_0.1_24 21415578    0.9549937</code></pre>
</div>
<div id="create-design-matrix" class="section level2">
<h2><strong>📌 Create Design Matrix</strong></h2>
<pre class="r"><code># ✅ Create Design Matrix
design &lt;- model.matrix(~ 0 + Metadata_2$Drug_time)
colnames(design) &lt;- gsub(&quot;Metadata_2\\$Drug_time&quot;, &quot;&quot;, colnames(design))
design</code></pre>
<pre><code>    CX.5461_0.1_24 CX.5461_0.1_3 CX.5461_0.1_48 CX.5461_0.5_24 CX.5461_0.5_3
1                0             1              0              0             0
2                0             0              0              0             0
3                0             0              0              0             0
4                0             0              0              0             0
5                0             0              0              0             0
6                0             0              1              0             0
7                0             0              0              0             0
8                0             0              0              0             0
9                0             0              0              0             0
10               0             0              0              0             0
11               0             0              0              0             0
12               0             0              0              0             0
13               0             0              0              0             0
14               0             0              1              0             0
15               0             0              0              0             0
16               0             0              0              0             0
17               0             0              0              0             0
18               0             0              0              0             0
19               0             0              0              0             0
20               0             1              0              0             0
21               0             0              0              0             1
22               0             0              0              0             1
23               0             0              0              0             0
24               0             0              0              0             0
25               0             0              0              0             0
26               0             0              0              0             0
27               1             0              0              0             0
28               0             0              0              1             0
29               0             0              0              0             0
30               0             0              0              0             0
31               0             0              0              0             0
32               0             0              0              0             0
33               0             0              0              0             0
34               0             0              1              0             0
35               0             0              0              0             0
36               0             0              0              0             0
37               0             0              0              0             0
38               0             0              0              0             0
39               0             0              0              0             0
40               0             1              0              0             0
41               0             0              0              0             1
42               0             0              0              0             0
43               0             0              0              0             0
44               0             0              0              0             0
45               0             0              0              0             0
46               0             0              0              0             0
47               1             0              0              0             0
48               0             0              0              1             0
49               0             0              0              0             0
50               0             0              0              0             0
51               0             0              0              0             0
52               0             0              0              0             0
53               0             0              1              0             0
54               0             0              0              0             0
55               0             0              0              0             0
56               0             0              0              0             0
57               0             0              0              0             0
58               0             0              0              0             0
59               0             0              0              0             0
60               0             1              0              0             0
61               0             0              0              0             1
62               0             0              0              0             0
63               0             0              0              0             0
64               0             0              0              0             0
65               0             0              0              0             0
66               0             0              0              0             0
67               1             0              0              0             0
68               0             0              0              1             0
69               0             0              0              0             0
70               0             0              0              0             0
71               0             0              0              0             0
72               0             0              0              0             0
73               0             0              1              0             0
74               0             0              0              0             0
75               0             0              0              0             0
76               1             0              0              0             0
77               0             0              0              0             0
78               0             0              0              0             0
79               0             0              0              0             0
80               0             1              0              0             0
81               0             0              0              0             1
82               0             0              0              0             0
83               0             0              0              0             0
84               0             0              0              0             0
85               0             0              0              0             0
86               1             0              0              0             0
87               0             0              0              1             0
88               0             0              0              1             0
89               0             0              0              0             0
90               0             0              0              0             0
91               0             0              0              0             0
92               0             0              0              0             0
93               0             0              1              0             0
94               0             0              0              0             0
95               0             0              0              0             0
96               0             0              0              0             0
97               0             0              0              0             0
98               0             0              0              0             0
99               0             0              0              0             0
100              0             1              0              0             0
101              0             0              0              0             1
102              0             0              0              0             0
103              0             0              0              0             0
104              0             0              0              0             0
105              0             0              0              0             0
106              1             0              0              0             0
107              0             0              0              1             0
108              0             0              0              0             0
    CX.5461_0.5_48 DOX_0.1_24 DOX_0.1_3 DOX_0.1_48 DOX_0.5_24 DOX_0.5_3
1                0          0         0          0          0         0
2                0          0         0          0          1         0
3                0          0         0          0          1         0
4                0          0         0          0          0         0
5                0          0         0          0          0         0
6                0          0         0          0          0         0
7                1          0         0          0          0         0
8                0          0         0          1          0         0
9                0          0         0          0          0         0
10               0          0         0          0          0         0
11               0          0         0          0          0         0
12               0          0         0          0          0         0
13               0          0         0          0          0         0
14               0          0         0          0          0         0
15               1          0         0          0          0         0
16               0          0         0          1          0         0
17               0          0         0          0          0         0
18               0          0         0          0          0         0
19               0          0         0          0          0         0
20               0          0         0          0          0         0
21               0          0         0          0          0         0
22               0          0         0          0          0         0
23               0          0         1          0          0         0
24               0          0         0          0          0         1
25               0          0         0          0          0         0
26               0          0         0          0          0         0
27               0          0         0          0          0         0
28               0          0         0          0          0         0
29               0          1         0          0          0         0
30               0          0         0          0          1         0
31               0          0         0          0          0         0
32               0          0         1          0          0         0
33               0          0         0          0          0         0
34               0          0         0          0          0         0
35               1          0         0          0          0         0
36               0          0         0          1          0         0
37               0          0         0          0          0         0
38               0          0         0          0          0         0
39               0          0         0          0          0         0
40               0          0         0          0          0         0
41               0          0         0          0          0         0
42               0          0         1          0          0         0
43               0          0         0          0          0         1
44               0          0         0          0          0         1
45               0          0         0          0          0         0
46               0          0         0          0          0         0
47               0          0         0          0          0         0
48               0          0         0          0          0         0
49               0          1         0          0          0         0
50               0          0         0          0          1         0
51               0          0         0          0          0         0
52               0          0         0          0          0         0
53               0          0         0          0          0         0
54               0          0         0          0          0         0
55               1          0         0          0          0         0
56               0          0         0          1          0         0
57               0          0         0          0          0         0
58               0          0         0          0          0         0
59               0          0         0          0          0         0
60               0          0         0          0          0         0
61               0          0         0          0          0         0
62               0          0         1          0          0         0
63               0          0         0          0          0         1
64               0          0         0          0          0         0
65               0          0         0          0          0         0
66               0          0         0          0          0         0
67               0          0         0          0          0         0
68               0          0         0          0          0         0
69               0          1         0          0          0         0
70               0          0         0          0          1         0
71               0          0         0          0          0         0
72               0          0         0          0          0         0
73               0          0         0          0          0         0
74               1          0         0          0          0         0
75               0          0         0          1          0         0
76               0          0         0          0          0         0
77               0          0         0          0          0         0
78               0          0         0          0          0         0
79               0          0         0          0          0         0
80               0          0         0          0          0         0
81               0          0         0          0          0         0
82               0          0         1          0          0         0
83               0          0         0          0          0         1
84               0          0         0          0          0         0
85               0          0         0          0          0         0
86               0          0         0          0          0         0
87               0          0         0          0          0         0
88               0          0         0          0          0         0
89               0          1         0          0          0         0
90               0          0         0          0          1         0
91               0          0         0          0          0         0
92               0          0         0          0          0         0
93               0          0         0          0          0         0
94               1          0         0          0          0         0
95               0          0         0          1          0         0
96               0          0         0          0          0         0
97               0          0         0          0          0         0
98               0          1         0          0          0         0
99               0          0         0          0          0         0
100              0          0         0          0          0         0
101              0          0         0          0          0         0
102              0          0         1          0          0         0
103              0          0         0          0          0         1
104              0          0         0          0          0         0
105              0          0         0          0          0         0
106              0          0         0          0          0         0
107              0          0         0          0          0         0
108              0          1         0          0          0         0
    DOX_0.5_48 VEH_0.1_24 VEH_0.1_3 VEH_0.1_48 VEH_0.5_24 VEH_0.5_3 VEH_0.5_48
1            0          0         0          0          0         0          0
2            0          0         0          0          0         0          0
3            0          0         0          0          0         0          0
4            0          1         0          0          0         0          0
5            0          0         0          0          1         0          0
6            0          0         0          0          0         0          0
7            0          0         0          0          0         0          0
8            0          0         0          0          0         0          0
9            1          0         0          0          0         0          0
10           0          0         0          1          0         0          0
11           0          0         0          0          0         0          1
12           0          1         0          0          0         0          0
13           0          0         0          0          1         0          0
14           0          0         0          0          0         0          0
15           0          0         0          0          0         0          0
16           0          0         0          0          0         0          0
17           1          0         0          0          0         0          0
18           0          0         0          1          0         0          0
19           0          0         0          0          0         0          1
20           0          0         0          0          0         0          0
21           0          0         0          0          0         0          0
22           0          0         0          0          0         0          0
23           0          0         0          0          0         0          0
24           0          0         0          0          0         0          0
25           0          0         1          0          0         0          0
26           0          0         0          0          0         1          0
27           0          0         0          0          0         0          0
28           0          0         0          0          0         0          0
29           0          0         0          0          0         0          0
30           0          0         0          0          0         0          0
31           0          1         0          0          0         0          0
32           0          0         0          0          0         0          0
33           0          0         0          0          1         0          0
34           0          0         0          0          0         0          0
35           0          0         0          0          0         0          0
36           0          0         0          0          0         0          0
37           1          0         0          0          0         0          0
38           0          0         0          1          0         0          0
39           0          0         0          0          0         0          1
40           0          0         0          0          0         0          0
41           0          0         0          0          0         0          0
42           0          0         0          0          0         0          0
43           0          0         0          0          0         0          0
44           0          0         0          0          0         0          0
45           0          0         1          0          0         0          0
46           0          0         0          0          0         1          0
47           0          0         0          0          0         0          0
48           0          0         0          0          0         0          0
49           0          0         0          0          0         0          0
50           0          0         0          0          0         0          0
51           0          1         0          0          0         0          0
52           0          0         0          0          1         0          0
53           0          0         0          0          0         0          0
54           0          0         1          0          0         0          0
55           0          0         0          0          0         0          0
56           0          0         0          0          0         0          0
57           1          0         0          0          0         0          0
58           0          0         0          1          0         0          0
59           0          0         0          0          0         0          1
60           0          0         0          0          0         0          0
61           0          0         0          0          0         0          0
62           0          0         0          0          0         0          0
63           0          0         0          0          0         0          0
64           0          0         1          0          0         0          0
65           0          0         0          0          0         1          0
66           0          0         0          0          0         1          0
67           0          0         0          0          0         0          0
68           0          0         0          0          0         0          0
69           0          0         0          0          0         0          0
70           0          0         0          0          0         0          0
71           0          1         0          0          0         0          0
72           0          0         0          0          1         0          0
73           0          0         0          0          0         0          0
74           0          0         0          0          0         0          0
75           0          0         0          0          0         0          0
76           0          0         0          0          0         0          0
77           1          0         0          0          0         0          0
78           0          0         0          1          0         0          0
79           0          0         0          0          0         0          1
80           0          0         0          0          0         0          0
81           0          0         0          0          0         0          0
82           0          0         0          0          0         0          0
83           0          0         0          0          0         0          0
84           0          0         1          0          0         0          0
85           0          0         0          0          0         1          0
86           0          0         0          0          0         0          0
87           0          0         0          0          0         0          0
88           0          0         0          0          0         0          0
89           0          0         0          0          0         0          0
90           0          0         0          0          0         0          0
91           0          1         0          0          0         0          0
92           0          0         0          0          1         0          0
93           0          0         0          0          0         0          0
94           0          0         0          0          0         0          0
95           0          0         0          0          0         0          0
96           1          0         0          0          0         0          0
97           0          0         0          1          0         0          0
98           0          0         0          0          0         0          0
99           0          0         0          0          0         0          1
100          0          0         0          0          0         0          0
101          0          0         0          0          0         0          0
102          0          0         0          0          0         0          0
103          0          0         0          0          0         0          0
104          0          0         1          0          0         0          0
105          0          0         0          0          0         1          0
106          0          0         0          0          0         0          0
107          0          0         0          0          0         0          0
108          0          0         0          0          0         0          0
attr(,&quot;assign&quot;)
 [1] 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
attr(,&quot;contrasts&quot;)
attr(,&quot;contrasts&quot;)$`Metadata_2$Drug_time`
[1] &quot;contr.treatment&quot;</code></pre>
</div>
<div id="voom-transformation-and-linear-modeling"
class="section level2">
<h2><strong>📌 Voom Transformation and Linear Modeling</strong></h2>
<pre class="r"><code># ✅ Duplicate Correlation for Individual Effect
corfit &lt;- duplicateCorrelation(object = dge$counts, design = design, block = Metadata_2$Ind)

# ✅ Voom Transformation
v &lt;- voom(dge, design, block = Metadata_2$Ind, correlation = corfit$consensus.correlation, plot = TRUE)</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/Voom-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Voom-1">
Past versions of Voom-1.png
</button>
</p>
<div id="fig-Voom-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/92099e7a5fa068d2357cb772c18c0e40b0953661/docs/figure/DGE_Analysis.Rmd/Voom-1.png" target="_blank">92099e7</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d6850ef5338055d6359fa111d4d8140bf7eb511f/docs/figure/DGE_Analysis.Rmd/Voom-1.png" target="_blank">d6850ef</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># ✅ Fit Linear Model
fit &lt;- lmFit(v, design, block = Metadata_2$Ind, correlation = corfit$consensus.correlation)</code></pre>
</div>
<div id="create-contrast-matrix" class="section level2">
<h2><strong>📌 Create Contrast Matrix</strong></h2>
<pre class="r"><code>contrast_matrix &lt;- makeContrasts(
  CX_0.1_3vsVEH_0.1_3 = CX.5461_0.1_3 - VEH_0.1_3,
  CX_0.1_24vsVEH_0.1_24 = CX.5461_0.1_24 - VEH_0.1_24,
  CX_0.1_48vsVEH_0.1_48 = CX.5461_0.1_48 - VEH_0.1_48,
  CX_0.5_3vsVEH_0.5_3 = CX.5461_0.5_3 - VEH_0.5_3,
  CX_0.5_24vsVEH_0.5_24 = CX.5461_0.5_24 - VEH_0.5_24,
  CX_0.5_48vsVEH_0.5_48 = CX.5461_0.5_48 - VEH_0.5_48,
  DOX_0.1_3vsVEH_0.1_3 = DOX_0.1_3 - VEH_0.1_3,
  DOX_0.1_24vsVEH_0.1_24 = DOX_0.1_24 - VEH_0.1_24,
  DOX_0.1_48vsVEH_0.1_48 = DOX_0.1_48 - VEH_0.1_48,
  DOX_0.5_3vsVEH_0.5_3 = DOX_0.5_3 - VEH_0.5_3,
  DOX_0.5_24vsVEH_0.5_24 = DOX_0.5_24 - VEH_0.5_24,
  DOX_0.5_48vsVEH_0.5_48 = DOX_0.5_48 - VEH_0.5_48,
  levels = design
)</code></pre>
</div>
<div id="fit-model-and-generate-results" class="section level2">
<h2><strong>📌 Fit Model and Generate Results</strong></h2>
<pre class="r"><code># ✅ Apply Contrasts
fit2 &lt;- contrasts.fit(fit, contrast_matrix)
fit2 &lt;- eBayes(fit2)</code></pre>
</div>
<div id="mean-variance-trend-plot" class="section level2">
<h2><strong>📌 Mean-Variance Trend Plot</strong></h2>
<pre class="r"><code># Plot Mean-Variance Trend
plotSA(fit2, main = &quot;Final model: Mean-Variance trend&quot;)</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/fit2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fit2-1">
Past versions of fit2-1.png
</button>
</p>
<div id="fig-fit2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/92099e7a5fa068d2357cb772c18c0e40b0953661/docs/figure/DGE_Analysis.Rmd/fit2-1.png" target="_blank">92099e7</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/1ab310a2868484e3894cf13650eefbd3753c6e8a/docs/figure/DGE_Analysis.Rmd/fit2-1.png" target="_blank">1ab310a</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="summary-of-results" class="section level2">
<h2><strong>📌Summary of Results</strong></h2>
<pre class="r"><code># ✅ Summary of Results
results_summary &lt;- decideTests(fit2, adjust.method = &quot;BH&quot;, p.value = 0.05)
summary(results_summary)</code></pre>
<pre><code>       CX_0.1_3vsVEH_0.1_3 CX_0.1_24vsVEH_0.1_24 CX_0.1_48vsVEH_0.1_48
Down                     0                   204                   294
NotSig               14278                 14074                 13961
Up                       1                     1                    24
       CX_0.5_3vsVEH_0.5_3 CX_0.5_24vsVEH_0.5_24 CX_0.5_48vsVEH_0.5_48
Down                     1                   259                   337
NotSig               14277                 14001                 13909
Up                       1                    19                    33
       DOX_0.1_3vsVEH_0.1_3 DOX_0.1_24vsVEH_0.1_24 DOX_0.1_48vsVEH_0.1_48
Down                      1                   2528                   1696
NotSig                14277                   9215                  10818
Up                        1                   2536                   1765
       DOX_0.5_3vsVEH_0.5_3 DOX_0.5_24vsVEH_0.5_24 DOX_0.5_48vsVEH_0.5_48
Down                    476                   5067                   5177
NotSig                13746                   4451                   4151
Up                       57                   4761                   4951</code></pre>
</div>
<div id="extract-differentially-expressed-genes" class="section level2">
<h2><strong>📌 Extract Differentially Expressed Genes</strong></h2>
<pre class="r"><code># Generate Top Table for Specific Comparisons

Toptable_CX_0.1_3 &lt;- topTable(fit = fit2, coef = &quot;CX_0.1_3vsVEH_0.1_3&quot;, number = nrow(x), adjust.method = &quot;BH&quot;, p.value = 1, sort.by = &quot;none&quot;)

Toptable_CX_0.1_24 &lt;- topTable(fit = fit2, coef = &quot;CX_0.1_24vsVEH_0.1_24&quot;, number = nrow(x), adjust.method = &quot;BH&quot;, p.value = 1, sort.by = &quot;none&quot;)

Toptable_CX_0.1_48 &lt;- topTable(fit = fit2, coef = &quot;CX_0.1_48vsVEH_0.1_48&quot;, number = nrow(x), adjust.method = &quot;BH&quot;, p.value = 1, sort.by = &quot;none&quot;)

Toptable_CX_0.5_3 &lt;- topTable(fit = fit2, coef = &quot;CX_0.5_3vsVEH_0.5_3&quot;, number = nrow(x), adjust.method = &quot;BH&quot;, p.value = 1, sort.by = &quot;none&quot;)

Toptable_CX_0.5_24 &lt;- topTable(fit = fit2, coef = &quot;CX_0.5_24vsVEH_0.5_24&quot;, number = nrow(x), adjust.method = &quot;BH&quot;, p.value = 1, sort.by = &quot;none&quot;)

Toptable_CX_0.5_48 &lt;- topTable(fit = fit2, coef = &quot;CX_0.5_48vsVEH_0.5_48&quot;, number = nrow(x), adjust.method = &quot;BH&quot;, p.value = 1, sort.by = &quot;none&quot;)

Toptable_DOX_0.1_3 &lt;- topTable(fit = fit2, coef = &quot;DOX_0.1_3vsVEH_0.1_3&quot;, number = nrow(x), adjust.method = &quot;BH&quot;, p.value = 1, sort.by = &quot;none&quot;)

Toptable_DOX_0.1_24 &lt;- topTable(fit = fit2, coef = &quot;DOX_0.1_24vsVEH_0.1_24&quot;, number = nrow(x), adjust.method = &quot;BH&quot;, p.value = 1, sort.by = &quot;none&quot;)

Toptable_DOX_0.1_48 &lt;- topTable(fit = fit2, coef = &quot;DOX_0.1_48vsVEH_0.1_48&quot;, number = nrow(x), adjust.method = &quot;BH&quot;, p.value = 1, sort.by = &quot;none&quot;)

Toptable_DOX_0.5_3 &lt;- topTable(fit = fit2, coef = &quot;DOX_0.5_3vsVEH_0.5_3&quot;, number = nrow(x), adjust.method = &quot;BH&quot;, p.value = 1, sort.by = &quot;none&quot;)

Toptable_DOX_0.5_24 &lt;- topTable(fit = fit2, coef = &quot;DOX_0.5_24vsVEH_0.5_24&quot;, number = nrow(x), adjust.method = &quot;BH&quot;, p.value = 1, sort.by = &quot;none&quot;)

Toptable_DOX_0.5_48 &lt;- topTable(fit = fit2, coef = &quot;DOX_0.5_48vsVEH_0.5_48&quot;, number = nrow(x), adjust.method = &quot;BH&quot;, p.value = 1, sort.by = &quot;none&quot;)</code></pre>
</div>
<div id="save-all-toptables-as-an-rds-file" class="section level2">
<h2><strong>📌 Save all Toptables as an RDS file</strong></h2>
<pre class="r"><code>saveRDS(list(
  CX_0.1_3 = Toptable_CX_0.1_3,
  CX_0.1_24 = Toptable_CX_0.1_24,
  CX_0.1_48 = Toptable_CX_0.1_48,
  CX_0.5_3 = Toptable_CX_0.5_3,
  CX_0.5_24 = Toptable_CX_0.5_24,
  CX_0.5_48 = Toptable_CX_0.5_48,
  DOX_0.1_3 = Toptable_DOX_0.1_3,
  DOX_0.1_24 = Toptable_DOX_0.1_24,
  DOX_0.1_48 = Toptable_DOX_0.1_48,
  DOX_0.5_3 = Toptable_DOX_0.5_3,
  DOX_0.5_24 = Toptable_DOX_0.5_24,
  DOX_0.5_48 = Toptable_DOX_0.5_48
), file = &quot;data/Toptable_list.rds&quot;)</code></pre>
</div>
<div id="volcano-plots-for-differential-expression-analysis"
class="section level2">
<h2><strong>📌 Volcano Plots for Differential Expression
Analysis</strong></h2>
<pre class="r"><code># Define a function to generate volcano plots
generate_volcano_plot &lt;- function(toptable, title) {
  
  # Add Significance Labels
  toptable$Significance &lt;- &quot;Not Significant&quot;
  toptable$Significance[toptable$logFC &gt; 0 &amp; toptable$adj.P.Val &lt; 0.05] &lt;- &quot;Upregulated&quot;
  toptable$Significance[toptable$logFC &lt; 0 &amp; toptable$adj.P.Val &lt; 0.05] &lt;- &quot;Downregulated&quot;

  # Generate Volcano Plot
  ggplot(toptable, aes(x = logFC, y = -log10(P.Value), color = Significance)) +
    geom_point(alpha = 0.4, size = 2) + 
    scale_color_manual(values = c(&quot;Downregulated&quot; = &quot;red&quot;, &quot;Upregulated&quot; = &quot;blue&quot;, &quot;Not Significant&quot; = &quot;gray&quot;)) +
    xlim(-5, 5) +
    labs(title = title, x = &quot;log2 Fold Change&quot;, y = &quot;-log10 P-value&quot;) +
    theme(legend.position = &quot;none&quot;, 
          plot.title = element_text(size = rel(1.5), hjust = 0.5),
          axis.title = element_text(size = rel(1.25))) +
    theme_bw()
}

# Generate volcano plots for all comparisons
volcano_plots &lt;- list(
  &quot;CX_0.1_3&quot; = generate_volcano_plot(Toptable_CX_0.1_3, &quot;Volcano Plot CX_0.1_3 (adj P-val&lt;0.05)&quot;),
  &quot;CX_0.1_24&quot; = generate_volcano_plot(Toptable_CX_0.1_24, &quot;Volcano Plot CX_0.1_24 (adj P-val&lt;0.05)&quot;),
  &quot;CX_0.1_48&quot; = generate_volcano_plot(Toptable_CX_0.1_48, &quot;Volcano Plot CX_0.1_48 (adj P-val&lt;0.05)&quot;),
  &quot;CX_0.5_3&quot; = generate_volcano_plot(Toptable_CX_0.5_3, &quot;Volcano Plot CX_0.5_3 (adj P-val&lt;0.05)&quot;),
  &quot;CX_0.5_24&quot; = generate_volcano_plot(Toptable_CX_0.5_24, &quot;Volcano Plot CX_0.5_24 (adj P-val&lt;0.05)&quot;),
  &quot;CX_0.5_48&quot; = generate_volcano_plot(Toptable_CX_0.5_48, &quot;Volcano Plot CX_0.5_48 (adj P-val&lt;0.05)&quot;),
  &quot;DOX_0.1_3&quot; = generate_volcano_plot(Toptable_DOX_0.1_3, &quot;Volcano Plot DOX_0.1_3 (adj P-val&lt;0.05)&quot;),
  &quot;DOX_0.1_24&quot; = generate_volcano_plot(Toptable_DOX_0.1_24, &quot;Volcano Plot DOX_0.1_24 (adj P-val&lt;0.05)&quot;),
  &quot;DOX_0.1_48&quot; = generate_volcano_plot(Toptable_DOX_0.1_48, &quot;Volcano Plot DOX_0.1_48 (adj P-val&lt;0.05)&quot;),
  &quot;DOX_0.5_3&quot; = generate_volcano_plot(Toptable_DOX_0.5_3, &quot;Volcano Plot DOX_0.5_3 (adj P-val&lt;0.05)&quot;),
  &quot;DOX_0.5_24&quot; = generate_volcano_plot(Toptable_DOX_0.5_24, &quot;Volcano Plot DOX_0.5_24 (adj P-val&lt;0.05)&quot;),
  &quot;DOX_0.5_48&quot; = generate_volcano_plot(Toptable_DOX_0.5_48, &quot;Volcano Plot DOX_0.5_48 (adj P-val&lt;0.05)&quot;)
)

# Display each volcano plot
for (plot_name in names(volcano_plots)) {
  print(volcano_plots[[plot_name]])
}</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-volcano-plot-1">
Past versions of "volcano plot-1.png"
</button>
</p>
<div id="fig-no-spaces-volcano-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/92099e7a5fa068d2357cb772c18c0e40b0953661/docs/figure/DGE_Analysis.Rmd/volcano plot-1.png" target="_blank">92099e7</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/9ba22e3aa55034dcd7d1f2f13b35956b8e87abbf/docs/figure/DGE_Analysis.Rmd/volcano plot-1.png" target="_blank">9ba22e3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DGE_Analysis.Rmd/volcano%20plot-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-volcano-plot-2">
Past versions of "volcano plot-2.png"
</button>
</p>
<div id="fig-no-spaces-volcano-plot-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/92099e7a5fa068d2357cb772c18c0e40b0953661/docs/figure/DGE_Analysis.Rmd/volcano plot-2.png" target="_blank">92099e7</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/9ba22e3aa55034dcd7d1f2f13b35956b8e87abbf/docs/figure/DGE_Analysis.Rmd/volcano plot-2.png" target="_blank">9ba22e3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DGE_Analysis.Rmd/volcano%20plot-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-volcano-plot-3">
Past versions of "volcano plot-3.png"
</button>
</p>
<div id="fig-no-spaces-volcano-plot-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/92099e7a5fa068d2357cb772c18c0e40b0953661/docs/figure/DGE_Analysis.Rmd/volcano plot-3.png" target="_blank">92099e7</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/9ba22e3aa55034dcd7d1f2f13b35956b8e87abbf/docs/figure/DGE_Analysis.Rmd/volcano plot-3.png" target="_blank">9ba22e3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DGE_Analysis.Rmd/volcano%20plot-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-volcano-plot-4">
Past versions of "volcano plot-4.png"
</button>
</p>
<div id="fig-no-spaces-volcano-plot-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/92099e7a5fa068d2357cb772c18c0e40b0953661/docs/figure/DGE_Analysis.Rmd/volcano plot-4.png" target="_blank">92099e7</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/9ba22e3aa55034dcd7d1f2f13b35956b8e87abbf/docs/figure/DGE_Analysis.Rmd/volcano plot-4.png" target="_blank">9ba22e3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DGE_Analysis.Rmd/volcano%20plot-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-volcano-plot-5">
Past versions of "volcano plot-5.png"
</button>
</p>
<div id="fig-no-spaces-volcano-plot-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/92099e7a5fa068d2357cb772c18c0e40b0953661/docs/figure/DGE_Analysis.Rmd/volcano plot-5.png" target="_blank">92099e7</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/9ba22e3aa55034dcd7d1f2f13b35956b8e87abbf/docs/figure/DGE_Analysis.Rmd/volcano plot-5.png" target="_blank">9ba22e3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Warning: Removed 3 rows containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/volcano%20plot-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-volcano-plot-6">
Past versions of "volcano plot-6.png"
</button>
</p>
<div id="fig-no-spaces-volcano-plot-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/92099e7a5fa068d2357cb772c18c0e40b0953661/docs/figure/DGE_Analysis.Rmd/volcano plot-6.png" target="_blank">92099e7</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/9ba22e3aa55034dcd7d1f2f13b35956b8e87abbf/docs/figure/DGE_Analysis.Rmd/volcano plot-6.png" target="_blank">9ba22e3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DGE_Analysis.Rmd/volcano%20plot-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-volcano-plot-7">
Past versions of "volcano plot-7.png"
</button>
</p>
<div id="fig-no-spaces-volcano-plot-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/92099e7a5fa068d2357cb772c18c0e40b0953661/docs/figure/DGE_Analysis.Rmd/volcano plot-7.png" target="_blank">92099e7</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/9ba22e3aa55034dcd7d1f2f13b35956b8e87abbf/docs/figure/DGE_Analysis.Rmd/volcano plot-7.png" target="_blank">9ba22e3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/volcano%20plot-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-volcano-plot-8">
Past versions of "volcano plot-8.png"
</button>
</p>
<div id="fig-no-spaces-volcano-plot-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/92099e7a5fa068d2357cb772c18c0e40b0953661/docs/figure/DGE_Analysis.Rmd/volcano plot-8.png" target="_blank">92099e7</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/9ba22e3aa55034dcd7d1f2f13b35956b8e87abbf/docs/figure/DGE_Analysis.Rmd/volcano plot-8.png" target="_blank">9ba22e3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Warning: Removed 34 rows containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/volcano%20plot-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-volcano-plot-9">
Past versions of "volcano plot-9.png"
</button>
</p>
<div id="fig-no-spaces-volcano-plot-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/92099e7a5fa068d2357cb772c18c0e40b0953661/docs/figure/DGE_Analysis.Rmd/volcano plot-9.png" target="_blank">92099e7</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/9ba22e3aa55034dcd7d1f2f13b35956b8e87abbf/docs/figure/DGE_Analysis.Rmd/volcano plot-9.png" target="_blank">9ba22e3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DGE_Analysis.Rmd/volcano%20plot-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-volcano-plot-10">
Past versions of "volcano plot-10.png"
</button>
</p>
<div id="fig-no-spaces-volcano-plot-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/92099e7a5fa068d2357cb772c18c0e40b0953661/docs/figure/DGE_Analysis.Rmd/volcano plot-10.png" target="_blank">92099e7</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/9ba22e3aa55034dcd7d1f2f13b35956b8e87abbf/docs/figure/DGE_Analysis.Rmd/volcano plot-10.png" target="_blank">9ba22e3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Warning: Removed 28 rows containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/volcano%20plot-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-volcano-plot-11">
Past versions of "volcano plot-11.png"
</button>
</p>
<div id="fig-no-spaces-volcano-plot-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/92099e7a5fa068d2357cb772c18c0e40b0953661/docs/figure/DGE_Analysis.Rmd/volcano plot-11.png" target="_blank">92099e7</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/9ba22e3aa55034dcd7d1f2f13b35956b8e87abbf/docs/figure/DGE_Analysis.Rmd/volcano plot-11.png" target="_blank">9ba22e3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Warning: Removed 24 rows containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/volcano%20plot-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-volcano-plot-12">
Past versions of "volcano plot-12.png"
</button>
</p>
<div id="fig-no-spaces-volcano-plot-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/92099e7a5fa068d2357cb772c18c0e40b0953661/docs/figure/DGE_Analysis.Rmd/volcano plot-12.png" target="_blank">92099e7</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/9ba22e3aa55034dcd7d1f2f13b35956b8e87abbf/docs/figure/DGE_Analysis.Rmd/volcano plot-12.png" target="_blank">9ba22e3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dna-damage-marker-expression-changes-between-cx5461-and-dox"
class="section level2">
<h2><strong>📌 DNA Damage Marker expression changes between CX5461 and
DOX</strong></h2>
<pre class="r"><code># 📌 DDR Gene Expression Heatmap — CX-5461 and DOX (68 genes, with categories)

# Load libraries
library(tidyverse)
library(ComplexHeatmap)
library(circlize)
library(grid)
library(org.Hs.eg.db)
library(reshape2)

# Load DEG files
load_deg &lt;- function(path) read.csv(path)

CX_0.1_3 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;)
CX_0.1_24 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;)
CX_0.1_48 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;)
CX_0.5_3 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)
DOX_0.1_3 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;)
DOX_0.1_24 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;)
DOX_0.1_48 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;)
DOX_0.5_3 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

# Final Entrez IDs and categories (68 genes)
entrez_category &lt;- tribble(
  ~ENTREZID, ~Category,
  317, &quot;Apoptosis&quot;, 355, &quot;Apoptosis&quot;, 581, &quot;Apoptosis&quot;, 637, &quot;Apoptosis&quot;,
  836, &quot;Apoptosis&quot;, 841, &quot;Apoptosis&quot;, 842, &quot;Apoptosis&quot;, 27113, &quot;Apoptosis&quot;,
  5366, &quot;Apoptosis&quot;, 54205, &quot;Apoptosis&quot;, 55367, &quot;Apoptosis&quot;, 8795, &quot;Apoptosis&quot;,
  1026, &quot;Cell Cycle / Checkpoint&quot;, 1027, &quot;Cell Cycle / Checkpoint&quot;, 595, &quot;Cell Cycle / Checkpoint&quot;,
  894, &quot;Cell Cycle / Checkpoint&quot;, 896, &quot;Cell Cycle / Checkpoint&quot;, 898, &quot;Cell Cycle / Checkpoint&quot;,
  9133, &quot;Cell Cycle / Checkpoint&quot;, 9134, &quot;Cell Cycle / Checkpoint&quot;, 891, &quot;Cell Cycle / Checkpoint&quot;,
  983, &quot;Cell Cycle / Checkpoint&quot;, 1017, &quot;Cell Cycle / Checkpoint&quot;, 1019, &quot;Cell Cycle / Checkpoint&quot;,
  1020, &quot;Cell Cycle / Checkpoint&quot;, 1021, &quot;Cell Cycle / Checkpoint&quot;, 993, &quot;Cell Cycle / Checkpoint&quot;,
  995, &quot;Cell Cycle / Checkpoint&quot;, 1869, &quot;Cell Cycle / Checkpoint&quot;, 4609, &quot;Cell Cycle / Checkpoint&quot;,
  5925, &quot;Cell Cycle / Checkpoint&quot;, 9874, &quot;Cell Cycle / Checkpoint&quot;, 11011, &quot;Cell Cycle / Checkpoint&quot;,
  1385, &quot;Cell Cycle / Checkpoint&quot;,
  472, &quot;Damage Sensors / Signal Transducers&quot;, 545, &quot;Damage Sensors / Signal Transducers&quot;,
  5591, &quot;Damage Sensors / Signal Transducers&quot;, 5810, &quot;Damage Sensors / Signal Transducers&quot;,
  5883, &quot;Damage Sensors / Signal Transducers&quot;, 5884, &quot;Damage Sensors / Signal Transducers&quot;,
  6118, &quot;Damage Sensors / Signal Transducers&quot;, 4361, &quot;Damage Sensors / Signal Transducers&quot;,
  10111, &quot;Damage Sensors / Signal Transducers&quot;, 4683, &quot;Damage Sensors / Signal Transducers&quot;,
  84126, &quot;Damage Sensors / Signal Transducers&quot;, 3014, &quot;Damage Sensors / Signal Transducers&quot;,
  672, &quot;DNA Repair&quot;, 2177, &quot;DNA Repair&quot;, 5888, &quot;DNA Repair&quot;, 5893, &quot;DNA Repair&quot;,
  1647, &quot;DNA Repair&quot;, 4616, &quot;DNA Repair&quot;, 10912, &quot;DNA Repair&quot;, 1111, &quot;DNA Repair&quot;,
  11200, &quot;DNA Repair&quot;, 1643, &quot;DNA Repair&quot;, 8243, &quot;DNA Repair&quot;, 5981, &quot;DNA Repair&quot;,
  7157, &quot;p53 Regulators / Targets&quot;, 4193, &quot;p53 Regulators / Targets&quot;, 5371, &quot;p53 Regulators / Targets&quot;,
  27244, &quot;p53 Regulators / Targets&quot;, 50484, &quot;p53 Regulators / Targets&quot;,
  5916, &quot;DOX Cardiotoxicity&quot;, 7799, &quot;DOX Cardiotoxicity&quot;, 4292, &quot;DOX Cardiotoxicity&quot;,
  207, &quot;Miscellaneous / Broad&quot;, 25, &quot;Miscellaneous / Broad&quot;
)

entrez_ids &lt;- entrez_category$ENTREZID

# Extract relevant DEG values
extract_data &lt;- function(df, name) {
  df %&gt;%
    filter(Entrez_ID %in% entrez_ids) %&gt;%
    mutate(
      Gene = mapIds(org.Hs.eg.db, as.character(Entrez_ID),
                    column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
      Condition = name,
      Signif = ifelse(adj.P.Val &lt; 0.05, &quot;*&quot;, &quot;&quot;)
    )
}

# DEG list
deg_list &lt;- list(
  &quot;CX_0.1_3&quot; = CX_0.1_3, &quot;CX_0.1_24&quot; = CX_0.1_24, &quot;CX_0.1_48&quot; = CX_0.1_48,
  &quot;CX_0.5_3&quot; = CX_0.5_3, &quot;CX_0.5_24&quot; = CX_0.5_24, &quot;CX_0.5_48&quot; = CX_0.5_48,
  &quot;DOX_0.1_3&quot; = DOX_0.1_3, &quot;DOX_0.1_24&quot; = DOX_0.1_24, &quot;DOX_0.1_48&quot; = DOX_0.1_48,
  &quot;DOX_0.5_3&quot; = DOX_0.5_3, &quot;DOX_0.5_24&quot; = DOX_0.5_24, &quot;DOX_0.5_48&quot; = DOX_0.5_48
)

# Combine all DEGs and annotate
all_data &lt;- bind_rows(mapply(extract_data, deg_list, names(deg_list), SIMPLIFY = FALSE)) %&gt;%
  left_join(entrez_category, by = c(&quot;Entrez_ID&quot; = &quot;ENTREZID&quot;))

# Create matrices
logFC_mat &lt;- acast(all_data, Gene ~ Condition, value.var = &quot;logFC&quot;)
signif_mat &lt;- acast(all_data, Gene ~ Condition, value.var = &quot;Signif&quot;)

# Set desired order
desired_order &lt;- c(&quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;,
                   &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
                   &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;,
                   &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;)

logFC_mat &lt;- logFC_mat[, desired_order, drop = FALSE]
signif_mat &lt;- signif_mat[, desired_order, drop = FALSE]

# Column annotation
meta &lt;- str_split_fixed(colnames(logFC_mat), &quot;_&quot;, 3)
col_annot &lt;- HeatmapAnnotation(
  Drug = meta[, 1],
  Conc = meta[, 2],
  Time = meta[, 3],
  col = list(
    Drug = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;),
    Conc = c(&quot;0.1&quot; = &quot;lightgreen&quot;, &quot;0.5&quot; = &quot;darkgreen&quot;),
    Time = c(&quot;3&quot; = &quot;yellow&quot;, &quot;24&quot; = &quot;orange&quot;, &quot;48&quot; = &quot;purple&quot;)
  ),
  annotation_height = unit(c(1, 1, 1), &quot;cm&quot;)
)

# Row annotation
gene_order_df &lt;- all_data %&gt;%
  distinct(Gene, Category) %&gt;%
  arrange(factor(Category, levels = sort(unique(entrez_category$Category))), Gene)

ordered_genes &lt;- gene_order_df$Gene
logFC_mat &lt;- logFC_mat[ordered_genes, ]
signif_mat &lt;- signif_mat[ordered_genes, ]

category_colors &lt;- structure(
  c(&quot;darkorange&quot;, &quot;steelblue&quot;, &quot;darkgreen&quot;, &quot;firebrick&quot;, &quot;gold&quot;, &quot;mediumpurple&quot;, &quot;gray60&quot;),
  names = sort(unique(entrez_category$Category))
)

ha_left &lt;- rowAnnotation(
  Category = gene_order_df$Category,
  col = list(Category = category_colors),
  annotation_name_side = &quot;top&quot;
)

# Final Heatmap
Heatmap(logFC_mat,
        name = &quot;logFC&quot;,
        top_annotation = col_annot,
        left_annotation = ha_left,
        cluster_columns = FALSE,
        cluster_rows = FALSE,
        show_row_names = TRUE,
        show_column_names = FALSE,
        row_names_gp = gpar(fontsize = 10),
        column_title = &quot;DDR Gene Expression Response (n = 68)\nCX-5461 and DOX&quot;,
        column_title_gp = gpar(fontsize = 14, fontface = &quot;bold&quot;),
        cell_fun = function(j, i, x, y, width, height, fill) {
          grid.text(signif_mat[i, j], x, y, gp = gpar(fontsize = 9))
        }
)</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/DNA-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-DNA-1">
Past versions of DNA-1.png
</button>
</p>
<div id="fig-DNA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/5a1db8113750a17ebafa6fec8521e12848f7e4fa/docs/figure/DGE_Analysis.Rmd/DNA-1.png" target="_blank">5a1db81</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3d33304d85bf2b6b87a3d324bc3a9d5e93720619/docs/figure/DGE_Analysis.Rmd/DNA-1.png" target="_blank">3d33304</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/72226f194b3d254ad4121b6ddbafec94f6adb811/docs/figure/DGE_Analysis.Rmd/DNA-1.png" target="_blank">72226f1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/92099e7a5fa068d2357cb772c18c0e40b0953661/docs/figure/DGE_Analysis.Rmd/DNA-1.png" target="_blank">92099e7</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="logfc-boxplots-of-dna-damage-response-genes"
class="section level2">
<h2><strong>📌 LogFC boxplots of DNA Damage Response Genes</strong></h2>
<pre class="r"><code># Load Libraries
library(tidyverse)
library(org.Hs.eg.db)

# DEG File Paths 
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

# DDR Entrez IDs
entrez_ids &lt;- c(
  10111, 1017, 1019, 1020, 1021, 1026, 1027, 10912, 11011, 1111,
  11200, 1385, 1643, 1647, 1869, 207, 2177, 25, 27113, 27244,
  3014, 317, 355, 4193, 4292, 4361, 4609, 4616, 4683, 472, 50484,
  5366, 5371, 54205, 545, 55367, 5591, 581, 5810, 5883, 5884,
  5888, 5893, 5925, 595, 5981, 6118, 637, 672, 7157, 7799,
  8243, 836, 841, 84126, 842, 8795, 891, 894, 896, 898,
  9133, 9134, 983, 9874, 993, 995, 5916
) %&gt;% as.character()

# Load and Filter DEGs
ddr_data_list &lt;- map2_dfr(deg_files, names(deg_files), function(file, label) {
  read_csv(file, show_col_types = FALSE) %&gt;%
    mutate(
      Entrez_ID = as.character(Entrez_ID),
      Condition = label,
      Drug = ifelse(str_detect(label, &quot;CX&quot;), &quot;CX.5461&quot;, &quot;DOX&quot;)
    ) %&gt;%
    filter(Entrez_ID %in% entrez_ids)
})

# Complete Combinations
all_conditions &lt;- names(deg_files)
all_combos &lt;- crossing(
  Entrez_ID = entrez_ids,
  Condition = all_conditions
) %&gt;%
  mutate(
    Drug = ifelse(str_detect(Condition, &quot;CX&quot;), &quot;CX.5461&quot;, &quot;DOX&quot;)
  )

# Merge with DEG Data 
complete_ddr &lt;- all_combos %&gt;%
  left_join(ddr_data_list, by = c(&quot;Entrez_ID&quot;, &quot;Condition&quot;, &quot;Drug&quot;)) %&gt;%
  mutate(
    logFC = ifelse(is.na(logFC), 0, logFC),
    adj.P.Val = ifelse(is.na(adj.P.Val), 1, adj.P.Val)
  )

#Map Gene Symbols
complete_ddr &lt;- complete_ddr %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;)
  )

# Order Condition Labels
complete_ddr$Condition &lt;- factor(complete_ddr$Condition, levels = c(
  &quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;,
  &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
  &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;,
  &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;
))

#Wilcoxon Test: CX vs DOX Per Condition 

condition_pairs &lt;- tibble(
  cx = c(&quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;, &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;),
  dox = c(&quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;, &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;)
)

wilcox_results &lt;- map2_dfr(condition_pairs$cx, condition_pairs$dox, function(cx_label, dox_label) {
  cx_vals &lt;- complete_ddr %&gt;% filter(Condition == cx_label) %&gt;% pull(logFC)
  dox_vals &lt;- complete_ddr %&gt;% filter(Condition == dox_label) %&gt;% pull(logFC)
  
  test &lt;- tryCatch(wilcox.test(cx_vals, dox_vals), error = function(e) NULL)
  pval &lt;- if (!is.null(test)) test$p.value else NA
  
  tibble(
    Condition = dox_label,
    p_value = signif(pval, 3),
    label = case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01  ~ &quot;**&quot;,
      pval &lt; 0.05  ~ &quot;*&quot;,
      TRUE         ~ &quot;&quot;
    ),
    y_pos = max(c(cx_vals, dox_vals), na.rm = TRUE) + 0.5
  )
})

# Final Boxplot
ggplot(complete_ddr, aes(x = Condition, y = logFC, fill = Drug)) +
  geom_boxplot(outlier.size = 0.6) +
  geom_text(data = wilcox_results,
            aes(x = Condition, y = y_pos, label = label),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;LogFC of DNA Damage Response Genes&quot;,
    x = &quot;Condition&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text.x = element_text(angle = 45, hjust = 1, size = 10),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/DDR-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-DDR-1">
Past versions of DDR-1.png
</button>
</p>
<div id="fig-DDR-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/92099e7a5fa068d2357cb772c18c0e40b0953661/docs/figure/DGE_Analysis.Rmd/DDR-1.png" target="_blank">92099e7</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="p53-target-gene-expression-changes-between-cx5461-and-dox"
class="section level2">
<h2><strong>📌 P53 target gene expression changes between CX5461 and
DOX</strong></h2>
<pre class="r"><code># Load libraries
library(tidyverse)
library(ComplexHeatmap)
library(circlize)
library(grid)
library(org.Hs.eg.db)
library(reshape2)

# Load DEG files
load_deg &lt;- function(path) read.csv(path)

CX_0.1_3 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;)
CX_0.1_24 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;)
CX_0.1_48 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;)
CX_0.5_3 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)
DOX_0.1_3 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;)
DOX_0.1_24 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;)
DOX_0.1_48 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;)
DOX_0.5_3 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

# P53 target Entrez IDs (shortened for clarity – replace with your full list)
entrez_ids &lt;- c(1026,50484,4193,9766,9518,7832,1643,1647,1263,57103,51065,8795,51499,64393,581,
                5228,5429,8493,55959,7508,64782,282991,355,53836,4814,10769,9050,27244,9540,94241,
                26154,57763,900,26999,55332,26263,23479,23612,29950,9618,10346,8824,134147,55294,
                22824,4254,6560,467,27113,60492,8444,60401,1969,220965,2232,3976,55191,84284,93129,
                5564,7803,83667,7779,132671,7039,51768,137695,93134,7633,10973,340485,307,27350,
                23245,3732,29965,1363,1435,196513,8507,8061,2517,51278,53354,54858,23228,5366,5912,
                6236,51222,26152,59,1907,50650,91012,780,9249,11072,144455,64787,116151,27165,2876,
                57822,55733,57722,121457,375449,85377,4851,5875,127544,29901,84958,8797,8793,441631,
                220001,54541,5889,5054,25816,25987,5111,98,317,598,604,10904,1294,80315,53944,
                1606,2770,3628,3675,3985,4035,4163,84552,29085,55367,5371,5791,54884,5980,8794,
                1462,50808,220,583,694,1056,9076,10978,54677,1612,55040,114907,2274,127707,4000,
                8079,4646,4747,27445,5143,80055,79156,5360,5364,23654,5565,5613,5625,10076,56963,
                6004,390,255488,6326,6330,23513,7869,283130,204962,83959,6548,6774,9263,10228,
                22954,10475,85363,494514,10142,79714,1006,8446,9648,79828,5507,55240,63874,25841,
                9289,84883,154810,51321,421,8553,655,119032,84280,10950,824,839,57828,857,8812,
                8837,94027,113189,22837,132864,10898,3300,81704,1847,1849,1947,9538,24139,5168,
                147965,115548,9873,23768,2632,2817,3280,3265,23308,3490,51477,182,3856,8844,144811,
                9404,4043,9848,2872,23041,740,343263,4638,26509,4792,22861,57523,55214,80025,164091,
                57060,64065,51090,5453,8496,333926,55671,5900,55544,23179,8601,389,6223,55800,6385,
                4088,6643,122809,257397,285343,7011,54790,374618,55362,51754,7157,9537,22906,7205,
                80705,219699,55245,83719,7748,25946,118738)

# Function to extract relevant data
extract_data &lt;- function(df, name) {
  df %&gt;%
    filter(Entrez_ID %in% entrez_ids) %&gt;%
    mutate(Gene = mapIds(org.Hs.eg.db, as.character(Entrez_ID),
                         column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
           Condition = name,
           Signif = ifelse(adj.P.Val &lt; 0.05, &quot;*&quot;, &quot;&quot;))
}

# Collect all data
deg_list &lt;- list(
  &quot;CX_0.1_3&quot; = CX_0.1_3, &quot;CX_0.1_24&quot; = CX_0.1_24, &quot;CX_0.1_48&quot; = CX_0.1_48,
  &quot;CX_0.5_3&quot; = CX_0.5_3, &quot;CX_0.5_24&quot; = CX_0.5_24, &quot;CX_0.5_48&quot; = CX_0.5_48,
  &quot;DOX_0.1_3&quot; = DOX_0.1_3, &quot;DOX_0.1_24&quot; = DOX_0.1_24, &quot;DOX_0.1_48&quot; = DOX_0.1_48,
  &quot;DOX_0.5_3&quot; = DOX_0.5_3, &quot;DOX_0.5_24&quot; = DOX_0.5_24, &quot;DOX_0.5_48&quot; = DOX_0.5_48
)

all_data &lt;- bind_rows(mapply(extract_data, deg_list, names(deg_list), SIMPLIFY = FALSE))

# Create matrices
logFC_mat &lt;- acast(all_data, Gene ~ Condition, value.var = &quot;logFC&quot;)
signif_mat &lt;- acast(all_data, Gene ~ Condition, value.var = &quot;Signif&quot;)

# Desired column order
desired_order &lt;- c(&quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;, 
                   &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
                   &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;, 
                   &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;)

logFC_mat &lt;- logFC_mat[, desired_order]
signif_mat &lt;- signif_mat[, desired_order]

# Column annotation
meta &lt;- str_split_fixed(colnames(logFC_mat), &quot;_&quot;, 3)
col_annot &lt;- HeatmapAnnotation(
  Drug = meta[, 1],
  Conc = meta[, 2],
  Time = meta[, 3],
  col = list(
    Drug = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;),
    Conc = c(&quot;0.1&quot; = &quot;lightgreen&quot;, &quot;0.5&quot; = &quot;darkgreen&quot;),
    Time = c(&quot;3&quot; = &quot;yellow&quot;, &quot;24&quot; = &quot;orange&quot;, &quot;48&quot; = &quot;purple&quot;)
  ),
  annotation_height = unit(c(2, 2, 2), &quot;cm&quot;)
)

# Draw heatmap
Heatmap(logFC_mat,
        name = &quot;logFC&quot;,
        top_annotation = col_annot,
        cluster_columns = FALSE,
        cluster_rows = TRUE,
        show_row_names = TRUE,
        show_column_names = FALSE,
        cell_fun = function(j, i, x, y, width, height, fill) {
          grid.text(signif_mat[i, j], x, y, gp = gpar(fontsize = 9))
        },
        column_title = &quot;P53 Target Genes\nCX-5461 and DOX Expression&quot;,
        column_title_gp = gpar(fontsize = 14, fontface = &quot;bold&quot;)
)</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/P53_target-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-P53_target-1">
Past versions of P53_target-1.png
</button>
</p>
<div id="fig-P53_target-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/92099e7a5fa068d2357cb772c18c0e40b0953661/docs/figure/DGE_Analysis.Rmd/P53_target-1.png" target="_blank">92099e7</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="logfc-boxplots-of-p53-target-genes" class="section level2">
<h2><strong>📌 LogFC boxplots of P53 target Genes</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)

# ----------------- DEG File Paths -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

# ----------------- P53 Target Entrez IDs -----------------
entrez_ids &lt;- c(
  1026, 50484, 4193, 9766, 9518, 7832, 1643, 1647, 1263, 57103, 51065, 8795, 51499, 64393, 581, 5228, 5429, 8493, 55959,
  7508, 64782, 282991, 355, 53836, 4814, 10769, 9050, 27244, 9540, 94241, 26154, 57763, 900, 26999, 55332, 26263, 23479,
  23612, 29950, 9618, 10346, 8824, 134147, 55294, 22824, 4254, 6560, 467, 27113, 60492, 8444, 60401, 1969, 220965, 2232,
  3976, 55191, 84284, 93129, 5564, 7803, 83667, 7779, 132671, 7039, 51768, 137695, 93134, 7633, 10973, 340485, 307,
  27350, 23245, 3732, 29965, 1363, 1435, 196513, 8507, 8061, 2517, 51278, 53354, 54858, 23228, 5366, 5912, 6236, 51222,
  26152, 59, 1907, 50650, 91012, 780, 9249, 11072, 144455, 64787, 116151, 27165, 2876, 57822, 55733, 57722, 121457,
  375449, 85377, 4851, 5875, 127544, 29901, 84958, 8797, 8793, 441631, 220001, 54541, 5889, 5054, 25816, 25987, 5111,
  98, 317, 598, 604, 10904, 1294, 80315, 53944, 1606, 2770, 3628, 3675, 3985, 4035, 4163, 84552, 29085, 55367, 5371,
  5791, 54884, 5980, 8794, 1462, 50808, 220, 583, 694, 1056, 9076, 10978, 54677, 1612, 55040, 114907, 2274, 127707,
  4000, 8079, 4646, 4747, 27445, 5143, 80055, 79156, 5360, 5364, 23654, 5565, 5613, 5625, 10076, 56963, 6004, 390,
  255488, 6326, 6330, 23513, 7869, 283130, 204962, 83959, 6548, 6774, 9263, 10228, 22954, 10475, 85363, 494514, 10142,
  79714, 1006, 8446, 9648, 79828, 5507, 55240, 63874, 25841, 9289, 84883, 154810, 51321, 421, 8553, 655, 119032, 84280,
  10950, 824, 839, 57828, 857, 8812, 8837, 94027, 113189, 22837, 132864, 10898, 3300, 81704, 1847, 1849, 1947, 9538,
  24139, 5168, 147965, 115548, 9873, 23768, 2632, 2817, 3280, 3265, 23308, 3490, 51477, 182, 3856, 8844, 144811, 9404,
  4043, 9848, 2872, 23041, 740, 343263, 4638, 26509, 4792, 22861, 57523, 55214, 80025, 164091, 57060, 64065, 51090,
  5453, 8496, 333926, 55671, 5900, 55544, 23179, 8601, 389, 6223, 55800, 6385, 4088, 6643, 122809, 257397, 285343,
  7011, 54790, 374618, 55362, 51754, 7157, 9537, 22906, 7205, 80705, 219699, 55245, 83719, 7748, 25946, 118738
) %&gt;% as.character()

# ----------------- Load and Filter DEGs -----------------
p53_data_list &lt;- map2_dfr(deg_files, names(deg_files), function(file, label) {
  read_csv(file, show_col_types = FALSE) %&gt;%
    mutate(
      Entrez_ID = as.character(Entrez_ID),
      Condition = label,
      Drug = ifelse(str_detect(label, &quot;CX&quot;), &quot;CX.5461&quot;, &quot;DOX&quot;)
    ) %&gt;%
    filter(Entrez_ID %in% entrez_ids)
})

# ----------------- Complete Combinations -----------------
all_conditions &lt;- names(deg_files)
all_combos &lt;- crossing(
  Entrez_ID = entrez_ids,
  Condition = all_conditions
) %&gt;%
  mutate(
    Drug = ifelse(str_detect(Condition, &quot;CX&quot;), &quot;CX.5461&quot;, &quot;DOX&quot;)
  )

# ----------------- Merge with DEG Data -----------------
complete_p53 &lt;- all_combos %&gt;%
  left_join(p53_data_list, by = c(&quot;Entrez_ID&quot;, &quot;Condition&quot;, &quot;Drug&quot;)) %&gt;%
  mutate(
    logFC = ifelse(is.na(logFC), 0, logFC),
    adj.P.Val = ifelse(is.na(adj.P.Val), 1, adj.P.Val)
  )

# ----------------- Map Gene Symbols -----------------
complete_p53 &lt;- complete_p53 %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;)
  )

# ----------------- Order Condition Labels -----------------
complete_p53$Condition &lt;- factor(complete_p53$Condition, levels = c(
  &quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;,
  &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
  &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;,
  &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;
))

# ----------------- Wilcoxon Tests: CX vs DOX -----------------
condition_pairs &lt;- tibble(
  cx = c(&quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;, &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;),
  dox = c(&quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;, &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;)
)

wilcox_results &lt;- map2_dfr(condition_pairs$cx, condition_pairs$dox, function(cx_label, dox_label) {
  cx_vals &lt;- complete_p53 %&gt;% filter(Condition == cx_label) %&gt;% pull(logFC)
  dox_vals &lt;- complete_p53 %&gt;% filter(Condition == dox_label) %&gt;% pull(logFC)
  
  test &lt;- tryCatch(wilcox.test(cx_vals, dox_vals), error = function(e) NULL)
  pval &lt;- if (!is.null(test)) test$p.value else NA
  
  tibble(
    Condition = dox_label,
    p_value = signif(pval, 3),
    label = case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01  ~ &quot;**&quot;,
      pval &lt; 0.05  ~ &quot;*&quot;,
      TRUE         ~ &quot;&quot;
    ),
    y_pos = max(c(cx_vals, dox_vals), na.rm = TRUE) + 0.5
  )
})

# ----------------- Final Plot -----------------
ggplot(complete_p53, aes(x = Condition, y = logFC, fill = Drug)) +
  geom_boxplot(outlier.size = 0.6) +
  geom_text(data = wilcox_results,
            aes(x = Condition, y = y_pos, label = label),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;LogFC of P53 Target Genes&quot;,
    x = &quot;Condition&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text.x = element_text(angle = 45, hjust = 1, size = 10),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/P53_target_box-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-P53_target_box-1">
Past versions of P53_target_box-1.png
</button>
</p>
<div id="fig-P53_target_box-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/43e348d0bc195290fd54602de2e15faafc29c815/docs/figure/DGE_Analysis.Rmd/P53_target_box-1.png" target="_blank">43e348d</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="logfc-boxplots-of-top2b-target-genes" class="section level2">
<h2><strong>📌 LogFC boxplots of TOP2B target Genes</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)

# ----------------- DEG File Paths -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

# ----------------- TOP2B Entrez IDs -----------------
top2b_entrez_ids &lt;- read_csv(&quot;data/TOP2B_target_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load &amp; Combine DEG Data -----------------
top2b_data_list &lt;- map2_dfr(deg_files, names(deg_files), function(file, label) {
  read_csv(file, show_col_types = FALSE) %&gt;%
    mutate(Entrez_ID = as.character(Entrez_ID)) %&gt;%  # FIX: Ensure type match
    filter(Entrez_ID %in% top2b_entrez_ids) %&gt;%
    mutate(
      Condition = label,
      Drug = ifelse(str_detect(label, &quot;CX&quot;), &quot;CX.5461&quot;, &quot;DOX&quot;)
    )
})

# ----------------- Create All Gene × Condition Combos -----------------
all_conditions &lt;- names(deg_files)
all_combos &lt;- crossing(
  Entrez_ID = top2b_entrez_ids,
  Condition = all_conditions
) %&gt;%
  mutate(
    Drug = ifelse(str_detect(Condition, &quot;CX&quot;), &quot;CX.5461&quot;, &quot;DOX&quot;)
  )

# ----------------- Merge &amp; Impute logFC -----------------
complete_top2b &lt;- all_combos %&gt;%
  left_join(top2b_data_list, by = c(&quot;Entrez_ID&quot;, &quot;Condition&quot;, &quot;Drug&quot;)) %&gt;%
  mutate(
    logFC = ifelse(is.na(logFC), 0, logFC),
    adj.P.Val = ifelse(is.na(adj.P.Val), 1, adj.P.Val),
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;)
  )

# ----------------- Order X-axis Conditions -----------------
complete_top2b$Condition &lt;- factor(complete_top2b$Condition, levels = c(
  &quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;,
  &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
  &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;,
  &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;
))

# ----------------- Wilcoxon Test: CX vs DOX per condition -----------------
condition_pairs &lt;- tibble(
  cx = c(&quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;, &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;),
  dox = c(&quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;, &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;)
)

wilcox_results &lt;- map2_dfr(condition_pairs$cx, condition_pairs$dox, function(cx_label, dox_label) {
  cx_vals &lt;- complete_top2b %&gt;% filter(Condition == cx_label) %&gt;% pull(logFC)
  dox_vals &lt;- complete_top2b %&gt;% filter(Condition == dox_label) %&gt;% pull(logFC)
  
  test &lt;- tryCatch(wilcox.test(cx_vals, dox_vals), error = function(e) NULL)
  pval &lt;- if (!is.null(test)) test$p.value else NA
  
  tibble(
    Condition = dox_label,
    p_value = signif(pval, 3),
    label = case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01  ~ &quot;**&quot;,
      pval &lt; 0.05  ~ &quot;*&quot;,
      TRUE         ~ &quot;&quot;
    ),
    y_pos = max(c(cx_vals, dox_vals), na.rm = TRUE) + 0.5
  )
})

# ----------------- Final Boxplot with Wilcoxon Stars -----------------
ggplot(complete_top2b, aes(x = Condition, y = logFC, fill = Drug)) +
  geom_boxplot(outlier.size = 0.6) +
  geom_text(data = wilcox_results,
            aes(x = Condition, y = y_pos, label = label),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;LogFC of TOP2B Target Genes&quot;,
    x = &quot;Treatment Condition&quot;,
    y = &quot;Log Fold Change&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text.x = element_text(angle = 45, hjust = 1, size = 10),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/TOP2B_target_box-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-TOP2B_target_box-1">
Past versions of TOP2B_target_box-1.png
</button>
</p>
<div id="fig-TOP2B_target_box-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/711098f70cf35739ae12795badc60904e4505b4b/docs/figure/DGE_Analysis.Rmd/TOP2B_target_box-1.png" target="_blank">711098f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/fdcc3300b8cc1c2d328633dd483aabd9364227ea/docs/figure/DGE_Analysis.Rmd/TOP2B_target_box-1.png" target="_blank">fdcc330</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="top2b-target-gene-expression-changes-between-cx5461-and-dox-ipsc-cm-chip-seq"
class="section level2">
<h2><strong>📌 TOP2B target gene expression changes between CX5461 and
DOX (ipSC-CM ChIP Seq)</strong></h2>
<pre class="r"><code># Load libraries
library(tidyverse)
library(ComplexHeatmap)
library(circlize)
library(grid)
library(org.Hs.eg.db)
library(reshape2)

# Load DEG files
load_deg &lt;- function(path) read.csv(path)

CX_0.1_3 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;)
CX_0.1_24 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;)
CX_0.1_48 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;)
CX_0.5_3 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)
DOX_0.1_3 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;)
DOX_0.1_24 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;)
DOX_0.1_48 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;)
DOX_0.5_3 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

# TOP2B target Entrez IDs
entrez_ids &lt;- c(
  8365, 8367, 8334, 79171, 1876, 2802, 3012, 10024, 85450, 6945,
  100288069, 51367, 8665, 9441, 148304, 51433, 29882, 84893, 4751, 1662,
  9601, 2872, 255812, 1871, 10523, 23435, 8896, 5566, 65095, 388
)

# Function to extract relevant data
extract_data &lt;- function(df, name) {
  df %&gt;%
    filter(Entrez_ID %in% entrez_ids) %&gt;%
    mutate(
      Gene = mapIds(org.Hs.eg.db, as.character(Entrez_ID),
                    column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
      Condition = name,
      Signif = ifelse(adj.P.Val &lt; 0.05, &quot;*&quot;, &quot;&quot;)
    )
}

# Combine DEG data
deg_list &lt;- list(
  &quot;CX_0.1_3&quot; = CX_0.1_3, &quot;CX_0.1_24&quot; = CX_0.1_24, &quot;CX_0.1_48&quot; = CX_0.1_48,
  &quot;CX_0.5_3&quot; = CX_0.5_3, &quot;CX_0.5_24&quot; = CX_0.5_24, &quot;CX_0.5_48&quot; = CX_0.5_48,
  &quot;DOX_0.1_3&quot; = DOX_0.1_3, &quot;DOX_0.1_24&quot; = DOX_0.1_24, &quot;DOX_0.1_48&quot; = DOX_0.1_48,
  &quot;DOX_0.5_3&quot; = DOX_0.5_3, &quot;DOX_0.5_24&quot; = DOX_0.5_24, &quot;DOX_0.5_48&quot; = DOX_0.5_48
)

all_data &lt;- bind_rows(mapply(extract_data, deg_list, names(deg_list), SIMPLIFY = FALSE))

# Create logFC and significance matrices
logFC_mat &lt;- acast(all_data, Gene ~ Condition, value.var = &quot;logFC&quot;)
signif_mat &lt;- acast(all_data, Gene ~ Condition, value.var = &quot;Signif&quot;)

# Desired condition order
desired_order &lt;- c(&quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;,
                   &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
                   &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;,
                   &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;)

logFC_mat &lt;- logFC_mat[, desired_order, drop = FALSE]
signif_mat &lt;- signif_mat[, desired_order, drop = FALSE]

# Top annotation
meta &lt;- str_split_fixed(colnames(logFC_mat), &quot;_&quot;, 3)
col_annot &lt;- HeatmapAnnotation(
  Drug = meta[, 1],
  Conc = meta[, 2],
  Time = meta[, 3],
  col = list(
    Drug = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;),
    Conc = c(&quot;0.1&quot; = &quot;lightgreen&quot;, &quot;0.5&quot; = &quot;darkgreen&quot;),
    Time = c(&quot;3&quot; = &quot;yellow&quot;, &quot;24&quot; = &quot;orange&quot;, &quot;48&quot; = &quot;purple&quot;)
  ),
  annotation_height = unit(c(2, 2, 2), &quot;cm&quot;)
)

# Plot heatmap
Heatmap(logFC_mat,
        name = &quot;logFC&quot;,
        top_annotation = col_annot,
        cluster_columns = FALSE,
        cluster_rows = TRUE,
        show_row_names = TRUE,
        show_column_names = FALSE,
        cell_fun = function(j, i, x, y, width, height, fill) {
          grid.text(signif_mat[i, j], x, y, gp = gpar(fontsize = 9))
        },
        column_title = &quot;TOP2B Target Genes\nCX-5461 and DOX Expression&quot;,
        column_title_gp = gpar(fontsize = 14, fontface = &quot;bold&quot;)
)</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/TOP2B_target_heatmap-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-TOP2B_target_heatmap-1">
Past versions of TOP2B_target_heatmap-1.png
</button>
</p>
<div id="fig-TOP2B_target_heatmap-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/328c60f4df9618213e2b4ce7339b3810e00c835e/docs/figure/DGE_Analysis.Rmd/TOP2B_target_heatmap-1.png" target="_blank">328c60f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="logfc-boxplots-of-top2a-target-genes-chip-atlas"
class="section level2">
<h2><strong>📌 LogFC boxplots of TOP2A target Genes (ChIP
Atlas)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)

# ----------------- DEG File Paths -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

# ----------------- Load TOP2A Entrez IDs -----------------
top2a_entrez_ids &lt;- read_csv(&quot;data/TOP2A_target_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load &amp; Filter DEGs -----------------
top2a_data_list &lt;- map2_dfr(deg_files, names(deg_files), function(file, label) {
  read_csv(file, show_col_types = FALSE) %&gt;%
    mutate(Entrez_ID = as.character(Entrez_ID)) %&gt;%
    filter(Entrez_ID %in% top2a_entrez_ids) %&gt;%
    mutate(
      Condition = label,
      Drug = ifelse(str_detect(label, &quot;CX&quot;), &quot;CX.5461&quot;, &quot;DOX&quot;)
    )
})

# ----------------- Create All Gene × Condition Combos -----------------
all_conditions &lt;- names(deg_files)
all_combos &lt;- crossing(
  Entrez_ID = top2a_entrez_ids,
  Condition = all_conditions
) %&gt;%
  mutate(
    Drug = ifelse(str_detect(Condition, &quot;CX&quot;), &quot;CX.5461&quot;, &quot;DOX&quot;)
  )

# ----------------- Merge &amp; Impute logFC -----------------
complete_top2a &lt;- all_combos %&gt;%
  left_join(top2a_data_list, by = c(&quot;Entrez_ID&quot;, &quot;Condition&quot;, &quot;Drug&quot;)) %&gt;%
  mutate(
    logFC = ifelse(is.na(logFC), 0, logFC),
    adj.P.Val = ifelse(is.na(adj.P.Val), 1, adj.P.Val),
    Gene = mapIds(org.Hs.eg.db, keys = as.character(Entrez_ID),
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;)
  )

# ----------------- Order X-axis Conditions -----------------
complete_top2a$Condition &lt;- factor(complete_top2a$Condition, levels = c(
  &quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;,
  &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
  &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;,
  &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;
))

# ----------------- Wilcoxon Test: CX vs DOX per condition -----------------
condition_pairs &lt;- tibble(
  cx = c(&quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;, &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;),
  dox = c(&quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;, &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;)
)

wilcox_results &lt;- map2_dfr(condition_pairs$cx, condition_pairs$dox, function(cx_label, dox_label) {
  cx_vals &lt;- complete_top2a %&gt;% filter(Condition == cx_label) %&gt;% pull(logFC)
  dox_vals &lt;- complete_top2a %&gt;% filter(Condition == dox_label) %&gt;% pull(logFC)
  
  test &lt;- tryCatch(wilcox.test(cx_vals, dox_vals), error = function(e) NULL)
  pval &lt;- if (!is.null(test)) test$p.value else NA
  
  tibble(
    Condition = dox_label,
    p_value = signif(pval, 3),
    label = case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01 ~ &quot;**&quot;,
      pval &lt; 0.05 ~ &quot;*&quot;,
      TRUE ~ &quot;&quot;
    ),
    y_pos = max(c(cx_vals, dox_vals), na.rm = TRUE) + 0.5
  )
})

# ----------------- Final Boxplot -----------------
ggplot(complete_top2a, aes(x = Condition, y = logFC, fill = Drug)) +
  geom_boxplot(outlier.size = 0.6) +
  geom_text(data = wilcox_results,
            aes(x = Condition, y = y_pos, label = label),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;LogFC of TOP2A Target Genes (ChIP Atlas)&quot;,
    x = &quot;Treatment Condition&quot;,
    y = &quot;Log Fold Change&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text.x = element_text(angle = 45, hjust = 1, size = 10),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/TOP2A_target_box-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-TOP2A_target_box-1">
Past versions of TOP2A_target_box-1.png
</button>
</p>
<div id="fig-TOP2A_target_box-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/711098f70cf35739ae12795badc60904e4505b4b/docs/figure/DGE_Analysis.Rmd/TOP2A_target_box-1.png" target="_blank">711098f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="top2a-target-gene-expression-changes-between-cx5461-and-dox-chip_atlas"
class="section level2">
<h2><strong>📌 TOP2A target gene expression changes between CX5461 and
DOX (ChIP_Atlas)</strong></h2>
<pre class="r"><code># Load libraries
library(tidyverse)
library(ComplexHeatmap)
library(circlize)
library(grid)
library(org.Hs.eg.db)
library(reshape2)

# Load DEG files
load_deg &lt;- function(path) read.csv(path)

CX_0.1_3 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;)
CX_0.1_24 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;)
CX_0.1_48 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;)
CX_0.5_3 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- load_deg(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)
DOX_0.1_3 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;)
DOX_0.1_24 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;)
DOX_0.1_48 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;)
DOX_0.5_3 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- load_deg(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

# TOP2A ChIP Atlas Entrez IDs
entrez_ids &lt;- c(
  9856, 100462981, 4437, 1719, 256691, 100462977, 100463486, 91522, 8214, 51567,
  55856, 3800, 84898, 729857, 400966, 56999, 7317, 729515, 55684, 84960, 7337,
  64901, 140735, 23443, 100463487, 79643, 2976, 90522, 9424, 23189
)

# Extract relevant data
extract_data &lt;- function(df, name) {
  df %&gt;%
    filter(Entrez_ID %in% entrez_ids) %&gt;%
    mutate(
      Gene = mapIds(org.Hs.eg.db, as.character(Entrez_ID),
                    column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
      Condition = name,
      Signif = ifelse(adj.P.Val &lt; 0.05, &quot;*&quot;, &quot;&quot;)
    )
}

# Combine DEG data
deg_list &lt;- list(
  &quot;CX_0.1_3&quot; = CX_0.1_3, &quot;CX_0.1_24&quot; = CX_0.1_24, &quot;CX_0.1_48&quot; = CX_0.1_48,
  &quot;CX_0.5_3&quot; = CX_0.5_3, &quot;CX_0.5_24&quot; = CX_0.5_24, &quot;CX_0.5_48&quot; = CX_0.5_48,
  &quot;DOX_0.1_3&quot; = DOX_0.1_3, &quot;DOX_0.1_24&quot; = DOX_0.1_24, &quot;DOX_0.1_48&quot; = DOX_0.1_48,
  &quot;DOX_0.5_3&quot; = DOX_0.5_3, &quot;DOX_0.5_24&quot; = DOX_0.5_24, &quot;DOX_0.5_48&quot; = DOX_0.5_48
)

all_data &lt;- bind_rows(mapply(extract_data, deg_list, names(deg_list), SIMPLIFY = FALSE))

# Create logFC and significance matrices
logFC_mat &lt;- acast(all_data, Gene ~ Condition, value.var = &quot;logFC&quot;)
signif_mat &lt;- acast(all_data, Gene ~ Condition, value.var = &quot;Signif&quot;)

# Desired condition order
desired_order &lt;- c(&quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;,
                   &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
                   &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;,
                   &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;)

logFC_mat &lt;- logFC_mat[, desired_order, drop = FALSE]
signif_mat &lt;- signif_mat[, desired_order, drop = FALSE]

# Column annotation
meta &lt;- str_split_fixed(colnames(logFC_mat), &quot;_&quot;, 3)
col_annot &lt;- HeatmapAnnotation(
  Drug = meta[, 1],
  Conc = meta[, 2],
  Time = meta[, 3],
  col = list(
    Drug = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;),
    Conc = c(&quot;0.1&quot; = &quot;lightgreen&quot;, &quot;0.5&quot; = &quot;darkgreen&quot;),
    Time = c(&quot;3&quot; = &quot;yellow&quot;, &quot;24&quot; = &quot;orange&quot;, &quot;48&quot; = &quot;purple&quot;)
  ),
  annotation_height = unit(c(2, 2, 2), &quot;cm&quot;)
)

# Draw heatmap
Heatmap(logFC_mat,
        name = &quot;logFC&quot;,
        top_annotation = col_annot,
        cluster_columns = FALSE,
        cluster_rows = TRUE,
        show_row_names = TRUE,
        show_column_names = FALSE,
        cell_fun = function(j, i, x, y, width, height, fill) {
          grid.text(signif_mat[i, j], x, y, gp = gpar(fontsize = 9))
        },
        column_title = &quot;TOP2A Target Genes (ChIP Atlas)\nCX-5461 and DOX Expression&quot;,
        column_title_gp = gpar(fontsize = 14, fontface = &quot;bold&quot;)
)</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/TOP2A_target_heatmap-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-TOP2A_target_heatmap-1">
Past versions of TOP2A_target_heatmap-1.png
</button>
</p>
<div id="fig-TOP2A_target_heatmap-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/328c60f4df9618213e2b4ce7339b3810e00c835e/docs/figure/DGE_Analysis.Rmd/TOP2A_target_heatmap-1.png" target="_blank">328c60f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="logfc-boxplots-of-top2a-target-genes-rpe-1-cell-line"
class="section level2">
<h2><strong>📌 LogFC boxplots of TOP2A target Genes (RPE-1 cell
line)</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)

# ----------------- DEG File Paths -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

# ----------------- Load TOP2A (RPE-1) Entrez IDs -----------------
top2a_lit_ids &lt;- read_csv(&quot;data/TOP2A_target_lit_mapped.csv&quot;, show_col_types = FALSE) %&gt;%
  pull(Entrez_ID) %&gt;%
  unique() %&gt;%
  as.character()

# ----------------- Load &amp; Filter DEGs -----------------
top2a_lit_data &lt;- map2_dfr(deg_files, names(deg_files), function(file, label) {
  read_csv(file, show_col_types = FALSE) %&gt;%
    mutate(Entrez_ID = as.character(Entrez_ID)) %&gt;%
    filter(Entrez_ID %in% top2a_lit_ids) %&gt;%
    mutate(
      Condition = label,
      Drug = ifelse(str_detect(label, &quot;CX&quot;), &quot;CX.5461&quot;, &quot;DOX&quot;)
    )
})

# ----------------- Create All Gene × Condition Combos -----------------
all_conditions &lt;- names(deg_files)
all_combos &lt;- crossing(
  Entrez_ID = top2a_lit_ids,
  Condition = all_conditions
) %&gt;%
  mutate(
    Drug = ifelse(str_detect(Condition, &quot;CX&quot;), &quot;CX.5461&quot;, &quot;DOX&quot;)
  )

# ----------------- Merge &amp; Impute logFC -----------------
complete_top2a_lit &lt;- all_combos %&gt;%
  left_join(top2a_lit_data, by = c(&quot;Entrez_ID&quot;, &quot;Condition&quot;, &quot;Drug&quot;)) %&gt;%
  mutate(
    logFC = ifelse(is.na(logFC), 0, logFC),
    adj.P.Val = ifelse(is.na(adj.P.Val), 1, adj.P.Val),
    Gene = mapIds(org.Hs.eg.db, keys = as.character(Entrez_ID),
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;)
  )

# ----------------- Order Conditions -----------------
complete_top2a_lit$Condition &lt;- factor(complete_top2a_lit$Condition, levels = c(
  &quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;,
  &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
  &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;,
  &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;
))

# ----------------- Wilcoxon Test: CX vs DOX -----------------
condition_pairs &lt;- tibble(
  cx = c(&quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;, &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;),
  dox = c(&quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;, &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;)
)

wilcox_results &lt;- map2_dfr(condition_pairs$cx, condition_pairs$dox, function(cx_label, dox_label) {
  cx_vals &lt;- complete_top2a_lit %&gt;% filter(Condition == cx_label) %&gt;% pull(logFC)
  dox_vals &lt;- complete_top2a_lit %&gt;% filter(Condition == dox_label) %&gt;% pull(logFC)
  
  test &lt;- tryCatch(wilcox.test(cx_vals, dox_vals), error = function(e) NULL)
  pval &lt;- if (!is.null(test)) test$p.value else NA
  
  tibble(
    Condition = dox_label,
    p_value = signif(pval, 3),
    label = case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01 ~ &quot;**&quot;,
      pval &lt; 0.05 ~ &quot;*&quot;,
      TRUE ~ &quot;&quot;
    ),
    y_pos = max(c(cx_vals, dox_vals), na.rm = TRUE) + 0.5
  )
})

# ----------------- Final Boxplot -----------------
ggplot(complete_top2a_lit, aes(x = Condition, y = logFC, fill = Drug)) +
  geom_boxplot(outlier.size = 0.6) +
  geom_text(data = wilcox_results,
            aes(x = Condition, y = y_pos, label = label),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;LogFC of TOP2A Target Genes\n(RPE-1, Literature)&quot;,
    x = &quot;Treatment Condition&quot;,
    y = &quot;Log Fold Change&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text.x = element_text(angle = 45, hjust = 1, size = 10),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/TOP2A_target_box_2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-TOP2A_target_box_2-1">
Past versions of TOP2A_target_box_2-1.png
</button>
</p>
<div id="fig-TOP2A_target_box_2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/d11a2ea01708a96de53fa57d625c8ec8b2d13700/docs/figure/DGE_Analysis.Rmd/TOP2A_target_box_2-1.png" target="_blank">d11a2ea</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="logfc-boxplots-of-anthracycline-cardiotoxicity-genes"
class="section level2">
<h2><strong>📌 LogFC boxplots of Anthracycline cardiotoxicity
genes</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)

# ----------------- DEG File Paths -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

# ----------------- AC Cardiotoxicity Entrez IDs -----------------
entrez_ids &lt;- c(
  6272, 8029, 11128, 79899, 54477, 121665, 5095, 22863, 57161, 4692,
  8214, 23151, 56606, 108, 22999, 56895, 9603, 3181, 4023, 10499,
  92949, 4363, 10057, 5243, 5244, 5880, 1535, 2950, 847, 5447,
  3038, 3077, 4846, 3958, 23327, 29899, 23155, 80856, 55020, 78996,
  23262, 150383, 9620, 79730, 344595, 5066, 6251, 3482, 9588, 339416,
  7292, 55157, 87769, 23409, 720, 3107, 54535, 1590, 80059, 7991,
  57110, 8803, 323, 54826, 5916, 23371, 283337, 64078, 80010, 1933,
  10818, 51020
) %&gt;% as.character()

# ----------------- Load and Filter DEGs -----------------
ac_data_list &lt;- map2_dfr(deg_files, names(deg_files), function(file, label) {
  read_csv(file, show_col_types = FALSE) %&gt;%
    mutate(
      Entrez_ID = as.character(Entrez_ID),
      Condition = label,
      Drug = ifelse(str_detect(label, &quot;CX&quot;), &quot;CX.5461&quot;, &quot;DOX&quot;)
    ) %&gt;%
    filter(Entrez_ID %in% entrez_ids)
})

# ----------------- Create Full Gene × Condition Table -----------------
all_conditions &lt;- names(deg_files)
all_combos &lt;- crossing(
  Entrez_ID = entrez_ids %&gt;% as.character(),
  Condition = all_conditions
) %&gt;%
  mutate(
    Drug = ifelse(str_detect(Condition, &quot;CX&quot;), &quot;CX.5461&quot;, &quot;DOX&quot;)
  )

# ----------------- Merge and Fill Missing Values -----------------
complete_ac &lt;- all_combos %&gt;%
  left_join(ac_data_list, by = c(&quot;Entrez_ID&quot;, &quot;Condition&quot;, &quot;Drug&quot;)) %&gt;%
  mutate(
    logFC = ifelse(is.na(logFC), 0, logFC),
    adj.P.Val = ifelse(is.na(adj.P.Val), 1, adj.P.Val)
  )

# ----------------- Annotate Gene Symbols -----------------
complete_ac &lt;- complete_ac %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;)
  )

# ----------------- Order Conditions -----------------
complete_ac$Condition &lt;- factor(complete_ac$Condition, levels = c(
  &quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;,
  &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
  &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;,
  &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;
))

# ----------------- Wilcoxon Test: CX vs DOX (paired by condition) -----------------
condition_pairs &lt;- tibble(
  cx = c(&quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;, &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;),
  dox = c(&quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;, &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;)
)

wilcox_results &lt;- map2_dfr(condition_pairs$cx, condition_pairs$dox, function(cx_label, dox_label) {
  cx_vals &lt;- complete_ac %&gt;% filter(Condition == cx_label) %&gt;% pull(logFC)
  dox_vals &lt;- complete_ac %&gt;% filter(Condition == dox_label) %&gt;% pull(logFC)
  
  test &lt;- tryCatch(wilcox.test(cx_vals, dox_vals), error = function(e) NULL)
  pval &lt;- if (!is.null(test)) test$p.value else NA
  
  tibble(
    Condition = dox_label,
    p_value = signif(pval, 3),
    label = case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01  ~ &quot;**&quot;,
      pval &lt; 0.05  ~ &quot;*&quot;,
      TRUE         ~ &quot;&quot;
    ),
    y_pos = max(c(cx_vals, dox_vals), na.rm = TRUE) + 0.5
  )
})

# ----------------- Plot Boxplot with Wilcoxon Stars -----------------
ggplot(complete_ac, aes(x = Condition, y = logFC, fill = Drug)) +
  geom_boxplot(outlier.size = 0.6) +
  geom_text(data = wilcox_results,
            aes(x = Condition, y = y_pos, label = label),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;LogFC of AC Cardiotoxicity Genes&quot;,
    x = &quot;Condition&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text.x = element_text(angle = 45, hjust = 1, size = 10),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/AC_cardiotox_box-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="logfc-boxplots-of-dox-cardiotoxicity-genes"
class="section level2">
<h2><strong>📌 LogFC boxplots of DOX cardiotoxicity genes</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)

# ----------------- DEG File Paths -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

# ----------------- DOX Cardiotoxicity Entrez IDs -----------------
entrez_ids &lt;- c(
  847, 873, 2064, 2878, 2944, 3038, 4846, 51196, 5880, 6687,
  7799, 4292, 5916, 3077, 51310, 9154, 64078, 5244, 10057, 10060,
  89845, 56853, 4625, 1573, 79890
) %&gt;% as.character()

# ----------------- Load and Filter DEGs -----------------
tox_data_list &lt;- map2_dfr(deg_files, names(deg_files), function(file, label) {
  read_csv(file, show_col_types = FALSE) %&gt;%
    mutate(
      Entrez_ID = as.character(Entrez_ID),
      Condition = label,
      Drug = ifelse(str_detect(label, &quot;CX&quot;), &quot;CX.5461&quot;, &quot;DOX&quot;)
    ) %&gt;%
    filter(Entrez_ID %in% entrez_ids)
})

# ----------------- Create Full Gene × Condition Table -----------------
all_conditions &lt;- names(deg_files)
all_combos &lt;- crossing(
  Entrez_ID = entrez_ids,
  Condition = all_conditions
) %&gt;%
  mutate(
    Drug = ifelse(str_detect(Condition, &quot;CX&quot;), &quot;CX.5461&quot;, &quot;DOX&quot;)
  )

# ----------------- Merge and Fill Missing Values -----------------
complete_doxcardio &lt;- all_combos %&gt;%
  left_join(tox_data_list, by = c(&quot;Entrez_ID&quot;, &quot;Condition&quot;, &quot;Drug&quot;)) %&gt;%
  mutate(
    logFC = ifelse(is.na(logFC), 0, logFC),
    adj.P.Val = ifelse(is.na(adj.P.Val), 1, adj.P.Val)
  )

# ----------------- Annotate Gene Symbols -----------------
complete_doxcardio &lt;- complete_doxcardio %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;)
  )

# ----------------- Order Conditions -----------------
complete_doxcardio$Condition &lt;- factor(complete_doxcardio$Condition, levels = c(
  &quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;,
  &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
  &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;,
  &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;
))

# ----------------- Wilcoxon Test: CX vs DOX (paired by condition) -----------------
condition_pairs &lt;- tibble(
  cx = c(&quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;, &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;),
  dox = c(&quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;, &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;)
)

wilcox_results &lt;- map2_dfr(condition_pairs$cx, condition_pairs$dox, function(cx_label, dox_label) {
  cx_vals &lt;- complete_doxcardio %&gt;% filter(Condition == cx_label) %&gt;% pull(logFC)
  dox_vals &lt;- complete_doxcardio %&gt;% filter(Condition == dox_label) %&gt;% pull(logFC)
  
  test &lt;- tryCatch(wilcox.test(cx_vals, dox_vals), error = function(e) NULL)
  pval &lt;- if (!is.null(test)) test$p.value else NA
  
  tibble(
    Condition = dox_label,
    p_value = signif(pval, 3),
    label = case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01  ~ &quot;**&quot;,
      pval &lt; 0.05  ~ &quot;*&quot;,
      TRUE         ~ &quot;&quot;
    ),
    y_pos = max(c(cx_vals, dox_vals), na.rm = TRUE) + 0.5
  )
})

# ----------------- Final Boxplot -----------------
ggplot(complete_doxcardio, aes(x = Condition, y = logFC, fill = Drug)) +
  geom_boxplot(outlier.size = 0.6) +
  geom_text(data = wilcox_results,
            aes(x = Condition, y = y_pos, label = label),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;LogFC of DOX Cardiotoxicity Genes&quot;,
    x = &quot;Condition&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text.x = element_text(angle = 45, hjust = 1, size = 10),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/DOX_cardiotox_box-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="logfc-boxplots-of-heart-specific-genes" class="section level2">
<h2><strong>📌 LogFC boxplots of Heart specific genes</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)

# ----------------- DEG File Paths -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

# ----------------- Load Heart-Specific Genes and Map to Entrez IDs -----------------
heart_genes &lt;- read.csv(&quot;data/Human_Heart_Genes.csv&quot;, stringsAsFactors = FALSE)
heart_genes$Entrez_ID &lt;- mapIds(
  org.Hs.eg.db,
  keys = heart_genes$Gene,
  column = &quot;ENTREZID&quot;,
  keytype = &quot;SYMBOL&quot;,
  multiVals = &quot;first&quot;
)
entrez_ids &lt;- na.omit(heart_genes$Entrez_ID) %&gt;% as.character()

# ----------------- Load and Filter DEGs -----------------
heart_data_list &lt;- map2_dfr(deg_files, names(deg_files), function(file, label) {
  read_csv(file, show_col_types = FALSE) %&gt;%
    mutate(
      Entrez_ID = as.character(Entrez_ID),
      Condition = label,
      Drug = ifelse(str_detect(label, &quot;CX&quot;), &quot;CX.5461&quot;, &quot;DOX&quot;)
    ) %&gt;%
    filter(Entrez_ID %in% entrez_ids)
})

# ----------------- Create Full Gene × Condition Table -----------------
all_conditions &lt;- names(deg_files)
all_combos &lt;- crossing(
  Entrez_ID = entrez_ids,
  Condition = all_conditions
) %&gt;%
  mutate(
    Drug = ifelse(str_detect(Condition, &quot;CX&quot;), &quot;CX.5461&quot;, &quot;DOX&quot;)
  )

# ----------------- Merge and Fill Missing Values -----------------
complete_heart &lt;- all_combos %&gt;%
  left_join(heart_data_list, by = c(&quot;Entrez_ID&quot;, &quot;Condition&quot;, &quot;Drug&quot;)) %&gt;%
  mutate(
    logFC = ifelse(is.na(logFC), 0, logFC),
    adj.P.Val = ifelse(is.na(adj.P.Val), 1, adj.P.Val)
  )

# ----------------- Annotate Gene Symbols -----------------
complete_heart &lt;- complete_heart %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;)
  )

# ----------------- Order Conditions -----------------
complete_heart$Condition &lt;- factor(complete_heart$Condition, levels = c(
  &quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;,
  &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
  &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;,
  &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;
))

# ----------------- Wilcoxon Test: CX vs DOX (matched) -----------------
condition_pairs &lt;- tibble(
  cx = c(&quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;, &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;),
  dox = c(&quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;, &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;)
)

wilcox_results &lt;- map2_dfr(condition_pairs$cx, condition_pairs$dox, function(cx_label, dox_label) {
  cx_vals &lt;- complete_heart %&gt;% filter(Condition == cx_label) %&gt;% pull(logFC)
  dox_vals &lt;- complete_heart %&gt;% filter(Condition == dox_label) %&gt;% pull(logFC)
  
  test &lt;- tryCatch(wilcox.test(cx_vals, dox_vals), error = function(e) NULL)
  pval &lt;- if (!is.null(test)) test$p.value else NA
  
  tibble(
    Condition = dox_label,
    p_value = signif(pval, 3),
    label = case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01  ~ &quot;**&quot;,
      pval &lt; 0.05  ~ &quot;*&quot;,
      TRUE         ~ &quot;&quot;
    ),
    y_pos = max(c(cx_vals, dox_vals), na.rm = TRUE) + 0.5
  )
})

# ----------------- Final Boxplot -----------------
ggplot(complete_heart, aes(x = Condition, y = logFC, fill = Drug)) +
  geom_boxplot(outlier.size = 0.6) +
  geom_text(data = wilcox_results,
            aes(x = Condition, y = y_pos, label = label),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;LogFC of Heart-Specific Genes&quot;,
    x = &quot;Condition&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text.x = element_text(angle = 45, hjust = 1, size = 10),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/Heart_Specific_box-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div
id="atrial-fibrillation-gene-expression-changes-between-cx5461-and-dox"
class="section level2">
<h2><strong>📌 Atrial fibrillation gene expression changes between
CX5461 and DOX</strong></h2>
<pre class="r"><code>library(tidyverse) 
library(ggfortify)
library(cluster)
library(edgeR)
library(limma)
library(Homo.sapiens)
library(BiocParallel)
library(qvalue)
library(pheatmap)
library(clusterProfiler)
library(AnnotationDbi)
library(org.Hs.eg.db)
library(RColorBrewer)
library(readr)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(ComplexHeatmap)
library(circlize)
library(grid)
library(reshape2)
library(dplyr)

# Load UCSC transcript database
txdb &lt;- TxDb.Hsapiens.UCSC.hg38.knownGene

# Load DEGs Data
CX_0.1_3 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;)
CX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;)
CX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;)
CX_0.5_3 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)

DOX_0.1_3 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;)
DOX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;)
DOX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;)
DOX_0.5_3 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

Entrez_IDs &lt;- c(
  3425, 53834, 9464, 79658, 55876, 3781, 9472, 11017, 64395, 23353,
  57794, 79591, 2702, 51306, 857, 10418, 9644, 115509, 89797, 10728,
  282996, 8497, 5695, 9026, 57666, 143384, 5883, 90102, 84909, 221037,
  29119, 79933, 391, 81575, 8738, 5396, 26010, 744, 56981, 80724,
  5781, 6910, 4598, 79068, 2250, 56978, 3752, 5208, 9736, 26778,
  80315, 64374, 79159, 11078, 5308, 3782, 10654, 79776, 6310, 8082,
  2995, 23224, 79741, 5708, 26136, 9967, 84033, 9148, 29982, 81488,
  1788, 9208, 2908, 1021, 5339, 5727, 23118, 10090, 57585, 221154,
  1073, 29998, 817, 858, 3486, 5747, 2104, 55105, 4194, 3570,
  1607, 6586, 5570, 2070, 27145, 4646, 3159, 9748, 79991, 221035,
  5033, 80212, 196385, 79600, 11046, 54014, 151636, 64778, 92597, 3983,
  7090, 9267, 7273, 26018, 9948, 2131, 55870, 5523, 5318, 6239,
  3480, 55777, 2066, 11165, 2028, 57619, 2690, 23414, 55818, 84700,
  28965, 80204, 463, 5108, 222553, 387119, 28981, 11113, 10301, 995,
  23030, 2697, 996, 339500, 54805, 9960, 91404, 145781, 100820829, 84542,
  2176, 51684, 9513, 7473, 4666, 23150, 5915, 5062, 4016, 23039,
  159686, 1839, 5201, 93166, 64753, 29959, 5496, 23245, 5069, 56916,
  92344, 23092, 3992, 9415, 10554, 7456, 9570, 57178, 23143, 161176,
  5424, 2034, 10277, 11278, 79803, 6653, 4756, 132660, 5430, 9031,
  57158, 285761, 8110, 387700, 1829, 4126, 7323, 51308, 7332, 6598,
  3757, 6187, 6660, 10529, 6920, 115286, 8451, 8943, 4137, 7514,
  4801, 9709, 23177, 8671, 29915, 26207, 3680, 490, 493856, 58489,
  54897, 4625, 22955, 84952, 10221, 2263, 84641, 4892, 1026, 84650,
  8382, 221656, 2969, 144453, 117177, 2626, 8476, 161882, 51807, 4624,
  9612, 55795, 51043, 144348, 51232, 8729, 3899, 11155, 23316, 79006,
  146330, 6403, 6331, 4300, 427, 845, 3313, 113622, 5789, 376132,
  29841, 8462, 203859, 401397, 5506, 55521, 5819, 4059, 6934, 57727,
  23066, 79568, 83478, 10087, 9586, 222194, 10466, 10499, 58499, 79720,
  4772, 10794, 125919, 602, 27332, 59345, 340359, 3705, 64710, 57801,
  10818, 143684, 149281, 55013, 23095, 93649, 84034, 23347, 440926, 11124,
  23293, 51426, 832, 7068, 57646, 152002, 7531, 1398, 100101267, 221937,
  26873, 3709, 10576, 27040, 201176, 284403, 307, 6525, 5387, 1808,
  114907, 1952, 4091, 5096, 23451, 51207, 142891, 4084, 3797, 1185,
  5334, 217, 2042, 7781, 253461, 152404, 2992, 153478, 6695, 26249,
  662, 23036, 22852, 23411, 23387, 6786, 7690, 10021, 8925, 5595,
  63892, 23466, 11149, 139411, 755, 55347, 22820, 55111, 84444, 6258,
  57623, 1387, 3762, 5861, 3339, 114822, 129787, 196528, 54437, 10395,
  4023, 5095, 894, 3156, 1877, 283871, 2673, 23157, 10512, 2258,
  79750, 84665, 26091, 5978, 8751, 79695, 2768, 155382, 84163, 7709,
  4092, 1837, 2064, 4629, 55122, 8882, 150962, 23013, 8742, 489,
  201134, 55114, 84264, 64428, 4089, 861, 7422
)


# Subset the toptable based on the entrez IDs and select specific columns

subset_toptable1 &lt;- CX_0.1_3[CX_0.1_3$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]


subset_toptable2 &lt;- CX_0.1_24[CX_0.1_24$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]


subset_toptable3 &lt;- CX_0.1_48[CX_0.1_48$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]


subset_toptable3 &lt;- CX_0.1_48[CX_0.1_48$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]

subset_toptable4 &lt;- CX_0.5_3[CX_0.5_3$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]

subset_toptable5 &lt;- CX_0.5_24[CX_0.5_24$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]

subset_toptable6 &lt;- CX_0.5_48[CX_0.5_48$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]

subset_toptable7 &lt;- DOX_0.1_3[DOX_0.1_3$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]

subset_toptable8 &lt;- DOX_0.1_24[DOX_0.1_24$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]

subset_toptable9 &lt;- DOX_0.1_48[DOX_0.1_48$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]

subset_toptable10 &lt;- DOX_0.5_3[DOX_0.5_3$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]

subset_toptable11 &lt;- DOX_0.5_24[DOX_0.5_24$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]

subset_toptable12 &lt;- DOX_0.5_48[DOX_0.5_48$Entrez_ID %in% Entrez_IDs, c(&quot;Entrez_ID&quot;, &quot;logFC&quot;, &quot;adj.P.Val&quot;)]

# Function to add columns and combine data
add_metadata &lt;- function(data, drug, conc, time) {
  data %&gt;%
    mutate(Drug = drug, Conc = conc, Time = time)
}

# Add metadata and combine all subsets
combined_data &lt;- bind_rows(
  add_metadata(subset_toptable1, &quot;CX&quot;, 0.1, 3),
  add_metadata(subset_toptable2, &quot;CX&quot;, 0.1, 24),
  add_metadata(subset_toptable3, &quot;CX&quot;, 0.1, 48),
  add_metadata(subset_toptable4, &quot;CX&quot;, 0.5, 3),
  add_metadata(subset_toptable5, &quot;CX&quot;, 0.5, 24),
  add_metadata(subset_toptable6, &quot;CX&quot;, 0.5, 48),
  add_metadata(subset_toptable7, &quot;DOX&quot;, 0.1, 3),
  add_metadata(subset_toptable8, &quot;DOX&quot;, 0.1, 24),
  add_metadata(subset_toptable9, &quot;DOX&quot;, 0.1, 48),
  add_metadata(subset_toptable10, &quot;DOX&quot;, 0.5, 3),
  add_metadata(subset_toptable11, &quot;DOX&quot;, 0.5, 24),
  add_metadata(subset_toptable12, &quot;DOX&quot;, 0.5, 48)
)

# Convert Entrez IDs to Gene symbols
combined_data &lt;- combined_data %&gt;%
  mutate(Gene = mapIds(
    org.Hs.eg.db,
    keys = as.character(Entrez_ID),
    column = &quot;SYMBOL&quot;,
    keytype = &quot;ENTREZID&quot;,
    multiVals = &quot;first&quot;
  ))

# Reorder columns
final_data &lt;- dplyr::select(combined_data, Entrez_ID, Gene, logFC, adj.P.Val, Drug, Conc, Time)

# Assuming your dataframe is named data
# Add a column for significance stars
final_data &lt;- final_data %&gt;%
  mutate(Significance = ifelse(adj.P.Val &lt; 0.05, &quot;*&quot;, &quot;&quot;))

# Create a matrix for the heatmap (logFC values)
logFC_matrix &lt;- acast(final_data, Gene ~ paste(Drug, Conc, Time, sep = &quot;_&quot;), value.var = &quot;logFC&quot;)

# Create a matrix for the significance annotations
signif_matrix &lt;- acast(final_data, Gene ~ paste(Drug, Conc, Time, sep = &quot;_&quot;), value.var = &quot;Significance&quot;)

# Split column names into Drug, Conc, and Time
colnames_split &lt;- strsplit(colnames(logFC_matrix), &quot;_&quot;)
drug &lt;- sapply(colnames_split, function(x) x[1])
conc &lt;- sapply(colnames_split, function(x) x[2])
time &lt;- sapply(colnames_split, function(x) x[3])

# Create the desired column order: CX 0.1 3hr, CX 0.5 3hr, CX 0.1 24hr, CX 0.5 24hr, CX 0.1 48h, CX 0.5 48h,
# DOX 0.1 3hr, DOX 0.5 3hr, DOX 0.1 24hr, DOX 0.5 24hr, DOX 0.1 48h, DOX 0.5 48h
desired_order &lt;- c(&quot;CX_0.1_3&quot;, &quot;CX_0.5_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.1_48&quot;, &quot;CX_0.5_48&quot;,
                   &quot;DOX_0.1_3&quot;, &quot;DOX_0.5_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.1_48&quot;, &quot;DOX_0.5_48&quot;)

# Reorder columns in the matrix based on the desired order
column_names &lt;- paste(drug, conc, time, sep = &quot;_&quot;)
column_order &lt;- match(desired_order, column_names)
logFC_matrix &lt;- logFC_matrix[, column_order]
signif_matrix &lt;- signif_matrix[, column_order]
drug &lt;- drug[column_order]
conc &lt;- conc[column_order]
time &lt;- time[column_order]

# Prepare annotations matching the column structure
ha_top &lt;- HeatmapAnnotation(
  Drug = drug,
  Conc = conc,
  Time = time,
  col = list(Drug = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;),
             Conc = c(&quot;0.1&quot; = &quot;lightgreen&quot;, &quot;0.5&quot; = &quot;darkgreen&quot;),
             Time = c(&quot;3&quot; = &quot;yellow&quot;, &quot;24&quot; = &quot;orange&quot;, &quot;48&quot; = &quot;purple&quot;)),
  annotation_height = unit(c(2, 2, 2), &quot;cm&quot;)
)

# Create the heatmap
heatmap &lt;- Heatmap(logFC_matrix, name = &quot;logFC&quot;, top_annotation = ha_top,
                   cell_fun = function(j, i, x, y, width, height, fill) {
                     grid.text(signif_matrix[i, j], x, y, gp = gpar(fontsize = 10))
                   },
                   show_row_names = TRUE, show_column_names = FALSE,
                   column_title = &quot;Genes in Atrial fibrillation GWAS associated loci\nin response to CX5461 and DOX&quot;,
                   column_title_gp = gpar(fontsize = 16, fontface = &quot;bold&quot;),
                   cluster_columns = FALSE)  # Disable column clustering

# Draw the heatmap
draw(heatmap, heatmap_legend_side = &quot;right&quot;, annotation_legend_side = &quot;right&quot;)</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/AF-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-AF-1">
Past versions of AF-1.png
</button>
</p>
<div id="fig-AF-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/92099e7a5fa068d2357cb772c18c0e40b0953661/docs/figure/DGE_Analysis.Rmd/AF-1.png" target="_blank">92099e7</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="logfc-boxplots-of-atrial-fibrillation-gwas-genes"
class="section level2">
<h2><strong>📌 LogFC boxplots of Atrial fibrillation GWAS
genes</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)

# ----------------- DEG File Paths -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

# ----------------- AF Entrez IDs -----------------
entrez_ids &lt;- c(
  3425, 53834, 9464, 79658, 55876, 3781, 9472, 11017, 64395, 23353,
  57794, 79591, 2702, 51306, 857, 10418, 9644, 115509, 89797, 10728,
  282996, 8497, 5695, 9026, 57666, 143384, 5883, 90102, 84909, 221037,
  29119, 79933, 391, 81575, 8738, 5396, 26010, 744, 56981, 80724,
  5781, 6910, 4598, 79068, 2250, 56978, 3752, 5208, 9736, 26778,
  80315, 64374, 79159, 11078, 5308, 3782, 10654, 79776, 6310, 8082,
  2995, 23224, 79741, 5708, 26136, 9967, 84033, 9148, 29982, 81488,
  1788, 9208, 2908, 1021, 5339, 5727, 23118, 10090, 57585, 221154,
  1073, 29998, 817, 858, 3486, 5747, 2104, 55105, 4194, 3570,
  1607, 6586, 5570, 2070, 27145, 4646, 3159, 9748, 79991, 221035,
  5033, 80212, 196385, 79600, 11046, 54014, 151636, 64778, 92597, 3983,
  7090, 9267, 7273, 26018, 9948, 2131, 55870, 5523, 5318, 6239,
  3480, 55777, 2066, 11165, 2028, 57619, 2690, 23414, 55818, 84700,
  28965, 80204, 463, 5108, 222553, 387119, 28981, 11113, 10301, 995,
  23030, 2697, 996, 339500, 54805, 9960, 91404, 145781, 100820829, 84542,
  2176, 51684, 9513, 7473, 4666, 23150, 5915, 5062, 4016, 23039,
  159686, 1839, 5201, 93166, 64753, 29959, 5496, 23245, 5069, 56916,
  92344, 23092, 3992, 9415, 10554, 7456, 9570, 57178, 23143, 161176,
  5424, 2034, 10277, 11278, 79803, 6653, 4756, 132660, 5430, 9031,
  57158, 285761, 8110, 387700, 1829, 4126, 7323, 51308, 7332, 6598,
  3757, 6187, 6660, 10529, 6920, 115286, 8451, 8943, 4137, 7514,
  4801, 9709, 23177, 8671, 29915, 26207, 3680, 490, 493856, 58489,
  54897, 4625, 22955, 84952, 10221, 2263, 84641, 4892, 1026, 84650,
  8382, 221656, 2969, 144453, 117177, 2626, 8476, 161882, 51807, 4624,
  9612, 55795, 51043, 144348, 51232, 8729, 3899, 11155, 23316, 79006,
  146330, 6403, 6331, 4300, 427, 845, 3313, 113622, 5789, 376132,
  29841, 8462, 203859, 401397, 5506, 55521, 5819, 4059, 6934, 57727,
  23066, 79568, 83478, 10087, 9586, 222194, 10466, 10499, 58499, 79720,
  4772, 10794, 125919, 602, 27332, 59345, 340359, 3705, 64710, 57801,
  10818, 143684, 149281, 55013, 23095, 93649, 84034, 23347, 440926, 11124,
  23293, 51426, 832, 7068, 57646, 152002, 7531, 1398, 100101267, 221937,
  26873, 3709, 10576, 27040, 201176, 284403, 307, 6525, 5387, 1808,
  114907, 1952, 4091, 5096, 23451, 51207, 142891, 4084, 3797, 1185,
  5334, 217, 2042, 7781, 253461, 152404, 2992, 153478, 6695, 26249,
  662, 23036, 22852, 23411, 23387, 6786, 7690, 10021, 8925, 5595,
  63892, 23466, 11149, 139411, 755, 55347, 22820, 55111, 84444, 6258,
  57623, 1387, 3762, 5861, 3339, 114822, 129787, 196528, 54437, 10395,
  4023, 5095, 894, 3156, 1877, 283871, 2673, 23157, 10512, 2258,
  79750, 84665, 26091, 5978, 8751, 79695, 2768, 155382, 84163, 7709,
  4092, 1837, 2064, 4629, 55122, 8882, 150962, 23013, 8742, 489,
  201134, 55114, 84264, 64428, 4089, 861, 7422
) %&gt;% as.character()

# ----------------- Load and Filter DEGs -----------------
af_data_list &lt;- map2_dfr(deg_files, names(deg_files), function(file, label) {
  read_csv(file, show_col_types = FALSE) %&gt;%
    mutate(
      Entrez_ID = as.character(Entrez_ID),
      Condition = label,
      Drug = ifelse(str_detect(label, &quot;CX&quot;), &quot;CX.5461&quot;, &quot;DOX&quot;)
    ) %&gt;%
    filter(Entrez_ID %in% entrez_ids)
})

# ----------------- Complete Combinations -----------------
all_conditions &lt;- names(deg_files)
all_combos &lt;- crossing(
  Entrez_ID = entrez_ids,
  Condition = all_conditions
) %&gt;%
  mutate(
    Drug = ifelse(str_detect(Condition, &quot;CX&quot;), &quot;CX.5461&quot;, &quot;DOX&quot;)
  )

# ----------------- Merge with Data -----------------
complete_af &lt;- all_combos %&gt;%
  left_join(af_data_list, by = c(&quot;Entrez_ID&quot;, &quot;Condition&quot;, &quot;Drug&quot;)) %&gt;%
  mutate(
    logFC = ifelse(is.na(logFC), 0, logFC),
    adj.P.Val = ifelse(is.na(adj.P.Val), 1, adj.P.Val)
  )

# ----------------- Gene Symbol Mapping -----------------
complete_af &lt;- complete_af %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;)
  )

# ----------------- Order Conditions -----------------
complete_af$Condition &lt;- factor(complete_af$Condition, levels = c(
  &quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;,
  &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
  &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;,
  &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;
))

# ----------------- Wilcoxon Tests -----------------
condition_pairs &lt;- tibble(
  cx = c(&quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;, &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;),
  dox = c(&quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;, &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;)
)

wilcox_results &lt;- map2_dfr(condition_pairs$cx, condition_pairs$dox, function(cx_label, dox_label) {
  cx_vals &lt;- complete_af %&gt;% filter(Condition == cx_label) %&gt;% pull(logFC)
  dox_vals &lt;- complete_af %&gt;% filter(Condition == dox_label) %&gt;% pull(logFC)
  
  test &lt;- tryCatch(wilcox.test(cx_vals, dox_vals), error = function(e) NULL)
  pval &lt;- if (!is.null(test)) test$p.value else NA
  
  tibble(
    Condition = dox_label,
    p_value = signif(pval, 3),
    label = case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01  ~ &quot;**&quot;,
      pval &lt; 0.05  ~ &quot;*&quot;,
      TRUE         ~ &quot;&quot;
    ),
    y_pos = max(c(cx_vals, dox_vals), na.rm = TRUE) + 0.5
  )
})

# ----------------- Final Plot -----------------
ggplot(complete_af, aes(x = Condition, y = logFC, fill = Drug)) +
  geom_boxplot(outlier.size = 0.6) +
  geom_text(data = wilcox_results,
            aes(x = Condition, y = y_pos, label = label),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;LogFC of AF-Associated Genes&quot;,
    x = &quot;Condition&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text.x = element_text(angle = 45, hjust = 1, size = 10),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/AF_box-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="heart-failure-gene-expression-changes-between-cx5461-and-dox"
class="section level2">
<h2><strong>📌 Heart Failure gene expression changes between CX5461 and
DOX</strong></h2>
<pre class="r"><code># Load necessary libraries
library(tidyverse)
library(ComplexHeatmap)
library(circlize)
library(org.Hs.eg.db)
library(reshape2)
library(grid)

# Set 30 Entrez IDs
Entrez_IDs &lt;- c(
  9709, 8882, 4023, 29959, 5496, 3992, 9415, 5308, 1026, 54437,
  79068, 10221, 9031, 1187, 1952, 3705, 84722, 7273, 23293, 155382,
  9531, 602, 27258, 84163, 81846, 79933, 56911, 64753, 93210, 1021
)

# rsID mapping for row annotation
rsid_mapping &lt;- data.frame(
  Entrez_ID = Entrez_IDs,
  rsID = c(&quot;rs247617&quot;, &quot;rs964184&quot;, &quot;rs765547&quot;, &quot;rs1728918&quot;, &quot;rs1728918&quot;, &quot;rs174547&quot;,
           &quot;rs174547&quot;, &quot;rs59788391&quot;, &quot;rs3176326&quot;, &quot;rs7632505&quot;, &quot;rs11642015&quot;, &quot;rs2980853&quot;,
           &quot;rs2074755&quot;, &quot;rs28579893&quot;, &quot;rs602633&quot;, &quot;rs4905014&quot;, &quot;rs602633&quot;, &quot;rs2562845&quot;,
           &quot;rs216193&quot;, &quot;rs799165&quot;, &quot;rs17617337&quot;, &quot;rs4803750&quot;, &quot;rs11710541&quot;, &quot;rs202209188&quot;,
           &quot;rs2957657&quot;, &quot;rs34163229&quot;, &quot;rs56968346&quot;, &quot;rs3807132&quot;, &quot;rs12150603&quot;, &quot;rs2282979&quot;
  )
)

# Load DEGs
load_deg &lt;- function(name) {
  read.csv(paste0(&quot;data/DEGs/Toptable_&quot;, name, &quot;.csv&quot;))
}
samples &lt;- c(&quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;, &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
             &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;, &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;)
deg_list &lt;- lapply(samples, load_deg)
names(deg_list) &lt;- samples

# Subset and annotate
# Subset and annotate
get_subset &lt;- function(df, name) {
  parts &lt;- strsplit(name, &quot;_&quot;)[[1]]
  df %&gt;%
    filter(Entrez_ID %in% Entrez_IDs) %&gt;%
    dplyr::select(Entrez_ID, logFC, adj.P.Val) %&gt;%
    mutate(Drug = parts[1], Conc = parts[2], Time = parts[3])
}
combined_data &lt;- bind_rows(mapply(get_subset, deg_list, names(deg_list), SIMPLIFY = FALSE))

# Add gene symbol and significance
combined_data &lt;- combined_data %&gt;%
  mutate(Gene = mapIds(org.Hs.eg.db, keys = as.character(Entrez_ID),
                       column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;),
         Significance = ifelse(adj.P.Val &lt; 0.05, &quot;*&quot;, &quot;&quot;))

# Join rsID mapping
combined_data &lt;- left_join(combined_data, rsid_mapping, by = &quot;Entrez_ID&quot;)

# Preserve gene order
ordered_genes &lt;- combined_data %&gt;%
  distinct(Entrez_ID, Gene) %&gt;%
  arrange(factor(Entrez_ID, levels = Entrez_IDs)) %&gt;%
  pull(Gene)

# Create logFC and significance matrices
logFC_matrix &lt;- acast(combined_data, Gene ~ paste(Drug, Conc, Time, sep = &quot;_&quot;), value.var = &quot;logFC&quot;)
signif_matrix &lt;- acast(combined_data, Gene ~ paste(Drug, Conc, Time, sep = &quot;_&quot;), value.var = &quot;Significance&quot;)
logFC_matrix &lt;- logFC_matrix[ordered_genes, ]
signif_matrix &lt;- signif_matrix[ordered_genes, ]

# Desired order: group 0.1 first, then 0.5 for both drugs
desired_order &lt;- c(&quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;, &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
                   &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;, &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;)
logFC_matrix &lt;- logFC_matrix[, desired_order]
signif_matrix &lt;- signif_matrix[, desired_order]

# Split column info
split_cols &lt;- strsplit(colnames(logFC_matrix), &quot;_&quot;)
drug &lt;- sapply(split_cols, `[`, 1)
conc &lt;- sapply(split_cols, `[`, 2)
time &lt;- sapply(split_cols, `[`, 3)

# Annotations
ha_top &lt;- HeatmapAnnotation(
  Drug = drug,
  Conc = conc,
  Time = time,
  col = list(
    Drug = c(&quot;CX&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;),
    Conc = c(&quot;0.1&quot; = &quot;lightgreen&quot;, &quot;0.5&quot; = &quot;darkgreen&quot;),
    Time = c(&quot;3&quot; = &quot;yellow&quot;, &quot;24&quot; = &quot;orange&quot;, &quot;48&quot; = &quot;purple&quot;)
  ),
  annotation_height = unit(c(2, 2, 2), &quot;cm&quot;)
)

# rsID annotation
rsid_vector &lt;- combined_data %&gt;%
  distinct(Gene, rsID) %&gt;%
  filter(Gene %in% ordered_genes) %&gt;%
  arrange(match(Gene, ordered_genes)) %&gt;%
  pull(rsID)

ha_left &lt;- rowAnnotation(
  rsID = anno_text(rsid_vector, location = 0, just = &quot;left&quot;, gp = gpar(fontsize = 9)),
  annotation_name_side = &quot;top&quot;,
  width = unit(5, &quot;cm&quot;)
)

# Heatmap
Heatmap(logFC_matrix,
        name = &quot;logFC&quot;,
        top_annotation = ha_top,
        left_annotation = ha_left,
        show_row_names = TRUE,
        show_column_names = FALSE,
        cluster_columns = FALSE,
        cluster_rows = FALSE,
        cell_fun = function(j, i, x, y, width, height, fill) {
          grid.text(signif_matrix[i, j], x, y, gp = gpar(fontsize = 10))
        },
        column_title = &quot;Heart Failure GWAS Genes: Response to CX-5461 and DOX&quot;,
        column_title_gp = gpar(fontsize = 16, fontface = &quot;bold&quot;)
)</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/HF-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-HF-1">
Past versions of HF-1.png
</button>
</p>
<div id="fig-HF-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/92099e7a5fa068d2357cb772c18c0e40b0953661/docs/figure/DGE_Analysis.Rmd/HF-1.png" target="_blank">92099e7</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="logfc-boxplots-of-heart-failure-gwas-genes"
class="section level2">
<h2><strong>📌 LogFC boxplots of Heart Failure GWAS genes</strong></h2>
<pre class="r"><code># ----------------- Load Libraries -----------------
library(tidyverse)
library(org.Hs.eg.db)

# ----------------- DEG File Paths -----------------
deg_files &lt;- list(
  &quot;CX_0.1_3&quot; = &quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;,
  &quot;CX_0.1_24&quot; = &quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot; = &quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot; = &quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot; = &quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot; = &quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot; = &quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot; = &quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;,
  &quot;DOX_0.5_24&quot; = &quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;,
  &quot;DOX_0.5_48&quot; = &quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;
)

# ----------------- Heart Failure Entrez IDs -----------------
entrez_ids &lt;- c(
  9709, 8882, 4023, 29959, 5496, 3992, 9415, 5308, 1026, 54437, 79068, 10221,
  9031, 1187, 1952, 3705, 84722, 7273, 23293, 155382, 9531, 602, 27258, 84163,
  81846, 79933, 56911, 64753, 93210, 1021, 283450, 5998, 57602, 114991, 7073,
  3156, 100101267, 22996, 285025, 11080, 11124, 54810, 7531, 27241, 4774, 57794,
  463, 91319, 6598, 9640, 2186, 26010, 80816, 571, 88, 51652, 64788, 90523, 2969,
  7781, 80777, 10725, 23387, 817, 134728, 8842, 949, 6934, 129787, 10327, 202052,
  2318, 5578, 6801, 6311, 10019, 80724, 217, 84909, 388591, 55101, 9839, 27161,
  5310, 387119, 4641, 5587, 55188, 222553, 9960, 22852, 10087, 9570, 54497,
  200942, 26249, 4137, 375056, 5409, 64116, 8291, 22876, 339855, 4864, 5142,
  221692, 55023, 51426, 6146, 84251, 8189, 27332, 57099, 1869, 1112, 23327,
  11264, 6001
) %&gt;% as.character()

# ----------------- Load and Filter DEGs -----------------
hf_data_list &lt;- map2_dfr(deg_files, names(deg_files), function(file, label) {
  read_csv(file, show_col_types = FALSE) %&gt;%
    mutate(
      Entrez_ID = as.character(Entrez_ID),
      Condition = label,
      Drug = ifelse(str_detect(label, &quot;CX&quot;), &quot;CX.5461&quot;, &quot;DOX&quot;)
    ) %&gt;%
    filter(Entrez_ID %in% entrez_ids)
})

# ----------------- Complete Combinations -----------------
all_conditions &lt;- names(deg_files)
all_combos &lt;- crossing(
  Entrez_ID = entrez_ids,
  Condition = all_conditions
) %&gt;%
  mutate(
    Drug = ifelse(str_detect(Condition, &quot;CX&quot;), &quot;CX.5461&quot;, &quot;DOX&quot;)
  )

# ----------------- Merge with Data -----------------
complete_hf &lt;- all_combos %&gt;%
  left_join(hf_data_list, by = c(&quot;Entrez_ID&quot;, &quot;Condition&quot;, &quot;Drug&quot;)) %&gt;%
  mutate(
    logFC = ifelse(is.na(logFC), 0, logFC),
    adj.P.Val = ifelse(is.na(adj.P.Val), 1, adj.P.Val)
  )

# ----------------- Gene Symbol Mapping -----------------
complete_hf &lt;- complete_hf %&gt;%
  mutate(
    Gene = mapIds(org.Hs.eg.db, keys = Entrez_ID,
                  column = &quot;SYMBOL&quot;, keytype = &quot;ENTREZID&quot;, multiVals = &quot;first&quot;)
  )

# ----------------- Order Conditions -----------------
complete_hf$Condition &lt;- factor(complete_hf$Condition, levels = c(
  &quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;,
  &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
  &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;,
  &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;
))

# ----------------- Wilcoxon Tests -----------------
condition_pairs &lt;- tibble(
  cx = c(&quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;, &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;),
  dox = c(&quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;, &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;)
)

wilcox_results &lt;- map2_dfr(condition_pairs$cx, condition_pairs$dox, function(cx_label, dox_label) {
  cx_vals &lt;- complete_hf %&gt;% filter(Condition == cx_label) %&gt;% pull(logFC)
  dox_vals &lt;- complete_hf %&gt;% filter(Condition == dox_label) %&gt;% pull(logFC)
  
  test &lt;- tryCatch(wilcox.test(cx_vals, dox_vals), error = function(e) NULL)
  pval &lt;- if (!is.null(test)) test$p.value else NA
  
  tibble(
    Condition = dox_label,
    p_value = signif(pval, 3),
    label = case_when(
      pval &lt; 0.001 ~ &quot;***&quot;,
      pval &lt; 0.01  ~ &quot;**&quot;,
      pval &lt; 0.05  ~ &quot;*&quot;,
      TRUE         ~ &quot;&quot;
    ),
    y_pos = max(c(cx_vals, dox_vals), na.rm = TRUE) + 0.5
  )
})

# ----------------- Final Plot -----------------
ggplot(complete_hf, aes(x = Condition, y = logFC, fill = Drug)) +
  geom_boxplot(outlier.size = 0.6) +
  geom_text(data = wilcox_results,
            aes(x = Condition, y = y_pos, label = label),
            inherit.aes = FALSE,
            size = 4, vjust = 0) +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +
  labs(
    title = &quot;LogFC of Heart Failure-Associated Genes&quot;,
    x = &quot;Condition&quot;,
    y = &quot;logFC&quot;,
    fill = &quot;Drug&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text.x = element_text(angle = 45, hjust = 1, size = 10),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/DGE_Analysis.Rmd/HF_box-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 26100)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] TxDb.Hsapiens.UCSC.hg38.knownGene_3.18.0
 [2] RColorBrewer_1.1-3                      
 [3] clusterProfiler_4.10.1                  
 [4] pheatmap_1.0.12                         
 [5] qvalue_2.34.0                           
 [6] BiocParallel_1.36.0                     
 [7] cluster_2.1.8.1                         
 [8] ggfortify_0.4.17                        
 [9] reshape2_1.4.4                          
[10] circlize_0.4.16                         
[11] ComplexHeatmap_2.18.0                   
[12] Homo.sapiens_1.3.1                      
[13] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2 
[14] org.Hs.eg.db_3.18.0                     
[15] GO.db_3.18.0                            
[16] OrganismDbi_1.44.0                      
[17] GenomicFeatures_1.54.4                  
[18] GenomicRanges_1.54.1                    
[19] GenomeInfoDb_1.38.8                     
[20] AnnotationDbi_1.64.1                    
[21] IRanges_2.36.0                          
[22] S4Vectors_0.40.2                        
[23] Biobase_2.62.0                          
[24] BiocGenerics_0.48.1                     
[25] biomaRt_2.58.2                          
[26] scales_1.3.0                            
[27] lubridate_1.9.4                         
[28] forcats_1.0.0                           
[29] stringr_1.5.1                           
[30] dplyr_1.1.4                             
[31] purrr_1.0.4                             
[32] readr_2.1.5                             
[33] tidyr_1.3.1                             
[34] tibble_3.2.1                            
[35] ggplot2_3.5.2                           
[36] tidyverse_2.0.0                         
[37] data.table_1.17.0                       
[38] edgeR_4.0.16                            
[39] limma_3.58.1                            

loaded via a namespace (and not attached):
  [1] splines_4.3.0               later_1.3.2                
  [3] BiocIO_1.12.0               ggplotify_0.1.2            
  [5] bitops_1.0-9                filelock_1.0.3             
  [7] polyclip_1.10-7             graph_1.80.0               
  [9] XML_3.99-0.18               lifecycle_1.0.4            
 [11] doParallel_1.0.17           rprojroot_2.0.4            
 [13] lattice_0.22-7              vroom_1.6.5                
 [15] MASS_7.3-60                 magrittr_2.0.3             
 [17] sass_0.4.10                 rmarkdown_2.29             
 [19] jquerylib_0.1.4             yaml_2.3.10                
 [21] httpuv_1.6.15               cowplot_1.1.3              
 [23] DBI_1.2.3                   abind_1.4-8                
 [25] zlibbioc_1.48.2             ggraph_2.2.1               
 [27] RCurl_1.98-1.17             yulab.utils_0.2.0          
 [29] tweenr_2.0.3                rappdirs_0.3.3             
 [31] git2r_0.36.2                GenomeInfoDbData_1.2.11    
 [33] enrichplot_1.22.0           ggrepel_0.9.6              
 [35] tidytree_0.4.6              codetools_0.2-20           
 [37] DelayedArray_0.28.0         DOSE_3.28.2                
 [39] xml2_1.3.8                  ggforce_0.4.2              
 [41] tidyselect_1.2.1            shape_1.4.6.1              
 [43] aplot_0.2.5                 farver_2.1.2               
 [45] viridis_0.6.5               matrixStats_1.5.0          
 [47] BiocFileCache_2.10.2        GenomicAlignments_1.38.2   
 [49] jsonlite_2.0.0              GetoptLong_1.0.5           
 [51] tidygraph_1.3.1             iterators_1.0.14           
 [53] foreach_1.5.2               tools_4.3.0                
 [55] progress_1.2.3              treeio_1.26.0              
 [57] Rcpp_1.0.12                 glue_1.7.0                 
 [59] gridExtra_2.3               SparseArray_1.2.4          
 [61] xfun_0.52                   MatrixGenerics_1.14.0      
 [63] withr_3.0.2                 BiocManager_1.30.25        
 [65] fastmap_1.2.0               digest_0.6.34              
 [67] gridGraphics_0.5-1          timechange_0.3.0           
 [69] R6_2.6.1                    colorspace_2.1-0           
 [71] Cairo_1.6-2                 RSQLite_2.3.9              
 [73] generics_0.1.3              rtracklayer_1.62.0         
 [75] graphlayouts_1.2.2          prettyunits_1.2.0          
 [77] httr_1.4.7                  S4Arrays_1.2.1             
 [79] scatterpie_0.2.4            whisker_0.4.1              
 [81] pkgconfig_2.0.3             gtable_0.3.6               
 [83] blob_1.2.4                  workflowr_1.7.1            
 [85] XVector_0.42.0              shadowtext_0.1.4           
 [87] htmltools_0.5.8.1           fgsea_1.28.0               
 [89] RBGL_1.78.0                 clue_0.3-66                
 [91] png_0.1-8                   ggfun_0.1.8                
 [93] knitr_1.50                  rstudioapi_0.17.1          
 [95] tzdb_0.5.0                  rjson_0.2.23               
 [97] nlme_3.1-168                curl_6.2.2                 
 [99] cachem_1.1.0                GlobalOptions_0.1.2        
[101] parallel_4.3.0              HDO.db_0.99.1              
[103] restfulr_0.0.15             pillar_1.10.2              
[105] vctrs_0.6.5                 promises_1.3.2             
[107] dbplyr_2.5.0                evaluate_1.0.3             
[109] magick_2.8.6                cli_3.6.1                  
[111] locfit_1.5-9.12             compiler_4.3.0             
[113] Rsamtools_2.18.0            rlang_1.1.3                
[115] crayon_1.5.3                labeling_0.4.3             
[117] plyr_1.8.9                  fs_1.6.3                   
[119] stringi_1.8.3               viridisLite_0.4.2          
[121] munsell_0.5.1               Biostrings_2.70.3          
[123] lazyeval_0.2.2              GOSemSim_2.28.1            
[125] Matrix_1.6-1.1              patchwork_1.3.0            
[127] hms_1.1.3                   bit64_4.6.0-1              
[129] KEGGREST_1.42.0             statmod_1.5.0              
[131] SummarizedExperiment_1.32.0 igraph_2.1.4               
[133] memoise_2.0.1               bslib_0.9.0                
[135] ggtree_3.10.1               fastmatch_1.1-6            
[137] bit_4.6.0                   gson_0.1.0                 
[139] ape_5.8-1                  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
