<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Macrophage dataset comparison</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CX5461_Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Macrophage dataset comparison</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-03-06
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>CX5461_Project/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree0aec578ba029d6b945b4201439463119a9a53e32targetblank0aec578a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/0aec578ba029d6b945b4201439463119a9a53e32" target="_blank">0aec578</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree0aec578ba029d6b945b4201439463119a9a53e32targetblank0aec578a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/0aec578ba029d6b945b4201439463119a9a53e32" target="_blank">0aec578</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Unstaged changes:
    Modified:   analysis/Macrophage.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Macrophage.Rmd</code>) and HTML
(<code>docs/Macrophage.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/07caa2fdb1c88047c32f0d632105c92d1f23a0e3/analysis/Macrophage.Rmd" target="_blank">07caa2f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/07caa2fdb1c88047c32f0d632105c92d1f23a0e3/docs/Macrophage.html" target="_blank">07caa2f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/10d28dd7ba8a410e15ba473a9c14c68fd3eb544a/analysis/Macrophage.Rmd" target="_blank">10d28dd</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/10d28dd7ba8a410e15ba473a9c14c68fd3eb544a/docs/Macrophage.html" target="_blank">10d28dd</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/46aa22b9d3ecdb0423e688fbc2c17281dcc9bddd/analysis/Macrophage.Rmd" target="_blank">46aa22b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-01
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/46aa22b9d3ecdb0423e688fbc2c17281dcc9bddd/docs/Macrophage.html" target="_blank">46aa22b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-01
</td>
<td>
Commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="proportion-of-macrophage-de-genes-in-corrmotif-clusters"
class="section level2">
<h2><strong>📌 Proportion of Macrophage DE genes in Corrmotif
clusters</strong></h2>
</div>
<div id="load-required-libraries" class="section level2">
<h2><strong>📌 Load Required Libraries</strong></h2>
<pre class="r"><code>library(ggplot2)</code></pre>
<pre><code>Warning: package &#39;ggplot2&#39; was built under R version 4.3.3</code></pre>
<pre class="r"><code>library(dplyr)</code></pre>
<pre><code>Warning: package &#39;dplyr&#39; was built under R version 4.3.2</code></pre>
<pre class="r"><code>library(tidyr)</code></pre>
<pre><code>Warning: package &#39;tidyr&#39; was built under R version 4.3.3</code></pre>
<pre class="r"><code>library(org.Hs.eg.db)</code></pre>
<pre><code>Warning: package &#39;AnnotationDbi&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;BiocGenerics&#39; was built under R version 4.3.1</code></pre>
<pre><code>Warning: package &#39;Biobase&#39; was built under R version 4.3.1</code></pre>
<pre><code>Warning: package &#39;IRanges&#39; was built under R version 4.3.1</code></pre>
<pre><code>Warning: package &#39;S4Vectors&#39; was built under R version 4.3.1</code></pre>
<pre class="r"><code>library(clusterProfiler)</code></pre>
<pre><code>Warning: package &#39;clusterProfiler&#39; was built under R version 4.3.3</code></pre>
<pre class="r"><code>library(biomaRt)</code></pre>
<pre><code>Warning: package &#39;biomaRt&#39; was built under R version 4.3.2</code></pre>
<pre class="r"><code>library(gprofiler2)</code></pre>
<pre><code>Warning: package &#39;gprofiler2&#39; was built under R version 4.3.3</code></pre>
<pre class="r"><code>library(AnnotationDbi)</code></pre>
</div>
<div id="load-data" class="section level2">
<h2><strong>📌 Load Data</strong></h2>
<pre class="r"><code># Read the file
file_path &lt;- &quot;data/Macrophage/IFN_gamma_primed_macrophages_DEGs.csv&quot;  

IFN_gamma_primed_macrophages_DEGs &lt;- read.csv(file_path, header = TRUE)

# Extract rat gene symbols (replace &quot;Symbol&quot; with the actual column name for gene symbols)
rat_genes &lt;- IFN_gamma_primed_macrophages_DEGs$Symbol

# Map rat gene symbols to human homologs
homologs &lt;- gorth(query = rat_genes,
                  source_organism = &quot;rnorvegicus&quot;,  # Rat organism code
                  target_organism = &quot;hsapiens&quot;)  # Human organism code


macrophage_new&lt;- data.frame(homologs$ortholog_name)

# Map gene symbols to Entrez IDs using org.Hs.eg.db
macrophage_new &lt;- macrophage_new %&gt;%
  mutate(Entrez_ID = mapIds(org.Hs.eg.db,
                            keys = homologs.ortholog_name,
                            column = &quot;ENTREZID&quot;,
                            keytype = &quot;SYMBOL&quot;,
                            multiVals = &quot;first&quot;))</code></pre>
</div>
<div id="proportion-of-de-genes-across-response-groups"
class="section level2">
<h2><strong>📌 Proportion of DE genes across response
groups</strong></h2>
<pre class="r"><code># Load the saved datasets
prob_all_1 &lt;- read.csv(&quot;data/prob_all_1.csv&quot;)$Entrez_ID
prob_all_2 &lt;- read.csv(&quot;data/prob_all_2.csv&quot;)$Entrez_ID
prob_all_3 &lt;- read.csv(&quot;data/prob_all_3.csv&quot;)$Entrez_ID
prob_all_4 &lt;- read.csv(&quot;data/prob_all_4.csv&quot;)$Entrez_ID

# Example Response Groups Data (Replace with actual data)
response_groups &lt;- list(
  &quot;Non Response&quot; = prob_all_1, # Replace &#39;prob_all_1&#39;, &#39;prob_all_2&#39;, etc. with your actual response group dataframes
  &quot;CX_DOX Shared Late Response&quot; = prob_all_2,
  &quot;DOX-Specific Response&quot; = prob_all_3,
  &quot;Late High Dose DOX-Specific Response&quot; = prob_all_4
)

# Combine response groups into a single dataframe
response_groups_df &lt;- bind_rows(
  lapply(response_groups, function(ids) {
    data.frame(Entrez_ID = ids)
  }),
  .id = &quot;Set&quot;
)

# Step 2: Match Overlap Genes with Response Groups
# Classify genes as DEG (match) or Non-DEG (no match)
response_groups_df &lt;- response_groups_df %&gt;%
  mutate(
    DEG_Status = ifelse(Entrez_ID %in% macrophage_new$Entrez_ID, &quot;DEG&quot;, &quot;Non-DEG&quot;)
  )

# Step 3: Calculate Proportions
proportion_data &lt;- response_groups_df %&gt;%
  group_by(Set, DEG_Status) %&gt;%
  summarize(Count = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Set) %&gt;%
  mutate(Percentage = (Count / sum(Count)) * 100)

# Step 4: Perform Chi-Square Tests (Refactored Version)
# Get counts for the Non Response group
non_response_counts &lt;- proportion_data %&gt;%
  filter(Set == &quot;Non Response&quot;) %&gt;%
  dplyr::select(DEG_Status, Count) %&gt;%
  {setNames(.$Count, .$DEG_Status)}  # Create named vector for Non Response counts

# Perform chi-square test for selected response groups
chi_results &lt;- proportion_data %&gt;%
  filter(Set != &quot;Non Response&quot;) %&gt;% # Exclude &quot;Non Response&quot;
  group_by(Set) %&gt;%
  summarize(
    p_value = {
      # Extract counts for the current response group
      group_counts &lt;- Count[DEG_Status %in% c(&quot;DEG&quot;, &quot;Non-DEG&quot;)]
      # Ensure there are no missing categories, fill with 0 if missing
      if (!&quot;DEG&quot; %in% DEG_Status) group_counts &lt;- c(group_counts, 0)
      if (!&quot;Non-DEG&quot; %in% DEG_Status) group_counts &lt;- c(0, group_counts)
      # Create contingency table
      contingency_table &lt;- matrix(c(
        group_counts[1], group_counts[2],
        non_response_counts[&quot;DEG&quot;], non_response_counts[&quot;Non-DEG&quot;]
      ), nrow = 2, byrow = TRUE)
      # Debugging: Print the contingency table
      print(paste(&quot;Set:&quot;, unique(Set)))
      print(&quot;Contingency Table:&quot;)
      print(contingency_table)
      # Perform chi-square test
      if (all(contingency_table &gt;= 0 &amp; is.finite(contingency_table))) {
        chisq.test(contingency_table)$p.value
      } else {
        NA
      }
    },
    .groups = &quot;drop&quot;
  ) %&gt;%
  mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))

# Step 5: Merge Results and Plot Proportions
# Merge chi-square results back into proportion data
proportion_data &lt;- proportion_data %&gt;%
  left_join(chi_results %&gt;% dplyr::select(Set, Significance), by = &quot;Set&quot;)

# Define the correct order for response groups
response_order &lt;- c(
  &quot;Non Response&quot;,
  &quot;CX_DOX Shared Late Response&quot;,
  &quot;DOX-Specific Response&quot;,
  &quot;Late High Dose DOX-Specific Response&quot;
)
proportion_data$Set &lt;- factor(proportion_data$Set, levels = response_order)

# Plot proportions with significance stars
ggplot(proportion_data, aes(x = Set, y = Percentage, fill = DEG_Status)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(
    data = proportion_data %&gt;% distinct(Set, Significance),
    aes(x = Set, y = 105, label = Significance), # Position stars above bars
    inherit.aes = FALSE,
    size = 6,
    color = &quot;black&quot;,
    hjust = 0.5
  ) +
  scale_fill_manual(values = c(&quot;DEG&quot; = &quot;#e41a1c&quot;, &quot;Non-DEG&quot; = &quot;#377eb8&quot;)) +
  labs(
    title = &quot;Proportion of DEGs and Non-DEGs Across Response Groups\n(IFN_gamma_primed_macrophages_DEGs)&quot;,
    x = &quot;Response Groups&quot;,
    y = &quot;Percentage&quot;,
    fill = &quot;Category&quot;
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
    legend.title = element_blank(),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2)
  )</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_text()`).</code></pre>
<p><img src="figure/Macrophage.Rmd/load_data_tissue-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-load_data_tissue-1">
Past versions of load_data_tissue-1.png
</button>
</p>
<div id="fig-load_data_tissue-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/46aa22b9d3ecdb0423e688fbc2c17281dcc9bddd/docs/figure/Macrophage.Rmd/load_data_tissue-1.png" target="_blank">46aa22b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="proportion-of-macrophage-genes-in-cx-and-dox-degs"
class="section level2">
<h2><strong>📌 Proportion of Macrophage genes in CX and DOX
DEGs</strong></h2>
<pre class="r"><code># Load necessary libraries
library(dplyr)
library(ggplot2)
library(tidyr)
library(org.Hs.eg.db)
library(AnnotationDbi)

# Load DEGs Data
CX_0.1_3 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;)
CX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;)
CX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;)
CX_0.5_3 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)

DOX_0.1_3 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;)
DOX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;)
DOX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;)
DOX_0.5_3 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

# Extract Significant DEGs
DEG1 &lt;- as.character(CX_0.1_3$Entrez_ID[CX_0.1_3$adj.P.Val &lt; 0.05])
DEG2 &lt;- as.character(CX_0.1_24$Entrez_ID[CX_0.1_24$adj.P.Val &lt; 0.05])
DEG3 &lt;- as.character(CX_0.1_48$Entrez_ID[CX_0.1_48$adj.P.Val &lt; 0.05])
DEG4 &lt;- as.character(CX_0.5_3$Entrez_ID[CX_0.5_3$adj.P.Val &lt; 0.05])
DEG5 &lt;- as.character(CX_0.5_24$Entrez_ID[CX_0.5_24$adj.P.Val &lt; 0.05])
DEG6 &lt;- as.character(CX_0.5_48$Entrez_ID[CX_0.5_48$adj.P.Val &lt; 0.05])
DEG7 &lt;- as.character(DOX_0.1_3$Entrez_ID[DOX_0.1_3$adj.P.Val &lt; 0.05])
DEG8 &lt;- as.character(DOX_0.1_24$Entrez_ID[DOX_0.1_24$adj.P.Val &lt; 0.05])
DEG9 &lt;- as.character(DOX_0.1_48$Entrez_ID[DOX_0.1_48$adj.P.Val &lt; 0.05])
DEG10 &lt;- as.character(DOX_0.5_3$Entrez_ID[DOX_0.5_3$adj.P.Val &lt; 0.05])
DEG11 &lt;- as.character(DOX_0.5_24$Entrez_ID[DOX_0.5_24$adj.P.Val &lt; 0.05])
DEG12 &lt;- as.character(DOX_0.5_48$Entrez_ID[DOX_0.5_48$adj.P.Val &lt; 0.05])

# Define CX-5461 DEG lists
CX_DEGs &lt;- list(
  &quot;CX_0.1_3&quot; = DEG1, &quot;CX_0.1_24&quot; = DEG2, &quot;CX_0.1_48&quot; = DEG3,
  &quot;CX_0.5_3&quot; = DEG4, &quot;CX_0.5_24&quot; = DEG5, &quot;CX_0.5_48&quot; = DEG6
)

# Define DOX DEG lists
DOX_DEGs &lt;- list(
  &quot;DOX_0.1_3&quot; = DEG7, &quot;DOX_0.1_24&quot; = DEG8, &quot;DOX_0.1_48&quot; = DEG9,
  &quot;DOX_0.5_3&quot; = DEG10, &quot;DOX_0.5_24&quot; = DEG11, &quot;DOX_0.5_48&quot; = DEG12
)

# Load Macrophage dataset (Use `Entrez_ID` for matching)
Macrophage_genes &lt;- na.omit(macrophage_new$Entrez_ID)  # Keep only Entrez_IDs

total_macrophage_genes &lt;- length(Macrophage_genes)  # Total Macrophage genes count

# Function to calculate the proportion of DEGs in Macrophage genes
calculate_macrophage_proportion &lt;- function(deg_list, drug_name) {
  data.frame(
    Sample = names(deg_list),
    Drug = drug_name,
    Macrophage_Target_DEGs = sapply(deg_list, function(ids) sum(ids %in% Macrophage_genes)),
    Non_Macrophage_Target_DEGs = sapply(deg_list, function(ids) total_macrophage_genes - sum(ids %in% Macrophage_genes))
  ) %&gt;%
    mutate(
      Yes_Proportion = (Macrophage_Target_DEGs / total_macrophage_genes) * 100,
      No_Proportion = (Non_Macrophage_Target_DEGs / total_macrophage_genes) * 100
    )
}

# Calculate Proportions
CX_proportion &lt;- calculate_macrophage_proportion(CX_DEGs, &quot;CX-5461&quot;)
DOX_proportion &lt;- calculate_macrophage_proportion(DOX_DEGs, &quot;DOX&quot;)

# Combine Data
proportion_data &lt;- bind_rows(CX_proportion, DOX_proportion)

# Convert to long format for visualization
proportion_long &lt;- proportion_data %&gt;%
  dplyr::select(Sample, Drug, Yes_Proportion, No_Proportion) %&gt;%
  pivot_longer(cols = c(Yes_Proportion, No_Proportion), names_to = &quot;Category&quot;, values_to = &quot;Percentage&quot;) %&gt;%
  mutate(Category = ifelse(Category == &quot;Yes_Proportion&quot;, &quot;Yes&quot;, &quot;No&quot;))

# Ensure correct order of samples on X-axis
sample_order &lt;- c(
  &quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;, &quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;,
  &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;, &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;
)
proportion_long$Sample &lt;- factor(proportion_long$Sample, levels = sample_order, ordered = TRUE)

# Ensure &quot;Yes&quot; is at the Bottom and &quot;No&quot; is at the Top
proportion_long$Category &lt;- factor(proportion_long$Category, levels = c(&quot;Yes&quot;, &quot;No&quot;))

# Generate Stacked Bar Plot
ggplot(proportion_long, aes(x = Sample, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +  # Stacked bars
  facet_wrap(~Drug, scales = &quot;free_x&quot;) +  # Separate CX-5461 and DOX
  scale_y_continuous(labels = scales::percent_format(scale = 1), limits = c(0, 100)) +  # Y-axis as percentage
  scale_fill_manual(values = c(&quot;Yes&quot; = &quot;#e41a1c&quot;, &quot;No&quot; = &quot;#377eb8&quot;)) +  # Yes (Red), No (Blue)
  labs(
    title = &quot;Proportion of IFN gamma primed Macrophage Genes in CX-5461 and DOX DEGs&quot;,
    x = &quot;Samples (CX-5461 and DOX)&quot;,
    y = &quot;Percentage&quot;,
    fill = &quot;Category&quot;
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
    legend.title = element_blank(),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),  
    strip.background = element_blank(),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Macrophage.Rmd/load_DEGs-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-load_DEGs-1">
Past versions of load_DEGs-1.png
</button>
</p>
<div id="fig-load_DEGs-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/07caa2fdb1c88047c32f0d632105c92d1f23a0e3/docs/figure/Macrophage.Rmd/load_DEGs-1.png" target="_blank">07caa2f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="correlation-of-macrophage-genes-with-cx-and-dox-expressed-genes"
class="section level2">
<h2><strong>📌 Correlation of Macrophage genes with CX and DOX expressed
genes</strong></h2>
</div>
<div id="correlation-of-macrophage-genes-with-cx-expressed-genes"
class="section level2">
<h2><strong>📌 Correlation of Macrophage genes with CX expressed
genes</strong></h2>
<pre class="r"><code>macrophage &lt;- read.csv(&quot;data/Macrophage/IFN_gamma_primed_macrophages_DEGs.csv&quot;)

# **Step 2: Extract Rat Gene Symbols**
rat_genes &lt;- unique(macrophage$Symbol)

# **Step 3: Map Rat Gene Symbols to Human Orthologs**
homologs &lt;- gorth(query = rat_genes,
                  source_organism = &quot;rnorvegicus&quot;,  # Rat organism code
                  target_organism = &quot;hsapiens&quot;,  # Human organism code
                  numeric_ns = &quot;&quot;)  # Ensure proper namespace handling

# **Step 4: Check Column Names**
print(colnames(homologs))  # Check available columns</code></pre>
<pre><code>[1] &quot;input_number&quot;  &quot;input&quot;         &quot;input_ensg&quot;    &quot;ensg_number&quot;  
[5] &quot;ortholog_name&quot; &quot;ortholog_ensg&quot; &quot;description&quot;  </code></pre>
<pre class="r"><code># **Step 5: Add Human Orthologs Directly to Macrophage Dataset**
# Create a lookup table for faster mapping
homolog_map &lt;- setNames(homologs$ortholog_name, homologs$input)

# Assign the mapped human genes directly into a new column in macrophage dataset
macrophage$Human_Symbol &lt;- homolog_map[macrophage$Symbol]

# **Step 6: Remove Unmapped Entries (Optional)**
macrophage &lt;- macrophage %&gt;% filter(!is.na(Human_Symbol))

# **Step 7: Save Final Dataset**
write.csv(macrophage, &quot;C:/Work/Postdoc_UTMB/CX-5461 Project/Transcriptome literatures/Macrophage/macrophage_Mapped.csv&quot;, row.names = FALSE)

# **Step 8: Display Output**
print(head(macrophage))  # Show first few rows of updated dataset</code></pre>
<pre><code>    Symbol     logFC Human_Symbol
1   Abcb11  1.449252       ABCB11
2    Abcb4  1.590701        ABCB1
3 AC095390 -2.856968         PUS1
4 AC117058 -5.244999        MCF2L
5   Adgrg5  1.919416       ADGRG5
6      Alk -3.442091          ALK</code></pre>
<pre class="r"><code># **Step 6: Filter Out NA Values (Only Mapped Genes)**
macrophage &lt;- macrophage %&gt;% filter(!is.na(Human_Symbol))

# Ensure Human_Symbol is the first column and remove Symbol
macrophage &lt;- macrophage %&gt;%
  dplyr::select(Human_Symbol, everything(), -Symbol)

# **Step 1: Map Gene Symbols to Entrez IDs using org.Hs.eg.db**
macrophage &lt;- macrophage %&gt;%
  mutate(Entrez_ID = mapIds(org.Hs.eg.db,
                            keys = Human_Symbol,
                            column = &quot;ENTREZID&quot;,
                            keytype = &quot;SYMBOL&quot;,
                            multiVals = &quot;first&quot;))

# **Step 2: Convert Entrez_ID to character to avoid merge issues**
macrophage$Entrez_ID &lt;- as.character(macrophage$Entrez_ID)

CX_0.1_3$Entrez_ID &lt;- as.character(CX_0.1_3$Entrez_ID)
CX_0.5_3$Entrez_ID &lt;- as.character(CX_0.5_3$Entrez_ID)
CX_0.1_24$Entrez_ID &lt;- as.character(CX_0.1_24$Entrez_ID)
CX_0.5_24$Entrez_ID &lt;- as.character(CX_0.5_24$Entrez_ID)
CX_0.1_48$Entrez_ID &lt;- as.character(CX_0.1_48$Entrez_ID)
CX_0.5_48$Entrez_ID &lt;- as.character(CX_0.5_48$Entrez_ID)

# **Step 3: Merge Macrophage dataset with CX at different concentrations &amp; timepoints**
merged_CX_0.1_3 &lt;- merge(macrophage, CX_0.1_3, by = &quot;Entrez_ID&quot;)
merged_CX_0.5_3 &lt;- merge(macrophage, CX_0.5_3, by = &quot;Entrez_ID&quot;)
merged_CX_0.1_24 &lt;- merge(macrophage, CX_0.1_24, by = &quot;Entrez_ID&quot;)
merged_CX_0.5_24 &lt;- merge(macrophage, CX_0.5_24, by = &quot;Entrez_ID&quot;)
merged_CX_0.1_48 &lt;- merge(macrophage, CX_0.1_48, by = &quot;Entrez_ID&quot;)
merged_CX_0.5_48 &lt;- merge(macrophage, CX_0.5_48, by = &quot;Entrez_ID&quot;)

# **Step 4: Remove NA values**
merged_CX_0.1_3 &lt;- na.omit(merged_CX_0.1_3)
merged_CX_0.5_3 &lt;- na.omit(merged_CX_0.5_3)
merged_CX_0.1_24 &lt;- na.omit(merged_CX_0.1_24)
merged_CX_0.5_24 &lt;- na.omit(merged_CX_0.5_24)
merged_CX_0.1_48 &lt;- na.omit(merged_CX_0.1_48)
merged_CX_0.5_48 &lt;- na.omit(merged_CX_0.5_48)

# **Step 5: Rename columns to avoid conflicts**
colnames(merged_CX_0.1_3) &lt;- colnames(merged_CX_0.5_3) &lt;-
  colnames(merged_CX_0.1_24) &lt;- colnames(merged_CX_0.5_24) &lt;-
  colnames(merged_CX_0.1_48) &lt;- colnames(merged_CX_0.5_48) &lt;-
  c(&quot;Entrez_ID&quot;, &quot;Symbol_Macrophage&quot;, &quot;logFC_Macrophage&quot;, &quot;logFC_CX&quot;, &quot;AveExpr_CX&quot;, &quot;t_CX&quot;, &quot;P.Value_CX&quot;, &quot;adj.P.Val_CX&quot;, &quot;B_CX&quot;)

# **Step 6: Add timepoint and concentration labels for faceting**
merged_CX_0.1_3$Timepoint &lt;- &quot;3hr&quot;
merged_CX_0.5_3$Timepoint &lt;- &quot;3hr&quot;
merged_CX_0.1_24$Timepoint &lt;- &quot;24hr&quot;
merged_CX_0.5_24$Timepoint &lt;- &quot;24hr&quot;
merged_CX_0.1_48$Timepoint &lt;- &quot;48hr&quot;
merged_CX_0.5_48$Timepoint &lt;- &quot;48hr&quot;

merged_CX_0.1_3$Concentration &lt;- &quot;0.1 µM&quot;
merged_CX_0.5_3$Concentration &lt;- &quot;0.5 µM&quot;
merged_CX_0.1_24$Concentration &lt;- &quot;0.1 µM&quot;
merged_CX_0.5_24$Concentration &lt;- &quot;0.5 µM&quot;
merged_CX_0.1_48$Concentration &lt;- &quot;0.1 µM&quot;
merged_CX_0.5_48$Concentration &lt;- &quot;0.5 µM&quot;

# **Step 7: Combine all datasets into a single data frame**
merged_data_macrophage &lt;- rbind(
  merged_CX_0.1_3[, c(&quot;Entrez_ID&quot;, &quot;logFC_CX&quot;, &quot;logFC_Macrophage&quot;, &quot;Timepoint&quot;, &quot;Concentration&quot;)],
  merged_CX_0.5_3[, c(&quot;Entrez_ID&quot;, &quot;logFC_CX&quot;, &quot;logFC_Macrophage&quot;, &quot;Timepoint&quot;, &quot;Concentration&quot;)],
  merged_CX_0.1_24[, c(&quot;Entrez_ID&quot;, &quot;logFC_CX&quot;, &quot;logFC_Macrophage&quot;, &quot;Timepoint&quot;, &quot;Concentration&quot;)],
  merged_CX_0.5_24[, c(&quot;Entrez_ID&quot;, &quot;logFC_CX&quot;, &quot;logFC_Macrophage&quot;, &quot;Timepoint&quot;, &quot;Concentration&quot;)],
  merged_CX_0.1_48[, c(&quot;Entrez_ID&quot;, &quot;logFC_CX&quot;, &quot;logFC_Macrophage&quot;, &quot;Timepoint&quot;, &quot;Concentration&quot;)],
  merged_CX_0.5_48[, c(&quot;Entrez_ID&quot;, &quot;logFC_CX&quot;, &quot;logFC_Macrophage&quot;, &quot;Timepoint&quot;, &quot;Concentration&quot;)]
)

# **Ensure timepoints are in correct order**
merged_data_macrophage$Timepoint &lt;- factor(merged_data_macrophage$Timepoint, levels = c(&quot;3hr&quot;, &quot;24hr&quot;, &quot;48hr&quot;))

# **Step 8: Compute correlations for each facet**
correlations &lt;- merged_data_macrophage %&gt;%
  group_by(Concentration, Timepoint) %&gt;%
  summarise(
    r_value = cor(logFC_CX, logFC_Macrophage, method = &quot;pearson&quot;),
    p_value = cor.test(logFC_CX, logFC_Macrophage, method = &quot;pearson&quot;)$p.value,
    .groups = &quot;drop&quot;
  )

# **Step 9: Create correlation annotation data**
correlation_data &lt;- correlations %&gt;%
  mutate(
    x = 1.5,  # Adjusted to fit within fixed axis range (-5 to 2)
    y = max(merged_data_macrophage$logFC_Macrophage, na.rm = TRUE) * 0.85,
    label = paste0(&quot;r = &quot;, round(r_value, 3), &quot;\np = &quot;, signif(p_value, 3))
  )

# **Step 10: Create styled scatter plot with fixed X-axis range and ordered timepoints**
scatter_plot_macrophage &lt;- ggplot(merged_data_macrophage, aes(x = logFC_CX, y = logFC_Macrophage)) +
  geom_point(alpha = 0.6, color = &quot;black&quot;) +
  geom_smooth(method = &quot;lm&quot;, color = &quot;black&quot;, se = FALSE) +
  scale_x_continuous(limits = c(-6, 2)) +  # Fixed X-axis range
  labs(
    title = &quot;Correlation between CX and IFN_gamma_primed_macrophages logFC&quot;,
    x = &quot;logFC (CX)&quot;,
    y = &quot;logFC (Macrophage)&quot;
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 2),  # Outer border
    strip.background = element_rect(fill = &quot;white&quot;, color = &quot;black&quot;, linewidth = 1.5),
    strip.text = element_text(size = 12, face = &quot;bold&quot;, color = &quot;black&quot;)
  ) +
  facet_grid(Timepoint ~ Concentration, scales = &quot;fixed&quot;) +  # Ensures correct timepoint order
  geom_text(data = correlation_data,
            aes(x = x, y = y, label = label),
            inherit.aes = FALSE, size = 3, fontface = &quot;bold&quot;)

# **Step 11: Display the plot**
print(scatter_plot_macrophage)</code></pre>
<pre><code>Warning: Removed 1 row containing non-finite outside the scale range
(`stat_smooth()`).</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/Macrophage.Rmd/Data_prep-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Data_prep-1">
Past versions of Data_prep-1.png
</button>
</p>
<div id="fig-Data_prep-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/46aa22b9d3ecdb0423e688fbc2c17281dcc9bddd/docs/figure/Macrophage.Rmd/Data_prep-1.png" target="_blank">46aa22b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="correlation-of-myeloma-genes-with-dox-expressed-genes"
class="section level2">
<h2><strong>📌 Correlation of Myeloma genes with DOX expressed
genes</strong></h2>
<pre class="r"><code># **Step 2: Convert Entrez_ID to character to avoid merge issues**
macrophage$Entrez_ID &lt;- as.character(macrophage$Entrez_ID)

DOX_0.1_3$Entrez_ID &lt;- as.character(DOX_0.1_3$Entrez_ID)
DOX_0.5_3$Entrez_ID &lt;- as.character(DOX_0.5_3$Entrez_ID)
DOX_0.1_24$Entrez_ID &lt;- as.character(DOX_0.1_24$Entrez_ID)
DOX_0.5_24$Entrez_ID &lt;- as.character(DOX_0.5_24$Entrez_ID)
DOX_0.1_48$Entrez_ID &lt;- as.character(DOX_0.1_48$Entrez_ID)
DOX_0.5_48$Entrez_ID &lt;- as.character(DOX_0.5_48$Entrez_ID)

# **Step 3: Merge Macrophage dataset with DOX at different concentrations &amp; timepoints**
merged_DOX_0.1_3 &lt;- merge(macrophage, DOX_0.1_3, by = &quot;Entrez_ID&quot;)
merged_DOX_0.5_3 &lt;- merge(macrophage, DOX_0.5_3, by = &quot;Entrez_ID&quot;)
merged_DOX_0.1_24 &lt;- merge(macrophage, DOX_0.1_24, by = &quot;Entrez_ID&quot;)
merged_DOX_0.5_24 &lt;- merge(macrophage, DOX_0.5_24, by = &quot;Entrez_ID&quot;)
merged_DOX_0.1_48 &lt;- merge(macrophage, DOX_0.1_48, by = &quot;Entrez_ID&quot;)
merged_DOX_0.5_48 &lt;- merge(macrophage, DOX_0.5_48, by = &quot;Entrez_ID&quot;)

# **Step 4: Remove NA values**
merged_DOX_0.1_3 &lt;- na.omit(merged_DOX_0.1_3)
merged_DOX_0.5_3 &lt;- na.omit(merged_DOX_0.5_3)
merged_DOX_0.1_24 &lt;- na.omit(merged_DOX_0.1_24)
merged_DOX_0.5_24 &lt;- na.omit(merged_DOX_0.5_24)
merged_DOX_0.1_48 &lt;- na.omit(merged_DOX_0.1_48)
merged_DOX_0.5_48 &lt;- na.omit(merged_DOX_0.5_48)

# **Step 5: Rename columns to avoid conflicts**
colnames(merged_DOX_0.1_3) &lt;- colnames(merged_DOX_0.5_3) &lt;-
  colnames(merged_DOX_0.1_24) &lt;- colnames(merged_DOX_0.5_24) &lt;-
  colnames(merged_DOX_0.1_48) &lt;- colnames(merged_DOX_0.5_48) &lt;-
  c(&quot;Entrez_ID&quot;, &quot;Symbol_Macrophage&quot;, &quot;logFC_Macrophage&quot;, &quot;logFC_DOX&quot;, &quot;AveExpr_DOX&quot;, &quot;t_DOX&quot;, &quot;P.Value_DOX&quot;, &quot;adj.P.Val_DOX&quot;, &quot;B_DOX&quot;)

# **Step 6: Add timepoint and concentration labels for faceting**
merged_DOX_0.1_3$Timepoint &lt;- &quot;3hr&quot;
merged_DOX_0.5_3$Timepoint &lt;- &quot;3hr&quot;
merged_DOX_0.1_24$Timepoint &lt;- &quot;24hr&quot;
merged_DOX_0.5_24$Timepoint &lt;- &quot;24hr&quot;
merged_DOX_0.1_48$Timepoint &lt;- &quot;48hr&quot;
merged_DOX_0.5_48$Timepoint &lt;- &quot;48hr&quot;

merged_DOX_0.1_3$Concentration &lt;- &quot;0.1 µM&quot;
merged_DOX_0.5_3$Concentration &lt;- &quot;0.5 µM&quot;
merged_DOX_0.1_24$Concentration &lt;- &quot;0.1 µM&quot;
merged_DOX_0.5_24$Concentration &lt;- &quot;0.5 µM&quot;
merged_DOX_0.1_48$Concentration &lt;- &quot;0.1 µM&quot;
merged_DOX_0.5_48$Concentration &lt;- &quot;0.5 µM&quot;

# **Step 7: Combine all datasets into a single data frame**
merged_data_macrophage_dox &lt;- rbind(
  merged_DOX_0.1_3[, c(&quot;Entrez_ID&quot;, &quot;logFC_DOX&quot;, &quot;logFC_Macrophage&quot;, &quot;Timepoint&quot;, &quot;Concentration&quot;)],
  merged_DOX_0.5_3[, c(&quot;Entrez_ID&quot;, &quot;logFC_DOX&quot;, &quot;logFC_Macrophage&quot;, &quot;Timepoint&quot;, &quot;Concentration&quot;)],
  merged_DOX_0.1_24[, c(&quot;Entrez_ID&quot;, &quot;logFC_DOX&quot;, &quot;logFC_Macrophage&quot;, &quot;Timepoint&quot;, &quot;Concentration&quot;)],
  merged_DOX_0.5_24[, c(&quot;Entrez_ID&quot;, &quot;logFC_DOX&quot;, &quot;logFC_Macrophage&quot;, &quot;Timepoint&quot;, &quot;Concentration&quot;)],
  merged_DOX_0.1_48[, c(&quot;Entrez_ID&quot;, &quot;logFC_DOX&quot;, &quot;logFC_Macrophage&quot;, &quot;Timepoint&quot;, &quot;Concentration&quot;)],
  merged_DOX_0.5_48[, c(&quot;Entrez_ID&quot;, &quot;logFC_DOX&quot;, &quot;logFC_Macrophage&quot;, &quot;Timepoint&quot;, &quot;Concentration&quot;)]
)

# **Ensure timepoints are in correct order**
merged_data_macrophage_dox$Timepoint &lt;- factor(merged_data_macrophage_dox$Timepoint, levels = c(&quot;3hr&quot;, &quot;24hr&quot;, &quot;48hr&quot;))

# **Step 8: Compute correlations for each facet**
correlations &lt;- merged_data_macrophage_dox %&gt;%
  group_by(Concentration, Timepoint) %&gt;%
  summarise(
    r_value = cor(logFC_DOX, logFC_Macrophage, method = &quot;pearson&quot;),
    p_value = cor.test(logFC_DOX, logFC_Macrophage, method = &quot;pearson&quot;)$p.value,
    .groups = &quot;drop&quot;
  )

# **Step 9: Create correlation annotation data**
correlation_data &lt;- correlations %&gt;%
  mutate(
    x = 1.5,  # Adjusted to fit within fixed axis range (-5 to 2)
    y = max(merged_data_macrophage_dox$logFC_Macrophage, na.rm = TRUE) * 0.85,
    label = paste0(&quot;r = &quot;, round(r_value, 3), &quot;\np = &quot;, signif(p_value, 3))
  )

# **Step 10: Create styled scatter plot with fixed X-axis range and ordered timepoints**
scatter_plot_macrophage_dox &lt;- ggplot(merged_data_macrophage_dox, aes(x = logFC_DOX, y = logFC_Macrophage)) +
  geom_point(alpha = 0.6, color = &quot;black&quot;) +
  geom_smooth(method = &quot;lm&quot;, color = &quot;black&quot;, se = FALSE) +
  #scale_x_continuous(limits = c(-5, 2)) +  # Fixed X-axis range
  labs(
    title = &quot;Correlation between DOX and IFN_gamma_primed_macrophages logFC&quot;,
    x = &quot;logFC (DOX)&quot;,
    y = &quot;logFC (Macrophage)&quot;
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = 14, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 2),  # Outer border
    strip.background = element_rect(fill = &quot;white&quot;, color = &quot;black&quot;, linewidth = 1.5),
    strip.text = element_text(size = 12, face = &quot;bold&quot;, color = &quot;black&quot;)
  ) +
  facet_grid(Timepoint ~ Concentration, scales = &quot;fixed&quot;) +  # Ensures correct timepoint order
  geom_text(data = correlation_data,
            aes(x = x, y = y, label = label),
            inherit.aes = FALSE, size = 3, fontface = &quot;bold&quot;)

# **Step 11: Display the plot**
print(scatter_plot_macrophage_dox)</code></pre>
<p><img src="figure/Macrophage.Rmd/Data_DOX-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Data_DOX-1">
Past versions of Data_DOX-1.png
</button>
</p>
<div id="fig-Data_DOX-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/46aa22b9d3ecdb0423e688fbc2c17281dcc9bddd/docs/figure/Macrophage.Rmd/Data_DOX-1.png" target="_blank">46aa22b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="proportion-of-macrophage-genes-in-corrmotif-0.1-micromolar"
class="section level2">
<h2><strong>📌 Proportion of Macrophage Genes in corrmotif 0.1
micromolar</strong></h2>
<pre class="r"><code># Load necessary libraries
library(dplyr)
library(ggplot2)
library(tidyr)
library(org.Hs.eg.db)
library(gprofiler2)

# Read Macrophage Dataset
file_path &lt;- &quot;data/Macrophage/IFN_gamma_primed_macrophages_DEGs.csv&quot;  
IFN_gamma_primed_macrophages_DEGs &lt;- read.csv(file_path, header = TRUE)

# Extract rat gene symbols
rat_genes &lt;- IFN_gamma_primed_macrophages_DEGs$Symbol

# Map rat gene symbols to human homologs
homologs &lt;- gorth(query = rat_genes,
                  source_organism = &quot;rnorvegicus&quot;,  # Rat organism code
                  target_organism = &quot;hsapiens&quot;)  # Human organism code

# Convert to dataframe and map gene symbols to Entrez IDs
macrophage_new &lt;- data.frame(homologs$ortholog_name) %&gt;%
  mutate(Entrez_ID = mapIds(org.Hs.eg.db,
                            keys = homologs.ortholog_name,
                            column = &quot;ENTREZID&quot;,
                            keytype = &quot;SYMBOL&quot;,
                            multiVals = &quot;first&quot;)) %&gt;%
  na.omit()

# Load Corrmotif Groups for 0.1 Concentration
prob_groups_0.1 &lt;- list(
  &quot;Non Response (0.1)&quot; = read.csv(&quot;data/prob_1_0.1.csv&quot;)$Entrez_ID,
  &quot;CX_DOX mid-late (0.1)&quot; = read.csv(&quot;data/prob_2_0.1.csv&quot;)$Entrez_ID,
   &quot;DOX only mid-late (0.1)&quot;= read.csv(&quot;data/prob_3_0.1.csv&quot;)$Entrez_ID
)

# Function to calculate proportions in macrophage dataset
calculate_proportion &lt;- function(deg_list, group_name) {
  total_group_genes &lt;- length(deg_list)
  matched_genes &lt;- sum(deg_list %in% macrophage_new$Entrez_ID)
  
  data.frame(
    Group = group_name,
    Macrophage_DEGs = matched_genes,
    Non_Macrophage_DEGs = total_group_genes - matched_genes
  ) %&gt;%
    mutate(
      Yes_Proportion = (Macrophage_DEGs / total_group_genes) * 100,
      No_Proportion = (Non_Macrophage_DEGs / total_group_genes) * 100
    )
}

# Calculate Proportions for Each Group
proportion_data &lt;- bind_rows(
  calculate_proportion(prob_groups_0.1[[&quot;Non Response (0.1)&quot;]], &quot;Non Response (0.1)&quot;),
   calculate_proportion(prob_groups_0.1[[&quot;CX_DOX mid-late (0.1)&quot;]], &quot;CX_DOX mid-late (0.1)&quot;),
  calculate_proportion(prob_groups_0.1[[&quot;DOX only mid-late (0.1)&quot;]], &quot;DOX only mid-late (0.1)&quot;)
 
)

# Convert Data to Long Format
proportion_long &lt;- proportion_data %&gt;%
  pivot_longer(cols = c(Yes_Proportion, No_Proportion), names_to = &quot;Category&quot;, values_to = &quot;Percentage&quot;) %&gt;%
  mutate(Category = ifelse(Category == &quot;Yes_Proportion&quot;, &quot;Yes&quot;, &quot;No&quot;))

# Perform Chi-Square Test Comparing Groups to Non-Response
test_results &lt;- data.frame(Group = character(), P_Value = numeric())

target_groups &lt;- c(&quot;CX_DOX mid-late (0.1)&quot;,&quot;DOX only mid-late (0.1)&quot;)
for (grp in target_groups) {
  contingency_table &lt;- matrix(
    c(
      sum(prob_groups_0.1[[grp]] %in% macrophage_new$Entrez_ID),
      length(prob_groups_0.1[[grp]]) - sum(prob_groups_0.1[[grp]] %in% macrophage_new$Entrez_ID),
      sum(prob_groups_0.1[[&quot;Non Response (0.1)&quot;]] %in% macrophage_new$Entrez_ID),
      length(prob_groups_0.1[[&quot;Non Response (0.1)&quot;]]) - sum(prob_groups_0.1[[&quot;Non Response (0.1)&quot;]] %in% macrophage_new$Entrez_ID)
    ),
    nrow = 2, byrow = TRUE
  )
  test_result &lt;- chisq.test(contingency_table)
  test_results &lt;- rbind(test_results, data.frame(Group = grp, P_Value = test_result$p.value))
}

# Add Significance Stars
test_results$Significant &lt;- ifelse(test_results$P_Value &lt; 0.05, &quot;*&quot;, &quot;&quot;)
proportion_long &lt;- left_join(proportion_long, test_results, by = &quot;Group&quot;)

# Ensure Correct Ordering
proportion_long$Group &lt;- factor(proportion_long$Group, levels = c(&quot;Non Response (0.1)&quot;, &quot;CX_DOX mid-late (0.1)&quot;,&quot;DOX only mid-late (0.1)&quot;))
proportion_long$Category &lt;- factor(proportion_long$Category, levels = c(&quot;Yes&quot;, &quot;No&quot;))

# Generate Stacked Bar Plot with Significance Annotations
ggplot(proportion_long, aes(x = Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(data = subset(proportion_long, Significant == &quot;*&quot;),
            aes(x = Group, y = 102, label = &quot;*&quot;),
            size = 6, color = &quot;black&quot;, fontface = &quot;bold&quot;) +
  scale_y_continuous(labels = scales::percent_format(scale = 1), limits = c(0, 105)) +
  scale_fill_manual(values = c(&quot;Yes&quot; = &quot;#e41a1c&quot;, &quot;No&quot; = &quot;#377eb8&quot;)) +
  labs(
    title = &quot;Proportion of Macrophage Genes in\n0.1 Corrmotif Response Groups&quot;,
    x = &quot;Group&quot;,
    y = &quot;Percentage&quot;,
    fill = &quot;Category&quot;
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
    legend.title = element_blank(),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    strip.background = element_blank(),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Macrophage.Rmd/prop%20DNA-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-prop-DNA-1">
Past versions of "prop DNA-1.png"
</button>
</p>
<div id="fig-no-spaces-prop-DNA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/10d28dd7ba8a410e15ba473a9c14c68fd3eb544a/docs/figure/Macrophage.Rmd/prop DNA-1.png" target="_blank">10d28dd</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="proportion-of-macrophage-genes-in-corrmotif-0.5-micromolar"
class="section level2">
<h2><strong>📌 Proportion of Macrophage Genes in corrmotif 0.5
micromolar</strong></h2>
<pre class="r"><code># Load necessary libraries
library(dplyr)
library(ggplot2)
library(tidyr)
library(org.Hs.eg.db)
library(gprofiler2)

# Read Macrophage Dataset
file_path &lt;- &quot;data/Macrophage/IFN_gamma_primed_macrophages_DEGs.csv&quot;  
IFN_gamma_primed_macrophages_DEGs &lt;- read.csv(file_path, header = TRUE)

# Extract rat gene symbols
rat_genes &lt;- IFN_gamma_primed_macrophages_DEGs$Symbol

# Map rat gene symbols to human homologs
homologs &lt;- gorth(query = rat_genes,
                  source_organism = &quot;rnorvegicus&quot;,  # Rat organism code
                  target_organism = &quot;hsapiens&quot;)  # Human organism code

# Convert to dataframe and map gene symbols to Entrez IDs
macrophage_new &lt;- data.frame(homologs$ortholog_name) %&gt;%
  mutate(Entrez_ID = mapIds(org.Hs.eg.db,
                            keys = homologs.ortholog_name,
                            column = &quot;ENTREZID&quot;,
                            keytype = &quot;SYMBOL&quot;,
                            multiVals = &quot;first&quot;)) %&gt;%
  na.omit()

# Load Corrmotif Groups for 0.5 Concentration
prob_groups_0.5 &lt;- list(
  &quot;Non Response (0.5)&quot; = read.csv(&quot;data/prob_1_0.5.csv&quot;)$Entrez_ID,
  &quot;DOX-specific response (0.5)&quot; = read.csv(&quot;data/prob_2_0.5.csv&quot;)$Entrez_ID,
  &quot;DOX only mid-late response (0.5)&quot; = read.csv(&quot;data/prob_3_0.5.csv&quot;)$Entrez_ID,
  &quot;CX DOX (early) response (0.5)&quot; = read.csv(&quot;data/prob_4_0.5.csv&quot;)$Entrez_ID,
  &quot;DOX + CX (mid-late) response (0.5)&quot; = read.csv(&quot;data/prob_5_0.5.csv&quot;)$Entrez_ID
)

# Function to calculate proportions in macrophage dataset
calculate_proportion &lt;- function(deg_list, group_name) {
  total_group_genes &lt;- length(deg_list)
  matched_genes &lt;- sum(deg_list %in% macrophage_new$Entrez_ID)
  
  data.frame(
    Group = group_name,
    Macrophage_DEGs = matched_genes,
    Non_Macrophage_DEGs = total_group_genes - matched_genes
  ) %&gt;%
    mutate(
      Yes_Proportion = (Macrophage_DEGs / total_group_genes) * 100,
      No_Proportion = (Non_Macrophage_DEGs / total_group_genes) * 100
    )
}

# Calculate Proportions for Each Group
proportion_data &lt;- bind_rows(
  calculate_proportion(prob_groups_0.5[[&quot;Non Response (0.5)&quot;]], &quot;Non Response (0.5)&quot;),
  calculate_proportion(prob_groups_0.5[[&quot;DOX-specific response (0.5)&quot;]], &quot;DOX-specific response (0.5)&quot;),
  calculate_proportion(prob_groups_0.5[[&quot;DOX only mid-late response (0.5)&quot;]], &quot;DOX only mid-late response (0.5)&quot;),
  calculate_proportion(prob_groups_0.5[[&quot;CX DOX (early) response (0.5)&quot;]], &quot;CX DOX (early) response (0.5)&quot;),
  calculate_proportion(prob_groups_0.5[[&quot;DOX + CX (mid-late) response (0.5)&quot;]], &quot;DOX + CX (mid-late) response (0.5)&quot;)
)

# Convert Data to Long Format
proportion_long &lt;- proportion_data %&gt;%
  pivot_longer(cols = c(Yes_Proportion, No_Proportion), names_to = &quot;Category&quot;, values_to = &quot;Percentage&quot;) %&gt;%
  mutate(Category = ifelse(Category == &quot;Yes_Proportion&quot;, &quot;Yes&quot;, &quot;No&quot;))

# Perform Chi-Square Test Comparing Groups to Non-Response
test_results &lt;- data.frame(Group = character(), P_Value = numeric())

target_groups &lt;- c(&quot;DOX-specific response (0.5)&quot;, &quot;DOX only mid-late response (0.5)&quot;, &quot;CX DOX (early) response (0.5)&quot;, &quot;DOX + CX (mid-late) response (0.5)&quot;)
for (grp in target_groups) {
  contingency_table &lt;- matrix(
    c(
      sum(prob_groups_0.5[[grp]] %in% macrophage_new$Entrez_ID),
      length(prob_groups_0.5[[grp]]) - sum(prob_groups_0.5[[grp]] %in% macrophage_new$Entrez_ID),
      sum(prob_groups_0.5[[&quot;Non Response (0.5)&quot;]] %in% macrophage_new$Entrez_ID),
      length(prob_groups_0.5[[&quot;Non Response (0.5)&quot;]]) - sum(prob_groups_0.5[[&quot;Non Response (0.5)&quot;]] %in% macrophage_new$Entrez_ID)
    ),
    nrow = 2, byrow = TRUE
  )
  test_result &lt;- chisq.test(contingency_table)
  test_results &lt;- rbind(test_results, data.frame(Group = grp, P_Value = test_result$p.value))
}</code></pre>
<pre><code>Warning in chisq.test(contingency_table): Chi-squared approximation may be
incorrect
Warning in chisq.test(contingency_table): Chi-squared approximation may be
incorrect
Warning in chisq.test(contingency_table): Chi-squared approximation may be
incorrect</code></pre>
<pre class="r"><code># Add Significance Stars
test_results$Significant &lt;- ifelse(test_results$P_Value &lt; 0.05, &quot;*&quot;, &quot;&quot;)
proportion_long &lt;- left_join(proportion_long, test_results, by = &quot;Group&quot;)

# Ensure Correct Ordering
proportion_long$Group &lt;- factor(proportion_long$Group, levels = c(&quot;Non Response (0.5)&quot;, &quot;DOX-specific response (0.5)&quot;, &quot;DOX only mid-late response (0.5)&quot;, &quot;CX DOX (early) response (0.5)&quot;, &quot;DOX + CX (mid-late) response (0.5)&quot;))
proportion_long$Category &lt;- factor(proportion_long$Category, levels = c(&quot;Yes&quot;, &quot;No&quot;))

# Generate Stacked Bar Plot with Significance Annotations
ggplot(proportion_long, aes(x = Group, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(data = subset(proportion_long, Significant == &quot;*&quot;),
            aes(x = Group, y = 102, label = &quot;*&quot;),
            size = 6, color = &quot;black&quot;, fontface = &quot;bold&quot;) +
  scale_y_continuous(labels = scales::percent_format(scale = 1), limits = c(0, 105)) +
  scale_fill_manual(values = c(&quot;Yes&quot; = &quot;#e41a1c&quot;, &quot;No&quot; = &quot;#377eb8&quot;)) +
  labs(
    title = &quot;Proportion of Macrophage Genes in\n0.5 Corrmotif Response Groups&quot;,
    x = &quot;Group&quot;,
    y = &quot;Percentage&quot;,
    fill = &quot;Category&quot;
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
    legend.title = element_blank(),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.2),
    strip.background = element_blank(),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Macrophage.Rmd/prop%200.5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-prop-0.5-1">
Past versions of "prop 0.5-1.png"
</button>
</p>
<div id="fig-no-spaces-prop-0.5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/10d28dd7ba8a410e15ba473a9c14c68fd3eb544a/docs/figure/Macrophage.Rmd/prop 0.5-1.png" target="_blank">10d28dd</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 22631)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] gprofiler2_0.2.3       biomaRt_2.58.2         clusterProfiler_4.10.1
 [4] org.Hs.eg.db_3.18.0    AnnotationDbi_1.64.1   IRanges_2.36.0        
 [7] S4Vectors_0.40.1       Biobase_2.62.0         BiocGenerics_0.48.1   
[10] tidyr_1.3.1            dplyr_1.1.4            ggplot2_3.5.1         

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3      rstudioapi_0.17.1       jsonlite_1.8.9         
  [4] magrittr_2.0.3          farver_2.1.2            rmarkdown_2.29         
  [7] fs_1.6.3                zlibbioc_1.48.0         vctrs_0.6.5            
 [10] memoise_2.0.1           RCurl_1.98-1.13         ggtree_3.10.1          
 [13] htmltools_0.5.8.1       progress_1.2.3          curl_6.0.1             
 [16] gridGraphics_0.5-1      sass_0.4.9              bslib_0.8.0            
 [19] htmlwidgets_1.6.4       plyr_1.8.9              plotly_4.10.4          
 [22] cachem_1.0.8            whisker_0.4.1           igraph_2.1.1           
 [25] lifecycle_1.0.4         pkgconfig_2.0.3         Matrix_1.6-1.1         
 [28] R6_2.5.1                fastmap_1.1.1           gson_0.1.0             
 [31] GenomeInfoDbData_1.2.11 digest_0.6.34           aplot_0.2.3            
 [34] enrichplot_1.22.0       colorspace_2.1-0        patchwork_1.3.0        
 [37] rprojroot_2.0.4         RSQLite_2.3.3           labeling_0.4.3         
 [40] filelock_1.0.3          mgcv_1.9-1              httr_1.4.7             
 [43] polyclip_1.10-7         compiler_4.3.0          bit64_4.0.5            
 [46] withr_3.0.2             BiocParallel_1.36.0     viridis_0.6.5          
 [49] DBI_1.2.3               ggforce_0.4.2           MASS_7.3-60            
 [52] rappdirs_0.3.3          HDO.db_0.99.1           tools_4.3.0            
 [55] ape_5.8                 scatterpie_0.2.4        httpuv_1.6.15          
 [58] glue_1.7.0              nlme_3.1-166            GOSemSim_2.28.1        
 [61] promises_1.3.0          grid_4.3.0              shadowtext_0.1.4       
 [64] reshape2_1.4.4          fgsea_1.28.0            generics_0.1.3         
 [67] gtable_0.3.6            data.table_1.14.10      hms_1.1.3              
 [70] xml2_1.3.6              tidygraph_1.3.1         XVector_0.42.0         
 [73] ggrepel_0.9.6           pillar_1.10.1           stringr_1.5.1          
 [76] yulab.utils_0.1.8       later_1.3.2             splines_4.3.0          
 [79] tweenr_2.0.3            BiocFileCache_2.10.2    treeio_1.26.0          
 [82] lattice_0.22-5          bit_4.0.5               tidyselect_1.2.1       
 [85] GO.db_3.18.0            Biostrings_2.70.1       knitr_1.49             
 [88] git2r_0.35.0            gridExtra_2.3           xfun_0.50              
 [91] graphlayouts_1.2.0      stringi_1.8.3           workflowr_1.7.1        
 [94] lazyeval_0.2.2          ggfun_0.1.8             yaml_2.3.10            
 [97] evaluate_1.0.3          codetools_0.2-20        ggraph_2.2.1           
[100] tibble_3.2.1            qvalue_2.34.0           ggplotify_0.1.2        
[103] cli_3.6.1               munsell_0.5.1           jquerylib_0.1.4        
[106] Rcpp_1.0.12             GenomeInfoDb_1.38.8     dbplyr_2.5.0           
[109] png_0.1-8               XML_3.99-0.17           parallel_4.3.0         
[112] blob_1.2.4              prettyunits_1.2.0       DOSE_3.28.2            
[115] bitops_1.0-7            viridisLite_0.4.2       tidytree_0.4.6         
[118] scales_1.3.0            purrr_1.0.2             crayon_1.5.3           
[121] rlang_1.1.3             cowplot_1.1.3           fastmatch_1.1-4        
[124] KEGGREST_1.42.0        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
