<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Overlap of GO Terms in DEGs</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Paul_CX_2025</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Overlap of GO Terms in DEGs</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-02
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>Paul_CX_2025/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttreec5f10c43ccea82d6a53f4730efcb8ed5e21d8742targetblankc5f10c4a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/tree/c5f10c43ccea82d6a53f4730efcb8ed5e21d8742" target="_blank">c5f10c4</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttreec5f10c43ccea82d6a53f4730efcb8ed5e21d8742targetblankc5f10c4a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sayanpaul01/Paul_CX_2025/tree/c5f10c43ccea82d6a53f4730efcb8ed5e21d8742" target="_blank">c5f10c4</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    0.1 box.svg
    Ignored:    Rplot04.svg
    Ignored:    analysis/Corrmotif_Conc.html

Untracked files:
    Untracked:  0.1 density.svg
    Untracked:  0.1.emf
    Untracked:  0.1.svg
    Untracked:  0.5 box.svg
    Untracked:  0.5 density.svg
    Untracked:  0.5.svg
    Untracked:  Additional/
    Untracked:  Autosome factors.svg
    Untracked:  CX_5461_Pattern_Genes_24hr.csv
    Untracked:  CX_5461_Pattern_Genes_3hr.csv
    Untracked:  Cell viability box plot.svg
    Untracked:  DEG GO terms.svg
    Untracked:  DNA damage associated GO terms.svg
    Untracked:  DRC1.svg
    Untracked:  Figure 1.jpeg
    Untracked:  Figure 1.pdf
    Untracked:  Figure_CM_Purity.pdf
    Untracked:  G Quadruplex DEGs.svg
    Untracked:  PC2 Vs PC3 Autosome.svg
    Untracked:  PCA autosome.svg
    Untracked:  Rplot 18.svg
    Untracked:  Rplot.svg
    Untracked:  Rplot01.svg
    Untracked:  Rplot02.svg
    Untracked:  Rplot03.svg
    Untracked:  Rplot05.svg
    Untracked:  Rplot06.svg
    Untracked:  Rplot07.svg
    Untracked:  Rplot08.jpeg
    Untracked:  Rplot08.svg
    Untracked:  Rplot09.svg
    Untracked:  Rplot10.svg
    Untracked:  Rplot11.svg
    Untracked:  Rplot12.svg
    Untracked:  Rplot13.svg
    Untracked:  Rplot14.svg
    Untracked:  Rplot15.svg
    Untracked:  Rplot16.svg
    Untracked:  Rplot17.svg
    Untracked:  Rplot18.svg
    Untracked:  Rplot19.svg
    Untracked:  Rplot20.svg
    Untracked:  Rplot21.svg
    Untracked:  Rplot22.svg
    Untracked:  Rplot23.svg
    Untracked:  Rplot24.svg
    Untracked:  TOP2B.bed
    Untracked:  TS HPA (Violin).svg
    Untracked:  TS HPA.svg
    Untracked:  TS_HA.svg
    Untracked:  TS_HV.svg
    Untracked:  Violin HA.svg
    Untracked:  Violin HV (CX vs DOX).svg
    Untracked:  Violin HV.svg
    Untracked:  data/AF.csv
    Untracked:  data/AF_Mapped.csv
    Untracked:  data/AF_genes.csv
    Untracked:  data/Annotated_DOX_Gene_Table.csv
    Untracked:  data/BP/
    Untracked:  data/CAD_genes.csv
    Untracked:  data/Cardiotox.csv
    Untracked:  data/Cardiotox_mapped.csv
    Untracked:  data/Corrmotif_GO/
    Untracked:  data/DOX_Vald.csv
    Untracked:  data/DOX_Vald_Mapped.csv
    Untracked:  data/DOX_alt.csv
    Untracked:  data/Entrez_Cardiotox.csv
    Untracked:  data/Entrez_Cardiotox_Mapped.csv
    Untracked:  data/GWAS.xlsx
    Untracked:  data/GWAS_SNPs.bed
    Untracked:  data/HF.csv
    Untracked:  data/HF_Mapped.csv
    Untracked:  data/HF_genes.csv
    Untracked:  data/Hypertension_genes.csv
    Untracked:  data/MI_genes.csv
    Untracked:  data/P53_Target_mapped.csv
    Untracked:  data/Sample_annotated.csv
    Untracked:  data/Samples.csv
    Untracked:  data/Samples.xlsx
    Untracked:  data/TOP2A.bed
    Untracked:  data/TOP2A_target.csv
    Untracked:  data/TOP2A_target_lit.csv
    Untracked:  data/TOP2A_target_lit_mapped.csv
    Untracked:  data/TOP2A_target_mapped.csv
    Untracked:  data/TOP2B.bed
    Untracked:  data/TOP2B_target.csv
    Untracked:  data/TOP2B_target_heatmap.csv
    Untracked:  data/TOP2B_target_heatmap_mapped.csv
    Untracked:  data/TOP2B_target_mapped.csv
    Untracked:  data/TS.csv
    Untracked:  data/TS_HPA.csv
    Untracked:  data/TS_HPA_mapped.csv
    Untracked:  data/Toptable_CX_0.1_24.csv
    Untracked:  data/Toptable_CX_0.1_3.csv
    Untracked:  data/Toptable_CX_0.1_48.csv
    Untracked:  data/Toptable_CX_0.5_24.csv
    Untracked:  data/Toptable_CX_0.5_3.csv
    Untracked:  data/Toptable_CX_0.5_48.csv
    Untracked:  data/Toptable_DOX_0.1_24.csv
    Untracked:  data/Toptable_DOX_0.1_3.csv
    Untracked:  data/Toptable_DOX_0.1_48.csv
    Untracked:  data/Toptable_DOX_0.5_24.csv
    Untracked:  data/Toptable_DOX_0.5_3.csv
    Untracked:  data/Toptable_DOX_0.5_48.csv
    Untracked:  data/count.tsv
    Untracked:  data/ts_data_mapped
    Untracked:  results/
    Untracked:  run_bedtools.bat

Unstaged changes:
    Deleted:    analysis/Actox.Rmd
    Modified:   analysis/Overlap_GO_DEG.Rmd
    Modified:   data/DOX_0.5_48 (Combined).csv
    Modified:   data/Total_number_of_Mapped_reads_by_Individuals.csv
    Modified:   data/count.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Overlap_GO_DEG.Rmd</code>) and
HTML (<code>docs/Overlap_GO_DEG.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/d82f62c5a078ce0e438bc3f9a25972c823c5ce9c/analysis/Overlap_GO_DEG.Rmd" target="_blank">d82f62c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-23
</td>
<td>
COmmit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/d82f62c5a078ce0e438bc3f9a25972c823c5ce9c/docs/Overlap_GO_DEG.html" target="_blank">d82f62c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-23
</td>
<td>
COmmit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/bb4b782392d13afe42ba79ea6117aba336465842/docs/Overlap_GO_DEG.html" target="_blank">bb4b782</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/2e99243ed543b0c8d7b534b36297e34dc55b914f/docs/Overlap_GO_DEG.html" target="_blank">2e99243</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/7f9dc0289c28242d041bc31e19e61a31b6cdaac4/analysis/Overlap_GO_DEG.Rmd" target="_blank">7f9dc02</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-20
</td>
<td>
Added Overlap of GO Terms in DEGs and updated index
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-required-libraries" class="section level2">
<h2><strong>📌 Load Required Libraries</strong></h2>
<pre class="r"><code>library(ggVennDiagram)
library(ggplot2)
library(dplyr)</code></pre>
</div>
<div id="read-and-process-go-terms" class="section level2">
<h2><strong>📌 Read and Process GO Terms</strong></h2>
<pre class="r"><code># Load GO term data for different conditions
# 📦 Load Required Library
library(dplyr)

# 📁 Load GO Term Enrichment Data (Filtered for p.adjust &lt; 0.05)

CX_0.1_24_GO &lt;- read.csv(&quot;data/BP/Combined_Terms/GO_All_CX_0.1_24.csv&quot;)
CX_0.1_48_GO &lt;- read.csv(&quot;data/BP/Combined_Terms/GO_All_CX_0.1_48.csv&quot;)
CX_0.5_3_GO  &lt;- read.csv(&quot;data/BP/Combined_Terms/GO_All_CX_0.5_3.csv&quot;)
CX_0.5_24_GO &lt;- read.csv(&quot;data/BP/Combined_Terms/GO_All_CX_0.5_24.csv&quot;)
CX_0.5_48_GO &lt;- read.csv(&quot;data/BP/Combined_Terms/GO_All_CX_0.5_48.csv&quot;)

DOX_0.1_3_GO  &lt;- read.csv(&quot;data/BP/Combined_Terms/GO_All_DOX_0.1_3.csv&quot;)
DOX_0.1_24_GO &lt;- read.csv(&quot;data/BP/Combined_Terms/GO_All_DOX_0.1_24.csv&quot;)
DOX_0.1_48_GO &lt;- read.csv(&quot;data/BP/Combined_Terms/GO_All_DOX_0.1_48.csv&quot;)
DOX_0.5_3_GO  &lt;- read.csv(&quot;data/BP/Combined_Terms/GO_All_DOX_0.5_3.csv&quot;)
DOX_0.5_24_GO &lt;- read.csv(&quot;data/BP/Combined_Terms/GO_All_DOX_0.5_24.csv&quot;)
DOX_0.5_48_GO &lt;- read.csv(&quot;data/BP/Combined_Terms/GO_All_DOX_0.5_48.csv&quot;)

# ✅ Extract Significant GO Term IDs (p.adjust &lt; 0.05)

DEG2_GO  &lt;- CX_0.1_24_GO  %&gt;% filter(p.adjust &lt; 0.05) %&gt;% pull(ID)
DEG3_GO  &lt;- CX_0.1_48_GO  %&gt;% filter(p.adjust &lt; 0.05) %&gt;% pull(ID)
DEG4_GO  &lt;- CX_0.5_3_GO   %&gt;% filter(p.adjust &lt; 0.05) %&gt;% pull(ID)
DEG5_GO  &lt;- CX_0.5_24_GO  %&gt;% filter(p.adjust &lt; 0.05) %&gt;% pull(ID)
DEG6_GO  &lt;- CX_0.5_48_GO  %&gt;% filter(p.adjust &lt; 0.05) %&gt;% pull(ID)

DEG7_GO  &lt;- DOX_0.1_3_GO  %&gt;% filter(p.adjust &lt; 0.05) %&gt;% pull(ID)
DEG8_GO  &lt;- DOX_0.1_24_GO %&gt;% filter(p.adjust &lt; 0.05) %&gt;% pull(ID)
DEG9_GO  &lt;- DOX_0.1_48_GO %&gt;% filter(p.adjust &lt; 0.05) %&gt;% pull(ID)
DEG10_GO &lt;- DOX_0.5_3_GO  %&gt;% filter(p.adjust &lt; 0.05) %&gt;% pull(ID)
DEG11_GO &lt;- DOX_0.5_24_GO %&gt;% filter(p.adjust &lt; 0.05) %&gt;% pull(ID)
DEG12_GO &lt;- DOX_0.5_48_GO %&gt;% filter(p.adjust &lt; 0.05) %&gt;% pull(ID)</code></pre>
</div>
<div id="overlap-of-go-terms-across-the-drugs" class="section level2">
<h2><strong>📌 Overlap of GO Terms across the drugs</strong></h2>
</div>
<div id="overlap-of-go-terms-in-cx-5461-vs-veh" class="section level2">
<h2><strong>📌 Overlap of GO Terms in CX-5461 vs VEH</strong></h2>
<pre class="r"><code>CX_Datasets &lt;- list(
  &quot;CX_0.1_24&quot; = DEG2_GO,
  &quot;CX_0.1_48&quot; = DEG3_GO,
  &quot;CX_0.5_3&quot; = DEG4_GO,
  &quot;CX_0.5_24&quot; = DEG5_GO,
  &quot;CX_0.5_48&quot; = DEG6_GO
)

ggVennDiagram(CX_Datasets, label = &quot;count&quot;) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;),
    legend.title = element_text(size = 12, face = &quot;bold&quot;),
    legend.text = element_text(size = 10)
  ) +
  labs(title = &quot;CX-5461 GO Term Overlap&quot;)</code></pre>
<p><img src="figure/Overlap_GO_DEG.Rmd/Venn1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Venn1-1">
Past versions of Venn1-1.png
</button>
</p>
<div id="fig-Venn1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/d82f62c5a078ce0e438bc3f9a25972c823c5ce9c/docs/figure/Overlap_GO_DEG.Rmd/Venn1-1.png" target="_blank">d82f62c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/2e99243ed543b0c8d7b534b36297e34dc55b914f/docs/figure/Overlap_GO_DEG.Rmd/Venn1-1.png" target="_blank">2e99243</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="overlap-of-go-terms-in-dox-vs-veh" class="section level2">
<h2><strong>📌 Overlap of GO Terms in DOX vs VEH</strong></h2>
<pre class="r"><code>DOX_Datasets &lt;- list(
  &quot;DOX_0.1_3&quot; = DEG7_GO,
  &quot;DOX_0.1_24&quot; = DEG8_GO,
  &quot;DOX_0.1_48&quot; = DEG9_GO,
  &quot;DOX_0.5_3&quot; = DEG10_GO,
  &quot;DOX_0.5_24&quot; = DEG11_GO,
  &quot;DOX_0.5_48&quot; = DEG12_GO
)

ggVennDiagram(DOX_Datasets, label = &quot;count&quot;) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;),
    legend.title = element_text(size = 12, face = &quot;bold&quot;),
    legend.text = element_text(size = 10)
  ) +
  labs(title = &quot;DOX GO Term Overlap&quot;)</code></pre>
<p><img src="figure/Overlap_GO_DEG.Rmd/Venn2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Venn2-1">
Past versions of Venn2-1.png
</button>
</p>
<div id="fig-Venn2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/d82f62c5a078ce0e438bc3f9a25972c823c5ce9c/docs/figure/Overlap_GO_DEG.Rmd/Venn2-1.png" target="_blank">d82f62c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/2e99243ed543b0c8d7b534b36297e34dc55b914f/docs/figure/Overlap_GO_DEG.Rmd/Venn2-1.png" target="_blank">2e99243</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="overlap-of-go-terms-across-concentrations"
class="section level2">
<h2><strong>📌 Overlap of GO Terms across concentrations</strong></h2>
</div>
<div id="µm-concentration" class="section level2">
<h2><strong>📌 0.1 µM Concentration</strong></h2>
<pre class="r"><code>Concentration_0_1 &lt;- list(
  &quot;CX_0.1_24&quot; = DEG2_GO,
  &quot;CX_0.1_48&quot; = DEG3_GO,
  &quot;DOX_0.1_3&quot; = DEG7_GO,
  &quot;DOX_0.1_24&quot; = DEG8_GO,
  &quot;DOX_0.1_48&quot; = DEG9_GO
)

ggVennDiagram(Concentration_0_1, label = &quot;count&quot;) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;),
    legend.title = element_text(size = 12, face = &quot;bold&quot;),
    legend.text = element_text(size = 10)
  ) +
  labs(title = &quot;0.1 µM GO Term Overlap&quot;)</code></pre>
<p><img src="figure/Overlap_GO_DEG.Rmd/Venn3-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Venn3-1">
Past versions of Venn3-1.png
</button>
</p>
<div id="fig-Venn3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/d82f62c5a078ce0e438bc3f9a25972c823c5ce9c/docs/figure/Overlap_GO_DEG.Rmd/Venn3-1.png" target="_blank">d82f62c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/2e99243ed543b0c8d7b534b36297e34dc55b914f/docs/figure/Overlap_GO_DEG.Rmd/Venn3-1.png" target="_blank">2e99243</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="µm-concentration-1" class="section level2">
<h2><strong>📌 0.5 µM Concentration</strong></h2>
<pre class="r"><code>Concentration_0_5 &lt;- list(
  &quot;CX_0.5_3&quot; = DEG4_GO,
  &quot;CX_0.5_24&quot; = DEG5_GO,
  &quot;CX_0.5_48&quot; = DEG6_GO,
  &quot;DOX_0.5_3&quot; = DEG10_GO,
  &quot;DOX_0.5_24&quot; = DEG11_GO,
  &quot;DOX_0.5_48&quot; = DEG12_GO
)

ggVennDiagram(Concentration_0_5, label = &quot;count&quot;) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;),
    legend.title = element_text(size = 12, face = &quot;bold&quot;),
    legend.text = element_text(size = 10)
  ) +
  labs(title = &quot;0.5 µM GO Term Overlap&quot;)</code></pre>
<p><img src="figure/Overlap_GO_DEG.Rmd/Venn4-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Venn4-1">
Past versions of Venn4-1.png
</button>
</p>
<div id="fig-Venn4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/d82f62c5a078ce0e438bc3f9a25972c823c5ce9c/docs/figure/Overlap_GO_DEG.Rmd/Venn4-1.png" target="_blank">d82f62c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/2e99243ed543b0c8d7b534b36297e34dc55b914f/docs/figure/Overlap_GO_DEG.Rmd/Venn4-1.png" target="_blank">2e99243</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="overlap-of-go-terms-across-timepoints" class="section level2">
<h2><strong>📌 Overlap of GO Terms Across Timepoints</strong></h2>
</div>
<div id="hours" class="section level2">
<h2><strong>📌3 Hours</strong></h2>
<pre class="r"><code>Timepoint_3hrs &lt;- list(
  &quot;CX_0.5_3&quot; = DEG4_GO,
  &quot;DOX_0.1_3&quot; = DEG7_GO,
  &quot;DOX_0.5_3&quot; = DEG10_GO
)

ggVennDiagram(Timepoint_3hrs, label = &quot;count&quot;) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;),
    legend.title = element_text(size = 12, face = &quot;bold&quot;),
    legend.text = element_text(size = 10)
  ) +
  labs(title = &quot;3 Hours GO Term Overlap&quot;)</code></pre>
<p><img src="figure/Overlap_GO_DEG.Rmd/Venn5-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Venn5-1">
Past versions of Venn5-1.png
</button>
</p>
<div id="fig-Venn5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/d82f62c5a078ce0e438bc3f9a25972c823c5ce9c/docs/figure/Overlap_GO_DEG.Rmd/Venn5-1.png" target="_blank">d82f62c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/2e99243ed543b0c8d7b534b36297e34dc55b914f/docs/figure/Overlap_GO_DEG.Rmd/Venn5-1.png" target="_blank">2e99243</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="hours-1" class="section level2">
<h2><strong>📌24 Hours</strong></h2>
<pre class="r"><code>Timepoint_24hrs &lt;- list(
  &quot;CX_0.1_24&quot; = DEG2_GO,
  &quot;CX_0.5_24&quot; = DEG5_GO,
  &quot;DOX_0.1_24&quot; = DEG8_GO,
  &quot;DOX_0.5_24&quot; = DEG11_GO
)

ggVennDiagram(Timepoint_24hrs, label = &quot;count&quot;) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;),
    legend.title = element_text(size = 12, face = &quot;bold&quot;),
    legend.text = element_text(size = 10)
  ) +
  labs(title = &quot;24 Hours GO Term Overlap&quot;)</code></pre>
<p><img src="figure/Overlap_GO_DEG.Rmd/Venn6-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Venn6-1">
Past versions of Venn6-1.png
</button>
</p>
<div id="fig-Venn6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/d82f62c5a078ce0e438bc3f9a25972c823c5ce9c/docs/figure/Overlap_GO_DEG.Rmd/Venn6-1.png" target="_blank">d82f62c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/2e99243ed543b0c8d7b534b36297e34dc55b914f/docs/figure/Overlap_GO_DEG.Rmd/Venn6-1.png" target="_blank">2e99243</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="hours-2" class="section level2">
<h2><strong>📌48 Hours</strong></h2>
<pre class="r"><code>Timepoint_48hrs &lt;- list(
  &quot;CX_0.1_48&quot; = DEG3_GO,
  &quot;CX_0.5_48&quot; = DEG6_GO,
  &quot;DOX_0.1_48&quot; = DEG9_GO,
  &quot;DOX_0.5_48&quot; = DEG12_GO
)

ggVennDiagram(Timepoint_48hrs, label = &quot;count&quot;) +
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;),
    legend.title = element_text(size = 12, face = &quot;bold&quot;),
    legend.text = element_text(size = 10)
  ) +
  labs(title = &quot;48 Hours GO Term Overlap&quot;)</code></pre>
<p><img src="figure/Overlap_GO_DEG.Rmd/Venn7-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Venn7-1">
Past versions of Venn7-1.png
</button>
</p>
<div id="fig-Venn7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/d82f62c5a078ce0e438bc3f9a25972c823c5ce9c/docs/figure/Overlap_GO_DEG.Rmd/Venn7-1.png" target="_blank">d82f62c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/2e99243ed543b0c8d7b534b36297e34dc55b914f/docs/figure/Overlap_GO_DEG.Rmd/Venn7-1.png" target="_blank">2e99243</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="overlapping-of-bp-terms-across-all-samples-upset-plot"
class="section level2">
<h2><strong>📌Overlapping of BP terms across all samples Upset
plot</strong></h2>
<pre class="r"><code># 📦 Load Required Libraries
library(UpSetR)
library(dplyr)
library(tidyr)

# ✅ Define GO Enrichment Files (Combined_Terms structure, but now All_Terms paths)
go_files &lt;- list(
  &quot;CX_0.1_24&quot;  = &quot;data/BP/All_Terms/GO_BP_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot;  = &quot;data/BP/All_Terms/GO_BP_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot;   = &quot;data/BP/All_Terms/GO_BP_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot;  = &quot;data/BP/All_Terms/GO_BP_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot;  = &quot;data/BP/All_Terms/GO_BP_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot;  = &quot;data/BP/All_Terms/GO_BP_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/BP/All_Terms/GO_BP_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/BP/All_Terms/GO_BP_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot;  = &quot;data/BP/All_Terms/GO_BP_DOX_0.5_3.csv&quot;
)

# ✅ Read and filter (p.adjust &lt; 0.05), extract GO term IDs
gene_sets &lt;- lapply(go_files, function(file) {
  read.csv(file) %&gt;%
    filter(p.adjust &lt; 0.05) %&gt;%
    pull(ID) %&gt;%
    as.character()
})

# 🧮 Build binary matrix
all_ids &lt;- unique(unlist(gene_sets))
binary_matrix &lt;- data.frame(GO_ID = all_ids)

for (name in names(gene_sets)) {
  binary_matrix[[name]] &lt;- as.integer(binary_matrix$GO_ID %in% gene_sets[[name]])
}

# ✅ Prepare matrix for UpSetR
upset_input &lt;- binary_matrix[, -1]
colnames(upset_input) &lt;- names(gene_sets)

# 🎯 Plot UpSet
par(mar = c(10, 4, 2, 2))
upset(upset_input,
      sets = colnames(upset_input),
      order.by = &quot;freq&quot;,
      sets.bar.color = &quot;#56B4E9&quot;,
      mainbar.y.label = &quot;Number of Shared BP (p.adjust &lt; 0.05)&quot;,
      sets.x.label = &quot;BP per Condition&quot;,
      text.scale = 1.2,
      nintersects = 30)</code></pre>
<p><img src="figure/Overlap_GO_DEG.Rmd/Upset1-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Upset1-1">
Past versions of Upset1-1.png
</button>
</p>
<div id="fig-Upset1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/d82f62c5a078ce0e438bc3f9a25972c823c5ce9c/docs/figure/Overlap_GO_DEG.Rmd/Upset1-1.png" target="_blank">d82f62c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="identifying-unique-go-terms" class="section level2">
<h2><strong>📌Identifying Unique GO terms</strong></h2>
<pre class="r"><code># 📦 Load Required Libraries
# 📦 Load Required Libraries
library(dplyr)

# ✅ Step 1: Define File Paths for All_Terms
go_files &lt;- list(
  &quot;CX_0.1_24&quot;  = &quot;data/BP/All_Terms/GO_BP_CX_0.1_24.csv&quot;,
  &quot;CX_0.1_48&quot;  = &quot;data/BP/All_Terms/GO_BP_CX_0.1_48.csv&quot;,
  &quot;CX_0.5_3&quot;   = &quot;data/BP/All_Terms/GO_BP_CX_0.5_3.csv&quot;,
  &quot;CX_0.5_24&quot;  = &quot;data/BP/All_Terms/GO_BP_CX_0.5_24.csv&quot;,
  &quot;CX_0.5_48&quot;  = &quot;data/BP/All_Terms/GO_BP_CX_0.5_48.csv&quot;,
  &quot;DOX_0.1_3&quot;  = &quot;data/BP/All_Terms/GO_BP_DOX_0.1_3.csv&quot;,
  &quot;DOX_0.1_24&quot; = &quot;data/BP/All_Terms/GO_BP_DOX_0.1_24.csv&quot;,
  &quot;DOX_0.1_48&quot; = &quot;data/BP/All_Terms/GO_BP_DOX_0.1_48.csv&quot;,
  &quot;DOX_0.5_3&quot;  = &quot;data/BP/All_Terms/GO_BP_DOX_0.5_3.csv&quot;
)

# ✅ Step 2: Read and Filter Each File (p.adjust &lt; 0.05)
go_data &lt;- lapply(go_files, function(file) {
  df &lt;- read.csv(file)
  df %&gt;% filter(p.adjust &lt; 0.05)
})

# ✅ Step 3: Identify Unique GO Terms in Each Condition
unique_go_terms &lt;- list()

for (set_name in names(go_data)) {
  current_ids &lt;- go_data[[set_name]]$ID
  other_ids &lt;- unlist(lapply(go_data[names(go_data) != set_name], `[[`, &quot;ID&quot;), use.names = FALSE)
  unique_ids &lt;- setdiff(current_ids, other_ids)
  
  if (length(unique_ids) &gt; 0) {
    unique_go_terms[[set_name]] &lt;- unique_ids
  }
}

# ✅ Step 4: Map Unique IDs to Description and p.adjust
mapped_unique_go_terms &lt;- list()

for (set_name in names(unique_go_terms)) {
  df &lt;- go_data[[set_name]]
  unique_ids &lt;- unique_go_terms[[set_name]]
  
  mapped_df &lt;- df %&gt;%
    filter(ID %in% unique_ids) %&gt;%
    dplyr::select(GO_ID = ID, Function = Description, p.adjust)
  
  mapped_unique_go_terms[[set_name]] &lt;- mapped_df
}

# 🎯 Final Output: List of data.frames with GO_ID, Function, and p.adjust
mapped_unique_go_terms</code></pre>
<pre><code>$CX_0.1_24
       GO_ID                              Function   p.adjust
1 GO:0032147 activation of protein kinase activity 0.01634492
2 GO:0061351   neural precursor cell proliferation 0.01728494
3 GO:0030865    cortical cytoskeleton organization 0.02949342
4 GO:0021987           cerebral cortex development 0.02957858
5 GO:0007283                       spermatogenesis 0.04378504

$CX_0.1_48
        GO_ID                                                  Function
1  GO:1902115                          regulation of organelle assembly
2  GO:0009994                                    oocyte differentiation
3  GO:0007281                                     germ cell development
4  GO:0048469                                           cell maturation
5  GO:0070979                         protein K11-linked ubiquitination
6  GO:0007141                                            male meiosis I
7  GO:0046653                        tetrahydrofolate metabolic process
8  GO:0042558           pteridine-containing compound metabolic process
9  GO:0048477                                                 oogenesis
10 GO:0006266                                              DNA ligation
11 GO:0009163                           nucleoside biosynthetic process
12 GO:0034404 nucleobase-containing small molecule biosynthetic process
     p.adjust
1  0.02367495
2  0.03009502
3  0.03449739
4  0.03686756
5  0.03897505
6  0.03928009
7  0.03928009
8  0.04342249
9  0.04369792
10 0.04621861
11 0.04621861
12 0.04621861

$CX_0.5_3
       GO_ID                                      Function   p.adjust
1 GO:0060218       hematopoietic stem cell differentiation 0.01231022
2 GO:0002244 hematopoietic progenitor cell differentiation 0.01449870
3 GO:0048863                     stem cell differentiation 0.02000410
4 GO:0007517                      muscle organ development 0.02281494

$CX_0.5_24
       GO_ID                                                        Function
1 GO:0008630 intrinsic apoptotic signaling pathway in response to DNA damage
2 GO:0071539                              protein localization to centrosome
3 GO:1905508           protein localization to microtubule organizing center
    p.adjust
1 0.01956480
2 0.03302088
3 0.03630493

$CX_0.5_48
        GO_ID
1  GO:0031507
2  GO:0070828
3  GO:0032200
4  GO:0006305
5  GO:0006306
6  GO:0009129
7  GO:0051446
8  GO:0031100
9  GO:0009219
10 GO:0006544
11 GO:0040029
12 GO:0042276
13 GO:0140719
14 GO:2001032
15 GO:0009116
16 GO:0002562
17 GO:0016444
18 GO:0045911
19 GO:0051444
20 GO:0090399
21 GO:0031571
22 GO:0044819
23 GO:0045739
                                                                                       Function
1                                                                     heterochromatin formation
2                                                                  heterochromatin organization
3                                                                         telomere organization
4                                                                                DNA alkylation
5                                                                               DNA methylation
6                                         pyrimidine nucleoside monophosphate metabolic process
7                                                     positive regulation of meiotic cell cycle
8                                                                     animal organ regeneration
9                                              pyrimidine deoxyribonucleotide metabolic process
10                                                                    glycine metabolic process
11                                                     epigenetic regulation of gene expression
12                                                            error-prone translesion synthesis
13                                                       constitutive heterochromatin formation
14                       regulation of double-strand break repair via nonhomologous end joining
15                                                                 nucleoside metabolic process
16 somatic diversification of immune receptors via germline recombination within a single locus
17                                                               somatic cell DNA recombination
18                                                     positive regulation of DNA recombination
19                                negative regulation of ubiquitin-protein transferase activity
20                                                                       replicative senescence
21                                                   mitotic G1 DNA damage checkpoint signaling
22                                                 mitotic G1/S transition checkpoint signaling
23                                                            positive regulation of DNA repair
       p.adjust
1  0.0006775733
2  0.0013367179
3  0.0050904693
4  0.0067535333
5  0.0067535333
6  0.0098485800
7  0.0098485800
8  0.0137145772
9  0.0148082118
10 0.0188576687
11 0.0247093018
12 0.0247093018
13 0.0315207935
14 0.0380501591
15 0.0392803558
16 0.0402832146
17 0.0402832146
18 0.0402832146
19 0.0467989120
20 0.0467989120
21 0.0477331220
22 0.0477331220
23 0.0489508320

$DOX_0.1_3
       GO_ID                                Function   p.adjust
1 GO:1900034 regulation of cellular response to heat 0.01049825
2 GO:0034605               cellular response to heat 0.01670754
3 GO:0009408                        response to heat 0.01723368
4 GO:0009266        response to temperature stimulus 0.02025016

$DOX_0.1_24
        GO_ID
1  GO:0070372
2  GO:0001759
3  GO:0070371
4  GO:0032609
5  GO:0032649
6  GO:0008585
7  GO:0061458
8  GO:0010712
9  GO:0048608
10 GO:0046545
11 GO:0032729
12 GO:0010818
13 GO:0070374
14 GO:0040008
15 GO:0014856
16 GO:0046660
17 GO:0033002
18 GO:0042327
19 GO:0010714
20 GO:0009123
21 GO:0032965
22 GO:0048771
23 GO:0031128
24 GO:0032733
25 GO:0009156
26 GO:0002250
27 GO:0007169
28 GO:0003156
29 GO:0032967
30 GO:0001934
31 GO:0043410
32 GO:0048247
33 GO:0009410
34 GO:0110110
35 GO:0031952
36 GO:0002460
37 GO:0009161
38 GO:0034103
39 GO:0050869
40 GO:0030330
41 GO:0060485
42 GO:0002706
43 GO:0003014
44 GO:1901652
45 GO:0030316
46 GO:0030856
47 GO:0008217
48 GO:2000242
                                                                                                                    Function
1                                                                                        regulation of ERK1 and ERK2 cascade
2                                                                                                            organ induction
3                                                                                                      ERK1 and ERK2 cascade
4                                                                                              type II interferon production
5                                                                                regulation of type II interferon production
6                                                                                                   female gonad development
7                                                                                            reproductive system development
8                                                                                   regulation of collagen metabolic process
9                                                                                         reproductive structure development
10                                                                      development of primary female sexual characteristics
11                                                                      positive regulation of type II interferon production
12                                                                                                         T cell chemotaxis
13                                                                              positive regulation of ERK1 and ERK2 cascade
14                                                                                                      regulation of growth
15                                                                                        skeletal muscle cell proliferation
16                                                                                                female sex differentiation
17                                                                                                 muscle cell proliferation
18                                                                                    positive regulation of phosphorylation
19                                                                         positive regulation of collagen metabolic process
20                                                                                nucleoside monophosphate metabolic process
21                                                                               regulation of collagen biosynthetic process
22                                                                                                         tissue remodeling
23                                                                                                   developmental induction
24                                                                          positive regulation of interleukin-10 production
25                                                                         ribonucleoside monophosphate biosynthetic process
26                                                                                                  adaptive immune response
27                                                          transmembrane receptor protein tyrosine kinase signaling pathway
28                                                                                      regulation of animal organ formation
29                                                                      positive regulation of collagen biosynthetic process
30                                                                            positive regulation of protein phosphorylation
31                                                                                       positive regulation of MAPK cascade
32                                                                                                     lymphocyte chemotaxis
33                                                                                           response to xenobiotic stimulus
34                                                                         positive regulation of animal organ morphogenesis
35                                                                                 regulation of protein autophosphorylation
36 adaptive immune response based on somatic recombination of immune receptors built from immunoglobulin superfamily domains
37                                                                            ribonucleoside monophosphate metabolic process
38                                                                                           regulation of tissue remodeling
39                                                                                  negative regulation of B cell activation
40                                                            DNA damage response, signal transduction by p53 class mediator
41                                                                                                    mesenchyme development
42                                                                                regulation of lymphocyte mediated immunity
43                                                                                                      renal system process
44                                                                                                       response to peptide
45                                                                                                osteoclast differentiation
46                                                                             regulation of epithelial cell differentiation
47                                                                                              regulation of blood pressure
48                                                                               negative regulation of reproductive process
       p.adjust
1  0.0007516882
2  0.0014565624
3  0.0018451390
4  0.0024696279
5  0.0024696279
6  0.0028351080
7  0.0033494853
8  0.0042734147
9  0.0043969608
10 0.0053562247
11 0.0056220914
12 0.0060875264
13 0.0064855995
14 0.0064855995
15 0.0066569208
16 0.0073987218
17 0.0086612503
18 0.0089181320
19 0.0100403845
20 0.0105322006
21 0.0114881038
22 0.0117167502
23 0.0122542224
24 0.0122965743
25 0.0139144050
26 0.0149844616
27 0.0150191959
28 0.0173895646
29 0.0173895646
30 0.0173895646
31 0.0178954330
32 0.0207281068
33 0.0207281068
34 0.0216769106
35 0.0222648297
36 0.0241443316
37 0.0275870140
38 0.0306074768
39 0.0309911358
40 0.0349368983
41 0.0388673988
42 0.0397535216
43 0.0440793528
44 0.0442630140
45 0.0472091401
46 0.0489826882
47 0.0492409369
48 0.0494289130

$DOX_0.1_48
        GO_ID
1  GO:1901136
2  GO:0006516
3  GO:0008045
4  GO:0030903
5  GO:0086010
6  GO:0034644
7  GO:0001508
8  GO:0043062
9  GO:0097205
10 GO:0051899
11 GO:0002377
12 GO:0030198
13 GO:0008016
14 GO:0061337
15 GO:0045229
16 GO:0003094
17 GO:0006026
18 GO:0044409
19 GO:0046718
20 GO:0006027
21 GO:0071482
22 GO:0099638
23 GO:0086012
24 GO:0002381
25 GO:0001570
26 GO:0086001
27 GO:0086002
                                                                        Function
1                                      carbohydrate derivative catabolic process
2                                                 glycoprotein catabolic process
3                                                     motor neuron axon guidance
4                                                          notochord development
5                                membrane depolarization during action potential
6                                                        cellular response to UV
7                                                               action potential
8                                           extracellular structure organization
9                                                               renal filtration
10                                                       membrane depolarization
11                                                     immunoglobulin production
12                                             extracellular matrix organization
13                                               regulation of heart contraction
14                                                            cardiac conduction
15                                 external encapsulating structure organization
16                                                         glomerular filtration
17                                                 aminoglycan catabolic process
18                                                               entry into host
19                                                    viral entry into host cell
20                                           glycosaminoglycan catabolic process
21                                           cellular response to light stimulus
22                                 endosome to plasma membrane protein transport
23           membrane depolarization during cardiac muscle cell action potential
24 immunoglobulin production involved in immunoglobulin-mediated immune response
25                                                                vasculogenesis
26                                          cardiac muscle cell action potential
27                  cardiac muscle cell action potential involved in contraction
       p.adjust
1  0.0001051443
2  0.0004795775
3  0.0066131033
4  0.0099227357
5  0.0161688486
6  0.0161688486
7  0.0164282137
8  0.0167556978
9  0.0211116001
10 0.0220454960
11 0.0222151167
12 0.0228958876
13 0.0252354205
14 0.0254108130
15 0.0255011743
16 0.0262879898
17 0.0262879898
18 0.0272494111
19 0.0290812554
20 0.0329595937
21 0.0405838543
22 0.0423210689
23 0.0426459913
24 0.0428790400
25 0.0430052946
26 0.0430052946
27 0.0430680536

$DOX_0.5_3
        GO_ID
1  GO:0060411
2  GO:0003007
3  GO:0060840
4  GO:0035270
5  GO:0030098
6  GO:0003279
7  GO:0045165
8  GO:0090092
9  GO:0003281
10 GO:0048844
11 GO:0003151
12 GO:1903131
13 GO:0071559
14 GO:0060419
15 GO:0017015
16 GO:1903844
17 GO:0035019
18 GO:0035904
19 GO:0014706
20 GO:0030217
21 GO:0071560
22 GO:0060412
23 GO:0048738
24 GO:0003170
25 GO:0055017
26 GO:0035265
27 GO:0060976
28 GO:0046649
29 GO:0002521
30 GO:1902105
31 GO:0030278
32 GO:1903706
33 GO:0035051
34 GO:0042110
35 GO:0002065
36 GO:0033689
37 GO:0003181
38 GO:0009896
39 GO:0035909
40 GO:0007179
41 GO:0017148
42 GO:0048568
43 GO:0002067
44 GO:0060420
45 GO:0060537
46 GO:0005978
47 GO:0009250
48 GO:0045785
49 GO:0003171
                                                                                 Function
1                                                            cardiac septum morphogenesis
2                                                                     heart morphogenesis
3                                                                      artery development
4                                                            endocrine system development
5                                                              lymphocyte differentiation
6                                                              cardiac septum development
7                                                                    cell fate commitment
8  regulation of transmembrane receptor protein serine/threonine kinase signaling pathway
9                                                          ventricular septum development
10                                                                   artery morphogenesis
11                                                            outflow tract morphogenesis
12                                                       mononuclear cell differentiation
13                                            response to transforming growth factor beta
14                                                                           heart growth
15               regulation of transforming growth factor beta receptor signaling pathway
16            regulation of cellular response to transforming growth factor beta stimulus
17                                               somatic stem cell population maintenance
18                                                                      aorta development
19                                                     striated muscle tissue development
20                                                                 T cell differentiation
21                          cellular response to transforming growth factor beta stimulus
22                                                       ventricular septum morphogenesis
23                                                      cardiac muscle tissue development
24                                                                heart valve development
25                                                           cardiac muscle tissue growth
26                                                                           organ growth
27                                                       coronary vasculature development
28                                                                  lymphocyte activation
29                                                              leukocyte differentiation
30                                                regulation of leukocyte differentiation
31                                                             regulation of ossification
32                                                              regulation of hemopoiesis
33                                                             cardiocyte differentiation
34                                                                      T cell activation
35                                      columnar/cuboidal epithelial cell differentiation
36                                        negative regulation of osteoblast proliferation
37                                                   atrioventricular valve morphogenesis
38                                               positive regulation of catabolic process
39                                                                    aorta morphogenesis
40                             transforming growth factor beta receptor signaling pathway
41                                                     negative regulation of translation
42                                                            embryonic organ development
43                                              glandular epithelial cell differentiation
44                                                             regulation of heart growth
45                                                              muscle tissue development
46                                                          glycogen biosynthetic process
47                                                            glucan biosynthetic process
48                                                   positive regulation of cell adhesion
49                                                     atrioventricular valve development
     p.adjust
1  0.01860398
2  0.01860398
3  0.01860398
4  0.01860398
5  0.01860398
6  0.01860398
7  0.02065923
8  0.02065923
9  0.02065923
10 0.02065923
11 0.02103458
12 0.02103458
13 0.02103458
14 0.02144515
15 0.02542938
16 0.02542938
17 0.02542938
18 0.02542938
19 0.02587649
20 0.02658237
21 0.02658237
22 0.02658237
23 0.02658237
24 0.02658237
25 0.02658237
26 0.02672722
27 0.02687238
28 0.02725889
29 0.02804421
30 0.02935587
31 0.03300580
32 0.03337939
33 0.03456539
34 0.03688428
35 0.04054107
36 0.04216232
37 0.04216232
38 0.04216232
39 0.04830795
40 0.04830795
41 0.04830795
42 0.04830795
43 0.04830795
44 0.04830795
45 0.04830795
46 0.04830795
47 0.04830795
48 0.04933983
49 0.04933983</code></pre>
</div>
<div id="top-unique-go-terms" class="section level2">
<h2><strong>📌TOP Unique GO terms</strong></h2>
<pre class="r"><code># 📦 Load Required Libraries
library(dplyr)
library(readr)

# ✅ Step 1: Load and Filter GO Term Data (p.adjust &lt; 0.05) from All_Terms
CX_0.1_24_GO  &lt;- read_csv(&quot;data/BP/All_Terms/GO_BP_CX_0.1_24.csv&quot;)  %&gt;% filter(p.adjust &lt; 0.05)
CX_0.1_48_GO  &lt;- read_csv(&quot;data/BP/All_Terms/GO_BP_CX_0.1_48.csv&quot;)  %&gt;% filter(p.adjust &lt; 0.05)
CX_0.5_3_GO   &lt;- read_csv(&quot;data/BP/All_Terms/GO_BP_CX_0.5_3.csv&quot;)   %&gt;% filter(p.adjust &lt; 0.05)
CX_0.5_24_GO  &lt;- read_csv(&quot;data/BP/All_Terms/GO_BP_CX_0.5_24.csv&quot;)  %&gt;% filter(p.adjust &lt; 0.05)
CX_0.5_48_GO  &lt;- read_csv(&quot;data/BP/All_Terms/GO_BP_CX_0.5_48.csv&quot;)  %&gt;% filter(p.adjust &lt; 0.05)
DOX_0.1_3_GO  &lt;- read_csv(&quot;data/BP/All_Terms/GO_BP_DOX_0.1_3.csv&quot;)  %&gt;% filter(p.adjust &lt; 0.05)
DOX_0.1_24_GO &lt;- read_csv(&quot;data/BP/All_Terms/GO_BP_DOX_0.1_24.csv&quot;) %&gt;% filter(p.adjust &lt; 0.05)
DOX_0.1_48_GO &lt;- read_csv(&quot;data/BP/All_Terms/GO_BP_DOX_0.1_48.csv&quot;) %&gt;% filter(p.adjust &lt; 0.05)
DOX_0.5_3_GO  &lt;- read_csv(&quot;data/BP/All_Terms/GO_BP_DOX_0.5_3.csv&quot;)  %&gt;% filter(p.adjust &lt; 0.05)

# ✅ Step 2: Create Named List of Filtered Data Frames
go_files &lt;- list(
  `CX_0.1_24`  = CX_0.1_24_GO,
  `CX_0.1_48`  = CX_0.1_48_GO,
  `CX_0.5_3`   = CX_0.5_3_GO,
  `CX_0.5_24`  = CX_0.5_24_GO,
  `CX_0.5_48`  = CX_0.5_48_GO,
  `DOX_0.1_3`  = DOX_0.1_3_GO,
  `DOX_0.1_24` = DOX_0.1_24_GO,
  `DOX_0.1_48` = DOX_0.1_48_GO,
  `DOX_0.5_3`  = DOX_0.5_3_GO
)

# ✅ Step 3: Identify Unique GO Term IDs per Condition
unique_go_terms &lt;- list()

for (set_name in names(go_files)) {
  current_ids &lt;- go_files[[set_name]]$ID
  other_ids &lt;- unlist(lapply(go_files[names(go_files) != set_name], `[[`, &quot;ID&quot;), use.names = FALSE)
  unique_ids &lt;- setdiff(current_ids, other_ids)
  
  if (length(unique_ids) &gt; 0) {
    unique_go_terms[[set_name]] &lt;- unique_ids
  }
}

# ✅ Step 4: Map Unique GO IDs to Description and p.adjust
mapped_unique_go_terms &lt;- list()

for (set_name in names(unique_go_terms)) {
  source_df &lt;- go_files[[set_name]]
  unique_ids &lt;- unique_go_terms[[set_name]]
  
  mapped_df &lt;- source_df %&gt;%
    filter(ID %in% unique_ids) %&gt;%
    dplyr::select(GO_ID = ID, Function = Description, p.adjust)
  
  if (nrow(mapped_df) &gt; 0) {
    mapped_unique_go_terms[[set_name]] &lt;- mapped_df
  }
}

# 🎯 Final Output: Named list of data.frames with unique + significant GO terms per condition
mapped_unique_go_terms</code></pre>
<pre><code>$CX_0.1_24
# A tibble: 5 × 3
  GO_ID      Function                              p.adjust
  &lt;chr&gt;      &lt;chr&gt;                                    &lt;dbl&gt;
1 GO:0032147 activation of protein kinase activity   0.0163
2 GO:0061351 neural precursor cell proliferation     0.0173
3 GO:0030865 cortical cytoskeleton organization      0.0295
4 GO:0021987 cerebral cortex development             0.0296
5 GO:0007283 spermatogenesis                         0.0438

$CX_0.1_48
# A tibble: 12 × 3
   GO_ID      Function                                                  p.adjust
   &lt;chr&gt;      &lt;chr&gt;                                                        &lt;dbl&gt;
 1 GO:1902115 regulation of organelle assembly                            0.0237
 2 GO:0009994 oocyte differentiation                                      0.0301
 3 GO:0007281 germ cell development                                       0.0345
 4 GO:0048469 cell maturation                                             0.0369
 5 GO:0070979 protein K11-linked ubiquitination                           0.0390
 6 GO:0007141 male meiosis I                                              0.0393
 7 GO:0046653 tetrahydrofolate metabolic process                          0.0393
 8 GO:0042558 pteridine-containing compound metabolic process             0.0434
 9 GO:0048477 oogenesis                                                   0.0437
10 GO:0006266 DNA ligation                                                0.0462
11 GO:0009163 nucleoside biosynthetic process                             0.0462
12 GO:0034404 nucleobase-containing small molecule biosynthetic process   0.0462

$CX_0.5_3
# A tibble: 4 × 3
  GO_ID      Function                                      p.adjust
  &lt;chr&gt;      &lt;chr&gt;                                            &lt;dbl&gt;
1 GO:0060218 hematopoietic stem cell differentiation         0.0123
2 GO:0002244 hematopoietic progenitor cell differentiation   0.0145
3 GO:0048863 stem cell differentiation                       0.0200
4 GO:0007517 muscle organ development                        0.0228

$CX_0.5_24
# A tibble: 3 × 3
  GO_ID      Function                                                   p.adjust
  &lt;chr&gt;      &lt;chr&gt;                                                         &lt;dbl&gt;
1 GO:0008630 intrinsic apoptotic signaling pathway in response to DNA …   0.0196
2 GO:0071539 protein localization to centrosome                           0.0330
3 GO:1905508 protein localization to microtubule organizing center        0.0363

$CX_0.5_48
# A tibble: 23 × 3
   GO_ID      Function                                              p.adjust
   &lt;chr&gt;      &lt;chr&gt;                                                    &lt;dbl&gt;
 1 GO:0031507 heterochromatin formation                             0.000678
 2 GO:0070828 heterochromatin organization                          0.00134 
 3 GO:0032200 telomere organization                                 0.00509 
 4 GO:0006305 DNA alkylation                                        0.00675 
 5 GO:0006306 DNA methylation                                       0.00675 
 6 GO:0009129 pyrimidine nucleoside monophosphate metabolic process 0.00985 
 7 GO:0051446 positive regulation of meiotic cell cycle             0.00985 
 8 GO:0031100 animal organ regeneration                             0.0137  
 9 GO:0009219 pyrimidine deoxyribonucleotide metabolic process      0.0148  
10 GO:0006544 glycine metabolic process                             0.0189  
# ℹ 13 more rows

$DOX_0.1_3
# A tibble: 4 × 3
  GO_ID      Function                                p.adjust
  &lt;chr&gt;      &lt;chr&gt;                                      &lt;dbl&gt;
1 GO:1900034 regulation of cellular response to heat   0.0105
2 GO:0034605 cellular response to heat                 0.0167
3 GO:0009408 response to heat                          0.0172
4 GO:0009266 response to temperature stimulus          0.0203

$DOX_0.1_24
# A tibble: 48 × 3
   GO_ID      Function                                             p.adjust
   &lt;chr&gt;      &lt;chr&gt;                                                   &lt;dbl&gt;
 1 GO:0070372 regulation of ERK1 and ERK2 cascade                  0.000752
 2 GO:0001759 organ induction                                      0.00146 
 3 GO:0070371 ERK1 and ERK2 cascade                                0.00185 
 4 GO:0032609 type II interferon production                        0.00247 
 5 GO:0032649 regulation of type II interferon production          0.00247 
 6 GO:0008585 female gonad development                             0.00284 
 7 GO:0061458 reproductive system development                      0.00335 
 8 GO:0010712 regulation of collagen metabolic process             0.00427 
 9 GO:0048608 reproductive structure development                   0.00440 
10 GO:0046545 development of primary female sexual characteristics 0.00536 
# ℹ 38 more rows

$DOX_0.1_48
# A tibble: 27 × 3
   GO_ID      Function                                        p.adjust
   &lt;chr&gt;      &lt;chr&gt;                                              &lt;dbl&gt;
 1 GO:1901136 carbohydrate derivative catabolic process       0.000105
 2 GO:0006516 glycoprotein catabolic process                  0.000480
 3 GO:0008045 motor neuron axon guidance                      0.00661 
 4 GO:0030903 notochord development                           0.00992 
 5 GO:0086010 membrane depolarization during action potential 0.0162  
 6 GO:0034644 cellular response to UV                         0.0162  
 7 GO:0001508 action potential                                0.0164  
 8 GO:0043062 extracellular structure organization            0.0168  
 9 GO:0097205 renal filtration                                0.0211  
10 GO:0051899 membrane depolarization                         0.0220  
# ℹ 17 more rows

$DOX_0.5_3
# A tibble: 49 × 3
   GO_ID      Function                                                  p.adjust
   &lt;chr&gt;      &lt;chr&gt;                                                        &lt;dbl&gt;
 1 GO:0060411 cardiac septum morphogenesis                                0.0186
 2 GO:0003007 heart morphogenesis                                         0.0186
 3 GO:0060840 artery development                                          0.0186
 4 GO:0035270 endocrine system development                                0.0186
 5 GO:0030098 lymphocyte differentiation                                  0.0186
 6 GO:0003279 cardiac septum development                                  0.0186
 7 GO:0045165 cell fate commitment                                        0.0207
 8 GO:0090092 regulation of transmembrane receptor protein serine/thre…   0.0207
 9 GO:0003281 ventricular septum development                              0.0207
10 GO:0048844 artery morphogenesis                                        0.0207
# ℹ 39 more rows</code></pre>
</div>
<div id="plot-unique-go-terms" class="section level2">
<h2><strong>📌Plot Unique GO terms</strong></h2>
<pre class="r"><code># 📦 Load Required Libraries
library(dplyr)
library(ggplot2)
library(ggpubr)
library(stringr)

# 🧾 Assumes: mapped_unique_go_terms is a list of data.frames with GO_ID, Function, p.adjust

# 🔁 Step 1: Extract Top 10 Terms by p.adjust per Condition
plot_data &lt;- list()

for (set_name in names(mapped_unique_go_terms)) {
  unique_df &lt;- mapped_unique_go_terms[[set_name]]
  
  if (nrow(unique_df) &gt; 0) {
    unique_df &lt;- unique_df %&gt;%
      mutate(Sample = set_name,
             NegLog10Padj = -log10(p.adjust))
    
    top10 &lt;- unique_df %&gt;%
      slice_min(order_by = p.adjust, n = 10, with_ties = FALSE) %&gt;%
      dplyr::select(Sample, Function, NegLog10Padj)  #  fully qualified here
    
    plot_data[[set_name]] &lt;- top10
  }
}

# 🔗 Step 2: Combine and Format Labels
plot_df &lt;- bind_rows(plot_data) %&gt;%
  mutate(Function = str_trunc(Function, width = 60))

# 🎨 Step 3: Faceted Bar Plot
ggplot(plot_df, aes(x = NegLog10Padj, y = reorder(Function, NegLog10Padj))) +
  geom_bar(stat = &quot;identity&quot;, fill = &quot;#3182bd&quot;) +
  facet_wrap(~ Sample, scales = &quot;free_y&quot;, ncol = 2, strip.position = &quot;top&quot;) +
  labs(
    x = &quot;-log10(p.adjust)&quot;,
    y = &quot;GO Term&quot;,
    title = &quot;Top Unique GO Biological Process Terms per Condition&quot;
  ) +
  theme_pubr(base_size = 14) +
  theme(
    strip.background = element_rect(colour = &quot;black&quot;, fill = &quot;grey90&quot;, size = 1),
    strip.text = element_text(face = &quot;bold&quot;),
    axis.text.y = element_text(size = 10),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, size = 1)
  )</code></pre>
<p><img src="figure/Overlap_GO_DEG.Rmd/Unique2-1.png" width="1152" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 26100)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] stringr_1.5.1       ggpubr_0.6.0        readr_2.1.5        
[4] tidyr_1.3.1         UpSetR_1.4.0        dplyr_1.1.4        
[7] ggplot2_3.5.2       ggVennDiagram_1.5.2

loaded via a namespace (and not attached):
 [1] utf8_1.2.4        sass_0.4.10       generics_0.1.3    rstatix_0.7.2    
 [5] stringi_1.8.3     hms_1.1.3         digest_0.6.34     magrittr_2.0.3   
 [9] evaluate_1.0.3    grid_4.3.0        fastmap_1.2.0     rprojroot_2.0.4  
[13] workflowr_1.7.1   plyr_1.8.9        jsonlite_2.0.0    whisker_0.4.1    
[17] backports_1.5.0   Formula_1.2-5     gridExtra_2.3     promises_1.3.2   
[21] purrr_1.0.4       scales_1.3.0      jquerylib_0.1.4   abind_1.4-8      
[25] cli_3.6.1         crayon_1.5.3      rlang_1.1.3       bit64_4.6.0-1    
[29] munsell_0.5.1     withr_3.0.2       cachem_1.1.0      yaml_2.3.10      
[33] parallel_4.3.0    tools_4.3.0       tzdb_0.5.0        ggsignif_0.6.4   
[37] colorspace_2.1-0  httpuv_1.6.15     broom_1.0.8       vctrs_0.6.5      
[41] R6_2.6.1          lifecycle_1.0.4   git2r_0.36.2      car_3.1-3        
[45] bit_4.6.0         fs_1.6.3          vroom_1.6.5       pkgconfig_2.0.3  
[49] pillar_1.10.2     bslib_0.9.0       later_1.3.2       gtable_0.3.6     
[53] glue_1.7.0        Rcpp_1.0.12       xfun_0.52         tibble_3.2.1     
[57] tidyselect_1.2.1  rstudioapi_0.17.1 knitr_1.50        farver_2.1.2     
[61] htmltools_0.5.8.1 carData_3.0-5     rmarkdown_2.29    labeling_0.4.3   
[65] compiler_4.3.0   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
