<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Venn Diagram Overlap of DE Genes</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Paul_CX_2025</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Venn Diagram Overlap of DE Genes</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-07-03
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Paul_CX_2025/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree98c546964aac1b29220c891e2a4ad037b8b48ecctargetblank98c5469a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/tree/98c546964aac1b29220c891e2a4ad037b8b48ecc" target="_blank">98c5469</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree98c546964aac1b29220c891e2a4ad037b8b48ecctargetblank98c5469a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sayanpaul01/Paul_CX_2025/tree/98c546964aac1b29220c891e2a4ad037b8b48ecc" target="_blank">98c5469</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    0.1 box.svg
    Ignored:    Rplot04.svg

Untracked files:
    Untracked:  0.1 density.svg
    Untracked:  0.1.emf
    Untracked:  0.1.svg
    Untracked:  0.5 box.svg
    Untracked:  0.5 density.svg
    Untracked:  0.5.svg
    Untracked:  Additional/
    Untracked:  Autosome factors.svg
    Untracked:  CX_5461_Pattern_Genes_24hr.csv
    Untracked:  CX_5461_Pattern_Genes_3hr.csv
    Untracked:  Cell viability box plot.svg
    Untracked:  DEG GO terms.svg
    Untracked:  DNA damage associated GO terms.svg
    Untracked:  DRC1.svg
    Untracked:  Figure 1.jpeg
    Untracked:  Figure 1.pdf
    Untracked:  Figure_CM_Purity.pdf
    Untracked:  G Quadruplex DEGs.svg
    Untracked:  PC2 Vs PC3 Autosome.svg
    Untracked:  PCA autosome.svg
    Untracked:  Rplot 18.svg
    Untracked:  Rplot.svg
    Untracked:  Rplot01.svg
    Untracked:  Rplot02.svg
    Untracked:  Rplot03.svg
    Untracked:  Rplot05.svg
    Untracked:  Rplot06.svg
    Untracked:  Rplot07.svg
    Untracked:  Rplot08.jpeg
    Untracked:  Rplot08.svg
    Untracked:  Rplot09.svg
    Untracked:  Rplot10.svg
    Untracked:  Rplot11.svg
    Untracked:  Rplot12.svg
    Untracked:  Rplot13.svg
    Untracked:  Rplot14.svg
    Untracked:  Rplot15.svg
    Untracked:  Rplot16.svg
    Untracked:  Rplot17.svg
    Untracked:  Rplot18.svg
    Untracked:  Rplot19.svg
    Untracked:  Rplot20.svg
    Untracked:  Rplot21.svg
    Untracked:  Rplot22.svg
    Untracked:  Rplot23.svg
    Untracked:  Rplot24.svg
    Untracked:  S15.pdf
    Untracked:  TOP2B.bed
    Untracked:  TS HPA (Violin).svg
    Untracked:  TS HPA.svg
    Untracked:  TS_HA.svg
    Untracked:  TS_HV.svg
    Untracked:  Violin HA.svg
    Untracked:  Violin HV (CX vs DOX).svg
    Untracked:  Violin HV.svg
    Untracked:  data/AF.csv
    Untracked:  data/AF_Mapped.csv
    Untracked:  data/AF_genes.csv
    Untracked:  data/Annotated_DOX_Gene_Table.csv
    Untracked:  data/BP/
    Untracked:  data/CAD_genes.csv
    Untracked:  data/Cardiotox.csv
    Untracked:  data/Cardiotox_mapped.csv
    Untracked:  data/Col_DEG_proportion_data.csv
    Untracked:  data/Col_DEGs.csv
    Untracked:  data/Corrmotif_GO/
    Untracked:  data/DOX_Vald.csv
    Untracked:  data/DOX_Vald_Mapped.csv
    Untracked:  data/DOX_alt.csv
    Untracked:  data/Entrez_Cardiotox.csv
    Untracked:  data/Entrez_Cardiotox_Mapped.csv
    Untracked:  data/GWAS.xlsx
    Untracked:  data/GWAS_SNPs.bed
    Untracked:  data/HF.csv
    Untracked:  data/HF_Mapped.csv
    Untracked:  data/HF_genes.csv
    Untracked:  data/Hypertension_genes.csv
    Untracked:  data/MI_genes.csv
    Untracked:  data/P53_Target_mapped.csv
    Untracked:  data/Sample_annotated.csv
    Untracked:  data/Samples.csv
    Untracked:  data/Samples.xlsx
    Untracked:  data/TOP2A.bed
    Untracked:  data/TOP2A_target.csv
    Untracked:  data/TOP2A_target_lit.csv
    Untracked:  data/TOP2A_target_lit_mapped.csv
    Untracked:  data/TOP2A_target_mapped.csv
    Untracked:  data/TOP2B.bed
    Untracked:  data/TOP2B_target.csv
    Untracked:  data/TOP2B_target_heatmap.csv
    Untracked:  data/TOP2B_target_heatmap_mapped.csv
    Untracked:  data/TOP2B_target_mapped.csv
    Untracked:  data/TS.csv
    Untracked:  data/TS_HPA.csv
    Untracked:  data/TS_HPA_mapped.csv
    Untracked:  data/Toptable_CX_0.1_24.csv
    Untracked:  data/Toptable_CX_0.1_3.csv
    Untracked:  data/Toptable_CX_0.1_48.csv
    Untracked:  data/Toptable_CX_0.5_24.csv
    Untracked:  data/Toptable_CX_0.5_3.csv
    Untracked:  data/Toptable_CX_0.5_48.csv
    Untracked:  data/Toptable_DOX_0.1_24.csv
    Untracked:  data/Toptable_DOX_0.1_3.csv
    Untracked:  data/Toptable_DOX_0.1_48.csv
    Untracked:  data/Toptable_DOX_0.5_24.csv
    Untracked:  data/Toptable_DOX_0.5_3.csv
    Untracked:  data/Toptable_DOX_0.5_48.csv
    Untracked:  data/count.tsv
    Untracked:  data/heatmap.csv
    Untracked:  data/ts_data_mapped
    Untracked:  results/
    Untracked:  run_bedtools.bat

Unstaged changes:
    Deleted:    analysis/Actox.Rmd
    Modified:   analysis/_site.yml
    Modified:   data/DEGs/Toptable_DOX_0.5_3.csv
    Modified:   data/DOX_0.5_48 (Combined).csv
    Modified:   data/Total_number_of_Mapped_reads_by_Individuals.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Overlap_DEGs.Rmd</code>) and HTML
(<code>docs/Overlap_DEGs.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/52c525bcd24cafbe37054b67c1523f1c8b72b7e6/analysis/Overlap_DEGs.Rmd" target="_blank">52c525b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-20
</td>
<td>
Added Overlap of GO Terms across drugs, concentrations, and timepoints
and updated index
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/6a1580c39867acf0c70c50f2e49e7680a80dd15d/docs/Overlap_DEGs.html" target="_blank">6a1580c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/0c8c19f81315e5ad34517a280f2f87ed65851be8/analysis/Overlap_DEGs.Rmd" target="_blank">0c8c19f</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-19
</td>
<td>
Added UpSet plots for DEGs across samples, drugs, concentrations, and
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/0ea6c0c0ea9a55a087085290ad025ad37795834f/docs/Overlap_DEGs.html" target="_blank">0ea6c0c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/bef2c813b7e8db5ef938d8ac5011e79827118a95/analysis/Overlap_DEGs.Rmd" target="_blank">bef2c81</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-19
</td>
<td>
Added UpSet plots for DEGs across samples, drugs, concentrations, and
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/1dcb0c817da88bf51b0956d2e16bd053f3b40b56/docs/Overlap_DEGs.html" target="_blank">1dcb0c8</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/af515756fe0be3881d4cc85ef259954fe654a547/analysis/Overlap_DEGs.Rmd" target="_blank">af51575</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-19
</td>
<td>
Added Overlap of DEGs analysis and updated index.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="overlap-of-de-genes" class="section level2">
<h2><strong>📌 overlap of DE genes</strong></h2>
</div>
<div id="read-and-process-deg-data" class="section level2">
<h2><strong>📌 Read and Process DEG Data</strong></h2>
<pre class="r"><code># Load DEGs Data
CX_0.1_3 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;)
CX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;)
CX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;)
CX_0.5_3 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)

DOX_0.1_3 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;)
DOX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;)
DOX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;)
DOX_0.5_3 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

# Extract Significant DEGs
DEG1 &lt;- CX_0.1_3$Entrez_ID[CX_0.1_3$adj.P.Val &lt; 0.05]
DEG2 &lt;- CX_0.1_24$Entrez_ID[CX_0.1_24$adj.P.Val &lt; 0.05]
DEG3 &lt;- CX_0.1_48$Entrez_ID[CX_0.1_48$adj.P.Val &lt; 0.05]
DEG4 &lt;- CX_0.5_3$Entrez_ID[CX_0.5_3$adj.P.Val &lt; 0.05]
DEG5 &lt;- CX_0.5_24$Entrez_ID[CX_0.5_24$adj.P.Val &lt; 0.05]
DEG6 &lt;- CX_0.5_48$Entrez_ID[CX_0.5_48$adj.P.Val &lt; 0.05]

DEG7 &lt;- DOX_0.1_3$Entrez_ID[DOX_0.1_3$adj.P.Val &lt; 0.05]
DEG8 &lt;- DOX_0.1_24$Entrez_ID[DOX_0.1_24$adj.P.Val &lt; 0.05]
DEG9 &lt;- DOX_0.1_48$Entrez_ID[DOX_0.1_48$adj.P.Val &lt; 0.05]
DEG10 &lt;- DOX_0.5_3$Entrez_ID[DOX_0.5_3$adj.P.Val &lt; 0.05]
DEG11 &lt;- DOX_0.5_24$Entrez_ID[DOX_0.5_24$adj.P.Val &lt; 0.05]
DEG12 &lt;- DOX_0.5_48$Entrez_ID[DOX_0.5_48$adj.P.Val &lt; 0.05]</code></pre>
</div>
<div id="load-libraries" class="section level2">
<h2><strong>📌 Load Libraries</strong></h2>
<pre class="r"><code>library(ggplot2)
library(ggVennDiagram)
library(UpSetR)</code></pre>
</div>
<div id="across-drugs" class="section level2">
<h2><strong>📌 Across Drugs</strong></h2>
</div>
<div id="venn-diagram-cx-5461-vs-veh-0.1-µm" class="section level2">
<h2><strong>📌Venn Diagram: CX-5461 vs VEH (0.1 µM)</strong></h2>
<pre class="r"><code>venntest &lt;- list(DEG1, DEG2, DEG3)
ggVennDiagram(
  venntest,
  category.names = c(&quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;),
  fill = c(&quot;red&quot;, &quot;blue&quot;, &quot;green&quot;)
) + ggtitle(&quot;CX-5461 Vs VEH 0.1 micromolar&quot;)+
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;),  # Increase title size
    text = element_text(size = 16)  # Increase text size globally
  )</code></pre>
<p><img src="figure/Overlap_DEGs.Rmd/venn1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-venn1-1">
Past versions of venn1-1.png
</button>
</p>
<div id="fig-venn1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/1dcb0c817da88bf51b0956d2e16bd053f3b40b56/docs/figure/Overlap_DEGs.Rmd/venn1-1.png" target="_blank">1dcb0c8</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="venn-diagram-cx-5461-vs-veh-0.5-µm" class="section level2">
<h2><strong>📌Venn Diagram: CX-5461 vs VEH (0.5 µM)</strong></h2>
<pre class="r"><code>venntest1 &lt;- list(DEG4, DEG5, DEG6)
ggVennDiagram(
  venntest1,
  category.names = c(&quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;),
  fill = c(&quot;red&quot;, &quot;blue&quot;, &quot;green&quot;)
) + ggtitle(&quot;CX-5461 Vs VEH 0.5 micromolar&quot;)+
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;),  # Increase title size
    text = element_text(size = 16)  # Increase text size globally
  )</code></pre>
<p><img src="figure/Overlap_DEGs.Rmd/venn2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-venn2-1">
Past versions of venn2-1.png
</button>
</p>
<div id="fig-venn2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/1dcb0c817da88bf51b0956d2e16bd053f3b40b56/docs/figure/Overlap_DEGs.Rmd/venn2-1.png" target="_blank">1dcb0c8</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="venn-diagram-dox-vs-veh-0.1-µm" class="section level2">
<h2><strong>📌Venn Diagram: DOX vs VEH (0.1 µM)</strong></h2>
<pre class="r"><code>venntest2 &lt;- list(DEG7, DEG8, DEG9)
ggVennDiagram(
  venntest2,
  category.names = c(&quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;),
  fill = c(&quot;red&quot;, &quot;blue&quot;, &quot;green&quot;)
) + ggtitle(&quot;DOX Vs VEH 0.1 micromolar&quot;)+
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;),  # Increase title size
    text = element_text(size = 16)  # Increase text size globally
  )</code></pre>
<p><img src="figure/Overlap_DEGs.Rmd/venn3-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-venn3-1">
Past versions of venn3-1.png
</button>
</p>
<div id="fig-venn3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/1dcb0c817da88bf51b0956d2e16bd053f3b40b56/docs/figure/Overlap_DEGs.Rmd/venn3-1.png" target="_blank">1dcb0c8</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="venn-diagram-dox-vs-veh-0.5-µm" class="section level2">
<h2><strong>📌Venn Diagram: DOX vs VEH (0.5 µM)</strong></h2>
<pre class="r"><code>venntest3 &lt;- list(DEG10, DEG11, DEG12)
ggVennDiagram(
  venntest3,
  category.names = c(&quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;),
  fill = c(&quot;red&quot;, &quot;blue&quot;, &quot;green&quot;)
) + ggtitle(&quot;DOX Vs VEH 0.5 micromolar&quot;)+
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;),  # Increase title size
    text = element_text(size = 16)  # Increase text size globally
  )</code></pre>
<p><img src="figure/Overlap_DEGs.Rmd/venn4-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-venn4-1">
Past versions of venn4-1.png
</button>
</p>
<div id="fig-venn4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/1dcb0c817da88bf51b0956d2e16bd053f3b40b56/docs/figure/Overlap_DEGs.Rmd/venn4-1.png" target="_blank">1dcb0c8</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="across-concentrations" class="section level2">
<h2><strong>📌 Across Concentrations</strong></h2>
</div>
<div id="venn-diagram-0.1-micromolar" class="section level2">
<h2><strong>📌Venn Diagram: 0.1 Micromolar</strong></h2>
<pre class="r"><code>venntest7 &lt;- list(DEG1, DEG2, DEG3, DEG7, DEG8, DEG9)
ggVennDiagram(
  venntest7, label = &quot;count&quot;,
  category.names = c(&quot;CX_0.1_3&quot;, &quot;CX_0.1_24&quot;, &quot;CX_0.1_48&quot;, &quot;DOX_0.1_3&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.1_48&quot;)
) + ggtitle(&quot;0.1 micromolar&quot;)+
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;),  # Increase title size
    text = element_text(size = 16)  # Increase text size globally
  )</code></pre>
<p><img src="figure/Overlap_DEGs.Rmd/venn5-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-venn5-1">
Past versions of venn5-1.png
</button>
</p>
<div id="fig-venn5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/1dcb0c817da88bf51b0956d2e16bd053f3b40b56/docs/figure/Overlap_DEGs.Rmd/venn5-1.png" target="_blank">1dcb0c8</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="venn-diagram-0.5-micromolar" class="section level2">
<h2><strong>📌Venn Diagram: 0.5 Micromolar</strong></h2>
<pre class="r"><code>venntest8 &lt;- list(DEG4, DEG5, DEG6, DEG10, DEG11, DEG12)
ggVennDiagram(
  venntest8, label = &quot;count&quot;,
  category.names = c(&quot;CX_0.5_3&quot;, &quot;CX_0.5_24&quot;, &quot;CX_0.5_48&quot;, &quot;DOX_0.5_3&quot;, &quot;DOX_0.5_24&quot;, &quot;DOX_0.5_48&quot;)
) + ggtitle(&quot;0.5 micromolar&quot;)+
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;),  # Increase title size
    text = element_text(size = 16)  # Increase text size globally
  )</code></pre>
<p><img src="figure/Overlap_DEGs.Rmd/venn6-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-venn6-1">
Past versions of venn6-1.png
</button>
</p>
<div id="fig-venn6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/1dcb0c817da88bf51b0956d2e16bd053f3b40b56/docs/figure/Overlap_DEGs.Rmd/venn6-1.png" target="_blank">1dcb0c8</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="across-timepoints" class="section level2">
<h2><strong>📌 Across Timepoints</strong></h2>
</div>
<div id="venn-diagram-3-hour-timepoint" class="section level2">
<h2><strong>📌Venn Diagram: 3-hour Timepoint</strong></h2>
<pre class="r"><code>venntest4 &lt;- list(DEG1, DEG4, DEG7, DEG10)
ggVennDiagram(
  venntest4, label_percent_digit = 2,
  category.names = c(&quot;CX_0.1_3&quot;, &quot;CX_0.5_3&quot;, &quot;DOX_0.1_3&quot;, &quot;DOX_0.5_3&quot;)
) + ggtitle(&quot;3hr&quot;)+
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;),  # Increase title size
    text = element_text(size = 16)  # Increase text size globally
  )</code></pre>
<p><img src="figure/Overlap_DEGs.Rmd/venn7-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-venn7-1">
Past versions of venn7-1.png
</button>
</p>
<div id="fig-venn7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/1dcb0c817da88bf51b0956d2e16bd053f3b40b56/docs/figure/Overlap_DEGs.Rmd/venn7-1.png" target="_blank">1dcb0c8</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="venn-diagram-24-hour-timepoint" class="section level2">
<h2><strong>📌Venn Diagram: 24-hour Timepoint</strong></h2>
<pre class="r"><code>venntest5 &lt;- list(DEG2, DEG5, DEG8, DEG11)
ggVennDiagram(
  venntest5, label_percent_digit = 2,
  category.names = c(&quot;CX_0.1_24&quot;, &quot;CX_0.5_24&quot;, &quot;DOX_0.1_24&quot;, &quot;DOX_0.5_24&quot;)
) + ggtitle(&quot;24hr&quot;)+
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;),  # Increase title size
    text = element_text(size = 16)  # Increase text size globally
  )</code></pre>
<p><img src="figure/Overlap_DEGs.Rmd/venn8-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-venn8-1">
Past versions of venn8-1.png
</button>
</p>
<div id="fig-venn8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/1dcb0c817da88bf51b0956d2e16bd053f3b40b56/docs/figure/Overlap_DEGs.Rmd/venn8-1.png" target="_blank">1dcb0c8</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="venn-diagram-48-hour-timepoint" class="section level2">
<h2><strong>📌Venn Diagram: 48-hour Timepoint</strong></h2>
<pre class="r"><code>venntest6 &lt;- list(DEG3, DEG6, DEG9, DEG12)
ggVennDiagram(
  venntest6, label_percent_digit = 2,
  category.names = c(&quot;CX_0.1_48&quot;, &quot;CX_0.5_48&quot;, &quot;DOX_0.1_48&quot;, &quot;DOX_0.5_48&quot;)
) + ggtitle(&quot;48hr&quot;)+
  theme(
    plot.title = element_text(size = 16, face = &quot;bold&quot;),  # Increase title size
    text = element_text(size = 16)  # Increase text size globally
  )</code></pre>
<p><img src="figure/Overlap_DEGs.Rmd/venn9-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-venn9-1">
Past versions of venn9-1.png
</button>
</p>
<div id="fig-venn9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/1dcb0c817da88bf51b0956d2e16bd053f3b40b56/docs/figure/Overlap_DEGs.Rmd/venn9-1.png" target="_blank">1dcb0c8</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="overlap-of-degs-upset-plot" class="section level2">
<h2><strong>📌 Overlap of DEGs (Upset Plot)</strong></h2>
</div>
<div id="loading-data" class="section level2">
<h2><strong>📌 Loading data</strong></h2>
<pre class="r"><code># Extract Significant DEGs
# Create a list of DEGs for each sample
DEG_list &lt;- list(
  CX_0.1_3 = CX_0.1_3$Entrez_ID[CX_0.1_3$adj.P.Val &lt; 0.05],
  CX_0.1_24 = CX_0.1_24$Entrez_ID[CX_0.1_24$adj.P.Val &lt; 0.05],
  CX_0.1_48 = CX_0.1_48$Entrez_ID[CX_0.1_48$adj.P.Val &lt; 0.05],
  CX_0.5_3 = CX_0.5_3$Entrez_ID[CX_0.5_3$adj.P.Val &lt; 0.05],
  CX_0.5_24 = CX_0.5_24$Entrez_ID[CX_0.5_24$adj.P.Val &lt; 0.05],
  CX_0.5_48 = CX_0.5_48$Entrez_ID[CX_0.5_48$adj.P.Val &lt; 0.05],
  DOX_0.1_3 = DOX_0.1_3$Entrez_ID[DOX_0.1_3$adj.P.Val &lt; 0.05],
  DOX_0.1_24 = DOX_0.1_24$Entrez_ID[DOX_0.1_24$adj.P.Val &lt; 0.05],
  DOX_0.1_48 = DOX_0.1_48$Entrez_ID[DOX_0.1_48$adj.P.Val &lt; 0.05],
  DOX_0.5_3 = DOX_0.5_3$Entrez_ID[DOX_0.5_3$adj.P.Val &lt; 0.05],
  DOX_0.5_24 = DOX_0.5_24$Entrez_ID[DOX_0.5_24$adj.P.Val &lt; 0.05],
  DOX_0.5_48 = DOX_0.5_48$Entrez_ID[DOX_0.5_48$adj.P.Val &lt; 0.05]
)

# Convert list to binary matrix
DEG_matrix &lt;- fromList(DEG_list)

# Define order of sets
set_order &lt;- names(DEG_list)</code></pre>
</div>
<div
id="upset-plot-of-degs-across-samplesshow-all-intersections-till-lowest-size-5"
class="section level2">
<h2><strong>📌 UpSet Plot of DEGs Across Samples(Show all intersections
till lowest size 5)</strong></h2>
<pre class="r"><code>upset(
  DEG_matrix,
  sets = set_order,  # Specify the exact order of sets
  order.by = &quot;freq&quot;,  # Order intersections by frequency
  main.bar.color = &quot;blue&quot;,  # Color for the intersection bars
  matrix.color = &quot;black&quot;,  # Color for matrix dots
  sets.bar.color = rainbow(length(DEG_list)),  # Assign different colors to set size bars
  keep.order = TRUE,  # Keep the specified order of sets
  number.angles = 0,  # Angle of numbers in intersection size bars
  point.size = 2.5,  # Size of points in the matrix
  text.scale = 1,  # Scale for text elements
  show.numbers = &quot;yes&quot;  # Show intersection size numbers directly
)</code></pre>
<p><img src="figure/Overlap_DEGs.Rmd/upset2-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-upset2-1">
Past versions of upset2-1.png
</button>
</p>
<div id="fig-upset2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/0ea6c0c0ea9a55a087085290ad025ad37795834f/docs/figure/Overlap_DEGs.Rmd/upset2-1.png" target="_blank">0ea6c0c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="upset-plot-of-degs-across-samplesshow-all-intersections-till-last-lowest-size"
class="section level2">
<h2><strong>📌 UpSet Plot of DEGs Across Samples(Show all intersections
till last lowest size)</strong></h2>
<pre class="r"><code>upset(
  DEG_matrix,
  sets = set_order,  # Specify the exact order of sets
  order.by = &quot;freq&quot;,  # Order intersections by frequency
  main.bar.color = &quot;blue&quot;,  # Color for the intersection bars
  matrix.color = &quot;black&quot;,  # Color for matrix dots
  sets.bar.color = rainbow(length(DEG_list)),  # Assign different colors to set size bars
  keep.order = TRUE,  # Keep the specified order of sets
  number.angles = 0,  # Angle of numbers in intersection size bars
  point.size = 2.5,  # Size of points in the matrix
  text.scale = 1,  # Scale for text elements
  show.numbers = &quot;yes&quot;,  # Show intersection size numbers directly
  nintersects = NA  # Show all intersections including those with lowest size
)</code></pre>
<p><img src="figure/Overlap_DEGs.Rmd/upset3-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-upset3-1">
Past versions of upset3-1.png
</button>
</p>
<div id="fig-upset3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/6a1580c39867acf0c70c50f2e49e7680a80dd15d/docs/figure/Overlap_DEGs.Rmd/upset3-1.png" target="_blank">6a1580c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/0ea6c0c0ea9a55a087085290ad025ad37795834f/docs/figure/Overlap_DEGs.Rmd/upset3-1.png" target="_blank">0ea6c0c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="drug-specific-de-genes" class="section level2">
<h2><strong>📌 Drug Specific DE genes</strong></h2>
</div>
<div id="upset-plot-cx-5461-vs-dox" class="section level2">
<h2><strong>📌 UpSet Plot: CX-5461 vs DOX</strong></h2>
<pre class="r"><code># Combine DEGs under CX-5461 and DOX
CX_DEGs &lt;- unique(unlist(DEG_list[1:6]))
DOX_DEGs &lt;- unique(unlist(DEG_list[7:12]))

# Create binary matrix for drug-specific DEGs
DEG_matrix_drug &lt;- fromList(list(CX_5461 = CX_DEGs, DOX = DOX_DEGs))

# Generate the UpSet plot for drugs
upset(
  DEG_matrix_drug,
  sets = c(&quot;CX_5461&quot;, &quot;DOX&quot;),
  order.by = &quot;freq&quot;,
  main.bar.color = &quot;darkgreen&quot;,
  point.size = 3,
  text.scale = 1.5,
  matrix.color = &quot;purple&quot;,
  sets.bar.color = c(&quot;blue&quot;, &quot;red&quot;)
)</code></pre>
<p><img src="figure/Overlap_DEGs.Rmd/upset4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-upset4-1">
Past versions of upset4-1.png
</button>
</p>
<div id="fig-upset4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/0ea6c0c0ea9a55a087085290ad025ad37795834f/docs/figure/Overlap_DEGs.Rmd/upset4-1.png" target="_blank">0ea6c0c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="conc.-specific-de-genes" class="section level2">
<h2><strong>📌 Conc. Specific DE genes</strong></h2>
</div>
<div id="upset-plot-concentration-specific-degs" class="section level2">
<h2><strong>📌 UpSet Plot: Concentration-Specific DEGs</strong></h2>
<pre class="r"><code># Combine DEGs under concentrations 0.1 and 0.5
DEG_0.1 &lt;- unique(unlist(DEG_list[c(1, 2, 3, 7, 8, 9)]))
DEG_0.5 &lt;- unique(unlist(DEG_list[c(4, 5, 6, 10, 11, 12)]))

# Create binary matrix for concentration-specific DEGs
DEG_matrix_concentration &lt;- fromList(list(Concentration_0.1 = DEG_0.1, Concentration_0.5 = DEG_0.5))

# Generate the UpSet plot for concentration
upset(
  DEG_matrix_concentration,
  sets = c(&quot;Concentration_0.1&quot;, &quot;Concentration_0.5&quot;),
  order.by = &quot;freq&quot;,
  main.bar.color = &quot;darkorange&quot;,
  matrix.color = &quot;darkblue&quot;,
  sets.bar.color = c(&quot;cyan&quot;, &quot;magenta&quot;),
  text.scale = 1.5,
  keep.order = TRUE
)</code></pre>
<p><img src="figure/Overlap_DEGs.Rmd/upset5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-upset5-1">
Past versions of upset5-1.png
</button>
</p>
<div id="fig-upset5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/0ea6c0c0ea9a55a087085290ad025ad37795834f/docs/figure/Overlap_DEGs.Rmd/upset5-1.png" target="_blank">0ea6c0c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="timepoints-specific-de-genes" class="section level2">
<h2><strong>📌 Timepoints Specific DE genes</strong></h2>
</div>
<div id="upset-plot-timepoint-specific-degs" class="section level2">
<h2><strong>📌 UpSet Plot: Timepoint-Specific DEGs</strong></h2>
<pre class="r"><code># Combine DEGs under timepoints 3hr, 24hr, and 48hr
DEG_3hr &lt;- unique(unlist(DEG_list[c(1, 4, 7, 10)]))
DEG_24hr &lt;- unique(unlist(DEG_list[c(2, 5, 8, 11)]))
DEG_48hr &lt;- unique(unlist(DEG_list[c(3, 6, 9, 12)]))

# Create binary matrix for timepoint-specific DEGs
DEG_matrix_timepoint &lt;- fromList(list(Timepoint_3hr = DEG_3hr, Timepoint_24hr = DEG_24hr, Timepoint_48hr = DEG_48hr))

# Generate the UpSet plot for timepoints
upset(
  DEG_matrix_timepoint,
  sets = c(&quot;Timepoint_3hr&quot;, &quot;Timepoint_24hr&quot;, &quot;Timepoint_48hr&quot;),
  order.by = &quot;freq&quot;,
  main.bar.color = &quot;darkgreen&quot;,
  matrix.color = &quot;darkred&quot;,
  sets.bar.color = c(&quot;blue&quot;, &quot;orange&quot;, &quot;purple&quot;),
  text.scale = 1.5,
  keep.order = TRUE
)</code></pre>
<p><img src="figure/Overlap_DEGs.Rmd/upset6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-upset6-1">
Past versions of upset6-1.png
</button>
</p>
<div id="fig-upset6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/0ea6c0c0ea9a55a087085290ad025ad37795834f/docs/figure/Overlap_DEGs.Rmd/upset6-1.png" target="_blank">0ea6c0c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 26100)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] UpSetR_1.4.0        ggVennDiagram_1.5.2 ggplot2_3.5.2      
[4] workflowr_1.7.1    

loaded via a namespace (and not attached):
 [1] gtable_0.3.6      jsonlite_2.0.0    dplyr_1.1.4       compiler_4.3.0   
 [5] promises_1.3.2    tidyselect_1.2.1  Rcpp_1.0.12       stringr_1.5.1    
 [9] git2r_0.36.2      gridExtra_2.3     callr_3.7.6       later_1.3.2      
[13] jquerylib_0.1.4   scales_1.3.0      yaml_2.3.10       fastmap_1.2.0    
[17] plyr_1.8.9        R6_2.6.1          labeling_0.4.3    generics_0.1.3   
[21] knitr_1.50        tibble_3.2.1      munsell_0.5.1     rprojroot_2.0.4  
[25] bslib_0.9.0       pillar_1.10.2     rlang_1.1.3       cachem_1.1.0     
[29] stringi_1.8.3     httpuv_1.6.15     xfun_0.52         getPass_0.2-4    
[33] fs_1.6.3          sass_0.4.10       cli_3.6.1         withr_3.0.2      
[37] magrittr_2.0.3    ps_1.8.1          digest_0.6.34     grid_4.3.0       
[41] processx_3.8.6    rstudioapi_0.17.1 lifecycle_1.0.4   vctrs_0.6.5      
[45] evaluate_1.0.3    glue_1.7.0        farver_2.1.2      whisker_0.4.1    
[49] colorspace_2.1-0  rmarkdown_2.29    httr_1.4.7        tools_4.3.0      
[53] pkgconfig_2.0.3   htmltools_0.5.8.1</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
