<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Your Name" />

<meta name="date" content="2025-05-22" />

<title>PCA Analysis</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CX5461_Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">PCA Analysis</h1>
<h4 class="author">Your Name</h4>
<h4 class="date">2025-05-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-05-22
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>CX5461_Project/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttreeedfc7e151f7cd8075206029a0f81739fc1c00305targetblankedfc7e1a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/edfc7e151f7cd8075206029a0f81739fc1c00305" target="_blank">edfc7e1</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttreeedfc7e151f7cd8075206029a0f81739fc1c00305targetblankedfc7e1a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/edfc7e151f7cd8075206029a0f81739fc1c00305" target="_blank">edfc7e1</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    0.1 box.svg
    Ignored:    Rplot04.svg
    Ignored:    analysis/Corrmotif_Conc.html
    Ignored:    analysis/DGE_Analysis.html

Untracked files:
    Untracked:  0.1 density.svg
    Untracked:  0.1.emf
    Untracked:  0.1.svg
    Untracked:  0.5 box.svg
    Untracked:  0.5 density.svg
    Untracked:  0.5.svg
    Untracked:  Additional/
    Untracked:  CX_5461_Pattern_Genes_24hr.csv
    Untracked:  CX_5461_Pattern_Genes_3hr.csv
    Untracked:  Cell viability box plot.svg
    Untracked:  DRC1.svg
    Untracked:  Figure 1.jpeg
    Untracked:  Figure 1.pdf
    Untracked:  Figure_CM_Purity.pdf
    Untracked:  Rplot.svg
    Untracked:  Rplot01.svg
    Untracked:  Rplot02.svg
    Untracked:  Rplot03.svg
    Untracked:  Rplot05.svg
    Untracked:  Rplot06.svg
    Untracked:  Rplot07.svg
    Untracked:  Rplot08.jpeg
    Untracked:  Rplot08.svg
    Untracked:  Rplot09.svg
    Untracked:  Rplot10.svg
    Untracked:  Rplot11.svg
    Untracked:  TOP2B.bed
    Untracked:  TS HPA (Violin).svg
    Untracked:  TS HPA.svg
    Untracked:  TS_HA.svg
    Untracked:  TS_HV.svg
    Untracked:  Violin HA.svg
    Untracked:  Violin HV (CX vs DOX).svg
    Untracked:  Violin HV.svg
    Untracked:  data/AF.csv
    Untracked:  data/AF_Mapped.csv
    Untracked:  data/AF_genes.csv
    Untracked:  data/Annotated_DOX_Gene_Table.csv
    Untracked:  data/CAD_genes.csv
    Untracked:  data/Cardiotox.csv
    Untracked:  data/Cardiotox_mapped.csv
    Untracked:  data/DOX_Vald.csv
    Untracked:  data/DOX_Vald_Mapped.csv
    Untracked:  data/DOX_alt.csv
    Untracked:  data/Entrez_Cardiotox.csv
    Untracked:  data/Entrez_Cardiotox_Mapped.csv
    Untracked:  data/GWAS.xlsx
    Untracked:  data/GWAS_SNPs.bed
    Untracked:  data/HF.csv
    Untracked:  data/HF_Mapped.csv
    Untracked:  data/HF_genes.csv
    Untracked:  data/Hypertension_genes.csv
    Untracked:  data/MI_genes.csv
    Untracked:  data/P53_Target_mapped.csv
    Untracked:  data/Sample_annotated.csv
    Untracked:  data/Samples.csv
    Untracked:  data/Samples.xlsx
    Untracked:  data/TOP2A.bed
    Untracked:  data/TOP2A_target.csv
    Untracked:  data/TOP2A_target_lit.csv
    Untracked:  data/TOP2A_target_lit_mapped.csv
    Untracked:  data/TOP2A_target_mapped.csv
    Untracked:  data/TOP2B.bed
    Untracked:  data/TOP2B_target.csv
    Untracked:  data/TOP2B_target_heatmap.csv
    Untracked:  data/TOP2B_target_heatmap_mapped.csv
    Untracked:  data/TOP2B_target_mapped.csv
    Untracked:  data/TS.csv
    Untracked:  data/TS_HPA.csv
    Untracked:  data/TS_HPA_mapped.csv
    Untracked:  data/Toptable_CX_0.1_24.csv
    Untracked:  data/Toptable_CX_0.1_3.csv
    Untracked:  data/Toptable_CX_0.1_48.csv
    Untracked:  data/Toptable_CX_0.5_24.csv
    Untracked:  data/Toptable_CX_0.5_3.csv
    Untracked:  data/Toptable_CX_0.5_48.csv
    Untracked:  data/Toptable_DOX_0.1_24.csv
    Untracked:  data/Toptable_DOX_0.1_3.csv
    Untracked:  data/Toptable_DOX_0.1_48.csv
    Untracked:  data/Toptable_DOX_0.5_24.csv
    Untracked:  data/Toptable_DOX_0.5_3.csv
    Untracked:  data/Toptable_DOX_0.5_48.csv
    Untracked:  data/count.tsv
    Untracked:  data/ts_data_mapped
    Untracked:  results/
    Untracked:  run_bedtools.bat

Unstaged changes:
    Deleted:    analysis/Actox.Rmd
    Modified:   data/count.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/PCA.Rmd</code>) and HTML
(<code>docs/PCA.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/analysis/PCA.Rmd" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/PCA.html" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/PCA.html" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/analysis/PCA.Rmd" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/PCA.html" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/PCA.html" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/56e44e607a92add89b054abf771e904ed50cd9f5/analysis/PCA.Rmd" target="_blank">56e44e6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
<td>
Fixed duplicate chunk labels in PCA analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/773671b22a4d8f6d93cf10e7a7d48152263e29af/docs/PCA.html" target="_blank">773671b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/91e6c2cbf0317a6c47c3dd473681f181f986688e/analysis/PCA.Rmd" target="_blank">91e6c2c</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-01
</td>
<td>
Fixed duplicate row names issue in count matrix
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-required-libraries" class="section level3">
<h3><strong>Load Required Libraries</strong></h3>
<pre class="r"><code>library(edgeR)</code></pre>
<pre><code>Warning: package &#39;edgeR&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;limma&#39; was built under R version 4.3.1</code></pre>
<pre class="r"><code>library(ggplot2)
library(reshape2)
library(dplyr)</code></pre>
<pre><code>Warning: package &#39;dplyr&#39; was built under R version 4.3.2</code></pre>
<pre class="r"><code>library(Biobase)</code></pre>
<pre><code>Warning: package &#39;Biobase&#39; was built under R version 4.3.1</code></pre>
<pre><code>Warning: package &#39;BiocGenerics&#39; was built under R version 4.3.1</code></pre>
<pre class="r"><code>library(limma)
library(tidyverse)</code></pre>
<pre><code>Warning: package &#39;tidyverse&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;tidyr&#39; was built under R version 4.3.3</code></pre>
<pre><code>Warning: package &#39;readr&#39; was built under R version 4.3.3</code></pre>
<pre><code>Warning: package &#39;purrr&#39; was built under R version 4.3.3</code></pre>
<pre><code>Warning: package &#39;stringr&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;lubridate&#39; was built under R version 4.3.3</code></pre>
<pre class="r"><code>library(scales)</code></pre>
<pre><code>Warning: package &#39;scales&#39; was built under R version 4.3.2</code></pre>
<pre class="r"><code>library(biomaRt)</code></pre>
<pre><code>Warning: package &#39;biomaRt&#39; was built under R version 4.3.2</code></pre>
<pre class="r"><code>library(ggrepel)</code></pre>
<pre><code>Warning: package &#39;ggrepel&#39; was built under R version 4.3.3</code></pre>
<pre class="r"><code>library(corrplot)</code></pre>
<pre><code>Warning: package &#39;corrplot&#39; was built under R version 4.3.3</code></pre>
<pre class="r"><code>library(Hmisc)</code></pre>
<pre><code>Warning: package &#39;Hmisc&#39; was built under R version 4.3.3</code></pre>
<pre class="r"><code>library(org.Hs.eg.db)</code></pre>
<pre><code>Warning: package &#39;AnnotationDbi&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;IRanges&#39; was built under R version 4.3.1</code></pre>
<pre><code>Warning: package &#39;S4Vectors&#39; was built under R version 4.3.2</code></pre>
<pre class="r"><code>library(AnnotationDbi)
library(tidyr)
library(ggfortify)</code></pre>
<p>📍 Load the Count Matrix CSV file</p>
</div>
<div id="color-palettes" class="section level2">
<h2><strong>📌Color palettes</strong></h2>
<pre class="r"><code>### 📌 Color palettes (updated)
drug_conc_palette &lt;- c(
  &quot;CX-5461_0.1&quot; = &quot;gold&quot;,  # light green
  &quot;CX-5461_0.5&quot; = &quot;green4&quot;,  # dark green
  &quot;DOX_0.1&quot;     = &quot;salmon2&quot;,  # peach
  &quot;DOX_0.5&quot;     = &quot;red3&quot;,  # burnt orange
  &quot;VEH_0.1&quot;     = &quot;lightblue3&quot;,  # sky blue
  &quot;VEH_0.5&quot;     = &quot;darkblue&quot;   # navy blue
)
drug_palc &lt;- c(&quot;#8B006D&quot;,&quot;#DF707E&quot;,&quot;#F1B72B&quot;, &quot;#3386DD&quot;,&quot;#707031&quot;,&quot;#41B333&quot;)
drug_palc1 &lt;- c(&quot;#8B006D&quot;,&quot;#F1B72B&quot;, &quot;#3386DD&quot;,&quot;#707031&quot;)
drug_palc2 &lt;- c(&quot;#8B006D&quot;,&quot;#F1B72B&quot;, &quot;#3386DD&quot;)</code></pre>
</div>
<div id="load-metadata" class="section level2">
<h2><strong>📌Load Metadata</strong></h2>
</div>
<div id="pca-of-unfiltered-log2cpm" class="section level2">
<h2><strong>📌PCA of Unfiltered log2(CPM)</strong></h2>
<pre class="r"><code>prcomp_res &lt;- prcomp(t(matrix), center = TRUE)

ggplot2::autoplot(prcomp_res, data = Metadata,
                  colour = &quot;Drug_Conc&quot;, shape = &quot;Time&quot;, size = 4, x = 1, y = 2) +
  ggrepel::geom_text_repel(label = Indiv) +
  scale_color_manual(values = drug_conc_palette) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Unfiltered&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 67 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_unfiltered-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_unfiltered-1">
Past versions of pca_unfiltered-1.png
</button>
</p>
<div id="fig-pca_unfiltered-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_unfiltered-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_unfiltered-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_unfiltered-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/773671b22a4d8f6d93cf10e7a7d48152263e29af/docs/figure/PCA.Rmd/pca_unfiltered-1.png" target="_blank">773671b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res &lt;- prcomp(t(lcpm %&gt;% as.matrix()), center = TRUE)

ggplot2::autoplot(prcomp_res, data = Metadata, colour = &quot;Condition&quot;, shape = &quot;Time&quot;, size =4, x=2, y=3) +
  ggrepel::geom_text_repel(label=Indiv) +
  scale_color_manual(values=drug_palc) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Unfiltered&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 33 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_unfiltered2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_unfiltered2-1">
Past versions of pca_unfiltered2-1.png
</button>
</p>
<div id="fig-pca_unfiltered2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_unfiltered2-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/PCA.Rmd/pca_unfiltered2-1.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_unfiltered2-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_unfiltered2-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/773671b22a4d8f6d93cf10e7a7d48152263e29af/docs/figure/PCA.Rmd/pca_unfiltered2-1.png" target="_blank">773671b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res &lt;- prcomp(t(lcpm %&gt;% as.matrix()), center = TRUE)

ggplot2::autoplot(prcomp_res, data = Metadata, colour = &quot;Condition&quot;, shape = &quot;Time&quot;, size =4, x=3, y=4) +
  ggrepel::geom_text_repel(label=Indiv) +
  scale_color_manual(values=drug_palc) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Unfiltered&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 38 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_unfiltered3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_unfiltered3-1">
Past versions of pca_unfiltered3-1.png
</button>
</p>
<div id="fig-pca_unfiltered3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_unfiltered3-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/PCA.Rmd/pca_unfiltered3-1.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_unfiltered3-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_unfiltered3-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/773671b22a4d8f6d93cf10e7a7d48152263e29af/docs/figure/PCA.Rmd/pca_unfiltered3-1.png" target="_blank">773671b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pca-of-filtered-log2cpm-rowmeans-0" class="section level2">
<h2><strong>📌PCA of Filtered log2(CPM) (RowMeans &gt; 0)</strong></h2>
<pre class="r"><code>prcomp_res1 &lt;- prcomp(t(filcpm_matrix %&gt;% as.matrix()), center = TRUE)

ggplot2::autoplot(prcomp_res1, data = Metadata, colour = &quot;Drug_Conc&quot;, shape = &quot;Time&quot;, size =4, x=1, y=2) +
  ggrepel::geom_text_repel(label=Indiv) +
  scale_color_manual(values=drug_conc_palette) +
  ggtitle(expression(&quot;PCA of gene expression (log2 cpm)&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 51 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_filtered_0-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_filtered_0-1">
Past versions of pca_filtered_0-1.png
</button>
</p>
<div id="fig-pca_filtered_0-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_filtered_0-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/PCA.Rmd/pca_filtered_0-1.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_filtered_0-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_filtered_0-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/773671b22a4d8f6d93cf10e7a7d48152263e29af/docs/figure/PCA.Rmd/pca_filtered_0-1.png" target="_blank">773671b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res1 &lt;- prcomp(t(filcpm_matrix %&gt;% as.matrix()), center = TRUE)

ggplot2::autoplot(prcomp_res1, data = Metadata, colour = &quot;Drug_Conc&quot;, shape = &quot;Time&quot;, size =4, x=2, y=3) +
  ggrepel::geom_text_repel(label=Indiv) +
  scale_color_manual(values=drug_conc_palette) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) filtered (Rowmeans &gt;0)&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 22 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_filtered_0_1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_filtered_0_1-1">
Past versions of pca_filtered_0_1-1.png
</button>
</p>
<div id="fig-pca_filtered_0_1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_filtered_0_1-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/PCA.Rmd/pca_filtered_0_1-1.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_filtered_0_1-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_filtered_0_1-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/773671b22a4d8f6d93cf10e7a7d48152263e29af/docs/figure/PCA.Rmd/pca_filtered_0_1-1.png" target="_blank">773671b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res1 &lt;- prcomp(t(filcpm_matrix %&gt;% as.matrix()), center = TRUE)

ggplot2::autoplot(prcomp_res1, data = Metadata, colour = &quot;Condition&quot;, shape = &quot;Time&quot;, size =4, x=3, y=4) +
  ggrepel::geom_text_repel(label=Indiv) +
  scale_color_manual(values=drug_palc) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) filtered (Rowmeans &gt;0)&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 26 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_filtered_0_2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_filtered_0_2-1">
Past versions of pca_filtered_0_2-1.png
</button>
</p>
<div id="fig-pca_filtered_0_2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_filtered_0_2-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/PCA.Rmd/pca_filtered_0_2-1.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_filtered_0_2-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_filtered_0_2-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/773671b22a4d8f6d93cf10e7a7d48152263e29af/docs/figure/PCA.Rmd/pca_filtered_0_2-1.png" target="_blank">773671b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pca-of-filtered-log2cpm-rowmeans-0.5" class="section level2">
<h2><strong>📌PCA of Filtered log2(CPM) (RowMeans &gt;
0.5)</strong></h2>
<pre class="r"><code>prcomp_res2 &lt;- prcomp(t(filcpm_matrix1 %&gt;% as.matrix()), center = TRUE)

ggplot2::autoplot(prcomp_res2, data = Metadata, colour = &quot;Condition&quot;, shape = &quot;Time&quot;, size =4, x=1, y=2) +
  ggrepel::geom_text_repel(label=Indiv) +
  scale_color_manual(values=drug_palc) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) filtered (Rowmeans &gt;0.5)&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 54 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_filtered_0.5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_filtered_0.5-1">
Past versions of pca_filtered_0.5-1.png
</button>
</p>
<div id="fig-pca_filtered_0.5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_filtered_0.5-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/PCA.Rmd/pca_filtered_0.5-1.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_filtered_0.5-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_filtered_0.5-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/773671b22a4d8f6d93cf10e7a7d48152263e29af/docs/figure/PCA.Rmd/pca_filtered_0.5-1.png" target="_blank">773671b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res2 &lt;- prcomp(t(filcpm_matrix1 %&gt;% as.matrix()), center = TRUE)

ggplot2::autoplot(prcomp_res2, data = Metadata, colour = &quot;Condition&quot;, shape = &quot;Time&quot;, size =4, x=2, y=3) +
  ggrepel::geom_text_repel(label=Indiv) +
  scale_color_manual(values=drug_palc) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) filtered (Rowmeans &gt;0.5)&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 28 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_filtered_0.5_1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_filtered_0.5_1-1">
Past versions of pca_filtered_0.5_1-1.png
</button>
</p>
<div id="fig-pca_filtered_0.5_1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_filtered_0.5_1-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/PCA.Rmd/pca_filtered_0.5_1-1.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_filtered_0.5_1-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_filtered_0.5_1-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/773671b22a4d8f6d93cf10e7a7d48152263e29af/docs/figure/PCA.Rmd/pca_filtered_0.5_1-1.png" target="_blank">773671b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res2 &lt;- prcomp(t(filcpm_matrix1 %&gt;% as.matrix()), center = TRUE)

ggplot2::autoplot(prcomp_res2, data = Metadata, colour = &quot;Condition&quot;, shape = &quot;Time&quot;, size =4, x=3, y=4) +
  ggrepel::geom_text_repel(label=Indiv) +
  scale_color_manual(values=drug_palc) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) filtered (Rowmeans &gt;0.5)&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 26 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_filtered_0.5_2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_filtered_0.5_2-1">
Past versions of pca_filtered_0.5_2-1.png
</button>
</p>
<div id="fig-pca_filtered_0.5_2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_filtered_0.5_2-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/PCA.Rmd/pca_filtered_0.5_2-1.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_filtered_0.5_2-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_filtered_0.5_2-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/773671b22a4d8f6d93cf10e7a7d48152263e29af/docs/figure/PCA.Rmd/pca_filtered_0.5_2-1.png" target="_blank">773671b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pca-of-filtered-log2cpm-rowmeans-1" class="section level2">
<h2><strong>📌PCA of Filtered log2(CPM) (RowMeans &gt; 1)</strong></h2>
<pre class="r"><code>prcomp_res3 &lt;- prcomp(t(filcpm_matrix2 %&gt;% as.matrix()), center = TRUE)

ggplot2::autoplot(prcomp_res3, data = Metadata, colour = &quot;Condition&quot;, shape = &quot;Time&quot;, size =4, x=1, y=2) +
  ggrepel::geom_text_repel(label=Indiv) +
  scale_color_manual(values=drug_palc) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) filtered (Rowmeans &gt;1)&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 60 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_filtered_1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_filtered_1-1">
Past versions of pca_filtered_1-1.png
</button>
</p>
<div id="fig-pca_filtered_1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_filtered_1-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/PCA.Rmd/pca_filtered_1-1.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_filtered_1-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_filtered_1-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/773671b22a4d8f6d93cf10e7a7d48152263e29af/docs/figure/PCA.Rmd/pca_filtered_1-1.png" target="_blank">773671b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res3 &lt;- prcomp(t(filcpm_matrix2 %&gt;% as.matrix()), center = TRUE)

ggplot2::autoplot(prcomp_res3, data = Metadata, colour = &quot;Condition&quot;, shape = &quot;Time&quot;, size =4, x=2, y=3) +
  ggrepel::geom_text_repel(label=Indiv) +
  scale_color_manual(values=drug_palc) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) filtered (Rowmeans &gt;1)&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 31 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_filtered_1_1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_filtered_1_1-1">
Past versions of pca_filtered_1_1-1.png
</button>
</p>
<div id="fig-pca_filtered_1_1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_filtered_1_1-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/PCA.Rmd/pca_filtered_1_1-1.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_filtered_1_1-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_filtered_1_1-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/773671b22a4d8f6d93cf10e7a7d48152263e29af/docs/figure/PCA.Rmd/pca_filtered_1_1-1.png" target="_blank">773671b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res3 &lt;- prcomp(t(filcpm_matrix2 %&gt;% as.matrix()), center = TRUE)

ggplot2::autoplot(prcomp_res3, data = Metadata, colour = &quot;Condition&quot;, shape = &quot;Time&quot;, size =4, x=3, y=4) +
  ggrepel::geom_text_repel(label=Indiv) +
  scale_color_manual(values=drug_palc) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) filtered (Rowmeans &gt;1)&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 16 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_filtered_1_2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_filtered_1_2-1">
Past versions of pca_filtered_1_2-1.png
</button>
</p>
<div id="fig-pca_filtered_1_2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_filtered_1_2-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/PCA.Rmd/pca_filtered_1_2-1.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_filtered_1_2-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_filtered_1_2-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/773671b22a4d8f6d93cf10e7a7d48152263e29af/docs/figure/PCA.Rmd/pca_filtered_1_2-1.png" target="_blank">773671b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pca-analysis-by-drugs" class="section level2">
<h2><strong>📌 PCA Analysis by Drugs</strong></h2>
</div>
<div id="pca-analysis-cx-5461-veh" class="section level2">
<h2><strong>📌 PCA Analysis: CX-5461 &amp; VEH </strong></h2>
<pre class="r"><code>selected_columns &lt;- grepl(&quot;VEH|CX.5461&quot;, colnames(matrix))
subset_matrix_CX &lt;- matrix[, selected_columns]

subset_meta &lt;- subset(Metadata, Metadata$Drug %in% c(&quot;VEH&quot;, &quot;CX-5461&quot;))

prcomp_res4 &lt;- prcomp(t(subset_matrix_CX), center = TRUE)

ggplot2::autoplot(prcomp_res4, data = as.data.frame(subset_meta), colour = &quot;Condition&quot;, shape = &quot;Time&quot;, size = 4) +
  ggrepel::geom_text_repel(label = subset_meta$Ind) +  # ✅ Corrected label
  scale_color_manual(values = drug_palc1) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Unfiltered (CX-5461 vs VEH)&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 4 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_drug-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_drug-1">
Past versions of pca_drug-1.png
</button>
</p>
<div id="fig-pca_drug-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_drug-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_drug-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_drug-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res5 &lt;- prcomp(t(subset_matrix_CX[rowMeans(subset_matrix_CX) &gt; 0, ]), center = TRUE)

ggplot2::autoplot(prcomp_res5, data = as.data.frame(subset_meta), colour = &quot;Condition&quot;, shape = &quot;Time&quot;, size = 4) +
  ggrepel::geom_text_repel(label = subset_meta$Ind) +
  scale_color_manual(values = drug_palc1) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 0 (CX-5461 vs VEH)&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 2 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_drug_1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_drug_1-1">
Past versions of pca_drug_1-1.png
</button>
</p>
<div id="fig-pca_drug_1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_drug_1-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_drug_1-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_drug_1-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res6 &lt;- prcomp(t(subset_matrix_CX[rowMeans(subset_matrix_CX) &gt; 0.5, ]), center = TRUE)

ggplot2::autoplot(prcomp_res6, data = as.data.frame(subset_meta), colour = &quot;Condition&quot;, shape = &quot;Time&quot;, size = 4) +
  ggrepel::geom_text_repel(label = subset_meta$Ind) +
  scale_color_manual(values = drug_palc1) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 0.5 (CX-5461 vs VEH)&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 2 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_drug_2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_drug_2-1">
Past versions of pca_drug_2-1.png
</button>
</p>
<div id="fig-pca_drug_2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_drug_2-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_drug_2-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_drug_2-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res7 &lt;- prcomp(t(subset_matrix_CX[rowMeans(subset_matrix_CX) &gt; 1, ]), center = TRUE)

ggplot2::autoplot(prcomp_res7, data = as.data.frame(subset_meta), colour = &quot;Condition&quot;, shape = &quot;Time&quot;, size = 4) +
  ggrepel::geom_text_repel(label = subset_meta$Ind) +
  scale_color_manual(values = drug_palc1) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 1 (CX-5461 vs VEH)&quot;)) +
  theme_bw()</code></pre>
<p><img src="figure/PCA.Rmd/pca_drug_3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_drug_3-1">
Past versions of pca_drug_3-1.png
</button>
</p>
<div id="fig-pca_drug_3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_drug_3-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/PCA.Rmd/pca_drug_3-1.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_drug_3-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_drug_3-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pca-analysis-dox-veh" class="section level2">
<h2><strong>📌 PCA Analysis: DOX &amp; VEH </strong></h2>
<pre class="r"><code>selected_columns &lt;- grepl(&quot;VEH|DOX&quot;, colnames(matrix))
subset_matrix_DOX &lt;- matrix[, selected_columns]

subset_meta_dox &lt;- subset(Metadata, Metadata$Drug %in% c(&quot;VEH&quot;, &quot;DOX&quot;))

prcomp_res8 &lt;- prcomp(t(subset_matrix_DOX), center = TRUE)

ggplot2::autoplot(prcomp_res8, data = as.data.frame(subset_meta_dox), colour = &quot;Condition&quot;, shape = &quot;Time&quot;, size = 4) +
  ggrepel::geom_text_repel(label = subset_meta_dox$Ind) +
  scale_color_manual(values = drug_palc1) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Unfiltered (DOX vs VEH)&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 42 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_drug_dox_veh-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_drug_dox_veh-1">
Past versions of pca_drug_dox_veh-1.png
</button>
</p>
<div id="fig-pca_drug_dox_veh-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_drug_dox_veh-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_drug_dox_veh-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_drug_dox_veh-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res9 &lt;- prcomp(t(subset_matrix_DOX[rowMeans(subset_matrix_DOX) &gt; 0, ]), center = TRUE)

ggplot2::autoplot(prcomp_res9, data = as.data.frame(subset_meta_dox), colour = &quot;Condition&quot;, shape = &quot;Time&quot;, size = 4) +
  ggrepel::geom_text_repel(label = subset_meta_dox$Ind) +
  scale_color_manual(values = drug_palc1) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 0 (DOX vs VEH)&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 32 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_drug_dox_veh_1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_drug_dox_veh_1-1">
Past versions of pca_drug_dox_veh_1-1.png
</button>
</p>
<div id="fig-pca_drug_dox_veh_1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_drug_dox_veh_1-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_drug_dox_veh_1-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_drug_dox_veh_1-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res10 &lt;- prcomp(t(subset_matrix_DOX[rowMeans(subset_matrix_DOX) &gt; 0.5, ]), center = TRUE)

ggplot2::autoplot(prcomp_res10, data = as.data.frame(subset_meta_dox), colour = &quot;Condition&quot;, shape = &quot;Time&quot;, size = 4) +
  ggrepel::geom_text_repel(label = subset_meta_dox$Ind) +
  scale_color_manual(values = drug_palc1) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 0.5 (DOX vs VEH)&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 30 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_drug_dox_veh_2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_drug_dox_veh_2-1">
Past versions of pca_drug_dox_veh_2-1.png
</button>
</p>
<div id="fig-pca_drug_dox_veh_2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_drug_dox_veh_2-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_drug_dox_veh_2-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_drug_dox_veh_2-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res11 &lt;- prcomp(t(subset_matrix_DOX[rowMeans(subset_matrix_DOX) &gt; 1, ]), center = TRUE)

ggplot2::autoplot(prcomp_res11, data = as.data.frame(subset_meta_dox), colour = &quot;Condition&quot;, shape = &quot;Time&quot;, size = 4) +
  ggrepel::geom_text_repel(label = subset_meta_dox$Ind) +
  scale_color_manual(values = drug_palc1) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 1 (DOX vs VEH)&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 34 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_drug_dox_veh_3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_drug_dox_veh_3-1">
Past versions of pca_drug_dox_veh_3-1.png
</button>
</p>
<div id="fig-pca_drug_dox_veh_3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_drug_dox_veh_3-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_drug_dox_veh_3-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_drug_dox_veh_3-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pca-analysis-cx-5461-dox" class="section level2">
<h2><strong>📌 PCA Analysis: CX-5461 &amp; DOX</strong></h2>
<pre class="r"><code>selected_columns &lt;- grepl(&quot;CX.5461|DOX&quot;, colnames(matrix))
subset_matrix_CX_DOX &lt;- matrix[, selected_columns]

subset_meta_cx_dox &lt;- subset(Metadata, Metadata$Drug %in% c(&quot;CX-5461&quot;, &quot;DOX&quot;))

prcomp_res12 &lt;- prcomp(t(subset_matrix_CX_DOX), center = TRUE)

ggplot2::autoplot(prcomp_res12, data = as.data.frame(subset_meta_cx_dox), colour = &quot;Condition&quot;, shape = &quot;Time&quot;, size = 4) +
  ggrepel::geom_text_repel(label = subset_meta_cx_dox$Ind) +
  scale_color_manual(values = drug_palc1) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Unfiltered (CX-5461 vs DOX)&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 30 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_drug_cx_dox-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_drug_cx_dox-1">
Past versions of pca_drug_cx_dox-1.png
</button>
</p>
<div id="fig-pca_drug_cx_dox-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_drug_cx_dox-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_drug_cx_dox-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_drug_cx_dox-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res13 &lt;- prcomp(t(subset_matrix_CX_DOX[rowMeans(subset_matrix_CX_DOX) &gt; 0, ]), center = TRUE)

ggplot2::autoplot(prcomp_res13, data = as.data.frame(subset_meta_cx_dox), colour = &quot;Condition&quot;, shape = &quot;Time&quot;, size = 4) +
  ggrepel::geom_text_repel(label = subset_meta_cx_dox$Ind) +
  scale_color_manual(values = drug_palc1) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 0 (CX-5461 vs DOX)&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 16 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_drug_cx_dox_1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_drug_cx_dox_1-1">
Past versions of pca_drug_cx_dox_1-1.png
</button>
</p>
<div id="fig-pca_drug_cx_dox_1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_drug_cx_dox_1-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/PCA.Rmd/pca_drug_cx_dox_1-1.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_drug_cx_dox_1-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_drug_cx_dox_1-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res14 &lt;- prcomp(t(subset_matrix_CX_DOX[rowMeans(subset_matrix_CX_DOX) &gt; 0.5, ]), center = TRUE)

ggplot2::autoplot(prcomp_res14, data = as.data.frame(subset_meta_cx_dox), colour = &quot;Condition&quot;, shape = &quot;Time&quot;, size = 4) +
  ggrepel::geom_text_repel(label = subset_meta_cx_dox$Ind) +
  scale_color_manual(values = drug_palc1) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 0.5 (CX-5461 vs DOX)&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 15 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_drug_cx_dox_2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_drug_cx_dox_2-1">
Past versions of pca_drug_cx_dox_2-1.png
</button>
</p>
<div id="fig-pca_drug_cx_dox_2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_drug_cx_dox_2-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/PCA.Rmd/pca_drug_cx_dox_2-1.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_drug_cx_dox_2-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_drug_cx_dox_2-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res15 &lt;- prcomp(t(subset_matrix_CX_DOX[rowMeans(subset_matrix_CX_DOX) &gt; 1, ]), center = TRUE)

ggplot2::autoplot(prcomp_res15, data = as.data.frame(subset_meta_cx_dox), colour = &quot;Condition&quot;, shape = &quot;Time&quot;, size = 4) +
  ggrepel::geom_text_repel(label = subset_meta_cx_dox$Ind) +
  scale_color_manual(values = drug_palc1) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 1 (CX-5461 vs DOX)&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 21 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_drug_cx_dox_3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_drug_cx_dox_3-1">
Past versions of pca_drug_cx_dox_3-1.png
</button>
</p>
<div id="fig-pca_drug_cx_dox_3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_drug_cx_dox_3-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_drug_cx_dox_3-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_drug_cx_dox_3-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pca-analysis-by-timepoints" class="section level2">
<h2><strong>📌 PCA Analysis by Timepoints</strong></h2>
</div>
<div id="hour-timepoint" class="section level2">
<h2><strong>📌 3-Hour Timepoint</strong></h2>
<pre class="r"><code>selected_columns &lt;- grepl(&quot;_3&quot;, colnames(matrix))
subset_matrix_3hr &lt;- matrix[, selected_columns]
subset_meta_3hr &lt;- subset(Metadata, Metadata$Time == &quot;3hr&quot;)

filtered_matrix_3hr &lt;- subset_matrix_3hr[rowMeans(subset_matrix_3hr) &gt; 0, ]

if (nrow(filtered_matrix_3hr) &gt; 2) {
  prcomp_res_3hr &lt;- prcomp(t(filtered_matrix_3hr), center = TRUE)

  ggplot2::autoplot(prcomp_res_3hr, data = as.data.frame(subset_meta_3hr),
                    colour = &quot;Drug_Conc&quot;, shape = &quot;Drug&quot;, size = 4) +
    ggrepel::geom_text_repel(aes(label = Ind)) +
    scale_color_manual(values = drug_conc_palette) +
    ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 0 (3 Hours)&quot;)) +
    theme_bw()
} else {
  print(&quot;No genes passed the rowMeans &gt; 0 filter for 3hr samples.&quot;)
}</code></pre>
<p><img src="figure/PCA.Rmd/pca_3hr-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_3hr-1">
Past versions of pca_3hr-1.png
</button>
</p>
<div id="fig-pca_3hr-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_3hr-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_3hr-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_3hr-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res_3hr &lt;- prcomp(t(subset_matrix_3hr[rowMeans(subset_matrix_3hr) &gt; 0, ]), center = TRUE)

ggplot2::autoplot(prcomp_res_3hr, data = as.data.frame(subset_meta_3hr), colour = &quot;Condition&quot;, shape = &quot;Drug&quot;, size = 4, x=2, y=3) +
  ggrepel::geom_text_repel(label = subset_meta_3hr$Ind) +
  scale_color_manual(values = drug_palc) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 0 (3 Hours)&quot;)) +
  theme_bw()</code></pre>
<p><img src="figure/PCA.Rmd/pca_3hr_2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_3hr_2-1">
Past versions of pca_3hr_2-1.png
</button>
</p>
<div id="fig-pca_3hr_2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_3hr_2-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_3hr_2-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res_3hr &lt;- prcomp(t(subset_matrix_3hr[rowMeans(subset_matrix_3hr) &gt; 0, ]), center = TRUE)

ggplot2::autoplot(prcomp_res_3hr, data = as.data.frame(subset_meta_3hr), colour = &quot;Condition&quot;, shape = &quot;Drug&quot;, size = 4, x=3, y=4) +
  ggrepel::geom_text_repel(label = subset_meta_3hr$Ind) +
  scale_color_manual(values = drug_palc) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 0 (3 Hours)&quot;)) +
  theme_bw()</code></pre>
<p><img src="figure/PCA.Rmd/pca_3hr_3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_3hr_3-1">
Past versions of pca_3hr_3-1.png
</button>
</p>
<div id="fig-pca_3hr_3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_3hr_3-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_3hr_3-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="hour-timepoint-1" class="section level2">
<h2><strong>📌 24-Hour Timepoint</strong></h2>
<pre class="r"><code># 📌 Subset for 24hr samples
selected_columns &lt;- grepl(&quot;_24&quot;, colnames(matrix))
subset_matrix_24hr &lt;- matrix[, selected_columns]
subset_meta_24hr &lt;- subset(Metadata, Metadata$Time == &quot;24hr&quot;)  # match your relabeled timepoints

# 📌 Filter low-expression genes
filtered_matrix_24hr &lt;- subset_matrix_24hr[rowMeans(subset_matrix_24hr) &gt; 0, ]

# 📌 Run PCA if genes remain
if (nrow(filtered_matrix_24hr) &gt; 2) {
  prcomp_res_24hr &lt;- prcomp(t(filtered_matrix_24hr), center = TRUE)

  ggplot2::autoplot(prcomp_res_24hr, data = as.data.frame(subset_meta_24hr),
                    colour = &quot;Drug_Conc&quot;, shape = &quot;Drug&quot;, size = 4) +
    ggrepel::geom_text_repel(aes(label = Ind)) +
    scale_color_manual(values = drug_conc_palette) +
    ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 0 (24 Hours)&quot;)) +
    theme_bw()
} else {
  message(&quot;⚠️ No genes passed the rowMeans &gt; 0 filter for 24-hour samples.&quot;)
}</code></pre>
<p><img src="figure/PCA.Rmd/pca_24hr-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_24hr-1">
Past versions of pca_24hr-1.png
</button>
</p>
<div id="fig-pca_24hr-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_24hr-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_24hr-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_24hr-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res_24hr &lt;- prcomp(t(subset_matrix_24hr[rowMeans(subset_matrix_24hr) &gt; 0, ]), center = TRUE)

ggplot2::autoplot(prcomp_res_24hr, data = as.data.frame(subset_meta_24hr), colour = &quot;Condition&quot;, shape = &quot;Drug&quot;, size = 4, x=2, y=3) +
  ggrepel::geom_text_repel(label = subset_meta_24hr$Ind) +
  scale_color_manual(values = drug_palc) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 0 (24 Hours)&quot;)) +
  theme_bw()</code></pre>
<p><img src="figure/PCA.Rmd/pca_24hr_1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_24hr_1-1">
Past versions of pca_24hr_1-1.png
</button>
</p>
<div id="fig-pca_24hr_1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_24hr_1-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_24hr_1-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res_24hr &lt;- prcomp(t(subset_matrix_24hr[rowMeans(subset_matrix_24hr) &gt; 0, ]), center = TRUE)

ggplot2::autoplot(prcomp_res_24hr, data = as.data.frame(subset_meta_24hr), colour = &quot;Condition&quot;, shape = &quot;Drug&quot;, size = 4, x=3, y=4) +
  ggrepel::geom_text_repel(label = subset_meta_24hr$Ind) +
  scale_color_manual(values = drug_palc) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 0 (24 Hours)&quot;)) +
  theme_bw()</code></pre>
<p><img src="figure/PCA.Rmd/pca_24hr_2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_24hr_2-1">
Past versions of pca_24hr_2-1.png
</button>
</p>
<div id="fig-pca_24hr_2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_24hr_2-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_24hr_2-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="hour-timepoint-2" class="section level2">
<h2><strong>📌 48-Hour Timepoint</strong></h2>
<pre class="r"><code># 📌 Subset for 48hr samples
selected_columns &lt;- grepl(&quot;_48&quot;, colnames(matrix))
subset_matrix_48hr &lt;- matrix[, selected_columns]
subset_meta_48hr &lt;- subset(Metadata, Metadata$Time == &quot;48hr&quot;)  # must match relabeled levels

# 📌 Filter low-expression genes
filtered_matrix_48hr &lt;- subset_matrix_48hr[rowMeans(subset_matrix_48hr) &gt; 0, ]

# 📌 Run PCA only if data is valid
if (nrow(filtered_matrix_48hr) &gt; 2) {
  prcomp_res_48hr_1 &lt;- prcomp(t(filtered_matrix_48hr), center = TRUE)

  ggplot2::autoplot(prcomp_res_48hr_1, data = as.data.frame(subset_meta_48hr),
                    colour = &quot;Drug_Conc&quot;, shape = &quot;Drug&quot;, size = 4) +
    ggrepel::geom_text_repel(aes(label = Ind)) +
    scale_color_manual(values = drug_conc_palette) +
    ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 0 (48 Hours)&quot;)) +
    theme_bw()
} else {
  message(&quot;⚠️ No genes passed the rowMeans &gt; 0 filter for 48-hour samples.&quot;)
}</code></pre>
<pre><code>Warning: ggrepel: 1 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_48hr-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_48hr-1">
Past versions of pca_48hr-1.png
</button>
</p>
<div id="fig-pca_48hr-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_48hr-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_48hr-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_48hr-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res_48hr_1 &lt;- prcomp(t(subset_matrix_48hr[rowMeans(subset_matrix_48hr) &gt; 0, ]), center = TRUE)

ggplot2::autoplot(prcomp_res_48hr_1, data = as.data.frame(subset_meta_48hr), colour = &quot;Condition&quot;, shape = &quot;Drug&quot;, size = 4, x=2, y=3) +
  ggrepel::geom_text_repel(label = subset_meta_48hr$Ind) +
  scale_color_manual(values = drug_palc) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 0 (48 Hours)&quot;)) +
  theme_bw()</code></pre>
<p><img src="figure/PCA.Rmd/pca_48hr_1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_48hr_1-1">
Past versions of pca_48hr_1-1.png
</button>
</p>
<div id="fig-pca_48hr_1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_48hr_1-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_48hr_1-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res_48hr_1 &lt;- prcomp(t(subset_matrix_48hr[rowMeans(subset_matrix_48hr) &gt; 0, ]), center = TRUE)

ggplot2::autoplot(prcomp_res_48hr_1, data = as.data.frame(subset_meta_48hr), colour = &quot;Condition&quot;, shape = &quot;Drug&quot;, size = 4, x=3, y=4) +
  ggrepel::geom_text_repel(label = subset_meta_48hr$Ind) +
  scale_color_manual(values = drug_palc) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 0 (48 Hours)&quot;)) +
  theme_bw()</code></pre>
<p><img src="figure/PCA.Rmd/pca_48hr_2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_48hr_2-1">
Past versions of pca_48hr_2-1.png
</button>
</p>
<div id="fig-pca_48hr_2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_48hr_2-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/PCA.Rmd/pca_48hr_2-1.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_48hr_2-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_48hr_2-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pca-analysis-by-concentrations" class="section level2">
<h2><strong>📌 PCA Analysis by Concentrations</strong></h2>
</div>
<div id="µm-concentration" class="section level2">
<h2><strong>📌0.1 µM Concentration</strong></h2>
<pre class="r"><code>selected_columns &lt;- grepl(&quot;_0.1_&quot;, colnames(matrix))
subset_matrix_0.1 &lt;- matrix[, selected_columns]

subset_meta_0.1 &lt;- subset(Metadata, Metadata$Conc. == 0.1)

prcomp_res_0.1 &lt;- prcomp(t(subset_matrix_0.1[rowMeans(subset_matrix_0.1) &gt; 0, ]), center = TRUE)

ggplot2::autoplot(prcomp_res_0.1, data = as.data.frame(subset_meta_0.1), colour = &quot;Drug&quot;, shape = &quot;Time&quot;, size = 4) +
  ggrepel::geom_text_repel(label = subset_meta_0.1$Ind) +
  scale_color_manual(values = drug_palc) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 0 (0.1 µM)&quot;)) +
  theme_bw()</code></pre>
<p><img src="figure/PCA.Rmd/pca_0.1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_0.1-1">
Past versions of pca_0.1-1.png
</button>
</p>
<div id="fig-pca_0.1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_0.1-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/PCA.Rmd/pca_0.1-1.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_0.1-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_0.1-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res_0.1 &lt;- prcomp(t(subset_matrix_0.1[rowMeans(subset_matrix_0.1) &gt; 0, ]), center = TRUE)

ggplot2::autoplot(prcomp_res_0.1, data = as.data.frame(subset_meta_0.1), colour = &quot;Drug&quot;, shape = &quot;Time&quot;, size = 4, x=2, y=3) +
  ggrepel::geom_text_repel(label = subset_meta_0.1$Ind) +
  scale_color_manual(values = drug_palc) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 0 (0.1 µM)&quot;)) +
  theme_bw()</code></pre>
<p><img src="figure/PCA.Rmd/pca_0.1_1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_0.1_1-1">
Past versions of pca_0.1_1-1.png
</button>
</p>
<div id="fig-pca_0.1_1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_0.1_1-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/PCA.Rmd/pca_0.1_1-1.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_0.1_1-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_0.1_1-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res_0.1 &lt;- prcomp(t(subset_matrix_0.1[rowMeans(subset_matrix_0.1) &gt; 0, ]), center = TRUE)

ggplot2::autoplot(prcomp_res_0.1, data = as.data.frame(subset_meta_0.1), colour = &quot;Drug&quot;, shape = &quot;Time&quot;, size = 4, x=3, y=4) +
  ggrepel::geom_text_repel(label = subset_meta_0.1$Ind) +
  scale_color_manual(values = drug_palc) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 0 (0.1 µM)&quot;)) +
  theme_bw()</code></pre>
<p><img src="figure/PCA.Rmd/pca_0.1_2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_0.1_2-1">
Past versions of pca_0.1_2-1.png
</button>
</p>
<div id="fig-pca_0.1_2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_0.1_2-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/PCA.Rmd/pca_0.1_2-1.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_0.1_2-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_0.1_2-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="µm-concentration-1" class="section level2">
<h2><strong>📌0.5 µM Concentration</strong></h2>
<pre class="r"><code>selected_columns &lt;- grepl(&quot;_0.5_&quot;, colnames(matrix))
subset_matrix_0.5 &lt;- matrix[, selected_columns]

subset_meta_0.5 &lt;- subset(Metadata, Metadata$Conc. == 0.5)

prcomp_res_0.5 &lt;- prcomp(t(subset_matrix_0.5[rowMeans(subset_matrix_0.5) &gt; 0, ]), center = TRUE)

ggplot2::autoplot(prcomp_res_0.5, data = as.data.frame(subset_meta_0.5), colour = &quot;Drug&quot;, shape = &quot;Time&quot;, size = 4) +
  ggrepel::geom_text_repel(label = subset_meta_0.5$Ind) +
  scale_color_manual(values = drug_palc) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 0 (0.5 µM)&quot;)) +
  theme_bw()</code></pre>
<pre><code>Warning: ggrepel: 11 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/PCA.Rmd/pca_0.5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_0.5-1">
Past versions of pca_0.5-1.png
</button>
</p>
<div id="fig-pca_0.5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_0.5-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/PCA.Rmd/pca_0.5-1.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_0.5-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_0.5-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res_0.5 &lt;- prcomp(t(subset_matrix_0.5[rowMeans(subset_matrix_0.5) &gt; 0, ]), center = TRUE)

ggplot2::autoplot(prcomp_res_0.5, data = as.data.frame(subset_meta_0.5), colour = &quot;Drug&quot;, shape = &quot;Time&quot;, size = 4, x=2, y=3) +
  ggrepel::geom_text_repel(label = subset_meta_0.5$Ind) +
  scale_color_manual(values = drug_palc) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 0 (0.5 µM)&quot;)) +
  theme_bw()</code></pre>
<p><img src="figure/PCA.Rmd/pca_0.5_1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_0.5_1-1">
Past versions of pca_0.5_1-1.png
</button>
</p>
<div id="fig-pca_0.5_1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_0.5_1-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/PCA.Rmd/pca_0.5_1-1.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_0.5_1-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_0.5_1-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prcomp_res_0.5 &lt;- prcomp(t(subset_matrix_0.5[rowMeans(subset_matrix_0.5) &gt; 0, ]), center = TRUE)

ggplot2::autoplot(prcomp_res_0.5, data = as.data.frame(subset_meta_0.5), colour = &quot;Drug&quot;, shape = &quot;Time&quot;, size = 4, x=3, y=4) +
  ggrepel::geom_text_repel(label = subset_meta_0.5$Ind) +
  scale_color_manual(values = drug_palc) +
  ggtitle(expression(&quot;PCA of log&quot;[2]*&quot;(cpm) Filtered rowMeans &gt; 0 (0.5 µM)&quot;)) +
  theme_bw()</code></pre>
<p><img src="figure/PCA.Rmd/pca_0.5_2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_0.5_2-1">
Past versions of pca_0.5_2-1.png
</button>
</p>
<div id="fig-pca_0.5_2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/edfc7e151f7cd8075206029a0f81739fc1c00305/docs/figure/PCA.Rmd/pca_0.5_2-1.png" target="_blank">edfc7e1</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-05-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/PCA.Rmd/pca_0.5_2-1.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/0e53214d0a3f61e5f67eaa087ebc4b48c2451b35/docs/figure/PCA.Rmd/pca_0.5_2-1.png" target="_blank">0e53214</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/3f3d8c0195de9d12de0d0f5dc0692c19093a8a86/docs/figure/PCA.Rmd/pca_0.5_2-1.png" target="_blank">3f3d8c0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 26100)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggfortify_0.4.17     org.Hs.eg.db_3.18.0  AnnotationDbi_1.64.1
 [4] IRanges_2.36.0       S4Vectors_0.40.2     Hmisc_5.2-3         
 [7] corrplot_0.95        ggrepel_0.9.6        biomaRt_2.58.2      
[10] scales_1.3.0         lubridate_1.9.4      forcats_1.0.0       
[13] stringr_1.5.1        purrr_1.0.4          readr_2.1.5         
[16] tidyr_1.3.1          tibble_3.2.1         tidyverse_2.0.0     
[19] Biobase_2.62.0       BiocGenerics_0.48.1  dplyr_1.1.4         
[22] reshape2_1.4.4       ggplot2_3.5.2        edgeR_4.0.16        
[25] limma_3.58.1         workflowr_1.7.1     

loaded via a namespace (and not attached):
 [1] DBI_1.2.3               bitops_1.0-9            gridExtra_2.3          
 [4] rlang_1.1.3             magrittr_2.0.3          git2r_0.36.2           
 [7] compiler_4.3.0          RSQLite_2.3.9           getPass_0.2-4          
[10] png_0.1-8               callr_3.7.6             vctrs_0.6.5            
[13] pkgconfig_2.0.3         crayon_1.5.3            fastmap_1.2.0          
[16] backports_1.5.0         dbplyr_2.5.0            XVector_0.42.0         
[19] labeling_0.4.3          promises_1.3.2          rmarkdown_2.29         
[22] tzdb_0.5.0              ps_1.8.1                bit_4.6.0              
[25] xfun_0.52               zlibbioc_1.48.2         cachem_1.1.0           
[28] GenomeInfoDb_1.38.8     jsonlite_2.0.0          progress_1.2.3         
[31] blob_1.2.4              later_1.3.2             prettyunits_1.2.0      
[34] cluster_2.1.8.1         R6_2.6.1                bslib_0.9.0            
[37] stringi_1.8.3           rpart_4.1.24            jquerylib_0.1.4        
[40] Rcpp_1.0.12             knitr_1.50              base64enc_0.1-3        
[43] httpuv_1.6.15           nnet_7.3-20             timechange_0.3.0       
[46] tidyselect_1.2.1        rstudioapi_0.17.1       yaml_2.3.10            
[49] curl_6.2.2              processx_3.8.6          lattice_0.22-7         
[52] plyr_1.8.9              withr_3.0.2             KEGGREST_1.42.0        
[55] evaluate_1.0.3          foreign_0.8-90          BiocFileCache_2.10.2   
[58] xml2_1.3.8              Biostrings_2.70.3       pillar_1.10.2          
[61] filelock_1.0.3          whisker_0.4.1           checkmate_2.3.2        
[64] generics_0.1.3          rprojroot_2.0.4         RCurl_1.98-1.17        
[67] hms_1.1.3               munsell_0.5.1           glue_1.7.0             
[70] tools_4.3.0             data.table_1.17.0       locfit_1.5-9.12        
[73] fs_1.6.3                XML_3.99-0.18           grid_4.3.0             
[76] colorspace_2.1-0        GenomeInfoDbData_1.2.11 htmlTable_2.4.3        
[79] Formula_1.2-5           cli_3.6.1               rappdirs_0.3.3         
[82] gtable_0.3.6            sass_0.4.10             digest_0.6.34          
[85] farver_2.1.2            htmlwidgets_1.6.4       memoise_2.0.1          
[88] htmltools_0.5.8.1       lifecycle_1.0.4         httr_1.4.7             
[91] statmod_1.5.0           bit64_4.6.0-1          </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
