<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Tissue</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CX5461_Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Tissue</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-04-30
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>CX5461_Project/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree6f9e594c722182678bafc5ddb7c9cdeb3f27dfbftargetblank6f9e594a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/6f9e594c722182678bafc5ddb7c9cdeb3f27dfbf" target="_blank">6f9e594</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree6f9e594c722182678bafc5ddb7c9cdeb3f27dfbftargetblank6f9e594a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/6f9e594c722182678bafc5ddb7c9cdeb3f27dfbf" target="_blank">6f9e594</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    0.1 box.svg
    Ignored:    Rplot04.svg

Untracked files:
    Untracked:  0.1 density.svg
    Untracked:  0.1.emf
    Untracked:  0.1.svg
    Untracked:  0.5 box.svg
    Untracked:  0.5 density.svg
    Untracked:  0.5.svg
    Untracked:  CX_5461_Pattern_Genes_24hr.csv
    Untracked:  CX_5461_Pattern_Genes_3hr.csv
    Untracked:  Cell viability box plot.svg
    Untracked:  DRC1.svg
    Untracked:  Figure 1.jpeg
    Untracked:  Figure 1.pdf
    Untracked:  Figure_CM_Purity.pdf
    Untracked:  Rplot.svg
    Untracked:  Rplot01.svg
    Untracked:  Rplot02.svg
    Untracked:  Rplot03.svg
    Untracked:  Rplot05.svg
    Untracked:  Rplot06.svg
    Untracked:  Rplot07.svg
    Untracked:  Rplot08.jpeg
    Untracked:  Rplot08.svg
    Untracked:  Rplot09.svg
    Untracked:  Rplot10.svg
    Untracked:  Rplot11.svg
    Untracked:  TS HPA (Violin).svg
    Untracked:  TS HPA.svg
    Untracked:  TS_HA.svg
    Untracked:  TS_HV.svg
    Untracked:  Violin HA.svg
    Untracked:  Violin HV (CX vs DOX).svg
    Untracked:  Violin HV.svg
    Untracked:  data/AF.csv
    Untracked:  data/AF_Mapped.csv
    Untracked:  data/AF_genes.csv
    Untracked:  data/CAD_genes.csv
    Untracked:  data/Cardiotox.csv
    Untracked:  data/DDR.csv
    Untracked:  data/DDR_Mapped.csv
    Untracked:  data/DOX_Vald.csv
    Untracked:  data/DOX_Vald_Mapped.csv
    Untracked:  data/Entrez_Cardiotox.csv
    Untracked:  data/Entrez_Cardiotox_Mapped.csv
    Untracked:  data/GWAS.xlsx
    Untracked:  data/GWAS_SNPs.bed
    Untracked:  data/HF.csv
    Untracked:  data/HF_Mapped.csv
    Untracked:  data/HF_genes.csv
    Untracked:  data/Hypertension_genes.csv
    Untracked:  data/MI_genes.csv
    Untracked:  data/TS.csv
    Untracked:  data/TS_HPA.csv
    Untracked:  data/TS_HPA_mapped.csv
    Untracked:  data/Toptable_CX_0.1_24.csv
    Untracked:  data/Toptable_CX_0.1_3.csv
    Untracked:  data/Toptable_CX_0.1_48.csv
    Untracked:  data/Toptable_CX_0.5_24.csv
    Untracked:  data/Toptable_CX_0.5_3.csv
    Untracked:  data/Toptable_CX_0.5_48.csv
    Untracked:  data/Toptable_DOX_0.1_24.csv
    Untracked:  data/Toptable_DOX_0.1_3.csv
    Untracked:  data/Toptable_DOX_0.1_48.csv
    Untracked:  data/Toptable_DOX_0.5_24.csv
    Untracked:  data/Toptable_DOX_0.5_3.csv
    Untracked:  data/Toptable_DOX_0.5_48.csv
    Untracked:  data/ts_data_mapped
    Untracked:  results/
    Untracked:  run_bedtools.bat

Unstaged changes:
    Deleted:    analysis/Actox.Rmd
    Modified:   analysis/Tissue.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Tissue.Rmd</code>) and HTML
(<code>docs/Tissue.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/02edb70eb41c6124c5323344461f1e8cca22d7dd/analysis/Tissue.Rmd" target="_blank">02edb70</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/02edb70eb41c6124c5323344461f1e8cca22d7dd/docs/Tissue.html" target="_blank">02edb70</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/analysis/Tissue.Rmd" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/Tissue.html" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/1c1e1e44938af2a636ee23b538d65aa1398a443f/analysis/Tissue.Rmd" target="_blank">1c1e1e4</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-28
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/cb007742d4f6d631433a383831b038bee876f5df/analysis/Tissue.Rmd" target="_blank">cb00774</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-28
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/cb007742d4f6d631433a383831b038bee876f5df/docs/Tissue.html" target="_blank">cb00774</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-28
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/fa752956d8eacadb49c22072788db50fd9058d01/analysis/Tissue.Rmd" target="_blank">fa75295</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-28
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/fa752956d8eacadb49c22072788db50fd9058d01/docs/Tissue.html" target="_blank">fa75295</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-28
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/cf2940840cc495cff92a381a19b0eb2816d2af74/analysis/Tissue.Rmd" target="_blank">cf29408</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-28
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/cf2940840cc495cff92a381a19b0eb2816d2af74/docs/Tissue.html" target="_blank">cf29408</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-28
</td>
<td>
Commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="tissue-specificity-analysis-correlation-heatmap"
class="section level2">
<h2><strong>📌 Tissue specificity analysis (Correlation
Heatmap)</strong></h2>
</div>
<div id="load-required-libraries" class="section level2">
<h2><strong>📌 Load Required Libraries</strong></h2>
<pre class="r"><code>library(ggplot2)
library(dplyr)</code></pre>
<pre><code>Warning: package &#39;dplyr&#39; was built under R version 4.3.2</code></pre>
<pre class="r"><code>library(tidyr)</code></pre>
<pre><code>Warning: package &#39;tidyr&#39; was built under R version 4.3.3</code></pre>
<pre class="r"><code>library(org.Hs.eg.db)</code></pre>
<pre><code>Warning: package &#39;AnnotationDbi&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;BiocGenerics&#39; was built under R version 4.3.1</code></pre>
<pre><code>Warning: package &#39;Biobase&#39; was built under R version 4.3.1</code></pre>
<pre><code>Warning: package &#39;IRanges&#39; was built under R version 4.3.1</code></pre>
<pre><code>Warning: package &#39;S4Vectors&#39; was built under R version 4.3.2</code></pre>
<pre class="r"><code>library(clusterProfiler)</code></pre>
<pre><code>Warning: package &#39;clusterProfiler&#39; was built under R version 4.3.3</code></pre>
<pre class="r"><code>library(biomaRt)</code></pre>
<pre><code>Warning: package &#39;biomaRt&#39; was built under R version 4.3.2</code></pre>
<pre class="r"><code>library(pheatmap)</code></pre>
<pre><code>Warning: package &#39;pheatmap&#39; was built under R version 4.3.1</code></pre>
</div>
<div id="load-data" class="section level2">
<h2><strong>📌 Load Data</strong></h2>
<pre class="r"><code># Read the CSV file into R
file_path &lt;- &quot;data/count.csv&quot;
df &lt;- read.csv(file_path, check.names = FALSE)

# Remove &#39;x&#39; from column headers (if applicable)
colnames(df) &lt;- gsub(&quot;^x&quot;, &quot;&quot;, colnames(df))

# Step 1: Calculate IPSC_CM (Median across VEH samples)
# Select columns with &#39;VEH&#39; in their names
veh_columns &lt;- grep(&quot;VEH&quot;, colnames(df), value = TRUE)

# Calculate the median logCPM across VEH samples
df$IPSC_CM &lt;- apply(df[, veh_columns], 1, median, na.rm = TRUE)

# Create a new dataframe with Entrez_ID and IPSC_CM
veh_median_df &lt;- df[, c(&quot;Entrez_ID&quot;, &quot;IPSC_CM&quot;)]

# Step 2: Read the Tissue_Gtex dataset
Tissue_Gtex &lt;- read.csv(&quot;data/Tissue_Gtex.csv&quot;)

# Step 3: Convert Ensembl IDs to Entrez IDs using biomaRt
mart &lt;- useMart(&quot;ensembl&quot;, dataset = &quot;hsapiens_gene_ensembl&quot;)

gene_ids &lt;- Tissue_Gtex$gene.id
conversion &lt;- getBM(
  attributes = c(&quot;ensembl_gene_id&quot;, &quot;entrezgene_id&quot;),
  filters = &quot;ensembl_gene_id&quot;,
  values = gene_ids,
  mart = mart
)

# Merge Entrez IDs back into Tissue_Gtex
Tissue_Gtex &lt;- merge(Tissue_Gtex, conversion, by.x = &quot;gene.id&quot;, by.y = &quot;ensembl_gene_id&quot;, all.x = TRUE)

# Rename for consistency
colnames(Tissue_Gtex)[colnames(Tissue_Gtex) == &quot;entrezgene_id&quot;] &lt;- &quot;Entrez_ID&quot;

# Step 4: Ensure Entrez_ID column types match before merging
veh_median_df$Entrez_ID &lt;- as.character(veh_median_df$Entrez_ID)
Tissue_Gtex$Entrez_ID &lt;- as.character(Tissue_Gtex$Entrez_ID)

# Step 5: Merge iPSC_CM with Tissue_Gtex by Entrez_ID
merged_df &lt;- merge(veh_median_df, Tissue_Gtex, by = &quot;Entrez_ID&quot;, all.x = TRUE)

# Step 6: Remove rows with NA values
cleaned_df &lt;- na.omit(merged_df) </code></pre>
</div>
<div id="correlation-heatmap-tissue-specificity" class="section level2">
<h2><strong>📌 Correlation heatmap (Tissue specificity)</strong></h2>
<pre class="r"><code># Filter relevant tissue columns
tissue_cols &lt;- colnames(cleaned_df)[which(colnames(cleaned_df) %in% c(
  &quot;IPSC_CM&quot;, &quot;Adrenal.Gland&quot;, &quot;Spleen&quot;, &quot;Heart...Atrial&quot;, &quot;Pancreas&quot;, &quot;Artery&quot;, &quot;Breast&quot;,
  &quot;small.Intestine&quot;, &quot;Colon&quot;, &quot;Nerve...Tibial&quot;, &quot;Esophagus&quot;, &quot;Muscle...Skeletal&quot;,
  &quot;Thyroid&quot;, &quot;Heart..Ventricle&quot;, &quot;Stomach&quot;, &quot;Uterus&quot;, &quot;Vagina&quot;, &quot;Skin&quot;, &quot;Ovary&quot;, &quot;Liver&quot;, 
  &quot;Lung&quot;, &quot;Brain&quot;, &quot;Pituitary&quot;, &quot;Testis&quot;, &quot;Prostate&quot;, &quot;Salivary.Gland&quot;))]

data_subset &lt;- cleaned_df[, tissue_cols]

# Compute Pearson and Spearman correlations
pearson_corr &lt;- cor(data_subset, method = &quot;pearson&quot;, use = &quot;complete.obs&quot;)
spearman_corr &lt;- cor(data_subset, method = &quot;spearman&quot;, use = &quot;complete.obs&quot;)

# Reorder tissues by highest correlation with IPSC_CM
order_pearson &lt;- order(pearson_corr[&quot;IPSC_CM&quot;, ], decreasing = TRUE)
order_spearman &lt;- order(spearman_corr[&quot;IPSC_CM&quot;, ], decreasing = TRUE)

pearson_corr &lt;- pearson_corr[order_pearson, order_pearson]
spearman_corr &lt;- spearman_corr[order_spearman, order_spearman]

# Plot Pearson correlation heatmap
pheatmap(pearson_corr, 
         cluster_rows = TRUE,
         cluster_cols = TRUE,
         main = &quot;Pearson Correlation Heatmap&quot;, 
         color = colorRampPalette(c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;))(100), 
         display_numbers = TRUE,
         number_format = &quot;%.2f&quot;,
         fontsize_number = 8)</code></pre>
<p><img src="figure/Tissue.Rmd/Genes%20of%20Interest-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Genes-of-Interest-1">
Past versions of "Genes of Interest-1.png"
</button>
</p>
<div id="fig-no-spaces-Genes-of-Interest-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/Tissue.Rmd/Genes of Interest-1.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/cf2940840cc495cff92a381a19b0eb2816d2af74/docs/figure/Tissue.Rmd/Genes of Interest-1.png" target="_blank">cf29408</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Optional: Plot Spearman correlation heatmap
pheatmap(spearman_corr, 
         cluster_rows = TRUE,
         cluster_cols = TRUE,
         main = &quot;Spearman Correlation Heatmap&quot;, 
         color = colorRampPalette(c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;))(100), 
         display_numbers = TRUE,
         number_format = &quot;%.2f&quot;,
         fontsize_number = 8)</code></pre>
<p><img src="figure/Tissue.Rmd/Genes%20of%20Interest-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Genes-of-Interest-2">
Past versions of "Genes of Interest-2.png"
</button>
</p>
<div id="fig-no-spaces-Genes-of-Interest-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/Tissue.Rmd/Genes of Interest-2.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/cf2940840cc495cff92a381a19b0eb2816d2af74/docs/figure/Tissue.Rmd/Genes of Interest-2.png" target="_blank">cf29408</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="correlation-plot-tissue-specificity" class="section level2">
<h2><strong>📌 Correlation Plot (Tissue specificity)</strong></h2>
<pre class="r"><code># Load libraries
library(ComplexHeatmap)</code></pre>
<pre><code>Warning: package &#39;ComplexHeatmap&#39; was built under R version 4.3.1</code></pre>
<pre class="r"><code>library(circlize)</code></pre>
<pre><code>Warning: package &#39;circlize&#39; was built under R version 4.3.3</code></pre>
<pre class="r"><code>library(grid)

# Compute correlation (Pearson)
cor_values &lt;- cor(data_subset, method = &quot;pearson&quot;, use = &quot;complete.obs&quot;)

# Extract and sort correlations with median-based IPSC_CM
ipsc_cm_corr &lt;- cor_values[&quot;IPSC_CM&quot;, ]
ipsc_cm_corr_sorted &lt;- sort(ipsc_cm_corr, decreasing = TRUE)

# Create matrix for heatmap
corr_matrix &lt;- matrix(ipsc_cm_corr_sorted, ncol = 1)
rownames(corr_matrix) &lt;- names(ipsc_cm_corr_sorted)
colnames(corr_matrix) &lt;- &quot;IPSC_CM&quot;

# Define color function: blue → white → red
col_fun &lt;- colorRamp2(
  c(0.5, 0.75, 1.0),
  c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;)
)

# Plot heatmap
Heatmap(
  corr_matrix,
  name = &quot;Corr.&quot;,
  col = col_fun,
  cluster_rows = FALSE,
  cluster_columns = FALSE,
  show_column_names = TRUE,
  show_row_names = TRUE,
  row_names_side = &quot;right&quot;,
  column_names_side = &quot;bottom&quot;,
  column_title = &quot;Pearson Correlation with IPSC_CM (Median)&quot;,
  column_title_gp = gpar(fontsize = 14, fontface = &quot;bold&quot;),
  heatmap_width = unit(5, &quot;cm&quot;),
  heatmap_height = unit(12, &quot;cm&quot;),
  cell_fun = function(j, i, x, y, width, height, fill) {
    grid.text(sprintf(&quot;%.2f&quot;, corr_matrix[i, j]), x, y, gp = gpar(fontsize = 9))
  }
)</code></pre>
<p><img src="figure/Tissue.Rmd/Genes%20of%20Interest1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Genes-of-Interest1-1">
Past versions of "Genes of Interest1-1.png"
</button>
</p>
<div id="fig-no-spaces-Genes-of-Interest1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/02edb70eb41c6124c5323344461f1e8cca22d7dd/docs/figure/Tissue.Rmd/Genes of Interest1-1.png" target="_blank">02edb70</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="tissue-specific-gene-proportions-in-corrmotif-clusters"
class="section level2">
<h2><strong>📌 Tissue specific gene proportions in corrmotif
clusters</strong></h2>
</div>
<div id="load-datasets" class="section level2">
<h2><strong>📌 Load Datasets</strong></h2>
<pre class="r"><code># Load your Heart_genes dataset
heart_genes &lt;- read.csv(&quot;data/Human_Heart_Genes.csv&quot;, stringsAsFactors = FALSE)

# Convert Gene names to Entrez IDs in heart_genes
heart_genes$Entrez_ID &lt;- mapIds(
  org.Hs.eg.db, 
  keys = heart_genes$Gene, 
  column = &quot;ENTREZID&quot;, 
  keytype = &quot;SYMBOL&quot;, 
  multiVals = &quot;first&quot;
)

# Create a vector of Entrez IDs specific to the heart
heart_entrez_ids &lt;- na.omit(heart_genes$Entrez_ID)


# Load the saved datasets
prob_all_1 &lt;- read.csv(&quot;data/prob_all_1.csv&quot;)$Entrez_ID
prob_all_2 &lt;- read.csv(&quot;data/prob_all_2.csv&quot;)$Entrez_ID
prob_all_3 &lt;- read.csv(&quot;data/prob_all_3.csv&quot;)$Entrez_ID
prob_all_4 &lt;- read.csv(&quot;data/prob_all_4.csv&quot;)$Entrez_ID</code></pre>
</div>
<div id="tissue-specific-gene-proportions-analysis"
class="section level2">
<h2><strong>📌 Tissue specific gene proportions analysis</strong></h2>
<pre class="r"><code># Example Response Groups Data (Replace with actual data)
response_groups &lt;- list(
  &quot;Non Response&quot; = prob_all_1, # Replace &#39;prob_all_1&#39;, &#39;prob_all_2&#39;, etc. with your actual response group dataframes
  &quot;CX_DOX Shared Late Response&quot; = prob_all_2,
  &quot;DOX-Specific Response&quot; = prob_all_3,
  &quot;Late High Dose DOX-Specific Response&quot; = prob_all_4
)

# Combine all response groups into a single dataframe
response_groups_df &lt;- bind_rows(
  lapply(response_groups, function(ids) data.frame(Entrez_ID = ids)),
  .id = &quot;Set&quot;
)

# Categorize genes into Heart-specific and Non-Heart-specific
response_groups_df &lt;- response_groups_df %&gt;%
  mutate(
    Category = ifelse(Entrez_ID %in% heart_entrez_ids, &quot;Heart-specific Genes&quot;, &quot;Non-Heart-specific Genes&quot;)
  )

# Calculate counts for Heart-specific and Non-Heart-specific genes in each response group
proportion_data &lt;- response_groups_df %&gt;%
  group_by(Set, Category) %&gt;%
  summarize(Count = n(), .groups = &quot;drop&quot;)

# Reference counts for &quot;Non Response&quot;
non_response_counts &lt;- proportion_data %&gt;%
  filter(Set == &quot;Non Response&quot;) %&gt;%
  dplyr::select(Category, Count) %&gt;%
  {setNames(.$Count, .$Category)} # Create named vector for &quot;Non Response&quot; counts

# Perform Chi-square test for selected response groups compared to &quot;Non Response&quot;
chi_results &lt;- proportion_data %&gt;%
  filter(Set != &quot;Non Response&quot;) %&gt;% # Exclude &quot;Non Response&quot;
  group_by(Set) %&gt;%
  summarize(
    p_value = {
      # Extract counts for the current response group
      group_counts &lt;- Count[Category %in% c(&quot;Heart-specific Genes&quot;, &quot;Non-Heart-specific Genes&quot;)]
      # Ensure there are no missing categories, fill with 0 if missing
      if (length(group_counts) &lt; 2) group_counts &lt;- c(group_counts, 0)
      # Create contingency table
      contingency_table &lt;- matrix(c(
        group_counts[1], group_counts[2],
        non_response_counts[&quot;Heart-specific Genes&quot;], non_response_counts[&quot;Non-Heart-specific Genes&quot;]
      ), nrow = 2)
      # Print the contingency table for debugging
      print(paste(&quot;Set:&quot;, unique(Set)))
      print(&quot;Contingency Table:&quot;)
      print(contingency_table)
      # Perform chi-square test
      if (all(contingency_table &gt;= 0 &amp; is.finite(contingency_table))) {
        chisq.test(contingency_table)$p.value
      } else {
        NA
      }
    },
    .groups = &quot;drop&quot;
  ) %&gt;%
  mutate(Significance = ifelse(!is.na(p_value) &amp; p_value &lt; 0.05, &quot;*&quot;, &quot;&quot;))</code></pre>
<pre><code>[1] &quot;Set: CX_DOX Shared Late Response&quot;
[1] &quot;Contingency Table:&quot;
     [,1] [,2]
[1,]    2  123
[2,]  412 6908
[1] &quot;Set: DOX-Specific Response&quot;
[1] &quot;Contingency Table:&quot;
     [,1] [,2]
[1,]   62  123
[2,] 1469 6908
[1] &quot;Set: Late High Dose DOX-Specific Response&quot;
[1] &quot;Contingency Table:&quot;
     [,1] [,2]
[1,]  159  123
[2,] 4715 6908</code></pre>
<pre class="r"><code># Merge chi-square results back into the proportion data
proportion_data &lt;- proportion_data %&gt;%
  left_join(chi_results %&gt;% dplyr::select(Set, p_value, Significance), by = &quot;Set&quot;)

# Calculate proportions for plotting
proportion_data &lt;- proportion_data %&gt;%
  group_by(Set) %&gt;%
  mutate(Percentage = (Count / sum(Count)) * 100)

# Define the correct order for response groups
response_order &lt;- c(
  &quot;Non Response&quot;,
  &quot;CX_DOX Shared Late Response&quot;,
  &quot;DOX-Specific Response&quot;,
  &quot;Late High Dose DOX-Specific Response&quot;
)
proportion_data$Set &lt;- factor(proportion_data$Set, levels = response_order)

# Create the proportion plot with significance stars
ggplot(proportion_data, aes(x = Set, y = Percentage, fill = Category)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  geom_text(
    data = proportion_data %&gt;% distinct(Set, Significance), # Add stars for significant groups
    aes(x = Set, y = 105, label = Significance), # Position stars above the bars
    inherit.aes = FALSE,
    size = 6,
    color = &quot;black&quot;,
    hjust = 0.5
  ) +
  scale_fill_manual(values = c(
    &quot;Heart-specific Genes&quot; = &quot;#4daf4a&quot;,  # Green
    &quot;Non-Heart-specific Genes&quot; = &quot;#377eb8&quot;   # Blue
  )) +
  labs(
    title = &quot;Proportion of Heart and Non-Heart-Specific Genes Across Response Sets&quot;,
    x = &quot;Response Sets&quot;,
    y = &quot;Percentage of Genes&quot;,
    fill = &quot;Gene Category&quot;
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;),
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1.5) # Add border
  )</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_text()`).</code></pre>
<p><img src="figure/Tissue.Rmd/Prop-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Prop-1">
Past versions of Prop-1.png
</button>
</p>
<div id="fig-Prop-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/ffaf948eb32474671f8ad4c457bdd152ee6476de/docs/figure/Tissue.Rmd/Prop-1.png" target="_blank">ffaf948</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/cb007742d4f6d631433a383831b038bee876f5df/docs/figure/Tissue.Rmd/Prop-1.png" target="_blank">cb00774</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/fa752956d8eacadb49c22072788db50fd9058d01/docs/figure/Tissue.Rmd/Prop-1.png" target="_blank">fa75295</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 26100)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] circlize_0.4.16        ComplexHeatmap_2.18.0  pheatmap_1.0.12       
 [4] biomaRt_2.58.2         clusterProfiler_4.10.1 org.Hs.eg.db_3.18.0   
 [7] AnnotationDbi_1.64.1   IRanges_2.36.0         S4Vectors_0.40.2      
[10] Biobase_2.62.0         BiocGenerics_0.48.1    tidyr_1.3.1           
[13] dplyr_1.1.4            ggplot2_3.5.2         

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3      shape_1.4.6.1           rstudioapi_0.17.1      
  [4] jsonlite_2.0.0          magrittr_2.0.3          magick_2.8.6           
  [7] farver_2.1.2            rmarkdown_2.29          GlobalOptions_0.1.2    
 [10] fs_1.6.3                zlibbioc_1.48.2         vctrs_0.6.5            
 [13] Cairo_1.6-2             memoise_2.0.1           RCurl_1.98-1.17        
 [16] ggtree_3.10.1           htmltools_0.5.8.1       progress_1.2.3         
 [19] curl_6.2.2              gridGraphics_0.5-1      sass_0.4.10            
 [22] bslib_0.9.0             plyr_1.8.9              cachem_1.1.0           
 [25] whisker_0.4.1           igraph_2.1.4            iterators_1.0.14       
 [28] lifecycle_1.0.4         pkgconfig_2.0.3         Matrix_1.6-1.1         
 [31] R6_2.6.1                fastmap_1.2.0           gson_0.1.0             
 [34] clue_0.3-66             GenomeInfoDbData_1.2.11 digest_0.6.34          
 [37] aplot_0.2.5             enrichplot_1.22.0       colorspace_2.1-0       
 [40] patchwork_1.3.0         rprojroot_2.0.4         RSQLite_2.3.9          
 [43] labeling_0.4.3          filelock_1.0.3          httr_1.4.7             
 [46] polyclip_1.10-7         compiler_4.3.0          doParallel_1.0.17      
 [49] bit64_4.6.0-1           withr_3.0.2             BiocParallel_1.36.0    
 [52] viridis_0.6.5           DBI_1.2.3               ggforce_0.4.2          
 [55] MASS_7.3-60             rappdirs_0.3.3          rjson_0.2.23           
 [58] HDO.db_0.99.1           tools_4.3.0             ape_5.8-1              
 [61] scatterpie_0.2.4        httpuv_1.6.15           glue_1.7.0             
 [64] nlme_3.1-168            GOSemSim_2.28.1         promises_1.3.2         
 [67] shadowtext_0.1.4        cluster_2.1.8.1         reshape2_1.4.4         
 [70] fgsea_1.28.0            generics_0.1.3          gtable_0.3.6           
 [73] data.table_1.17.0       hms_1.1.3               xml2_1.3.8             
 [76] tidygraph_1.3.1         XVector_0.42.0          foreach_1.5.2          
 [79] ggrepel_0.9.6           pillar_1.10.2           stringr_1.5.1          
 [82] yulab.utils_0.2.0       later_1.3.2             splines_4.3.0          
 [85] tweenr_2.0.3            BiocFileCache_2.10.2    treeio_1.26.0          
 [88] lattice_0.22-7          bit_4.6.0               tidyselect_1.2.1       
 [91] GO.db_3.18.0            Biostrings_2.70.3       knitr_1.50             
 [94] git2r_0.36.2            gridExtra_2.3           xfun_0.52              
 [97] graphlayouts_1.2.2      matrixStats_1.5.0       stringi_1.8.3          
[100] workflowr_1.7.1         lazyeval_0.2.2          ggfun_0.1.8            
[103] yaml_2.3.10             evaluate_1.0.3          codetools_0.2-20       
[106] ggraph_2.2.1            tibble_3.2.1            qvalue_2.34.0          
[109] ggplotify_0.1.2         cli_3.6.1               munsell_0.5.1          
[112] jquerylib_0.1.4         Rcpp_1.0.12             GenomeInfoDb_1.38.8    
[115] dbplyr_2.5.0            png_0.1-8               XML_3.99-0.18          
[118] parallel_4.3.0          blob_1.2.4              prettyunits_1.2.0      
[121] DOSE_3.28.2             bitops_1.0-9            viridisLite_0.4.2      
[124] tidytree_0.4.6          scales_1.3.0            purrr_1.0.4            
[127] crayon_1.5.3            GetoptLong_1.0.5        rlang_1.1.3            
[130] cowplot_1.1.3           fastmatch_1.1-6         KEGGREST_1.42.0        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
