---
title: "Corrmotif Conc Analysis"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---
## **ðŸ“Œ 0.1 Micromolar**

## **ðŸ“Œ Fit Limma Model Functions**
```{r fit_limma_model_functions, echo=TRUE, results='hide',message=FALSE, warning=FALSE}
## Fit limma model using code as it is found in the original cormotif code. It has
## only been modified to add names to the matrix of t values, as well as the
## limma fits

limmafit.default <- function(exprs,groupid,compid) {
  limmafits  <- list()
  compnum    <- nrow(compid)
  genenum    <- nrow(exprs)
  limmat     <- matrix(0,genenum,compnum)
  limmas2    <- rep(0,compnum)
  limmadf    <- rep(0,compnum)
  limmav0    <- rep(0,compnum)
  limmag1num <- rep(0,compnum)
  limmag2num <- rep(0,compnum)

  rownames(limmat)  <- rownames(exprs)
  colnames(limmat)  <- rownames(compid)
  names(limmas2)    <- rownames(compid)
  names(limmadf)    <- rownames(compid)
  names(limmav0)    <- rownames(compid)
  names(limmag1num) <- rownames(compid)
  names(limmag2num) <- rownames(compid)

  for(i in 1:compnum) {
    selid1 <- which(groupid == compid[i,1])
    selid2 <- which(groupid == compid[i,2])
    eset   <- new("ExpressionSet", exprs=cbind(exprs[,selid1],exprs[,selid2]))
    g1num  <- length(selid1)
    g2num  <- length(selid2)
    designmat <- cbind(base=rep(1,(g1num+g2num)), delta=c(rep(0,g1num),rep(1,g2num)))
    fit <- lmFit(eset,designmat)
    fit <- eBayes(fit)
    limmat[,i] <- fit$t[,2]
    limmas2[i] <- fit$s2.prior
    limmadf[i] <- fit$df.prior
    limmav0[i] <- fit$var.prior[2]
    limmag1num[i] <- g1num
    limmag2num[i] <- g2num
    limmafits[[i]] <- fit

    # log odds
    # w<-sqrt(1+fit$var.prior[2]/(1/g1num+1/g2num))
    # log(0.99)+dt(fit$t[1,2],g1num+g2num-2+fit$df.prior,log=TRUE)-log(0.01)-dt(fit$t[1,2]/w, g1num+g2num-2+fit$df.prior, log=TRUE)+log(w)
  }
  names(limmafits) <- rownames(compid)
  limmacompnum<-nrow(compid)
  result<-list(t       = limmat,
               v0      = limmav0,
               df0     = limmadf,
               s20     = limmas2,
               g1num   = limmag1num,
               g2num   = limmag2num,
               compnum = limmacompnum,
               fits    = limmafits)
}

limmafit.counts <-
  function (exprs, groupid, compid, norm.factor.method = "TMM", voom.normalize.method = "none")
  {
    limmafits  <- list()
    compnum    <- nrow(compid)
    genenum    <- nrow(exprs)
    limmat     <- matrix(NA,genenum,compnum)
    limmas2    <- rep(0,compnum)
    limmadf    <- rep(0,compnum)
    limmav0    <- rep(0,compnum)
    limmag1num <- rep(0,compnum)
    limmag2num <- rep(0,compnum)

    rownames(limmat)  <- rownames(exprs)
    colnames(limmat)  <- rownames(compid)
    names(limmas2)    <- rownames(compid)
    names(limmadf)    <- rownames(compid)
    names(limmav0)    <- rownames(compid)
    names(limmag1num) <- rownames(compid)
    names(limmag2num) <- rownames(compid)

    for (i in 1:compnum) {
      message(paste("Running limma for comparision",i,"/",compnum))
      selid1 <- which(groupid == compid[i, 1])
      selid2 <- which(groupid == compid[i, 2])
      # make a new count data frame
      counts <- cbind(exprs[, selid1], exprs[, selid2])

      # remove NAs
      not.nas <- which(apply(counts, 1, function(x) !any(is.na(x))) == TRUE)

      # runn voom/limma
      d <- DGEList(counts[not.nas,])
      d <- calcNormFactors(d, method = norm.factor.method)
      g1num <- length(selid1)
      g2num <- length(selid2)
      designmat <- cbind(base = rep(1, (g1num + g2num)), delta = c(rep(0,
                                                                       g1num), rep(1, g2num)))

      y <- voom(d, designmat, normalize.method = voom.normalize.method)
      fit <- lmFit(y, designmat)
      fit <- eBayes(fit)

      limmafits[[i]] <- fit
      limmat[not.nas, i] <- fit$t[, 2]
      limmas2[i] <- fit$s2.prior
      limmadf[i] <- fit$df.prior
      limmav0[i] <- fit$var.prior[2]
      limmag1num[i] <- g1num
      limmag2num[i] <- g2num
    }
    limmacompnum <- nrow(compid)
    names(limmafits) <- rownames(compid)
    result <- list(t       = limmat,
                   v0      = limmav0,
                   df0     = limmadf,
                   s20     = limmas2,
                   g1num   = limmag1num,
                   g2num   = limmag2num,
                   compnum = limmacompnum,
                   fits    = limmafits)
  }

limmafit.list <-
  function (fitlist, cmp.idx=2)
  {
    compnum    <- length(fitlist)

    genes <- c()
    for (i in 1:compnum) genes <- unique(c(genes, rownames(fitlist[[i]])))

    genenum    <- length(genes)
    limmat     <- matrix(NA,genenum,compnum)
    limmas2    <- rep(0,compnum)
    limmadf    <- rep(0,compnum)
    limmav0    <- rep(0,compnum)
    limmag1num <- rep(0,compnum)
    limmag2num <- rep(0,compnum)

    rownames(limmat)  <- genes
    colnames(limmat)  <- names(fitlist)
    names(limmas2)    <- names(fitlist)
    names(limmadf)    <- names(fitlist)
    names(limmav0)    <- names(fitlist)
    names(limmag1num) <- names(fitlist)
    names(limmag2num) <- names(fitlist)

    for (i in 1:compnum) {
      this.t <- fitlist[[i]]$t[,cmp.idx]
      limmat[names(this.t),i] <- this.t

      limmas2[i]    <- fitlist[[i]]$s2.prior
      limmadf[i]    <- fitlist[[i]]$df.prior
      limmav0[i]    <- fitlist[[i]]$var.prior[cmp.idx]
      limmag1num[i] <- sum(fitlist[[i]]$design[,cmp.idx]==0)
      limmag2num[i] <- sum(fitlist[[i]]$design[,cmp.idx]==1)
    }

    limmacompnum <- compnum
    result <- list(t       = limmat,
                   v0      = limmav0,
                   df0     = limmadf,
                   s20     = limmas2,
                   g1num   = limmag1num,
                   g2num   = limmag2num,
                   compnum = limmacompnum,
                   fits    = limmafits)

  }

## Rank genes based on statistics
generank<-function(x) {
  xcol<-ncol(x)
  xrow<-nrow(x)
  result<-matrix(0,xrow,xcol)
  z<-(1:1:xrow)
  for(i in 1:xcol) {
    y<-sort(x[,i],decreasing=TRUE,na.last=TRUE)
    result[,i]<-match(x[,i],y)
    result[,i]<-order(result[,i])
  }
  result
}

## Log-likelihood for moderated t under H0
modt.f0.loglike<-function(x,df) {
  a<-dt(x, df, log=TRUE)
  result<-as.vector(a)
  flag<-which(is.na(result)==TRUE)
  result[flag]<-0
  result
}

## Log-likelihood for moderated t under H1
## param=c(df,g1num,g2num,v0)
modt.f1.loglike<-function(x,param) {
  df<-param[1]
  g1num<-param[2]
  g2num<-param[3]
  v0<-param[4]
  w<-sqrt(1+v0/(1/g1num+1/g2num))
  dt(x/w, df, log=TRUE)-log(w)
  a<-dt(x/w, df, log=TRUE)-log(w)
  result<-as.vector(a)
  flag<-which(is.na(result)==TRUE)
  result[flag]<-0
  result
}

## Correlation Motif Fit
cmfit.X<-function(x, type, K=1, tol=1e-3, max.iter=100) {
  ## initialize
  xrow <- nrow(x)
  xcol <- ncol(x)
  loglike0 <- list()
  loglike1 <- list()
  p <- rep(1, K)/K
  q <- matrix(runif(K * xcol), K, xcol)
  q[1, ] <- rep(0.01, xcol)
  for (i in 1:xcol) {
    f0 <- type[[i]][[1]]
    f0param <- type[[i]][[2]]
    f1 <- type[[i]][[3]]
    f1param <- type[[i]][[4]]
    loglike0[[i]] <- f0(x[, i], f0param)
    loglike1[[i]] <- f1(x[, i], f1param)
  }
  condlike <- list()
  for (i in 1:xcol) {
    condlike[[i]] <- matrix(0, xrow, K)
  }
  loglike.old <- -1e+10
  for (i.iter in 1:max.iter) {
    if ((i.iter%%50) == 0) {
      print(paste("We have run the first ", i.iter, " iterations for K=",
                  K, sep = ""))
    }
    err <- tol + 1
    clustlike <- matrix(0, xrow, K)
    #templike <- matrix(0, xrow, 2)
    templike1 <- rep(0, xrow)
    templike2 <- rep(0, xrow)
    for (j in 1:K) {
      for (i in 1:xcol) {
        templike1 <- log(q[j, i]) + loglike1[[i]]
        templike2 <- log(1 - q[j, i]) + loglike0[[i]]
        tempmax <- Rfast::Pmax(templike1, templike2)

        templike1 <- exp(templike1 - tempmax)
        templike2 <- exp(templike2 - tempmax)

        tempsum <- templike1 + templike2
        clustlike[, j] <- clustlike[, j] + tempmax +
          log(tempsum)
        condlike[[i]][, j] <- templike1/tempsum
      }
      clustlike[, j] <- clustlike[, j] + log(p[j])
    }
    #tempmax <- apply(clustlike, 1, max)
    tempmax <- Rfast::rowMaxs(clustlike, value=TRUE)
    for (j in 1:K) {
      clustlike[, j] <- exp(clustlike[, j] - tempmax)
    }
    #tempsum <- apply(clustlike, 1, sum)
    tempsum <- Rfast::rowsums(clustlike)
    for (j in 1:K) {
      clustlike[, j] <- clustlike[, j]/tempsum
    }
    #p.new <- (apply(clustlike, 2, sum) + 1)/(xrow + K)
    p.new <- (Rfast::colsums(clustlike) + 1)/(xrow + K)
    q.new <- matrix(0, K, xcol)
    for (j in 1:K) {
      clustpsum <- sum(clustlike[, j])
      for (i in 1:xcol) {
        q.new[j, i] <- (sum(clustlike[, j] * condlike[[i]][,
                                                           j]) + 1)/(clustpsum + 2)
      }
    }
    err.p <- max(abs(p.new - p)/p)
    err.q <- max(abs(q.new - q)/q)
    err <- max(err.p, err.q)
    loglike.new <- (sum(tempmax + log(tempsum)) + sum(log(p.new)) +
                      sum(log(q.new) + log(1 - q.new)))/xrow
    p <- p.new
    q <- q.new
    loglike.old <- loglike.new
    if (err < tol) {
      break
    }
  }
  clustlike <- matrix(0, xrow, K)
  for (j in 1:K) {
    for (i in 1:xcol) {
      templike1 <- log(q[j, i]) + loglike1[[i]]
      templike2 <- log(1 - q[j, i]) + loglike0[[i]]
      tempmax <- Rfast::Pmax(templike1, templike2)

      templike1 <- exp(templike1 - tempmax)
      templike2 <- exp(templike2 - tempmax)

      tempsum <- templike1 + templike2
      clustlike[, j] <- clustlike[, j] + tempmax + log(tempsum)
      condlike[[i]][, j] <- templike1/tempsum
    }
    clustlike[, j] <- clustlike[, j] + log(p[j])
  }
  #tempmax <- apply(clustlike, 1, max)
  tempmax <- Rfast::rowMaxs(clustlike, value=TRUE)
  for (j in 1:K) {
    clustlike[, j] <- exp(clustlike[, j] - tempmax)
  }
  #tempsum <- apply(clustlike, 1, sum)
  tempsum <- Rfast::rowsums(clustlike)
  for (j in 1:K) {
    clustlike[, j] <- clustlike[, j]/tempsum
  }
  p.post <- matrix(0, xrow, xcol)
  for (j in 1:K) {
    for (i in 1:xcol) {
      p.post[, i] <- p.post[, i] + clustlike[, j] * condlike[[i]][,
                                                                  j]
    }
  }
  loglike.old <- loglike.old - (sum(log(p)) + sum(log(q) +
                                                    log(1 - q)))/xrow
  loglike.old <- loglike.old * xrow
  result <- list(p.post = p.post, motif.prior = p, motif.q = q,
                 loglike = loglike.old, clustlike=clustlike, condlike=condlike)
}

## Fit using (0,0,...,0) and (1,1,...,1)
cmfitall<-function(x, type, tol=1e-3, max.iter=100) {
  ## initialize
  xrow<-nrow(x)
  xcol<-ncol(x)
  loglike0<-list()
  loglike1<-list()
  p<-0.01

  ## compute loglikelihood
  L0<-matrix(0,xrow,1)
  L1<-matrix(0,xrow,1)
  for(i in 1:xcol) {
    f0<-type[[i]][[1]]
    f0param<-type[[i]][[2]]
    f1<-type[[i]][[3]]
    f1param<-type[[i]][[4]]
    loglike0[[i]]<-f0(x[,i],f0param)
    loglike1[[i]]<-f1(x[,i],f1param)
    L0<-L0+loglike0[[i]]
    L1<-L1+loglike1[[i]]
  }


  ## EM algorithm to get MLE of p and q
  loglike.old <- -1e10
  for(i.iter in 1:max.iter) {
    if((i.iter%%50) == 0) {
      print(paste("We have run the first ", i.iter, " iterations",sep=""))
    }
    err<-tol+1

    ## compute posterior cluster membership
    clustlike<-matrix(0,xrow,2)
    clustlike[,1]<-log(1-p)+L0
    clustlike[,2]<-log(p)+L1

    tempmax<-apply(clustlike,1,max)
    for(j in 1:2) {
      clustlike[,j]<-exp(clustlike[,j]-tempmax)
    }
    tempsum<-apply(clustlike,1,sum)


    ## update motif occurrence rate
    for(j in 1:2) {
      clustlike[,j]<-clustlike[,j]/tempsum
    }

    p.new<-(sum(clustlike[,2])+1)/(xrow+2)

    ## evaluate convergence
    err<-abs(p.new-p)/p


    ## evaluate whether the log.likelihood increases
    loglike.new<-(sum(tempmax+log(tempsum))+log(p.new)+log(1-p.new))/xrow

    loglike.old<-loglike.new
    p<-p.new

    if(err<tol) {
      break;
    }
  }

  ## compute posterior p
  clustlike<-matrix(0,xrow,2)
  clustlike[,1]<-log(1-p)+L0
  clustlike[,2]<-log(p)+L1

  tempmax<-apply(clustlike,1,max)
  for(j in 1:2) {
    clustlike[,j]<-exp(clustlike[,j]-tempmax)
  }
  tempsum<-apply(clustlike,1,sum)

  for(j in 1:2) {
    clustlike[,j]<-clustlike[,j]/tempsum
  }

  p.post<-matrix(0,xrow,xcol)
  for(i in 1:xcol) {
    p.post[,i]<-clustlike[,2]
  }

  ## return

  #calculate back loglikelihood
  loglike.old<-loglike.old-(log(p)+log(1-p))/xrow
  loglike.old<-loglike.old*xrow
  result<-list(p.post=p.post, motif.prior=p, loglike=loglike.old)
}

## Fit each dataset separately
cmfitsep<-function(x, type, tol=1e-3, max.iter=100) {
  ## initialize
  xrow<-nrow(x)
  xcol<-ncol(x)
  loglike0<-list()
  loglike1<-list()
  p<-0.01*rep(1,xcol)
  loglike.final<-rep(0,xcol)

  ## compute loglikelihood
  for(i in 1:xcol) {
    f0<-type[[i]][[1]]
    f0param<-type[[i]][[2]]
    f1<-type[[i]][[3]]
    f1param<-type[[i]][[4]]
    loglike0[[i]]<-f0(x[,i],f0param)
    loglike1[[i]]<-f1(x[,i],f1param)
  }

  p.post<-matrix(0,xrow,xcol)

  ## EM algorithm to get MLE of p
  for(coli in 1:xcol) {
    loglike.old <- -1e10
    for(i.iter in 1:max.iter) {
      if((i.iter%%50) == 0) {
        print(paste("We have run the first ", i.iter, " iterations",sep=""))
      }
      err<-tol+1

      ## compute posterior cluster membership
      clustlike<-matrix(0,xrow,2)
      clustlike[,1]<-log(1-p[coli])+loglike0[[coli]]
      clustlike[,2]<-log(p[coli])+loglike1[[coli]]

      tempmax<-apply(clustlike,1,max)
      for(j in 1:2) {
        clustlike[,j]<-exp(clustlike[,j]-tempmax)
      }
      tempsum<-apply(clustlike,1,sum)

      ## evaluate whether the log.likelihood increases
      loglike.new<-sum(tempmax+log(tempsum))/xrow

      ## update motif occurrence rate
      for(j in 1:2) {
        clustlike[,j]<-clustlike[,j]/tempsum
      }

      p.new<-(sum(clustlike[,2]))/(xrow)

      ## evaluate convergence
      err<-abs(p.new-p[coli])/p[coli]
      loglike.old<-loglike.new
      p[coli]<-p.new

      if(err<tol) {
        break;
      }
    }

    ## compute posterior p
    clustlike<-matrix(0,xrow,2)
    clustlike[,1]<-log(1-p[coli])+loglike0[[coli]]
    clustlike[,2]<-log(p[coli])+loglike1[[coli]]

    tempmax<-apply(clustlike,1,max)
    for(j in 1:2) {
      clustlike[,j]<-exp(clustlike[,j]-tempmax)
    }
    tempsum<-apply(clustlike,1,sum)

    for(j in 1:2) {
      clustlike[,j]<-clustlike[,j]/tempsum
    }

    p.post[,coli]<-clustlike[,2]
    loglike.final[coli]<-loglike.old
  }


  ## return
  loglike.final<-loglike.final*xrow
  result<-list(p.post=p.post, motif.prior=p, loglike=loglike.final)
}

## Fit the full model
cmfitfull<-function(x, type, tol=1e-3, max.iter=100) {
  ## initialize
  xrow<-nrow(x)
  xcol<-ncol(x)
  loglike0<-list()
  loglike1<-list()
  K<-2^xcol
  p<-rep(1,K)/K
  pattern<-rep(0,xcol)
  patid<-matrix(0,K,xcol)

  ## compute loglikelihood
  for(i in 1:xcol) {
    f0<-type[[i]][[1]]
    f0param<-type[[i]][[2]]
    f1<-type[[i]][[3]]
    f1param<-type[[i]][[4]]
    loglike0[[i]]<-f0(x[,i],f0param)
    loglike1[[i]]<-f1(x[,i],f1param)
  }
  L<-matrix(0,xrow,K)
  for(i in 1:K)
  {
    patid[i,]<-pattern
    for(j in 1:xcol) {
      if(pattern[j] < 0.5) {
        L[,i]<-L[,i]+loglike0[[j]]
      } else {
        L[,i]<-L[,i]+loglike1[[j]]
      }
    }

    if(i < K) {
      pattern[xcol]<-pattern[xcol]+1
      j<-xcol
      while(pattern[j] > 1) {
        pattern[j]<-0
        j<-j-1
        pattern[j]<-pattern[j]+1
      }
    }
  }

  ## EM algorithm to get MLE of p and q
  loglike.old <- -1e10
  for(i.iter in 1:max.iter) {
    if((i.iter%%50) == 0) {
      print(paste("We have run the first ", i.iter, " iterations",sep=""))
    }
    err<-tol+1

    ## compute posterior cluster membership
    clustlike<-matrix(0,xrow,K)
    for(j in 1:K) {
      clustlike[,j]<-log(p[j])+L[,j]
    }

    tempmax<-apply(clustlike,1,max)
    for(j in 1:K) {
      clustlike[,j]<-exp(clustlike[,j]-tempmax)
    }
    tempsum<-apply(clustlike,1,sum)


    ## update motif occurrence rate
    for(j in 1:K) {
      clustlike[,j]<-clustlike[,j]/tempsum
    }

    p.new<-(apply(clustlike,2,sum)+1)/(xrow+K)

    ## evaluate convergence
    err<-max(abs(p.new-p)/p)

    ## evaluate whether the log.likelihood increases
    loglike.new<-(sum(tempmax+log(tempsum))+sum(log(p.new)))/xrow

    loglike.old<-loglike.new
    p<-p.new

    if(err<tol) {
      break;
    }
  }

  ## compute posterior p
  clustlike<-matrix(0,xrow,K)
  for(j in 1:K) {
    clustlike[,j]<-log(p[j])+L[,j]
  }

  tempmax<-apply(clustlike,1,max)
  for(j in 1:K) {
    clustlike[,j]<-exp(clustlike[,j]-tempmax)
  }
  tempsum<-apply(clustlike,1,sum)

  for(j in 1:K) {
    clustlike[,j]<-clustlike[,j]/tempsum
  }

  p.post<-matrix(0,xrow,xcol)
  for(j in 1:K) {
    for(i in 1:xcol) {
      if(patid[j,i] > 0.5) {
        p.post[,i]<-p.post[,i]+clustlike[,j]
      }
    }
  }

  ## return
  #calculate back loglikelihood
  loglike.old<-loglike.old-sum(log(p))/xrow
  loglike.old<-loglike.old*xrow
  result<-list(p.post=p.post, motif.prior=p, loglike=loglike.old)
}

generatetype<-function(limfitted)
{
  jtype<-list()
  df<-limfitted$g1num+limfitted$g2num-2+limfitted$df0
  for(j in 1:limfitted$compnum)
  {
    jtype[[j]]<-list(f0=modt.f0.loglike, f0.param=df[j], f1=modt.f1.loglike, f1.param=c(df[j],limfitted$g1num[j],limfitted$g2num[j],limfitted$v0[j]))
  }
  jtype
}

cormotiffit <- function(exprs, groupid=NULL, compid=NULL, K=1, tol=1e-3,
                        max.iter=100, BIC=TRUE, norm.factor.method="TMM",
                        voom.normalize.method = "none", runtype=c("logCPM","counts","limmafits"), each=3)
{
  # first I want to do some typechecking. Input can be either a normalized
  # matrix, a count matrix, or a list of limma fits. Dispatch the correct
  # limmafit accordingly.
  # todo: add some typechecking here
  limfitted <- list()
  if (runtype=="counts") {
    limfitted <- limmafit.counts(exprs,groupid,compid, norm.factor.method, voom.normalize.method)
  } else if (runtype=="logCPM") {
    limfitted <- limmafit.default(exprs,groupid,compid)
  } else if (runtype=="limmafits") {
    limfitted <- limmafit.list(exprs)
  } else {
    stop("runtype must be one of 'logCPM', 'counts', or 'limmafits'")
  }


  jtype<-generatetype(limfitted)
  fitresult<-list()
  ks <- rep(K, each = each)
  fitresult <- bplapply(1:length(ks), function(i, x, type, ks, tol, max.iter) {
    cmfit.X(x, type, K = ks[i], tol = tol, max.iter = max.iter)
  }, x=limfitted$t, type=jtype, ks=ks, tol=tol, max.iter=max.iter)

  best.fitresults <- list()
  for (i in 1:length(K)) {
    w.k <- which(ks==K[i])
    this.bic <- c()
    for (j in w.k) this.bic[j] <- -2 * fitresult[[j]]$loglike + (K[i] - 1 + K[i] * limfitted$compnum) * log(dim(limfitted$t)[1])
    w.min <- which(this.bic == min(this.bic, na.rm = TRUE))[1]
    best.fitresults[[i]] <- fitresult[[w.min]]
  }
  fitresult <- best.fitresults

  bic <- rep(0, length(K))
  aic <- rep(0, length(K))
  loglike <- rep(0, length(K))
  for (i in 1:length(K)) loglike[i] <- fitresult[[i]]$loglike
  for (i in 1:length(K)) bic[i] <- -2 * fitresult[[i]]$loglike + (K[i] - 1 + K[i] * limfitted$compnum) * log(dim(limfitted$t)[1])
  for (i in 1:length(K)) aic[i] <- -2 * fitresult[[i]]$loglike + 2 * (K[i] - 1 + K[i] * limfitted$compnum)
  if(BIC==TRUE) {
    bestflag=which(bic==min(bic))
  }
  else {
    bestflag=which(aic==min(aic))
  }
  result<-list(bestmotif=fitresult[[bestflag]],bic=cbind(K,bic),
               aic=cbind(K,aic),loglike=cbind(K,loglike), allmotifs=fitresult)

}

cormotiffitall<-function(exprs,groupid,compid, tol=1e-3, max.iter=100)
{
  limfitted<-limmafit(exprs,groupid,compid)
  jtype<-generatetype(limfitted)
  fitresult<-cmfitall(limfitted$t,type=jtype,tol=1e-3,max.iter=max.iter)
}

cormotiffitsep<-function(exprs,groupid,compid, tol=1e-3, max.iter=100)
{
  limfitted<-limmafit(exprs,groupid,compid)
  jtype<-generatetype(limfitted)
  fitresult<-cmfitsep(limfitted$t,type=jtype,tol=1e-3,max.iter=max.iter)
}

cormotiffitfull<-function(exprs,groupid,compid, tol=1e-3, max.iter=100)
{
  limfitted<-limmafit(exprs,groupid,compid)
  jtype<-generatetype(limfitted)
  fitresult<-cmfitfull(limfitted$t,type=jtype,tol=1e-3,max.iter=max.iter)
}

plotIC<-function(fitted_cormotif)
{
  oldpar<-par(mfrow=c(1,2))
  plot(fitted_cormotif$bic[,1], fitted_cormotif$bic[,2], type="b",xlab="Motif Number", ylab="BIC", main="BIC")
  plot(fitted_cormotif$aic[,1], fitted_cormotif$aic[,2], type="b",xlab="Motif Number", ylab="AIC", main="AIC")
}

plotMotif<-function(fitted_cormotif,title="")
{
  layout(matrix(1:2,ncol=2))
  u<-1:dim(fitted_cormotif$bestmotif$motif.q)[2]
  v<-1:dim(fitted_cormotif$bestmotif$motif.q)[1]
  image(u,v,t(fitted_cormotif$bestmotif$motif.q),
        col=gray(seq(from=1,to=0,by=-0.1)),xlab="Study",yaxt = "n",
        ylab="Corr. Motifs",main=paste(title,"pattern",sep=" "))
  axis(2,at=1:length(v))
  for(i in 1:(length(u)+1))
  {
    abline(v=(i-0.5))
  }
  for(i in 1:(length(v)+1))
  {
    abline(h=(i-0.5))
  }
  Ng=10000
  if(is.null(fitted_cormotif$bestmotif$p.post)!=TRUE)
    Ng=nrow(fitted_cormotif$bestmotif$p.post)
  genecount=floor(fitted_cormotif$bestmotif$motif.p*Ng)
  NK=nrow(fitted_cormotif$bestmotif$motif.q)
  plot(0,0.7,pch=".",xlim=c(0,1.2),ylim=c(0.75,NK+0.25),
       frame.plot=FALSE,axes=FALSE,xlab="No. of genes",ylab="", main=paste(title,"frequency",sep=" "))
  segments(0,0.7,fitted_cormotif$bestmotif$motif.p[1],0.7)
  rect(0,1:NK-0.3,fitted_cormotif$bestmotif$motif.p,1:NK+0.3,
       col="dark grey")
  mtext(1:NK,at=1:NK,side=2,cex=0.8)
  text(fitted_cormotif$bestmotif$motif.p+0.15,1:NK,
       labels=floor(fitted_cormotif$bestmotif$motif.p*Ng))
}
```

## **ðŸ“Œ Load Required Libraries**
```{r load_libraries, echo=TRUE, results='hide',message=FALSE, warning=FALSE}
library(Cormotif)
library(Rfast)
library(dplyr)
library(BiocParallel)
library(gprofiler2)
library(ggplot2)
```

## **ðŸ“Œ Corrmotif Model 0.1 Micromolar**
## **ðŸ“Œ Load Corrmotif Data**
```{r load_data, echo=TRUE, results='hide', message=FALSE}
# Read the Corrmotif Results
Corrmotif <- read.csv("data/Corrmotif/CX5461.csv")
Corrmotif_df <- data.frame(Corrmotif)
rownames(Corrmotif_df) <- Corrmotif_df$Gene

# Filter for 0.1 Concentration Only
exprs.corrmotif <- as.matrix(Corrmotif_df[, grep("0.1", colnames(Corrmotif_df))])


# Read group and comparison IDs
groupid <- read.csv("data/Corrmotif/groupid.csv")
groupid_df <- data.frame(groupid[, grep("0.1", colnames(groupid))])


compid <- read.csv("data/Corrmotif/Compid.csv")
compid_df <- compid[compid$Cond1 %in% unique(as.numeric(groupid_df)) & compid$Cond2 %in% unique(as.numeric(groupid_df)), ]
```

## **ðŸ“Œ Corrmotif Model 0.1 Micromolar (K=1:8)**
## **ðŸ“Œ Fit Corrmotif Model (K=1:8) (0.1 Micromolar)**
```{r Model, eval=FALSE,echo=TRUE, message=FALSE}
set.seed(11111)
# Fit Corrmotif Model (K = 1 to 8)
set.seed(11111)
motif.fitted_0.1 <- cormotiffit(
  exprs = exprs.corrmotif,
  groupid = groupid_df,
  compid = compid_df,
  K = 1:8,
  max.iter = 1000,
  BIC = TRUE,
  runtype = "logCPM"
)

gene_prob_0.1 <- motif.fitted_0.1$bestmotif$p.post
rownames(gene_prob_0.1) <- rownames(Corrmotif_df)
motif_prob_0.1 <- motif.fitted_0.1$bestmotif$clustlike
rownames(motif_prob_0.1) <- rownames(gene_prob_0.1)
write.csv(motif_prob_0.1,"data/cormotif_probability_genelist_0.1.csv")
```

## **ðŸ“Œ Plot motif (0.1 Micromolar)**
```{r Model1, echo=TRUE, message=FALSE}
cormotif_0.1 <- readRDS("data/Corrmotif/cormotif_0.1.RDS")

cormotif_0.1$bic

plotIC(cormotif_0.1)

plotMotif(cormotif_0.1)

```

## **ðŸ“Œ Extract Gene Probabilities (0.1 Micromolar)**
```{r Prob, echo=TRUE, message=FALSE}
# Extract posterior probabilities for genes
gene_prob_tran_0.1 <- cormotif_0.1$bestmotif$p.post

rownames(gene_prob_tran_0.1) <- rownames(Corrmotif_df)

# Define gene probability groups
prob_1_0.1  <- rownames(gene_prob_tran_0.1[(gene_prob_tran_0.1[,1] <0.5 & gene_prob_tran_0.1[,2] <0.5 & gene_prob_tran_0.1[,3] <0.5 & gene_prob_tran_0.1[,4] <0.5 & gene_prob_tran_0.1[,5] < 0.5 & gene_prob_tran_0.1[,6]<0.5),])
length(prob_1_0.1)

prob_2_0.1  <- rownames(gene_prob_tran_0.1[(gene_prob_tran_0.1[,1] <0.5 & gene_prob_tran_0.1[,2] >0.5 & gene_prob_tran_0.1[,3] >0.5 & gene_prob_tran_0.1[,4] <0.5 & gene_prob_tran_0.1[,5] > 0.5 & gene_prob_tran_0.1[,6]>0.5),])
length(prob_2_0.1)

prob_3_0.1  <- rownames(gene_prob_tran_0.1[(gene_prob_tran_0.1[,1] <0.5 & gene_prob_tran_0.1[,2] <0.5 & gene_prob_tran_0.1[,3] <0.5 & gene_prob_tran_0.1[,4] <0.5 & gene_prob_tran_0.1[,5] > 0.5 & gene_prob_tran_0.1[,6]>0.5),])
length(prob_3_0.1)
```

## **ðŸ“Œ Distribution of Gene Clusters Identified by Corrmotif (0.1 micromolar)**
```{r pie-chart_0.1, echo=TRUE, message=FALSE, warning=FALSE}

# Load necessary library
library(ggplot2)

# Data
data <- data.frame(
  Category = c("Non response (0.1 ÂµM)", "CX-DOX mid-late response (0.1 ÂµM)", "DOX only mid-late (0.1 ÂµM)"),
  Value = c(12308, 415, 1551)
)

# Define custom colors
custom_colors <- c("Non response (0.1 ÂµM)" = "#FF9999",
                   "CX-DOX mid-late response (0.1 ÂµM)" = "#66B2FF",
                   "DOX only mid-late (0.1 ÂµM)" = "#99FF99")

# Create pie chart
ggplot(data, aes(x = "", y = Value, fill = Category)) +
  geom_bar(width = 1, stat = "identity") +
  coord_polar("y", start = 0) +
  geom_text(aes(label = Value),
            position = position_stack(vjust = 0.5),
            size = 4, color = "black") +
  labs(title = "Pie Chart (0.1 micromolar Corrmotif)", x = NULL, y = NULL) +
  theme_void() +
  scale_fill_manual(values = custom_colors)
```

## **ðŸ“Œ Save Corr-motif datasets for Gene Ontology analysis (0.1 Micromolar)**
```{r box, echo=TRUE, results='hide',message=FALSE, warning=FALSE}
write.csv(data.frame(Entrez_ID = prob_1_0.1), "data/prob_1_0.1.csv", row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_2_0.1), "data/prob_2_0.1.csv", row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_3_0.1), "data/prob_3_0.1.csv", row.names = FALSE)
```

## **ðŸ“Œ 0.1 Micromolar abs logFC**
```{r 0.1abs logFC, echo=TRUE, message=FALSE, warning=FALSE, fig.width=10,fig.height=8}
# Load Required Libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.1 <- as.character(read.csv("data/prob_1_0.1.csv")$Entrez_ID)
prob_2_0.1 <- as.character(read.csv("data/prob_2_0.1.csv")$Entrez_ID)
prob_3_0.1 <- as.character(read.csv("data/prob_3_0.1.csv")$Entrez_ID)

# Load Datasets (Only 0.1 Micromolar)
CX_0.1_3  <- read.csv("data/DEGs/Toptable_CX_0.1_3.csv")
CX_0.1_24 <- read.csv("data/DEGs/Toptable_CX_0.1_24.csv")
CX_0.1_48 <- read.csv("data/DEGs/Toptable_CX_0.1_48.csv")

DOX_0.1_3  <- read.csv("data/DEGs/Toptable_DOX_0.1_3.csv")
DOX_0.1_24 <- read.csv("data/DEGs/Toptable_DOX_0.1_24.csv")
DOX_0.1_48 <- read.csv("data/DEGs/Toptable_DOX_0.1_48.csv")

# Combine All 0.1 Micromolar Datasets into a Single Dataframe
all_toptables_0.1 <- bind_rows(
  CX_0.1_3  %>% mutate(Drug = "CX.5461", Timepoint = "Timepoint: 3 hours"),
  CX_0.1_24 %>% mutate(Drug = "CX.5461", Timepoint = "Timepoint: 24 hours"),
  CX_0.1_48 %>% mutate(Drug = "CX.5461", Timepoint = "Timepoint: 48 hours"),
  DOX_0.1_3  %>% mutate(Drug = "DOX", Timepoint = "Timepoint: 3 hours"),
  DOX_0.1_24 %>% mutate(Drug = "DOX", Timepoint = "Timepoint: 24 hours"),
  DOX_0.1_48 %>% mutate(Drug = "DOX", Timepoint = "Timepoint: 48 hours")
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.1$Entrez_ID <- as.character(all_toptables_0.1$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.1 <- all_toptables_0.1 %>%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ "Non response\n(0.1 micromolar)",
      Entrez_ID %in% prob_2_0.1 ~ "CX-DOX mid-late response\n(0.1 micromolar)",
      Entrez_ID %in% prob_3_0.1 ~ "DOX only mid-late response\n(0.1 micromolar)",
      TRUE ~ NA_character_
    )
  ) %>%
  filter(!is.na(Response_Group))

# Compute Absolute logFC
all_toptables_0.1 <- all_toptables_0.1 %>%
  mutate(absFC = abs(logFC))

# Convert Factors for Proper Ordering (Reversed Order for Response Groups)
all_toptables_0.1 <- all_toptables_0.1 %>%
  mutate(
    Drug = factor(Drug, levels = c("CX.5461", "DOX")),
    Timepoint = factor(Timepoint, levels = c("Timepoint: 3 hours", "Timepoint: 24 hours", "Timepoint: 48 hours")),
    Response_Group = factor(Response_Group,
                            levels = c(
                              "DOX only mid-late response\n(0.1 micromolar)",
                              "CX-DOX mid-late response\n(0.1 micromolar)",
                              "Non response\n(0.1 micromolar)"  # Reversed Order
                            ))
  )

# **Plot the Boxplot with Faceted Labels Wrapping Correctly**
ggplot(all_toptables_0.1, aes(x = Drug, y = absFC, fill = Drug)) +
  geom_boxplot() +
  scale_fill_manual(values = c("CX.5461" = "blue", "DOX" = "red")) +  # Custom color palette
  facet_grid(Response_Group ~ Timepoint, labeller = label_wrap_gen(width = 20)) +  # Ensure Proper Wrapping
  theme_bw() +
  labs(
    x = "Drugs",
    y = "|Log Fold Change|",
    title = "|Log Fold| for 0.1 micromolar"
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = "black"),
    axis.line = element_line(linewidth = 1.5),
    strip.background = element_rect(fill = "gray"),  # Gray background for facet labels
    strip.text = element_text(size = 12, color = "black", face = "bold"),  # Bold styling for facet labels
    axis.text.x = element_text(size = 10, color = "black", angle = 15)
  )
```

## **ðŸ“Œ 0.1 Micromolar mean (Abs logFC) across timepoints**
```{r 0.1 mean, echo=TRUE, message=FALSE, warning=FALSE, fig.height=12, fig.width=10}

# Load Required Libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.1 <- as.character(read.csv("data/prob_1_0.1.csv")$Entrez_ID)
prob_2_0.1 <- as.character(read.csv("data/prob_2_0.1.csv")$Entrez_ID)
prob_3_0.1 <- as.character(read.csv("data/prob_3_0.1.csv")$Entrez_ID)

# Load Datasets (Only 0.1 Micromolar)
CX_0.1_3  <- read.csv("data/DEGs/Toptable_CX_0.1_3.csv")
CX_0.1_24 <- read.csv("data/DEGs/Toptable_CX_0.1_24.csv")
CX_0.1_48 <- read.csv("data/DEGs/Toptable_CX_0.1_48.csv")

DOX_0.1_3  <- read.csv("data/DEGs/Toptable_DOX_0.1_3.csv")
DOX_0.1_24 <- read.csv("data/DEGs/Toptable_DOX_0.1_24.csv")
DOX_0.1_48 <- read.csv("data/DEGs/Toptable_DOX_0.1_48.csv")

# Combine All 0.1 Micromolar Datasets into a Single Dataframe
all_toptables_0.1 <- bind_rows(
  CX_0.1_3  %>% mutate(Drug = "CX.5461", Timepoint = "3"),
  CX_0.1_24 %>% mutate(Drug = "CX.5461", Timepoint = "24"),
  CX_0.1_48 %>% mutate(Drug = "CX.5461", Timepoint = "48"),
  DOX_0.1_3  %>% mutate(Drug = "DOX", Timepoint = "3"),
  DOX_0.1_24 %>% mutate(Drug = "DOX", Timepoint = "24"),
  DOX_0.1_48 %>% mutate(Drug = "DOX", Timepoint = "48")
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.1$Entrez_ID <- as.character(all_toptables_0.1$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.1 <- all_toptables_0.1 %>%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ "Non response\n(0.1 micromolar)",
      Entrez_ID %in% prob_2_0.1 ~ "CX-DOX mid-late response\n(0.1 micromolar)",
      Entrez_ID %in% prob_3_0.1 ~ "DOX only mid-late response\n(0.1 micromolar)",
      TRUE ~ NA_character_
    )
  ) %>%
  filter(!is.na(Response_Group))

# Compute Mean Absolute logFC for Line Plot
data_summary <- all_toptables_0.1 %>%
  mutate(abs_logFC = abs(logFC)) %>%  # Take absolute log fold change
  group_by(Response_Group, Drug, Timepoint) %>%
  dplyr::summarize(mean_abs_logFC = mean(abs_logFC, na.rm = TRUE), .groups = "drop") %>%
  as.data.frame()

# **Ensure all timepoints exist in the summary**
timepoints_full <- expand.grid(
  Response_Group = unique(all_toptables_0.1$Response_Group),
  Drug = unique(all_toptables_0.1$Drug),
  Timepoint = c("3", "24", "48")
)

# **Merge to keep missing timepoints**
data_summary <- full_join(timepoints_full, data_summary, by = c("Response_Group", "Drug", "Timepoint"))

# **Replace NA mean_abs_logFC with 0 if no genes were present**
data_summary$mean_abs_logFC[is.na(data_summary$mean_abs_logFC)] <- 0

# Convert Factors for Proper Ordering (Reversed Order for Response Groups)
data_summary <- data_summary %>%
  mutate(
    Timepoint = factor(Timepoint, levels = c("3", "24", "48"), labels = c("3 hours", "24 hours", "48 hours")),
    Response_Group = factor(Response_Group, levels = c(
      "DOX only mid-late response\n(0.1 micromolar)",
      "CX-DOX mid-late response\n(0.1 micromolar)",
      "Non response\n(0.1 micromolar)"  # Reversed Order
    ))
  )

# Define custom drug palette
drug_palette <- c("CX.5461" = "blue", "DOX" = "red")

# **Plot the Line Plot for Absolute logFC**
ggplot(data_summary, aes(x = Timepoint, y = mean_abs_logFC, group = Drug, color = Drug)) +
  geom_point(size = 3) +
  geom_line(size = 1.2) +
  scale_color_manual(values = drug_palette) +
  ylim(0, 2.5) +  # Adjust the Y-axis for better visualization
  facet_wrap(~ Response_Group, ncol = 1) +  # Facet by Response Group (Reversed Order)
  theme_bw() +
  labs(
    x = "Timepoints",
    y = "Mean |Log Fold Change|",
    title = "Mean |Log Fold Change| Across Response Groups (0.1 micromolar)",
    color = "Drug"
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = "black"),
    axis.text = element_text(size = 12, color = "black"),
    strip.text = element_text(size = 12, color = "black", face = "bold"),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )
```


## **ðŸ“Œ 0.1 Micromolar logFC**
```{r 0.1 logFC, echo=TRUE, message=FALSE, warning=FALSE, fig.width=10,fig.height=8}
# Load required libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.1 <- as.character(read.csv("data/prob_1_0.1.csv")$Entrez_ID)
prob_2_0.1 <- as.character(read.csv("data/prob_2_0.1.csv")$Entrez_ID)
prob_3_0.1 <- as.character(read.csv("data/prob_3_0.1.csv")$Entrez_ID)

# Load Datasets (Only 0.1 Micromolar)
CX_0.1_3  <- read.csv("data/DEGs/Toptable_CX_0.1_3.csv")
CX_0.1_24 <- read.csv("data/DEGs/Toptable_CX_0.1_24.csv")
CX_0.1_48 <- read.csv("data/DEGs/Toptable_CX_0.1_48.csv")

DOX_0.1_3  <- read.csv("data/DEGs/Toptable_DOX_0.1_3.csv")
DOX_0.1_24 <- read.csv("data/DEGs/Toptable_DOX_0.1_24.csv")
DOX_0.1_48 <- read.csv("data/DEGs/Toptable_DOX_0.1_48.csv")

# Combine All 0.1 Micromolar Datasets into a Single Dataframe
all_toptables_0.1 <- bind_rows(
  CX_0.1_3  %>% mutate(Drug = "CX.5461", Timepoint = "Timepoint: 3 hours"),
  CX_0.1_24 %>% mutate(Drug = "CX.5461", Timepoint = "Timepoint: 24 hours"),
  CX_0.1_48 %>% mutate(Drug = "CX.5461", Timepoint = "Timepoint: 48 hours"),
  DOX_0.1_3  %>% mutate(Drug = "DOX", Timepoint = "Timepoint: 3 hours"),
  DOX_0.1_24 %>% mutate(Drug = "DOX", Timepoint = "Timepoint: 24 hours"),
  DOX_0.1_48 %>% mutate(Drug = "DOX", Timepoint = "Timepoint: 48 hours")
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.1$Entrez_ID <- as.character(all_toptables_0.1$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.1 <- all_toptables_0.1 %>%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ "Non response\n(0.1 micromolar)",
      Entrez_ID %in% prob_2_0.1 ~ "CX-DOX mid-late response\n(0.1 micromolar)",
      Entrez_ID %in% prob_3_0.1 ~ "DOX only mid-late response\n(0.1 micromolar)",
      TRUE ~ NA_character_
    )
  ) %>%
  filter(!is.na(Response_Group))

# Convert factors to ensure correct ordering (Reversed Order for Response Groups)
all_toptables_0.1 <- all_toptables_0.1 %>%
  mutate(
    Timepoint = factor(Timepoint, levels = c("Timepoint: 3 hours", "Timepoint: 24 hours", "Timepoint: 48 hours")),
    Response_Group = factor(Response_Group, levels = c(
      "DOX only mid-late response\n(0.1 micromolar)",
      "CX-DOX mid-late response\n(0.1 micromolar)",
      "Non response\n(0.1 micromolar)"  # Reversed Order
    ))
  )

# **Plot the Boxplot**
ggplot(all_toptables_0.1, aes(x = Drug, y = logFC, fill = Drug)) +
  geom_boxplot() +
  scale_fill_manual(values = c("CX.5461" = "blue", "DOX" = "red")) +
  facet_grid(Response_Group ~ Timepoint) +
  theme_bw() +
  labs(x = "Drugs", y = "Log Fold Change", title = "Log Fold Change for 0.1 Micromolar") +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    strip.text = element_text(size = 12, face = "bold")
  )
```

## **ðŸ“Œ 0.1 Micromolar mean logFC across timepoints**
```{r 0.1 mean2, echo=TRUE, message=FALSE, warning=FALSE, fig.height=12, fig.width=10}

# Compute Mean logFC for Line Plot
data_summary <- all_toptables_0.1 %>%
  group_by(Response_Group, Drug, Timepoint) %>%
  dplyr::summarize(mean_logFC = mean(logFC, na.rm = TRUE), .groups = "drop") %>%
  as.data.frame()

# Convert factors to ensure correct ordering (Reversed Order for Response Groups)
data_summary <- data_summary %>%
  mutate(
    Timepoint = factor(Timepoint, levels = c("Timepoint: 3 hours", "Timepoint: 24 hours", "Timepoint: 48 hours")),
    Response_Group = factor(Response_Group, levels = c(
      "DOX only mid-late response\n(0.1 micromolar)",
      "CX-DOX mid-late response\n(0.1 micromolar)",
      "Non response\n(0.1 micromolar)"  # Reversed Order
    ))
  )

# **Plot the Line Plot**
ggplot(data_summary, aes(x = Timepoint, y = mean_logFC, group = Drug, color = Drug)) +
  geom_point(size = 3) +
  geom_line(size = 1.2) +
  scale_color_manual(values = c("CX.5461" = "blue", "DOX" = "red")) +
  ylim(-2, 1.5) +  # Adjust the Y-axis for better visualization
  facet_wrap(~ Response_Group, ncol = 1) +  # Facet by Response Group (Reversed Order)
  theme_bw() +
  labs(
    x = "Timepoints",
    y = "Mean Log Fold Change",
    title = "Mean Log Fold Change Across Response Groups (0.1 micromolar)",
    color = "Drug"
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = "black"),
    axis.text = element_text(size = 12, color = "black"),
    strip.text = element_text(size = 12, color = "black", face = "bold"),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )
```



## **ðŸ“Œ Corrmotif Model 0.5 Micromolar**
## **ðŸ“Œ Load Corrmotif Data**
```{r load_data_0.5, echo=TRUE, results='hide', message=FALSE}
# Read the Corrmotif Results
Corrmotif <- read.csv("data/Corrmotif/CX5461.csv")
Corrmotif_df <- data.frame(Corrmotif)
rownames(Corrmotif_df) <- Corrmotif_df$Gene

# Filter for 0.5 Concentration Only
exprs.corrmotif <- as.matrix(Corrmotif_df[, grep("0.5", colnames(Corrmotif_df))])


# Read group and comparison IDs
groupid <- read.csv("data/Corrmotif/groupid.csv")
groupid_df <- data.frame(groupid[, grep("0.5", colnames(groupid))])


compid <- read.csv("data/Corrmotif/Compid.csv")
compid_df <- compid[compid$Cond1 %in% unique(as.numeric(groupid_df)) & compid$Cond2 %in% unique(as.numeric(groupid_df)), ]
```

## **ðŸ“Œ Corrmotif Model 0.5 Micromolar (K=1:8)**
## **ðŸ“Œ Fit Corrmotif Model (K=1:8) (0.5 Micromolar)**
```{r Model_0.5, eval=FALSE, echo=TRUE, message=FALSE}
# Fit Corrmotif Model (K = 1 to 8)
set.seed(11111)
motif.fitted_0.5 <- cormotiffit(
  exprs = exprs.corrmotif,
  groupid = groupid_df,
  compid = compid_df,
  K = 1:8,
  max.iter = 1000,
  BIC = TRUE,
  runtype = "logCPM"
)

gene_prob_0.5 <- motif.fitted_0.5$bestmotif$p.post
rownames(gene_prob_0.5) <- rownames(Corrmotif_df)
motif_prob_0.5 <- motif.fitted_0.5$bestmotif$clustlike
rownames(motif_prob_0.5) <- rownames(gene_prob_0.5)
write.csv(motif_prob_0.5,"data/cormotif_probability_genelist_0.5.csv")
```

## **ðŸ“Œ Plot motif (0.5 Micromolar)**
```{r Motif_0.5, echo=TRUE, message=FALSE}
cormotif_0.5 <- readRDS("data/Corrmotif/cormotif_0.5.RDS")

cormotif_0.5$bic

plotIC(cormotif_0.5)

plotMotif(cormotif_0.5)

```

## **ðŸ“Œ Extract Gene Probabilities (0.5 Micromolar)**
```{r Prob_0.5, echo=TRUE, message=FALSE}
# Extract posterior probabilities for genes
gene_prob_tran_0.5 <- cormotif_0.5$bestmotif$p.post

rownames(gene_prob_tran_0.5) <- rownames(Corrmotif_df)

# Define gene probability groups
prob_1_0.5  <- rownames(gene_prob_tran_0.5[(gene_prob_tran_0.5[,1] <0.5 & gene_prob_tran_0.5[,2] <0.5 & gene_prob_tran_0.5[,3] <0.5 & gene_prob_tran_0.5[,4] <0.5 & gene_prob_tran_0.5[,5] < 0.5 & gene_prob_tran_0.5[,6]<0.5),])
length(prob_1_0.5)

prob_2_0.5  <- rownames(gene_prob_tran_0.5[(gene_prob_tran_0.5[,1] <0.5 & gene_prob_tran_0.5[,2] <0.5 & gene_prob_tran_0.5[,3] <0.5 & gene_prob_tran_0.5[,4] >0.5 & gene_prob_tran_0.5[,5] > 0.5 & gene_prob_tran_0.5[,6]>=0.02),])
length(prob_2_0.5)

prob_3_0.5  <- rownames(gene_prob_tran_0.5[(gene_prob_tran_0.5[,1] <0.5 & gene_prob_tran_0.5[,2] <0.5 & gene_prob_tran_0.5[,3] <0.5 & gene_prob_tran_0.5[,4] <0.5 & gene_prob_tran_0.5[,5] > 0.5 & gene_prob_tran_0.5[,6]>0.5),])
length(prob_3_0.5)

prob_4_0.5  <- rownames(gene_prob_tran_0.5[(gene_prob_tran_0.5[,1] >= 0.1 & gene_prob_tran_0.5[,2] > 0.5 & gene_prob_tran_0.5[,3] > 0.5 & gene_prob_tran_0.5[,4] >= 0.02 & gene_prob_tran_0.5[,5] < 0.5 & gene_prob_tran_0.5[,6] < 0.5),]) 

length(prob_4_0.5)

prob_5_0.5  <- rownames(gene_prob_tran_0.5[(gene_prob_tran_0.5[,1] <0.5 & gene_prob_tran_0.5[,2] >0.5 & gene_prob_tran_0.5[,3] >0.5 & gene_prob_tran_0.5[,4] >=0.02 & gene_prob_tran_0.5[,4] <0.5 & gene_prob_tran_0.5[,5] > 0.5 & gene_prob_tran_0.5[,6]>0.5),])
length(prob_5_0.5)
```
## **ðŸ“Œ Distribution of Gene Clusters Identified by Corrmotif (0.5 micromolar)**
```{r pie-chart_0.5, echo=TRUE, message=FALSE, warning=FALSE}

# Load necessary library
library(ggplot2)

# Data
data <- data.frame(
  Category = c("Non response (0.5)", "DOX-specific response (0.5 ÂµM)", "DOX only mid-late response (0.5 ÂµM)", "CX + DOX (early) response (0.5 ÂµM)", "DOX + CX (mid-late) response (0.5 ÂµM)"),
  Value = c(7134,179,6450,142,221)
)

# Add values to category names (to be displayed in the legend)
data$Category <- paste0(data$Category, " (", data$Value, ")")

# Define custom colors with updated category names
custom_colors <- setNames(
  c("#FF9999", "#FF66CC", "#66B2FF", "#99FF99", "#FFD700"),
  data$Category  # Ensures color names match updated categories
)

# Create pie chart without number labels inside
ggplot(data, aes(x = "", y = Value, fill = Category)) +
  geom_bar(width = 1, stat = "identity") +
  coord_polar("y", start = 0) +
  labs(title = "Pie Chart (0.5 micromolar Corrmotif)", x = NULL, y = NULL) +
  theme_void() +
  scale_fill_manual(values = custom_colors)
```

## **ðŸ“Œ Combined pie charts for concentrations**
```{r pie-chart_comb, echo=TRUE, message=FALSE, warning=FALSE}
# Load necessary libraries
library(ggplot2)
library(dplyr)

# Data for 0.1 ÂµM
data_0.1 <- data.frame(
  Category = c("Non response", "CX-DOX mid-late response", "DOX only mid-late"),
  Value = c(12308, 415, 1551),
  Concentration = "0.1 ÂµM"
)

# Data for 0.5 ÂµM
data_0.5 <- data.frame(
  Category = c("Non response", "DOX specific response", "DOX only mid-late response", 
               "CX + DOX (early) response", "DOX + CX (mid-late) response"),
  Value = c(7134, 179, 6450, 142, 221),
  Concentration = "0.5 ÂµM"
)

# Combine both datasets
combined_data <- bind_rows(data_0.1, data_0.5)

# Add values to category names (for legend only)
combined_data$Category_Legend <- paste0(combined_data$Category, " (", combined_data$Value, ")")

# Define custom colors for updated categories
custom_colors <- c(
  "Non response (12308)" = "#FF9999",
  "CX-DOX mid-late response (415)" = "#66B2FF",
  "DOX only mid-late (1551)" = "#99FF99",
  "Non response (7134)" = "#FF9999",
  "DOX specific response (179)" = "#FF66CC",
  "DOX only mid-late response (6450)" = "#99FF99",
  "CX + DOX (early) response (142)" = "#FFD700",
  "DOX + CX (mid-late) response (221)" = "#8A2BE2"
)

# Ensure categories appear in a specific order
combined_data$Category_Legend <- factor(combined_data$Category_Legend, levels = names(custom_colors))

# Create faceted pie charts without numbers inside the slices
pie_chart <- ggplot(combined_data, aes(x = "", y = Value, fill = Category_Legend)) +
  geom_bar(width = 1, stat = "identity") +
  coord_polar("y", start = 0) +
  facet_wrap(~ Concentration) +  # Facet by concentration (side-by-side)
  labs(title = "Corrmotif Pie Charts for 0.1 ÂµM and 0.5 ÂµM", x = NULL, y = NULL, fill = "Category") +
  theme_void() +
  theme(
    panel.border = element_rect(color = "black", fill = NA, linewidth = 1.5),  # Adds box around facets
    strip.background = element_rect(fill = "white", color = "black", linewidth = 1),  # Box for facet titles
    strip.text = element_text(size = 12, face = "bold", color = "black"),
    legend.title = element_text(size = 12, face = "bold"),  # Bold legend title
    legend.text = element_text(size = 10)  # Adjust legend text size
  ) +
  scale_fill_manual(values = custom_colors)

# Display the plot
print(pie_chart)
```


## **ðŸ“Œ Save Corr-motif datasets for Gene Ontology analysis (0.5 Micromolar)**
```{r box_0.5, echo=TRUE, results='hide',message=FALSE, warning=FALSE}
write.csv(data.frame(Entrez_ID = prob_1_0.5), "data/prob_1_0.5.csv", row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_2_0.5), "data/prob_2_0.5.csv", row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_3_0.5), "data/prob_3_0.5.csv", row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_4_0.5), "data/prob_4_0.5.csv", row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_5_0.5), "data/prob_5_0.5.csv", row.names = FALSE)
```

## **ðŸ“Œ 0.5 Micromolar abs logFC**
```{r 0.5abs logFC, echo=TRUE, message=FALSE, warning=FALSE, fig.width=12,fig.height=12}
# Load Response Groups from CSV Files
prob_1_0.5 <- as.character(read.csv("data/prob_1_0.5.csv")$Entrez_ID)
prob_2_0.5 <- as.character(read.csv("data/prob_2_0.5.csv")$Entrez_ID)
prob_3_0.5 <- as.character(read.csv("data/prob_3_0.5.csv")$Entrez_ID)
prob_4_0.5 <- as.character(read.csv("data/prob_4_0.5.csv")$Entrez_ID)
prob_5_0.5 <- as.character(read.csv("data/prob_5_0.5.csv")$Entrez_ID)

# Load Datasets (Only 0.5 Micromolar)
CX_0.5_3  <- read.csv("data/DEGs/Toptable_CX_0.5_3.csv")
CX_0.5_24 <- read.csv("data/DEGs/Toptable_CX_0.5_24.csv")
CX_0.5_48 <- read.csv("data/DEGs/Toptable_CX_0.5_48.csv")

DOX_0.5_3  <- read.csv("data/DEGs/Toptable_DOX_0.5_3.csv")
DOX_0.5_24 <- read.csv("data/DEGs/Toptable_DOX_0.5_24.csv")
DOX_0.5_48 <- read.csv("data/DEGs/Toptable_DOX_0.5_48.csv")

# Convert datasets to DataFrames
Toptable_CX_0.5_3_df  <- data.frame(CX_0.5_3)
Toptable_CX_0.5_24_df <- data.frame(CX_0.5_24)
Toptable_CX_0.5_48_df <- data.frame(CX_0.5_48)

Toptable_DOX_0.5_3_df  <- data.frame(DOX_0.5_3)
Toptable_DOX_0.5_24_df <- data.frame(DOX_0.5_24)
Toptable_DOX_0.5_48_df <- data.frame(DOX_0.5_48)

# Combine All 0.5 Micromolar Datasets into a Single Dataframe
all_toptables_0.5 <- bind_rows(
  Toptable_CX_0.5_3_df  %>% mutate(Drug = "CX.5461", Timepoint = "3"),
  Toptable_CX_0.5_24_df %>% mutate(Drug = "CX.5461", Timepoint = "24"),
  Toptable_CX_0.5_48_df %>% mutate(Drug = "CX.5461", Timepoint = "48"),
  Toptable_DOX_0.5_3_df  %>% mutate(Drug = "DOX", Timepoint = "3"),
  Toptable_DOX_0.5_24_df %>% mutate(Drug = "DOX", Timepoint = "24"),
  Toptable_DOX_0.5_48_df %>% mutate(Drug = "DOX", Timepoint = "48")
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.5$Entrez_ID <- as.character(all_toptables_0.5$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.5 <- all_toptables_0.5 %>%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ "Non response\n(0.5 micromolar)",
      Entrez_ID %in% prob_2_0.5 ~ "DOX-specific response\n(0.5 micromolar)",
      Entrez_ID %in% prob_3_0.5 ~ "DOX only mid-late response\n(0.5 micromolar)",
      Entrez_ID %in% prob_4_0.5 ~ "CX + DOX (early) response\n(0.5 micromolar)",
      Entrez_ID %in% prob_5_0.5 ~ "DOX + CX (mid-late) response\n(0.5 micromolar)",
      TRUE ~ NA_character_
    )
  )

# Remove NA Values (Genes Not in Response Groups)
all_toptables_0.5 <- all_toptables_0.5 %>% filter(!is.na(Response_Group))

# Compute Absolute logFC
all_toptables_0.5 <- all_toptables_0.5 %>%
  mutate(absFC = abs(logFC))

# Convert Factors for Proper Ordering (Reversed Order for Response Groups)
all_toptables_0.5 <- all_toptables_0.5 %>%
  mutate(
    Drug = factor(Drug, levels = c("CX.5461", "DOX")),
    Timepoint = factor(Timepoint, levels = c("3", "24", "48"),
                       labels = c("Timepoint: 3 hours", "Timepoint: 24 hours", "Timepoint: 48 hours")),
    Response_Group = factor(Response_Group,
                            levels = c("DOX + CX (mid-late) response\n(0.5 micromolar)",
                                       "CX + DOX (early) response\n(0.5 micromolar)",
                                       "DOX only mid-late response\n(0.5 micromolar)",
                                       "DOX-specific response\n(0.5 micromolar)",
                                       "Non response\n(0.5 micromolar)")) # Reversed Order
  )

# **Plot the Boxplot with Faceted Labels Wrapping Correctly**
ggplot(all_toptables_0.5, aes(x = Drug, y = absFC, fill = Drug)) +
  geom_boxplot() +
  scale_fill_manual(values = c("CX.5461" = "blue", "DOX" = "red")) +  # Custom color palette
  facet_grid(Response_Group ~ Timepoint, labeller = label_wrap_gen(width = 20)) +  # Ensure Proper Wrapping
  theme_bw() +
  labs(
    x = "Drugs",
    y = "|Log Fold Change|",
    title = "|Log Fold| for 0.5 micromolar"
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = "black"),
    axis.line = element_line(linewidth = 1.5),
    strip.background = element_rect(fill = "gray"),  # Gray background for facet labels
    strip.text = element_text(size = 12, color = "black", face = "bold"),  # Bold styling for facet labels
    axis.text.x = element_text(size = 10, color = "black", angle = 15)
  )
```

## **ðŸ“Œ 0.5 Micromolar mean (Abs logFC) across timepoints**
```{r 0.5 mean, echo=TRUE, message=FALSE, warning=FALSE, fig.height=12, fig.width=10}

# Load required libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.5 <- as.character(read.csv("data/prob_1_0.5.csv")$Entrez_ID)
prob_2_0.5 <- as.character(read.csv("data/prob_2_0.5.csv")$Entrez_ID)
prob_3_0.5 <- as.character(read.csv("data/prob_3_0.5.csv")$Entrez_ID)
prob_4_0.5 <- as.character(read.csv("data/prob_4_0.5.csv")$Entrez_ID)
prob_5_0.5 <- as.character(read.csv("data/prob_5_0.5.csv")$Entrez_ID)

# Load Datasets (Only 0.5 Micromolar)
CX_0.5_3  <- read.csv("data/DEGs/Toptable_CX_0.5_3.csv")
CX_0.5_24 <- read.csv("data/DEGs/Toptable_CX_0.5_24.csv")
CX_0.5_48 <- read.csv("data/DEGs/Toptable_CX_0.5_48.csv")

DOX_0.5_3  <- read.csv("data/DEGs/Toptable_DOX_0.5_3.csv")
DOX_0.5_24 <- read.csv("data/DEGs/Toptable_DOX_0.5_24.csv")
DOX_0.5_48 <- read.csv("data/DEGs/Toptable_DOX_0.5_48.csv")

# Combine All 0.5 Micromolar Datasets into a Single Dataframe
all_toptables_0.5 <- bind_rows(
  CX_0.5_3  %>% mutate(Drug = "CX.5461", Timepoint = "3"),
  CX_0.5_24 %>% mutate(Drug = "CX.5461", Timepoint = "24"),
  CX_0.5_48 %>% mutate(Drug = "CX.5461", Timepoint = "48"),
  DOX_0.5_3  %>% mutate(Drug = "DOX", Timepoint = "3"),
  DOX_0.5_24 %>% mutate(Drug = "DOX", Timepoint = "24"),
  DOX_0.5_48 %>% mutate(Drug = "DOX", Timepoint = "48")
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.5$Entrez_ID <- as.character(all_toptables_0.5$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.5 <- all_toptables_0.5 %>%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ "Non response\n(0.5 micromolar)",
      Entrez_ID %in% prob_2_0.5 ~ "DOX-specific response\n(0.5 micromolar)",
      Entrez_ID %in% prob_3_0.5 ~ "DOX only mid-late response\n(0.5 micromolar)",
      Entrez_ID %in% prob_4_0.5 ~ "CX + DOX (early) response\n(0.5 micromolar)",
      Entrez_ID %in% prob_5_0.5 ~ "DOX + CX (mid-late) response\n(0.5 micromolar)",
      TRUE ~ NA_character_
    )
  ) %>%
  filter(!is.na(Response_Group))  # Remove NA values

# Compute Mean Absolute logFC for Line Plot
data_summary <- all_toptables_0.5 %>%
  mutate(abs_logFC = abs(logFC)) %>%
  group_by(Response_Group, Drug, Timepoint) %>%
  dplyr::summarize(mean_abs_logFC = mean(abs_logFC, na.rm = TRUE), .groups = "drop") %>%
  as.data.frame()

# **Ensure all timepoints exist in the summary**
timepoints_full <- expand.grid(
  Response_Group = unique(all_toptables_0.5$Response_Group),
  Drug = unique(all_toptables_0.5$Drug),
  Timepoint = c("3", "24", "48")
)

# **Merge to keep missing timepoints**
data_summary <- full_join(timepoints_full, data_summary, by = c("Response_Group", "Drug", "Timepoint"))

# **Replace NA mean_abs_logFC with 0 if no genes were present**
data_summary$mean_abs_logFC[is.na(data_summary$mean_abs_logFC)] <- 0

# Convert Factors for Proper Ordering (Reversed Order for Response Groups)
data_summary <- data_summary %>%
  mutate(
    Timepoint = factor(Timepoint, levels = c("3", "24", "48"), labels = c("3 hours", "24 hours", "48 hours")),
    Response_Group = factor(Response_Group, levels = c(
      "DOX + CX (mid-late) response\n(0.5 micromolar)",
                                       "CX + DOX (early) response\n(0.5 micromolar)",
                                       "DOX only mid-late response\n(0.5 micromolar)",
                                       "DOX-specific response\n(0.5 micromolar)",
                                       "Non response\n(0.5 micromolar)"  # Reversed order
    ))
  )

# Define custom drug palette
drug_palette <- c("CX.5461" = "blue", "DOX" = "red")

# **Plot the Line Plot for Mean Absolute logFC**
ggplot(data_summary, aes(x = Timepoint, y = mean_abs_logFC, group = Drug, color = Drug)) +
  geom_point(size = 3) +
  geom_line(size = 1.2) +
  scale_color_manual(values = drug_palette) +
  ylim(0, 2.5) +  # Adjust the Y-axis for better visualization
  facet_wrap(~ Response_Group, ncol = 1) +  # Facet by Response Group (Reversed Order)
  theme_bw() +
  labs(
    x = "Timepoints",
    y = "Mean |Log Fold Change|",
    title = "Mean |Log Fold Change| Across Response Groups (0.5 micromolar)",
    color = "Drug"
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = "black"),
    axis.text = element_text(size = 12, color = "black"),
    strip.text = element_text(size = 12, color = "black", face = "bold"),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )
```

## **ðŸ“Œ 0.5 Micromolar logFC**
```{r 0.5logFC, echo=TRUE, message=FALSE, warning=FALSE, fig.width=12,fig.height=14}
# Load required libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.5 <- as.character(read.csv("data/prob_1_0.5.csv")$Entrez_ID)
prob_2_0.5 <- as.character(read.csv("data/prob_2_0.5.csv")$Entrez_ID)
prob_3_0.5 <- as.character(read.csv("data/prob_3_0.5.csv")$Entrez_ID)
prob_4_0.5 <- as.character(read.csv("data/prob_4_0.5.csv")$Entrez_ID)
prob_5_0.5 <- as.character(read.csv("data/prob_5_0.5.csv")$Entrez_ID)

# Load Datasets (Only 0.5 Micromolar)
CX_0.5_3  <- read.csv("data/DEGs/Toptable_CX_0.5_3.csv")
CX_0.5_24 <- read.csv("data/DEGs/Toptable_CX_0.5_24.csv")
CX_0.5_48 <- read.csv("data/DEGs/Toptable_CX_0.5_48.csv")

DOX_0.5_3  <- read.csv("data/DEGs/Toptable_DOX_0.5_3.csv")
DOX_0.5_24 <- read.csv("data/DEGs/Toptable_DOX_0.5_24.csv")
DOX_0.5_48 <- read.csv("data/DEGs/Toptable_DOX_0.5_48.csv")

# Combine All 0.5 Micromolar Datasets into a Single Dataframe
all_toptables_0.5 <- bind_rows(
  CX_0.5_3  %>% mutate(Drug = "CX.5461", Timepoint = "Timepoint: 3 hours"),
  CX_0.5_24 %>% mutate(Drug = "CX.5461", Timepoint = "Timepoint: 24 hours"),
  CX_0.5_48 %>% mutate(Drug = "CX.5461", Timepoint = "Timepoint: 48 hours"),
  DOX_0.5_3  %>% mutate(Drug = "DOX", Timepoint = "Timepoint: 3 hours"),
  DOX_0.5_24 %>% mutate(Drug = "DOX", Timepoint = "Timepoint: 24 hours"),
  DOX_0.5_48 %>% mutate(Drug = "DOX", Timepoint = "Timepoint: 48 hours")
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.5$Entrez_ID <- as.character(all_toptables_0.5$Entrez_ID)

# Assign Response Groups
all_toptables_0.5 <- all_toptables_0.5 %>%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ "Non response\n(0.5 micromolar)",
      Entrez_ID %in% prob_2_0.5 ~ "DOX-specific response\n(0.5 micromolar)",
      Entrez_ID %in% prob_3_0.5 ~ "DOX only mid-late response\n(0.5 micromolar)",
      Entrez_ID %in% prob_4_0.5 ~ "CX + DOX (early) response\n(0.5 micromolar)",
      Entrez_ID %in% prob_5_0.5 ~ "DOX + CX (mid-late) response\n(0.5 micromolar)",
      TRUE ~ NA_character_
    )
  ) %>%
  filter(!is.na(Response_Group))

# Convert factors to ensure correct ordering (Reversed Order for Response Groups)
all_toptables_0.5 <- all_toptables_0.5 %>%
  mutate(
    Timepoint = factor(Timepoint, levels = c("Timepoint: 3 hours", "Timepoint: 24 hours", "Timepoint: 48 hours")),
    Response_Group = factor(Response_Group, levels = c(
      "DOX + CX (mid-late) response\n(0.5 micromolar)",
                                       "CX + DOX (early) response\n(0.5 micromolar)",
                                       "DOX only mid-late response\n(0.5 micromolar)",
                                       "DOX-specific response\n(0.5 micromolar)",
                                       "Non response\n(0.5 micromolar)"  # Reversed Order
    ))
  )

# **Plot the Boxplot**
ggplot(all_toptables_0.5, aes(x = Drug, y = logFC, fill = Drug)) +
  geom_boxplot() +
  scale_fill_manual(values = c("CX.5461" = "blue", "DOX" = "red")) +
  facet_grid(Response_Group ~ Timepoint) +
  theme_bw() +
  labs(x = "Drugs", y = "Log Fold Change", title = "Log Fold Change for 0.5 Micromolar") +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    strip.text = element_text(size = 12, face = "bold")
  )
```

## **ðŸ“Œ 0.5 Micromolar mean logFC across timepoints**
```{r 0.5 mean2, echo=TRUE, message=FALSE, warning=FALSE, fig.height=12, fig.width=10}

# Load Required Libraries
library(dplyr)
library(ggplot2)

# Load Response Groups from CSV Files
prob_1_0.5 <- as.character(read.csv("data/prob_1_0.5.csv")$Entrez_ID)
prob_2_0.5 <- as.character(read.csv("data/prob_2_0.5.csv")$Entrez_ID)
prob_3_0.5 <- as.character(read.csv("data/prob_3_0.5.csv")$Entrez_ID)
prob_4_0.5 <- as.character(read.csv("data/prob_4_0.5.csv")$Entrez_ID)
prob_5_0.5 <- as.character(read.csv("data/prob_5_0.5.csv")$Entrez_ID)

# Load Datasets (Only 0.5 Micromolar)
CX_0.5_3  <- read.csv("data/DEGs/Toptable_CX_0.5_3.csv")
CX_0.5_24 <- read.csv("data/DEGs/Toptable_CX_0.5_24.csv")
CX_0.5_48 <- read.csv("data/DEGs/Toptable_CX_0.5_48.csv")

DOX_0.5_3  <- read.csv("data/DEGs/Toptable_DOX_0.5_3.csv")
DOX_0.5_24 <- read.csv("data/DEGs/Toptable_DOX_0.5_24.csv")
DOX_0.5_48 <- read.csv("data/DEGs/Toptable_DOX_0.5_48.csv")

# Combine All 0.5 Micromolar Datasets into a Single Dataframe
all_toptables_0.5 <- bind_rows(
  CX_0.5_3  %>% mutate(Drug = "CX.5461", Timepoint = "Timepoint: 3 hours"),
  CX_0.5_24 %>% mutate(Drug = "CX.5461", Timepoint = "Timepoint: 24 hours"),
  CX_0.5_48 %>% mutate(Drug = "CX.5461", Timepoint = "Timepoint: 48 hours"),
  DOX_0.5_3  %>% mutate(Drug = "DOX", Timepoint = "Timepoint: 3 hours"),
  DOX_0.5_24 %>% mutate(Drug = "DOX", Timepoint = "Timepoint: 24 hours"),
  DOX_0.5_48 %>% mutate(Drug = "DOX", Timepoint = "Timepoint: 48 hours")
)

# Convert `Entrez_ID` to Character to Avoid `%in%` Issues
all_toptables_0.5$Entrez_ID <- as.character(all_toptables_0.5$Entrez_ID)

# Assign Response Groups with Line Breaks for Better Plotting
all_toptables_0.5 <- all_toptables_0.5 %>%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ "Non response\n(0.5 micromolar)",
      Entrez_ID %in% prob_2_0.5 ~ "DOX-specific response\n(0.5 micromolar)",
      Entrez_ID %in% prob_3_0.5 ~ "DOX only mid-late response\n(0.5 micromolar)",
      Entrez_ID %in% prob_4_0.5 ~ "CX + DOX (early) response\n(0.5 micromolar)",
      Entrez_ID %in% prob_5_0.5 ~ "DOX + CX (mid-late) response\n(0.5 micromolar)",
      TRUE ~ NA_character_
    )
  ) %>%
  filter(!is.na(Response_Group))

# Compute Mean logFC for Line Plot
data_summary_0.5 <- all_toptables_0.5 %>%
  group_by(Response_Group, Drug, Timepoint) %>%
  dplyr::summarize(mean_logFC = mean(logFC, na.rm = TRUE), .groups = "drop") %>%
  as.data.frame()

# **Ensure all timepoints exist in the summary**
timepoints_full <- expand.grid(
  Response_Group = unique(all_toptables_0.5$Response_Group),
  Drug = unique(all_toptables_0.5$Drug),
  Timepoint = c("Timepoint: 3 hours", "Timepoint: 24 hours", "Timepoint: 48 hours")
)

# **Merge to keep missing timepoints**
data_summary_0.5 <- full_join(timepoints_full, data_summary_0.5, by = c("Response_Group", "Drug", "Timepoint"))

# **Replace NA mean_logFC with 0 if no genes were present**
data_summary_0.5$mean_logFC[is.na(data_summary_0.5$mean_logFC)] <- 0

# Convert Factors for Proper Ordering (Reversed Order for Response Groups)
data_summary_0.5 <- data_summary_0.5 %>%
  mutate(
    Timepoint = factor(Timepoint, levels = c("Timepoint: 3 hours", "Timepoint: 24 hours", "Timepoint: 48 hours")),
    Response_Group = factor(Response_Group, levels = c(
      "DOX + CX (mid-late) response\n(0.5 micromolar)",
                                       "CX + DOX (early) response\n(0.5 micromolar)",
                                       "DOX only mid-late response\n(0.5 micromolar)",
                                       "DOX-specific response\n(0.5 micromolar)",
                                       "Non response\n(0.5 micromolar)"  # Reversed Order
    ))
  )

# Define custom drug palette
drug_palette <- c("CX.5461" = "blue", "DOX" = "red")

# **Plot the Line Plot for Mean logFC**
ggplot(data_summary_0.5, aes(x = Timepoint, y = mean_logFC, group = Drug, color = Drug)) +
  geom_point(size = 3) +
  geom_line(size = 1.2) +
  scale_color_manual(values = drug_palette) +
  ylim(-2, 1.5) +  # Adjust the Y-axis for better visualization
  facet_wrap(~ Response_Group, ncol = 1) +  # Facet by Response Group (Reversed Order)
  theme_bw() +
  labs(
    x = "Timepoints",
    y = "Mean Log Fold Change",
    title = "Mean Log Fold Change Across Response Groups (0.5 micromolar)",
    color = "Drug"
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = "black"),
    axis.text = element_text(size = 12, color = "black"),
    strip.text = element_text(size = 12, color = "black", face = "bold"),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )
```

## **ðŸ“Œ Proportion of DNA Damage Repair Genes (0.1 micromolar)**
```{r prop DNA, echo=TRUE, message=FALSE}
# Load necessary libraries
library(dplyr)
library(ggplot2)
library(tidyr)
library(org.Hs.eg.db)

# **ðŸ”¹ Read DNA Damage Repair Gene List**
DNA_damage <- read.csv("data/DNA_Damage.csv", stringsAsFactors = FALSE)

# Convert gene symbols to Entrez IDs
DNA_damage <- DNA_damage %>%
  mutate(Entrez_ID = mapIds(org.Hs.eg.db,
                            keys = DNA_damage$Symbol,
                            column = "ENTREZID",
                            keytype = "SYMBOL",
                            multiVals = "first"))

DNA_damage_genes <- na.omit(DNA_damage$Entrez_ID)

# **ðŸ”¹ Load Corrmotif Groups for 0.1 Concentration**
prob_groups_0.1 <- list(
  "Non Response (0.1)" = read.csv("data/prob_1_0.1.csv")$Entrez_ID,
   "CX_DOX mid-late (0.1)" = read.csv("data/prob_2_0.1.csv")$Entrez_ID,
  "DOX only mid-late (0.1)"= read.csv("data/prob_3_0.1.csv")$Entrez_ID
)

# **ðŸ”¹ Create Dataframe for Corrmotif Groups**
corrmotif_df_0.1 <- bind_rows(
  lapply(prob_groups_0.1, function(ids) {
    data.frame(Entrez_ID = ids)
  }),
  .id = "Response_Group"
)

# **ðŸ”¹ Match Entrez_IDs with DNA Damage Repair Genes**
corrmotif_df_0.1 <- corrmotif_df_0.1 %>%
  mutate(DNA_Damage = ifelse(Entrez_ID %in% DNA_damage_genes, "Yes", "No"))

# **ðŸ”¹ Count DNA Damage Repair Genes in Each Response Group**
proportion_data <- corrmotif_df_0.1 %>%
  group_by(Response_Group, DNA_Damage) %>%
  summarise(Count = n(), .groups = "drop") %>%
  group_by(Response_Group) %>%
  mutate(Percentage = (Count / sum(Count)) * 100)

# **ðŸ”¹ Ensure "Yes" is at the Bottom and "No" at the Top**
proportion_data$DNA_Damage <- factor(proportion_data$DNA_Damage, levels = c("Yes", "No"))

# **ðŸ”¹ Set Order of Response Groups for X-axis**
response_order <- c("Non Response (0.1)", "CX_DOX mid-late (0.1)","DOX only mid-late (0.1)")
proportion_data$Response_Group <- factor(proportion_data$Response_Group, levels = response_order)

# **ðŸ”¹ Perform Chi-Square Tests for "DOX only mid-late (0.1)" and "CX_DOX mid-late (0.1)" vs "Non Response (0.1)"**
non_response_counts <- proportion_data %>%
  filter(Response_Group == "Non Response (0.1)") %>%
  dplyr::select(DNA_Damage, Count) %>%
  {setNames(.$Count, .$DNA_Damage)}  # Convert to named vector

chi_results <- proportion_data %>%
  filter(Response_Group %in% c("CX_DOX mid-late (0.1)","DOX only mid-late (0.1)")) %>%
  group_by(Response_Group) %>%
  summarise(
    p_value = {
      group_counts <- Count[DNA_Damage %in% c("Yes", "No")]
      if (!"Yes" %in% DNA_Damage) group_counts <- c(group_counts, 0)
      if (!"No" %in% DNA_Damage) group_counts <- c(0, group_counts)

      contingency_table <- matrix(c(
        group_counts[1], group_counts[2],
        non_response_counts["Yes"], non_response_counts["No"]
      ), nrow = 2, byrow = TRUE)

      # Perform chi-square test if all values are valid
      if (all(contingency_table >= 0 & is.finite(contingency_table))) {
        chisq.test(contingency_table)$p.value
      } else {
        NA
      }
    },
    .groups = "drop"
  ) %>%
  mutate(Significance = ifelse(!is.na(p_value) & p_value < 0.05, "*", ""))

# **ðŸ”¹ Merge Chi-Square Results into Proportion Data**
proportion_data <- proportion_data %>%
  left_join(chi_results %>% dplyr::select(Response_Group, Significance), by = "Response_Group")

# **ðŸ”¹ Set Star Position Uniform Across Groups at 105%**
star_positions <- data.frame(
  Response_Group = c("CX_DOX mid-late (0.1)", "DOX only mid-late (0.1)"),
  y_pos = 105,  # Fixed at 105% of Y-axis
  Significance = chi_results$Significance
)

# **ðŸ”¹ Generate Proportion Plot with Chi-Square Stars**
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = DNA_Damage)) +
  geom_bar(stat = "identity", position = "stack") +  # Stacked bars
  geom_text(
    data = star_positions,
    aes(x = Response_Group, y = y_pos, label = Significance),  # Place stars at fixed 105%
    inherit.aes = FALSE,
    size = 6, color = "black", fontface = "bold", vjust = 0  # Keeps stars aligned
  ) +
  scale_y_continuous(labels = scales::percent_format(scale = 1), limits = c(0, 110)) +  # Fixed Y-axis to 100%
  scale_fill_manual(values = c("Yes" = "#e41a1c", "No" = "#377eb8")) +  # Yes (Red), No (Blue)
  labs(
    title = "Proportion of DNA Damage Repair Genes in\n0.1 Corrmotif Response Groups",
    x = "Response Groups (0.1 Concentration)",
    y = "Percentage",
    fill = "DNA Damage Repair"
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = "black"),
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
    legend.title = element_blank(),
    panel.border = element_rect(color = "black", fill = NA, linewidth = 1.2),
    strip.background = element_blank(),
    strip.text = element_text(size = 12, face = "bold")
  )
```

## **ðŸ“Œ Proportion of DNA Damage Repair Genes (0.5 micromolar) **
```{r prop DNA 0.5, echo=TRUE, message=FALSE}
# Load necessary libraries
library(dplyr)
library(ggplot2)
library(tidyr)
library(org.Hs.eg.db)

# **ðŸ”¹ Read DNA Damage Repair Gene List**
DNA_damage <- read.csv("data/DNA_Damage.csv", stringsAsFactors = FALSE)

# Convert gene symbols to Entrez IDs
DNA_damage <- DNA_damage %>%
  mutate(Entrez_ID = mapIds(org.Hs.eg.db,
                            keys = DNA_damage$Symbol,
                            column = "ENTREZID",
                            keytype = "SYMBOL",
                            multiVals = "first"))

DNA_damage_genes <- na.omit(DNA_damage$Entrez_ID)

# **ðŸ”¹ Load Corrmotif Groups for 0.5 Concentration**
prob_groups_0.5 <- list(
  "Non Response (0.5)" = read.csv("data/prob_1_0.5.csv")$Entrez_ID,
  "DOX-specific response (0.5)" = read.csv("data/prob_2_0.5.csv")$Entrez_ID,
  "DOX only mid-late response (0.5)" = read.csv("data/prob_3_0.5.csv")$Entrez_ID,
  "CX DOX (early) response (0.5)" = read.csv("data/prob_4_0.5.csv")$Entrez_ID,
  "DOX + CX (mid-late) response (0.5)" = read.csv("data/prob_5_0.5.csv")$Entrez_ID
)

# **ðŸ”¹ Create Dataframe for Corrmotif Groups**
corrmotif_df_0.5 <- bind_rows(
  lapply(prob_groups_0.5, function(ids) {
    data.frame(Entrez_ID = ids)
  }),
  .id = "Response_Group"
)

# **ðŸ”¹ Match Entrez_IDs with DNA Damage Repair Genes**
corrmotif_df_0.5 <- corrmotif_df_0.5 %>%
  mutate(DNA_Damage = ifelse(Entrez_ID %in% DNA_damage_genes, "Yes", "No"))

# **ðŸ”¹ Count DNA Damage Repair Genes in Each Response Group**
proportion_data <- corrmotif_df_0.5 %>%
  group_by(Response_Group, DNA_Damage) %>%
  summarise(Count = n(), .groups = "drop") %>%
  group_by(Response_Group) %>%
  mutate(Percentage = (Count / sum(Count)) * 100)

# **ðŸ”¹ Ensure "Yes" is at the Bottom and "No" at the Top**
proportion_data$DNA_Damage <- factor(proportion_data$DNA_Damage, levels = c("Yes", "No"))

# **ðŸ”¹ Set Order of Response Groups for X-axis**
response_order <- c("Non Response (0.5)", "DOX-specific response (0.5)", "DOX only mid-late response (0.5)", 
                    "CX DOX (early) response (0.5)", "DOX + CX (mid-late) response (0.5)")
proportion_data$Response_Group <- factor(proportion_data$Response_Group, levels = response_order)

# **ðŸ”¹ Perform Chi-Square Tests for Each Response Group vs Non-Response**
non_response_counts <- proportion_data %>%
  filter(Response_Group == "Non Response (0.5)") %>%
  dplyr::select(DNA_Damage, Count) %>%
  {setNames(.$Count, .$DNA_Damage)}  # Convert to named vector

# **Comparing Each Group Against "Non Response (0.5)"**
chi_results <- proportion_data %>%
  filter(Response_Group %in% c("DOX-specific response (0.5)", "DOX only mid-late response (0.5)", 
                               "CX DOX (early) response (0.5)", "DOX + CX (mid-late) response (0.5)")) %>%
  group_by(Response_Group) %>%
  summarise(
    p_value = {
      group_counts <- Count[DNA_Damage %in% c("Yes", "No")]
      if (!"Yes" %in% DNA_Damage) group_counts <- c(group_counts, 0)
      if (!"No" %in% DNA_Damage) group_counts <- c(0, group_counts)

      contingency_table <- matrix(c(
        group_counts[1], group_counts[2],  # Response group counts
        non_response_counts["Yes"], non_response_counts["No"]  # Non-response counts
      ), nrow = 2, byrow = TRUE)

      # Perform chi-square test if all values are valid
      if (all(contingency_table >= 0 & is.finite(contingency_table))) {
        chisq.test(contingency_table)$p.value
      } else {
        NA
      }
    },
    .groups = "drop"
  ) %>%
  mutate(Significance = ifelse(!is.na(p_value) & p_value < 0.05, "*", ""))

# **ðŸ”¹ Merge Chi-Square Results into Proportion Data**
proportion_data <- proportion_data %>%
  left_join(chi_results %>% dplyr::select(Response_Group, Significance), by = "Response_Group")

# **ðŸ”¹ Set Star Position Uniform Across Groups at 105%**
star_positions <- data.frame(
  Response_Group = c("DOX-specific response (0.5)", "DOX only mid-late response (0.5)", 
                     "CX DOX (early) response (0.5)", "DOX + CX (mid-late) response (0.5)"),
  y_pos = 105,  # Fixed at 105% of Y-axis
  Significance = chi_results$Significance
)

# **ðŸ”¹ Generate Proportion Plot with Chi-Square Stars**
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = DNA_Damage)) +
  geom_bar(stat = "identity", position = "stack") +  # Stacked bars
  geom_text(
    data = star_positions,
    aes(x = Response_Group, y = y_pos, label = Significance),  # Place stars at fixed 105%
    inherit.aes = FALSE,
    size = 6, color = "black", fontface = "bold", vjust = 0  # Keeps stars aligned
  ) +
  scale_y_continuous(labels = scales::percent_format(scale = 1), limits = c(0, 110)) +  # **Y-axis now limited to 110% for visibility**
  scale_fill_manual(values = c("Yes" = "#e41a1c", "No" = "#377eb8")) +  # Yes (Red), No (Blue)
  labs(
    title = "Proportion of DNA Damage Repair Genes in\n0.5 Corrmotif Response Groups",
    x = "Response Groups (0.5 Concentration)",
    y = "Percentage",
    fill = "DNA Damage Repair Genes"
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = "black"),
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
    legend.title = element_blank(),
    panel.border = element_rect(color = "black", fill = NA, linewidth = 1.2),
    strip.background = element_blank(),
    strip.text = element_text(size = 12, face = "bold")
  )
```

## **ðŸ“Œ Proportion of DNA Damage Repair Genes (Combined) **
```{r prop DNA combined, echo=TRUE, message=FALSE, , fig.width=10, fig.height=6}
# Load Required Libraries
library(dplyr)
library(ggplot2)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load DNA Damage Genes -----------------
DNA_damage <- read.csv("data/DNA_Damage.csv", stringsAsFactors = FALSE)

DNA_damage$Entrez_ID <- mapIds(
  org.Hs.eg.db,
  keys = DNA_damage$Symbol,
  column = "ENTREZID",
  keytype = "SYMBOL",
  multiVals = "first"
)
dna_damage_ids <- na.omit(DNA_damage$Entrez_ID)

# ----------------- Load CorrMotif Groups -----------------
# 0.1 ÂµM
prob_1_0.1 <- as.character(read.csv("data/prob_1_0.1.csv")$Entrez_ID)
prob_2_0.1 <- as.character(read.csv("data/prob_2_0.1.csv")$Entrez_ID)
prob_3_0.1 <- as.character(read.csv("data/prob_3_0.1.csv")$Entrez_ID)

# 0.5 ÂµM
prob_1_0.5 <- as.character(read.csv("data/prob_1_0.5.csv")$Entrez_ID)
prob_2_0.5 <- as.character(read.csv("data/prob_2_0.5.csv")$Entrez_ID)
prob_3_0.5 <- as.character(read.csv("data/prob_3_0.5.csv")$Entrez_ID)
prob_4_0.5 <- as.character(read.csv("data/prob_4_0.5.csv")$Entrez_ID)
prob_5_0.5 <- as.character(read.csv("data/prob_5_0.5.csv")$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 <- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %>%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ "Non response",
      Entrez_ID %in% prob_2_0.1 ~ "CX-DOX mid-late response",
      Entrez_ID %in% prob_3_0.1 ~ "DOX only mid-late response"
    ),
    Category = ifelse(Entrez_ID %in% dna_damage_ids, "Yes", "No"),
    Concentration = "0.1 ÂµM"
  )

df_0.5 <- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %>%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ "Non response",
      Entrez_ID %in% prob_2_0.5 ~ "DOX-specific response",
      Entrez_ID %in% prob_3_0.5 ~ "DOX only mid-late response",
      Entrez_ID %in% prob_4_0.5 ~ "CX total + DOX early response",
      Entrez_ID %in% prob_5_0.5 ~ "DOX early + CX-DOX mid-late response"
    ),
    Category = ifelse(Entrez_ID %in% dna_damage_ids, "Yes", "No"),
    Concentration = "0.5 ÂµM"
  )

# ----------------- Combine All -----------------
df_combined <- bind_rows(df_0.1, df_0.5)

# ----------------- Calculate Proportions -----------------
proportion_data <- df_combined %>%
  group_by(Concentration, Response_Group, Category) %>%
  summarise(Count = n(), .groups = "drop") %>%
  group_by(Concentration, Response_Group) %>%
  mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Set Order: "Yes" at bottom -----------------
proportion_data$Category <- factor(proportion_data$Category, levels = c("Yes", "No"))

# ----------------- Chi-square or Fisher's Test -----------------
get_chi_or_fisher <- function(df, ref_group) {
  ref_counts <- df %>%
    ungroup() %>%
    filter(Response_Group == ref_group) %>%
    dplyr::select(Category, Count) %>%
    {setNames(.$Count, .$Category)}

  df %>%
    filter(Response_Group != ref_group) %>%
    group_by(Response_Group) %>%
    summarise(
      p_value = {
        group_counts <- Count[Category %in% c("Yes", "No")]
        if (length(group_counts) < 2) group_counts <- c(group_counts, 0)
        contingency_table <- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts["Yes"], ref_counts["No"]
        ), nrow = 2)

        if (any(contingency_table < 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = "drop"
    ) %>%
    mutate(Significance = ifelse(!is.na(p_value) & p_value < 0.05, "*", ""))
}

# Perform tests
chi_0.1 <- get_chi_or_fisher(proportion_data %>% filter(Concentration == "0.1 ÂµM"), "Non response")
chi_0.5 <- get_chi_or_fisher(proportion_data %>% filter(Concentration == "0.5 ÂµM"), "Non response")

chi_all <- bind_rows(
  chi_0.1 %>% mutate(Concentration = "0.1 ÂµM"),
  chi_0.5 %>% mutate(Concentration = "0.5 ÂµM")
)

proportion_data <- proportion_data %>%
  left_join(chi_all, by = c("Concentration", "Response_Group"))

# ----------------- Factor Levels -----------------
proportion_data$Concentration <- factor(proportion_data$Concentration, levels = c("0.1 ÂµM", "0.5 ÂµM"))
proportion_data$Response_Group <- factor(proportion_data$Response_Group, levels = c(
  "Non response",
  "CX-DOX mid-late response",
  "DOX-specific response",
  "DOX only mid-late response",
  "CX total + DOX early response",
  "DOX early + CX-DOX mid-late response"
))

# ----------------- Star Positions -----------------
label_data <- proportion_data %>%
  group_by(Concentration, Response_Group) %>%
  summarise(Significance = dplyr::first(Significance), .groups = "drop") %>%
  filter(Significance == "*") %>%
  mutate(y_pos = 105)

# ----------------- Stack Order -----------------
proportion_data <- proportion_data %>%
  arrange(Concentration, Response_Group, Category)

# ----------------- Plot -----------------
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = "identity", position = "stack") +
  geom_text(
    data = label_data,
    aes(x = Response_Group, y = y_pos, label = Significance),
    inherit.aes = FALSE,
    size = 5, color = "black"
  ) +
  facet_wrap(~ Concentration, scales = "free_x") +
  scale_fill_manual(values = c("Yes" = "#e41a1c", "No" = "#377eb8")) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +
  labs(
    title = "DNA Damage Gene Proportions Across\nCormotif Clusters (0.1 and 0.5 ÂµM)",
    x = "Response Groups",
    y = "Percentage of Genes",
    fill = "DNA Damage Gene"
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = "bold"),
    axis.title.x = element_text(size = 14, face = "bold"),
    axis.title.y = element_text(size = 14, face = "bold"),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = "bold"),
    panel.border = element_rect(color = "black", fill = NA, linewidth = 1.2),
    panel.spacing = unit(1.2, "lines")
  )
```

## **ðŸ“Œ Proportion of P53 target Genes Genes (0.1 micromolar)**
```{r prop P53 0.1, echo=TRUE, message=FALSE}
# Load necessary libraries
library(dplyr)
library(ggplot2)
library(tidyr)
library(org.Hs.eg.db)

# **ðŸ”¹ Read P53 Target Gene List**
P53_Target <- read.csv("data/P53_Target.csv", stringsAsFactors = FALSE)

# Convert gene symbols to Entrez IDs
P53_Target <- P53_Target %>%
  mutate(Entrez_ID = mapIds(org.Hs.eg.db,
                            keys = P53_Target$Symbol,
                            column = "ENTREZID",
                            keytype = "SYMBOL",
                            multiVals = "first"))

P53_Target_genes <- na.omit(P53_Target$Entrez_ID)

# **ðŸ”¹ Load Corrmotif Groups for 0.1 Concentration**
prob_groups_0.1 <- list(
  "Non Response (0.1)" = read.csv("data/prob_1_0.1.csv")$Entrez_ID,
  "CX_DOX mid-late (0.1)" = read.csv("data/prob_2_0.1.csv")$Entrez_ID,
 "DOX only mid-late (0.1)"= read.csv("data/prob_3_0.1.csv")$Entrez_ID
)

# **ðŸ”¹ Create Dataframe for Corrmotif Groups**
corrmotif_df_0.1 <- bind_rows(
  lapply(prob_groups_0.1, function(ids) {
    data.frame(Entrez_ID = ids)
  }),
  .id = "Response_Group"
)

# **ðŸ”¹ Match Entrez_IDs with P53 Target Genes**
corrmotif_df_0.1 <- corrmotif_df_0.1 %>%
  mutate(P53_Target = ifelse(Entrez_ID %in% P53_Target_genes, "Yes", "No"))

# **ðŸ”¹ Count P53 Target Genes in Each Response Group**
proportion_data <- corrmotif_df_0.1 %>%
  group_by(Response_Group, P53_Target) %>%
  summarise(Count = n(), .groups = "drop") %>%
  group_by(Response_Group) %>%
  mutate(Percentage = (Count / sum(Count)) * 100)

# **ðŸ”¹ Ensure "Yes" is at the Bottom and "No" at the Top**
proportion_data$P53_Target <- factor(proportion_data$P53_Target, levels = c("Yes", "No"))

# **ðŸ”¹ Set Order of Response Groups for X-axis**
response_order <- c("Non Response (0.1)", "CX_DOX mid-late (0.1)", "DOX only mid-late (0.1)")
proportion_data$Response_Group <- factor(proportion_data$Response_Group, levels = response_order)

# **ðŸ”¹ Perform Chi-Square Tests for "DOX only mid-late (0.1)" and "CX_DOX mid-late (0.1)" vs "Non Response (0.1)"**
non_response_counts <- proportion_data %>%
  filter(Response_Group == "Non Response (0.1)") %>%
  dplyr::select(P53_Target, Count) %>%
  {setNames(.$Count, .$P53_Target)}  # Convert to named vector

chi_results <- proportion_data %>%
  filter(Response_Group %in% c("CX_DOX mid-late (0.1)","DOX only mid-late (0.1)")) %>%
  group_by(Response_Group) %>%
  summarise(
    p_value = {
      group_counts <- Count[P53_Target %in% c("Yes", "No")]
      if (!"Yes" %in% P53_Target) group_counts <- c(group_counts, 0)
      if (!"No" %in% P53_Target) group_counts <- c(0, group_counts)

      contingency_table <- matrix(c(
        group_counts[1], group_counts[2],
        non_response_counts["Yes"], non_response_counts["No"]
      ), nrow = 2, byrow = TRUE)

      # Perform chi-square test if all values are valid
      if (all(contingency_table >= 0 & is.finite(contingency_table))) {
        chisq.test(contingency_table)$p.value
      } else {
        NA
      }
    },
    .groups = "drop"
  ) %>%
  mutate(Significance = ifelse(!is.na(p_value) & p_value < 0.05, "*", ""))

# **ðŸ”¹ Merge Chi-Square Results into Proportion Data**
proportion_data <- proportion_data %>%
  left_join(chi_results %>% dplyr::select(Response_Group, Significance), by = "Response_Group")

# **ðŸ”¹ Set Star Position Uniform Across Groups at 105%**
star_positions <- data.frame(
  Response_Group = c("CX_DOX mid-late (0.1)","DOX only mid-late (0.1)"),
  y_pos = 105,  # Fixed at 105% of Y-axis
  Significance = chi_results$Significance
)

# **ðŸ”¹ Generate Proportion Plot with Chi-Square Stars**
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = P53_Target)) +
  geom_bar(stat = "identity", position = "stack") +  # Stacked bars
  geom_text(
    data = star_positions,
    aes(x = Response_Group, y = y_pos, label = Significance),  # Place stars at fixed 105%
    inherit.aes = FALSE,
    size = 6, color = "black", fontface = "bold", vjust = 0  # Keeps stars aligned
  ) +
  scale_y_continuous(labels = scales::percent_format(scale = 1), limits = c(0, 110)) +  # Fixed Y-axis to 100%
  scale_fill_manual(values = c("Yes" = "#e41a1c", "No" = "#377eb8")) +  # Yes (Red), No (Blue)
  labs(
    title = "Proportion of P53 Target Genes in\n0.1 Corrmotif Response Groups",
    x = "Response Groups (0.1 Concentration)",
    y = "Percentage",
    fill = "P53 Target Genes"
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = "black"),
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
    legend.title = element_blank(),
    panel.border = element_rect(color = "black", fill = NA, linewidth = 1.2),
    strip.background = element_blank(),
    strip.text = element_text(size = 12, face = "bold")
  )
```

## **ðŸ“Œ Proportion of P53 target Genes (0.5 micromolar) **
```{r prop P53 0.5, echo=TRUE, message=FALSE}
# Load necessary libraries
library(dplyr)
library(ggplot2)
library(tidyr)
library(org.Hs.eg.db)

# **ðŸ”¹ Read P53 Target Gene List**
P53_Target <- read.csv("data/P53_Target.csv", stringsAsFactors = FALSE)

# Convert gene symbols to Entrez IDs
P53_Target <- P53_Target %>%
  mutate(Entrez_ID = mapIds(org.Hs.eg.db,
                            keys = P53_Target$Symbol,
                            column = "ENTREZID",
                            keytype = "SYMBOL",
                            multiVals = "first"))

P53_Target_genes <- na.omit(P53_Target$Entrez_ID)

# **ðŸ”¹ Load Corrmotif Groups for 0.5 Concentration**
prob_groups_0.5 <- list(
  "Non Response (0.5)" = read.csv("data/prob_1_0.5.csv")$Entrez_ID,
  "DOX-specific response (0.5)" = read.csv("data/prob_2_0.5.csv")$Entrez_ID,
  "DOX only mid-late response (0.5)" = read.csv("data/prob_3_0.5.csv")$Entrez_ID,
  "CX DOX (early) response (0.5)" = read.csv("data/prob_4_0.5.csv")$Entrez_ID,
  "DOX + CX (mid-late) response (0.5)" = read.csv("data/prob_5_0.5.csv")$Entrez_ID
)

# **ðŸ”¹ Create Dataframe for Corrmotif Groups**
corrmotif_df_0.5 <- bind_rows(
  lapply(prob_groups_0.5, function(ids) {
    data.frame(Entrez_ID = ids)
  }),
  .id = "Response_Group"
)

# **ðŸ”¹ Match Entrez_IDs with P53 Target Genes**
corrmotif_df_0.5 <- corrmotif_df_0.5 %>%
  mutate(P53_Target = ifelse(Entrez_ID %in% P53_Target_genes, "Yes", "No"))

# **ðŸ”¹ Count P53 Target Genes in Each Response Group**
proportion_data <- corrmotif_df_0.5 %>%
  group_by(Response_Group, P53_Target) %>%
  summarise(Count = n(), .groups = "drop") %>%
  group_by(Response_Group) %>%
  mutate(Percentage = (Count / sum(Count)) * 100)

# **ðŸ”¹ Ensure "Yes" is at the Bottom and "No" at the Top**
proportion_data$P53_Target <- factor(proportion_data$P53_Target, levels = c("Yes", "No"))

# **ðŸ”¹ Set Order of Response Groups for X-axis**
response_order <- c("Non Response (0.5)", "DOX-specific response (0.5)", "DOX only mid-late response (0.5)", 
                    "CX DOX (early) response (0.5)", "DOX + CX (mid-late) response (0.5)")
proportion_data$Response_Group <- factor(proportion_data$Response_Group, levels = response_order)

# **ðŸ”¹ Perform Chi-Square Tests for Each Response Group vs Non-Response**
non_response_counts <- proportion_data %>%
  filter(Response_Group == "Non Response (0.5)") %>%
  dplyr::select(P53_Target, Count) %>%
  {setNames(.$Count, .$P53_Target)}  # Convert to named vector

# **Comparing Each Group Against "Non Response (0.5)"**
chi_results <- proportion_data %>%
  filter(Response_Group %in% c("DOX-specific response (0.5)", "DOX only mid-late response (0.5)", 
                               "CX DOX (early) response (0.5)", "DOX + CX (mid-late) response (0.5)")) %>%
  group_by(Response_Group) %>%
  summarise(
    p_value = {
      group_counts <- Count[P53_Target %in% c("Yes", "No")]
      if (!"Yes" %in% P53_Target) group_counts <- c(group_counts, 0)
      if (!"No" %in% P53_Target) group_counts <- c(0, group_counts)

      contingency_table <- matrix(c(
        group_counts[1], group_counts[2],  # Response group counts
        non_response_counts["Yes"], non_response_counts["No"]  # Non-response counts
      ), nrow = 2, byrow = TRUE)

      # Perform chi-square test if all values are valid
      if (all(contingency_table >= 0 & is.finite(contingency_table))) {
        chisq.test(contingency_table)$p.value
      } else {
        NA
      }
    },
    .groups = "drop"
  ) %>%
  mutate(Significance = ifelse(!is.na(p_value) & p_value < 0.05, "*", ""))

# **ðŸ”¹ Merge Chi-Square Results into Proportion Data**
proportion_data <- proportion_data %>%
  left_join(chi_results %>% dplyr::select(Response_Group, Significance), by = "Response_Group")

# **ðŸ”¹ Set Star Position Uniform Across Groups at 105%**
star_positions <- data.frame(
  Response_Group = c("DOX-specific response (0.5)", "DOX only mid-late response (0.5)", 
                     "CX DOX (early) response (0.5)", "DOX + CX (mid-late) response (0.5)"),
  y_pos = 105,  # Fixed at 105% of Y-axis
  Significance = chi_results$Significance
)

# **ðŸ”¹ Generate Proportion Plot with Chi-Square Stars**
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = P53_Target)) +
  geom_bar(stat = "identity", position = "stack") +  # Stacked bars
  geom_text(
    data = star_positions,
    aes(x = Response_Group, y = y_pos, label = Significance),  # Place stars at fixed 105%
    inherit.aes = FALSE,
    size = 6, color = "black", fontface = "bold", vjust = 0  # Keeps stars aligned
  ) +
  scale_y_continuous(labels = scales::percent_format(scale = 1), limits = c(0, 110)) +  # **Y-axis now limited to 110% for visibility**
  scale_fill_manual(values = c("Yes" = "#e41a1c", "No" = "#377eb8")) +  # Yes (Red), No (Blue)
  labs(
    title = "Proportion of P53 Target Genes in\n0.5 Corrmotif Response Groups",
    x = "Response Groups (0.5 Concentration)",
    y = "Percentage",
    fill = "P53 Target Genes"
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = "black"),
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
    legend.title = element_blank(),
    panel.border = element_rect(color = "black", fill = NA, linewidth = 1.2),
    strip.background = element_blank(),
    strip.text = element_text(size = 12, face = "bold")
  )
```

## **ðŸ“Œ Proportion of Heart Specific Genes Genes (0.1 micromolar)**
```{r prop heart 0.1, echo=TRUE, message=FALSE}
# Load Required Libraries
library(dplyr)
library(ggplot2)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load Heart-Specific Genes -----------------
heart_genes <- read.csv("data/Human_Heart_Genes.csv", stringsAsFactors = FALSE)

# Convert SYMBOL to Entrez ID
heart_genes$Entrez_ID <- mapIds(
  org.Hs.eg.db,
  keys = heart_genes$Gene,
  column = "ENTREZID",
  keytype = "SYMBOL",
  multiVals = "first"
)
heart_entrez_ids <- na.omit(heart_genes$Entrez_ID)

# ----------------- Load CorrMotif Groups for 0.1 ÂµM -----------------
prob_1_0.1 <- as.character(read.csv("data/prob_1_0.1.csv")$Entrez_ID)
prob_2_0.1 <- as.character(read.csv("data/prob_2_0.1.csv")$Entrez_ID)
prob_3_0.1 <- as.character(read.csv("data/prob_3_0.1.csv")$Entrez_ID)

# Combine into one dataframe
all_ids_0.1 <- unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))
df_0.1 <- data.frame(Entrez_ID = all_ids_0.1, stringsAsFactors = FALSE)

# ----------------- Annotate Response Group -----------------
df_0.1 <- df_0.1 %>%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ "Non response\n(0.1 ÂµM)",
      Entrez_ID %in% prob_2_0.1 ~ "CX-DOX mid-late response\n(0.1 ÂµM)",
      Entrez_ID %in% prob_3_0.1 ~ "DOX only mid-late response\n(0.1 ÂµM)"
    ),
    Category = ifelse(Entrez_ID %in% heart_entrez_ids, "Heart-specific Genes", "Non-Heart-specific Genes")
  )

# ----------------- Proportion Calculation -----------------
proportion_0.1 <- df_0.1 %>%
  group_by(Response_Group, Category) %>%
  summarise(Count = n(), .groups = "drop") %>%
  group_by(Response_Group) %>%
  mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Chi-square Test (Optional) -----------------
ref_counts_0.1 <- proportion_0.1 %>%
  dplyr::ungroup() %>%
  dplyr::filter(Response_Group == "Non response\n(0.1 ÂµM)") %>%
  dplyr::select(Category, Count) %>%
  {setNames(.$Count, .$Category)}

chi_0.1 <- proportion_0.1 %>%
  filter(Response_Group != "Non response\n(0.1 ÂµM)") %>%
  group_by(Response_Group) %>%
  summarise(
    p_value = {
      group_counts <- Count[Category %in% c("Heart-specific Genes", "Non-Heart-specific Genes")]
      if (length(group_counts) < 2) group_counts <- c(group_counts, 0)
      contingency_table <- matrix(c(
        group_counts[1], group_counts[2],
        ref_counts_0.1["Heart-specific Genes"], ref_counts_0.1["Non-Heart-specific Genes"]
      ), nrow = 2)
      print(paste("Set:", unique(Response_Group)))
      print("Contingency Table:")
      print(contingency_table)
      if (all(contingency_table >= 0 & is.finite(contingency_table))) {
        chisq.test(contingency_table)$p.value
      } else {
        NA
      }
    },
    .groups = "drop"
  ) %>%
  mutate(Significance = ifelse(!is.na(p_value) & p_value < 0.05, "*", ""))

# Merge with proportion data
proportion_0.1 <- proportion_0.1 %>%
  left_join(chi_0.1, by = "Response_Group")

# ----------------- Plot -----------------
proportion_0.1$Response_Group <- factor(
  proportion_0.1$Response_Group,
  levels = c(
    "Non response\n(0.1 ÂµM)",
    "CX-DOX mid-late response\n(0.1 ÂµM)",
    "DOX only mid-late response\n(0.1 ÂµM)"
  )
)

ggplot(proportion_0.1, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = "identity", position = "stack") +
  geom_text(
    data = proportion_0.1 %>% distinct(Response_Group, Significance),
    aes(x = Response_Group, y = 105, label = Significance),
    inherit.aes = FALSE,
    size = 6,
    color = "black"
  ) +
  scale_fill_manual(values = c(
    "Heart-specific Genes" = "#4daf4a",
    "Non-Heart-specific Genes" = "#377eb8"
  )) +
  labs(
    title = "Heart-Specific Gene Proportions\nAcross 0.1 ÂµM Cormotif Clusters",
    x = "Response Groups",
    y = "Percentage of Genes",
    fill = "Gene Category"
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = rel(1.4), hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
    panel.border = element_rect(color = "black", fill = NA, linewidth = 1.2)
  )
```

## **ðŸ“Œ Proportion of Heart Specific Genes Genes (0.5 micromolar)**
```{r prop heart 0.5, echo=TRUE, message=FALSE}
# ----------------- Load CorrMotif Groups for 0.5 ÂµM -----------------
prob_1_0.5 <- as.character(read.csv("data/prob_1_0.5.csv")$Entrez_ID)
prob_2_0.5 <- as.character(read.csv("data/prob_2_0.5.csv")$Entrez_ID)
prob_3_0.5 <- as.character(read.csv("data/prob_3_0.5.csv")$Entrez_ID)
prob_4_0.5 <- as.character(read.csv("data/prob_4_0.5.csv")$Entrez_ID)
prob_5_0.5 <- as.character(read.csv("data/prob_5_0.5.csv")$Entrez_ID)

# Combine into one dataframe
all_ids_0.5 <- unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))
df_0.5 <- data.frame(Entrez_ID = all_ids_0.5, stringsAsFactors = FALSE)

# ----------------- Annotate Response Group -----------------
df_0.5 <- df_0.5 %>%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ "Non response\n(0.5 ÂµM)",
      Entrez_ID %in% prob_2_0.5 ~ "DOX-specific\nresponse (0.5 ÂµM)",
      Entrez_ID %in% prob_3_0.5 ~ "DOX only mid-late\nresponse (0.5 ÂµM)",
      Entrez_ID %in% prob_4_0.5 ~ "CX total + DOX early\nresponse (0.5 ÂµM)",
      Entrez_ID %in% prob_5_0.5 ~ "DOX early + CX-DOX mid-late\nresponse (0.5 ÂµM)"
    ),
    Category = ifelse(Entrez_ID %in% heart_entrez_ids, "Heart-specific Genes", "Non-Heart-specific Genes")
  )

# ----------------- Proportion Calculation -----------------
proportion_0.5 <- df_0.5 %>%
  group_by(Response_Group, Category) %>%
  summarise(Count = n(), .groups = "drop") %>%
  group_by(Response_Group) %>%
  mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Chi-square Test (vs Non-response) -----------------
ref_counts_0.5 <- proportion_0.5 %>%
  dplyr::ungroup() %>%
  dplyr::filter(Response_Group == "Non response\n(0.5 ÂµM)") %>%
  dplyr::select(Category, Count) %>%
  {setNames(.$Count, .$Category)}

chi_0.5 <- proportion_0.5 %>%
  filter(Response_Group != "Non response\n(0.5 ÂµM)") %>%
  group_by(Response_Group) %>%
  summarise(
    p_value = {
      group_counts <- Count[Category %in% c("Heart-specific Genes", "Non-Heart-specific Genes")]
      if (length(group_counts) < 2) group_counts <- c(group_counts, 0)
      contingency_table <- matrix(c(
        group_counts[1], group_counts[2],
        ref_counts_0.5["Heart-specific Genes"], ref_counts_0.5["Non-Heart-specific Genes"]
      ), nrow = 2)
      print(paste("Set:", unique(Response_Group)))
      print("Contingency Table:")
      print(contingency_table)
      if (all(contingency_table >= 0 & is.finite(contingency_table))) {
        chisq.test(contingency_table)$p.value
      } else {
        NA
      }
    },
    .groups = "drop"
  ) %>%
  mutate(Significance = ifelse(!is.na(p_value) & p_value < 0.05, "*", ""))

# Merge with proportion data
proportion_0.5 <- proportion_0.5 %>%
  left_join(chi_0.5, by = "Response_Group")

# ----------------- Plot -----------------
proportion_0.5$Response_Group <- factor(
  proportion_0.5$Response_Group,
  levels = c(
    "Non response\n(0.5 ÂµM)",
    "DOX-specific\nresponse (0.5 ÂµM)",
    "DOX only mid-late\nresponse (0.5 ÂµM)",
    "CX total + DOX early\nresponse (0.5 ÂµM)",
    "DOX early + CX-DOX mid-late\nresponse (0.5 ÂµM)"
  )
)

ggplot(proportion_0.5, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = "identity", position = "stack") +
  geom_text(
    data = proportion_0.5 %>% distinct(Response_Group, Significance),
    aes(x = Response_Group, y = 105, label = Significance),
    inherit.aes = FALSE,
    size = 6,
    color = "black"
  ) +
  scale_fill_manual(values = c(
    "Heart-specific Genes" = "#4daf4a",
    "Non-Heart-specific Genes" = "#377eb8"
  )) +
  labs(
    title = "Heart-Specific Gene Proportions\nAcross 0.5 ÂµM Cormotif Clusters",
    x = "Response Groups",
    y = "Percentage of Genes",
    fill = "Gene Category"
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(size = rel(1.4), hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
    panel.border = element_rect(color = "black", fill = NA, linewidth = 1.2)
  )
```

## **ðŸ“Œ Proportion of Heart Specific Genes Genes (Combined)**
```{r prop heart comb, echo=TRUE, message=FALSE, fig.width=10, fig.height=6}
# Load Required Libraries
library(dplyr)
library(ggplot2)
library(org.Hs.eg.db)
library(AnnotationDbi)

# ----------------- Load Heart-Specific Genes -----------------
heart_genes <- read.csv("data/Human_Heart_Genes.csv", stringsAsFactors = FALSE)
heart_genes$Entrez_ID <- mapIds(
  org.Hs.eg.db,
  keys = heart_genes$Gene,
  column = "ENTREZID",
  keytype = "SYMBOL",
  multiVals = "first"
)
heart_entrez_ids <- na.omit(heart_genes$Entrez_ID)

# ----------------- Load CorrMotif Groups -----------------
# 0.1 ÂµM
prob_1_0.1 <- as.character(read.csv("data/prob_1_0.1.csv")$Entrez_ID)
prob_2_0.1 <- as.character(read.csv("data/prob_2_0.1.csv")$Entrez_ID)
prob_3_0.1 <- as.character(read.csv("data/prob_3_0.1.csv")$Entrez_ID)

# 0.5 ÂµM
prob_1_0.5 <- as.character(read.csv("data/prob_1_0.5.csv")$Entrez_ID)
prob_2_0.5 <- as.character(read.csv("data/prob_2_0.5.csv")$Entrez_ID)
prob_3_0.5 <- as.character(read.csv("data/prob_3_0.5.csv")$Entrez_ID)
prob_4_0.5 <- as.character(read.csv("data/prob_4_0.5.csv")$Entrez_ID)
prob_5_0.5 <- as.character(read.csv("data/prob_5_0.5.csv")$Entrez_ID)

# ----------------- Annotate CorrMotif Groups -----------------
df_0.1 <- data.frame(Entrez_ID = unique(c(prob_1_0.1, prob_2_0.1, prob_3_0.1))) %>%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ "Non response",
      Entrez_ID %in% prob_2_0.1 ~ "CX-DOX mid-late response",
      Entrez_ID %in% prob_3_0.1 ~ "DOX only mid-late response"
    ),
    Category = ifelse(Entrez_ID %in% heart_entrez_ids, "Heart-specific Genes", "Non-Heart-specific Genes"),
    Concentration = "0.1 ÂµM"
  )

df_0.5 <- data.frame(Entrez_ID = unique(c(prob_1_0.5, prob_2_0.5, prob_3_0.5, prob_4_0.5, prob_5_0.5))) %>%
  mutate(
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ "Non response",
      Entrez_ID %in% prob_2_0.5 ~ "DOX-specific response",
      Entrez_ID %in% prob_3_0.5 ~ "DOX only mid-late response",
      Entrez_ID %in% prob_4_0.5 ~ "CX total + DOX early response",
      Entrez_ID %in% prob_5_0.5 ~ "DOX early + CX-DOX mid-late response"
    ),
    Category = ifelse(Entrez_ID %in% heart_entrez_ids, "Heart-specific Genes", "Non-Heart-specific Genes"),
    Concentration = "0.5 ÂµM"
  )

# ----------------- Combine Data -----------------
df_combined <- bind_rows(df_0.1, df_0.5)

# ----------------- Calculate Proportions -----------------
proportion_data <- df_combined %>%
  dplyr::group_by(Concentration, Response_Group, Category) %>%
  dplyr::summarise(Count = n(), .groups = "drop") %>%
  dplyr::group_by(Concentration, Response_Group) %>%
  dplyr::mutate(Percentage = Count / sum(Count) * 100)

# ----------------- Chi-square or Fisher's Test -----------------
get_chi_or_fisher <- function(df, ref_group) {
  ref_counts <- df %>%
    dplyr::ungroup() %>%
    dplyr::filter(Response_Group == ref_group) %>%
    dplyr::select(Category, Count) %>%
    {setNames(.$Count, .$Category)}
  
  df %>%
    dplyr::filter(Response_Group != ref_group) %>%
    dplyr::group_by(Response_Group) %>%
    dplyr::summarise(
      p_value = {
        group_counts <- Count[Category %in% c("Heart-specific Genes", "Non-Heart-specific Genes")]
        if (length(group_counts) < 2) group_counts <- c(group_counts, 0)
        contingency_table <- matrix(c(
          group_counts[1], group_counts[2],
          ref_counts["Heart-specific Genes"], ref_counts["Non-Heart-specific Genes"]
        ), nrow = 2)
        if (any(contingency_table < 5)) {
          fisher.test(contingency_table)$p.value
        } else {
          chisq.test(contingency_table)$p.value
        }
      },
      .groups = "drop"
    ) %>%
    dplyr::mutate(Significance = ifelse(!is.na(p_value) & p_value < 0.05, "*", ""))
}

# Run test for each concentration
chi_0.1 <- get_chi_or_fisher(proportion_data %>% dplyr::filter(Concentration == "0.1 ÂµM"), "Non response")
chi_0.5 <- get_chi_or_fisher(proportion_data %>% dplyr::filter(Concentration == "0.5 ÂµM"), "Non response")

chi_all <- dplyr::bind_rows(
  chi_0.1 %>% dplyr::mutate(Concentration = "0.1 ÂµM"),
  chi_0.5 %>% dplyr::mutate(Concentration = "0.5 ÂµM")
)

# Merge test results into proportions
proportion_data <- proportion_data %>%
  dplyr::left_join(chi_all, by = c("Concentration", "Response_Group"))

# ----------------- Reorder Factor Levels -----------------
proportion_data$Concentration <- factor(proportion_data$Concentration, levels = c("0.1 ÂµM", "0.5 ÂµM"))
proportion_data$Response_Group <- factor(proportion_data$Response_Group, levels = c(
  "Non response",
  "CX-DOX mid-late response",
  "DOX-specific response",
  "DOX only mid-late response",
  "CX total + DOX early response",
  "DOX early + CX-DOX mid-late response"
))

# ----------------- Compute Y-position for Significance Labels -----------------
label_data <- proportion_data %>%
  dplyr::group_by(Concentration, Response_Group) %>%
  dplyr::summarise(Significance = dplyr::first(Significance),
                   ymax = sum(Percentage), .groups = "drop") %>%
  dplyr::filter(!is.na(Significance)) %>%
  dplyr::mutate(y_pos = ymax + 3)

# ----------------- Final Plot -----------------
# --- Position stars above 100
label_data <- proportion_data %>%
  dplyr::group_by(Concentration, Response_Group) %>%
  dplyr::summarise(Significance = dplyr::first(Significance), .groups = "drop") %>%
  dplyr::filter(!is.na(Significance)) %>%
  dplyr::mutate(y_pos = 105)  # stars go at 105

# --- Plot
ggplot(proportion_data, aes(x = Response_Group, y = Percentage, fill = Category)) +
  geom_bar(stat = "identity", position = "stack") +
  geom_text(
    data = label_data,
    aes(x = Response_Group, y = y_pos, label = Significance),
    inherit.aes = FALSE,
    size = 5,
    color = "black"
  ) +
  facet_wrap(~ Concentration, scales = "free_x") +
  scale_fill_manual(values = c(
    "Heart-specific Genes" = "#4daf4a",
    "Non-Heart-specific Genes" = "#377eb8"
  )) +
  scale_y_continuous(limits = c(0, 110), expand = c(0, 0)) +  # allow space for stars and border
  labs(
    title = "Heart-Specific Gene Proportions Across\nCormotif Clusters (0.1 and 0.5)",
    x = "Response Groups",
    y = "Percentage of Genes",
    fill = "Gene Category"
  ) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(size = 16, hjust = 0.5, face = "bold"),
    axis.title.x = element_text(size = 14, face = "bold"),
    axis.title.y = element_text(size = 14, face = "bold"),
    axis.text.x = element_text(size = 11, angle = 45, hjust = 1),
    axis.text.y = element_text(size = 12),
    legend.title = element_text(size = 13),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 14, face = "bold"),
    panel.border = element_rect(color = "black", fill = NA, linewidth = 1.2),  # draws full box
    panel.spacing = unit(1.2, "lines")  # adds vertical spacing to avoid cramped look
  )

```


## **ðŸ“Œ Atrial fibrillation gene expression changes between CX5461 and DOX**
```{r AF, echo=TRUE, message=FALSE, warning=FALSE, , fig.height=65, fig.width=10}
library(tidyverse) 
library(ggfortify)
library(cluster)
library(edgeR)
library(limma)
library(Homo.sapiens)
library(BiocParallel)
library(qvalue)
library(pheatmap)
library(clusterProfiler)
library(AnnotationDbi)
library(org.Hs.eg.db)
library(RColorBrewer)
library(readr)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(ComplexHeatmap)
library(circlize)
library(grid)
library(reshape2)
library(dplyr)

# Load UCSC transcript database
txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene

# Load DEGs Data
CX_0.1_3 <- read.csv("data/DEGs/Toptable_CX_0.1_3.csv")
CX_0.1_24 <- read.csv("data/DEGs/Toptable_CX_0.1_24.csv")
CX_0.1_48 <- read.csv("data/DEGs/Toptable_CX_0.1_48.csv")
CX_0.5_3 <- read.csv("data/DEGs/Toptable_CX_0.5_3.csv")
CX_0.5_24 <- read.csv("data/DEGs/Toptable_CX_0.5_24.csv")
CX_0.5_48 <- read.csv("data/DEGs/Toptable_CX_0.5_48.csv")

DOX_0.1_3 <- read.csv("data/DEGs/Toptable_DOX_0.1_3.csv")
DOX_0.1_24 <- read.csv("data/DEGs/Toptable_DOX_0.1_24.csv")
DOX_0.1_48 <- read.csv("data/DEGs/Toptable_DOX_0.1_48.csv")
DOX_0.5_3 <- read.csv("data/DEGs/Toptable_DOX_0.5_3.csv")
DOX_0.5_24 <- read.csv("data/DEGs/Toptable_DOX_0.5_24.csv")
DOX_0.5_48 <- read.csv("data/DEGs/Toptable_DOX_0.5_48.csv")

Entrez_IDs <- c(
  3425, 53834, 9464, 79658, 55876, 3781, 9472, 11017, 64395, 23353,
  57794, 79591, 2702, 51306, 857, 10418, 9644, 115509, 89797, 10728,
  282996, 8497, 5695, 9026, 57666, 143384, 5883, 90102, 84909, 221037,
  29119, 79933, 391, 81575, 8738, 5396, 26010, 744, 56981, 80724,
  5781, 6910, 4598, 79068, 2250, 56978, 3752, 5208, 9736, 26778,
  80315, 64374, 79159, 11078, 5308, 3782, 10654, 79776, 6310, 8082,
  2995, 23224, 79741, 5708, 26136, 9967, 84033, 9148, 29982, 81488,
  1788, 9208, 2908, 1021, 5339, 5727, 23118, 10090, 57585, 221154,
  1073, 29998, 817, 858, 3486, 5747, 2104, 55105, 4194, 3570,
  1607, 6586, 5570, 2070, 27145, 4646, 3159, 9748, 79991, 221035,
  5033, 80212, 196385, 79600, 11046, 54014, 151636, 64778, 92597, 3983,
  7090, 9267, 7273, 26018, 9948, 2131, 55870, 5523, 5318, 6239,
  3480, 55777, 2066, 11165, 2028, 57619, 2690, 23414, 55818, 84700,
  28965, 80204, 463, 5108, 222553, 387119, 28981, 11113, 10301, 995,
  23030, 2697, 996, 339500, 54805, 9960, 91404, 145781, 100820829, 84542,
  2176, 51684, 9513, 7473, 4666, 23150, 5915, 5062, 4016, 23039,
  159686, 1839, 5201, 93166, 64753, 29959, 5496, 23245, 5069, 56916,
  92344, 23092, 3992, 9415, 10554, 7456, 9570, 57178, 23143, 161176,
  5424, 2034, 10277, 11278, 79803, 6653, 4756, 132660, 5430, 9031,
  57158, 285761, 8110, 387700, 1829, 4126, 7323, 51308, 7332, 6598,
  3757, 6187, 6660, 10529, 6920, 115286, 8451, 8943, 4137, 7514,
  4801, 9709, 23177, 8671, 29915, 26207, 3680, 490, 493856, 58489,
  54897, 4625, 22955, 84952, 10221, 2263, 84641, 4892, 1026, 84650,
  8382, 221656, 2969, 144453, 117177, 2626, 8476, 161882, 51807, 4624,
  9612, 55795, 51043, 144348, 51232, 8729, 3899, 11155, 23316, 79006,
  146330, 6403, 6331, 4300, 427, 845, 3313, 113622, 5789, 376132,
  29841, 8462, 203859, 401397, 5506, 55521, 5819, 4059, 6934, 57727,
  23066, 79568, 83478, 10087, 9586, 222194, 10466, 10499, 58499, 79720,
  4772, 10794, 125919, 602, 27332, 59345, 340359, 3705, 64710, 57801,
  10818, 143684, 149281, 55013, 23095, 93649, 84034, 23347, 440926, 11124,
  23293, 51426, 832, 7068, 57646, 152002, 7531, 1398, 100101267, 221937,
  26873, 3709, 10576, 27040, 201176, 284403, 307, 6525, 5387, 1808,
  114907, 1952, 4091, 5096, 23451, 51207, 142891, 4084, 3797, 1185,
  5334, 217, 2042, 7781, 253461, 152404, 2992, 153478, 6695, 26249,
  662, 23036, 22852, 23411, 23387, 6786, 7690, 10021, 8925, 5595,
  63892, 23466, 11149, 139411, 755, 55347, 22820, 55111, 84444, 6258,
  57623, 1387, 3762, 5861, 3339, 114822, 129787, 196528, 54437, 10395,
  4023, 5095, 894, 3156, 1877, 283871, 2673, 23157, 10512, 2258,
  79750, 84665, 26091, 5978, 8751, 79695, 2768, 155382, 84163, 7709,
  4092, 1837, 2064, 4629, 55122, 8882, 150962, 23013, 8742, 489,
  201134, 55114, 84264, 64428, 4089, 861, 7422
)


# Subset the toptable based on the entrez IDs and select specific columns

subset_toptable1 <- CX_0.1_3[CX_0.1_3$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]


subset_toptable2 <- CX_0.1_24[CX_0.1_24$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]


subset_toptable3 <- CX_0.1_48[CX_0.1_48$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]


subset_toptable3 <- CX_0.1_48[CX_0.1_48$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]

subset_toptable4 <- CX_0.5_3[CX_0.5_3$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]

subset_toptable5 <- CX_0.5_24[CX_0.5_24$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]

subset_toptable6 <- CX_0.5_48[CX_0.5_48$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]

subset_toptable7 <- DOX_0.1_3[DOX_0.1_3$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]

subset_toptable8 <- DOX_0.1_24[DOX_0.1_24$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]

subset_toptable9 <- DOX_0.1_48[DOX_0.1_48$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]

subset_toptable10 <- DOX_0.5_3[DOX_0.5_3$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]

subset_toptable11 <- DOX_0.5_24[DOX_0.5_24$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]

subset_toptable12 <- DOX_0.5_48[DOX_0.5_48$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]

# Function to add columns and combine data
add_metadata <- function(data, drug, conc, time) {
  data %>%
    mutate(Drug = drug, Conc = conc, Time = time)
}

# Add metadata and combine all subsets
combined_data <- bind_rows(
  add_metadata(subset_toptable1, "CX", 0.1, 3),
  add_metadata(subset_toptable2, "CX", 0.1, 24),
  add_metadata(subset_toptable3, "CX", 0.1, 48),
  add_metadata(subset_toptable4, "CX", 0.5, 3),
  add_metadata(subset_toptable5, "CX", 0.5, 24),
  add_metadata(subset_toptable6, "CX", 0.5, 48),
  add_metadata(subset_toptable7, "DOX", 0.1, 3),
  add_metadata(subset_toptable8, "DOX", 0.1, 24),
  add_metadata(subset_toptable9, "DOX", 0.1, 48),
  add_metadata(subset_toptable10, "DOX", 0.5, 3),
  add_metadata(subset_toptable11, "DOX", 0.5, 24),
  add_metadata(subset_toptable12, "DOX", 0.5, 48)
)

# Convert Entrez IDs to Gene symbols
combined_data <- combined_data %>%
  mutate(Gene = mapIds(
    org.Hs.eg.db,
    keys = as.character(Entrez_ID),
    column = "SYMBOL",
    keytype = "ENTREZID",
    multiVals = "first"
  ))

# Reorder columns
final_data <- dplyr::select(combined_data, Entrez_ID, Gene, logFC, adj.P.Val, Drug, Conc, Time)

# Assuming your dataframe is named data
# Add a column for significance stars
final_data <- final_data %>%
  mutate(Significance = ifelse(adj.P.Val < 0.05, "*", ""))

# Create a matrix for the heatmap (logFC values)
logFC_matrix <- acast(final_data, Gene ~ paste(Drug, Conc, Time, sep = "_"), value.var = "logFC")

# Create a matrix for the significance annotations
signif_matrix <- acast(final_data, Gene ~ paste(Drug, Conc, Time, sep = "_"), value.var = "Significance")

# Split column names into Drug, Conc, and Time
colnames_split <- strsplit(colnames(logFC_matrix), "_")
drug <- sapply(colnames_split, function(x) x[1])
conc <- sapply(colnames_split, function(x) x[2])
time <- sapply(colnames_split, function(x) x[3])

# Create the desired column order: CX 0.1 3hr, CX 0.5 3hr, CX 0.1 24hr, CX 0.5 24hr, CX 0.1 48h, CX 0.5 48h,
# DOX 0.1 3hr, DOX 0.5 3hr, DOX 0.1 24hr, DOX 0.5 24hr, DOX 0.1 48h, DOX 0.5 48h
desired_order <- c("CX_0.1_3", "CX_0.5_3", "CX_0.1_24", "CX_0.5_24", "CX_0.1_48", "CX_0.5_48",
                   "DOX_0.1_3", "DOX_0.5_3", "DOX_0.1_24", "DOX_0.5_24", "DOX_0.1_48", "DOX_0.5_48")

# Reorder columns in the matrix based on the desired order
column_names <- paste(drug, conc, time, sep = "_")
column_order <- match(desired_order, column_names)
logFC_matrix <- logFC_matrix[, column_order]
signif_matrix <- signif_matrix[, column_order]
drug <- drug[column_order]
conc <- conc[column_order]
time <- time[column_order]

# Prepare annotations matching the column structure
ha_top <- HeatmapAnnotation(
  Drug = drug,
  Conc = conc,
  Time = time,
  col = list(Drug = c("CX" = "blue", "DOX" = "red"),
             Conc = c("0.1" = "lightgreen", "0.5" = "darkgreen"),
             Time = c("3" = "yellow", "24" = "orange", "48" = "purple")),
  annotation_height = unit(c(2, 2, 2), "cm")
)

# Create the heatmap
heatmap <- Heatmap(logFC_matrix, name = "logFC", top_annotation = ha_top,
                   cell_fun = function(j, i, x, y, width, height, fill) {
                     grid.text(signif_matrix[i, j], x, y, gp = gpar(fontsize = 10))
                   },
                   show_row_names = TRUE, show_column_names = FALSE,
                   column_title = "Genes in Atrial fibrillation GWAS associated loci\nin response to CX5461 and DOX",
                   column_title_gp = gpar(fontsize = 16, fontface = "bold"),
                   cluster_columns = FALSE)  # Disable column clustering

# Draw the heatmap
draw(heatmap, heatmap_legend_side = "right", annotation_legend_side = "right")
```

## **ðŸ“Œ Heart Failure gene expression changes between CX5461 and DOX**
```{r HF, echo=TRUE, message=FALSE, warning=FALSE, , fig.height=25, fig.width=10}
library(tidyverse) 
library(ggfortify)
library(cluster)
library(edgeR)
library(limma)
library(Homo.sapiens)
library(BiocParallel)
library(qvalue)
library(pheatmap)
library(clusterProfiler)
library(AnnotationDbi)
library(org.Hs.eg.db)
library(RColorBrewer)
library(readr)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(ComplexHeatmap)
library(circlize)
library(grid)
library(reshape2)
library(dplyr)

# Load UCSC transcript database
txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene

# Load DEGs Data
CX_0.1_3 <- read.csv("data/DEGs/Toptable_CX_0.1_3.csv")
CX_0.1_24 <- read.csv("data/DEGs/Toptable_CX_0.1_24.csv")
CX_0.1_48 <- read.csv("data/DEGs/Toptable_CX_0.1_48.csv")
CX_0.5_3 <- read.csv("data/DEGs/Toptable_CX_0.5_3.csv")
CX_0.5_24 <- read.csv("data/DEGs/Toptable_CX_0.5_24.csv")
CX_0.5_48 <- read.csv("data/DEGs/Toptable_CX_0.5_48.csv")

DOX_0.1_3 <- read.csv("data/DEGs/Toptable_DOX_0.1_3.csv")
DOX_0.1_24 <- read.csv("data/DEGs/Toptable_DOX_0.1_24.csv")
DOX_0.1_48 <- read.csv("data/DEGs/Toptable_DOX_0.1_48.csv")
DOX_0.5_3 <- read.csv("data/DEGs/Toptable_DOX_0.5_3.csv")
DOX_0.5_24 <- read.csv("data/DEGs/Toptable_DOX_0.5_24.csv")
DOX_0.5_48 <- read.csv("data/DEGs/Toptable_DOX_0.5_48.csv")

Entrez_IDs <- c(
  57794, 285025, 9960, 8189, 949, 23327, 283450, 6801, 222553, 387119,
  27241, 22876, 202052, 2186, 6146, 7073, 79068, 11124, 27258, 26249,
  7531, 93210, 463, 64788, 88, 10725, 80777, 129787, 339855, 55101,
  84909, 6001, 4864, 54497, 5308, 375056, 29959, 5496, 3992, 9415,
  9531, 817, 8842, 57099, 27161, 4774, 80724, 217, 84163, 9031,
  23293, 1021, 57602, 64116, 4137, 9709, 7273, 571, 1187, 81846,
  10221, 1026, 6311, 10019, 79933, 2318, 11080, 54810, 80816, 2969,
  9640, 5409, 64753, 26010, 5578, 6934, 10087, 602, 51652, 8291,
  27332, 3705, 114991, 5998, 56911, 51426, 91319, 6598, 11264, 1869,
  1952, 84722, 10327, 7781, 90523, 100101267, 22852, 23387, 1112, 9839,
  200942, 54437, 4023, 9570, 3156, 84251, 5142, 55188, 155382, 22996,
  4641, 388591, 221692, 55023, 134728, 5587, 8882, 5310
)

# Subset the toptable based on the entrez IDs and select specific columns

subset_toptable1 <- CX_0.1_3[CX_0.1_3$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]


subset_toptable2 <- CX_0.1_24[CX_0.1_24$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]


subset_toptable3 <- CX_0.1_48[CX_0.1_48$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]


subset_toptable3 <- CX_0.1_48[CX_0.1_48$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]

subset_toptable4 <- CX_0.5_3[CX_0.5_3$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]

subset_toptable5 <- CX_0.5_24[CX_0.5_24$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]

subset_toptable6 <- CX_0.5_48[CX_0.5_48$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]

subset_toptable7 <- DOX_0.1_3[DOX_0.1_3$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]

subset_toptable8 <- DOX_0.1_24[DOX_0.1_24$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]

subset_toptable9 <- DOX_0.1_48[DOX_0.1_48$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]

subset_toptable10 <- DOX_0.5_3[DOX_0.5_3$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]

subset_toptable11 <- DOX_0.5_24[DOX_0.5_24$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]

subset_toptable12 <- DOX_0.5_48[DOX_0.5_48$Entrez_ID %in% Entrez_IDs, c("Entrez_ID", "logFC", "adj.P.Val")]

# Function to add columns and combine data
add_metadata <- function(data, drug, conc, time) {
  data %>%
    mutate(Drug = drug, Conc = conc, Time = time)
}

# Add metadata and combine all subsets
combined_data <- bind_rows(
  add_metadata(subset_toptable1, "CX", 0.1, 3),
  add_metadata(subset_toptable2, "CX", 0.1, 24),
  add_metadata(subset_toptable3, "CX", 0.1, 48),
  add_metadata(subset_toptable4, "CX", 0.5, 3),
  add_metadata(subset_toptable5, "CX", 0.5, 24),
  add_metadata(subset_toptable6, "CX", 0.5, 48),
  add_metadata(subset_toptable7, "DOX", 0.1, 3),
  add_metadata(subset_toptable8, "DOX", 0.1, 24),
  add_metadata(subset_toptable9, "DOX", 0.1, 48),
  add_metadata(subset_toptable10, "DOX", 0.5, 3),
  add_metadata(subset_toptable11, "DOX", 0.5, 24),
  add_metadata(subset_toptable12, "DOX", 0.5, 48)
)

# Convert Entrez IDs to Gene symbols
combined_data <- combined_data %>%
  mutate(Gene = mapIds(
    org.Hs.eg.db,
    keys = as.character(Entrez_ID),
    column = "SYMBOL",
    keytype = "ENTREZID",
    multiVals = "first"
  ))

# Reorder columns
final_data <- dplyr::select(combined_data, Entrez_ID, Gene, logFC, adj.P.Val, Drug, Conc, Time)

# Assuming your dataframe is named data
# Add a column for significance stars
final_data <- final_data %>%
  mutate(Significance = ifelse(adj.P.Val < 0.05, "*", ""))

# Create a matrix for the heatmap (logFC values)
logFC_matrix <- acast(final_data, Gene ~ paste(Drug, Conc, Time, sep = "_"), value.var = "logFC")

# Create a matrix for the significance annotations
signif_matrix <- acast(final_data, Gene ~ paste(Drug, Conc, Time, sep = "_"), value.var = "Significance")

# Split column names into Drug, Conc, and Time
colnames_split <- strsplit(colnames(logFC_matrix), "_")
drug <- sapply(colnames_split, function(x) x[1])
conc <- sapply(colnames_split, function(x) x[2])
time <- sapply(colnames_split, function(x) x[3])

# Create the desired column order: CX 0.1 3hr, CX 0.5 3hr, CX 0.1 24hr, CX 0.5 24hr, CX 0.1 48h, CX 0.5 48h,
# DOX 0.1 3hr, DOX 0.5 3hr, DOX 0.1 24hr, DOX 0.5 24hr, DOX 0.1 48h, DOX 0.5 48h
desired_order <- c("CX_0.1_3", "CX_0.5_3", "CX_0.1_24", "CX_0.5_24", "CX_0.1_48", "CX_0.5_48",
                   "DOX_0.1_3", "DOX_0.5_3", "DOX_0.1_24", "DOX_0.5_24", "DOX_0.1_48", "DOX_0.5_48")

# Reorder columns in the matrix based on the desired order
column_names <- paste(drug, conc, time, sep = "_")
column_order <- match(desired_order, column_names)
logFC_matrix <- logFC_matrix[, column_order]
signif_matrix <- signif_matrix[, column_order]
drug <- drug[column_order]
conc <- conc[column_order]
time <- time[column_order]

# Prepare annotations matching the column structure
ha_top <- HeatmapAnnotation(
  Drug = drug,
  Conc = conc,
  Time = time,
  col = list(Drug = c("CX" = "blue", "DOX" = "red"),
             Conc = c("0.1" = "lightgreen", "0.5" = "darkgreen"),
             Time = c("3" = "yellow", "24" = "orange", "48" = "purple")),
  annotation_height = unit(c(2, 2, 2), "cm")
)

# Create the heatmap
heatmap <- Heatmap(logFC_matrix, name = "logFC", top_annotation = ha_top,
                   cell_fun = function(j, i, x, y, width, height, fill) {
                     grid.text(signif_matrix[i, j], x, y, gp = gpar(fontsize = 10))
                   },
                   show_row_names = TRUE, show_column_names = FALSE,
                   column_title = "Genes in Heart failure GWAS associated loci\nin response to CX5461 and DOX",
                   column_title_gp = gpar(fontsize = 16, fontface = "bold"),
                   cluster_columns = FALSE)  # Disable column clustering

# Draw the heatmap
draw(heatmap, heatmap_legend_side = "right", annotation_legend_side = "right")
```

## **ðŸ“Œ Corr-motif Boxplots**
## **ðŸ“Œ Non response (0.1 micromolar)**
```{r box1, echo=TRUE, message=FALSE, warning=FALSE}
# Load libraries
library(dplyr)
library(ggplot2)
library(reshape2)

# Load response group files
prob_1_0.1 <- as.character(read.csv("data/prob_1_0.1.csv")$Entrez_ID)

# Load expression dataset
boxplot1 <- read.csv("data/boxplot1.csv", check.names = FALSE)
boxplot1 <- as.data.frame(boxplot1)

# Choose a gene from the non-response group
target_entrez <- 92342  # Ensure this is from prob_1_0.1

# Stop if the gene is not in the non-response group
if (!(target_entrez %in% prob_1_0.1)) {
  stop("Selected gene is not in the non-response group for 0.1 micromolar.")
}

# Filter for selected gene
gene_data <- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if(nrow(gene_data) == 0) stop("No data found for selected ENTIREZID.")

# Reshape to long format
gene_data_long <- melt(gene_data,
                       id.vars = c("ENTREZID", "SYMBOL", "GENENAME"),
                       variable.name = "Sample",
                       value.name = "log2CPM")

# Extract metadata from sample names
gene_data_long <- gene_data_long %>%
  mutate(
    Time = sub(".*_(\\d+)$", "\\1", Sample),
    Concentration = sub(".*_(0\\.\\d)_\\d+$", "\\1", Sample),
    Drug = sub(".*_(CX\\.5461|DOX|VEH)_.*", "\\1", Sample),
    Indv = sub("^([0-9]+\\.[0-9]+)_.*", "\\1", Sample)
  )

# Filter for 0.1 micromolar only
gene_data_long <- gene_data_long %>% filter(Concentration == "0.1")

# Convert to factors
gene_data_long$Time <- factor(gene_data_long$Time, levels = c("3", "24", "48"))
gene_data_long$Concentration <- factor(gene_data_long$Concentration, levels = "0.1")

# Map individual IDs
indv_mapping <- c("75.1" = "1", "78.1" = "2", "87.1" = "3", 
                  "17.3" = "4", "84.1" = "5", "90.1" = "6")

gene_data_long <- gene_data_long %>%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], "Unknown"))

# Define color palette for drugs
drug_palette <- c("CX.5461" = "#08306B", "DOX" = "#E7298A", "VEH" = "green")

# Extract gene symbol for labeling
gene_symbol <- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +  # Only facets by Time now
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle("Non response (0.1 micromolar)") +
  labs(
    x = "Drugs",
    y = paste(gene_symbol, " log2CPM")
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = "black"),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = "black"),
    axis.text.x = element_text(size = 10, color = "black", angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = "bold")
  )
```

## **ðŸ“Œ CX-DOX Mid-Late Response (0.1 micromolar)**
```{r box2, echo=TRUE, message=FALSE, warning=FALSE}
# Load libraries
library(dplyr)
library(ggplot2)
library(reshape2)

# Load response group files
prob_2_0.1 <- as.character(read.csv("data/prob_2_0.1.csv")$Entrez_ID)

prob_2_0.1

# Load expression dataset
boxplot1 <- read.csv("data/boxplot1.csv", check.names = FALSE)
boxplot1 <- as.data.frame(boxplot1)

# Choose a gene from the CX-DOX mid-late response group
target_entrez <- 4521  # Replace with your desired Entrez ID from prob_2_0.1

# Stop if the gene is not in the CX-DOX mid-late response group
if (!(target_entrez %in% prob_2_0.1)) {
  stop("Selected gene is not in the CX-DOX mid-late response group for 0.1 micromolar.")
}

# Filter for selected gene
gene_data <- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if(nrow(gene_data) == 0) stop("No data found for selected ENTREZID.")

# Reshape to long format
gene_data_long <- melt(gene_data,
                       id.vars = c("ENTREZID", "SYMBOL", "GENENAME"),
                       variable.name = "Sample",
                       value.name = "log2CPM")

# Extract metadata from sample names
gene_data_long <- gene_data_long %>%
  mutate(
    Time = sub(".*_(\\d+)$", "\\1", Sample),
    Concentration = sub(".*_(0\\.\\d)_\\d+$", "\\1", Sample),
    Drug = sub(".*_(CX\\.5461|DOX|VEH)_.*", "\\1", Sample),
    Indv = sub("^([0-9]+\\.[0-9]+)_.*", "\\1", Sample)
  )

# Filter for 0.1 micromolar only
gene_data_long <- gene_data_long %>% filter(Concentration == "0.1")

# Convert to factors
gene_data_long$Time <- factor(gene_data_long$Time, levels = c("3", "24", "48"))
gene_data_long$Concentration <- factor(gene_data_long$Concentration, levels = "0.1")

# Map individual IDs
indv_mapping <- c("75.1" = "1", "78.1" = "2", "87.1" = "3", 
                  "17.3" = "4", "84.1" = "5", "90.1" = "6")

gene_data_long <- gene_data_long %>%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], "Unknown"))

# Define color palette for drugs
drug_palette <- c("CX.5461" = "#08306B", "DOX" = "#E7298A", "VEH" = "green")

# Extract gene symbol for labeling
gene_symbol <- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +  # Only facets by Time
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle("CX-DOX mid-late\nresponse (0.1 micromolar)") +
  labs(
    x = "Drugs",
    y = paste(gene_symbol, " log2CPM")
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = "black"),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = "black"),
    axis.text.x = element_text(size = 10, color = "black", angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = "bold")
  )
```

## **ðŸ“Œ DOX-only mid-late response (0.1 micromolar)**
```{r box3, echo=TRUE, message=FALSE, warning=FALSE}

# Load libraries
library(dplyr)
library(ggplot2)
library(reshape2)

# Load response group files
prob_3_0.1 <- as.character(read.csv("data/prob_3_0.1.csv")$Entrez_ID)

# Print gene list if needed
prob_3_0.1

# Load expression dataset
boxplot1 <- read.csv("data/boxplot1.csv", check.names = FALSE)
boxplot1 <- as.data.frame(boxplot1)

# Choose a gene from the DOX-only mid-late response group
target_entrez <- 1674  # Replace with your desired Entrez ID from prob_3_0.1

# Stop if the gene is not in the DOX-only mid-late response group
if (!(target_entrez %in% prob_3_0.1)) {
  stop("Selected gene is not in the DOX-only mid-late response group for 0.1 micromolar.")
}

# Filter for selected gene
gene_data <- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if(nrow(gene_data) == 0) stop("No data found for selected ENTREZID.")

# Reshape to long format
gene_data_long <- reshape2::melt(gene_data,
                                 id.vars = c("ENTREZID", "SYMBOL", "GENENAME"),
                                 variable.name = "Sample",
                                 value.name = "log2CPM")

# Extract metadata from sample names
gene_data_long <- gene_data_long %>%
  mutate(
    Time = sub(".*_(\\d+)$", "\\1", Sample),
    Concentration = sub(".*_(0\\.\\d)_\\d+$", "\\1", Sample),
    Drug = sub(".*_(CX\\.5461|DOX|VEH)_.*", "\\1", Sample),
    Indv = sub("^([0-9]+\\.[0-9]+)_.*", "\\1", Sample)
  )

# Filter for 0.1 micromolar only
gene_data_long <- gene_data_long %>% filter(Concentration == "0.1")

# Convert to factors
gene_data_long$Time <- factor(gene_data_long$Time, levels = c("3", "24", "48"))
gene_data_long$Concentration <- factor(gene_data_long$Concentration, levels = "0.1")

# Map individual IDs
indv_mapping <- c("75.1" = "1", "78.1" = "2", "87.1" = "3", 
                  "17.3" = "4", "84.1" = "5", "90.1" = "6")

gene_data_long <- gene_data_long %>%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], "Unknown"))

# Define color palette for drugs
drug_palette <- c("CX.5461" = "#08306B", "DOX" = "#E7298A", "VEH" = "green")

# Extract gene symbol for labeling
gene_symbol <- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle("DOX-only mid-late\nresponse (0.1 micromolar)") +
  labs(
    x = "Drugs",
    y = paste(gene_symbol, " log2CPM")
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = "black"),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = "black"),
    axis.text.x = element_text(size = 10, color = "black", angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = "bold")
  )
```

## **ðŸ“Œ Non response (0.5 micromolar)**
```{r box4, echo=TRUE, message=FALSE, warning=FALSE}

# Load libraries
library(dplyr)
library(ggplot2)
library(reshape2)

# Load response group: Non response (0.5 ÂµM)
prob_1_0.5 <- as.character(read.csv("data/prob_1_0.5.csv")$Entrez_ID)

# Print genes (optional)
prob_1_0.5

# Load expression dataset
boxplot1 <- read.csv("data/boxplot1.csv", check.names = FALSE)
boxplot1 <- as.data.frame(boxplot1)

# Choose a gene from the non-response group (0.5 ÂµM)
target_entrez <- 9446  # Replace with your desired Entrez ID from prob_1_0.5

# Check if it's in the group
if (!(target_entrez %in% prob_1_0.5)) {
  stop("Selected gene is not in the Non response group for 0.5 micromolar.")
}

# Filter for selected gene
gene_data <- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if(nrow(gene_data) == 0) stop("No data found for selected ENTREZID.")

# Reshape to long format
gene_data_long <- melt(gene_data,
                       id.vars = c("ENTREZID", "SYMBOL", "GENENAME"),
                       variable.name = "Sample",
                       value.name = "log2CPM")

# Extract metadata from sample names
gene_data_long <- gene_data_long %>%
  mutate(
    Time = sub(".*_(\\d+)$", "\\1", Sample),
    Concentration = sub(".*_(0\\.\\d)_\\d+$", "\\1", Sample),
    Drug = sub(".*_(CX\\.5461|DOX|VEH)_.*", "\\1", Sample),
    Indv = sub("^([0-9]+\\.[0-9]+)_.*", "\\1", Sample)
  )

# Filter for 0.5 micromolar only
gene_data_long <- gene_data_long %>% filter(Concentration == "0.5")

# Convert to factors
gene_data_long$Time <- factor(gene_data_long$Time, levels = c("3", "24", "48"))
gene_data_long$Concentration <- factor(gene_data_long$Concentration, levels = "0.5")

# Map individual IDs
indv_mapping <- c("75.1" = "1", "78.1" = "2", "87.1" = "3", 
                  "17.3" = "4", "84.1" = "5", "90.1" = "6")

gene_data_long <- gene_data_long %>%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], "Unknown"))

# Define color palette for drugs
drug_palette <- c("CX.5461" = "#08306B", "DOX" = "#E7298A", "VEH" = "green")

# Extract gene symbol for labeling
gene_symbol <- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle("Non response (0.5 micromolar)") +
  labs(
    x = "Drugs",
    y = paste(gene_symbol, " log2CPM")
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = "black"),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = "black"),
    axis.text.x = element_text(size = 10, color = "black", angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = "bold")
  )
```
## **ðŸ“Œ DOX-specific response (0.5 ÂµM)**
```{r box5, echo=TRUE, message=FALSE, warning=FALSE}

# Load libraries
library(dplyr)
library(ggplot2)
library(reshape2)

# Load response group: DOX-specific response (0.5 ÂµM)
prob_2_0.5 <- as.character(read.csv("data/prob_2_0.5.csv")$Entrez_ID)

# View gene list (optional)
prob_2_0.5

# Load expression dataset
boxplot1 <- read.csv("data/boxplot1.csv", check.names = FALSE)
boxplot1 <- as.data.frame(boxplot1)

# Choose a gene from the DOX-specific response group (0.5 ÂµM)
target_entrez <- 114821  # Replace with your desired Entrez ID from prob_2_0.5

# Check membership
if (!(target_entrez %in% prob_2_0.5)) {
  stop("Selected gene is not in the DOX-specific response group for 0.5 micromolar.")
}

# Filter for selected gene
gene_data <- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if(nrow(gene_data) == 0) stop("No data found for selected ENTREZID.")

# Reshape to long format
gene_data_long <- melt(gene_data,
                       id.vars = c("ENTREZID", "SYMBOL", "GENENAME"),
                       variable.name = "Sample",
                       value.name = "log2CPM")

# Extract metadata from sample names
gene_data_long <- gene_data_long %>%
  mutate(
    Time = sub(".*_(\\d+)$", "\\1", Sample),
    Concentration = sub(".*_(0\\.\\d)_\\d+$", "\\1", Sample),
    Drug = sub(".*_(CX\\.5461|DOX|VEH)_.*", "\\1", Sample),
    Indv = sub("^([0-9]+\\.[0-9]+)_.*", "\\1", Sample)
  )

# Filter for 0.5 micromolar only
gene_data_long <- gene_data_long %>% filter(Concentration == "0.5")

# Convert to factors
gene_data_long$Time <- factor(gene_data_long$Time, levels = c("3", "24", "48"))
gene_data_long$Concentration <- factor(gene_data_long$Concentration, levels = "0.5")

# Map individual IDs
indv_mapping <- c("75.1" = "1", "78.1" = "2", "87.1" = "3", 
                  "17.3" = "4", "84.1" = "5", "90.1" = "6")

gene_data_long <- gene_data_long %>%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], "Unknown"))

# Define color palette for drugs
drug_palette <- c("CX.5461" = "#08306B", "DOX" = "#E7298A", "VEH" = "green")

# Extract gene symbol for labeling
gene_symbol <- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle("DOX-specific\nresponse (0.5 micromolar)") +
  labs(
    x = "Drugs",
    y = paste(gene_symbol, " log2CPM")
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = "black"),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = "black"),
    axis.text.x = element_text(size = 10, color = "black", angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = "bold")
  )
```

## **ðŸ“Œ DOX only mid-late response (0.5 ÂµM)**
```{r box6, echo=TRUE, message=FALSE, warning=FALSE}

# Load libraries
library(dplyr)
library(ggplot2)
library(reshape2)

# Load response group: DOX only mid-late response (0.5 ÂµM)
prob_3_0.5 <- as.character(read.csv("data/prob_3_0.5.csv")$Entrez_ID)

# View gene list (optional)
prob_3_0.5

# Load expression dataset
boxplot1 <- read.csv("data/boxplot1.csv", check.names = FALSE)
boxplot1 <- as.data.frame(boxplot1)

# Choose a gene from the DOX only mid-late response group (0.5 ÂµM)
target_entrez <- 4892  # Replace with your desired Entrez ID from prob_3_0.5

# Check membership
if (!(target_entrez %in% prob_3_0.5)) {
  stop("Selected gene is not in the DOX only mid-late response group for 0.5 micromolar.")
}

# Filter for selected gene
gene_data <- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if(nrow(gene_data) == 0) stop("No data found for selected ENTREZID.")

# Reshape to long format
gene_data_long <- melt(gene_data,
                       id.vars = c("ENTREZID", "SYMBOL", "GENENAME"),
                       variable.name = "Sample",
                       value.name = "log2CPM")

# Extract metadata from sample names
gene_data_long <- gene_data_long %>%
  mutate(
    Time = sub(".*_(\\d+)$", "\\1", Sample),
    Concentration = sub(".*_(0\\.\\d)_\\d+$", "\\1", Sample),
    Drug = sub(".*_(CX\\.5461|DOX|VEH)_.*", "\\1", Sample),
    Indv = sub("^([0-9]+\\.[0-9]+)_.*", "\\1", Sample)
  )

# Filter for 0.5 micromolar only
gene_data_long <- gene_data_long %>% filter(Concentration == "0.5")

# Convert to factors
gene_data_long$Time <- factor(gene_data_long$Time, levels = c("3", "24", "48"))
gene_data_long$Concentration <- factor(gene_data_long$Concentration, levels = "0.5")

# Map individual IDs
indv_mapping <- c("75.1" = "1", "78.1" = "2", "87.1" = "3", 
                  "17.3" = "4", "84.1" = "5", "90.1" = "6")

gene_data_long <- gene_data_long %>%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], "Unknown"))

# Define color palette for drugs
drug_palette <- c("CX.5461" = "#08306B", "DOX" = "#E7298A", "VEH" = "green")

# Extract gene symbol for labeling
gene_symbol <- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle("DOX only mid-late\nresponse (0.5 micromolar)") +
  labs(
    x = "Drugs",
    y = paste(gene_symbol, " log2CPM")
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = "black"),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = "black"),
    axis.text.x = element_text(size = 10, color = "black", angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = "bold")
  )
```

## **ðŸ“Œ CX total + DOX early response (0.5 micromolar)**
```{r box7, echo=TRUE, message=FALSE, warning=FALSE}

# Load libraries
library(dplyr)
library(ggplot2)
library(reshape2)

# Load response group: CX total + DOX early response (0.5 ÂµM)
prob_4_0.5 <- as.character(read.csv("data/prob_4_0.5.csv")$Entrez_ID)

# View gene list (optional)
prob_4_0.5

# Load expression dataset
boxplot1 <- read.csv("data/boxplot1.csv", check.names = FALSE)
boxplot1 <- as.data.frame(boxplot1)

# Choose a gene from the CX total + DOX early response group (0.5 ÂµM)
target_entrez <- 983  # Replace with your desired Entrez ID from prob_4_0.5

# Check membership
if (!(target_entrez %in% prob_4_0.5)) {
  stop("Selected gene is not in the CX total + DOX early response group for 0.5 micromolar.")
}

# Filter for selected gene
gene_data <- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if(nrow(gene_data) == 0) stop("No data found for selected ENTREZID.")

# Reshape to long format
gene_data_long <- melt(gene_data,
                       id.vars = c("ENTREZID", "SYMBOL", "GENENAME"),
                       variable.name = "Sample",
                       value.name = "log2CPM")

# Extract metadata from sample names
gene_data_long <- gene_data_long %>%
  mutate(
    Time = sub(".*_(\\d+)$", "\\1", Sample),
    Concentration = sub(".*_(0\\.\\d)_\\d+$", "\\1", Sample),
    Drug = sub(".*_(CX\\.5461|DOX|VEH)_.*", "\\1", Sample),
    Indv = sub("^([0-9]+\\.[0-9]+)_.*", "\\1", Sample)
  )

# Filter for 0.5 micromolar only
gene_data_long <- gene_data_long %>% filter(Concentration == "0.5")

# Convert to factors
gene_data_long$Time <- factor(gene_data_long$Time, levels = c("3", "24", "48"))
gene_data_long$Concentration <- factor(gene_data_long$Concentration, levels = "0.5")

# Map individual IDs
indv_mapping <- c("75.1" = "1", "78.1" = "2", "87.1" = "3", 
                  "17.3" = "4", "84.1" = "5", "90.1" = "6")

gene_data_long <- gene_data_long %>%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], "Unknown"))

# Define color palette for drugs
drug_palette <- c("CX.5461" = "#08306B", "DOX" = "#E7298A", "VEH" = "green")

# Extract gene symbol for labeling
gene_symbol <- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle("CX total+ DOX early\nresponse (0.5 micromolar)") +
  labs(
    x = "Drugs",
    y = paste(gene_symbol, " log2CPM")
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = "black"),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = "black"),
    axis.text.x = element_text(size = 10, color = "black", angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = "bold")
  )

```

## **ðŸ“Œ DOX early + CX-DOX mid-late response (0.5 ÂµM)**
```{r box8, echo=TRUE, message=FALSE, warning=FALSE}

# Load libraries
library(dplyr)
library(ggplot2)
library(reshape2)

# Load response group: DOX early + CX-DOX mid-late response (0.5 ÂµM)
prob_5_0.5 <- as.character(read.csv("data/prob_5_0.5.csv")$Entrez_ID)

# View gene list (optional)
prob_5_0.5

# Load expression dataset
boxplot1 <- read.csv("data/boxplot1.csv", check.names = FALSE)
boxplot1 <- as.data.frame(boxplot1)

# Choose a gene from the DOX early + CX-DOX mid-late response group (0.5 ÂµM)
target_entrez <- 672  # Replace with your desired Entrez ID from prob_5_0.5

# Check membership
if (!(target_entrez %in% prob_5_0.5)) {
  stop("Selected gene is not in the DOX early + CX-DOX mid-late response group for 0.5 micromolar.")
}

# Filter for selected gene
gene_data <- boxplot1[boxplot1$ENTREZID == target_entrez, ]
if(nrow(gene_data) == 0) stop("No data found for selected ENTREZID.")

# Reshape to long format
gene_data_long <- melt(gene_data,
                       id.vars = c("ENTREZID", "SYMBOL", "GENENAME"),
                       variable.name = "Sample",
                       value.name = "log2CPM")

# Extract metadata from sample names
gene_data_long <- gene_data_long %>%
  mutate(
    Time = sub(".*_(\\d+)$", "\\1", Sample),
    Concentration = sub(".*_(0\\.\\d)_\\d+$", "\\1", Sample),
    Drug = sub(".*_(CX\\.5461|DOX|VEH)_.*", "\\1", Sample),
    Indv = sub("^([0-9]+\\.[0-9]+)_.*", "\\1", Sample)
  )

# Filter for 0.5 micromolar only
gene_data_long <- gene_data_long %>% filter(Concentration == "0.5")

# Convert to factors
gene_data_long$Time <- factor(gene_data_long$Time, levels = c("3", "24", "48"))
gene_data_long$Concentration <- factor(gene_data_long$Concentration, levels = "0.5")

# Map individual IDs
indv_mapping <- c("75.1" = "1", "78.1" = "2", "87.1" = "3", 
                  "17.3" = "4", "84.1" = "5", "90.1" = "6")

gene_data_long <- gene_data_long %>%
  mutate(Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], "Unknown"))

# Define color palette for drugs
drug_palette <- c("CX.5461" = "#08306B", "DOX" = "#E7298A", "VEH" = "green")

# Extract gene symbol for labeling
gene_symbol <- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(. ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle("DOX early+ CX-DOX mid-late\nresponse (0.5 micromolar)") +
  labs(
    x = "Drugs",
    y = paste(gene_symbol, " log2CPM")
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = "black"),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = "black"),
    axis.text.x = element_text(size = 10, color = "black", angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = "bold")
  )
```

## **ðŸ“Œ Corrmotif conc boxplot for manuscript 0.1**
```{r box9, echo=TRUE, message=FALSE, warning=FALSE,fig.width=10}

# Load Required Libraries
library(dplyr)
library(ggplot2)

# ----------------- Load Response Groups -----------------

# 0.1 ÂµM
prob_1_0.1 <- as.character(read.csv("data/prob_1_0.1.csv")$Entrez_ID)
prob_2_0.1 <- as.character(read.csv("data/prob_2_0.1.csv")$Entrez_ID)
prob_3_0.1 <- as.character(read.csv("data/prob_3_0.1.csv")$Entrez_ID)

# ----------------- Load DEG Data -----------------

# Helper function to load and label
load_deg <- function(path, drug, time) {
  read.csv(path) %>%
    mutate(Drug = drug, Timepoint = time)
}

# 0.1 ÂµM DEG data
deg_0.1 <- bind_rows(
  load_deg("data/DEGs/Toptable_CX_0.1_3.csv", "CX.5461", 3),
  load_deg("data/DEGs/Toptable_CX_0.1_24.csv", "CX.5461", 24),
  load_deg("data/DEGs/Toptable_CX_0.1_48.csv", "CX.5461", 48),
  load_deg("data/DEGs/Toptable_DOX_0.1_3.csv", "DOX", 3),
  load_deg("data/DEGs/Toptable_DOX_0.1_24.csv", "DOX", 24),
  load_deg("data/DEGs/Toptable_DOX_0.1_48.csv", "DOX", 48)
) %>%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.1 ~ "Non response\n(0.1 ÂµM)",
      Entrez_ID %in% prob_2_0.1 ~ "CX-DOX mid-late response\n(0.1 ÂµM)",
      Entrez_ID %in% prob_3_0.1 ~ "DOX only mid-late response\n(0.1 ÂµM)",
      TRUE ~ NA_character_
    )
  ) %>%
  filter(!is.na(Response_Group))

# ----------------- Set Factor Levels -----------------

deg_0.1 <- deg_0.1 %>%
  mutate(
    Timepoint = factor(Timepoint, levels = c(3, 24, 48)),
    Response_Group = factor(Response_Group, levels = c(
      "Non response\n(0.1 ÂµM)",
      "CX-DOX mid-late response\n(0.1 ÂµM)",
      "DOX only mid-late response\n(0.1 ÂµM)"
    ))
  )

# ----------------- Plot Boxplot (Grouped by Timepoint, Faceted by Response Group) -----------------

ggplot(deg_0.1, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75)) +
  scale_fill_manual(values = c("CX.5461" = "blue", "DOX" = "red")) +
  facet_grid(. ~ Response_Group, scales = "free_x") +
  theme_bw() +
  labs(
    x = "Timepoint (hours)",
    y = "Log Fold Change",
    title = "Log Fold Change Distribution by Drug and Response Group (0.1 ÂµM)",
    fill = "Drug"
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = "bold"),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )

```

## **ðŸ“Œ Corrmotif conc boxplot for manuscript 0.5**
```{r box10, echo=TRUE, message=FALSE, warning=FALSE,fig.width=12}
# Load Required Libraries
library(dplyr)
library(ggplot2)

# ----------------- Load Response Groups -----------------

# 0.5 ÂµM
prob_1_0.5 <- as.character(read.csv("data/prob_1_0.5.csv")$Entrez_ID)
prob_2_0.5 <- as.character(read.csv("data/prob_2_0.5.csv")$Entrez_ID)
prob_3_0.5 <- as.character(read.csv("data/prob_3_0.5.csv")$Entrez_ID)
prob_4_0.5 <- as.character(read.csv("data/prob_4_0.5.csv")$Entrez_ID)
prob_5_0.5 <- as.character(read.csv("data/prob_5_0.5.csv")$Entrez_ID)

# ----------------- Load DEG Data -----------------

# Helper function to load and label
load_deg <- function(path, drug, time) {
  read.csv(path) %>%
    mutate(Drug = drug, Timepoint = time)
}

# 0.5 ÂµM DEG data
deg_0.5 <- bind_rows(
  load_deg("data/DEGs/Toptable_CX_0.5_3.csv", "CX.5461", 3),
  load_deg("data/DEGs/Toptable_CX_0.5_24.csv", "CX.5461", 24),
  load_deg("data/DEGs/Toptable_CX_0.5_48.csv", "CX.5461", 48),
  load_deg("data/DEGs/Toptable_DOX_0.5_3.csv", "DOX", 3),
  load_deg("data/DEGs/Toptable_DOX_0.5_24.csv", "DOX", 24),
  load_deg("data/DEGs/Toptable_DOX_0.5_48.csv", "DOX", 48)
) %>%
  mutate(
    Entrez_ID = as.character(Entrez_ID),
    Response_Group = case_when(
      Entrez_ID %in% prob_1_0.5 ~ "Non response\n(0.5 ÂµM)",
      Entrez_ID %in% prob_2_0.5 ~ "DOX-specific\nresponse (0.5 ÂµM)",
      Entrez_ID %in% prob_3_0.5 ~ "DOX only mid-late\nresponse (0.5 ÂµM)",
      Entrez_ID %in% prob_4_0.5 ~ "CX + DOX (early)\nresponse (0.5 ÂµM)",
      Entrez_ID %in% prob_5_0.5 ~ "DOX + CX (mid-late)\nresponse (0.5 ÂµM)",
      TRUE ~ NA_character_
    )
  ) %>%
  filter(!is.na(Response_Group))

# ----------------- Set Factor Levels -----------------

deg_0.5 <- deg_0.5 %>%
  mutate(
    Timepoint = factor(Timepoint, levels = c(3, 24, 48)),
    Response_Group = factor(Response_Group, levels = c(
      "Non response\n(0.5 ÂµM)",
      "DOX-specific\nresponse (0.5 ÂµM)",
      "DOX only mid-late\nresponse (0.5 ÂµM)",
      "CX + DOX (early)\nresponse (0.5 ÂµM)",
      "DOX + CX (mid-late)\nresponse (0.5 ÂµM)"
    ))
  )

# ----------------- Plot Boxplot (Grouped by Timepoint, Faceted by Response Group) -----------------

ggplot(deg_0.5, aes(x = Timepoint, y = logFC, fill = Drug)) +
  geom_boxplot(position = position_dodge(width = 0.75)) +
  scale_fill_manual(values = c("CX.5461" = "blue", "DOX" = "red")) +
  facet_grid(. ~ Response_Group, scales = "free_x") +
  theme_bw() +
  labs(
    x = "Timepoint (hours)",
    y = "Log Fold Change",
    title = "Log Fold Change Distribution by Drug and Response Group (0.5 ÂµM)",
    fill = "Drug"
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 14),
    axis.text = element_text(size = 12),
    strip.text = element_text(size = 11, face = "bold"),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )
```














